{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import os\n",
    "import pandas as pd\n",
    "from Bio import SeqIO\n",
    "from Bio.Seq import Seq  # Import Seq from Bio.Seq\n",
    "\n",
    "riboDir = \"/users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>header</th>\n",
       "      <th>seq</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ENST00000431238.7:X:+|1|259:15:192|noncoding|CTG</td>\n",
       "      <td>CTGTACCGGCTGCATCCGGAGCAGGGCATGCCGGCCGGCGTGTGCG...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ENST00000431238.7:X:+|2|259:24:192|noncoding|CTG</td>\n",
       "      <td>CTGCATCCGGAGCAGGGCATGCCGGCCGGCGTGTGCGTGGACGCTG...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ENST00000431238.7:X:+|3|259:42:192|noncoding|ATG</td>\n",
       "      <td>ATGCCGGCCGGCGTGTGCGTGGACGCTGCGGGGAAGCTCTGGGTGG...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ENST00000431238.7:X:+|4|259:54:192|noncoding|GTG</td>\n",
       "      <td>GTGTGCGTGGACGCTGCGGGGAAGCTCTGGGTGGCCTCTGCGTCGA...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ENST00000431238.7:X:+|5|259:60:192|noncoding|GTG</td>\n",
       "      <td>GTGGACGCTGCGGGGAAGCTCTGGGTGGCCTCTGCGTCGATAGAGG...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6910424</th>\n",
       "      <td>ENST00000839853.1:KI270755.1:+|51|781:650:764|...</td>\n",
       "      <td>TTGTCTCTACGTGACAGTCTGTCTTGTGAGGTCGAGCTCCCTGAAG...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6910425</th>\n",
       "      <td>ENST00000839853.1:KI270755.1:+|52|781:660:711|...</td>\n",
       "      <td>GTGACAGTCTGTCTTGTGAGGTCGAGCTCCCTGAAGGCAGCAAGCA...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6910426</th>\n",
       "      <td>ENST00000839853.1:KI270755.1:+|53|781:668:764|...</td>\n",
       "      <td>CTGTCTTGTGAGGTCGAGCTCCCTGAAGGCAGCAAGCATGTAATTG...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6910427</th>\n",
       "      <td>ENST00000839853.1:KI270755.1:+|54|781:675:711|...</td>\n",
       "      <td>GTGAGGTCGAGCTCCCTGAAGGCAGCAAGCATGTAA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6910428</th>\n",
       "      <td>ENST00000839853.1:KI270755.1:+|55|781:711:753|...</td>\n",
       "      <td>TTGGTAAATTCTCATCAGTTAGCCATGTCTGGAACATGGTAG</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>6910429 rows Ã— 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                    header  \\\n",
       "0         ENST00000431238.7:X:+|1|259:15:192|noncoding|CTG   \n",
       "1         ENST00000431238.7:X:+|2|259:24:192|noncoding|CTG   \n",
       "2         ENST00000431238.7:X:+|3|259:42:192|noncoding|ATG   \n",
       "3         ENST00000431238.7:X:+|4|259:54:192|noncoding|GTG   \n",
       "4         ENST00000431238.7:X:+|5|259:60:192|noncoding|GTG   \n",
       "...                                                    ...   \n",
       "6910424  ENST00000839853.1:KI270755.1:+|51|781:650:764|...   \n",
       "6910425  ENST00000839853.1:KI270755.1:+|52|781:660:711|...   \n",
       "6910426  ENST00000839853.1:KI270755.1:+|53|781:668:764|...   \n",
       "6910427  ENST00000839853.1:KI270755.1:+|54|781:675:711|...   \n",
       "6910428  ENST00000839853.1:KI270755.1:+|55|781:711:753|...   \n",
       "\n",
       "                                                       seq  \n",
       "0        CTGTACCGGCTGCATCCGGAGCAGGGCATGCCGGCCGGCGTGTGCG...  \n",
       "1        CTGCATCCGGAGCAGGGCATGCCGGCCGGCGTGTGCGTGGACGCTG...  \n",
       "2        ATGCCGGCCGGCGTGTGCGTGGACGCTGCGGGGAAGCTCTGGGTGG...  \n",
       "3        GTGTGCGTGGACGCTGCGGGGAAGCTCTGGGTGGCCTCTGCGTCGA...  \n",
       "4        GTGGACGCTGCGGGGAAGCTCTGGGTGGCCTCTGCGTCGATAGAGG...  \n",
       "...                                                    ...  \n",
       "6910424  TTGTCTCTACGTGACAGTCTGTCTTGTGAGGTCGAGCTCCCTGAAG...  \n",
       "6910425  GTGACAGTCTGTCTTGTGAGGTCGAGCTCCCTGAAGGCAGCAAGCA...  \n",
       "6910426  CTGTCTTGTGAGGTCGAGCTCCCTGAAGGCAGCAAGCATGTAATTG...  \n",
       "6910427               GTGAGGTCGAGCTCCCTGAAGGCAGCAAGCATGTAA  \n",
       "6910428         TTGGTAAATTCTCATCAGTTAGCCATGTCTGGAACATGGTAG  \n",
       "\n",
       "[6910429 rows x 2 columns]"
      ]
     },
     "execution_count": 2,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "## Create fasta of translated sequences from candidates\n",
    "specie = \"human\"\n",
    "fasta=os.path.join(\"/users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47\",specie,\"RiboNovel_MultMap_1to1/Annotation/candidateORF.fa\")\n",
    "fasta_df_input = pd.read_csv(fasta, sep=\"\\t\", header=None)\n",
    "fasta_df = pd.DataFrame({'header':fasta_df_input[0].iloc[::2].values, 'seq':fasta_df_input[0].iloc[1::2].values})\n",
    "fasta_df['header'] = fasta_df.header.str[1:]\n",
    "fasta_df\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "human_brain_ribo_2_r1\n",
      "human_liver_ribo_2_r1\n",
      "human_brain_ribo_3_r1\n",
      "human_liver_ribo_1_r1\n",
      "human_testis_ribo_3_r1\n",
      "human_liver_ribo_3_r1\n",
      "human_testis_ribo_2_r1\n",
      "human_testis_ribo_1_r1\n",
      "human_brain_ribo_1_r1\n"
     ]
    }
   ],
   "source": [
    "dict_translated = dict()\n",
    "dict_translated_prot = dict()\n",
    "\n",
    "for infolder_results in os.listdir(os.path.join(riboDir,specie,\"RiboNovel_MultMap_1to1/RibORF\")):\n",
    "    print(infolder_results)\n",
    "    outfile=os.path.join(riboDir,specie,\"RiboNovel_MultMap_1to1/RibORF\",infolder_results,\"repre.valid.pred.pvalue.parameters.fa\")\n",
    "    outfile_prot=os.path.join(riboDir,specie,\"RiboNovel_MultMap_1to1/RibORF\",infolder_results,\"repre.valid.pred.pvalue.parameters.PROTEIN.fa\")\n",
    "\n",
    "    seqs_df = pd.read_csv(os.path.join(riboDir,specie,\"RiboNovel_MultMap_1to1/RibORF\",infolder_results,\"repre.valid.pred.pvalue.parameters.txt\"), sep=\"\\t\")\n",
    "    candidates_in_repre = fasta_df[fasta_df['header'].isin(seqs_df.orfID.values.tolist())]\n",
    "\n",
    "    with open(outfile, 'w') as out:\n",
    "        with open(outfile_prot, 'w') as out_prot:\n",
    "            for index, row in candidates_in_repre.iterrows():\n",
    "                    out.write(\">%s\\n%s\\n\" %(row['header'], row['seq']))\n",
    "                    ## add to non-redundant dictionary\n",
    "                    dict_translated[row['header']] = row['seq']\n",
    "\n",
    "                    ## translate to proteins\n",
    "                    out_prot.write(\">%s\\n%s\\n\" %(row['header'], str(Seq(row['seq']).translate())))\n",
    "                    ## add to non-redundant dictionary\n",
    "                    dict_translated_prot[row['header']] = str(Seq(row['seq']).translate())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "outfile_prot = os.path.join(riboDir,specie,\"RiboNovel_MultMap_1to1/RibORF/repre.valid.pred.pvalue.parameters.allsamples.noredundant.PROTEIN.fa\")\n",
    "outfile = os.path.join(riboDir,specie,\"RiboNovel_MultMap_1to1/RibORF/repre.valid.pred.pvalue.parameters.allsamples.noredundant.fa\")\n",
    "\n",
    "with open(outfile, 'w') as out:\n",
    "    with open(outfile_prot, 'w') as out_prot:\n",
    "        for key, value in dict_translated.items():\n",
    "            out.write(\">%s\\n%s\\n\" %(key, value))\n",
    "\n",
    "        ## proteins\n",
    "        for key, value in dict_translated_prot.items():\n",
    "            out_prot.write(\">%s\\n%s\\n\" %(key, value))\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "## get transcript Length\n",
    "## Tab delimited text file with two columns. The first column shows the transcript length, and the second shows the ORF length (e.g. 1000\t36)\n",
    "df_translated = pd.DataFrame(dict_translated.items(), columns=['header', 'seq'])\n",
    "## get transcript length\n",
    "df_translated['transcriptLength'] = df_translated['header'].str.split(\"\\|\", expand=True)[2]\n",
    "df_translated['transcriptLength'] = df_translated['transcriptLength'].str.split(\":\", expand=True)[0]\n",
    "## get orf length\n",
    "df_translated['orfLength_info'] = df_translated['header'].str.split(\"\\|\", expand=True)[2]\n",
    "df_translated['endORF'] = df_translated['orfLength_info'].str.split(\":\", expand=True)[2]\n",
    "df_translated['startORF'] = df_translated['orfLength_info'].str.split(\":\", expand=True)[1]\n",
    "df_translated['orfLength'] = df_translated['endORF'].astype('int64') - df_translated['startORF'].astype('int64')\n",
    "df_translated\n",
    "\n",
    "transcript_length = df_translated[['transcriptLength', 'orfLength']]\n",
    "transcript_length.to_csv(os.path.join(riboDir,specie,\"RiboNovel_MultMap_1to1/RibORF/length_transcript_orf.allsamples.noredundant.txt\"), header=None, sep=\"\\t\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.10"
  },
  "orig_nbformat": 4
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
