---
title: "Preparation"
author: "Marta Espinosa"
date: "2024-10-22"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
library(dplyr)
library(tidyr)
library(ggplot2)
library(stringr)
library(purrr)
library(ggvenn)
library(ggpubr)
library(ggside)
library(bioseq)
# library(ggthemr)
library(ggbeeswarm)
library(ggbreak)
library(ggrepel)
library(rcartocolor)

annot = read.csv("/users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/human/newReference_Resconstructed/1transcript_1gene.reconstructed.csv")
# names(gene_transcript) = c("gene_id"gene_type"gene_name")
# gene_transcript$gene_id = gsub("\\..*"gene_transcript$gene_id)
plots_wd="/projects_eg/projects/marta/TestisRestricted_Microproteins_TSA/with_TranscriptomeReconstruction/PreparationPlots"

cancers = c("BRCA","BLCA","LUAD","KIRC","KIRP","PRAD","LUSC","COAD","LIHC")
tcga_projects=c("TCGA-BRCA","TCGA-LUSC","TCGA-PRAD","TCGA-KIRC","TCGA-KIRP","TCGA-LUAD","TCGA-BLCA")#,"TCGA-LIHC"]
other_projects=c("GSE102101_KIRC","GSE133624_BLCA","GSE22260_PRAD","PRJEB2449_PRAD","SRP238334_KIRC","GSE214846_LIHC","GSE229705_LUAD","TCGA_COAD","SRP107326_COAD")
manuscript_projects = c("liver_adjacent_totalRNA_LIHC","hcc_normal_totalRNA_LIHC","GSE193567_LIHC","LIHC_TCGA_LIHC")
# deleted_projects=c("GSE103001_BRCA"GSE89223_PRAD")
all_projects = c(tcga_projects,other_projects,manuscript_projects)

cancers_dir = "/users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/cancers"

CTDB = read.csv("/projects_eg/projects/marta/CTdatabase_list.csv", sep="\t")
CTDB = CTDB %>% mutate(X_A = case_when(grepl("^X", chr) ~ "X",
                                       TRUE ~ "A"))

total_num_patients = data.frame(ctype = c("BRCA","BLCA","LUAD","KIRC","LUSC","PRAD","LIHC","COAD"),
                                total_n = c(109,38,179,142,49,75,182,144))

# Function to compute the required statistics
compute_stats <- function(row) {
  mean_value <- mean(row)
  median_value <- median(row)
  sd_value <- sd(row)
  q1_value <- quantile(row, 0.25)
  q3_value <- quantile(row, 0.75)
  
  return(c(mean = mean_value, median = median_value, sd = sd_value, Q1 = q1_value, Q3 = q3_value))
}

# Function to calculate log2ratio of relative frequency
calculate_log2ratio_df <- function(group1_seqs, group2_seqs) {
  all_amino_acids <- c("A","R","N","D","C","Q","E","G","H","U","L","K","M","F","P","S","T","W","Y","V", "*")
  log2ratios <- numeric(length(all_amino_acids))
  
  for (aa in all_amino_acids) {
    count_group1 <- sum(sapply(group1_seqs, function(seq) sum(unlist(strsplit(as.character(seq), "")) == aa)))
    count_group2 <- sum(sapply(group2_seqs, function(seq) sum(unlist(strsplit(as.character(seq), "")) == aa)))
    
    freq_group1 <- count_group1 / sum(sapply(group1_seqs, nchar))
    freq_group2 <- count_group2 / sum(sapply(group2_seqs, nchar))
    
    log2ratios[aa] <- log2(freq_group1 / freq_group2)
  }
  
  log2ratio_df <- data.frame(Aminoacid = all_amino_acids, log2ratio = log2ratios)
  return(log2ratio_df)
}

```

## TestisORFs

`/home/marta/PROJECT_SCRIPTS/TestisRestricted_MP/Q1.1_TestisORFs_in1.Rmd`

```{r testis, echo=F}
testis = read.csv("/projects_eg/projects/marta/TestisRestricted_Microproteins_TSA/with_TranscriptomeReconstruction/Q1_TestisORFs/human/ribORF_humanTestis_in1.csv")
print("ORFs")
table(testis %>% select(orfID, gene_id, gene_type) %>% unique() %>% pull(gene_type))
print("Genes")
table(testis %>% select(gene_id, gene_type) %>% unique() %>% pull(gene_type))
```

## TestisRestricted - No Present in Proteome (/)BLASTP)

`/home/marta/PROJECT_SCRIPTS/TestisRestricted_MP/Q2_TestisRestricted_in1.Rmd`
`/home/marta/PROJECT_SCRIPTS/TestisRestricted_MP/Q2.1_TestisRestricted_BLASTP_ncORFSvsCDS.ipynb`

```{r testisRestricted_NoProteome, echo=F}
testisRestr = read.csv('/projects_eg/projects/marta/TestisRestricted_Microproteins_TSA/with_TranscriptomeReconstruction/Q2_TestisRestricted/human/testisRestricted_GTEx_translatedONLYtestis.csv')
print("ORFs")
table(testisRestr %>% select(orfID, gene_id, gene_type) %>% unique() %>% pull(gene_type))
print("Genes")
table(testisRestr %>% select(gene_id, gene_type) %>% unique() %>% pull(gene_type))

testisRestr = read.csv('/projects_eg/projects/marta/TestisRestricted_Microproteins_TSA/with_TranscriptomeReconstruction/Q2_TestisRestricted/human/testisRestricted_GTEx_translatedONLYtestis.noProteome.csv')
print("ORFs")
table(testisRestr %>% select(orfID, gene_id, gene_type) %>% unique() %>% pull(gene_type))
print("Genes")
table(testisRestr %>% select(gene_id, gene_type) %>% unique() %>% pull(gene_type))
```

## TumorReactivated

- Expressed 3 times more in tumor than in normal in at least 5% of the patients of at least 1 cancer type of the studied ones.

```{r tumorReactivated, echo=FALSE}
tumorReact = read.csv("/projects_eg/projects/marta/TestisRestricted_Microproteins_TSA/with_TranscriptomeReconstruction/Q4_TestisRestricted_TumorSpecific/human/TOv3x_5percent_TestisRestrictedGTEx_Translated_Ctypes.csv")
filtered_data_TOv3x_long = read.csv("/projects_eg/projects/marta/TestisRestricted_Microproteins_TSA/with_TranscriptomeReconstruction/Q4_TestisRestricted_TumorSpecific/human/TOv3xdata_long_5percent.csv")

tumorReact = merge(tumorReact, total_num_patients, by="ctype")
print("ORFs")
table(tumorReact %>% select(orfID, gene_id, gene_type) %>% unique() %>% pull(gene_type))
print("Genes")
table(tumorReact %>% select(gene_id, gene_type) %>% unique() %>% pull(gene_type))

print("Per cancertype")
ctypes_tumorReact = tumorReact %>% mutate(coding_ncoding = case_when(gene_type == "protein_coding" ~ "protein_coding",
                                                                     TRUE ~ "noncoding")) %>% select(gene_id, ctype, coding_ncoding) %>% unique()
table(ctypes_tumorReact$coding_ncoding, ctypes_tumorReact$ctype)
table(ctypes_tumorReact$ctype)
```

```{r correlation_npatients_nTSTR, echo=F}
tumorReact %>% select(gene_id, total_n, ctype) %>% group_by(ctype, total_n) %>% count() %>%
  ggplot(., aes(x=total_n, y=n, color=ctype)) +
  geom_point(stat="identity", size=5) +
  geom_text(aes(label=ctype), hjust = -0.3, size=3, fontface = "bold") +
  scale_x_continuous(limits=c(25,200)) +
  labs(x="Total number of patients",
       y="Number of TS/TR genes",
       title="Correlation between TS/TR and\nthe number of patients",
       color="CancerType") +
    scale_color_manual(values=c("BRCA"="#E69F00", "BLCA"="#56A3A6", "LUAD"="#0079E3", "KIRC"="#F0C27B", "LUSC"="#0072B2", "PRAD"="#D55E00", "LIHC"="#CC79A7", "COAD"="#8B7355")) +
  theme_minimal() +
  theme(legend.position="none")

ggsave(file.path(plots_wd, "PNG/TSTR_numpatients_scatter.png"), height=5.08, width=4.80)

```

## Correlation testis expression vs tumor expression?

```{r testis_tumor_TPMs, echo=F}
# raw_counts = read.table("/users/genomics/saraa/projectTestis/featureCounts/results_geneID/featureCounts_geneID.txt header=TRUE)
testis_TPMs = read.csv("/users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/human/featureCounts_gffcompare/table_of_counts_TPMs_withLength.csv", header=TRUE)
names(testis_TPMs)[1] = "transcript_id"
## For novels, remove those shorter than 300 pb
testis_TPMs = testis_TPMs %>%
  filter(!(startsWith(transcript_id, "TCONS") & Length < 300))

testis_TPMs = merge(testis_TPMs, annot, by="transcript_id", all.x=T)
testis_TPMs = testis_TPMs %>% unique()
testis_TPMs = testis_TPMs[,!grepl("brain", names(testis_TPMs))]
testis_TPMs = testis_TPMs[,!grepl("liver", names(testis_TPMs))]
testis_TPMs = testis_TPMs %>% select(-Length)

## Select only numeric columns
numeric_df = testis_TPMs[, sapply(testis_TPMs, is.numeric)]
## Apply the function to each row
stats_per_row = t(apply(numeric_df, 1, compute_stats))
stats_df = as.data.frame(stats_per_row)

testis_TPMs_stats = cbind(testis_TPMs %>% select(transcript_id, gene_id, gene_type, gene_name), stats_df)
testis_TPMs_stats %>% head
names(testis_TPMs_stats) = c("transcript_id","gene_id","gene_type","gene_name","mean_testis","median_testis","sd_testis","Q1.25%_testis","Q3.75%_testis")


### Testis-Tumor
testis_tumor_TPMs = merge(filtered_data_TOv3x_long, testis_TPMs_stats, by=c("transcript_id","gene_id","gene_type","gene_name"))
## remove the zeros, show mean of patients for which it is Overexpressed
testis_tumor_TPMs = testis_tumor_TPMs %>% subset(normal_tumor == "tumor") %>% subset(forced_TPM > 0) 
testis_tumor_TPMs = testis_tumor_TPMs %>% group_by(gene_name, gene_type, ctype, mean_testis, median_testis) %>% mutate(mean_tumor = mean(forced_TPM),
                                                                                          median_tumor = median(forced_TPM))
testis_tumor_TPMs = testis_tumor_TPMs %>% select(gene_name, gene_type, mean_tumor, median_tumor, mean_testis, median_testis) %>% unique()

### MEAN tumor zeros removed
testis_tumor_TPMs %>% mutate(coding_noncoding = case_when(gene_type == "protein_coding" ~ "protein_coding",
                                                          TRUE ~ "noncoding")) %>%
  ggplot(., aes(x=mean_tumor, y=mean_testis, color=coding_noncoding)) +
  geom_point() +
  geom_text_repel(aes(label=gene_name)) +
  labs(x="Mean TPM Tumor Samples (zeros removed)",
       y="Mean TPM Testis Samples",
       title="Correlation expression Testis vs Tumor") +
  scale_x_continuous(limits=c(0,275)) +
  scale_color_manual(values=c("#DFA398","#3B3960")) +
  theme_minimal() +
  facet_wrap(~ ctype)

### MEAN tumor zeros removed
testis_tumor_TPMs %>% mutate(coding_noncoding = case_when(gene_type == "protein_coding" ~ "protein_coding",
                                                          TRUE ~ "noncoding")) %>%
  ggplot(., aes(x=log(mean_tumor), y=log(mean_testis), color=coding_noncoding)) +
  geom_point() +
  # geom_smooth(method = "lm", se = FALSE, aes(color=coding_noncoding)) +
   # stat_cor(label.y = c(5.2, 4.8, 4.3, 3.8), label.x = c(1,1,1,1), aes(group=coding_noncoding, color = coding_noncoding), size=3) +
  labs(x="log(Mean TPM) Tumor Samples (zeros removed)",
       y="log(Mean TPM) Testis Samples",
       title="Correlation expression Testis vs Tumor") +
  scale_y_continuous(limits=c(-2.5,6)) +
  scale_x_continuous(limits=c(-2.5,6)) +
  geom_vline(xintercept=0, size=.2) +
  geom_hline(yintercept=0, size=.2) +
  scale_color_manual(values=c("#DFA398","#3B3960")) +
  theme_minimal() +
  theme(legend.position="top") +
  facet_wrap(~ ctype)
ggsave(file.path(plots_wd, "PNG/TSTR_TestisExpr_TumorExpr.png"), width=7, height=7)
```

## Comparisons, from TestisSpecific, which are Reactivated and which are not

```{r comparisons, echo=F}
testisRestr = testisRestr %>% 
  mutate(tumorReactivated = case_when(gene_id %in% tumorReact$gene_id ~ "Reactivated",
                                      TRUE ~ "NoReactivated")) %>%
  mutate(coding_noncoding = case_when(gene_type == "protein_coding" ~ "protein_coding",
                                                          TRUE ~ "noncoding"))

### length
testisRestr %>% select(gene_id, coding_noncoding, length_aa, tumorReactivated) %>% unique() %>%
  ggplot(., aes(y=length_aa, x=coding_noncoding, fill=tumorReactivated)) +
  geom_boxplot() +
  stat_compare_means() +
  labs(x="",
       y="Length (AA)",
       title = "Length differences between TS/TR and TS/nonTR",
       fill="Reactivation\nin tumor") +
  scale_fill_manual(values=c("#899DA4","#DC863B")) +
  scale_y_continuous(trans="log10") +
  theme_minimal()

### translation efficiency
testis_TE = read.csv("/projects_eg/projects/marta/TestisRestricted_Microproteins_TSA/with_TranscriptomeReconstruction/Q1_TestisORFs/human/mean_translationEfficiency_testis.csv")
testisRestr = merge(testisRestr, testis_TE %>% select(orfID, mean_TE), by="orfID")

testisRestr %>% select(gene_id, coding_noncoding, mean_TE, tumorReactivated) %>% unique() %>%
  ggplot(., aes(y=mean_TE, x=coding_noncoding, fill=tumorReactivated)) +
  geom_boxplot() +
  stat_compare_means() +
  scale_y_continuous(trans="log10") +
  labs(x="",
       y="Mean Translation Efficiency",
       title = "TE differences between TS/TR and TS/nonTR",
       fill="Reactivation\nin tumor") +
  scale_fill_manual(values=c("#899DA4","#DC863B")) +
  theme_minimal()

### evolutionary conservation
conservation = read.csv("/users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/EvolutionaryOrigin_MMseqs/Clustering/allHuman_matrix_oldestAge.tsv", sep="\t")
testisRestr = merge(testisRestr, conservation %>% select(orfID, oldest_species), by="orfID", all.x = T)
testisRestr[is.na(testisRestr)] = "human"

testisRestr = testisRestr %>% mutate(age_categorical = case_when(oldest_species == "macaca"  ~ "primate",
                                                                 oldest_species == "human"  ~ "primate",
                                                                 # oldest_species == "mouse" ~ "primate",
                                                   TRUE ~ "mammal"))

testisRestr %>% select(gene_id, coding_noncoding, age_categorical, tumorReactivated) %>% unique() %>% group_by(coding_noncoding, tumorReactivated, age_categorical) %>% count() %>% group_by(coding_noncoding, tumorReactivated) %>% mutate(proportion = n / sum(n)) %>%
  ggplot(., aes(y=proportion, x=age_categorical, fill=tumorReactivated, group=tumorReactivated)) +
  geom_bar(stat="identity", position="dodge") +
  facet_grid(tumorReactivated ~ coding_noncoding)

### aminoacid composition
hph=c("A","V","L","I","M","F","Y","W")
polar=c("S","T","N","Q")
positively_charged=c("R","H","K")
negatively_charged=c("D","E")
special_cases=c("C","G","P")

aacomposition_testisRestr = testisRestr %>% select(tumorReactivated, orfID, coding_noncoding, ORFpep) %>% unique()
## remove starting codon
aacomposition_testisRestr$ORFpep = substring(aacomposition_testisRestr$ORFpep, 2)
aacomposition_testisRestr$ORFpep = gsub("\\*","",as.character(aacomposition_testisRestr$ORFpep))


#### non-coding
nc_React = aacomposition_testisRestr %>% subset(coding_noncoding == "noncoding") %>% subset(tumorReactivated == "Reactivated")
nc_React$ORFpep = as.character(nc_React$ORFpep)
nc_NoReact = aacomposition_testisRestr %>% subset(coding_noncoding == "noncoding") %>% subset(tumorReactivated != "Reactivated")
nc_NoReact$ORFpep = as.character(nc_NoReact$ORFpep)

log2ratios_noncoding = calculate_log2ratio_df(nc_React, nc_NoReact)

log2ratios_noncoding %>% 
  mutate(aa_type = case_when(Aminoacid %in% hph ~ "hidrophobic",
                             Aminoacid %in% polar ~ "polar",
                             Aminoacid %in% positively_charged ~ "positive",
                             Aminoacid %in% negatively_charged ~ "negative",
                             Aminoacid %in% special_cases ~ "special")) %>%
  subset(Aminoacid != "*") %>% 
  drop_na() %>%
ggplot(., aes(x=reorder(Aminoacid,log2ratio), y=log2ratio, fill = aa_type)) +
  geom_bar(stat = "identity") +
  labs(y="Log2Ratio\n(Reactivated/No Reactivated)",
       x="Aminoacid",
       title="Relative enrichement of aminoacids in\nNon-coding genes Reactivated in cancer") +
  scale_fill_carto_d(palette = "Burg") +
  theme_classic()

#### coding
c_React = aacomposition_testisRestr %>% subset(coding_noncoding == "protein_coding") %>% subset(tumorReactivated == "Reactivated")
c_React$ORFpep = as.character(c_React$ORFpep)
c_NoReact = aacomposition_testisRestr %>% subset(coding_noncoding == "protein_coding") %>% subset(tumorReactivated != "Reactivated")
c_NoReact$ORFpep = as.character(c_NoReact$ORFpep)

log2ratios_coding = calculate_log2ratio_df(c_React, c_NoReact)
log2ratios_coding %>% 
  mutate(aa_type = case_when(Aminoacid %in% hph ~ "hidrophobic",
                             Aminoacid %in% polar ~ "polar",
                             Aminoacid %in% positively_charged ~ "positive",
                             Aminoacid %in% negatively_charged ~ "negative",
                             Aminoacid %in% special_cases ~ "special")) %>%
  subset(Aminoacid != "*") %>% drop_na() %>% 
ggplot(., aes(x=reorder(Aminoacid,log2ratio), y=log2ratio, fill = aa_type)) +
  geom_bar(stat = "identity") +
  labs(y="Log2Ratio\n(Reactivated/No Reactivated)",
       x="Aminoacid",
       title="Relative enrichement of aminoacids in\nProteinCoding genes Reactivated in cancer") +
  scale_fill_carto_d(palette = "Burg") +
  theme_classic()
```