---
title: "Tumor samples | Adjacent controls"
author: "Marta Espinosa"
date: "2024-10-01"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
library(dplyr)
library(tidyr)
library(ggplot2)
library(stringr)
library(purrr)
library(ggvenn)
library(ggpubr)
library(ggside)
library(bioseq)
library(rcartocolor)
library(ggbeeswarm)
library(ggbreak)
library(pheatmap)
library(stats)
library(ggfortify)

annot = read.csv("/users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/human/newReference_Resconstructed/1transcript_1gene.reconstructed.csv")
# names(gene_transcript) = c("gene_id"gene_type"gene_name")
# gene_transcript$gene_id = gsub("\\..*"gene_transcript$gene_id)

plots_wd = "/projects_eg/projects/marta/TestisRestricted_Microproteins_TSA/with_TranscriptomeReconstruction/Q0_TumorAdjacent/human/plots"
save_wd = "/projects_eg/projects/marta/TestisRestricted_Microproteins_TSA/with_TranscriptomeReconstruction/Q0_TumorAdjacent/human"

cancers = c("BRCA","BLCA","LUAD","KIRC","PRAD","LUSC","COAD","LIHC")
tcga_projects=c("TCGA-BRCA","TCGA-LUSC","TCGA-PRAD","TCGA-KIRC","TCGA-KIRP","TCGA-LUAD","TCGA-BLCA")#,"TCGA-LIHC"]
other_projects=c("GSE102101_KIRC","GSE133624_BLCA","GSE22260_PRAD","PRJEB2449_PRAD","SRP238334_KIRC","GSE214846_LIHC","GSE229705_LUAD","TCGA_COAD","SRP107326_COAD")
manuscript_projects = c("liver_adjacent_totalRNA_LIHC","hcc_normal_totalRNA_LIHC","GSE193567_LIHC","LIHC_TCGA_LIHC")
# deleted_projects=c("GSE103001_BRCA"GSE89223_PRAD")
all_projects = c(tcga_projects,other_projects,manuscript_projects)

cancers_dir = "/users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/cancers"

save_pheatmap_pdf <- function(x, filename, width=7, height=7) {
   stopifnot(!missing(x))
   stopifnot(!missing(filename))
   pdf(filename, width=width, height=height)
   grid::grid.newpage()
   grid::grid.draw(x$gtable)
   dev.off()
}

```

## PCA

```{r input, echo=F}
cancers_fc = data.frame("transcript_id" = character(),
                             "Length" = numeric(),
                             "sample" = character(),
                             "TPM"= numeric(),
                             "ctype" = character(),
                             "project" = character(),
                             "normal_tumor" = character(),
                             stringsAsFactors = F)

for(ctype_var in cancers) {
  print(ctype_var)
  
  ## table of counts input ##
  fc = read.csv(paste0(cancers_dir,"/merged_fc_",ctype_var,".csv"))
  names(fc) = gsub("\\.","-",names(fc))
  fc = merge(fc, annot, by="transcript_id")
  fc_long = fc %>% pivot_longer(cols=-c(transcript_id,gene_id, gene_name, gene_type, Length), names_to = "sample", values_to = "TPM") %>% unique()
  fc_long$ctype = ctype_var

  fc_long = fc_long %>% subset(gene_type == "protein_coding" | gene_type == "lncRNA")# | gene_type == "novel" | gene_type == "lncRNA" | gene_type == "processed_pseudogene")
  ## patients info ##
  temp_patients = read.csv(paste0(cancers_dir,"/merged_patients_",ctype_var,".csv"))
  temp_patients = temp_patients %>% pivot_longer(cols=c("normal","tumor"), names_to = "normal_tumor", values_to = "sample")

  fc_complete = merge(fc_long, temp_patients, by="sample")
  fc_complete = fc_complete %>% unique() 
  
  cancers_fc = rbind(cancers_fc, fc_complete)
  
  ####### PCA #######
  tpm = fc_complete %>% select(gene_id, sample, TPM) %>% unique() %>% pivot_wider(names_from = "sample", values_from = "TPM") 

  
  # Remove gene_id column and ensure it's numeric
  tpm_matrix <- as.matrix(tpm %>% select(-gene_id))
  # Set rownames to gene_ids
  rownames(tpm_matrix) <- tpm$gene_id
  V <- apply(tpm_matrix, 1, var)

  ######## normal-tumor
  col_annot = fc_complete %>% select(sample, normal_tumor) %>% unique()
  rownames(col_annot) = col_annot$sample
  col_annot$sample = NULL
  colors_annot = list("normal_tumor" = c("normal" = "blue","tumor" = "red"))
  # Extract the color information from col_annot
  point_colors <- colors_annot$normal_tumor[col_annot$normal_tumor]
  
  ######## project
  col_annot2 = fc_complete %>% select(sample, project) %>% unique()
  rownames(col_annot2) = col_annot2$sample
  col_annot2$sample = NULL
  
  

  # Automatically assign shapes to projects based on their unique values
  unique_projects <- unique(col_annot2$project)
  # Assign shapes from a predefined set of `pch` values (you can extend this if needed)
  shape_mapping <- setNames(seq(1, length(unique_projects)), unique_projects)
  # Create a vector of shapes based on the project in col_annot2
  point_shapes <- shape_mapping[col_annot2$project]


  # sort the results by variance in decreasing order 
  # and select the top 100 genes 
  selectedGenes <- names(V[order(V, decreasing = T)][1:100])
  ph = pheatmap(tpm_matrix[selectedGenes,], 
           scale = 'row', 
           show_rownames = FALSE, 
           annotation_col= col_annot,
           annotation_colors = colors_annot)
  save_pheatmap_pdf(ph, paste0(plots_wd,"/PDF/pheatmap_normal_tumor_",ctype_var,".pdf"))
  
  # transpose the matrix
  M <- t(tpm_matrix[selectedGenes,])
  # transform the counts to log2 scale
  M <- log2(M + 1)
  # compute PCA
  pcaResults <- prcomp(M, scale=T)
  
  pcaResults$samples<-rownames(M)
  png(filename= paste0(plots_wd,"/PNG/pca_normal_tumor_",ctype_var,".png"))
  plot(pcaResults$x[,1], pcaResults$x[,2],
       col = point_colors,  # Color points based on normal/tumor status,
       pch = point_shapes,            # Shape of points based on project
       xlab = "PC1", 
       ylab = "PC2", 
       main = "PCA Plot Colored by Normal/Tumor Status")
  
  legend("bottomright", legend = names(shape_mapping),
       pch = shape_mapping, title = "Projects")
  dev.off()
  }
```

```{r BRCA, echo=F}
BRCA = c("BRCA1","BRCA2","PTEN")
# 10.3390/ijms23073754
######################## BRCA ######################
ctype_var = "BRCA"
temp_fc = cancers_fc %>% subset(ctype == ctype_var)

biomarkers_expr = temp_fc %>% subset(gene_name %in% BRCA)
biomarkers_expr$logTPM = log(biomarkers_expr$TPM)
 ggplot(biomarkers_expr, aes(x=gene_name, y=logTPM, fill=normal_tumor)) +
  geom_boxplot(position = position_dodge(width=.8), alpha=.7) +
  geom_jitter(aes(color=normal_tumor), position=position_jitterdodge(dodge.width = 0.8), alpha=.4, size=1) +
  scale_fill_manual(values=c("#FFB900","#5773CC")) +
  scale_color_manual(values=c("#FFB900","#5773CC")) +
  stat_compare_means(label = "p.signif", paired=TRUE, label.y=2) +
  labs(x="Dataset",
       y="logTPM",
       title="BRCA BIOMARKERS") +
  theme_classic()

ggsave(file.path(plots_wd,"PNG/BRCA_biomarkers_ctype.png"))
ggsave(file.path(plots_wd,"PDF/BRCA_biomarkers_ctype.pdf"))
```

```{r HCC, echo=F}
LIHC = c("TERT","THBS4","AFP","MTM1")
# https://www.science.org/doi/10.1126/sciadv.adn3628
######################## BRCA ######################
ctype_var = "LIHC"
temp_fc = cancers_fc %>% subset(ctype == ctype_var)

biomarkers_expr = temp_fc %>% subset(gene_name %in% LIHC)
biomarkers_expr$logTPM = log(biomarkers_expr$TPM)
ggplot(biomarkers_expr, aes(x=gene_name, y=logTPM, fill=normal_tumor)) +
  geom_boxplot(position = position_dodge(width=.8)) +
  geom_jitter(aes(color=normal_tumor), position=position_jitterdodge(dodge.width = 0.8), alpha=.4, size=1) +
  scale_fill_manual(values=c("#FFB900","#5773CC")) +
  scale_color_manual(values=c("#FFB900","#5773CC")) +
  stat_compare_means(label = "p.signif", paired=TRUE, label.y=3.5) +
  labs(x="Dataset",
       y="logTPM",
       title="LIHC BIOMARKERS") +
  theme_classic() +
  theme(legend.position = "top") 

ggsave(file.path(plots_wd,"PNG/LIHC_biomarkers_ctype.png"), width=7, height=7)
ggsave(file.path(plots_wd,"PDF/LIHC_biomarkers_ctype.pdf"), width=7, height=7)

```

```{r BLCA, echo=F}
BLCA = c("PLK1", "CDK1","CNN1")
# https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5363876/
# https://www.ncbi.nlm.nih.gov/pmc/articles/PMC8918393/
######################## BRCA ######################
ctype_var = "BLCA"
temp_fc = cancers_fc %>% subset(ctype == ctype_var)

biomarkers_expr = temp_fc %>% subset(gene_name %in% BLCA)
biomarkers_expr$logTPM = log(biomarkers_expr$TPM)
ggplot(biomarkers_expr, aes(x=gene_name, y=logTPM, fill=normal_tumor)) +
  geom_boxplot(position = position_dodge(width=.8)) +
  geom_jitter(aes(color=normal_tumor), position=position_jitterdodge(dodge.width = 0.8), alpha=.4, size=1) +
  scale_fill_manual(values=c("#FFB900","#5773CC")) +
  scale_color_manual(values=c("#FFB900","#5773CC")) +
  stat_compare_means(label = "p.signif", paired=TRUE, label.y=3.5) +
  labs(x="Dataset",
       y="logTPM",
       title="BLCA BIOMARKERS") +
  theme_classic() +
  theme(legend.position = "top") 

ggsave(file.path(plots_wd,"PNG/BLCA_biomarkers_ctype.png"))
ggsave(file.path(plots_wd,"PDF/BLCA_biomarkers_ctype.pdf"))

```

```{r LUAD, echo=F}
LUAD = c("ABCA8","ADAMTS8","CEP55","PYCR1")
# https://www.nature.com/articles/s41598-022-21050-5/figures/5
######################## BRCA ######################
ctype_var = "LUAD"
temp_fc = cancers_fc %>% subset(ctype == ctype_var)

biomarkers_expr = temp_fc %>% subset(gene_name %in% LUAD)
biomarkers_expr$logTPM = log(biomarkers_expr$TPM)
ggplot(biomarkers_expr, aes(x=gene_name, y=logTPM, fill=normal_tumor)) +
  geom_boxplot(position = position_dodge(width=.8)) +
  geom_jitter(aes(color=normal_tumor), position=position_jitterdodge(dodge.width = 0.8), alpha=.4, size=1) +
  scale_fill_manual(values=c("#FFB900","#5773CC")) +
  scale_color_manual(values=c("#FFB900","#5773CC")) +
  stat_compare_means(label = "p.signif", paired=TRUE, label.y=3.5) +
  labs(x="Dataset",
       y="logTPM",
       title="LUAD BIOMARKERS") +
  theme_classic() +
  theme(legend.position = "top") 

ggsave(file.path(plots_wd,"PNG/LUAD_biomarkers_ctype.png"))
ggsave(file.path(plots_wd,"PDF/LUAD_biomarkers_ctype.pdf"))

```

```{r LUSC, echo=F}
LUSC = c("ABCA8","ADAMTS8","CEP55","PYCR1")
# https://www.nature.com/articles/s41598-022-21050-5/figures/5
######################## BRCA ######################
ctype_var = "LUSC"
temp_fc = cancers_fc %>% subset(ctype == ctype_var)

biomarkers_expr = temp_fc %>% subset(gene_name %in% LUSC)
biomarkers_expr$logTPM = log(biomarkers_expr$TPM)
ggplot(biomarkers_expr, aes(x=gene_name, y=logTPM, fill=normal_tumor)) +
  geom_boxplot(position = position_dodge(width=.8)) +
  geom_jitter(aes(color=normal_tumor), position=position_jitterdodge(dodge.width = 0.8), alpha=.4, size=1) +
  scale_fill_manual(values=c("#FFB900","#5773CC")) +
  scale_color_manual(values=c("#FFB900","#5773CC")) +
  stat_compare_means(label = "p.signif", paired=TRUE, label.y=3.5) +
  labs(x="Dataset",
       y="logTPM",
       title="LUSC BIOMARKERS") +
  theme_classic() +
  theme(legend.position = "top") 

ggsave(file.path(plots_wd,"PNG/LUSC_biomarkers_ctype.png"))
ggsave(file.path(plots_wd,"PDF/LUSC_biomarkers_ctype.pdf"))
```

```{r COAD, echo=F}
COAD = c("TRIB3","STMN4","FAM135B")
# https://www.ncbi.nlm.nih.gov/pmc/articles/PMC9130243/
######################## BRCA ######################
ctype_var = "COAD"
temp_fc = cancers_fc %>% subset(ctype == ctype_var)

biomarkers_expr = temp_fc %>% subset(gene_name %in% COAD)
biomarkers_expr$logTPM = log(biomarkers_expr$TPM)
ggplot(biomarkers_expr, aes(x=gene_name, y=logTPM, fill=normal_tumor)) +
  geom_boxplot(position = position_dodge(width=.8)) +
  geom_jitter(aes(color=normal_tumor), position=position_jitterdodge(dodge.width = 0.8), alpha=.4, size=1) +
  scale_fill_manual(values=c("#FFB900","#5773CC")) +
  scale_color_manual(values=c("#FFB900","#5773CC")) +
  stat_compare_means(label = "p.signif", paired=TRUE, label.y=3) +
  labs(x="Dataset",
       y="logTPM",
       title="COAD BIOMARKERS") +
  theme_classic() +
  theme(legend.position = "top") 

ggsave(file.path(plots_wd,"PNG/COAD_biomarkers_ctype.png"))
ggsave(file.path(plots_wd,"PDF/COAD_biomarkers_ctype.pdf"))
```

```{r PRAD, echo=F}
PRAD = c("AR","GOLM1","PCA3")
# https://www.nature.com/articles/s41598-022-21050-5/figures/5
######################## BRCA ######################
ctype_var = "PRAD"
temp_fc = cancers_fc %>% subset(ctype == ctype_var)

biomarkers_expr = temp_fc %>% subset(gene_name %in% PRAD)
biomarkers_expr$logTPM = log(biomarkers_expr$TPM)
ggplot(biomarkers_expr, aes(x=gene_name, y=logTPM, fill=normal_tumor)) +
  geom_boxplot(position = position_dodge(width=.8)) +
  geom_jitter(aes(color=normal_tumor), position=position_jitterdodge(dodge.width = 0.8), alpha=.4, size=1) +
  scale_fill_manual(values=c("#FFB900","#5773CC")) +
  scale_color_manual(values=c("#FFB900","#5773CC")) +
  stat_compare_means(label = "p.signif", paired=TRUE, label.y=3) +
  labs(x="Dataset",
       y="logTPM",
       title="PRAD BIOMARKERS") +
  theme_classic() +
  theme(legend.position = "top") 

ggsave(file.path(plots_wd,"PNG/PRAD_biomarkers_ctype.png"))
ggsave(file.path(plots_wd,"PDF/PRAD_biomarkers_ctype.pdf"))
```

```{r KIRC, echo=F}
KIRC = c("CAV1","ALB","VEGFA")
# https://www.nature.com/articles/s41598-022-21050-5/figures/5
######################## BRCA ######################
ctype_var = "KIRC"
temp_fc = cancers_fc %>% subset(ctype == ctype_var)

biomarkers_expr = temp_fc %>% subset(gene_name %in% KIRC)
biomarkers_expr$logTPM = log(biomarkers_expr$TPM)
ggplot(biomarkers_expr, aes(x=gene_name, y=logTPM, fill=normal_tumor)) +
  geom_boxplot(position = position_dodge(width=.8)) +
  geom_jitter(aes(color=normal_tumor), position=position_jitterdodge(dodge.width = 0.8), alpha=.4, size=1) +
  scale_fill_manual(values=c("#FFB900","#5773CC")) +
  scale_color_manual(values=c("#FFB900","#5773CC")) +
  stat_compare_means(label = "p.signif", paired=TRUE, label.y=3) +
  labs(x="Dataset",
       y="logTPM",
       title="KIRC BIOMARKERS") +
  theme_classic() +
  theme(legend.position = "top")

ggsave(file.path(plots_wd,"PNG/KIRC_biomarkers_ctype.png"))
ggsave(file.path(plots_wd,"PDF/KIRC_biomarkers_ctype.pdf"))
```

```{r normalTumor, echo=F}
#### TAA ####
filtered_data_TAA_long = read.csv("/projects_eg/projects/marta/TestisRestricted_Microproteins_TSA/with_TranscriptomeReconstruction/Q3_TestisRestrictedTumors/human/TAAdata_long.csv")
filtered_data_TAA_long = filtered_data_TAA_long %>% subset(ctype != "KIRP")
temporary = filtered_data_TAA_long %>% select(gene_name, ctype) %>% unique()
table(temporary$ctype)


ggplot(filtered_data_TAA_long, aes(x=gene_name, y=logTPM, color=normal_tumor)) +
  geom_quasirandom(dodge.width= .8, size=.5) +
  scale_color_manual(values=c("#FFB900","#5773CC")) +
  geom_hline(yintercept=log(1)) +
  labs(x="",
       title="Testis-Restricted genes expressed in cancer\nExpression in tumor and adjacent samples") +
  theme_classic() +
  theme(axis.text.x = element_blank(),
        axis.ticks.x = element_blank()) +
  facet_wrap(~ ctype, ncol=1, scales="free")
ggsave(file.path(plots_wd, "PNG/TAA_normaltumor_ctypes_genenames.png"), width=8, height=14)
  

#### TOv3x ####
filtered_data_TOV3x_long = read.csv("/projects_eg/projects/marta/TestisRestricted_Microproteins_TSA/with_TranscriptomeReconstruction/Q4_TestisRestricted_TumorSpecific/human/TOv3xdata_long_5percent.csv")
filtered_data_TOV3x_long = filtered_data_TOV3x_long %>% subset(ctype != "KIRP")
temporary = filtered_data_TOV3x_long %>% select(gene_name, ctype) %>% unique()
table(temporary$ctype)


ggplot(filtered_data_TOV3x_long, aes(x=gene_name, y=logTPM, color=normal_tumor)) +
  geom_quasirandom(dodge.width= .8, size=.5) +
  scale_color_manual(values=c("#FFB900","#5773CC")) +
  geom_hline(yintercept=log(1)) +
  labs(x="",
       title="Testis-Restricted genes TOv3x in cancer\nExpression in tumor and adjacent samples") +
  theme_classic() +
  theme(axis.text.x = element_text(angle=90)) +
  facet_wrap(~ ctype, ncol=1, scales="free")
ggsave(file.path(plots_wd, "PNG/TOv3x_normaltumor_ctypes_genenames.png"), width=8, height=14)

#### TOv3x + log2ratio 3x ####
TOV3xlog2ratio3x=read.csv("/projects_eg/projects/marta/TestisRestricted_Microproteins_TSA/with_TranscriptomeReconstruction/Q4_TestisRestricted_TumorSpecific/human/TOv3x_5percent_TestisRestrictedGTEx_Translated_Ctypes_log2ratio3xMEAN.csv")
filtered_data_TOV3xlog2ratio3x_long = filtered_data_TOV3x_long %>% subset(gene_name %in% TOV3xlog2ratio3x$gene_name)
filtered_data_TOV3xlog2ratio3x_long = filtered_data_TOV3xlog2ratio3x_long %>% subset(ctype != "KIRP")
temporary = filtered_data_TOV3xlog2ratio3x_long %>% select(gene_name, ctype) %>% unique()
table(temporary$ctype)

ggplot(filtered_data_TOV3xlog2ratio3x_long, aes(x=gene_name, y=logTPM, color=normal_tumor)) +
  geom_quasirandom(dodge.width= .8, size=.5) +
  scale_color_manual(values=c("#FFB900","#5773CC")) +
  geom_hline(yintercept=log(1)) +
  labs(x="",
       title="Testis-Restricted genes TOv3x & log2ratio 3x in cancer\nExpression in tumor and adjacent samples") +
  theme_classic() +
  theme(axis.text.x = element_text(angle=90)) +
  facet_wrap(~ ctype, ncol=1, scales="free")
ggsave(file.path(plots_wd, "PNG/TOv3xlog2ratio_normaltumor_ctypes_genenames.png"), width=8, height=14)


#### PER CANCER TYPE
for(ctype_var in cancers) {
  print(ctype_var)
  ### TAA ###
  print(nrow(filtered_data_TAA_long %>% subset(ctype == ctype_var) %>% select(gene_name) %>% unique()))
  filtered_data_TAA_long %>% subset(ctype == ctype_var) %>%
    ggplot(., aes(x=gene_name, y=logTPM, color=normal_tumor)) +
      geom_quasirandom(dodge.width= .8, size=.5) +
      scale_color_manual(values=c("#FFB900","#5773CC")) +
      geom_hline(yintercept=log(1)) +
      labs(x="",
           title=paste0("Testis-Restricted genes expressed in cancer | ",ctype_var,"\nExpression in tumor and adjacent samples")) +
      theme_classic() +
      theme(axis.text.x = element_text(angle=90),
          legend.position = "top")
  ggsave(paste0(plots_wd, "/PNG/TAA_normaltumor_ctypes_genenames_",ctype_var,".png"), width=8, height=7)
  

  #### TOv3x ####
  # print(nrow(filtered_data_TOV3x_long %>% subset(ctype == ctype_var) %>% select(gene_name) %>% unique()))
  # filtered_data_TOV3x_long %>% subset(ctype == ctype_var) %>%
  #   ggplot(., aes(x=gene_name, y=logTPM, color=normal_tumor)) +
  #     geom_quasirandom(dodge.width= .8, size=.5) +
  #     scale_color_manual(values=c("#FFB900","#5773CC")) +
  #     geom_hline(yintercept=log(1)) +
  #     labs(x="",
  #          title=paste0("Testis-Restricted genes TOv3x in cancer | ",ctype_var,"\nExpression in tumor and adjacent samples")) +
  #     theme_classic() +
  # theme(axis.text.x = element_text(angle=90),
  #         legend.position = "top")
  # ggsave(paste0(plots_wd, "/PNG/TOv3x_normaltumor_ctypes_genenames_",ctype_var,".png"), width=8, height=7)

  #### TOv3x + log2ratio 3x ####
  print(nrow(filtered_data_TOV3xlog2ratio3x_long %>% subset(ctype == ctype_var) %>% select(gene_name) %>% unique()))
  temp_filtered_data_TOV3xlog2ratio3x_long = filtered_data_TOV3xlog2ratio3x_long %>% subset(ctype == ctype_var)
  
  ggplot(temp_filtered_data_TOV3xlog2ratio3x_long, aes(x=gene_name, y=logTPM, color=normal_tumor)) +
    geom_quasirandom(dodge.width= .8, size=.5) +
    scale_color_manual(values=c("#FFB900","#5773CC")) +
    geom_hline(yintercept=log(1)) +
    labs(x="",
         title=paste0("Testis-Restricted genes TOv3x & log2ratio 3x in cancer | ",ctype_var,"\nExpression in tumor and adjacent samples")) +
    theme_classic() +
  theme(axis.text.x = element_text(angle=90),
          legend.position = "top")
  ggsave(paste0(plots_wd, "/PNG/TOv3xlog2ratio_normaltumor_ctypes_genenames_",ctype_var,".png"), width=8, height=7)
}

```




