{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Evolutionary origin\n",
    "\n",
    "Macaca - Mouse - Human"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>specie</th>\n",
       "      <th>age</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>human</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>macaca</td>\n",
       "      <td>25</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>mouse</td>\n",
       "      <td>90</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>opossum</td>\n",
       "      <td>180</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>platypus</td>\n",
       "      <td>200</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>chicken</td>\n",
       "      <td>310</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "     specie  age\n",
       "1     human    0\n",
       "2    macaca   25\n",
       "3     mouse   90\n",
       "4   opossum  180\n",
       "5  platypus  200\n",
       "6   chicken  310"
      ]
     },
     "execution_count": 1,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import os,re,glob\n",
    "import pandas as pd\n",
    "import numpy as np\n",
    "from collections import Counter\n",
    "from Bio.Seq import Seq\n",
    "from Bio.SeqRecord import SeqRecord\n",
    "from Bio import SeqIO\n",
    "\n",
    "users_dir = \"/users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction\"\n",
    "GENOMEDIR = \"/genomics/users/marta/genomes\"\n",
    "species=[\"mouse\",\"macaca\",\"platypus\",\"chicken\",\"opossum\"]\n",
    "\n",
    "## annotation file\n",
    "annotation=\"/users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/human/newReference_Resconstructed/gencode.v38.gffcompare.TestisLiverBrain.annotation.sorted.1transcript.sorted.NOchr.gtf\"\n",
    "transcript_gene=pd.read_csv(\"/users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/human/newReference_Resconstructed/1transcript_1gene.reconstructed.csv\")\n",
    "\n",
    "## evoDir\n",
    "evoDir=os.path.join(users_dir,\"EvolutionaryOrigin_MMseqs\")\n",
    "\n",
    "def translate_dna_to_protein(dna_seq):\n",
    "    return str(Seq(dna_seq).translate())\n",
    "\n",
    "# Function to save DF in fasta format\n",
    "def create_seqrecord_PROT(row):\n",
    "    return SeqRecord(Seq(row['protein']), id=row['header'][:-4], description=\"\")\n",
    "def create_seqrecord_DNA(row):\n",
    "    return SeqRecord(Seq(row['seq']), id=row['header'], description=\"\")\n",
    "\n",
    "species_and_years_dict = {\"specie\" : [\"human\",\"macaca\",\"mouse\",\"opossum\",\"platypus\",\"chicken\"],\n",
    "                          \"age\" : [0,25,90,180,200,310]}\n",
    "species_and_years = pd.DataFrame(data=species_and_years_dict, index=[1,2,3,4,5,6])\n",
    "species_and_years\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "## get repre sequences - both DNA and PROTEINS\n",
    "for s in species:\n",
    "    print(s)\n",
    "    riboseq_dir = os.path.join(users_dir,s,\"RiboSeq/RiboQC_RiboNovel\")\n",
    "\n",
    "    ## get all candidates\n",
    "    print(\"Reading all the candidate ORF sequences\")\n",
    "    df = pd.read_csv(os.path.join(riboseq_dir,\"Annotation/candidateORF.fa\"), header=None, sep=\"\\t\")\n",
    "    ## convert to df\n",
    "    candidates_fasta = pd.DataFrame({'header':df[0].iloc[::2].values, 'seq':df[0].iloc[1::2].values})\n",
    "    candidates_fasta['header'] = candidates_fasta['header'].str[1:]\n",
    "    candidates_fasta = candidates_fasta[~candidates_fasta['header'].str.contains(\"iORF\")]\n",
    "\n",
    "    all_repre_candidates = pd.DataFrame()\n",
    "    for sample in os.listdir(os.path.join(riboseq_dir,\"RibORF\")):\n",
    "        if sample.endswith(\"r1\"):\n",
    "            print(sample)\n",
    "\n",
    "            repre = pd.read_csv(os.path.join(riboseq_dir,\"RibORF\",sample,\"repre.valid.pred.pvalue.parameters.txt\"), sep=\"\\t\")\n",
    "            repre_fasta = candidates_fasta[candidates_fasta['header'].isin(repre.orfID.values.tolist())]\n",
    "            repre_fasta['header'] = s + \"_\" +repre_fasta['header']\n",
    "\n",
    "            ## keep only the coding and the non-coding\n",
    "            repre_fasta = repre_fasta[repre_fasta['header'].str.contains(\"canonical\")| repre_fasta['header'].str.contains(\"noncoding\")]\n",
    "\n",
    "            repre_fasta['protein'] = repre_fasta['seq'].apply(translate_dna_to_protein)\n",
    "            all_repre_candidates = pd.concat([all_repre_candidates, repre_fasta])\n",
    "\n",
    "            ## translate to protein\n",
    "            print(\"Translating to protein\")\n",
    "            out_proteins = os.path.join(riboseq_dir,\"RibORF\",sample,\"repre.valid.pred.pvalue.parameters.PROTEIN.fa\")\n",
    "            seq_records_proteins = repre_fasta.apply(create_seqrecord_PROT, axis=1).tolist()\n",
    "            SeqIO.write(seq_records_proteins, out_proteins, \"fasta\")\n",
    "\n",
    "            ## keep the dna sequences as well\n",
    "            print(\"Saving the original DNA ORF\")\n",
    "            out_dna = os.path.join(riboseq_dir,\"RibORF\",sample,\"repre.valid.pred.pvalue.parameters.DNA.fa\")\n",
    "            seq_records_dna = repre_fasta.apply(create_seqrecord_DNA, axis=1).tolist()\n",
    "            SeqIO.write(seq_records_dna, out_dna, \"fasta\")\n",
    "\n",
    "            ## get only the nonredundant\n",
    "            print(\"Getting the non-redundant\")\n",
    "            all_repre_candidates.drop_duplicates(inplace=True)\n",
    "            out_nonredundant_proteins = os.path.join(riboseq_dir,\"RibORF\")+\"/\"+s+\"_repre.valid.pred.pvalue.parameters.PROTEIN.noredunant.fa\"\n",
    "            seq_records_proteins = all_repre_candidates.apply(create_seqrecord_PROT, axis=1).tolist()\n",
    "            SeqIO.write(seq_records_proteins, out_nonredundant_proteins, \"fasta\")"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## MMseqs2\n",
    "\n",
    "Against all expressed in mouse, macaca, platypus, opossum in any tissue (non-redunant set)"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### TESTIS-EXPRESSED"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "15040"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "## Testis-expressed ORFs human\n",
    "TestisExpressed_translatedINtestis = pd.read_csv(\"/projects_eg/projects/marta/TestisRestricted_Microproteins_TSA/with_TranscriptomeReconstruction/Q1_TestisORFs/human/ribORF_humanTestis_in1.csv\")\n",
    "TestisExpressed_translatedINtestis\n",
    "\n",
    "def create_seqrecord_PROThuman(row):\n",
    "    return SeqRecord(Seq(row['ORFpep']), id=row['orfID'], description=\"\")\n",
    "\n",
    "out_nonredundant_proteins = \"/projects_eg/projects/marta/TestisRestricted_Microproteins_TSA/with_TranscriptomeReconstruction/Q1_TestisORFs/human/ribORF_humanTestis_in1.fa\"\n",
    "seq_records_proteins = TestisExpressed_translatedINtestis.apply(create_seqrecord_PROThuman, axis=1).tolist()\n",
    "SeqIO.write(seq_records_proteins, out_nonredundant_proteins, \"fasta\")"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Creating database of human (expressed in testis)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/EvolutionaryOrigin_MMseqs/MMseqsDB/ribORF_humanTestis_in1_DB exists and will be overwritten\n",
      "createdb /projects_eg/projects/marta/TestisRestricted_Microproteins_TSA/with_TranscriptomeReconstruction/Q1_TestisORFs/human/ribORF_humanTestis_in1.fa /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/EvolutionaryOrigin_MMseqs/MMseqsDB/ribORF_humanTestis_in1_DB \n",
      "\n",
      "MMseqs Version:       \t14.7e284\n",
      "Database type         \t0\n",
      "Shuffle input database\ttrue\n",
      "Createdb mode         \t0\n",
      "Write lookup file     \t1\n",
      "Offset of numeric ids \t0\n",
      "Compressed            \t0\n",
      "Verbosity             \t3\n",
      "\n",
      "Converting sequences\n",
      "[=\n",
      "Time for merging to ribORF_humanTestis_in1_DB_h: 0h 0m 1s 432ms\n",
      "Time for merging to ribORF_humanTestis_in1_DB: 0h 0m 0s 874ms\n",
      "Database type: Aminoacid\n",
      "Time for processing: 0h 0m 6s 137ms\n"
     ]
    }
   ],
   "source": [
    "%%bash -s \"$evoDir\" \"$users_dir\"\n",
    "\n",
    "# mkdir $1/MMseqsDB\n",
    "\n",
    "module load Miniconda3/4.9.2\n",
    "source /soft/system/software/Miniconda3/4.9.2/bin/activate MMseqs2\n",
    "\n",
    "mmseqs createdb /projects_eg/projects/marta/TestisRestricted_Microproteins_TSA/with_TranscriptomeReconstruction/Q1_TestisORFs/human/ribORF_humanTestis_in1.fa $1/MMseqsDB/ribORF_humanTestis_in1_DB\n"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Creating a joint database with all the specie, everything translated, no matter the tissue"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/EvolutionaryOrigin_MMseqs/MMseqsDB/speciesDB exists and will be overwritten\n",
      "createdb /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/chicken/RiboSeq/RiboQC_RiboNovel/RibORF/chicken_repre.valid.pred.pvalue.parameters.PROTEIN.noredunant.fa /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/macaca/RiboSeq/RiboQC_RiboNovel/RibORF/macaca_repre.valid.pred.pvalue.parameters.PROTEIN.noredunant.fa /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/mouse/RiboSeq/RiboQC_RiboNovel/RibORF/mouse_repre.valid.pred.pvalue.parameters.PROTEIN.noredunant.fa /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/opossum/RiboSeq/RiboQC_RiboNovel/RibORF/opossum_repre.valid.pred.pvalue.parameters.PROTEIN.noredunant.fa /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/platypus/RiboSeq/RiboQC_RiboNovel/RibORF/platypus_repre.valid.pred.pvalue.parameters.PROTEIN.noredunant.fa /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/EvolutionaryOrigin_MMseqs/MMseqsDB/speciesDB \n",
      "\n",
      "MMseqs Version:       \t14.7e284\n",
      "Database type         \t0\n",
      "Shuffle input database\ttrue\n",
      "Createdb mode         \t0\n",
      "Write lookup file     \t1\n",
      "Offset of numeric ids \t0\n",
      "Compressed            \t0\n",
      "Verbosity             \t3\n",
      "\n",
      "Converting sequences\n",
      "[===========\n",
      "Time for merging to speciesDB_h: 0h 0m 0s 611ms\n",
      "Time for merging to speciesDB: 0h 0m 1s 385ms\n",
      "Database type: Aminoacid\n",
      "Time for processing: 0h 0m 5s 183ms\n"
     ]
    }
   ],
   "source": [
    "%%bash -s \"$evoDir\" \"$users_dir\"\n",
    "\n",
    "module load Miniconda3/4.9.2\n",
    "source /soft/system/software/Miniconda3/4.9.2/bin/activate MMseqs2\n",
    "\n",
    "# species=\"platypus mouse macaca chicken\"\n",
    "# for s in $species; do\n",
    "#     echo $s\n",
    "\n",
    "#     ### create DB\n",
    "#     mmseqs createdb $2/$s/RiboSeq/RiboQC_RiboNovel/RibORF/repre.valid.pred.pvalue.parameters.PROTEIN.noredunant.fa $1/MMseqsDB/${s}_noredunantProts_DB\n",
    "\n",
    "# done\n",
    "\n",
    "mmseqs createdb $2/*/RiboSeq/RiboQC_RiboNovel*/RibORF/*_repre.valid.pred.pvalue.parameters.PROTEIN.noredunant.fa $1/MMseqsDB/speciesDB\n",
    "\n"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Creating a joint database with all the specie AND human for clustering, everything translated for the species, no matter the tissue, but only translated in 1 for testis"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/EvolutionaryOrigin_MMseqs/MMseqsDB/species_and_humanDB exists and will be overwritten\n",
      "createdb /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/chicken/RiboSeq/RiboQC_RiboNovel/RibORF/chicken_repre.valid.pred.pvalue.parameters.PROTEIN.noredunant.fa /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/macaca/RiboSeq/RiboQC_RiboNovel/RibORF/macaca_repre.valid.pred.pvalue.parameters.PROTEIN.noredunant.fa /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/mouse/RiboSeq/RiboQC_RiboNovel/RibORF/mouse_repre.valid.pred.pvalue.parameters.PROTEIN.noredunant.fa /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/opossum/RiboSeq/RiboQC_RiboNovel/RibORF/opossum_repre.valid.pred.pvalue.parameters.PROTEIN.noredunant.fa /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/platypus/RiboSeq/RiboQC_RiboNovel/RibORF/platypus_repre.valid.pred.pvalue.parameters.PROTEIN.noredunant.fa /projects_eg/projects/marta/TestisRestricted_Microproteins_TSA/with_TranscriptomeReconstruction/Q1_TestisORFs/human/ribORF_humanTestis_in1.fa /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/EvolutionaryOrigin_MMseqs/MMseqsDB/species_and_humanDB \n",
      "\n",
      "MMseqs Version:       \t14.7e284\n",
      "Database type         \t0\n",
      "Shuffle input database\ttrue\n",
      "Createdb mode         \t0\n",
      "Write lookup file     \t1\n",
      "Offset of numeric ids \t0\n",
      "Compressed            \t0\n",
      "Verbosity             \t3\n",
      "\n",
      "Converting sequences\n",
      "[=============\n",
      "Time for merging to species_and_humanDB_h: 0h 0m 1s 114ms\n",
      "Time for merging to species_and_humanDB: 0h 0m 1s 663ms\n",
      "Database type: Aminoacid\n",
      "Time for processing: 0h 0m 7s 479ms\n"
     ]
    }
   ],
   "source": [
    "%%bash -s \"$evoDir\" \"$users_dir\"\n",
    "\n",
    "module load Miniconda3/4.9.2\n",
    "source /soft/system/software/Miniconda3/4.9.2/bin/activate MMseqs2\n",
    "\n",
    "mmseqs createdb $2/*/RiboSeq/RiboQC_RiboNovel*/RibORF/*_repre.valid.pred.pvalue.parameters.PROTEIN.noredunant.fa /projects_eg/projects/marta/TestisRestricted_Microproteins_TSA/with_TranscriptomeReconstruction/Q1_TestisORFs/human/ribORF_humanTestis_in1.fa $1/MMseqsDB/species_and_humanDB\n",
    "\n"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Searching"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "search -a -s 6 /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/EvolutionaryOrigin_MMseqs/MMseqsDB/ribORF_humanTestis_in1_DB /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/EvolutionaryOrigin_MMseqs/MMseqsDB/speciesDB /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/EvolutionaryOrigin_MMseqs/Searching/resultDB /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/EvolutionaryOrigin_MMseqs/tmp \n",
      "\n",
      "MMseqs Version:                        \t14.7e284\n",
      "Substitution matrix                    \taa:blosum62.out,nucl:nucleotide.out\n",
      "Add backtrace                          \ttrue\n",
      "Alignment mode                         \t2\n",
      "Alignment mode                         \t0\n",
      "Allow wrapped scoring                  \tfalse\n",
      "E-value threshold                      \t0.001\n",
      "Seq. id. threshold                     \t0\n",
      "Min alignment length                   \t0\n",
      "Seq. id. mode                          \t0\n",
      "Alternative alignments                 \t0\n",
      "Coverage threshold                     \t0\n",
      "Coverage mode                          \t0\n",
      "Max sequence length                    \t65535\n",
      "Compositional bias                     \t1\n",
      "Compositional bias                     \t1\n",
      "Max reject                             \t2147483647\n",
      "Max accept                             \t2147483647\n",
      "Include identical seq. id.             \tfalse\n",
      "Preload mode                           \t0\n",
      "Pseudo count a                         \tsubstitution:1.100,context:1.400\n",
      "Pseudo count b                         \tsubstitution:4.100,context:5.800\n",
      "Score bias                             \t0\n",
      "Realign hits                           \tfalse\n",
      "Realign score bias                     \t-0.2\n",
      "Realign max seqs                       \t2147483647\n",
      "Correlation score weight               \t0\n",
      "Gap open cost                          \taa:11,nucl:5\n",
      "Gap extension cost                     \taa:1,nucl:2\n",
      "Zdrop                                  \t40\n",
      "Threads                                \t24\n",
      "Compressed                             \t0\n",
      "Verbosity                              \t3\n",
      "Seed substitution matrix               \taa:VTML80.out,nucl:nucleotide.out\n",
      "Sensitivity                            \t6\n",
      "k-mer length                           \t0\n",
      "k-score                                \tseq:2147483647,prof:2147483647\n",
      "Alphabet size                          \taa:21,nucl:5\n",
      "Max results per query                  \t300\n",
      "Split database                         \t0\n",
      "Split mode                             \t2\n",
      "Split memory limit                     \t0\n",
      "Diagonal scoring                       \ttrue\n",
      "Exact k-mer matching                   \t0\n",
      "Mask residues                          \t1\n",
      "Mask residues probability              \t0.9\n",
      "Mask lower case residues               \t0\n",
      "Minimum diagonal score                 \t15\n",
      "Selected taxa                          \t\n",
      "Spaced k-mers                          \t1\n",
      "Spaced k-mer pattern                   \t\n",
      "Local temporary path                   \t\n",
      "Rescore mode                           \t0\n",
      "Remove hits by seq. id. and coverage   \tfalse\n",
      "Sort results                           \t0\n",
      "Mask profile                           \t1\n",
      "Profile E-value threshold              \t0.1\n",
      "Global sequence weighting              \tfalse\n",
      "Allow deletions                        \tfalse\n",
      "Filter MSA                             \t1\n",
      "Use filter only at N seqs              \t0\n",
      "Maximum seq. id. threshold             \t0.9\n",
      "Minimum seq. id.                       \t0.0\n",
      "Minimum score per column               \t-20\n",
      "Minimum coverage                       \t0\n",
      "Select N most diverse seqs             \t1000\n",
      "Pseudo count mode                      \t0\n",
      "Gap pseudo count                       \t10\n",
      "Min codons in orf                      \t30\n",
      "Max codons in length                   \t32734\n",
      "Max orf gaps                           \t2147483647\n",
      "Contig start mode                      \t2\n",
      "Contig end mode                        \t2\n",
      "Orf start mode                         \t1\n",
      "Forward frames                         \t1,2,3\n",
      "Reverse frames                         \t1,2,3\n",
      "Translation table                      \t1\n",
      "Translate orf                          \t0\n",
      "Use all table starts                   \tfalse\n",
      "Offset of numeric ids                  \t0\n",
      "Create lookup                          \t0\n",
      "Add orf stop                           \tfalse\n",
      "Overlap between sequences              \t0\n",
      "Sequence split mode                    \t1\n",
      "Header split mode                      \t0\n",
      "Chain overlapping alignments           \t0\n",
      "Merge query                            \t1\n",
      "Search type                            \t0\n",
      "Search iterations                      \t1\n",
      "Start sensitivity                      \t4\n",
      "Search steps                           \t1\n",
      "Exhaustive search mode                 \tfalse\n",
      "Filter results during exhaustive search\t0\n",
      "Strand selection                       \t1\n",
      "LCA search mode                        \tfalse\n",
      "Disk space limit                       \t0\n",
      "MPI runner                             \t\n",
      "Force restart with latest tmp          \tfalse\n",
      "Remove temporary files                 \tfalse\n",
      "\n",
      "prefilter /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/EvolutionaryOrigin_MMseqs/MMseqsDB/ribORF_humanTestis_in1_DB /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/EvolutionaryOrigin_MMseqs/MMseqsDB/speciesDB /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/EvolutionaryOrigin_MMseqs/tmp/10268209260104502318/pref_0 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' --seed-sub-mat 'aa:VTML80.out,nucl:nucleotide.out' -k 0 --k-score seq:2147483647,prof:2147483647 --alph-size aa:21,nucl:5 --max-seq-len 65535 --max-seqs 300 --split 0 --split-mode 2 --split-memory-limit 0 -c 0 --cov-mode 0 --comp-bias-corr 1 --comp-bias-corr-scale 1 --diag-score 1 --exact-kmer-matching 0 --mask 1 --mask-prob 0.9 --mask-lower-case 0 --min-ungapped-score 15 --add-self-matches 0 --spaced-kmer-mode 1 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --threads 24 --compressed 0 -v 3 -s 6.0 \n",
      "\n",
      "Query database size: 15040 type: Aminoacid\n",
      "Estimated memory consumption: 1G\n",
      "Target database size: 115413 type: Aminoacid\n",
      "Index table k-mer threshold: 109 at k-mer size 6 \n",
      "Index table: counting k-mers\n",
      "[=================================================================] 115.41K 0s 654ms\n",
      "Index table: Masked residues: 1929565\n",
      "Index table: fill\n",
      "[=================================================================] 115.41K 0s 434ms\n",
      "Index statistics\n",
      "Entries:          42415036\n",
      "DB size:          730 MB\n",
      "Avg k-mer size:   0.662735\n",
      "Top 10 k-mers\n",
      "    DVVFEE\t478\n",
      "    LLLLLL\t278\n",
      "    TFDAFT\t246\n",
      "    VMEYLV\t232\n",
      "    LYDMNY\t224\n",
      "    LLPQLY\t183\n",
      "    VHSGRT\t150\n",
      "    LGGFVY\t134\n",
      "    SSSSGS\t132\n",
      "    IGGFVY\t132\n",
      "Time for index table init: 0h 0m 1s 545ms\n",
      "Process prefiltering step 1 of 1\n",
      "\n",
      "k-mer similarity threshold: 109\n",
      "Starting prefiltering scores calculation (step 1 of 1)\n",
      "Query db start 1 to 15040\n",
      "Target db start 1 to 115413\n",
      "[=================================================================] 15.04K 10s 680ms\n",
      "\n",
      "418.101239 k-mers per position\n",
      "138466 DB matches per sequence\n",
      "3 overflows\n",
      "0 queries produce too many hits (truncated result)\n",
      "263 sequences passed prefiltering per query sequence\n",
      "300 median result list length\n",
      "246 sequences with 0 size result lists\n",
      "Time for merging to pref_0: 0h 0m 0s 270ms\n",
      "Time for processing: 0h 0m 14s 547ms\n",
      "align /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/EvolutionaryOrigin_MMseqs/MMseqsDB/ribORF_humanTestis_in1_DB /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/EvolutionaryOrigin_MMseqs/MMseqsDB/speciesDB /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/EvolutionaryOrigin_MMseqs/tmp/10268209260104502318/pref_0 /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/EvolutionaryOrigin_MMseqs/Searching/resultDB --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' -a 1 --alignment-mode 2 --alignment-output-mode 0 --wrapped-scoring 0 -e 0.001 --min-seq-id 0 --min-aln-len 0 --seq-id-mode 0 --alt-ali 0 -c 0 --cov-mode 0 --max-seq-len 65535 --comp-bias-corr 1 --comp-bias-corr-scale 1 --max-rejected 2147483647 --max-accept 2147483647 --add-self-matches 0 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --score-bias 0 --realign 0 --realign-score-bias -0.2 --realign-max-seqs 2147483647 --corr-score-weight 0 --gap-open aa:11,nucl:5 --gap-extend aa:1,nucl:2 --zdrop 40 --threads 24 --compressed 0 -v 3 \n",
      "\n",
      "Compute score, coverage and sequence identity\n",
      "Query database size: 15040 type: Aminoacid\n",
      "Target database size: 115413 type: Aminoacid\n",
      "Calculation of alignments\n",
      "[=================================================================] 15.04K 58s 47ms\n",
      "Time for merging to resultDB: 0h 0m 0s 207ms\n",
      "3957772 alignments calculated\n",
      "966626 sequence pairs passed the thresholds (0.244235 of overall calculated)\n",
      "64.270348 hits per query sequence\n",
      "Time for processing: 0h 0m 59s 968ms\n",
      "convertalis /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/EvolutionaryOrigin_MMseqs/MMseqsDB/ribORF_humanTestis_in1_DB /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/EvolutionaryOrigin_MMseqs/MMseqsDB/speciesDB /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/EvolutionaryOrigin_MMseqs/Searching/resultDB /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/EvolutionaryOrigin_MMseqs/Searching/resultDB.m8 \n",
      "\n",
      "MMseqs Version:        \t14.7e284\n",
      "Substitution matrix    \taa:blosum62.out,nucl:nucleotide.out\n",
      "Alignment format       \t0\n",
      "Format alignment output\tquery,target,fident,alnlen,mismatch,gapopen,qstart,qend,tstart,tend,evalue,bits\n",
      "Translation table      \t1\n",
      "Gap open cost          \taa:11,nucl:5\n",
      "Gap extension cost     \taa:1,nucl:2\n",
      "Database output        \tfalse\n",
      "Preload mode           \t0\n",
      "Search type            \t0\n",
      "Threads                \t24\n",
      "Compressed             \t0\n",
      "Verbosity              \t3\n",
      "\n",
      "[=================================================================] 15.04K 1s 52ms\n",
      "Time for merging to resultDB.m8: 0h 0m 2s 576ms\n",
      "Time for processing: 0h 0m 6s 251ms\n"
     ]
    }
   ],
   "source": [
    "%%bash -s \"$evoDir\" \"$users_dir\"\n",
    "\n",
    "module load Miniconda3/4.9.2\n",
    "source /soft/system/software/Miniconda3/4.9.2/bin/activate MMseqs2\n",
    "mkdir -p $1/Searching\n",
    "\n",
    "rm $1/Searching/resultDB*\n",
    "## The alignment consists of two steps the prefilter and alignment. To run the search, type:\n",
    "mmseqs search -a -s 6 $1/MMseqsDB/ribORF_humanTestis_in1_DB $1/MMseqsDB/speciesDB $1/Searching/resultDB $1/tmp\n",
    "\n",
    "# Then, convert the result database into a BLAST tab formatted file (option -m 8 in legacy blast, -outfmt 6 in blast+):\n",
    "# Format alignment output\tquery,target,fident,alnlen,mismatch,gapopen,qstart,qend,tstart,tend,evalue,bits\n",
    "\n",
    "mmseqs convertalis $1/MMseqsDB/ribORF_humanTestis_in1_DB $1/MMseqsDB/speciesDB $1/Searching/resultDB $1/Searching/resultDB.m8"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Clustering"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "cluster -c 0.5 --min-seq-id 0.5 /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/EvolutionaryOrigin_MMseqs/MMseqsDB/species_and_humanDB /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/EvolutionaryOrigin_MMseqs/Clustering/species_and_human_cluDB /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/EvolutionaryOrigin_MMseqs/tmp \n",
      "\n",
      "MMseqs Version:                     \t14.7e284\n",
      "Substitution matrix                 \taa:blosum62.out,nucl:nucleotide.out\n",
      "Seed substitution matrix            \taa:VTML80.out,nucl:nucleotide.out\n",
      "Sensitivity                         \t4\n",
      "k-mer length                        \t0\n",
      "k-score                             \tseq:2147483647,prof:2147483647\n",
      "Alphabet size                       \taa:21,nucl:5\n",
      "Max sequence length                 \t65535\n",
      "Max results per query               \t20\n",
      "Split database                      \t0\n",
      "Split mode                          \t2\n",
      "Split memory limit                  \t0\n",
      "Coverage threshold                  \t0.5\n",
      "Coverage mode                       \t0\n",
      "Compositional bias                  \t1\n",
      "Compositional bias                  \t1\n",
      "Diagonal scoring                    \ttrue\n",
      "Exact k-mer matching                \t0\n",
      "Mask residues                       \t1\n",
      "Mask residues probability           \t0.9\n",
      "Mask lower case residues            \t0\n",
      "Minimum diagonal score              \t15\n",
      "Selected taxa                       \t\n",
      "Include identical seq. id.          \tfalse\n",
      "Spaced k-mers                       \t1\n",
      "Preload mode                        \t0\n",
      "Pseudo count a                      \tsubstitution:1.100,context:1.400\n",
      "Pseudo count b                      \tsubstitution:4.100,context:5.800\n",
      "Spaced k-mer pattern                \t\n",
      "Local temporary path                \t\n",
      "Threads                             \t24\n",
      "Compressed                          \t0\n",
      "Verbosity                           \t3\n",
      "Add backtrace                       \tfalse\n",
      "Alignment mode                      \t3\n",
      "Alignment mode                      \t0\n",
      "Allow wrapped scoring               \tfalse\n",
      "E-value threshold                   \t0.001\n",
      "Seq. id. threshold                  \t0.5\n",
      "Min alignment length                \t0\n",
      "Seq. id. mode                       \t0\n",
      "Alternative alignments              \t0\n",
      "Max reject                          \t2147483647\n",
      "Max accept                          \t2147483647\n",
      "Score bias                          \t0\n",
      "Realign hits                        \tfalse\n",
      "Realign score bias                  \t-0.2\n",
      "Realign max seqs                    \t2147483647\n",
      "Correlation score weight            \t0\n",
      "Gap open cost                       \taa:11,nucl:5\n",
      "Gap extension cost                  \taa:1,nucl:2\n",
      "Zdrop                               \t40\n",
      "Rescore mode                        \t0\n",
      "Remove hits by seq. id. and coverage\tfalse\n",
      "Sort results                        \t0\n",
      "Cluster mode                        \t0\n",
      "Max connected component depth       \t1000\n",
      "Similarity type                     \t2\n",
      "Single step clustering              \tfalse\n",
      "Cascaded clustering steps           \t3\n",
      "Cluster reassign                    \tfalse\n",
      "Remove temporary files              \tfalse\n",
      "Force restart with latest tmp       \tfalse\n",
      "MPI runner                          \t\n",
      "k-mers per sequence                 \t21\n",
      "Scale k-mers per sequence           \taa:0.000,nucl:0.200\n",
      "Adjust k-mer length                 \tfalse\n",
      "Shift hash                          \t67\n",
      "Include only extendable             \tfalse\n",
      "Skip repeating k-mers               \tfalse\n",
      "\n",
      "Set cluster sensitivity to -s 3.000000\n",
      "Set cluster mode SET COVER\n",
      "Set cluster iterations to 3\n",
      "mergeclusters /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/EvolutionaryOrigin_MMseqs/MMseqsDB/species_and_humanDB /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/EvolutionaryOrigin_MMseqs/Clustering/species_and_human_cluDB /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/EvolutionaryOrigin_MMseqs/tmp/15556508981604495086/clu_redundancy /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/EvolutionaryOrigin_MMseqs/tmp/15556508981604495086/clu_step0 /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/EvolutionaryOrigin_MMseqs/tmp/15556508981604495086/clu_step1 /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/EvolutionaryOrigin_MMseqs/tmp/15556508981604495086/clu_step2 --threads 24 --compressed 0 -v 3 \n",
      "\n",
      "Clustering step 1\n",
      "[=================================================================] 71.74K 0s 12ms\n",
      "Clustering step 2\n",
      "[=================================================================] 64.19K 0s 30ms\n",
      "Clustering step 3\n",
      "[=================================================================] 63.74K 0s 41ms\n",
      "Clustering step 4\n",
      "[=================================================================] 63.66K 0s 52ms\n",
      "Write merged clustering\n",
      "[=================================================================] 130.45K 0s 729ms\n",
      "Time for merging to species_and_human_cluDB: 0h 0m 0s 436ms\n",
      "Time for processing: 0h 0m 1s 411ms\n",
      "createtsv /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/EvolutionaryOrigin_MMseqs/MMseqsDB/species_and_humanDB /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/EvolutionaryOrigin_MMseqs/MMseqsDB/species_and_humanDB /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/EvolutionaryOrigin_MMseqs/Clustering/species_and_human_cluDB /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/EvolutionaryOrigin_MMseqs/Clustering/species_and_human_cluDB.tsv \n",
      "\n",
      "MMseqs Version:                 \t14.7e284\n",
      "First sequence as representative\tfalse\n",
      "Target column                   \t1\n",
      "Add full header                 \tfalse\n",
      "Sequence source                 \t0\n",
      "Database output                 \tfalse\n",
      "Threads                         \t24\n",
      "Compressed                      \t0\n",
      "Verbosity                       \t3\n",
      "\n",
      "Time for merging to species_and_human_cluDB.tsv: 0h 0m 0s 872ms\n",
      "Time for processing: 0h 0m 2s 287ms\n"
     ]
    }
   ],
   "source": [
    "%%bash -s \"$evoDir\" \"$users_dir\"\n",
    "\n",
    "module load Miniconda3/4.9.2\n",
    "source /soft/system/software/Miniconda3/4.9.2/bin/activate MMseqs2\n",
    "\n",
    "mkdir -p $1/Clustering\n",
    "\n",
    "rm -r $1/Clustering/species_and_human_cluDB*\n",
    "\n",
    "## cluster all species and human\n",
    "mmseqs cluster -c 0.5 --min-seq-id 0.5 $1/MMseqsDB/species_and_humanDB $1/Clustering/species_and_human_cluDB $1/tmp\n",
    "mmseqs createtsv $1/MMseqsDB/species_and_humanDB $1/MMseqsDB/species_and_humanDB $1/Clustering/species_and_human_cluDB $1/Clustering/species_and_human_cluDB.tsv"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Analysis"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "all human\n",
      "13368\n",
      "all human after annotation\n",
      "13368\n",
      "noncoding all\n",
      "1819\n",
      "1590\n",
      "Conserved noncoding human ORFs\n",
      "86\n",
      "Non-Conserved noncoding human ORFs\n",
      "1590\n",
      "Canonical all\n",
      "11549\n",
      "2547\n",
      "Conserved canonical human ORFs\n",
      "2132\n",
      "Conserved canonical human ORFs as noncoding in other species\n",
      "2107\n",
      "Conserved canonical human ORFs as canonical in other species\n",
      "244\n",
      "Non-Conserved canonical human ORFs\n",
      "2547\n"
     ]
    }
   ],
   "source": [
    "clustering = pd.read_csv(os.path.join(evoDir,\"Clustering/species_and_human_cluDB.tsv\"), header=None, sep=\"\\t\")\n",
    "clustering = clustering[clustering[0].str.startswith(\"ENST\")]\n",
    "print(\"all human\")\n",
    "print(len(clustering))\n",
    "clustering['transcript_id'] = clustering[0].str.split(\":\", expand=True)[0]\n",
    "clustering['transcript_id'] = clustering[0].str.split(\".\", expand=True)[0]\n",
    "\n",
    "clustering = clustering.merge(transcript_gene, on=\"transcript_id\")\n",
    "print(\"all human after annotation\")\n",
    "print(len(clustering))\n",
    "clustering = clustering[[\"gene_name\",\"gene_type\",0,1]]\n",
    "\n",
    "## non-coding\n",
    "noncoding = clustering[clustering[0].str.contains(\"noncoding\")]\n",
    "noncoding.to_csv(os.path.join(evoDir,\"Clustering/noncodingHuman_speciesDB.tsv\"), header=None, index=None, sep=\"\\t\")\n",
    "print(\"noncoding all\")\n",
    "print(len(noncoding))\n",
    "print(len(set(noncoding[0].values.tolist())))\n",
    "\n",
    "noncoding_conserved = noncoding[~noncoding[1].str.startswith(\"ENST\")]\n",
    "noncoding_conserved.to_csv(os.path.join(evoDir,\"Clustering/noncodingHuman_Conserved.tsv\"), header=None, index=None, sep=\"\\t\")\n",
    "print(\"Conserved noncoding human ORFs\")\n",
    "print(len(set(noncoding_conserved[0].values.tolist())))\n",
    "noncoding_conserved.columns = ['gene_name','gene_type','orfID',1]\n",
    "\n",
    "noncoding_NonConserved = noncoding[noncoding[1].str.startswith(\"ENST\")]\n",
    "noncoding_NonConserved.to_csv(os.path.join(evoDir,\"Clustering/noncodingHuman_NonConserved.tsv\"), header=None, index=None, sep=\"\\t\")\n",
    "print(\"Non-Conserved noncoding human ORFs\")\n",
    "print(len(set(noncoding_NonConserved[0].values.tolist())))\n",
    "noncoding_NonConserved.columns = ['gene_name','gene_type','orfID',1]\n",
    "\n",
    "\n",
    "## canonical\n",
    "canonical = clustering[clustering[0].str.contains(\"canonical\")]\n",
    "print(\"Canonical all\")\n",
    "print(len(canonical))\n",
    "print(len(set(canonical[0].values.tolist())))\n",
    "canonical.to_csv(os.path.join(evoDir,\"Clustering/canonicalHuman_speciesDB.tsv\"), header=None, index=None, sep=\"\\t\")\n",
    "\n",
    "canonical_conserved = canonical[~canonical[1].str.startswith(\"ENST\")]\n",
    "canonical_conserved.to_csv(os.path.join(evoDir,\"Clustering/canonicalHuman_Conserved.tsv\"), header=None, index=None, sep=\"\\t\")\n",
    "print(\"Conserved canonical human ORFs\")\n",
    "print(len(set(canonical_conserved[0].values.tolist())))\n",
    "canonical_conserved.columns = ['gene_name','gene_type','orfID',1]\n",
    "\n",
    "canonical_conserved_noncoding = canonical_conserved[~canonical_conserved[1].str.contains(\"noncoding\")]\n",
    "canonical_conserved_noncoding.to_csv(os.path.join(evoDir,\"Clustering/canonicalHuman_ConservedAsNoncoding.tsv\"), header=None, index=None, sep=\"\\t\")\n",
    "print(\"Conserved canonical human ORFs as noncoding in other species\")\n",
    "print(len(set(canonical_conserved_noncoding['orfID'].values.tolist())))\n",
    "\n",
    "canonical_conserved_canonical = canonical_conserved[~canonical_conserved[1].str.contains(\"canonical\")]\n",
    "canonical_conserved_canonical.to_csv(os.path.join(evoDir,\"Clustering/canonicalHuman_ConservedAsCanonical.tsv\"), header=None, index=None, sep=\"\\t\")\n",
    "print(\"Conserved canonical human ORFs as canonical in other species\")\n",
    "print(len(set(canonical_conserved_canonical['orfID'].values.tolist())))\n",
    "\n",
    "canonical_NonConserved = canonical[canonical[1].str.startswith(\"ENST\")]\n",
    "canonical_NonConserved.to_csv(os.path.join(evoDir,\"Clustering/canonicalHuman_NonConserved.tsv\"), header=None, index=None, sep=\"\\t\")\n",
    "print(\"Non-Conserved canonical human ORFs\")\n",
    "print(len(set(canonical_NonConserved[0].values.tolist())))\n",
    "canonical_NonConserved.columns = ['gene_name','gene_type','orfID',1]\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "86\n",
      "86\n",
      "86\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_3596687/3091240473.py:4: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  noncoding_conserved_to_matrix['specie'] = noncoding_conserved_to_matrix[1].str.split(\"_\", expand=True)[0]\n",
      "/tmp/ipykernel_3596687/3091240473.py:6: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  noncoding_conserved_to_matrix['presence'] = 1\n",
      "/tmp/ipykernel_3596687/3091240473.py:26: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  noncoding_conserved_matrix_ordered['sum'] = noncoding_conserved_matrix_ordered.select_dtypes(include='number').sum(axis=1)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "86\n"
     ]
    }
   ],
   "source": [
    "noncoding_conserved_to_matrix = noncoding_conserved\n",
    "print(len(set(noncoding_conserved_to_matrix.orfID.values.tolist())))\n",
    "# Assuming noncoding_conserved_to_matrix is already defined\n",
    "noncoding_conserved_to_matrix['specie'] = noncoding_conserved_to_matrix[1].str.split(\"_\", expand=True)[0]\n",
    "noncoding_conserved_to_matrix = noncoding_conserved_to_matrix[['orfID', 'specie']]\n",
    "noncoding_conserved_to_matrix['presence'] = 1\n",
    "\n",
    "print(len(set(noncoding_conserved_to_matrix.orfID.values.tolist())))\n",
    "noncoding_conserved_to_matrix\n",
    "\n",
    "# Pivoting the data without creating an 'id', and keeping 'orfID' as the unique identifier\n",
    "noncoding_conserved_matrix = noncoding_conserved_to_matrix.pivot_table(index='orfID', columns='specie', values='presence', aggfunc='max')\n",
    "\n",
    "# Fill NaN values with 0\n",
    "noncoding_conserved_matrix = noncoding_conserved_matrix.fillna(0)\n",
    "\n",
    "# Reset index so 'orfID' becomes a column instead of an index\n",
    "noncoding_conserved_matrix = noncoding_conserved_matrix.reset_index()\n",
    "\n",
    "# Rename columns as necessary\n",
    "noncoding_conserved_matrix.columns = [\"orfID\", \"chicken\", \"macaca\", \"mouse\", \"opossum\", \"platypus\"]\n",
    "\n",
    "# Reorder columns for the final matrix\n",
    "noncoding_conserved_matrix_ordered = noncoding_conserved_matrix[[\"orfID\", \"macaca\", \"mouse\", \"opossum\", \"platypus\", \"chicken\"]]\n",
    "\n",
    "# Calculate sum across species columns to identify how many species are present\n",
    "noncoding_conserved_matrix_ordered['sum'] = noncoding_conserved_matrix_ordered.select_dtypes(include='number').sum(axis=1)\n",
    "\n",
    "# Sort by the sum column to order by the number of species present\n",
    "noncoding_conserved_matrix_ordered = noncoding_conserved_matrix_ordered.sort_values(by=\"sum\", ascending=False)\n",
    "print(len(set(noncoding_conserved_matrix_ordered.orfID.values.tolist())))\n",
    "\n",
    "# Display the result\n",
    "noncoding_conserved_matrix_ordered =  noncoding_conserved_matrix_ordered.merge(noncoding_conserved[['gene_name','gene_type','orfID']], on=\"orfID\")\n",
    "noncoding_conserved_matrix_ordered.drop_duplicates(inplace=True)\n",
    "\n",
    "noncoding_conserved_matrix_ordered.to_csv(os.path.join(evoDir,\"Clustering/noncodingHuman_Conserved_matrix.tsv\"), sep=\"\\t\", index=None)\n",
    "print(len(set(noncoding_conserved_matrix_ordered.orfID.values.tolist())))\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "1590\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_3596687/4161199752.py:3: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  noncoding_conserved_touse['specie'] = noncoding_conserved_touse[1].str.split(\"_\", expand=True)[0]\n",
      "/tmp/ipykernel_3596687/4161199752.py:8: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  noncoding_to_matrix['presence'] = np.where(noncoding_to_matrix['specie'].isna(), 0, 1)\n",
      "/tmp/ipykernel_3596687/4161199752.py:36: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  noncoding_matrix_ordered['sum'] = noncoding_matrix_ordered.select_dtypes(include='number').sum(axis=1)\n"
     ]
    }
   ],
   "source": [
    "### all noncoding\n",
    "noncoding_conserved_touse = noncoding_conserved\n",
    "noncoding_conserved_touse['specie'] = noncoding_conserved_touse[1].str.split(\"_\", expand=True)[0]\n",
    "# noncoding_NonConserved.drop(\"specie\",axis=1, inplace=True)\n",
    "noncoding_to_matrix_raw = pd.concat([noncoding_conserved, noncoding_NonConserved])\n",
    "\n",
    "noncoding_to_matrix = noncoding_to_matrix_raw[['orfID', 'specie']]\n",
    "noncoding_to_matrix['presence'] = np.where(noncoding_to_matrix['specie'].isna(), 0, 1)\n",
    "\n",
    "\n",
    "human = pd.DataFrame({\n",
    "    'orfID': noncoding_to_matrix['orfID'].unique(),\n",
    "    'specie': 'human',\n",
    "    'presence': 1\n",
    "})\n",
    "\n",
    "noncoding_to_matrix = pd.concat([human, noncoding_to_matrix], ignore_index=True)\n",
    "noncoding_to_matrix.dropna(inplace=True)\n",
    "\n",
    "# Pivoting the data without creating an 'id', and keeping 'orfID' as the unique identifier\n",
    "noncoding_matrix = noncoding_to_matrix.pivot_table(index='orfID', columns='specie', values='presence', aggfunc='max')\n",
    "noncoding_matrix\n",
    "# Fill NaN values with 0\n",
    "noncoding_matrix = noncoding_matrix.fillna(0)\n",
    "\n",
    "# Reset index so 'orfID' becomes a column instead of an index\n",
    "noncoding_matrix = noncoding_matrix.reset_index()\n",
    "\n",
    "# Rename columns as necessary\n",
    "noncoding_matrix.columns = [\"orfID\", \"chicken\", \"human\", \"macaca\", \"mouse\", \"opossum\", \"platypus\"]\n",
    "\n",
    "# Reorder columns for the final matrix\n",
    "noncoding_matrix_ordered = noncoding_matrix[[\"orfID\", \"human\",\"macaca\", \"mouse\", \"opossum\", \"platypus\", \"chicken\"]]\n",
    "\n",
    "# Calculate sum across species columns to identify how many species are present\n",
    "noncoding_matrix_ordered['sum'] = noncoding_matrix_ordered.select_dtypes(include='number').sum(axis=1)\n",
    "\n",
    "# Sort by the sum column to order by the number of species present\n",
    "noncoding_matrix_ordered = noncoding_matrix_ordered.sort_values(by=\"sum\", ascending=False)\n",
    "\n",
    "# Display the result\n",
    "noncoding_matrix_ordered =  noncoding_matrix_ordered.merge(noncoding_to_matrix_raw[['gene_name','gene_type','orfID']], on=\"orfID\")\n",
    "noncoding_matrix_ordered.drop_duplicates(inplace=True)\n",
    "\n",
    "noncoding_matrix_ordered.to_csv(os.path.join(evoDir,\"Clustering/noncodingHuman_matrix.tsv\"), sep=\"\\t\", index=None)\n",
    "print(len(set(noncoding_matrix_ordered.orfID.values.tolist())))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_3062604/2062637614.py:4: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  canonical_conserved_to_matrix['specie'] = canonical_conserved_to_matrix[1].str.split(\"_\", expand=True)[0]\n",
      "/tmp/ipykernel_3062604/2062637614.py:6: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  canonical_conserved_to_matrix['presence'] = 1\n",
      "/tmp/ipykernel_3062604/2062637614.py:24: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  canonical_conserved_matrix_ordered['sum'] = canonical_conserved_matrix_ordered.select_dtypes(include='number').sum(axis=1)\n"
     ]
    }
   ],
   "source": [
    "canonical_conserved_to_matrix = canonical_conserved\n",
    "\n",
    "# Assuming noncoding_conserved_to_matrix is already defined\n",
    "canonical_conserved_to_matrix['specie'] = canonical_conserved_to_matrix[1].str.split(\"_\", expand=True)[0]\n",
    "canonical_conserved_to_matrix = canonical_conserved_to_matrix[['orfID', 'specie']]\n",
    "canonical_conserved_to_matrix['presence'] = 1\n",
    "\n",
    "# Pivoting the data without creating an 'id', and keeping 'orfID' as the unique identifier\n",
    "canonical_conserved_matrix = canonical_conserved_to_matrix.pivot_table(index='orfID', columns='specie', values='presence', aggfunc='max')\n",
    "\n",
    "# Fill NaN values with 0\n",
    "canonical_conserved_matrix = canonical_conserved_matrix.fillna(0)\n",
    "\n",
    "# Reset index so 'orfID' becomes a column instead of an index\n",
    "canonical_conserved_matrix = canonical_conserved_matrix.reset_index()\n",
    "\n",
    "# Rename columns as necessary\n",
    "canonical_conserved_matrix.columns = [\"orfID\", \"chicken\", \"macaca\", \"mouse\", \"opossum\", \"platypus\"]\n",
    "\n",
    "# Reorder columns for the final matrix\n",
    "canonical_conserved_matrix_ordered = canonical_conserved_matrix[[\"orfID\", \"macaca\", \"mouse\", \"opossum\", \"platypus\", \"chicken\"]]\n",
    "\n",
    "# Calculate sum across species columns to identify how many species are present\n",
    "canonical_conserved_matrix_ordered['sum'] = canonical_conserved_matrix_ordered.select_dtypes(include='number').sum(axis=1)\n",
    "\n",
    "# Sort by the sum column to order by the number of species present\n",
    "canonical_conserved_matrix_ordered = canonical_conserved_matrix_ordered.sort_values(by=\"sum\", ascending=False)\n",
    "\n",
    "# Display the result\n",
    "canonical_conserved_matrix_ordered =  canonical_conserved_matrix_ordered.merge(canonical_conserved[['gene_name','gene_type','orfID']], on=\"orfID\")\n",
    "canonical_conserved_matrix_ordered.drop_duplicates(inplace=True)\n",
    "\n",
    "canonical_conserved_matrix_ordered.to_csv(os.path.join(evoDir,\"Clustering/canonicalHuman_Conserved_matrix.tsv\"), sep=\"\\t\", index=None)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_3062604/2806917182.py:3: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  canonical_conserved_touse['specie'] = canonical_conserved_touse[1].str.split(\"_\", expand=True)[0]\n",
      "/tmp/ipykernel_3062604/2806917182.py:8: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  canonical_to_matrix['presence'] = np.where(canonical_to_matrix['specie'].isna(), 0, 1)\n",
      "/tmp/ipykernel_3062604/2806917182.py:36: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  canonical_matrix_ordered['sum'] = canonical_matrix_ordered.select_dtypes(include='number').sum(axis=1)\n"
     ]
    }
   ],
   "source": [
    "### all canonical\n",
    "canonical_conserved_touse = canonical_conserved\n",
    "canonical_conserved_touse['specie'] = canonical_conserved_touse[1].str.split(\"_\", expand=True)[0]\n",
    "# noncoding_NonConserved.drop(\"specie\",axis=1, inplace=True)\n",
    "canonical_to_matrix_raw = pd.concat([canonical_conserved, canonical_NonConserved])\n",
    "\n",
    "canonical_to_matrix = canonical_to_matrix_raw[['orfID', 'specie']]\n",
    "canonical_to_matrix['presence'] = np.where(canonical_to_matrix['specie'].isna(), 0, 1)\n",
    "\n",
    "\n",
    "human = pd.DataFrame({\n",
    "    'orfID': canonical_to_matrix['orfID'].unique(),\n",
    "    'specie': 'human',\n",
    "    'presence': 1\n",
    "})\n",
    "\n",
    "canonical_to_matrix = pd.concat([human, canonical_to_matrix], ignore_index=True)\n",
    "canonical_to_matrix.dropna(inplace=True)\n",
    "\n",
    "# Pivoting the data without creating an 'id', and keeping 'orfID' as the unique identifier\n",
    "canonical_matrix = canonical_to_matrix.pivot_table(index='orfID', columns='specie', values='presence', aggfunc='max')\n",
    "canonical_matrix\n",
    "# Fill NaN values with 0\n",
    "canonical_matrix = canonical_matrix.fillna(0)\n",
    "\n",
    "# Reset index so 'orfID' becomes a column instead of an index\n",
    "canonical_matrix = canonical_matrix.reset_index()\n",
    "\n",
    "# Rename columns as necessary\n",
    "canonical_matrix.columns = [\"orfID\", \"chicken\", \"human\", \"macaca\", \"mouse\", \"opossum\", \"platypus\"]\n",
    "\n",
    "# Reorder columns for the final matrix\n",
    "canonical_matrix_ordered = canonical_matrix[[\"orfID\", \"human\",\"macaca\", \"mouse\", \"opossum\", \"platypus\", \"chicken\"]]\n",
    "\n",
    "# Calculate sum across species columns to identify how many species are present\n",
    "canonical_matrix_ordered['sum'] = canonical_matrix_ordered.select_dtypes(include='number').sum(axis=1)\n",
    "\n",
    "# Sort by the sum column to order by the number of species present\n",
    "canonical_matrix_ordered = canonical_matrix_ordered.sort_values(by=\"sum\", ascending=False)\n",
    "\n",
    "# Display the result\n",
    "canonical_matrix_ordered =  canonical_matrix_ordered.merge(canonical_to_matrix_raw[['gene_name','gene_type','orfID']], on=\"orfID\")\n",
    "canonical_matrix_ordered.drop_duplicates(inplace=True)\n",
    "\n",
    "canonical_matrix_ordered.to_csv(os.path.join(evoDir,\"Clustering/canonicalHuman_matrix.tsv\"), sep=\"\\t\", index=None)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Find the oldest specie containg the ORF translated"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [],
   "source": [
    "species_ordered = ['chicken','platypus','opossum','mouse','macaca','human']\n",
    "def find_last_species(row):\n",
    "    for sp in species_ordered:\n",
    "        if row[sp] == 1.0:  # Check if the value is 1.0\n",
    "            return sp\n",
    "    return None  # Return None if no species has a 1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Apply the function to each row to create a new column 'oldest_species'\n",
    "canonical_matrix_ordered['oldest_species'] = canonical_matrix_ordered.apply(find_last_species, axis=1)\n",
    "canonical_matrix_ordered = species_and_years.merge(canonical_matrix_ordered, left_on=[\"specie\"], right_on=\"oldest_species\")\n",
    "canonical_matrix_ordered[['orfID','gene_name','gene_type','oldest_species','age']].to_csv(os.path.join(evoDir,\"Clustering/canonicalHuman_matrix_oldestAge.tsv\"), sep=\"\\t\", index=None)\n",
    "\n",
    "noncoding_matrix_ordered['oldest_species'] = noncoding_matrix_ordered.apply(find_last_species, axis=1)\n",
    "noncoding_matrix_ordered = species_and_years.merge(noncoding_matrix_ordered, left_on=[\"specie\"], right_on=\"oldest_species\")\n",
    "noncoding_matrix_ordered[['orfID','gene_name','gene_type','oldest_species','age']].to_csv(os.path.join(evoDir,\"Clustering/noncodingHuman_matrix_oldestAge.tsv\"), sep=\"\\t\", index=None)\n",
    "\n",
    "complete_matrix_age = pd.concat([noncoding_matrix_ordered[['orfID','gene_name','gene_type','oldest_species','age']], canonical_matrix_ordered[['orfID','gene_name','gene_type','oldest_species','age']]])\n",
    "complete_matrix_age.to_csv(os.path.join(evoDir,\"Clustering/allHuman_matrix_oldestAge.tsv\"), sep=\"\\t\", index=None)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.16"
  },
  "orig_nbformat": 4
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
