{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Evolutionary origin\n",
    "\n",
    "Macaca - Mouse - Human"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import os,re,glob\n",
    "import pandas as pd\n",
    "import numpy as np\n",
    "from collections import Counter\n",
    "from Bio.Seq import Seq\n",
    "from Bio.SeqRecord import SeqRecord\n",
    "from Bio import SeqIO\n",
    "\n",
    "users_dir = \"/users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction\"\n",
    "GENOMEDIR = \"/genomics/users/marta/genomes\"\n",
    "species=[\"human\",\"mouse\",\"macaca\",\"opossum\",\"platypus\",\"chicken\"]\n",
    "\n",
    "## annotation file\n",
    "annotation=\"/users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/human/newReference_Resconstructed/gencode.v38.gffcompare.TestisLiverBrain.annotation.sorted.1transcript.sorted.NOchr.gtf\"\n",
    "transcript_gene=pd.read_csv(\"/users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/human/newReference_Resconstructed/1transcript_1gene.reconstructed.csv\")\n",
    "\n",
    "## evoDir\n",
    "evoDir=os.path.join(users_dir,\"EvolutionaryOrigin\")\n",
    "\n",
    "def translate_dna_to_protein(dna_seq):\n",
    "    return str(Seq(dna_seq).translate())\n",
    "\n",
    "# Function to save DF in fasta format\n",
    "def create_seqrecord_PROT(row):\n",
    "    return SeqRecord(Seq(row['protein']), id=row['header'][:-4], description=\"\")\n",
    "def create_seqrecord_DNA(row):\n",
    "    return SeqRecord(Seq(row['seq']), id=row['header'], description=\"\")\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "## get repre sequences - both DNA and PROTEINS\n",
    "for s in [\"platypus\",\"mouse\",\"macaca\",\"chicken\"]:\n",
    "    print(s)\n",
    "    # riboseq_dir = os.path.join(users_dir,s,\"RiboSeq/RiboNovel\")\n",
    "    riboseq_dir = os.path.join(users_dir,s,\"RiboSeq/RiboQC_RiboNovel\")\n",
    "\n",
    "    ## get all candidates\n",
    "    print(\"Reading all the candidate ORF sequences\")\n",
    "    df = pd.read_csv(os.path.join(riboseq_dir,\"Annotation/candidateORF.fa\"), header=None, sep=\"\\t\")\n",
    "    ## convert to df\n",
    "    candidates_fasta = pd.DataFrame({'header':df[0].iloc[::2].values, 'seq':df[0].iloc[1::2].values})\n",
    "    candidates_fasta['header'] = candidates_fasta['header'].str[1:]\n",
    "    candidates_fasta\n",
    "\n",
    "    all_repre_candidates = pd.DataFrame()\n",
    "    for sample in os.listdir(os.path.join(riboseq_dir,\"RibORF\")):\n",
    "        if sample.endswith(\"r1\"):\n",
    "            print(sample)\n",
    "\n",
    "            repre = pd.read_csv(os.path.join(riboseq_dir,\"RibORF\",sample,\"repre.valid.pred.pvalue.parameters.txt\"), sep=\"\\t\")\n",
    "            repre_fasta = candidates_fasta[candidates_fasta['header'].isin(repre.orfID.values.tolist())]\n",
    "            repre_fasta['protein'] = repre_fasta['seq'].apply(translate_dna_to_protein)\n",
    "            all_repre_candidates = pd.concat([all_repre_candidates, repre_fasta])\n",
    "\n",
    "            ## translate to protein\n",
    "            print(\"Translating to protein\")\n",
    "            out_proteins = os.path.join(riboseq_dir,\"RibORF\",sample,\"repre.valid.pred.pvalue.parameters.PROTEIN.fa\")\n",
    "            seq_records_proteins = repre_fasta.apply(create_seqrecord_PROT, axis=1).tolist()\n",
    "            SeqIO.write(seq_records_proteins, out_proteins, \"fasta\")\n",
    "\n",
    "            ## keep the dna sequences as well\n",
    "            print(\"Saving the original DNA ORF\")\n",
    "            out_dna = os.path.join(riboseq_dir,\"RibORF\",sample,\"repre.valid.pred.pvalue.parameters.DNA.fa\")\n",
    "            seq_records_dna = repre_fasta.apply(create_seqrecord_DNA, axis=1).tolist()\n",
    "            SeqIO.write(seq_records_dna, out_dna, \"fasta\")\n",
    "\n",
    "            ## get only the nonredundant\n",
    "            print(\"Getting the non-redundant\")\n",
    "            all_repre_candidates.drop_duplicates(inplace=True)\n",
    "            out_nonredundant_proteins = os.path.join(riboseq_dir,\"RibORF\")+\"/\"+s+\"_repre.valid.pred.pvalue.parameters.PROTEIN.noredunant.fa\"\n",
    "            seq_records_proteins = all_repre_candidates.apply(create_seqrecord_PROT, axis=1).tolist()\n",
    "            SeqIO.write(seq_records_proteins, out_nonredundant_proteins, \"fasta\")"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## MMseqs2\n",
    "\n",
    "Against all expressed in mouse, macaca, platypus, opossum in any tissue (non-redunant set)"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### TESTIS-EXPRESSED"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "15040"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "## Testis-expressed ORFs human\n",
    "TestisExpressed_translatedINtestis = pd.read_csv(\"/projects_eg/projects/marta/TestisRestricted_Microproteins_TSA/with_TranscriptomeReconstruction/Q1_TestisORFs/human/ribORF_humanTestis_in1.csv\")\n",
    "TestisExpressed_translatedINtestis\n",
    "\n",
    "def create_seqrecord_PROThuman(row):\n",
    "    return SeqRecord(Seq(row['ORFpep']), id=row['orfID'], description=\"\")\n",
    "\n",
    "out_nonredundant_proteins = \"/projects_eg/projects/marta/TestisRestricted_Microproteins_TSA/with_TranscriptomeReconstruction/Q1_TestisORFs/human/ribORF_humanTestis_in1.fa\"\n",
    "seq_records_proteins = TestisExpressed_translatedINtestis.apply(create_seqrecord_PROThuman, axis=1).tolist()\n",
    "SeqIO.write(seq_records_proteins, out_nonredundant_proteins, \"fasta\")"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Creating database of human (expressed in testis)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "createdb /projects_eg/projects/marta/TestisRestricted_Microproteins_TSA/with_TranscriptomeReconstruction/Q1_TestisORFs/human/ribORF_humanTestis_in1.fa /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/EvolutionaryOrigin/MMseqsDB/ribORF_humanTestis_in1_DB \n",
      "\n",
      "MMseqs Version:       \t14.7e284\n",
      "Database type         \t0\n",
      "Shuffle input database\ttrue\n",
      "Createdb mode         \t0\n",
      "Write lookup file     \t1\n",
      "Offset of numeric ids \t0\n",
      "Compressed            \t0\n",
      "Verbosity             \t3\n",
      "\n",
      "Converting sequences\n",
      "[=\n",
      "Time for merging to ribORF_humanTestis_in1_DB_h: 0h 0m 0s 518ms\n",
      "Time for merging to ribORF_humanTestis_in1_DB: 0h 0m 0s 708ms\n",
      "Database type: Aminoacid\n",
      "Time for processing: 0h 0m 4s 150ms\n"
     ]
    }
   ],
   "source": [
    "%%bash -s \"$evoDir\" \"$users_dir\"\n",
    "\n",
    "# mkdir $1/MMseqsDB\n",
    "\n",
    "module load Miniconda3/4.9.2\n",
    "source /soft/system/software/Miniconda3/4.9.2/bin/activate MMseqs2\n",
    "\n",
    "mmseqs createdb /projects_eg/projects/marta/TestisRestricted_Microproteins_TSA/with_TranscriptomeReconstruction/Q1_TestisORFs/human/ribORF_humanTestis_in1.fa $1/MMseqsDB/ribORF_humanTestis_in1_DB\n"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Creating a database per specie, everything translated, no matter the tissue"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "createdb /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/chicken/RiboSeq/RiboQC_RiboNovel/RibORF/chicken_repre.valid.pred.pvalue.parameters.PROTEIN.noredunant.fa /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/macaca/RiboSeq/RiboQC_RiboNovel/RibORF/macaca_repre.valid.pred.pvalue.parameters.PROTEIN.noredunant.fa /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/mouse/RiboSeq/RiboQC_RiboNovel/RibORF/mouse_repre.valid.pred.pvalue.parameters.PROTEIN.noredunant.fa /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/platypus/RiboSeq/RiboQC_RiboNovel/RibORF/platypus_repre.valid.pred.pvalue.parameters.PROTEIN.noredunant.fa /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/EvolutionaryOrigin/MMseqsDB/speciesDB \n",
      "\n",
      "MMseqs Version:       \t14.7e284\n",
      "Database type         \t0\n",
      "Shuffle input database\ttrue\n",
      "Createdb mode         \t0\n",
      "Write lookup file     \t1\n",
      "Offset of numeric ids \t0\n",
      "Compressed            \t0\n",
      "Verbosity             \t3\n",
      "\n",
      "Converting sequences\n",
      "[===============\n",
      "Time for merging to speciesDB_h: 0h 0m 1s 864ms\n",
      "Time for merging to speciesDB: 0h 0m 2s 168ms\n",
      "Database type: Aminoacid\n",
      "Time for processing: 0h 0m 12s 910ms\n"
     ]
    }
   ],
   "source": [
    "%%bash -s \"$evoDir\" \"$users_dir\"\n",
    "\n",
    "module load Miniconda3/4.9.2\n",
    "source /soft/system/software/Miniconda3/4.9.2/bin/activate MMseqs2\n",
    "\n",
    "# species=\"platypus mouse macaca chicken\"\n",
    "# for s in $species; do\n",
    "#     echo $s\n",
    "\n",
    "#     ### create DB\n",
    "#     mmseqs createdb $2/$s/RiboSeq/RiboQC_RiboNovel/RibORF/repre.valid.pred.pvalue.parameters.PROTEIN.noredunant.fa $1/MMseqsDB/${s}_noredunantProts_DB\n",
    "\n",
    "# done\n",
    "\n",
    "mmseqs createdb $2/*/RiboSeq/RiboQC_RiboNovel/RibORF/*_repre.valid.pred.pvalue.parameters.PROTEIN.noredunant.fa $1/MMseqsDB/speciesDB\n"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Analysis"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>orfID</th>\n",
       "      <th>ORFseq</th>\n",
       "      <th>gene_id</th>\n",
       "      <th>gene_name</th>\n",
       "      <th>transcript_id</th>\n",
       "      <th>gene_type</th>\n",
       "      <th>orfType</th>\n",
       "      <th>length</th>\n",
       "      <th>length_aa</th>\n",
       "      <th>start_codon</th>\n",
       "      <th>ORFpep</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ENST00000000412.8:12:-|10|2450:160:994|canonic...</td>\n",
       "      <td>ATGTTCCCTTTCTACAGCTGCTGGAGGACTGGACTGCTACTACTAC...</td>\n",
       "      <td>ENSG00000003056</td>\n",
       "      <td>M6PR</td>\n",
       "      <td>ENST00000000412</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>canonical</td>\n",
       "      <td>834</td>\n",
       "      <td>278</td>\n",
       "      <td>ATG</td>\n",
       "      <td>MFPFYSCWRTGLLLLLLAVAVRESWQTEEKTCDLVGEKGKESEKEL...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ENST00000001008.6:12:+|5|3715:171:1551|canonic...</td>\n",
       "      <td>ATGACAGCCGAGGAGATGAAGGCGACCGAGAGCGGGGCGCAGTCGG...</td>\n",
       "      <td>ENSG00000004478</td>\n",
       "      <td>FKBP4</td>\n",
       "      <td>ENST00000001008</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>canonical</td>\n",
       "      <td>1380</td>\n",
       "      <td>460</td>\n",
       "      <td>ATG</td>\n",
       "      <td>MTAEEMKATESGAQSAPLPMEGVDISPKQDEGVLKVIKREGTGTEM...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ENST00000001146.7:2:-|1|4556:29:1568|canonical...</td>\n",
       "      <td>ATGCTCTTTGAGGGCTTGGATCTGGTGTCGGCGCTGGCCACCCTCG...</td>\n",
       "      <td>ENSG00000003137</td>\n",
       "      <td>CYP26B1</td>\n",
       "      <td>ENST00000001146</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>canonical</td>\n",
       "      <td>1539</td>\n",
       "      <td>513</td>\n",
       "      <td>ATG</td>\n",
       "      <td>MLFEGLDLVSALATLAACLVSVTLLLAVSQQLWQLRWAATRDKSCK...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ENST00000002165.11:6:-|3|2385:87:1491|canonica...</td>\n",
       "      <td>ATGCGGCCCCAGGAGCTCCCCAGGCTCGCGTTCCCGTTGCTGCTGT...</td>\n",
       "      <td>ENSG00000001036</td>\n",
       "      <td>FUCA2</td>\n",
       "      <td>ENST00000002165</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>canonical</td>\n",
       "      <td>1404</td>\n",
       "      <td>468</td>\n",
       "      <td>ATG</td>\n",
       "      <td>MRPQELPRLAFPLLLLLLLLLPPPPCPAHSATRFDPTWESLDARQL...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ENST00000002596.6:4:-|24|7160:305:1229|canonic...</td>\n",
       "      <td>ATGGCCGCGCTGCTCCTGGGCGCGGTGCTGCTGGTGGCCCAGCCCC...</td>\n",
       "      <td>ENSG00000002587</td>\n",
       "      <td>HS3ST1</td>\n",
       "      <td>ENST00000002596</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>canonical</td>\n",
       "      <td>924</td>\n",
       "      <td>308</td>\n",
       "      <td>ATG</td>\n",
       "      <td>MAALLLGAVLLVAQPQLVPSRPAELGQQELLRKAGTLQDDVRDGVA...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12516</th>\n",
       "      <td>TCONS_00001992:8:+|9|1191:153:675|noncoding|ATG</td>\n",
       "      <td>ATGACCGGGTCAGCGGAAATTTCAGAGCCAGCCAGCAGCTCAGGTC...</td>\n",
       "      <td>XLOC_001676</td>\n",
       "      <td>XLOC_001676</td>\n",
       "      <td>TCONS_00001992</td>\n",
       "      <td>novel</td>\n",
       "      <td>noncoding</td>\n",
       "      <td>522</td>\n",
       "      <td>174</td>\n",
       "      <td>ATG</td>\n",
       "      <td>MTGSAEISEPASSSGPRGRGRNGTAPRRQPRGHSRATADSGTSLSA...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12517</th>\n",
       "      <td>TCONS_00002050:9:-|86|10655:1396:1420|noncodin...</td>\n",
       "      <td>CTGTGTAAGTGCATCAGAACATAA</td>\n",
       "      <td>XLOC_001809</td>\n",
       "      <td>XLOC_001809</td>\n",
       "      <td>TCONS_00002050</td>\n",
       "      <td>novel</td>\n",
       "      <td>noncoding</td>\n",
       "      <td>24</td>\n",
       "      <td>8</td>\n",
       "      <td>CTG</td>\n",
       "      <td>MCKCIRT*</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12518</th>\n",
       "      <td>TCONS_00002051:9:-|9|437:109:145|noncoding|ATG</td>\n",
       "      <td>ATGGCTTTACCCATGGAGATTTTTGTTGCTGGCTAA</td>\n",
       "      <td>XLOC_001810</td>\n",
       "      <td>XLOC_001810</td>\n",
       "      <td>TCONS_00002051</td>\n",
       "      <td>novel</td>\n",
       "      <td>noncoding</td>\n",
       "      <td>36</td>\n",
       "      <td>12</td>\n",
       "      <td>ATG</td>\n",
       "      <td>MALPMEIFVAG*</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12519</th>\n",
       "      <td>TCONS_00002066:9:+|15|2910:249:327|noncoding|ATG</td>\n",
       "      <td>ATGTGCTTAATAAATATACGCCGAATGACTGAACTGCAACCAAGGA...</td>\n",
       "      <td>XLOC_001758</td>\n",
       "      <td>XLOC_001758</td>\n",
       "      <td>TCONS_00002066</td>\n",
       "      <td>novel</td>\n",
       "      <td>noncoding</td>\n",
       "      <td>78</td>\n",
       "      <td>26</td>\n",
       "      <td>ATG</td>\n",
       "      <td>MCLINIRRMTELQPRTLYTCFTDLV*</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12520</th>\n",
       "      <td>TCONS_00002105:9:+|25|1451:469:511|noncoding|TTG</td>\n",
       "      <td>TTGTACGTTGGTGGAGACGTGCCTTTTTTCCTTTGCACTTAA</td>\n",
       "      <td>XLOC_001725</td>\n",
       "      <td>XLOC_001725</td>\n",
       "      <td>TCONS_00002105</td>\n",
       "      <td>novel</td>\n",
       "      <td>noncoding</td>\n",
       "      <td>42</td>\n",
       "      <td>14</td>\n",
       "      <td>TTG</td>\n",
       "      <td>MYVGGDVPFFLCT*</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>12521 rows Ã— 11 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                   orfID  \\\n",
       "0      ENST00000000412.8:12:-|10|2450:160:994|canonic...   \n",
       "1      ENST00000001008.6:12:+|5|3715:171:1551|canonic...   \n",
       "2      ENST00000001146.7:2:-|1|4556:29:1568|canonical...   \n",
       "3      ENST00000002165.11:6:-|3|2385:87:1491|canonica...   \n",
       "4      ENST00000002596.6:4:-|24|7160:305:1229|canonic...   \n",
       "...                                                  ...   \n",
       "12516    TCONS_00001992:8:+|9|1191:153:675|noncoding|ATG   \n",
       "12517  TCONS_00002050:9:-|86|10655:1396:1420|noncodin...   \n",
       "12518     TCONS_00002051:9:-|9|437:109:145|noncoding|ATG   \n",
       "12519   TCONS_00002066:9:+|15|2910:249:327|noncoding|ATG   \n",
       "12520   TCONS_00002105:9:+|25|1451:469:511|noncoding|TTG   \n",
       "\n",
       "                                                  ORFseq          gene_id  \\\n",
       "0      ATGTTCCCTTTCTACAGCTGCTGGAGGACTGGACTGCTACTACTAC...  ENSG00000003056   \n",
       "1      ATGACAGCCGAGGAGATGAAGGCGACCGAGAGCGGGGCGCAGTCGG...  ENSG00000004478   \n",
       "2      ATGCTCTTTGAGGGCTTGGATCTGGTGTCGGCGCTGGCCACCCTCG...  ENSG00000003137   \n",
       "3      ATGCGGCCCCAGGAGCTCCCCAGGCTCGCGTTCCCGTTGCTGCTGT...  ENSG00000001036   \n",
       "4      ATGGCCGCGCTGCTCCTGGGCGCGGTGCTGCTGGTGGCCCAGCCCC...  ENSG00000002587   \n",
       "...                                                  ...              ...   \n",
       "12516  ATGACCGGGTCAGCGGAAATTTCAGAGCCAGCCAGCAGCTCAGGTC...      XLOC_001676   \n",
       "12517                           CTGTGTAAGTGCATCAGAACATAA      XLOC_001809   \n",
       "12518               ATGGCTTTACCCATGGAGATTTTTGTTGCTGGCTAA      XLOC_001810   \n",
       "12519  ATGTGCTTAATAAATATACGCCGAATGACTGAACTGCAACCAAGGA...      XLOC_001758   \n",
       "12520         TTGTACGTTGGTGGAGACGTGCCTTTTTTCCTTTGCACTTAA      XLOC_001725   \n",
       "\n",
       "         gene_name    transcript_id       gene_type    orfType  length  \\\n",
       "0             M6PR  ENST00000000412  protein_coding  canonical     834   \n",
       "1            FKBP4  ENST00000001008  protein_coding  canonical    1380   \n",
       "2          CYP26B1  ENST00000001146  protein_coding  canonical    1539   \n",
       "3            FUCA2  ENST00000002165  protein_coding  canonical    1404   \n",
       "4           HS3ST1  ENST00000002596  protein_coding  canonical     924   \n",
       "...            ...              ...             ...        ...     ...   \n",
       "12516  XLOC_001676   TCONS_00001992           novel  noncoding     522   \n",
       "12517  XLOC_001809   TCONS_00002050           novel  noncoding      24   \n",
       "12518  XLOC_001810   TCONS_00002051           novel  noncoding      36   \n",
       "12519  XLOC_001758   TCONS_00002066           novel  noncoding      78   \n",
       "12520  XLOC_001725   TCONS_00002105           novel  noncoding      42   \n",
       "\n",
       "       length_aa start_codon  \\\n",
       "0            278         ATG   \n",
       "1            460         ATG   \n",
       "2            513         ATG   \n",
       "3            468         ATG   \n",
       "4            308         ATG   \n",
       "...          ...         ...   \n",
       "12516        174         ATG   \n",
       "12517          8         CTG   \n",
       "12518         12         ATG   \n",
       "12519         26         ATG   \n",
       "12520         14         TTG   \n",
       "\n",
       "                                                  ORFpep  \n",
       "0      MFPFYSCWRTGLLLLLLAVAVRESWQTEEKTCDLVGEKGKESEKEL...  \n",
       "1      MTAEEMKATESGAQSAPLPMEGVDISPKQDEGVLKVIKREGTGTEM...  \n",
       "2      MLFEGLDLVSALATLAACLVSVTLLLAVSQQLWQLRWAATRDKSCK...  \n",
       "3      MRPQELPRLAFPLLLLLLLLLPPPPCPAHSATRFDPTWESLDARQL...  \n",
       "4      MAALLLGAVLLVAQPQLVPSRPAELGQQELLRKAGTLQDDVRDGVA...  \n",
       "...                                                  ...  \n",
       "12516  MTGSAEISEPASSSGPRGRGRNGTAPRRQPRGHSRATADSGTSLSA...  \n",
       "12517                                           MCKCIRT*  \n",
       "12518                                       MALPMEIFVAG*  \n",
       "12519                         MCLINIRRMTELQPRTLYTCFTDLV*  \n",
       "12520                                     MYVGGDVPFFLCT*  \n",
       "\n",
       "[12521 rows x 11 columns]"
      ]
     },
     "execution_count": 2,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ribORF_in_2 = pd.read_csv(\"/projects_eg/projects/marta/TestisRestricted_Microproteins_TSA/with_TranscriptomeReconstruction/ribORF_humanTestis_in2.csv\")\n",
    "ribORF_in_2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th>orfID</th>\n",
       "      <th>ORFseq</th>\n",
       "      <th>gene_id</th>\n",
       "      <th>gene_name</th>\n",
       "      <th>transcript_id</th>\n",
       "      <th>orfType</th>\n",
       "      <th>length</th>\n",
       "      <th>length_aa</th>\n",
       "      <th>start_codon</th>\n",
       "      <th>ORFpep</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>gene_type</th>\n",
       "      <th>mouse</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th rowspan=\"2\" valign=\"top\">lncRNA</th>\n",
       "      <th>conserved_mouse</th>\n",
       "      <td>110</td>\n",
       "      <td>110</td>\n",
       "      <td>110</td>\n",
       "      <td>110</td>\n",
       "      <td>110</td>\n",
       "      <td>110</td>\n",
       "      <td>110</td>\n",
       "      <td>110</td>\n",
       "      <td>110</td>\n",
       "      <td>110</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>no</th>\n",
       "      <td>394</td>\n",
       "      <td>394</td>\n",
       "      <td>394</td>\n",
       "      <td>394</td>\n",
       "      <td>394</td>\n",
       "      <td>394</td>\n",
       "      <td>394</td>\n",
       "      <td>394</td>\n",
       "      <td>394</td>\n",
       "      <td>394</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>novel</th>\n",
       "      <th>no</th>\n",
       "      <td>63</td>\n",
       "      <td>63</td>\n",
       "      <td>63</td>\n",
       "      <td>63</td>\n",
       "      <td>63</td>\n",
       "      <td>63</td>\n",
       "      <td>63</td>\n",
       "      <td>63</td>\n",
       "      <td>63</td>\n",
       "      <td>63</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"2\" valign=\"top\">processed_pseudogene</th>\n",
       "      <th>conserved_mouse</th>\n",
       "      <td>109</td>\n",
       "      <td>109</td>\n",
       "      <td>109</td>\n",
       "      <td>109</td>\n",
       "      <td>109</td>\n",
       "      <td>109</td>\n",
       "      <td>109</td>\n",
       "      <td>109</td>\n",
       "      <td>109</td>\n",
       "      <td>109</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>no</th>\n",
       "      <td>14</td>\n",
       "      <td>14</td>\n",
       "      <td>14</td>\n",
       "      <td>14</td>\n",
       "      <td>14</td>\n",
       "      <td>14</td>\n",
       "      <td>14</td>\n",
       "      <td>14</td>\n",
       "      <td>14</td>\n",
       "      <td>14</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"2\" valign=\"top\">protein_coding</th>\n",
       "      <th>conserved_mouse</th>\n",
       "      <td>11533</td>\n",
       "      <td>11533</td>\n",
       "      <td>11533</td>\n",
       "      <td>11533</td>\n",
       "      <td>11533</td>\n",
       "      <td>11533</td>\n",
       "      <td>11533</td>\n",
       "      <td>11533</td>\n",
       "      <td>11533</td>\n",
       "      <td>11533</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>no</th>\n",
       "      <td>298</td>\n",
       "      <td>298</td>\n",
       "      <td>298</td>\n",
       "      <td>298</td>\n",
       "      <td>298</td>\n",
       "      <td>298</td>\n",
       "      <td>298</td>\n",
       "      <td>298</td>\n",
       "      <td>298</td>\n",
       "      <td>298</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                      orfID  ORFseq  gene_id  gene_name  \\\n",
       "gene_type            mouse                                                \n",
       "lncRNA               conserved_mouse    110     110      110        110   \n",
       "                     no                 394     394      394        394   \n",
       "novel                no                  63      63       63         63   \n",
       "processed_pseudogene conserved_mouse    109     109      109        109   \n",
       "                     no                  14      14       14         14   \n",
       "protein_coding       conserved_mouse  11533   11533    11533      11533   \n",
       "                     no                 298     298      298        298   \n",
       "\n",
       "                                      transcript_id  orfType  length  \\\n",
       "gene_type            mouse                                             \n",
       "lncRNA               conserved_mouse            110      110     110   \n",
       "                     no                         394      394     394   \n",
       "novel                no                          63       63      63   \n",
       "processed_pseudogene conserved_mouse            109      109     109   \n",
       "                     no                          14       14      14   \n",
       "protein_coding       conserved_mouse          11533    11533   11533   \n",
       "                     no                         298      298     298   \n",
       "\n",
       "                                      length_aa  start_codon  ORFpep  \n",
       "gene_type            mouse                                            \n",
       "lncRNA               conserved_mouse        110          110     110  \n",
       "                     no                     394          394     394  \n",
       "novel                no                      63           63      63  \n",
       "processed_pseudogene conserved_mouse        109          109     109  \n",
       "                     no                      14           14      14  \n",
       "protein_coding       conserved_mouse      11533        11533   11533  \n",
       "                     no                     298          298     298  "
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "## Analysis in Mouse\n",
    "human_mouse = pd.read_csv(os.path.join(evoDir,\"TestisExpressedHuman_vs_mouse.txt\"), sep=\"\\t\", header=None)\n",
    "human_mouse.columns = [\"qseqid\",\"sseqid\",\"pident\",\"length\",\"mismatch\",\"gapopen\",\"qstart\",\"qend\",\"sstart\",\"send\",\"evalue\",\"bitscore\",\"qlen\",\"slen\",\"qframe\",\"sframe\",\"sstrand\",\"qcovs\"]\n",
    "\n",
    "ribORF_in_2['mouse'] = np.where(ribORF_in_2['orfID'].isin(human_mouse.qseqid.values.tolist()), \"conserved_mouse\",\"no\")\n",
    "ribORF_in_2.groupby(['gene_type','mouse']).count()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th>orfID</th>\n",
       "      <th>ORFseq</th>\n",
       "      <th>gene_id</th>\n",
       "      <th>gene_name</th>\n",
       "      <th>transcript_id</th>\n",
       "      <th>orfType</th>\n",
       "      <th>length</th>\n",
       "      <th>length_aa</th>\n",
       "      <th>start_codon</th>\n",
       "      <th>ORFpep</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>gene_type</th>\n",
       "      <th>macaca</th>\n",
       "      <th>mouse</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th rowspan=\"4\" valign=\"top\">lncRNA</th>\n",
       "      <th rowspan=\"2\" valign=\"top\">conserved_macaca</th>\n",
       "      <th>conserved_mouse</th>\n",
       "      <td>109</td>\n",
       "      <td>109</td>\n",
       "      <td>109</td>\n",
       "      <td>109</td>\n",
       "      <td>109</td>\n",
       "      <td>109</td>\n",
       "      <td>109</td>\n",
       "      <td>109</td>\n",
       "      <td>109</td>\n",
       "      <td>109</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>no</th>\n",
       "      <td>62</td>\n",
       "      <td>62</td>\n",
       "      <td>62</td>\n",
       "      <td>62</td>\n",
       "      <td>62</td>\n",
       "      <td>62</td>\n",
       "      <td>62</td>\n",
       "      <td>62</td>\n",
       "      <td>62</td>\n",
       "      <td>62</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"2\" valign=\"top\">no</th>\n",
       "      <th>conserved_mouse</th>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>no</th>\n",
       "      <td>332</td>\n",
       "      <td>332</td>\n",
       "      <td>332</td>\n",
       "      <td>332</td>\n",
       "      <td>332</td>\n",
       "      <td>332</td>\n",
       "      <td>332</td>\n",
       "      <td>332</td>\n",
       "      <td>332</td>\n",
       "      <td>332</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"2\" valign=\"top\">novel</th>\n",
       "      <th>conserved_macaca</th>\n",
       "      <th>no</th>\n",
       "      <td>8</td>\n",
       "      <td>8</td>\n",
       "      <td>8</td>\n",
       "      <td>8</td>\n",
       "      <td>8</td>\n",
       "      <td>8</td>\n",
       "      <td>8</td>\n",
       "      <td>8</td>\n",
       "      <td>8</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>no</th>\n",
       "      <th>no</th>\n",
       "      <td>55</td>\n",
       "      <td>55</td>\n",
       "      <td>55</td>\n",
       "      <td>55</td>\n",
       "      <td>55</td>\n",
       "      <td>55</td>\n",
       "      <td>55</td>\n",
       "      <td>55</td>\n",
       "      <td>55</td>\n",
       "      <td>55</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"4\" valign=\"top\">processed_pseudogene</th>\n",
       "      <th rowspan=\"2\" valign=\"top\">conserved_macaca</th>\n",
       "      <th>conserved_mouse</th>\n",
       "      <td>107</td>\n",
       "      <td>107</td>\n",
       "      <td>107</td>\n",
       "      <td>107</td>\n",
       "      <td>107</td>\n",
       "      <td>107</td>\n",
       "      <td>107</td>\n",
       "      <td>107</td>\n",
       "      <td>107</td>\n",
       "      <td>107</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>no</th>\n",
       "      <td>7</td>\n",
       "      <td>7</td>\n",
       "      <td>7</td>\n",
       "      <td>7</td>\n",
       "      <td>7</td>\n",
       "      <td>7</td>\n",
       "      <td>7</td>\n",
       "      <td>7</td>\n",
       "      <td>7</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"2\" valign=\"top\">no</th>\n",
       "      <th>conserved_mouse</th>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>no</th>\n",
       "      <td>7</td>\n",
       "      <td>7</td>\n",
       "      <td>7</td>\n",
       "      <td>7</td>\n",
       "      <td>7</td>\n",
       "      <td>7</td>\n",
       "      <td>7</td>\n",
       "      <td>7</td>\n",
       "      <td>7</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"4\" valign=\"top\">protein_coding</th>\n",
       "      <th rowspan=\"2\" valign=\"top\">conserved_macaca</th>\n",
       "      <th>conserved_mouse</th>\n",
       "      <td>11374</td>\n",
       "      <td>11374</td>\n",
       "      <td>11374</td>\n",
       "      <td>11374</td>\n",
       "      <td>11374</td>\n",
       "      <td>11374</td>\n",
       "      <td>11374</td>\n",
       "      <td>11374</td>\n",
       "      <td>11374</td>\n",
       "      <td>11374</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>no</th>\n",
       "      <td>269</td>\n",
       "      <td>269</td>\n",
       "      <td>269</td>\n",
       "      <td>269</td>\n",
       "      <td>269</td>\n",
       "      <td>269</td>\n",
       "      <td>269</td>\n",
       "      <td>269</td>\n",
       "      <td>269</td>\n",
       "      <td>269</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"2\" valign=\"top\">no</th>\n",
       "      <th>conserved_mouse</th>\n",
       "      <td>159</td>\n",
       "      <td>159</td>\n",
       "      <td>159</td>\n",
       "      <td>159</td>\n",
       "      <td>159</td>\n",
       "      <td>159</td>\n",
       "      <td>159</td>\n",
       "      <td>159</td>\n",
       "      <td>159</td>\n",
       "      <td>159</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>no</th>\n",
       "      <td>29</td>\n",
       "      <td>29</td>\n",
       "      <td>29</td>\n",
       "      <td>29</td>\n",
       "      <td>29</td>\n",
       "      <td>29</td>\n",
       "      <td>29</td>\n",
       "      <td>29</td>\n",
       "      <td>29</td>\n",
       "      <td>29</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                       orfID  ORFseq  gene_id  \\\n",
       "gene_type            macaca           mouse                                     \n",
       "lncRNA               conserved_macaca conserved_mouse    109     109      109   \n",
       "                                      no                  62      62       62   \n",
       "                     no               conserved_mouse      1       1        1   \n",
       "                                      no                 332     332      332   \n",
       "novel                conserved_macaca no                   8       8        8   \n",
       "                     no               no                  55      55       55   \n",
       "processed_pseudogene conserved_macaca conserved_mouse    107     107      107   \n",
       "                                      no                   7       7        7   \n",
       "                     no               conserved_mouse      2       2        2   \n",
       "                                      no                   7       7        7   \n",
       "protein_coding       conserved_macaca conserved_mouse  11374   11374    11374   \n",
       "                                      no                 269     269      269   \n",
       "                     no               conserved_mouse    159     159      159   \n",
       "                                      no                  29      29       29   \n",
       "\n",
       "                                                       gene_name  \\\n",
       "gene_type            macaca           mouse                        \n",
       "lncRNA               conserved_macaca conserved_mouse        109   \n",
       "                                      no                      62   \n",
       "                     no               conserved_mouse          1   \n",
       "                                      no                     332   \n",
       "novel                conserved_macaca no                       8   \n",
       "                     no               no                      55   \n",
       "processed_pseudogene conserved_macaca conserved_mouse        107   \n",
       "                                      no                       7   \n",
       "                     no               conserved_mouse          2   \n",
       "                                      no                       7   \n",
       "protein_coding       conserved_macaca conserved_mouse      11374   \n",
       "                                      no                     269   \n",
       "                     no               conserved_mouse        159   \n",
       "                                      no                      29   \n",
       "\n",
       "                                                       transcript_id  orfType  \\\n",
       "gene_type            macaca           mouse                                     \n",
       "lncRNA               conserved_macaca conserved_mouse            109      109   \n",
       "                                      no                          62       62   \n",
       "                     no               conserved_mouse              1        1   \n",
       "                                      no                         332      332   \n",
       "novel                conserved_macaca no                           8        8   \n",
       "                     no               no                          55       55   \n",
       "processed_pseudogene conserved_macaca conserved_mouse            107      107   \n",
       "                                      no                           7        7   \n",
       "                     no               conserved_mouse              2        2   \n",
       "                                      no                           7        7   \n",
       "protein_coding       conserved_macaca conserved_mouse          11374    11374   \n",
       "                                      no                         269      269   \n",
       "                     no               conserved_mouse            159      159   \n",
       "                                      no                          29       29   \n",
       "\n",
       "                                                       length  length_aa  \\\n",
       "gene_type            macaca           mouse                                \n",
       "lncRNA               conserved_macaca conserved_mouse     109        109   \n",
       "                                      no                   62         62   \n",
       "                     no               conserved_mouse       1          1   \n",
       "                                      no                  332        332   \n",
       "novel                conserved_macaca no                    8          8   \n",
       "                     no               no                   55         55   \n",
       "processed_pseudogene conserved_macaca conserved_mouse     107        107   \n",
       "                                      no                    7          7   \n",
       "                     no               conserved_mouse       2          2   \n",
       "                                      no                    7          7   \n",
       "protein_coding       conserved_macaca conserved_mouse   11374      11374   \n",
       "                                      no                  269        269   \n",
       "                     no               conserved_mouse     159        159   \n",
       "                                      no                   29         29   \n",
       "\n",
       "                                                       start_codon  ORFpep  \n",
       "gene_type            macaca           mouse                                 \n",
       "lncRNA               conserved_macaca conserved_mouse          109     109  \n",
       "                                      no                        62      62  \n",
       "                     no               conserved_mouse            1       1  \n",
       "                                      no                       332     332  \n",
       "novel                conserved_macaca no                         8       8  \n",
       "                     no               no                        55      55  \n",
       "processed_pseudogene conserved_macaca conserved_mouse          107     107  \n",
       "                                      no                         7       7  \n",
       "                     no               conserved_mouse            2       2  \n",
       "                                      no                         7       7  \n",
       "protein_coding       conserved_macaca conserved_mouse        11374   11374  \n",
       "                                      no                       269     269  \n",
       "                     no               conserved_mouse          159     159  \n",
       "                                      no                        29      29  "
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "## Analysis in Macaca\n",
    "human_macaca = pd.read_csv(os.path.join(evoDir,\"TestisExpressedHuman_vs_macaca.txt\"), sep=\"\\t\", header=None)\n",
    "human_macaca.columns = [\"qseqid\",\"sseqid\",\"pident\",\"length\",\"mismatch\",\"gapopen\",\"qstart\",\"qend\",\"sstart\",\"send\",\"evalue\",\"bitscore\",\"qlen\",\"slen\",\"qframe\",\"sframe\",\"sstrand\",\"qcovs\"]\n",
    "\n",
    "ribORF_in_2['macaca'] = np.where(ribORF_in_2['orfID'].isin(human_macaca.qseqid.values.tolist()), \"conserved_macaca\",\"no\")\n",
    "# ribORF_in_2_macaca.drop(columns=['mouse'], inplace=True)\n",
    "ribORF_in_2.to_csv(os.path.join(evoDir,\"ribORF_humanTestis_in2.evolution.csv\"), index=None)\n",
    "ribORF_in_2.groupby(['gene_type','macaca','mouse']).count()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.16"
  },
  "orig_nbformat": 4
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
