{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Evolutionary origin\n",
    "\n",
    "Macaca - Mouse - Human"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [],
   "source": [
    "import os,re,glob\n",
    "import pandas as pd\n",
    "import numpy as np\n",
    "from collections import Counter\n",
    "from Bio.Seq import Seq\n",
    "from Bio.SeqRecord import SeqRecord\n",
    "from Bio import SeqIO\n",
    "\n",
    "users_dir = \"/users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction\"\n",
    "GENOMEDIR = \"/genomics/users/marta/genomes\"\n",
    "species=[\"mouse\",\"macaca\",\"opossum\",\"platypus\",\"chicken\"]\n",
    "\n",
    "## annotation file\n",
    "annotation=\"/users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/human/newReference_Resconstructed/gencode.v38.gffcompare.TestisLiverBrain.annotation.sorted.1transcript.sorted.NOchr.gtf\"\n",
    "transcript_gene=pd.read_csv(\"/users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/human/newReference_Resconstructed/1transcript_1gene.reconstructed.csv\")\n",
    "\n",
    "## evoDir\n",
    "evoDir=os.path.join(users_dir,\"EvolutionaryOrigin_OrthoFinder\")\n",
    "\n",
    "def translate_dna_to_protein(dna_seq):\n",
    "    return str(Seq(dna_seq).translate())\n",
    "\n",
    "# Function to save DF in fasta format\n",
    "def create_seqrecord_PROT(row):\n",
    "    return SeqRecord(Seq(row['protein']), id=row['header'][:-4], description=\"\")\n",
    "def create_seqrecord_DNA(row):\n",
    "    return SeqRecord(Seq(row['seq']), id=row['header'], description=\"\")\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "## get repre sequences - both DNA and PROTEINS\n",
    "for s in species:\n",
    "    print(s)\n",
    "    riboseq_dir = os.path.join(users_dir,s,\"RiboSeq/RiboQC_RiboNovel\")\n",
    "\n",
    "    ## get all candidates\n",
    "    print(\"Reading all the candidate ORF sequences\")\n",
    "    df = pd.read_csv(os.path.join(riboseq_dir,\"Annotation/candidateORF.fa\"), header=None, sep=\"\\t\")\n",
    "    ## convert to df\n",
    "    candidates_fasta = pd.DataFrame({'header':df[0].iloc[::2].values, 'seq':df[0].iloc[1::2].values})\n",
    "    candidates_fasta['header'] = candidates_fasta['header'].str[1:]\n",
    "    candidates_fasta = candidates_fasta[~candidates_fasta['header'].str.contains(\"iORF\")]\n",
    "\n",
    "    all_repre_candidates = pd.DataFrame()\n",
    "    for sample in os.listdir(os.path.join(riboseq_dir,\"RibORF\")):\n",
    "        if sample.endswith(\"r1\"):\n",
    "            print(sample)\n",
    "\n",
    "            repre = pd.read_csv(os.path.join(riboseq_dir,\"RibORF\",sample,\"repre.valid.pred.pvalue.parameters.txt\"), sep=\"\\t\")\n",
    "            repre_fasta = candidates_fasta[candidates_fasta['header'].isin(repre.orfID.values.tolist())]\n",
    "            repre_fasta['header'] = s + \"_\" +candidates_fasta['header']\n",
    "\n",
    "            repre_fasta['protein'] = repre_fasta['seq'].apply(translate_dna_to_protein)\n",
    "            all_repre_candidates = pd.concat([all_repre_candidates, repre_fasta])\n",
    "\n",
    "            ## translate to protein\n",
    "            print(\"Translating to protein\")\n",
    "            out_proteins = os.path.join(riboseq_dir,\"RibORF\",sample,\"repre.valid.pred.pvalue.parameters.PROTEIN.fa\")\n",
    "            seq_records_proteins = repre_fasta.apply(create_seqrecord_PROT, axis=1).tolist()\n",
    "            SeqIO.write(seq_records_proteins, out_proteins, \"fasta\")\n",
    "\n",
    "            ## keep the dna sequences as well\n",
    "            print(\"Saving the original DNA ORF\")\n",
    "            out_dna = os.path.join(riboseq_dir,\"RibORF\",sample,\"repre.valid.pred.pvalue.parameters.DNA.fa\")\n",
    "            seq_records_dna = repre_fasta.apply(create_seqrecord_DNA, axis=1).tolist()\n",
    "            SeqIO.write(seq_records_dna, out_dna, \"fasta\")\n",
    "\n",
    "            ## get only the nonredundant\n",
    "            print(\"Getting the non-redundant\")\n",
    "            all_repre_candidates.drop_duplicates(inplace=True)\n",
    "            out_nonredundant_proteins = os.path.join(riboseq_dir,\"RibORF\")+\"/\"+s+\"_repre.valid.pred.pvalue.parameters.PROTEIN.noredunant.fa\"\n",
    "            seq_records_proteins = all_repre_candidates.apply(create_seqrecord_PROT, axis=1).tolist()\n",
    "            SeqIO.write(seq_records_proteins, out_nonredundant_proteins, \"fasta\")"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Orthofinder\n",
    "\n",
    "Against all expressed in mouse, macaca, platypus, opossum in any tissue (non-redunant set)"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### TESTIS-EXPRESSED"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "15040"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "## Testis-expressed ORFs human\n",
    "TestisExpressed_translatedINtestis = pd.read_csv(\"/projects_eg/projects/marta/TestisRestricted_Microproteins_TSA/with_TranscriptomeReconstruction/Q1_TestisORFs/human/ribORF_humanTestis_in1.csv\")\n",
    "TestisExpressed_translatedINtestis\n",
    "\n",
    "def create_seqrecord_PROThuman(row):\n",
    "    return SeqRecord(Seq(row['ORFpep']), id=row['orfID'], description=\"\")\n",
    "\n",
    "out_nonredundant_proteins = \"/projects_eg/projects/marta/TestisRestricted_Microproteins_TSA/with_TranscriptomeReconstruction/Q1_TestisORFs/human/ribORF_humanTestis_in1.fa\"\n",
    "seq_records_proteins = TestisExpressed_translatedINtestis.apply(create_seqrecord_PROThuman, axis=1).tolist()\n",
    "SeqIO.write(seq_records_proteins, out_nonredundant_proteins, \"fasta\")"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Creating database of human (expressed in testis)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "createdb /projects_eg/projects/marta/TestisRestricted_Microproteins_TSA/with_TranscriptomeReconstruction/Q1_TestisORFs/human/ribORF_humanTestis_in1.fa /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/EvolutionaryOrigin/MMseqsDB/ribORF_humanTestis_in1_DB \n",
      "\n",
      "MMseqs Version:       \t14.7e284\n",
      "Database type         \t0\n",
      "Shuffle input database\ttrue\n",
      "Createdb mode         \t0\n",
      "Write lookup file     \t1\n",
      "Offset of numeric ids \t0\n",
      "Compressed            \t0\n",
      "Verbosity             \t3\n",
      "\n",
      "Converting sequences\n",
      "[=\n",
      "Time for merging to ribORF_humanTestis_in1_DB_h: 0h 0m 0s 518ms\n",
      "Time for merging to ribORF_humanTestis_in1_DB: 0h 0m 0s 708ms\n",
      "Database type: Aminoacid\n",
      "Time for processing: 0h 0m 4s 150ms\n"
     ]
    }
   ],
   "source": [
    "%%bash -s \"$evoDir\" \"$users_dir\"\n",
    "\n",
    "# mkdir $1/MMseqsDB\n",
    "\n",
    "module load Miniconda3/4.9.2\n",
    "source /soft/system/software/Miniconda3/4.9.2/bin/activate MMseqs2\n",
    "\n",
    "mmseqs createdb /projects_eg/projects/marta/TestisRestricted_Microproteins_TSA/with_TranscriptomeReconstruction/Q1_TestisORFs/human/ribORF_humanTestis_in1.fa $1/MMseqsDB/ribORF_humanTestis_in1_DB\n"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Creating a joint database with all the specie, everything translated, no matter the tissue"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/EvolutionaryOrigin/MMseqsDB/speciesDB exists and will be overwritten\n",
      "createdb /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/chicken/RiboSeq/RiboQC_RiboNovel/RibORF/chicken_repre.valid.pred.pvalue.parameters.PROTEIN.noredunant.fa /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/macaca/RiboSeq/RiboQC_RiboNovel/RibORF/macaca_repre.valid.pred.pvalue.parameters.PROTEIN.noredunant.fa /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/mouse/RiboSeq/RiboQC_RiboNovel/RibORF/mouse_repre.valid.pred.pvalue.parameters.PROTEIN.noredunant.fa /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/opossum/RiboSeq/RiboQC_RiboNovel_RiboDetector/RibORF/opossum_repre.valid.pred.pvalue.parameters.PROTEIN.noredunant.fa /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/platypus/RiboSeq/RiboQC_RiboNovel/RibORF/platypus_repre.valid.pred.pvalue.parameters.PROTEIN.noredunant.fa /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/EvolutionaryOrigin/MMseqsDB/speciesDB \n",
      "\n",
      "MMseqs Version:       \t14.7e284\n",
      "Database type         \t0\n",
      "Shuffle input database\ttrue\n",
      "Createdb mode         \t0\n",
      "Write lookup file     \t1\n",
      "Offset of numeric ids \t0\n",
      "Compressed            \t0\n",
      "Verbosity             \t3\n",
      "\n",
      "Converting sequences\n",
      "[================\n",
      "Time for merging to speciesDB_h: 0h 0m 0s 652ms\n",
      "Time for merging to speciesDB: 0h 0m 1s 49ms\n",
      "Database type: Aminoacid\n",
      "Time for processing: 0h 0m 4s 945ms\n"
     ]
    }
   ],
   "source": [
    "%%bash -s \"$evoDir\" \"$users_dir\"\n",
    "\n",
    "module load Miniconda3/4.9.2\n",
    "source /soft/system/software/Miniconda3/4.9.2/bin/activate MMseqs2\n",
    "\n",
    "# species=\"platypus mouse macaca chicken\"\n",
    "# for s in $species; do\n",
    "#     echo $s\n",
    "\n",
    "#     ### create DB\n",
    "#     mmseqs createdb $2/$s/RiboSeq/RiboQC_RiboNovel/RibORF/repre.valid.pred.pvalue.parameters.PROTEIN.noredunant.fa $1/MMseqsDB/${s}_noredunantProts_DB\n",
    "\n",
    "# done\n",
    "\n",
    "mmseqs createdb $2/*/RiboSeq/RiboQC_RiboNovel*/RibORF/*_repre.valid.pred.pvalue.parameters.PROTEIN.noredunant.fa $1/MMseqsDB/speciesDB\n",
    "\n"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Creating a joint database with all the specie AND human for clustering, everything translated for the species, no matter the tissue, but only translated in 1 for testis"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/EvolutionaryOrigin/MMseqsDB/species_and_humanDB exists and will be overwritten\n",
      "createdb /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/chicken/RiboSeq/RiboQC_RiboNovel/RibORF/chicken_repre.valid.pred.pvalue.parameters.PROTEIN.noredunant.fa /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/macaca/RiboSeq/RiboQC_RiboNovel/RibORF/macaca_repre.valid.pred.pvalue.parameters.PROTEIN.noredunant.fa /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/mouse/RiboSeq/RiboQC_RiboNovel/RibORF/mouse_repre.valid.pred.pvalue.parameters.PROTEIN.noredunant.fa /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/opossum/RiboSeq/RiboQC_RiboNovel_RiboDetector/RibORF/opossum_repre.valid.pred.pvalue.parameters.PROTEIN.noredunant.fa /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/platypus/RiboSeq/RiboQC_RiboNovel/RibORF/platypus_repre.valid.pred.pvalue.parameters.PROTEIN.noredunant.fa /projects_eg/projects/marta/TestisRestricted_Microproteins_TSA/with_TranscriptomeReconstruction/Q1_TestisORFs/human/ribORF_humanTestis_in1.fa /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/EvolutionaryOrigin/MMseqsDB/species_and_humanDB \n",
      "\n",
      "MMseqs Version:       \t14.7e284\n",
      "Database type         \t0\n",
      "Shuffle input database\ttrue\n",
      "Createdb mode         \t0\n",
      "Write lookup file     \t1\n",
      "Offset of numeric ids \t0\n",
      "Compressed            \t0\n",
      "Verbosity             \t3\n",
      "\n",
      "Converting sequences\n",
      "[==================\n",
      "Time for merging to species_and_humanDB_h: 0h 0m 0s 692ms\n",
      "Time for merging to species_and_humanDB: 0h 0m 1s 400ms\n",
      "Database type: Aminoacid\n",
      "Time for processing: 0h 0m 6s 41ms\n"
     ]
    }
   ],
   "source": [
    "%%bash -s \"$evoDir\" \"$users_dir\"\n",
    "\n",
    "module load Miniconda3/4.9.2\n",
    "source /soft/system/software/Miniconda3/4.9.2/bin/activate MMseqs2\n",
    "\n",
    "mmseqs createdb $2/*/RiboSeq/RiboQC_RiboNovel*/RibORF/*_repre.valid.pred.pvalue.parameters.PROTEIN.noredunant.fa /projects_eg/projects/marta/TestisRestricted_Microproteins_TSA/with_TranscriptomeReconstruction/Q1_TestisORFs/human/ribORF_humanTestis_in1.fa $1/MMseqsDB/species_and_humanDB\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "search -a -s 6 /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/EvolutionaryOrigin/MMseqsDB/ribORF_humanTestis_in1_DB /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/EvolutionaryOrigin/MMseqsDB/speciesDB /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/EvolutionaryOrigin/Searching/resultDB /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/EvolutionaryOrigin/tmp \n",
      "\n",
      "MMseqs Version:                        \t14.7e284\n",
      "Substitution matrix                    \taa:blosum62.out,nucl:nucleotide.out\n",
      "Add backtrace                          \ttrue\n",
      "Alignment mode                         \t2\n",
      "Alignment mode                         \t0\n",
      "Allow wrapped scoring                  \tfalse\n",
      "E-value threshold                      \t0.001\n",
      "Seq. id. threshold                     \t0\n",
      "Min alignment length                   \t0\n",
      "Seq. id. mode                          \t0\n",
      "Alternative alignments                 \t0\n",
      "Coverage threshold                     \t0\n",
      "Coverage mode                          \t0\n",
      "Max sequence length                    \t65535\n",
      "Compositional bias                     \t1\n",
      "Compositional bias                     \t1\n",
      "Max reject                             \t2147483647\n",
      "Max accept                             \t2147483647\n",
      "Include identical seq. id.             \tfalse\n",
      "Preload mode                           \t0\n",
      "Pseudo count a                         \tsubstitution:1.100,context:1.400\n",
      "Pseudo count b                         \tsubstitution:4.100,context:5.800\n",
      "Score bias                             \t0\n",
      "Realign hits                           \tfalse\n",
      "Realign score bias                     \t-0.2\n",
      "Realign max seqs                       \t2147483647\n",
      "Correlation score weight               \t0\n",
      "Gap open cost                          \taa:11,nucl:5\n",
      "Gap extension cost                     \taa:1,nucl:2\n",
      "Zdrop                                  \t40\n",
      "Threads                                \t24\n",
      "Compressed                             \t0\n",
      "Verbosity                              \t3\n",
      "Seed substitution matrix               \taa:VTML80.out,nucl:nucleotide.out\n",
      "Sensitivity                            \t6\n",
      "k-mer length                           \t0\n",
      "k-score                                \tseq:2147483647,prof:2147483647\n",
      "Alphabet size                          \taa:21,nucl:5\n",
      "Max results per query                  \t300\n",
      "Split database                         \t0\n",
      "Split mode                             \t2\n",
      "Split memory limit                     \t0\n",
      "Diagonal scoring                       \ttrue\n",
      "Exact k-mer matching                   \t0\n",
      "Mask residues                          \t1\n",
      "Mask residues probability              \t0.9\n",
      "Mask lower case residues               \t0\n",
      "Minimum diagonal score                 \t15\n",
      "Selected taxa                          \t\n",
      "Spaced k-mers                          \t1\n",
      "Spaced k-mer pattern                   \t\n",
      "Local temporary path                   \t\n",
      "Rescore mode                           \t0\n",
      "Remove hits by seq. id. and coverage   \tfalse\n",
      "Sort results                           \t0\n",
      "Mask profile                           \t1\n",
      "Profile E-value threshold              \t0.1\n",
      "Global sequence weighting              \tfalse\n",
      "Allow deletions                        \tfalse\n",
      "Filter MSA                             \t1\n",
      "Use filter only at N seqs              \t0\n",
      "Maximum seq. id. threshold             \t0.9\n",
      "Minimum seq. id.                       \t0.0\n",
      "Minimum score per column               \t-20\n",
      "Minimum coverage                       \t0\n",
      "Select N most diverse seqs             \t1000\n",
      "Pseudo count mode                      \t0\n",
      "Gap pseudo count                       \t10\n",
      "Min codons in orf                      \t30\n",
      "Max codons in length                   \t32734\n",
      "Max orf gaps                           \t2147483647\n",
      "Contig start mode                      \t2\n",
      "Contig end mode                        \t2\n",
      "Orf start mode                         \t1\n",
      "Forward frames                         \t1,2,3\n",
      "Reverse frames                         \t1,2,3\n",
      "Translation table                      \t1\n",
      "Translate orf                          \t0\n",
      "Use all table starts                   \tfalse\n",
      "Offset of numeric ids                  \t0\n",
      "Create lookup                          \t0\n",
      "Add orf stop                           \tfalse\n",
      "Overlap between sequences              \t0\n",
      "Sequence split mode                    \t1\n",
      "Header split mode                      \t0\n",
      "Chain overlapping alignments           \t0\n",
      "Merge query                            \t1\n",
      "Search type                            \t0\n",
      "Search iterations                      \t1\n",
      "Start sensitivity                      \t4\n",
      "Search steps                           \t1\n",
      "Exhaustive search mode                 \tfalse\n",
      "Filter results during exhaustive search\t0\n",
      "Strand selection                       \t1\n",
      "LCA search mode                        \tfalse\n",
      "Disk space limit                       \t0\n",
      "MPI runner                             \t\n",
      "Force restart with latest tmp          \tfalse\n",
      "Remove temporary files                 \tfalse\n",
      "\n",
      "prefilter /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/EvolutionaryOrigin/MMseqsDB/ribORF_humanTestis_in1_DB /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/EvolutionaryOrigin/MMseqsDB/speciesDB /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/EvolutionaryOrigin/tmp/4956806787674595260/pref_0 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' --seed-sub-mat 'aa:VTML80.out,nucl:nucleotide.out' -k 0 --k-score seq:2147483647,prof:2147483647 --alph-size aa:21,nucl:5 --max-seq-len 65535 --max-seqs 300 --split 0 --split-mode 2 --split-memory-limit 0 -c 0 --cov-mode 0 --comp-bias-corr 1 --comp-bias-corr-scale 1 --diag-score 1 --exact-kmer-matching 0 --mask 1 --mask-prob 0.9 --mask-lower-case 0 --min-ungapped-score 15 --add-self-matches 0 --spaced-kmer-mode 1 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --threads 24 --compressed 0 -v 3 -s 6.0 \n",
      "\n",
      "Query database size: 15040 type: Aminoacid\n",
      "Estimated memory consumption: 1G\n",
      "Target database size: 168668 type: Aminoacid\n",
      "Index table k-mer threshold: 109 at k-mer size 6 \n",
      "Index table: counting k-mers\n",
      "[=================================================================] 168.67K 0s 734ms\n",
      "Index table: Masked residues: 2079201\n",
      "Index table: fill\n",
      "[=================================================================] 168.67K 0s 521ms\n",
      "Index statistics\n",
      "Entries:          46152190\n",
      "DB size:          752 MB\n",
      "Avg k-mer size:   0.721128\n",
      "Top 10 k-mers\n",
      "    DVVFEE\t528\n",
      "    LLLLLL\t302\n",
      "    IHDKNI\t286\n",
      "    TFDAFT\t279\n",
      "    VMEYLV\t250\n",
      "    LYDMNY\t241\n",
      "    LLPQLY\t199\n",
      "    LEYNSL\t146\n",
      "    SSSSGS\t143\n",
      "    LGGFVY\t142\n",
      "Time for index table init: 0h 0m 1s 695ms\n",
      "Process prefiltering step 1 of 1\n",
      "\n",
      "k-mer similarity threshold: 109\n",
      "Starting prefiltering scores calculation (step 1 of 1)\n",
      "Query db start 1 to 15040\n",
      "Target db start 1 to 168668\n",
      "[=================================================================] 15.04K 11s 473ms\n",
      "\n",
      "418.101239 k-mers per position\n",
      "151070 DB matches per sequence\n",
      "4 overflows\n",
      "0 queries produce too many hits (truncated result)\n",
      "266 sequences passed prefiltering per query sequence\n",
      "300 median result list length\n",
      "238 sequences with 0 size result lists\n",
      "Time for merging to pref_0: 0h 0m 0s 560ms\n",
      "Time for processing: 0h 0m 15s 783ms\n",
      "align /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/EvolutionaryOrigin/MMseqsDB/ribORF_humanTestis_in1_DB /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/EvolutionaryOrigin/MMseqsDB/speciesDB /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/EvolutionaryOrigin/tmp/4956806787674595260/pref_0 /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/EvolutionaryOrigin/Searching/resultDB --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' -a 1 --alignment-mode 2 --alignment-output-mode 0 --wrapped-scoring 0 -e 0.001 --min-seq-id 0 --min-aln-len 0 --seq-id-mode 0 --alt-ali 0 -c 0 --cov-mode 0 --max-seq-len 65535 --comp-bias-corr 1 --comp-bias-corr-scale 1 --max-rejected 2147483647 --max-accept 2147483647 --add-self-matches 0 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --score-bias 0 --realign 0 --realign-score-bias -0.2 --realign-max-seqs 2147483647 --corr-score-weight 0 --gap-open aa:11,nucl:5 --gap-extend aa:1,nucl:2 --zdrop 40 --threads 24 --compressed 0 -v 3 \n",
      "\n",
      "Compute score, coverage and sequence identity\n",
      "Query database size: 15040 type: Aminoacid\n",
      "Target database size: 168668 type: Aminoacid\n",
      "Calculation of alignments\n",
      "[=================================================================] 15.04K 1m 2s 303ms\n",
      "Time for merging to resultDB: 0h 0m 0s 194ms\n",
      "4005217 alignments calculated\n",
      "996267 sequence pairs passed the thresholds (0.248742 of overall calculated)\n",
      "66.241158 hits per query sequence\n",
      "Time for processing: 0h 1m 5s 258ms\n"
     ]
    }
   ],
   "source": [
    "%%bash -s \"$evoDir\" \"$users_dir\"\n",
    "\n",
    "module load Miniconda3/4.9.2\n",
    "source /soft/system/software/Miniconda3/4.9.2/bin/activate MMseqs2\n",
    "mkdir -p $1/Searching\n",
    "\n",
    "## The alignment consists of two steps the prefilter and alignment. To run the search, type:\n",
    "mmseqs search -a -s 6 $1/MMseqsDB/ribORF_humanTestis_in1_DB $1/MMseqsDB/speciesDB $1/Searching/resultDB $1/tmp"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "convertalis /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/EvolutionaryOrigin/MMseqsDB/ribORF_humanTestis_in1_DB /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/EvolutionaryOrigin/MMseqsDB/speciesDB /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/EvolutionaryOrigin/Searching/resultDB /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/EvolutionaryOrigin/Searching/resultDB.m8 \n",
      "\n",
      "MMseqs Version:        \t14.7e284\n",
      "Substitution matrix    \taa:blosum62.out,nucl:nucleotide.out\n",
      "Alignment format       \t0\n",
      "Format alignment output\tquery,target,fident,alnlen,mismatch,gapopen,qstart,qend,tstart,tend,evalue,bits\n",
      "Translation table      \t1\n",
      "Gap open cost          \taa:11,nucl:5\n",
      "Gap extension cost     \taa:1,nucl:2\n",
      "Database output        \tfalse\n",
      "Preload mode           \t0\n",
      "Search type            \t0\n",
      "Threads                \t24\n",
      "Compressed             \t0\n",
      "Verbosity              \t3\n",
      "\n",
      "[=================================================================] 15.04K 0s 763ms\n",
      "Time for merging to resultDB.m8: 0h 0m 3s 83ms\n",
      "Time for processing: 0h 0m 5s 890ms\n"
     ]
    }
   ],
   "source": [
    "%%bash -s \"$evoDir\" \"$users_dir\"\n",
    "\n",
    "module load Miniconda3/4.9.2\n",
    "source /soft/system/software/Miniconda3/4.9.2/bin/activate MMseqs2\n",
    "\n",
    "# Then, convert the result database into a BLAST tab formatted file (option -m 8 in legacy blast, -outfmt 6 in blast+):\n",
    "# Format alignment output\tquery,target,fident,alnlen,mismatch,gapopen,qstart,qend,tstart,tend,evalue,bits\n",
    "\n",
    "mmseqs convertalis $1/MMseqsDB/ribORF_humanTestis_in1_DB $1/MMseqsDB/speciesDB $1/Searching/resultDB $1/Searching/resultDB.m8"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "cluster /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/EvolutionaryOrigin/MMseqsDB/species_and_humanDB /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/EvolutionaryOrigin/Clustering/species_and_human_cluDB /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/EvolutionaryOrigin/tmp \n",
      "\n",
      "MMseqs Version:                     \t14.7e284\n",
      "Substitution matrix                 \taa:blosum62.out,nucl:nucleotide.out\n",
      "Seed substitution matrix            \taa:VTML80.out,nucl:nucleotide.out\n",
      "Sensitivity                         \t4\n",
      "k-mer length                        \t0\n",
      "k-score                             \tseq:2147483647,prof:2147483647\n",
      "Alphabet size                       \taa:21,nucl:5\n",
      "Max sequence length                 \t65535\n",
      "Max results per query               \t20\n",
      "Split database                      \t0\n",
      "Split mode                          \t2\n",
      "Split memory limit                  \t0\n",
      "Coverage threshold                  \t0.8\n",
      "Coverage mode                       \t0\n",
      "Compositional bias                  \t1\n",
      "Compositional bias                  \t1\n",
      "Diagonal scoring                    \ttrue\n",
      "Exact k-mer matching                \t0\n",
      "Mask residues                       \t1\n",
      "Mask residues probability           \t0.9\n",
      "Mask lower case residues            \t0\n",
      "Minimum diagonal score              \t15\n",
      "Selected taxa                       \t\n",
      "Include identical seq. id.          \tfalse\n",
      "Spaced k-mers                       \t1\n",
      "Preload mode                        \t0\n",
      "Pseudo count a                      \tsubstitution:1.100,context:1.400\n",
      "Pseudo count b                      \tsubstitution:4.100,context:5.800\n",
      "Spaced k-mer pattern                \t\n",
      "Local temporary path                \t\n",
      "Threads                             \t24\n",
      "Compressed                          \t0\n",
      "Verbosity                           \t3\n",
      "Add backtrace                       \tfalse\n",
      "Alignment mode                      \t3\n",
      "Alignment mode                      \t0\n",
      "Allow wrapped scoring               \tfalse\n",
      "E-value threshold                   \t0.001\n",
      "Seq. id. threshold                  \t0\n",
      "Min alignment length                \t0\n",
      "Seq. id. mode                       \t0\n",
      "Alternative alignments              \t0\n",
      "Max reject                          \t2147483647\n",
      "Max accept                          \t2147483647\n",
      "Score bias                          \t0\n",
      "Realign hits                        \tfalse\n",
      "Realign score bias                  \t-0.2\n",
      "Realign max seqs                    \t2147483647\n",
      "Correlation score weight            \t0\n",
      "Gap open cost                       \taa:11,nucl:5\n",
      "Gap extension cost                  \taa:1,nucl:2\n",
      "Zdrop                               \t40\n",
      "Rescore mode                        \t0\n",
      "Remove hits by seq. id. and coverage\tfalse\n",
      "Sort results                        \t0\n",
      "Cluster mode                        \t0\n",
      "Max connected component depth       \t1000\n",
      "Similarity type                     \t2\n",
      "Single step clustering              \tfalse\n",
      "Cascaded clustering steps           \t3\n",
      "Cluster reassign                    \tfalse\n",
      "Remove temporary files              \tfalse\n",
      "Force restart with latest tmp       \tfalse\n",
      "MPI runner                          \t\n",
      "k-mers per sequence                 \t21\n",
      "Scale k-mers per sequence           \taa:0.000,nucl:0.200\n",
      "Adjust k-mer length                 \tfalse\n",
      "Shift hash                          \t67\n",
      "Include only extendable             \tfalse\n",
      "Skip repeating k-mers               \tfalse\n",
      "\n",
      "Set cluster sensitivity to -s 6.000000\n",
      "Set cluster mode SET COVER\n",
      "Set cluster iterations to 3\n",
      "linclust /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/EvolutionaryOrigin/MMseqsDB/species_and_humanDB /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/EvolutionaryOrigin/tmp/1320681769031790363/clu_redundancy /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/EvolutionaryOrigin/tmp/1320681769031790363/linclust --cluster-mode 0 --max-iterations 1000 --similarity-type 2 --threads 24 --compressed 0 -v 3 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' -a 0 --alignment-mode 3 --alignment-output-mode 0 --wrapped-scoring 0 -e 0.001 --min-seq-id 0 --min-aln-len 0 --seq-id-mode 0 --alt-ali 0 -c 0.8 --cov-mode 0 --max-seq-len 65535 --comp-bias-corr 1 --comp-bias-corr-scale 1 --max-rejected 2147483647 --max-accept 2147483647 --add-self-matches 0 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --score-bias 0 --realign 0 --realign-score-bias -0.2 --realign-max-seqs 2147483647 --corr-score-weight 0 --gap-open aa:11,nucl:5 --gap-extend aa:1,nucl:2 --zdrop 40 --alph-size aa:13,nucl:5 --kmer-per-seq 21 --spaced-kmer-mode 1 --kmer-per-seq-scale aa:0.000,nucl:0.200 --adjust-kmer-len 0 --mask 0 --mask-prob 0.9 --mask-lower-case 0 -k 0 --hash-shift 67 --split-memory-limit 0 --include-only-extendable 0 --ignore-multi-kmer 0 --rescore-mode 0 --filter-hits 0 --sort-results 0 --remove-tmp-files 0 --force-reuse 0 \n",
      "\n",
      "kmermatcher /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/EvolutionaryOrigin/MMseqsDB/species_and_humanDB /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/EvolutionaryOrigin/tmp/1320681769031790363/linclust/12210831225304955152/pref --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' --alph-size aa:13,nucl:5 --min-seq-id 0 --kmer-per-seq 21 --spaced-kmer-mode 1 --kmer-per-seq-scale aa:0.000,nucl:0.200 --adjust-kmer-len 0 --mask 0 --mask-prob 0.9 --mask-lower-case 0 --cov-mode 0 -k 0 -c 0.8 --max-seq-len 65535 --hash-shift 67 --split-memory-limit 0 --include-only-extendable 0 --ignore-multi-kmer 0 --threads 24 --compressed 0 -v 3 \n",
      "\n",
      "kmermatcher /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/EvolutionaryOrigin/MMseqsDB/species_and_humanDB /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/EvolutionaryOrigin/tmp/1320681769031790363/linclust/12210831225304955152/pref --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' --alph-size aa:13,nucl:5 --min-seq-id 0 --kmer-per-seq 21 --spaced-kmer-mode 1 --kmer-per-seq-scale aa:0.000,nucl:0.200 --adjust-kmer-len 0 --mask 0 --mask-prob 0.9 --mask-lower-case 0 --cov-mode 0 -k 0 -c 0.8 --max-seq-len 65535 --hash-shift 67 --split-memory-limit 0 --include-only-extendable 0 --ignore-multi-kmer 0 --threads 24 --compressed 0 -v 3 \n",
      "\n",
      "Database size: 183708 type: Aminoacid\n",
      "Reduced amino acid alphabet: (A S T) (C) (D B N) (E Q Z) (F Y) (G) (H) (I V) (K R) (L J M) (P) (W) (X) \n",
      "\n",
      "Generate k-mers list for 1 split\n",
      "[=================================================================] 183.71K 0s 520ms\n",
      "Sort kmer 0h 0m 0s 38ms\n",
      "Sort by rep. sequence 0h 0m 0s 44ms\n",
      "Time for fill: 0h 0m 0s 24ms\n",
      "Time for merging to pref: 0h 0m 0s 79ms\n",
      "Time for processing: 0h 0m 0s 922ms\n",
      "rescorediagonal /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/EvolutionaryOrigin/MMseqsDB/species_and_humanDB /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/EvolutionaryOrigin/MMseqsDB/species_and_humanDB /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/EvolutionaryOrigin/tmp/1320681769031790363/linclust/12210831225304955152/pref /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/EvolutionaryOrigin/tmp/1320681769031790363/linclust/12210831225304955152/pref_rescore1 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' --rescore-mode 0 --wrapped-scoring 0 --filter-hits 0 -e 0.001 -c 0.8 -a 0 --cov-mode 0 --min-seq-id 0.5 --min-aln-len 0 --seq-id-mode 0 --add-self-matches 0 --sort-results 0 --db-load-mode 0 --threads 24 --compressed 0 -v 3 \n",
      "\n",
      "[=================================================================] 183.71K 0s 58ms\n",
      "Time for merging to pref_rescore1: 0h 0m 0s 477ms\n",
      "Time for processing: 0h 0m 3s 203ms\n",
      "clust /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/EvolutionaryOrigin/MMseqsDB/species_and_humanDB /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/EvolutionaryOrigin/tmp/1320681769031790363/linclust/12210831225304955152/pref_rescore1 /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/EvolutionaryOrigin/tmp/1320681769031790363/linclust/12210831225304955152/pre_clust --cluster-mode 0 --max-iterations 1000 --similarity-type 2 --threads 24 --compressed 0 -v 3 \n",
      "\n",
      "Clustering mode: Set Cover\n",
      "[=================================================================] 183.71K 0s 15ms\n",
      "Sort entries\n",
      "Find missing connections\n",
      "Found 76707 new connections.\n",
      "Reconstruct initial order\n",
      "[=================================================================] 183.71K 0s 12ms\n",
      "Add missing connections\n",
      "[=================================================================] 183.71K 0s 5ms\n",
      "\n",
      "Time for read in: 0h 0m 0s 85ms\n",
      "Total time: 0h 0m 0s 190ms\n",
      "\n",
      "Size of the sequence database: 183708\n",
      "Size of the alignment database: 183708\n",
      "Number of clusters: 137665\n",
      "\n",
      "Writing results 0h 0m 0s 26ms\n",
      "Time for merging to pre_clust: 0h 0m 0s 40ms\n",
      "Time for processing: 0h 0m 0s 408ms\n",
      "createsubdb /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/EvolutionaryOrigin/tmp/1320681769031790363/linclust/12210831225304955152/order_redundancy /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/EvolutionaryOrigin/MMseqsDB/species_and_humanDB /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/EvolutionaryOrigin/tmp/1320681769031790363/linclust/12210831225304955152/input_step_redundancy -v 3 --subdb-mode 1 \n",
      "\n",
      "Time for merging to input_step_redundancy: 0h 0m 0s 53ms\n",
      "Time for processing: 0h 0m 0s 287ms\n",
      "createsubdb /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/EvolutionaryOrigin/tmp/1320681769031790363/linclust/12210831225304955152/order_redundancy /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/EvolutionaryOrigin/tmp/1320681769031790363/linclust/12210831225304955152/pref /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/EvolutionaryOrigin/tmp/1320681769031790363/linclust/12210831225304955152/pref_filter1 -v 3 --subdb-mode 1 \n",
      "\n",
      "Time for merging to pref_filter1: 0h 0m 0s 41ms\n",
      "Time for processing: 0h 0m 0s 320ms\n",
      "filterdb /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/EvolutionaryOrigin/tmp/1320681769031790363/linclust/12210831225304955152/pref_filter1 /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/EvolutionaryOrigin/tmp/1320681769031790363/linclust/12210831225304955152/pref_filter2 --filter-file /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/EvolutionaryOrigin/tmp/1320681769031790363/linclust/12210831225304955152/order_redundancy --threads 24 --compressed 0 -v 3 \n",
      "\n",
      "Filtering using file(s)\n",
      "[=================================================================] 137.66K 0s 29ms\n",
      "Time for merging to pref_filter2: 0h 0m 0s 375ms\n",
      "Time for processing: 0h 0m 1s 74ms\n",
      "rescorediagonal /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/EvolutionaryOrigin/tmp/1320681769031790363/linclust/12210831225304955152/input_step_redundancy /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/EvolutionaryOrigin/tmp/1320681769031790363/linclust/12210831225304955152/input_step_redundancy /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/EvolutionaryOrigin/tmp/1320681769031790363/linclust/12210831225304955152/pref_filter2 /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/EvolutionaryOrigin/tmp/1320681769031790363/linclust/12210831225304955152/pref_rescore2 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' --rescore-mode 1 --wrapped-scoring 0 --filter-hits 1 -e 0.001 -c 0.8 -a 0 --cov-mode 0 --min-seq-id 0 --min-aln-len 0 --seq-id-mode 0 --add-self-matches 0 --sort-results 0 --db-load-mode 0 --threads 24 --compressed 0 -v 3 \n",
      "\n",
      "Can not find any score per column for coverage 0.800000 and sequence identity 0.000000. No hit will be filtered.\n",
      "[=================================================================] 137.66K 0s 93ms\n",
      "Time for merging to pref_rescore2: 0h 0m 0s 384ms\n",
      "Time for processing: 0h 0m 1s 198ms\n",
      "align /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/EvolutionaryOrigin/tmp/1320681769031790363/linclust/12210831225304955152/input_step_redundancy /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/EvolutionaryOrigin/tmp/1320681769031790363/linclust/12210831225304955152/input_step_redundancy /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/EvolutionaryOrigin/tmp/1320681769031790363/linclust/12210831225304955152/pref_rescore2 /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/EvolutionaryOrigin/tmp/1320681769031790363/linclust/12210831225304955152/aln --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' -a 0 --alignment-mode 3 --alignment-output-mode 0 --wrapped-scoring 0 -e 0.001 --min-seq-id 0 --min-aln-len 0 --seq-id-mode 0 --alt-ali 0 -c 0.8 --cov-mode 0 --max-seq-len 65535 --comp-bias-corr 1 --comp-bias-corr-scale 1 --max-rejected 2147483647 --max-accept 2147483647 --add-self-matches 0 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --score-bias 0 --realign 0 --realign-score-bias -0.2 --realign-max-seqs 2147483647 --corr-score-weight 0 --gap-open aa:11,nucl:5 --gap-extend aa:1,nucl:2 --zdrop 40 --threads 24 --compressed 0 -v 3 \n",
      "\n",
      "Compute score, coverage and sequence identity\n",
      "Query database size: 137665 type: Aminoacid\n",
      "Target database size: 137665 type: Aminoacid\n",
      "Calculation of alignments\n",
      "[=================================================================] 137.66K 6s 339ms\n",
      "Time for merging to aln: 0h 0m 2s 58ms\n",
      "163524 alignments calculated\n",
      "157234 sequence pairs passed the thresholds (0.961535 of overall calculated)\n",
      "1.142149 hits per query sequence\n",
      "Time for processing: 0h 0m 9s 207ms\n",
      "clust /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/EvolutionaryOrigin/tmp/1320681769031790363/linclust/12210831225304955152/input_step_redundancy /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/EvolutionaryOrigin/tmp/1320681769031790363/linclust/12210831225304955152/aln /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/EvolutionaryOrigin/tmp/1320681769031790363/linclust/12210831225304955152/clust --cluster-mode 0 --max-iterations 1000 --similarity-type 2 --threads 24 --compressed 0 -v 3 \n",
      "\n",
      "Clustering mode: Set Cover\n",
      "[=================================================================] 137.66K 0s 11ms\n",
      "Sort entries\n",
      "Find missing connections\n",
      "Found 19569 new connections.\n",
      "Reconstruct initial order\n",
      "[=================================================================] 137.66K 0s 11ms\n",
      "Add missing connections\n",
      "[=================================================================] 137.66K 0s 3ms\n",
      "\n",
      "Time for read in: 0h 0m 0s 83ms\n",
      "Total time: 0h 0m 0s 219ms\n",
      "\n",
      "Size of the sequence database: 137665\n",
      "Size of the alignment database: 137665\n",
      "Number of clusters: 123893\n",
      "\n",
      "Writing results 0h 0m 0s 30ms\n",
      "Time for merging to clust: 0h 0m 0s 77ms\n",
      "Time for processing: 0h 0m 0s 487ms\n",
      "mergeclusters /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/EvolutionaryOrigin/MMseqsDB/species_and_humanDB /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/EvolutionaryOrigin/tmp/1320681769031790363/clu_redundancy /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/EvolutionaryOrigin/tmp/1320681769031790363/linclust/12210831225304955152/pre_clust /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/EvolutionaryOrigin/tmp/1320681769031790363/linclust/12210831225304955152/clust --threads 24 --compressed 0 -v 3 \n",
      "\n",
      "Clustering step 1\n",
      "[=================================================================] 137.66K 0s 41ms\n",
      "Clustering step 2\n",
      "[=================================================================] 123.89K 0s 117ms\n",
      "Write merged clustering\n",
      "[=================================================================] 183.71K 0s 738ms\n",
      "Time for merging to clu_redundancy: 0h 0m 1s 102ms\n",
      "Time for processing: 0h 0m 2s 313ms\n",
      "createsubdb /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/EvolutionaryOrigin/tmp/1320681769031790363/clu_redundancy /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/EvolutionaryOrigin/MMseqsDB/species_and_humanDB /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/EvolutionaryOrigin/tmp/1320681769031790363/input_step_redundancy -v 3 --subdb-mode 1 \n",
      "\n",
      "Time for merging to input_step_redundancy: 0h 0m 0s 87ms\n",
      "Time for processing: 0h 0m 0s 310ms\n",
      "prefilter /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/EvolutionaryOrigin/tmp/1320681769031790363/input_step_redundancy /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/EvolutionaryOrigin/tmp/1320681769031790363/input_step_redundancy /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/EvolutionaryOrigin/tmp/1320681769031790363/pref_step0 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' --seed-sub-mat 'aa:VTML80.out,nucl:nucleotide.out' -s 1 -k 0 --k-score seq:2147483647,prof:2147483647 --alph-size aa:21,nucl:5 --max-seq-len 65535 --max-seqs 20 --split 0 --split-mode 2 --split-memory-limit 0 -c 0.8 --cov-mode 0 --comp-bias-corr 0 --comp-bias-corr-scale 1 --diag-score 0 --exact-kmer-matching 0 --mask 1 --mask-prob 0.9 --mask-lower-case 0 --min-ungapped-score 0 --add-self-matches 0 --spaced-kmer-mode 1 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --threads 24 --compressed 0 -v 3 \n",
      "\n",
      "Query database size: 123893 type: Aminoacid\n",
      "Estimated memory consumption: 1G\n",
      "Target database size: 123893 type: Aminoacid\n",
      "Index table k-mer threshold: 154 at k-mer size 6 \n",
      "Index table: counting k-mers\n",
      "[=================================================================] 123.89K 0s 265ms\n",
      "Index table: Masked residues: 1041307\n",
      "Index table: fill\n",
      "[=================================================================] 123.89K 0s 124ms\n",
      "Index statistics\n",
      "Entries:          10041494\n",
      "DB size:          545 MB\n",
      "Avg k-mer size:   0.156898\n",
      "Top 10 k-mers\n",
      "    TFDAFT\t175\n",
      "    LYDMNY\t162\n",
      "    VADTEW\t125\n",
      "    YRVLYR\t89\n",
      "    ECKFSS\t65\n",
      "    FTEWLS\t54\n",
      "    VHSGRT\t51\n",
      "    CLPQLY\t49\n",
      "    PPPPPP\t48\n",
      "    VAYSEW\t48\n",
      "Time for index table init: 0h 0m 0s 832ms\n",
      "Process prefiltering step 1 of 1\n",
      "\n",
      "k-mer similarity threshold: 154\n",
      "Starting prefiltering scores calculation (step 1 of 1)\n",
      "Query db start 1 to 123893\n",
      "Target db start 1 to 123893\n",
      "[=================================================================] 123.89K 0s 437ms\n",
      "\n",
      "2.164017 k-mers per position\n",
      "321 DB matches per sequence\n",
      "0 overflows\n",
      "0 queries produce too many hits (truncated result)\n",
      "3 sequences passed prefiltering per query sequence\n",
      "1 median result list length\n",
      "0 sequences with 0 size result lists\n",
      "Time for merging to pref_step0: 0h 0m 0s 405ms\n",
      "Time for processing: 0h 0m 2s 620ms\n",
      "align /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/EvolutionaryOrigin/tmp/1320681769031790363/input_step_redundancy /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/EvolutionaryOrigin/tmp/1320681769031790363/input_step_redundancy /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/EvolutionaryOrigin/tmp/1320681769031790363/pref_step0 /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/EvolutionaryOrigin/tmp/1320681769031790363/aln_step0 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' -a 0 --alignment-mode 3 --alignment-output-mode 0 --wrapped-scoring 0 -e 0.001 --min-seq-id 0 --min-aln-len 0 --seq-id-mode 0 --alt-ali 0 -c 0.8 --cov-mode 0 --max-seq-len 65535 --comp-bias-corr 0 --comp-bias-corr-scale 1 --max-rejected 2147483647 --max-accept 2147483647 --add-self-matches 0 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --score-bias 0 --realign 0 --realign-score-bias -0.2 --realign-max-seqs 2147483647 --corr-score-weight 0 --gap-open aa:11,nucl:5 --gap-extend aa:1,nucl:2 --zdrop 40 --threads 24 --compressed 0 -v 3 \n",
      "\n",
      "Compute score, coverage and sequence identity\n",
      "Query database size: 123893 type: Aminoacid\n",
      "Target database size: 123893 type: Aminoacid\n",
      "Calculation of alignments\n",
      "[=================================================================] 123.89K 5s 716ms\n",
      "Time for merging to aln_step0: 0h 0m 1s 146ms\n",
      "208757 alignments calculated\n",
      "161600 sequence pairs passed the thresholds (0.774106 of overall calculated)\n",
      "1.304351 hits per query sequence\n",
      "Time for processing: 0h 0m 8s 208ms\n",
      "clust /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/EvolutionaryOrigin/tmp/1320681769031790363/input_step_redundancy /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/EvolutionaryOrigin/tmp/1320681769031790363/aln_step0 /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/EvolutionaryOrigin/tmp/1320681769031790363/clu_step0 --cluster-mode 0 --max-iterations 1000 --similarity-type 2 --threads 24 --compressed 0 -v 3 \n",
      "\n",
      "Clustering mode: Set Cover\n",
      "[=================================================================] 123.89K 0s 9ms\n",
      "Sort entries\n",
      "Find missing connections\n",
      "Found 2419 new connections.\n",
      "Reconstruct initial order\n",
      "[=================================================================] 123.89K 0s 10ms\n",
      "Add missing connections\n",
      "[=================================================================] 123.89K 0s 3ms\n",
      "\n",
      "Time for read in: 0h 0m 0s 61ms\n",
      "Total time: 0h 0m 0s 195ms\n",
      "\n",
      "Size of the sequence database: 123893\n",
      "Size of the alignment database: 123893\n",
      "Number of clusters: 113244\n",
      "\n",
      "Writing results 0h 0m 0s 20ms\n",
      "Time for merging to clu_step0: 0h 0m 0s 50ms\n",
      "Time for processing: 0h 0m 0s 645ms\n",
      "createsubdb /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/EvolutionaryOrigin/tmp/1320681769031790363/clu_step0 /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/EvolutionaryOrigin/tmp/1320681769031790363/input_step_redundancy /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/EvolutionaryOrigin/tmp/1320681769031790363/input_step1 -v 3 --subdb-mode 1 \n",
      "\n",
      "Time for merging to input_step1: 0h 0m 0s 279ms\n",
      "Time for processing: 0h 0m 0s 511ms\n",
      "prefilter /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/EvolutionaryOrigin/tmp/1320681769031790363/input_step1 /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/EvolutionaryOrigin/tmp/1320681769031790363/input_step1 /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/EvolutionaryOrigin/tmp/1320681769031790363/pref_step1 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' --seed-sub-mat 'aa:VTML80.out,nucl:nucleotide.out' -s 3.5 -k 0 --k-score seq:2147483647,prof:2147483647 --alph-size aa:21,nucl:5 --max-seq-len 65535 --max-seqs 20 --split 0 --split-mode 2 --split-memory-limit 0 -c 0.8 --cov-mode 0 --comp-bias-corr 1 --comp-bias-corr-scale 1 --diag-score 1 --exact-kmer-matching 0 --mask 1 --mask-prob 0.9 --mask-lower-case 0 --min-ungapped-score 15 --add-self-matches 0 --spaced-kmer-mode 1 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --threads 24 --compressed 0 -v 3 \n",
      "\n",
      "Query database size: 113244 type: Aminoacid\n",
      "Estimated memory consumption: 1G\n",
      "Target database size: 113244 type: Aminoacid\n",
      "Index table k-mer threshold: 131 at k-mer size 6 \n",
      "Index table: counting k-mers\n",
      "[=================================================================] 113.24K 0s 236ms\n",
      "Index table: Masked residues: 656166\n",
      "Index table: fill\n",
      "[=================================================================] 113.24K 0s 157ms\n",
      "Index statistics\n",
      "Entries:          13557140\n",
      "DB size:          565 MB\n",
      "Avg k-mer size:   0.211830\n",
      "Top 10 k-mers\n",
      "    DVVFEE\t128\n",
      "    VMEYLV\t66\n",
      "    LYDMNY\t61\n",
      "    QRLRML\t50\n",
      "    LLGPGK\t49\n",
      "    TFDAFS\t48\n",
      "    LLLLLP\t44\n",
      "    APPPPP\t43\n",
      "    QIELEQ\t40\n",
      "    GGGGGP\t38\n",
      "Time for index table init: 0h 0m 0s 816ms\n",
      "Process prefiltering step 1 of 1\n",
      "\n",
      "k-mer similarity threshold: 131\n",
      "Starting prefiltering scores calculation (step 1 of 1)\n",
      "Query db start 1 to 113244\n",
      "Target db start 1 to 113244\n",
      "[=================================================================] 113.24K 2s 211ms\n",
      "\n",
      "27.399431 k-mers per position\n",
      "1020 DB matches per sequence\n",
      "0 overflows\n",
      "0 queries produce too many hits (truncated result)\n",
      "6 sequences passed prefiltering per query sequence\n",
      "4 median result list length\n",
      "0 sequences with 0 size result lists\n",
      "Time for merging to pref_step1: 0h 0m 0s 418ms\n",
      "Time for processing: 0h 0m 4s 594ms\n",
      "align /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/EvolutionaryOrigin/tmp/1320681769031790363/input_step1 /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/EvolutionaryOrigin/tmp/1320681769031790363/input_step1 /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/EvolutionaryOrigin/tmp/1320681769031790363/pref_step1 /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/EvolutionaryOrigin/tmp/1320681769031790363/aln_step1 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' -a 0 --alignment-mode 3 --alignment-output-mode 0 --wrapped-scoring 0 -e 0.001 --min-seq-id 0 --min-aln-len 0 --seq-id-mode 0 --alt-ali 0 -c 0.8 --cov-mode 0 --max-seq-len 65535 --comp-bias-corr 1 --comp-bias-corr-scale 1 --max-rejected 2147483647 --max-accept 2147483647 --add-self-matches 0 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --score-bias 0 --realign 0 --realign-score-bias -0.2 --realign-max-seqs 2147483647 --corr-score-weight 0 --gap-open aa:11,nucl:5 --gap-extend aa:1,nucl:2 --zdrop 40 --threads 24 --compressed 0 -v 3 \n",
      "\n",
      "Compute score, coverage and sequence identity\n",
      "Query database size: 113244 type: Aminoacid\n",
      "Target database size: 113244 type: Aminoacid\n",
      "Calculation of alignments\n",
      "[=================================================================] 113.24K 1s 104ms\n",
      "Time for merging to aln_step1: 0h 0m 0s 348ms\n",
      "206425 alignments calculated\n",
      "115960 sequence pairs passed the thresholds (0.561754 of overall calculated)\n",
      "1.023984 hits per query sequence\n",
      "Time for processing: 0h 0m 2s 13ms\n",
      "clust /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/EvolutionaryOrigin/tmp/1320681769031790363/input_step1 /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/EvolutionaryOrigin/tmp/1320681769031790363/aln_step1 /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/EvolutionaryOrigin/tmp/1320681769031790363/clu_step1 --cluster-mode 0 --max-iterations 1000 --similarity-type 2 --threads 24 --compressed 0 -v 3 \n",
      "\n",
      "Clustering mode: Set Cover\n",
      "[=================================================================] 113.24K 0s 12ms\n",
      "Sort entries\n",
      "Find missing connections\n",
      "Found 702 new connections.\n",
      "Reconstruct initial order\n",
      "[=================================================================] 113.24K 0s 9ms\n",
      "Add missing connections\n",
      "[=================================================================] 113.24K 0s 2ms\n",
      "\n",
      "Time for read in: 0h 0m 0s 72ms\n",
      "Total time: 0h 0m 0s 182ms\n",
      "\n",
      "Size of the sequence database: 113244\n",
      "Size of the alignment database: 113244\n",
      "Number of clusters: 111980\n",
      "\n",
      "Writing results 0h 0m 0s 19ms\n",
      "Time for merging to clu_step1: 0h 0m 0s 36ms\n",
      "Time for processing: 0h 0m 0s 354ms\n",
      "createsubdb /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/EvolutionaryOrigin/tmp/1320681769031790363/clu_step1 /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/EvolutionaryOrigin/tmp/1320681769031790363/input_step1 /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/EvolutionaryOrigin/tmp/1320681769031790363/input_step2 -v 3 --subdb-mode 1 \n",
      "\n",
      "Time for merging to input_step2: 0h 0m 0s 41ms\n",
      "Time for processing: 0h 0m 0s 234ms\n",
      "prefilter /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/EvolutionaryOrigin/tmp/1320681769031790363/input_step2 /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/EvolutionaryOrigin/tmp/1320681769031790363/input_step2 /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/EvolutionaryOrigin/tmp/1320681769031790363/pref_step2 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' --seed-sub-mat 'aa:VTML80.out,nucl:nucleotide.out' -s 6 -k 0 --k-score seq:2147483647,prof:2147483647 --alph-size aa:21,nucl:5 --max-seq-len 65535 --max-seqs 20 --split 0 --split-mode 2 --split-memory-limit 0 -c 0.8 --cov-mode 0 --comp-bias-corr 1 --comp-bias-corr-scale 1 --diag-score 1 --exact-kmer-matching 0 --mask 1 --mask-prob 0.9 --mask-lower-case 0 --min-ungapped-score 15 --add-self-matches 0 --spaced-kmer-mode 1 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --threads 24 --compressed 0 -v 3 \n",
      "\n",
      "Query database size: 111980 type: Aminoacid\n",
      "Estimated memory consumption: 1G\n",
      "Target database size: 111980 type: Aminoacid\n",
      "Index table k-mer threshold: 109 at k-mer size 6 \n",
      "Index table: counting k-mers\n",
      "[=================================================================] 111.98K 0s 204ms\n",
      "Index table: Masked residues: 608700\n",
      "Index table: fill\n",
      "[=================================================================] 111.98K 0s 154ms\n",
      "Index statistics\n",
      "Entries:          13414932\n",
      "DB size:          565 MB\n",
      "Avg k-mer size:   0.209608\n",
      "Top 10 k-mers\n",
      "    DVVFEE\t98\n",
      "    LLLLLL\t95\n",
      "    SSSSSS\t77\n",
      "    LPLLLL\t52\n",
      "    LLPLLL\t52\n",
      "    ALLLLL\t48\n",
      "    VMEYLV\t48\n",
      "    SSSLSS\t44\n",
      "    SSLSSS\t41\n",
      "    SSSSLS\t40\n",
      "Time for index table init: 0h 0m 0s 761ms\n",
      "Process prefiltering step 1 of 1\n",
      "\n",
      "k-mer similarity threshold: 109\n",
      "Starting prefiltering scores calculation (step 1 of 1)\n",
      "Query db start 1 to 111980\n",
      "Target db start 1 to 111980\n",
      "[=================================================================] 111.98K 15s 656ms\n",
      "\n",
      "281.633132 k-mers per position\n",
      "10350 DB matches per sequence\n",
      "2 overflows\n",
      "0 queries produce too many hits (truncated result)\n",
      "14 sequences passed prefiltering per query sequence\n",
      "20 median result list length\n",
      "0 sequences with 0 size result lists\n",
      "Time for merging to pref_step2: 0h 0m 0s 641ms\n",
      "Time for processing: 0h 0m 18s 272ms\n",
      "align /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/EvolutionaryOrigin/tmp/1320681769031790363/input_step2 /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/EvolutionaryOrigin/tmp/1320681769031790363/input_step2 /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/EvolutionaryOrigin/tmp/1320681769031790363/pref_step2 /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/EvolutionaryOrigin/tmp/1320681769031790363/aln_step2 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' -a 0 --alignment-mode 3 --alignment-output-mode 0 --wrapped-scoring 0 -e 0.001 --min-seq-id 0 --min-aln-len 0 --seq-id-mode 0 --alt-ali 0 -c 0.8 --cov-mode 0 --max-seq-len 65535 --comp-bias-corr 1 --comp-bias-corr-scale 1 --max-rejected 2147483647 --max-accept 2147483647 --add-self-matches 0 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --score-bias 0 --realign 0 --realign-score-bias -0.2 --realign-max-seqs 2147483647 --corr-score-weight 0 --gap-open aa:11,nucl:5 --gap-extend aa:1,nucl:2 --zdrop 40 --threads 24 --compressed 0 -v 3 \n",
      "\n",
      "Compute score, coverage and sequence identity\n",
      "Query database size: 111980 type: Aminoacid\n",
      "Target database size: 111980 type: Aminoacid\n",
      "Calculation of alignments\n",
      "[=================================================================] 111.98K 1s 221ms\n",
      "Time for merging to aln_step2: 0h 0m 0s 388ms\n",
      "293874 alignments calculated\n",
      "112676 sequence pairs passed the thresholds (0.383416 of overall calculated)\n",
      "1.006215 hits per query sequence\n",
      "Time for processing: 0h 0m 2s 895ms\n",
      "clust /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/EvolutionaryOrigin/tmp/1320681769031790363/input_step2 /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/EvolutionaryOrigin/tmp/1320681769031790363/aln_step2 /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/EvolutionaryOrigin/tmp/1320681769031790363/clu_step2 --cluster-mode 0 --max-iterations 1000 --similarity-type 2 --threads 24 --compressed 0 -v 3 \n",
      "\n",
      "Clustering mode: Set Cover\n",
      "[=================================================================] 111.98K 0s 8ms\n",
      "Sort entries\n",
      "Find missing connections\n",
      "Found 216 new connections.\n",
      "Reconstruct initial order\n",
      "[=================================================================] 111.98K 0s 7ms\n",
      "Add missing connections\n",
      "[=================================================================] 111.98K 0s 2ms\n",
      "\n",
      "Time for read in: 0h 0m 0s 66ms\n",
      "Total time: 0h 0m 0s 158ms\n",
      "\n",
      "Size of the sequence database: 111980\n",
      "Size of the alignment database: 111980\n",
      "Number of clusters: 111546\n",
      "\n",
      "Writing results 0h 0m 0s 19ms\n",
      "Time for merging to clu_step2: 0h 0m 0s 44ms\n",
      "Time for processing: 0h 0m 0s 338ms\n",
      "mergeclusters /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/EvolutionaryOrigin/MMseqsDB/species_and_humanDB /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/EvolutionaryOrigin/Clustering/species_and_human_cluDB /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/EvolutionaryOrigin/tmp/1320681769031790363/clu_redundancy /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/EvolutionaryOrigin/tmp/1320681769031790363/clu_step0 /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/EvolutionaryOrigin/tmp/1320681769031790363/clu_step1 /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/EvolutionaryOrigin/tmp/1320681769031790363/clu_step2 --threads 24 --compressed 0 -v 3 \n",
      "\n",
      "Clustering step 1\n",
      "[=================================================================] 123.89K 0s 59ms\n",
      "Clustering step 2\n",
      "[=================================================================] 113.24K 0s 86ms\n",
      "Clustering step 3\n",
      "[=================================================================] 111.98K 0s 121ms\n",
      "Clustering step 4\n",
      "[=================================================================] 111.55K 0s 183ms\n",
      "Write merged clustering\n",
      "[=================================================================] 183.71K 0s 902ms\n",
      "Time for merging to species_and_human_cluDB: 0h 0m 0s 314ms\n",
      "Time for processing: 0h 0m 1s 404ms\n",
      "createtsv /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/EvolutionaryOrigin/MMseqsDB/species_and_humanDB /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/EvolutionaryOrigin/MMseqsDB/species_and_humanDB /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/EvolutionaryOrigin/Clustering/species_and_human_cluDB /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/EvolutionaryOrigin/Clustering/species_and_human_cluDB.tsv \n",
      "\n",
      "MMseqs Version:                 \t14.7e284\n",
      "First sequence as representative\tfalse\n",
      "Target column                   \t1\n",
      "Add full header                 \tfalse\n",
      "Sequence source                 \t0\n",
      "Database output                 \tfalse\n",
      "Threads                         \t24\n",
      "Compressed                      \t0\n",
      "Verbosity                       \t3\n",
      "\n",
      "Time for merging to species_and_human_cluDB.tsv: 0h 0m 0s 819ms\n",
      "Time for processing: 0h 0m 1s 670ms\n"
     ]
    }
   ],
   "source": [
    "%%bash -s \"$evoDir\" \"$users_dir\"\n",
    "\n",
    "module load Miniconda3/4.9.2\n",
    "source /soft/system/software/Miniconda3/4.9.2/bin/activate MMseqs2\n",
    "\n",
    "mkdir -p $1/Clustering\n",
    "\n",
    "## cluster all species and human\n",
    "mmseqs cluster $1/MMseqsDB/species_and_humanDB $1/Clustering/species_and_human_cluDB $1/tmp\n",
    "mmseqs createtsv $1/MMseqsDB/species_and_humanDB $1/MMseqsDB/species_and_humanDB $1/Clustering/species_and_human_cluDB $1/Clustering/species_and_human_cluDB.tsv"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Analysis"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>orfID</th>\n",
       "      <th>ORFseq</th>\n",
       "      <th>gene_id</th>\n",
       "      <th>gene_name</th>\n",
       "      <th>transcript_id</th>\n",
       "      <th>gene_type</th>\n",
       "      <th>orfType</th>\n",
       "      <th>length</th>\n",
       "      <th>length_aa</th>\n",
       "      <th>start_codon</th>\n",
       "      <th>ORFpep</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ENST00000000412.8:12:-|10|2450:160:994|canonic...</td>\n",
       "      <td>ATGTTCCCTTTCTACAGCTGCTGGAGGACTGGACTGCTACTACTAC...</td>\n",
       "      <td>ENSG00000003056</td>\n",
       "      <td>M6PR</td>\n",
       "      <td>ENST00000000412</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>canonical</td>\n",
       "      <td>834</td>\n",
       "      <td>278</td>\n",
       "      <td>ATG</td>\n",
       "      <td>MFPFYSCWRTGLLLLLLAVAVRESWQTEEKTCDLVGEKGKESEKEL...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ENST00000001008.6:12:+|5|3715:171:1551|canonic...</td>\n",
       "      <td>ATGACAGCCGAGGAGATGAAGGCGACCGAGAGCGGGGCGCAGTCGG...</td>\n",
       "      <td>ENSG00000004478</td>\n",
       "      <td>FKBP4</td>\n",
       "      <td>ENST00000001008</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>canonical</td>\n",
       "      <td>1380</td>\n",
       "      <td>460</td>\n",
       "      <td>ATG</td>\n",
       "      <td>MTAEEMKATESGAQSAPLPMEGVDISPKQDEGVLKVIKREGTGTEM...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ENST00000001146.7:2:-|1|4556:29:1568|canonical...</td>\n",
       "      <td>ATGCTCTTTGAGGGCTTGGATCTGGTGTCGGCGCTGGCCACCCTCG...</td>\n",
       "      <td>ENSG00000003137</td>\n",
       "      <td>CYP26B1</td>\n",
       "      <td>ENST00000001146</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>canonical</td>\n",
       "      <td>1539</td>\n",
       "      <td>513</td>\n",
       "      <td>ATG</td>\n",
       "      <td>MLFEGLDLVSALATLAACLVSVTLLLAVSQQLWQLRWAATRDKSCK...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ENST00000002165.11:6:-|3|2385:87:1491|canonica...</td>\n",
       "      <td>ATGCGGCCCCAGGAGCTCCCCAGGCTCGCGTTCCCGTTGCTGCTGT...</td>\n",
       "      <td>ENSG00000001036</td>\n",
       "      <td>FUCA2</td>\n",
       "      <td>ENST00000002165</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>canonical</td>\n",
       "      <td>1404</td>\n",
       "      <td>468</td>\n",
       "      <td>ATG</td>\n",
       "      <td>MRPQELPRLAFPLLLLLLLLLPPPPCPAHSATRFDPTWESLDARQL...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ENST00000002596.6:4:-|24|7160:305:1229|canonic...</td>\n",
       "      <td>ATGGCCGCGCTGCTCCTGGGCGCGGTGCTGCTGGTGGCCCAGCCCC...</td>\n",
       "      <td>ENSG00000002587</td>\n",
       "      <td>HS3ST1</td>\n",
       "      <td>ENST00000002596</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>canonical</td>\n",
       "      <td>924</td>\n",
       "      <td>308</td>\n",
       "      <td>ATG</td>\n",
       "      <td>MAALLLGAVLLVAQPQLVPSRPAELGQQELLRKAGTLQDDVRDGVA...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12516</th>\n",
       "      <td>TCONS_00001992:8:+|9|1191:153:675|noncoding|ATG</td>\n",
       "      <td>ATGACCGGGTCAGCGGAAATTTCAGAGCCAGCCAGCAGCTCAGGTC...</td>\n",
       "      <td>XLOC_001676</td>\n",
       "      <td>XLOC_001676</td>\n",
       "      <td>TCONS_00001992</td>\n",
       "      <td>novel</td>\n",
       "      <td>noncoding</td>\n",
       "      <td>522</td>\n",
       "      <td>174</td>\n",
       "      <td>ATG</td>\n",
       "      <td>MTGSAEISEPASSSGPRGRGRNGTAPRRQPRGHSRATADSGTSLSA...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12517</th>\n",
       "      <td>TCONS_00002050:9:-|86|10655:1396:1420|noncodin...</td>\n",
       "      <td>CTGTGTAAGTGCATCAGAACATAA</td>\n",
       "      <td>XLOC_001809</td>\n",
       "      <td>XLOC_001809</td>\n",
       "      <td>TCONS_00002050</td>\n",
       "      <td>novel</td>\n",
       "      <td>noncoding</td>\n",
       "      <td>24</td>\n",
       "      <td>8</td>\n",
       "      <td>CTG</td>\n",
       "      <td>MCKCIRT*</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12518</th>\n",
       "      <td>TCONS_00002051:9:-|9|437:109:145|noncoding|ATG</td>\n",
       "      <td>ATGGCTTTACCCATGGAGATTTTTGTTGCTGGCTAA</td>\n",
       "      <td>XLOC_001810</td>\n",
       "      <td>XLOC_001810</td>\n",
       "      <td>TCONS_00002051</td>\n",
       "      <td>novel</td>\n",
       "      <td>noncoding</td>\n",
       "      <td>36</td>\n",
       "      <td>12</td>\n",
       "      <td>ATG</td>\n",
       "      <td>MALPMEIFVAG*</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12519</th>\n",
       "      <td>TCONS_00002066:9:+|15|2910:249:327|noncoding|ATG</td>\n",
       "      <td>ATGTGCTTAATAAATATACGCCGAATGACTGAACTGCAACCAAGGA...</td>\n",
       "      <td>XLOC_001758</td>\n",
       "      <td>XLOC_001758</td>\n",
       "      <td>TCONS_00002066</td>\n",
       "      <td>novel</td>\n",
       "      <td>noncoding</td>\n",
       "      <td>78</td>\n",
       "      <td>26</td>\n",
       "      <td>ATG</td>\n",
       "      <td>MCLINIRRMTELQPRTLYTCFTDLV*</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12520</th>\n",
       "      <td>TCONS_00002105:9:+|25|1451:469:511|noncoding|TTG</td>\n",
       "      <td>TTGTACGTTGGTGGAGACGTGCCTTTTTTCCTTTGCACTTAA</td>\n",
       "      <td>XLOC_001725</td>\n",
       "      <td>XLOC_001725</td>\n",
       "      <td>TCONS_00002105</td>\n",
       "      <td>novel</td>\n",
       "      <td>noncoding</td>\n",
       "      <td>42</td>\n",
       "      <td>14</td>\n",
       "      <td>TTG</td>\n",
       "      <td>MYVGGDVPFFLCT*</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>12521 rows × 11 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                   orfID  \\\n",
       "0      ENST00000000412.8:12:-|10|2450:160:994|canonic...   \n",
       "1      ENST00000001008.6:12:+|5|3715:171:1551|canonic...   \n",
       "2      ENST00000001146.7:2:-|1|4556:29:1568|canonical...   \n",
       "3      ENST00000002165.11:6:-|3|2385:87:1491|canonica...   \n",
       "4      ENST00000002596.6:4:-|24|7160:305:1229|canonic...   \n",
       "...                                                  ...   \n",
       "12516    TCONS_00001992:8:+|9|1191:153:675|noncoding|ATG   \n",
       "12517  TCONS_00002050:9:-|86|10655:1396:1420|noncodin...   \n",
       "12518     TCONS_00002051:9:-|9|437:109:145|noncoding|ATG   \n",
       "12519   TCONS_00002066:9:+|15|2910:249:327|noncoding|ATG   \n",
       "12520   TCONS_00002105:9:+|25|1451:469:511|noncoding|TTG   \n",
       "\n",
       "                                                  ORFseq          gene_id  \\\n",
       "0      ATGTTCCCTTTCTACAGCTGCTGGAGGACTGGACTGCTACTACTAC...  ENSG00000003056   \n",
       "1      ATGACAGCCGAGGAGATGAAGGCGACCGAGAGCGGGGCGCAGTCGG...  ENSG00000004478   \n",
       "2      ATGCTCTTTGAGGGCTTGGATCTGGTGTCGGCGCTGGCCACCCTCG...  ENSG00000003137   \n",
       "3      ATGCGGCCCCAGGAGCTCCCCAGGCTCGCGTTCCCGTTGCTGCTGT...  ENSG00000001036   \n",
       "4      ATGGCCGCGCTGCTCCTGGGCGCGGTGCTGCTGGTGGCCCAGCCCC...  ENSG00000002587   \n",
       "...                                                  ...              ...   \n",
       "12516  ATGACCGGGTCAGCGGAAATTTCAGAGCCAGCCAGCAGCTCAGGTC...      XLOC_001676   \n",
       "12517                           CTGTGTAAGTGCATCAGAACATAA      XLOC_001809   \n",
       "12518               ATGGCTTTACCCATGGAGATTTTTGTTGCTGGCTAA      XLOC_001810   \n",
       "12519  ATGTGCTTAATAAATATACGCCGAATGACTGAACTGCAACCAAGGA...      XLOC_001758   \n",
       "12520         TTGTACGTTGGTGGAGACGTGCCTTTTTTCCTTTGCACTTAA      XLOC_001725   \n",
       "\n",
       "         gene_name    transcript_id       gene_type    orfType  length  \\\n",
       "0             M6PR  ENST00000000412  protein_coding  canonical     834   \n",
       "1            FKBP4  ENST00000001008  protein_coding  canonical    1380   \n",
       "2          CYP26B1  ENST00000001146  protein_coding  canonical    1539   \n",
       "3            FUCA2  ENST00000002165  protein_coding  canonical    1404   \n",
       "4           HS3ST1  ENST00000002596  protein_coding  canonical     924   \n",
       "...            ...              ...             ...        ...     ...   \n",
       "12516  XLOC_001676   TCONS_00001992           novel  noncoding     522   \n",
       "12517  XLOC_001809   TCONS_00002050           novel  noncoding      24   \n",
       "12518  XLOC_001810   TCONS_00002051           novel  noncoding      36   \n",
       "12519  XLOC_001758   TCONS_00002066           novel  noncoding      78   \n",
       "12520  XLOC_001725   TCONS_00002105           novel  noncoding      42   \n",
       "\n",
       "       length_aa start_codon  \\\n",
       "0            278         ATG   \n",
       "1            460         ATG   \n",
       "2            513         ATG   \n",
       "3            468         ATG   \n",
       "4            308         ATG   \n",
       "...          ...         ...   \n",
       "12516        174         ATG   \n",
       "12517          8         CTG   \n",
       "12518         12         ATG   \n",
       "12519         26         ATG   \n",
       "12520         14         TTG   \n",
       "\n",
       "                                                  ORFpep  \n",
       "0      MFPFYSCWRTGLLLLLLAVAVRESWQTEEKTCDLVGEKGKESEKEL...  \n",
       "1      MTAEEMKATESGAQSAPLPMEGVDISPKQDEGVLKVIKREGTGTEM...  \n",
       "2      MLFEGLDLVSALATLAACLVSVTLLLAVSQQLWQLRWAATRDKSCK...  \n",
       "3      MRPQELPRLAFPLLLLLLLLLPPPPCPAHSATRFDPTWESLDARQL...  \n",
       "4      MAALLLGAVLLVAQPQLVPSRPAELGQQELLRKAGTLQDDVRDGVA...  \n",
       "...                                                  ...  \n",
       "12516  MTGSAEISEPASSSGPRGRGRNGTAPRRQPRGHSRATADSGTSLSA...  \n",
       "12517                                           MCKCIRT*  \n",
       "12518                                       MALPMEIFVAG*  \n",
       "12519                         MCLINIRRMTELQPRTLYTCFTDLV*  \n",
       "12520                                     MYVGGDVPFFLCT*  \n",
       "\n",
       "[12521 rows x 11 columns]"
      ]
     },
     "execution_count": 2,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ribORF_in_2 = pd.read_csv(\"/projects_eg/projects/marta/TestisRestricted_Microproteins_TSA/with_TranscriptomeReconstruction/ribORF_humanTestis_in2.csv\")\n",
    "ribORF_in_2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th>orfID</th>\n",
       "      <th>ORFseq</th>\n",
       "      <th>gene_id</th>\n",
       "      <th>gene_name</th>\n",
       "      <th>transcript_id</th>\n",
       "      <th>orfType</th>\n",
       "      <th>length</th>\n",
       "      <th>length_aa</th>\n",
       "      <th>start_codon</th>\n",
       "      <th>ORFpep</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>gene_type</th>\n",
       "      <th>mouse</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th rowspan=\"2\" valign=\"top\">lncRNA</th>\n",
       "      <th>conserved_mouse</th>\n",
       "      <td>110</td>\n",
       "      <td>110</td>\n",
       "      <td>110</td>\n",
       "      <td>110</td>\n",
       "      <td>110</td>\n",
       "      <td>110</td>\n",
       "      <td>110</td>\n",
       "      <td>110</td>\n",
       "      <td>110</td>\n",
       "      <td>110</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>no</th>\n",
       "      <td>394</td>\n",
       "      <td>394</td>\n",
       "      <td>394</td>\n",
       "      <td>394</td>\n",
       "      <td>394</td>\n",
       "      <td>394</td>\n",
       "      <td>394</td>\n",
       "      <td>394</td>\n",
       "      <td>394</td>\n",
       "      <td>394</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>novel</th>\n",
       "      <th>no</th>\n",
       "      <td>63</td>\n",
       "      <td>63</td>\n",
       "      <td>63</td>\n",
       "      <td>63</td>\n",
       "      <td>63</td>\n",
       "      <td>63</td>\n",
       "      <td>63</td>\n",
       "      <td>63</td>\n",
       "      <td>63</td>\n",
       "      <td>63</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"2\" valign=\"top\">processed_pseudogene</th>\n",
       "      <th>conserved_mouse</th>\n",
       "      <td>109</td>\n",
       "      <td>109</td>\n",
       "      <td>109</td>\n",
       "      <td>109</td>\n",
       "      <td>109</td>\n",
       "      <td>109</td>\n",
       "      <td>109</td>\n",
       "      <td>109</td>\n",
       "      <td>109</td>\n",
       "      <td>109</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>no</th>\n",
       "      <td>14</td>\n",
       "      <td>14</td>\n",
       "      <td>14</td>\n",
       "      <td>14</td>\n",
       "      <td>14</td>\n",
       "      <td>14</td>\n",
       "      <td>14</td>\n",
       "      <td>14</td>\n",
       "      <td>14</td>\n",
       "      <td>14</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"2\" valign=\"top\">protein_coding</th>\n",
       "      <th>conserved_mouse</th>\n",
       "      <td>11533</td>\n",
       "      <td>11533</td>\n",
       "      <td>11533</td>\n",
       "      <td>11533</td>\n",
       "      <td>11533</td>\n",
       "      <td>11533</td>\n",
       "      <td>11533</td>\n",
       "      <td>11533</td>\n",
       "      <td>11533</td>\n",
       "      <td>11533</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>no</th>\n",
       "      <td>298</td>\n",
       "      <td>298</td>\n",
       "      <td>298</td>\n",
       "      <td>298</td>\n",
       "      <td>298</td>\n",
       "      <td>298</td>\n",
       "      <td>298</td>\n",
       "      <td>298</td>\n",
       "      <td>298</td>\n",
       "      <td>298</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                      orfID  ORFseq  gene_id  gene_name  \\\n",
       "gene_type            mouse                                                \n",
       "lncRNA               conserved_mouse    110     110      110        110   \n",
       "                     no                 394     394      394        394   \n",
       "novel                no                  63      63       63         63   \n",
       "processed_pseudogene conserved_mouse    109     109      109        109   \n",
       "                     no                  14      14       14         14   \n",
       "protein_coding       conserved_mouse  11533   11533    11533      11533   \n",
       "                     no                 298     298      298        298   \n",
       "\n",
       "                                      transcript_id  orfType  length  \\\n",
       "gene_type            mouse                                             \n",
       "lncRNA               conserved_mouse            110      110     110   \n",
       "                     no                         394      394     394   \n",
       "novel                no                          63       63      63   \n",
       "processed_pseudogene conserved_mouse            109      109     109   \n",
       "                     no                          14       14      14   \n",
       "protein_coding       conserved_mouse          11533    11533   11533   \n",
       "                     no                         298      298     298   \n",
       "\n",
       "                                      length_aa  start_codon  ORFpep  \n",
       "gene_type            mouse                                            \n",
       "lncRNA               conserved_mouse        110          110     110  \n",
       "                     no                     394          394     394  \n",
       "novel                no                      63           63      63  \n",
       "processed_pseudogene conserved_mouse        109          109     109  \n",
       "                     no                      14           14      14  \n",
       "protein_coding       conserved_mouse      11533        11533   11533  \n",
       "                     no                     298          298     298  "
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "## Analysis in Mouse\n",
    "human_mouse = pd.read_csv(os.path.join(evoDir,\"TestisExpressedHuman_vs_mouse.txt\"), sep=\"\\t\", header=None)\n",
    "human_mouse.columns = [\"qseqid\",\"sseqid\",\"pident\",\"length\",\"mismatch\",\"gapopen\",\"qstart\",\"qend\",\"sstart\",\"send\",\"evalue\",\"bitscore\",\"qlen\",\"slen\",\"qframe\",\"sframe\",\"sstrand\",\"qcovs\"]\n",
    "\n",
    "ribORF_in_2['mouse'] = np.where(ribORF_in_2['orfID'].isin(human_mouse.qseqid.values.tolist()), \"conserved_mouse\",\"no\")\n",
    "ribORF_in_2.groupby(['gene_type','mouse']).count()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th>orfID</th>\n",
       "      <th>ORFseq</th>\n",
       "      <th>gene_id</th>\n",
       "      <th>gene_name</th>\n",
       "      <th>transcript_id</th>\n",
       "      <th>orfType</th>\n",
       "      <th>length</th>\n",
       "      <th>length_aa</th>\n",
       "      <th>start_codon</th>\n",
       "      <th>ORFpep</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>gene_type</th>\n",
       "      <th>macaca</th>\n",
       "      <th>mouse</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th rowspan=\"4\" valign=\"top\">lncRNA</th>\n",
       "      <th rowspan=\"2\" valign=\"top\">conserved_macaca</th>\n",
       "      <th>conserved_mouse</th>\n",
       "      <td>109</td>\n",
       "      <td>109</td>\n",
       "      <td>109</td>\n",
       "      <td>109</td>\n",
       "      <td>109</td>\n",
       "      <td>109</td>\n",
       "      <td>109</td>\n",
       "      <td>109</td>\n",
       "      <td>109</td>\n",
       "      <td>109</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>no</th>\n",
       "      <td>62</td>\n",
       "      <td>62</td>\n",
       "      <td>62</td>\n",
       "      <td>62</td>\n",
       "      <td>62</td>\n",
       "      <td>62</td>\n",
       "      <td>62</td>\n",
       "      <td>62</td>\n",
       "      <td>62</td>\n",
       "      <td>62</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"2\" valign=\"top\">no</th>\n",
       "      <th>conserved_mouse</th>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>no</th>\n",
       "      <td>332</td>\n",
       "      <td>332</td>\n",
       "      <td>332</td>\n",
       "      <td>332</td>\n",
       "      <td>332</td>\n",
       "      <td>332</td>\n",
       "      <td>332</td>\n",
       "      <td>332</td>\n",
       "      <td>332</td>\n",
       "      <td>332</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"2\" valign=\"top\">novel</th>\n",
       "      <th>conserved_macaca</th>\n",
       "      <th>no</th>\n",
       "      <td>8</td>\n",
       "      <td>8</td>\n",
       "      <td>8</td>\n",
       "      <td>8</td>\n",
       "      <td>8</td>\n",
       "      <td>8</td>\n",
       "      <td>8</td>\n",
       "      <td>8</td>\n",
       "      <td>8</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>no</th>\n",
       "      <th>no</th>\n",
       "      <td>55</td>\n",
       "      <td>55</td>\n",
       "      <td>55</td>\n",
       "      <td>55</td>\n",
       "      <td>55</td>\n",
       "      <td>55</td>\n",
       "      <td>55</td>\n",
       "      <td>55</td>\n",
       "      <td>55</td>\n",
       "      <td>55</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"4\" valign=\"top\">processed_pseudogene</th>\n",
       "      <th rowspan=\"2\" valign=\"top\">conserved_macaca</th>\n",
       "      <th>conserved_mouse</th>\n",
       "      <td>107</td>\n",
       "      <td>107</td>\n",
       "      <td>107</td>\n",
       "      <td>107</td>\n",
       "      <td>107</td>\n",
       "      <td>107</td>\n",
       "      <td>107</td>\n",
       "      <td>107</td>\n",
       "      <td>107</td>\n",
       "      <td>107</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>no</th>\n",
       "      <td>7</td>\n",
       "      <td>7</td>\n",
       "      <td>7</td>\n",
       "      <td>7</td>\n",
       "      <td>7</td>\n",
       "      <td>7</td>\n",
       "      <td>7</td>\n",
       "      <td>7</td>\n",
       "      <td>7</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"2\" valign=\"top\">no</th>\n",
       "      <th>conserved_mouse</th>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>no</th>\n",
       "      <td>7</td>\n",
       "      <td>7</td>\n",
       "      <td>7</td>\n",
       "      <td>7</td>\n",
       "      <td>7</td>\n",
       "      <td>7</td>\n",
       "      <td>7</td>\n",
       "      <td>7</td>\n",
       "      <td>7</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"4\" valign=\"top\">protein_coding</th>\n",
       "      <th rowspan=\"2\" valign=\"top\">conserved_macaca</th>\n",
       "      <th>conserved_mouse</th>\n",
       "      <td>11374</td>\n",
       "      <td>11374</td>\n",
       "      <td>11374</td>\n",
       "      <td>11374</td>\n",
       "      <td>11374</td>\n",
       "      <td>11374</td>\n",
       "      <td>11374</td>\n",
       "      <td>11374</td>\n",
       "      <td>11374</td>\n",
       "      <td>11374</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>no</th>\n",
       "      <td>269</td>\n",
       "      <td>269</td>\n",
       "      <td>269</td>\n",
       "      <td>269</td>\n",
       "      <td>269</td>\n",
       "      <td>269</td>\n",
       "      <td>269</td>\n",
       "      <td>269</td>\n",
       "      <td>269</td>\n",
       "      <td>269</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"2\" valign=\"top\">no</th>\n",
       "      <th>conserved_mouse</th>\n",
       "      <td>159</td>\n",
       "      <td>159</td>\n",
       "      <td>159</td>\n",
       "      <td>159</td>\n",
       "      <td>159</td>\n",
       "      <td>159</td>\n",
       "      <td>159</td>\n",
       "      <td>159</td>\n",
       "      <td>159</td>\n",
       "      <td>159</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>no</th>\n",
       "      <td>29</td>\n",
       "      <td>29</td>\n",
       "      <td>29</td>\n",
       "      <td>29</td>\n",
       "      <td>29</td>\n",
       "      <td>29</td>\n",
       "      <td>29</td>\n",
       "      <td>29</td>\n",
       "      <td>29</td>\n",
       "      <td>29</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                       orfID  ORFseq  gene_id  \\\n",
       "gene_type            macaca           mouse                                     \n",
       "lncRNA               conserved_macaca conserved_mouse    109     109      109   \n",
       "                                      no                  62      62       62   \n",
       "                     no               conserved_mouse      1       1        1   \n",
       "                                      no                 332     332      332   \n",
       "novel                conserved_macaca no                   8       8        8   \n",
       "                     no               no                  55      55       55   \n",
       "processed_pseudogene conserved_macaca conserved_mouse    107     107      107   \n",
       "                                      no                   7       7        7   \n",
       "                     no               conserved_mouse      2       2        2   \n",
       "                                      no                   7       7        7   \n",
       "protein_coding       conserved_macaca conserved_mouse  11374   11374    11374   \n",
       "                                      no                 269     269      269   \n",
       "                     no               conserved_mouse    159     159      159   \n",
       "                                      no                  29      29       29   \n",
       "\n",
       "                                                       gene_name  \\\n",
       "gene_type            macaca           mouse                        \n",
       "lncRNA               conserved_macaca conserved_mouse        109   \n",
       "                                      no                      62   \n",
       "                     no               conserved_mouse          1   \n",
       "                                      no                     332   \n",
       "novel                conserved_macaca no                       8   \n",
       "                     no               no                      55   \n",
       "processed_pseudogene conserved_macaca conserved_mouse        107   \n",
       "                                      no                       7   \n",
       "                     no               conserved_mouse          2   \n",
       "                                      no                       7   \n",
       "protein_coding       conserved_macaca conserved_mouse      11374   \n",
       "                                      no                     269   \n",
       "                     no               conserved_mouse        159   \n",
       "                                      no                      29   \n",
       "\n",
       "                                                       transcript_id  orfType  \\\n",
       "gene_type            macaca           mouse                                     \n",
       "lncRNA               conserved_macaca conserved_mouse            109      109   \n",
       "                                      no                          62       62   \n",
       "                     no               conserved_mouse              1        1   \n",
       "                                      no                         332      332   \n",
       "novel                conserved_macaca no                           8        8   \n",
       "                     no               no                          55       55   \n",
       "processed_pseudogene conserved_macaca conserved_mouse            107      107   \n",
       "                                      no                           7        7   \n",
       "                     no               conserved_mouse              2        2   \n",
       "                                      no                           7        7   \n",
       "protein_coding       conserved_macaca conserved_mouse          11374    11374   \n",
       "                                      no                         269      269   \n",
       "                     no               conserved_mouse            159      159   \n",
       "                                      no                          29       29   \n",
       "\n",
       "                                                       length  length_aa  \\\n",
       "gene_type            macaca           mouse                                \n",
       "lncRNA               conserved_macaca conserved_mouse     109        109   \n",
       "                                      no                   62         62   \n",
       "                     no               conserved_mouse       1          1   \n",
       "                                      no                  332        332   \n",
       "novel                conserved_macaca no                    8          8   \n",
       "                     no               no                   55         55   \n",
       "processed_pseudogene conserved_macaca conserved_mouse     107        107   \n",
       "                                      no                    7          7   \n",
       "                     no               conserved_mouse       2          2   \n",
       "                                      no                    7          7   \n",
       "protein_coding       conserved_macaca conserved_mouse   11374      11374   \n",
       "                                      no                  269        269   \n",
       "                     no               conserved_mouse     159        159   \n",
       "                                      no                   29         29   \n",
       "\n",
       "                                                       start_codon  ORFpep  \n",
       "gene_type            macaca           mouse                                 \n",
       "lncRNA               conserved_macaca conserved_mouse          109     109  \n",
       "                                      no                        62      62  \n",
       "                     no               conserved_mouse            1       1  \n",
       "                                      no                       332     332  \n",
       "novel                conserved_macaca no                         8       8  \n",
       "                     no               no                        55      55  \n",
       "processed_pseudogene conserved_macaca conserved_mouse          107     107  \n",
       "                                      no                         7       7  \n",
       "                     no               conserved_mouse            2       2  \n",
       "                                      no                         7       7  \n",
       "protein_coding       conserved_macaca conserved_mouse        11374   11374  \n",
       "                                      no                       269     269  \n",
       "                     no               conserved_mouse          159     159  \n",
       "                                      no                        29      29  "
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "## Analysis in Macaca\n",
    "human_macaca = pd.read_csv(os.path.join(evoDir,\"TestisExpressedHuman_vs_macaca.txt\"), sep=\"\\t\", header=None)\n",
    "human_macaca.columns = [\"qseqid\",\"sseqid\",\"pident\",\"length\",\"mismatch\",\"gapopen\",\"qstart\",\"qend\",\"sstart\",\"send\",\"evalue\",\"bitscore\",\"qlen\",\"slen\",\"qframe\",\"sframe\",\"sstrand\",\"qcovs\"]\n",
    "\n",
    "ribORF_in_2['macaca'] = np.where(ribORF_in_2['orfID'].isin(human_macaca.qseqid.values.tolist()), \"conserved_macaca\",\"no\")\n",
    "# ribORF_in_2_macaca.drop(columns=['mouse'], inplace=True)\n",
    "ribORF_in_2.to_csv(os.path.join(evoDir,\"ribORF_humanTestis_in2.evolution.csv\"), index=None)\n",
    "ribORF_in_2.groupby(['gene_type','macaca','mouse']).count()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.16"
  },
  "orig_nbformat": 4
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
