{
 "cells": [
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## PHYLOSTRATIGRAPHY"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "28\n"
     ]
    }
   ],
   "source": [
    "import os,re,glob\n",
    "import pandas as pd\n",
    "import numpy as np\n",
    "from collections import Counter\n",
    "from Bio.Seq import Seq\n",
    "from Bio.SeqRecord import SeqRecord\n",
    "from Bio import SeqIO\n",
    "\n",
    "users_dir = \"/users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction\"\n",
    "GENOMEDIR = \"/genomics/users/marta/genomes\"\n",
    "\n",
    "## annotation file\n",
    "annotation=\"/users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/human/newReference_Resconstructed/gencode.v38.gffcompare.TestisLiverBrain.annotation.sorted.1transcript.sorted.NOchr.gtf\"\n",
    "transcript_gene=pd.read_csv(\"/users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/human/newReference_Resconstructed/1transcript_1gene.reconstructed.csv\")\n",
    "\n",
    "## evoDir\n",
    "evoDir=os.path.join(users_dir,\"EvolutionaryOrigin\")\n",
    "phyloDir = os.path.join(users_dir,\"Phylostratigraphy\")\n",
    "\n",
    "lineage=[\"cellular organisms\", \"Eukaryota\", \"Opisthokonta\", \"Metazoa\", \"Eumetazoa\",\n",
    "    \"Bilateria\", \"Deuterostomia\", \"Chordata\", \"Vertebrata\",\n",
    "    \"Gnathostomata\", \"Teleostomi\", \"Euteleostomi\", \"Sarcopterygii\",\n",
    "    \"Dipnotetrapodomorpha\", \"Tetrapoda\", \"Amniota\", \"Mammalia\", \"Theria\",\n",
    "    \"Eutheria\", \"Boreoeutheria\", \"Euarchontoglires\", \"Primates\",\n",
    "    \"Simiiformes\", \"Catarrhini\", \"Hominoidea\", \"Hominidae\", \"Homininae\", \"Homo sapiens\"]\n",
    "print(len(lineage))"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Testis"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "11827\n"
     ]
    }
   ],
   "source": [
    "ribORF_in_2 = pd.read_csv(\"/projects_eg/projects/marta/TestisRestricted_Microproteins_TSA/with_TranscriptomeReconstruction/ribORF_humanTestis_in2.csv\")\n",
    "\n",
    "ribORF_in_2_noncanonical = ribORF_in_2[ribORF_in_2['gene_type'] != \"protein_coding\"]\n",
    "ribORF_in_2_noncanonical.to_csv(os.path.join(evoDir,\"ribORF_humanTestis_in2.noncanonical.csv\"), index=None)\n",
    "\n",
    "ribORF_in_2_canonical = ribORF_in_2[ribORF_in_2['gene_type'] == \"protein_coding\"]\n",
    "print(len(set(ribORF_in_2_canonical.gene_name.values.tolist())))\n",
    "ribORF_in_2_canonical.to_csv(os.path.join(evoDir,\"ribORF_humanTestis_in2.canonical.csv\"), index=None)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {},
   "outputs": [],
   "source": [
    "input_file = \"/projects_eg/projects/marta/TestisRestricted_Microproteins_TSA/with_TranscriptomeReconstruction/ribORF_humanTestis_in2.fa\"\n",
    "output_file_canonical = \"/projects_eg/projects/marta/TestisRestricted_Microproteins_TSA/with_TranscriptomeReconstruction/ribORF_humanTestis_in2.canonical.fa\"\n",
    "output_file_noncanonical = \"/projects_eg/projects/marta/TestisRestricted_Microproteins_TSA/with_TranscriptomeReconstruction/ribORF_humanTestis_in2.noncanonical.fa\"\n",
    "\n",
    "fasta_sequences = SeqIO.parse(open(input_file),'fasta')\n",
    "\n",
    "with open(output_file_canonical, \"w\") as out_c:\n",
    "    with open(output_file_noncanonical, \"w\") as out_nc:\n",
    "        for fasta in fasta_sequences:\n",
    "            if \"noncoding\" in fasta.id:\n",
    "                out_nc.write(\">%s\\n%s\\n\" %(fasta.id, str(fasta.seq)))\n",
    "            else:\n",
    "                out_c.write(\">%s\\n%s\\n\" %(fasta.id, str(fasta.seq)))\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "14341\n",
      "11511\n",
      "Missing PCGs in DENSE\n",
      "316\n"
     ]
    }
   ],
   "source": [
    "DENSE=pd.read_csv(\"/projects_eg/projects/marta/DENSE_AnneLopes/denovotable.csv\")\n",
    "DENSE_candidates = DENSE[DENSE['Gene'].isin(ribORF_in_2_canonical.gene_name.values.tolist())]\n",
    "DENSE_candidates = DENSE_candidates[['Gene','Phylostratum']].drop_duplicates()\n",
    "print(len(DENSE_candidates))\n",
    "print(len(set(DENSE_candidates.Gene.values.tolist())))\n",
    "DENSE_candidates.Phylostratum = DENSE_candidates.Phylostratum.astype(\"category\")\n",
    "DENSE_candidates.Phylostratum = DENSE_candidates.Phylostratum.cat.set_categories(lineage)\n",
    "  ## 'sort' changed to 'sort_values'\n",
    "DENSE_candidates.sort_values([\"Phylostratum\"]).to_csv(os.path.join(phyloDir, \"DENSE_PCGs.csv\"), index=None)\n",
    "\n",
    "\n",
    "missing_PCGs_in_DENSE = ribORF_in_2_canonical[~ribORF_in_2_canonical['gene_name'].isin(DENSE.Gene.values.tolist())]\n",
    "print(\"Missing PCGs in DENSE\")\n",
    "print(len(set(missing_PCGs_in_DENSE.gene_name.values.tolist())))\n",
    "missing_PCGs_in_DENSE.to_csv(os.path.join(phyloDir, \"DENSE_missingPCGs.csv\"), index=None)\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "14341\n"
     ]
    }
   ],
   "source": [
    "print(len(DENSE_candidates))\n",
    "summary = DENSE_candidates.groupby(\"Phylostratum\").count().reset_index()\n",
    "summary.columns = ['Phylostratum','n_PCGs']\n",
    "summary = summary[summary['Phylostratum'] != \"Unknown\"]\n",
    "summary = summary[summary[\"Phylostratum\"] != \"Possible contamination or HGT\"]\n",
    "\n",
    "# Sorting the DataFrame\n",
    "summary.Phylostratum = summary.Phylostratum.astype(\"category\")\n",
    "summary.Phylostratum = summary.Phylostratum.cat.set_categories(lineage)\n",
    "summary.sort_values([\"Phylostratum\"])  ## 'sort' changed to 'sort_values'\n",
    "summary.to_csv(os.path.join(phyloDir, \"summary_DENSE_PCGs.csv\"), index=None)"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## GenEra \n",
    "\n",
    "`https://github.com/josuebarrera/GenEra/wiki/Running-GenEra`\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Process is interrupted.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "# >>>>>>>>>>>>>>>>>>>>>> ERROR REPORT <<<<<<<<<<<<<<<<<<<<<<\n",
      "\n",
      "    Traceback (most recent call last):\n",
      "      File \"/soft/system/software/Miniconda3/24.5.0/lib/python3.12/site-packages/conda/exception_handler.py\", line 18, in __call__\n",
      "        return func(*args, **kwargs)\n",
      "               ^^^^^^^^^^^^^^^^^^^^^\n",
      "      File \"/soft/system/software/Miniconda3/24.5.0/lib/python3.12/site-packages/conda/cli/main.py\", line 110, in main_sourced\n",
      "        print(activator.execute(), end=\"\")\n",
      "    BrokenPipeError: [Errno 32] Broken pipe\n",
      "\n",
      "`$ /soft/system/software/Miniconda3/24.5.0/bin/conda shell.bash hook`\n",
      "\n",
      "  environment variables:\n",
      "                 CIO_TEST=<not set>\n",
      "               CONDA_ROOT=/soft/system/software/Miniconda3/24.5.0\n",
      "           CURL_CA_BUNDLE=<not set>\n",
      "               LD_PRELOAD=<not set>\n",
      "  LMOD_DEFAULT_MODULEPATH=/soft/system/modules/all:/etc/lmod/modules:/usr/share/lmod/lmod/module\n",
      "                          files\n",
      "                  MANPATH=/soft/system/software/Miniconda3/24.5.0/share/man:/usr/share/lmod/lmod\n",
      "                          /share/man::\n",
      "               MODULEPATH=/soft/system/modules/all:/etc/lmod/modules:/usr/share/lmod/lmod/module\n",
      "                          files\n",
      "                     PATH=/soft/system/software/Miniconda3/24.5.0:/soft/system/software/Minicond\n",
      "                          a3/24.5.0/bin:/soft/system/software/Miniconda3/4.9.2/bin:/genomics/use\n",
      "                          rs/marta/tools/hisat-genotype:/genomics/users/marta/tools/hisat-genoty\n",
      "                          pe/hisat2:/home/marta/.local/bin:/genomics/users/marta/tools/hisat-\n",
      "                          genotype:/genomics/users/marta/tools/hisat-genotype/hisat2:/usr/local/\n",
      "                          sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/loca\n",
      "                          l/games:/snap/bin:/home/marta/.dotnet/tools:/opt/bin:/home/marta/.dotn\n",
      "                          et/tools:/opt/bin:/home/marta/.local/bin:/home/marta/bin:/genomics/use\n",
      "                          rs/marta/tools/hisat-genotype:/genomics/users/marta/tools/hisat-genoty\n",
      "                          pe/hisat2:/home/marta/.local/bin:/genomics/users/marta/tools/hisat-\n",
      "                          genotype:/genomics/users/marta/tools/hisat-genotype/hisat2:/usr/local/\n",
      "                          sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/loca\n",
      "                          l/games:/snap/bin:/home/marta/.dotnet/tools:/opt/bin:/home/marta/.dotn\n",
      "                          et/tools:/opt/bin:/home/marta/.local/bin:/home/marta/bin\n",
      "          PKG_CONFIG_PATH=/soft/system/software/Miniconda3/24.5.0/lib/pkgconfig\n",
      "         PYTHONIOENCODING=utf-8\n",
      "               PYTHONPATH=/genomics/users/marta/tools/hisat-\n",
      "                          genotype/hisatgenotype_modules:/genomics/users/marta/tools/hisat-\n",
      "                          genotype/hisatgenotype_modules:/genomics/users/marta/tools/hisat-\n",
      "                          genotype/hisatgenotype_modules:/genomics/users/marta/tools/hisat-\n",
      "                          genotype/hisatgenotype_modules::/genomics/users/marta/tools/hisat-\n",
      "                          genotype/hisatgenotype_modules:/genomics/users/marta/tools/hisat-\n",
      "                          genotype/hisatgenotype_modules:/genomics/users/marta/tools/hisat-\n",
      "                          genotype/hisatgenotype_modules:/genomics/users/marta/tools/hisat-\n",
      "                          genotype/hisatgenotype_modules:\n",
      "         PYTHONUNBUFFERED=1\n",
      "       REQUESTS_CA_BUNDLE=<not set>\n",
      "            SSL_CERT_FILE=<not set>\n",
      "   VSCODE_CODE_CACHE_PATH=/home/marta/.config/Code/CachedData/5e805b79fcb6ba4c2d23712967df89a089\n",
      "                          da575b\n",
      "               WINDOWPATH=2\n",
      "\n",
      "     active environment : None\n",
      "       user config file : /home/marta/.condarc\n",
      " populated config files : \n",
      "          conda version : 24.5.0\n",
      "    conda-build version : not installed\n",
      "         python version : 3.12.4.final.0\n",
      "                 solver : libmamba (default)\n",
      "       virtual packages : __archspec=1=cascadelake\n",
      "                          __conda=24.5.0=0\n",
      "                          __cuda=12.2=0\n",
      "                          __glibc=2.31=0\n",
      "                          __linux=5.14.0=0\n",
      "                          __unix=0=0\n",
      "       base environment : /soft/system/software/Miniconda3/24.5.0  (read only)\n",
      "      conda av data dir : /soft/system/software/Miniconda3/24.5.0/etc/conda\n",
      "  conda av metadata url : None\n",
      "           channel URLs : https://repo.anaconda.com/pkgs/main/linux-64\n",
      "                          https://repo.anaconda.com/pkgs/main/noarch\n",
      "                          https://repo.anaconda.com/pkgs/r/linux-64\n",
      "                          https://repo.anaconda.com/pkgs/r/noarch\n",
      "          package cache : /soft/system/software/Miniconda3/24.5.0/pkgs\n",
      "                          /home/marta/.conda/pkgs\n",
      "       envs directories : /home/marta/.conda/envs\n",
      "                          /soft/system/software/Miniconda3/24.5.0/envs\n",
      "               platform : linux-64\n",
      "             user-agent : conda/24.5.0 requests/2.32.2 CPython/3.12.4 Linux/5.14.0-1056-oem ubuntu/20.04.5 glibc/2.31 solver/libmamba conda-libmamba-solver/24.1.0 libmambapy/1.5.8\n",
      "                UID:GID : 2334:2300\n",
      "             netrc file : None\n",
      "           offline mode : False\n",
      "\n",
      "\n",
      "An unexpected error has occurred. Conda has prepared the above report.\n",
      "If you suspect this error is being caused by a malfunctioning plugin,\n",
      "consider using the --no-plugins option to turn off plugins.\n",
      "\n",
      "Example: conda --no-plugins install <package>\n",
      "\n",
      "Alternatively, you can set the CONDA_NO_PLUGINS environment variable on\n",
      "the command line to run the command without plugins enabled.\n",
      "\n",
      "Example: CONDA_NO_PLUGINS=true conda install <package>\n",
      "\n"
     ]
    }
   ],
   "source": [
    "%%bash -s \"$phyloDir\"\n",
    "\n",
    "OUTDIR=$1/GenEra\n",
    "mkdir -p $OUTDIR\n",
    "mkdir -p $OUTDIR/output\n",
    "\n",
    "module load  Miniconda3/24.5.0\n",
    "eval \"$(/soft/system/software/Miniconda3/24.5.0/bin/conda shell.bash hook)\"\n",
    "conda activate genEra\n",
    "\n",
    "genEra -q /projects_eg/projects/marta/TestisRestricted_Microproteins_TSA/with_TranscriptomeReconstruction/ribORF_humanTestis_in2.noncanonical.fa \\\n",
    "    -t 9606 -b /datasets/diamond_databases/nr/nr -d /datasets/diamond_databases/nr/nr -n 20 -e 1e-4 -r $OUTDIR/ncbi_lineages_2022-07-28.csv\n",
    "\n",
    "# genEra -q /projects_eg/projects/marta/TestisRestricted_Microproteins_TSA/with_TranscriptomeReconstruction/ribORF_humanTestis_in2.noncanonical.fa \\\n",
    "#     -t 9606 -p /home/marta/PROJECT_SCRIPTS/TestisRestricted_MP/tmp_9606_16006/9606_Diamond_results.bout -d /datasets/diamond_databases/nr/nr -n 20 -e 1e-4 -r $OUTDIR/ncbi_lineages_2022-07-28.csv    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "metadata": {},
   "outputs": [],
   "source": [
    "PCGs_ages = pd.read_csv(os.path.join(phyloDir, \"summary_DENSE_PCGs.csv\"))\n",
    "PCGs_ages\n",
    "\n",
    "ncORFs_ages = pd.read_csv(os.path.join(phyloDir, \"GenEra/output/9606_gene_age_summary.tsv\"), sep=\"\\t\")\n",
    "ncORFs_ages = ncORFs_ages[[\"#number_of_genes\",\"phylostratum\"]]\n",
    "ncORFs_ages.columns = ['n_ncORFs','Phylostratum']\n",
    "\n",
    "ages = PCGs_ages.merge(ncORFs_ages, on=\"Phylostratum\")\n",
    "ages.Phylostratum = ages.Phylostratum.astype(\"category\")\n",
    "ages.Phylostratum = ages.Phylostratum.cat.set_categories(lineage)\n",
    "ages.sort_values([\"Phylostratum\"], inplace=True)  ## 'sort' changed to 'sort_values'\n",
    "ages.drop_duplicates(inplace=True)\n",
    "ages.to_csv(os.path.join(phyloDir,\"summary_all_genes_ages.csv\"), index=None)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "metadata": {},
   "outputs": [],
   "source": [
    "ncORFs_genes_ages = pd.read_csv(os.path.join(phyloDir, \"GenEra/output/9606_gene_ages.tsv\"), sep=\"\\t\", header=None)\n",
    "ncORFs_genes_ages['transcript_id'] = ncORFs_genes_ages[0].str.split(\":\", expand=True)[0]\n",
    "ncORFs_genes_ages['transcript_id'] = ncORFs_genes_ages['transcript_id'].str.split(\".\", expand=True)[0]\n",
    "ncORFs_genes_ages = ncORFs_genes_ages.merge(transcript_gene, on=\"transcript_id\")\n",
    "ncORFs_genes_ages = ncORFs_genes_ages[['gene_name','gene_type',1,0]]\n",
    "ncORFs_genes_ages.columns=['gene_name','gene_type','Phylostratum','orfID']\n",
    "ncORFs_genes_ages\n",
    "\n",
    "\n",
    "\n",
    "PCGs_gene_ages = pd.read_csv(os.path.join(phyloDir, \"DENSE_PCGs.csv\"))\n",
    "PCGs_gene_ages.columns=['gene_name','Phylostratum']\n",
    "PCGs_gene_ages['gene_type'] = \"protein_coding\"\n",
    "PCGs_gene_ages['orfID'] = PCGs_gene_ages['gene_name'] + \"_ORF1\"\n",
    "\n",
    "\n",
    "all_genes = pd.concat([PCGs_gene_ages, ncORFs_genes_ages])\n",
    "\n",
    "all_genes.Phylostratum = all_genes.Phylostratum.astype(\"category\")\n",
    "all_genes.Phylostratum = all_genes.Phylostratum.cat.set_categories(lineage)\n",
    "all_genes.sort_values([\"Phylostratum\"], inplace=True)  ## 'sort' changed to 'sort_values'\n",
    "all_genes.drop_duplicates(inplace=True)\n",
    "all_genes.to_csv(os.path.join(phyloDir,\"all_genes_phyostratum.csv\"), index=None)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene_name</th>\n",
       "      <th>Phylostratum</th>\n",
       "      <th>gene_type</th>\n",
       "      <th>orfID</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>A2M</td>\n",
       "      <td>cellular organisms</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>A2M_ORF1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5846</th>\n",
       "      <td>ZNF814</td>\n",
       "      <td>cellular organisms</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>ZNF814_ORF1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5845</th>\n",
       "      <td>NEDD4L</td>\n",
       "      <td>cellular organisms</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>NEDD4L_ORF1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5844</th>\n",
       "      <td>PEX5</td>\n",
       "      <td>cellular organisms</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>PEX5_ORF1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5843</th>\n",
       "      <td>QDPR</td>\n",
       "      <td>cellular organisms</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>QDPR_ORF1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1577</th>\n",
       "      <td>XLOC_000607</td>\n",
       "      <td>NaN</td>\n",
       "      <td>novel</td>\n",
       "      <td>TCONS_00000707:15:+|540|22729:8123:8150|noncod...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1582</th>\n",
       "      <td>XLOC_000828</td>\n",
       "      <td>NaN</td>\n",
       "      <td>novel</td>\n",
       "      <td>TCONS_00000993:18:-|64|1055:791:806|noncoding|GTG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1588</th>\n",
       "      <td>XLOC_000954</td>\n",
       "      <td>NaN</td>\n",
       "      <td>novel</td>\n",
       "      <td>TCONS_00001162:2:+|62|9974:848:863|noncoding|TTG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1596</th>\n",
       "      <td>XLOC_001498</td>\n",
       "      <td>NaN</td>\n",
       "      <td>novel</td>\n",
       "      <td>TCONS_00001784:6:-|13|690:130:331|noncoding|ATG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1598</th>\n",
       "      <td>XLOC_001809</td>\n",
       "      <td>NaN</td>\n",
       "      <td>novel</td>\n",
       "      <td>TCONS_00002050:9:-|86|10655:1396:1420|noncodin...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>15940 rows × 4 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        gene_name        Phylostratum       gene_type  \\\n",
       "0             A2M  cellular organisms  protein_coding   \n",
       "5846       ZNF814  cellular organisms  protein_coding   \n",
       "5845       NEDD4L  cellular organisms  protein_coding   \n",
       "5844         PEX5  cellular organisms  protein_coding   \n",
       "5843         QDPR  cellular organisms  protein_coding   \n",
       "...           ...                 ...             ...   \n",
       "1577  XLOC_000607                 NaN           novel   \n",
       "1582  XLOC_000828                 NaN           novel   \n",
       "1588  XLOC_000954                 NaN           novel   \n",
       "1596  XLOC_001498                 NaN           novel   \n",
       "1598  XLOC_001809                 NaN           novel   \n",
       "\n",
       "                                                  orfID  \n",
       "0                                              A2M_ORF1  \n",
       "5846                                        ZNF814_ORF1  \n",
       "5845                                        NEDD4L_ORF1  \n",
       "5844                                          PEX5_ORF1  \n",
       "5843                                          QDPR_ORF1  \n",
       "...                                                 ...  \n",
       "1577  TCONS_00000707:15:+|540|22729:8123:8150|noncod...  \n",
       "1582  TCONS_00000993:18:-|64|1055:791:806|noncoding|GTG  \n",
       "1588   TCONS_00001162:2:+|62|9974:848:863|noncoding|TTG  \n",
       "1596    TCONS_00001784:6:-|13|690:130:331|noncoding|ATG  \n",
       "1598  TCONS_00002050:9:-|86|10655:1396:1420|noncodin...  \n",
       "\n",
       "[15940 rows x 4 columns]"
      ]
     },
     "execution_count": 40,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "all_genes"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Phylostratigraphy Tumor-expressed"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene_name</th>\n",
       "      <th>Phylostratum</th>\n",
       "      <th>gene_type</th>\n",
       "      <th>orfID</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>A2M</td>\n",
       "      <td>cellular organisms</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>A2M_ORF1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ZNF814</td>\n",
       "      <td>cellular organisms</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>ZNF814_ORF1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>NEDD4L</td>\n",
       "      <td>cellular organisms</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>NEDD4L_ORF1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>PEX5</td>\n",
       "      <td>cellular organisms</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>PEX5_ORF1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>QDPR</td>\n",
       "      <td>cellular organisms</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>QDPR_ORF1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15935</th>\n",
       "      <td>XLOC_000607</td>\n",
       "      <td>NaN</td>\n",
       "      <td>novel</td>\n",
       "      <td>TCONS_00000707:15:+|540|22729:8123:8150|noncod...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15936</th>\n",
       "      <td>XLOC_000828</td>\n",
       "      <td>NaN</td>\n",
       "      <td>novel</td>\n",
       "      <td>TCONS_00000993:18:-|64|1055:791:806|noncoding|GTG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15937</th>\n",
       "      <td>XLOC_000954</td>\n",
       "      <td>NaN</td>\n",
       "      <td>novel</td>\n",
       "      <td>TCONS_00001162:2:+|62|9974:848:863|noncoding|TTG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15938</th>\n",
       "      <td>XLOC_001498</td>\n",
       "      <td>NaN</td>\n",
       "      <td>novel</td>\n",
       "      <td>TCONS_00001784:6:-|13|690:130:331|noncoding|ATG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15939</th>\n",
       "      <td>XLOC_001809</td>\n",
       "      <td>NaN</td>\n",
       "      <td>novel</td>\n",
       "      <td>TCONS_00002050:9:-|86|10655:1396:1420|noncodin...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>15940 rows × 4 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "         gene_name        Phylostratum       gene_type  \\\n",
       "0              A2M  cellular organisms  protein_coding   \n",
       "1           ZNF814  cellular organisms  protein_coding   \n",
       "2           NEDD4L  cellular organisms  protein_coding   \n",
       "3             PEX5  cellular organisms  protein_coding   \n",
       "4             QDPR  cellular organisms  protein_coding   \n",
       "...            ...                 ...             ...   \n",
       "15935  XLOC_000607                 NaN           novel   \n",
       "15936  XLOC_000828                 NaN           novel   \n",
       "15937  XLOC_000954                 NaN           novel   \n",
       "15938  XLOC_001498                 NaN           novel   \n",
       "15939  XLOC_001809                 NaN           novel   \n",
       "\n",
       "                                                   orfID  \n",
       "0                                               A2M_ORF1  \n",
       "1                                            ZNF814_ORF1  \n",
       "2                                            NEDD4L_ORF1  \n",
       "3                                              PEX5_ORF1  \n",
       "4                                              QDPR_ORF1  \n",
       "...                                                  ...  \n",
       "15935  TCONS_00000707:15:+|540|22729:8123:8150|noncod...  \n",
       "15936  TCONS_00000993:18:-|64|1055:791:806|noncoding|GTG  \n",
       "15937   TCONS_00001162:2:+|62|9974:848:863|noncoding|TTG  \n",
       "15938    TCONS_00001784:6:-|13|690:130:331|noncoding|ATG  \n",
       "15939  TCONS_00002050:9:-|86|10655:1396:1420|noncodin...  \n",
       "\n",
       "[15940 rows x 4 columns]"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "all_genes = pd.read_csv(os.path.join(phyloDir,\"all_genes_phyostratum.csv\"))\n",
    "all_genes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>transcript_id</th>\n",
       "      <th>gene_id</th>\n",
       "      <th>gene_name</th>\n",
       "      <th>gene_type</th>\n",
       "      <th>Length</th>\n",
       "      <th>n</th>\n",
       "      <th>ctype</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ENST00000000412</td>\n",
       "      <td>ENSG00000003056</td>\n",
       "      <td>M6PR</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>2450</td>\n",
       "      <td>131</td>\n",
       "      <td>BRCA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ENST00000001008</td>\n",
       "      <td>ENSG00000004478</td>\n",
       "      <td>FKBP4</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>3715</td>\n",
       "      <td>131</td>\n",
       "      <td>BRCA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ENST00000001146</td>\n",
       "      <td>ENSG00000003137</td>\n",
       "      <td>CYP26B1</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>4556</td>\n",
       "      <td>83</td>\n",
       "      <td>BRCA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ENST00000002165</td>\n",
       "      <td>ENSG00000001036</td>\n",
       "      <td>FUCA2</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>2385</td>\n",
       "      <td>131</td>\n",
       "      <td>BRCA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ENST00000002596</td>\n",
       "      <td>ENSG00000002587</td>\n",
       "      <td>HS3ST1</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>7160</td>\n",
       "      <td>63</td>\n",
       "      <td>BRCA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>191011</th>\n",
       "      <td>TCONS_00002143</td>\n",
       "      <td>XLOC_001856</td>\n",
       "      <td>XLOC_001856</td>\n",
       "      <td>novel</td>\n",
       "      <td>244</td>\n",
       "      <td>140</td>\n",
       "      <td>COAD</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>191012</th>\n",
       "      <td>TCONS_00002144</td>\n",
       "      <td>XLOC_001848</td>\n",
       "      <td>XLOC_001848</td>\n",
       "      <td>novel</td>\n",
       "      <td>237</td>\n",
       "      <td>143</td>\n",
       "      <td>COAD</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>191013</th>\n",
       "      <td>TCONS_00002148</td>\n",
       "      <td>XLOC_001844</td>\n",
       "      <td>XLOC_001844</td>\n",
       "      <td>novel</td>\n",
       "      <td>249</td>\n",
       "      <td>110</td>\n",
       "      <td>COAD</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>191014</th>\n",
       "      <td>TCONS_00002149</td>\n",
       "      <td>XLOC_001845</td>\n",
       "      <td>XLOC_001845</td>\n",
       "      <td>novel</td>\n",
       "      <td>301</td>\n",
       "      <td>142</td>\n",
       "      <td>COAD</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>191015</th>\n",
       "      <td>TCONS_00002153</td>\n",
       "      <td>XLOC_001861</td>\n",
       "      <td>XLOC_001861</td>\n",
       "      <td>novel</td>\n",
       "      <td>259</td>\n",
       "      <td>63</td>\n",
       "      <td>COAD</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>191016 rows × 7 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "          transcript_id          gene_id    gene_name       gene_type  Length  \\\n",
       "0       ENST00000000412  ENSG00000003056         M6PR  protein_coding    2450   \n",
       "1       ENST00000001008  ENSG00000004478        FKBP4  protein_coding    3715   \n",
       "2       ENST00000001146  ENSG00000003137      CYP26B1  protein_coding    4556   \n",
       "3       ENST00000002165  ENSG00000001036        FUCA2  protein_coding    2385   \n",
       "4       ENST00000002596  ENSG00000002587       HS3ST1  protein_coding    7160   \n",
       "...                 ...              ...          ...             ...     ...   \n",
       "191011   TCONS_00002143      XLOC_001856  XLOC_001856           novel     244   \n",
       "191012   TCONS_00002144      XLOC_001848  XLOC_001848           novel     237   \n",
       "191013   TCONS_00002148      XLOC_001844  XLOC_001844           novel     249   \n",
       "191014   TCONS_00002149      XLOC_001845  XLOC_001845           novel     301   \n",
       "191015   TCONS_00002153      XLOC_001861  XLOC_001861           novel     259   \n",
       "\n",
       "          n ctype  \n",
       "0       131  BRCA  \n",
       "1       131  BRCA  \n",
       "2        83  BRCA  \n",
       "3       131  BRCA  \n",
       "4        63  BRCA  \n",
       "...     ...   ...  \n",
       "191011  140  COAD  \n",
       "191012  143  COAD  \n",
       "191013  110  COAD  \n",
       "191014  142  COAD  \n",
       "191015   63  COAD  \n",
       "\n",
       "[191016 rows x 7 columns]"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "tumorexpressed = pd.read_csv(os.path.join(users_dir, \"cancers/tumorexpressed/cancertypes/tumor_1FPKM_n10percent_pancancer.csv\"))\n",
    "tumorexpressed"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_3998815/1877913046.py:2: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  phylo_tumorexpressed.Phylostratum = phylo_tumorexpressed.Phylostratum.astype(\"category\")\n",
      "/tmp/ipykernel_3998815/1877913046.py:3: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  phylo_tumorexpressed.Phylostratum = phylo_tumorexpressed.Phylostratum.cat.set_categories(lineage)\n",
      "/tmp/ipykernel_3998815/1877913046.py:4: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  phylo_tumorexpressed.sort_values([\"Phylostratum\"], inplace=True)  ## 'sort' changed to 'sort_values'\n"
     ]
    }
   ],
   "source": [
    "phylo_tumorexpressed = all_genes[all_genes['gene_name'].isin(tumorexpressed.gene_name.values.tolist())]\n",
    "phylo_tumorexpressed.Phylostratum = phylo_tumorexpressed.Phylostratum.astype(\"category\")\n",
    "phylo_tumorexpressed.Phylostratum = phylo_tumorexpressed.Phylostratum.cat.set_categories(lineage)\n",
    "phylo_tumorexpressed.sort_values([\"Phylostratum\"], inplace=True)  ## 'sort' changed to 'sort_values'\n",
    "phylo_tumorexpressed.to_csv(os.path.join(phyloDir,\"all_genes_phyostratum.tumorexpressed.csv\"), index=None)"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.16"
  },
  "orig_nbformat": 4
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
