{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 42,
   "metadata": {},
   "outputs": [],
   "source": [
    "import os,re,glob\n",
    "import pandas as pd\n",
    "import numpy as np\n",
    "\n",
    "dir=\"/projects_eg/projects/marta/TestisRestricted_Microproteins_TSA/with_TranscriptomeReconstruction\"\n",
    "annot = pd.read_csv(\"/users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/human/newReference_Resconstructed/1transcript_1gene.reconstructed.csv\")\n"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Mudge 2022"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "Mudge2022_SS=pd.read_csv(\"/projects_eg/projects/marta/Mudge2022/Supp3_singlestudy_ORFs.csv\")\n",
    "Mudge2022_RS=pd.read_csv(\"/projects_eg/projects/marta/Mudge2022/Supp2_replicated_ORFs.csv\")\n",
    "Mudge2022  = pd.concat([Mudge2022_RS, Mudge2022_SS])\n",
    "Mudge2022"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene_id</th>\n",
       "      <th>gene_name_x</th>\n",
       "      <th>gene_name_y</th>\n",
       "      <th>gene_type</th>\n",
       "      <th>ORFpep</th>\n",
       "      <th>alternative_clustered_ORFs</th>\n",
       "      <th>number_studies_MS</th>\n",
       "      <th>MS_peptides</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ENSG00000133980</td>\n",
       "      <td>VRTN</td>\n",
       "      <td>VRTN</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>MTSRNQLVQKVLQELQEAVECEGLEGLIGASLEAKQVLSSFTLPTC...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ENSG00000133980</td>\n",
       "      <td>VRTN</td>\n",
       "      <td>VRTN</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>MTSRNQLVQKVLQELQEAVECEGLEGLIGASLEAKQVLSSFTLPTC...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ENSG00000198054</td>\n",
       "      <td>DSCR8</td>\n",
       "      <td>DSCR8</td>\n",
       "      <td>lncRNA</td>\n",
       "      <td>MPNPSFILLSKNEGAWTQLCYCEKGSSFIQDGIC*</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ENSG00000198054</td>\n",
       "      <td>DSCR8</td>\n",
       "      <td>DSCR8</td>\n",
       "      <td>lncRNA</td>\n",
       "      <td>MGGKKRRWSCHLEHA*</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ENSG00000234593</td>\n",
       "      <td>KAZN-AS1</td>\n",
       "      <td>KAZN-AS1</td>\n",
       "      <td>lncRNA</td>\n",
       "      <td>MFPSTLWKFCSFALCNKSHCCSLFGVSLSVRAVILAAKVSGFTSET...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>ENSG00000234593</td>\n",
       "      <td>KAZN-AS1</td>\n",
       "      <td>KAZN-AS1</td>\n",
       "      <td>lncRNA</td>\n",
       "      <td>MFPSTLWKFCSFALCNKSHCCSLFGVSLSVRAVILAAKVSGFTSET...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>ENSG00000234593</td>\n",
       "      <td>KAZN-AS1</td>\n",
       "      <td>KAZN-AS1</td>\n",
       "      <td>lncRNA</td>\n",
       "      <td>MFPSTLWKFCSFALCNKSHCCSLFGVSLSVRAVILAAKVSGFTSET...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>ENSG00000236716</td>\n",
       "      <td>RP11-478H13.1</td>\n",
       "      <td>AL355376.1</td>\n",
       "      <td>lncRNA</td>\n",
       "      <td>MTQRSRCETGPFLPCLRQGSRDCFRGDVRGQQGPGREEESRVTVRG...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>ENSG00000248307</td>\n",
       "      <td>LINC00616</td>\n",
       "      <td>LINC00616</td>\n",
       "      <td>lncRNA</td>\n",
       "      <td>MVELYHHFTGYRQGPPGFRKAAMTPQ*</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>ENSG00000248307</td>\n",
       "      <td>LINC00616</td>\n",
       "      <td>LINC00616</td>\n",
       "      <td>lncRNA</td>\n",
       "      <td>MPWDLCSGRYCIDYKW*</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>ENSG00000253408</td>\n",
       "      <td>IKBKB-DT</td>\n",
       "      <td>AC083973.1</td>\n",
       "      <td>lncRNA</td>\n",
       "      <td>MGSLGHPACSTIGIPGKVTESGNSPHLGTDSEQEDSSDSL*</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>ENSG00000263711</td>\n",
       "      <td>LINC02864</td>\n",
       "      <td>LINC02864</td>\n",
       "      <td>lncRNA</td>\n",
       "      <td>MLALFHFHLPPWDDAVRRPSVDASPSTLNFPDAELYASIFLCCMAP...</td>\n",
       "      <td>MLALFHFHLPPWDDAVRRPSVDASPSTLNFPDAEL*</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>ENSG00000263711</td>\n",
       "      <td>LINC02864</td>\n",
       "      <td>LINC02864</td>\n",
       "      <td>lncRNA</td>\n",
       "      <td>MLALFHFHLPPWDDAVRRPSVDASPSTLNFPDAELYASIFLCCMAP...</td>\n",
       "      <td>MLALFHFHLPPWDDAVRRPSVDASPSTLNFPDAEL*</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "            gene_id    gene_name_x gene_name_y       gene_type  \\\n",
       "0   ENSG00000133980           VRTN        VRTN  protein_coding   \n",
       "1   ENSG00000133980           VRTN        VRTN  protein_coding   \n",
       "2   ENSG00000198054          DSCR8       DSCR8          lncRNA   \n",
       "3   ENSG00000198054          DSCR8       DSCR8          lncRNA   \n",
       "4   ENSG00000234593       KAZN-AS1    KAZN-AS1          lncRNA   \n",
       "5   ENSG00000234593       KAZN-AS1    KAZN-AS1          lncRNA   \n",
       "6   ENSG00000234593       KAZN-AS1    KAZN-AS1          lncRNA   \n",
       "7   ENSG00000236716  RP11-478H13.1  AL355376.1          lncRNA   \n",
       "8   ENSG00000248307      LINC00616   LINC00616          lncRNA   \n",
       "9   ENSG00000248307      LINC00616   LINC00616          lncRNA   \n",
       "10  ENSG00000253408       IKBKB-DT  AC083973.1          lncRNA   \n",
       "11  ENSG00000263711      LINC02864   LINC02864          lncRNA   \n",
       "12  ENSG00000263711      LINC02864   LINC02864          lncRNA   \n",
       "\n",
       "                                               ORFpep  \\\n",
       "0   MTSRNQLVQKVLQELQEAVECEGLEGLIGASLEAKQVLSSFTLPTC...   \n",
       "1   MTSRNQLVQKVLQELQEAVECEGLEGLIGASLEAKQVLSSFTLPTC...   \n",
       "2                 MPNPSFILLSKNEGAWTQLCYCEKGSSFIQDGIC*   \n",
       "3                                    MGGKKRRWSCHLEHA*   \n",
       "4   MFPSTLWKFCSFALCNKSHCCSLFGVSLSVRAVILAAKVSGFTSET...   \n",
       "5   MFPSTLWKFCSFALCNKSHCCSLFGVSLSVRAVILAAKVSGFTSET...   \n",
       "6   MFPSTLWKFCSFALCNKSHCCSLFGVSLSVRAVILAAKVSGFTSET...   \n",
       "7   MTQRSRCETGPFLPCLRQGSRDCFRGDVRGQQGPGREEESRVTVRG...   \n",
       "8                         MVELYHHFTGYRQGPPGFRKAAMTPQ*   \n",
       "9                                   MPWDLCSGRYCIDYKW*   \n",
       "10          MGSLGHPACSTIGIPGKVTESGNSPHLGTDSEQEDSSDSL*   \n",
       "11  MLALFHFHLPPWDDAVRRPSVDASPSTLNFPDAELYASIFLCCMAP...   \n",
       "12  MLALFHFHLPPWDDAVRRPSVDASPSTLNFPDAELYASIFLCCMAP...   \n",
       "\n",
       "              alternative_clustered_ORFs  number_studies_MS MS_peptides  \n",
       "0                                    NaN                0.0         NaN  \n",
       "1                                    NaN                0.0         NaN  \n",
       "2                                    NaN                0.0         NaN  \n",
       "3                                    NaN                0.0         NaN  \n",
       "4                                    NaN                0.0         NaN  \n",
       "5                                    NaN                0.0         NaN  \n",
       "6                                    NaN                0.0         NaN  \n",
       "7                                    NaN                0.0         NaN  \n",
       "8                                    NaN                0.0         NaN  \n",
       "9                                    NaN                0.0         NaN  \n",
       "10                                   NaN                0.0         NaN  \n",
       "11  MLALFHFHLPPWDDAVRRPSVDASPSTLNFPDAEL*                0.0         NaN  \n",
       "12  MLALFHFHLPPWDDAVRRPSVDASPSTLNFPDAEL*                0.0         NaN  "
      ]
     },
     "execution_count": 33,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "TestisRestricted = pd.read_csv(os.path.join(dir,\"Q2_TestisRestricted/human/testisRestricted_GTEx_translatedONLYtestis.csv\"))\n",
    "TestisRestricted['Mudge2022'] = np.where(TestisRestricted['gene_id'].isin(Mudge2022.gene_id.values.tolist()), \"yes\", \"no\")\n",
    "\n",
    "TestisRestricted.groupby([\"Mudge2022\",\"gene_type\"]).count()\n",
    "\n",
    "TestisRestricted_Mudge2022 = TestisRestricted.merge(Mudge2022, on=\"gene_id\")\n",
    "TestisRestricted_Mudge2022\n",
    "TestisRestricted_Mudge2022[['gene_id','gene_name_x','gene_name_y','gene_type','ORFpep','alternative_clustered_ORFs','number_studies_MS','MS_peptides']]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene_id</th>\n",
       "      <th>gene_name_x</th>\n",
       "      <th>gene_name_y</th>\n",
       "      <th>gene_type</th>\n",
       "      <th>ORFpep</th>\n",
       "      <th>alternative_clustered_ORFs</th>\n",
       "      <th>number_studies_MS</th>\n",
       "      <th>MS_peptides</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ENSG00000253408</td>\n",
       "      <td>IKBKB-DT</td>\n",
       "      <td>AC083973.1</td>\n",
       "      <td>lncRNA</td>\n",
       "      <td>MGSLGHPACSTIGIPGKVTESGNSPHLGTDSEQEDSSDSL*</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ENSG00000253408</td>\n",
       "      <td>IKBKB-DT</td>\n",
       "      <td>AC083973.1</td>\n",
       "      <td>lncRNA</td>\n",
       "      <td>MGSLGHPACSTIGIPGKVTESGNSPHLGTDSEQEDSSDSL*</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ENSG00000253408</td>\n",
       "      <td>IKBKB-DT</td>\n",
       "      <td>AC083973.1</td>\n",
       "      <td>lncRNA</td>\n",
       "      <td>MGSLGHPACSTIGIPGKVTESGNSPHLGTDSEQEDSSDSL*</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ENSG00000263711</td>\n",
       "      <td>LINC02864</td>\n",
       "      <td>LINC02864</td>\n",
       "      <td>lncRNA</td>\n",
       "      <td>MLALFHFHLPPWDDAVRRPSVDASPSTLNFPDAELYASIFLCCMAP...</td>\n",
       "      <td>MLALFHFHLPPWDDAVRRPSVDASPSTLNFPDAEL*</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ENSG00000263711</td>\n",
       "      <td>LINC02864</td>\n",
       "      <td>LINC02864</td>\n",
       "      <td>lncRNA</td>\n",
       "      <td>MLALFHFHLPPWDDAVRRPSVDASPSTLNFPDAELYASIFLCCMAP...</td>\n",
       "      <td>MLALFHFHLPPWDDAVRRPSVDASPSTLNFPDAEL*</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>ENSG00000263711</td>\n",
       "      <td>LINC02864</td>\n",
       "      <td>LINC02864</td>\n",
       "      <td>lncRNA</td>\n",
       "      <td>MLALFHFHLPPWDDAVRRPSVDASPSTLNFPDAELYASIFLCCMAP...</td>\n",
       "      <td>MLALFHFHLPPWDDAVRRPSVDASPSTLNFPDAEL*</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>ENSG00000263711</td>\n",
       "      <td>LINC02864</td>\n",
       "      <td>LINC02864</td>\n",
       "      <td>lncRNA</td>\n",
       "      <td>MLALFHFHLPPWDDAVRRPSVDASPSTLNFPDAELYASIFLCCMAP...</td>\n",
       "      <td>MLALFHFHLPPWDDAVRRPSVDASPSTLNFPDAEL*</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>ENSG00000263711</td>\n",
       "      <td>LINC02864</td>\n",
       "      <td>LINC02864</td>\n",
       "      <td>lncRNA</td>\n",
       "      <td>MLALFHFHLPPWDDAVRRPSVDASPSTLNFPDAELYASIFLCCMAP...</td>\n",
       "      <td>MLALFHFHLPPWDDAVRRPSVDASPSTLNFPDAEL*</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>ENSG00000263711</td>\n",
       "      <td>LINC02864</td>\n",
       "      <td>LINC02864</td>\n",
       "      <td>lncRNA</td>\n",
       "      <td>MLALFHFHLPPWDDAVRRPSVDASPSTLNFPDAELYASIFLCCMAP...</td>\n",
       "      <td>MLALFHFHLPPWDDAVRRPSVDASPSTLNFPDAEL*</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>ENSG00000263711</td>\n",
       "      <td>LINC02864</td>\n",
       "      <td>LINC02864</td>\n",
       "      <td>lncRNA</td>\n",
       "      <td>MLALFHFHLPPWDDAVRRPSVDASPSTLNFPDAELYASIFLCCMAP...</td>\n",
       "      <td>MLALFHFHLPPWDDAVRRPSVDASPSTLNFPDAEL*</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>ENSG00000263711</td>\n",
       "      <td>LINC02864</td>\n",
       "      <td>LINC02864</td>\n",
       "      <td>lncRNA</td>\n",
       "      <td>MLALFHFHLPPWDDAVRRPSVDASPSTLNFPDAELYASIFLCCMAP...</td>\n",
       "      <td>MLALFHFHLPPWDDAVRRPSVDASPSTLNFPDAEL*</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>ENSG00000263711</td>\n",
       "      <td>LINC02864</td>\n",
       "      <td>LINC02864</td>\n",
       "      <td>lncRNA</td>\n",
       "      <td>MLALFHFHLPPWDDAVRRPSVDASPSTLNFPDAELYASIFLCCMAP...</td>\n",
       "      <td>MLALFHFHLPPWDDAVRRPSVDASPSTLNFPDAEL*</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>ENSG00000263711</td>\n",
       "      <td>LINC02864</td>\n",
       "      <td>LINC02864</td>\n",
       "      <td>lncRNA</td>\n",
       "      <td>MLALFHFHLPPWDDAVRRPSVDASPSTLNFPDAELYASIFLCCMAP...</td>\n",
       "      <td>MLALFHFHLPPWDDAVRRPSVDASPSTLNFPDAEL*</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "            gene_id gene_name_x gene_name_y gene_type  \\\n",
       "0   ENSG00000253408    IKBKB-DT  AC083973.1    lncRNA   \n",
       "1   ENSG00000253408    IKBKB-DT  AC083973.1    lncRNA   \n",
       "2   ENSG00000253408    IKBKB-DT  AC083973.1    lncRNA   \n",
       "3   ENSG00000263711   LINC02864   LINC02864    lncRNA   \n",
       "4   ENSG00000263711   LINC02864   LINC02864    lncRNA   \n",
       "5   ENSG00000263711   LINC02864   LINC02864    lncRNA   \n",
       "6   ENSG00000263711   LINC02864   LINC02864    lncRNA   \n",
       "7   ENSG00000263711   LINC02864   LINC02864    lncRNA   \n",
       "8   ENSG00000263711   LINC02864   LINC02864    lncRNA   \n",
       "9   ENSG00000263711   LINC02864   LINC02864    lncRNA   \n",
       "10  ENSG00000263711   LINC02864   LINC02864    lncRNA   \n",
       "11  ENSG00000263711   LINC02864   LINC02864    lncRNA   \n",
       "12  ENSG00000263711   LINC02864   LINC02864    lncRNA   \n",
       "\n",
       "                                               ORFpep  \\\n",
       "0           MGSLGHPACSTIGIPGKVTESGNSPHLGTDSEQEDSSDSL*   \n",
       "1           MGSLGHPACSTIGIPGKVTESGNSPHLGTDSEQEDSSDSL*   \n",
       "2           MGSLGHPACSTIGIPGKVTESGNSPHLGTDSEQEDSSDSL*   \n",
       "3   MLALFHFHLPPWDDAVRRPSVDASPSTLNFPDAELYASIFLCCMAP...   \n",
       "4   MLALFHFHLPPWDDAVRRPSVDASPSTLNFPDAELYASIFLCCMAP...   \n",
       "5   MLALFHFHLPPWDDAVRRPSVDASPSTLNFPDAELYASIFLCCMAP...   \n",
       "6   MLALFHFHLPPWDDAVRRPSVDASPSTLNFPDAELYASIFLCCMAP...   \n",
       "7   MLALFHFHLPPWDDAVRRPSVDASPSTLNFPDAELYASIFLCCMAP...   \n",
       "8   MLALFHFHLPPWDDAVRRPSVDASPSTLNFPDAELYASIFLCCMAP...   \n",
       "9   MLALFHFHLPPWDDAVRRPSVDASPSTLNFPDAELYASIFLCCMAP...   \n",
       "10  MLALFHFHLPPWDDAVRRPSVDASPSTLNFPDAELYASIFLCCMAP...   \n",
       "11  MLALFHFHLPPWDDAVRRPSVDASPSTLNFPDAELYASIFLCCMAP...   \n",
       "12  MLALFHFHLPPWDDAVRRPSVDASPSTLNFPDAELYASIFLCCMAP...   \n",
       "\n",
       "              alternative_clustered_ORFs  number_studies_MS MS_peptides  \n",
       "0                                    NaN                0.0         NaN  \n",
       "1                                    NaN                0.0         NaN  \n",
       "2                                    NaN                0.0         NaN  \n",
       "3   MLALFHFHLPPWDDAVRRPSVDASPSTLNFPDAEL*                0.0         NaN  \n",
       "4   MLALFHFHLPPWDDAVRRPSVDASPSTLNFPDAEL*                0.0         NaN  \n",
       "5   MLALFHFHLPPWDDAVRRPSVDASPSTLNFPDAEL*                0.0         NaN  \n",
       "6   MLALFHFHLPPWDDAVRRPSVDASPSTLNFPDAEL*                0.0         NaN  \n",
       "7   MLALFHFHLPPWDDAVRRPSVDASPSTLNFPDAEL*                0.0         NaN  \n",
       "8   MLALFHFHLPPWDDAVRRPSVDASPSTLNFPDAEL*                0.0         NaN  \n",
       "9   MLALFHFHLPPWDDAVRRPSVDASPSTLNFPDAEL*                0.0         NaN  \n",
       "10  MLALFHFHLPPWDDAVRRPSVDASPSTLNFPDAEL*                0.0         NaN  \n",
       "11  MLALFHFHLPPWDDAVRRPSVDASPSTLNFPDAEL*                0.0         NaN  \n",
       "12  MLALFHFHLPPWDDAVRRPSVDASPSTLNFPDAEL*                0.0         NaN  "
      ]
     },
     "execution_count": 32,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "TSTR = pd.read_csv(os.path.join(dir,\"Q4_TestisRestricted_TumorSpecific/human/TOv3x_5percent_TestisRestrictedGTEx_Translated_Ctypes_log2ratio3xMEAN.csv\"))\n",
    "TSTR\n",
    "TSTR['Mudge2022'] = np.where(TSTR['gene_id'].isin(Mudge2022.gene_id.values.tolist()), \"yes\", \"no\")\n",
    "\n",
    "TSTR.groupby([\"Mudge2022\",\"gene_type\"]).count()\n",
    "\n",
    "TSTR_Mudge2022 = TSTR.merge(Mudge2022, on=\"gene_id\")\n",
    "TSTR_Mudge2022\n",
    "TSTR_Mudge2022[['gene_id','gene_name_x','gene_name_y','gene_type','ORFpep','alternative_clustered_ORFs','number_studies_MS','MS_peptides']]"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Chen 2022"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "19606\n"
     ]
    }
   ],
   "source": [
    "Chen2020=pd.read_csv(\"/data/genomics/marta/Chen_SuppTableS2.csv\")\n",
    "Chen2020\n",
    "def save_to_fasta(df, filename):\n",
    "    with open(filename, 'w') as f:\n",
    "        for index, row in df.iterrows():\n",
    "            f.write(f\">{row['ORF name']}\\n\")\n",
    "            f.write(f\"{row['Amino acid sequence']}\\n\")\n",
    "\n",
    "\n",
    "## select protein coding sequences\n",
    "Chen2020_to_fasta = Chen2020[['ORF name','Amino acid sequence']]\n",
    "print(len(Chen2020_to_fasta))\n",
    "\n",
    "## save the fasta sequences\n",
    "save_to_fasta(Chen2020_to_fasta, os.path.join(dir,\"Q2_TestisRestricted/human/blastp_Chen/Chen_SuppTableS2.fa\"))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "Building a new DB, current time: 11/11/2024 12:50:36\n",
      "New DB name:   /projects_eg/projects/marta/TestisRestricted_Microproteins_TSA/with_TranscriptomeReconstruction/Q2_TestisRestricted/human/blastp_Chen/Chen_SuppTableS2\n",
      "New DB title:  /projects_eg/projects/marta/TestisRestricted_Microproteins_TSA/with_TranscriptomeReconstruction/Q2_TestisRestricted/human/blastp_Chen/Chen_SuppTableS2.fa\n",
      "Sequence type: Protein\n",
      "Keep MBits: T\n",
      "Maximum file size: 1000000000B\n",
      "Adding sequences from FASTA; added 19606 sequences in 0.295758 seconds.\n",
      "\n",
      "\n"
     ]
    }
   ],
   "source": [
    "%%bash -s \"$dir\"\n",
    "\n",
    "## create blastp database\n",
    "module load BLAST+/2.12.0-Linux_x86_64\n",
    "\n",
    "outdir=$1/Q2_TestisRestricted/human/blastp_Chen\n",
    "\n",
    "makeblastdb -in $outdir/Chen_SuppTableS2.fa -out $outdir/Chen_SuppTableS2 -parse_seqids -dbtype prot\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash -s \"$dir\"\n",
    "## TestisRestricted\n",
    "module load BLAST+/2.12.0-Linux_x86_64\n",
    "\n",
    "outdir=$1/Q2_TestisRestricted/human/blastp_Chen\n",
    "\n",
    "blastp -query $1/Q2_TestisRestricted/human/testisRestricted_GTEx_translatedONLYtestis.NonCanonical.fa \\\n",
    "-db $outdir/Chen_SuppTableS2 \\\n",
    "-task 'blastp-short' \\\n",
    "-out $outdir/TestisRestrictedGTExncORFs_Chen.txt \\\n",
    "-evalue 0.0001 \\\n",
    "-outfmt \"6 qseqid sseqid pident length mismatch gapopen qstart qend sstart send evalue bitscore qlen slen qframe sframe sstrand qcovs\"\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>0</th>\n",
       "      <th>1</th>\n",
       "      <th>2</th>\n",
       "      <th>3</th>\n",
       "      <th>4</th>\n",
       "      <th>5</th>\n",
       "      <th>6</th>\n",
       "      <th>7</th>\n",
       "      <th>8</th>\n",
       "      <th>9</th>\n",
       "      <th>...</th>\n",
       "      <th>12</th>\n",
       "      <th>13</th>\n",
       "      <th>14</th>\n",
       "      <th>15</th>\n",
       "      <th>16</th>\n",
       "      <th>17</th>\n",
       "      <th>transcript_id</th>\n",
       "      <th>gene_id</th>\n",
       "      <th>gene_type</th>\n",
       "      <th>gene_name</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>0 rows × 22 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, transcript_id, gene_id, gene_type, gene_name]\n",
       "Index: []\n",
       "\n",
       "[0 rows x 22 columns]"
      ]
     },
     "execution_count": 49,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "TSTR = pd.read_csv(os.path.join(dir,\"Q4_TestisRestricted_TumorSpecific/human/TOv3x_5percent_TestisRestrictedGTEx_Translated_Ctypes_log2ratio3xMEAN.csv\"))\n",
    "blastp_Chen = pd.read_csv(os.path.join(dir,\"Q2_TestisRestricted/human/blastp_Chen/TestisRestrictedGTExncORFs_Chen.txt\"), sep=\"\\t\", header=None)\n",
    "blastp_Chen['transcript_id'] = blastp_Chen[0].str.split(\"|\", expand=True)[1]\n",
    "blastp_Chen['transcript_id'] = blastp_Chen['transcript_id'].str.split(\":\", expand=True)[0]\n",
    "blastp_Chen['transcript_id'] = blastp_Chen['transcript_id'].str.split(\".\", expand=True)[0]\n",
    "blastp_Chen_annot = blastp_Chen.merge(annot, on=\"transcript_id\")\n",
    "blastp_Chen_annot\n",
    "\n",
    "TSTR_Chen = blastp_Chen_annot[blastp_Chen_annot['gene_name'].isin(TSTR.gene_name.values.tolist())]\n",
    "TSTR_Chen"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.16"
  },
  "orig_nbformat": 4
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
