{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import os,re,glob\n",
    "import pandas as pd\n",
    "import numpy as np\n",
    "from collections import Counter\n",
    "from rna_seq_normalization import Normalization as Norm\n",
    "from functools import reduce\n",
    "\n",
    "users_dir = \"/users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction\"\n",
    "specie = \"human\"\n",
    "# GENOMEDIR = \"/genomics/users/marta/genomes\"\n",
    "GENOMEDIR = \"/data/genomics/marta/genomes\"\n",
    "\n",
    "### cancer data\n",
    "cancer_dir = users_dir + \"/cancers\"\n",
    "raw_cancer_dir=\"/users/genomics/marta/TCGA_RNASeq\"\n",
    "tcga_projects=[\"TCGA-BRCA\",\"TCGA-LUSC\",\"TCGA-PRAD\",\"TCGA-KIRC\",\"TCGA-KIRP\",\"TCGA-LUAD\",\"TCGA-BLCA\"]#,\"TCGA-LIHC\"]\n",
    "other_projects=[\"GSE102101_KIRC\",\"GSE133624_BLCA\",\"GSE22260_PRAD\",\"PRJEB2449_PRAD\",\"SRP238334_KIRC\",\"GSE214846_LIHC\",\"GSE229705_LUAD\",\"TCGA_COAD\",\"SRP107326_COAD\"]\n",
    "# deleted=[\"GSE103001_BRCA\",\"GSE89223_PRAD\"]\n",
    "manuscript_projects = [\"liver_adjacent_totalRNA_LIHC\",\"hcc_normal_totalRNA_LIHC\",\"GSE193567_LIHC\",\"LIHC_TCGA_LIHC\"]\n",
    "all_projects = tcga_projects + other_projects + manuscript_projects\n",
    "\n",
    "cancertypes = [\"BRCA\",\"BLCA\",\"LUAD\",\"KIRC\",\"PRAD\",\"LUSC\",\"LIHC\",\"COAD\"]\n",
    "## annotation file\n",
    "annotation=\"/users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/human/newReference_Resconstructed/gencode.v38.gffcompare.TestisLiverBrain.annotation.sorted.1transcript.sorted.NOchr.gtf\"\n",
    "transcript_gene=pd.read_csv(\"/users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/human/newReference_Resconstructed/1transcript_1gene.reconstructed.csv\")\n",
    "\n",
    "def count_greater_than_one(row):\n",
    "    return (row > 1).sum()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Unnamed: 0</th>\n",
       "      <th>gene_name</th>\n",
       "      <th>gene_id</th>\n",
       "      <th>transcript_id</th>\n",
       "      <th>gene_type</th>\n",
       "      <th>Length</th>\n",
       "      <th>num_patients_overexpr</th>\n",
       "      <th>percentage_num_patients_overexpr</th>\n",
       "      <th>ctype</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>0</td>\n",
       "      <td>MAGEC2</td>\n",
       "      <td>ENSG00000046774</td>\n",
       "      <td>ENST00000247452</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>1994.0</td>\n",
       "      <td>8.0</td>\n",
       "      <td>7.34</td>\n",
       "      <td>BRCA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>4</td>\n",
       "      <td>ACTL8</td>\n",
       "      <td>ENSG00000117148</td>\n",
       "      <td>ENST00000375406</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>1841.0</td>\n",
       "      <td>27.0</td>\n",
       "      <td>24.77</td>\n",
       "      <td>BRCA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>7</td>\n",
       "      <td>MAGEA10</td>\n",
       "      <td>ENSG00000124260</td>\n",
       "      <td>ENST00000244096</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>2741.0</td>\n",
       "      <td>6.0</td>\n",
       "      <td>5.50</td>\n",
       "      <td>BRCA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>9</td>\n",
       "      <td>SSX1</td>\n",
       "      <td>ENSG00000126752</td>\n",
       "      <td>ENST00000376919</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>1223.0</td>\n",
       "      <td>6.0</td>\n",
       "      <td>5.50</td>\n",
       "      <td>BRCA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>11</td>\n",
       "      <td>LIN28A</td>\n",
       "      <td>ENSG00000131914</td>\n",
       "      <td>ENST00000326279</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>3975.0</td>\n",
       "      <td>7.0</td>\n",
       "      <td>6.42</td>\n",
       "      <td>BRCA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>257</th>\n",
       "      <td>77</td>\n",
       "      <td>LINC02864</td>\n",
       "      <td>ENSG00000263711</td>\n",
       "      <td>ENST00000581862</td>\n",
       "      <td>lncRNA</td>\n",
       "      <td>3634.0</td>\n",
       "      <td>13.0</td>\n",
       "      <td>9.03</td>\n",
       "      <td>COAD</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>258</th>\n",
       "      <td>86</td>\n",
       "      <td>XLOC_000310</td>\n",
       "      <td>XLOC_000310</td>\n",
       "      <td>TCONS_00000389</td>\n",
       "      <td>novel</td>\n",
       "      <td>322.0</td>\n",
       "      <td>10.0</td>\n",
       "      <td>6.94</td>\n",
       "      <td>COAD</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>259</th>\n",
       "      <td>88</td>\n",
       "      <td>XLOC_000930</td>\n",
       "      <td>XLOC_000930</td>\n",
       "      <td>TCONS_00001039</td>\n",
       "      <td>novel</td>\n",
       "      <td>6255.0</td>\n",
       "      <td>55.0</td>\n",
       "      <td>38.19</td>\n",
       "      <td>COAD</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>260</th>\n",
       "      <td>91</td>\n",
       "      <td>XLOC_001049</td>\n",
       "      <td>XLOC_001049</td>\n",
       "      <td>TCONS_00001208</td>\n",
       "      <td>novel</td>\n",
       "      <td>2720.0</td>\n",
       "      <td>29.0</td>\n",
       "      <td>20.14</td>\n",
       "      <td>COAD</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>261</th>\n",
       "      <td>99</td>\n",
       "      <td>XLOC_001676</td>\n",
       "      <td>XLOC_001676</td>\n",
       "      <td>TCONS_00001992</td>\n",
       "      <td>novel</td>\n",
       "      <td>1191.0</td>\n",
       "      <td>9.0</td>\n",
       "      <td>6.25</td>\n",
       "      <td>COAD</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>262 rows × 9 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     Unnamed: 0    gene_name          gene_id    transcript_id  \\\n",
       "0             0       MAGEC2  ENSG00000046774  ENST00000247452   \n",
       "1             4        ACTL8  ENSG00000117148  ENST00000375406   \n",
       "2             7      MAGEA10  ENSG00000124260  ENST00000244096   \n",
       "3             9         SSX1  ENSG00000126752  ENST00000376919   \n",
       "4            11       LIN28A  ENSG00000131914  ENST00000326279   \n",
       "..          ...          ...              ...              ...   \n",
       "257          77    LINC02864  ENSG00000263711  ENST00000581862   \n",
       "258          86  XLOC_000310      XLOC_000310   TCONS_00000389   \n",
       "259          88  XLOC_000930      XLOC_000930   TCONS_00001039   \n",
       "260          91  XLOC_001049      XLOC_001049   TCONS_00001208   \n",
       "261          99  XLOC_001676      XLOC_001676   TCONS_00001992   \n",
       "\n",
       "          gene_type  Length  num_patients_overexpr  \\\n",
       "0    protein_coding  1994.0                    8.0   \n",
       "1    protein_coding  1841.0                   27.0   \n",
       "2    protein_coding  2741.0                    6.0   \n",
       "3    protein_coding  1223.0                    6.0   \n",
       "4    protein_coding  3975.0                    7.0   \n",
       "..              ...     ...                    ...   \n",
       "257          lncRNA  3634.0                   13.0   \n",
       "258           novel   322.0                   10.0   \n",
       "259           novel  6255.0                   55.0   \n",
       "260           novel  2720.0                   29.0   \n",
       "261           novel  1191.0                    9.0   \n",
       "\n",
       "     percentage_num_patients_overexpr ctype  \n",
       "0                                7.34  BRCA  \n",
       "1                               24.77  BRCA  \n",
       "2                                5.50  BRCA  \n",
       "3                                5.50  BRCA  \n",
       "4                                6.42  BRCA  \n",
       "..                                ...   ...  \n",
       "257                              9.03  COAD  \n",
       "258                              6.94  COAD  \n",
       "259                             38.19  COAD  \n",
       "260                             20.14  COAD  \n",
       "261                              6.25  COAD  \n",
       "\n",
       "[262 rows x 9 columns]"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "candidates=pd.read_csv(os.path.join(cancer_dir,\"log2ratio3x/cancertypes/TOv3x_5percent_TestisRestrictedGTEx_Translated_Ctypes_log2ratio3xMEANgenes.csv\"))\n",
    "candidates"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>header</th>\n",
       "      <th>seq</th>\n",
       "      <th>transcript_id</th>\n",
       "      <th>ORFtype</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>&gt;ENST00000381163.7:X:+|14|3393:283:1789|canoni...</td>\n",
       "      <td>MSETEFHHGAQAGLELLRSSNSPTSASQSAGMTVTDQAFVTLATND...</td>\n",
       "      <td>ENST00000381163</td>\n",
       "      <td>canonical</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>&gt;ENST00000381154.6:X:-|4|5145:65:1847|canonica...</td>\n",
       "      <td>MRSAARRGRAAPAARDSLPVLLFLCLLLKTCEPKTANAFKPNILLI...</td>\n",
       "      <td>ENST00000381154</td>\n",
       "      <td>canonical</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>&gt;ENST00000683290.1:X:-|31|2976:345:2190|canoni...</td>\n",
       "      <td>MRPVINRCAPGSLDLMLPQAASEGIVFHSLQISLCFRSWLPAMLAV...</td>\n",
       "      <td>ENST00000683290</td>\n",
       "      <td>canonical</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>&gt;ENST00000381127.6:X:+|19|2165:222:1995|canoni...</td>\n",
       "      <td>MRPRRPLVFMSLVCALLNTCQAHRVHDDKPNIVLIMVDDLGIGDLG...</td>\n",
       "      <td>ENST00000381127</td>\n",
       "      <td>canonical</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>&gt;ENST00000217939.7:X:-|6|9804:92:8654|extensio...</td>\n",
       "      <td>MLPAHRSRVRRLPKDRYPRIRHPDKMPKRAHWGALSVVLILLWGHP...</td>\n",
       "      <td>ENST00000217939</td>\n",
       "      <td>extension</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>43012</th>\n",
       "      <td>&gt;ENST00000468564.6:22:+|19|5066:195:372|noncod...</td>\n",
       "      <td>VLKSVGDLQGVWPSAPQGPHALVIPTLLPGGKKKYLPPTSRKDPKF...</td>\n",
       "      <td>ENST00000468564</td>\n",
       "      <td>noncoding</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>43013</th>\n",
       "      <td>&gt;ENST00000468564.6:22:+|312|5066:3913:3970|non...</td>\n",
       "      <td>MRAHGTSLMNYLSWLRRK*</td>\n",
       "      <td>ENST00000468564</td>\n",
       "      <td>noncoding</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>43014</th>\n",
       "      <td>&gt;ENST00000381176.5:22:-|350|6853:4262:4343|dOR...</td>\n",
       "      <td>MSSFMLSLSSVYGVIVVCFRISIQNQ*</td>\n",
       "      <td>ENST00000381176</td>\n",
       "      <td>dORF</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>43015</th>\n",
       "      <td>&gt;ENST00000608025.2:22:-|4|1001:41:164|noncodin...</td>\n",
       "      <td>MGTPPRPPATTPLGQPRISSAAPAPAAGTEARVRSGRAGA*</td>\n",
       "      <td>ENST00000608025</td>\n",
       "      <td>noncoding</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>43016</th>\n",
       "      <td>&gt;TCONS_00000005:GL000219.1:-|2|393:55:319|nonc...</td>\n",
       "      <td>MRSLPSLCSREPRRRGGSCTSGGSLLSSFRTHSTPLESGGSRGAES...</td>\n",
       "      <td>TCONS_00000005</td>\n",
       "      <td>noncoding</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>43017 rows × 4 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                  header  \\\n",
       "0      >ENST00000381163.7:X:+|14|3393:283:1789|canoni...   \n",
       "1      >ENST00000381154.6:X:-|4|5145:65:1847|canonica...   \n",
       "2      >ENST00000683290.1:X:-|31|2976:345:2190|canoni...   \n",
       "3      >ENST00000381127.6:X:+|19|2165:222:1995|canoni...   \n",
       "4      >ENST00000217939.7:X:-|6|9804:92:8654|extensio...   \n",
       "...                                                  ...   \n",
       "43012  >ENST00000468564.6:22:+|19|5066:195:372|noncod...   \n",
       "43013  >ENST00000468564.6:22:+|312|5066:3913:3970|non...   \n",
       "43014  >ENST00000381176.5:22:-|350|6853:4262:4343|dOR...   \n",
       "43015  >ENST00000608025.2:22:-|4|1001:41:164|noncodin...   \n",
       "43016  >TCONS_00000005:GL000219.1:-|2|393:55:319|nonc...   \n",
       "\n",
       "                                                     seq    transcript_id  \\\n",
       "0      MSETEFHHGAQAGLELLRSSNSPTSASQSAGMTVTDQAFVTLATND...  ENST00000381163   \n",
       "1      MRSAARRGRAAPAARDSLPVLLFLCLLLKTCEPKTANAFKPNILLI...  ENST00000381154   \n",
       "2      MRPVINRCAPGSLDLMLPQAASEGIVFHSLQISLCFRSWLPAMLAV...  ENST00000683290   \n",
       "3      MRPRRPLVFMSLVCALLNTCQAHRVHDDKPNIVLIMVDDLGIGDLG...  ENST00000381127   \n",
       "4      MLPAHRSRVRRLPKDRYPRIRHPDKMPKRAHWGALSVVLILLWGHP...  ENST00000217939   \n",
       "...                                                  ...              ...   \n",
       "43012  VLKSVGDLQGVWPSAPQGPHALVIPTLLPGGKKKYLPPTSRKDPKF...  ENST00000468564   \n",
       "43013                                MRAHGTSLMNYLSWLRRK*  ENST00000468564   \n",
       "43014                        MSSFMLSLSSVYGVIVVCFRISIQNQ*  ENST00000381176   \n",
       "43015          MGTPPRPPATTPLGQPRISSAAPAPAAGTEARVRSGRAGA*  ENST00000608025   \n",
       "43016  MRSLPSLCSREPRRRGGSCTSGGSLLSSFRTHSTPLESGGSRGAES...   TCONS_00000005   \n",
       "\n",
       "         ORFtype  \n",
       "0      canonical  \n",
       "1      canonical  \n",
       "2      canonical  \n",
       "3      canonical  \n",
       "4      extension  \n",
       "...          ...  \n",
       "43012  noncoding  \n",
       "43013  noncoding  \n",
       "43014       dORF  \n",
       "43015  noncoding  \n",
       "43016  noncoding  \n",
       "\n",
       "[43017 rows x 4 columns]"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "human_seqs_riboseq_testis = pd.read_csv(\"/users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/human/240724_RiboNovel/RibORF/repre.valid.pred.pvalue.parameters.allsamples.noredundant.PROTEIN.fa\", header=None, sep=\"\\t\")\n",
    "df = pd.DataFrame({'header':human_seqs_riboseq_testis[0].iloc[::2].values, 'seq':human_seqs_riboseq_testis[0].iloc[1::2].values})\n",
    "df['transcript_id'] = df['header'].str.split(\":\", expand=True)[0]\n",
    "df['transcript_id'] = df['transcript_id'].str.split(\"\\.\", expand=True)[0]\n",
    "df['transcript_id'] = df['transcript_id'].str.split(\">\", expand=True)[1]\n",
    "df['ORFtype'] = df['header'].str.split(\"\\|\", expand=True)[3]\n",
    "df\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'ACTL8',\n",
       " 'ADAD1',\n",
       " 'CBY2',\n",
       " 'COX7B2',\n",
       " 'DCAF4L2',\n",
       " 'DCAF8L2',\n",
       " 'GOLGA6L7',\n",
       " 'KCNU1',\n",
       " 'LUZP4',\n",
       " 'LYZL2',\n",
       " 'MAGEA10',\n",
       " 'MAGEB1',\n",
       " 'MAGEC1',\n",
       " 'MAGEC2',\n",
       " 'MEIOB',\n",
       " 'PAGE2',\n",
       " 'PRSS54',\n",
       " 'RBM46',\n",
       " 'RFPL4B',\n",
       " 'RNF17',\n",
       " 'SOX30',\n",
       " 'SPATA12',\n",
       " 'SSX3',\n",
       " 'TFDP3',\n",
       " 'TSBP1',\n",
       " 'VCX',\n",
       " 'ZFP42'}"
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ORFs_from_candidates = df.merge(candidates, on=\"transcript_id\")\n",
    "set(ORFs_from_candidates[(ORFs_from_candidates['gene_type'] == \"protein_coding\") & (ORFs_from_candidates['ORFtype'] != \"canonical\")].gene_name.values.tolist())"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.16"
  },
  "orig_nbformat": 4
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
