{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import os,re,glob\n",
    "import pandas as pd\n",
    "import numpy as np\n",
    "from collections import Counter\n",
    "from rna_seq_normalization import Normalization as Norm\n",
    "from functools import reduce\n",
    "\n",
    "users_dir = \"/users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction\"\n",
    "specie = \"human\"\n",
    "# GENOMEDIR = \"/genomics/users/marta/genomes\"\n",
    "GENOMEDIR = \"/data/genomics/marta/genomes\"\n",
    "\n",
    "### cancer data\n",
    "cancer_dir = users_dir + \"/cancers\"\n",
    "raw_cancer_dir=\"/users/genomics/marta/TCGA_RNASeq\"\n",
    "tcga_projects=[\"TCGA-BRCA\",\"TCGA-LUSC\",\"TCGA-PRAD\",\"TCGA-KIRC\",\"TCGA-KIRP\",\"TCGA-LUAD\",\"TCGA-BLCA\"]#,\"TCGA-LIHC\"]\n",
    "other_projects=[\"GSE102101_KIRC\",\"GSE133624_BLCA\",\"GSE22260_PRAD\",\"PRJEB2449_PRAD\",\"SRP238334_KIRC\",\"GSE214846_LIHC\",\"GSE229705_LUAD\",\"TCGA_COAD\",\"SRP107326_COAD\"]\n",
    "# deleted=[\"GSE103001_BRCA\",\"GSE89223_PRAD\"]\n",
    "manuscript_projects = [\"liver_adjacent_totalRNA_LIHC\",\"hcc_normal_totalRNA_LIHC\",\"GSE193567_LIHC\",\"LIHC_TCGA_LIHC\"]\n",
    "all_projects = tcga_projects + other_projects + manuscript_projects\n",
    "\n",
    "cancertypes = [\"BRCA\",\"BLCA\",\"LUAD\",\"KIRC\",\"KIRP\",\"PRAD\",\"LUSC\",\"LIHC\",\"COAD\"]\n",
    "## annotation file\n",
    "annotation=\"/users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/human/newReference_Resconstructed/gencode.v38.gffcompare.TestisLiverBrain.annotation.sorted.1transcript.sorted.NOchr.gtf\"\n",
    "transcript_gene=pd.read_csv(\"/users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/human/newReference_Resconstructed/1transcript_1gene.reconstructed.csv\")\n",
    "\n",
    "def count_greater_than_one(row):\n",
    "    return (row > 1).sum()"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Quantify with TCGAData and new reference\n",
    "`featureCounts_newRef_cancer.sh`\n",
    "\n",
    "output in: `/users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/cancers/featureCounts`"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## TPM"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "TCGA-PRAD\n",
      "TCGA-LUSC\n",
      "GSE89223_PRAD\n",
      "TCGA-BLCA\n",
      "GSE102101_KIRC\n",
      "TCGA_COAD_SE\n",
      "GSE103001_BRCA\n",
      "TCGA-BRCA\n",
      "GSE214846_LIHC\n",
      "liver_adjacent_totalRNA_LIHC\n",
      "GSE193567_LIHC\n",
      "TCGA_COAD_PE\n",
      "TCGA-KIRP\n",
      "GSE22260_PRAD\n",
      "GSE229705_LUAD\n",
      "GSE133624_BLCA\n",
      "PRJEB2449_PRAD\n",
      "TCGA-LUAD\n",
      "SRP107326_COAD\n",
      "hcc_normal_totalRNA_LIHC\n",
      "TCGA-KIRC\n",
      "LIHC_TCGA_LIHC\n",
      "SRP238334_KIRC\n"
     ]
    }
   ],
   "source": [
    "for file in os.listdir(os.path.join(cancer_dir,\"featureCounts\")):\n",
    "    if file.endswith(\".txt\"):\n",
    "\n",
    "        proj = file.split(\"featureCounts_\")[-1]\n",
    "        cancer_type = proj[:-4]\n",
    "        print(cancer_type)\n",
    "        toc = pd.read_csv(os.path.join(cancer_dir,\"featureCounts\",file), sep=\"\\t\", comment=\"#\")\n",
    "        toc = toc[toc['Geneid'].str.contains('PAR_')==False]\n",
    "        toc['Geneid']=toc['Geneid'].str.split('.').str[0]\n",
    "        toc.rename(columns={'Geneid':'transcript_id'}, inplace=True)\n",
    "\n",
    "        filter_col = [col for col in toc if col.startswith('/')]\n",
    "        for col in filter_col:\n",
    "            new_col=col.split(\"Aligned\")[0]\n",
    "            new_col=new_col.split(\"/\")[-1]\n",
    "            if proj in tcga_projects:\n",
    "                new_col=cancer_type+new_col[4:]\n",
    "            toc.rename(columns={col:new_col}, inplace=True)\n",
    "\n",
    "        length = toc['Length']\n",
    "        genes = toc['transcript_id']\n",
    "        # we are only interested in the columns with counts\n",
    "        counts = toc\n",
    "        counts.drop([\"Chr\",\"Start\",\"End\",\"Strand\",\"Length\",\"transcript_id\"],axis=1, inplace=True)\n",
    "        # calculate TPMs\n",
    "        tpm_df = Norm.tpm(counts, length)\n",
    "        # add transcript_id and length again\n",
    "        tpms = pd.concat([genes,tpm_df, length], axis=1)\n",
    "        tpms.to_csv(os.path.join(cancer_dir,\"featureCounts/table_of_counts_TPMs_\")+cancer_type+\".csv\", index=None)\n",
    " "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "19\n",
      "66\n",
      "83\n"
     ]
    }
   ],
   "source": [
    "## merge single-end and paired-end TCGA COAD\n",
    "SE_COAD = pd.read_csv(os.path.join(cancer_dir,\"featureCounts/table_of_counts_TPMs_TCGA_COAD_SE.csv\"))\n",
    "print(len(SE_COAD.columns))\n",
    "PE_COAD = pd.read_csv(os.path.join(cancer_dir,\"featureCounts/table_of_counts_TPMs_TCGA_COAD_PE.csv\"))\n",
    "print(len(PE_COAD.columns))\n",
    "\n",
    "COAD = SE_COAD.merge(PE_COAD, on=[\"transcript_id\",\"Length\"])\n",
    "COAD.to_csv(os.path.join(cancer_dir,\"featureCounts/table_of_counts_TPMs_TCGA_COAD.csv\"), index=None)\n",
    "print(len(COAD.columns))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "TCGA-BRCA\n",
      "109  patients\n",
      "lncRNA:  10685\n",
      "PROTEIN CODING:  16230\n",
      "NOVEL:  478\n",
      "TCGA-LUSC\n",
      "49  patients\n",
      "lncRNA:  9922\n",
      "PROTEIN CODING:  16280\n",
      "NOVEL:  456\n",
      "TCGA-PRAD\n",
      "52  patients\n",
      "lncRNA:  7393\n",
      "PROTEIN CODING:  15071\n",
      "NOVEL:  385\n",
      "TCGA-KIRC\n",
      "71  patients\n",
      "lncRNA:  7978\n",
      "PROTEIN CODING:  15463\n",
      "NOVEL:  391\n",
      "TCGA-KIRP\n",
      "32  patients\n",
      "lncRNA:  7911\n",
      "PROTEIN CODING:  15436\n",
      "NOVEL:  390\n",
      "TCGA-LUAD\n",
      "56  patients\n",
      "lncRNA:  10065\n",
      "PROTEIN CODING:  16087\n",
      "NOVEL:  431\n",
      "TCGA-BLCA\n",
      "18  patients\n",
      "lncRNA:  7942\n",
      "PROTEIN CODING:  15521\n",
      "NOVEL:  373\n",
      "GSE102101_KIRC\n",
      "         transcript_id   SRR5885319   SRR5885320   SRR5885321   SRR5885322  \\\n",
      "0       TCONS_00002128   918.580901   259.244976   743.867429   460.021130   \n",
      "1      ENST00000387314  2739.699672   347.541269  1844.771818   399.913980   \n",
      "2      ENST00000389680  4460.676619  1093.170912  3362.743728   897.972000   \n",
      "3       TCONS_00002129  8547.016124  2386.216764  6447.811071  1860.799667   \n",
      "4      ENST00000387342  1675.129075    55.297117  1637.638926   179.115200   \n",
      "...                ...          ...          ...          ...          ...   \n",
      "57817  ENST00000425340     0.290033     0.554147     0.284691     0.488158   \n",
      "57818  ENST00000318083     0.768010     2.113511     0.516160     1.961413   \n",
      "57819  ENST00000222145     4.884411    15.923205     4.564286    44.418950   \n",
      "57820  ENST00000597553     0.075058     0.251131     0.010810     0.057150   \n",
      "57821  ENST00000645652     1.206211     2.161318     0.857057     3.291924   \n",
      "\n",
      "        SRR5885323   SRR5885324   SRR5885325   SRR5885326    SRR5885327  ...  \\\n",
      "0       923.944413   177.298415   692.077866   624.661320    584.826593  ...   \n",
      "1      2053.274934    77.230563  3658.421912  1391.376020   2883.337777  ...   \n",
      "2      4442.686431  2758.408904  4077.609350  1990.561240   5795.766066  ...   \n",
      "3      7328.929933  5996.260342  7933.823765  3670.249934  10347.541760  ...   \n",
      "4      2054.837670   139.310342  2323.725917   809.874473   2850.285977  ...   \n",
      "...            ...          ...          ...          ...           ...  ...   \n",
      "57817     0.335331     0.678456     0.190318     0.201065      0.268940  ...   \n",
      "57818     0.584004     0.906701     0.772746     0.326323      1.037990  ...   \n",
      "57819     5.226093     2.178751     4.819890    29.358246     11.587327  ...   \n",
      "57820     0.108715     0.029786     0.112578     0.473820      0.217379  ...   \n",
      "57821     1.590090     0.252309     1.172961     4.587906      2.279800  ...   \n",
      "\n",
      "        SRR5885330   SRR5885331   SRR5885332   SRR5885333   SRR5885334  \\\n",
      "0       439.369036   797.556412   656.307230  1033.818397   166.641997   \n",
      "1       879.506075  2806.598495   309.802531  1943.082671   388.161550   \n",
      "2      1351.419336  4590.661419  1339.741749  4978.252390  1025.974976   \n",
      "3      2625.861869  8801.770102  2976.449629  9102.214452  2171.265484   \n",
      "4       740.021930  1989.880215   175.314813  2853.146667   884.812157   \n",
      "...            ...          ...          ...          ...          ...   \n",
      "57817     0.245418     1.014466     0.356033     0.509935     0.261629   \n",
      "57818     0.705571     1.038982     1.102326     0.742010     0.511899   \n",
      "57819    22.996611     7.701499    21.141202     7.019601     3.454505   \n",
      "57820     0.265973     0.076155     0.048094     0.052676     0.030630   \n",
      "57821     3.451275     2.145412     3.187480     1.913340     0.498158   \n",
      "\n",
      "        SRR5885335   SRR5885336   SRR5885337   SRR5885338  Length  \n",
      "0       361.809075   479.531540   384.086701   333.196033     577  \n",
      "1       876.287161   287.919837   247.171930   185.214178      71  \n",
      "2      3451.523793  1485.611644   957.617042   963.030804     954  \n",
      "3      6515.608146  3194.680002  1996.892435  2069.194113     353  \n",
      "4      1748.659573   398.057737   139.627210    83.524140      69  \n",
      "...            ...          ...          ...          ...     ...  \n",
      "57817     0.597710     0.478485     1.217722     0.572233    3116  \n",
      "57818     0.659238     0.841126     2.217711     1.268144    1858  \n",
      "57819    12.079384    36.050265    20.868438    14.432325    3199  \n",
      "57820     0.022695     0.097187     0.227208     0.200978    2218  \n",
      "57821     2.414571     2.436827     4.374586     1.897017    3273  \n",
      "\n",
      "[57822 rows x 22 columns]\n",
      "10  patients\n",
      "lncRNA:  6587\n",
      "PROTEIN CODING:  14516\n",
      "NOVEL:  307\n",
      "GSE133624_BLCA\n",
      "         transcript_id   SRR9623608    SRR9623609   SRR9623610   SRR9623611  \\\n",
      "0       TCONS_00002128   108.385307    852.984285   156.137544    93.721125   \n",
      "1      ENST00000387314   272.289135   1635.960214   608.499676   260.698700   \n",
      "2      ENST00000389680  2438.399309  24047.524097  4530.412218  1914.214171   \n",
      "3       TCONS_00002129    10.524498     34.156272    42.607543     7.196981   \n",
      "4      ENST00000387342  3013.198276   9622.721675   859.475916  3154.190046   \n",
      "...                ...          ...           ...          ...          ...   \n",
      "57817  ENST00000425340    12.761823     46.937946     1.473971     3.610699   \n",
      "57818  ENST00000318083     0.925714      0.846434     2.336134     1.692908   \n",
      "57819  ENST00000222145    10.129523      2.270789    55.378069     3.933008   \n",
      "57820  ENST00000597553     0.465278      0.033764     0.318575     0.000000   \n",
      "57821  ENST00000645652     4.813620      2.356734     5.782694     2.328629   \n",
      "\n",
      "        SRR9623612   SRR9623613   SRR9623614   SRR9623615   SRR9623616  ...  \\\n",
      "0       255.205639   356.503370   181.451170   398.973718    33.944136  ...   \n",
      "1      1286.703070   581.121283   620.092498  2993.284486   382.848295  ...   \n",
      "2      3829.225827  4409.134668  2274.749398  6606.713308  2600.765992  ...   \n",
      "3        16.363727    33.438897    10.864212    35.722138     8.319126  ...   \n",
      "4      1172.022294   796.502440  1498.455104  1876.555015  8916.530847  ...   \n",
      "...            ...          ...          ...          ...          ...  ...   \n",
      "57817     2.566780     0.364915     5.858446     1.431195    23.031431  ...   \n",
      "57818     1.004424     4.808486     2.889717     2.814048     1.358484  ...   \n",
      "57819    22.668328    16.282871    25.966723     5.047466     5.849344  ...   \n",
      "57820     0.681131     0.219711     0.138325     0.069332     0.043769  ...   \n",
      "57821     2.253594     5.277337     2.694975     0.328890     2.128153  ...   \n",
      "\n",
      "        SRR9623664   SRR9623665   SRR9623666   SRR9623667   SRR9623668  \\\n",
      "0        69.414415    99.689929   159.526060    43.102901   162.225990   \n",
      "1       429.602783   533.860143   162.920359   535.969147   396.284772   \n",
      "2      3909.185525  8249.823162  6488.476439  3507.963709  5613.576329   \n",
      "3         5.243165    25.431387     9.528483     5.854370    13.802747   \n",
      "4      3521.419014  4205.539944   687.216127   636.192440  3268.137684   \n",
      "...            ...          ...          ...          ...          ...   \n",
      "57817     0.593979     2.080742     1.237414     1.120613     0.220234   \n",
      "57818     0.597687     1.878991     0.750616     1.073913     0.849501   \n",
      "57819     3.355691     1.091330     2.820935     5.301754     1.029696   \n",
      "57820     0.634191     0.037477     0.036987     0.032129     0.000000   \n",
      "57821     1.967893     1.599984     1.303383     1.219267     2.683773   \n",
      "\n",
      "        SRR9623669   SRR9623670   SRR9623671   SRR9623672  Length  \n",
      "0       122.081166   461.984536   265.348790    61.894582     577  \n",
      "1       465.118751   215.304699  1965.858193   362.504960      71  \n",
      "2      6837.079613  7151.824137  5390.182063  5380.131581     954  \n",
      "3        13.808953    16.164506    13.897252    11.432910     353  \n",
      "4      1958.182315  2897.446491  2848.488402  1649.862319      69  \n",
      "...            ...          ...          ...          ...     ...  \n",
      "57817     3.459229     0.542583     1.041114     1.270754    3116  \n",
      "57818     0.591223     1.289096     1.533094     1.188524    1858  \n",
      "57819     9.400198     3.435283    20.183100     2.808817    3199  \n",
      "57820     0.061908     0.730497     0.249717     0.240321    2218  \n",
      "57821     3.628919     5.660592     2.248272     2.326535    3273  \n",
      "\n",
      "[57822 rows x 67 columns]\n",
      "20  patients\n",
      "lncRNA:  8586\n",
      "PROTEIN CODING:  15619\n",
      "NOVEL:  425\n",
      "GSE22260_PRAD\n",
      "         transcript_id    SRR057629    SRR057630    SRR057631    SRR057632  \\\n",
      "0       TCONS_00002128   610.659219   442.216724   745.537895   584.418454   \n",
      "1      ENST00000387314   371.295495   322.201811   409.745342   364.339747   \n",
      "2      ENST00000389680  1163.286234  1842.723132  2093.818153  2072.393724   \n",
      "3       TCONS_00002129  2502.685052  3726.314150  4218.837323  4047.455557   \n",
      "4      ENST00000387342   149.095681   420.802031   300.715705   127.198323   \n",
      "...                ...          ...          ...          ...          ...   \n",
      "57817  ENST00000425340     0.619039     5.082625     1.578934     2.520161   \n",
      "57818  ENST00000318083     3.460577     2.841309     3.108505     0.745852   \n",
      "57819  ENST00000222145     7.838709     6.876046     6.686814     2.021579   \n",
      "57820  ENST00000597553     0.289890     0.396690     0.675103     0.208265   \n",
      "57821  ENST00000645652     3.536069     8.333524     7.712043     4.092879   \n",
      "\n",
      "         SRR057633    SRR057634    SRR057635    SRR057636    SRR057637  ...  \\\n",
      "0      1034.804786   670.227377   791.245139   375.762416   718.991046  ...   \n",
      "1       239.802076   334.653697   552.292252   393.747574   450.378472  ...   \n",
      "2      2350.868176  1574.236829  2014.072060  1550.510623  2263.396568  ...   \n",
      "3      4483.926357  2938.427491  4086.314489  3176.627024  4467.321997  ...   \n",
      "4       147.484468  1187.426912   216.495514   283.612383   469.530722  ...   \n",
      "...            ...          ...          ...          ...          ...  ...   \n",
      "57817     1.695736     2.629411     2.397014     3.289654     1.957911  ...   \n",
      "57818     1.474600     1.763885     0.000000     2.006177     2.717427  ...   \n",
      "57819     4.649341     4.866260     2.918527     2.524602     6.115910  ...   \n",
      "57820     0.176466     0.000000     0.000000     0.140046     0.474243  ...   \n",
      "57821     3.467960     3.003940     4.564067     3.416569     4.306468  ...   \n",
      "\n",
      "         SRR057650    SRR057651    SRR057652    SRR057653    SRR057654  \\\n",
      "0       621.441281   964.951560   672.740914   863.537697   642.222866   \n",
      "1       220.588012   252.965493   326.113943   987.424094   286.733358   \n",
      "2      1839.559935  2391.759125  1394.841895  2268.927718  1657.063234   \n",
      "3      3775.912919  4625.816819  2496.366636  4624.629317  3297.648096   \n",
      "4       489.802978   390.446739   868.525112   489.878879   301.674683   \n",
      "...            ...          ...          ...          ...          ...   \n",
      "57817     5.555313     2.401657    17.484022     2.008846     2.349087   \n",
      "57818     3.105553     2.416651     2.932197     1.347592     4.801373   \n",
      "57819     5.668852     2.339344    12.347031     6.261520    11.440692   \n",
      "57820     1.114927     0.337401     0.614069     0.564433     0.515650   \n",
      "57821     6.044382     3.201035     2.912935     4.972461     3.773934   \n",
      "\n",
      "         SRR057655    SRR057656    SRR057657    SRR057658  Length  \n",
      "0      1071.063314   649.757581   585.535908   505.191667     577  \n",
      "1       263.245664   188.226455   301.759179   189.440147      71  \n",
      "2      1221.923217  1123.429087  1309.472301  1433.223410     954  \n",
      "3      2349.254364  1986.223289  2476.772792  2635.956763     353  \n",
      "4       518.197513   338.944014   417.988606   418.622668      69  \n",
      "...            ...          ...          ...          ...     ...  \n",
      "57817    10.366483     2.374189     3.349734     2.052112    3116  \n",
      "57818     1.968152     1.926622     3.252379     4.390930    1858  \n",
      "57819    18.099325    11.861343    14.940297    11.235021    3199  \n",
      "57820     0.732757     0.322783     1.238405     0.198824    2218  \n",
      "57821     6.331198     7.510039     5.874573     3.570511    3273  \n",
      "\n",
      "[57822 rows x 32 columns]\n",
      "10  patients\n",
      "lncRNA:  10545\n",
      "PROTEIN CODING:  14884\n",
      "NOVEL:  482\n",
      "PRJEB2449_PRAD\n",
      "         transcript_id    ERR031017    ERR031018    ERR031019    ERR031022  \\\n",
      "0       TCONS_00002128  1716.817397  1056.752537  1095.016818  1168.531177   \n",
      "1      ENST00000387314  1464.985458   869.071038  1379.944792  1104.468820   \n",
      "2      ENST00000389680  4519.450508  5019.290804  2942.696981  4297.255532   \n",
      "3       TCONS_00002129  7925.354590  9438.424680  4879.136628  8378.920745   \n",
      "4      ENST00000387342   721.647412   480.012532   920.429782   287.235960   \n",
      "...                ...          ...          ...          ...          ...   \n",
      "57817  ENST00000425340     0.375823     0.394686     7.134345     0.607075   \n",
      "57818  ENST00000318083     0.579860     1.095585     1.712705     0.971831   \n",
      "57819  ENST00000222145     1.691253     2.372955     5.324015     6.840542   \n",
      "57820  ENST00000597553     0.000000     0.133840     0.000000     0.000000   \n",
      "57821  ENST00000645652     0.379263     0.518280     1.903438     2.259277   \n",
      "\n",
      "         ERR031023    ERR031024    ERR031025    ERR031026    ERR031027  ...  \\\n",
      "0      2368.487087  1746.314363  1371.433155  1078.489284  1676.356585  ...   \n",
      "1      1234.575365   651.252560   654.606242   517.383102  1642.486716  ...   \n",
      "2      4789.007857  3190.419036  1940.900485  3132.292442  3728.102010  ...   \n",
      "3      8362.614528  5574.118098  3671.952382  6276.629085  6719.216691  ...   \n",
      "4       847.843201  1024.025325   321.282711   229.821782   627.101414  ...   \n",
      "...            ...          ...          ...          ...          ...  ...   \n",
      "57817     6.967323     1.121567     2.676709     0.771489     0.986759  ...   \n",
      "57818     0.688563     1.374540     1.565256     0.953358     1.113274  ...   \n",
      "57819     4.290070     2.843216     5.934948     2.557643     2.035905  ...   \n",
      "57820     0.017479     0.020201     0.000000     0.000000     0.012602  ...   \n",
      "57821     1.480606     0.821360     1.240626     1.262794     0.657598  ...   \n",
      "\n",
      "         ERR031041    ERR031042    ERR031043    ERR031044    ERR299295  \\\n",
      "0      1177.716157  1067.397132  1114.200160  1245.134949   991.632147   \n",
      "1       530.541244   580.810517   355.690071   435.003098   700.519115   \n",
      "2      2655.669420  4999.424088  2745.537477  3424.734293  2997.144244   \n",
      "3      4734.127696  6879.501947  5512.683348  6729.755289  4983.178550   \n",
      "4       316.058514   186.795720   196.535464   150.457776   303.052007   \n",
      "...            ...          ...          ...          ...          ...   \n",
      "57817     2.560141     0.144940     0.851225     0.368867     6.477851   \n",
      "57818     1.143251     0.149584     0.683623     0.518840     2.378680   \n",
      "57819     3.157724     0.792776     1.541498     0.996758     6.062337   \n",
      "57820     0.127692     0.000000     0.101059     0.016716     0.208182   \n",
      "57821     3.922812     1.316188     0.924533     2.220321     1.954937   \n",
      "\n",
      "         ERR299296    ERR299297    ERR299298    ERR299299  Length  \n",
      "0      1167.793835  1412.534636  1009.410290  1177.499293     577  \n",
      "1      1462.146952   501.506455  1332.384089  1395.077759      71  \n",
      "2      4166.086614  4693.259953  5234.234573  2928.272633     954  \n",
      "3      7875.582012  8203.813845  9666.968297  5159.563141     353  \n",
      "4       380.721415   416.587338   593.978848   383.467451      69  \n",
      "...            ...          ...          ...          ...     ...  \n",
      "57817     1.802772     0.914171     0.788904     1.750139    3116  \n",
      "57818     1.482048     0.603961     0.866825     0.478314    1858  \n",
      "57819     6.059914     0.634110     1.722355     3.055888    3199  \n",
      "57820     0.024830     0.000000     0.114652     0.109276    2218  \n",
      "57821     1.127371     0.672520     0.673366     1.468714    3273  \n",
      "\n",
      "[57822 rows x 30 columns]\n",
      "13  patients\n",
      "lncRNA:  6886\n",
      "PROTEIN CODING:  13721\n",
      "NOVEL:  335\n",
      "SRP238334_KIRC\n",
      "         transcript_id  SRR10744251  SRR10744252  SRR10744253  SRR10744254  \\\n",
      "0       TCONS_00002128    30.483816   100.570978    24.148639    20.214487   \n",
      "1      ENST00000387314     0.000000     2.233105     0.000000     0.000000   \n",
      "2      ENST00000389680     0.121698     0.166195     0.546469     0.092975   \n",
      "3       TCONS_00002129     0.000000     0.000000     0.000000     0.000000   \n",
      "4      ENST00000387342     0.000000     0.000000     0.000000     0.000000   \n",
      "...                ...          ...          ...          ...          ...   \n",
      "57817  ENST00000425340     0.055889     0.000000     0.015210     0.000000   \n",
      "57818  ENST00000318083     0.781082     2.560018     1.198873     0.286430   \n",
      "57819  ENST00000222145    12.684269    22.377467     5.066791     0.720895   \n",
      "57820  ENST00000597553     0.052344     0.536126     0.042736     0.000000   \n",
      "57821  ENST00000645652     2.039644     3.536261     0.709531     0.257449   \n",
      "\n",
      "       SRR10744255  SRR10744256  SRR10744257  SRR10744258  SRR10744259  ...  \\\n",
      "0        22.461645    74.257695    19.909971    26.004881    30.372973  ...   \n",
      "1         0.000000     0.417919     0.000000     0.000000     0.000000  ...   \n",
      "2         0.330007     2.052795     1.102435     4.370629     0.292364  ...   \n",
      "3         0.000000     0.084057     0.000000     0.000000     0.000000  ...   \n",
      "4         0.000000     0.000000     0.000000     0.000000     0.000000  ...   \n",
      "...            ...          ...          ...          ...          ...  ...   \n",
      "57817     0.016839     0.009523     0.000000     0.000000     0.000000  ...   \n",
      "57818     1.835642     0.383280     0.500738     0.427452     1.050813  ...   \n",
      "57819    11.334027     2.773365     8.269736    11.411410    14.967344  ...   \n",
      "57820     0.236570     0.013378     0.182376     0.051153     0.272461  ...   \n",
      "57821     2.292506     0.407959     2.150460     1.083276     3.195643  ...   \n",
      "\n",
      "       SRR10744431  SRR10744432  SRR10744433  SRR10744434  SRR10744435  \\\n",
      "0       258.681087   304.073423   221.452895   248.487427   168.715946   \n",
      "1         1.343713     1.096332     3.204584     1.686110     2.489314   \n",
      "2         1.800069     1.346282     2.051068     3.137154     1.370949   \n",
      "3         0.000000     0.000000     0.000000     0.000000     0.000000   \n",
      "4         0.000000     0.564055     0.000000     0.000000     0.000000   \n",
      "...            ...          ...          ...          ...          ...   \n",
      "57817     0.107161     0.012490     0.029207     0.012806     0.056721   \n",
      "57818     1.694467     1.110198     1.983807     0.880564     0.951245   \n",
      "57819     5.248840     2.810400     4.153638     3.280685     5.458593   \n",
      "57820     0.086027     0.105283     0.020516     0.215895     0.079685   \n",
      "57821     1.224245     0.570776     1.292996     1.207014     1.241995   \n",
      "\n",
      "       SRR10744436  SRR10744437  SRR10744438  SRR10744439  Length  \n",
      "0       377.560886     9.030744   142.691705    33.597419     577  \n",
      "1         0.531132    49.905674     2.973388     0.000000      71  \n",
      "2         0.751044   126.936672     1.814577     0.253214     954  \n",
      "3         0.000000   269.836592     0.000000     0.000000     353  \n",
      "4         0.000000    24.165748     0.000000     0.000000      69  \n",
      "...            ...          ...          ...          ...     ...  \n",
      "57817     0.012102     0.200670     0.054200    12.694645    3116  \n",
      "57818     0.872737     0.560898     0.886255     1.462660    1858  \n",
      "57819     4.055129     5.538141     3.629597     6.116563    3199  \n",
      "57820     0.153018     0.187944     0.152289     0.108912    2218  \n",
      "57821     1.255859     0.191045     1.135211     0.977927    3273  \n",
      "\n",
      "[57822 rows x 191 columns]\n",
      "61  patients\n",
      "lncRNA:  8972\n",
      "PROTEIN CODING:  15239\n",
      "NOVEL:  447\n",
      "GSE214846_LIHC\n",
      "         transcript_id  SRR21814873  SRR21814874  SRR21814875  SRR21814876  \\\n",
      "0       TCONS_00002128   254.814989   147.975031   100.908100   114.811310   \n",
      "1      ENST00000387314   411.567920   271.478678   320.436542   326.175557   \n",
      "2      ENST00000389680  2054.220214  4363.840515  2479.121544  3332.944976   \n",
      "3       TCONS_00002129  4171.863747  8312.998258  4867.715812  7093.542537   \n",
      "4      ENST00000387342   387.354788   391.936402  1519.424595   328.944066   \n",
      "...                ...          ...          ...          ...          ...   \n",
      "57817  ENST00000425340     0.034797     0.023520     0.089404     0.103635   \n",
      "57818  ENST00000318083     0.087536     0.000000    15.093694     0.049658   \n",
      "57819  ENST00000222145     2.474285     1.786976     4.412286     0.418206   \n",
      "57820  ENST00000597553     0.000000     0.000000     0.083734     0.000000   \n",
      "57821  ENST00000645652     0.463792     0.111960     0.624182     0.084569   \n",
      "\n",
      "       SRR21814877  SRR21814878  SRR21814879  SRR21814880  SRR21814881  ...  \\\n",
      "0       109.153627   161.653742   103.433218   229.794075   123.789341  ...   \n",
      "1       208.009131   261.783578   146.731303   403.706865   223.680612  ...   \n",
      "2      2404.514767  3857.662869  2299.517709  3011.419328  2002.371562  ...   \n",
      "3      5050.166261  6703.560983  4753.804848  6362.082143  4393.535123  ...   \n",
      "4       315.961420   268.547742   349.956053   789.183657   227.873918  ...   \n",
      "...            ...          ...          ...          ...          ...  ...   \n",
      "57817     0.000000     0.054724     0.000000     0.204871     0.126784  ...   \n",
      "57818     0.042057     0.030592     0.043466     0.171792     0.297675  ...   \n",
      "57819     0.708374     1.403673     3.079903     2.454534     2.074702  ...   \n",
      "57820     0.000000     0.000000     0.036411     0.000000     0.035623  ...   \n",
      "57821     0.143247     0.156297     0.518161     0.409592     0.144842  ...   \n",
      "\n",
      "        SRR21814994   SRR21814995  SRR21814996  SRR21814997  SRR21814998  \\\n",
      "0         51.870556    172.663104    67.932063   123.137563   229.365579   \n",
      "1         75.228604    243.470816   320.881401   372.747494  1312.527032   \n",
      "2       8572.882153   4814.915927   850.506008  2671.315971  3461.945700   \n",
      "3      16159.349602  10040.989536  1873.943517  5886.305359  7052.207912   \n",
      "4        384.376436   1900.139454   379.579665   370.881891  1559.000227   \n",
      "...             ...           ...          ...          ...          ...   \n",
      "57817      4.905960      0.428803     1.986190     0.102021     0.800359   \n",
      "57818      0.446077      0.853970     1.834455     0.556066     0.984324   \n",
      "57819      3.972631      4.202876     2.243081     1.565147     2.988440   \n",
      "57820      0.083039      0.414158     0.121319     0.143327     0.149920   \n",
      "57821      0.872226      0.280661     0.383664     0.339947     1.422339   \n",
      "\n",
      "       SRR21814999  SRR21815000  SRR21815001   SRR21815002  Length  \n",
      "0        36.225361   104.594923    45.963096    245.344641     577  \n",
      "1       120.777368   151.788910   224.855876    203.291312      71  \n",
      "2      1284.481232  1868.037036  1306.118440   5005.379092     954  \n",
      "3      2998.584701  4248.291682  2899.651902  11003.067425     353  \n",
      "4       671.102069   422.452944  3025.555384    822.308784      69  \n",
      "...            ...          ...          ...           ...     ...  \n",
      "57817     6.226372     7.114844     1.931804      0.365093    3116  \n",
      "57818     6.057557     8.949080     0.375626      1.951666    1858  \n",
      "57819     9.616601     1.122956     7.008586     29.027529    3199  \n",
      "57820     2.223055     0.000000     0.196662      0.000000    2218  \n",
      "57821    17.815861     0.689899     5.917230      2.541681    3273  \n",
      "\n",
      "[57822 rows x 132 columns]\n",
      "65  patients\n",
      "lncRNA:  10739\n",
      "PROTEIN CODING:  15963\n",
      "NOVEL:  515\n",
      "GSE229705_LUAD\n",
      "         transcript_id    SRR24166098    SRR24166099    SRR24166100  \\\n",
      "0       TCONS_00002128    1409.415912    1360.110168     853.023972   \n",
      "1      ENST00000387314    1110.569326    3949.909747    2292.361317   \n",
      "2      ENST00000389680   44801.585872   85546.677735   74889.765230   \n",
      "3       TCONS_00002129  109358.192159  199266.893079  179249.695809   \n",
      "4      ENST00000387342    4740.667364    8567.356268    9720.998707   \n",
      "...                ...            ...            ...            ...   \n",
      "57817  ENST00000425340       1.553321       0.245235       4.579991   \n",
      "57818  ENST00000318083       0.189457       0.000000       0.293167   \n",
      "57819  ENST00000222145       1.678076       4.299696       1.089749   \n",
      "57820  ENST00000597553       0.277736       0.137809       0.098233   \n",
      "57821  ENST00000645652       2.097223       0.980579       2.429784   \n",
      "\n",
      "         SRR24166101   SRR24166102    SRR24166103    SRR24166104  SRR24166105  \\\n",
      "0         429.052602    640.860710     982.191351     803.426650    54.000297   \n",
      "1        1503.370014   1585.352197    4542.432841    1552.219479    24.455897   \n",
      "2       95232.404141  34762.724193  118318.645932   86672.141409   410.430960   \n",
      "3      226065.959923  82446.989545  284681.984299  198131.629854   860.532727   \n",
      "4       13977.761680   1899.258514    8909.089182    4962.855101    13.048396   \n",
      "...              ...           ...            ...            ...          ...   \n",
      "57817       0.287859      1.091767       0.234300       3.536829     0.268302   \n",
      "57818       0.120690      0.557252       0.196469       0.142356     0.363431   \n",
      "57819       5.958295      1.710752       4.165038       1.322904     3.196400   \n",
      "57820       0.707707      0.000000       0.000000       0.000000     0.304443   \n",
      "57821       2.740509      2.259558       3.401685       1.643181     1.503120   \n",
      "\n",
      "         SRR24166106  ...    SRR24166335    SRR24166336    SRR24166337  \\\n",
      "0        1490.824494  ...    1631.938425     454.129653     748.092908   \n",
      "1        2749.751941  ...    3570.036608    2112.486650    1750.153016   \n",
      "2       46152.736178  ...   69505.111193   65032.120581   58148.390145   \n",
      "3      107454.488401  ...  166278.642319  151541.539077  140961.962335   \n",
      "4        4192.737710  ...    7637.046275    2532.013590    2668.579823   \n",
      "...              ...  ...            ...            ...            ...   \n",
      "57817       5.600960  ...       0.107034       2.572663       0.523655   \n",
      "57818       0.039802  ...       0.359006       0.000000       0.202663   \n",
      "57819       0.901568  ...       8.062503       3.290207      11.221501   \n",
      "57820       0.000000  ...       0.200491       0.027590       0.679077   \n",
      "57821       3.389169  ...       4.585471       1.888358       2.569378   \n",
      "\n",
      "         SRR24166338   SRR24166339    SRR24166340   SRR24166341  \\\n",
      "0         853.686699   1134.937323    2274.337275   1086.279971   \n",
      "1        2708.922040    984.581732    2682.814532   2398.523100   \n",
      "2       71438.503611  22974.785699   84335.583970  39700.235648   \n",
      "3      172901.693918  57345.626731  207453.744919  96935.296295   \n",
      "4        5781.009990   2579.940110    5869.162352   3250.524776   \n",
      "...              ...           ...            ...           ...   \n",
      "57817       0.879385      0.198827       0.047277      0.325696   \n",
      "57818       0.000000      0.055575       0.079288      0.054622   \n",
      "57819       1.794716      5.551824       0.276304      7.518729   \n",
      "57820       0.000000      0.232772       0.000000      0.045756   \n",
      "57821       2.192674      2.744699       0.000000      3.720879   \n",
      "\n",
      "         SRR24166342    SRR24166343  Length  \n",
      "0        2867.589002    1893.339036     577  \n",
      "1        3600.948822    2576.339114      71  \n",
      "2       89972.592602   50378.855859     954  \n",
      "3      217710.631374  123280.733375     353  \n",
      "4       16818.896887    5483.572357      69  \n",
      "...              ...            ...     ...  \n",
      "57817       0.859442       0.513192    3116  \n",
      "57818       0.045042       0.191258    1858  \n",
      "57819       0.444732      12.413616    3199  \n",
      "57820       0.000000       1.041398    2218  \n",
      "57821       0.613662       3.365741    3273  \n",
      "\n",
      "[57822 rows x 248 columns]\n",
      "123  patients\n",
      "lncRNA:  17050\n",
      "PROTEIN CODING:  16255\n",
      "NOVEL:  690\n",
      "TCGA_COAD\n",
      "         transcript_id  03fec96d-2ca4-40ba-9ba5-e1904285a66b  \\\n",
      "0       TCONS_00002128                            219.769525   \n",
      "1      ENST00000387314                             54.869871   \n",
      "2      ENST00000389680                            932.185335   \n",
      "3       TCONS_00002129                           1697.359829   \n",
      "4      ENST00000387342                            141.150756   \n",
      "...                ...                                   ...   \n",
      "57817  ENST00000425340                             29.255714   \n",
      "57818  ENST00000318083                              0.681444   \n",
      "57819  ENST00000222145                              5.449681   \n",
      "57820  ENST00000597553                              0.043911   \n",
      "57821  ENST00000645652                              4.433764   \n",
      "\n",
      "       1f04bbb9-7526-495d-a40d-f49c6bbf7402  \\\n",
      "0                                344.056523   \n",
      "1                                100.973138   \n",
      "2                               1652.586842   \n",
      "3                               3243.772572   \n",
      "4                                149.738085   \n",
      "...                                     ...   \n",
      "57817                             39.856865   \n",
      "57818                              0.964625   \n",
      "57819                              9.194863   \n",
      "57820                              0.047533   \n",
      "57821                              7.376405   \n",
      "\n",
      "       3f805d4a-24d4-4df9-90f5-5dcf9f979fd3  \\\n",
      "0                                214.356695   \n",
      "1                                 38.488557   \n",
      "2                                732.428004   \n",
      "3                               1528.406700   \n",
      "4                                154.972827   \n",
      "...                                     ...   \n",
      "57817                             37.634126   \n",
      "57818                              2.302072   \n",
      "57819                              4.196878   \n",
      "57820                              0.214270   \n",
      "57821                              8.748491   \n",
      "\n",
      "       417630a9-9bfc-4e9f-84d7-47f62f8f37f1  \\\n",
      "0                                508.031572   \n",
      "1                                 55.753943   \n",
      "2                               1422.713065   \n",
      "3                               2866.174650   \n",
      "4                                639.896143   \n",
      "...                                     ...   \n",
      "57817                             51.076124   \n",
      "57818                              0.710178   \n",
      "57819                              2.094108   \n",
      "57820                              0.000000   \n",
      "57821                              5.333985   \n",
      "\n",
      "       43ba1922-8387-40ec-9dca-308d2e60ed45  \\\n",
      "0                                368.040513   \n",
      "1                                 62.638264   \n",
      "2                               1584.026394   \n",
      "3                               3172.230322   \n",
      "4                                127.564944   \n",
      "...                                     ...   \n",
      "57817                             27.504333   \n",
      "57818                              3.490673   \n",
      "57819                              7.993770   \n",
      "57820                              0.041773   \n",
      "57821                             13.021729   \n",
      "\n",
      "       5e64bb5c-a7bd-4b23-a39c-d4ff5ffe2252  \\\n",
      "0                                274.361251   \n",
      "1                                 37.663314   \n",
      "2                               1293.320305   \n",
      "3                               3111.192463   \n",
      "4                                108.514012   \n",
      "...                                     ...   \n",
      "57817                             34.842192   \n",
      "57818                              1.295310   \n",
      "57819                              3.009298   \n",
      "57820                              0.000000   \n",
      "57821                              1.388928   \n",
      "\n",
      "       699eb7a7-c181-4061-b21d-45f86aa569b4  \\\n",
      "0                                363.474915   \n",
      "1                                120.347951   \n",
      "2                               1144.070853   \n",
      "3                               2150.834191   \n",
      "4                                238.040883   \n",
      "...                                     ...   \n",
      "57817                             40.340853   \n",
      "57818                              1.584056   \n",
      "57819                              1.988452   \n",
      "57820                              0.000000   \n",
      "57821                              9.137325   \n",
      "\n",
      "       abb984a8-1f9b-426b-ae7f-f6e3df4597f5  \\\n",
      "0                                329.246765   \n",
      "1                                184.730761   \n",
      "2                               1985.448742   \n",
      "3                               3973.604304   \n",
      "4                                190.085276   \n",
      "...                                     ...   \n",
      "57817                             22.328519   \n",
      "57818                              0.772094   \n",
      "57819                              2.370310   \n",
      "57820                              0.000000   \n",
      "57821                              3.944683   \n",
      "\n",
      "       abf49519-3682-4c89-9197-77219375a029  ...  \\\n",
      "0                                298.356826  ...   \n",
      "1                                 25.402561  ...   \n",
      "2                                538.238738  ...   \n",
      "3                                901.280001  ...   \n",
      "4                                365.944146  ...   \n",
      "...                                     ...  ...   \n",
      "57817                             28.767015  ...   \n",
      "57818                              1.358996  ...   \n",
      "57819                              5.694335  ...   \n",
      "57820                              0.040658  ...   \n",
      "57821                              6.667687  ...   \n",
      "\n",
      "       d48667cd-c1ce-4602-8cfa-84a84e474479  \\\n",
      "0                                 90.000655   \n",
      "1                                111.477120   \n",
      "2                                775.171072   \n",
      "3                               1476.222690   \n",
      "4                               1278.360732   \n",
      "...                                     ...   \n",
      "57817                             49.940708   \n",
      "57818                              0.804646   \n",
      "57819                              3.491331   \n",
      "57820                              0.019825   \n",
      "57821                              8.289164   \n",
      "\n",
      "       d58ca2b2-a17c-47f4-b5d0-84e24cd2574c  \\\n",
      "0                                129.851314   \n",
      "1                                291.374294   \n",
      "2                                936.020012   \n",
      "3                               1586.718240   \n",
      "4                                437.420565   \n",
      "...                                     ...   \n",
      "57817                             53.599898   \n",
      "57818                              0.522432   \n",
      "57819                              1.517158   \n",
      "57820                              0.000000   \n",
      "57821                              0.296571   \n",
      "\n",
      "       dee5c1f1-db2b-44cc-8b1f-5867a736055a  \\\n",
      "0                                494.743628   \n",
      "1                                245.341435   \n",
      "2                               1199.746881   \n",
      "3                               2737.099856   \n",
      "4                               2788.189879   \n",
      "...                                     ...   \n",
      "57817                             17.094532   \n",
      "57818                              1.647006   \n",
      "57819                              9.902302   \n",
      "57820                              0.015161   \n",
      "57821                              5.167988   \n",
      "\n",
      "       df187a91-a1f5-42ad-bb2f-8ca57fa5dffc  \\\n",
      "0                                129.706093   \n",
      "1                                153.495676   \n",
      "2                                972.432896   \n",
      "3                               1806.423326   \n",
      "4                                326.597776   \n",
      "...                                     ...   \n",
      "57817                             39.351754   \n",
      "57818                              1.226132   \n",
      "57819                              5.033136   \n",
      "57820                              0.013880   \n",
      "57821                              0.790104   \n",
      "\n",
      "       e1ac23ad-cdcf-4195-9ed8-1c40d9729a17  \\\n",
      "0                                767.962367   \n",
      "1                               1071.930197   \n",
      "2                               5180.185571   \n",
      "3                               8834.206160   \n",
      "4                                643.181958   \n",
      "...                                     ...   \n",
      "57817                             32.815997   \n",
      "57818                              0.628570   \n",
      "57819                              2.585966   \n",
      "57820                              0.000000   \n",
      "57821                              0.406382   \n",
      "\n",
      "       e730e97c-c9af-45a1-9bed-3ea3d5a31075  \\\n",
      "0                                270.094645   \n",
      "1                                414.648725   \n",
      "2                               1761.171932   \n",
      "3                               3511.529040   \n",
      "4                                808.950179   \n",
      "...                                     ...   \n",
      "57817                             22.550036   \n",
      "57818                              0.491834   \n",
      "57819                              2.123151   \n",
      "57820                              0.000000   \n",
      "57821                              0.392392   \n",
      "\n",
      "       e907b5dd-1f1f-4d20-b049-b300c812406e  \\\n",
      "0                                170.524189   \n",
      "1                                152.101016   \n",
      "2                               2579.424901   \n",
      "3                               5675.064743   \n",
      "4                                999.178061   \n",
      "...                                     ...   \n",
      "57817                             12.168515   \n",
      "57818                              0.885677   \n",
      "57819                              4.436761   \n",
      "57820                              0.200938   \n",
      "57821                              2.115852   \n",
      "\n",
      "       ec6de687-aed9-43a8-a158-12f183fbf917  \\\n",
      "0                                405.529165   \n",
      "1                               1430.716527   \n",
      "2                               2232.840077   \n",
      "3                               3646.729590   \n",
      "4                                917.437882   \n",
      "...                                     ...   \n",
      "57817                             38.500137   \n",
      "57818                              0.705738   \n",
      "57819                              4.360608   \n",
      "57820                              0.000000   \n",
      "57821                              1.372370   \n",
      "\n",
      "       ef895091-fa01-458e-9279-09ec9d38b17b  \\\n",
      "0                                485.962948   \n",
      "1                                371.320192   \n",
      "2                               1058.579409   \n",
      "3                               2221.952416   \n",
      "4                               1622.902029   \n",
      "...                                     ...   \n",
      "57817                             14.803324   \n",
      "57818                              2.159681   \n",
      "57819                              7.244790   \n",
      "57820                              0.016908   \n",
      "57821                              2.142630   \n",
      "\n",
      "       f27ac5c0-97f7-4f30-b762-992a5f393f6b  \n",
      "0                                211.954337  \n",
      "1                                325.669479  \n",
      "2                               1784.380315  \n",
      "3                               3242.845748  \n",
      "4                                434.928927  \n",
      "...                                     ...  \n",
      "57817                             74.020072  \n",
      "57818                              0.560719  \n",
      "57819                              2.750097  \n",
      "57820                              0.000000  \n",
      "57821                              0.627771  \n",
      "\n",
      "[57822 rows x 83 columns]\n",
      "40  patients\n",
      "lncRNA:  11074\n",
      "PROTEIN CODING:  15307\n",
      "NOVEL:  466\n",
      "SRP107326_COAD\n",
      "         transcript_id    SRR5580910   SRR5580911   SRR5580912   SRR5580913  \\\n",
      "0       TCONS_00002128   1464.098333   406.718016   220.271993    52.603563   \n",
      "1      ENST00000387314   6046.729136  2184.048669  3352.498424   229.420931   \n",
      "2      ENST00000389680  11513.474398  4143.026036  9456.536696  3313.010601   \n",
      "3       TCONS_00002129    111.414813    37.408662    14.878888    13.413998   \n",
      "4      ENST00000387342   2645.918118  2105.186021  3627.824381  2620.797771   \n",
      "...                ...           ...          ...          ...          ...   \n",
      "57817  ENST00000425340     33.749510    40.611071    48.056781    35.695914   \n",
      "57818  ENST00000318083      0.092033     0.731031     0.120291     0.713584   \n",
      "57819  ENST00000222145      1.247247     1.934231     0.547280     0.562474   \n",
      "57820  ENST00000597553      0.000000     0.068042     0.000000     0.000000   \n",
      "57821  ENST00000645652      0.121905     4.034604     0.125191     2.314765   \n",
      "\n",
      "        SRR5580914   SRR5580915   SRR5580916   SRR5580917   SRR5580920  ...  \\\n",
      "0       711.241543   243.006500   113.863233    60.490379   500.764527  ...   \n",
      "1      3112.528874   502.405070   918.519837   905.782253  1172.996874  ...   \n",
      "2      9085.677880  3520.153965  4499.347473  2410.699464  5493.512274  ...   \n",
      "3        67.123364    25.713695    50.538407    26.086225    16.519893  ...   \n",
      "4      3708.018034  2417.515598   917.075373  3244.908759   775.658203  ...   \n",
      "...            ...          ...          ...          ...          ...  ...   \n",
      "57817    44.908041    45.943766    59.249162    28.813308    56.102722  ...   \n",
      "57818     0.042938     1.071343     0.180409     1.079151     0.162742  ...   \n",
      "57819     0.748168     2.264965     1.327246     3.319602     0.783181  ...   \n",
      "57820     0.000000     0.035898     0.083959     0.167407     0.000000  ...   \n",
      "57821     0.109688     3.600394     0.068276     5.014303     0.118780  ...   \n",
      "\n",
      "        SRR5581286   SRR5581287   SRR5581289   SRR5581300   SRR5581301  \\\n",
      "0       168.378069   370.349160   145.883906   195.441181  2235.147437   \n",
      "1       738.637806   940.024808  2879.330426   145.006423   957.819149   \n",
      "2      4149.045718  4337.603113  5454.307359  1646.841287  6012.376343   \n",
      "3      2021.255076  2670.347580    23.418483    69.219685    21.405464   \n",
      "4      1358.325254  1391.749360  2265.795362   445.639062  2242.016587   \n",
      "...            ...          ...          ...          ...          ...   \n",
      "57817    36.402326    65.769720    28.749095    62.270548    36.600504   \n",
      "57818     0.973299     0.362384     0.139914     1.034348     0.162672   \n",
      "57819     1.988290     1.262849     1.934056     5.084970     1.810888   \n",
      "57820     0.056229     0.000000     0.000000     0.030945     0.000000   \n",
      "57821     8.497335     0.154287     0.301817     5.934645     1.492911   \n",
      "\n",
      "        SRR5581322   SRR5581323   SRR5581324   SRR5581325  Length  \n",
      "0       252.312472   176.623206   500.049133    80.846745     577  \n",
      "1       557.024647   224.667313  5545.011309   454.191441      71  \n",
      "2      2380.969757  1588.163895  9168.420742  2603.626575     954  \n",
      "3      3927.062353  2442.278557   169.390896    37.041745     353  \n",
      "4       655.461849  1318.908169  2545.549272  1778.515215      69  \n",
      "...            ...          ...          ...          ...     ...  \n",
      "57817    39.284228    27.061672    50.488576    26.028531    3116  \n",
      "57818     0.177677     1.210739     0.377630     4.707539    1858  \n",
      "57819     2.167110     2.429254     3.216837     2.209430    3199  \n",
      "57820     0.000000     0.030734     0.000000     0.199165    2218  \n",
      "57821     0.302588     4.082183     0.357285     8.961826    3273  \n",
      "\n",
      "[57822 rows x 210 columns]\n",
      "104  patients\n",
      "lncRNA:  8304\n",
      "PROTEIN CODING:  15455\n",
      "NOVEL:  422\n",
      "liver_adjacent_totalRNA_LIHC\n",
      "11  patients\n",
      "lncRNA:  5657\n",
      "PROTEIN CODING:  13889\n",
      "NOVEL:  360\n",
      "hcc_normal_totalRNA_LIHC\n",
      "50  patients\n",
      "lncRNA:  10528\n",
      "PROTEIN CODING:  15957\n",
      "NOVEL:  543\n",
      "GSE193567_LIHC\n",
      "9  patients\n",
      "lncRNA:  3916\n",
      "PROTEIN CODING:  13362\n",
      "NOVEL:  285\n",
      "LIHC_TCGA_LIHC\n",
      "47  patients\n",
      "lncRNA:  7115\n",
      "PROTEIN CODING:  15181\n",
      "NOVEL:  361\n"
     ]
    }
   ],
   "source": [
    "## filter patients we are interested in, from those not TCGA\n",
    "for proj in all_projects:\n",
    "    print(proj)\n",
    "    if proj in tcga_projects:\n",
    "        cancer_type = proj[5:]\n",
    "\n",
    "        fc = pd.read_csv(os.path.join(cancer_dir,\"featureCounts/table_of_counts_TPMs_\")+proj+\".csv\")\n",
    "        tumor_transcripts = list()\n",
    "        patients=pd.read_csv(os.path.join(\"/users/genomics/marta/TCGA_RNASeq\",proj,\"results/QC_patients1.csv\"))\n",
    "        print(len(patients),\" patients\")\n",
    "        # patients['normal'] = patients['normal'].str.replace('TCGA', cancer_type)\n",
    "        # patients['tumor'] = patients['tumor'].str.replace('TCGA', cancer_type)\n",
    "\n",
    "    elif proj in other_projects:\n",
    "        fc = pd.read_csv(os.path.join(cancer_dir,\"featureCounts/table_of_counts_TPMs_\")+proj+\".csv\")\n",
    "        print(fc)\n",
    "        tumor_transcripts = list()\n",
    "        patients=pd.read_csv(os.path.join(\"/users/genomics/marta/cancers_RNASeq\",proj,\"results/patients.csv\"))\n",
    "        print(len(patients),\" patients\")\n",
    "\n",
    "        samples_to_keep = patients.tumor.values.tolist() + patients.normal.values.tolist() + ['Length','transcript_id']\n",
    "        fc_selected = fc[fc.columns.intersection(samples_to_keep)]\n",
    "\n",
    "    elif proj in manuscript_projects:\n",
    "        fc = pd.read_csv(os.path.join(cancer_dir,\"featureCounts/table_of_counts_TPMs_\")+proj+\".csv\")\n",
    "        tumor_transcripts = list()\n",
    "\n",
    "        if \"liver_adjacent_totalRNA\" in proj:\n",
    "            patients=pd.read_csv(os.path.join(\"/projects_eg/projects/marta/liver_adjacent_totalRNA/results/clean_patients.csv\"))\n",
    "            # patients.rename(columns={'adjacent':'normal'}, inplace=True)\n",
    "        elif \"hcc_normal_totalRNA\" in proj:\n",
    "            patients=pd.read_csv(os.path.join(\"/projects_eg/projects/marta/hcc_normal_totalRNA/results/clean_patients.csv\"))\n",
    "            # patients.rename(columns={'adjacent':'normal'}, inplace=True)\n",
    "        elif \"GSE193567\" in proj:\n",
    "            patients=pd.read_csv(os.path.join(\"/projects_eg/projects/marta/GSE193567/results/clean_patients.csv\"))        \n",
    "            # patients.rename(columns={'adjacent':'normal'}, inplace=True)\n",
    "        elif \"LIHC_TCGA\" in proj:\n",
    "            patients=pd.read_csv(os.path.join(\"/projects_eg/projects/marta/LIHC_TCGA/results/clean_patients.csv\"))    \n",
    "            patients['normal'] = patients['patient'] + \"_normal\"\n",
    "            patients['tumor'] = patients['patient'] + \"_tumor\"\n",
    "        print(len(patients),\" patients\")\n",
    "\n",
    "    for index,patient in patients.iterrows():\n",
    "        if proj in tcga_projects:\n",
    "            patient_fc = fc[[\"transcript_id\",\"Length\",patient['tumor']]]\n",
    "\n",
    "            tumor_patient_fc = patient_fc[patient_fc[patient['tumor']] >= 1 ]\n",
    "            tumor_transcripts.extend(tumor_patient_fc.transcript_id.values.tolist())\n",
    "        if proj in other_projects:\n",
    "            patient_fc = fc[[\"transcript_id\",\"Length\",patient['tumor']]]\n",
    "\n",
    "            tumor_patient_fc = patient_fc[patient_fc[patient['tumor']] >= 1 ]\n",
    "            tumor_transcripts.extend(tumor_patient_fc.transcript_id.values.tolist())       \n",
    "        if proj in manuscript_projects:\n",
    "            patient_fc = fc[[\"transcript_id\",\"Length\",patient['tumor']]]\n",
    "\n",
    "            tumor_patient_fc = patient_fc[patient_fc[patient['tumor']] >= 1 ]\n",
    "            tumor_transcripts.extend(tumor_patient_fc.transcript_id.values.tolist())                        \n",
    "\n",
    "    tumor1FPKM = fc[fc['transcript_id'].isin(tumor_transcripts)]\n",
    "    # print(tumor1FPKM)\n",
    "    known = tumor1FPKM[tumor1FPKM['transcript_id'].str.contains('ENST')]\n",
    "    novel = tumor1FPKM[tumor1FPKM['transcript_id'].str.contains('TCONS')]\n",
    "    merged = tumor1FPKM.merge(transcript_gene, on=['transcript_id'], how=\"inner\")\n",
    "    merged = merged[merged['gene_type'].isin(['lncRNA','processed_pseudogene','novel','protein_coding'])]\n",
    "\n",
    "    lncRNA = merged[merged['gene_type'].isin(['lncRNA','processed_pseudogene'])]\n",
    "    print(\"lncRNA: \",len(lncRNA))\n",
    "\n",
    "    cds = merged[merged['gene_type'] == \"protein_coding\"]\n",
    "    print(\"PROTEIN CODING: \",len(cds))\n",
    "\n",
    "    novel = merged[merged['gene_type'] == \"novel\"]\n",
    "    ## length limitations\n",
    "    novel = novel[novel['Length'].astype(int) < 91667]\n",
    "    novel = novel[novel['Length'].astype(int) > 300]\n",
    "\n",
    "    print(\"NOVEL: \",len(novel))\n",
    "\n",
    "    # # tumor1FPKM = pd.concat([novel, lncRNA, cds], ignore_index = True)\n",
    "    # merged[['transcript_id','gene_id','gene_name','gene_type','Length']].to_csv(os.path.join(cancer_dir,\"tumorexpressed/tumor_1FPKM_\")+cancer_type+\".csv\",index=False)\n",
    "\n",
    "    ## keep only tumor samples\n",
    "    if proj in tcga_projects:\n",
    "        tumorONLY_merged = merged[merged.columns.drop(list(merged.filter(regex='normal')))]\n",
    "    else:\n",
    "        tumorONLY_merged = merged.drop(columns=patients.normal.values.tolist())\n",
    "\n",
    "    tumorONLY_merged = tumorONLY_merged[['transcript_id','gene_id','gene_id', 'gene_type', 'gene_name','Length', *patients.tumor.values.tolist()]]\n",
    "    tumorONLY_merged.to_csv(os.path.join(cancer_dir,\"tumorexpressed/tumor_1FPKM_table_of_counts_\")+proj+\".csv\",index=False)\n",
    "\n",
    "    ###### GET TUMOR-EXPRESSED > 1 TPM\n",
    "    # Filter only integer columns\n",
    "    int_columns = tumorONLY_merged.select_dtypes(include='int64')\n",
    "    \n",
    "    # Create a new column with the count of int columns per row whose value is > 1\n",
    "    # tumorONLY_merged['n'] = int_columns.apply(lambda row: (row > 1).sum(), axis=1)\n",
    "    tumorONLY_merged['n'] = tumorONLY_merged.drop(columns=['transcript_id','gene_id','gene_name','gene_type','Length']).apply(count_greater_than_one, axis=1)\n",
    "    tumorONLY_merged = tumorONLY_merged[['transcript_id','gene_id','gene_name','gene_type','Length','n']]\n",
    "    tumorONLY_merged.sort_values(by=['n'], ascending=False).to_csv(os.path.join(cancer_dir,\"tumorexpressed/tumor_1FPKM_n_\")+proj+\".csv\",index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "BRCA\n",
      "TCGA-BRCA\n",
      "BLCA\n",
      "TCGA-BLCA\n",
      "GSE133624_BLCA\n",
      "LUAD\n",
      "TCGA-LUAD\n",
      "GSE229705_LUAD\n",
      "KIRC\n",
      "TCGA-KIRC\n",
      "GSE102101_KIRC\n",
      "SRP238334_KIRC\n",
      "KIRP\n",
      "TCGA-KIRP\n",
      "PRAD\n",
      "TCGA-PRAD\n",
      "GSE22260_PRAD\n",
      "PRJEB2449_PRAD\n",
      "LUSC\n",
      "TCGA-LUSC\n",
      "LIHC\n",
      "GSE214846_LIHC\n",
      "liver_adjacent_totalRNA_LIHC\n",
      "hcc_normal_totalRNA_LIHC\n",
      "GSE193567_LIHC\n",
      "LIHC_TCGA_LIHC\n",
      "COAD\n",
      "TCGA_COAD\n",
      "SRP107326_COAD\n"
     ]
    }
   ],
   "source": [
    "## create a merged table of counts per ctype and a big patients file\n",
    "## CAN THEY BE CONSIDERED AS ONE?\n",
    "for ctype in cancertypes:\n",
    "    print(ctype)\n",
    "    \n",
    "    # Lists to store DataFrames for counts and patients\n",
    "    fc_list = []\n",
    "    patients_list = []\n",
    "    \n",
    "    for proj in all_projects:\n",
    "        if proj in tcga_projects and ctype in proj:\n",
    "            print(proj)\n",
    "            # Read the CSV files for TCGA projects\n",
    "            patients = pd.read_csv(os.path.join(\"/users/genomics/marta/TCGA_RNASeq\", proj, \"results/QC_patients1.csv\"))\n",
    "            patients['project'] = proj  # Add the project column\n",
    "            fc = pd.read_csv(os.path.join(cancer_dir, \"featureCounts\", f\"table_of_counts_TPMs_{proj}.csv\"))\n",
    "            fc_list.append(fc)\n",
    "            patients_list.append(patients)\n",
    "        \n",
    "        elif proj in other_projects and ctype in proj:\n",
    "            print(proj)\n",
    "            # Read the CSV files for other projects\n",
    "            patients = pd.read_csv(os.path.join(\"/users/genomics/marta/cancers_RNASeq\", proj, \"results/patients.csv\"))\n",
    "            patients['project'] = proj  # Add the project column\n",
    "            fc = pd.read_csv(os.path.join(cancer_dir, \"featureCounts\", f\"table_of_counts_TPMs_{proj}.csv\"))\n",
    "            fc_list.append(fc)\n",
    "            patients_list.append(patients)\n",
    "        \n",
    "        elif proj in manuscript_projects and ctype in proj:\n",
    "            print(proj)\n",
    "            # Read the CSV files for manuscript projects\n",
    "            if proj == \"LIHC_TCGA_LIHC\":\n",
    "                patients = pd.read_csv(os.path.join(\"/projects_eg/projects/marta\", proj[:-5], \"results/clean_patients.csv\"))\n",
    "                patients['normal'] = patients['patient'] + \"_normal\"\n",
    "                patients['tumor'] = patients['patient'] + \"_tumor\"\n",
    "\n",
    "            else:\n",
    "                patients = pd.read_csv(os.path.join(\"/projects_eg/projects/marta\", proj[:-5], \"results/clean_patients.csv\"))\n",
    "\n",
    "            patients['project'] = proj  # Add the project column\n",
    "            fc = pd.read_csv(os.path.join(cancer_dir, \"featureCounts\", f\"table_of_counts_TPMs_{proj}.csv\"))\n",
    "            fc_list.append(fc)\n",
    "            patients_list.append(patients)\n",
    "    \n",
    "    # Merge all fc DataFrames by 'transcript_id' and 'Length'\n",
    "    if fc_list:\n",
    "        merged_fc = reduce(lambda left, right: pd.merge(left, right, on=[\"transcript_id\", \"Length\"], how='outer'), fc_list)\n",
    "        # Save the merged fc DataFrame\n",
    "        merged_fc.to_csv(os.path.join(cancer_dir, f\"merged_fc_{ctype}.csv\"), index=False)\n",
    "    \n",
    "    # Combine all patient DataFrames into one big DataFrame\n",
    "    if patients_list:\n",
    "        merged_patients = pd.concat(patients_list, ignore_index=True)\n",
    "        # Save the merged patients DataFrame\n",
    "        merged_patients.to_csv(os.path.join(cancer_dir, f\"merged_patients_{ctype}.csv\"), index=False)\n"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Here to run Q3.1 - TPM distribution to know if we can treat them as one per cancer type or if there is bias according to the dataset"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "BRCA\n",
      "TCGA-BRCA\n",
      "109\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_1161236/3785118530.py:54: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  percent10['ctype'] = ctype\n",
      "/tmp/ipykernel_1161236/3785118530.py:62: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  percent10['ctype'] = ctype\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "BLCA\n",
      "TCGA-BLCA\n",
      "GSE133624_BLCA\n",
      "38\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_1161236/3785118530.py:54: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  percent10['ctype'] = ctype\n",
      "/tmp/ipykernel_1161236/3785118530.py:62: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  percent10['ctype'] = ctype\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "LUAD\n",
      "TCGA-LUAD\n",
      "GSE229705_LUAD\n",
      "179\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_1161236/3785118530.py:54: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  percent10['ctype'] = ctype\n",
      "/tmp/ipykernel_1161236/3785118530.py:62: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  percent10['ctype'] = ctype\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "KIRC\n",
      "TCGA-KIRC\n",
      "GSE102101_KIRC\n",
      "SRP238334_KIRC\n",
      "142\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_1161236/3785118530.py:54: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  percent10['ctype'] = ctype\n",
      "/tmp/ipykernel_1161236/3785118530.py:62: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  percent10['ctype'] = ctype\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "KIRP\n",
      "TCGA-KIRP\n",
      "32\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_1161236/3785118530.py:54: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  percent10['ctype'] = ctype\n",
      "/tmp/ipykernel_1161236/3785118530.py:62: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  percent10['ctype'] = ctype\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PRAD\n",
      "TCGA-PRAD\n",
      "GSE22260_PRAD\n",
      "PRJEB2449_PRAD\n",
      "75\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_1161236/3785118530.py:54: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  percent10['ctype'] = ctype\n",
      "/tmp/ipykernel_1161236/3785118530.py:62: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  percent10['ctype'] = ctype\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "LUSC\n",
      "TCGA-LUSC\n",
      "49\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_1161236/3785118530.py:54: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  percent10['ctype'] = ctype\n",
      "/tmp/ipykernel_1161236/3785118530.py:62: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  percent10['ctype'] = ctype\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "LIHC\n",
      "GSE214846_LIHC\n",
      "liver_adjacent_totalRNA_LIHC\n",
      "hcc_normal_totalRNA_LIHC\n",
      "GSE193567_LIHC\n",
      "LIHC_TCGA_LIHC\n",
      "182\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_1161236/3785118530.py:54: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  percent10['ctype'] = ctype\n",
      "/tmp/ipykernel_1161236/3785118530.py:62: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  percent10['ctype'] = ctype\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "COAD\n",
      "TCGA_COAD\n",
      "SRP107326_COAD\n",
      "144\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_1161236/3785118530.py:54: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  percent10['ctype'] = ctype\n",
      "/tmp/ipykernel_1161236/3785118530.py:62: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  percent10['ctype'] = ctype\n"
     ]
    }
   ],
   "source": [
    "## Expressed in > 10% of the patients\n",
    "df_pancancer = pd.DataFrame(columns = ['transcript_id','gene_id','gene_name','gene_type','Length','n'])\n",
    "\n",
    "## group by cancertype\n",
    "for ctype in cancertypes:\n",
    "    print(ctype)\n",
    "    ctype_df = pd.DataFrame(columns = ['transcript_id','gene_id','gene_name','gene_type','Length','n'])\n",
    "\n",
    "    num_patients_ctype = 0\n",
    "    for proj in all_projects:\n",
    "        if proj in tcga_projects:\n",
    "            if ctype in proj:\n",
    "                print(proj)\n",
    "                fc = pd.read_csv(os.path.join(cancer_dir,\"featureCounts/table_of_counts_TPMs_\")+proj+\".csv\")\n",
    "                patients=pd.read_csv(os.path.join(\"/users/genomics/marta/TCGA_RNASeq\",proj,\"results/QC_patients1.csv\"))\n",
    "                ## count how many patients per cancer type (not dataset)\n",
    "                num_patients_ctype = num_patients_ctype + len(patients)\n",
    "                input = pd.read_csv(os.path.join(cancer_dir,\"tumorexpressed/tumor_1FPKM_n_\")+proj+\".csv\")\n",
    "                ctype_df = pd.concat([ctype_df, input])\n",
    "\n",
    "        elif proj in other_projects:\n",
    "            if ctype in proj:\n",
    "                print(proj)\n",
    "                fc = pd.read_csv(os.path.join(cancer_dir,\"featureCounts/table_of_counts_TPMs_\")+proj+\".csv\")\n",
    "                tumor_transcripts = list()\n",
    "                patients=pd.read_csv(os.path.join(\"/users/genomics/marta/cancers_RNASeq\",proj,\"results/patients.csv\"))\n",
    "                ## count how many patients per cancer type (not dataset)\n",
    "                num_patients_ctype = num_patients_ctype + len(patients)\n",
    "\n",
    "                input = pd.read_csv(os.path.join(cancer_dir,\"tumorexpressed/tumor_1FPKM_n_\")+proj+\".csv\")\n",
    "                ctype_df = pd.concat([ctype_df, input])\n",
    "        elif proj in manuscript_projects:\n",
    "            if ctype in proj:\n",
    "                print(proj)                \n",
    "                fc = pd.read_csv(os.path.join(cancer_dir,\"featureCounts/table_of_counts_TPMs_\")+proj+\".csv\")\n",
    "                tumor_transcripts = list()\n",
    "                patients=pd.read_csv(os.path.join(\"/projects_eg/projects/marta\",proj[:-5],\"results/clean_patients.csv\"))\n",
    "                ## count how many patients per cancer type (not dataset)\n",
    "                num_patients_ctype = num_patients_ctype + len(patients)\n",
    "\n",
    "                input = pd.read_csv(os.path.join(cancer_dir,\"tumorexpressed/tumor_1FPKM_n_\")+proj+\".csv\")\n",
    "                ctype_df = pd.concat([ctype_df, input])\n",
    "\n",
    "\n",
    "    print(num_patients_ctype)\n",
    "    ctype_df_summed = ctype_df\n",
    "    ctype_df_summed = ctype_df_summed.groupby([\"transcript_id\",\"gene_id\",\"gene_name\",\"gene_type\",\"Length\"])[\"n\"].sum().reset_index()\n",
    "    ctype_df_summed.drop_duplicates(inplace=True)\n",
    "    # print(ctype_df_summed)\n",
    "\n",
    "    percent10 = ctype_df_summed[ctype_df_summed['n'] > (num_patients_ctype*0.1)]\n",
    "    percent10.sort_values(by=['n'], ascending=False).to_csv(os.path.join(cancer_dir,\"tumorexpressed/cancertypes/tumor_1FPKM_n10percent_\")+ctype+\".csv\",index=False)\n",
    "\n",
    "    percent10['ctype'] = ctype\n",
    "    df_pancancer = pd.concat([df_pancancer, percent10])\n",
    "\n",
    "    df_pancancer.to_csv(os.path.join(cancer_dir,\"tumorexpressed/cancertypes/tumor_1FPKM_n10percent_pancancer.csv\"),index=False)\n",
    "\n",
    "    percent10 = ctype_df_summed[ctype_df_summed['n'] > (num_patients_ctype*0.05)]\n",
    "    percent10.sort_values(by=['n'], ascending=False).to_csv(os.path.join(cancer_dir,\"tumorexpressed/cancertypes/tumor_1FPKM_n5percent_\")+ctype+\".csv\",index=False)\n",
    "\n",
    "    percent10['ctype'] = ctype\n",
    "    df_pancancer = pd.concat([df_pancancer, percent10])\n",
    "\n",
    "    df_pancancer.to_csv(os.path.join(cancer_dir,\"tumorexpressed/cancertypes/tumor_1FPKM_n5percent_pancancer.csv\"),index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "BRCA\n",
      "TCGA-BRCA\n",
      "109\n",
      "109\n",
      "BLCA\n",
      "TCGA-BLCA\n",
      "18\n",
      "GSE133624_BLCA\n",
      "20\n",
      "38\n",
      "LUAD\n",
      "TCGA-LUAD\n",
      "56\n",
      "GSE229705_LUAD\n",
      "123\n",
      "179\n",
      "KIRC\n",
      "TCGA-KIRC\n",
      "71\n",
      "GSE102101_KIRC\n",
      "10\n",
      "SRP238334_KIRC\n",
      "61\n",
      "142\n",
      "KIRP\n",
      "TCGA-KIRP\n",
      "32\n",
      "32\n",
      "PRAD\n",
      "TCGA-PRAD\n",
      "52\n",
      "GSE22260_PRAD\n",
      "10\n",
      "PRJEB2449_PRAD\n",
      "13\n",
      "75\n",
      "LUSC\n",
      "TCGA-LUSC\n",
      "49\n",
      "49\n",
      "LIHC\n",
      "GSE214846_LIHC\n",
      "65\n",
      "liver_adjacent_totalRNA_LIHC\n",
      "11\n",
      "hcc_normal_totalRNA_LIHC\n",
      "50\n",
      "GSE193567_LIHC\n",
      "9\n",
      "LIHC_TCGA_LIHC\n",
      "47\n",
      "182\n",
      "COAD\n",
      "TCGA_COAD\n",
      "40\n",
      "SRP107326_COAD\n",
      "104\n",
      "144\n"
     ]
    }
   ],
   "source": [
    "#Select normal samples and create a file with min max expression in normal samples per cancer type\n",
    "tolerance_dictionary = dict()\n",
    "\n",
    "## group by cancertype\n",
    "for ctype in cancertypes:\n",
    "    print(ctype)\n",
    "    num_patients_ctype = 0\n",
    "    for proj in all_projects:\n",
    "        if proj in tcga_projects:\n",
    "            if ctype in proj:\n",
    "                print(proj)\n",
    "\n",
    "                # fc = pd.read_csv(os.path.join(cancer_dir,\"featureCounts/table_of_counts_TPMs_\")+proj+\".csv\")\n",
    "                patients=pd.read_csv(os.path.join(\"/users/genomics/marta/TCGA_RNASeq\",proj,\"results/QC_patients1.csv\"))\n",
    "                print(len(patients))\n",
    "                num_patients_ctype = num_patients_ctype + len(patients)\n",
    "\n",
    "        elif proj in other_projects:\n",
    "            if ctype in proj:\n",
    "                print(proj)\n",
    "                # fc = pd.read_csv(os.path.join(cancer_dir,\"featureCounts/table_of_counts_TPMs_\")+proj+\".csv\")\n",
    "                patients=pd.read_csv(os.path.join(\"/users/genomics/marta/cancers_RNASeq\",proj,\"results/patients.csv\"))\n",
    "                print(len(patients))\n",
    "                num_patients_ctype = num_patients_ctype + len(patients)\n",
    "\n",
    "        elif proj in manuscript_projects:\n",
    "            if ctype in proj:\n",
    "                print(proj)\n",
    "                # fc = pd.read_csv(os.path.join(cancer_dir,\"featureCounts/table_of_counts_TPMs_\")+proj+\".csv\")\n",
    "                patients=pd.read_csv(os.path.join(\"/projects_eg/projects/marta\",proj[:-5],\"results/clean_patients.csv\"))\n",
    "                print(len(patients))\n",
    "                num_patients_ctype = num_patients_ctype + len(patients)\n",
    "\n",
    "    print(num_patients_ctype)\n",
    "\n",
    "    tolerance_dictionary[ctype] = num_patients_ctype*0.05\n",
    "    # tolerance_dictionary[proj] = len(patients)*0.05"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'BRCA': 5.45,\n",
       " 'BLCA': 1.9000000000000001,\n",
       " 'LUAD': 8.950000000000001,\n",
       " 'KIRC': 7.1000000000000005,\n",
       " 'KIRP': 1.6,\n",
       " 'PRAD': 3.75,\n",
       " 'LUSC': 2.45,\n",
       " 'LIHC': 9.1,\n",
       " 'COAD': 7.2}"
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "tolerance_dictionary"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Function to count values above 0.1 in a row\n",
    "def count_values_above_01(row):\n",
    "    return sum(row > 0.1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "BRCA\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_2285107/3238322726.py:43: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  summary_tolerance_accepted.drop([\"counts_01\"], axis=1, inplace=True)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Done!\n",
      "BLCA\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_2285107/3238322726.py:43: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  summary_tolerance_accepted.drop([\"counts_01\"], axis=1, inplace=True)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Done!\n",
      "LUAD\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_2285107/3238322726.py:43: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  summary_tolerance_accepted.drop([\"counts_01\"], axis=1, inplace=True)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Done!\n",
      "KIRC\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_2285107/3238322726.py:43: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  summary_tolerance_accepted.drop([\"counts_01\"], axis=1, inplace=True)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Done!\n",
      "KIRP\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_2285107/3238322726.py:43: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  summary_tolerance_accepted.drop([\"counts_01\"], axis=1, inplace=True)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Done!\n",
      "PRAD\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_2285107/3238322726.py:43: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  summary_tolerance_accepted.drop([\"counts_01\"], axis=1, inplace=True)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Done!\n",
      "LUSC\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_2285107/3238322726.py:43: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  summary_tolerance_accepted.drop([\"counts_01\"], axis=1, inplace=True)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Done!\n",
      "LIHC\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_2285107/3238322726.py:43: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  summary_tolerance_accepted.drop([\"counts_01\"], axis=1, inplace=True)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Done!\n",
      "COAD\n",
      "Done!\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_2285107/3238322726.py:43: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  summary_tolerance_accepted.drop([\"counts_01\"], axis=1, inplace=True)\n"
     ]
    }
   ],
   "source": [
    "## TUMOR-SPECIFIC - per cancer type\n",
    "for ctype in cancertypes:\n",
    "    print(ctype)\n",
    "    tumorspecific_candidates = pd.DataFrame()\n",
    "\n",
    "    patients = pd.read_csv(cancer_dir+\"/merged_patients_\"+ctype+\".csv\")\n",
    "    fc=pd.read_csv(cancer_dir+\"/merged_fc_\"+ctype+\".csv\")\n",
    "\n",
    "    fc_info = fc.merge(transcript_gene, on=['transcript_id'], how=\"inner\")\n",
    "    fc_info = fc_info[fc_info['gene_type'].isin(['lncRNA','processed_pseudogene','novel','protein_coding'])]\n",
    "\n",
    "    ## select only normal samples\n",
    "    fc_normals = fc_info.drop(columns=patients.tumor.values.tolist())\n",
    "    \n",
    "    # Select only the columns with float data types\n",
    "    float_columns = fc_normals.select_dtypes(include=['float64'])\n",
    "\n",
    "    # Apply the function only to the float columns and store the result in a new column\n",
    "    fc_normals['counts_01'] = float_columns.apply(count_values_above_01, axis=1)\n",
    "    # fc_normals_info = pd.concat([fc_normals, fc_info[['gene_id','transcript_id','gene_name','gene_type']]], axis=1)\n",
    "    fc_normals_info = fc_normals[['gene_id','transcript_id','gene_name','gene_type','counts_01']]\n",
    "\n",
    "\n",
    "    for index,patient in patients.iterrows():\n",
    "        patient_fc = fc_info[[\"transcript_id\",\"gene_id\",\"gene_name\",\"gene_type\",\"Length\",patient['tumor'],patient['normal']]] \n",
    "        tumor1TPM = patient_fc[patient_fc[patient['tumor']] > 1 ]\n",
    "        tumor1TPM_normal01TPM = tumor1TPM[tumor1TPM[patient['normal']] < 0.1 ]\n",
    "\n",
    "        try:\n",
    "            os.makedirs(os.path.join(cancer_dir,\"tumorspecific/cancertypes\",ctype,patient['patient']))\n",
    "        except:\n",
    "            pass\n",
    "        tumor1TPM_normal01TPM.to_csv(os.path.join(cancer_dir,\"tumorspecific/cancertypes\",ctype,patient['patient'],\"tumorspecific_TPMs.csv\"), index=None)\n",
    "        tumorspecific_candidates = pd.concat([tumorspecific_candidates,tumor1TPM_normal01TPM[[\"transcript_id\",\"gene_id\",\"gene_name\",\"gene_type\",\"Length\"]]])\n",
    "\n",
    "\n",
    "    summary = tumorspecific_candidates.groupby([\"transcript_id\",\"gene_id\",\"gene_name\",\"gene_type\",\"Length\"]).size().reset_index(name='n')\n",
    "    summary.sort_values(by=\"n\", ascending=False).to_csv(os.path.join(cancer_dir,\"tumorspecific/cancertypes\")+\"/tumorspecific_n_\"+ctype+\".csv\", index=False)\n",
    "    ### tolerance 5% only\n",
    "    summary_tolerance = fc_normals_info.merge(summary, on=[\"gene_id\",\"transcript_id\",\"gene_name\",\"gene_type\"], how=\"right\")\n",
    "    # ## Expression > 0.1 in less than 5%\n",
    "    summary_tolerance_accepted = summary_tolerance[summary_tolerance['counts_01'] < tolerance_dictionary[ctype]]\n",
    "    summary_tolerance_accepted.drop([\"counts_01\"], axis=1, inplace=True)\n",
    "    summary_tolerance_accepted.sort_values(by=\"n\", ascending=False).to_csv(os.path.join(cancer_dir,\"tumorspecific/cancertypes\")+\"/tumorspecific_n_\"+ctype+\"_TOLERANCE5PERCENT.csv\", index=False)\n",
    "    print(\"Done!\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# ## TUMOR-SPECIFIC - per dataset\n",
    "\n",
    "# for proj in all_projects:\n",
    "#     tumorspecific_candidates = pd.DataFrame()\n",
    "\n",
    "#     if proj in tcga_projects:\n",
    "#         print(proj)\n",
    "#         fc = pd.read_csv(os.path.join(cancer_dir,\"featureCounts/table_of_counts_TPMs_\")+proj+\".csv\")\n",
    "#         patients=pd.read_csv(os.path.join(\"/users/genomics/marta/TCGA_RNASeq\",proj,\"results/QC_patients1.csv\"))\n",
    "\n",
    "#     elif proj in other_projects:\n",
    "#         print(proj)\n",
    "#         fc = pd.read_csv(os.path.join(cancer_dir,\"featureCounts/table_of_counts_TPMs_\")+proj+\".csv\")\n",
    "#         patients=pd.read_csv(os.path.join(\"/users/genomics/marta/cancers_RNASeq\",proj,\"results/patients.csv\"))\n",
    "\n",
    "#     elif proj in manuscript_projects:\n",
    "#         print(proj)\n",
    "#         fc = pd.read_csv(os.path.join(cancer_dir,\"featureCounts/table_of_counts_TPMs_\")+proj+\".csv\")\n",
    "\n",
    "#         if \"LIHC_TCGA\" in proj:\n",
    "#             patients=pd.read_csv(os.path.join(\"/projects_eg/projects/marta/LIHC_TCGA/results/clean_patients.csv\"))    \n",
    "#             patients['normal'] = patients['patient'] + \"_normal\"\n",
    "#             patients['tumor'] = patients['patient'] + \"_tumor\"\n",
    "        \n",
    "#             normals = [col for col in fc_info.columns if 'normal' in col]\n",
    "\n",
    "#     fc_info = fc.merge(transcript_gene, on=['transcript_id'], how=\"inner\")\n",
    "#     fc_info = fc_info[fc_info['gene_type'].isin(['lncRNA','processed_pseudogene','novel','protein_coding'])]\n",
    "\n",
    "#     ## keep only normal samples\n",
    "#     if proj in tcga_projects:\n",
    "#         fc_normals = fc_info[fc_info.columns.drop(list(fc_info.filter(regex='tumor')))]\n",
    "#     else:\n",
    "#         fc_normals = fc_info.drop(columns=patients.tumor.values.tolist())\n",
    "    \n",
    "#     # Select only the columns with float data types\n",
    "#     float_columns = fc_normals.select_dtypes(include=['float64'])\n",
    "\n",
    "#     # Apply the function only to the float columns and store the result in a new column\n",
    "#     fc_normals['counts_01'] = float_columns.apply(count_values_above_01, axis=1)\n",
    "\n",
    "#     fc_normals_info = pd.concat([fc_normals, fc_info[['gene_id','transcript_id','gene_name','gene_type']]], axis=1)\n",
    "#     fc_normals_info = fc_normals_info[['gene_id','transcript_id','gene_name','gene_type','counts_01']]\n",
    "\n",
    "\n",
    "#     for index,patient in patients.iterrows():\n",
    "#         if proj in tcga_projects:\n",
    "#             patient_fc = fc_info[[\"transcript_id\",\"gene_id\",\"gene_name\",\"gene_type\",\"Length\",patient['tumor'],patient['normal']]] \n",
    "#             tumor1TPM = patient_fc[patient_fc[patient['tumor']] > 1 ]\n",
    "#             tumor1TPM_normal01TPM = tumor1TPM[tumor1TPM[patient['normal']] < 0.1 ]\n",
    "#         else:\n",
    "#             patient_fc = fc_info[[\"transcript_id\",\"gene_id\",\"gene_name\",\"gene_type\",\"Length\",patient['tumor'],patient['adjacent']]] \n",
    "#             tumor1TPM = patient_fc[patient_fc[patient['tumor']] > 1 ]\n",
    "#             tumor1TPM_normal01TPM = tumor1TPM[tumor1TPM[patient['normal']] < 0.1 ]\n",
    "\n",
    "#         try:\n",
    "#             os.makedirs(os.path.join(cancer_dir,\"tumorspecific\",proj,patient['patient']))\n",
    "#         except:\n",
    "#             print(\"Directory exists\")\n",
    "#         tumor1TPM_normal01TPM.to_csv(os.path.join(cancer_dir,\"tumorspecific\",proj,patient['patient'],\"tumorspecific_TPMs.csv\"), index=None)\n",
    "#         tumorspecific_candidates = pd.concat([tumorspecific_candidates,tumor1TPM_normal01TPM[[\"transcript_id\",\"gene_id\",\"gene_name\",\"gene_type\",\"Length\"]]])\n",
    "\n",
    "\n",
    "# summary = tumorspecific_candidates.groupby([\"transcript_id\",\"gene_id\",\"gene_name\",\"gene_type\",\"Length\"]).size().reset_index(name='n')\n",
    "# summary.sort_values(by=\"n\", ascending=False).to_csv(os.path.join(cancer_dir,\"tumorspecific\")+\"/tumorspecific_n_\"+proj+\".csv\", index=False)\n",
    "# print(len(summary))\n",
    "# ### tolerance 5% only\n",
    "# summary_tolerance = fc_normals_info.merge(summary, on=[\"gene_id\",\"transcript_id\"], how=\"right\")\n",
    "# # ## Expression > 0.1 in less than 5%\n",
    "# summary_tolerance_accepted = summary_tolerance[summary_tolerance['counts_01'] < tolerance_dictionary[proj]]\n",
    "# summary_tolerance_accepted.drop([\"counts_01\"], axis=1, inplace=True)\n",
    "# summary_tolerance_accepted.sort_values(by=\"n\", ascending=False).to_csv(os.path.join(cancer_dir,\"tumorspecific\")+\"/tumorspecific_n_\"+proj+\"_TOLERANCE5PERCENT.csv\", index=False)\n",
    "# print(len(summary_tolerance_accepted))\n",
    "# print(\"Done!\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## GTEx"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [],
   "source": [
    "GTEx=pd.read_csv(os.path.join(GENOMEDIR,\"GTEx/GTEx_Analysis_2017-06-05_v8_RNASeQCv1.1.9_gene_median_tpm.gct\"), sep=\"\\t\", skiprows=2)\n",
    "GTEx.drop('Description', inplace=True, axis=1)\n",
    "GTEx['Name'] = GTEx['Name'].str[:-2]\n",
    "for ctype in cancertypes:\n",
    "    tumorspecific = pd.read_csv(os.path.join(cancer_dir,\"tumorspecific/cancertypes\")+\"/tumorspecific_n_\"+ctype+\"_TOLERANCE5PERCENT.csv\")\n",
    "    ## outfiles\n",
    "    output_full = os.path.join(os.path.join(cancer_dir,\"tumorspecific/cancertypes/GTEx\")+\"/tumorspecific_n_\"+ctype+\"_TOLERANCE5PERCENT_GTExfull.csv\")\n",
    "    output_shared = os.path.join(os.path.join(cancer_dir,\"tumorspecific/cancertypes/GTEx\")+\"/tumorspecific_n_\"+ctype+\"_TOLERANCE5PERCENT_inGTEx.csv\")\n",
    "    output_absent = os.path.join(os.path.join(cancer_dir,\"tumorspecific/cancertypes/GTEx\")+\"/tumorspecific_n_\"+ctype+\"_TOLERANCE5PERCENT_noGTEx.csv\")\n",
    "\n",
    "    full = GTEx[GTEx['Name'].isin(tumorspecific.gene_id.values.tolist())]\n",
    "    full.to_csv(output_full, index=False)\n",
    "\n",
    "    absent = tumorspecific[~tumorspecific['gene_id'].isin(GTEx.Name.values.tolist())]\n",
    "    absent.to_csv(output_absent, index=False)\n",
    "\n",
    "    shared = tumorspecific[tumorspecific['gene_id'].isin(GTEx.Name.values.tolist())]\n",
    "    shared.to_csv(output_shared, index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "BRCA\n",
      "BLCA\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_2285107/588818480.py:22: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  selected_in_GTEx.drop('gt', axis=1, inplace=True)\n",
      "/tmp/ipykernel_2285107/588818480.py:22: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  selected_in_GTEx.drop('gt', axis=1, inplace=True)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "LUAD\n",
      "KIRC\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_2285107/588818480.py:22: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  selected_in_GTEx.drop('gt', axis=1, inplace=True)\n",
      "/tmp/ipykernel_2285107/588818480.py:22: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  selected_in_GTEx.drop('gt', axis=1, inplace=True)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "KIRP\n",
      "PRAD\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_2285107/588818480.py:22: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  selected_in_GTEx.drop('gt', axis=1, inplace=True)\n",
      "/tmp/ipykernel_2285107/588818480.py:22: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  selected_in_GTEx.drop('gt', axis=1, inplace=True)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "LUSC\n",
      "LIHC\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_2285107/588818480.py:22: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  selected_in_GTEx.drop('gt', axis=1, inplace=True)\n",
      "/tmp/ipykernel_2285107/588818480.py:22: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  selected_in_GTEx.drop('gt', axis=1, inplace=True)\n",
      "/tmp/ipykernel_2285107/588818480.py:22: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  selected_in_GTEx.drop('gt', axis=1, inplace=True)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "COAD\n"
     ]
    }
   ],
   "source": [
    "##  MEDIAN TPM = 0.5\n",
    "for ctype in cancertypes:\n",
    "    print(ctype)\n",
    "    ### GTEx output\n",
    "    fullGTEx = pd.read_csv(os.path.join(os.path.join(cancer_dir,\"tumorspecific/cancertypes/GTEx\")+\"/tumorspecific_n_\"+ctype+\"_TOLERANCE5PERCENT_GTExfull.csv\"))\n",
    "    ### present in GTEx\n",
    "    inGTEx = pd.read_csv(os.path.join(os.path.join(cancer_dir,\"tumorspecific/cancertypes/GTEx\")+\"/tumorspecific_n_\"+ctype+\"_TOLERANCE5PERCENT_inGTEx.csv\"))\n",
    "    ### absent in GTEx\n",
    "    noGTEx = pd.read_csv(os.path.join(os.path.join(cancer_dir,\"tumorspecific/cancertypes/GTEx\")+\"/tumorspecific_n_\"+ctype+\"_TOLERANCE5PERCENT_noGTEx.csv\"))\n",
    "\n",
    "    to_save = fullGTEx[['Name','Testis','Ovary']]\n",
    "    fullGTEx.drop(['Testis','Ovary','Name'], axis=1, inplace=True)\n",
    "\n",
    "    # True False greater than 0\n",
    "    inGTEx_05 = fullGTEx.gt(0.5)\n",
    "\n",
    "    # count True columns (greater than 0)\n",
    "    fullGTEx['gt'] = inGTEx_05[(inGTEx_05 == True)].count(axis='columns')\n",
    "\n",
    "    # get rows with no grater than 0\n",
    "    selected_in_GTEx = fullGTEx.loc[fullGTEx['gt'] == 0]\n",
    "    selected_in_GTEx.drop('gt', axis=1, inplace=True)\n",
    "\n",
    "    selected_in_GTEx = pd.merge(to_save, selected_in_GTEx , left_index=True, right_index=True)\n",
    "    selected_in_GTEx.rename(columns={'Name':'gene_id'}, inplace=True)\n",
    "    selected_in_GTEx.to_csv(os.path.join(cancer_dir,\"tumorspecific/cancertypes/GTEx\")+\"/GTExvalues_\"+ctype+\"_TOLERANCE5PERCENT.csv\", index=False)\n",
    "\n",
    "    afterGTEx = pd.concat([inGTEx[inGTEx['gene_id'].isin(selected_in_GTEx.gene_id.values.tolist())], noGTEx])\n",
    "    afterGTEx.to_csv(os.path.join(cancer_dir,\"tumorspecific/cancertypes/GTEx\")+\"/tumorspecific_n_GTEx_\"+ctype+\"_TOLERANCE5PERCENT.csv\", index=False)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.16"
  },
  "orig_nbformat": 4
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
