---
title: "3. Testis-specific translated ORFs in tumors (pancancer)"
author: "Marta Espinosa"
date: "2024-07-10"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
# packages <- c("dplyr "tidyr "ggplot2 "stringr "purrr "ggvenn "ggpubr 
#               "ggside "bioseq "ggthemr "ggbeeswarm "ggbreak")
# 
# # Install packages that are not already installed
# installed_packages <- installed.packages()
# for (pkg in packages) {
#   if (!(pkg %in% installed_packages[, "Package"])) {
#     install.packages(pkg)
#   }
# }
# 
# # Load the packages
# lapply(packages, library, character.only = TRUE)

library(dplyr)
library(tidyr)
library(ggplot2)
library(stringr)
library(purrr)
library(ggvenn)
library(ggpubr)
library(ggside)
library(bioseq)
library(rcartocolor)
library(ggbeeswarm)
library(ggbreak)

annot = read.csv("/users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/human/newReference_Resconstructed/1transcript_1gene.reconstructed.csv")
# names(gene_transcript) = c("gene_id"gene_type"gene_name")
# gene_transcript$gene_id = gsub("\\..*"gene_transcript$gene_id)

pre_save_wd = "/projects_eg/projects/marta/TestisRestricted_Microproteins_TSA/with_TranscriptomeReconstruction/Q2_TestisRestricted/human"
plots_wd = "/projects_eg/projects/marta/TestisRestricted_Microproteins_TSA/with_TranscriptomeReconstruction/Q3_TestisRestrictedTumors/human/plots"
save_wd = "/projects_eg/projects/marta/TestisRestricted_Microproteins_TSA/with_TranscriptomeReconstruction/Q3_TestisRestrictedTumors/human"

cancers = c("BRCA","BLCA","LUAD","KIRC","KIRP","PRAD","LUSC","COAD","LIHC")
tcga_projects=c("TCGA-BRCA","TCGA-LUSC","TCGA-PRAD","TCGA-KIRC","TCGA-KIRP","TCGA-LUAD","TCGA-BLCA")#,"TCGA-LIHC"]
other_projects=c("GSE102101_KIRC","GSE133624_BLCA","GSE22260_PRAD","PRJEB2449_PRAD","SRP238334_KIRC","GSE214846_LIHC","GSE229705_LUAD","TCGA_COAD","SRP107326_COAD")
manuscript_projects = c("liver_adjacent_totalRNA_LIHC","hcc_normal_totalRNA_LIHC","GSE193567_LIHC","LIHC_TCGA_LIHC")
# deleted_projects=c("GSE103001_BRCA"GSE89223_PRAD")
all_projects = c(tcga_projects,other_projects,manuscript_projects)


cancers_dir = "/users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/cancers"
```

## Testis-Specific Translated ORFs

```{r input_translation_data}
print("Translated IN testis")
translated_IN_testis = read.csv(file.path(pre_save_wd,"testisRestricted_GTEx_translatedINtestis.csv"))
table(translated_IN_testis$gene_type)
print("Corresponding to how many genes?")
translated_IN_testis_genes = translated_IN_testis %>% select(gene_type, gene_name) %>% unique()
table(translated_IN_testis_genes$gene_type)

print("Translated ONLY in testis")
translated_only_testis = read.csv(file.path(pre_save_wd,"testisRestricted_GTEx_translatedONLYtestis.csv"))
table(translated_only_testis$gene_type)
print("Corresponding to how many genes?")
translated_only_testis_genes = translated_only_testis %>% select(gene_type, gene_name) %>% unique()
table(translated_only_testis_genes$gene_type)

## add column to translated IN testis
translated_IN_testis = translated_IN_testis %>% mutate(ONLYtranslation_testis = case_when(transcript_id %in% translated_only_testis$transcript_id ~ "ONLYTestis",
                                                                                          TRUE ~ "several_tissues"))
table(translated_IN_testis$ONLYtranslation_testis, translated_IN_testis$gene_type)
```

## Tumor Associated Antigens (TAA)

```{r cancer_TAA}
TAA_pancancer_10percent = read.csv(paste0(cancers_dir,"/tumorexpressed/cancertypes/tumor_1FPKM_n10percent_pancancer.csv"))
ggplot(TAA_pancancer_10percent %>% group_by(gene_type, ctype) %>% count(), aes(x=n, y=ctype, fill=gene_type)) +
  geom_bar(stat="identity") +
    scale_fill_manual(values=c("protein_coding"="#CC79A7",
                             "lncRNA" = "#009E73",
                             "processed_pseudogene" = "#0090B2",
                             "novel" = "#E69F00")) +
  ggtitle("Tumor-Expressed genes (> 1 TPM in > 10% patients)") +
  theme_classic()
```

```{r}
TAA_translated = merge(TAA_pancancer_10percent, translated_IN_testis, by=c("transcript_id","gene_name","gene_id","gene_type"))
table(TAA_translated$gene_type, TAA_translated$ONLYtranslation_testis)

TAA_translated_genes = TAA_translated %>% select(transcript_id, gene_name, gene_type, ONLYtranslation_testis) %>% unique()
table(TAA_translated_genes$gene_type, TAA_translated_genes$ONLYtranslation_testis)

```

## Is the expression in testis and in tumor correlated?

```{r stats_function, echo=F}
# Function to compute the required statistics
compute_stats <- function(row) {
  mean_value <- mean(row)
  median_value <- median(row)
  sd_value <- sd(row)
  q1_value <- quantile(row, 0.25)
  q3_value <- quantile(row, 0.75)
  
  return(c(mean = mean_value, median = median_value, sd = sd_value, Q1 = q1_value, Q3 = q3_value))
}

```

```{r TPMS_testis, echo=F}
## Load counts data
# raw_counts = read.table("/users/genomics/saraa/projectTestis/featureCounts/results_geneID/featureCounts_geneID.txt header=TRUE)
testis_TPMs = read.csv("/users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/human/featureCounts_gffcompare/table_of_counts_TPMs_withLength.csv", header=TRUE)

## For novels, remove those shorter than 300 pb
testis_TPMs = testis_TPMs %>%
  filter(!(startsWith(gene_id, "TCONS") & Length < 300))
names(testis_TPMs)[1] = "transcript_id"

testis_TPMs = merge(testis_TPMs, annot, by="transcript_id", all.x=T)
testis_TPMs = testis_TPMs %>% unique()
testis_TPMs = testis_TPMs[,!grepl("brain", names(testis_TPMs))]
testis_TPMs = testis_TPMs[,!grepl("liver", names(testis_TPMs))]
testis_TPMs = testis_TPMs %>% select(-Length)

## Select only numeric columns
numeric_df = testis_TPMs[, sapply(testis_TPMs, is.numeric)]
## Apply the function to each row
stats_per_row = t(apply(numeric_df, 1, compute_stats))
stats_df = as.data.frame(stats_per_row)

testis_TPMs_stats = cbind(testis_TPMs %>% select(transcript_id, gene_id, gene_type, gene_name), stats_df)
testis_TPMs_stats %>% head
names(testis_TPMs_stats) = c("transcript_id","gene_id","gene_type","gene_name","mean_testis","median_testis","sd_testis","Q1.25%_testis","Q3.75%_testis")

```


```{r TPMS_cancer, echo=F}
testis_tumor_TPMs = data.frame(transcript_id = character(),
                               stringsAsFactors = F)

for(i in 1:length(all_projects)) {
  TAA_tumor = read.csv(paste0(cancers_dir,"/tumorexpressed/tumor_1FPKM_table_of_counts_",all_projects[i],".csv"))
  TAA_tumor = TAA_tumor %>% select(-Length)
  
  ## Select only numeric columns
  numeric_df = TAA_tumor[, sapply(TAA_tumor, is.numeric)]
  ## Apply the function to each row
  stats_per_row = t(apply(numeric_df, 1, compute_stats))
  stats_df = as.data.frame(stats_per_row)

  TAA_tumor_stats = cbind(TAA_tumor %>% select(transcript_id), stats_df)
  names(TAA_tumor_stats) = c("transcript_id","mean_tumor","median_tumor","sd_tumor","Q1.25%_tumor","Q3.75%_tumor")
  TAA_tumor_stats_testis = merge(testis_TPMs_stats, TAA_tumor_stats, by="transcript_id")
  TAA_tumor_stats_testis$cancertype = cancers[i]
  
  ## rbind
  testis_tumor_TPMs = rbind(testis_tumor_TPMs, TAA_tumor_stats_testis)
}
```

```{r correlation_TPM, echo=F}
ggplot(testis_tumor_TPMs %>% drop_na(), aes(x=mean_tumor, y=mean_testis)) +
  geom_point(size=.03) +
  geom_smooth(method = "lm", se = FALSE, aes(color=gene_type)) +
  scale_x_continuous(trans="log10") +
  scale_y_continuous(trans="log10") +
  scale_color_manual(values=c("protein_coding"="#CC79A7",
                             "lncRNA" = "#009E73",
                             "processed_pseudogene" = "#0090B2",
                             "novel" = "#E69F00")) +
  labs(title="Correlation TPMs between tumor and testis") +
  theme_classic() +
  theme(legend.position = "top") +
  facet_wrap(~ cancertype, nrow=2) +
  stat_cor(label.y = c(5.2, 4.8, 4.3, 3.8), aes(group=gene_type, color = gene_type), size=3) 
ggsave(file.path(plots_wd,"PNG/correlationTPM_testis_tumor.cancertype.png"), width=12.52, height=5.64)
ggsave(file.path(plots_wd,"PDF/correlationTPM_testis_tumor.cancertype.pdf"), width=12.52, height=5.64)


# ggplot(testis_tumor_TPMs %>% drop_na(), aes(x=mean_tumor, y=mean_testis)) +
#   geom_point(size=.03) +
#   geom_smooth(method = "lm se = FALSE, aes(color=gene_type)) +
#   scale_x_continuous(trans="log10") +
#   scale_y_continuous(trans="log10") +
#   scale_color_manual(values=c("protein_coding"="#CC79A7
#                              "lncRNA" = "#
#                              "processed_pseudogene" = "#0090B2
#                              "novel" = "#E69F00")) +
#   labs(title="Correlation TPMs between tumor and testis") +
#   theme_classic() +
#   theme(legend.position = "top") +
#   # facet_wrap(~ cancertype, nrow=2) +
#   stat_cor(label.y = c(5.2, 4.8, 4.3, 3.8), aes(group=gene_type, color = gene_type), size=3) 
```

## Phylistratigraphy

```{r phylostratigraphy, echo=F}
# all_genes_tumorexpressed = read.csv("/users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/Phylostratigraphy/all_genes_phyostratum.tumorexpressed.csv")
# all_genes_tumorexpressed$Phylostratum = replace(all_genes_tumorexpressed$Phylostratum, all_genes_tumorexpressed$Phylostratum == " "NA") 
# all_genes_tumorexpressed = all_genes_tumorexpressed %>% mutate(ORFtype = case_when(gene_type == "protein_coding" ~ "CDS
#                                                                                   TRUE ~ "ncORF"))
# 
# ggplot(all_genes_tumorexpressed, aes(x=factor(Phylostratum, levels = unique(Phylostratum)), fill=ORFtype)) +
#   geom_bar(stat="count position="dodge") +
#   scale_y_break(c(2000, 6500)) +  # Make sure this is a valid range
#   scale_fill_manual(values=c("CDS"="#CC79A7"ncORF" = "#FD8F24"), labels = c("ncORFs"CDS")) +
#   labs(x="Phylostratum
#        y="Number of genes
#        fill="Gene type
#        title="Evolutionary ages of all translated ORFs in testis reactivated in cancer") +
#   theme_classic() +
#   theme(axis.text.x = element_text(angle=90))
# ggsave(file.path(plots_wd,"PNG/evoAge_translatedin2.tumorexpressed.png"), width=7.97, height=4.59)
# ggsave(file.path(plots_wd,"PDF/evoAge_translatedin2.tumorexpressed.pdf"), width=7.97, height=4.59)
```

```{r novel_phyl, echo=F}
# novels_genes = all_genes_tumorexpressed %>% subset(gene_type == "novel")
# table(novels_genes$Phylostratum)
```

```{r by_biotype, echo=F}
# ggplot(all_genes_tumorexpressed, aes(x=factor(Phylostratum, levels = unique(Phylostratum)), fill=gene_type)) +
#   geom_bar() +
#     labs(x="Phylostratum
#        y="Number of genes
#        title="Evolutionary ages of all translated ORFs in testis reactivated in tumor") +
#     scale_fill_manual(values=c("protein_coding"="#CC79A7
#                         "lncRNA" = "#
#                         "processed_pseudogene" = "#0090B2
#                         "novel" = "#E69F00")) +
#     theme_classic() +
#   theme(axis.text.x = element_text(angle = 90),
#         legend.position = "none") +
#   facet_wrap(~ gene_type, nrow=4, scales = "free_y")
# ggsave(file.path(plots_wd,"PNG/evoAge_translatedin2_biotype.tumorexpressed.png"), width=11.48, height=6.84)
# ggsave(file.path(plots_wd,"PDF/evoAge_translatedin2_biotype.tumorexpressed.pdf"), width=11.48, height=6.84)
```

```{r comparison, echo=F}
# all_genes_tumorexpressed = read.csv("/users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/Phylostratigraphy/all_genes_phyostratum.tumorexpressed.csv")
# all_genes_tumorexpressed$Phylostratum = replace(all_genes_tumorexpressed$Phylostratum, all_genes_tumorexpressed$Phylostratum == " "NA") 
# all_genes_tumorexpressed = all_genes_tumorexpressed%>% mutate(ORFtype = case_when(gene_type == "protein_coding" ~ "CDS
#                                                                                   TRUE ~ "ncORF"))
# 
# 
# all_genes = read.csv("/users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/Phylostratigraphy/all_genes_phyostratum.csv")
# all_genes$Phylostratum = replace(all_genes$Phylostratum, all_genes$Phylostratum == " "NA") 
# all_genes = all_genes %>% mutate(ORFtype = case_when(gene_type == "protein_coding" ~ "CDS
#                                                                              TRUE ~ "ncORF"))
# all_genes = all_genes %>% mutate(tumorexpressed = case_when(gene_name %in% all_genes_tumorexpressed$gene_name ~ "yes
#                                                             TRUE ~ "no"))
# 
# CDS = ggplot(all_genes %>% subset(ORFtype == "CDS"), aes(x=factor(Phylostratum, levels = unique(Phylostratum)), fill=tumorexpressed)) +
#   geom_bar(stat="count position="dodge") +
#     scale_y_break(c(2000, 6500)) +
#   # scale_fill_manual(values=c("CDS"="#CC79A7"ncORF" = "#FD8F24"), labels = c("ncORFs"CDS")) +
#   labs(x="
#        y="Number of genes
#        title="CDSs in testis") +
#   theme_classic() +
#   theme(axis.text.x = element_blank())
# 
# 
# ncORF = ggplot(all_genes %>% subset(ORFtype == "ncORF"), aes(x=factor(Phylostratum, levels = unique(Phylostratum)), fill=tumorexpressed)) +
#   geom_bar(stat="count position="dodge") +
#     # scale_y_break(c(2000, 6500)) +
#   # scale_fill_manual(values=c("CDS"="#CC79A7"ncORF" = "#FD8F24"), labels = c("ncORFs"CDS")) +
#   labs(x="Phylostratum
#        y="Number of genes
#        title="Translated ncORFs in testis") +
#   theme_classic() +
#   theme(axis.text.x = element_text(angle=90))
# 
# ggarrange(CDS, ncORF, nrow = 2)
```

```{r comparison_ratio, echo=F}
# normalized_df <- all_genes %>%
#   group_by(Phylostratum, tumorexpressed) %>%
#   summarise(count = n()) %>%
#   group_by(tumorexpressed) %>% 
#   mutate(total = sum(count)) %>%
#   mutate(proportion = count / total) %>%
#   ungroup()
# 
# # Plot the proportions
# ggplot(normalized_df, aes(x = factor(Phylostratum, levels = c("cellular organisms "Eukaryota "Opisthokonta "Metazoa "Eumetazoa
#     "Bilateria "Deuterostomia "Chordata "Vertebrata"Gnathostomata "Teleostomi "Euteleostomi "Sarcopterygii
#     "Dipnotetrapodomorpha "Tetrapoda "Amniota "Mammalia "Theria"Eutheria "Boreoeutheria "Euarchontoglires "Primates
#     "Simiiformes "Catarrhini "Hominoidea "Hominidae "Homininae "Homo sapiens"NA")), y = proportion, fill = tumorexpressed)) +
#   geom_bar(stat = "identity position = "dodge") +
#     scale_fill_manual(values=c("yes" = "#B1283A "no"="#A8A6A7")) +
#   labs(title = "Are younger ORFs more easily reactivated?
#        x = "Phylostratum
#        y = "Proportion
#        fill = "Tumor Expressed") +
#   theme_classic() +
#     theme(axis.text.x = element_text(angle=90))
# ggsave(file.path(plots_wd,"PNG/Translatedin2_reactivated_evoAge.png"), height=3.81, width=9.61)
# ggsave(file.path(plots_wd,"PDF/Translatedin2_reactivated_evoAge.pdf"), height=3.81, width=9.61)
# 
# #### STATISTICAL TEST - CHI-SQUARED ####
# # Create a contingency table
# contingency_table <- table(all_genes$Phylostratum, all_genes$tumorexpressed)
# # Perform Chi-squared test
# chi_test <- chisq.test(contingency_table)
# # View the test result
# print(chi_test)
# # Chi-squared/Fisher's Exact Test: A significant p-value (e.g., < 0.05) indicates that the distribution of tumor-expressed genes is not independent of Phylostratum, implying some Phylostratum categories may be more associated with tumor expression.
# 
# 
# #### LOGISTIC REGRESSION ####
# # Convert Phylostratum to a factor with ordered levels if not already
# all_genes$Phylostratum <- factor(all_genes$Phylostratum, levels = unique(all_genes$Phylostratum))
# all_genes$tumorexpressed_binary <- ifelse(all_genes$tumorexpressed == "yes 1, 0)
# 
# # Fit logistic regression model
# logistic_model <- glm(tumorexpressed_binary ~ Phylostratum, data = all_genes, family = binomial)
# # Error in eval(family$initialize) : y values must be 0 <= y <= 1
# 
# # View summary of the model
# summary(logistic_model)
# # Logistic Regression: The regression coefficients will tell you how each Phylostratum level influences the odds of tumor expression. If certain Phylostratum levels have significantly positive or negative coefficients, it indicates a higher or lower likelihood of tumor expression for those levels.
# 

```
