---
title: "3. Testis-specific translated ORFs in tumors"
author: "Marta Espinosa"
date: "2024-07-10"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
# packages <- c("dplyr "tidyr "ggplot2 "stringr "purrr "ggvenn "ggpubr 
#               "ggside "bioseq "ggthemr "ggbeeswarm "ggbreak")
# 
# # Install packages that are not already installed
# installed_packages <- installed.packages()
# for (pkg in packages) {
#   if (!(pkg %in% installed_packages[, "Package"])) {
#     install.packages(pkg)
#   }
# }
# 
# # Load the packages
# lapply(packages, library, character.only = TRUE)

library(dplyr)
library(tidyr)
library(ggplot2)
library(stringr)
library(purrr)
library(ggvenn)
library(ggpubr)
library(ggside)
library(bioseq)
# library(ggthemr)
library(ggbeeswarm)
library(ggbreak)
library(GGally)

annot = read.csv("/users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/human/newReference_Resconstructed/transID_geneID_isoforms_selected.1to1.csv")
# names(gene_transcript) = c("gene_id"gene_type"gene_name")
# gene_transcript$gene_id = gsub("\\..*"gene_transcript$gene_id)

pre_save_wd = "/projects_eg/projects/marta/TestisRestricted_Microproteins_TSA/with_TranscriptomeReconstruction/Multimap_altORFs/Q2_TestisRestricted/human"
plots_wd = "/projects_eg/projects/marta/TestisRestricted_Microproteins_TSA/with_TranscriptomeReconstruction/Multimap_altORFs/Q3_TestisRestrictedTumors/human/plots"
save_wd = "/projects_eg/projects/marta/TestisRestricted_Microproteins_TSA/with_TranscriptomeReconstruction/Multimap_altORFs/Q3_TestisRestrictedTumors/human"

cancers = c("BRCA","BLCA","LUAD","KIRC","PRAD","LUSC","COAD","LIHC")
tcga_projects=c("TCGA-BRCA","TCGA-LUSC","TCGA-PRAD","TCGA-KIRC","TCGA-LUAD","TCGA-BLCA")#,"TCGA-LIHC"]
other_projects=c("GSE102101_KIRC","GSE133624_BLCA","GSE22260_PRAD","PRJEB2449_PRAD","SRP238334_KIRC","GSE214846_LIHC","GSE229705_LUAD","TCGA_COAD","SRP107326_COAD")
manuscript_projects = c("liver_adjacent_totalRNA_LIHC","hcc_normal_totalRNA_LIHC","GSE193567_LIHC","LIHC_TCGA_LIHC")
# deleted_projects=c("GSE103001_BRCA"GSE89223_PRAD")
all_projects = c(tcga_projects,other_projects,manuscript_projects)

prev_cancers_dir = "/users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/cancers"
cancers_dir = "/users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/cancers"

CTDB = read.csv("/projects_eg/projects/marta/CTdatabase_list.csv", sep="\t")
names(CTDB)[3] = "position"
```

## Testis-Specific Translated ORFs

```{r input_translation_data}
print("Translated IN testis")
translated_IN_testis = read.csv(file.path(pre_save_wd,"testisRestricted_GTEx_translatedINtestis.csv"))
table(translated_IN_testis$gene_type)
print("Corresponding to how many genes?")
translated_IN_testis_genes = translated_IN_testis %>% select(gene_type, gene_name) %>% unique()
table(translated_IN_testis_genes$gene_type)

print("Translated ONLY in testis")
translated_only_testis = read.csv(file.path(pre_save_wd,"testisRestricted_GTEx_translatedONLYtestis.csv"))
table(translated_only_testis$gene_type)
print("Corresponding to how many genes?")
translated_only_testis_genes = translated_only_testis %>% select(gene_type, gene_name) %>% unique()
table(translated_only_testis_genes$gene_type)

## add column to translated IN testis
translated_IN_testis = translated_IN_testis %>% mutate(ONLYtranslation_testis = case_when(transcript_id %in% translated_only_testis$transcript_id ~ "ONLYTestis",
                                                                                          TRUE ~ "several_tissues"))
table(translated_IN_testis$ONLYtranslation_testis, translated_IN_testis$gene_type)
table(translated_IN_testis$ONLYtranslation_testis, translated_IN_testis$geneORFtype)

```

## Tumor Associated Antigens (TAA)

```{r cancer_TAA}
TAA_10percent = data.frame(transcript_id = character(),
                           gene_id = character(),
                           gene_name = character(),
                           gene_type = character(),
                           Length = character(), 
                           n = numeric(),
                           ctype = factor(),
                           stringsAsFactors = F)
for(c in 1:length(cancers)) {
  temp = read.csv(paste0(prev_cancers_dir,"/tumorexpressed/cancertypes/tumor_1FPKM_n10percent_",cancers[c],".csv"))
  temp$ctype = cancers[c]
  
  TAA_10percent = rbind(TAA_10percent, temp)
  
}


ggplot(TAA_10percent %>% group_by(gene_type, ctype) %>% count(), aes(x=n, y=ctype, fill=gene_type)) +
  geom_bar(stat="identity") +
    scale_fill_manual(values=c("protein_coding"="#CC79A7",
                             "lncRNA" = "#009E73",
                             "processed_pseudogene" = "#0090B2",
                             "novel" = "#E69F00")) +
  ggtitle("Tumor-Expressed genes per cancer type (> 1 TPM in > 10% patients)") +
  theme_classic()
```

```{r inputs, echo=F}
TAA_translated = merge(TAA_10percent, translated_IN_testis, by=c("transcript_id","gene_name","gene_id","gene_type"))
TAA_translated = TAA_translated %>% subset(ONLYtranslation_testis == "ONLYTestis") 
table(TAA_translated$gene_type, TAA_translated$ctype)

TAA_translated_genes = TAA_translated %>% subset(ONLYtranslation_testis == "ONLYTestis") %>% select(transcript_id, gene_name, gene_type, ctype) %>% unique()
table(TAA_translated_genes$gene_type, TAA_translated_genes$ctype)
table(TAA_translated_genes$ctype)

```

```{r adjacents, echo=F}
# data_completed = data.frame(transcript_id = character(),
#                               Length = numeric(),
#                               sample = character(),
#                               TPM = numeric(),
#                               patient = factor(),
#                             normal_tumor = factor(),
#                             project = character(),
#                             ctype = character(),
#                               stringsAsFactors = F)
# 
# unfiltered_allfc = data.frame(transcript_id = character(),
#                               Length = numeric(),
#                               sample = character(),
#                               TPM = numeric(),
#                               patient = factor(),
#                             normal_tumor = factor(),
#                             project = character(),
#                             ctype = character(),
#                               stringsAsFactors = F)
# for (ctype_var in cancers) {
#   print(ctype_var)
# 
#   ## table of counts
#   df = read.csv(paste0(cancers_dir,"/merged_fc_",ctype_var,".csv"))
#   ## pivot to make it easier to join
#   df_long = df %>% pivot_longer(cols=-c(Length, transcript_id), values_to = "TPM", names_to = "sample")
#   df_long$sample = gsub("^X","", df_long$sample)
#   df_long$sample = gsub("\\.","-",df_long$sample)
# 
#   ## patients
#   temp_patients = read.csv(paste0(cancers_dir,"/merged_patients_",ctype_var,".csv"))
#   temp_patients = temp_patients %>% pivot_longer(cols=c("normal","tumor"), names_to = "normal_tumor", values_to = "sample")
# 
#   ## combine
#   complete_tableofcounts = merge(temp_patients, df_long, by="sample")
#   complete_tableofcounts$ctype = ctype_var
# 
#   unfiltered_allfc = rbind(unfiltered_allfc, complete_tableofcounts)
#   ## select the tumor-associated per ctype
#   temp_TAA_translated_genes = TAA_translated_genes %>% subset(ctype == ctype_var)
#   TAA_complete_tableofcounts = complete_tableofcounts %>% subset(transcript_id %in% temp_TAA_translated_genes$transcript_id)
#   TAA_complete_tableofcounts = merge(TAA_complete_tableofcounts, annot, by="transcript_id")
# 
#   data_completed = rbind(data_completed, TAA_complete_tableofcounts)
# }
# write.csv(unfiltered_allfc, file.path(save_wd,"unfiltered_tablesofcounts_long.csv"), row.names=F)
# 
# # data_completed = merge(data_completed, annot, by="transcript_id")
# data_completed_wide = data_completed %>% select(-c(sample)) %>% pivot_wider(names_from="normal_tumor", values_from = "TPM")
# filtered_data_TAA = data_completed_wide %>% filter(tumor > 1)
# 
# filtered_data_TAA_long = filtered_data_TAA %>% pivot_longer(cols=c(normal, tumor), values_to = "TPM", names_to = "normal_tumor")
# filtered_data_TAA_long = filtered_data_TAA_long %>% mutate(forced_TPM = case_when(TPM == 0 ~ 1e-10,
#                                                                                   TRUE ~ TPM))
# filtered_data_TAA_long$logTPM = log(filtered_data_TAA_long$forced_TPM)
# write.csv(filtered_data_TAA_long, file.path(save_wd,"TAAdata_long.csv"), row.names=F)
```

```{r plots, echo=F}
filtered_data_TAA_long = read.csv(file.path(save_wd,"TAAdata_long.csv"))
# filtered_data_TAA_long$logTPM = log(filtered_data_TAA_long$TPM)

temporary = filtered_data_TAA_long %>% select(gene_name, ctype) %>% unique()
table(temporary$ctype)

ggplot(filtered_data_TAA_long, aes(x=ctype, y=logTPM, fill=normal_tumor)) +
    geom_quasirandom(dodge.width= .8, size=.5) +
  scale_fill_manual(values=c("#FFB900","#5773CC")) +
  geom_hline(yintercept=log(0.1)) +
  geom_hline(yintercept=log(1)) +
  labs(x="",
       title="Testis-Restricted genes reactivated in cancer\nExpression in tumor and adjacent samples") +
  theme_classic() 
ggsave(file.path(plots_wd,"PNG/TestisRestricted_Reactivated_AdjacentTumor.zeros.png"))
ggsave(file.path(plots_wd,"PDF/TestisRestricted_Reactivated_AdjacentTumor.zeros.pdf"))

ggplot(filtered_data_TAA_long, aes(x=project, y=logTPM, fill=normal_tumor)) +
    geom_quasirandom(dodge.width= .8, size=.5) +
  scale_fill_manual(values=c("#FFB900","#5773CC")) +
  geom_hline(yintercept=log(0.1)) +
  geom_hline(yintercept=log(1)) +
  labs(x="",
       title="Testis-Restricted genes reactivated in cancer\nExpression in tumor and adjacent samples") +
  theme_classic() +
  theme(axis.text.x = element_text(angle=90)) +
  facet_wrap(~ctype, scales="free_x",nrow=1)
ggsave(file.path(plots_wd,"PNG/TestisRestricted_Reactivated_AdjacentTumor_perproject.zeros.png"))
ggsave(file.path(plots_wd,"PDF/TestisRestricted_Reactivated_AdjacentTumor_perproject.zeros.pdf"))

```

```{r explore_TCGA, echo=F}
TCGAdata = filtered_data_TAA_long %>% subset(grepl("TCGA", project)) %>% select(-TPM)

for(proj in unique(TCGAdata$project)) {
  temp_TCGAdata = TCGAdata %>% subset(project == proj)

  ggplot(temp_TCGAdata, aes(x=patient, y=logTPM, fill=normal_tumor)) +
    geom_quasirandom(dodge.width= .8, size=.5) +
    scale_fill_manual(values=c("#FFB900","#5773CC")) +
    geom_hline(yintercept=log(0.1)) +
    geom_hline(yintercept=log(1)) +
    labs(x="",
         title="Testis-Restricted genes reactivated in cancer\nExpression in tumor and adjacent samples") +
    theme_classic() +
    theme(axis.text.x = element_text(angle=90),
          legend.position = "top") +
    facet_wrap(~ctype, scales="free_x",ncol=1)
  ggsave(paste0(plots_wd,"/PNG/TestisRestricted_Reactivated_AdjacentTumor_",proj,".png"), width=14)
  ggsave(paste0(plots_wd,"/PDF/TestisRestricted_Reactivated_AdjacentTumor_",proj,".pdf"), width=14)

}
```

```{r CTA, echo=F}
CTA_filtered_TAA = filtered_data_TAA_long %>% subset(gene_type == "protein_coding")
ggplot(CTA_filtered_TAA, aes(x=ctype, y=logTPM, fill=normal_tumor)) +
    geom_quasirandom(dodge.width= .8, size=.5) +
  scale_fill_manual(values=c("#FFB900","#5773CC")) +
  geom_hline(yintercept=log(0.1)) +
  geom_hline(yintercept=log(1)) +
  labs(x="",
       title="Testis-Restricted genes reactivated in cancer\nExpression in tumor and adjacent samples") +
  theme_classic() 
ggsave(file.path(plots_wd,"PNG/CT_TAA_Reactivated_AdjacentTumor.zeros.png"))
ggsave(file.path(plots_wd,"PDF/CT_TAA_Reactivated_AdjacentTumor.zeros.pdf"))

ggplot(CTA_filtered_TAA, aes(x=project, y=logTPM, fill=normal_tumor)) +
    geom_quasirandom(dodge.width= .8, size=.5) +
  scale_fill_manual(values=c("#FFB900","#5773CC")) +
  geom_hline(yintercept=log(0.1)) +
  geom_hline(yintercept=log(1)) +
  labs(x="",
       title="Testis-Restricted genes reactivated in cancer\nExpression in tumor and adjacent samples") +
  theme_classic() +
  theme(axis.text.x = element_text(angle=90)) +
  facet_wrap(~ctype, scales="free_x",nrow=1)
ggsave(file.path(plots_wd,"PNG/CT_TAA_Reactivated_AdjacentTumor_perproject.zeros.png"))
ggsave(file.path(plots_wd,"PDF/CT_TAA_Reactivated_AdjacentTumor_perproject.zeros.pdf"))
```

## CTAB from database

```{r CTAB_list, echo=F}
unfiltered_allfc = read.csv(file.path(save_wd,"unfiltered_tablesofcounts_long.csv"))

annot_CTDB = merge(CTDB, annot, by="gene_name")

CTDB_fc = unfiltered_allfc %>% subset(transcript_id %in% annot_CTDB$transcript_id)
CTDB_fc = merge(CTDB_fc, annot_CTDB, by="transcript_id")

CTDB_fc = CTDB_fc %>% mutate(forced_TPM = case_when(TPM == 0 ~ 1e-10, TRUE ~ TPM))
CTDB_fc$logTPM = log(CTDB_fc$forced_TPM)
CTDB_fc$chr = gsub("p.*", "", CTDB_fc$chr)
CTDB_fc$chr = gsub("q.*", "", CTDB_fc$chr)
CTDB_fc = CTDB_fc %>% mutate(chr_X = case_when(chr == "X" ~ "X", TRUE ~ "nonX"))
length(unique(CTDB_fc$gene_name))

ggplot(CTDB_fc, aes(x=project, y=logTPM, fill=normal_tumor)) +
  geom_violin(position = position_dodge(width=.8)) +
  scale_fill_manual(values=c("#FFB900","#5773CC")) +
  geom_hline(yintercept=log(0.1)) +
  geom_hline(yintercept=log(1)) +
  labs(x="",
       title="CT from list\nExpression in tumor and adjacent samples") +
  theme_classic() +
  theme(axis.text.x = element_text(angle=90)) +
  facet_wrap(~ctype, scales="free_x",nrow=1)
ggsave(file.path(plots_wd,"PNG/CTDB_Reactivated_AdjacentTumor_perproject.zeros.png"))
ggsave(file.path(plots_wd,"PDF/CTDB_Reactivated_AdjacentTumor_perproject.zeros.pdf"))

ggplot(CTDB_fc %>% subset(grepl("AGE", gene_name)), aes(x=project, y=logTPM, fill=normal_tumor)) +
  geom_violin(position = position_dodge(width=.8)) +
  scale_fill_manual(values=c("#FFB900","#5773CC")) +
  geom_hline(yintercept=log(0.1)) +
  geom_hline(yintercept=log(1)) +
  labs(x="",
       title="CT from list\nExpression in tumor and adjacent samples") +
  theme_classic() +
  theme(axis.text.x = element_text(angle=90), legend.position = "top") +
  facet_wrap(~ctype, scales="free_x",nrow=1)
ggsave(file.path(plots_wd,"PNG/CTDB_AGE_Reactivated_AdjacentTumor_perproject.zeros.png"))
ggsave(file.path(plots_wd,"PDF/CTDB_AGE_Reactivated_AdjacentTumor_perproject.zeros.pdf"))

ggplot(CTDB_fc %>% subset(chr_X == "X"), aes(x=project, y=logTPM, fill=normal_tumor)) +
  geom_violin(position = position_dodge(width=.8)) +
  scale_fill_manual(values=c("#FFB900","#5773CC")) +
  geom_hline(yintercept=log(0.1)) +
  geom_hline(yintercept=log(1)) +
  labs(x="",
       title="CT-X from list\nExpression in tumor and adjacent samples") +
  theme_classic() +
  theme(axis.text.x = element_text(angle=90), legend.position = "top") +
  facet_wrap(~ctype, scales="free_x",nrow=1)
ggsave(file.path(plots_wd,"PNG/CTDB_Xchr_Reactivated_AdjacentTumor_perproject.zeros.png"))
ggsave(file.path(plots_wd,"PDF/CTDB_Xchr_Reactivated_AdjacentTumor_perproject.zeros.pdf"))

ggplot(CTDB_fc %>% subset(chr_X == "nonX"), aes(x=project, y=logTPM, fill=normal_tumor)) +
  geom_violin(position = position_dodge(width=.8)) +
  scale_fill_manual(values=c("#FFB900","#5773CC")) +
  geom_hline(yintercept=log(0.1)) +
  geom_hline(yintercept=log(1)) +
  labs(x="",
       title="CT non-X from list\nExpression in tumor and adjacent samples") +
  theme_classic() +
  theme(axis.text.x = element_text(angle=90), legend.position = "top") +
  facet_wrap(~ctype, scales="free_x",nrow=1)
ggsave(file.path(plots_wd,"PNG/CTDB_nonXchr_Reactivated_AdjacentTumor_perproject.zeros.png"))
ggsave(file.path(plots_wd,"PDF/CTDB_nonXchr_Reactivated_AdjacentTumor_perproject.zeros.pdf"))


for(proj in all_projects) {
 print(proj)
 temp = CTDB_fc %>% subset(ctype == ctype_var)

 ggplot(temp %>% subset(chr_X == "nonX"), aes(x=gene_family, y=logTPM, color=normal_tumor)) +
    geom_quasirandom(dodge.width= .8, size=.5) +
   scale_color_manual(values=c("#FFB900","#5773CC")) +
   geom_hline(yintercept=log(0.1)) +
   geom_hline(yintercept=log(1)) +
   labs(title=paste0(proj,"CT non-X from list\nExpression in tumor and adjacent samples")) +
   theme_classic() +
   theme(axis.text.x = element_text(angle=90), legend.position = "top") 
#   facet_wrap(~project, scales="free_x",ncol=1)
 ggsave(paste0(plots_wd,"/PNG/",proj,"_CTDB_nonXchr_Reactivated_AdjacentTumor_geneName.zeros.png"), width=14)
 ggsave(paste0(plots_wd,"/PDF/",proj,"_CTDB_nonXchr_Reactivated_AdjacentTumor_geneName.zeros.pdf"), width=14)

  ggplot(temp %>% subset(chr_X == "X"), aes(x=gene_family, y=logTPM, color=normal_tumor)) +
    geom_quasirandom(dodge.width= .8, size=.5) +
   scale_color_manual(values=c("#FFB900","#5773CC")) +
   geom_hline(yintercept=log(0.1)) +
   geom_hline(yintercept=log(1)) +
   labs(title=paste0(proj,"CT-X from list\nExpression in tumor and adjacent samples")) +
   theme_classic() +
   theme(axis.text.x = element_text(angle=90), legend.position = "top") 
#   facet_wrap(~gene_family, scales="free_x")
 ggsave(paste0(plots_wd,"/PNG/",proj,"_CTDB_Xchr_Reactivated_AdjacentTumor_geneName.zeros.png"), width=14)
 ggsave(paste0(plots_wd,"/PDF/",proj,"_CTDB_Xchr_Reactivated_AdjacentTumor_geneName.zeros.pdf"), width=14)
}



```

```{r TSA_plot, echo=F}
# data_completed_TSA = data.frame(transcript_id = character(),
#                               Length = numeric(),
#                               sample = character(),
#                               TPM = numeric(),
#                               patient = factor(),
#                             normal_tumor = factor(),
#                             project = character(),
#                             ctype = character(),
#                               stringsAsFactors = F)
# 
# TSA_testisSpecific_orfs = read.csv("/projects_eg/projects/marta/TestisRestricted_Microproteins_TSA/with_TranscriptomeReconstruction/Q4_TestisRestricted_TumorSpecific/human/TSA_TestisRestrictedGTEx_Translated_Ctypes.csv")
# 
# 
# for (ctype_var in cancers) {
#   temp = data_completed %>% subset(ctype == ctype_var)
#   temp = temp %>% subset(transcript_id %in% TSA_testisSpecific_orfs$transcript_id)
# 
#   data_completed_TSA = rbind(data_completed_TSA, temp)
# }
# 
# data_completed_TSA_wide = data_completed_TSA %>% select(-c(sample)) %>% pivot_wider(names_from="normal_tumor", values_from = "TPM")
# filtered_data = data_completed_TSA_wide %>% filter(normal < 0.1, tumor > 1)
# 
# filtered_data_long = filtered_data %>% pivot_longer(cols=c(normal, tumor), values_to = "TPM", names_to = "normal_tumor")
# filtered_data_long$logTPM = log(filtered_data_long$TPM)
# temporary = filtered_data_long %>% select(gene_name, ctype) %>% unique()
# table(temporary$ctype)
# ggplot(filtered_data_long, aes(x=ctype, y=logTPM, fill=normal_tumor)) +
#     geom_quasirandom(dodge.width= .8, size=.5) +
#   scale_fill_manual(values=c("#FFB900","#5773CC")) +
#   geom_hline(yintercept=log(0.1)) +
#   geom_hline(yintercept=log(1)) +
#   labs(x="",
#        title="Testis/Tumor Specific genes\nExpression in tumor and adjacent samples") +
#   theme_classic() 
# 
# ggsave("/projects_eg/projects/marta/TestisRestricted_Microproteins_TSA/with_TranscriptomeReconstruction/Q4_TestisRestricted_TumorSpecific/human/plots/PNG/TestisTumorSpecific_AdjacentTumor.png")
# ggsave("/projects_eg/projects/marta/TestisRestricted_Microproteins_TSA/with_TranscriptomeReconstruction/Q4_TestisRestricted_TumorSpecific/human/plots/PDF/TestisTumorSpecific_AdjacentTumor.pdf")
```

## Is there a correlation between the number of Translated TestisRestrictedGTEx TumorAssociated and the number of patients per ctype?

```{r num_patients_numTTS, echo=F}
total_num_patients = data.frame(ctype = c("BRCA","BLCA","LUAD","KIRC","LUSC","PRAD","LIHC","COAD"),
                                total_n = c(109,38,179,142,49,75,182,144))

total_burden_TestisRestrGTEx_TumorAss = TAA_translated_genes %>% select(ctype, gene_name, gene_type) %>% unique() %>%
  group_by(ctype) %>% count()

full_df = merge(total_burden_TestisRestrGTEx_TumorAss, total_num_patients, by="ctype")

ggplot(full_df, aes(x=total_n, y=n, color=ctype)) +
  geom_point(shape = 21, size=5,stroke = 3) +
  scale_color_manual(values=c("BRCA"="#E69F00", "BLCA"="#56A3A6", "LUAD"="#0079E3", "KIRC"="#F0C27B", "LUSC"="#0072B2", "PRAD"="#D55E00", "LIHC"="#CC79A7", "COAD"="#8B7355")) +
  stat_smooth(method = "lm", 
              formula = y ~ x, 
              geom = "smooth", color="grey") +
  stat_cor(p.accuracy = 0.001, r.accuracy = 0.01)+
  labs(x="Number of patients per cancer type",
       y="Number of genes",
       title="Correlation between number of patients and\nTranslated TestisRestrictedGTEx TumorAssociated genes") +
  theme_minimal()
ggsave(file.path(plots_wd,"PNG/corr_patients_TumorAss.png"))
ggsave(file.path(plots_wd,"PDF/corr_patients_TumorAss.pdf"))
```


## Is the expression in testis and in tumor correlated?

```{r stats_function, echo=F}
# Function to compute the required statistics
compute_stats <- function(row) {
  mean_value <- mean(row)
  median_value <- median(row)
  sd_value <- sd(row)
  q1_value <- quantile(row, 0.25)
  q3_value <- quantile(row, 0.75)
  
  return(c(mean = mean_value, median = median_value, sd = sd_value, Q1 = q1_value, Q3 = q3_value))
}

```

```{r TPMS_testis, echo=F}
## Load counts data
# raw_counts = read.table("/users/genomics/saraa/projectTestis/featureCounts/results_geneID/featureCounts_geneID.txt header=TRUE)
testis_TPMs = read.csv("/users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/human/featureCounts_gffcompare/table_of_counts_TPMs_withLength.csv", header=TRUE)

## For novels, remove those shorter than 300 pb
testis_TPMs = testis_TPMs %>%
  filter(!(startsWith(gene_id, "TCONS") & Length < 300))
names(testis_TPMs)[1] = "transcript_id"

testis_TPMs = merge(testis_TPMs, annot, by="transcript_id", all.x=T)
testis_TPMs = testis_TPMs %>% unique()
testis_TPMs = testis_TPMs[,!grepl("brain", names(testis_TPMs))]
testis_TPMs = testis_TPMs[,!grepl("liver", names(testis_TPMs))]
testis_TPMs = testis_TPMs %>% select(-Length)

## Select only numeric columns
numeric_df = testis_TPMs[, sapply(testis_TPMs, is.numeric)]
## Apply the function to each row
stats_per_row = t(apply(numeric_df, 1, compute_stats))
stats_df = as.data.frame(stats_per_row)

testis_TPMs_stats = cbind(testis_TPMs %>% select(transcript_id, gene_id, gene_type, gene_name), stats_df)
testis_TPMs_stats %>% head
names(testis_TPMs_stats) = c("transcript_id","gene_id","gene_type","gene_name","mean_testis","median_testis","sd_testis","Q1.25%_testis","Q3.75%_testis")

```


```{r TPMS_cancer, echo=F}
testis_tumor_TPMs = data.frame(transcript_id = character(),
                               stringsAsFactors = F)

for(i in 1:length(all_projects)) {
  TAA_tumor = read.csv(paste0(prev_cancers_dir,"/tumorexpressed/tumor_1FPKM_table_of_counts_",all_projects[i],".csv"))
  TAA_tumor = TAA_tumor %>% select(-Length)
  
  ## Select only numeric columns
  numeric_df = TAA_tumor[, sapply(TAA_tumor, is.numeric)]
  ## Apply the function to each row
  stats_per_row = t(apply(numeric_df, 1, compute_stats))
  stats_df = as.data.frame(stats_per_row)

  TAA_tumor_stats = cbind(TAA_tumor %>% select(transcript_id), stats_df)
  names(TAA_tumor_stats) = c("transcript_id","mean_tumor","median_tumor","sd_tumor","Q1.25%_tumor","Q3.75%_tumor")
  TAA_tumor_stats_testis = merge(testis_TPMs_stats, TAA_tumor_stats, by="transcript_id")
  TAA_tumor_stats_testis$cancertype = cancers[i]
  
  ## rbind
  testis_tumor_TPMs = rbind(testis_tumor_TPMs, TAA_tumor_stats_testis)
}
```

```{r correlation_TPM, echo=F}
ggplot(testis_tumor_TPMs %>% drop_na(), aes(x=mean_tumor, y=mean_testis)) +
  geom_point(size=.03) +
  geom_smooth(method = "lm", se = FALSE, aes(color=gene_type)) +
  scale_x_continuous(trans="log10") +
  scale_y_continuous(trans="log10") +
    scale_color_manual(values=c("protein_coding"="#CC79A7",
                           "lncRNA" = "#009E73",
                           "processed_pseudogene" = "#0090B2",
                           "novel" = "#E69F00")) +
  labs(title="Correlation TPMs between tumor and testis") +
  theme_classic() +
  theme(legend.position = "top") +
  facet_wrap(~ cancertype, nrow=2) +
  stat_cor(label.y = c(5.2, 4.8, 4.3, 3.8), aes(group=gene_type, color = gene_type), size=3) 
ggsave(file.path(plots_wd,"PNG/correlationTPM_testis_tumor.cancertype.png"), width=12.52, height=5.64)
ggsave(file.path(plots_wd,"PDF/correlationTPM_testis_tumor.cancertype.pdf"), width=12.52, height=5.64)


testis_tumor_TPMs_pancancer = testis_tumor_TPMs %>% select(-cancertype) %>% group_by(transcript_id, gene_id, gene_name, gene_type) %>% mutate(mean_mean_testis = mean(mean_testis),
                                                                                                           mean_mean_tumor = mean(mean_tumor)) %>% unique()

ggplot(testis_tumor_TPMs_pancancer %>% drop_na(), aes(x=mean_mean_tumor, y=mean_mean_testis, color=gene_type)) +
  geom_point(size=.03) +
  geom_smooth(method = "lm", se = FALSE, color="black") +
  scale_x_continuous(trans="log10") +
  scale_y_continuous(trans="log10") +
    scale_color_manual(values=c("protein_coding"="#CC79A7",
                           "lncRNA" = "#009E73",
                           "processed_pseudogene" = "#0090B2",
                           "novel" = "#E69F00")) +
  labs(title="Correlation TPMs between tumor and testis") +
  theme_classic() +
  theme(legend.position = "top") +
  facet_wrap(~ gene_type, nrow=2, scales="free") +
  stat_cor(label.y = c(5.2, 4.8, 4.3, 3.8), aes(group=gene_type), size=3, color="black") 
ggsave(file.path(plots_wd,"PNG/correlationTPM_testis_tumor.meancancers.png"), width=7, height=7)
ggsave(file.path(plots_wd,"PDF/correlationTPM_testis_tumor.meancancers.pdf"), width=7, height=7)

ggplot(testis_tumor_TPMs_pancancer %>% drop_na(), aes(x=mean_mean_tumor, y=mean_mean_testis, color=gene_type)) +
  geom_point(size=.5, alpha=.5) +
  geom_smooth(method = "lm", se = FALSE, aes(color=gene_type)) +
  scale_x_continuous(trans="log10") +
  # xlim(0,100000) +
  # ylim(0,100000) +
  scale_y_continuous(trans="log10") +
    scale_color_manual(values=c("protein_coding"="#CC79A7",
                           "lncRNA" = "#009E73",
                           "processed_pseudogene" = "#0090B2",
                           "novel" = "#E69F00")) +
  labs(title="Correlation TPMs between tumor and testis") +
  theme_classic() +
  theme(legend.position = "top") +
  stat_cor(label.y = c(5.2, 4.8, 4.3, 3.8), aes(group=gene_type, color=gene_type), size=3) 

ggsave(file.path(plots_wd,"PNG/correlationTPM_testis_tumor.meancancers.onlydots.png"), width=7, height=7)
ggsave(file.path(plots_wd,"PDF/correlationTPM_testis_tumor.meancancers.onlydots.pdf"), width=7, height=7)
```
