---
title: "3. Testis-specific translated ORFs in tumors"
author: "Marta Espinosa"
date: "2024-07-10"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
# packages <- c("dplyr", "tidyr", "ggplot2", "stringr", "purrr", "ggvenn", "ggpubr", 
#               "ggside", "bioseq", "ggthemr", "ggbeeswarm", "ggbreak")
# 
# # Install packages that are not already installed
# installed_packages <- installed.packages()
# for (pkg in packages) {
#   if (!(pkg %in% installed_packages[, "Package"])) {
#     install.packages(pkg)
#   }
# }
# 
# # Load the packages
# lapply(packages, library, character.only = TRUE)

library(dplyr)
library(tidyr)
library(ggplot2)
library(stringr)
library(purrr)
library(ggvenn)
library(ggpubr)
library(ggside)
library(bioseq)
# library(ggthemr)
library(ggbeeswarm)
library(ggbreak)

annot = read.csv("/users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/human/newReference_Resconstructed/1transcript_1gene.reconstructed.csv")
# names(gene_transcript) = c("gene_id","gene_type","gene_name")
# gene_transcript$gene_id = gsub("\\..*","",gene_transcript$gene_id)

pre_save_wd = "/projects_eg/projects/marta/TestisRestricted_Microproteins_TSA/with_TranscriptomeReconstruction/Q2_TestisRestricted/human"
plots_wd = "/projects_eg/projects/marta/TestisRestricted_Microproteins_TSA/with_TranscriptomeReconstruction/Q3_TestisRestrictedTumors/human/plots"
save_wd = "/projects_eg/projects/marta/TestisRestricted_Microproteins_TSA/with_TranscriptomeReconstruction/Q3_TestisRestrictedTumors/human"

cancers = c("BRCA","BLCA","LUAD","KIRC","KIRP","PRAD","LUSC","COAD","LIHC")
tcga_projects=c("TCGA-BRCA","TCGA-LUSC","TCGA-PRAD","TCGA-KIRC","TCGA-KIRP","TCGA-LUAD","TCGA-BLCA")#,"TCGA-LIHC"]
other_projects=c("GSE102101_KIRC","GSE133624_BLCA","GSE22260_PRAD","PRJEB2449_PRAD","SRP238334_KIRC","GSE214846_LIHC","GSE229705_LUAD","TCGA_COAD","SRP107326_COAD")
manuscript_projects = c("liver_adjacent_totalRNA_LIHC","hcc_normal_totalRNA_LIHC","GSE193567_LIHC","LIHC_TCGA_LIHC")
# deleted_projects=c("GSE103001_BRCA","GSE89223_PRAD")
all_projects = c(tcga_projects,other_projects,manuscript_projects)

cancers_dir = "/users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/cancers"
```

## Testis-Specific Translated ORFs

```{r input_translation_data}
print("Translated IN testis")
translated_IN_testis = read.csv(file.path(pre_save_wd,"testisRestricted_GTEx_translatedINtestis.csv"))
table(translated_IN_testis$gene_type)
print("Corresponding to how many genes?")
translated_IN_testis_genes = translated_IN_testis %>% select(gene_type, gene_name) %>% unique()
table(translated_IN_testis_genes$gene_type)

print("Translated ONLY in testis")
translated_only_testis = read.csv(file.path(pre_save_wd,"testisRestricted_GTEx_translatedONLYtestis.csv"))
table(translated_only_testis$gene_type)
print("Corresponding to how many genes?")
translated_only_testis_genes = translated_only_testis %>% select(gene_type, gene_name) %>% unique()
table(translated_only_testis_genes$gene_type)

## add column to translated IN testis
translated_IN_testis = translated_IN_testis %>% mutate(ONLYtranslation_testis = case_when(transcript_id %in% translated_only_testis$transcript_id ~ "ONLYTestis",
                                                                                          TRUE ~ "several_tissues"))
table(translated_IN_testis$ONLYtranslation_testis, translated_IN_testis$gene_type)
```

## Tumor Associated Antigens (TAA)

```{r cancer_TAA}
TAA_10percent = data.frame(transcript_id = character(),
                           gene_id = character(),
                           gene_name = character(),
                           gene_type = character(),
                           Length = character(), 
                           n = numeric(),
                           ctype = factor(),
                           stringsAsFactors = F)
for(c in 1:length(cancers)) {
  temp = read.csv(paste0(cancers_dir,"/tumorexpressed/cancertypes/tumor_1FPKM_n10percent_",cancers[c],".csv"))
  temp$ctype = cancers[c]
  
  TAA_10percent = rbind(TAA_10percent, temp)
  
}


ggplot(TAA_10percent %>% group_by(gene_type, ctype) %>% count(), aes(x=n, y=ctype, fill=gene_type)) +
  geom_bar(stat="identity") +
    scale_fill_manual(values=c("protein_coding"="#849DB1",
                             "lncRNA" = "#B66353",
                             "processed_pseudogene" = "#FBB04E",
                             "novel" = "#B09C85")) +
  ggtitle("Tumor-Expressed genes per cancer type (> 1 TPM in > 10% patients)") +
  theme_classic()
```

```{r}
TAA_translated = merge(TAA_10percent, translated_IN_testis, by=c("transcript_id","gene_name","gene_id","gene_type"))
TAA_translated = TAA_translated %>% subset(ONLYtranslation_testis == "ONLYTestis") 
table(TAA_translated$gene_type, TAA_translated$ctype)

TAA_translated_genes = TAA_translated %>% subset(ONLYtranslation_testis == "ONLYTestis") %>% select(transcript_id, gene_name, gene_type, ctype) %>% unique()
table(TAA_translated_genes$gene_type, TAA_translated_genes$ctype)

```
