---
title: "3. Testis-specific translated ORFs in tumors"
author: "Marta Espinosa"
date: "2024-07-10"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
# packages <- c("dplyr", "tidyr", "ggplot2", "stringr", "purrr", "ggvenn", "ggpubr", 
#               "ggside", "bioseq", "ggthemr", "ggbeeswarm", "ggbreak")
# 
# # Install packages that are not already installed
# installed_packages <- installed.packages()
# for (pkg in packages) {
#   if (!(pkg %in% installed_packages[, "Package"])) {
#     install.packages(pkg)
#   }
# }
# 
# # Load the packages
# lapply(packages, library, character.only = TRUE)

library(dplyr)
library(tidyr)
library(ggplot2)
library(stringr)
library(purrr)
library(ggvenn)
library(ggpubr)
library(ggside)
library(bioseq)
# library(ggthemr)
library(ggbeeswarm)
library(ggbreak)
library(GGally)

annot = read.csv("/users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/human/newReference_Resconstructed/1transcript_1gene.reconstructed.csv")
# names(gene_transcript) = c("gene_id","gene_type","gene_name")
# gene_transcript$gene_id = gsub("\\..*","",gene_transcript$gene_id)

pre_save_wd = "/projects_eg/projects/marta/TestisRestricted_Microproteins_TSA/with_TranscriptomeReconstruction/Q2_TestisRestricted/human"
plots_wd = "/projects_eg/projects/marta/TestisRestricted_Microproteins_TSA/with_TranscriptomeReconstruction/Q3_TestisRestrictedTumors/human/plots"
save_wd = "/projects_eg/projects/marta/TestisRestricted_Microproteins_TSA/with_TranscriptomeReconstruction/Q3_TestisRestrictedTumors/human"

cancers = c("BRCA","BLCA","LUAD","KIRC","KIRP","PRAD","LUSC","COAD","LIHC")
tcga_projects=c("TCGA-BRCA","TCGA-LUSC","TCGA-PRAD","TCGA-KIRC","TCGA-KIRP","TCGA-LUAD","TCGA-BLCA")#,"TCGA-LIHC"]
other_projects=c("GSE102101_KIRC","GSE133624_BLCA","GSE22260_PRAD","PRJEB2449_PRAD","SRP238334_KIRC","GSE214846_LIHC","GSE229705_LUAD","TCGA_COAD","SRP107326_COAD")
manuscript_projects = c("liver_adjacent_totalRNA_LIHC","hcc_normal_totalRNA_LIHC","GSE193567_LIHC","LIHC_TCGA_LIHC")
# deleted_projects=c("GSE103001_BRCA","GSE89223_PRAD")
all_projects = c(tcga_projects,other_projects,manuscript_projects)

cancers_dir = "/users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/cancers"
```

## Testis-Specific Translated ORFs

```{r input_translation_data}
print("Translated IN testis")
translated_IN_testis = read.csv(file.path(pre_save_wd,"testisRestricted_GTEx_translatedINtestis.csv"))
table(translated_IN_testis$gene_type)
print("Corresponding to how many genes?")
translated_IN_testis_genes = translated_IN_testis %>% select(gene_type, gene_name) %>% unique()
table(translated_IN_testis_genes$gene_type)

print("Translated ONLY in testis")
translated_only_testis = read.csv(file.path(pre_save_wd,"testisRestricted_GTEx_translatedONLYtestis.csv"))
table(translated_only_testis$gene_type)
print("Corresponding to how many genes?")
translated_only_testis_genes = translated_only_testis %>% select(gene_type, gene_name) %>% unique()
table(translated_only_testis_genes$gene_type)

## add column to translated IN testis
translated_IN_testis = translated_IN_testis %>% mutate(ONLYtranslation_testis = case_when(transcript_id %in% translated_only_testis$transcript_id ~ "ONLYTestis",
                                                                                          TRUE ~ "several_tissues"))
table(translated_IN_testis$ONLYtranslation_testis, translated_IN_testis$gene_type)
```

## Tumor Associated Antigens (TAA)

```{r cancer_TAA}
TAA_10percent = data.frame(transcript_id = character(),
                           gene_id = character(),
                           gene_name = character(),
                           gene_type = character(),
                           Length = character(), 
                           n = numeric(),
                           ctype = factor(),
                           stringsAsFactors = F)
for(c in 1:length(cancers)) {
  temp = read.csv(paste0(cancers_dir,"/tumorexpressed/cancertypes/tumor_1FPKM_n10percent_",cancers[c],".csv"))
  temp$ctype = cancers[c]
  
  TAA_10percent = rbind(TAA_10percent, temp)
  
}


ggplot(TAA_10percent %>% group_by(gene_type, ctype) %>% count(), aes(x=n, y=ctype, fill=gene_type)) +
  geom_bar(stat="identity") +
    scale_fill_manual(values=c("protein_coding"="#849DB1",
                             "lncRNA" = "#B66353",
                             "processed_pseudogene" = "#FBB04E",
                             "novel" = "#B09C85")) +
  ggtitle("Tumor-Expressed genes per cancer type (> 1 TPM in > 10% patients)") +
  theme_classic()
```

```{r}
TAA_translated = merge(TAA_10percent, translated_IN_testis, by=c("transcript_id","gene_name","gene_id","gene_type"))
TAA_translated = TAA_translated %>% subset(ONLYtranslation_testis == "ONLYTestis") 
table(TAA_translated$gene_type, TAA_translated$ctype)

TAA_translated_genes = TAA_translated %>% subset(ONLYtranslation_testis == "ONLYTestis") %>% select(transcript_id, gene_name, gene_type, ctype) %>% unique()
table(TAA_translated_genes$gene_type, TAA_translated_genes$ctype)

table(TAA_translated_genes$ctype)

```

```{r adjacents, echo=F}
data_completed = data.frame(transcript_id = character(),
                              Length = numeric(),
                              sample = character(),
                              TPM = numeric(),
                              patient = factor(),
                            normal_tumor = factor(),
                            project = character(),
                            ctype = character(),
                              stringsAsFactors = F)


for (ctype_var in cancers) {
  print(ctype_var)
  
  ## table of counts
  df = read.csv(paste0(cancers_dir,"/merged_fc_",ctype_var,".csv"))
  ## pivot to make it easier to join
  df_long = df %>% pivot_longer(cols=-c(Length, transcript_id), values_to = "TPM", names_to = "sample")
  df_long$sample = gsub("^X","", df_long$sample)
  df_long$sample = gsub("\\.","-",df_long$sample)
  
  ## patients
  temp_patients = read.csv(paste0(cancers_dir,"/merged_patients_",ctype_var,".csv"))
  temp_patients = temp_patients %>% pivot_longer(cols=c("normal","tumor"), names_to = "normal_tumor", values_to = "sample")
  
  ## combine
  complete_tableofcounts = merge(temp_patients, df_long, by="sample")
  complete_tableofcounts$ctype = ctype_var
  
  ## select the tumor-associated per ctype
  temp_TAA_translated_genes = TAA_translated_genes %>% subset(ctype == ctype_var)
  TAA_complete_tableofcounts = complete_tableofcounts %>% subset(transcript_id %in% temp_TAA_translated_genes$transcript_id)
  
  data_completed = rbind(data_completed, TAA_complete_tableofcounts)
}

data_completed = merge(data_completed, annot, by="transcript_id")
data_completed_wide = data_completed %>% select(-c(sample)) %>% pivot_wider(names_from="normal_tumor", values_from = "TPM")
filtered_data_TAA = data_completed_wide %>% filter(tumor > 1)

filtered_data_TAA_long = filtered_data_TAA %>% pivot_longer(cols=c(normal, tumor), values_to = "TPM", names_to = "normal_tumor")
filtered_data_TAA_long$logTPM = log(filtered_data_TAA_long$TPM)
temporary = filtered_data_TAA_long %>% select(gene_name, ctype) %>% unique()
table(temporary$ctype)

ggplot(filtered_data_TAA_long, aes(x=ctype, y=logTPM, fill=normal_tumor)) +
  geom_boxplot(position = position_dodge(width=.8)) +
  scale_fill_manual(values=c("#FFB900","#5773CC")) +
  geom_hline(yintercept=log(0.1)) +
  geom_hline(yintercept=log(1)) +
  labs(x="",
       title="Testis-Restricted genes reactivated in cancer\nExpression in tumor and adjacent samples") +
  theme_classic() 
ggsave(file.path(plots_wd,"PNG/TestisRestricted_Reactivated_AdjacentTumor.png"))
ggsave(file.path(plots_wd,"PDF/TestisRestricted_Reactivated_AdjacentTumor.pdf"))
```


```{r TSA_plot, echo=F}
data_completed_TSA = data.frame(transcript_id = character(),
                              Length = numeric(),
                              sample = character(),
                              TPM = numeric(),
                              patient = factor(),
                            normal_tumor = factor(),
                            project = character(),
                            ctype = character(),
                              stringsAsFactors = F)

TSA_testisSpecific_orfs = read.csv("/projects_eg/projects/marta/TestisRestricted_Microproteins_TSA/with_TranscriptomeReconstruction/Q4_TestisRestricted_TumorSpecific/human/TSA_TestisRestrictedGTEx_Translated_Ctypes.csv")


for (ctype_var in cancers) {
  temp = data_completed %>% subset(ctype == ctype_var)
  temp = temp %>% subset(transcript_id %in% TSA_testisSpecific_orfs$transcript_id)

  data_completed_TSA = rbind(data_completed_TSA, temp)
}

data_completed_TSA_wide = data_completed_TSA %>% select(-c(sample)) %>% pivot_wider(names_from="normal_tumor", values_from = "TPM")
filtered_data = data_completed_TSA_wide %>% filter(normal < 0.1, tumor > 1)

filtered_data_long = filtered_data %>% pivot_longer(cols=c(normal, tumor), values_to = "TPM", names_to = "normal_tumor")
filtered_data_long$logTPM = log(filtered_data_long$TPM)
temporary = filtered_data_long %>% select(gene_name, ctype) %>% unique()
table(temporary$ctype)
ggplot(filtered_data_long, aes(x=ctype, y=logTPM, fill=normal_tumor)) +
  geom_boxplot(position = position_dodge(width=.8)) +
  scale_fill_manual(values=c("#FFB900","#5773CC")) +
  geom_hline(yintercept=log(0.1)) +
  geom_hline(yintercept=log(1)) +
  labs(x="",
       title="Testis/Tumor Specific genes\nExpression in tumor and adjacent samples") +
  theme_classic() 

ggsave("/projects_eg/projects/marta/TestisRestricted_Microproteins_TSA/with_TranscriptomeReconstruction/Q4_TestisRestricted_TumorSpecific/human/plots/PNG/TestisTumorSpecific_AdjacentTumor.png")
ggsave("/projects_eg/projects/marta/TestisRestricted_Microproteins_TSA/with_TranscriptomeReconstruction/Q4_TestisRestricted_TumorSpecific/human/plots/PDF/TestisTumorSpecific_AdjacentTumor.pdf")
```

## Is there a correlation between the number of Translated TestisRestrictedGTEx TumorAssociated and the number of patients per ctype?

```{r num_patients_numTTS, echo=F}
total_num_patients = data.frame(ctype = c("BRCA","BLCA","LUAD","KIRC","LUSC","PRAD","LIHC","COAD"),
                                total_n = c(109,38,179,142,49,75,182,144))

total_burden_TestisRestrGTEx_TumorAss = TAA_translated_genes %>% select(ctype, gene_name, gene_type) %>% unique() %>%
  group_by(ctype) %>% count()

full_df = merge(total_burden_TestisRestrGTEx_TumorAss, total_num_patients, by="ctype")

ggplot(full_df, aes(x=total_n, y=n, color=ctype)) +
  geom_point(shape = 21, size=5,stroke = 3) +
  scale_color_manual(values=c("BRCA"="#E69F00", "BLCA"="#56A3A6", "LUAD"="#009E73", "KIRC"="#F0C27B", "LUSC"="#0072B2", "PRAD"="#D55E00", "LIHC"="#CC79A7", "COAD"="#8B7355")) +
  stat_smooth(method = "lm", 
              formula = y ~ x, 
              geom = "smooth", color="grey") +
  stat_cor(p.accuracy = 0.001, r.accuracy = 0.01)+
  labs(x="Number of patients per cancer type",
       y="Number of genes",
       title="Correlation between number of patients and\nTranslated TestisRestrictedGTEx TumorAssociated genes") +
  theme_minimal()
ggsave(file.path(plots_wd,"PNG/corr_patients_TumorAss.png"))
ggsave(file.path(plots_wd,"PDF/corr_patients_TumorAss.pdf"))
```

