---
title: "3. Tumor-specific translated ORFs in tumors"
author: "Marta Espinosa"
date: "2024-07-10"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
library(dplyr)
library(tidyr)
library(ggplot2)
library(stringr)
library(purrr)
library(ggvenn)
library(ggpubr)
library(ggside)
library(bioseq)
library(ggthemr)
library(ggbeeswarm)

annot = read.csv("/users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/human/newReference_Resconstructed/1transcript_1gene.reconstructed.csv")
# names(gene_transcript) = c("gene_id","gene_type","gene_name")
# gene_transcript$gene_id = gsub("\\..*","",gene_transcript$gene_id)

plots_wd = "/projects_eg/projects/marta/TestisRestricted_Microproteins_TSA/with_TranscriptomeReconstruction/plots"
save_wd = "/projects_eg/projects/marta/TestisRestricted_Microproteins_TSA/with_TranscriptomeReconstruction"

cancers = c("TCGA-KIRC","TCGA-KIRP","TCGA-LUAD","TCGA-LUSC","TCGA-PRAD","TCGA-BRCA","TCGA-LIHC")
cancers_dir = "/users/genomics/marta/TCGA_RNASeq"
```

## Testis-Specific Translated ORFs

```{r input_translation_data}
translated_IN_testis = read.csv(file.path(save_wd,"testisRestricted_GTEx_translatedINtestis.csv"))
table(translated_IN_testis$gene_type)

translated_only_testis = read.csv(file.path(save_wd,"testisRestricted_GTEx_translatedONLYtestis.csv"))
table(translated_only_testis$gene_type)

## add column to translated IN testis
translated_IN_testis = translated_IN_testis %>% mutate(ONLYtranslation_testis = case_when(transcript_id %in% translated_only_testis$transcript_id ~ "ONLYTestis",
                                                                                          TRUE ~ "several_tissues"))
table(translated_IN_testis$ONLYtranslation_testis, translated_IN_testis$gene_type)
```

## Tumor Associated Antigens (TAA)

```{r cancer_TAA}
## need to include novels
TAA_df = data.frame(transcript_id = character(),
                    gene_id = character())
for(i in 1:length(cancers)) {
  TAA = read.csv(file.path(cancers_dir,cancers[i],"analysis/07_quantification/TPM/tumor_1TPM_n_TOLERABILITY5.csv"))
  TAA$cancertype = cancers[i]
  
  TAA_df = rbind(TAA_df, TAA)
}

ggplot(TAA_df %>% group_by(gene_type, cancertype) %>% count(), aes(x=n, y=cancertype, fill=gene_type)) +
  geom_bar(stat="identity") +
    scale_fill_manual(values=c("protein_coding"="#849DB1",
                             "lncRNA" = "#B66353",
                             "processed_pseudogene" = "#FBB04E",
                             "novel" = "#B09C85")) +
  theme_classic()
```

```{r}
TAA_translated = merge(TAA_df, translated_IN_testis, by=c("transcript_id","gene_name","gene_id","gene_type"))
TAA_translated_genes = TAA_translated %>% select(transcript_id, gene_name, gene_type, ONLYtranslation_testis) %>% unique()
table(TAA_translated_genes$gene_type, TAA_translated_genes$ONLYtranslation_testis)
```