---
title: "Q4.3_Heatmap_TTS_ctype"
author: "Marta Espinosa"
date: "2024-09-18"
output: html_document
---

```{r setup, include=FALSE}
library(tidyverse)
library(dplyr)
library(ggplot2)
library(rcartocolor)
require(QualityGraphs)
require(RColorBrewer)
require(stringr)
library("pheatmap")
library(ggpubr)
library(Hmisc)
library(corrplot)

annot = read.csv("/users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/human/newReference_Resconstructed/1transcript_1gene.reconstructed.csv")
chr_annot = read.csv("/users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/human/newReference_Resconstructed/gene_transcript_chr.csv")
chr_annot$gene_id = gsub("\\..*","",chr_annot$gene_id)
chr_annot$gene_id = gsub("_PAR*","",chr_annot$gene_id)
chr_annot$transcript_id = gsub("\\..*","",chr_annot$transcript_id)
annot = merge(annot, chr_annot, by=c("gene_id", "transcript_id"))

plots_wd = "/projects_eg/projects/marta/TestisRestricted_Microproteins_TSA/with_TranscriptomeReconstruction/Q4_TestisRestricted_TumorSpecific/human/plots"
save_wd = "/projects_eg/projects/marta/TestisRestricted_Microproteins_TSA/with_TranscriptomeReconstruction/Q4_TestisRestricted_TumorSpecific/human"

cancers = c("BRCA","BLCA","LUAD","KIRC","PRAD","LUSC","COAD","LIHC")#"KIRP"
tcga_projects=c("TCGA-BRCA","TCGA-LUSC","TCGA-PRAD","TCGA-KIRC","TCGA-KIRP","TCGA-LUAD","TCGA-BLCA")#,"TCGA-LIHC"]
other_projects=c("GSE102101_KIRC","GSE133624_BLCA","GSE22260_PRAD","PRJEB2449_PRAD","SRP238334_KIRC","GSE214846_LIHC","GSE229705_LUAD","TCGA_COAD","SRP107326_COAD")
manuscript_projects = c("liver_adjacent_totalRNA_LIHC","hcc_normal_totalRNA_LIHC","GSE193567_LIHC","LIHC_TCGA_LIHC")
# deleted_projects=c("GSE103001_BRCA","GSE89223_PRAD")
all_projects = c(tcga_projects,other_projects,manuscript_projects)

cancers_dir = "/users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/cancers"

ctype_patients = data.frame(ctype = c("BRCA","BLCA","LUAD","KIRC","PRAD","LUSC","COAD","LIHC"),
                            num_patients = c(109,38,179,142,49,75,182,144))

save_pheatmap_pdf <- function(x, filename, width=7, height=7) {
   stopifnot(!missing(x))
   stopifnot(!missing(filename))
   pdf(filename, width=width, height=height)
   grid::grid.newpage()
   grid::grid.draw(x$gtable)
   dev.off()
}

CTDB = read.csv("/projects_eg/projects/marta/CTdatabase_list.csv", sep="\t")

# Extract the correlation coefficients
# res2$r
# Extract p-values
# res2$P
# A simple function to format the correlation matrix
# ++++++++++++++++++++++++++++
# flattenCorrMatrix
# ++++++++++++++++++++++++++++
# cormat : matrix of the correlation coefficients
# pmat : matrix of the correlation p-values
flattenCorrMatrix <- function(cormat, pmat) {
  ut <- upper.tri(cormat)
  data.frame(
    row = rownames(cormat)[row(cormat)[ut]],
    column = rownames(cormat)[col(cormat)[ut]],
    cor  =(cormat)[ut],
    p = pmat[ut]
  )
}
# flattenCorrMatrix(res2$r, res2$P)
```

## HEATMAP with Testis/Tumor-Specific genes per cancer type

```{r import, echo=F}
data_completed_TSA = read.csv(file.path(save_wd,"TTS_TPM_patients.csv"))
```


```{r force_zero, echo=TRUE}
# data_completed_TSA_wide = data_completed_TSA %>% select(-c(X, sample)) %>% pivot_wider(names_from="normal_tumor", values_from = "TPM")
# # ## if the gene is not TS in that patient, although it is expressed in the tumor sample, we will force the expression to 0
# data_completed_TSA_forced = data_completed_TSA_wide %>% mutate(forced_tumor_expr = case_when(normal < 0.1 & tumor > 1 ~ tumor,
#                                                                                   TRUE ~ 0))
# write.csv(data_completed_TSA_forced, file.path(save_wd,"TTS_forcedTPM_patients.csv"), row.names=F)
```

```{r import_TTS, echo=F}
data_completed_TSA_forced = read.csv(file.path(save_wd,"TTS_forcedTPM_patients.csv"))
```

```{r heatmap_corr_per_ctype_5percent, echo=F}
############ 5 percent #################
for (ctype_var in cancers) {
  print(ctype_var)
  ## how many patients in that ctype?
  n_ctype = ctype_patients %>% subset(ctype == ctype_var) %>% pull(num_patients)
  
  df = data_completed_TSA_forced %>% subset(ctype == ctype_var)
  
  ## in how many patients is the gene TS?
  counting_1TPM = df %>% select(gene_name, gene_type, patient, forced_tumor_expr) %>% subset(forced_tumor_expr > 0) %>% group_by(gene_name) %>% count()
  df = merge(df, counting_1TPM, by="gene_name")
  df$total_n = n_ctype
  
  ## calculate the percentage and keep only those TS in > 5% 
  df$percentage_n = (df$n / n_ctype)* 100
  df = df %>% unique()
  df_5percent = df %>% subset(percentage_n > 5)
  ## if dataframe not empty
  if (dim(df_5percent)[1] > 0) {

    df_5percent = df_5percent %>% unique()
    to_table = df_5percent %>% select(gene_name, gene_type) %>% unique()
    print(table(to_table$gene_type))
    
    ## keep expression data to plot
    df_5percent_expr = df_5percent %>% select(gene_name, patient, forced_tumor_expr) %>% pivot_wider(names_from = patient, values_from = forced_tumor_expr) %>% unique()
    df_5percent_expr = as.data.frame(df_5percent_expr)
  
    df_5percent_expr_values = mutate_all(df_5percent_expr[,-1], function(x) as.numeric(as.character(x)))
    ## get log2TPM
    df_log = log2(df_5percent_expr_values+1)
  
    ## convert to matrix
    matrix = as.matrix(df_log)
    rownames(matrix) = df_5percent_expr$gene_name
    matrix = na.omit(matrix)
    nozeros_matrix_log = matrix[,colSums(matrix) > 0]
    
    
    
    ## row annotation
    row_annot = annot %>% subset(gene_name %in% df_5percent_expr$gene_name)
    row_annot = row_annot %>% select(gene_name, gene_type)
    row_annot = unique(row_annot)
    rownames(row_annot) = row_annot$gene_name
    
    row_annot_chr = merge(annot, row_annot, by=c("gene_name"), all.y=T)
    row_annot_chr = row_annot_chr %>% select(gene_name, chr) %>% unique()
    rownames(row_annot_chr) = row_annot_chr$gene_name
    row_annot_chr = row_annot_chr %>% mutate(X_nonX = case_when(chr == "X" ~ "X",
                                                                TRUE ~ "nonX"))
    row_annot_chr = row_annot_chr %>% select(X_nonX)
    row_annot = row_annot %>% select(-c(gene_name))

    row_annotation = data.frame("X_nonX" = row_annot_chr,
                                "gene_type" = row_annot)
    ## col annotation
    col_annot = df_5percent %>% select(patient, project) %>% subset(patient %in% colnames(df_5percent_expr_values))
    col_annot = unique(col_annot)
    rownames(col_annot) = col_annot$patient
    col_annot = col_annot %>% select(-c(patient))
  
    biotypecolors = list("gene_type" = c("protein_coding"="#849DB1",
                               "lncRNA" = "#B66353",
                               "processed_pseudogene" = "#FBB04E",
                               "novel" = "#B09C85"),
                           "X_nonX" = c("X" = c("red"), "nonX" = "grey"))
  
    
    ## heatmap_all_geneTypes
    ph = pheatmap(nozeros_matrix_log, clustering_distance_cols = "correlation",clustering_distance_rows = "euclidean", clustering_method = "complete",
           # treeheight_row = 0, treeheight_col = 0,
           annotation_row = row_annotation,
           annotation_col= col_annot,
           annotation_names_col = FALSE,
           annotation_names_row = FALSE,
           show_colnames = FALSE,
           # cutree_cols = 3,
           annotation_colors = biotypecolors,
           color=colorRampPalette(c("lightyellow", "red"))(50),
           fontsize = 6,
           main=paste0("Testis/Tumor Specific genes ", ctype_var))
    col.order = ph$tree_col$order
    save_pheatmap_pdf(ph, paste0(plots_wd,"/PDF/pheatmap_5percent_",ctype_var,".pdf"))
    # save_pheatmap_pdf(ph, paste0(plots_wd,"/PDF/pheatmap_",ctype_var,".pdf"))

    
    lncRNA = df_5percent %>% subset(gene_type != "protein_coding") %>% select(gene_name) %>% unique() %>% subset(gene_name %in% df_5percent_expr$gene_name)
    CTA = df_5percent %>% subset(gene_type == "protein_coding") %>% select(gene_name) %>% unique() %>% subset(gene_name %in% df_5percent_expr$gene_name)
  
  
    # ## CTA
    # CTA_matrix = nozeros_matrix_log[CTA$gene_name, col.order]
    # CTA_matrix_non_zero <- CTA_matrix[, apply(CTA_matrix, 2, function(x) any(x != 0))]
    # 
    # cta_ph = pheatmap(CTA_matrix_non_zero, clustering_distance_cols = "correlation",clustering_distance_rows = "euclidean", clustering_method = "complete",
    #        treeheight_row = 0, treeheight_col = 0,
    #        annotation_row = row_annotation,
    #        annotation_col= col_annot,
    #        annotation_names_col = FALSE,
    #        annotation_names_row = FALSE,
    #        show_colnames = FALSE,
    #        # cutree_cols = 3,
    #        annotation_colors = biotypecolors,
    #        color=colorRampPalette(c("lightyellow", "red"))(50),
    #        fontsize = 6,
    #        main=paste0("CTA Testis/Tumor Specific genes ", ctype_var))
    # 
    # save_pheatmap_pdf(cta_ph, paste0(plots_wd,"/PDF/pheatmap_",ctype_var,"_CTA.pdf"))
    # save_pheatmap_pdf(cta_ph, paste0(plots_wd,"/PDF/pheatmap_5percent_",ctype_var,"_CTA.pdf"))


    # ## lncRNA
    # lncRNA_matrix = nozeros_matrix_log[lncRNA$gene_name, col.order]
    # lncRNA_matrix_non_zero <- lncRNA_matrix[, apply(lncRNA_matrix, 2, function(x) any(x != 0))]
    # 
    # lncRNA_ph = pheatmap(lncRNA_matrix_non_zero, clustering_distance_cols = "correlation",clustering_distance_rows = "euclidean", clustering_method = "complete",
    #        treeheight_row = 0, treeheight_col = 0,
    #        annotation_row = row_annotation,
    #        annotation_col= col_annot,
    #        annotation_names_col = FALSE,
    #        annotation_names_row = FALSE,
    #        show_colnames = FALSE,
    #        # cutree_cols = 3,
    #        annotation_colors = biotypecolors,
    #        color=colorRampPalette(c("lightyellow", "red"))(50),
    #        fontsize = 6,
    #        main=paste0("lncRNA Testis/Tumor Specific genes ", ctype_var))
    # 
    # save_pheatmap_pdf(lncRNA_ph, paste0(plots_wd,"/PDF/pheatmap_",ctype_var,"_nonCTA.pdf"))
    # save_pheatmap_pdf(lncRNA_ph, paste0(plots_wd,"/PDF/pheatmap_5percent_",ctype_var,"_nonCTA.pdf"))

    
    ##### correlations
    if (nrow(nozeros_matrix_log) > 4) {
      transposed = t(nozeros_matrix_log)
      res2 <- rcorr(transposed)
      cormat <- round(cor(transposed),2)
      
      ##### ----> CORRPLOT 1 - ALL GENES <----- #####
      p <- {
        corrplot(cormat, type = "lower", order = "hclust", method="circle",
               tl.col = "black", col = rev(COL2('RdBu', 10)), tl.cex=0.6, title=paste0("Correlation all TTS genes ", ctype_var),  mar=c(0,0,1,0));
        recordPlot() 
      }
      
      png(paste0(plots_wd,"/PNG/corrplot_allgenes_TTS_",ctype_var,".png"))
      replayPlot(p)
      dev.off()
      pdf(paste0(plots_wd,"/PDF/corrplot_allgenes_TTS_",ctype_var,".pdf"))
      replayPlot(p)
      dev.off()
    
      ### getting correlation values
      cormat = data.frame(cormat)
      cormat$A = rownames(cormat)
      cormat = cormat %>% pivot_longer(cols=-c(A), names_to = "B", values_to = "correlation")
      
      biotypes = df_5percent %>% select(gene_name, gene_type) %>% distinct()
      lncRNA_biotypes = biotypes %>% subset(gene_type != "protein_coding")
      proteincoding_biotypes = biotypes %>% subset(gene_type == "protein_coding")
      
      cormat = cormat %>% mutate(A_biotype = case_when(A %in% proteincoding_biotypes$gene_name ~ "protein_coding", 
                                                       TRUE ~ "non-coding")) 
      cormat = cormat %>%
        mutate(B_biotype = case_when(B %in% lncRNA_biotypes$gene_name ~ "non-coding", 
                                     TRUE ~ "protein_coding")) 
      
      cormat_lncRNA = cormat %>% subset(A_biotype == "non-coding" & B_biotype == "non-coding")
      
      cormat_proteincoding = cormat %>% subset(A_biotype == "protein_coding" & B_biotype == "protein_coding")
      print("non-coding")
      print(mean(cormat_lncRNA$correlation))
      print("PCG")
      print(mean(cormat_proteincoding$correlation))
    }
    
  ####### 2. non-CTA ####### 
  gene_names_ctype = df_5percent %>% subset(gene_type != "protein_coding") %>% select(gene_name) %>% unique() %>% subset(gene_name %in% df_5percent_expr$gene_name) %>% pull(gene_name)
  lncRNA_matrix = nozeros_matrix_log[gene_names_ctype,]
  
  if (length(gene_names_ctype) > 4) {
      lncRNA_transposed = t(lncRNA_matrix)
      res2 <- rcorr(lncRNA_transposed)
      # A simple function to format the correlation matrix
      # ++++++++++++++++++++++++++++
      # flattenCorrMatrix
      # ++++++++++++++++++++++++++++
      # cormat : matrix of the correlation coefficients
      # pmat : matrix of the correlation p-values
      flattenCorrMatrix(res2$r, res2$P)
      
      lncRNA_cormat <- round(cor(lncRNA_transposed),2)
      dim(lncRNA_cormat)
      ##### ----> CORRPLOT 2 - NON-CTA <----- #####
        p <- {
      corrplot(lncRNA_cormat, type = "lower", order = "hclust", method="circle",
               tl.col = "black", col = rev(COL2('RdBu', 10)), tl.cex=0.6, title=paste0("Correlation non-CTA TTS genes ", ctype_var),  mar=c(0,0,1,0));
        recordPlot() 
      }
      
      png(paste0(plots_wd,"/PNG/corrplot_nonCTA_TTS_",ctype_var,".png"))
      replayPlot(p)
      dev.off()
      pdf(paste0(plots_wd,"/PDF/corrplot_nonCTA_TTS_",ctype_var,".pdf"))
      replayPlot(p)
      dev.off()
  }
  
  ####### 3. CTA ####### 
  gene_names_ctype = df_5percent %>% subset(gene_type == "protein_coding") %>% select(gene_name) %>% unique() %>% subset(gene_name %in% df_5percent_expr$gene_name) %>% pull(gene_name)
  CTA_matrix = nozeros_matrix_log[gene_names_ctype,]
  if (length(gene_names_ctype) > 4) {

    CTA_transposed = t(CTA_matrix)
    res2 <- rcorr(CTA_transposed)
    # A simple function to format the correlation matrix
    # ++++++++++++++++++++++++++++
    # flattenCorrMatrix
    # ++++++++++++++++++++++++++++
    # cormat : matrix of the correlation coefficients
    # pmat : matrix of the correlation p-values
    flattenCorrMatrix(res2$r, res2$P)
    
    CTA_cormat <- round(cor(CTA_transposed),2)
    dim(CTA_cormat)
    
    ##### ----> CORRPLOT 3 - CTA <----- #####
        p <- {
    corrplot(CTA_cormat, type = "lower", order = "hclust", method="circle",
             tl.col = "black", col = rev(COL2('RdBu', 10)), tl.cex=0.6, title=paste0("Correlation CTA TTS genes ", ctype_var),  mar=c(0,0,1,0));
      recordPlot() 
    }
    
    png(paste0(plots_wd,"/PNG/corrplot_CTA_TTS_",ctype_var,".png"))
    replayPlot(p)
    dev.off()
    pdf(paste0(plots_wd,"/PDF/corrplot_CTA_TTS_",ctype_var,".pdf"))
    replayPlot(p)
    dev.off()
    
    CTA_cormat = data.frame(CTA_cormat)
    CTA_cormat$A = rownames(CTA_cormat)
    CTA_cormat = CTA_cormat %>% pivot_longer(cols=-c(A), names_to = "B", values_to = "correlation")
    ## which group we want to compare?
    x_nonX_df = df_5percent %>% select(gene_name, chr) %>% mutate(X_nonX = case_when(chr == "X" ~ "X",
                                                                TRUE ~ "nonX")) %>% select(-chr)
    chr_X = x_nonX_df %>% subset(X_nonX == "X")
    chr_nonX = x_nonX_df %>% subset(X_nonX == "nonX")
    
    
    CTA_cormat = CTA_cormat %>% mutate(A_chr = case_when(A %in% chr_X$gene_name ~ "X", 
                                                     TRUE ~ "non-X")) 
    CTA_cormat = CTA_cormat %>%
      mutate(B_chr = case_when(B %in% chr_nonX$gene_name ~ "non-X", 
                                   TRUE ~ "X")) 
    
    CTA_cormat %>% head
    CTA_cormat_nonX = CTA_cormat %>% subset(A_chr == "non-X" & B_chr == "non-X")
    CTA_cormat_X = CTA_cormat %>% subset(A_chr == "X" & B_chr == "X")
    print("nonX")
    print(mean(CTA_cormat_nonX$correlation))
    print("X")
    print(mean(CTA_cormat_X$correlation))
  }
  ####### 4. CTA-Xchr ####### 
  gene_names_ctype = df_5percent %>% subset(gene_type == "protein_coding")  %>% subset(chr == "X") %>% select(gene_name) %>% unique() %>% subset(gene_name %in% df_5percent_expr$gene_name) %>% pull(gene_name)
  CTA_matrix_X = nozeros_matrix_log[gene_names_ctype,]
  print("Num CT-X")
  print(nrow(CTA_matrix_X))
  
  if (length(gene_names_ctype) > 4) {

    CTAX_transposed = t(CTA_matrix_X)
    res2 <- rcorr(CTAX_transposed)
    # A simple function to format the correlation matrix
    # ++++++++++++++++++++++++++++
    # flattenCorrMatrix
    # ++++++++++++++++++++++++++++
    # cormat : matrix of the correlation coefficients
    # pmat : matrix of the correlation p-values
    flattenCorrMatrix(res2$r, res2$P)
    
    CTAX_cormat <- round(cor(CTAX_transposed),2)
    dim(CTAX_cormat)
    ##### ----> CORRPLOT 4 - CTA-X chr <----- #####
          p <- {
    corrplot(CTAX_cormat, type = "lower", order = "hclust", method="circle",
             tl.col = "black", col = rev(COL2('RdBu', 10)), tl.cex=0.6, title=paste0("Correlation CTA-X chr TTS genes ", ctype_var),  mar=c(0,0,1,0));
      recordPlot() 
    }
    
    png(paste0(plots_wd,"/PNG/corrplot_CTAX_TTS_",ctype_var,".png"))
    replayPlot(p)
    dev.off()
    pdf(paste0(plots_wd,"/PDF/corrplot_CTAX_TTS_",ctype_var,".pdf"))
    replayPlot(p)
    dev.off()
  
  }
  
  ####### 5. CTA-nonXchr ####### 
  gene_names_ctype = df_5percent %>% subset(gene_type == "protein_coding")  %>% subset(chr != "X") %>% select(gene_name) %>% unique() %>% subset(gene_name %in% df_5percent_expr$gene_name) %>% pull(gene_name)
  CTA_matrix_nonX = nozeros_matrix_log[gene_names_ctype,]
  print("Num CT-nonX")
  print(nrow(CTA_matrix_nonX))
  if (length(gene_names_ctype) > 4) {
    CTAnonX_transposed = t(CTA_matrix_nonX)
    res2 <- rcorr(CTAnonX_transposed)
    # A simple function to format the correlation matrix
    # ++++++++++++++++++++++++++++
    # flattenCorrMatrix
    # ++++++++++++++++++++++++++++
    # cormat : matrix of the correlation coefficients
    # pmat : matrix of the correlation p-values
    flattenCorrMatrix(res2$r, res2$P)
      
    CTAnonX_cormat <- round(cor(CTAnonX_transposed),2)
    dim(CTAnonX_cormat)
    ##### ----> CORRPLOT 5 - CTA-nonX chr <----- #####
    p <- {
    corrplot(CTAnonX_cormat, type = "lower", order = "hclust", method="circle",
             tl.col = "black", col = rev(COL2('RdBu', 10)), tl.cex=0.6, title=paste0("Correlation CTA-nonX chr TTS genes ", ctype_var),  mar=c(0,0,1,0));
      recordPlot() 
    }
    
    png(paste0(plots_wd,"/PNG/corrplot_CTAnonX_TTS_",ctype_var,".png"))
    replayPlot(p)
    dev.off()
    pdf(paste0(plots_wd,"/PDF/corrplot_CTAnonX_TTS_",ctype_var,".pdf"))
    replayPlot(p)
    dev.off()
  }
  
  ####### 6. lncRNA ####### 
  gene_names_ctype = df_5percent %>% subset(gene_type == "lncRNA")%>% select(gene_name) %>% unique() %>% subset(gene_name %in% df_5percent_expr$gene_name) %>% pull(gene_name)
  lncRNA_matrix = nozeros_matrix_log[gene_names_ctype,]
  
  if (length(gene_names_ctype) > 4) {
    lncRNA_transposed = t(lncRNA_matrix)
    res2 <- rcorr(lncRNA_transposed)
    # A simple function to format the correlation matrix
    # ++++++++++++++++++++++++++++
    # flattenCorrMatrix
    # ++++++++++++++++++++++++++++
    # cormat : matrix of the correlation coefficients
    # pmat : matrix of the correlation p-values
    flattenCorrMatrix(res2$r, res2$P)
    
    lncRNA_cormat <- round(cor(lncRNA_transposed),2)
    dim(lncRNA_cormat)
    ##### ----> CORRPLOT 6 - lncRNA <----- #####
    p <- {
    corrplot(lncRNA_cormat, type = "lower", order = "hclust", method="circle",
             tl.col = "black", col = rev(COL2('RdBu', 10)), tl.cex=0.6, title=paste0("Correlation lncRNA TTS genes ", ctype_var),  mar=c(0,0,1,0));
      recordPlot() 
    }
    
    png(paste0(plots_wd,"/PNG/corrplot_lncRNA_TTS_",ctype_var,".png"))
    replayPlot(p)
    dev.off()
    pdf(paste0(plots_wd,"/PDF/corrplot_lncRNA_TTS_",ctype_var,".pdf"))
    replayPlot(p)
    dev.off()
  }
  }
}
```

```{r heatmap_corr_per_ctype_all, echo=F}
############ no minimum %  #################
for (ctype_var in cancers) {
  print(ctype_var)
  ## how many patients in that ctype?
  n_ctype = ctype_patients %>% subset(ctype == ctype_var) %>% pull(num_patients)
  
  df = data_completed_TSA_forced %>% subset(ctype == ctype_var)
  
  ## in how many patients is the gene TS?
  counting_1TPM = df %>% select(gene_name, gene_type, patient, forced_tumor_expr) %>% subset(forced_tumor_expr > 0) %>% group_by(gene_name) %>% count()
  df = merge(df, counting_1TPM, by="gene_name")
  df$total_n = n_ctype
  
  ## calculate the percentage and keep only those TS in > 5% 
  df$percentage_n = (df$n / n_ctype)* 100
  df = df %>% unique()
  df_5percent = df #%>% subset(percentage_n > 5)
  ## if dataframe not empty
  if (dim(df_5percent)[1] > 0) {

    df_5percent = df_5percent %>% unique()
    to_table = df_5percent %>% select(gene_name, gene_type) %>% unique()
    print(table(to_table$gene_type))
    
    ## keep expression data to plot
    df_5percent_expr = df_5percent %>% select(gene_name, patient, forced_tumor_expr) %>% pivot_wider(names_from = patient, values_from = forced_tumor_expr) %>% unique()
    df_5percent_expr = as.data.frame(df_5percent_expr)
  
    df_5percent_expr_values = mutate_all(df_5percent_expr[,-1], function(x) as.numeric(as.character(x)))
    ## get log2TPM
    df_log = log2(df_5percent_expr_values+1)
  
    ## convert to matrix
    matrix = as.matrix(df_log)
    rownames(matrix) = df_5percent_expr$gene_name
    matrix = na.omit(matrix)
    nozeros_matrix_log = matrix[,colSums(matrix) > 0]
    
    
    
    ## row annotation
    row_annot = annot %>% subset(gene_name %in% df_5percent_expr$gene_name)
    row_annot = row_annot %>% select(gene_name, gene_type)
    row_annot = unique(row_annot)
    rownames(row_annot) = row_annot$gene_name
    
    row_annot_chr = merge(annot, row_annot, by=c("gene_name"), all.y=T)
    row_annot_chr = row_annot_chr %>% select(gene_name, chr) %>% unique()
    rownames(row_annot_chr) = row_annot_chr$gene_name
    row_annot_chr = row_annot_chr %>% mutate(X_nonX = case_when(chr == "X" ~ "X",
                                                                TRUE ~ "nonX"))
    row_annot_chr = row_annot_chr %>% select(X_nonX)
    row_annot = row_annot %>% select(-c(gene_name))

    row_annotation = data.frame("X_nonX" = row_annot_chr,
                                "gene_type" = row_annot)
    ## col annotation
    col_annot = df_5percent %>% select(patient, project) %>% subset(patient %in% colnames(df_5percent_expr_values))
    col_annot = unique(col_annot)
    rownames(col_annot) = col_annot$patient
    col_annot = col_annot %>% select(-c(patient))
  
    biotypecolors = list("gene_type" = c("protein_coding"="#849DB1",
                               "lncRNA" = "#B66353",
                               "processed_pseudogene" = "#FBB04E",
                               "novel" = "#B09C85"),
                           "X_nonX" = c("X" = c("red"), "nonX" = "grey"))
  
    
    ## heatmap_all_geneTypes
    ph = pheatmap(nozeros_matrix_log, clustering_distance_cols = "correlation",clustering_distance_rows = "euclidean", clustering_method = "complete",
           # treeheight_row = 0, treeheight_col = 0,
           annotation_row = row_annotation,
           annotation_col= col_annot,
           annotation_names_col = FALSE,
           annotation_names_row = FALSE,
           show_colnames = FALSE,
           # cutree_cols = 3,
           annotation_colors = biotypecolors,
           color=colorRampPalette(c("lightyellow", "red"))(50),
           fontsize = 6,
           main=paste0("Testis/Tumor Specific genes ", ctype_var))
    col.order = ph$tree_col$order
    # save_pheatmap_pdf(ph, paste0(plots_wd,"/PDF/pheatmap_5percent_",ctype_var,".pdf"))
    save_pheatmap_pdf(ph, paste0(plots_wd,"/PDF/pheatmap_",ctype_var,".pdf"))


  ##### correlations
    if (nrow(nozeros_matrix_log) > 4) {
      transposed = t(nozeros_matrix_log)
      res2 <- rcorr(transposed)
      cormat <- round(cor(transposed),2)

    
      ### getting correlation values
      cormat = data.frame(cormat)
      cormat$A = rownames(cormat)
      cormat = cormat %>% pivot_longer(cols=-c(A), names_to = "B", values_to = "correlation")
      
      biotypes = df_5percent %>% select(gene_name, gene_type) %>% distinct()
      lncRNA_biotypes = biotypes %>% subset(gene_type != "protein_coding")
      proteincoding_biotypes = biotypes %>% subset(gene_type == "protein_coding")
      
      cormat = cormat %>% mutate(A_biotype = case_when(A %in% proteincoding_biotypes$gene_name ~ "protein_coding", 
                                                       TRUE ~ "non-coding")) 
      cormat = cormat %>%
        mutate(B_biotype = case_when(B %in% lncRNA_biotypes$gene_name ~ "non-coding", 
                                     TRUE ~ "protein_coding")) 
      
      cormat_lncRNA = cormat %>% subset(A_biotype == "non-coding" & B_biotype == "non-coding")
      
      cormat_proteincoding = cormat %>% subset(A_biotype == "protein_coding" & B_biotype == "protein_coding")
      print("non-coding")
      print(mean(cormat_lncRNA$correlation))
      print("PCG")
      print(mean(cormat_proteincoding$correlation))
    }
    
  ####### 2. non-CTA ####### 
  gene_names_ctype = df_5percent %>% subset(gene_type != "protein_coding") %>% select(gene_name) %>% unique() %>% subset(gene_name %in% df_5percent_expr$gene_name) %>% pull(gene_name)
  lncRNA_matrix = nozeros_matrix_log[gene_names_ctype,]
  
  if (length(gene_names_ctype) > 4) {
      lncRNA_transposed = t(lncRNA_matrix)
      res2 <- rcorr(lncRNA_transposed)
      # A simple function to format the correlation matrix
      # ++++++++++++++++++++++++++++
      # flattenCorrMatrix
      # ++++++++++++++++++++++++++++
      # cormat : matrix of the correlation coefficients
      # pmat : matrix of the correlation p-values
      flattenCorrMatrix(res2$r, res2$P)
      
      lncRNA_cormat <- round(cor(lncRNA_transposed),2)
      dim(lncRNA_cormat)
  }
  
  ####### 3. CTA ####### 
  gene_names_ctype = df_5percent %>% subset(gene_type == "protein_coding") %>% select(gene_name) %>% unique() %>% subset(gene_name %in% df_5percent_expr$gene_name) %>% pull(gene_name)
  CTA_matrix = nozeros_matrix_log[gene_names_ctype,]
  if (length(gene_names_ctype) > 4) {

    CTA_transposed = t(CTA_matrix)
    res2 <- rcorr(CTA_transposed)
    # A simple function to format the correlation matrix
    # ++++++++++++++++++++++++++++
    # flattenCorrMatrix
    # ++++++++++++++++++++++++++++
    # cormat : matrix of the correlation coefficients
    # pmat : matrix of the correlation p-values
    flattenCorrMatrix(res2$r, res2$P)
    
    CTA_cormat <- round(cor(CTA_transposed),2)
    dim(CTA_cormat)

    
    CTA_cormat = data.frame(CTA_cormat)
    CTA_cormat$A = rownames(CTA_cormat)
    CTA_cormat = CTA_cormat %>% pivot_longer(cols=-c(A), names_to = "B", values_to = "correlation")
    ## which group we want to compare?
    x_nonX_df = df_5percent %>% select(gene_name, chr) %>% mutate(X_nonX = case_when(chr == "X" ~ "X",
                                                                TRUE ~ "nonX")) %>% select(-chr)
    chr_X = x_nonX_df %>% subset(X_nonX == "X")
    chr_nonX = x_nonX_df %>% subset(X_nonX == "nonX")
    
    
    CTA_cormat = CTA_cormat %>% mutate(A_chr = case_when(A %in% chr_X$gene_name ~ "X", 
                                                     TRUE ~ "non-X")) 
    CTA_cormat = CTA_cormat %>%
      mutate(B_chr = case_when(B %in% chr_nonX$gene_name ~ "non-X", 
                                   TRUE ~ "X")) 
    
    CTA_cormat %>% head
    CTA_cormat_nonX = CTA_cormat %>% subset(A_chr == "non-X" & B_chr == "non-X")
    CTA_cormat_X = CTA_cormat %>% subset(A_chr == "X" & B_chr == "X")
    print("nonX")
    print(mean(CTA_cormat_nonX$correlation))
    print("X")
    print(mean(CTA_cormat_X$correlation))
  }
  ####### 4. CTA-Xchr ####### 
  gene_names_ctype = df_5percent %>% subset(gene_type == "protein_coding")  %>% subset(chr == "X") %>% select(gene_name) %>% unique() %>% subset(gene_name %in% df_5percent_expr$gene_name) %>% pull(gene_name)
  CTA_matrix_X = nozeros_matrix_log[gene_names_ctype,]
  print("Num CT-X")
  print(nrow(CTA_matrix_X))
  if (length(gene_names_ctype) > 4) {

    CTAX_transposed = t(CTA_matrix_X)
    res2 <- rcorr(CTAX_transposed)
    # A simple function to format the correlation matrix
    # ++++++++++++++++++++++++++++
    # flattenCorrMatrix
    # ++++++++++++++++++++++++++++
    # cormat : matrix of the correlation coefficients
    # pmat : matrix of the correlation p-values
    flattenCorrMatrix(res2$r, res2$P)
    
    CTAX_cormat <- round(cor(CTAX_transposed),2)
    dim(CTAX_cormat)
  
  }
  
  ####### 5. CTA-nonXchr ####### 
  gene_names_ctype = df_5percent %>% subset(gene_type == "protein_coding")  %>% subset(chr != "X") %>% select(gene_name) %>% unique() %>% subset(gene_name %in% df_5percent_expr$gene_name) %>% pull(gene_name)
  CTA_matrix_nonX = nozeros_matrix_log[gene_names_ctype,]
  print("Num CT-nonX")
  print(nrow(CTA_matrix_nonX))
  if (length(gene_names_ctype) > 4) {
    CTAnonX_transposed = t(CTA_matrix_nonX)
    res2 <- rcorr(CTAnonX_transposed)
    # A simple function to format the correlation matrix
    # ++++++++++++++++++++++++++++
    # flattenCorrMatrix
    # ++++++++++++++++++++++++++++
    # cormat : matrix of the correlation coefficients
    # pmat : matrix of the correlation p-values
    flattenCorrMatrix(res2$r, res2$P)
      
    CTAnonX_cormat <- round(cor(CTAnonX_transposed),2)
    dim(CTAnonX_cormat)
  }
  
  ####### 6. lncRNA ####### 
  gene_names_ctype = df_5percent %>% subset(gene_type == "lncRNA")%>% select(gene_name) %>% unique() %>% subset(gene_name %in% df_5percent_expr$gene_name) %>% pull(gene_name)
  lncRNA_matrix = nozeros_matrix_log[gene_names_ctype,]
  
  if (length(gene_names_ctype) > 4) {
    lncRNA_transposed = t(lncRNA_matrix)
    res2 <- rcorr(lncRNA_transposed)
    # A simple function to format the correlation matrix
    # ++++++++++++++++++++++++++++
    # flattenCorrMatrix
    # ++++++++++++++++++++++++++++
    # cormat : matrix of the correlation coefficients
    # pmat : matrix of the correlation p-values
    flattenCorrMatrix(res2$r, res2$P)
    
    lncRNA_cormat <- round(cor(lncRNA_transposed),2)
    dim(lncRNA_cormat)

  }
  }
}
```

```{r plot_corr_all, echo=F}
corrplot_values_all = read.csv(file.path(save_wd,"corrplot_values.csv"))
names(corrplot_values_all)[1] = "type"
corrplot_values_all = corrplot_values_all %>% pivot_longer(cols=-c("type"), names_to = "ctype", values_to = "coexpression_corr")
corrplot_values_all$coexpression_corr = as.numeric(corrplot_values_all$coexpression_corr)
######### plot
ggplot(corrplot_values_all %>% subset(type != "CT"), aes(x=ctype, y=coexpression_corr, color=type)) +
  geom_point(size=5) +
  scale_color_manual(values = c("Non-coding" = "#bcbddc",
                                "CT-nonX" = "#807dba",
                                "CT-X" = "#3f007d")) +
  labs(y="Coexpression coefficient",
       x="",
       color="Condition",
       title="Comparison of coexpression Correlation per condition") +
  theme_minimal()
ggsave(file.path(plots_wd,"PNG/coexpr_coef_TSAall.png"))
ggsave(file.path(plots_wd,"PDF/coexpr_coef_TSAall.pdf"))
```
