---
title: "4. Testis-specific / Tumor-specific translated ncORFs"
author: "Marta Espinosa"
date: "2024-07-10"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
library(dplyr)
library(tidyr)
library(ggplot2)
library(stringr)
library(purrr)
library(ggvenn)
library(ggpubr)
library(ggside)
library(bioseq)
# library(ggthemr)
library(ggbeeswarm)
library(ggbreak)

annot = read.csv("/users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/human/newReference_Resconstructed/1transcript_1gene.reconstructed.csv")
# names(gene_transcript) = c("gene_id","gene_type","gene_name")
# gene_transcript$gene_id = gsub("\\..*","",gene_transcript$gene_id)

Q2_wd = "/projects_eg/projects/marta/TestisRestricted_Microproteins_TSA/with_TranscriptomeReconstruction/Q2_TestisRestricted/human"

pre_save_wd = "/projects_eg/projects/marta/TestisRestricted_Microproteins_TSA/with_TranscriptomeReconstruction/Q3_TestisRestrictedTumors/human"
plots_wd = "/projects_eg/projects/marta/TestisRestricted_Microproteins_TSA/with_TranscriptomeReconstruction/Q4_TestisRestricted_TumorSpecific/human/plots"
save_wd = "/projects_eg/projects/marta/TestisRestricted_Microproteins_TSA/with_TranscriptomeReconstruction/Q4_TestisRestricted_TumorSpecific/human"

cancers = c("BRCA","BLCA","LUAD","KIRC","KIRP","PRAD","LUSC","COAD","LIHC")
tcga_projects=c("TCGA-BRCA","TCGA-LUSC","TCGA-PRAD","TCGA-KIRC","TCGA-KIRP","TCGA-LUAD","TCGA-BLCA")#,"TCGA-LIHC"]
other_projects=c("GSE102101_KIRC","GSE133624_BLCA","GSE22260_PRAD","PRJEB2449_PRAD","SRP238334_KIRC","GSE214846_LIHC","GSE229705_LUAD","TCGA_COAD","SRP107326_COAD")
manuscript_projects = c("liver_adjacent_totalRNA_LIHC","hcc_normal_totalRNA_LIHC","GSE193567_LIHC","LIHC_TCGA_LIHC")
# deleted_projects=c("GSE103001_BRCA","GSE89223_PRAD")
all_projects = c(tcga_projects,other_projects,manuscript_projects)

cancers_dir = "/users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/cancers"
```

## Testis-Specific Translated ORFs

```{r input_translation_data}
print("TestisRestricted with GTEx")
testisRestricted_GTEx = read.csv(file.path(Q2_wd,"testisRestricted_GTEx.csv"))
table(testisRestricted_GTEx$gene_type)
print("Translated IN testis")
translated_IN_testis = read.csv(file.path(Q2_wd,"testisRestricted_GTEx_translatedINtestis.csv"))
table(translated_IN_testis$gene_type)
print("Corresponding to how many genes?")
translated_IN_testis_genes = translated_IN_testis %>% select(gene_type, gene_name) %>% unique()
table(translated_IN_testis_genes$gene_type)

print("Translated ONLY in testis")
translated_only_testis = read.csv(file.path(Q2_wd,"testisRestricted_GTEx_translatedONLYtestis.csv"))
table(translated_only_testis$gene_type)
print("Corresponding to how many genes?")
translated_only_testis_genes = translated_only_testis %>% select(gene_type, gene_name) %>% unique()
table(translated_only_testis_genes$gene_type)

## add column to translated IN testis
translated_IN_testis = translated_IN_testis %>% mutate(ONLYtranslation_testis = case_when(transcript_id %in% translated_only_testis$transcript_id ~ "ONLYTestis",
                                                                                          TRUE ~ "several_tissues"))
table(translated_IN_testis$ONLYtranslation_testis, translated_IN_testis$gene_type)
```

## Tumor Specific Antigens (TSA)

```{r cancer_TSA, echo=F}
TSA_GTEX_all = data.frame("gene_id" = character(),
                    "transcript_id" = character(),
                    "gene_name" = character(),
                    "gene_type" = factor(), 
                    "Length" = numeric(),
                    "n"=numeric(),
                    "ctype" =character())

for (ctype in cancers) {
  print(ctype)
  
  TSA_GTEx = read.csv(paste0(file.path(cancers_dir,"tumorspecific/cancertypes/GTEx"),"/tumorspecific_n_GTEx_",ctype,"_TOLERANCE5PERCENT.csv"))
  TSA_GTEx$ctype = ctype
  
  TSA_GTEX_all = rbind(TSA_GTEX_all, TSA_GTEx)
}
table(TSA_GTEX_all$ctype, TSA_GTEX_all$gene_type)


ggplot(TSA_GTEX_all %>% unique() %>% group_by(gene_type, ctype) %>% count(), aes(x=n, y=ctype, fill=gene_type)) +
  geom_bar(stat="identity", position="dodge") +
    scale_fill_manual(values=c("protein_coding"="#849DB1",
                             "lncRNA" = "#B66353",
                             "processed_pseudogene" = "#FBB04E",
                             "novel" = "#B09C85")) +
  ggtitle("Tumor-Specific genes") +
  theme_classic()
ggsave(file.path(plots_wd,"PNG/tumorspecific_ctype.png"), width=5.41, height=4.66)
ggsave(file.path(plots_wd,"PDF/tumorspecific_ctype.pdf"), width=5.41, height=4.66)

ggplot(TSA_GTEX_all %>% unique() %>% group_by(gene_type, ctype, n) %>% count(), aes(x=n, y=nn, color=gene_type)) +
  geom_point(stat="identity", position="dodge") +
  geom_line() +
  scale_color_manual(values=c("protein_coding"="#849DB1",
                             "lncRNA" = "#B66353",
                             "processed_pseudogene" = "#FBB04E",
                             "novel" = "#B09C85")) +
  scale_y_continuous(trans="log10") +
  labs(title="Tumor-Specific genes",
       y="Number of genes",
       x="Number of patients") +
  theme_classic() +
  theme(legend.position = "top") +
  facet_wrap(~ ctype, scales="free", ncol=2)
ggsave(file.path(plots_wd,"PNG/tumorspecific_ctype_numpatients.png"), width=6.32, height=7.86)
ggsave(file.path(plots_wd,"PDF/tumorspecific_ctype_numpatients.pdf"), width=6.32, height=7.86)
```

```{r}
TSA_testis = TSA_GTEX_all %>% mutate(testisRestricted_GTEx = case_when(gene_id %in% testisRestricted_GTEx$gene_id ~ "testisRestricted_GTEx",
                                                                       TRUE ~ "no"))

TSA_testis_proportions = TSA_testis %>%
  group_by(gene_type, ctype, testisRestricted_GTEx) %>% 
  summarise(count = n()) %>%                            
  mutate(proportion = count / sum(count)) %>%           
  ungroup()                                             

average_proportions = TSA_testis_proportions %>% subset(testisRestricted_GTEx == "testisRestricted_GTEx") %>%
  group_by(gene_type) %>%
  summarise(avg_proportion = mean(proportion))


ggplot(TSA_testis_proportions, aes(x=ctype, y=proportion, fill=testisRestricted_GTEx)) +
  geom_bar(stat="identity") +
      scale_fill_manual(values=c("testisRestricted_GTEx" = "#B1283A", "no"="#A8A6A7")) +
  theme_classic() +
  theme(axis.text.x = element_text(angle=90),
        legend.position = "top") +
  facet_wrap(~ gene_type) +
  geom_hline(data = average_proportions, aes(yintercept = avg_proportion), color="black", linetype="dashed") +
    geom_text(data = average_proportions, aes(x = Inf, y = avg_proportion, label = round(avg_proportion, 2)),
            color = "black", vjust = -0.5, hjust = 1.1, size = 3, inherit.aes = FALSE) +
  labs(title = "Proportion of Testis-Restricted genes",
       y="Proportion",
       y="Cancer Type")
ggsave(file.path(plots_wd,"PNG/tumorspecific_testisrestricted_proportions.png"), height=5.25, width=6.29)
ggsave(file.path(plots_wd,"PDF/tumorspecific_testisrestricted_proportions.pdf"), height=5.25, width=6.29)
```

## Those that are testis-specific, do they tend to be more shared across cancer types?

```{r testisExpressed_translated_tumorspecific, echo=F}
TSA_testis = TSA_GTEX_all %>% mutate(testisRestricted_GTEx = case_when(gene_id %in% testisRestricted_GTEx$gene_id ~ "testisRestricted_GTEx",
                                                                       TRUE ~ "no"))
TSA_testis = TSA_testis %>% mutate(translatedONLYtestis = case_when(gene_id %in% translated_only_testis$gene_id ~ "yes",
                                                                    TRUE ~ "no"))

## only the specifics
TSA_testisSpecific = TSA_testis %>% subset(testisRestricted_GTEx == "testisRestricted_GTEx" & translatedONLYtestis == "yes")
table(TSA_testisSpecific$gene_type)
acrossCancers_TSA_testisSpecific = TSA_testisSpecific %>% group_by(gene_name, gene_type) %>% count()
acrossCancers_TSA_testisSpecific = acrossCancers_TSA_testisSpecific %>% group_by(gene_type, n) %>% count()
names(acrossCancers_TSA_testisSpecific) = c("gene_type","n_cancertypes","num_genes")
acrossCancers_TSA_testisSpecific$type = "specific"

acrossCancers_all = TSA_testis %>% group_by(gene_name, gene_type) %>% count()

acrossCancers_all = acrossCancers_all %>% group_by(gene_type, n) %>% count()
names(acrossCancers_all) = c("gene_type","n_cancertypes","num_genes")
acrossCancers_all$type = "all"

## normalize because non specific are more
acrossCancers_normalized = acrossCancers %>%
  group_by(type) %>%
  mutate(total_genes = sum(num_genes),
         normalized_num_genes = num_genes / total_genes)

# acrossCancers = rbind(acrossCancers_TSA_testisSpecific, acrossCancers_all)

ggplot(acrossCancers_normalized, aes(x=n_cancertypes, y=normalized_num_genes, color=type)) +
  geom_point() +
  geom_line() +
  scale_color_manual(values=c("specific" = "#B1283A", "all"="#A8A6A7")) +
  theme_classic() +
  facet_wrap(~ gene_type)

## statistical model
model <- lm(num_genes ~ n_cancertypes * type, data = acrossCancers)
summary(model)

########## INTERPRETATION ##########
# The coefficient for n_cancertypes is -174.49, which means that for each additional cancer type, the number of genes is expected to decrease by approximately 174.49, holding type constant. This coefficient is statistically significant (p-value < 0.001), suggesting a strong negative relationship between the number of cancer types and the number of genes.


# In conclusion, rows with type == "specific" do tend to have more genes with higher n_cancertypes compared to other types, but the relationship is less negative due to the interaction effect. This means that while "specific" types generally have fewer genes when n_cancertypes is low, this trend diminishes as n_cancertypes increases.

model_gene_type <- lm(num_genes ~ n_cancertypes * type * gene_type, data = acrossCancers)
summary(model_gene_type)

#Novel Genes: "Novel" gene types also see an increase in the number of genes with more cancer types, highlighting their unique behavior compared to other gene types.
```


```{r plot_to_delete, echo=F}
# shared_counts <- TSA_testis %>%
#   group_by(transcript_id, gene_type, TestisINFO) %>%
#   summarise(unique_ctypes = n_distinct(ctype)) %>%
#   ungroup()
# 
# ggplot(shared_counts, aes(x=factor(TestisINFO, levels=c("no","TestisRestrictedUNTranslated","TestisRestrictedTranslated")), y=unique_ctypes, fill=TestisINFO)) +
#   geom_boxplot() +
#   scale_fill_manual(values=c("TestisRestrictedTranslated" = "#B1283A","TestisRestrictedUNTranslated"="#de97a0", "no"="#A8A6A7")) +
#   theme_classic() +
#   labs(title="Those testis-specific, do they tend to be\nmore shared across cancer types?",
#        y="Number of cancer types",
#        x="Testis information") +
#   theme(axis.text.x=element_blank(),
#         axis.ticks.x=element_blank(),
#         legend.position = "top") +
#   facet_wrap(~ gene_type) +
#     stat_compare_means(comparisons = list(c("no", "TestisRestrictedUNTranslated"), 
#                                         c("no", "TestisRestrictedTranslated"), 
#                                         c("TestisRestrictedUNTranslated", "TestisRestrictedTranslated")),
#                      method = "wilcox.test", 
#                      label = "p.signif",
#                      aes(label = ..p.adj..),
#                      hide.ns = TRUE)
# ggsave(file.path(plots_wd,"PNG/tumorspecific_testisrestricted_numctypes.png"), height=8.26, width=7.08)
# ggsave(file.path(plots_wd,"PDF/tumorspecific_testisrestricted_numctypes.pdf"), height=8.26, width=7.08)  
```


##  How many patients express the tumor-specific transcripts ?

```{r num_patients, echo=F}
TSA_testisSpecific

summary_stats <- TSA_testisSpecific %>%
  group_by(ctype, gene_type) %>%
  summarise(
    # min_length = min(n, na.rm = TRUE),
    # q1_length = quantile(n, 0.25, na.rm = TRUE),
    median_length = median(n, na.rm = TRUE),
    mean_length = mean(n, na.rm = TRUE),
    # q3_length = quantile(n, 0.75, na.rm = TRUE),
    max_length = max(n, na.rm = TRUE),
    .groups = 'drop'
  )

print(summary_stats)

# ggplot(TSA_testisSpecific, aes(x=n, color=gene_type)) +
#   geom_point(stat="count", position="dodge") +
# 
#     scale_color_manual(values=c("protein_coding"="#849DB1",
#                              "lncRNA" = "#B66353",
#                              "processed_pseudogene" = "#FBB04E",
#                              "novel" = "#B09C85")) +
#   scale_y_continuous(trans="log10") +
#   labs(title="Tumor-Specific genes translated in Testis",
#        y="Number of genes",
#        x="Number of patients") +
#   theme_classic() +
#   theme(legend.position = "top") +
#   facet_wrap(~ ctype, scales="free", ncol=2)
# ggsave(file.path(plots_wd,"PNG/tumorspecific_testistranslated_ctypes.png"), width=5.38, height=6.84)
# ggsave(file.path(plots_wd,"PDF/tumorspecific_testistranslated_ctypes.pdf"), width=5.38, height=6.84)

# TSA_testis_bigN = TSA_testis %>% subset(TestisINFO == "TestisRestrictedTranslated") %>% group_by(gene_name) %>% mutate(bigN = sum(n)) %>% select(gene_name, gene_type, bigN, TestisINFO) %>% unique() %>% mutate(bigN_proportion = (bigN/760)*100)
# 
# ggplot(TSA_testis_bigN, aes(x=gene_type, y=bigN_proportion, fill=gene_type)) +
#   geom_violin() +
#     scale_fill_manual(values=c("protein_coding"="#849DB1",
#                              "lncRNA" = "#B66353",
#                              "processed_pseudogene" = "#FBB04E",
#                              "novel" = "#B09C85")) +
#   labs(y="Percentage of patients (out of 760)",
#        x="",
#        title="How many patients express the tumor-specific transcripts?") +
#   theme_classic() +
#   theme(legend.position = "none")
# ggsave(file.path(plots_wd,"PNG/tumorspecificTranslated_percentagepatients.png"), width=5.38, height=4.26)
# ggsave(file.path(plots_wd,"PDF/tumorspecificTranslated_percentagepatients.pdf"), width=5.38, height=4.26)
#
# summary(TSA_testis_bigN$bigN_proportion)
# summary(TSA_testis_bigN$bigN)
```
