---
title: "4. Testis-specific / Tumor-specific translated ncORFs"
author: "Marta Espinosa"
date: "2024-07-10"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
library(dplyr)
library(tidyr)
library(ggplot2)
library(stringr)
library(purrr)
library(ggvenn)
library(ggpubr)
library(ggside)
library(bioseq)
# library(ggthemr)
library(ggbeeswarm)
library(ggbreak)

annot = read.csv("/users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/human/newReference_Resconstructed/1transcript_1gene.reconstructed.csv")
# names(gene_transcript) = c("gene_id","gene_type","gene_name")
# gene_transcript$gene_id = gsub("\\..*","",gene_transcript$gene_id)

Q2_wd = "/projects_eg/projects/marta/TestisRestricted_Microproteins_TSA/with_TranscriptomeReconstruction/Q2_TestisRestricted/human"

pre_save_wd = "/projects_eg/projects/marta/TestisRestricted_Microproteins_TSA/with_TranscriptomeReconstruction/Q3_TestisRestrictedTumors/human"
plots_wd = "/projects_eg/projects/marta/TestisRestricted_Microproteins_TSA/with_TranscriptomeReconstruction/Q4_TestisRestricted_TumorSpecific/human/plots"
save_wd = "/projects_eg/projects/marta/TestisRestricted_Microproteins_TSA/with_TranscriptomeReconstruction/Q4_TestisRestricted_TumorSpecific/human"

cancers = c("BRCA","BLCA","LUAD","KIRC","KIRP","PRAD","LUSC","COAD","LIHC")
tcga_projects=c("TCGA-BRCA","TCGA-LUSC","TCGA-PRAD","TCGA-KIRC","TCGA-KIRP","TCGA-LUAD","TCGA-BLCA")#,"TCGA-LIHC"]
other_projects=c("GSE102101_KIRC","GSE133624_BLCA","GSE22260_PRAD","PRJEB2449_PRAD","SRP238334_KIRC","GSE214846_LIHC","GSE229705_LUAD","TCGA_COAD","SRP107326_COAD")
manuscript_projects = c("liver_adjacent_totalRNA_LIHC","hcc_normal_totalRNA_LIHC","GSE193567_LIHC","LIHC_TCGA_LIHC")
# deleted_projects=c("GSE103001_BRCA","GSE89223_PRAD")
all_projects = c(tcga_projects,other_projects,manuscript_projects)

cancers_dir = "/users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/cancers"
```

## Testis-Specific Translated ORFs

```{r input_translation_data}
print("TestisRestricted with GTEx")
testisRestricted_GTEx = read.csv(file.path(Q2_wd,"testisRestricted_GTEx.csv"))
table(testisRestricted_GTEx$gene_type)
print("Translated IN testis")
translated_IN_testis = read.csv(file.path(Q2_wd,"testisRestricted_GTEx_translatedINtestis.csv"))
table(translated_IN_testis$gene_type)
print("Corresponding to how many genes?")
translated_IN_testis_genes = translated_IN_testis %>% select(gene_type, gene_name) %>% unique()
table(translated_IN_testis_genes$gene_type)

print("Translated ONLY in testis")
translated_only_testis = read.csv(file.path(Q2_wd,"testisRestricted_GTEx_translatedONLYtestis.csv"))
table(translated_only_testis$gene_type)
print("Corresponding to how many genes?")
translated_only_testis_genes = translated_only_testis %>% select(gene_type, gene_name) %>% unique()
table(translated_only_testis_genes$gene_type)

## add column to translated IN testis
translated_IN_testis = translated_IN_testis %>% mutate(ONLYtranslation_testis = case_when(transcript_id %in% translated_only_testis$transcript_id ~ "ONLYTestis",
                                                                                          TRUE ~ "several_tissues"))
table(translated_IN_testis$ONLYtranslation_testis, translated_IN_testis$gene_type)
```

## Tumor Specific Antigens (TSA)

```{r cancer_TSA, echo=F}
TSA_GTEX_all = data.frame("gene_id" = character(),
                    "transcript_id" = character(),
                    "gene_name" = character(),
                    "gene_type" = factor(), 
                    "Length" = numeric(),
                    "n"=numeric(),
                    "ctype" =character())

for (ctype in cancers) {
  print(ctype)
  
  TSA_GTEx = read.csv(paste0(file.path(cancers_dir,"tumorspecific/cancertypes/GTEx"),"/tumorspecific_n_GTEx_",ctype,"_TOLERANCE5PERCENT.csv"))
  TSA_GTEx$ctype = ctype
  
  TSA_GTEX_all = rbind(TSA_GTEX_all, TSA_GTEx)
}
table(TSA_GTEX_all$ctype, TSA_GTEX_all$gene_type)


ggplot(TSA_GTEX_all %>% unique() %>% group_by(gene_type, ctype) %>% count(), aes(x=n, y=ctype, fill=gene_type)) +
  geom_bar(stat="identity", position="dodge") +
    scale_fill_manual(values=c("protein_coding"="#849DB1",
                             "lncRNA" = "#B66353",
                             "processed_pseudogene" = "#FBB04E",
                             "novel" = "#B09C85")) +
  ggtitle("Tumor-Specific genes") +
  theme_classic()
ggsave(file.path(plots_wd,"PNG/tumorspecific_ctype.png"), width=5.41, height=4.66)
ggsave(file.path(plots_wd,"PDF/tumorspecific_ctype.pdf"), width=5.41, height=4.66)

ggplot(TSA_GTEX_all %>% unique() %>% group_by(gene_type, ctype, n) %>% count(), aes(x=n, y=nn, color=gene_type)) +
  geom_point(stat="identity", position="dodge") +
  geom_line() +
  scale_color_manual(values=c("protein_coding"="#849DB1",
                             "lncRNA" = "#B66353",
                             "processed_pseudogene" = "#FBB04E",
                             "novel" = "#B09C85")) +
  scale_y_continuous(trans="log10") +
  labs(title="Tumor-Specific genes",
       y="Number of genes",
       x="Number of patients") +
  theme_classic() +
  theme(legend.position = "top") +
  facet_wrap(~ ctype, scales="free", ncol=2)
ggsave(file.path(plots_wd,"PNG/tumorspecific_ctype_numpatients.png"), width=6.32, height=7.86)
ggsave(file.path(plots_wd,"PDF/tumorspecific_ctype_numpatients.pdf"), width=6.32, height=7.86)
```

```{r TestisRestricted_TumorSpecific, echo=F}
TSA_testis = TSA_GTEX_all %>% mutate(testisRestricted_GTEx = case_when(gene_id %in% testisRestricted_GTEx$gene_id ~ "testisRestricted_GTEx",
                                                                       TRUE ~ "no"))

TSA_testis_proportions = TSA_testis %>%
  group_by(gene_type, ctype, testisRestricted_GTEx) %>% 
  summarise(count = n()) %>%                            
  mutate(proportion = count / sum(count)) %>%           
  ungroup()                                             

average_proportions = TSA_testis_proportions %>% subset(testisRestricted_GTEx == "testisRestricted_GTEx") %>%
  group_by(gene_type) %>%
  summarise(avg_proportion = mean(proportion))


ggplot(TSA_testis_proportions, aes(x=ctype, y=proportion, fill=testisRestricted_GTEx)) +
  geom_bar(stat="identity") +
      scale_fill_manual(values=c("testisRestricted_GTEx" = "#B1283A", "no"="#A8A6A7")) +
  theme_classic() +
  theme(axis.text.x = element_text(angle=90),
        legend.position = "top") +
  facet_wrap(~ gene_type) +
  geom_hline(data = average_proportions, aes(yintercept = avg_proportion), color="black", linetype="dashed") +
    geom_text(data = average_proportions, aes(x = Inf, y = avg_proportion, label = round(avg_proportion, 2)),
            color = "black", vjust = -0.5, hjust = 1.1, size = 3, inherit.aes = FALSE) +
  labs(title = "Proportion of Testis-Restricted genes",
       y="Proportion",
       y="Cancer Type")
ggsave(file.path(plots_wd,"PNG/tumorspecific_testisrestricted_proportions.png"), height=5.25, width=6.29)
ggsave(file.path(plots_wd,"PDF/tumorspecific_testisrestricted_proportions.pdf"), height=5.25, width=6.29)
```

## Those that are testis-specific, do they tend to be more shared across cancer types?

```{r testisExpressed_translated_tumorspecific, echo=F}
TSA_testis = TSA_GTEX_all %>% mutate(testisRestricted_GTEx = case_when(gene_id %in% testisRestricted_GTEx$gene_id ~ "testisRestricted_GTEx",
                                                                       TRUE ~ "no"))
TSA_testis = TSA_testis %>% mutate(translatedONLYtestis = case_when(gene_id %in% translated_only_testis$gene_id ~ "yes",
                                                                    TRUE ~ "no"))

## only the specifics
TSA_testisSpecific = TSA_testis %>% subset(testisRestricted_GTEx == "testisRestricted_GTEx" & translatedONLYtestis == "yes")
table(TSA_testisSpecific$gene_type)
table(TSA_testisSpecific$gene_type, TSA_testisSpecific$ctype)

TSA_testisSpecific_orfs = translated_only_testis %>% subset(gene_id %in% TSA_testisSpecific$gene_id)
TSA_testisSpecific_orfs = merge(TSA_testisSpecific_orfs, TSA_testisSpecific, by=c("gene_id", "gene_name", "gene_type", "transcript_id"))
table(TSA_testisSpecific_orfs$gene_type, TSA_testisSpecific_orfs$ctype)
write.csv(TSA_testisSpecific_orfs, file.path(save_wd,"TSA_TestisRestrictedGTEx_Translated_Ctypes.csv"), row.names = F)
```

```{r upset, echo=F}
library(UpSetR)

to_upset = TSA_testisSpecific %>% select(ctype, gene_name) %>% unique()
to_upset$presence = 1
to_upset = to_upset %>% pivot_wider(names_from = "ctype", values_from = "presence")
to_upset[is.na(to_upset)] = 0
to_upset = as.data.frame(to_upset)
rownames(to_upset) = to_upset$gene_name
to_upset$gene_name = NULL

geneName_geneType = annot %>% subset(gene_name %in% rownames(to_upset)) %>% select(gene_name, gene_type)

upset(to_upset, 
      sets=colnames(to_upset)
)
```

```{r upset_CTA, echo=F}
CTA_geneName_geneType = geneName_geneType %>% subset(gene_type == "protein_coding")


to_upset_CTA = to_upset %>% subset(rownames(to_upset) %in% CTA_geneName_geneType$gene_name)
upset(to_upset_CTA, 
      sets=colnames(to_upset_CTA))
```

```{r upset_nonCTA, echo=F}
nonCTA_geneName_geneType = geneName_geneType %>% subset(gene_type != "protein_coding")


to_upset_nonCTA = to_upset %>% subset(rownames(to_upset) %in% nonCTA_geneName_geneType$gene_name)
upset(to_upset_nonCTA, 
      sets=colnames(to_upset_nonCTA)
)
```

```{r upset_novel, echo=F}
novel_geneName_geneType = geneName_geneType %>% subset(gene_type == "novel")

to_upset_novel = to_upset %>% subset(rownames(to_upset) %in% novel_geneName_geneType$gene_name)
upset(to_upset_novel, 
      sets=cancers)
```

## Is there a correlation between the number of TTS and the number of patients per ctype?

```{r num_patients_numTTS, echo=F}
total_num_patients = data.frame(ctype = c("BRCA","BLCA","LUAD","KIRC","LUSC","PRAD","LIHC","COAD"),
                                total_n = c(109,38,179,142,49,75,182,144))

total_burden_TTS = TSA_testisSpecific %>% select(ctype, gene_name, gene_type) %>% unique() %>%
  group_by(ctype) %>% count()

full_df = merge(total_burden_TTS, total_num_patients, by="ctype")

ggplot(full_df, aes(x=total_n, y=n, color=ctype)) +
  geom_point(shape = 21, size=5,stroke = 3) +
  scale_color_manual(values=c("BRCA"="#E69F00", "BLCA"="#56A3A6", "LUAD"="#009E73", "KIRC"="#F0C27B", "LUSC"="#0072B2", "PRAD"="#D55E00", "LIHC"="#CC79A7", "COAD"="#8B7355")) +
  stat_smooth(method = "lm", 
              formula = y ~ x, 
              geom = "smooth", color="grey") +
  stat_cor(p.accuracy = 0.001, r.accuracy = 0.01)+
  labs(x="Number of patients per cancer type",
       y="Number of genes",
       title="Correlation between number of patients and TTS genes") +
  theme_minimal()
ggsave(file.path(plots_wd,"PNG/corr_patients_TTS.png"))
ggsave(file.path(plots_wd,"PDF/corr_patients_TTS.pdf"))
```
