{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "import os,re\n",
    "import pandas as pd\n",
    "from collections import Counter\n",
    "from rna_seq_normalization import Normalization as Norm\n",
    "from functools import reduce\n",
    "\n",
    "annot = pd.read_csv(\"/users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/human/newReference_Resconstructed/1transcript_1gene.reconstructed.csv\")\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Quantify in thymus"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "        ==========     _____ _    _ ____  _____  ______          _____  \n",
      "        =====         / ____| |  | |  _ \\|  __ \\|  ____|   /\\   |  __ \\ \n",
      "          =====      | (___ | |  | | |_) | |__) | |__     /  \\  | |  | |\n",
      "            ====      \\___ \\| |  | |  _ <|  _  /|  __|   / /\\ \\ | |  | |\n",
      "              ====    ____) | |__| | |_) | | \\ \\| |____ / ____ \\| |__| |\n",
      "        ==========   |_____/ \\____/|____/|_|  \\_\\______/_/    \\_\\_____/\n",
      "\t  v2.0.3\n",
      "\n",
      "//========================== featureCounts setting ===========================\\\\\n",
      "||                                                                            ||\n",
      "||             Input files : 30 BAM files                                     ||\n",
      "||                                                                            ||\n",
      "||                           SRR8668611Aligned.sortedByCoord.out.bam          ||\n",
      "||                           SRR8668612Aligned.sortedByCoord.out.bam          ||\n",
      "||                           SRR8668613Aligned.sortedByCoord.out.bam          ||\n",
      "||                           SRR8668614Aligned.sortedByCoord.out.bam          ||\n",
      "||                           SRR8668615Aligned.sortedByCoord.out.bam          ||\n",
      "||                           SRR8668616Aligned.sortedByCoord.out.bam          ||\n",
      "||                           SRR8668617Aligned.sortedByCoord.out.bam          ||\n",
      "||                           SRR8668618Aligned.sortedByCoord.out.bam          ||\n",
      "||                           SRR8668619Aligned.sortedByCoord.out.bam          ||\n",
      "||                           SRR8668620Aligned.sortedByCoord.out.bam          ||\n",
      "||                           SRR8668621Aligned.sortedByCoord.out.bam          ||\n",
      "||                           SRR8668622Aligned.sortedByCoord.out.bam          ||\n",
      "||                           SRR8668623Aligned.sortedByCoord.out.bam          ||\n",
      "||                           SRR8668624Aligned.sortedByCoord.out.bam          ||\n",
      "||                           SRR8668625Aligned.sortedByCoord.out.bam          ||\n",
      "||                           SRR8668626Aligned.sortedByCoord.out.bam          ||\n",
      "||                           SRR8668627Aligned.sortedByCoord.out.bam          ||\n",
      "||                           SRR8668628Aligned.sortedByCoord.out.bam          ||\n",
      "||                           SRR8668629Aligned.sortedByCoord.out.bam          ||\n",
      "||                           SRR8668630Aligned.sortedByCoord.out.bam          ||\n",
      "||                           SRR8668631Aligned.sortedByCoord.out.bam          ||\n",
      "||                           SRR8668632Aligned.sortedByCoord.out.bam          ||\n",
      "||                           SRR8668633Aligned.sortedByCoord.out.bam          ||\n",
      "||                           SRR8668634Aligned.sortedByCoord.out.bam          ||\n",
      "||                           SRR18941362Aligned.sortedByCoord.out.bam         ||\n",
      "||                           SRR18941363Aligned.sortedByCoord.out.bam         ||\n",
      "||                           SRR18941364Aligned.sortedByCoord.out.bam         ||\n",
      "||                           SRR18941365Aligned.sortedByCoord.out.bam         ||\n",
      "||                           SRR18941366Aligned.sortedByCoord.out.bam         ||\n",
      "||                           SRR18941367Aligned.sortedByCoord.out.bam         ||\n",
      "||                                                                            ||\n",
      "||             Output file : gffcompare_stranded_featureCounts.txt            ||\n",
      "||                 Summary : gffcompare_stranded_featureCounts.txt.summary    ||\n",
      "||              Paired-end : yes                                              ||\n",
      "||        Count read pairs : yes                                              ||\n",
      "||              Annotation : gencode.v38.gffcompare.TestisLiverBrain.anno ... ||\n",
      "||      Dir for temp files : /users/genomics/marta/TestisProject_SaraRazq ... ||\n",
      "||                                                                            ||\n",
      "||                 Threads : 10                                               ||\n",
      "||                   Level : meta-feature level                               ||\n",
      "||      Multimapping reads : not counted                                      ||\n",
      "|| Multi-overlapping reads : counted                                          ||\n",
      "||   Min overlapping bases : 1                                                ||\n",
      "||                                                                            ||\n",
      "\\\\============================================================================//\n",
      "\n",
      "//================================= Running ==================================\\\\\n",
      "||                                                                            ||\n",
      "|| Load annotation file gencode.v38.gffcompare.TestisLiverBrain.annotatio ... ||\n",
      "||    Features : 265746                                                       ||\n",
      "||    Meta-features : 57866                                                   ||\n",
      "||    Chromosomes/contigs : 22                                                ||\n",
      "||                                                                            ||\n",
      "|| Process BAM file SRR8668611Aligned.sortedByCoord.out.bam...                ||\n",
      "||    Strand specific : reversely stranded                                    ||\n",
      "||    Paired-end reads are included.                                          ||\n",
      "||    Total alignments : 14323394                                             ||\n",
      "||    Successfully assigned alignments : 10350934 (72.3%)                     ||\n",
      "||    Running time : 1.05 minutes                                             ||\n",
      "||                                                                            ||\n",
      "|| Process BAM file SRR8668612Aligned.sortedByCoord.out.bam...                ||\n",
      "||    Strand specific : reversely stranded                                    ||\n",
      "||    Paired-end reads are included.                                          ||\n",
      "||    Total alignments : 14290397                                             ||\n",
      "||    Successfully assigned alignments : 10331106 (72.3%)                     ||\n",
      "||    Running time : 0.75 minutes                                             ||\n",
      "||                                                                            ||\n",
      "|| Process BAM file SRR8668613Aligned.sortedByCoord.out.bam...                ||\n",
      "||    Strand specific : reversely stranded                                    ||\n",
      "||    Paired-end reads are included.                                          ||\n",
      "||    Total alignments : 14669485                                             ||\n",
      "||    Successfully assigned alignments : 10602789 (72.3%)                     ||\n",
      "||    Running time : 0.83 minutes                                             ||\n",
      "||                                                                            ||\n",
      "|| Process BAM file SRR8668614Aligned.sortedByCoord.out.bam...                ||\n",
      "||    Strand specific : reversely stranded                                    ||\n",
      "||    Paired-end reads are included.                                          ||\n",
      "||    Total alignments : 14241669                                             ||\n",
      "||    Successfully assigned alignments : 10299156 (72.3%)                     ||\n",
      "||    Running time : 0.52 minutes                                             ||\n",
      "||                                                                            ||\n",
      "|| Process BAM file SRR8668615Aligned.sortedByCoord.out.bam...                ||\n",
      "||    Strand specific : reversely stranded                                    ||\n",
      "||    Paired-end reads are included.                                          ||\n",
      "||    Total alignments : 13118328                                             ||\n",
      "||    Successfully assigned alignments : 9461930 (72.1%)                      ||\n",
      "||    Running time : 0.57 minutes                                             ||\n",
      "||                                                                            ||\n",
      "|| Process BAM file SRR8668616Aligned.sortedByCoord.out.bam...                ||\n",
      "||    Strand specific : reversely stranded                                    ||\n",
      "||    Paired-end reads are included.                                          ||\n",
      "||    Total alignments : 13008581                                             ||\n",
      "||    Successfully assigned alignments : 9384062 (72.1%)                      ||\n",
      "||    Running time : 0.63 minutes                                             ||\n",
      "||                                                                            ||\n",
      "|| Process BAM file SRR8668617Aligned.sortedByCoord.out.bam...                ||\n",
      "||    Strand specific : reversely stranded                                    ||\n",
      "||    Paired-end reads are included.                                          ||\n",
      "||    Total alignments : 13429055                                             ||\n",
      "||    Successfully assigned alignments : 9690210 (72.2%)                      ||\n",
      "||    Running time : 0.44 minutes                                             ||\n",
      "||                                                                            ||\n",
      "|| Process BAM file SRR8668618Aligned.sortedByCoord.out.bam...                ||\n",
      "||    Strand specific : reversely stranded                                    ||\n",
      "||    Paired-end reads are included.                                          ||\n",
      "||    Total alignments : 13007491                                             ||\n",
      "||    Successfully assigned alignments : 9386757 (72.2%)                      ||\n",
      "||    Running time : 0.26 minutes                                             ||\n",
      "||                                                                            ||\n",
      "|| Process BAM file SRR8668619Aligned.sortedByCoord.out.bam...                ||\n",
      "||    Strand specific : reversely stranded                                    ||\n",
      "||    Paired-end reads are included.                                          ||\n",
      "||    Total alignments : 20222557                                             ||\n",
      "||    Successfully assigned alignments : 15444044 (76.4%)                     ||\n",
      "||    Running time : 0.77 minutes                                             ||\n",
      "||                                                                            ||\n",
      "|| Process BAM file SRR8668620Aligned.sortedByCoord.out.bam...                ||\n",
      "||    Strand specific : reversely stranded                                    ||\n",
      "||    Paired-end reads are included.                                          ||\n",
      "||    Total alignments : 20104349                                             ||\n",
      "||    Successfully assigned alignments : 15359344 (76.4%)                     ||\n",
      "||    Running time : 1.35 minutes                                             ||\n",
      "||                                                                            ||\n",
      "|| Process BAM file SRR8668621Aligned.sortedByCoord.out.bam...                ||\n",
      "||    Strand specific : reversely stranded                                    ||\n",
      "||    Paired-end reads are included.                                          ||\n",
      "||    Total alignments : 20273457                                             ||\n",
      "||    Successfully assigned alignments : 15484612 (76.4%)                     ||\n",
      "||    Running time : 1.58 minutes                                             ||\n",
      "||                                                                            ||\n",
      "|| Process BAM file SRR8668622Aligned.sortedByCoord.out.bam...                ||\n",
      "||    Strand specific : reversely stranded                                    ||\n",
      "||    Paired-end reads are included.                                          ||\n",
      "||    Total alignments : 19991944                                             ||\n",
      "||    Successfully assigned alignments : 15271104 (76.4%)                     ||\n",
      "||    Running time : 2.28 minutes                                             ||\n",
      "||                                                                            ||\n",
      "|| Process BAM file SRR8668623Aligned.sortedByCoord.out.bam...                ||\n",
      "||    Strand specific : reversely stranded                                    ||\n",
      "||    Paired-end reads are included.                                          ||\n",
      "||    Total alignments : 25424739                                             ||\n",
      "||    Successfully assigned alignments : 19967313 (78.5%)                     ||\n",
      "||    Running time : 1.62 minutes                                             ||\n",
      "||                                                                            ||\n",
      "|| Process BAM file SRR8668624Aligned.sortedByCoord.out.bam...                ||\n",
      "||    Strand specific : reversely stranded                                    ||\n",
      "||    Paired-end reads are included.                                          ||\n",
      "||    Total alignments : 25248963                                             ||\n",
      "||    Successfully assigned alignments : 19832054 (78.5%)                     ||\n",
      "||    Running time : 1.21 minutes                                             ||\n",
      "||                                                                            ||\n",
      "|| Process BAM file SRR8668625Aligned.sortedByCoord.out.bam...                ||\n",
      "||    Strand specific : reversely stranded                                    ||\n",
      "||    Paired-end reads are included.                                          ||\n",
      "||    Total alignments : 25466756                                             ||\n",
      "||    Successfully assigned alignments : 20007006 (78.6%)                     ||\n",
      "||    Running time : 1.06 minutes                                             ||\n",
      "||                                                                            ||\n",
      "|| Process BAM file SRR8668626Aligned.sortedByCoord.out.bam...                ||\n",
      "||    Strand specific : reversely stranded                                    ||\n",
      "||    Paired-end reads are included.                                          ||\n",
      "||    Total alignments : 25101355                                             ||\n",
      "||    Successfully assigned alignments : 19717007 (78.5%)                     ||\n",
      "||    Running time : 1.09 minutes                                             ||\n",
      "||                                                                            ||\n",
      "|| Process BAM file SRR8668627Aligned.sortedByCoord.out.bam...                ||\n",
      "||    Strand specific : reversely stranded                                    ||\n",
      "||    Paired-end reads are included.                                          ||\n",
      "||    Total alignments : 14335170                                             ||\n",
      "||    Successfully assigned alignments : 11394635 (79.5%)                     ||\n",
      "||    Running time : 0.77 minutes                                             ||\n",
      "||                                                                            ||\n",
      "|| Process BAM file SRR8668628Aligned.sortedByCoord.out.bam...                ||\n",
      "||    Strand specific : reversely stranded                                    ||\n",
      "||    Paired-end reads are included.                                          ||\n",
      "||    Total alignments : 14043222                                             ||\n",
      "||    Successfully assigned alignments : 11163838 (79.5%)                     ||\n",
      "||    Running time : 0.52 minutes                                             ||\n",
      "||                                                                            ||\n",
      "|| Process BAM file SRR8668629Aligned.sortedByCoord.out.bam...                ||\n",
      "||    Strand specific : reversely stranded                                    ||\n",
      "||    Paired-end reads are included.                                          ||\n",
      "||    Total alignments : 14365008                                             ||\n",
      "||    Successfully assigned alignments : 11422311 (79.5%)                     ||\n",
      "||    Running time : 0.90 minutes                                             ||\n",
      "||                                                                            ||\n",
      "|| Process BAM file SRR8668630Aligned.sortedByCoord.out.bam...                ||\n",
      "||    Strand specific : reversely stranded                                    ||\n",
      "||    Paired-end reads are included.                                          ||\n",
      "||    Total alignments : 13966065                                             ||\n",
      "||    Successfully assigned alignments : 11103055 (79.5%)                     ||\n",
      "||    Running time : 0.73 minutes                                             ||\n",
      "||                                                                            ||\n",
      "|| Process BAM file SRR8668631Aligned.sortedByCoord.out.bam...                ||\n",
      "||    Strand specific : reversely stranded                                    ||\n",
      "||    Paired-end reads are included.                                          ||\n",
      "||    Total alignments : 21587241                                             ||\n",
      "||    Successfully assigned alignments : 16936151 (78.5%)                     ||\n",
      "||    Running time : 2.02 minutes                                             ||\n",
      "||                                                                            ||\n",
      "|| Process BAM file SRR8668632Aligned.sortedByCoord.out.bam...                ||\n",
      "||    Strand specific : reversely stranded                                    ||\n",
      "||    Paired-end reads are included.                                          ||\n",
      "||    Total alignments : 21502897                                             ||\n",
      "||    Successfully assigned alignments : 16874316 (78.5%)                     ||\n",
      "||    Running time : 1.81 minutes                                             ||\n",
      "||                                                                            ||\n",
      "|| Process BAM file SRR8668633Aligned.sortedByCoord.out.bam...                ||\n",
      "||    Strand specific : reversely stranded                                    ||\n",
      "||    Paired-end reads are included.                                          ||\n",
      "||    Total alignments : 21702446                                             ||\n",
      "||    Successfully assigned alignments : 17030296 (78.5%)                     ||\n",
      "||    Running time : 1.47 minutes                                             ||\n",
      "||                                                                            ||\n",
      "|| Process BAM file SRR8668634Aligned.sortedByCoord.out.bam...                ||\n",
      "||    Strand specific : reversely stranded                                    ||\n",
      "||    Paired-end reads are included.                                          ||\n",
      "||    Total alignments : 21349413                                             ||\n",
      "||    Successfully assigned alignments : 16752951 (78.5%)                     ||\n",
      "||    Running time : 0.89 minutes                                             ||\n",
      "||                                                                            ||\n",
      "|| Process BAM file SRR18941362Aligned.sortedByCoord.out.bam...               ||\n",
      "||    Strand specific : reversely stranded                                    ||\n",
      "||    Paired-end reads are included.                                          ||\n",
      "||    Total alignments : 73502276                                             ||\n",
      "||    Successfully assigned alignments : 52256841 (71.1%)                     ||\n",
      "||    Running time : 1.19 minutes                                             ||\n",
      "||                                                                            ||\n",
      "|| Process BAM file SRR18941363Aligned.sortedByCoord.out.bam...               ||\n",
      "||    Strand specific : reversely stranded                                    ||\n",
      "||    Paired-end reads are included.                                          ||\n",
      "||    Total alignments : 70831124                                             ||\n",
      "||    Successfully assigned alignments : 52567545 (74.2%)                     ||\n",
      "||    Running time : 1.70 minutes                                             ||\n",
      "||                                                                            ||\n",
      "|| Process BAM file SRR18941364Aligned.sortedByCoord.out.bam...               ||\n",
      "||    Strand specific : reversely stranded                                    ||\n",
      "||    Paired-end reads are included.                                          ||\n",
      "||    Total alignments : 71334282                                             ||\n",
      "||    Successfully assigned alignments : 52077294 (73.0%)                     ||\n",
      "||    Running time : 2.62 minutes                                             ||\n",
      "||                                                                            ||\n",
      "|| Process BAM file SRR18941365Aligned.sortedByCoord.out.bam...               ||\n",
      "||    Strand specific : reversely stranded                                    ||\n",
      "||    Paired-end reads are included.                                          ||\n",
      "||    Total alignments : 76583894                                             ||\n",
      "||    Successfully assigned alignments : 55929783 (73.0%)                     ||\n",
      "||    Running time : 2.74 minutes                                             ||\n",
      "||                                                                            ||\n",
      "|| Process BAM file SRR18941366Aligned.sortedByCoord.out.bam...               ||\n",
      "||    Strand specific : reversely stranded                                    ||\n",
      "||    Paired-end reads are included.                                          ||\n",
      "||    Total alignments : 148726438                                            ||\n",
      "||    Successfully assigned alignments : 103458676 (69.6%)                    ||\n",
      "||    Running time : 5.18 minutes                                             ||\n",
      "||                                                                            ||\n",
      "|| Process BAM file SRR18941367Aligned.sortedByCoord.out.bam...               ||\n",
      "||    Strand specific : reversely stranded                                    ||\n",
      "||    Paired-end reads are included.                                          ||\n",
      "||    Total alignments : 153956013                                            ||\n",
      "||    Successfully assigned alignments : 106610800 (69.2%)                    ||\n",
      "||    Running time : 6.46 minutes                                             ||\n",
      "||                                                                            ||\n",
      "|| Write the final count table.                                               ||\n",
      "|| Write the read assignment summary.                                         ||\n",
      "||                                                                            ||\n",
      "|| Summary of counting results can be found in file \"/users/genomics/marta/T  ||\n",
      "|| estisProject_SaraRazquin/with_TranscriptomeReconstruction/human/quantific  ||\n",
      "|| ation_thymus/gffcompare_stranded_featureCounts.txt.summary\"                ||\n",
      "||                                                                            ||\n",
      "\\\\============================================================================//\n",
      "\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "###PREPARING NEEDED DATA\n",
    "thymus_dir=/projects_eg/projects/marta/thymus\n",
    "outdir=/users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/human/quantification_thymus\n",
    "\n",
    "mkdir -p $outdir\n",
    "AnnotGTF=/users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/human/newReference_Resconstructed/gencode.v38.gffcompare.TestisLiverBrain.annotation.sorted.1transcript.sorted.NOchr.gtf\n",
    "\n",
    "module load Subread/2.0.3\n",
    "########################\n",
    "\n",
    "# countReadPairs may need to be removed in case of single-end reads\n",
    "featureCounts -T 10 -p -s 2 -g transcript_id -O --countReadPairs -a $AnnotGTF -o ${outdir}/gffcompare_stranded_featureCounts.txt $thymus_dir/GSE*/analysis/05_STAR/uniquely_mapped_2pass_BAM_files/*Aligned.sortedByCoord.out.bam \n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "### modify headers\n",
    "file = \"/users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/human/quantification_thymus/gffcompare_stranded_featureCounts.txt\"\n",
    "\n",
    "toc = pd.read_csv(file, sep=\"\\t\", comment=\"#\")\n",
    "toc = toc[toc['Geneid'].str.contains('PAR_')==False]\n",
    "toc['Geneid']=toc['Geneid'].str.split('.').str[0]\n",
    "toc.rename(columns={'Geneid':'transcript_id'}, inplace=True)\n",
    "\n",
    "filter_col = [col for col in toc if col.startswith('/')]\n",
    "for col in filter_col:\n",
    "    new_col=col.split(\"Aligned\")[0]\n",
    "    new_col=new_col.split(\"/\")[-1]\n",
    "    toc.rename(columns={col:new_col}, inplace=True)\n",
    "\n",
    "length = toc['Length']\n",
    "genes = toc['transcript_id']\n",
    "# we are only interested in the columns with counts\n",
    "counts = toc\n",
    "counts.drop([\"Chr\",\"Start\",\"End\",\"Strand\",\"Length\",\"transcript_id\"],axis=1, inplace=True)\n",
    "# calculate TPMs\n",
    "tpm_df = Norm.tpm(counts, length)\n",
    "# add transcript_id and length again\n",
    "tpms = pd.concat([genes,tpm_df, length], axis=1)\n",
    "tpms.to_csv(\"/users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/human/quantification_thymus/table_of_counts_TPMs_thymus.csv\", index=None)\n",
    " "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Candidates"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_157873/1926467606.py:3: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  tumorReact_genes.drop_duplicates(inplace=True)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>transcript_id</th>\n",
       "      <th>gene_id</th>\n",
       "      <th>gene_name</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ENST00000265007</td>\n",
       "      <td>ENSG00000039600</td>\n",
       "      <td>SOX30</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ENST00000247452</td>\n",
       "      <td>ENSG00000046774</td>\n",
       "      <td>MAGEC2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>ENST00000376150</td>\n",
       "      <td>ENSG00000068985</td>\n",
       "      <td>PAGE1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>ENST00000158009</td>\n",
       "      <td>ENSG00000073598</td>\n",
       "      <td>FNDC8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>ENST00000378988</td>\n",
       "      <td>ENSG00000099399</td>\n",
       "      <td>MAGEB2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>392</th>\n",
       "      <td>TCONS_00001462</td>\n",
       "      <td>XLOC_001220</td>\n",
       "      <td>XLOC_001220</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>395</th>\n",
       "      <td>TCONS_00001466</td>\n",
       "      <td>XLOC_001230</td>\n",
       "      <td>XLOC_001230</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>404</th>\n",
       "      <td>TCONS_00001617</td>\n",
       "      <td>XLOC_001342</td>\n",
       "      <td>XLOC_001342</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>409</th>\n",
       "      <td>TCONS_00001992</td>\n",
       "      <td>XLOC_001676</td>\n",
       "      <td>XLOC_001676</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>414</th>\n",
       "      <td>TCONS_00002105</td>\n",
       "      <td>XLOC_001725</td>\n",
       "      <td>XLOC_001725</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>136 rows × 3 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       transcript_id          gene_id    gene_name\n",
       "0    ENST00000265007  ENSG00000039600        SOX30\n",
       "4    ENST00000247452  ENSG00000046774       MAGEC2\n",
       "11   ENST00000376150  ENSG00000068985        PAGE1\n",
       "14   ENST00000158009  ENSG00000073598        FNDC8\n",
       "22   ENST00000378988  ENSG00000099399       MAGEB2\n",
       "..               ...              ...          ...\n",
       "392   TCONS_00001462      XLOC_001220  XLOC_001220\n",
       "395   TCONS_00001466      XLOC_001230  XLOC_001230\n",
       "404   TCONS_00001617      XLOC_001342  XLOC_001342\n",
       "409   TCONS_00001992      XLOC_001676  XLOC_001676\n",
       "414   TCONS_00002105      XLOC_001725  XLOC_001725\n",
       "\n",
       "[136 rows x 3 columns]"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "tumorReact = pd.read_csv(\"/projects_eg/projects/marta/TestisRestricted_Microproteins_TSA/with_TranscriptomeReconstruction/Q4_TestisRestricted_TumorSpecific/human/TOv3x_5percent_TestisRestrictedGTEx_Translated_Ctypes_log2ratio3xMEAN.csv\")\n",
    "tumorReact_genes = tumorReact[['transcript_id','gene_id','gene_name']]\n",
    "tumorReact_genes.drop_duplicates(inplace=True)\n",
    "tumorReact_genes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_157873/2258808333.py:2: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  tpms_candidates.drop([\"Length\"], axis=1, inplace=True)\n",
      "/tmp/ipykernel_157873/2258808333.py:6: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  tpms_candidates['mean'] = tpms_candidates.iloc[:, 1:].mean(axis=1)\n",
      "/tmp/ipykernel_157873/2258808333.py:7: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  tpms_candidates['median'] = tpms_candidates.iloc[:, 1:].median(axis=1)\n",
      "/tmp/ipykernel_157873/2258808333.py:8: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  tpms_candidates['max'] = tpms_candidates.iloc[:, 1:].max(axis=1)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>transcript_id</th>\n",
       "      <th>mean</th>\n",
       "      <th>median</th>\n",
       "      <th>max</th>\n",
       "      <th>gene_id</th>\n",
       "      <th>gene_type</th>\n",
       "      <th>gene_name</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ENST00000381059</td>\n",
       "      <td>0.161225</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.969202</td>\n",
       "      <td>ENSG00000182583</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>VCX</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ENST00000317103</td>\n",
       "      <td>0.072923</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.691067</td>\n",
       "      <td>ENSG00000177504</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>VCX2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ENST00000543214</td>\n",
       "      <td>0.056680</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.331745</td>\n",
       "      <td>ENSG00000183304</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>FAM9A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ENST00000327968</td>\n",
       "      <td>0.000875</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.013950</td>\n",
       "      <td>ENSG00000184735</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>DDX53</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ENST00000412172</td>\n",
       "      <td>0.000499</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.014967</td>\n",
       "      <td>ENSG00000224960</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>PPP4R3C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>131</th>\n",
       "      <td>ENST00000269881</td>\n",
       "      <td>0.332265</td>\n",
       "      <td>0.111484</td>\n",
       "      <td>2.686197</td>\n",
       "      <td>ENSG00000269058</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>CALR3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>132</th>\n",
       "      <td>ENST00000601693</td>\n",
       "      <td>0.006701</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.081271</td>\n",
       "      <td>ENSG00000196350</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>ZNF729</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>133</th>\n",
       "      <td>ENST00000596209</td>\n",
       "      <td>0.014091</td>\n",
       "      <td>0.011938</td>\n",
       "      <td>0.103742</td>\n",
       "      <td>ENSG00000213973</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>ZNF99</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>134</th>\n",
       "      <td>ENST00000600766</td>\n",
       "      <td>0.017706</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.076942</td>\n",
       "      <td>ENSG00000268696</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>ZNF723</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>135</th>\n",
       "      <td>TCONS_00001039</td>\n",
       "      <td>1.109857</td>\n",
       "      <td>0.694586</td>\n",
       "      <td>3.012705</td>\n",
       "      <td>XLOC_000930</td>\n",
       "      <td>novel</td>\n",
       "      <td>XLOC_000930</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>136 rows × 7 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       transcript_id      mean    median       max          gene_id  \\\n",
       "0    ENST00000381059  0.161225  0.000000  0.969202  ENSG00000182583   \n",
       "1    ENST00000317103  0.072923  0.000000  0.691067  ENSG00000177504   \n",
       "2    ENST00000543214  0.056680  0.000000  0.331745  ENSG00000183304   \n",
       "3    ENST00000327968  0.000875  0.000000  0.013950  ENSG00000184735   \n",
       "4    ENST00000412172  0.000499  0.000000  0.014967  ENSG00000224960   \n",
       "..               ...       ...       ...       ...              ...   \n",
       "131  ENST00000269881  0.332265  0.111484  2.686197  ENSG00000269058   \n",
       "132  ENST00000601693  0.006701  0.000000  0.081271  ENSG00000196350   \n",
       "133  ENST00000596209  0.014091  0.011938  0.103742  ENSG00000213973   \n",
       "134  ENST00000600766  0.017706  0.000000  0.076942  ENSG00000268696   \n",
       "135   TCONS_00001039  1.109857  0.694586  3.012705      XLOC_000930   \n",
       "\n",
       "          gene_type    gene_name  \n",
       "0    protein_coding          VCX  \n",
       "1    protein_coding         VCX2  \n",
       "2    protein_coding        FAM9A  \n",
       "3    protein_coding        DDX53  \n",
       "4    protein_coding      PPP4R3C  \n",
       "..              ...          ...  \n",
       "131  protein_coding        CALR3  \n",
       "132  protein_coding       ZNF729  \n",
       "133  protein_coding        ZNF99  \n",
       "134  protein_coding       ZNF723  \n",
       "135           novel  XLOC_000930  \n",
       "\n",
       "[136 rows x 7 columns]"
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "tpms_candidates = tpms[tpms['transcript_id'].isin(tumorReact_genes.transcript_id.values.tolist())]\n",
    "tpms_candidates.drop([\"Length\"], axis=1, inplace=True)\n",
    "tpms_candidates.to_csv(\"/users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/human/quantification_thymus/table_of_counts_TPMs_thymus_TOv3xlog2ratio3xmean.csv\")\n",
    "\n",
    "## compute mean, median, max\n",
    "tpms_candidates['mean'] = tpms_candidates.iloc[:, 1:].mean(axis=1)\n",
    "tpms_candidates['median'] = tpms_candidates.iloc[:, 1:].median(axis=1)\n",
    "tpms_candidates['max'] = tpms_candidates.iloc[:, 1:].max(axis=1)\n",
    "\n",
    "tpms_candidates_thymus = tpms_candidates[['transcript_id','mean','median','max']]\n",
    "tpms_candidates_thymus = tpms_candidates_thymus.merge(annot, on=\"transcript_id\")\n",
    "tpms_candidates_thymus.to_csv(\"/users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/human/quantification_thymus/thymus_summary_TOv3xlog2ratio3xmean.csv\")\n",
    "tpms_candidates_thymus"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.16"
  },
  "orig_nbformat": 4
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
