{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import os,re,glob\n",
    "import pandas as pd\n",
    "import numpy as np\n",
    "from collections import Counter\n",
    "from rna_seq_normalization import Normalization as Norm\n",
    "\n",
    "users_dir = \"/users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction\"\n",
    "specie = \"human\"\n",
    "GENOMEDIR = \"/genomics/users/marta/genomes\"\n",
    "\n",
    "### cancer data\n",
    "cancer_dir = users_dir + \"/cancers\"\n",
    "raw_cancer_dir=\"/users/genomics/marta/TCGA_RNASeq\"\n",
    "tcga_projects=[\"TCGA-BRCA\",\"TCGA-LUSC\",\"TCGA-PRAD\",\"TCGA-KIRC\",\"TCGA-KIRP\",\"TCGA-LUAD\"]#,\"TCGA-BLCA\",\"TCGA-LIHC\"]\n",
    "other_projects=[\"GSE102101_KIRC\",\"GSE133624_BLCA\",\"GSE214846_LIHC\",\"GSE22260_PRAD\",\"GSE229705_LUAD\",\"GSE89223_PRAD\",\"PRJEB2449_PRAD\",\"SRP107326_COAD\",\"SRP238334_KIRC\",\"TCGA_COAD\",\"GSE103001_BRCA\"]\n",
    "all_projects = tcga_projects + other_projects\n",
    "\n",
    "## annotation file\n",
    "annotation=\"/users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/human/newReference_Resconstructed/gencode.v38.gffcompare.TestisLiverBrain.annotation.sorted.1transcript.sorted.NOchr.gtf\"\n",
    "transcript_gene=pd.read_csv(\"/users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/human/newReference_Resconstructed/1transcript_1gene.reconstructed.csv\")\n",
    "\n",
    "def count_greater_than_one(row):\n",
    "    return (row > 1).sum()"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Quantify with TCGAData and new reference\n",
    "`featureCounts_newRef_cancer.sh`\n",
    "\n",
    "output in: `/users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/cancers/featureCounts`"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## TPM"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "TCGA-PRAD\n",
      "TCGA-LUSC\n",
      "GSE102101_KIRC\n",
      "BLCA\n",
      "TCGA-BRCA\n",
      "TCGA-KIRP\n",
      "GSE22260_PRAD\n",
      "GSE133624_BLCA\n",
      "PRJEB2449_PRAD\n",
      "TCGA-LUAD\n",
      "LIHC\n",
      "TCGA-KIRC\n",
      "SRP238334_KIRC\n"
     ]
    }
   ],
   "source": [
    "for file in os.listdir(os.path.join(cancer_dir,\"featureCounts\")):\n",
    "    if file.endswith(\".txt\"):\n",
    "\n",
    "        proj = file.split(\"featureCounts_\")[-1]\n",
    "        cancer_type = proj[:-4]\n",
    "        print(cancer_type)\n",
    "        toc = pd.read_csv(os.path.join(cancer_dir,\"featureCounts\",file), sep=\"\\t\", comment=\"#\")\n",
    "        toc['Geneid']=toc['Geneid'].str.split('.').str[0]\n",
    "        toc.rename(columns={'Geneid':'transcript_id'}, inplace=True)\n",
    "\n",
    "        filter_col = [col for col in toc if col.startswith('/')]\n",
    "        for col in filter_col:\n",
    "            new_col=col.split(\"Aligned\")[0]\n",
    "            new_col=new_col.split(\"/\")[-1]\n",
    "            if proj in tcga_projects:\n",
    "                new_col=cancer_type+new_col[4:]\n",
    "            toc.rename(columns={col:new_col}, inplace=True)\n",
    "\n",
    "        length = toc['Length']\n",
    "        genes = toc['transcript_id']\n",
    "        # we are only interested in the columns with counts\n",
    "        counts = toc\n",
    "        counts.drop([\"Chr\",\"Start\",\"End\",\"Strand\",\"Length\",\"transcript_id\"],axis=1, inplace=True)\n",
    "        # calculate TPMs\n",
    "        tpm_df = Norm.tpm(counts, length)\n",
    "        # add transcript_id and length again\n",
    "        tpms = pd.concat([genes,tpm_df, length], axis=1)\n",
    "        tpms.to_csv(os.path.join(cancer_dir,\"featureCounts/table_of_counts_TPMs_\")+cancer_type+\".csv\", index=None)\n",
    " "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "BRCA\n",
      "LUSC\n",
      "PRAD\n",
      "KIRC\n",
      "KIRP\n",
      "LUAD\n",
      "         transcript_id   SRR5885319   SRR5885320   SRR5885321   SRR5885322  \\\n",
      "0       TCONS_00002128   918.580901   259.244976   743.867429   460.021127   \n",
      "1      ENST00000387314  2739.699672   347.541269  1844.771818   399.913977   \n",
      "2      ENST00000389680  4460.676619  1093.170912  3362.743728   897.971994   \n",
      "3       TCONS_00002129  8547.016124  2386.216764  6447.811071  1860.799654   \n",
      "4      ENST00000387342  1675.129075    55.297117  1637.638926   179.115198   \n",
      "...                ...          ...          ...          ...          ...   \n",
      "57861  ENST00000425340     0.290033     0.554147     0.284691     0.488158   \n",
      "57862  ENST00000318083     0.768010     2.113511     0.516160     1.961413   \n",
      "57863  ENST00000222145     4.884411    15.923205     4.564286    44.418950   \n",
      "57864  ENST00000597553     0.075058     0.251131     0.010810     0.057150   \n",
      "57865  ENST00000645652     1.206211     2.161318     0.857057     3.291924   \n",
      "\n",
      "        SRR5885323   SRR5885324   SRR5885325   SRR5885326    SRR5885327  ...  \\\n",
      "0       923.944413   177.298415   692.077866   624.661301    584.826593  ...   \n",
      "1      2053.274934    77.230563  3658.421912  1391.375977   2883.337777  ...   \n",
      "2      4442.686431  2758.408904  4077.609350  1990.561178   5795.766066  ...   \n",
      "3      7328.929933  5996.260342  7933.823765  3670.249820  10347.541760  ...   \n",
      "4      2054.837670   139.310342  2323.725917   809.874448   2850.285977  ...   \n",
      "...            ...          ...          ...          ...           ...  ...   \n",
      "57861     0.335331     0.678456     0.190318     0.201065      0.268940  ...   \n",
      "57862     0.584004     0.906701     0.772746     0.326323      1.037990  ...   \n",
      "57863     5.226093     2.178751     4.819890    29.358245     11.587327  ...   \n",
      "57864     0.108715     0.029786     0.112578     0.473820      0.217379  ...   \n",
      "57865     1.590090     0.252309     1.172961     4.587905      2.279800  ...   \n",
      "\n",
      "        SRR5885330   SRR5885331   SRR5885332   SRR5885333   SRR5885334  \\\n",
      "0       439.369034   797.556412   656.307230  1033.818397   166.641996   \n",
      "1       879.506071  2806.598495   309.802531  1943.082671   388.161548   \n",
      "2      1351.419329  4590.661419  1339.741749  4978.252390  1025.974971   \n",
      "3      2625.861857  8801.770102  2976.449629  9102.214452  2171.265474   \n",
      "4       740.021926  1989.880215   175.314813  2853.146667   884.812153   \n",
      "...            ...          ...          ...          ...          ...   \n",
      "57861     0.245418     1.014466     0.356033     0.509935     0.261629   \n",
      "57862     0.705571     1.038982     1.102326     0.742010     0.511899   \n",
      "57863    22.996611     7.701499    21.141202     7.019601     3.454505   \n",
      "57864     0.265973     0.076155     0.048094     0.052676     0.030630   \n",
      "57865     3.451275     2.145412     3.187480     1.913340     0.498158   \n",
      "\n",
      "        SRR5885335   SRR5885336   SRR5885337   SRR5885338  Length  \n",
      "0       361.809075   479.531536   384.086701   333.196033     577  \n",
      "1       876.287161   287.919835   247.171930   185.214178      71  \n",
      "2      3451.523793  1485.611631   957.617042   963.030804     954  \n",
      "3      6515.608146  3194.679975  1996.892435  2069.194113     353  \n",
      "4      1748.659573   398.057734   139.627210    83.524140      69  \n",
      "...            ...          ...          ...          ...     ...  \n",
      "57861     0.597710     0.478485     1.217722     0.572233    3116  \n",
      "57862     0.659238     0.841126     2.217711     1.268144    1858  \n",
      "57863    12.079384    36.050265    20.868438    14.432325    3199  \n",
      "57864     0.022695     0.097187     0.227208     0.200978    2218  \n",
      "57865     2.414571     2.436827     4.374586     1.897017    3273  \n",
      "\n",
      "[57866 rows x 22 columns]\n",
      "          patientID    adjacent       tumor\n",
      "0  patient_12364284  SRR5885319  SRR5885320\n",
      "1  patient_17621953  SRR5885321  SRR5885322\n",
      "2  patient_20431713  SRR5885323  SRR5885324\n",
      "3  patient_40911432  SRR5885325  SRR5885326\n",
      "4  patient_57398667  SRR5885327  SRR5885328\n",
      "5  patient_70528835  SRR5885329  SRR5885330\n",
      "6  patient_74575859  SRR5885331  SRR5885332\n",
      "7  patient_75416923  SRR5885333  SRR5885334\n",
      "8  patient_77972083  SRR5885335  SRR5885336\n",
      "9  patient_86049102  SRR5885337  SRR5885338\n",
      "         transcript_id   SRR9623608    SRR9623609   SRR9623610   SRR9623611  \\\n",
      "0       TCONS_00002128   108.385307    852.984285   156.137544    93.721125   \n",
      "1      ENST00000387314   272.289135   1635.960214   608.499676   260.698700   \n",
      "2      ENST00000389680  2438.399309  24047.524097  4530.412218  1914.214171   \n",
      "3       TCONS_00002129    10.524498     34.156272    42.607543     7.196981   \n",
      "4      ENST00000387342  3013.198276   9622.721675   859.475916  3154.190046   \n",
      "...                ...          ...           ...          ...          ...   \n",
      "57861  ENST00000425340    12.761823     46.937946     1.473971     3.610699   \n",
      "57862  ENST00000318083     0.925714      0.846434     2.336134     1.692908   \n",
      "57863  ENST00000222145    10.129523      2.270789    55.378069     3.933008   \n",
      "57864  ENST00000597553     0.465278      0.033764     0.318575     0.000000   \n",
      "57865  ENST00000645652     4.813620      2.356734     5.782694     2.328629   \n",
      "\n",
      "        SRR9623612   SRR9623613   SRR9623614   SRR9623615   SRR9623616  ...  \\\n",
      "0       255.205639   356.503370   181.451170   398.973718    33.944136  ...   \n",
      "1      1286.703070   581.121283   620.092498  2993.284486   382.848295  ...   \n",
      "2      3829.225827  4409.134668  2274.749398  6606.713308  2600.765992  ...   \n",
      "3        16.363727    33.438897    10.864212    35.722138     8.319126  ...   \n",
      "4      1172.022294   796.502440  1498.455104  1876.555015  8916.530847  ...   \n",
      "...            ...          ...          ...          ...          ...  ...   \n",
      "57861     2.566780     0.364915     5.858446     1.431195    23.031431  ...   \n",
      "57862     1.004424     4.808486     2.889717     2.814048     1.358484  ...   \n",
      "57863    22.668328    16.282871    25.966723     5.047466     5.849344  ...   \n",
      "57864     0.681131     0.219711     0.138325     0.069332     0.043769  ...   \n",
      "57865     2.253594     5.277337     2.694975     0.328890     2.128153  ...   \n",
      "\n",
      "        SRR9623664   SRR9623665   SRR9623666   SRR9623667   SRR9623668  \\\n",
      "0        69.414415    99.689929   159.526060    43.102901   162.225990   \n",
      "1       429.602783   533.860143   162.920359   535.969147   396.284772   \n",
      "2      3909.185525  8249.823162  6488.476439  3507.963709  5613.576329   \n",
      "3         5.243165    25.431387     9.528483     5.854370    13.802747   \n",
      "4      3521.419014  4205.539944   687.216127   636.192440  3268.137684   \n",
      "...            ...          ...          ...          ...          ...   \n",
      "57861     0.593979     2.080742     1.237414     1.120613     0.220234   \n",
      "57862     0.597687     1.878991     0.750616     1.073913     0.849501   \n",
      "57863     3.355691     1.091330     2.820935     5.301754     1.029696   \n",
      "57864     0.634191     0.037477     0.036987     0.032129     0.000000   \n",
      "57865     1.967893     1.599984     1.303383     1.219267     2.683773   \n",
      "\n",
      "        SRR9623669   SRR9623670   SRR9623671   SRR9623672  Length  \n",
      "0       122.081166   461.984536   265.348790    61.894582     577  \n",
      "1       465.118751   215.304699  1965.858193   362.504960      71  \n",
      "2      6837.079613  7151.824137  5390.182063  5380.131581     954  \n",
      "3        13.808953    16.164506    13.897252    11.432910     353  \n",
      "4      1958.182315  2897.446491  2848.488402  1649.862319      69  \n",
      "...            ...          ...          ...          ...     ...  \n",
      "57861     3.459229     0.542583     1.041114     1.270754    3116  \n",
      "57862     0.591223     1.289096     1.533094     1.188524    1858  \n",
      "57863     9.400198     3.435283    20.183100     2.808817    3199  \n",
      "57864     0.061908     0.730497     0.249717     0.240321    2218  \n",
      "57865     3.628919     5.660592     2.248272     2.326535    3273  \n",
      "\n",
      "[57866 rows x 67 columns]\n",
      "   patientID    adjacent       tumor\n",
      "0      P0156  SRR9623608  SRR9623637\n",
      "1      P0453  SRR9623609  SRR9623638\n",
      "2      P0466  SRR9623610  SRR9623639\n",
      "3      P0474  SRR9623611  SRR9623640\n",
      "4      P0531  SRR9623612  SRR9623641\n",
      "5      P0555  SRR9623613  SRR9623642\n",
      "6      P0565  SRR9623614  SRR9623643\n",
      "7      P1037  SRR9623616  SRR9623645\n",
      "8      P1067  SRR9623618  SRR9623646\n",
      "9      P1495  SRR9623619  SRR9623647\n",
      "10     P1664  SRR9623620  SRR9623648\n",
      "11     P2224  SRR9623622  SRR9623649\n",
      "12     P2297  SRR9623623  SRR9623651\n",
      "13     P2347  SRR9623624  SRR9623652\n",
      "14     P2759  SRR9623626  SRR9623654\n",
      "15     P2804  SRR9623627  SRR9623656\n",
      "16     P3138  SRR9623629  SRR9623657\n",
      "17     P3143  SRR9623630  SRR9623658\n",
      "18     P3635  SRR9623634  SRR9623668\n",
      "19     P3740  SRR9623635  SRR9623670\n"
     ]
    },
    {
     "ename": "FileNotFoundError",
     "evalue": "[Errno 2] No such file or directory: '/users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/cancers/featureCounts/table_of_counts_TPMs_GSE214846_LIHC.csv'",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mFileNotFoundError\u001b[0m                         Traceback (most recent call last)",
      "Cell \u001b[0;32mIn[10], line 14\u001b[0m\n\u001b[1;32m     11\u001b[0m     patients[\u001b[39m'\u001b[39m\u001b[39mtumor\u001b[39m\u001b[39m'\u001b[39m] \u001b[39m=\u001b[39m patients[\u001b[39m'\u001b[39m\u001b[39mtumor\u001b[39m\u001b[39m'\u001b[39m]\u001b[39m.\u001b[39mstr\u001b[39m.\u001b[39mreplace(\u001b[39m'\u001b[39m\u001b[39mTCGA\u001b[39m\u001b[39m'\u001b[39m, cancer_type)\n\u001b[1;32m     13\u001b[0m \u001b[39mif\u001b[39;00m proj \u001b[39min\u001b[39;00m other_projects:\n\u001b[0;32m---> 14\u001b[0m     fc \u001b[39m=\u001b[39m pd\u001b[39m.\u001b[39;49mread_csv(os\u001b[39m.\u001b[39;49mpath\u001b[39m.\u001b[39;49mjoin(cancer_dir,\u001b[39m\"\u001b[39;49m\u001b[39mfeatureCounts/table_of_counts_TPMs_\u001b[39;49m\u001b[39m\"\u001b[39;49m)\u001b[39m+\u001b[39;49mproj\u001b[39m+\u001b[39;49m\u001b[39m\"\u001b[39;49m\u001b[39m.csv\u001b[39;49m\u001b[39m\"\u001b[39;49m)\n\u001b[1;32m     15\u001b[0m     \u001b[39mprint\u001b[39m(fc)\n\u001b[1;32m     16\u001b[0m     tumor_transcripts \u001b[39m=\u001b[39m \u001b[39mlist\u001b[39m()\n",
      "File \u001b[0;32m~/.local/lib/python3.8/site-packages/pandas/io/parsers/readers.py:912\u001b[0m, in \u001b[0;36mread_csv\u001b[0;34m(filepath_or_buffer, sep, delimiter, header, names, index_col, usecols, dtype, engine, converters, true_values, false_values, skipinitialspace, skiprows, skipfooter, nrows, na_values, keep_default_na, na_filter, verbose, skip_blank_lines, parse_dates, infer_datetime_format, keep_date_col, date_parser, date_format, dayfirst, cache_dates, iterator, chunksize, compression, thousands, decimal, lineterminator, quotechar, quoting, doublequote, escapechar, comment, encoding, encoding_errors, dialect, on_bad_lines, delim_whitespace, low_memory, memory_map, float_precision, storage_options, dtype_backend)\u001b[0m\n\u001b[1;32m    899\u001b[0m kwds_defaults \u001b[39m=\u001b[39m _refine_defaults_read(\n\u001b[1;32m    900\u001b[0m     dialect,\n\u001b[1;32m    901\u001b[0m     delimiter,\n\u001b[0;32m   (...)\u001b[0m\n\u001b[1;32m    908\u001b[0m     dtype_backend\u001b[39m=\u001b[39mdtype_backend,\n\u001b[1;32m    909\u001b[0m )\n\u001b[1;32m    910\u001b[0m kwds\u001b[39m.\u001b[39mupdate(kwds_defaults)\n\u001b[0;32m--> 912\u001b[0m \u001b[39mreturn\u001b[39;00m _read(filepath_or_buffer, kwds)\n",
      "File \u001b[0;32m~/.local/lib/python3.8/site-packages/pandas/io/parsers/readers.py:577\u001b[0m, in \u001b[0;36m_read\u001b[0;34m(filepath_or_buffer, kwds)\u001b[0m\n\u001b[1;32m    574\u001b[0m _validate_names(kwds\u001b[39m.\u001b[39mget(\u001b[39m\"\u001b[39m\u001b[39mnames\u001b[39m\u001b[39m\"\u001b[39m, \u001b[39mNone\u001b[39;00m))\n\u001b[1;32m    576\u001b[0m \u001b[39m# Create the parser.\u001b[39;00m\n\u001b[0;32m--> 577\u001b[0m parser \u001b[39m=\u001b[39m TextFileReader(filepath_or_buffer, \u001b[39m*\u001b[39;49m\u001b[39m*\u001b[39;49mkwds)\n\u001b[1;32m    579\u001b[0m \u001b[39mif\u001b[39;00m chunksize \u001b[39mor\u001b[39;00m iterator:\n\u001b[1;32m    580\u001b[0m     \u001b[39mreturn\u001b[39;00m parser\n",
      "File \u001b[0;32m~/.local/lib/python3.8/site-packages/pandas/io/parsers/readers.py:1407\u001b[0m, in \u001b[0;36mTextFileReader.__init__\u001b[0;34m(self, f, engine, **kwds)\u001b[0m\n\u001b[1;32m   1404\u001b[0m     \u001b[39mself\u001b[39m\u001b[39m.\u001b[39moptions[\u001b[39m\"\u001b[39m\u001b[39mhas_index_names\u001b[39m\u001b[39m\"\u001b[39m] \u001b[39m=\u001b[39m kwds[\u001b[39m\"\u001b[39m\u001b[39mhas_index_names\u001b[39m\u001b[39m\"\u001b[39m]\n\u001b[1;32m   1406\u001b[0m \u001b[39mself\u001b[39m\u001b[39m.\u001b[39mhandles: IOHandles \u001b[39m|\u001b[39m \u001b[39mNone\u001b[39;00m \u001b[39m=\u001b[39m \u001b[39mNone\u001b[39;00m\n\u001b[0;32m-> 1407\u001b[0m \u001b[39mself\u001b[39m\u001b[39m.\u001b[39m_engine \u001b[39m=\u001b[39m \u001b[39mself\u001b[39;49m\u001b[39m.\u001b[39;49m_make_engine(f, \u001b[39mself\u001b[39;49m\u001b[39m.\u001b[39;49mengine)\n",
      "File \u001b[0;32m~/.local/lib/python3.8/site-packages/pandas/io/parsers/readers.py:1661\u001b[0m, in \u001b[0;36mTextFileReader._make_engine\u001b[0;34m(self, f, engine)\u001b[0m\n\u001b[1;32m   1659\u001b[0m     \u001b[39mif\u001b[39;00m \u001b[39m\"\u001b[39m\u001b[39mb\u001b[39m\u001b[39m\"\u001b[39m \u001b[39mnot\u001b[39;00m \u001b[39min\u001b[39;00m mode:\n\u001b[1;32m   1660\u001b[0m         mode \u001b[39m+\u001b[39m\u001b[39m=\u001b[39m \u001b[39m\"\u001b[39m\u001b[39mb\u001b[39m\u001b[39m\"\u001b[39m\n\u001b[0;32m-> 1661\u001b[0m \u001b[39mself\u001b[39m\u001b[39m.\u001b[39mhandles \u001b[39m=\u001b[39m get_handle(\n\u001b[1;32m   1662\u001b[0m     f,\n\u001b[1;32m   1663\u001b[0m     mode,\n\u001b[1;32m   1664\u001b[0m     encoding\u001b[39m=\u001b[39;49m\u001b[39mself\u001b[39;49m\u001b[39m.\u001b[39;49moptions\u001b[39m.\u001b[39;49mget(\u001b[39m\"\u001b[39;49m\u001b[39mencoding\u001b[39;49m\u001b[39m\"\u001b[39;49m, \u001b[39mNone\u001b[39;49;00m),\n\u001b[1;32m   1665\u001b[0m     compression\u001b[39m=\u001b[39;49m\u001b[39mself\u001b[39;49m\u001b[39m.\u001b[39;49moptions\u001b[39m.\u001b[39;49mget(\u001b[39m\"\u001b[39;49m\u001b[39mcompression\u001b[39;49m\u001b[39m\"\u001b[39;49m, \u001b[39mNone\u001b[39;49;00m),\n\u001b[1;32m   1666\u001b[0m     memory_map\u001b[39m=\u001b[39;49m\u001b[39mself\u001b[39;49m\u001b[39m.\u001b[39;49moptions\u001b[39m.\u001b[39;49mget(\u001b[39m\"\u001b[39;49m\u001b[39mmemory_map\u001b[39;49m\u001b[39m\"\u001b[39;49m, \u001b[39mFalse\u001b[39;49;00m),\n\u001b[1;32m   1667\u001b[0m     is_text\u001b[39m=\u001b[39;49mis_text,\n\u001b[1;32m   1668\u001b[0m     errors\u001b[39m=\u001b[39;49m\u001b[39mself\u001b[39;49m\u001b[39m.\u001b[39;49moptions\u001b[39m.\u001b[39;49mget(\u001b[39m\"\u001b[39;49m\u001b[39mencoding_errors\u001b[39;49m\u001b[39m\"\u001b[39;49m, \u001b[39m\"\u001b[39;49m\u001b[39mstrict\u001b[39;49m\u001b[39m\"\u001b[39;49m),\n\u001b[1;32m   1669\u001b[0m     storage_options\u001b[39m=\u001b[39;49m\u001b[39mself\u001b[39;49m\u001b[39m.\u001b[39;49moptions\u001b[39m.\u001b[39;49mget(\u001b[39m\"\u001b[39;49m\u001b[39mstorage_options\u001b[39;49m\u001b[39m\"\u001b[39;49m, \u001b[39mNone\u001b[39;49;00m),\n\u001b[1;32m   1670\u001b[0m )\n\u001b[1;32m   1671\u001b[0m \u001b[39massert\u001b[39;00m \u001b[39mself\u001b[39m\u001b[39m.\u001b[39mhandles \u001b[39mis\u001b[39;00m \u001b[39mnot\u001b[39;00m \u001b[39mNone\u001b[39;00m\n\u001b[1;32m   1672\u001b[0m f \u001b[39m=\u001b[39m \u001b[39mself\u001b[39m\u001b[39m.\u001b[39mhandles\u001b[39m.\u001b[39mhandle\n",
      "File \u001b[0;32m~/.local/lib/python3.8/site-packages/pandas/io/common.py:859\u001b[0m, in \u001b[0;36mget_handle\u001b[0;34m(path_or_buf, mode, encoding, compression, memory_map, is_text, errors, storage_options)\u001b[0m\n\u001b[1;32m    854\u001b[0m \u001b[39melif\u001b[39;00m \u001b[39misinstance\u001b[39m(handle, \u001b[39mstr\u001b[39m):\n\u001b[1;32m    855\u001b[0m     \u001b[39m# Check whether the filename is to be opened in binary mode.\u001b[39;00m\n\u001b[1;32m    856\u001b[0m     \u001b[39m# Binary mode does not support 'encoding' and 'newline'.\u001b[39;00m\n\u001b[1;32m    857\u001b[0m     \u001b[39mif\u001b[39;00m ioargs\u001b[39m.\u001b[39mencoding \u001b[39mand\u001b[39;00m \u001b[39m\"\u001b[39m\u001b[39mb\u001b[39m\u001b[39m\"\u001b[39m \u001b[39mnot\u001b[39;00m \u001b[39min\u001b[39;00m ioargs\u001b[39m.\u001b[39mmode:\n\u001b[1;32m    858\u001b[0m         \u001b[39m# Encoding\u001b[39;00m\n\u001b[0;32m--> 859\u001b[0m         handle \u001b[39m=\u001b[39m \u001b[39mopen\u001b[39;49m(\n\u001b[1;32m    860\u001b[0m             handle,\n\u001b[1;32m    861\u001b[0m             ioargs\u001b[39m.\u001b[39;49mmode,\n\u001b[1;32m    862\u001b[0m             encoding\u001b[39m=\u001b[39;49mioargs\u001b[39m.\u001b[39;49mencoding,\n\u001b[1;32m    863\u001b[0m             errors\u001b[39m=\u001b[39;49merrors,\n\u001b[1;32m    864\u001b[0m             newline\u001b[39m=\u001b[39;49m\u001b[39m\"\u001b[39;49m\u001b[39m\"\u001b[39;49m,\n\u001b[1;32m    865\u001b[0m         )\n\u001b[1;32m    866\u001b[0m     \u001b[39melse\u001b[39;00m:\n\u001b[1;32m    867\u001b[0m         \u001b[39m# Binary mode\u001b[39;00m\n\u001b[1;32m    868\u001b[0m         handle \u001b[39m=\u001b[39m \u001b[39mopen\u001b[39m(handle, ioargs\u001b[39m.\u001b[39mmode)\n",
      "\u001b[0;31mFileNotFoundError\u001b[0m: [Errno 2] No such file or directory: '/users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/cancers/featureCounts/table_of_counts_TPMs_GSE214846_LIHC.csv'"
     ]
    }
   ],
   "source": [
    "## filter patients we are interested in, from those not TCGA\n",
    "for proj in all_projects:\n",
    "    if proj in tcga_projects:\n",
    "        cancer_type = proj[5:]\n",
    "        print(cancer_type)\n",
    "\n",
    "        fc = pd.read_csv(os.path.join(cancer_dir,\"featureCounts/table_of_counts_TPMs_\")+proj+\".csv\")\n",
    "        tumor_transcripts = list()\n",
    "        patients=pd.read_csv(os.path.join(\"/users/genomics/marta/TCGA_RNASeq\",proj,\"results/QC_patients.csv\"))\n",
    "        patients['normal'] = patients['normal'].str.replace('TCGA', cancer_type)\n",
    "        patients['tumor'] = patients['tumor'].str.replace('TCGA', cancer_type)\n",
    "\n",
    "    elif proj in other_projects:\n",
    "        fc = pd.read_csv(os.path.join(cancer_dir,\"featureCounts/table_of_counts_TPMs_\")+proj+\".csv\")\n",
    "        print(fc)\n",
    "        tumor_transcripts = list()\n",
    "        patients=pd.read_csv(os.path.join(\"/users/genomics/marta/cancers_RNASeq\",proj,\"results/patients.csv\"))\n",
    "\n",
    "        samples_to_keep = patients.tumor.values.tolist() + patients.normal.values.tolist() + ['Length','trnascript_id']\n",
    "        fc_selected = fc[fc.columns.intersection(samples_to_keep)]\n",
    "\n",
    "\n",
    "    for index,patient in patients.iterrows():\n",
    "        if proj in tcga_projects:\n",
    "            if \"_\" not in patient.iloc[0]: ## if there are two samples per patient, keep only one\n",
    "                patient_fc = fc[[\"transcript_id\",\"Length\",patient.iloc[2]]]\n",
    "\n",
    "                tumor_patient_fc = patient_fc[patient_fc[patient.iloc[2]] >= 1 ]\n",
    "                tumor_transcripts.extend(tumor_patient_fc.transcript_id.values.tolist())\n",
    "        if proj in other_projects:\n",
    "            patient_fc = fc[[\"transcript_id\",\"Length\",patient.iloc[2]]]\n",
    "\n",
    "            tumor_patient_fc = patient_fc[patient_fc[patient.iloc[2]] >= 1 ]\n",
    "            tumor_transcripts.extend(tumor_patient_fc.transcript_id.values.tolist())            \n",
    "\n",
    "    tumor1FPKM = fc[fc['transcript_id'].isin(tumor_transcripts)]\n",
    "\n",
    "    known = tumor1FPKM[tumor1FPKM['transcript_id'].str.contains('ENST')]\n",
    "    novel = tumor1FPKM[tumor1FPKM['transcript_id'].str.contains('TCONS')]\n",
    "    merged = tumor1FPKM.merge(transcript_gene, on=['transcript_id'], how=\"inner\")\n",
    "    merged = merged[merged['gene_type'].isin(['lncRNA','processed_pseudogene','novel','protein_coding'])]\n",
    "\n",
    "    lncRNA = merged[merged['gene_type'].isin(['lncRNA','processed_pseudogene'])]\n",
    "    print(\"lncRNA: \",len(lncRNA))\n",
    "\n",
    "    cds = merged[merged['gene_type'] == \"protein_coding\"]\n",
    "    print(\"PROTEIN CODING: \",len(cds))\n",
    "\n",
    "    novel = merged[merged['gene_type'] == \"novel\"]\n",
    "    novel = novel[novel['Length'].astype(int) < 91667]\n",
    "    novel = novel[novel['Length'].astype(int) > 300]\n",
    "\n",
    "    print(\"NOVEL: \",len(novel))\n",
    "\n",
    "    # # tumor1FPKM = pd.concat([novel, lncRNA, cds], ignore_index = True)\n",
    "    # merged.to_csv(os.path.join(cancer_dir,\"tumorexpressed/tumor_1FPKM_table_of_counts_\")+cancer_type+\".csv\",index=False)\n",
    "    # merged[['transcript_id','gene_id','gene_name','gene_type','Length']].to_csv(os.path.join(cancer_dir,\"tumorexpressed/tumor_1FPKM_\")+cancer_type+\".csv\",index=False)\n",
    "\n",
    "    tumorONLY_merged = merged[merged.columns.drop(list(merged.filter(regex='normal')))]\n",
    "    ###### GET TUMOR-EXPRESSED > 1 TPM\n",
    "    # Filter only integer columns\n",
    "    int_columns = tumorONLY_merged.select_dtypes(include='int64')\n",
    "\n",
    "    # Create a new column with the count of int columns per row whose value is > 1\n",
    "    # tumorONLY_merged['n'] = int_columns.apply(lambda row: (row > 1).sum(), axis=1)\n",
    "    tumorONLY_merged['n'] = tumorONLY_merged.drop(columns=['transcript_id','gene_id','gene_name','gene_type','Length']).apply(count_greater_than_one, axis=1)\n",
    "    print(tumorONLY_merged)\n",
    "    tumorONLY_merged = tumorONLY_merged[['transcript_id','gene_id','gene_name','gene_type','Length','n']]\n",
    "    tumorONLY_merged.sort_values(by=['n'], ascending=False).to_csv(os.path.join(cancer_dir,\"tumorexpressed/tumor_1FPKM_n_\")+proj+\".csv\",index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "BRCA\n",
      "lncRNA:  10685\n",
      "PROTEIN CODING:  16230\n",
      "NOVEL:  478\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_877837/2433886323.py:48: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  tumorONLY_merged['n'] = tumorONLY_merged.drop(columns=['transcript_id','gene_id','gene_name','gene_type','Length']).apply(count_greater_than_one, axis=1)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "         transcript_id  BRCA-A7-A0CE_tumor  BRCA-A7-A0CH_tumor  \\\n",
      "0       TCONS_00002128          267.709487          282.620818   \n",
      "3       TCONS_00002129         1892.387570         3628.566131   \n",
      "7      ENST00000361390         4922.762555         6200.533782   \n",
      "11     ENST00000361453         4632.671860         5120.310994   \n",
      "17     ENST00000361624        12520.123574        12805.411471   \n",
      "...                ...                 ...                 ...   \n",
      "33013  ENST00000600468            0.050594            0.058366   \n",
      "33014  ENST00000425340            6.479699           10.709122   \n",
      "33015  ENST00000318083            2.510292            1.574990   \n",
      "33016  ENST00000222145            4.556231            4.455789   \n",
      "33017  ENST00000645652            6.365758            1.211338   \n",
      "\n",
      "       BRCA-A7-A0D9_tumor  BRCA-A7-A0DB_2_tumor  BRCA-A7-A0DB_3_tumor  \\\n",
      "0              361.264756            185.775806             11.623270   \n",
      "3             4115.244777           3399.586815            963.154359   \n",
      "7             6418.095567          10891.813775             77.647977   \n",
      "11            5586.259399           8370.488514             29.430953   \n",
      "17           11672.918993          18843.253193            796.207614   \n",
      "...                   ...                   ...                   ...   \n",
      "33013            1.144771              0.126532              0.023628   \n",
      "33014            9.243799              7.595348              2.704196   \n",
      "33015            1.879093              1.354749              0.205675   \n",
      "33016            4.167124              5.805396              0.776473   \n",
      "33017            8.860999              0.665891              0.309405   \n",
      "\n",
      "       BRCA-A7-A0DB_tumor  BRCA-A7-A0DC_2_tumor  BRCA-A7-A0DC_tumor  \\\n",
      "0               96.524783             29.704647          227.476368   \n",
      "3             4385.939713           6390.519697       102138.145955   \n",
      "7             2797.729716             20.072529         1495.924116   \n",
      "11            1551.558168             14.024528         1338.766852   \n",
      "17            7880.224686             83.037266         3549.350361   \n",
      "...                   ...                   ...                 ...   \n",
      "33013            1.541314              0.039761            1.305553   \n",
      "33014            6.473731              0.383081            3.600660   \n",
      "33015            1.535856              0.275801            2.109164   \n",
      "33016            0.399878              0.290222            1.694886   \n",
      "33017            0.090193              1.897206            3.034303   \n",
      "\n",
      "       BRCA-A7-A13E_2_tumor  ...  BRCA-E9-A1RF_tumor  BRCA-E9-A1RH_tumor  \\\n",
      "0                458.229992  ...          493.523186          287.361643   \n",
      "3              20015.393025  ...         6756.367826         3460.016210   \n",
      "7               7829.294039  ...        10058.448711         7443.615679   \n",
      "11              7370.620060  ...        10733.516596         5345.088325   \n",
      "17             28498.287248  ...        15682.743348         9995.941267   \n",
      "...                     ...  ...                 ...                 ...   \n",
      "33013              0.052296  ...            0.179286            0.118307   \n",
      "33014              5.639154  ...            0.788246            0.221063   \n",
      "33015              0.648693  ...            1.487189            2.904119   \n",
      "33016              0.694041  ...            3.335109            0.762618   \n",
      "33017              1.821854  ...            3.853800            2.446586   \n",
      "\n",
      "       BRCA-E9-A1RI_tumor  BRCA-GI-A2C8_tumor  BRCA-GI-A2C9_tumor  Length  \\\n",
      "0              375.259616          557.708841          376.767538     577   \n",
      "3             5343.030078         7524.251942         3957.066005     353   \n",
      "7             8547.331594         5233.659577         4704.639182     956   \n",
      "11            6174.186225         6976.461926         7111.944671    1042   \n",
      "17           19182.445288        15523.834455        13945.291887    1542   \n",
      "...                   ...                 ...                 ...     ...   \n",
      "33013            0.282527            0.280756            0.925387    2426   \n",
      "33014            9.322336            2.267830            7.076410    3116   \n",
      "33015            1.528290            1.206674            2.797257    1858   \n",
      "33016            2.642516            6.076940            2.999381    3199   \n",
      "33017            6.172723            7.491630            6.793344    3273   \n",
      "\n",
      "               gene_id       gene_type    gene_name    n  \n",
      "0          XLOC_001837           novel  XLOC_001837  118  \n",
      "3          XLOC_001838           novel  XLOC_001838  118  \n",
      "7      ENSG00000198888  protein_coding       MT-ND1  118  \n",
      "11     ENSG00000198763  protein_coding       MT-ND2  118  \n",
      "17     ENSG00000198804  protein_coding       MT-CO1  118  \n",
      "...                ...             ...          ...  ...  \n",
      "33013  ENSG00000142233  protein_coding         NTN5    7  \n",
      "33014  ENSG00000176920  protein_coding         FUT2   95  \n",
      "33015  ENSG00000176909  protein_coding       MAMSTR   97  \n",
      "33016  ENSG00000105538  protein_coding       RASIP1  112  \n",
      "33017  ENSG00000174951  protein_coding         FUT1  107  \n",
      "\n",
      "[27680 rows x 124 columns]\n",
      "LUSC\n",
      "lncRNA:  9922\n",
      "PROTEIN CODING:  16280\n",
      "NOVEL:  456\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_877837/2433886323.py:48: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  tumorONLY_merged['n'] = tumorONLY_merged.drop(columns=['transcript_id','gene_id','gene_name','gene_type','Length']).apply(count_greater_than_one, axis=1)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "         transcript_id  LUSC-22-4593_tumor  LUSC-22-4609_tumor  \\\n",
      "0       TCONS_00002128           51.003975          134.036441   \n",
      "3       TCONS_00002129         1071.286059          908.034969   \n",
      "7      ENST00000361390         2250.151348         2307.340567   \n",
      "11     ENST00000361453         2767.517768         1845.464335   \n",
      "17     ENST00000361624         8874.354968         5926.298466   \n",
      "...                ...                 ...                 ...   \n",
      "31495  ENST00000425340            2.286746           11.556526   \n",
      "31496  ENST00000318083           17.205136            2.935649   \n",
      "31497  ENST00000222145            4.531111           12.698763   \n",
      "31498  ENST00000597553            0.055010            0.207989   \n",
      "31499  ENST00000645652            7.776261           10.438391   \n",
      "\n",
      "       LUSC-22-5471_tumor  LUSC-22-5472_tumor  LUSC-22-5478_tumor  \\\n",
      "0               37.784509          106.868616           61.738106   \n",
      "3             1294.795172         1241.888811          838.418900   \n",
      "7             3488.558778         2807.277190         2123.175150   \n",
      "11            4474.758575         2088.326497         2675.909984   \n",
      "17            5739.271021         6854.645918         6047.195291   \n",
      "...                   ...                 ...                 ...   \n",
      "31495           14.819419           56.275578            1.873596   \n",
      "31496            2.064176            3.769544            3.169962   \n",
      "31497            2.598929            7.761278            4.780489   \n",
      "31498            0.058025            0.217197            0.011647   \n",
      "31499            5.701619           13.541222            6.171995   \n",
      "\n",
      "       LUSC-22-5481_tumor  LUSC-22-5482_tumor  LUSC-22-5483_tumor  \\\n",
      "0              166.992083          109.668596           24.610517   \n",
      "3             1470.740541         1788.326728          739.202501   \n",
      "7             4606.423482         2594.561021         1299.746539   \n",
      "11            3186.186910         2631.456999         1648.253664   \n",
      "17           10575.119191         6177.674930         2807.678856   \n",
      "...                   ...                 ...                 ...   \n",
      "31495            1.327571           13.218464            6.839474   \n",
      "31496            2.453620            0.743072            2.490341   \n",
      "31497            9.782019            2.685393            5.286864   \n",
      "31498            0.000000            0.126798            0.184978   \n",
      "31499            2.132277           15.834000           11.017143   \n",
      "\n",
      "       LUSC-22-5489_tumor  ...  LUSC-77-8007_tumor  LUSC-77-8008_tumor  \\\n",
      "0               21.461007  ...          141.328165          164.953150   \n",
      "3              311.002405  ...         2272.587866         1493.855559   \n",
      "7              696.231860  ...         2920.242743         1829.991285   \n",
      "11             565.536215  ...         2633.466081         2778.349503   \n",
      "17            1211.082780  ...         6121.180443         6630.744777   \n",
      "...                   ...  ...                 ...                 ...   \n",
      "31495            1.319816  ...            4.916241           13.994148   \n",
      "31496            2.384317  ...            1.325072            2.552839   \n",
      "31497            3.138310  ...            3.591514            6.856489   \n",
      "31498            0.027267  ...            0.061667            0.082704   \n",
      "31499            1.972531  ...            2.674524            8.326796   \n",
      "\n",
      "       LUSC-85-7710_tumor  LUSC-90-7767_tumor  LUSC-92-7340_tumor  Length  \\\n",
      "0              133.967136           44.804136           57.270193     577   \n",
      "3             2348.221533          687.178820         2189.222832     353   \n",
      "7             1980.313277         1082.867227         1916.965072     956   \n",
      "11            2552.318350         2209.571489         3003.970569    1042   \n",
      "17            6882.403509         2162.213294         6390.661572    1542   \n",
      "...                   ...                 ...                 ...     ...   \n",
      "31495            5.374286            5.495565           16.143327    3116   \n",
      "31496            1.790675            2.794668            1.493300    1858   \n",
      "31497            2.103184           12.490284            4.718211    3199   \n",
      "31498            0.050001            0.000000            0.062546    2218   \n",
      "31499            6.415380            3.285441            8.587290    3273   \n",
      "\n",
      "               gene_id       gene_type    gene_name   n  \n",
      "0          XLOC_001837           novel  XLOC_001837  49  \n",
      "3          XLOC_001838           novel  XLOC_001838  49  \n",
      "7      ENSG00000198888  protein_coding       MT-ND1  49  \n",
      "11     ENSG00000198763  protein_coding       MT-ND2  49  \n",
      "17     ENSG00000198804  protein_coding       MT-CO1  49  \n",
      "...                ...             ...          ...  ..  \n",
      "31495  ENSG00000176920  protein_coding         FUT2  48  \n",
      "31496  ENSG00000176909  protein_coding       MAMSTR  45  \n",
      "31497  ENSG00000105538  protein_coding       RASIP1  49  \n",
      "31498  ENSG00000182264  protein_coding       IZUMO1   1  \n",
      "31499  ENSG00000174951  protein_coding         FUT1  49  \n",
      "\n",
      "[26926 rows x 55 columns]\n",
      "PRAD\n",
      "lncRNA:  7393\n",
      "PROTEIN CODING:  15071\n",
      "NOVEL:  385\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_877837/2433886323.py:48: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  tumorONLY_merged['n'] = tumorONLY_merged.drop(columns=['transcript_id','gene_id','gene_name','gene_type','Length']).apply(count_greater_than_one, axis=1)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "         transcript_id  PRAD-CH-5761_tumor  PRAD-CH-5767_tumor  \\\n",
      "0       TCONS_00002128          144.470157          200.289666   \n",
      "3       TCONS_00002129         2458.473628         7450.921126   \n",
      "7      ENST00000361390         6678.010599        10488.199868   \n",
      "11     ENST00000361453         9911.395469         8530.082248   \n",
      "17     ENST00000361624        19463.648535        20967.183057   \n",
      "...                ...                 ...                 ...   \n",
      "26675  ENST00000600468            0.644878            0.147132   \n",
      "26676  ENST00000425340            0.382016            0.193857   \n",
      "26677  ENST00000318083            0.713887            0.635445   \n",
      "26678  ENST00000222145            5.804824            2.531997   \n",
      "26679  ENST00000645652            4.301946            6.207849   \n",
      "\n",
      "       PRAD-CH-5768_tumor  PRAD-CH-5769_tumor  PRAD-EJ-7115_tumor  \\\n",
      "0              226.420020          328.384368          510.315150   \n",
      "3             1961.811550         3546.985440         3589.945905   \n",
      "7             5461.789674         4996.339904         8277.259977   \n",
      "11            5551.177461         5425.414583         8763.647800   \n",
      "17           15651.177379        15381.102123        17351.991813   \n",
      "...                   ...                 ...                 ...   \n",
      "26675            0.469524            0.132294            0.165845   \n",
      "26676            0.781878            1.544985            1.241549   \n",
      "26677            0.885530            1.014828            1.315929   \n",
      "26678            1.839689            2.420379            5.359776   \n",
      "26679            8.062427            2.929491            4.765798   \n",
      "\n",
      "       PRAD-EJ-7123_tumor  PRAD-EJ-7125_tumor  PRAD-EJ-7314_tumor  \\\n",
      "0              212.564533          275.590330          177.273543   \n",
      "3             1559.115893         3123.811178         2791.235329   \n",
      "7             7722.285403         5080.637821         7049.996821   \n",
      "11            7314.107847         2987.790372         6731.422496   \n",
      "17           13680.701926        20197.713531        16628.616867   \n",
      "...                   ...                 ...                 ...   \n",
      "26675            0.277338            0.097467            0.289995   \n",
      "26676            0.449843            0.177063            0.775502   \n",
      "26677            1.146718            1.505944            1.053629   \n",
      "26678            4.381717            5.753060            2.533875   \n",
      "26679           12.599556            9.969665            8.541884   \n",
      "\n",
      "       PRAD-EJ-7315_tumor  ...  PRAD-HC-8258_tumor  PRAD-HC-8259_tumor  \\\n",
      "0              442.886696  ...          456.656151          141.418266   \n",
      "3             3472.747781  ...         4055.726184         2022.448186   \n",
      "7             3014.547644  ...         5941.610557         4105.542366   \n",
      "11            4281.726955  ...         5565.248300         4257.806174   \n",
      "17           10855.784931  ...        15773.602013        15072.208189   \n",
      "...                   ...  ...                 ...                 ...   \n",
      "26675            0.215877  ...            0.709608            0.517655   \n",
      "26676            0.961757  ...            3.571063            0.275238   \n",
      "26677            0.782979  ...            1.785937            1.022099   \n",
      "26678            4.420258  ...            5.365798            2.661814   \n",
      "26679            6.302688  ...            7.150179            5.820918   \n",
      "\n",
      "       PRAD-HC-8260_tumor  PRAD-HC-8262_tumor  PRAD-J4-A83J_tumor  Length  \\\n",
      "0              427.166337          385.353889          260.385031     577   \n",
      "3             2783.852791         4218.043451         2500.076841     353   \n",
      "7             5080.728864         7608.681695         5559.774147     956   \n",
      "11            6870.076675         9598.100324         3960.828018    1042   \n",
      "17           12279.245270        21951.274220        12455.248326    1542   \n",
      "...                   ...                 ...                 ...     ...   \n",
      "26675            1.517422            0.539755            1.035167    2426   \n",
      "26676            2.190148            1.137102            0.817622    3116   \n",
      "26677            1.691731            1.091689            1.939285    1858   \n",
      "26678            4.328611            1.950337            3.413180    3199   \n",
      "26679            5.718859            4.502809            7.361463    3273   \n",
      "\n",
      "               gene_id       gene_type    gene_name   n  \n",
      "0          XLOC_001837           novel  XLOC_001837  54  \n",
      "3          XLOC_001838           novel  XLOC_001838  54  \n",
      "7      ENSG00000198888  protein_coding       MT-ND1  54  \n",
      "11     ENSG00000198763  protein_coding       MT-ND2  54  \n",
      "17     ENSG00000198804  protein_coding       MT-CO1  54  \n",
      "...                ...             ...          ...  ..  \n",
      "26675  ENSG00000142233  protein_coding         NTN5   4  \n",
      "26676  ENSG00000176920  protein_coding         FUT2  28  \n",
      "26677  ENSG00000176909  protein_coding       MAMSTR  43  \n",
      "26678  ENSG00000105538  protein_coding       RASIP1  54  \n",
      "26679  ENSG00000174951  protein_coding         FUT1  54  \n",
      "\n",
      "[23072 rows x 60 columns]\n",
      "KIRC\n",
      "lncRNA:  7978\n",
      "PROTEIN CODING:  15463\n",
      "NOVEL:  391\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_877837/2433886323.py:48: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  tumorONLY_merged['n'] = tumorONLY_merged.drop(columns=['transcript_id','gene_id','gene_name','gene_type','Length']).apply(count_greater_than_one, axis=1)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "         transcript_id  KIRC-A3-3358_tumor  KIRC-A3-3387_tumor  \\\n",
      "0       TCONS_00002128          304.946790          192.179407   \n",
      "3       TCONS_00002129         4986.046589         1906.270954   \n",
      "7      ENST00000361390        10920.358956         4938.040304   \n",
      "11     ENST00000361453        10844.425826         4016.360207   \n",
      "17     ENST00000361624        18364.112545        10536.273892   \n",
      "...                ...                 ...                 ...   \n",
      "27694  ENST00000594723            4.237029            2.839989   \n",
      "27695  ENST00000084798            5.050094            8.333380   \n",
      "27697  ENST00000318083            0.678024            1.242293   \n",
      "27698  ENST00000222145            7.716036           13.531194   \n",
      "27699  ENST00000645652            2.201132            2.337848   \n",
      "\n",
      "       KIRC-B0-4700_tumor  KIRC-B0-4712_tumor  KIRC-B0-5402_tumor  \\\n",
      "0              149.322490          212.805185          165.340737   \n",
      "3             5599.814533         5658.823521         5616.135322   \n",
      "7             5546.765339         5261.825391         8875.306515   \n",
      "11            8960.732524         6646.937654        14670.313316   \n",
      "17           20526.327943        15242.104863        16927.986756   \n",
      "...                   ...                 ...                 ...   \n",
      "27694            2.623518            2.962947            3.688874   \n",
      "27695            5.759942            6.682162            4.157847   \n",
      "27697            0.963740            0.814625            1.356689   \n",
      "27698           13.154039            8.229245           16.107319   \n",
      "27699            2.352491            1.965379            3.143078   \n",
      "\n",
      "       KIRC-B0-5690_tumor  KIRC-B0-5691_tumor  KIRC-B0-5694_tumor  \\\n",
      "0              455.322476          245.634296          745.087188   \n",
      "3             2202.681818         1394.874961         4522.461604   \n",
      "7             9175.036345         5250.993730        12494.464052   \n",
      "11            9050.063275         4744.769032        14616.686705   \n",
      "17           15649.538272        11531.022280        29768.962180   \n",
      "...                   ...                 ...                 ...   \n",
      "27694            8.826903           10.778038            5.827746   \n",
      "27695           13.353563           15.513494            4.752404   \n",
      "27697            1.918256            1.806667            0.870077   \n",
      "27698           78.546542           24.484220           17.550256   \n",
      "27699           15.882388            5.735759            3.313385   \n",
      "\n",
      "       KIRC-B0-5696_tumor  ...  KIRC-CZ-5985_tumor  KIRC-CZ-5986_tumor  \\\n",
      "0              409.678067  ...          148.953319          309.183104   \n",
      "3             4896.390905  ...         3099.828274         2664.354808   \n",
      "7             8350.987318  ...         3475.597393         9367.288272   \n",
      "11            8607.977454  ...         5605.231988         7149.978850   \n",
      "17           18035.079692  ...        14037.703908        16903.362349   \n",
      "...                   ...  ...                 ...                 ...   \n",
      "27694           12.174759  ...            3.874991           11.827101   \n",
      "27695            8.896044  ...            2.292932            7.290249   \n",
      "27697            1.414747  ...            0.203180            0.859672   \n",
      "27698           24.972723  ...            5.212030           10.295977   \n",
      "27699            3.618163  ...            1.191848            3.920943   \n",
      "\n",
      "       KIRC-CZ-5987_tumor  KIRC-CZ-5988_tumor  KIRC-CZ-5989_tumor  Length  \\\n",
      "0              362.569222          157.486368          207.815204     577   \n",
      "3             6106.733516         2897.694075         2317.226966     353   \n",
      "7            17976.406516         5193.904236         3771.783691     956   \n",
      "11           17845.022529         5938.088051        12877.614169    1042   \n",
      "17           29109.500932        11036.544847        19189.542239    1542   \n",
      "...                   ...                 ...                 ...     ...   \n",
      "27694            2.558514            2.288668            1.238388    5756   \n",
      "27695            8.730176            4.837638            2.846824    1715   \n",
      "27697            0.792616            1.689577            0.105109    1858   \n",
      "27698            4.680291           16.226786            2.350343    3199   \n",
      "27699            0.774911            4.555871            2.162951    3273   \n",
      "\n",
      "               gene_id       gene_type    gene_name   n  \n",
      "0          XLOC_001837           novel  XLOC_001837  71  \n",
      "3          XLOC_001838           novel  XLOC_001838  71  \n",
      "7      ENSG00000198888  protein_coding       MT-ND1  71  \n",
      "11     ENSG00000198763  protein_coding       MT-ND2  71  \n",
      "17     ENSG00000198804  protein_coding       MT-CO1  71  \n",
      "...                ...             ...          ...  ..  \n",
      "27694  ENSG00000105516  protein_coding          DBP  70  \n",
      "27695  ENSG00000063180  protein_coding         CA11  71  \n",
      "27697  ENSG00000176909  protein_coding       MAMSTR  18  \n",
      "27698  ENSG00000105538  protein_coding       RASIP1  71  \n",
      "27699  ENSG00000174951  protein_coding         FUT1  63  \n",
      "\n",
      "[24050 rows x 77 columns]\n",
      "KIRP\n",
      "lncRNA:  7911\n",
      "PROTEIN CODING:  15436\n",
      "NOVEL:  390\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_877837/2433886323.py:48: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  tumorONLY_merged['n'] = tumorONLY_merged.drop(columns=['transcript_id','gene_id','gene_name','gene_type','Length']).apply(count_greater_than_one, axis=1)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "         transcript_id  KIRP-A4-A4ZT_tumor  KIRP-A4-A57E_tumor  \\\n",
      "0       TCONS_00002128          294.209363          161.241017   \n",
      "3       TCONS_00002129         7291.239248        15292.071033   \n",
      "7      ENST00000361390        11544.965302         4320.608463   \n",
      "11     ENST00000361453         9045.966882         5539.460963   \n",
      "17     ENST00000361624        28101.293761        36216.373746   \n",
      "...                ...                 ...                 ...   \n",
      "27451  ENST00000084798            8.700903            1.310541   \n",
      "27453  ENST00000425340            0.249967            0.371580   \n",
      "27454  ENST00000318083            0.463341            1.008063   \n",
      "27455  ENST00000222145            2.114447           10.953976   \n",
      "27456  ENST00000645652            0.263027            0.208092   \n",
      "\n",
      "       KIRP-B9-4115_tumor  KIRP-BQ-5875_tumor  KIRP-BQ-5877_tumor  \\\n",
      "0              740.683401          521.762270         1138.027864   \n",
      "3             3068.542507        17463.736697          563.632531   \n",
      "7             4843.475902         3559.948577          949.240893   \n",
      "11            5175.146025         8679.856433         1434.880036   \n",
      "17           20224.580519        41171.558008         2813.041821   \n",
      "...                   ...                 ...                 ...   \n",
      "27451            8.797772            3.117847           32.204276   \n",
      "27453            0.527086            0.291250            0.256262   \n",
      "27454            0.777886            0.475247            0.698375   \n",
      "27455            0.698238            1.449139            3.640184   \n",
      "27456            0.227484            0.164869            1.240176   \n",
      "\n",
      "       KIRP-BQ-5878_tumor  KIRP-BQ-5879_tumor  KIRP-BQ-5882_tumor  \\\n",
      "0              592.057771          508.004909           59.533179   \n",
      "3             1080.438853         1704.694127         7599.588486   \n",
      "7             1133.221551         1748.151310         5020.449613   \n",
      "11            2681.834576         2031.767743         6353.397862   \n",
      "17            5186.017241         7695.153526        16241.602501   \n",
      "...                   ...                 ...                 ...   \n",
      "27451            8.200441           46.735267           19.843064   \n",
      "27453            0.630017            0.240563            0.538797   \n",
      "27454            2.580836            4.512563            1.921945   \n",
      "27455            1.438605            8.157849            3.340505   \n",
      "27456            0.216320            2.018799            1.294593   \n",
      "\n",
      "       KIRP-BQ-5884_tumor  ...  KIRP-GL-A59R_tumor  KIRP-GL-A9DE_tumor  \\\n",
      "0              176.609196  ...          248.589066         1062.695508   \n",
      "3             2869.436568  ...         3278.380114        11904.627730   \n",
      "7             5928.503067  ...         5163.950374        18695.867349   \n",
      "11            4344.946094  ...         9552.507556         7279.134527   \n",
      "17           15283.331106  ...        18094.606246        47871.810809   \n",
      "...                   ...  ...                 ...                 ...   \n",
      "27451           40.036445  ...           10.518544           16.461383   \n",
      "27453            0.193555  ...            0.827036            0.117899   \n",
      "27454            0.774059  ...            0.472840            0.517128   \n",
      "27455            2.385666  ...            2.398425            1.068898   \n",
      "27456            0.389802  ...            0.823154            0.112244   \n",
      "\n",
      "       KIRP-P4-A5E8_tumor  KIRP-P4-A5ED_tumor  KIRP-Y8-A8RY_tumor  Length  \\\n",
      "0             1381.522579          422.579139          796.244178     577   \n",
      "3             4553.466082         5322.131912        12061.613206     353   \n",
      "7             5270.039521        10925.991548        12260.714036     956   \n",
      "11            7270.877006        10616.854296        10946.226029    1042   \n",
      "17           18075.168186        34564.117695        36105.272507    1542   \n",
      "...                   ...                 ...                 ...     ...   \n",
      "27451           85.846158           15.114889           11.384795    1715   \n",
      "27453            0.472835            7.205833            0.704047    3116   \n",
      "27454            2.085238            2.883660            0.616608    1858   \n",
      "27455            1.518164            1.481224            7.154976    3199   \n",
      "27456            0.650222            3.122565            0.253215    3273   \n",
      "\n",
      "               gene_id       gene_type    gene_name   n  \n",
      "0          XLOC_001837           novel  XLOC_001837  32  \n",
      "3          XLOC_001838           novel  XLOC_001838  32  \n",
      "7      ENSG00000198888  protein_coding       MT-ND1  32  \n",
      "11     ENSG00000198763  protein_coding       MT-ND2  32  \n",
      "17     ENSG00000198804  protein_coding       MT-CO1  32  \n",
      "...                ...             ...          ...  ..  \n",
      "27451  ENSG00000063180  protein_coding         CA11  32  \n",
      "27453  ENSG00000176920  protein_coding         FUT2   4  \n",
      "27454  ENSG00000176909  protein_coding       MAMSTR  15  \n",
      "27455  ENSG00000105538  protein_coding       RASIP1  29  \n",
      "27456  ENSG00000174951  protein_coding         FUT1   9  \n",
      "\n",
      "[23964 rows x 38 columns]\n",
      "LUAD\n",
      "lncRNA:  10065\n",
      "PROTEIN CODING:  16087\n",
      "NOVEL:  431\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_877837/2433886323.py:48: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  tumorONLY_merged['n'] = tumorONLY_merged.drop(columns=['transcript_id','gene_id','gene_name','gene_type','Length']).apply(count_greater_than_one, axis=1)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "         transcript_id  LUAD-38-4625_tumor  LUAD-38-4626_tumor  \\\n",
      "0       TCONS_00002128           33.081218          167.607586   \n",
      "3       TCONS_00002129          559.679343         1482.097508   \n",
      "7      ENST00000361390         1205.134477         5597.900024   \n",
      "11     ENST00000361453         2538.791386         6117.085762   \n",
      "17     ENST00000361624         3116.938737        12994.530576   \n",
      "...                ...                 ...                 ...   \n",
      "31910  ENST00000084798           25.689259            7.925141   \n",
      "31912  ENST00000425340            4.456336            5.032938   \n",
      "31913  ENST00000318083            5.428071            1.052470   \n",
      "31914  ENST00000222145            2.926995           13.684228   \n",
      "31915  ENST00000645652            6.753087            5.483625   \n",
      "\n",
      "       LUAD-38-4627_tumor  LUAD-38-4632_tumor  LUAD-44-2655_tumor  \\\n",
      "0              112.342463           46.514476          176.747297   \n",
      "3             1980.756407         1469.437189         2027.146515   \n",
      "7             2562.342408         1400.686953         3073.669648   \n",
      "11            2158.531736         1652.384391         2743.003783   \n",
      "17            6905.110135         3569.193451         9536.072474   \n",
      "...                   ...                 ...                 ...   \n",
      "31910           17.661755           21.572719            6.777051   \n",
      "31912            1.054521            7.186961           11.709980   \n",
      "31913            1.736353            1.025131            0.409135   \n",
      "31914           11.737647            2.309440            5.352890   \n",
      "31915            1.971368            3.897241            2.701500   \n",
      "\n",
      "       LUAD-44-2657_tumor  LUAD-44-2661_tumor  LUAD-44-2662_2_tumor  \\\n",
      "0               86.103727          113.310437            117.098006   \n",
      "3             1034.782732         1269.114410           2973.631013   \n",
      "7             2154.912428         2121.849085           1205.811542   \n",
      "11            2024.547093         1516.359737            801.045048   \n",
      "17            5306.118002         5072.174173           3466.056922   \n",
      "...                   ...                 ...                   ...   \n",
      "31910           14.795581           10.507171              0.758958   \n",
      "31912            6.452550            6.217464              6.227814   \n",
      "31913            1.202377            0.879397              0.445802   \n",
      "31914            3.158202            1.583354              0.517849   \n",
      "31915            2.770986            3.858192              1.084587   \n",
      "\n",
      "       LUAD-44-2662_3_tumor  ...  LUAD-91-6831_tumor  LUAD-91-6835_tumor  \\\n",
      "0                 14.210802  ...           42.131053           94.278298   \n",
      "3               3237.693254  ...          650.942193         1053.292227   \n",
      "7                 25.751833  ...         1580.175705         3738.324094   \n",
      "11                16.843366  ...         3276.767598         2696.558594   \n",
      "17               312.343788  ...         3957.716524         7574.696624   \n",
      "...                     ...  ...                 ...                 ...   \n",
      "31910              0.949279  ...            3.939015           19.948458   \n",
      "31912              6.256888  ...           12.533113            8.984642   \n",
      "31913              0.224397  ...            4.166632            2.401713   \n",
      "31914              0.688895  ...           14.597112            6.011487   \n",
      "31915              1.207123  ...            6.508321            4.820566   \n",
      "\n",
      "       LUAD-91-6836_tumor  LUAD-91-6847_tumor  LUAD-91-6849_tumor  Length  \\\n",
      "0               66.807020          109.339130          124.250307     577   \n",
      "3             1596.335355         1908.029621         1397.513877     353   \n",
      "7             4486.071940         3639.097572         4514.471811     956   \n",
      "11            2175.129476         6286.330975         4203.969373    1042   \n",
      "17            7584.572763        13312.865033         7674.749259    1542   \n",
      "...                   ...                 ...                 ...     ...   \n",
      "31910           10.884314            0.989946            8.618103    1715   \n",
      "31912            4.965673            7.235630            7.164118    3116   \n",
      "31913            1.549356            2.266114            1.056241    1858   \n",
      "31914            1.926298            1.178188            3.450778    3199   \n",
      "31915            7.805836            2.935933            3.204121    3273   \n",
      "\n",
      "               gene_id       gene_type    gene_name   n  \n",
      "0          XLOC_001837           novel  XLOC_001837  70  \n",
      "3          XLOC_001838           novel  XLOC_001838  70  \n",
      "7      ENSG00000198888  protein_coding       MT-ND1  70  \n",
      "11     ENSG00000198763  protein_coding       MT-ND2  70  \n",
      "17     ENSG00000198804  protein_coding       MT-CO1  70  \n",
      "...                ...             ...          ...  ..  \n",
      "31910  ENSG00000063180  protein_coding         CA11  65  \n",
      "31912  ENSG00000176920  protein_coding         FUT2  69  \n",
      "31913  ENSG00000176909  protein_coding       MAMSTR  40  \n",
      "31914  ENSG00000105538  protein_coding       RASIP1  65  \n",
      "31915  ENSG00000174951  protein_coding         FUT1  68  \n",
      "\n",
      "[26832 rows x 76 columns]\n"
     ]
    }
   ],
   "source": [
    "for proj in projects:\n",
    "    cancer_type = proj[5:]\n",
    "    print(cancer_type)\n",
    "\n",
    "    fc = pd.read_csv(os.path.join(cancer_dir,\"featureCounts/table_of_counts_TPMs_\")+cancer_type+\".csv\")\n",
    "    tumor_transcripts = list()\n",
    "    patients=pd.read_csv(os.path.join(\"/users/genomics/marta/TCGA_RNASeq\",proj,\"results/QC_patients.csv\"))\n",
    "    patients['normal'] = patients['normal'].str.replace('TCGA', cancer_type)\n",
    "    patients['tumor'] = patients['tumor'].str.replace('TCGA', cancer_type)\n",
    "\n",
    "    for index,patient in patients.iterrows():\n",
    "        if \"_\" not in patient.iloc[0]: ## if there are two samples per patient, keep only one\n",
    "            patient_fc = fc[[\"transcript_id\",\"Length\",patient.iloc[2]]]\n",
    "\n",
    "            tumor_patient_fc = patient_fc[patient_fc[patient.iloc[2]] >= 1 ]\n",
    "            tumor_transcripts.extend(tumor_patient_fc.transcript_id.values.tolist())\n",
    "\n",
    "    tumor1FPKM = fc[fc['transcript_id'].isin(tumor_transcripts)]\n",
    "\n",
    "    known = tumor1FPKM[tumor1FPKM['transcript_id'].str.contains('ENST')]\n",
    "    novel = tumor1FPKM[tumor1FPKM['transcript_id'].str.contains('TCONS')]\n",
    "    merged = tumor1FPKM.merge(transcript_gene, on=['transcript_id'], how=\"inner\")\n",
    "    merged = merged[merged['gene_type'].isin(['lncRNA','processed_pseudogene','novel','protein_coding'])]\n",
    "\n",
    "    lncRNA = merged[merged['gene_type'].isin(['lncRNA','processed_pseudogene'])]\n",
    "    print(\"lncRNA: \",len(lncRNA))\n",
    "\n",
    "    cds = merged[merged['gene_type'] == \"protein_coding\"]\n",
    "    print(\"PROTEIN CODING: \",len(cds))\n",
    "\n",
    "    novel = merged[merged['gene_type'] == \"novel\"]\n",
    "    novel = novel[novel['Length'].astype(int) < 91667]\n",
    "    novel = novel[novel['Length'].astype(int) > 300]\n",
    "\n",
    "    print(\"NOVEL: \",len(novel))\n",
    "\n",
    "    # # tumor1FPKM = pd.concat([novel, lncRNA, cds], ignore_index = True)\n",
    "    # merged.to_csv(os.path.join(cancer_dir,\"tumorexpressed/tumor_1FPKM_table_of_counts_\")+cancer_type+\".csv\",index=False)\n",
    "    # merged[['transcript_id','gene_id','gene_name','gene_type','Length']].to_csv(os.path.join(cancer_dir,\"tumorexpressed/tumor_1FPKM_\")+cancer_type+\".csv\",index=False)\n",
    "\n",
    "    tumorONLY_merged = merged[merged.columns.drop(list(merged.filter(regex='normal')))]\n",
    "    ###### GET TUMOR-EXPRESSED > 1 TPM\n",
    "    # Filter only integer columns\n",
    "    int_columns = tumorONLY_merged.select_dtypes(include='int64')\n",
    "    \n",
    "    # Create a new column with the count of int columns per row whose value is > 1\n",
    "    # tumorONLY_merged['n'] = int_columns.apply(lambda row: (row > 1).sum(), axis=1)\n",
    "    tumorONLY_merged['n'] = tumorONLY_merged.drop(columns=['transcript_id','gene_id','gene_name','gene_type','Length']).apply(count_greater_than_one, axis=1)\n",
    "    print(tumorONLY_merged)\n",
    "    tumorONLY_merged = tumorONLY_merged[['transcript_id','gene_id','gene_name','gene_type','Length','n']]\n",
    "    tumorONLY_merged.sort_values(by=['n'], ascending=False).to_csv(os.path.join(cancer_dir,\"tumorexpressed/tumor_1FPKM_n_\")+cancer_type+\".csv\",index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>transcript_id</th>\n",
       "      <th>gene_id</th>\n",
       "      <th>gene_name</th>\n",
       "      <th>gene_type</th>\n",
       "      <th>Length</th>\n",
       "      <th>n</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>TCONS_00002128</td>\n",
       "      <td>XLOC_001837</td>\n",
       "      <td>XLOC_001837</td>\n",
       "      <td>novel</td>\n",
       "      <td>577</td>\n",
       "      <td>70</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>TCONS_00002129</td>\n",
       "      <td>XLOC_001838</td>\n",
       "      <td>XLOC_001838</td>\n",
       "      <td>novel</td>\n",
       "      <td>353</td>\n",
       "      <td>70</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>ENST00000361390</td>\n",
       "      <td>ENSG00000198888</td>\n",
       "      <td>MT-ND1</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>956</td>\n",
       "      <td>70</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>ENST00000361453</td>\n",
       "      <td>ENSG00000198763</td>\n",
       "      <td>MT-ND2</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>1042</td>\n",
       "      <td>70</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>ENST00000361624</td>\n",
       "      <td>ENSG00000198804</td>\n",
       "      <td>MT-CO1</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>1542</td>\n",
       "      <td>70</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31910</th>\n",
       "      <td>ENST00000084798</td>\n",
       "      <td>ENSG00000063180</td>\n",
       "      <td>CA11</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>1715</td>\n",
       "      <td>65</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31912</th>\n",
       "      <td>ENST00000425340</td>\n",
       "      <td>ENSG00000176920</td>\n",
       "      <td>FUT2</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>3116</td>\n",
       "      <td>69</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31913</th>\n",
       "      <td>ENST00000318083</td>\n",
       "      <td>ENSG00000176909</td>\n",
       "      <td>MAMSTR</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>1858</td>\n",
       "      <td>40</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31914</th>\n",
       "      <td>ENST00000222145</td>\n",
       "      <td>ENSG00000105538</td>\n",
       "      <td>RASIP1</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>3199</td>\n",
       "      <td>65</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31915</th>\n",
       "      <td>ENST00000645652</td>\n",
       "      <td>ENSG00000174951</td>\n",
       "      <td>FUT1</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>3273</td>\n",
       "      <td>68</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>26832 rows × 6 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "         transcript_id          gene_id    gene_name       gene_type  Length  \\\n",
       "0       TCONS_00002128      XLOC_001837  XLOC_001837           novel     577   \n",
       "3       TCONS_00002129      XLOC_001838  XLOC_001838           novel     353   \n",
       "7      ENST00000361390  ENSG00000198888       MT-ND1  protein_coding     956   \n",
       "11     ENST00000361453  ENSG00000198763       MT-ND2  protein_coding    1042   \n",
       "17     ENST00000361624  ENSG00000198804       MT-CO1  protein_coding    1542   \n",
       "...                ...              ...          ...             ...     ...   \n",
       "31910  ENST00000084798  ENSG00000063180         CA11  protein_coding    1715   \n",
       "31912  ENST00000425340  ENSG00000176920         FUT2  protein_coding    3116   \n",
       "31913  ENST00000318083  ENSG00000176909       MAMSTR  protein_coding    1858   \n",
       "31914  ENST00000222145  ENSG00000105538       RASIP1  protein_coding    3199   \n",
       "31915  ENST00000645652  ENSG00000174951         FUT1  protein_coding    3273   \n",
       "\n",
       "        n  \n",
       "0      70  \n",
       "3      70  \n",
       "7      70  \n",
       "11     70  \n",
       "17     70  \n",
       "...    ..  \n",
       "31910  65  \n",
       "31912  69  \n",
       "31913  40  \n",
       "31914  65  \n",
       "31915  68  \n",
       "\n",
       "[26832 rows x 6 columns]"
      ]
     },
     "execution_count": 21,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "tumorONLY_merged"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "#Select normal samples and create a file with min max expression in normal samples per cancer type\n",
    "tolerance_dictionary = dict()\n",
    "\n",
    "for proj in projects:\n",
    "    cancer_type = proj[5:]\n",
    "    fc = pd.read_csv(os.path.join(cancer_dir,\"featureCounts/table_of_counts_TPMs_\")+cancer_type+\".csv\")\n",
    "    normals = [col for col in fc.columns if 'normal' in col]\n",
    "    tolerance_dictionary[cancer_type] = len(normals)*0.05"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'BRCA': 5.6000000000000005,\n",
       " 'LUSC': 2.45,\n",
       " 'PRAD': 2.6,\n",
       " 'KIRC': 3.5500000000000003,\n",
       " 'KIRP': 1.6,\n",
       " 'LUAD': 2.9000000000000004}"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "tolerance_dictionary"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "BRCA\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_877837/4218154473.py:18: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  fc_normals['counts_01'] = fc_normals.apply(count_values_above_01, axis=1)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "LUSC\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_877837/4218154473.py:18: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  fc_normals['counts_01'] = fc_normals.apply(count_values_above_01, axis=1)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PRAD\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_877837/4218154473.py:18: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  fc_normals['counts_01'] = fc_normals.apply(count_values_above_01, axis=1)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "KIRC\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_877837/4218154473.py:18: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  fc_normals['counts_01'] = fc_normals.apply(count_values_above_01, axis=1)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "KIRP\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_877837/4218154473.py:18: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  fc_normals['counts_01'] = fc_normals.apply(count_values_above_01, axis=1)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "LUAD\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_877837/4218154473.py:18: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  fc_normals['counts_01'] = fc_normals.apply(count_values_above_01, axis=1)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "5956\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_877837/4218154473.py:45: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  summary_tolerance_accepted.drop([\"counts_01\"], axis=1, inplace=True)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "1225\n",
      "Done!\n"
     ]
    }
   ],
   "source": [
    "## TUMOR-SPECIFIC\n",
    "for proj in projects:\n",
    "    tumorspecific_candidates = pd.DataFrame()\n",
    "    cancer_type = proj[5:]\n",
    "    print(cancer_type)\n",
    "\n",
    "    ## import table of counts\n",
    "    fc = pd.read_csv(os.path.join(cancer_dir,\"featureCounts/table_of_counts_TPMs_\")+cancer_type+\".csv\")\n",
    "    fc_info = fc.merge(transcript_gene, on=['transcript_id'], how=\"inner\")\n",
    "    fc_info = fc_info[fc_info['gene_type'].isin(['lncRNA','processed_pseudogene','novel','protein_coding'])]\n",
    "\n",
    "    ## table of counts normal samples\n",
    "    normals = [col for col in fc_info.columns if 'normal' in col]\n",
    "    fc_normals = fc_info[normals]\n",
    "    # Function to count values above 0.1 in a row\n",
    "    def count_values_above_01(row):\n",
    "        return sum(row > 0.1)\n",
    "    fc_normals['counts_01'] = fc_normals.apply(count_values_above_01, axis=1)\n",
    "    fc_normals_info = pd.concat([fc_normals, fc_info[['gene_id','transcript_id','gene_name','gene_type']]], axis=1)\n",
    "    fc_normals_info = fc_normals_info[['gene_id','transcript_id','gene_name','gene_type','counts_01']]\n",
    "\n",
    "    ## import patients data\n",
    "    patients=pd.read_csv(os.path.join(\"/users/genomics/marta/TCGA_RNASeq\",proj,\"results/QC_patients.csv\"))\n",
    "    patients['normal'] = patients['normal'].str.replace('TCGA', cancer_type)\n",
    "    patients['tumor'] = patients['tumor'].str.replace('TCGA', cancer_type)\n",
    "\n",
    "    for index,patient in patients.iterrows():\n",
    "        if \"_\" not in patient.iloc[0]: ## if there are two samples per patient, keep only one\n",
    "            patient_fc = fc_info[[\"transcript_id\",\"gene_id\",\"gene_name\",\"gene_type\",\"Length\",patient.iloc[1],patient.iloc[2]]] ## patient - normal - tumor\n",
    "            tumor1TPM = patient_fc[patient_fc[patient.iloc[2]] > 1 ]\n",
    "            tumor1TPM_normal01TPM = tumor1TPM[tumor1TPM[patient.iloc[1]] < 0.1 ]\n",
    "            try:\n",
    "                os.makedirs(os.path.join(cancer_dir,\"tumorspecific\",cancer_type,patient.iloc[0]))\n",
    "            except:\n",
    "                pass\n",
    "            tumor1TPM_normal01TPM.to_csv(os.path.join(cancer_dir,\"tumorspecific\",cancer_type,patient.iloc[0],\"tumorspecific_TPMs.csv\"), index=None)\n",
    "            tumorspecific_candidates = pd.concat([tumorspecific_candidates,tumor1TPM_normal01TPM[[\"transcript_id\",\"gene_id\",\"gene_name\",\"gene_type\",\"Length\"]]])\n",
    "summary = tumorspecific_candidates.groupby([\"transcript_id\",\"gene_id\",\"gene_name\",\"gene_type\",\"Length\"]).size().reset_index(name='n')\n",
    "summary.sort_values(by=\"n\", ascending=False).to_csv(os.path.join(cancer_dir,\"tumorspecific\")+\"/tumorspecific_n_\"+cancer_type+\".csv\", index=False)\n",
    "print(len(summary))\n",
    "### tolerance 5% only\n",
    "summary_tolerance = fc_normals_info.merge(summary, on=[\"gene_id\",\"transcript_id\"], how=\"right\")\n",
    "# ## Expression > 0.1 in less than 5%\n",
    "summary_tolerance_accepted = summary_tolerance[summary_tolerance['counts_01'] < tolerance_dictionary[cancer_type]]\n",
    "summary_tolerance_accepted.drop([\"counts_01\"], axis=1, inplace=True)\n",
    "summary_tolerance_accepted.sort_values(by=\"n\", ascending=False).to_csv(os.path.join(cancer_dir,\"tumorspecific\")+\"/tumorspecific_n_\"+cancer_type+\"_TOLERANCE5PERCENT.csv\", index=False)\n",
    "print(len(summary_tolerance_accepted))\n",
    "print(\"Done!\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.16"
  },
  "orig_nbformat": 4
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
