{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {},
   "outputs": [],
   "source": [
    "import os,re,glob\n",
    "import pandas as pd\n",
    "import numpy as np\n",
    "from collections import Counter\n",
    "from rna_seq_normalization import Normalization as Norm\n",
    "\n",
    "users_dir = \"/users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction\"\n",
    "specie = \"human\"\n",
    "GENOMEDIR = \"/genomics/users/marta/genomes\"\n",
    "\n",
    "### cancer data\n",
    "cancer_dir = users_dir + \"/cancers\"\n",
    "raw_cancer_dir=\"/users/genomics/marta/TCGA_RNASeq\"\n",
    "projects=[\"TCGA-BRCA\",\"TCGA-LUSC\",\"TCGA-PRAD\",\"TCGA-KIRC\",\"TCGA-KIRP\",\"TCGA-LUAD\",\"TCGA-BLCA\",\"TCGA-LIHC\"]\n",
    "\n",
    "## annotation file\n",
    "annotation=\"/users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/human/newReference_Resconstructed/gencode.v38.gffcompare.TestisLiverBrain.annotation.sorted.1transcript.sorted.NOchr.gtf\"\n",
    "transcript_gene=pd.read_csv(\"/users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/human/newReference_Resconstructed/1transcript_1gene.reconstructed.csv\")"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Quantify with TCGAData and new reference\n",
    "`featureCounts_newRef_cancer.sh`\n",
    "\n",
    "output in: `/users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/cancers/featureCounts`"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## TPM"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PRAD\n",
      "KIRP\n",
      "LUAD\n"
     ]
    }
   ],
   "source": [
    "for file in os.listdir(os.path.join(cancer_dir,\"featureCounts\")):\n",
    "    if file.endswith(\".txt\"):\n",
    "\n",
    "        cancer_type = file[-8:-4]\n",
    "        print(cancer_type)\n",
    "        toc = pd.read_csv(os.path.join(cancer_dir,\"featureCounts\",file), sep=\"\\t\", comment=\"#\")\n",
    "        toc['Geneid']=toc['Geneid'].str.split('.').str[0]\n",
    "        toc.rename(columns={'Geneid':'transcript_id'}, inplace=True)\n",
    "\n",
    "        filter_col = [col for col in toc if col.startswith('/')]\n",
    "        for col in filter_col:\n",
    "            new_col=col.split(\"Aligned\")[0]\n",
    "            new_col=new_col.split(\"/\")[-1]\n",
    "            new_col=cancer_type+new_col[4:]\n",
    "            toc.rename(columns={col:new_col}, inplace=True)\n",
    "\n",
    "        length = toc['Length']\n",
    "        genes = toc['transcript_id']\n",
    "        # we are only interested in the columns with counts\n",
    "        counts = toc\n",
    "        counts.drop([\"Chr\",\"Start\",\"End\",\"Strand\",\"Length\",\"transcript_id\"],axis=1, inplace=True)\n",
    "        # calculate TPMs\n",
    "        tpm_df = Norm.tpm(counts, length)\n",
    "        # add transcript_id and length again\n",
    "        tpms = pd.concat([genes,tpm_df, length], axis=1)\n",
    "        tpms.to_csv(os.path.join(cancer_dir,\"featureCounts/table_of_counts_TPMs_\")+cancer_type+\".csv\", index=None)\n",
    " "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PRAD\n",
      "lncRNA:  7393\n",
      "PROTEIN CODING:  15071\n",
      "NOVEL:  608\n",
      "         transcript_id  PRAD-CH-5761_tumor  PRAD-CH-5767_tumor  \\\n",
      "0       TCONS_00002128          144.470157          200.289666   \n",
      "1      ENST00000387314           95.324684          178.278104   \n",
      "2      ENST00000389680         1150.468400         4236.862487   \n",
      "3       TCONS_00002129         2458.473628         7450.921126   \n",
      "4      ENST00000387342          427.346993          293.274179   \n",
      "...                ...                 ...                 ...   \n",
      "26675  ENST00000600468            0.644878            0.147132   \n",
      "26676  ENST00000425340            0.382016            0.193857   \n",
      "26677  ENST00000318083            0.713887            0.635445   \n",
      "26678  ENST00000222145            5.804824            2.531997   \n",
      "26679  ENST00000645652            4.301946            6.207849   \n",
      "\n",
      "       PRAD-CH-5768_tumor  PRAD-CH-5769_tumor  PRAD-EJ-7115_tumor  \\\n",
      "0              226.420020          328.384368          510.315150   \n",
      "1              149.290514           76.280892          344.365686   \n",
      "2              913.533314         1777.517729         1696.794911   \n",
      "3             1961.811550         3546.985440         3589.945905   \n",
      "4              115.557252          270.361100          189.732795   \n",
      "...                   ...                 ...                 ...   \n",
      "26675            0.469524            0.132294            0.165845   \n",
      "26676            0.781878            1.544985            1.241549   \n",
      "26677            0.885530            1.014828            1.315929   \n",
      "26678            1.839689            2.420379            5.359776   \n",
      "26679            8.062427            2.929491            4.765798   \n",
      "\n",
      "       PRAD-EJ-7123_tumor  PRAD-EJ-7125_tumor  PRAD-EJ-7314_tumor  \\\n",
      "0              212.564533          275.590330          177.273543   \n",
      "1               96.737820          601.127168          124.506769   \n",
      "2              722.571871         1470.611209         1291.628683   \n",
      "3             1559.115893         3123.811178         2791.235329   \n",
      "4              180.394145           87.385342          104.177094   \n",
      "...                   ...                 ...                 ...   \n",
      "26675            0.277338            0.097467            0.289995   \n",
      "26676            0.449843            0.177063            0.775502   \n",
      "26677            1.146718            1.505944            1.053629   \n",
      "26678            4.381717            5.753060            2.533875   \n",
      "26679           12.599556            9.969665            8.541884   \n",
      "\n",
      "       PRAD-EJ-7315_tumor  ...  PRAD-HC-8258_2_tumor  PRAD-HC-8258_tumor  \\\n",
      "0              442.886696  ...            456.656151          456.656151   \n",
      "1              119.660294  ...            296.230184          296.230184   \n",
      "2             1889.223970  ...           1868.877335         1868.877335   \n",
      "3             3472.747781  ...           4055.726184         4055.726184   \n",
      "4              107.526783  ...            173.199449          173.199449   \n",
      "...                   ...  ...                   ...                 ...   \n",
      "26675            0.215877  ...              0.709608            0.709608   \n",
      "26676            0.961757  ...              3.571063            3.571063   \n",
      "26677            0.782979  ...              1.785937            1.785937   \n",
      "26678            4.420258  ...              5.365798            5.365798   \n",
      "26679            6.302688  ...              7.150179            7.150179   \n",
      "\n",
      "       PRAD-HC-8259_tumor  PRAD-HC-8260_tumor  PRAD-HC-8262_tumor  \\\n",
      "0              141.418266          427.166337          385.353889   \n",
      "1              300.691762          260.440658          126.930630   \n",
      "2              905.415309         1281.616353         1920.597266   \n",
      "3             2022.448186         2783.852791         4218.043451   \n",
      "4               52.825665           86.183506          117.958119   \n",
      "...                   ...                 ...                 ...   \n",
      "26675            0.517655            1.517422            0.539755   \n",
      "26676            0.275238            2.190148            1.137102   \n",
      "26677            1.022099            1.691731            1.091689   \n",
      "26678            2.661814            4.328611            1.950337   \n",
      "26679            5.820918            5.718859            4.502809   \n",
      "\n",
      "       PRAD-J4-A83J_tumor  Length          gene_id       gene_type  \\\n",
      "0              260.385031     577      XLOC_001837           novel   \n",
      "1              140.969930      71  ENSG00000210049         Mt_tRNA   \n",
      "2             1128.233908     954  ENSG00000211459         Mt_rRNA   \n",
      "3             2500.076841     353      XLOC_001838           novel   \n",
      "4              173.539742      69  ENSG00000210077         Mt_tRNA   \n",
      "...                   ...     ...              ...             ...   \n",
      "26675            1.035167    2426  ENSG00000142233  protein_coding   \n",
      "26676            0.817622    3116  ENSG00000176920  protein_coding   \n",
      "26677            1.939285    1858  ENSG00000176909  protein_coding   \n",
      "26678            3.413180    3199  ENSG00000105538  protein_coding   \n",
      "26679            7.361463    3273  ENSG00000174951  protein_coding   \n",
      "\n",
      "         gene_name  \n",
      "0      XLOC_001837  \n",
      "1            MT-TF  \n",
      "2          MT-RNR1  \n",
      "3      XLOC_001838  \n",
      "4            MT-TV  \n",
      "...            ...  \n",
      "26675         NTN5  \n",
      "26676         FUT2  \n",
      "26677       MAMSTR  \n",
      "26678       RASIP1  \n",
      "26679         FUT1  \n",
      "\n",
      "[26680 rows x 59 columns]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_783139/2959889540.py:45: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  tumorONLY_merged['n'] = int_columns.apply(lambda row: (row > 1).sum(), axis=1)\n"
     ]
    }
   ],
   "source": [
    "for proj in ['TCGA-PRAD']:\n",
    "    cancer_type = proj[5:]\n",
    "    print(cancer_type)\n",
    "\n",
    "    fc = pd.read_csv(os.path.join(cancer_dir,\"featureCounts/table_of_counts_TPMs_\")+cancer_type+\".csv\")\n",
    "    tumor_transcripts = list()\n",
    "    patients=pd.read_csv(os.path.join(\"/users/genomics/marta/TCGA_RNASeq\",proj,\"results/QC_patients.csv\"))\n",
    "    patients['normal'] = patients['normal'].str.replace('TCGA', cancer_type)\n",
    "    patients['tumor'] = patients['tumor'].str.replace('TCGA', cancer_type)\n",
    "\n",
    "    for index,patient in patients.iterrows():\n",
    "        patient_fc = fc[[\"transcript_id\",\"Length\",patient.iloc[2]]]\n",
    "\n",
    "        tumor_patient_fc = patient_fc[patient_fc[patient.iloc[2]] >= 1 ]\n",
    "        tumor_transcripts.extend(tumor_patient_fc.transcript_id.values.tolist())\n",
    "\n",
    "    tumor1FPKM = fc[fc['transcript_id'].isin(tumor_transcripts)]\n",
    "\n",
    "    known = tumor1FPKM[tumor1FPKM['transcript_id'].str.contains('ENST')]\n",
    "    novel = tumor1FPKM[tumor1FPKM['transcript_id'].str.contains('TCONS')]\n",
    "    merged = tumor1FPKM.merge(transcript_gene, on=['transcript_id'], how=\"inner\")\n",
    "\n",
    "    lncRNA = merged[merged['gene_type'].isin(['lncRNA','processed_pseudogene'])]\n",
    "    print(\"lncRNA: \",len(lncRNA))\n",
    "\n",
    "    cds = merged[merged['gene_type'] == \"protein_coding\"]\n",
    "    print(\"PROTEIN CODING: \",len(cds))\n",
    "\n",
    "    novel = merged[merged['gene_type'] == \"novel\"]\n",
    "    novel = novel[novel['Length'].astype(int) < 91667]\n",
    "    print(\"NOVEL: \",len(novel))\n",
    "\n",
    "    # tumor1FPKM = pd.concat([novel, lncRNA, cds], ignore_index = True)\n",
    "    merged.to_csv(os.path.join(cancer_dir,\"tumorexpressed/tumor_1FPKM_table_of_counts_\")+cancer_type+\".csv\",index=False)\n",
    "    merged[['transcript_id','gene_id','gene_name','gene_type','Length']].to_csv(os.path.join(cancer_dir,\"tumorexpressed/tumor_1FPKM_\")+cancer_type+\".csv\",index=False)\n",
    "\n",
    "    tumorONLY_merged = merged[merged.columns.drop(list(merged.filter(regex='normal')))]\n",
    "    print(tumorONLY_merged)\n",
    "\n",
    "    ###### GET TUMOR-EXPRESSED > 1 TPM\n",
    "    # Filter only integer columns\n",
    "    int_columns = tumorONLY_merged.select_dtypes(include='int64')\n",
    "\n",
    "    # Create a new column with the count of int columns per row whose value is > 1\n",
    "    tumorONLY_merged['n'] = int_columns.apply(lambda row: (row > 1).sum(), axis=1)\n",
    "    tumorONLY_merged = tumorONLY_merged[['transcript_id','gene_id','gene_name','gene_type','Length','n']]\n",
    "    tumorONLY_merged.sort_values(by=['n'], ascending=False).to_csv(os.path.join(cancer_dir,\"tumorexpressed/tumor_1FPKM_n_\")+cancer_type+\".csv\",index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>transcript_id</th>\n",
       "      <th>gene_id</th>\n",
       "      <th>gene_name</th>\n",
       "      <th>gene_type</th>\n",
       "      <th>Length</th>\n",
       "      <th>n</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>TCONS_00002128</td>\n",
       "      <td>XLOC_001837</td>\n",
       "      <td>XLOC_001837</td>\n",
       "      <td>novel</td>\n",
       "      <td>577</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ENST00000387314</td>\n",
       "      <td>ENSG00000210049</td>\n",
       "      <td>MT-TF</td>\n",
       "      <td>Mt_tRNA</td>\n",
       "      <td>71</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ENST00000389680</td>\n",
       "      <td>ENSG00000211459</td>\n",
       "      <td>MT-RNR1</td>\n",
       "      <td>Mt_rRNA</td>\n",
       "      <td>954</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>TCONS_00002129</td>\n",
       "      <td>XLOC_001838</td>\n",
       "      <td>XLOC_001838</td>\n",
       "      <td>novel</td>\n",
       "      <td>353</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ENST00000387342</td>\n",
       "      <td>ENSG00000210077</td>\n",
       "      <td>MT-TV</td>\n",
       "      <td>Mt_tRNA</td>\n",
       "      <td>69</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26675</th>\n",
       "      <td>ENST00000600468</td>\n",
       "      <td>ENSG00000142233</td>\n",
       "      <td>NTN5</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>2426</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26676</th>\n",
       "      <td>ENST00000425340</td>\n",
       "      <td>ENSG00000176920</td>\n",
       "      <td>FUT2</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>3116</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26677</th>\n",
       "      <td>ENST00000318083</td>\n",
       "      <td>ENSG00000176909</td>\n",
       "      <td>MAMSTR</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>1858</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26678</th>\n",
       "      <td>ENST00000222145</td>\n",
       "      <td>ENSG00000105538</td>\n",
       "      <td>RASIP1</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>3199</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26679</th>\n",
       "      <td>ENST00000645652</td>\n",
       "      <td>ENSG00000174951</td>\n",
       "      <td>FUT1</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>3273</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>26680 rows Ã— 6 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "         transcript_id          gene_id    gene_name       gene_type  Length  \\\n",
       "0       TCONS_00002128      XLOC_001837  XLOC_001837           novel     577   \n",
       "1      ENST00000387314  ENSG00000210049        MT-TF         Mt_tRNA      71   \n",
       "2      ENST00000389680  ENSG00000211459      MT-RNR1         Mt_rRNA     954   \n",
       "3       TCONS_00002129      XLOC_001838  XLOC_001838           novel     353   \n",
       "4      ENST00000387342  ENSG00000210077        MT-TV         Mt_tRNA      69   \n",
       "...                ...              ...          ...             ...     ...   \n",
       "26675  ENST00000600468  ENSG00000142233         NTN5  protein_coding    2426   \n",
       "26676  ENST00000425340  ENSG00000176920         FUT2  protein_coding    3116   \n",
       "26677  ENST00000318083  ENSG00000176909       MAMSTR  protein_coding    1858   \n",
       "26678  ENST00000222145  ENSG00000105538       RASIP1  protein_coding    3199   \n",
       "26679  ENST00000645652  ENSG00000174951         FUT1  protein_coding    3273   \n",
       "\n",
       "       n  \n",
       "0      1  \n",
       "1      1  \n",
       "2      1  \n",
       "3      1  \n",
       "4      1  \n",
       "...   ..  \n",
       "26675  1  \n",
       "26676  1  \n",
       "26677  1  \n",
       "26678  1  \n",
       "26679  1  \n",
       "\n",
       "[26680 rows x 6 columns]"
      ]
     },
     "execution_count": 41,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "## TUMOR-SPECIFIC"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.16"
  },
  "orig_nbformat": 4
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
