{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Let's predict the stability of ncORFs\n",
    "\n",
    "- pepScore"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "metadata": {},
   "outputs": [],
   "source": [
    "import os\n",
    "import pandas as pd\n",
    "from Bio import SeqIO\n",
    "from Bio.Seq import Seq  # Import Seq from Bio.Seq\n",
    "\n",
    "pepDir = \"/data/genomics/marta/tools/PepScore\"\n",
    "riboDir = \"/users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction\""
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "1. Calculate the false discovery rate (FDR) of ORF lengths considerating transcript lengths. \n",
    "\n",
    "`usage: perl CalculateFDR.ORFlength.pl -g SequenceFile -t lengthsOfInterest -o outputDir -s startCodon\n",
    "\n",
    "\t-g SequenceFile: input the random sequence file in the fasta format\n",
    "\n",
    "\t-t lengthsOfInterest: Tab delimited text file with two columns. The first column shows the transcript length, and the second shows the ORF length (e.g. 1000\t36)\n",
    "\t\n",
    "\t-o outputDir: output director\n",
    "\t\n",
    "\t-r repeat [optional]: the number of times generating random transcript sequences, default: 1000\n",
    "\t\n",
    "\t-s startCodon [optional]: start codon types, default: ATG\n",
    "\t\n",
    "\t-l orfLengthCutoff [optional]: cutoff of minimum candidate ORF length, default: 6`\n",
    "\n",
    "Example command line: perl CalculateFDR.ORFlength.pl -g rand.genome.fa -t transcript.length.txt -o outputDir"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>header</th>\n",
       "      <th>seq</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ENST00000431238.7:X:+|1|259:15:192|noncoding|CTG</td>\n",
       "      <td>CTGTACCGGCTGCATCCGGAGCAGGGCATGCCGGCCGGCGTGTGCG...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ENST00000431238.7:X:+|2|259:24:192|noncoding|CTG</td>\n",
       "      <td>CTGCATCCGGAGCAGGGCATGCCGGCCGGCGTGTGCGTGGACGCTG...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ENST00000431238.7:X:+|3|259:42:192|noncoding|ATG</td>\n",
       "      <td>ATGCCGGCCGGCGTGTGCGTGGACGCTGCGGGGAAGCTCTGGGTGG...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ENST00000431238.7:X:+|4|259:54:192|noncoding|GTG</td>\n",
       "      <td>GTGTGCGTGGACGCTGCGGGGAAGCTCTGGGTGGCCTCTGCGTCGA...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ENST00000431238.7:X:+|5|259:60:192|noncoding|GTG</td>\n",
       "      <td>GTGGACGCTGCGGGGAAGCTCTGGGTGGCCTCTGCGTCGATAGAGG...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8233062</th>\n",
       "      <td>ENST00000612925.1:KI270750.1:+|2|176:73:88|non...</td>\n",
       "      <td>CTGGATCCATCATAG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8233063</th>\n",
       "      <td>ENST00000612925.1:KI270750.1:+|3|176:102:150|n...</td>\n",
       "      <td>CTGGAATTGCCCCAAATGTGGGAAGCTCTACTGCAAAATTTTTGGTAG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8233064</th>\n",
       "      <td>ENST00000612925.1:KI270750.1:+|4|176:108:150|n...</td>\n",
       "      <td>TTGCCCCAAATGTGGGAAGCTCTACTGCAAAATTTTTGGTAG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8233065</th>\n",
       "      <td>ENST00000612925.1:KI270750.1:+|5|176:117:150|n...</td>\n",
       "      <td>ATGTGGGAAGCTCTACTGCAAAATTTTTGGTAG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8233066</th>\n",
       "      <td>ENST00000612925.1:KI270750.1:+|6|176:132:150|n...</td>\n",
       "      <td>CTGCAAAATTTTTGGTAG</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>8233067 rows Ã— 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                    header  \\\n",
       "0         ENST00000431238.7:X:+|1|259:15:192|noncoding|CTG   \n",
       "1         ENST00000431238.7:X:+|2|259:24:192|noncoding|CTG   \n",
       "2         ENST00000431238.7:X:+|3|259:42:192|noncoding|ATG   \n",
       "3         ENST00000431238.7:X:+|4|259:54:192|noncoding|GTG   \n",
       "4         ENST00000431238.7:X:+|5|259:60:192|noncoding|GTG   \n",
       "...                                                    ...   \n",
       "8233062  ENST00000612925.1:KI270750.1:+|2|176:73:88|non...   \n",
       "8233063  ENST00000612925.1:KI270750.1:+|3|176:102:150|n...   \n",
       "8233064  ENST00000612925.1:KI270750.1:+|4|176:108:150|n...   \n",
       "8233065  ENST00000612925.1:KI270750.1:+|5|176:117:150|n...   \n",
       "8233066  ENST00000612925.1:KI270750.1:+|6|176:132:150|n...   \n",
       "\n",
       "                                                       seq  \n",
       "0        CTGTACCGGCTGCATCCGGAGCAGGGCATGCCGGCCGGCGTGTGCG...  \n",
       "1        CTGCATCCGGAGCAGGGCATGCCGGCCGGCGTGTGCGTGGACGCTG...  \n",
       "2        ATGCCGGCCGGCGTGTGCGTGGACGCTGCGGGGAAGCTCTGGGTGG...  \n",
       "3        GTGTGCGTGGACGCTGCGGGGAAGCTCTGGGTGGCCTCTGCGTCGA...  \n",
       "4        GTGGACGCTGCGGGGAAGCTCTGGGTGGCCTCTGCGTCGATAGAGG...  \n",
       "...                                                    ...  \n",
       "8233062                                    CTGGATCCATCATAG  \n",
       "8233063   CTGGAATTGCCCCAAATGTGGGAAGCTCTACTGCAAAATTTTTGGTAG  \n",
       "8233064         TTGCCCCAAATGTGGGAAGCTCTACTGCAAAATTTTTGGTAG  \n",
       "8233065                  ATGTGGGAAGCTCTACTGCAAAATTTTTGGTAG  \n",
       "8233066                                 CTGCAAAATTTTTGGTAG  \n",
       "\n",
       "[8233067 rows x 2 columns]"
      ]
     },
     "execution_count": 35,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "## Create fasta of translated sequences from candidates\n",
    "specie = \"human\"\n",
    "fasta=os.path.join(\"/users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction\",specie,\"240708_RiboNovel/Annotation/candidateORF.fa\")\n",
    "fasta_df_input = pd.read_csv(fasta, sep=\"\\t\", header=None)\n",
    "fasta_df = pd.DataFrame({'header':fasta_df_input[0].iloc[::2].values, 'seq':fasta_df_input[0].iloc[1::2].values})\n",
    "fasta_df['header'] = fasta_df.header.str[1:]\n",
    "fasta_df\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "human_brain_ribo_3_r1\n",
      "human_testis_ribo_3_r1\n",
      "human_liver_ribo_1_r1\n",
      "human_brain_ribo_1_r1\n",
      "human_testis_ribo_2_r1\n",
      "human_testis_ribo_1_r1\n",
      "human_liver_ribo_3_r1\n",
      "human_liver_ribo_2_r1\n",
      "human_brain_ribo_2_r1\n"
     ]
    }
   ],
   "source": [
    "dict_translated = dict()\n",
    "dict_translated_prot = dict()\n",
    "\n",
    "for infolder_results in os.listdir(os.path.join(riboDir,specie,\"240724_RiboNovel/RibORF\")):\n",
    "    print(infolder_results)\n",
    "    outfile=os.path.join(riboDir,specie,\"240724_RiboNovel/RibORF\",infolder_results,\"repre.valid.pred.pvalue.parameters.fa\")\n",
    "    outfile_prot=os.path.join(riboDir,specie,\"240724_RiboNovel/RibORF\",infolder_results,\"repre.valid.pred.pvalue.parameters.PROTEIN.fa\")\n",
    "\n",
    "    seqs_df = pd.read_csv(os.path.join(riboDir,specie,\"240724_RiboNovel/RibORF\",infolder_results,\"repre.valid.pred.pvalue.parameters.txt\"), sep=\"\\t\")\n",
    "    candidates_in_repre = fasta_df[fasta_df['header'].isin(seqs_df.orfID.values.tolist())]\n",
    "\n",
    "    with open(outfile, 'w') as out:\n",
    "        with open(outfile_prot, 'w') as out_prot:\n",
    "            for index, row in candidates_in_repre.iterrows():\n",
    "                    out.write(\">%s\\n%s\\n\" %(row['header'], row['seq']))\n",
    "                    ## add to non-redundant dictionary\n",
    "                    dict_translated[row['header']] = row['seq']\n",
    "\n",
    "                    ## translate to proteins\n",
    "                    out_prot.write(\">%s\\n%s\\n\" %(row['header'], str(Seq(row['seq']).translate())))\n",
    "                    ## add to non-redundant dictionary\n",
    "                    dict_translated_prot[row['header']] = str(Seq(row['seq']).translate())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "metadata": {},
   "outputs": [],
   "source": [
    "outfile_prot = os.path.join(riboDir,specie,\"240724_RiboNovel/RibORF/repre.valid.pred.pvalue.parameters.allsamples.noredundant.PROTEIN.fa\")\n",
    "outfile = os.path.join(riboDir,specie,\"240724_RiboNovel/RibORF/repre.valid.pred.pvalue.parameters.allsamples.noredundant.fa\")\n",
    "\n",
    "with open(outfile, 'w') as out:\n",
    "    with open(outfile_prot, 'w') as out_prot:\n",
    "        for key, value in dict_translated.items():\n",
    "            out.write(\">%s\\n%s\\n\" %(key, value))\n",
    "\n",
    "        ## proteins\n",
    "        for key, value in dict_translated_prot.items():\n",
    "            out_prot.write(\">%s\\n%s\\n\" %(key, value))\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 68,
   "metadata": {},
   "outputs": [],
   "source": [
    "## get transcript Length\n",
    "## Tab delimited text file with two columns. The first column shows the transcript length, and the second shows the ORF length (e.g. 1000\t36)\n",
    "df_translated = pd.DataFrame(dict_translated.items(), columns=['header', 'seq'])\n",
    "## get transcript length\n",
    "df_translated['transcriptLength'] = df_translated['header'].str.split(\"\\|\", expand=True)[2]\n",
    "df_translated['transcriptLength'] = df_translated['transcriptLength'].str.split(\":\", expand=True)[0]\n",
    "## get orf length\n",
    "df_translated['orfLength_info'] = df_translated['header'].str.split(\"\\|\", expand=True)[2]\n",
    "df_translated['endORF'] = df_translated['orfLength_info'].str.split(\":\", expand=True)[2]\n",
    "df_translated['startORF'] = df_translated['orfLength_info'].str.split(\":\", expand=True)[1]\n",
    "df_translated['orfLength'] = df_translated['endORF'].astype('int64') - df_translated['startORF'].astype('int64')\n",
    "df_translated\n",
    "\n",
    "transcript_length = df_translated[['transcriptLength', 'orfLength']]\n",
    "transcript_length.to_csv(os.path.join(riboDir,specie,\"240724_RiboNovel/RibORF/length_transcript_orf.allsamples.noredundant.txt\"), header=None, sep=\"\\t\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 70,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash -s \"$pepDir\" \"$riboDir\"\n",
    "\n",
    "specie=\"human\"\n",
    "\n",
    "module load Perl\n",
    "module load R/4.2.1-foss-2020b\n",
    "\n",
    "mkdir -p $2/$specie/stability/PepScore\n",
    "\n",
    "perl $1/CalculateFDR.ORFlength.pl -g $2/$specie/240724_RiboNovel/RibORF/repre.valid.pred.pvalue.parameters.allsamples.noredundant.fa -t $2/$specie/240724_RiboNovel/RibORF/length_transcript_orf.allsamples.noredundant.txt -o $2/$specie/stability/PepScore"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.16"
  },
  "orig_nbformat": 4
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
