{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "import os,re,glob\n",
    "import pandas as pd\n",
    "import numpy as np\n",
    "from collections import Counter\n",
    "\n",
    "users_dir = \"/users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction\"\n",
    "specie = \"mouse\"\n",
    "GENOMEDIR = \"/genomics/users/marta/genomes\"\n"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Bams are processed by Sara --> look for the scripts\n",
    "\n",
    "Outputs are in:\n",
    "\n",
    "`/users/genomics/saraa/projectTestis/STAR/{human/macaca/mouse}`"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Bam2Fastq + QC + STAR shiva\n",
    "\n",
    "/users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/macaca/STAR for macaca\n",
    "\n",
    "/datasets/marta/Wang2020/RNASeq/mouse* for mouse\n"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "\n",
    "## Stranded?\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Reading reference gene model /datasets/Common/ReferenceGenomes/Mouse/mm39/GenomesAndAnnotation/gencode.vM32.annotation.bed ... Done\n",
      "Loading SAM/BAM file ...  Finished\n",
      "Total 45866915 usable reads were sampled\n",
      "Reading reference gene model /datasets/Common/ReferenceGenomes/Mouse/mm39/GenomesAndAnnotation/gencode.vM32.annotation.bed ... Done\n",
      "Loading SAM/BAM file ...  Finished\n",
      "Total 39585215 usable reads were sampled\n",
      "Reading reference gene model /datasets/Common/ReferenceGenomes/Mouse/mm39/GenomesAndAnnotation/gencode.vM32.annotation.bed ... Done\n",
      "Loading SAM/BAM file ...  Finished\n",
      "Total 33823384 usable reads were sampled\n",
      "Reading reference gene model /datasets/Common/ReferenceGenomes/Mouse/mm39/GenomesAndAnnotation/gencode.vM32.annotation.bed ... Done\n",
      "Loading SAM/BAM file ...  Finished\n",
      "Total 52181407 usable reads were sampled\n",
      "Reading reference gene model /datasets/Common/ReferenceGenomes/Mouse/mm39/GenomesAndAnnotation/gencode.vM32.annotation.bed ... Done\n",
      "Loading SAM/BAM file ...  Finished\n",
      "Total 40253662 usable reads were sampled\n",
      "Reading reference gene model /datasets/Common/ReferenceGenomes/Mouse/mm39/GenomesAndAnnotation/gencode.vM32.annotation.bed ... Done\n",
      "Loading SAM/BAM file ...  Finished\n",
      "Total 26951506 usable reads were sampled\n",
      "Reading reference gene model /datasets/Common/ReferenceGenomes/Mouse/mm39/GenomesAndAnnotation/gencode.vM32.annotation.bed ... Done\n",
      "Loading SAM/BAM file ...  Finished\n",
      "Total 45866915 usable reads were sampled\n",
      "Reading reference gene model /datasets/Common/ReferenceGenomes/Mouse/mm39/GenomesAndAnnotation/gencode.vM32.annotation.bed ... Done\n",
      "Loading SAM/BAM file ...  Finished\n",
      "Total 38587530 usable reads were sampled\n",
      "Loading SAM/BAM file ...  Finished\n",
      "Total 33823384 usable reads were sampled\n",
      "Reading reference gene model /datasets/Common/ReferenceGenomes/Mouse/mm39/GenomesAndAnnotation/gencode.vM32.annotation.bed ... Done\n",
      "Reading reference gene model /datasets/Common/ReferenceGenomes/Mouse/mm39/GenomesAndAnnotation/gencode.vM32.annotation.bed ... Done\n",
      "Loading SAM/BAM file ...  Finished\n",
      "Total 40253662 usable reads were sampled\n",
      "Loading SAM/BAM file ...  Finished\n",
      "Total 42182017 usable reads were sampled\n",
      "Reading reference gene model /datasets/Common/ReferenceGenomes/Mouse/mm39/GenomesAndAnnotation/gencode.vM32.annotation.bed ... Done\n",
      "Reading reference gene model /datasets/Common/ReferenceGenomes/Mouse/mm39/GenomesAndAnnotation/gencode.vM32.annotation.bed ... Done\n",
      "Loading SAM/BAM file ...  Finished\n",
      "Total 26951506 usable reads were sampled\n",
      "Reading reference gene model /datasets/Common/ReferenceGenomes/Mouse/mm39/GenomesAndAnnotation/gencode.vM32.annotation.bed ... Done\n",
      "Loading SAM/BAM file ...  Finished\n",
      "Total 32756458 usable reads were sampled\n",
      "Reading reference gene model /datasets/Common/ReferenceGenomes/Mouse/mm39/GenomesAndAnnotation/gencode.vM32.annotation.bed ... Done\n",
      "Loading SAM/BAM file ...  Finished\n",
      "Total 38587530 usable reads were sampled\n",
      "Loading SAM/BAM file ...  Finished\n",
      "Total 31825603 usable reads were sampled\n",
      "Reading reference gene model /datasets/Common/ReferenceGenomes/Mouse/mm39/GenomesAndAnnotation/gencode.vM32.annotation.bed ... Done\n",
      "Reading reference gene model /datasets/Common/ReferenceGenomes/Mouse/mm39/GenomesAndAnnotation/gencode.vM32.annotation.bed ... Done\n",
      "Loading SAM/BAM file ...  Finished\n",
      "Total 42182017 usable reads were sampled\n",
      "Reading reference gene model /datasets/Common/ReferenceGenomes/Mouse/mm39/GenomesAndAnnotation/gencode.vM32.annotation.bed ... Done\n",
      "Loading SAM/BAM file ...  Finished\n",
      "Total 67176463 usable reads were sampled\n",
      "Reading reference gene model /datasets/Common/ReferenceGenomes/Mouse/mm39/GenomesAndAnnotation/gencode.vM32.annotation.bed ... Done\n",
      "Loading SAM/BAM file ...  Finished\n",
      "Total 32756458 usable reads were sampled\n",
      "Reading reference gene model /datasets/Common/ReferenceGenomes/Mouse/mm39/GenomesAndAnnotation/gencode.vM32.annotation.bed ... Done\n",
      "Loading SAM/BAM file ...  Finished\n",
      "Total 31825603 usable reads were sampled\n",
      "Reading reference gene model /datasets/Common/ReferenceGenomes/Mouse/mm39/GenomesAndAnnotation/gencode.vM32.annotation.bed ... Done\n",
      "Loading SAM/BAM file ...  Finished\n",
      "Total 46168112 usable reads were sampled\n",
      "Reading reference gene model /datasets/Common/ReferenceGenomes/Mouse/mm39/GenomesAndAnnotation/gencode.vM32.annotation.bed ... Done\n",
      "Loading SAM/BAM file ...  Finished\n",
      "Total 67176463 usable reads were sampled\n",
      "Reading reference gene model /datasets/Common/ReferenceGenomes/Mouse/mm39/GenomesAndAnnotation/gencode.vM32.annotation.bed ... Done\n",
      "Loading SAM/BAM file ...  Finished\n",
      "Total 63378036 usable reads were sampled\n",
      "Reading reference gene model /datasets/Common/ReferenceGenomes/Mouse/mm39/GenomesAndAnnotation/gencode.vM32.annotation.bed ... Done\n",
      "Loading SAM/BAM file ...  Finished\n",
      "Total 46168112 usable reads were sampled\n",
      "Reading reference gene model /datasets/Common/ReferenceGenomes/Mouse/mm39/GenomesAndAnnotation/gencode.vM32.annotation.bed ... Done\n",
      "Loading SAM/BAM file ...  Finished\n",
      "Total 44852410 usable reads were sampled\n",
      "Reading reference gene model /datasets/Common/ReferenceGenomes/Mouse/mm39/GenomesAndAnnotation/gencode.vM32.annotation.bed ... Done\n",
      "Loading SAM/BAM file ...  Finished\n",
      "Total 27547567 usable reads were sampled\n",
      "Reading reference gene model /datasets/Common/ReferenceGenomes/Mouse/mm39/GenomesAndAnnotation/gencode.vM32.annotation.bed ... Done\n",
      "Loading SAM/BAM file ...  Finished\n",
      "Total 63378036 usable reads were sampled\n",
      "Loading SAM/BAM file ...  Finished\n",
      "Total 33685861 usable reads were sampled\n",
      "Reading reference gene model /datasets/Common/ReferenceGenomes/Mouse/mm39/GenomesAndAnnotation/gencode.vM32.annotation.bed ... Done\n",
      "Reading reference gene model /datasets/Common/ReferenceGenomes/Mouse/mm39/GenomesAndAnnotation/gencode.vM32.annotation.bed ... Done\n",
      "Loading SAM/BAM file ...  Finished\n",
      "Total 44852410 usable reads were sampled\n",
      "Loading SAM/BAM file ...  Finished\n",
      "Total 40974879 usable reads were sampled\n",
      "Reading reference gene model /datasets/Common/ReferenceGenomes/Mouse/mm39/GenomesAndAnnotation/gencode.vM32.annotation.bed ... Done\n",
      "Reading reference gene model /datasets/Common/ReferenceGenomes/Mouse/mm39/GenomesAndAnnotation/gencode.vM32.annotation.bed ... Done\n",
      "Loading SAM/BAM file ...  Finished\n",
      "Total 27547567 usable reads were sampled\n",
      "Loading SAM/BAM file ...  Finished\n",
      "Total 28205046 usable reads were sampled\n",
      "Reading reference gene model /datasets/Common/ReferenceGenomes/Mouse/mm39/GenomesAndAnnotation/gencode.vM32.annotation.bed ... Done\n",
      "Reading reference gene model /datasets/Common/ReferenceGenomes/Mouse/mm39/GenomesAndAnnotation/gencode.vM32.annotation.bed ... Done\n",
      "Loading SAM/BAM file ...  Finished\n",
      "Total 33685861 usable reads were sampled\n",
      "Reading reference gene model /datasets/Common/ReferenceGenomes/Mouse/mm39/GenomesAndAnnotation/gencode.vM32.annotation.bed ... Done\n",
      "Loading SAM/BAM file ...  Finished\n",
      "Total 68090404 usable reads were sampled\n",
      "Loading SAM/BAM file ...  Finished\n",
      "Total 40974879 usable reads were sampled\n",
      "Reading reference gene model /datasets/Common/ReferenceGenomes/Mouse/mm39/GenomesAndAnnotation/gencode.vM32.annotation.bed ... Done\n",
      "Loading SAM/BAM file ...  Finished\n",
      "Total 28205046 usable reads were sampled\n",
      "Reading reference gene model /datasets/Common/ReferenceGenomes/Mouse/mm39/GenomesAndAnnotation/gencode.vM32.annotation.bed ... Done\n",
      "Loading SAM/BAM file ...  Finished\n",
      "Total 68090404 usable reads were sampled\n"
     ]
    }
   ],
   "source": [
    "%%bash -s \"$users_dir\" \"$GENOMEDIR\" \"$specie\"\n",
    "\n",
    "# BAMDIR=/users/genomics/saraa/projectTestis/STAR/$3\n",
    "BAMDIR=$1/$3/STAR\n",
    "\n",
    "OUTDIR=$1/$3/RSeQC\n",
    "mkdir -p $OUTDIR\n",
    "\n",
    "if [ \"$3\" == \"human\" ]; then\n",
    "    reference_annot=$2/Annot_files_GTF/gencode.v38.primary_assembly.annotation_gene.bed\n",
    "elif [ \"$3\" == \"macaca\" ]; then\n",
    "    reference_annot=/datasets/Common/ReferenceGenomes/Macaca/10.111/GenomesAndAnnotation/Macaca_mulatta.Mmul_10.111.bed\n",
    "elif [ \"$3\" == \"mouse\" ]; then\n",
    "    reference_annot=/datasets/Common/ReferenceGenomes/Mouse/mm39/GenomesAndAnnotation/gencode.vM32.annotation.bed\n",
    "fi\n",
    "\n",
    "for file in $BAMDIR/*Log.final.out; do\n",
    "    name=${file%%Log.*}\n",
    "    name=${name##*/}\n",
    "\n",
    "    ###################### get num of uniquely aligned reads #######################\n",
    "    uniq_mapped_reads=$(sed '9q;d' ${BAMDIR}/${name}Log.final.out | awk '{print $6}')\n",
    "    # echo $uniq_mapped_reads\n",
    "    ############################## RSEQC ###########################################\n",
    "    module load Miniconda3/4.9.2\n",
    "\n",
    "    infer_experiment.py -r /datasets/Common/ReferenceGenomes/Mouse/mm39/GenomesAndAnnotation/gencode.vM32.annotation.bed -i ${BAMDIR}/${name}Aligned.sortedByCoord.out.bam -s $uniq_mapped_reads > $OUTDIR/${name}.out\n",
    "\n",
    "done\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash -s \"$users_dir\" \"$specie\"\n",
    "\n",
    "OUTDIR=$1/$2/RSeQC\n",
    "\n",
    "# output for stringtie\n",
    "\n",
    "if [[ -f $OUTDIR/summary_stringtie.csv ]]; then\n",
    "    rm $OUTDIR/summary_stringtie.csv\n",
    "fi\n",
    "\n",
    "for file in $OUTDIR/*out; do\n",
    "    sample=${file##*/}\n",
    "    sample=${sample%%.out*}\n",
    "    echo -e $sample\"\\t\"$(python ~/Documents/scripts/Chris_decide_strandness_parameter.py -tab $file -tool stringtie) >> $OUTDIR/summary_stringtie.csv\n",
    "done\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash -s \"$users_dir\" \"$specie\"\n",
    "\n",
    "OUTDIR=$1/$2/RSeQC\n",
    "\n",
    "# output for featureCounts\n",
    "\n",
    "if [[ -f $OUTDIR/summary_featureCounts.csv ]]; then\n",
    "    rm $OUTDIR/summary_featureCounts.csv\n",
    "fi\n",
    "\n",
    "for file in $OUTDIR/*out; do\n",
    "    sample=${file##*/}\n",
    "    sample=${sample%%.out*}\n",
    "    echo -e $sample\"\\t\"$(python ~/Documents/scripts/Chris_decide_strandness_parameter.py -tab $file -tool featureCounts) >> $OUTDIR/summary_featureCounts.csv\n",
    "done\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Error creating output file /*bam_cov.gtf\n"
     ]
    },
    {
     "ename": "CalledProcessError",
     "evalue": "Command 'b'\\n## ------------------------------ done in shiva ---------------------------##\\nmodule load stringtie/2.2.0\\n\\nspecie=$2\\ndeclare -a bam_array\\n\\nif [ \"$specie\" == \"macaca\" ]; then\\n  BAMDIR=$1/$specie/STAR\\n  REF=/datasets/Common/ReferenceGenomes/Macaca/10.111/GenomesAndAnnotation/Macaca_mulatta.Mmul_10.111.gtf\\n\\n  for bam in $BAMDIR/*bam; do\\n      sample=${bam%%Aligned*}\\n      sample=${sample##*/}\\n\\n    ## in this case, we assume it\\'s always fr\\n    stringtie $bam -G $REF --conservative --fr -o $OUTDIR/${sample}.gtf -C $OUTDIR/${sample}_cov.gtf\\n  done\\n\\nelif [ \"$specie\" == \"mouse\" ]; then\\n  # BAMDIR=/datasets/marta/Wang2020/RNASeq\\n  BAMDIR=$1/$specie/STAR\\n\\n  REF=/datasets/Common/ReferenceGenomes/Mouse/mm39/GenomesAndAnnotation/gencode.vM32.annotation.gtf\\n\\n  for bam in $BAMDIR/*bam; do\\n      sample=${bam%%Aligned*}\\n      sample=${sample##*/}\\n\\n    ## in this case, we assume it\\'s always fr\\n    stringtie $bam -G $REF --conservative --fr -o $OUTDIR/${sample}.gtf -C $OUTDIR/${sample}_cov.gtf\\n  done\\n\\nfi\\n\\n'' returned non-zero exit status 1.",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mCalledProcessError\u001b[0m                        Traceback (most recent call last)",
      "\u001b[1;32m/home/marta/PROJECT_SCRIPTS/TestisRestricted_MP/transcriptome_reconstruction.ipynb Cell 8\u001b[0m in \u001b[0;36m<cell line: 1>\u001b[0;34m()\u001b[0m\n\u001b[0;32m----> <a href='vscode-notebook-cell:/home/marta/PROJECT_SCRIPTS/TestisRestricted_MP/transcriptome_reconstruction.ipynb#X10sZmlsZQ%3D%3D?line=0'>1</a>\u001b[0m get_ipython()\u001b[39m.\u001b[39;49mrun_cell_magic(\u001b[39m'\u001b[39;49m\u001b[39mbash\u001b[39;49m\u001b[39m'\u001b[39;49m, \u001b[39m'\u001b[39;49m\u001b[39m-s \u001b[39;49m\u001b[39m\"\u001b[39;49m\u001b[39m$users_dir\u001b[39;49m\u001b[39m\"\u001b[39;49m\u001b[39m \u001b[39;49m\u001b[39m\"\u001b[39;49m\u001b[39m$specie\u001b[39;49m\u001b[39m\"\u001b[39;49m\u001b[39m \u001b[39;49m\u001b[39m\"\u001b[39;49m\u001b[39m$GENOMEDIR\u001b[39;49m\u001b[39m\"\u001b[39;49m\u001b[39m'\u001b[39;49m, \u001b[39m'\u001b[39;49m\u001b[39m\\n\u001b[39;49;00m\u001b[39m## ------------------------------ done in shiva ---------------------------##\u001b[39;49m\u001b[39m\\n\u001b[39;49;00m\u001b[39mmodule load stringtie/2.2.0\u001b[39;49m\u001b[39m\\n\u001b[39;49;00m\u001b[39m\\n\u001b[39;49;00m\u001b[39mspecie=$2\u001b[39;49m\u001b[39m\\n\u001b[39;49;00m\u001b[39mdeclare -a bam_array\u001b[39;49m\u001b[39m\\n\u001b[39;49;00m\u001b[39m\\n\u001b[39;49;00m\u001b[39mif [ \u001b[39;49m\u001b[39m\"\u001b[39;49m\u001b[39m$specie\u001b[39;49m\u001b[39m\"\u001b[39;49m\u001b[39m == \u001b[39;49m\u001b[39m\"\u001b[39;49m\u001b[39mmacaca\u001b[39;49m\u001b[39m\"\u001b[39;49m\u001b[39m ]; then\u001b[39;49m\u001b[39m\\n\u001b[39;49;00m\u001b[39m  BAMDIR=$1/$specie/STAR\u001b[39;49m\u001b[39m\\n\u001b[39;49;00m\u001b[39m  REF=/datasets/Common/ReferenceGenomes/Macaca/10.111/GenomesAndAnnotation/Macaca_mulatta.Mmul_10.111.gtf\u001b[39;49m\u001b[39m\\n\u001b[39;49;00m\u001b[39m\\n\u001b[39;49;00m\u001b[39m  for bam in $BAMDIR/*bam; do\u001b[39;49m\u001b[39m\\n\u001b[39;49;00m\u001b[39m      sample=$\u001b[39;49m\u001b[39m{\u001b[39;49m\u001b[39mbam\u001b[39;49m\u001b[39m%%\u001b[39;49;00m\u001b[39mAligned*}\u001b[39;49m\u001b[39m\\n\u001b[39;49;00m\u001b[39m      sample=$\u001b[39;49m\u001b[39m{\u001b[39;49m\u001b[39msample##*/}\u001b[39;49m\u001b[39m\\n\u001b[39;49;00m\u001b[39m\\n\u001b[39;49;00m\u001b[39m    ## in this case, we assume it\u001b[39;49m\u001b[39m\\'\u001b[39;49;00m\u001b[39ms always fr\u001b[39;49m\u001b[39m\\n\u001b[39;49;00m\u001b[39m    stringtie $bam -G $REF --conservative --fr -o $OUTDIR/$\u001b[39;49m\u001b[39m{sample}\u001b[39;49;00m\u001b[39m.gtf -C $OUTDIR/$\u001b[39;49m\u001b[39m{sample}\u001b[39;49;00m\u001b[39m_cov.gtf\u001b[39;49m\u001b[39m\\n\u001b[39;49;00m\u001b[39m  done\u001b[39;49m\u001b[39m\\n\u001b[39;49;00m\u001b[39m\\n\u001b[39;49;00m\u001b[39melif [ \u001b[39;49m\u001b[39m\"\u001b[39;49m\u001b[39m$specie\u001b[39;49m\u001b[39m\"\u001b[39;49m\u001b[39m == \u001b[39;49m\u001b[39m\"\u001b[39;49m\u001b[39mmouse\u001b[39;49m\u001b[39m\"\u001b[39;49m\u001b[39m ]; then\u001b[39;49m\u001b[39m\\n\u001b[39;49;00m\u001b[39m  # BAMDIR=/datasets/marta/Wang2020/RNASeq\u001b[39;49m\u001b[39m\\n\u001b[39;49;00m\u001b[39m  BAMDIR=$1/$specie/STAR\u001b[39;49m\u001b[39m\\n\u001b[39;49;00m\u001b[39m\\n\u001b[39;49;00m\u001b[39m  REF=/datasets/Common/ReferenceGenomes/Mouse/mm39/GenomesAndAnnotation/gencode.vM32.annotation.gtf\u001b[39;49m\u001b[39m\\n\u001b[39;49;00m\u001b[39m\\n\u001b[39;49;00m\u001b[39m  for bam in $BAMDIR/*bam; do\u001b[39;49m\u001b[39m\\n\u001b[39;49;00m\u001b[39m      sample=$\u001b[39;49m\u001b[39m{\u001b[39;49m\u001b[39mbam\u001b[39;49m\u001b[39m%%\u001b[39;49;00m\u001b[39mAligned*}\u001b[39;49m\u001b[39m\\n\u001b[39;49;00m\u001b[39m      sample=$\u001b[39;49m\u001b[39m{\u001b[39;49m\u001b[39msample##*/}\u001b[39;49m\u001b[39m\\n\u001b[39;49;00m\u001b[39m\\n\u001b[39;49;00m\u001b[39m    ## in this case, we assume it\u001b[39;49m\u001b[39m\\'\u001b[39;49;00m\u001b[39ms always fr\u001b[39;49m\u001b[39m\\n\u001b[39;49;00m\u001b[39m    stringtie $bam -G $REF --conservative --fr -o $OUTDIR/$\u001b[39;49m\u001b[39m{sample}\u001b[39;49;00m\u001b[39m.gtf -C $OUTDIR/$\u001b[39;49m\u001b[39m{sample}\u001b[39;49;00m\u001b[39m_cov.gtf\u001b[39;49m\u001b[39m\\n\u001b[39;49;00m\u001b[39m  done\u001b[39;49m\u001b[39m\\n\u001b[39;49;00m\u001b[39m\\n\u001b[39;49;00m\u001b[39mfi\u001b[39;49m\u001b[39m\\n\u001b[39;49;00m\u001b[39m\\n\u001b[39;49;00m\u001b[39m'\u001b[39;49m)\n",
      "File \u001b[0;32m/soft/system/software/Miniconda3/4.9.2/lib/python3.8/site-packages/IPython/core/interactiveshell.py:2358\u001b[0m, in \u001b[0;36mInteractiveShell.run_cell_magic\u001b[0;34m(self, magic_name, line, cell)\u001b[0m\n\u001b[1;32m   2356\u001b[0m \u001b[39mwith\u001b[39;00m \u001b[39mself\u001b[39m\u001b[39m.\u001b[39mbuiltin_trap:\n\u001b[1;32m   2357\u001b[0m     args \u001b[39m=\u001b[39m (magic_arg_s, cell)\n\u001b[0;32m-> 2358\u001b[0m     result \u001b[39m=\u001b[39m fn(\u001b[39m*\u001b[39;49margs, \u001b[39m*\u001b[39;49m\u001b[39m*\u001b[39;49mkwargs)\n\u001b[1;32m   2359\u001b[0m \u001b[39mreturn\u001b[39;00m result\n",
      "File \u001b[0;32m/soft/system/software/Miniconda3/4.9.2/lib/python3.8/site-packages/IPython/core/magics/script.py:153\u001b[0m, in \u001b[0;36mScriptMagics._make_script_magic.<locals>.named_script_magic\u001b[0;34m(line, cell)\u001b[0m\n\u001b[1;32m    151\u001b[0m \u001b[39melse\u001b[39;00m:\n\u001b[1;32m    152\u001b[0m     line \u001b[39m=\u001b[39m script\n\u001b[0;32m--> 153\u001b[0m \u001b[39mreturn\u001b[39;00m \u001b[39mself\u001b[39;49m\u001b[39m.\u001b[39;49mshebang(line, cell)\n",
      "File \u001b[0;32m/soft/system/software/Miniconda3/4.9.2/lib/python3.8/site-packages/IPython/core/magics/script.py:305\u001b[0m, in \u001b[0;36mScriptMagics.shebang\u001b[0;34m(self, line, cell)\u001b[0m\n\u001b[1;32m    300\u001b[0m \u001b[39mif\u001b[39;00m args\u001b[39m.\u001b[39mraise_error \u001b[39mand\u001b[39;00m p\u001b[39m.\u001b[39mreturncode \u001b[39m!=\u001b[39m \u001b[39m0\u001b[39m:\n\u001b[1;32m    301\u001b[0m     \u001b[39m# If we get here and p.returncode is still None, we must have\u001b[39;00m\n\u001b[1;32m    302\u001b[0m     \u001b[39m# killed it but not yet seen its return code. We don't wait for it,\u001b[39;00m\n\u001b[1;32m    303\u001b[0m     \u001b[39m# in case it's stuck in uninterruptible sleep. -9 = SIGKILL\u001b[39;00m\n\u001b[1;32m    304\u001b[0m     rc \u001b[39m=\u001b[39m p\u001b[39m.\u001b[39mreturncode \u001b[39mor\u001b[39;00m \u001b[39m-\u001b[39m\u001b[39m9\u001b[39m\n\u001b[0;32m--> 305\u001b[0m     \u001b[39mraise\u001b[39;00m CalledProcessError(rc, cell)\n",
      "\u001b[0;31mCalledProcessError\u001b[0m: Command 'b'\\n## ------------------------------ done in shiva ---------------------------##\\nmodule load stringtie/2.2.0\\n\\nspecie=$2\\ndeclare -a bam_array\\n\\nif [ \"$specie\" == \"macaca\" ]; then\\n  BAMDIR=$1/$specie/STAR\\n  REF=/datasets/Common/ReferenceGenomes/Macaca/10.111/GenomesAndAnnotation/Macaca_mulatta.Mmul_10.111.gtf\\n\\n  for bam in $BAMDIR/*bam; do\\n      sample=${bam%%Aligned*}\\n      sample=${sample##*/}\\n\\n    ## in this case, we assume it\\'s always fr\\n    stringtie $bam -G $REF --conservative --fr -o $OUTDIR/${sample}.gtf -C $OUTDIR/${sample}_cov.gtf\\n  done\\n\\nelif [ \"$specie\" == \"mouse\" ]; then\\n  # BAMDIR=/datasets/marta/Wang2020/RNASeq\\n  BAMDIR=$1/$specie/STAR\\n\\n  REF=/datasets/Common/ReferenceGenomes/Mouse/mm39/GenomesAndAnnotation/gencode.vM32.annotation.gtf\\n\\n  for bam in $BAMDIR/*bam; do\\n      sample=${bam%%Aligned*}\\n      sample=${sample##*/}\\n\\n    ## in this case, we assume it\\'s always fr\\n    stringtie $bam -G $REF --conservative --fr -o $OUTDIR/${sample}.gtf -C $OUTDIR/${sample}_cov.gtf\\n  done\\n\\nfi\\n\\n'' returned non-zero exit status 1."
     ]
    }
   ],
   "source": [
    "%%bash -s \"$users_dir\" \"$specie\" \"$GENOMEDIR\"\n",
    "\n",
    "## ------------------------------ done in shiva ---------------------------##\n",
    "module load stringtie/2.2.0\n",
    "\n",
    "specie=$2\n",
    "\n",
    "if [ \"$specie\" == \"macaca\" ]; then\n",
    "  BAMDIR=$1/$specie/STAR\n",
    "  REF=/datasets/Common/ReferenceGenomes/Macaca/10.111/GenomesAndAnnotation/Macaca_mulatta.Mmul_10.111.gtf\n",
    "\n",
    "  for bam in $BAMDIR/*bam; do\n",
    "      sample=${bam%%Aligned*}\n",
    "      sample=${sample##*/}\n",
    "\n",
    "    ## in this case, we assume it's always fr\n",
    "    stringtie $bam -G $REF --conservative --fr -o $OUTDIR/${sample}.gtf \n",
    "  done\n",
    "\n",
    "elif [ \"$specie\" == \"mouse\" ]; then\n",
    "  # BAMDIR=/datasets/marta/Wang2020/RNASeq\n",
    "  BAMDIR=$1/$specie/STAR\n",
    "  REF=/datasets/Common/ReferenceGenomes/Mouse/mm39/GenomesAndAnnotation/gencode.vM32.annotation.gtf\n",
    "\n",
    "  for bam in $BAMDIR/*bam; do\n",
    "      sample=${bam%%Aligned*}\n",
    "      sample=${sample##*/}\n",
    "\n",
    "    ## in this case, we assume it's always fr\n",
    "    stringtie $bam -G $REF --conservative --fr -o $OUTDIR/${sample}.gtf \n",
    "  done\n",
    "\n",
    "fi\n",
    "\n"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Make summary"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/mouse/stringtieReconstruction/transcriptomeReconstruction.csv\n"
     ]
    }
   ],
   "source": [
    "%%bash -s \"$users_dir\" \"$specie\"\n",
    "## Total transcriptome + Presumably annotated + Presumably novel\n",
    "OUT=$1/$2/stringtieReconstruction/transcriptomeReconstruction.csv\n",
    "echo $OUT\n",
    "if [ -f \"$OUT\" ] ; then\n",
    "    rm \"$OUT\"\n",
    "fi\n",
    "echo -e \"Sample\\tTotal_transcripts\\tAnnotated_transcripts\\tNovel_transcript\" >> $OUT\n",
    "\n",
    "for file in $1/$2/stringtieReconstruction/*gtf; do\n",
    "    if [[ ${file} != *\"cov\"* ]]; then \n",
    "        name=\"${file%%.*}\"\n",
    "        name=\"${name##*/}\"\n",
    "        echo -e $name\"\\t\"$(awk '($3 == \"transcript\") {print}' $file | wc -l)\"\\t\"$(awk '($3 == \"transcript\") {print}' $file | grep \"reference\" | wc -l)\"\\t\"$(awk '($3 == \"transcript\") {print}' $file | grep -v \"reference\" | wc -l) >> $OUT\n",
    "    fi\n",
    "done\n"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "From here, I **need** to know if what is considered as novel is really novel. Intersect with annotated genes. Those that do not overlap at all, are novels for sure: maintain.\n",
    "\n",
    "For **human**:\n",
    "\n",
    "`$GENOMEDIR/Annot_files_GTF/gencode.v38.primary_assembly.annotation.gtf`\n",
    "\n",
    "For **macaca**:\n",
    "\n",
    "`/datasets/Common/ReferenceGenomes/Macaca/10.111/GenomesAndAnnotation/Macaca_mulatta.Mmul_10.111.bed`"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "mkdir: cannot create directory ‘/users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/mouse/stringtieReconstruction/NOoriented’: File exists\n",
      "mkdir: cannot create directory ‘/users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/mouse/stringtieReconstruction/oriented’: File exists\n"
     ]
    }
   ],
   "source": [
    "%%bash -s \"$users_dir\" \"$specie\"\n",
    "\n",
    "OUTDIR=$1/$2/stringtieReconstruction\n",
    "\n",
    "mkdir $OUTDIR/NOoriented\n",
    "mkdir $OUTDIR/oriented\n",
    "\n",
    "for file in $OUTDIR/*.gtf; do \n",
    "    if [[ ${file} != *\"cov\"* ]]; then \n",
    "        if [[ ${file} != *\"gene\"* ]]; then \n",
    "            name=${file##*/}\n",
    "            name=${name%%.*}\n",
    "            #separation of novel transcripts oriented not oriented\n",
    "            awk '($7 == \".\") {print}' $file > $OUTDIR/NOoriented/${name}_NOoriented.gtf\n",
    "            awk '($7 != \".\") {print}' $file > $OUTDIR/oriented/${name}_oriented.gtf\n",
    "        fi\n",
    "    fi\n",
    "done\n"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "**FORCING STRANDNESS**\n",
    "\n",
    "For the novel transcripts oriented, we force orientation when checking the overlapping with the reference."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash -s \"$users_dir\" \"$specie\" \"$GENOMEDIR\"\n",
    "\n",
    "OUTDIR=$1/$2/stringtieReconstruction\n",
    "\n",
    "module load BEDTools/2.30.0-GCC-10.2.0\n",
    "\n",
    "if [ \"$2\" == \"human\" ]; then\n",
    "    REF=$GENOMEDIR/Annot_files_GTF/gencode.v38.primary_assembly.annotation.gtf\n",
    "elif [ \"$2\" == \"macaca\" ]; then\n",
    "    REF=/datasets/Common/ReferenceGenomes/Macaca/10.111/GenomesAndAnnotation/Macaca_mulatta.Mmul_10.111.gtf\n",
    "elif [ \"$2\" == \"mouse\" ]; then\n",
    "    REF=/datasets/Common/ReferenceGenomes/Mouse/mm39/GenomesAndAnnotation/gencode.vM32.annotation.noChr.gtf\n",
    "fi\n",
    "\n",
    "for file in $OUTDIR/oriented/*oriented.gtf; do \n",
    "    name=${file##*/}\n",
    "    name=${name%%.*}\n",
    "    #select novel transcripts that do not overlap at all with the gencode annotation (-v)\n",
    "    bedtools intersect -v -s -a $file -b $REF > $OUTDIR/oriented/${name}_novel.gtf\n",
    "done\n"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "**WITHOUT FORCING STRANDNESS**\n",
    "\n",
    "For the novel transcripts NOT oriented, we do not force orientation when checking the overlapping with the reference."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash -s \"$users_dir\" \"$specie\" \"$GENOMEDIR\"\n",
    "\n",
    "OUTDIR=$1/$2/stringtieReconstruction\n",
    "\n",
    "module load BEDTools/2.30.0-GCC-10.2.0\n",
    "\n",
    "if [ \"$2\" == \"human\" ]; then\n",
    "    REF=$GENOMEDIR/Annot_files_GTF/gencode.v38.primary_assembly.annotation.gtf\n",
    "elif [ \"$2\" == \"macaca\" ]; then\n",
    "    REF=/datasets/Common/ReferenceGenomes/Macaca/10.111/GenomesAndAnnotation/Macaca_mulatta.Mmul_10.111.gtf\n",
    "elif [ \"$2\" == \"mouse\" ]; then\n",
    "    REF=/datasets/Common/ReferenceGenomes/Mouse/mm39/GenomesAndAnnotation/gencode.vM32.annotation.noChr.gtf\n",
    "fi\n",
    "\n",
    "for file in $OUTDIR/NOoriented/*oriented.gtf; do \n",
    "    name=${file##*/}\n",
    "    name=${name%%.*}\n",
    "\n",
    "    #select novel transcripts that do not overlap at all with the gencode annotation (-v)\n",
    "    bedtools intersect -v -a $file -b $REF > $OUTDIR/NOoriented/${name}_novel.gtf\n",
    "done"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "I want to see if there are \"alone exons\", I don't want them, only full transcripts. Check transcript line, store geneid (STRG.XX), see if later exons contain the same. If so, store in new file, if not, skip."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {},
   "outputs": [],
   "source": [
    "for file in os.listdir(os.path.join(users_dir,specie,\"stringtieReconstruction/NOoriented\")):\n",
    "    if file.endswith(\"novel.gtf\"):\n",
    "        outname=file[:-4]+\"_complete.gtf\"\n",
    "        with open(os.path.join(os.path.join(users_dir,specie,\"stringtieReconstruction/NOoriented\"),file)) as inp:\n",
    "            with open(os.path.join(os.path.join(users_dir,specie,\"stringtieReconstruction/NOoriented\"),outname), 'w') as out:\n",
    "                lines = inp.readlines()\n",
    "                t_gene_id = \"\"\n",
    "                for line in lines:\n",
    "                    tab_split_line=line.split(\"\\t\")\n",
    "                    coma_split_line=line.split(\";\")\n",
    "                    gene_id=re.findall(r'\"([^\"]*)\"',coma_split_line[0])\n",
    "                    if tab_split_line[2] == \"transcript\":\n",
    "                        t_gene_id = gene_id[0]\n",
    "                        out.write(line)\n",
    "                    if tab_split_line[2] == \"exon\":\n",
    "                        if gene_id[0] == t_gene_id:\n",
    "                            out.write(line)\n",
    "                        else:\n",
    "                            continue\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {},
   "outputs": [],
   "source": [
    "for file in os.listdir(os.path.join(users_dir,specie,\"stringtieReconstruction/oriented\")):\n",
    "    if file.endswith(\"novel.gtf\"):\n",
    "        outname=file[:-4]+\"_complete.gtf\"\n",
    "        with open(os.path.join(os.path.join(users_dir,specie,\"stringtieReconstruction/oriented\"),file)) as inp:\n",
    "            with open(os.path.join(os.path.join(users_dir,specie,\"stringtieReconstruction/oriented\"),outname), 'w') as out:\n",
    "                lines = inp.readlines()\n",
    "                t_gene_id = \"\"\n",
    "                for line in lines:\n",
    "                    tab_split_line=line.split(\"\\t\")\n",
    "                    coma_split_line=line.split(\";\")\n",
    "                    gene_id=re.findall(r'\"([^\"]*)\"',coma_split_line[0])\n",
    "                    if tab_split_line[2] == \"transcript\":\n",
    "                        t_gene_id = gene_id[0]\n",
    "                        out.write(line)\n",
    "                    if tab_split_line[2] == \"exon\":\n",
    "                        if gene_id[0] == t_gene_id:\n",
    "                            out.write(line)\n",
    "                        else:\n",
    "                            continue\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash -s \"$users_dir\" \"$specie\" \"$GENOMEDIR\"\n",
    "\n",
    "## Concatenate both oriented and unoriented\n",
    "OUTDIR=$1/$2/stringtieReconstruction/intersect_novel\n",
    "mkdir $OUTDIR\n",
    "\n",
    "\n",
    "for sample in $(cat $1/$2/stringtieReconstruction/transcriptomeReconstruction.csv | tail -n +2 | cut -f1); do\n",
    "    #concatenate both results in order to obtain a gtf file with all novel transcripts considered\n",
    "    cat $1/$2/stringtieReconstruction/oriented/${sample}_oriented_novel_complete.gtf $1/$2/stringtieReconstruction/NOoriented/${sample}_NOoriented_novel_complete.gtf > $OUTDIR/${sample}_novel.gtf\n",
    "done\n"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Now we use gffcompare so that we unify all novel transcripts into a single transcriptome"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Loading query file #1: /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/mouse/stringtieReconstruction/intersect_novel/mouse_brain_rna_1_novel.gtf\n",
      "  6 duplicate query transfrags discarded.\n",
      "Loading query file #2: /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/mouse/stringtieReconstruction/intersect_novel/mouse_brain_rna_2_novel.gtf\n",
      "  4 duplicate query transfrags discarded.\n",
      "Loading query file #3: /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/mouse/stringtieReconstruction/intersect_novel/mouse_brain_rna_3_novel.gtf\n",
      "  6 duplicate query transfrags discarded.\n",
      "Loading query file #4: /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/mouse/stringtieReconstruction/intersect_novel/mouse_elongatingSpermatids_rna_1_novel.gtf\n",
      "  6 duplicate query transfrags discarded.\n",
      "Loading query file #5: /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/mouse/stringtieReconstruction/intersect_novel/mouse_elongatingSpermatids_rna_2_novel.gtf\n",
      "  6 duplicate query transfrags discarded.\n",
      "Loading query file #6: /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/mouse/stringtieReconstruction/intersect_novel/mouse_liver_rna_1_novel.gtf\n",
      "  1 duplicate query transfrags discarded.\n",
      "Loading query file #7: /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/mouse/stringtieReconstruction/intersect_novel/mouse_liver_rna_2_novel.gtf\n",
      "  1 duplicate query transfrags discarded.\n",
      "Loading query file #8: /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/mouse/stringtieReconstruction/intersect_novel/mouse_liver_rna_3_novel.gtf\n",
      "  2 duplicate query transfrags discarded.\n",
      "Loading query file #9: /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/mouse/stringtieReconstruction/intersect_novel/mouse_liver_rna_4_novel.gtf\n",
      "  2 duplicate query transfrags discarded.\n",
      "Loading query file #10: /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/mouse/stringtieReconstruction/intersect_novel/mouse_liver_rna_5_novel.gtf\n",
      "Loading query file #11: /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/mouse/stringtieReconstruction/intersect_novel/mouse_roundSpermatids_rna_1_novel.gtf\n",
      "  13 duplicate query transfrags discarded.\n",
      "Loading query file #12: /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/mouse/stringtieReconstruction/intersect_novel/mouse_roundSpermatids_rna_2_novel.gtf\n",
      "  11 duplicate query transfrags discarded.\n",
      "Loading query file #13: /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/mouse/stringtieReconstruction/intersect_novel/mouse_spermatocytes_rna_1_novel.gtf\n",
      "  4 duplicate query transfrags discarded.\n",
      "Loading query file #14: /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/mouse/stringtieReconstruction/intersect_novel/mouse_spermatocytes_rna_2_novel.gtf\n",
      "  3 duplicate query transfrags discarded.\n",
      "Loading query file #15: /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/mouse/stringtieReconstruction/intersect_novel/mouse_spermatozoa_rna_1_novel.gtf\n",
      "Loading query file #16: /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/mouse/stringtieReconstruction/intersect_novel/mouse_spermatozoa_rna_2_novel.gtf\n",
      "  3 duplicate query transfrags discarded.\n",
      "Loading query file #17: /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/mouse/stringtieReconstruction/intersect_novel/mouse_testis_rna_1_novel.gtf\n",
      "  6 duplicate query transfrags discarded.\n",
      "Loading query file #18: /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/mouse/stringtieReconstruction/intersect_novel/mouse_testis_rna_2_novel.gtf\n",
      "  9 duplicate query transfrags discarded.\n",
      "Loading query file #19: /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/mouse/stringtieReconstruction/intersect_novel/mouse_testis_rna_3_novel.gtf\n",
      "  10 duplicate query transfrags discarded.\n"
     ]
    }
   ],
   "source": [
    "%%bash -s \"$users_dir\" \"$specie\"\n",
    "\n",
    "OUTDIR=$1/$2/gffcompare\n",
    "mkdir $OUTDIR\n",
    "\n",
    "module load gffcompare/0.12.2\n",
    "\n",
    "ls $1/$2/stringtieReconstruction/intersect_novel/*_novel.gtf > $OUTDIR/list_novel_assembled_files_stranded.txt\n",
    "\n",
    "cd $OUTDIR\n",
    "gffcompare -i list_novel_assembled_files_stranded.txt -o gffcmp.assembled.stranded\n",
    "\n",
    "\n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "  58257 +\n",
      "  58554 -\n",
      "      5 .\n"
     ]
    }
   ],
   "source": [
    "%%bash -s \"$users_dir\" \"$specie\"\n",
    "\n",
    "echo -e \"\\n\"$proj\n",
    "awk '$3 == \"transcript\" {print}' $1/$2/gffcompare/gffcmp.assembled.stranded.combined.gtf | cut -f 7 | sort | uniq -c\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash -s \"$users_dir\" \"$specie\"\n",
    "\n",
    "## miTranscriptome\n",
    "## for stranded datasets\n",
    "mitranscriptome=/genomics/users/marta/genomes/miTranscriptome/mitranscriptome.v2.gtf\n",
    "\n",
    "module load BEDTools\n",
    "\n",
    "gffcompare=$1/$2/gffcompare/gffcmp.assembled.stranded.combined.gtf\n",
    "miTranscriptomeOUTDIR=$1/$2/gffcompare/miTranscriptome\n",
    "\n",
    "mkdir $miTranscriptomeOUTDIR\n",
    "\n",
    "bedtools intersect -s -v -a $gffcompare -b $mitranscriptome > $miTranscriptomeOUTDIR/gffcmp.assembled.stranded.combined.NOmiTranscriptome.gtf\n",
    "bedtools intersect -s -wa -wb -a $gffcompare -b $mitranscriptome > $miTranscriptomeOUTDIR/gffcmp.assembled.stranded.combined.miTranscriptome.gtf\n",
    "bedtools intersect -s -u -a $gffcompare -b $mitranscriptome > $miTranscriptomeOUTDIR/gffcmp.assembled.stranded.combined.miTranscriptome.unique.gtf\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "    533 +\n",
      "    563 -\n"
     ]
    }
   ],
   "source": [
    "%%bash -s \"$users_dir\" \"$specie\"\n",
    "\n",
    "awk '$3 == \"transcript\" {print}' $1/$2/gffcompare/miTranscriptome/gffcmp.assembled.stranded.combined.miTranscriptome.unique.gtf | cut -f 7 | sort | uniq -c\n"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Let's create a new reference file with novel reconstructed transcripts"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash -s \"$users_dir\" \"$specie\" \"$GENOMEDIR\"\n",
    "\n",
    "export PATH=/genomics/users/marta/tools/gff3sort-master:$PATH\n",
    "\n",
    "if [ \"$2\" == \"human\" ]; then\n",
    "    ## reference \n",
    "    REF=$3/Annot_files_GTF/gencode.v38.primary_assembly.annotation.gtf \n",
    "\n",
    "    OUTDIR=$1/$2/newReference_Resconstructed\n",
    "    mkdir $OUTDIR\n",
    "\n",
    "    ## gffcompare\n",
    "    gffcompare=$1/$2/gffcompare/gffcmp.assembled.stranded.combined.gtf\n",
    "\n",
    "    ## concat\n",
    "    cat $gffcompare $REF > $OUTDIR/gencode.v38.gffcompare.TestisLiverBrain.annotation.gtf\n",
    "\n",
    "    #sort the newly assembled transcriptome\n",
    "    gff3sort.pl --precise --chr_order natural $OUTDIR/gencode.v38.gffcompare.TestisLiverBrain.annotation.gtf > $OUTDIR/gencode.v38.gffcompare.TestisLiverBrain.annotation.sorted.gtf\n",
    "    rm $OUTDIR/gencode.v38.gffcompare.TestisLiverBrain.annotation.gtf\n",
    "    \n",
    "elif [ \"$2\" == \"macaca\" ]; then\n",
    "    ## reference \n",
    "    REF=/datasets/Common/ReferenceGenomes/Macaca/10.111/GenomesAndAnnotation/Macaca_mulatta.Mmul_10.111.gtf\n",
    "    OUTDIR=$1/$2/newReference_Resconstructed\n",
    "    mkdir $OUTDIR\n",
    "\n",
    "    ## gffcompare\n",
    "    gffcompare=$1/$2/gffcompare/gffcmp.assembled.stranded.combined.gtf\n",
    "\n",
    "    ## concat\n",
    "    cat $gffcompare $REF > $OUTDIR/Macaca_mulatta.Mmul_10.111.gffcompare.TestisLiverBrain.annotation.gtf\n",
    "\n",
    "    #sort the newly assembled transcriptome\n",
    "    gff3sort.pl --precise --chr_order natural $OUTDIR/Macaca_mulatta.Mmul_10.111.gffcompare.TestisLiverBrain.annotation.gtf > $OUTDIR/Macaca_mulatta.Mmul_10.111.gffcompare.TestisLiverBrain.annotation.sorted.gtf\n",
    "    rm $OUTDIR/Macaca_mulatta.Mmul_10.111.gffcompare.TestisLiverBrain.annotation.gtf\n",
    "\n",
    "elif [ \"$2\" == \"mouse\" ]; then\n",
    "    ## reference \n",
    "    REF=/datasets/Common/ReferenceGenomes/Mouse/mm39/GenomesAndAnnotation/gencode.vM32.annotation.noChr.gtf\n",
    "    OUTDIR=$1/$2/newReference_Resconstructed\n",
    "    mkdir $OUTDIR\n",
    "\n",
    "    ## gffcompare\n",
    "    gffcompare=$1/$2/gffcompare/gffcmp.assembled.stranded.combined.gtf\n",
    "\n",
    "    ## concat\n",
    "    cat $gffcompare $REF > $OUTDIR/gencode.vM32.gffcompare.TestisLiverBrain.annotation.gtf\n",
    "\n",
    "    #sort the newly assembled transcriptome\n",
    "    gff3sort.pl --precise --chr_order natural $OUTDIR/gencode.vM32.gffcompare.TestisLiverBrain.annotation.gtf > $OUTDIR/gencode.vM32.gffcompare.TestisLiverBrain.annotation.sorted.gtf\n",
    "    rm $OUTDIR/gencode.vM32.gffcompare.TestisLiverBrain.annotation.gtf\n",
    "fi\n"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Get simplified annotation\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_1321744/2733866862.py:27: DtypeWarning: Columns (0) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  reference = pd.read_csv(annotation_GTF, sep=\"\\t\", header=None, comment=\"#\")\n",
      "/tmp/ipykernel_1321744/2733866862.py:33: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  exons['length'] = exons[4] - exons[3]\n"
     ]
    }
   ],
   "source": [
    "## Author: Marta Espinosa Camarena\n",
    "## Date: 21st May 24\n",
    "\n",
    "## This scripts attemps to build a simplified version of an annotation file.\n",
    "## By simplified we mean that we keep only 1 transcript per gene, being the longest transcript the selected one.\n",
    "## Running mode:\n",
    "## python get_simplified_annotation.py $absolute_path_to_annotation $outdir\n",
    "## make sure the output directory has writing permissions\n",
    "\n",
    "import pandas as pd\n",
    "import sys, os\n",
    "from pathlib import Path\n",
    "\n",
    "##----------------- importing requested data -----------------##\n",
    "# annotation_GTF = sys.argv[1]\n",
    "# annotation_GTF = \"/users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/human/newReference_Resconstructed/gencode.v38.gffcompare.TestisLiverBrain.annotation.sorted.gtf\" ## testing\n",
    "# annotation_GTF = \"/users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/macaca/newReference_Resconstructed/Macaca_mulatta.Mmul_10.111.gffcompare.TestisLiverBrain.annotation.sorted.gtf\"\n",
    "annotation_GTF = \"/users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/mouse/newReference_Resconstructed/gencode.vM32.gffcompare.TestisLiverBrain.annotation.sorted.gtf\"\n",
    "\n",
    "# outdir = sys.argv[2]\n",
    "# outdir = \"/users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/macaca/newReference_Resconstructed\" \n",
    "outdir = \"/users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/mouse/newReference_Resconstructed\" \n",
    "outfile = os.path.basename(annotation_GTF)\n",
    "outfile = outfile[:-4]\n",
    "\n",
    "## reading reference file and separate by transcripts and genes\n",
    "reference = pd.read_csv(annotation_GTF, sep=\"\\t\", header=None, comment=\"#\")\n",
    "reference['transcript_id'] = reference[8].str.extract('transcript_id \"([^\"]*)\"', expand=True)\n",
    "reference['gene_id'] = reference[8].str.extract('gene_id \"([^\"]*)\"', expand=True)\n",
    "\n",
    "## subset exons and calculate length\n",
    "exons = reference[reference[2] == \"exon\"]\n",
    "exons['length'] = exons[4] - exons[3]\n",
    "genes = reference[reference[2] == \"gene\"]\n",
    "\n",
    "## calculate length transcripts\n",
    "length_tid = exons.groupby('transcript_id', as_index=False)[['length']].sum()\n",
    "\n",
    "## make dataframe with geneid_transcriptid_length\n",
    "ref_w_length = length_tid.merge(reference, on=[\"transcript_id\"], how=\"right\").drop_duplicates()\n",
    "# ref_w_length = ref_w_length[['transcript_id','gene_id','length']]\n",
    "# ref_w_length.drop_duplicates(inplace=True)\n",
    "# ref_w_length.dropna(inplace=True)\n",
    "\n",
    "## select longest transcript per gene\n",
    "## Group by 'gene_id' and get the indices of the rows with the maximum 'length'\n",
    "## select datafrem rows corresponding to longest trancript_id - if different transcripts for the same gene with same length, we keep the first one\n",
    "max_length_indices = ref_w_length.groupby('gene_id')['length'].idxmax()\n",
    "\n",
    "longest_transcripts_gid = ref_w_length.loc[max_length_indices]\n",
    "longest_transcripts_gid = longest_transcripts_gid[['gene_id','transcript_id','length']]\n",
    "longest_transcripts_gid.drop_duplicates(inplace=True) ## no duplications\n",
    "\n",
    "reference_longest = longest_transcripts_gid.merge(ref_w_length[ref_w_length[2] != \"gene\"], on=['gene_id','transcript_id','length'], how=\"left\")\n",
    "\n",
    "\n",
    "full_reference = pd.concat([reference_longest, genes])\n",
    "full_reference.drop(['transcript_id','gene_id','length'], axis=1, inplace=True)\n",
    "full_reference.to_csv(outdir+\"/\"+outfile+\".1transcript.gtf\", sep=\"\\t\", header=None, index=None,  quoting = 3)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash -s \"$users_dir\" \"$specie\"\n",
    "\n",
    "export PATH=/genomics/users/marta/tools/gff3sort-master:$PATH\n",
    "\n",
    "if [ \"$2\" == \"human\" ]; then\n",
    "    reference=$1/$2/newReference_Resconstructed/gencode.v38.gffcompare.TestisLiverBrain.annotation.sorted.1transcript.gtf\n",
    "    out=$1/$2/newReference_Resconstructed/gencode.v38.gffcompare.TestisLiverBrain.annotation.sorted.1transcript.sorted.gtf\n",
    "elif [ \"$2\" == \"macaca\" ]; then\n",
    "    reference=$1/$2/newReference_Resconstructed/Macaca_mulatta.Mmul_10.111.gffcompare.TestisLiverBrain.annotation.sorted.1transcript.gtf\n",
    "    out=$1/$2/newReference_Resconstructed/Macaca_mulatta.Mmul_10.111.gffcompare.TestisLiverBrain.annotation.sorted.1transcript.sorted.gtf\n",
    "elif [ \"$2\" == \"mouse\" ]; then\n",
    "    reference=$1/$2/newReference_Resconstructed/gencode.vM32.gffcompare.TestisLiverBrain.annotation.sorted.1transcript.gtf\n",
    "    out=$1/$2/newReference_Resconstructed/gencode.vM32.gffcompare.TestisLiverBrain.annotation.sorted.1transcript.sorted.gtf\n",
    "fi\n",
    "\n",
    "gff3sort.pl --precise --chr_order natural $reference > $out\n"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Quantification\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "        ==========     _____ _    _ ____  _____  ______          _____  \n",
      "        =====         / ____| |  | |  _ \\|  __ \\|  ____|   /\\   |  __ \\ \n",
      "          =====      | (___ | |  | | |_) | |__) | |__     /  \\  | |  | |\n",
      "            ====      \\___ \\| |  | |  _ <|  _  /|  __|   / /\\ \\ | |  | |\n",
      "              ====    ____) | |__| | |_) | | \\ \\| |____ / ____ \\| |__| |\n",
      "        ==========   |_____/ \\____/|____/|_|  \\_\\______/_/    \\_\\_____/\n",
      "\t  v2.0.3\n",
      "\n",
      "//========================== featureCounts setting ===========================\\\\\n",
      "||                                                                            ||\n",
      "||             Input files : 19 BAM files                                     ||\n",
      "||                                                                            ||\n",
      "||                           mouse_brain_rna_1.bam                            ||\n",
      "||                           mouse_brain_rna_2.bam                            ||\n",
      "||                           mouse_brain_rna_3.bam                            ||\n",
      "||                           mouse_elongatingSpermatids_rna_1.bam             ||\n",
      "||                           mouse_elongatingSpermatids_rna_2.bam             ||\n",
      "||                           mouse_liver_rna_1.bam                            ||\n",
      "||                           mouse_liver_rna_2.bam                            ||\n",
      "||                           mouse_liver_rna_3.bam                            ||\n",
      "||                           mouse_liver_rna_4.bam                            ||\n",
      "||                           mouse_liver_rna_5.bam                            ||\n",
      "||                           mouse_roundSpermatids_rna_1.bam                  ||\n",
      "||                           mouse_roundSpermatids_rna_2.bam                  ||\n",
      "||                           mouse_spermatocytes_rna_1.bam                    ||\n",
      "||                           mouse_spermatocytes_rna_2.bam                    ||\n",
      "||                           mouse_spermatozoa_rna_1.bam                      ||\n",
      "||                           mouse_spermatozoa_rna_2.bam                      ||\n",
      "||                           mouse_testis_rna_1.bam                           ||\n",
      "||                           mouse_testis_rna_2.bam                           ||\n",
      "||                           mouse_testis_rna_3.bam                           ||\n",
      "||                                                                            ||\n",
      "||             Output file : gffcompare_stranded_featureCounts.txt            ||\n",
      "||                 Summary : gffcompare_stranded_featureCounts.txt.summary    ||\n",
      "||              Paired-end : no                                               ||\n",
      "||        Count read pairs : no                                               ||\n",
      "||              Annotation : gencode.vM32.gffcompare.TestisLiverBrain.ann ... ||\n",
      "||      Dir for temp files : /users/genomics/marta/TestisProject_SaraRazq ... ||\n",
      "||                                                                            ||\n",
      "||                 Threads : 10                                               ||\n",
      "||                   Level : meta-feature level                               ||\n",
      "||      Multimapping reads : not counted                                      ||\n",
      "|| Multi-overlapping reads : counted                                          ||\n",
      "||   Min overlapping bases : 1                                                ||\n",
      "||                                                                            ||\n",
      "\\\\============================================================================//\n",
      "\n",
      "//================================= Running ==================================\\\\\n",
      "||                                                                            ||\n",
      "|| Load annotation file gencode.vM32.gffcompare.TestisLiverBrain.annotati ... ||\n",
      "||    Features : 355105                                                       ||\n",
      "||    Meta-features : 84152                                                   ||\n",
      "||    Chromosomes/contigs : 40                                                ||\n",
      "||                                                                            ||\n",
      "|| Process BAM file mouse_brain_rna_1.bam...                                  ||\n",
      "||    Strand specific : stranded                                              ||\n",
      "||    Single-end reads are included.                                          ||\n",
      "||    Total alignments : 110640515                                            ||\n",
      "||    Successfully assigned alignments : 13496734 (12.2%)                     ||\n",
      "||    Running time : 5.77 minutes                                             ||\n",
      "||                                                                            ||\n",
      "|| Process BAM file mouse_brain_rna_2.bam...                                  ||\n",
      "||    Strand specific : stranded                                              ||\n",
      "||    Single-end reads are included.                                          ||\n",
      "||    Total alignments : 183394905                                            ||\n",
      "||    Successfully assigned alignments : 19686972 (10.7%)                     ||\n",
      "||    Running time : 11.24 minutes                                            ||\n",
      "||                                                                            ||\n",
      "|| Process BAM file mouse_brain_rna_3.bam...                                  ||\n",
      "||    Strand specific : stranded                                              ||\n",
      "||    Single-end reads are included.                                          ||\n",
      "||    Total alignments : 131577805                                            ||\n",
      "||    Successfully assigned alignments : 18344457 (13.9%)                     ||\n",
      "||    Running time : 11.20 minutes                                            ||\n",
      "||                                                                            ||\n",
      "|| Process BAM file mouse_elongatingSpermatids_rna_1.bam...                   ||\n",
      "||    Strand specific : stranded                                              ||\n",
      "||    Single-end reads are included.                                          ||\n",
      "||    Total alignments : 60095200                                             ||\n",
      "||    Successfully assigned alignments : 17934497 (29.8%)                     ||\n",
      "||    Running time : 3.87 minutes                                             ||\n",
      "||                                                                            ||\n",
      "|| Process BAM file mouse_elongatingSpermatids_rna_2.bam...                   ||\n",
      "||    Strand specific : stranded                                              ||\n",
      "||    Single-end reads are included.                                          ||\n",
      "||    Total alignments : 66918041                                             ||\n",
      "||    Successfully assigned alignments : 21207792 (31.7%)                     ||\n",
      "||    Running time : 2.47 minutes                                             ||\n",
      "||                                                                            ||\n",
      "|| Process BAM file mouse_liver_rna_1.bam...                                  ||\n",
      "||    Strand specific : stranded                                              ||\n",
      "||    Single-end reads are included.                                          ||\n",
      "||    Total alignments : 97114383                                             ||\n",
      "||    Successfully assigned alignments : 12900657 (13.3%)                     ||\n",
      "||    Running time : 2.88 minutes                                             ||\n",
      "||                                                                            ||\n",
      "|| Process BAM file mouse_liver_rna_2.bam...                                  ||\n",
      "||    Strand specific : stranded                                              ||\n",
      "||    Single-end reads are included.                                          ||\n",
      "||    Total alignments : 179307420                                            ||\n",
      "||    Successfully assigned alignments : 19798312 (11.0%)                     ||\n",
      "||    Running time : 0.75 minutes                                             ||\n",
      "||                                                                            ||\n",
      "|| Process BAM file mouse_liver_rna_3.bam...                                  ||\n",
      "||    Strand specific : stranded                                              ||\n",
      "||    Single-end reads are included.                                          ||\n",
      "||    Total alignments : 154012800                                            ||\n",
      "||    Successfully assigned alignments : 21595671 (14.0%)                     ||\n",
      "||    Running time : 0.97 minutes                                             ||\n",
      "||                                                                            ||\n",
      "|| Process BAM file mouse_liver_rna_4.bam...                                  ||\n",
      "||    Strand specific : stranded                                              ||\n",
      "||    Single-end reads are included.                                          ||\n",
      "||    Total alignments : 113395803                                            ||\n",
      "||    Successfully assigned alignments : 19348586 (17.1%)                     ||\n",
      "||    Running time : 0.75 minutes                                             ||\n",
      "||                                                                            ||\n",
      "|| Process BAM file mouse_liver_rna_5.bam...                                  ||\n",
      "||    Strand specific : stranded                                              ||\n",
      "||    Single-end reads are included.                                          ||\n",
      "||    Total alignments : 108125873                                            ||\n",
      "||    Successfully assigned alignments : 18801236 (17.4%)                     ||\n",
      "||    Running time : 0.72 minutes                                             ||\n",
      "||                                                                            ||\n",
      "|| Process BAM file mouse_roundSpermatids_rna_1.bam...                        ||\n",
      "||    Strand specific : stranded                                              ||\n",
      "||    Single-end reads are included.                                          ||\n",
      "||    Total alignments : 160604648                                            ||\n",
      "||    Successfully assigned alignments : 34705350 (21.6%)                     ||\n",
      "||    Running time : 1.16 minutes                                             ||\n",
      "||                                                                            ||\n",
      "|| Process BAM file mouse_roundSpermatids_rna_2.bam...                        ||\n",
      "||    Strand specific : stranded                                              ||\n",
      "||    Single-end reads are included.                                          ||\n",
      "||    Total alignments : 109107916                                            ||\n",
      "||    Successfully assigned alignments : 24606283 (22.6%)                     ||\n",
      "||    Running time : 0.46 minutes                                             ||\n",
      "||                                                                            ||\n",
      "|| Process BAM file mouse_spermatocytes_rna_1.bam...                          ||\n",
      "||    Strand specific : stranded                                              ||\n",
      "||    Single-end reads are included.                                          ||\n",
      "||    Total alignments : 139221399                                            ||\n",
      "||    Successfully assigned alignments : 32946614 (23.7%)                     ||\n",
      "||    Running time : 0.56 minutes                                             ||\n",
      "||                                                                            ||\n",
      "|| Process BAM file mouse_spermatocytes_rna_2.bam...                          ||\n",
      "||    Strand specific : stranded                                              ||\n",
      "||    Single-end reads are included.                                          ||\n",
      "||    Total alignments : 106394703                                            ||\n",
      "||    Successfully assigned alignments : 23694534 (22.3%)                     ||\n",
      "||    Running time : 0.40 minutes                                             ||\n",
      "||                                                                            ||\n",
      "|| Process BAM file mouse_spermatozoa_rna_1.bam...                            ||\n",
      "||    Strand specific : stranded                                              ||\n",
      "||    Single-end reads are included.                                          ||\n",
      "||    Total alignments : 179328471                                            ||\n",
      "||    Successfully assigned alignments : 17050865 (9.5%)                      ||\n",
      "||    Running time : 0.63 minutes                                             ||\n",
      "||                                                                            ||\n",
      "|| Process BAM file mouse_spermatozoa_rna_2.bam...                            ||\n",
      "||    Strand specific : stranded                                              ||\n",
      "||    Single-end reads are included.                                          ||\n",
      "||    Total alignments : 90116298                                             ||\n",
      "||    Successfully assigned alignments : 19822436 (22.0%)                     ||\n",
      "||    Running time : 0.34 minutes                                             ||\n",
      "||                                                                            ||\n",
      "|| Process BAM file mouse_testis_rna_1.bam...                                 ||\n",
      "||    Strand specific : stranded                                              ||\n",
      "||    Single-end reads are included.                                          ||\n",
      "||    Total alignments : 96817226                                             ||\n",
      "||    Successfully assigned alignments : 20675489 (21.4%)                     ||\n",
      "||    Running time : 0.44 minutes                                             ||\n",
      "||                                                                            ||\n",
      "|| Process BAM file mouse_testis_rna_2.bam...                                 ||\n",
      "||    Strand specific : stranded                                              ||\n",
      "||    Single-end reads are included.                                          ||\n",
      "||    Total alignments : 67326913                                             ||\n",
      "||    Successfully assigned alignments : 14746154 (21.9%)                     ||\n",
      "||    Running time : 0.27 minutes                                             ||\n",
      "||                                                                            ||\n",
      "|| Process BAM file mouse_testis_rna_3.bam...                                 ||\n",
      "||    Strand specific : stranded                                              ||\n",
      "||    Single-end reads are included.                                          ||\n",
      "||    Total alignments : 160548544                                            ||\n",
      "||    Successfully assigned alignments : 34778643 (21.7%)                     ||\n",
      "||    Running time : 0.72 minutes                                             ||\n",
      "||                                                                            ||\n",
      "|| Write the final count table.                                               ||\n",
      "|| Write the read assignment summary.                                         ||\n",
      "||                                                                            ||\n",
      "|| Summary of counting results can be found in file \"/users/genomics/marta/T  ||\n",
      "|| estisProject_SaraRazquin/with_TranscriptomeReconstruction/mouse/featureCo  ||\n",
      "|| unts_gffcompare/gffcompare_stranded_featureCounts.txt.summary\"             ||\n",
      "||                                                                            ||\n",
      "\\\\============================================================================//\n",
      "\n"
     ]
    }
   ],
   "source": [
    "%%bash -s \"$users_dir\" \"$specie\" \n",
    "\n",
    "###PREPARING NEEDED DATA\n",
    "# PROJECT=$1\n",
    "# DIR=$2/$PROJECT\n",
    "# p=$3 #single-end or paired-end dataset\n",
    "p=\"single-end\"\n",
    "strand=\"secondstrand\"\n",
    "CLUSTERDIR=$1/$2/featureCounts_gffcompare\n",
    "mkdir $CLUSTERDIR\n",
    "if [ \"$2\" == \"human\" ]; then\n",
    "    AnnotGTF=$1/$2/newReference_Resconstructed/gencode.v38.gffcompare.TestisLiverBrain.annotation.sorted.1transcript.sorted.gtf\n",
    "    BAMDIR=$1/$2/STAR\n",
    "\n",
    "elif [ \"$2\" == \"macaca\" ]; then\n",
    "    AnnotGTF=$1/$2/newReference_Resconstructed/Macaca_mulatta.Mmul_10.111.gffcompare.TestisLiverBrain.annotation.sorted.1transcript.gtf\n",
    "    BAMDIR=$1/$2/STAR\n",
    "\n",
    "elif [ \"$2\" == \"mouse\" ]; then\n",
    "    AnnotGTF=$1/$2/newReference_Resconstructed/gencode.vM32.gffcompare.TestisLiverBrain.annotation.sorted.1transcript.gtf\n",
    "    BAMDIR=/datasets/marta/Wang2020/RNASeq\n",
    "\n",
    "fi\n",
    "# BAMDIR=/users/genomics/saraa/projectTestis/STAR/$2\n",
    "\n",
    "module load Subread/2.0.3\n",
    "########################\n",
    "\n",
    "# countReadPairs may need to be removed in case of single-end reads\n",
    "if [ \"$2\" == \"mouse\" ]; then\n",
    "    if [ $p == \"paired-end\" ]; then\n",
    "        if [ $strand == \"firststrand\" ]; then\n",
    "            featureCounts -T 10 -p -s 2 -g transcript_id -O --countReadPairs -a $AnnotGTF -o ${CLUSTERDIR}/gffcompare_stranded_featureCounts.txt $BAMDIR/mouse*.bam \n",
    "        elif [ $strand == \"secondstrand\" ]; then\n",
    "            featureCounts -T 10 -p -s 1 -g transcript_id -O --countReadPairs -a $AnnotGTF -o ${CLUSTERDIR}/gffcompare_stranded_featureCounts.txt $BAMDIR/mouse*.bam \n",
    "        elif [ $strand == \"unstranded\" ]; then\n",
    "            featureCounts -T 10 -p -s 0 -g transcript_id -O --countReadPairs -a $AnnotGTF -o ${CLUSTERDIR}/gffcompare_stranded_featureCounts.txt $BAMDIR/mouse*.bam \n",
    "        fi\n",
    "    fi\n",
    "    if [ $p == \"single-end\" ]; then\n",
    "        if [ $strand == \"firststrand\" ]; then\n",
    "            featureCounts -T 10 -s 2 -g transcript_id -O -a $AnnotGTF -o ${CLUSTERDIR}/gffcompare_stranded_featureCounts.txt $BAMDIR/mouse*.bam \n",
    "        elif [ $strand == \"secondstrand\" ]; then\n",
    "            featureCounts -T 10 -s 1 -g transcript_id -O -a $AnnotGTF -o ${CLUSTERDIR}/gffcompare_stranded_featureCounts.txt $BAMDIR/mouse*.bam \n",
    "        elif [ $strand == \"unstranded\" ]; then\n",
    "            featureCounts -T 10 -s 0 -g transcript_id -O -a $AnnotGTF -o ${CLUSTERDIR}/gffcompare_stranded_featureCounts.txt $BAMDIR/mouse*.bam \n",
    "        fi\n",
    "    fi\n",
    "else\n",
    "    if [ $p == \"paired-end\" ]; then\n",
    "        if [ $strand == \"firststrand\" ]; then\n",
    "            featureCounts -T 10 -p -s 2 -g transcript_id -O --countReadPairs -a $AnnotGTF -o ${CLUSTERDIR}/gffcompare_stranded_featureCounts.txt $BAMDIR/*Aligned.sortedByCoord.out.bam \n",
    "        elif [ $strand == \"secondstrand\" ]; then\n",
    "            featureCounts -T 10 -p -s 1 -g transcript_id -O --countReadPairs -a $AnnotGTF -o ${CLUSTERDIR}/gffcompare_stranded_featureCounts.txt $BAMDIR/*Aligned.sortedByCoord.out.bam \n",
    "        elif [ $strand == \"unstranded\" ]; then\n",
    "            featureCounts -T 10 -p -s 0 -g transcript_id -O --countReadPairs -a $AnnotGTF -o ${CLUSTERDIR}/gffcompare_stranded_featureCounts.txt $BAMDIR/*Aligned.sortedByCoord.out.bam \n",
    "        fi\n",
    "    fi\n",
    "    if [ $p == \"single-end\" ]; then\n",
    "        if [ $strand == \"firststrand\" ]; then\n",
    "            featureCounts -T 10 -s 2 -g transcript_id -O -a $AnnotGTF -o ${CLUSTERDIR}/gffcompare_stranded_featureCounts.txt $BAMDIR/*Aligned.sortedByCoord.out.bam \n",
    "        elif [ $strand == \"secondstrand\" ]; then\n",
    "            featureCounts -T 10 -s 1 -g transcript_id -O -a $AnnotGTF -o ${CLUSTERDIR}/gffcompare_stranded_featureCounts.txt $BAMDIR/*Aligned.sortedByCoord.out.bam \n",
    "        elif [ $strand == \"unstranded\" ]; then\n",
    "            featureCounts -T 10 -s 0 -g transcript_id -O -a $AnnotGTF -o ${CLUSTERDIR}/gffcompare_stranded_featureCounts.txt $BAMDIR/*Aligned.sortedByCoord.out.bam \n",
    "        fi\n",
    "    fi\n",
    "fi\n"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## TPMs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "metadata": {},
   "outputs": [],
   "source": [
    "from rna_seq_normalization import Normalization as Norm"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "25\n"
     ]
    }
   ],
   "source": [
    "table_of_counts = pd.DataFrame()\n",
    "### modify headers\n",
    "\n",
    "df=pd.read_csv(os.path.join(users_dir,specie,\"featureCounts_gffcompare/gffcompare_stranded_featureCounts.txt\"), skiprows=1, sep=\"\\t\")\n",
    "df['Geneid']=df['Geneid'].str.split('.').str[0]\n",
    "filter_col = [col for col in df if col.startswith('/')]\n",
    "for col in filter_col:\n",
    "    new_col=col.split(\"Aligned\")[0]\n",
    "    new_col=new_col.split(\"/\")[-1]\n",
    "\n",
    "    df.rename(columns={col:new_col}, inplace=True)\n",
    "    \n",
    "cols_to_select = [\"Geneid\",\"Chr\",\"Start\",\"End\",\"Strand\",\"Length\"]\n",
    "\n",
    "print(len(df.columns))\n",
    "df.rename(columns={'Geneid':'gene_id'}, inplace=True)\n",
    "\n",
    "if table_of_counts.empty:\n",
    "    table_of_counts = df\n",
    "else:\n",
    "    table_of_counts = table_of_counts.merge(df, on=[\"gene_id\",\"Chr\",\"Start\",\"End\",\"Strand\",\"Length\"])\n",
    "\n",
    "table_of_counts.to_csv(os.path.join(users_dir,specie,\"featureCounts_gffcompare/table_of_counts.csv\"), index=None)\n",
    "\n",
    "# get TPMs\n",
    "length = table_of_counts['Length']\n",
    "genes = table_of_counts['gene_id']\n",
    "# we are only interested in the columns with counts\n",
    "counts = table_of_counts\n",
    "counts.drop([\"Chr\",\"Start\",\"End\",\"Strand\",\"Length\",\"gene_id\"],axis=1, inplace=True)\n",
    "# calculate TPMs\n",
    "tpm_df = Norm.tpm(counts, length)\n",
    "# add gene_names again\n",
    "tpms = pd.concat([genes,tpm_df], axis=1)\n",
    "tpms.to_csv(os.path.join(users_dir,specie,\"featureCounts_gffcompare/table_of_counts_TPMs.csv\"), index=None)\n",
    "# add length again\n",
    "tpms_length = pd.concat([genes,length,tpm_df], axis=1)\n",
    "tpms_length.to_csv(os.path.join(users_dir,specie,\"featureCounts_gffcompare/table_of_counts_TPMs_withLength.csv\"), index=None)"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Run RiboNovel with the same annotation"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.16"
  },
  "orig_nbformat": 4
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
