{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Immunopeptidomics new search\n",
    "\n",
    "With alterantive ORFs (uORFs, dORFs...)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chr</th>\n",
       "      <th>transcript_id</th>\n",
       "      <th>gene_id</th>\n",
       "      <th>gene_name</th>\n",
       "      <th>transcript_type</th>\n",
       "      <th>gene_type</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>M</td>\n",
       "      <td>ENST00000361390</td>\n",
       "      <td>ENSG00000198888</td>\n",
       "      <td>MT-ND1</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>protein_coding</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>M</td>\n",
       "      <td>ENST00000361453</td>\n",
       "      <td>ENSG00000198763</td>\n",
       "      <td>MT-ND2</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>protein_coding</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>M</td>\n",
       "      <td>ENST00000361624</td>\n",
       "      <td>ENSG00000198804</td>\n",
       "      <td>MT-CO1</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>protein_coding</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>M</td>\n",
       "      <td>ENST00000361739</td>\n",
       "      <td>ENSG00000198712</td>\n",
       "      <td>MT-CO2</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>protein_coding</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>M</td>\n",
       "      <td>ENST00000361851</td>\n",
       "      <td>ENSG00000228253</td>\n",
       "      <td>MT-ATP8</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>protein_coding</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>65159</th>\n",
       "      <td>KI270442.1</td>\n",
       "      <td>TCONS_00000004</td>\n",
       "      <td>XLOC_000003</td>\n",
       "      <td>XLOC_000003</td>\n",
       "      <td>novel</td>\n",
       "      <td>novel</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>65160</th>\n",
       "      <td>KI270442.1</td>\n",
       "      <td>TCONS_00000005</td>\n",
       "      <td>XLOC_000004</td>\n",
       "      <td>XLOC_000004</td>\n",
       "      <td>novel</td>\n",
       "      <td>novel</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>65161</th>\n",
       "      <td>KI270466.1</td>\n",
       "      <td>TCONS_00000006</td>\n",
       "      <td>XLOC_000005</td>\n",
       "      <td>XLOC_000005</td>\n",
       "      <td>novel</td>\n",
       "      <td>novel</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>65162</th>\n",
       "      <td>KI270467.1</td>\n",
       "      <td>TCONS_00000008</td>\n",
       "      <td>XLOC_000006</td>\n",
       "      <td>XLOC_000006</td>\n",
       "      <td>novel</td>\n",
       "      <td>novel</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>65163</th>\n",
       "      <td>KI270467.1</td>\n",
       "      <td>TCONS_00000007</td>\n",
       "      <td>XLOC_000007</td>\n",
       "      <td>XLOC_000007</td>\n",
       "      <td>novel</td>\n",
       "      <td>novel</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>65164 rows × 6 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "              chr    transcript_id          gene_id    gene_name  \\\n",
       "0               M  ENST00000361390  ENSG00000198888       MT-ND1   \n",
       "1               M  ENST00000361453  ENSG00000198763       MT-ND2   \n",
       "2               M  ENST00000361624  ENSG00000198804       MT-CO1   \n",
       "3               M  ENST00000361739  ENSG00000198712       MT-CO2   \n",
       "4               M  ENST00000361851  ENSG00000228253      MT-ATP8   \n",
       "...           ...              ...              ...          ...   \n",
       "65159  KI270442.1   TCONS_00000004      XLOC_000003  XLOC_000003   \n",
       "65160  KI270442.1   TCONS_00000005      XLOC_000004  XLOC_000004   \n",
       "65161  KI270466.1   TCONS_00000006      XLOC_000005  XLOC_000005   \n",
       "65162  KI270467.1   TCONS_00000008      XLOC_000006  XLOC_000006   \n",
       "65163  KI270467.1   TCONS_00000007      XLOC_000007  XLOC_000007   \n",
       "\n",
       "      transcript_type       gene_type  \n",
       "0      protein_coding  protein_coding  \n",
       "1      protein_coding  protein_coding  \n",
       "2      protein_coding  protein_coding  \n",
       "3      protein_coding  protein_coding  \n",
       "4      protein_coding  protein_coding  \n",
       "...               ...             ...  \n",
       "65159           novel           novel  \n",
       "65160           novel           novel  \n",
       "65161           novel           novel  \n",
       "65162           novel           novel  \n",
       "65163           novel           novel  \n",
       "\n",
       "[65164 rows x 6 columns]"
      ]
     },
     "execution_count": 1,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import os, re\n",
    "import pandas as pd\n",
    "from Bio import SeqIO\n",
    "from Bio.Seq import Seq  # Import Seq from Bio.Seq\n",
    "\n",
    "annot=pd.read_csv(\"/users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/human/newReference_Resconstructed/transID_geneID_isoforms_selected.1to1.csv\")\n",
    "annot"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "HCC_projs = [\"GSE193567\", \"GSE214846\", \"liver_adjacent_totalRNA\", \"hcc_normal_totalRNA\", \"LIHC_TCGA\"]\n",
    "\n",
    "all_genes = pd.DataFrame()\n",
    "\n",
    "for hcc_dataset in HCC_projs:\n",
    "    genes = pd.read_csv(\"/users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/cancers/tumorexpressed/tumor_1FPKM_n_\"+hcc_dataset+\"_LIHC.csv\")\n",
    "    genes = genes[[\"gene_id\",\"transcript_id\",\"gene_name\",\"gene_type\"]]\n",
    "\n",
    "    all_genes = pd.concat([all_genes, genes])\n",
    "\n",
    "all_genes.drop_duplicates(inplace=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene_id</th>\n",
       "      <th>transcript_id</th>\n",
       "      <th>gene_name</th>\n",
       "      <th>gene_type</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>XLOC_001021</td>\n",
       "      <td>TCONS_00001163</td>\n",
       "      <td>XLOC_001021</td>\n",
       "      <td>novel</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ENSG00000197958</td>\n",
       "      <td>ENST00000361436</td>\n",
       "      <td>RPL12</td>\n",
       "      <td>protein_coding</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ENSG00000136933</td>\n",
       "      <td>ENST00000373544</td>\n",
       "      <td>RABEPK</td>\n",
       "      <td>protein_coding</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ENSG00000044574</td>\n",
       "      <td>ENST00000324460</td>\n",
       "      <td>HSPA5</td>\n",
       "      <td>protein_coding</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ENSG00000165219</td>\n",
       "      <td>ENST00000297933</td>\n",
       "      <td>GAPVD1</td>\n",
       "      <td>protein_coding</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26984</th>\n",
       "      <td>ENSG00000288860</td>\n",
       "      <td>ENST00000689333</td>\n",
       "      <td>ENSG00000288860</td>\n",
       "      <td>lncRNA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27004</th>\n",
       "      <td>ENSG00000307634</td>\n",
       "      <td>ENST00000827594</td>\n",
       "      <td>ENSG00000307634</td>\n",
       "      <td>lncRNA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27012</th>\n",
       "      <td>ENSG00000154975</td>\n",
       "      <td>ENST00000451037</td>\n",
       "      <td>CA10</td>\n",
       "      <td>protein_coding</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27013</th>\n",
       "      <td>ENSG00000308371</td>\n",
       "      <td>ENST00000833657</td>\n",
       "      <td>ENSG00000308371</td>\n",
       "      <td>lncRNA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27018</th>\n",
       "      <td>ENSG00000301784</td>\n",
       "      <td>ENST00000781801</td>\n",
       "      <td>ENSG00000301784</td>\n",
       "      <td>lncRNA</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>39125 rows × 4 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "               gene_id    transcript_id        gene_name       gene_type\n",
       "0          XLOC_001021   TCONS_00001163      XLOC_001021           novel\n",
       "1      ENSG00000197958  ENST00000361436            RPL12  protein_coding\n",
       "2      ENSG00000136933  ENST00000373544           RABEPK  protein_coding\n",
       "3      ENSG00000044574  ENST00000324460            HSPA5  protein_coding\n",
       "4      ENSG00000165219  ENST00000297933           GAPVD1  protein_coding\n",
       "...                ...              ...              ...             ...\n",
       "26984  ENSG00000288860  ENST00000689333  ENSG00000288860          lncRNA\n",
       "27004  ENSG00000307634  ENST00000827594  ENSG00000307634          lncRNA\n",
       "27012  ENSG00000154975  ENST00000451037             CA10  protein_coding\n",
       "27013  ENSG00000308371  ENST00000833657  ENSG00000308371          lncRNA\n",
       "27018  ENSG00000301784  ENST00000781801  ENSG00000301784          lncRNA\n",
       "\n",
       "[39125 rows x 4 columns]"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "all_genes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "## Create fasta of translated sequences from candidates\n",
    "specie = \"human\"\n",
    "fasta=os.path.join(\"/users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47\",specie,\"RiboNovel_MultMap_1to1/Annotation/candidateORF.fa\")\n",
    "fasta_df_input = pd.read_csv(fasta, sep=\"\\t\", header=None)\n",
    "fasta_df = pd.DataFrame({'header':fasta_df_input[0].iloc[::2].values, 'seq':fasta_df_input[0].iloc[1::2].values})\n",
    "fasta_df['header'] = fasta_df.header.str[1:]\n",
    "fasta_df['transcript_id'] = fasta_df['header'].str.split(\":\", expand=True)[0]\n",
    "fasta_df['transcript_id'] = fasta_df['transcript_id'].str.split(\".\", expand=True)[0]\n",
    "fasta_df['coords'] = fasta_df['header'].str.split(\"\\|\", expand=True)[2]\n",
    "fasta_df['ORFtype'] = fasta_df['header'].str.split(\"\\|\", expand=True)[3]\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "fasta_df = fasta_df.merge(annot, on=\"transcript_id\")\n",
    "\n",
    "fasta_df['new_header'] = fasta_df['header']+\":\"+fasta_df['gene_name']+\":\"+fasta_df['gene_type']\n",
    "fasta_df.groupby(\"gene_type\").count()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "candidates_TAA = fasta_df[fasta_df['transcript_id'].isin(all_genes.transcript_id.values.tolist())]\n",
    "orftypes=[\"noncoding\",\"uORF\",\"dORF\",\"ouORF\",\"odORF\"]\n",
    "\n",
    "candidates_TAA_notcanonical = candidates_TAA[candidates_TAA[\"ORFtype\"].isin(orftypes)]\n",
    "candidates_TAA_notcanonical = candidates_TAA_notcanonical[~candidates_TAA_notcanonical[\"chr\"].str.contains(\"GL\")]\n",
    "candidates_TAA_notcanonical = candidates_TAA_notcanonical[~candidates_TAA_notcanonical[\"chr\"].str.contains(\"KI\")]\n",
    "\n",
    "candidates_TAA_notcanonical.groupby(\"gene_type\").count()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "to_fasta = candidates_TAA_notcanonical[['new_header','seq']]\n",
    "\n",
    "TAA_prot = \"/home/marta/241205_Immunopeptidomics_HCC/database/TAA.Wang2020candidates.PROTEIN.fa\"\n",
    "out_prot = open(TAA_prot,'w')\n",
    "\n",
    "for index, row in to_fasta.iterrows():\n",
    "    nucleotide_seq = Seq(row['seq'])  # Convert string to Biopython Seq object\n",
    "    protein_seq = nucleotide_seq.translate()  # Translate nucleotide to protein\n",
    "\n",
    "    out_prot.write(\">%s\\n%s\\n\" %(str(row['new_header']), str(protein_seq)[:-1]))\n",
    "\n",
    "out_prot.close()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "## remove redundancies of nested sequences\n",
    "from collections import defaultdict\n",
    "\n",
    "def remove_nested_sequences_by_transcript(fasta_file):\n",
    "    sequences = defaultdict(list)  # A dictionary to hold sequences by transcript ID\n",
    "    headers = []  # List to store the original headers\n",
    "    transcript_ids = []  # List to store the transcript IDs for each sequence\n",
    "    \n",
    "    # Read the sequences from the FASTA file\n",
    "    with open(fasta_file, 'r') as f:\n",
    "        header = None\n",
    "        sequence = []\n",
    "        \n",
    "        for line in f:\n",
    "            line = line.strip()\n",
    "            if line.startswith('>'):\n",
    "                if header is not None:\n",
    "                    # Store the sequence and header\n",
    "                    sequences[transcript_id].append((\"\".join(sequence), header))\n",
    "                header = line[1:]  # Remove the '>' symbol\n",
    "                sequence = []\n",
    "                transcript_id = header.split(':')[0]  # Extract the transcript ID\n",
    "            else:\n",
    "                sequence.append(line)\n",
    "        \n",
    "        # Add the last sequence\n",
    "        if header is not None:\n",
    "            sequences[transcript_id].append((\"\".join(sequence), header))\n",
    "    \n",
    "    # List to hold non-redundant sequences\n",
    "    unique_sequences = []\n",
    "    \n",
    "    # Process each transcript group\n",
    "    for transcript_id, seq_list in sequences.items():\n",
    "        # Sort sequences by length in descending order\n",
    "        seq_list.sort(key=lambda x: len(x[0]), reverse=True)\n",
    "        \n",
    "        # Iterate over sorted sequences, keeping only the longest non-redundant ones\n",
    "        for i, (seq_i, header_i) in enumerate(seq_list):\n",
    "            is_redundant = False\n",
    "            # Compare with previously added sequences within the same transcript\n",
    "            for j in range(i):\n",
    "                seq_j, _ = seq_list[j]\n",
    "                if seq_i in seq_j:\n",
    "                    is_redundant = True\n",
    "                    break\n",
    "            \n",
    "            # If not redundant, add the sequence to the output list\n",
    "            if not is_redundant:\n",
    "                unique_sequences.append((header_i, seq_i))\n",
    "    \n",
    "    # Write the output FASTA file\n",
    "    output_file = '/home/marta/241205_Immunopeptidomics_HCC/database/TAA.Wang2020candidates.PROTEIN.NOredundant.fa'\n",
    "    with open(output_file, 'w') as f:\n",
    "        for header, seq in unique_sequences:\n",
    "            f.write(f\">{header}\\n{seq}\\n\")\n",
    "    \n",
    "    print(f\"Non-redundant sequences saved to {output_file}\")\n",
    "\n",
    "# Example usage\n",
    "fasta_file = '/home/marta/241205_Immunopeptidomics_HCC/database/TAA.Wang2020candidates.PROTEIN.fa'\n",
    "remove_nested_sequences_by_transcript(fasta_file)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash \n",
    "\n",
    "DIR=/home/marta/241205_Immunopeptidomics_HCC\n",
    "\n",
    "## TAA + SP/TREMBL\n",
    "cat /home/marta/241205_Immunopeptidomics_HCC/database/uniprot_canonicalANDisoforms_oneline.fasta /home/marta/241205_Immunopeptidomics_HCC/database/TAA.Wang2020candidates.PROTEIN.NOredundant.fa > /home/marta/241205_Immunopeptidomics_HCC/database/TAA_SPTREMBL.fasta\n",
    "## remove empty lines\n",
    "sed -i '/^$/d' /home/marta/241205_Immunopeptidomics_HCC/database/TAA_SPTREMBL.fasta\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "\n",
    "grep \"^>\" /home/marta/241205_Immunopeptidomics_HCC/database/TAA_SPTREMBL.fasta | grep -v \"sp\" | grep -v \"tr\" | wc -l\n",
    "grep \"^>\" /home/marta/241205_Immunopeptidomics_HCC/database/TAA_SPTREMBL.fasta | grep  \"sp\\|tr\" | wc -l\n",
    "grep \"^>TC\" /home/marta/241205_Immunopeptidomics_HCC/database/TAA_SPTREMBL.fasta | wc -l\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene_id</th>\n",
       "      <th>gene_name</th>\n",
       "      <th>n</th>\n",
       "      <th>gene_type</th>\n",
       "      <th>RiboSeq</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ENSG00000046774</td>\n",
       "      <td>MAGEC2</td>\n",
       "      <td>25.0</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>117</th>\n",
       "      <td>ENSG00000068985</td>\n",
       "      <td>PAGE1</td>\n",
       "      <td>16.0</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>234</th>\n",
       "      <td>ENSG00000071677</td>\n",
       "      <td>PRLH</td>\n",
       "      <td>1.0</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>351</th>\n",
       "      <td>ENSG00000082929</td>\n",
       "      <td>LINC01587</td>\n",
       "      <td>3.0</td>\n",
       "      <td>lncRNA</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>468</th>\n",
       "      <td>ENSG00000083622</td>\n",
       "      <td>AC000061.1</td>\n",
       "      <td>1.0</td>\n",
       "      <td>lncRNA</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>160641</th>\n",
       "      <td>ENSG00000283436</td>\n",
       "      <td>LINC01958</td>\n",
       "      <td>3.0</td>\n",
       "      <td>lncRNA</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>160758</th>\n",
       "      <td>ENSG00000283480</td>\n",
       "      <td>AL512380.2</td>\n",
       "      <td>2.0</td>\n",
       "      <td>lncRNA</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>160875</th>\n",
       "      <td>ENSG00000283573</td>\n",
       "      <td>AL157371.2</td>\n",
       "      <td>2.0</td>\n",
       "      <td>lncRNA</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>160992</th>\n",
       "      <td>ENSG00000283599</td>\n",
       "      <td>BX276092.9</td>\n",
       "      <td>13.0</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>161109</th>\n",
       "      <td>ENSG00000283740</td>\n",
       "      <td>TAF11L11</td>\n",
       "      <td>1.0</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1376 rows × 5 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                gene_id   gene_name     n       gene_type  RiboSeq\n",
       "0       ENSG00000046774      MAGEC2  25.0  protein_coding        1\n",
       "117     ENSG00000068985       PAGE1  16.0  protein_coding        1\n",
       "234     ENSG00000071677        PRLH   1.0  protein_coding        1\n",
       "351     ENSG00000082929   LINC01587   3.0          lncRNA        0\n",
       "468     ENSG00000083622  AC000061.1   1.0          lncRNA        0\n",
       "...                 ...         ...   ...             ...      ...\n",
       "160641  ENSG00000283436   LINC01958   3.0          lncRNA        0\n",
       "160758  ENSG00000283480  AL512380.2   2.0          lncRNA        0\n",
       "160875  ENSG00000283573  AL157371.2   2.0          lncRNA        0\n",
       "160992  ENSG00000283599  BX276092.9  13.0  protein_coding        1\n",
       "161109  ENSG00000283740    TAF11L11   1.0  protein_coding        1\n",
       "\n",
       "[1376 rows x 5 columns]"
      ]
     },
     "execution_count": 2,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "TSASciAdv = pd.read_csv(\"/projects_eg/projects/marta/table_to_heatmap_noabundantcase3.csv\")\n",
    "TSASciAdv = TSASciAdv[['gene_id','gene_name','n','gene_type','RiboSeq']].drop_duplicates()\n",
    "TSASciAdv"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene_id</th>\n",
       "      <th>orfID_TSTR_RiboSeq</th>\n",
       "      <th>ctype</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ENSG00000117148</td>\n",
       "      <td></td>\n",
       "      <td>BRCA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ENSG00000117148</td>\n",
       "      <td></td>\n",
       "      <td>BLCA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ENSG00000117148</td>\n",
       "      <td></td>\n",
       "      <td>LUAD</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>ENSG00000117148</td>\n",
       "      <td></td>\n",
       "      <td>KIRC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>ENSG00000117148</td>\n",
       "      <td></td>\n",
       "      <td>PRAD</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1062</th>\n",
       "      <td>XLOC_000189</td>\n",
       "      <td>TCONS_00000243:11:+|5|322:61:301|noncoding|ATG</td>\n",
       "      <td>COAD</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1063</th>\n",
       "      <td>XLOC_000280</td>\n",
       "      <td>TCONS_00000334:13:+|6|443:103:166|noncoding|ATG</td>\n",
       "      <td>COAD</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1064</th>\n",
       "      <td>XLOC_000797</td>\n",
       "      <td>TCONS_00000933:5:+|46|6032:841:889|noncoding|ATG</td>\n",
       "      <td>COAD</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1065</th>\n",
       "      <td>ENSG00000241560</td>\n",
       "      <td>ENST00000467304.2:3:+|11|4530:178:454|noncodin...</td>\n",
       "      <td>COAD</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1066</th>\n",
       "      <td>ENSG00000241560</td>\n",
       "      <td>ENST00000467304.2:3:+|3|4530:41:146|noncoding|ATG</td>\n",
       "      <td>COAD</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>867 rows × 3 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "              gene_id                                 orfID_TSTR_RiboSeq ctype\n",
       "0     ENSG00000117148                                                     BRCA\n",
       "2     ENSG00000117148                                                     BLCA\n",
       "4     ENSG00000117148                                                     LUAD\n",
       "6     ENSG00000117148                                                     KIRC\n",
       "8     ENSG00000117148                                                     PRAD\n",
       "...               ...                                                ...   ...\n",
       "1062      XLOC_000189     TCONS_00000243:11:+|5|322:61:301|noncoding|ATG  COAD\n",
       "1063      XLOC_000280    TCONS_00000334:13:+|6|443:103:166|noncoding|ATG  COAD\n",
       "1064      XLOC_000797   TCONS_00000933:5:+|46|6032:841:889|noncoding|ATG  COAD\n",
       "1065  ENSG00000241560  ENST00000467304.2:3:+|11|4530:178:454|noncodin...  COAD\n",
       "1066  ENSG00000241560  ENST00000467304.2:3:+|3|4530:41:146|noncoding|ATG  COAD\n",
       "\n",
       "[867 rows x 3 columns]"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# TSTR = pd.read_csv(\"/users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/cancers/log2ratio3x/cancertypes/TOv3x_5percent_TestisRestrictedGTEx_Translated_Ctypes_log2ratio3xMEANgenes.csv\")\n",
    "TSTR = pd.read_csv(\"/users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/cancers/log2ratio3x/cancertypes/onlyStep1/log2ratio3x_1TPM_ORFs.csv\")\n",
    "TSTR = TSTR[['gene_id','gene_type','orfID','ctype']].drop_duplicates()\n",
    "TSTR.columns = [\"gene_id\",\"gene_type\",\"orfID_TSTR_RiboSeq\",\"ctype\"]\n",
    "# Update 'orfID_TSTR_RiboSeq' to empty if 'gene_type' is \"protein_coding\"\n",
    "TSTR.loc[TSTR['gene_type'] == 'protein_coding', 'orfID_TSTR_RiboSeq'] = ''\n",
    "TSTR.drop(\"gene_type\", axis=1, inplace=True)\n",
    "TSTR.drop_duplicates(inplace=True)\n",
    "TSTR"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>header</th>\n",
       "      <th>Entry</th>\n",
       "      <th>gene_name</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>&gt;tr|A0A075B6G3|A0A075B6G3_HUMAN Dystrophin OS=...</td>\n",
       "      <td>A0A075B6G3</td>\n",
       "      <td>DMD</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>&gt;tr|A0A087WV00|A0A087WV00_HUMAN Diacylglycerol...</td>\n",
       "      <td>A0A087WV00</td>\n",
       "      <td>DGKI</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>&gt;tr|A0A087WZT3|A0A087WZT3_HUMAN BOLA2-SMG1P6 r...</td>\n",
       "      <td>A0A087WZT3</td>\n",
       "      <td>BOLA2-SMG1P6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>&gt;sp|A0A087X1C5|CP2D7_HUMAN Putative cytochrome...</td>\n",
       "      <td>A0A087X1C5</td>\n",
       "      <td>CYP2D7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>&gt;tr|A0A0A0MQU7|A0A0A0MQU7_HUMAN HNF1 homeobox ...</td>\n",
       "      <td>A0A0A0MQU7</td>\n",
       "      <td>HNF1A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>131557</th>\n",
       "      <td>&gt;tr|X6RLU5|X6RLU5_HUMAN Calcium voltage-gated ...</td>\n",
       "      <td>X6RLU5</td>\n",
       "      <td>CACNA2D4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>131558</th>\n",
       "      <td>&gt;tr|X6RLV5|X6RLV5_HUMAN DEAD-box helicase 5 OS...</td>\n",
       "      <td>X6RLV5</td>\n",
       "      <td>DDX5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>131559</th>\n",
       "      <td>&gt;tr|X6RLY7|X6RLY7_HUMAN Calcium voltage-gated ...</td>\n",
       "      <td>X6RLY7</td>\n",
       "      <td>CACNA2D4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>131560</th>\n",
       "      <td>&gt;tr|X6RM00|X6RM00_HUMAN ELKS/RAB6-interacting/...</td>\n",
       "      <td>X6RM00</td>\n",
       "      <td>ERC1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>131561</th>\n",
       "      <td>&gt;tr|X6RM24|X6RM24_HUMAN Enolase superfamily me...</td>\n",
       "      <td>X6RM24</td>\n",
       "      <td>ENOSF1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>131562 rows × 3 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                   header       Entry  \\\n",
       "0       >tr|A0A075B6G3|A0A075B6G3_HUMAN Dystrophin OS=...  A0A075B6G3   \n",
       "1       >tr|A0A087WV00|A0A087WV00_HUMAN Diacylglycerol...  A0A087WV00   \n",
       "2       >tr|A0A087WZT3|A0A087WZT3_HUMAN BOLA2-SMG1P6 r...  A0A087WZT3   \n",
       "3       >sp|A0A087X1C5|CP2D7_HUMAN Putative cytochrome...  A0A087X1C5   \n",
       "4       >tr|A0A0A0MQU7|A0A0A0MQU7_HUMAN HNF1 homeobox ...  A0A0A0MQU7   \n",
       "...                                                   ...         ...   \n",
       "131557  >tr|X6RLU5|X6RLU5_HUMAN Calcium voltage-gated ...      X6RLU5   \n",
       "131558  >tr|X6RLV5|X6RLV5_HUMAN DEAD-box helicase 5 OS...      X6RLV5   \n",
       "131559  >tr|X6RLY7|X6RLY7_HUMAN Calcium voltage-gated ...      X6RLY7   \n",
       "131560  >tr|X6RM00|X6RM00_HUMAN ELKS/RAB6-interacting/...      X6RM00   \n",
       "131561  >tr|X6RM24|X6RM24_HUMAN Enolase superfamily me...      X6RM24   \n",
       "\n",
       "           gene_name  \n",
       "0                DMD  \n",
       "1               DGKI  \n",
       "2       BOLA2-SMG1P6  \n",
       "3             CYP2D7  \n",
       "4              HNF1A  \n",
       "...              ...  \n",
       "131557      CACNA2D4  \n",
       "131558          DDX5  \n",
       "131559      CACNA2D4  \n",
       "131560          ERC1  \n",
       "131561        ENOSF1  \n",
       "\n",
       "[131562 rows x 3 columns]"
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "annot = pd.read_csv(\"/users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/human/newReference_Resconstructed/transID_geneID_isoforms_selected.1to1.csv\")\n",
    "uniprotKB = pd.read_csv(\"/users/genomics/marta/HCC_proteomics/uniprotkb_AND_model_organism_9606_2024_02_28.tsv\", sep=\"\\t\")\n",
    "uniprotKB['Gene Names'] = uniprotKB['Gene Names'].str.split()\n",
    "uniprotKB = uniprotKB.explode('Gene Names')\n",
    "\n",
    "uniprotKB = uniprotKB[['Entry','Gene Names']]\n",
    "\n",
    "## database canonical\n",
    "sp_TrEMBL = pd.read_csv(\"/users/genomics/marta/241205_Immunopeptidomics_HCC/database/header.NOredundant.fa\", header=None, sep=\"\\t\")\n",
    "sp_TrEMBL['Entry'] = sp_TrEMBL[0].str.split(\"\\|\", expand=True)[1]\n",
    "sp_TrEMBL = sp_TrEMBL[~sp_TrEMBL[0].str.contains(\"ENST\")]\n",
    "sp_TrEMBL = sp_TrEMBL[~sp_TrEMBL[0].str.contains(\"TCONS\")]\n",
    "\n",
    "sp_TrEMBL = sp_TrEMBL.merge(uniprotKB, on=\"Entry\", how=\"left\")\n",
    "sp_TrEMBL.columns = ['header','Entry','gene_name']\n",
    "sp_TrEMBL.to_csv(\"/users/genomics/marta/241205_Immunopeptidomics_HCC/database/canonicalheaders_entries_genenames.csv\", index=None)\n",
    "\n",
    "sp_TrEMBL"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "46331\n",
      "19707\n"
     ]
    }
   ],
   "source": [
    "proteinNames = pd.read_csv(\"/users/genomics/marta/241205_Immunopeptidomics_HCC/database/canonicalheaders_entries_genenames.csv\")\n",
    "proteinNames.drop(\"header\", axis=1, inplace=True)\n",
    "print(len(set(proteinNames.gene_name.values.tolist())))\n",
    "\n",
    "## associate entry name with gene_name\n",
    "annotated_geneNames = annot.merge(proteinNames, on=\"gene_name\")\n",
    "print(len(set(annotated_geneNames.gene_name.values.tolist())))\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chr</th>\n",
       "      <th>transcript_id</th>\n",
       "      <th>gene_id</th>\n",
       "      <th>gene_name</th>\n",
       "      <th>transcript_type</th>\n",
       "      <th>gene_type</th>\n",
       "      <th>Entry</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>M</td>\n",
       "      <td>ENST00000361390</td>\n",
       "      <td>ENSG00000198888</td>\n",
       "      <td>MT-ND1</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>P03886</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>M</td>\n",
       "      <td>ENST00000361453</td>\n",
       "      <td>ENSG00000198763</td>\n",
       "      <td>MT-ND2</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>P03891</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>M</td>\n",
       "      <td>ENST00000361624</td>\n",
       "      <td>ENSG00000198804</td>\n",
       "      <td>MT-CO1</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>P00395</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>M</td>\n",
       "      <td>ENST00000361739</td>\n",
       "      <td>ENSG00000198712</td>\n",
       "      <td>MT-CO2</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>P00403</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>M</td>\n",
       "      <td>ENST00000361851</td>\n",
       "      <td>ENSG00000228253</td>\n",
       "      <td>MT-ATP8</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>P03928</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>80180</th>\n",
       "      <td>22</td>\n",
       "      <td>ENST00000395595</td>\n",
       "      <td>ENSG00000079974</td>\n",
       "      <td>RABL2B</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>C9JFZ0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>80181</th>\n",
       "      <td>22</td>\n",
       "      <td>ENST00000395595</td>\n",
       "      <td>ENSG00000079974</td>\n",
       "      <td>RABL2B</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>F2Z2T3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>80182</th>\n",
       "      <td>22</td>\n",
       "      <td>ENST00000395595</td>\n",
       "      <td>ENSG00000079974</td>\n",
       "      <td>RABL2B</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>F2Z3A9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>80183</th>\n",
       "      <td>22</td>\n",
       "      <td>ENST00000395595</td>\n",
       "      <td>ENSG00000079974</td>\n",
       "      <td>RABL2B</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>F2Z3J7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>80184</th>\n",
       "      <td>GL000194.1</td>\n",
       "      <td>ENST00000400754</td>\n",
       "      <td>ENSG00000274847</td>\n",
       "      <td>MAFIP</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>Q8WZ33</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>80185 rows × 7 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "              chr    transcript_id          gene_id gene_name transcript_type  \\\n",
       "0               M  ENST00000361390  ENSG00000198888    MT-ND1  protein_coding   \n",
       "1               M  ENST00000361453  ENSG00000198763    MT-ND2  protein_coding   \n",
       "2               M  ENST00000361624  ENSG00000198804    MT-CO1  protein_coding   \n",
       "3               M  ENST00000361739  ENSG00000198712    MT-CO2  protein_coding   \n",
       "4               M  ENST00000361851  ENSG00000228253   MT-ATP8  protein_coding   \n",
       "...           ...              ...              ...       ...             ...   \n",
       "80180          22  ENST00000395595  ENSG00000079974    RABL2B  protein_coding   \n",
       "80181          22  ENST00000395595  ENSG00000079974    RABL2B  protein_coding   \n",
       "80182          22  ENST00000395595  ENSG00000079974    RABL2B  protein_coding   \n",
       "80183          22  ENST00000395595  ENSG00000079974    RABL2B  protein_coding   \n",
       "80184  GL000194.1  ENST00000400754  ENSG00000274847     MAFIP  protein_coding   \n",
       "\n",
       "            gene_type   Entry  \n",
       "0      protein_coding  P03886  \n",
       "1      protein_coding  P03891  \n",
       "2      protein_coding  P00395  \n",
       "3      protein_coding  P00403  \n",
       "4      protein_coding  P03928  \n",
       "...               ...     ...  \n",
       "80180  protein_coding  C9JFZ0  \n",
       "80181  protein_coding  F2Z2T3  \n",
       "80182  protein_coding  F2Z3A9  \n",
       "80183  protein_coding  F2Z3J7  \n",
       "80184  protein_coding  Q8WZ33  \n",
       "\n",
       "[80185 rows x 7 columns]"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "annotated_geneNames"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene_name</th>\n",
       "      <th>transcript_id</th>\n",
       "      <th>n</th>\n",
       "      <th>Entry</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ACTL8</td>\n",
       "      <td>ENST00000375406</td>\n",
       "      <td>8.0</td>\n",
       "      <td>Q9H568</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>CALR3</td>\n",
       "      <td>ENST00000269881</td>\n",
       "      <td>2.0</td>\n",
       "      <td>Q96L12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>CALR3</td>\n",
       "      <td>ENST00000269881</td>\n",
       "      <td>2.0</td>\n",
       "      <td>M0R0Y8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>CCDC185</td>\n",
       "      <td>ENST00000366875</td>\n",
       "      <td>1.0</td>\n",
       "      <td>Q8N715</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>CETN1</td>\n",
       "      <td>ENST00000327228</td>\n",
       "      <td>1.0</td>\n",
       "      <td>Q12798</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>376</th>\n",
       "      <td>OR12D2</td>\n",
       "      <td>ENST00000642051</td>\n",
       "      <td>2.0</td>\n",
       "      <td>P58182</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>377</th>\n",
       "      <td>OR12D2</td>\n",
       "      <td>ENST00000642051</td>\n",
       "      <td>2.0</td>\n",
       "      <td>A0A140T931</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>378</th>\n",
       "      <td>OR12D2</td>\n",
       "      <td>ENST00000642051</td>\n",
       "      <td>2.0</td>\n",
       "      <td>A0A140T9F1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>379</th>\n",
       "      <td>CCDC195</td>\n",
       "      <td>ENST00000638102</td>\n",
       "      <td>1.0</td>\n",
       "      <td>A0A1B0GUA6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>380</th>\n",
       "      <td>TAF11L11</td>\n",
       "      <td>ENST00000707170</td>\n",
       "      <td>1.0</td>\n",
       "      <td>A0A1W2PQ09</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>183 rows × 4 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "    gene_name    transcript_id    n       Entry\n",
       "0       ACTL8  ENST00000375406  8.0      Q9H568\n",
       "8       CALR3  ENST00000269881  2.0      Q96L12\n",
       "9       CALR3  ENST00000269881  2.0      M0R0Y8\n",
       "14    CCDC185  ENST00000366875  1.0      Q8N715\n",
       "17      CETN1  ENST00000327228  1.0      Q12798\n",
       "..        ...              ...  ...         ...\n",
       "376    OR12D2  ENST00000642051  2.0      P58182\n",
       "377    OR12D2  ENST00000642051  2.0  A0A140T931\n",
       "378    OR12D2  ENST00000642051  2.0  A0A140T9F1\n",
       "379   CCDC195  ENST00000638102  1.0  A0A1B0GUA6\n",
       "380  TAF11L11  ENST00000707170  1.0  A0A1W2PQ09\n",
       "\n",
       "[183 rows x 4 columns]"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "TSTR_SciAdv = TSTR.merge(TSASciAdv, on=\"gene_id\", how=\"outer\")\n",
    "TSTR_SciAdv\n",
    "\n",
    "PCG_TSTR_SciAdv = TSTR_SciAdv[TSTR_SciAdv['gene_type'] == \"protein_coding\"]\n",
    "PCG_TSTR_SciAdv = PCG_TSTR_SciAdv.merge(annotated_geneNames, on=[\"gene_id\",\"gene_name\",\"gene_type\"])\n",
    "PCG_TSTR_SciAdv = PCG_TSTR_SciAdv[['gene_name','transcript_id','n','Entry']].drop_duplicates()\n",
    "PCG_TSTR_SciAdv"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'Q9H568|Q96L12|M0R0Y8|Q8N715|Q12798|Q5HYN5|Q8NA75|Q86TM3|Q7Z7J5|Q9BXU8|Q6NT46|P09681|Q8N7C0|Q7Z4W2|P43355|P43365|P43357|E7EMU0|O15479|O60732|Q9UBF1|O75459|Q7Z2X7|X6R922|X6RD31|Q9HBJ0|Q01851|Q16384|A0A0J9YWL9|P56180|A0A087X1C4|A6NN14|A0A0G2JQ93|A6NGE4|Q6PJQ5|A8MYU2|E5RGP7|E5RHP1|E5RJA6|Q6ZMV5|A1L453|Q6ZWI9|Q9NXZ1|Q86UG4|C9J020|H0Y8R6|Q01534|A6NC42|P43361|P43366|Q8TBY0|D6RF41|Q6ZQQ2|P0DMU9|P0DPQ3|Q5GH77|Q9BQM9|P11488|C9JCV8|O15480|P00540|A6NJL1|K7EJD1|K7ESD3|Q9H3M9|Q8NEA9|Q5JUK9|O14990|Q9BQY4|Q9BXU2|Q5H9I0|P0DPD5|A8MXY4|P81277|Q9BXT4|H9KV62|H9KV63|O43193|Q86VD1|Q9Y581|A0A3B3ITZ2|Q9BXI2|Q9UNG2|O60811|Q9BYW3|O95972|P49286|E9PR36|H0YDG4|P57773|Q969M2|A0A1W2PS75|P78386|F5GYI5|Q96Q77|M0QZ29|P01270|Q9NYA3|Q9H339|A0A2R8Y7B5|Q8TCW7|Q8WXF3|K7ENX1|P33032|Q7RTU3|Q9NZP2|Q5T619|A0A669KBK7|Q9H1F0|Q8NH53|Q8NH56|P56851|Q14507|A6NP11|A8MPY1|Q9Y5P0|Q6IFG1|Q8N0Y3|Q3LI61|Q8NH54|P59551|Q7Z6I5|Q6ZN17|A0A1B0GTK2|A0A1B0GVD3|Q6ZR62|Q07627|Q9NZK7|Q765I0|C9JU87|F2Z3I1|F8WCV4|A6NIM6|Q6IE38|H3BT63|Q8IYX0|Q9BYQ7|Q8NGY1|Q5VSR9|P0DUQ1|Q5VXH5|Q5W0N0|A0A494C027|A0A494C0F4|A0A494C1R2|Q495Y8|Q96RD3|J3KPH0|Q9H321|Q701N4|Q3LI60|Q3LI62|P0CB33|A6NER3|A6NDE8|P0DMU8|P21108|A0A0B4J207|A0PG75|O60810|A0A1W2PPW3|Q7Z4Y8|A0PK11|P0DJR0|A6NLW8|Q6UXS0|H3BQW0|H3BQX8|P43362|A0A075B794|A0A075B798|A0A075B7A9|Q9Y5P2|A0A1W2PPV0|A0A1W2PQG5|B6SEH9|A0A087X179|A6NER0|P58182|A0A140T931|A0A140T9F1|A0A1B0GUA6|A0A1W2PQ09'"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pattern_coding_genes = '|'.join(PCG_TSTR_SciAdv.Entry.values.tolist())\n",
    "pattern_coding_genes"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Melanoma Cancer, Chong"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\u001b[33mNextflow 24.10.4 is available - Please consider updating your version to it\u001b[m\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "N E X T F L O W  ~  version 23.10.1\n",
      "Launching `https://github.com/nf-core/mhcquant` [elated_boyd] DSL2 - revision: 1b3069246d [master]\n",
      "WARN: Access to undefined parameter `monochromeLogs` -- Initialise it to a default value eg. `params.monochromeLogs = some_value`\n",
      "\n",
      "\n",
      "------------------------------------------------------\n",
      "                                        ,--./,-.\n",
      "        ___     __   __   __   ___     /,-._.--~'\n",
      "  |\\ | |__  __ /  ` /  \\ |__) |__         }  {\n",
      "  | \\| |       \\__, \\__/ |  \\ |___     \\`-._,-`-,\n",
      "                                        `._,._,'\n",
      "  nf-core/mhcquant v2.6.0-g1b30692\n",
      "------------------------------------------------------\n",
      "Core Nextflow options\n",
      "  revision                 : master\n",
      "  runName                  : elated_boyd\n",
      "  containerEngine          : singularity\n",
      "  launchDir                : /home/marta/PROJECT_SCRIPTS/241205_Immunopeptidomics_HCC\n",
      "  workDir                  : /home/marta/PROJECT_SCRIPTS/241205_Immunopeptidomics_HCC/work\n",
      "  projectDir               : /home/marta/.nextflow/assets/nf-core/mhcquant\n",
      "  userName                 : marta\n",
      "  profile                  : singularity\n",
      "  configFiles              : \n",
      "\n",
      "Input/output options\n",
      "  input                    : /home/marta/immunopeptidomics_rawData/samplesheet_OD5P.tsv\n",
      "  outdir                   : /home/marta/241205_Immunopeptidomics_HCC/Melanoma_Chong/FDR5percent/OD5P\n",
      "\n",
      "Database Options\n",
      "  fasta                    : /home/marta/241205_Immunopeptidomics_HCC/database/TAA_SPTREMBL.fasta\n",
      "\n",
      "Database Search Settings\n",
      "  digest_mass_range        : 800:2500\n",
      "  prec_charge              : 2:3\n",
      "  fixed_mods               :  \n",
      "  default_params_file_comet:  \n",
      "\n",
      "Rescoring settings\n",
      "  fdr_threshold            : 0.05\n",
      "\n",
      "!! Only displaying parameters that differ from the pipeline defaults !!\n",
      "------------------------------------------------------\n",
      "If you use nf-core/mhcquant for your analysis please cite:\n",
      "\n",
      "* The pipeline\n",
      "  https://doi.org/10.1021/acs.jproteome.9b00313\n",
      "\n",
      "* The nf-core framework\n",
      "  https://doi.org/10.1038/s41587-020-0439-x\n",
      "\n",
      "* Software dependencies\n",
      "  https://github.com/nf-core/mhcquant/blob/master/CITATIONS.md\n",
      "------------------------------------------------------\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:MS... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:MS... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:MU... -\n",
      "Pulling Singularity image https://depot.galaxyproject.org/singularity/openms:3.1.0--h8964181_3 [cache /home/marta/PROJECT_SCRIPTS/241205_Immunopeptidomics_HCC/work/singularity/depot.galaxyproject.org-singularity-openms-3.1.0--h8964181_3.img]\n",
      "Pulling Singularity image https://depot.galaxyproject.org/singularity/thermorawfileparser:1.4.3--ha8f3691_0 [cache /home/marta/PROJECT_SCRIPTS/241205_Immunopeptidomics_HCC/work/singularity/depot.galaxyproject.org-singularity-thermorawfileparser-1.4.3--ha8f3691_0.img]\n",
      "WARN: Singularity cache directory has not been defined -- Remote image will be stored in the path: /home/marta/PROJECT_SCRIPTS/241205_Immunopeptidomics_HCC/work/singularity -- Use the environment variable NXF_SINGULARITY_CACHEDIR to specify a different location\n",
      "\n",
      "executor >  local (9)\n",
      "[a0/a5ff85] process > NFCORE_MHCQUANT:MHCQUANT:PR... [  0%] 0 of 8\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[65/488afc] process > NFCORE_MHCQUANT:MHCQUANT:OP... [  0%] 0 of 1\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:MS... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:MU... -\n",
      "Pulling Singularity image https://depot.galaxyproject.org/singularity/openms:3.1.0--h8964181_3 [cache /home/marta/PROJECT_SCRIPTS/241205_Immunopeptidomics_HCC/work/singularity/depot.galaxyproject.org-singularity-openms-3.1.0--h8964181_3.img]\n",
      "Pulling Singularity image https://depot.galaxyproject.org/singularity/thermorawfileparser:1.4.3--ha8f3691_0 [cache /home/marta/PROJECT_SCRIPTS/241205_Immunopeptidomics_HCC/work/singularity/depot.galaxyproject.org-singularity-thermorawfileparser-1.4.3--ha8f3691_0.img]\n",
      "WARN: Singularity cache directory has not been defined -- Remote image will be stored in the path: /home/marta/PROJECT_SCRIPTS/241205_Immunopeptidomics_HCC/work/singularity -- Use the environment variable NXF_SINGULARITY_CACHEDIR to specify a different location\n",
      "\n",
      "executor >  local (9)\n",
      "[a0/a5ff85] process > NFCORE_MHCQUANT:MHCQUANT:PR... [  0%] 0 of 8\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[65/488afc] process > NFCORE_MHCQUANT:MHCQUANT:OP... [100%] 1 of 1 ✔\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:MS... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:MU... -\n",
      "Pulling Singularity image https://depot.galaxyproject.org/singularity/openms:3.1.0--h8964181_3 [cache /home/marta/PROJECT_SCRIPTS/241205_Immunopeptidomics_HCC/work/singularity/depot.galaxyproject.org-singularity-openms-3.1.0--h8964181_3.img]\n",
      "Pulling Singularity image https://depot.galaxyproject.org/singularity/thermorawfileparser:1.4.3--ha8f3691_0 [cache /home/marta/PROJECT_SCRIPTS/241205_Immunopeptidomics_HCC/work/singularity/depot.galaxyproject.org-singularity-thermorawfileparser-1.4.3--ha8f3691_0.img]\n",
      "WARN: Singularity cache directory has not been defined -- Remote image will be stored in the path: /home/marta/PROJECT_SCRIPTS/241205_Immunopeptidomics_HCC/work/singularity -- Use the environment variable NXF_SINGULARITY_CACHEDIR to specify a different location\n",
      "\n",
      "executor >  local (9)\n",
      "[c4/948eaa] process > NFCORE_MHCQUANT:MHCQUANT:PR... [ 12%] 1 of 8\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[65/488afc] process > NFCORE_MHCQUANT:MHCQUANT:OP... [100%] 1 of 1 ✔\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:MS... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:MU... -\n",
      "Pulling Singularity image https://depot.galaxyproject.org/singularity/openms-thirdparty:3.1.0--h9ee0642_3 [cache /home/marta/PROJECT_SCRIPTS/241205_Immunopeptidomics_HCC/work/singularity/depot.galaxyproject.org-singularity-openms-thirdparty-3.1.0--h9ee0642_3.img]\n",
      "\n",
      "\n",
      "WARN: Singularity cache directory has not been defined -- Remote image will be stored in the path: /home/marta/PROJECT_SCRIPTS/241205_Immunopeptidomics_HCC/work/singularity -- Use the environment variable NXF_SINGULARITY_CACHEDIR to specify a different location\n",
      "\n",
      "\n",
      "executor >  local (9)\n",
      "[c4/948eaa] process > NFCORE_MHCQUANT:MHCQUANT:PR... [ 12%] 1 of 8\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[65/488afc] process > NFCORE_MHCQUANT:MHCQUANT:OP... [100%] 1 of 1 ✔\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... [  0%] 0 of 1\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:MS... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:MU... -\n",
      "Pulling Singularity image https://depot.galaxyproject.org/singularity/openms-thirdparty:3.1.0--h9ee0642_3 [cache /home/marta/PROJECT_SCRIPTS/241205_Immunopeptidomics_HCC/work/singularity/depot.galaxyproject.org-singularity-openms-thirdparty-3.1.0--h9ee0642_3.img]\n",
      "WARN: Singularity cache directory has not been defined -- Remote image will be stored in the path: /home/marta/PROJECT_SCRIPTS/241205_Immunopeptidomics_HCC/work/singularity -- Use the environment variable NXF_SINGULARITY_CACHEDIR to specify a different location\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "executor >  local (10)\n",
      "[dd/e1f701] process > NFCORE_MHCQUANT:MHCQUANT:PR... [ 25%] 2 of 8\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[65/488afc] process > NFCORE_MHCQUANT:MHCQUANT:OP... [100%] 1 of 1 ✔\n",
      "[7f/0b8433] process > NFCORE_MHCQUANT:MHCQUANT:OP... [  0%] 0 of 2\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:MS... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:MU... -\n",
      "Pulling Singularity image https://depot.galaxyproject.org/singularity/openms-thirdparty:3.1.0--h9ee0642_3 [cache /home/marta/PROJECT_SCRIPTS/241205_Immunopeptidomics_HCC/work/singularity/depot.galaxyproject.org-singularity-openms-thirdparty-3.1.0--h9ee0642_3.img]\n",
      "WARN: Singularity cache directory has not been defined -- Remote image will be stored in the path: /home/marta/PROJECT_SCRIPTS/241205_Immunopeptidomics_HCC/work/singularity -- Use the environment variable NXF_SINGULARITY_CACHEDIR to specify a different location\n",
      "\n",
      "executor >  local (10)\n",
      "[dd/e1f701] process > NFCORE_MHCQUANT:MHCQUANT:PR... [ 25%] 2 of 8\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[65/488afc] process > NFCORE_MHCQUANT:MHCQUANT:OP... [100%] 1 of 1 ✔\n",
      "[7f/0b8433] process > NFCORE_MHCQUANT:MHCQUANT:OP... [  0%] 0 of 2\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:MS... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:MU... -\n",
      "Pulling Singularity image https://depot.galaxyproject.org/singularity/openms-thirdparty:3.1.0--h9ee0642_3 [cache /home/marta/PROJECT_SCRIPTS/241205_Immunopeptidomics_HCC/work/singularity/depot.galaxyproject.org-singularity-openms-thirdparty-3.1.0--h9ee0642_3.img]\n",
      "WARN: Singularity cache directory has not been defined -- Remote image will be stored in the path: /home/marta/PROJECT_SCRIPTS/241205_Immunopeptidomics_HCC/work/singularity -- Use the environment variable NXF_SINGULARITY_CACHEDIR to specify a different location\n",
      "\n",
      "executor >  local (10)\n",
      "[25/2ecb00] process > NFCORE_MHCQUANT:MHCQUANT:PR... [ 37%] 3 of 8\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[65/488afc] process > NFCORE_MHCQUANT:MHCQUANT:OP... [100%] 1 of 1 ✔\n",
      "[7f/0b8433] process > NFCORE_MHCQUANT:MHCQUANT:OP... [  0%] 0 of 3\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:MS... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:MU... -\n",
      "Pulling Singularity image https://depot.galaxyproject.org/singularity/openms-thirdparty:3.1.0--h9ee0642_3 [cache /home/marta/PROJECT_SCRIPTS/241205_Immunopeptidomics_HCC/work/singularity/depot.galaxyproject.org-singularity-openms-thirdparty-3.1.0--h9ee0642_3.img]\n",
      "WARN: Singularity cache directory has not been defined -- Remote image will be stored in the path: /home/marta/PROJECT_SCRIPTS/241205_Immunopeptidomics_HCC/work/singularity -- Use the environment variable NXF_SINGULARITY_CACHEDIR to specify a different location\n",
      "\n",
      "executor >  local (10)\n",
      "[8d/800192] process > NFCORE_MHCQUANT:MHCQUANT:PR... [ 50%] 4 of 8\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[65/488afc] process > NFCORE_MHCQUANT:MHCQUANT:OP... [100%] 1 of 1 ✔\n",
      "[7f/0b8433] process > NFCORE_MHCQUANT:MHCQUANT:OP... [  0%] 0 of 4\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:MS... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:MU... -\n",
      "Pulling Singularity image https://depot.galaxyproject.org/singularity/openms-thirdparty:3.1.0--h9ee0642_3 [cache /home/marta/PROJECT_SCRIPTS/241205_Immunopeptidomics_HCC/work/singularity/depot.galaxyproject.org-singularity-openms-thirdparty-3.1.0--h9ee0642_3.img]\n",
      "WARN: Singularity cache directory has not been defined -- Remote image will be stored in the path: /home/marta/PROJECT_SCRIPTS/241205_Immunopeptidomics_HCC/work/singularity -- Use the environment variable NXF_SINGULARITY_CACHEDIR to specify a different location\n",
      "\n",
      "executor >  local (10)\n",
      "[07/5fd2ea] process > NFCORE_MHCQUANT:MHCQUANT:PR... [ 62%] 5 of 8\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[65/488afc] process > NFCORE_MHCQUANT:MHCQUANT:OP... [100%] 1 of 1 ✔\n",
      "[7f/0b8433] process > NFCORE_MHCQUANT:MHCQUANT:OP... [  0%] 0 of 5\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:MS... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:MU... -\n",
      "Pulling Singularity image https://depot.galaxyproject.org/singularity/openms-thirdparty:3.1.0--h9ee0642_3 [cache /home/marta/PROJECT_SCRIPTS/241205_Immunopeptidomics_HCC/work/singularity/depot.galaxyproject.org-singularity-openms-thirdparty-3.1.0--h9ee0642_3.img]\n",
      "WARN: Singularity cache directory has not been defined -- Remote image will be stored in the path: /home/marta/PROJECT_SCRIPTS/241205_Immunopeptidomics_HCC/work/singularity -- Use the environment variable NXF_SINGULARITY_CACHEDIR to specify a different location\n",
      "\n",
      "executor >  local (10)\n",
      "[a0/a5ff85] process > NFCORE_MHCQUANT:MHCQUANT:PR... [ 75%] 6 of 8\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[65/488afc] process > NFCORE_MHCQUANT:MHCQUANT:OP... [100%] 1 of 1 ✔\n",
      "[7f/0b8433] process > NFCORE_MHCQUANT:MHCQUANT:OP... [  0%] 0 of 6\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:MS... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:MU... -\n",
      "Pulling Singularity image https://depot.galaxyproject.org/singularity/openms-thirdparty:3.1.0--h9ee0642_3 [cache /home/marta/PROJECT_SCRIPTS/241205_Immunopeptidomics_HCC/work/singularity/depot.galaxyproject.org-singularity-openms-thirdparty-3.1.0--h9ee0642_3.img]\n",
      "WARN: Singularity cache directory has not been defined -- Remote image will be stored in the path: /home/marta/PROJECT_SCRIPTS/241205_Immunopeptidomics_HCC/work/singularity -- Use the environment variable NXF_SINGULARITY_CACHEDIR to specify a different location\n",
      "\n",
      "executor >  local (10)\n",
      "[53/3b5dc7] process > NFCORE_MHCQUANT:MHCQUANT:PR... [ 87%] 7 of 8\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[65/488afc] process > NFCORE_MHCQUANT:MHCQUANT:OP... [100%] 1 of 1 ✔\n",
      "[7f/0b8433] process > NFCORE_MHCQUANT:MHCQUANT:OP... [  0%] 0 of 7\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:MS... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:MU... -\n",
      "Pulling Singularity image https://depot.galaxyproject.org/singularity/openms-thirdparty:3.1.0--h9ee0642_3 [cache /home/marta/PROJECT_SCRIPTS/241205_Immunopeptidomics_HCC/work/singularity/depot.galaxyproject.org-singularity-openms-thirdparty-3.1.0--h9ee0642_3.img]\n",
      "WARN: Singularity cache directory has not been defined -- Remote image will be stored in the path: /home/marta/PROJECT_SCRIPTS/241205_Immunopeptidomics_HCC/work/singularity -- Use the environment variable NXF_SINGULARITY_CACHEDIR to specify a different location\n",
      "\n",
      "executor >  local (11)\n",
      "[20/43e5cd] process > NFCORE_MHCQUANT:MHCQUANT:PR... [100%] 8 of 8 ✔\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[65/488afc] process > NFCORE_MHCQUANT:MHCQUANT:OP... [100%] 1 of 1 ✔\n",
      "[55/fff2ca] process > NFCORE_MHCQUANT:MHCQUANT:OP... [  0%] 0 of 8\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:MS... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:MU... -\n",
      "Pulling Singularity image https://depot.galaxyproject.org/singularity/openms-thirdparty:3.1.0--h9ee0642_3 [cache /home/marta/PROJECT_SCRIPTS/241205_Immunopeptidomics_HCC/work/singularity/depot.galaxyproject.org-singularity-openms-thirdparty-3.1.0--h9ee0642_3.img]\n",
      "WARN: Singularity cache directory has not been defined -- Remote image will be stored in the path: /home/marta/PROJECT_SCRIPTS/241205_Immunopeptidomics_HCC/work/singularity -- Use the environment variable NXF_SINGULARITY_CACHEDIR to specify a different location\n",
      "\n",
      "executor >  local (11)\n",
      "[20/43e5cd] process > NFCORE_MHCQUANT:MHCQUANT:PR... [100%] 8 of 8 ✔\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[65/488afc] process > NFCORE_MHCQUANT:MHCQUANT:OP... [100%] 1 of 1 ✔\n",
      "[55/fff2ca] process > NFCORE_MHCQUANT:MHCQUANT:OP... [  0%] 0 of 8\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:MS... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:MU... -\n",
      "Pulling Singularity image https://depot.galaxyproject.org/singularity/openms-thirdparty:3.1.0--h9ee0642_3 [cache /home/marta/PROJECT_SCRIPTS/241205_Immunopeptidomics_HCC/work/singularity/depot.galaxyproject.org-singularity-openms-thirdparty-3.1.0--h9ee0642_3.img]\n",
      "WARN: Singularity cache directory has not been defined -- Remote image will be stored in the path: /home/marta/PROJECT_SCRIPTS/241205_Immunopeptidomics_HCC/work/singularity -- Use the environment variable NXF_SINGULARITY_CACHEDIR to specify a different location\n",
      "\n",
      "executor >  local (12)\n",
      "[20/43e5cd] process > NFCORE_MHCQUANT:MHCQUANT:PR... [100%] 8 of 8 ✔\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[65/488afc] process > NFCORE_MHCQUANT:MHCQUANT:OP... [100%] 1 of 1 ✔\n",
      "[7f/0b8433] process > NFCORE_MHCQUANT:MHCQUANT:OP... [ 12%] 1 of 8\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... [  0%] 0 of 1\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:MS... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:MU... -\n",
      "\n",
      "\n",
      "\n",
      "executor >  local (13)\n",
      "[20/43e5cd] process > NFCORE_MHCQUANT:MHCQUANT:PR... [100%] 8 of 8 ✔\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[65/488afc] process > NFCORE_MHCQUANT:MHCQUANT:OP... [100%] 1 of 1 ✔\n",
      "[ea/8d612f] process > NFCORE_MHCQUANT:MHCQUANT:OP... [ 25%] 2 of 8\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... [  0%] 0 of 2\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:MS... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:MU... -\n",
      "\n",
      "\n",
      "\n",
      "executor >  local (13)\n",
      "[20/43e5cd] process > NFCORE_MHCQUANT:MHCQUANT:PR... [100%] 8 of 8 ✔\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[65/488afc] process > NFCORE_MHCQUANT:MHCQUANT:OP... [100%] 1 of 1 ✔\n",
      "[ea/8d612f] process > NFCORE_MHCQUANT:MHCQUANT:OP... [ 25%] 2 of 8\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... [  0%] 0 of 2\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:MS... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:MU... -\n",
      "\n",
      "executor >  local (14)\n",
      "[20/43e5cd] process > NFCORE_MHCQUANT:MHCQUANT:PR... [100%] 8 of 8 ✔\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[65/488afc] process > NFCORE_MHCQUANT:MHCQUANT:OP... [100%] 1 of 1 ✔\n",
      "[27/b1736a] process > NFCORE_MHCQUANT:MHCQUANT:OP... [ 37%] 3 of 8\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... [  0%] 0 of 3\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:MS... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:MU... -\n",
      "\n",
      "executor >  local (14)\n",
      "[20/43e5cd] process > NFCORE_MHCQUANT:MHCQUANT:PR... [100%] 8 of 8 ✔\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[65/488afc] process > NFCORE_MHCQUANT:MHCQUANT:OP... [100%] 1 of 1 ✔\n",
      "[27/b1736a] process > NFCORE_MHCQUANT:MHCQUANT:OP... [ 37%] 3 of 8\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... [  0%] 0 of 3\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:MS... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:MU... -\n",
      "\n",
      "executor >  local (15)\n",
      "[20/43e5cd] process > NFCORE_MHCQUANT:MHCQUANT:PR... [100%] 8 of 8 ✔\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[65/488afc] process > NFCORE_MHCQUANT:MHCQUANT:OP... [100%] 1 of 1 ✔\n",
      "[36/8531c5] process > NFCORE_MHCQUANT:MHCQUANT:OP... [ 50%] 4 of 8\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... [  0%] 0 of 4\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:MS... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:MU... -\n",
      "\n",
      "executor >  local (16)\n",
      "[20/43e5cd] process > NFCORE_MHCQUANT:MHCQUANT:PR... [100%] 8 of 8 ✔\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[65/488afc] process > NFCORE_MHCQUANT:MHCQUANT:OP... [100%] 1 of 1 ✔\n",
      "[8d/397295] process > NFCORE_MHCQUANT:MHCQUANT:OP... [ 62%] 5 of 8\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... [  0%] 0 of 5\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:MS... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:MU... -\n",
      "\n",
      "executor >  local (16)\n",
      "[20/43e5cd] process > NFCORE_MHCQUANT:MHCQUANT:PR... [100%] 8 of 8 ✔\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[65/488afc] process > NFCORE_MHCQUANT:MHCQUANT:OP... [100%] 1 of 1 ✔\n",
      "[8d/397295] process > NFCORE_MHCQUANT:MHCQUANT:OP... [ 62%] 5 of 8\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... [  0%] 0 of 5\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:MS... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:MU... -\n",
      "\n",
      "executor >  local (17)\n",
      "[20/43e5cd] process > NFCORE_MHCQUANT:MHCQUANT:PR... [100%] 8 of 8 ✔\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[65/488afc] process > NFCORE_MHCQUANT:MHCQUANT:OP... [100%] 1 of 1 ✔\n",
      "[2e/2fe042] process > NFCORE_MHCQUANT:MHCQUANT:OP... [ 75%] 6 of 8\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... [  0%] 0 of 6\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:MS... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:MU... -\n",
      "\n",
      "executor >  local (17)\n",
      "[20/43e5cd] process > NFCORE_MHCQUANT:MHCQUANT:PR... [100%] 8 of 8 ✔\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[65/488afc] process > NFCORE_MHCQUANT:MHCQUANT:OP... [100%] 1 of 1 ✔\n",
      "[2e/2fe042] process > NFCORE_MHCQUANT:MHCQUANT:OP... [ 75%] 6 of 8\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... [  0%] 0 of 6\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:MS... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:MU... -\n",
      "\n",
      "executor >  local (24)\n",
      "[20/43e5cd] process > NFCORE_MHCQUANT:MHCQUANT:PR... [100%] 8 of 8 ✔\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[65/488afc] process > NFCORE_MHCQUANT:MHCQUANT:OP... [100%] 1 of 1 ✔\n",
      "[8d/397295] process > NFCORE_MHCQUANT:MHCQUANT:OP... [ 87%] 7 of 8\n",
      "[52/59a417] process > NFCORE_MHCQUANT:MHCQUANT:OP... [  0%] 0 of 7\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:MS... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:MU... -\n",
      "\n",
      "executor >  local (24)\n",
      "[20/43e5cd] process > NFCORE_MHCQUANT:MHCQUANT:PR... [100%] 8 of 8 ✔\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[65/488afc] process > NFCORE_MHCQUANT:MHCQUANT:OP... [100%] 1 of 1 ✔\n",
      "[8d/397295] process > NFCORE_MHCQUANT:MHCQUANT:OP... [ 87%] 7 of 8\n",
      "[52/59a417] process > NFCORE_MHCQUANT:MHCQUANT:OP... [  0%] 0 of 7\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:MS... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:MU... -\n",
      "\n",
      "executor >  local (25)\n",
      "[20/43e5cd] process > NFCORE_MHCQUANT:MHCQUANT:PR... [100%] 8 of 8 ✔\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[65/488afc] process > NFCORE_MHCQUANT:MHCQUANT:OP... [100%] 1 of 1 ✔\n",
      "[2e/2fe042] process > NFCORE_MHCQUANT:MHCQUANT:OP... [100%] 8 of 8 ✔\n",
      "[75/5a7b14] process > NFCORE_MHCQUANT:MHCQUANT:OP... [  0%] 0 of 8\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:MS... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:MU... -\n",
      "\n",
      "executor >  local (25)\n",
      "[20/43e5cd] process > NFCORE_MHCQUANT:MHCQUANT:PR... [100%] 8 of 8 ✔\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[65/488afc] process > NFCORE_MHCQUANT:MHCQUANT:OP... [100%] 1 of 1 ✔\n",
      "[2e/2fe042] process > NFCORE_MHCQUANT:MHCQUANT:OP... [100%] 8 of 8 ✔\n",
      "[75/5a7b14] process > NFCORE_MHCQUANT:MHCQUANT:OP... [  0%] 0 of 8\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:MS... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:MU... -\n",
      "\n",
      "executor >  local (25)\n",
      "[20/43e5cd] process > NFCORE_MHCQUANT:MHCQUANT:PR... [100%] 8 of 8 ✔\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[65/488afc] process > NFCORE_MHCQUANT:MHCQUANT:OP... [100%] 1 of 1 ✔\n",
      "[2e/2fe042] process > NFCORE_MHCQUANT:MHCQUANT:OP... [100%] 8 of 8 ✔\n",
      "[7c/bd7bb8] process > NFCORE_MHCQUANT:MHCQUANT:OP... [ 12%] 1 of 8\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:MS... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:MU... -\n",
      "\n",
      "executor >  local (25)\n",
      "[20/43e5cd] process > NFCORE_MHCQUANT:MHCQUANT:PR... [100%] 8 of 8 ✔\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[65/488afc] process > NFCORE_MHCQUANT:MHCQUANT:OP... [100%] 1 of 1 ✔\n",
      "[2e/2fe042] process > NFCORE_MHCQUANT:MHCQUANT:OP... [100%] 8 of 8 ✔\n",
      "[4d/705ccd] process > NFCORE_MHCQUANT:MHCQUANT:OP... [ 25%] 2 of 8\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:MS... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:MU... -\n",
      "\n",
      "executor >  local (25)\n",
      "[20/43e5cd] process > NFCORE_MHCQUANT:MHCQUANT:PR... [100%] 8 of 8 ✔\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[65/488afc] process > NFCORE_MHCQUANT:MHCQUANT:OP... [100%] 1 of 1 ✔\n",
      "[2e/2fe042] process > NFCORE_MHCQUANT:MHCQUANT:OP... [100%] 8 of 8 ✔\n",
      "[2a/055276] process > NFCORE_MHCQUANT:MHCQUANT:OP... [ 62%] 5 of 8\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:MS... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:MU... -\n",
      "\n",
      "executor >  local (25)\n",
      "[20/43e5cd] process > NFCORE_MHCQUANT:MHCQUANT:PR... [100%] 8 of 8 ✔\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[65/488afc] process > NFCORE_MHCQUANT:MHCQUANT:OP... [100%] 1 of 1 ✔\n",
      "[2e/2fe042] process > NFCORE_MHCQUANT:MHCQUANT:OP... [100%] 8 of 8 ✔\n",
      "[0f/eef3e6] process > NFCORE_MHCQUANT:MHCQUANT:OP... [ 75%] 6 of 8\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:MS... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:MU... -\n",
      "\n",
      "executor >  local (25)\n",
      "[20/43e5cd] process > NFCORE_MHCQUANT:MHCQUANT:PR... [100%] 8 of 8 ✔\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[65/488afc] process > NFCORE_MHCQUANT:MHCQUANT:OP... [100%] 1 of 1 ✔\n",
      "[2e/2fe042] process > NFCORE_MHCQUANT:MHCQUANT:OP... [100%] 8 of 8 ✔\n",
      "[f9/8220cd] process > NFCORE_MHCQUANT:MHCQUANT:OP... [ 87%] 7 of 8\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:MS... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:MU... -\n",
      "\n",
      "executor >  local (26)\n",
      "[20/43e5cd] process > NFCORE_MHCQUANT:MHCQUANT:PR... [100%] 8 of 8 ✔\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[65/488afc] process > NFCORE_MHCQUANT:MHCQUANT:OP... [100%] 1 of 1 ✔\n",
      "[2e/2fe042] process > NFCORE_MHCQUANT:MHCQUANT:OP... [100%] 8 of 8 ✔\n",
      "[75/5a7b14] process > NFCORE_MHCQUANT:MHCQUANT:OP... [100%] 8 of 8 ✔\n",
      "[e1/f70be5] process > NFCORE_MHCQUANT:MHCQUANT:OP... [  0%] 0 of 1\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:MS... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:MU... -\n",
      "\n",
      "executor >  local (26)\n",
      "[20/43e5cd] process > NFCORE_MHCQUANT:MHCQUANT:PR... [100%] 8 of 8 ✔\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[65/488afc] process > NFCORE_MHCQUANT:MHCQUANT:OP... [100%] 1 of 1 ✔\n",
      "[2e/2fe042] process > NFCORE_MHCQUANT:MHCQUANT:OP... [100%] 8 of 8 ✔\n",
      "[75/5a7b14] process > NFCORE_MHCQUANT:MHCQUANT:OP... [100%] 8 of 8 ✔\n",
      "[e1/f70be5] process > NFCORE_MHCQUANT:MHCQUANT:OP... [  0%] 0 of 1\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:MS... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:MU... -\n",
      "\n",
      "executor >  local (26)\n",
      "[20/43e5cd] process > NFCORE_MHCQUANT:MHCQUANT:PR... [100%] 8 of 8 ✔\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[65/488afc] process > NFCORE_MHCQUANT:MHCQUANT:OP... [100%] 1 of 1 ✔\n",
      "[2e/2fe042] process > NFCORE_MHCQUANT:MHCQUANT:OP... [100%] 8 of 8 ✔\n",
      "[75/5a7b14] process > NFCORE_MHCQUANT:MHCQUANT:OP... [100%] 8 of 8 ✔\n",
      "[e1/f70be5] process > NFCORE_MHCQUANT:MHCQUANT:OP... [100%] 1 of 1 ✔\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:MS... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:MU... -\n",
      "Pulling Singularity image https://depot.galaxyproject.org/singularity/ms2rescore:3.0.1--pyhdfd78af_2 [cache /home/marta/PROJECT_SCRIPTS/241205_Immunopeptidomics_HCC/work/singularity/depot.galaxyproject.org-singularity-ms2rescore-3.0.1--pyhdfd78af_2.img]\n",
      "WARN: Singularity cache directory has not been defined -- Remote image will be stored in the path: /home/marta/PROJECT_SCRIPTS/241205_Immunopeptidomics_HCC/work/singularity -- Use the environment variable NXF_SINGULARITY_CACHEDIR to specify a different location\n",
      "\n",
      "executor >  local (27)\n",
      "[20/43e5cd] process > NFCORE_MHCQUANT:MHCQUANT:PR... [100%] 8 of 8 ✔\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[65/488afc] process > NFCORE_MHCQUANT:MHCQUANT:OP... [100%] 1 of 1 ✔\n",
      "[2e/2fe042] process > NFCORE_MHCQUANT:MHCQUANT:OP... [100%] 8 of 8 ✔\n",
      "[75/5a7b14] process > NFCORE_MHCQUANT:MHCQUANT:OP... [100%] 8 of 8 ✔\n",
      "[e1/f70be5] process > NFCORE_MHCQUANT:MHCQUANT:OP... [100%] 1 of 1 ✔\n",
      "[7c/90decd] process > NFCORE_MHCQUANT:MHCQUANT:MS... [  0%] 0 of 1\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:MU... -\n",
      "Pulling Singularity image https://depot.galaxyproject.org/singularity/ms2rescore:3.0.1--pyhdfd78af_2 [cache /home/marta/PROJECT_SCRIPTS/241205_Immunopeptidomics_HCC/work/singularity/depot.galaxyproject.org-singularity-ms2rescore-3.0.1--pyhdfd78af_2.img]\n",
      "WARN: Singularity cache directory has not been defined -- Remote image will be stored in the path: /home/marta/PROJECT_SCRIPTS/241205_Immunopeptidomics_HCC/work/singularity -- Use the environment variable NXF_SINGULARITY_CACHEDIR to specify a different location\n",
      "\n",
      "executor >  local (28)\n",
      "[20/43e5cd] process > NFCORE_MHCQUANT:MHCQUANT:PR... [100%] 8 of 8 ✔\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[65/488afc] process > NFCORE_MHCQUANT:MHCQUANT:OP... [100%] 1 of 1 ✔\n",
      "[2e/2fe042] process > NFCORE_MHCQUANT:MHCQUANT:OP... [100%] 8 of 8 ✔\n",
      "[75/5a7b14] process > NFCORE_MHCQUANT:MHCQUANT:OP... [100%] 8 of 8 ✔\n",
      "[e1/f70be5] process > NFCORE_MHCQUANT:MHCQUANT:OP... [100%] 1 of 1 ✔\n",
      "[7c/90decd] process > NFCORE_MHCQUANT:MHCQUANT:MS... [100%] 1 of 1 ✔\n",
      "[07/3010fe] process > NFCORE_MHCQUANT:MHCQUANT:OP... [  0%] 0 of 1\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:MU... -\n",
      "\n",
      "\n",
      "\n",
      "executor >  local (29)\n",
      "[20/43e5cd] process > NFCORE_MHCQUANT:MHCQUANT:PR... [100%] 8 of 8 ✔\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[65/488afc] process > NFCORE_MHCQUANT:MHCQUANT:OP... [100%] 1 of 1 ✔\n",
      "[2e/2fe042] process > NFCORE_MHCQUANT:MHCQUANT:OP... [100%] 8 of 8 ✔\n",
      "[75/5a7b14] process > NFCORE_MHCQUANT:MHCQUANT:OP... [100%] 8 of 8 ✔\n",
      "[e1/f70be5] process > NFCORE_MHCQUANT:MHCQUANT:OP... [100%] 1 of 1 ✔\n",
      "[7c/90decd] process > NFCORE_MHCQUANT:MHCQUANT:MS... [100%] 1 of 1 ✔\n",
      "[07/3010fe] process > NFCORE_MHCQUANT:MHCQUANT:OP... [100%] 1 of 1 ✔\n",
      "[6d/72728e] process > NFCORE_MHCQUANT:MHCQUANT:OP... [  0%] 0 of 1\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:MU... -\n",
      "\n",
      "\n",
      "\n",
      "executor >  local (29)\n",
      "[20/43e5cd] process > NFCORE_MHCQUANT:MHCQUANT:PR... [100%] 8 of 8 ✔\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[65/488afc] process > NFCORE_MHCQUANT:MHCQUANT:OP... [100%] 1 of 1 ✔\n",
      "[2e/2fe042] process > NFCORE_MHCQUANT:MHCQUANT:OP... [100%] 8 of 8 ✔\n",
      "[75/5a7b14] process > NFCORE_MHCQUANT:MHCQUANT:OP... [100%] 8 of 8 ✔\n",
      "[e1/f70be5] process > NFCORE_MHCQUANT:MHCQUANT:OP... [100%] 1 of 1 ✔\n",
      "[7c/90decd] process > NFCORE_MHCQUANT:MHCQUANT:MS... [100%] 1 of 1 ✔\n",
      "[07/3010fe] process > NFCORE_MHCQUANT:MHCQUANT:OP... [100%] 1 of 1 ✔\n",
      "[6d/72728e] process > NFCORE_MHCQUANT:MHCQUANT:OP... [  0%] 0 of 1\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:MU... -\n",
      "\n",
      "executor >  local (30)\n",
      "[20/43e5cd] process > NFCORE_MHCQUANT:MHCQUANT:PR... [100%] 8 of 8 ✔\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[65/488afc] process > NFCORE_MHCQUANT:MHCQUANT:OP... [100%] 1 of 1 ✔\n",
      "[2e/2fe042] process > NFCORE_MHCQUANT:MHCQUANT:OP... [100%] 8 of 8 ✔\n",
      "[75/5a7b14] process > NFCORE_MHCQUANT:MHCQUANT:OP... [100%] 8 of 8 ✔\n",
      "[e1/f70be5] process > NFCORE_MHCQUANT:MHCQUANT:OP... [100%] 1 of 1 ✔\n",
      "[7c/90decd] process > NFCORE_MHCQUANT:MHCQUANT:MS... [100%] 1 of 1 ✔\n",
      "[07/3010fe] process > NFCORE_MHCQUANT:MHCQUANT:OP... [100%] 1 of 1 ✔\n",
      "[6d/72728e] process > NFCORE_MHCQUANT:MHCQUANT:OP... [100%] 1 of 1 ✔\n",
      "[91/8d5930] process > NFCORE_MHCQUANT:MHCQUANT:OP... [  0%] 0 of 1\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:MU... -\n",
      "\n",
      "executor >  local (30)\n",
      "[20/43e5cd] process > NFCORE_MHCQUANT:MHCQUANT:PR... [100%] 8 of 8 ✔\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[65/488afc] process > NFCORE_MHCQUANT:MHCQUANT:OP... [100%] 1 of 1 ✔\n",
      "[2e/2fe042] process > NFCORE_MHCQUANT:MHCQUANT:OP... [100%] 8 of 8 ✔\n",
      "[75/5a7b14] process > NFCORE_MHCQUANT:MHCQUANT:OP... [100%] 8 of 8 ✔\n",
      "[e1/f70be5] process > NFCORE_MHCQUANT:MHCQUANT:OP... [100%] 1 of 1 ✔\n",
      "[7c/90decd] process > NFCORE_MHCQUANT:MHCQUANT:MS... [100%] 1 of 1 ✔\n",
      "[07/3010fe] process > NFCORE_MHCQUANT:MHCQUANT:OP... [100%] 1 of 1 ✔\n",
      "[6d/72728e] process > NFCORE_MHCQUANT:MHCQUANT:OP... [100%] 1 of 1 ✔\n",
      "[91/8d5930] process > NFCORE_MHCQUANT:MHCQUANT:OP... [  0%] 0 of 1\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:MU... -\n",
      "\n",
      "executor >  local (31)\n",
      "[20/43e5cd] process > NFCORE_MHCQUANT:MHCQUANT:PR... [100%] 8 of 8 ✔\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[65/488afc] process > NFCORE_MHCQUANT:MHCQUANT:OP... [100%] 1 of 1 ✔\n",
      "[2e/2fe042] process > NFCORE_MHCQUANT:MHCQUANT:OP... [100%] 8 of 8 ✔\n",
      "[75/5a7b14] process > NFCORE_MHCQUANT:MHCQUANT:OP... [100%] 8 of 8 ✔\n",
      "[e1/f70be5] process > NFCORE_MHCQUANT:MHCQUANT:OP... [100%] 1 of 1 ✔\n",
      "[7c/90decd] process > NFCORE_MHCQUANT:MHCQUANT:MS... [100%] 1 of 1 ✔\n",
      "[07/3010fe] process > NFCORE_MHCQUANT:MHCQUANT:OP... [100%] 1 of 1 ✔\n",
      "[6d/72728e] process > NFCORE_MHCQUANT:MHCQUANT:OP... [100%] 1 of 1 ✔\n",
      "[91/8d5930] process > NFCORE_MHCQUANT:MHCQUANT:OP... [100%] 1 of 1 ✔\n",
      "[e7/4d060d] process > NFCORE_MHCQUANT:MHCQUANT:OP... [  0%] 0 of 1\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... [  0%] 0 of 1\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:MU... -\n",
      "\n",
      "executor >  local (32)\n",
      "[20/43e5cd] process > NFCORE_MHCQUANT:MHCQUANT:PR... [100%] 8 of 8 ✔\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[65/488afc] process > NFCORE_MHCQUANT:MHCQUANT:OP... [100%] 1 of 1 ✔\n",
      "[2e/2fe042] process > NFCORE_MHCQUANT:MHCQUANT:OP... [100%] 8 of 8 ✔\n",
      "[75/5a7b14] process > NFCORE_MHCQUANT:MHCQUANT:OP... [100%] 8 of 8 ✔\n",
      "[e1/f70be5] process > NFCORE_MHCQUANT:MHCQUANT:OP... [100%] 1 of 1 ✔\n",
      "[7c/90decd] process > NFCORE_MHCQUANT:MHCQUANT:MS... [100%] 1 of 1 ✔\n",
      "[07/3010fe] process > NFCORE_MHCQUANT:MHCQUANT:OP... [100%] 1 of 1 ✔\n",
      "[6d/72728e] process > NFCORE_MHCQUANT:MHCQUANT:OP... [100%] 1 of 1 ✔\n",
      "[91/8d5930] process > NFCORE_MHCQUANT:MHCQUANT:OP... [100%] 1 of 1 ✔\n",
      "[e7/4d060d] process > NFCORE_MHCQUANT:MHCQUANT:OP... [  0%] 0 of 1\n",
      "[b6/08e7ab] process > NFCORE_MHCQUANT:MHCQUANT:OP... [  0%] 0 of 1\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:MU... -\n",
      "\n",
      "executor >  local (32)\n",
      "[20/43e5cd] process > NFCORE_MHCQUANT:MHCQUANT:PR... [100%] 8 of 8 ✔\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[65/488afc] process > NFCORE_MHCQUANT:MHCQUANT:OP... [100%] 1 of 1 ✔\n",
      "[2e/2fe042] process > NFCORE_MHCQUANT:MHCQUANT:OP... [100%] 8 of 8 ✔\n",
      "[75/5a7b14] process > NFCORE_MHCQUANT:MHCQUANT:OP... [100%] 8 of 8 ✔\n",
      "[e1/f70be5] process > NFCORE_MHCQUANT:MHCQUANT:OP... [100%] 1 of 1 ✔\n",
      "[7c/90decd] process > NFCORE_MHCQUANT:MHCQUANT:MS... [100%] 1 of 1 ✔\n",
      "[07/3010fe] process > NFCORE_MHCQUANT:MHCQUANT:OP... [100%] 1 of 1 ✔\n",
      "[6d/72728e] process > NFCORE_MHCQUANT:MHCQUANT:OP... [100%] 1 of 1 ✔\n",
      "[91/8d5930] process > NFCORE_MHCQUANT:MHCQUANT:OP... [100%] 1 of 1 ✔\n",
      "[e7/4d060d] process > NFCORE_MHCQUANT:MHCQUANT:OP... [100%] 1 of 1 ✔\n",
      "[b6/08e7ab] process > NFCORE_MHCQUANT:MHCQUANT:OP... [  0%] 0 of 1\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:MU... -\n",
      "\n",
      "executor >  local (32)\n",
      "[20/43e5cd] process > NFCORE_MHCQUANT:MHCQUANT:PR... [100%] 8 of 8 ✔\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[65/488afc] process > NFCORE_MHCQUANT:MHCQUANT:OP... [100%] 1 of 1 ✔\n",
      "[2e/2fe042] process > NFCORE_MHCQUANT:MHCQUANT:OP... [100%] 8 of 8 ✔\n",
      "[75/5a7b14] process > NFCORE_MHCQUANT:MHCQUANT:OP... [100%] 8 of 8 ✔\n",
      "[e1/f70be5] process > NFCORE_MHCQUANT:MHCQUANT:OP... [100%] 1 of 1 ✔\n",
      "[7c/90decd] process > NFCORE_MHCQUANT:MHCQUANT:MS... [100%] 1 of 1 ✔\n",
      "[07/3010fe] process > NFCORE_MHCQUANT:MHCQUANT:OP... [100%] 1 of 1 ✔\n",
      "[6d/72728e] process > NFCORE_MHCQUANT:MHCQUANT:OP... [100%] 1 of 1 ✔\n",
      "[91/8d5930] process > NFCORE_MHCQUANT:MHCQUANT:OP... [100%] 1 of 1 ✔\n",
      "[e7/4d060d] process > NFCORE_MHCQUANT:MHCQUANT:OP... [100%] 1 of 1 ✔\n",
      "[b6/08e7ab] process > NFCORE_MHCQUANT:MHCQUANT:OP... [100%] 1 of 1 ✔\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:MU... -\n",
      "Pulling Singularity image https://depot.galaxyproject.org/singularity/multiqc:1.21--pyhdfd78af_0 [cache /home/marta/PROJECT_SCRIPTS/241205_Immunopeptidomics_HCC/work/singularity/depot.galaxyproject.org-singularity-multiqc-1.21--pyhdfd78af_0.img]\n",
      "WARN: Singularity cache directory has not been defined -- Remote image will be stored in the path: /home/marta/PROJECT_SCRIPTS/241205_Immunopeptidomics_HCC/work/singularity -- Use the environment variable NXF_SINGULARITY_CACHEDIR to specify a different location\n",
      "\n",
      "executor >  local (33)\n",
      "[20/43e5cd] process > NFCORE_MHCQUANT:MHCQUANT:PR... [100%] 8 of 8 ✔\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[65/488afc] process > NFCORE_MHCQUANT:MHCQUANT:OP... [100%] 1 of 1 ✔\n",
      "[2e/2fe042] process > NFCORE_MHCQUANT:MHCQUANT:OP... [100%] 8 of 8 ✔\n",
      "[75/5a7b14] process > NFCORE_MHCQUANT:MHCQUANT:OP... [100%] 8 of 8 ✔\n",
      "[e1/f70be5] process > NFCORE_MHCQUANT:MHCQUANT:OP... [100%] 1 of 1 ✔\n",
      "[7c/90decd] process > NFCORE_MHCQUANT:MHCQUANT:MS... [100%] 1 of 1 ✔\n",
      "[07/3010fe] process > NFCORE_MHCQUANT:MHCQUANT:OP... [100%] 1 of 1 ✔\n",
      "[6d/72728e] process > NFCORE_MHCQUANT:MHCQUANT:OP... [100%] 1 of 1 ✔\n",
      "[91/8d5930] process > NFCORE_MHCQUANT:MHCQUANT:OP... [100%] 1 of 1 ✔\n",
      "[e7/4d060d] process > NFCORE_MHCQUANT:MHCQUANT:OP... [100%] 1 of 1 ✔\n",
      "[b6/08e7ab] process > NFCORE_MHCQUANT:MHCQUANT:OP... [100%] 1 of 1 ✔\n",
      "[c8/a49581] process > NFCORE_MHCQUANT:MHCQUANT:MU... [  0%] 0 of 1\n",
      "Pulling Singularity image https://depot.galaxyproject.org/singularity/multiqc:1.21--pyhdfd78af_0 [cache /home/marta/PROJECT_SCRIPTS/241205_Immunopeptidomics_HCC/work/singularity/depot.galaxyproject.org-singularity-multiqc-1.21--pyhdfd78af_0.img]\n",
      "WARN: Singularity cache directory has not been defined -- Remote image will be stored in the path: /home/marta/PROJECT_SCRIPTS/241205_Immunopeptidomics_HCC/work/singularity -- Use the environment variable NXF_SINGULARITY_CACHEDIR to specify a different location\n",
      "\n",
      "executor >  local (33)\n",
      "[20/43e5cd] process > NFCORE_MHCQUANT:MHCQUANT:PR... [100%] 8 of 8 ✔\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[65/488afc] process > NFCORE_MHCQUANT:MHCQUANT:OP... [100%] 1 of 1 ✔\n",
      "[2e/2fe042] process > NFCORE_MHCQUANT:MHCQUANT:OP... [100%] 8 of 8 ✔\n",
      "[75/5a7b14] process > NFCORE_MHCQUANT:MHCQUANT:OP... [100%] 8 of 8 ✔\n",
      "[e1/f70be5] process > NFCORE_MHCQUANT:MHCQUANT:OP... [100%] 1 of 1 ✔\n",
      "[7c/90decd] process > NFCORE_MHCQUANT:MHCQUANT:MS... [100%] 1 of 1 ✔\n",
      "[07/3010fe] process > NFCORE_MHCQUANT:MHCQUANT:OP... [100%] 1 of 1 ✔\n",
      "[6d/72728e] process > NFCORE_MHCQUANT:MHCQUANT:OP... [100%] 1 of 1 ✔\n",
      "[91/8d5930] process > NFCORE_MHCQUANT:MHCQUANT:OP... [100%] 1 of 1 ✔\n",
      "[e7/4d060d] process > NFCORE_MHCQUANT:MHCQUANT:OP... [100%] 1 of 1 ✔\n",
      "[b6/08e7ab] process > NFCORE_MHCQUANT:MHCQUANT:OP... [100%] 1 of 1 ✔\n",
      "[c8/a49581] process > NFCORE_MHCQUANT:MHCQUANT:MU... [100%] 1 of 1 ✔\n",
      "Pulling Singularity image https://depot.galaxyproject.org/singularity/multiqc:1.21--pyhdfd78af_0 [cache /home/marta/PROJECT_SCRIPTS/241205_Immunopeptidomics_HCC/work/singularity/depot.galaxyproject.org-singularity-multiqc-1.21--pyhdfd78af_0.img]\n",
      "-[nf-core/mhcquant] Pipeline completed successfully-\n",
      "WARN: Singularity cache directory has not been defined -- Remote image will be stored in the path: /home/marta/PROJECT_SCRIPTS/241205_Immunopeptidomics_HCC/work/singularity -- Use the environment variable NXF_SINGULARITY_CACHEDIR to specify a different location\n",
      "\n",
      "executor >  local (33)\n",
      "[20/43e5cd] process > NFCORE_MHCQUANT:MHCQUANT:PR... [100%] 8 of 8 ✔\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[65/488afc] process > NFCORE_MHCQUANT:MHCQUANT:OP... [100%] 1 of 1 ✔\n",
      "[2e/2fe042] process > NFCORE_MHCQUANT:MHCQUANT:OP... [100%] 8 of 8 ✔\n",
      "[75/5a7b14] process > NFCORE_MHCQUANT:MHCQUANT:OP... [100%] 8 of 8 ✔\n",
      "[e1/f70be5] process > NFCORE_MHCQUANT:MHCQUANT:OP... [100%] 1 of 1 ✔\n",
      "[7c/90decd] process > NFCORE_MHCQUANT:MHCQUANT:MS... [100%] 1 of 1 ✔\n",
      "[07/3010fe] process > NFCORE_MHCQUANT:MHCQUANT:OP... [100%] 1 of 1 ✔\n",
      "[6d/72728e] process > NFCORE_MHCQUANT:MHCQUANT:OP... [100%] 1 of 1 ✔\n",
      "[91/8d5930] process > NFCORE_MHCQUANT:MHCQUANT:OP... [100%] 1 of 1 ✔\n",
      "[e7/4d060d] process > NFCORE_MHCQUANT:MHCQUANT:OP... [100%] 1 of 1 ✔\n",
      "[b6/08e7ab] process > NFCORE_MHCQUANT:MHCQUANT:OP... [100%] 1 of 1 ✔\n",
      "[c8/a49581] process > NFCORE_MHCQUANT:MHCQUANT:MU... [100%] 1 of 1 ✔\n",
      "Pulling Singularity image https://depot.galaxyproject.org/singularity/multiqc:1.21--pyhdfd78af_0 [cache /home/marta/PROJECT_SCRIPTS/241205_Immunopeptidomics_HCC/work/singularity/depot.galaxyproject.org-singularity-multiqc-1.21--pyhdfd78af_0.img]\n",
      "-[nf-core/mhcquant] Pipeline completed successfully-\n",
      "WARN: Singularity cache directory has not been defined -- Remote image will be stored in the path: /home/marta/PROJECT_SCRIPTS/241205_Immunopeptidomics_HCC/work/singularity -- Use the environment variable NXF_SINGULARITY_CACHEDIR to specify a different location\n",
      "Completed at: 05-Feb-2025 16:02:26\n",
      "Duration    : 1h 48m 4s\n",
      "CPU hours   : 21.9\n",
      "Succeeded   : 33\n",
      "\n",
      "\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "module load Java\n",
    "module load singularity\n",
    "module load Python\n",
    "\n",
    "FDR=0.05\n",
    "ref=/home/marta/241205_Immunopeptidomics_HCC/database/TAA_SPTREMBL.fasta\n",
    "\n",
    "# samples=\"Me275 Me290 OMM475 T1015A T1185B OD5P ONVC\"\n",
    "samples=\"OD5P\"\n",
    "\n",
    "for s in $samples; do\n",
    "    outdir=/home/marta/241205_Immunopeptidomics_HCC/Melanoma_Chong/FDR5percent/$s\n",
    "    mkdir -p $outdir\n",
    "\n",
    "    /home/marta/nextflow run nf-core/mhcquant -profile singularity --input /home/marta/immunopeptidomics_rawData/samplesheet_${s}.tsv --outdir $outdir --fasta $ref --fdr_threshold $FDR\n",
    "    # grep 'PEPTIDE' $outdir/${s}_A.tsv | grep -v 'UNASSIGNED' > $outdir/${s}_A_PEPTIDE.tsv\n",
    "    # grep -v 'non-unique' $outdir/${s}_A_PEPTIDE.tsv > $outdir/${s}_A_PEPTIDE.unique.tsv\n",
    "    grep -v 'non-unique' $outdir/${s}_A.tsv > $outdir/${s}_A_PEPTIDE.unique.tsv\n",
    "\n",
    "    if [ -s $outdir/${s}_A_PEPTIDE.unique.tsv ]; then\n",
    "        mv /home/marta/241205_Immunopeptidomics_HCC/Melanoma_Chong/FDR5percent/$s /users/genomics/marta/241205_Immunopeptidomics_HCC/Melanoma_Chong/FDR5percent\n",
    "    fi\n",
    "\n",
    "    rm -r work\n",
    "done"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "\n",
    "**Analysis | FDR 5%**"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Me290_A.tsv\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_3946535/3122408642.py:17: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  noncoding['transcript_id'] = noncoding['accessions'].str.split(\":\", expand=True)[0]\n",
      "/tmp/ipykernel_3946535/3122408642.py:18: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  noncoding['transcript_id'] = noncoding['transcript_id'].str.split(\".\", expand=True)[0]\n",
      "/tmp/ipykernel_3946535/3122408642.py:19: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  noncoding['sample'] = folder\n",
      "/tmp/ipykernel_3946535/3122408642.py:24: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  coding['sample'] = folder\n",
      "/tmp/ipykernel_3946535/3122408642.py:25: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  coding['Entry'] = coding['accessions'].apply(\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ONVC_A.tsv\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_3946535/3122408642.py:17: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  noncoding['transcript_id'] = noncoding['accessions'].str.split(\":\", expand=True)[0]\n",
      "/tmp/ipykernel_3946535/3122408642.py:18: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  noncoding['transcript_id'] = noncoding['transcript_id'].str.split(\".\", expand=True)[0]\n",
      "/tmp/ipykernel_3946535/3122408642.py:19: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  noncoding['sample'] = folder\n",
      "/tmp/ipykernel_3946535/3122408642.py:24: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  coding['sample'] = folder\n",
      "/tmp/ipykernel_3946535/3122408642.py:25: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  coding['Entry'] = coding['accessions'].apply(\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "OMM475_A.tsv\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_3946535/3122408642.py:17: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  noncoding['transcript_id'] = noncoding['accessions'].str.split(\":\", expand=True)[0]\n",
      "/tmp/ipykernel_3946535/3122408642.py:18: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  noncoding['transcript_id'] = noncoding['transcript_id'].str.split(\".\", expand=True)[0]\n",
      "/tmp/ipykernel_3946535/3122408642.py:19: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  noncoding['sample'] = folder\n",
      "/tmp/ipykernel_3946535/3122408642.py:24: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  coding['sample'] = folder\n",
      "/tmp/ipykernel_3946535/3122408642.py:25: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  coding['Entry'] = coding['accessions'].apply(\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Me275_A.tsv\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_3946535/3122408642.py:17: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  noncoding['transcript_id'] = noncoding['accessions'].str.split(\":\", expand=True)[0]\n",
      "/tmp/ipykernel_3946535/3122408642.py:18: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  noncoding['transcript_id'] = noncoding['transcript_id'].str.split(\".\", expand=True)[0]\n",
      "/tmp/ipykernel_3946535/3122408642.py:19: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  noncoding['sample'] = folder\n",
      "/tmp/ipykernel_3946535/3122408642.py:24: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  coding['sample'] = folder\n",
      "/tmp/ipykernel_3946535/3122408642.py:25: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  coding['Entry'] = coding['accessions'].apply(\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "OD5P_A.tsv\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_3946535/3122408642.py:17: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  noncoding['transcript_id'] = noncoding['accessions'].str.split(\":\", expand=True)[0]\n",
      "/tmp/ipykernel_3946535/3122408642.py:18: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  noncoding['transcript_id'] = noncoding['transcript_id'].str.split(\".\", expand=True)[0]\n",
      "/tmp/ipykernel_3946535/3122408642.py:19: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  noncoding['sample'] = folder\n",
      "/tmp/ipykernel_3946535/3122408642.py:24: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  coding['sample'] = folder\n",
      "/tmp/ipykernel_3946535/3122408642.py:25: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  coding['Entry'] = coding['accessions'].apply(\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "T1015A_A.tsv\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_3946535/3122408642.py:17: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  noncoding['transcript_id'] = noncoding['accessions'].str.split(\":\", expand=True)[0]\n",
      "/tmp/ipykernel_3946535/3122408642.py:18: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  noncoding['transcript_id'] = noncoding['transcript_id'].str.split(\".\", expand=True)[0]\n",
      "/tmp/ipykernel_3946535/3122408642.py:19: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  noncoding['sample'] = folder\n",
      "/tmp/ipykernel_3946535/3122408642.py:24: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  coding['sample'] = folder\n",
      "/tmp/ipykernel_3946535/3122408642.py:25: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  coding['Entry'] = coding['accessions'].apply(\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "T1185B_A.tsv\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_3946535/3122408642.py:17: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  noncoding['transcript_id'] = noncoding['accessions'].str.split(\":\", expand=True)[0]\n",
      "/tmp/ipykernel_3946535/3122408642.py:18: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  noncoding['transcript_id'] = noncoding['transcript_id'].str.split(\".\", expand=True)[0]\n",
      "/tmp/ipykernel_3946535/3122408642.py:19: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  noncoding['sample'] = folder\n",
      "/tmp/ipykernel_3946535/3122408642.py:24: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  coding['sample'] = folder\n",
      "/tmp/ipykernel_3946535/3122408642.py:25: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  coding['Entry'] = coding['accessions'].apply(\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sequence</th>\n",
       "      <th>sample</th>\n",
       "      <th>chr</th>\n",
       "      <th>accessions</th>\n",
       "      <th>gene_id</th>\n",
       "      <th>gene_name_x</th>\n",
       "      <th>gene_name_y</th>\n",
       "      <th>n</th>\n",
       "      <th>transcript_type</th>\n",
       "      <th>gene_type_x</th>\n",
       "      <th>protein_references</th>\n",
       "      <th>orfID_TSTR_RiboSeq</th>\n",
       "      <th>ctype</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>SAYGEPRKL</td>\n",
       "      <td>T1185B</td>\n",
       "      <td>X</td>\n",
       "      <td>sp|P43355|MAGA1_HUMAN</td>\n",
       "      <td>ENSG00000198681</td>\n",
       "      <td>MAGEA1</td>\n",
       "      <td>MAGEA1</td>\n",
       "      <td>40.0</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>unique</td>\n",
       "      <td></td>\n",
       "      <td>BRCA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>SAYGEPRKL</td>\n",
       "      <td>T1185B</td>\n",
       "      <td>X</td>\n",
       "      <td>sp|P43355|MAGA1_HUMAN</td>\n",
       "      <td>ENSG00000198681</td>\n",
       "      <td>MAGEA1</td>\n",
       "      <td>MAGEA1</td>\n",
       "      <td>40.0</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>unique</td>\n",
       "      <td></td>\n",
       "      <td>BLCA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>SAYGEPRKL</td>\n",
       "      <td>T1185B</td>\n",
       "      <td>X</td>\n",
       "      <td>sp|P43355|MAGA1_HUMAN</td>\n",
       "      <td>ENSG00000198681</td>\n",
       "      <td>MAGEA1</td>\n",
       "      <td>MAGEA1</td>\n",
       "      <td>40.0</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>unique</td>\n",
       "      <td></td>\n",
       "      <td>LUAD</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>SAYGEPRKL</td>\n",
       "      <td>T1185B</td>\n",
       "      <td>X</td>\n",
       "      <td>sp|P43355|MAGA1_HUMAN</td>\n",
       "      <td>ENSG00000198681</td>\n",
       "      <td>MAGEA1</td>\n",
       "      <td>MAGEA1</td>\n",
       "      <td>40.0</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>unique</td>\n",
       "      <td></td>\n",
       "      <td>PRAD</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>SAYGEPRKL</td>\n",
       "      <td>T1185B</td>\n",
       "      <td>X</td>\n",
       "      <td>sp|P43355|MAGA1_HUMAN</td>\n",
       "      <td>ENSG00000198681</td>\n",
       "      <td>MAGEA1</td>\n",
       "      <td>MAGEA1</td>\n",
       "      <td>40.0</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>unique</td>\n",
       "      <td></td>\n",
       "      <td>LUSC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>723</th>\n",
       "      <td>AGRMSKSLVI</td>\n",
       "      <td>OMM475</td>\n",
       "      <td>5</td>\n",
       "      <td>ENST00000296682.4:5:+|97|3193:1750:3004|odORF|...</td>\n",
       "      <td>ENSG00000164256</td>\n",
       "      <td>PRDM9</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>non-unique</td>\n",
       "      <td></td>\n",
       "      <td>BLCA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>724</th>\n",
       "      <td>AGRMSKSLVI</td>\n",
       "      <td>OMM475</td>\n",
       "      <td>5</td>\n",
       "      <td>ENST00000296682.4:5:+|97|3193:1750:3004|odORF|...</td>\n",
       "      <td>ENSG00000164256</td>\n",
       "      <td>PRDM9</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>non-unique</td>\n",
       "      <td></td>\n",
       "      <td>LUSC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>725</th>\n",
       "      <td>AGRMSKSLVI</td>\n",
       "      <td>OMM475</td>\n",
       "      <td>5</td>\n",
       "      <td>ENST00000296682.4:5:+|97|3193:1750:3004|odORF|...</td>\n",
       "      <td>ENSG00000164256</td>\n",
       "      <td>PRDM9</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>non-unique</td>\n",
       "      <td></td>\n",
       "      <td>LIHC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>726</th>\n",
       "      <td>PTSLCLELSY</td>\n",
       "      <td>ONVC</td>\n",
       "      <td>1</td>\n",
       "      <td>ENST00000608183.1:1:-|68|2631:1213:1444|noncod...</td>\n",
       "      <td>ENSG00000228918</td>\n",
       "      <td>LINC01344</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>lncRNA</td>\n",
       "      <td>lncRNA</td>\n",
       "      <td>unique</td>\n",
       "      <td>ENST00000608183.1:1:-|29|2631:358:436|noncodin...</td>\n",
       "      <td>BRCA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>727</th>\n",
       "      <td>LGGAMLFSL</td>\n",
       "      <td>Me290</td>\n",
       "      <td>11</td>\n",
       "      <td>ENST00000531481.1:11:-|71|4381:1537:1669|dORF|...</td>\n",
       "      <td>ENSG00000186714</td>\n",
       "      <td>CCDC73</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>unique</td>\n",
       "      <td></td>\n",
       "      <td>COAD</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>728 rows × 13 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       sequence  sample chr  \\\n",
       "0     SAYGEPRKL  T1185B   X   \n",
       "1     SAYGEPRKL  T1185B   X   \n",
       "2     SAYGEPRKL  T1185B   X   \n",
       "3     SAYGEPRKL  T1185B   X   \n",
       "4     SAYGEPRKL  T1185B   X   \n",
       "..          ...     ...  ..   \n",
       "723  AGRMSKSLVI  OMM475   5   \n",
       "724  AGRMSKSLVI  OMM475   5   \n",
       "725  AGRMSKSLVI  OMM475   5   \n",
       "726  PTSLCLELSY    ONVC   1   \n",
       "727   LGGAMLFSL   Me290  11   \n",
       "\n",
       "                                            accessions          gene_id  \\\n",
       "0                                sp|P43355|MAGA1_HUMAN  ENSG00000198681   \n",
       "1                                sp|P43355|MAGA1_HUMAN  ENSG00000198681   \n",
       "2                                sp|P43355|MAGA1_HUMAN  ENSG00000198681   \n",
       "3                                sp|P43355|MAGA1_HUMAN  ENSG00000198681   \n",
       "4                                sp|P43355|MAGA1_HUMAN  ENSG00000198681   \n",
       "..                                                 ...              ...   \n",
       "723  ENST00000296682.4:5:+|97|3193:1750:3004|odORF|...  ENSG00000164256   \n",
       "724  ENST00000296682.4:5:+|97|3193:1750:3004|odORF|...  ENSG00000164256   \n",
       "725  ENST00000296682.4:5:+|97|3193:1750:3004|odORF|...  ENSG00000164256   \n",
       "726  ENST00000608183.1:1:-|68|2631:1213:1444|noncod...  ENSG00000228918   \n",
       "727  ENST00000531481.1:11:-|71|4381:1537:1669|dORF|...  ENSG00000186714   \n",
       "\n",
       "    gene_name_x gene_name_y     n transcript_type     gene_type_x  \\\n",
       "0        MAGEA1      MAGEA1  40.0  protein_coding  protein_coding   \n",
       "1        MAGEA1      MAGEA1  40.0  protein_coding  protein_coding   \n",
       "2        MAGEA1      MAGEA1  40.0  protein_coding  protein_coding   \n",
       "3        MAGEA1      MAGEA1  40.0  protein_coding  protein_coding   \n",
       "4        MAGEA1      MAGEA1  40.0  protein_coding  protein_coding   \n",
       "..          ...         ...   ...             ...             ...   \n",
       "723       PRDM9         NaN   NaN  protein_coding  protein_coding   \n",
       "724       PRDM9         NaN   NaN  protein_coding  protein_coding   \n",
       "725       PRDM9         NaN   NaN  protein_coding  protein_coding   \n",
       "726   LINC01344         NaN   NaN          lncRNA          lncRNA   \n",
       "727      CCDC73         NaN   NaN  protein_coding  protein_coding   \n",
       "\n",
       "    protein_references                                 orfID_TSTR_RiboSeq  \\\n",
       "0               unique                                                      \n",
       "1               unique                                                      \n",
       "2               unique                                                      \n",
       "3               unique                                                      \n",
       "4               unique                                                      \n",
       "..                 ...                                                ...   \n",
       "723         non-unique                                                      \n",
       "724         non-unique                                                      \n",
       "725         non-unique                                                      \n",
       "726             unique  ENST00000608183.1:1:-|29|2631:358:436|noncodin...   \n",
       "727             unique                                                      \n",
       "\n",
       "    ctype  \n",
       "0    BRCA  \n",
       "1    BLCA  \n",
       "2    LUAD  \n",
       "3    PRAD  \n",
       "4    LUSC  \n",
       "..    ...  \n",
       "723  BLCA  \n",
       "724  LUSC  \n",
       "725  LIHC  \n",
       "726  BRCA  \n",
       "727  COAD  \n",
       "\n",
       "[728 rows x 13 columns]"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "dir=\"/users/genomics/marta/241205_Immunopeptidomics_HCC/Melanoma_Chong/FDR5percent\"\n",
    "\n",
    "beijer_output_FDR5 = pd.DataFrame()\n",
    "coding_TSTR = pd.DataFrame()\n",
    "\n",
    "folders = [ f.name for f in os.scandir(dir) if f.is_dir() ]\n",
    "\n",
    "for folder in folders:\n",
    "    for file in os.listdir(os.path.join(dir,folder)):\n",
    "        statinfo = os.stat(os.path.join(dir,folder,file))\n",
    "        \n",
    "        if file.endswith(\"_A.tsv\") and statinfo.st_size > 129:\n",
    "            print(file)\n",
    "            output = pd.read_csv(os.path.join(dir,folder,file), sep=\"\\t\")\n",
    "            output['sequence'] = output['sequence'].str.replace('(Oxidation)', '')\n",
    "            noncoding = output[output['accessions'].str.contains(\"ENS\") | output['accessions'].str.contains(\"TCONS\") ]\n",
    "            noncoding['transcript_id'] = noncoding['accessions'].str.split(\":\", expand=True)[0]\n",
    "            noncoding['transcript_id'] = noncoding['transcript_id'].str.split(\".\", expand=True)[0]\n",
    "            noncoding['sample'] = folder\n",
    "\n",
    "            ## protein coding genes\n",
    "            coding = output[output['accessions'].str.contains(pattern_coding_genes, na=False)] \n",
    "            coding\n",
    "            coding['sample'] = folder\n",
    "            coding['Entry'] = coding['accessions'].apply(\n",
    "                            lambda x: re.search(pattern_coding_genes, x).group(0) if re.search(pattern_coding_genes, x) else None\n",
    "                        )\n",
    "            coding = annotated_geneNames[['Entry','transcript_id']].merge(coding, on=\"Entry\")\n",
    "            coding.drop(\"Entry\", axis=1, inplace=True)\n",
    "            coding_TSTR = pd.concat([coding,coding_TSTR])    \n",
    "            beijer_output_FDR5 = pd.concat([noncoding,beijer_output_FDR5])\n",
    "\n",
    "## add protein-coding\n",
    "coding_TSTR = coding_TSTR.merge(PCG_TSTR_SciAdv[['Entry','transcript_id']])\n",
    "coding_TSTR = coding_TSTR[beijer_output_FDR5.columns]\n",
    "beijer_output_FDR5_TSA = pd.concat([coding_TSTR, beijer_output_FDR5])\n",
    "beijer_output_FDR5_TSA = beijer_output_FDR5_TSA[beijer_output_FDR5_TSA['target_decoy'] == \"target\"]\n",
    "\n",
    "## TSA Science Advances\n",
    "beijer_output_FDR5_TSA = beijer_output_FDR5_TSA.merge(annot, on=\"transcript_id\")\n",
    "beijer_output_FDR5_TSA = beijer_output_FDR5_TSA.merge(TSASciAdv, on=\"gene_id\", how=\"left\")\n",
    "\n",
    "beijer_output_FDR5_TSA = beijer_output_FDR5_TSA[['sequence','sample','chr','accessions','gene_id','gene_name_x','gene_name_y','n','transcript_type','gene_type_x','protein_references']].sort_values(by=\"n\", ascending=False)\n",
    "beijer_output_FDR5_TSA.to_csv(os.path.join(dir,\"FDR5percent_TSA_analysis.csv\"), index=None)\n",
    "\n",
    "## TSTR\n",
    "beijer_output_FDR5_TSA_TSTR = beijer_output_FDR5_TSA.merge(TSTR, on=\"gene_id\")\n",
    "beijer_output_FDR5_TSA_TSTR.to_csv(os.path.join(dir,\"FDR5percent_TSA_TSTR_analysis.csv\"), index=None)\n",
    "beijer_output_FDR5_TSA_TSTR"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Melanoma Bassani 2016"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\u001b[33mNextflow 24.10.4 is available - Please consider updating your version to it\u001b[m\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "N E X T F L O W  ~  version 23.10.1\n",
      "Launching `https://github.com/nf-core/mhcquant` [distraught_lattes] DSL2 - revision: 1b3069246d [master]\n",
      "WARN: Access to undefined parameter `monochromeLogs` -- Initialise it to a default value eg. `params.monochromeLogs = some_value`\n",
      "\n",
      "\n",
      "------------------------------------------------------\n",
      "                                        ,--./,-.\n",
      "        ___     __   __   __   ___     /,-._.--~'\n",
      "  |\\ | |__  __ /  ` /  \\ |__) |__         }  {\n",
      "  | \\| |       \\__, \\__/ |  \\ |___     \\`-._,-`-,\n",
      "                                        `._,._,'\n",
      "  nf-core/mhcquant v2.6.0-g1b30692\n",
      "------------------------------------------------------\n",
      "Core Nextflow options\n",
      "  revision                 : master\n",
      "  runName                  : distraught_lattes\n",
      "  containerEngine          : singularity\n",
      "  launchDir                : /home/marta/PROJECT_SCRIPTS/241205_Immunopeptidomics_HCC\n",
      "  workDir                  : /home/marta/PROJECT_SCRIPTS/241205_Immunopeptidomics_HCC/work\n",
      "  projectDir               : /home/marta/.nextflow/assets/nf-core/mhcquant\n",
      "  userName                 : marta\n",
      "  profile                  : singularity\n",
      "  configFiles              : \n",
      "\n",
      "Input/output options\n",
      "  input                    : /home/marta/immunopeptidomics_rawData/samplesheet_MM29.tsv\n",
      "  outdir                   : /home/marta/241205_Immunopeptidomics_HCC/Melanoma_Bassani/FDR5percent/MM29\n",
      "\n",
      "Database Options\n",
      "  fasta                    : /home/marta/241205_Immunopeptidomics_HCC/database/TAA_SPTREMBL.fasta\n",
      "\n",
      "Database Search Settings\n",
      "  digest_mass_range        : 800:2500\n",
      "  prec_charge              : 2:3\n",
      "  fixed_mods               :  \n",
      "  default_params_file_comet:  \n",
      "\n",
      "Rescoring settings\n",
      "  fdr_threshold            : 0.05\n",
      "\n",
      "!! Only displaying parameters that differ from the pipeline defaults !!\n",
      "------------------------------------------------------\n",
      "If you use nf-core/mhcquant for your analysis please cite:\n",
      "\n",
      "* The pipeline\n",
      "  https://doi.org/10.1021/acs.jproteome.9b00313\n",
      "\n",
      "* The nf-core framework\n",
      "  https://doi.org/10.1038/s41587-020-0439-x\n",
      "\n",
      "* Software dependencies\n",
      "  https://github.com/nf-core/mhcquant/blob/master/CITATIONS.md\n",
      "------------------------------------------------------\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:MS... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:MU... -\n",
      "\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:MS... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:MU... -\n",
      "Pulling Singularity image https://depot.galaxyproject.org/singularity/openms:3.1.0--h8964181_3 [cache /home/marta/PROJECT_SCRIPTS/241205_Immunopeptidomics_HCC/work/singularity/depot.galaxyproject.org-singularity-openms-3.1.0--h8964181_3.img]\n",
      "Pulling Singularity image https://depot.galaxyproject.org/singularity/thermorawfileparser:1.4.3--ha8f3691_0 [cache /home/marta/PROJECT_SCRIPTS/241205_Immunopeptidomics_HCC/work/singularity/depot.galaxyproject.org-singularity-thermorawfileparser-1.4.3--ha8f3691_0.img]\n",
      "WARN: Singularity cache directory has not been defined -- Remote image will be stored in the path: /home/marta/PROJECT_SCRIPTS/241205_Immunopeptidomics_HCC/work/singularity -- Use the environment variable NXF_SINGULARITY_CACHEDIR to specify a different location\n",
      "\n",
      "executor >  local (3)\n",
      "[57/f292b3] process > NFCORE_MHCQUANT:MHCQUANT:PR... [  0%] 0 of 2\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[9b/8be917] process > NFCORE_MHCQUANT:MHCQUANT:OP... [  0%] 0 of 1\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:MS... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:MU... -\n",
      "Pulling Singularity image https://depot.galaxyproject.org/singularity/openms:3.1.0--h8964181_3 [cache /home/marta/PROJECT_SCRIPTS/241205_Immunopeptidomics_HCC/work/singularity/depot.galaxyproject.org-singularity-openms-3.1.0--h8964181_3.img]\n",
      "Pulling Singularity image https://depot.galaxyproject.org/singularity/thermorawfileparser:1.4.3--ha8f3691_0 [cache /home/marta/PROJECT_SCRIPTS/241205_Immunopeptidomics_HCC/work/singularity/depot.galaxyproject.org-singularity-thermorawfileparser-1.4.3--ha8f3691_0.img]\n",
      "WARN: Singularity cache directory has not been defined -- Remote image will be stored in the path: /home/marta/PROJECT_SCRIPTS/241205_Immunopeptidomics_HCC/work/singularity -- Use the environment variable NXF_SINGULARITY_CACHEDIR to specify a different location\n",
      "\n",
      "executor >  local (3)\n",
      "[57/f292b3] process > NFCORE_MHCQUANT:MHCQUANT:PR... [  0%] 0 of 2\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[9b/8be917] process > NFCORE_MHCQUANT:MHCQUANT:OP... [  0%] 0 of 1\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:MS... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:MU... -\n",
      "Pulling Singularity image https://depot.galaxyproject.org/singularity/openms:3.1.0--h8964181_3 [cache /home/marta/PROJECT_SCRIPTS/241205_Immunopeptidomics_HCC/work/singularity/depot.galaxyproject.org-singularity-openms-3.1.0--h8964181_3.img]\n",
      "Pulling Singularity image https://depot.galaxyproject.org/singularity/thermorawfileparser:1.4.3--ha8f3691_0 [cache /home/marta/PROJECT_SCRIPTS/241205_Immunopeptidomics_HCC/work/singularity/depot.galaxyproject.org-singularity-thermorawfileparser-1.4.3--ha8f3691_0.img]\n",
      "WARN: Singularity cache directory has not been defined -- Remote image will be stored in the path: /home/marta/PROJECT_SCRIPTS/241205_Immunopeptidomics_HCC/work/singularity -- Use the environment variable NXF_SINGULARITY_CACHEDIR to specify a different location\n",
      "\n",
      "executor >  local (3)\n",
      "[57/f292b3] process > NFCORE_MHCQUANT:MHCQUANT:PR... [  0%] 0 of 2\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[9b/8be917] process > NFCORE_MHCQUANT:MHCQUANT:OP... [100%] 1 of 1 ✔\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:MS... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:MU... -\n",
      "Pulling Singularity image https://depot.galaxyproject.org/singularity/openms:3.1.0--h8964181_3 [cache /home/marta/PROJECT_SCRIPTS/241205_Immunopeptidomics_HCC/work/singularity/depot.galaxyproject.org-singularity-openms-3.1.0--h8964181_3.img]\n",
      "Pulling Singularity image https://depot.galaxyproject.org/singularity/thermorawfileparser:1.4.3--ha8f3691_0 [cache /home/marta/PROJECT_SCRIPTS/241205_Immunopeptidomics_HCC/work/singularity/depot.galaxyproject.org-singularity-thermorawfileparser-1.4.3--ha8f3691_0.img]\n",
      "WARN: Singularity cache directory has not been defined -- Remote image will be stored in the path: /home/marta/PROJECT_SCRIPTS/241205_Immunopeptidomics_HCC/work/singularity -- Use the environment variable NXF_SINGULARITY_CACHEDIR to specify a different location\n",
      "\n",
      "executor >  local (3)\n",
      "[57/f292b3] process > NFCORE_MHCQUANT:MHCQUANT:PR... [ 50%] 1 of 2\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[9b/8be917] process > NFCORE_MHCQUANT:MHCQUANT:OP... [100%] 1 of 1 ✔\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:MS... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:MU... -\n",
      "Pulling Singularity image https://depot.galaxyproject.org/singularity/openms-thirdparty:3.1.0--h9ee0642_3 [cache /home/marta/PROJECT_SCRIPTS/241205_Immunopeptidomics_HCC/work/singularity/depot.galaxyproject.org-singularity-openms-thirdparty-3.1.0--h9ee0642_3.img]\n",
      "\n",
      "\n",
      "WARN: Singularity cache directory has not been defined -- Remote image will be stored in the path: /home/marta/PROJECT_SCRIPTS/241205_Immunopeptidomics_HCC/work/singularity -- Use the environment variable NXF_SINGULARITY_CACHEDIR to specify a different location\n",
      "\n",
      "\n",
      "executor >  local (3)\n",
      "[37/2c2f5c] process > NFCORE_MHCQUANT:MHCQUANT:PR... [100%] 2 of 2 ✔\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[9b/8be917] process > NFCORE_MHCQUANT:MHCQUANT:OP... [100%] 1 of 1 ✔\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:MS... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:MU... -\n",
      "Pulling Singularity image https://depot.galaxyproject.org/singularity/openms-thirdparty:3.1.0--h9ee0642_3 [cache /home/marta/PROJECT_SCRIPTS/241205_Immunopeptidomics_HCC/work/singularity/depot.galaxyproject.org-singularity-openms-thirdparty-3.1.0--h9ee0642_3.img]\n",
      "WARN: Singularity cache directory has not been defined -- Remote image will be stored in the path: /home/marta/PROJECT_SCRIPTS/241205_Immunopeptidomics_HCC/work/singularity -- Use the environment variable NXF_SINGULARITY_CACHEDIR to specify a different location\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "executor >  local (5)\n",
      "[37/2c2f5c] process > NFCORE_MHCQUANT:MHCQUANT:PR... [100%] 2 of 2 ✔\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[9b/8be917] process > NFCORE_MHCQUANT:MHCQUANT:OP... [100%] 1 of 1 ✔\n",
      "[f1/d1d731] process > NFCORE_MHCQUANT:MHCQUANT:OP... [  0%] 0 of 2\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:MS... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:MU... -\n",
      "Pulling Singularity image https://depot.galaxyproject.org/singularity/openms-thirdparty:3.1.0--h9ee0642_3 [cache /home/marta/PROJECT_SCRIPTS/241205_Immunopeptidomics_HCC/work/singularity/depot.galaxyproject.org-singularity-openms-thirdparty-3.1.0--h9ee0642_3.img]\n",
      "WARN: Singularity cache directory has not been defined -- Remote image will be stored in the path: /home/marta/PROJECT_SCRIPTS/241205_Immunopeptidomics_HCC/work/singularity -- Use the environment variable NXF_SINGULARITY_CACHEDIR to specify a different location\n",
      "\n",
      "executor >  local (6)\n",
      "[37/2c2f5c] process > NFCORE_MHCQUANT:MHCQUANT:PR... [100%] 2 of 2 ✔\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[9b/8be917] process > NFCORE_MHCQUANT:MHCQUANT:OP... [100%] 1 of 1 ✔\n",
      "[8a/df33fb] process > NFCORE_MHCQUANT:MHCQUANT:OP... [ 50%] 1 of 2\n",
      "[a0/9d05a0] process > NFCORE_MHCQUANT:MHCQUANT:OP... [  0%] 0 of 1\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:MS... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:MU... -\n",
      "\n",
      "\n",
      "\n",
      "executor >  local (6)\n",
      "[37/2c2f5c] process > NFCORE_MHCQUANT:MHCQUANT:PR... [100%] 2 of 2 ✔\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[9b/8be917] process > NFCORE_MHCQUANT:MHCQUANT:OP... [100%] 1 of 1 ✔\n",
      "[8a/df33fb] process > NFCORE_MHCQUANT:MHCQUANT:OP... [ 50%] 1 of 2\n",
      "[a0/9d05a0] process > NFCORE_MHCQUANT:MHCQUANT:OP... [  0%] 0 of 1\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:MS... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:MU... -\n",
      "\n",
      "\n",
      "\n",
      "executor >  local (7)\n",
      "[37/2c2f5c] process > NFCORE_MHCQUANT:MHCQUANT:PR... [100%] 2 of 2 ✔\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[9b/8be917] process > NFCORE_MHCQUANT:MHCQUANT:OP... [100%] 1 of 1 ✔\n",
      "[f1/d1d731] process > NFCORE_MHCQUANT:MHCQUANT:OP... [100%] 2 of 2 ✔\n",
      "[9f/90726a] process > NFCORE_MHCQUANT:MHCQUANT:OP... [  0%] 0 of 2\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:MS... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:MU... -\n",
      "\n",
      "executor >  local (7)\n",
      "[37/2c2f5c] process > NFCORE_MHCQUANT:MHCQUANT:PR... [100%] 2 of 2 ✔\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[9b/8be917] process > NFCORE_MHCQUANT:MHCQUANT:OP... [100%] 1 of 1 ✔\n",
      "[f1/d1d731] process > NFCORE_MHCQUANT:MHCQUANT:OP... [100%] 2 of 2 ✔\n",
      "[9f/90726a] process > NFCORE_MHCQUANT:MHCQUANT:OP... [  0%] 0 of 2\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:MS... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:MU... -\n",
      "\n",
      "executor >  local (7)\n",
      "[37/2c2f5c] process > NFCORE_MHCQUANT:MHCQUANT:PR... [100%] 2 of 2 ✔\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[9b/8be917] process > NFCORE_MHCQUANT:MHCQUANT:OP... [100%] 1 of 1 ✔\n",
      "[f1/d1d731] process > NFCORE_MHCQUANT:MHCQUANT:OP... [100%] 2 of 2 ✔\n",
      "[a0/9d05a0] process > NFCORE_MHCQUANT:MHCQUANT:OP... [ 50%] 1 of 2\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:MS... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:MU... -\n",
      "\n",
      "executor >  local (8)\n",
      "[37/2c2f5c] process > NFCORE_MHCQUANT:MHCQUANT:PR... [100%] 2 of 2 ✔\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[9b/8be917] process > NFCORE_MHCQUANT:MHCQUANT:OP... [100%] 1 of 1 ✔\n",
      "[f1/d1d731] process > NFCORE_MHCQUANT:MHCQUANT:OP... [100%] 2 of 2 ✔\n",
      "[9f/90726a] process > NFCORE_MHCQUANT:MHCQUANT:OP... [100%] 2 of 2 ✔\n",
      "[54/2f47aa] process > NFCORE_MHCQUANT:MHCQUANT:OP... [  0%] 0 of 1\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:MS... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:MU... -\n",
      "\n",
      "executor >  local (8)\n",
      "[37/2c2f5c] process > NFCORE_MHCQUANT:MHCQUANT:PR... [100%] 2 of 2 ✔\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[9b/8be917] process > NFCORE_MHCQUANT:MHCQUANT:OP... [100%] 1 of 1 ✔\n",
      "[f1/d1d731] process > NFCORE_MHCQUANT:MHCQUANT:OP... [100%] 2 of 2 ✔\n",
      "[9f/90726a] process > NFCORE_MHCQUANT:MHCQUANT:OP... [100%] 2 of 2 ✔\n",
      "[54/2f47aa] process > NFCORE_MHCQUANT:MHCQUANT:OP... [  0%] 0 of 1\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:MS... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:MU... -\n",
      "\n",
      "executor >  local (8)\n",
      "[37/2c2f5c] process > NFCORE_MHCQUANT:MHCQUANT:PR... [100%] 2 of 2 ✔\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[9b/8be917] process > NFCORE_MHCQUANT:MHCQUANT:OP... [100%] 1 of 1 ✔\n",
      "[f1/d1d731] process > NFCORE_MHCQUANT:MHCQUANT:OP... [100%] 2 of 2 ✔\n",
      "[9f/90726a] process > NFCORE_MHCQUANT:MHCQUANT:OP... [100%] 2 of 2 ✔\n",
      "[54/2f47aa] process > NFCORE_MHCQUANT:MHCQUANT:OP... [100%] 1 of 1 ✔\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:MS... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:MU... -\n",
      "Pulling Singularity image https://depot.galaxyproject.org/singularity/ms2rescore:3.0.1--pyhdfd78af_2 [cache /home/marta/PROJECT_SCRIPTS/241205_Immunopeptidomics_HCC/work/singularity/depot.galaxyproject.org-singularity-ms2rescore-3.0.1--pyhdfd78af_2.img]\n",
      "WARN: Singularity cache directory has not been defined -- Remote image will be stored in the path: /home/marta/PROJECT_SCRIPTS/241205_Immunopeptidomics_HCC/work/singularity -- Use the environment variable NXF_SINGULARITY_CACHEDIR to specify a different location\n",
      "\n",
      "executor >  local (9)\n",
      "[37/2c2f5c] process > NFCORE_MHCQUANT:MHCQUANT:PR... [100%] 2 of 2 ✔\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[9b/8be917] process > NFCORE_MHCQUANT:MHCQUANT:OP... [100%] 1 of 1 ✔\n",
      "[f1/d1d731] process > NFCORE_MHCQUANT:MHCQUANT:OP... [100%] 2 of 2 ✔\n",
      "[9f/90726a] process > NFCORE_MHCQUANT:MHCQUANT:OP... [100%] 2 of 2 ✔\n",
      "[54/2f47aa] process > NFCORE_MHCQUANT:MHCQUANT:OP... [100%] 1 of 1 ✔\n",
      "[1e/d9ec75] process > NFCORE_MHCQUANT:MHCQUANT:MS... [  0%] 0 of 1\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:MU... -\n",
      "Pulling Singularity image https://depot.galaxyproject.org/singularity/ms2rescore:3.0.1--pyhdfd78af_2 [cache /home/marta/PROJECT_SCRIPTS/241205_Immunopeptidomics_HCC/work/singularity/depot.galaxyproject.org-singularity-ms2rescore-3.0.1--pyhdfd78af_2.img]\n",
      "WARN: Singularity cache directory has not been defined -- Remote image will be stored in the path: /home/marta/PROJECT_SCRIPTS/241205_Immunopeptidomics_HCC/work/singularity -- Use the environment variable NXF_SINGULARITY_CACHEDIR to specify a different location\n",
      "\n",
      "executor >  local (10)\n",
      "[37/2c2f5c] process > NFCORE_MHCQUANT:MHCQUANT:PR... [100%] 2 of 2 ✔\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[9b/8be917] process > NFCORE_MHCQUANT:MHCQUANT:OP... [100%] 1 of 1 ✔\n",
      "[f1/d1d731] process > NFCORE_MHCQUANT:MHCQUANT:OP... [100%] 2 of 2 ✔\n",
      "[9f/90726a] process > NFCORE_MHCQUANT:MHCQUANT:OP... [100%] 2 of 2 ✔\n",
      "[54/2f47aa] process > NFCORE_MHCQUANT:MHCQUANT:OP... [100%] 1 of 1 ✔\n",
      "[1e/d9ec75] process > NFCORE_MHCQUANT:MHCQUANT:MS... [100%] 1 of 1 ✔\n",
      "[6b/98300f] process > NFCORE_MHCQUANT:MHCQUANT:OP... [  0%] 0 of 1\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:MU... -\n",
      "\n",
      "\n",
      "\n",
      "executor >  local (10)\n",
      "[37/2c2f5c] process > NFCORE_MHCQUANT:MHCQUANT:PR... [100%] 2 of 2 ✔\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[9b/8be917] process > NFCORE_MHCQUANT:MHCQUANT:OP... [100%] 1 of 1 ✔\n",
      "[f1/d1d731] process > NFCORE_MHCQUANT:MHCQUANT:OP... [100%] 2 of 2 ✔\n",
      "[9f/90726a] process > NFCORE_MHCQUANT:MHCQUANT:OP... [100%] 2 of 2 ✔\n",
      "[54/2f47aa] process > NFCORE_MHCQUANT:MHCQUANT:OP... [100%] 1 of 1 ✔\n",
      "[1e/d9ec75] process > NFCORE_MHCQUANT:MHCQUANT:MS... [100%] 1 of 1 ✔\n",
      "[6b/98300f] process > NFCORE_MHCQUANT:MHCQUANT:OP... [  0%] 0 of 1\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:MU... -\n",
      "\n",
      "\n",
      "\n",
      "executor >  local (11)\n",
      "[37/2c2f5c] process > NFCORE_MHCQUANT:MHCQUANT:PR... [100%] 2 of 2 ✔\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[9b/8be917] process > NFCORE_MHCQUANT:MHCQUANT:OP... [100%] 1 of 1 ✔\n",
      "[f1/d1d731] process > NFCORE_MHCQUANT:MHCQUANT:OP... [100%] 2 of 2 ✔\n",
      "[9f/90726a] process > NFCORE_MHCQUANT:MHCQUANT:OP... [100%] 2 of 2 ✔\n",
      "[54/2f47aa] process > NFCORE_MHCQUANT:MHCQUANT:OP... [100%] 1 of 1 ✔\n",
      "[1e/d9ec75] process > NFCORE_MHCQUANT:MHCQUANT:MS... [100%] 1 of 1 ✔\n",
      "[6b/98300f] process > NFCORE_MHCQUANT:MHCQUANT:OP... [100%] 1 of 1 ✔\n",
      "[d1/eb6a7b] process > NFCORE_MHCQUANT:MHCQUANT:OP... [  0%] 0 of 1\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:MU... -\n",
      "\n",
      "executor >  local (11)\n",
      "[37/2c2f5c] process > NFCORE_MHCQUANT:MHCQUANT:PR... [100%] 2 of 2 ✔\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[9b/8be917] process > NFCORE_MHCQUANT:MHCQUANT:OP... [100%] 1 of 1 ✔\n",
      "[f1/d1d731] process > NFCORE_MHCQUANT:MHCQUANT:OP... [100%] 2 of 2 ✔\n",
      "[9f/90726a] process > NFCORE_MHCQUANT:MHCQUANT:OP... [100%] 2 of 2 ✔\n",
      "[54/2f47aa] process > NFCORE_MHCQUANT:MHCQUANT:OP... [100%] 1 of 1 ✔\n",
      "[1e/d9ec75] process > NFCORE_MHCQUANT:MHCQUANT:MS... [100%] 1 of 1 ✔\n",
      "[6b/98300f] process > NFCORE_MHCQUANT:MHCQUANT:OP... [100%] 1 of 1 ✔\n",
      "[d1/eb6a7b] process > NFCORE_MHCQUANT:MHCQUANT:OP... [  0%] 0 of 1\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:MU... -\n",
      "\n",
      "executor >  local (11)\n",
      "[37/2c2f5c] process > NFCORE_MHCQUANT:MHCQUANT:PR... [100%] 2 of 2 ✔\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[9b/8be917] process > NFCORE_MHCQUANT:MHCQUANT:OP... [100%] 1 of 1 ✔\n",
      "[f1/d1d731] process > NFCORE_MHCQUANT:MHCQUANT:OP... [100%] 2 of 2 ✔\n",
      "[9f/90726a] process > NFCORE_MHCQUANT:MHCQUANT:OP... [100%] 2 of 2 ✔\n",
      "[54/2f47aa] process > NFCORE_MHCQUANT:MHCQUANT:OP... [100%] 1 of 1 ✔\n",
      "[1e/d9ec75] process > NFCORE_MHCQUANT:MHCQUANT:MS... [100%] 1 of 1 ✔\n",
      "[6b/98300f] process > NFCORE_MHCQUANT:MHCQUANT:OP... [100%] 1 of 1 ✔\n",
      "[d1/eb6a7b] process > NFCORE_MHCQUANT:MHCQUANT:OP... [100%] 1 of 1 ✔\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:MU... -\n",
      "\n",
      "executor >  local (12)\n",
      "[37/2c2f5c] process > NFCORE_MHCQUANT:MHCQUANT:PR... [100%] 2 of 2 ✔\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[9b/8be917] process > NFCORE_MHCQUANT:MHCQUANT:OP... [100%] 1 of 1 ✔\n",
      "[f1/d1d731] process > NFCORE_MHCQUANT:MHCQUANT:OP... [100%] 2 of 2 ✔\n",
      "[9f/90726a] process > NFCORE_MHCQUANT:MHCQUANT:OP... [100%] 2 of 2 ✔\n",
      "[54/2f47aa] process > NFCORE_MHCQUANT:MHCQUANT:OP... [100%] 1 of 1 ✔\n",
      "[1e/d9ec75] process > NFCORE_MHCQUANT:MHCQUANT:MS... [100%] 1 of 1 ✔\n",
      "[6b/98300f] process > NFCORE_MHCQUANT:MHCQUANT:OP... [100%] 1 of 1 ✔\n",
      "[d1/eb6a7b] process > NFCORE_MHCQUANT:MHCQUANT:OP... [100%] 1 of 1 ✔\n",
      "[57/f1a810] process > NFCORE_MHCQUANT:MHCQUANT:OP... [  0%] 0 of 1\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:OP... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:MU... -\n",
      "\n",
      "executor >  local (14)\n",
      "[37/2c2f5c] process > NFCORE_MHCQUANT:MHCQUANT:PR... [100%] 2 of 2 ✔\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[9b/8be917] process > NFCORE_MHCQUANT:MHCQUANT:OP... [100%] 1 of 1 ✔\n",
      "[f1/d1d731] process > NFCORE_MHCQUANT:MHCQUANT:OP... [100%] 2 of 2 ✔\n",
      "[9f/90726a] process > NFCORE_MHCQUANT:MHCQUANT:OP... [100%] 2 of 2 ✔\n",
      "[54/2f47aa] process > NFCORE_MHCQUANT:MHCQUANT:OP... [100%] 1 of 1 ✔\n",
      "[1e/d9ec75] process > NFCORE_MHCQUANT:MHCQUANT:MS... [100%] 1 of 1 ✔\n",
      "[6b/98300f] process > NFCORE_MHCQUANT:MHCQUANT:OP... [100%] 1 of 1 ✔\n",
      "[d1/eb6a7b] process > NFCORE_MHCQUANT:MHCQUANT:OP... [100%] 1 of 1 ✔\n",
      "[57/f1a810] process > NFCORE_MHCQUANT:MHCQUANT:OP... [100%] 1 of 1 ✔\n",
      "[62/93c6e4] process > NFCORE_MHCQUANT:MHCQUANT:OP... [  0%] 0 of 1\n",
      "[38/29704d] process > NFCORE_MHCQUANT:MHCQUANT:OP... [  0%] 0 of 1\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:MU... -\n",
      "\n",
      "executor >  local (14)\n",
      "[37/2c2f5c] process > NFCORE_MHCQUANT:MHCQUANT:PR... [100%] 2 of 2 ✔\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[9b/8be917] process > NFCORE_MHCQUANT:MHCQUANT:OP... [100%] 1 of 1 ✔\n",
      "[f1/d1d731] process > NFCORE_MHCQUANT:MHCQUANT:OP... [100%] 2 of 2 ✔\n",
      "[9f/90726a] process > NFCORE_MHCQUANT:MHCQUANT:OP... [100%] 2 of 2 ✔\n",
      "[54/2f47aa] process > NFCORE_MHCQUANT:MHCQUANT:OP... [100%] 1 of 1 ✔\n",
      "[1e/d9ec75] process > NFCORE_MHCQUANT:MHCQUANT:MS... [100%] 1 of 1 ✔\n",
      "[6b/98300f] process > NFCORE_MHCQUANT:MHCQUANT:OP... [100%] 1 of 1 ✔\n",
      "[d1/eb6a7b] process > NFCORE_MHCQUANT:MHCQUANT:OP... [100%] 1 of 1 ✔\n",
      "[57/f1a810] process > NFCORE_MHCQUANT:MHCQUANT:OP... [100%] 1 of 1 ✔\n",
      "[62/93c6e4] process > NFCORE_MHCQUANT:MHCQUANT:OP... [  0%] 0 of 1\n",
      "[38/29704d] process > NFCORE_MHCQUANT:MHCQUANT:OP... [  0%] 0 of 1\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:MU... -\n",
      "\n",
      "executor >  local (14)\n",
      "[37/2c2f5c] process > NFCORE_MHCQUANT:MHCQUANT:PR... [100%] 2 of 2 ✔\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[9b/8be917] process > NFCORE_MHCQUANT:MHCQUANT:OP... [100%] 1 of 1 ✔\n",
      "[f1/d1d731] process > NFCORE_MHCQUANT:MHCQUANT:OP... [100%] 2 of 2 ✔\n",
      "[9f/90726a] process > NFCORE_MHCQUANT:MHCQUANT:OP... [100%] 2 of 2 ✔\n",
      "[54/2f47aa] process > NFCORE_MHCQUANT:MHCQUANT:OP... [100%] 1 of 1 ✔\n",
      "[1e/d9ec75] process > NFCORE_MHCQUANT:MHCQUANT:MS... [100%] 1 of 1 ✔\n",
      "[6b/98300f] process > NFCORE_MHCQUANT:MHCQUANT:OP... [100%] 1 of 1 ✔\n",
      "[d1/eb6a7b] process > NFCORE_MHCQUANT:MHCQUANT:OP... [100%] 1 of 1 ✔\n",
      "[57/f1a810] process > NFCORE_MHCQUANT:MHCQUANT:OP... [100%] 1 of 1 ✔\n",
      "[62/93c6e4] process > NFCORE_MHCQUANT:MHCQUANT:OP... [100%] 1 of 1 ✔\n",
      "[38/29704d] process > NFCORE_MHCQUANT:MHCQUANT:OP... [  0%] 0 of 1\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:MU... -\n",
      "\n",
      "executor >  local (14)\n",
      "[37/2c2f5c] process > NFCORE_MHCQUANT:MHCQUANT:PR... [100%] 2 of 2 ✔\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[9b/8be917] process > NFCORE_MHCQUANT:MHCQUANT:OP... [100%] 1 of 1 ✔\n",
      "[f1/d1d731] process > NFCORE_MHCQUANT:MHCQUANT:OP... [100%] 2 of 2 ✔\n",
      "[9f/90726a] process > NFCORE_MHCQUANT:MHCQUANT:OP... [100%] 2 of 2 ✔\n",
      "[54/2f47aa] process > NFCORE_MHCQUANT:MHCQUANT:OP... [100%] 1 of 1 ✔\n",
      "[1e/d9ec75] process > NFCORE_MHCQUANT:MHCQUANT:MS... [100%] 1 of 1 ✔\n",
      "[6b/98300f] process > NFCORE_MHCQUANT:MHCQUANT:OP... [100%] 1 of 1 ✔\n",
      "[d1/eb6a7b] process > NFCORE_MHCQUANT:MHCQUANT:OP... [100%] 1 of 1 ✔\n",
      "[57/f1a810] process > NFCORE_MHCQUANT:MHCQUANT:OP... [100%] 1 of 1 ✔\n",
      "[62/93c6e4] process > NFCORE_MHCQUANT:MHCQUANT:OP... [100%] 1 of 1 ✔\n",
      "[38/29704d] process > NFCORE_MHCQUANT:MHCQUANT:OP... [100%] 1 of 1 ✔\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:MU... -\n",
      "\n",
      "executor >  local (14)\n",
      "[37/2c2f5c] process > NFCORE_MHCQUANT:MHCQUANT:PR... [100%] 2 of 2 ✔\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[9b/8be917] process > NFCORE_MHCQUANT:MHCQUANT:OP... [100%] 1 of 1 ✔\n",
      "[f1/d1d731] process > NFCORE_MHCQUANT:MHCQUANT:OP... [100%] 2 of 2 ✔\n",
      "[9f/90726a] process > NFCORE_MHCQUANT:MHCQUANT:OP... [100%] 2 of 2 ✔\n",
      "[54/2f47aa] process > NFCORE_MHCQUANT:MHCQUANT:OP... [100%] 1 of 1 ✔\n",
      "[1e/d9ec75] process > NFCORE_MHCQUANT:MHCQUANT:MS... [100%] 1 of 1 ✔\n",
      "[6b/98300f] process > NFCORE_MHCQUANT:MHCQUANT:OP... [100%] 1 of 1 ✔\n",
      "[d1/eb6a7b] process > NFCORE_MHCQUANT:MHCQUANT:OP... [100%] 1 of 1 ✔\n",
      "[57/f1a810] process > NFCORE_MHCQUANT:MHCQUANT:OP... [100%] 1 of 1 ✔\n",
      "[62/93c6e4] process > NFCORE_MHCQUANT:MHCQUANT:OP... [100%] 1 of 1 ✔\n",
      "[38/29704d] process > NFCORE_MHCQUANT:MHCQUANT:OP... [100%] 1 of 1 ✔\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:MU... -\n",
      "Pulling Singularity image https://depot.galaxyproject.org/singularity/multiqc:1.21--pyhdfd78af_0 [cache /home/marta/PROJECT_SCRIPTS/241205_Immunopeptidomics_HCC/work/singularity/depot.galaxyproject.org-singularity-multiqc-1.21--pyhdfd78af_0.img]\n",
      "WARN: Singularity cache directory has not been defined -- Remote image will be stored in the path: /home/marta/PROJECT_SCRIPTS/241205_Immunopeptidomics_HCC/work/singularity -- Use the environment variable NXF_SINGULARITY_CACHEDIR to specify a different location\n",
      "\n",
      "executor >  local (15)\n",
      "[37/2c2f5c] process > NFCORE_MHCQUANT:MHCQUANT:PR... [100%] 2 of 2 ✔\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[9b/8be917] process > NFCORE_MHCQUANT:MHCQUANT:OP... [100%] 1 of 1 ✔\n",
      "[f1/d1d731] process > NFCORE_MHCQUANT:MHCQUANT:OP... [100%] 2 of 2 ✔\n",
      "[9f/90726a] process > NFCORE_MHCQUANT:MHCQUANT:OP... [100%] 2 of 2 ✔\n",
      "[54/2f47aa] process > NFCORE_MHCQUANT:MHCQUANT:OP... [100%] 1 of 1 ✔\n",
      "[1e/d9ec75] process > NFCORE_MHCQUANT:MHCQUANT:MS... [100%] 1 of 1 ✔\n",
      "[6b/98300f] process > NFCORE_MHCQUANT:MHCQUANT:OP... [100%] 1 of 1 ✔\n",
      "[d1/eb6a7b] process > NFCORE_MHCQUANT:MHCQUANT:OP... [100%] 1 of 1 ✔\n",
      "[57/f1a810] process > NFCORE_MHCQUANT:MHCQUANT:OP... [100%] 1 of 1 ✔\n",
      "[62/93c6e4] process > NFCORE_MHCQUANT:MHCQUANT:OP... [100%] 1 of 1 ✔\n",
      "[38/29704d] process > NFCORE_MHCQUANT:MHCQUANT:OP... [100%] 1 of 1 ✔\n",
      "[27/2a6b65] process > NFCORE_MHCQUANT:MHCQUANT:MU... [  0%] 0 of 1\n",
      "Pulling Singularity image https://depot.galaxyproject.org/singularity/multiqc:1.21--pyhdfd78af_0 [cache /home/marta/PROJECT_SCRIPTS/241205_Immunopeptidomics_HCC/work/singularity/depot.galaxyproject.org-singularity-multiqc-1.21--pyhdfd78af_0.img]\n",
      "WARN: Singularity cache directory has not been defined -- Remote image will be stored in the path: /home/marta/PROJECT_SCRIPTS/241205_Immunopeptidomics_HCC/work/singularity -- Use the environment variable NXF_SINGULARITY_CACHEDIR to specify a different location\n",
      "\n",
      "executor >  local (15)\n",
      "[37/2c2f5c] process > NFCORE_MHCQUANT:MHCQUANT:PR... [100%] 2 of 2 ✔\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[9b/8be917] process > NFCORE_MHCQUANT:MHCQUANT:OP... [100%] 1 of 1 ✔\n",
      "[f1/d1d731] process > NFCORE_MHCQUANT:MHCQUANT:OP... [100%] 2 of 2 ✔\n",
      "[9f/90726a] process > NFCORE_MHCQUANT:MHCQUANT:OP... [100%] 2 of 2 ✔\n",
      "[54/2f47aa] process > NFCORE_MHCQUANT:MHCQUANT:OP... [100%] 1 of 1 ✔\n",
      "[1e/d9ec75] process > NFCORE_MHCQUANT:MHCQUANT:MS... [100%] 1 of 1 ✔\n",
      "[6b/98300f] process > NFCORE_MHCQUANT:MHCQUANT:OP... [100%] 1 of 1 ✔\n",
      "[d1/eb6a7b] process > NFCORE_MHCQUANT:MHCQUANT:OP... [100%] 1 of 1 ✔\n",
      "[57/f1a810] process > NFCORE_MHCQUANT:MHCQUANT:OP... [100%] 1 of 1 ✔\n",
      "[62/93c6e4] process > NFCORE_MHCQUANT:MHCQUANT:OP... [100%] 1 of 1 ✔\n",
      "[38/29704d] process > NFCORE_MHCQUANT:MHCQUANT:OP... [100%] 1 of 1 ✔\n",
      "[27/2a6b65] process > NFCORE_MHCQUANT:MHCQUANT:MU... [100%] 1 of 1 ✔\n",
      "Pulling Singularity image https://depot.galaxyproject.org/singularity/multiqc:1.21--pyhdfd78af_0 [cache /home/marta/PROJECT_SCRIPTS/241205_Immunopeptidomics_HCC/work/singularity/depot.galaxyproject.org-singularity-multiqc-1.21--pyhdfd78af_0.img]\n",
      "-[nf-core/mhcquant] Pipeline completed successfully-\n",
      "WARN: Singularity cache directory has not been defined -- Remote image will be stored in the path: /home/marta/PROJECT_SCRIPTS/241205_Immunopeptidomics_HCC/work/singularity -- Use the environment variable NXF_SINGULARITY_CACHEDIR to specify a different location\n",
      "\n",
      "executor >  local (15)\n",
      "[37/2c2f5c] process > NFCORE_MHCQUANT:MHCQUANT:PR... [100%] 2 of 2 ✔\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[-        ] process > NFCORE_MHCQUANT:MHCQUANT:PR... -\n",
      "[9b/8be917] process > NFCORE_MHCQUANT:MHCQUANT:OP... [100%] 1 of 1 ✔\n",
      "[f1/d1d731] process > NFCORE_MHCQUANT:MHCQUANT:OP... [100%] 2 of 2 ✔\n",
      "[9f/90726a] process > NFCORE_MHCQUANT:MHCQUANT:OP... [100%] 2 of 2 ✔\n",
      "[54/2f47aa] process > NFCORE_MHCQUANT:MHCQUANT:OP... [100%] 1 of 1 ✔\n",
      "[1e/d9ec75] process > NFCORE_MHCQUANT:MHCQUANT:MS... [100%] 1 of 1 ✔\n",
      "[6b/98300f] process > NFCORE_MHCQUANT:MHCQUANT:OP... [100%] 1 of 1 ✔\n",
      "[d1/eb6a7b] process > NFCORE_MHCQUANT:MHCQUANT:OP... [100%] 1 of 1 ✔\n",
      "[57/f1a810] process > NFCORE_MHCQUANT:MHCQUANT:OP... [100%] 1 of 1 ✔\n",
      "[62/93c6e4] process > NFCORE_MHCQUANT:MHCQUANT:OP... [100%] 1 of 1 ✔\n",
      "[38/29704d] process > NFCORE_MHCQUANT:MHCQUANT:OP... [100%] 1 of 1 ✔\n",
      "[27/2a6b65] process > NFCORE_MHCQUANT:MHCQUANT:MU... [100%] 1 of 1 ✔\n",
      "Pulling Singularity image https://depot.galaxyproject.org/singularity/multiqc:1.21--pyhdfd78af_0 [cache /home/marta/PROJECT_SCRIPTS/241205_Immunopeptidomics_HCC/work/singularity/depot.galaxyproject.org-singularity-multiqc-1.21--pyhdfd78af_0.img]\n",
      "-[nf-core/mhcquant] Pipeline completed successfully-\n",
      "WARN: Singularity cache directory has not been defined -- Remote image will be stored in the path: /home/marta/PROJECT_SCRIPTS/241205_Immunopeptidomics_HCC/work/singularity -- Use the environment variable NXF_SINGULARITY_CACHEDIR to specify a different location\n",
      "Completed at: 06-Feb-2025 10:36:26\n",
      "Duration    : 26m 45s\n",
      "CPU hours   : 4.7\n",
      "Succeeded   : 15\n",
      "\n",
      "\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "module load Java\n",
    "module load singularity\n",
    "module load Python\n",
    "\n",
    "FDR=0.05\n",
    "ref=/home/marta/241205_Immunopeptidomics_HCC/database/TAA_SPTREMBL.fasta\n",
    "\n",
    "## HLA-I\n",
    "# samples=\"MM20 MM21 MM24 MM25 MM26 MM27 MM28 MM29 MM30 MM33 MM34 MM35 MM36 MM39 MM40 MM41 MM42 MM12 MM8 MM5 MM4 MM3 MM15\"\n",
    "samples=\"MM29\"\n",
    "\n",
    "for s in $samples; do\n",
    "    outdir=/home/marta/241205_Immunopeptidomics_HCC/Melanoma_Bassani/FDR5percent/$s\n",
    "    mkdir -p $outdir\n",
    "\n",
    "    /home/marta/nextflow run nf-core/mhcquant -profile singularity --input /home/marta/immunopeptidomics_rawData/samplesheet_${s}.tsv --outdir $outdir --fasta $ref --fdr_threshold $FDR\n",
    "    # grep 'PEPTIDE' $outdir/${s}_A.tsv | grep -v 'UNASSIGNED' > $outdir/${s}_A_PEPTIDE.tsv\n",
    "    # grep -v 'non-unique' $outdir/${s}_A_PEPTIDE.tsv > $outdir/${s}_A_PEPTIDE.unique.tsv\n",
    "    grep -v 'non-unique' $outdir/${s}_A.tsv > $outdir/${s}_A_PEPTIDE.unique.tsv\n",
    "\n",
    "    if [ -s $outdir/${s}_A_PEPTIDE.unique.tsv ]; then\n",
    "        mv /home/marta/241205_Immunopeptidomics_HCC/Melanoma_Bassani/FDR5percent/$s /users/genomics/marta/241205_Immunopeptidomics_HCC/Melanoma_Bassani/FDR5percent\n",
    "    fi\n",
    "\n",
    "    rm -r work\n",
    "done"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "MM24_A.tsv\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_3946535/3563408073.py:17: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  noncoding['transcript_id'] = noncoding['accessions'].str.split(\":\", expand=True)[0]\n",
      "/tmp/ipykernel_3946535/3563408073.py:18: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  noncoding['transcript_id'] = noncoding['transcript_id'].str.split(\".\", expand=True)[0]\n",
      "/tmp/ipykernel_3946535/3563408073.py:19: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  noncoding['sample'] = folder\n",
      "/tmp/ipykernel_3946535/3563408073.py:24: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  coding['sample'] = folder\n",
      "/tmp/ipykernel_3946535/3563408073.py:25: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  coding['Entry'] = coding['accessions'].apply(\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "MM30_A.tsv\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_3946535/3563408073.py:17: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  noncoding['transcript_id'] = noncoding['accessions'].str.split(\":\", expand=True)[0]\n",
      "/tmp/ipykernel_3946535/3563408073.py:18: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  noncoding['transcript_id'] = noncoding['transcript_id'].str.split(\".\", expand=True)[0]\n",
      "/tmp/ipykernel_3946535/3563408073.py:19: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  noncoding['sample'] = folder\n",
      "/tmp/ipykernel_3946535/3563408073.py:24: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  coding['sample'] = folder\n",
      "/tmp/ipykernel_3946535/3563408073.py:25: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  coding['Entry'] = coding['accessions'].apply(\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "MM29_A.tsv\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_3946535/3563408073.py:17: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  noncoding['transcript_id'] = noncoding['accessions'].str.split(\":\", expand=True)[0]\n",
      "/tmp/ipykernel_3946535/3563408073.py:18: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  noncoding['transcript_id'] = noncoding['transcript_id'].str.split(\".\", expand=True)[0]\n",
      "/tmp/ipykernel_3946535/3563408073.py:19: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  noncoding['sample'] = folder\n",
      "/tmp/ipykernel_3946535/3563408073.py:24: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  coding['sample'] = folder\n",
      "/tmp/ipykernel_3946535/3563408073.py:25: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  coding['Entry'] = coding['accessions'].apply(\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "MM34_A.tsv\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_3946535/3563408073.py:17: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  noncoding['transcript_id'] = noncoding['accessions'].str.split(\":\", expand=True)[0]\n",
      "/tmp/ipykernel_3946535/3563408073.py:18: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  noncoding['transcript_id'] = noncoding['transcript_id'].str.split(\".\", expand=True)[0]\n",
      "/tmp/ipykernel_3946535/3563408073.py:19: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  noncoding['sample'] = folder\n",
      "/tmp/ipykernel_3946535/3563408073.py:24: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  coding['sample'] = folder\n",
      "/tmp/ipykernel_3946535/3563408073.py:25: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  coding['Entry'] = coding['accessions'].apply(\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "MM33_A.tsv\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_3946535/3563408073.py:17: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  noncoding['transcript_id'] = noncoding['accessions'].str.split(\":\", expand=True)[0]\n",
      "/tmp/ipykernel_3946535/3563408073.py:18: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  noncoding['transcript_id'] = noncoding['transcript_id'].str.split(\".\", expand=True)[0]\n",
      "/tmp/ipykernel_3946535/3563408073.py:19: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  noncoding['sample'] = folder\n",
      "/tmp/ipykernel_3946535/3563408073.py:24: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  coding['sample'] = folder\n",
      "/tmp/ipykernel_3946535/3563408073.py:25: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  coding['Entry'] = coding['accessions'].apply(\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "MM39_A.tsv\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_3946535/3563408073.py:17: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  noncoding['transcript_id'] = noncoding['accessions'].str.split(\":\", expand=True)[0]\n",
      "/tmp/ipykernel_3946535/3563408073.py:18: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  noncoding['transcript_id'] = noncoding['transcript_id'].str.split(\".\", expand=True)[0]\n",
      "/tmp/ipykernel_3946535/3563408073.py:19: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  noncoding['sample'] = folder\n",
      "/tmp/ipykernel_3946535/3563408073.py:24: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  coding['sample'] = folder\n",
      "/tmp/ipykernel_3946535/3563408073.py:25: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  coding['Entry'] = coding['accessions'].apply(\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "MM28_A.tsv\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_3946535/3563408073.py:17: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  noncoding['transcript_id'] = noncoding['accessions'].str.split(\":\", expand=True)[0]\n",
      "/tmp/ipykernel_3946535/3563408073.py:18: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  noncoding['transcript_id'] = noncoding['transcript_id'].str.split(\".\", expand=True)[0]\n",
      "/tmp/ipykernel_3946535/3563408073.py:19: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  noncoding['sample'] = folder\n",
      "/tmp/ipykernel_3946535/3563408073.py:24: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  coding['sample'] = folder\n",
      "/tmp/ipykernel_3946535/3563408073.py:25: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  coding['Entry'] = coding['accessions'].apply(\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "MM36_A.tsv\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_3946535/3563408073.py:17: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  noncoding['transcript_id'] = noncoding['accessions'].str.split(\":\", expand=True)[0]\n",
      "/tmp/ipykernel_3946535/3563408073.py:18: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  noncoding['transcript_id'] = noncoding['transcript_id'].str.split(\".\", expand=True)[0]\n",
      "/tmp/ipykernel_3946535/3563408073.py:19: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  noncoding['sample'] = folder\n",
      "/tmp/ipykernel_3946535/3563408073.py:24: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  coding['sample'] = folder\n",
      "/tmp/ipykernel_3946535/3563408073.py:25: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  coding['Entry'] = coding['accessions'].apply(\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "MM25_A.tsv\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_3946535/3563408073.py:17: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  noncoding['transcript_id'] = noncoding['accessions'].str.split(\":\", expand=True)[0]\n",
      "/tmp/ipykernel_3946535/3563408073.py:18: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  noncoding['transcript_id'] = noncoding['transcript_id'].str.split(\".\", expand=True)[0]\n",
      "/tmp/ipykernel_3946535/3563408073.py:19: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  noncoding['sample'] = folder\n",
      "/tmp/ipykernel_3946535/3563408073.py:24: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  coding['sample'] = folder\n",
      "/tmp/ipykernel_3946535/3563408073.py:25: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  coding['Entry'] = coding['accessions'].apply(\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "MM5_A.tsv\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_3946535/3563408073.py:17: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  noncoding['transcript_id'] = noncoding['accessions'].str.split(\":\", expand=True)[0]\n",
      "/tmp/ipykernel_3946535/3563408073.py:18: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  noncoding['transcript_id'] = noncoding['transcript_id'].str.split(\".\", expand=True)[0]\n",
      "/tmp/ipykernel_3946535/3563408073.py:19: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  noncoding['sample'] = folder\n",
      "/tmp/ipykernel_3946535/3563408073.py:24: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  coding['sample'] = folder\n",
      "/tmp/ipykernel_3946535/3563408073.py:25: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  coding['Entry'] = coding['accessions'].apply(\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "MM8_A.tsv\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_3946535/3563408073.py:17: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  noncoding['transcript_id'] = noncoding['accessions'].str.split(\":\", expand=True)[0]\n",
      "/tmp/ipykernel_3946535/3563408073.py:18: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  noncoding['transcript_id'] = noncoding['transcript_id'].str.split(\".\", expand=True)[0]\n",
      "/tmp/ipykernel_3946535/3563408073.py:19: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  noncoding['sample'] = folder\n",
      "/tmp/ipykernel_3946535/3563408073.py:24: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  coding['sample'] = folder\n",
      "/tmp/ipykernel_3946535/3563408073.py:25: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  coding['Entry'] = coding['accessions'].apply(\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "MM42_A.tsv\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_3946535/3563408073.py:17: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  noncoding['transcript_id'] = noncoding['accessions'].str.split(\":\", expand=True)[0]\n",
      "/tmp/ipykernel_3946535/3563408073.py:18: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  noncoding['transcript_id'] = noncoding['transcript_id'].str.split(\".\", expand=True)[0]\n",
      "/tmp/ipykernel_3946535/3563408073.py:19: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  noncoding['sample'] = folder\n",
      "/tmp/ipykernel_3946535/3563408073.py:24: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  coding['sample'] = folder\n",
      "/tmp/ipykernel_3946535/3563408073.py:25: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  coding['Entry'] = coding['accessions'].apply(\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "MM21_A.tsv\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_3946535/3563408073.py:17: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  noncoding['transcript_id'] = noncoding['accessions'].str.split(\":\", expand=True)[0]\n",
      "/tmp/ipykernel_3946535/3563408073.py:18: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  noncoding['transcript_id'] = noncoding['transcript_id'].str.split(\".\", expand=True)[0]\n",
      "/tmp/ipykernel_3946535/3563408073.py:19: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  noncoding['sample'] = folder\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "MM35_A.tsv\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_3946535/3563408073.py:17: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  noncoding['transcript_id'] = noncoding['accessions'].str.split(\":\", expand=True)[0]\n",
      "/tmp/ipykernel_3946535/3563408073.py:18: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  noncoding['transcript_id'] = noncoding['transcript_id'].str.split(\".\", expand=True)[0]\n",
      "/tmp/ipykernel_3946535/3563408073.py:19: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  noncoding['sample'] = folder\n",
      "/tmp/ipykernel_3946535/3563408073.py:24: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  coding['sample'] = folder\n",
      "/tmp/ipykernel_3946535/3563408073.py:25: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  coding['Entry'] = coding['accessions'].apply(\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "MM3_A.tsv\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_3946535/3563408073.py:17: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  noncoding['transcript_id'] = noncoding['accessions'].str.split(\":\", expand=True)[0]\n",
      "/tmp/ipykernel_3946535/3563408073.py:18: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  noncoding['transcript_id'] = noncoding['transcript_id'].str.split(\".\", expand=True)[0]\n",
      "/tmp/ipykernel_3946535/3563408073.py:19: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  noncoding['sample'] = folder\n",
      "/tmp/ipykernel_3946535/3563408073.py:24: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  coding['sample'] = folder\n",
      "/tmp/ipykernel_3946535/3563408073.py:25: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  coding['Entry'] = coding['accessions'].apply(\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "MM4_A.tsv\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_3946535/3563408073.py:17: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  noncoding['transcript_id'] = noncoding['accessions'].str.split(\":\", expand=True)[0]\n",
      "/tmp/ipykernel_3946535/3563408073.py:18: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  noncoding['transcript_id'] = noncoding['transcript_id'].str.split(\".\", expand=True)[0]\n",
      "/tmp/ipykernel_3946535/3563408073.py:19: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  noncoding['sample'] = folder\n",
      "/tmp/ipykernel_3946535/3563408073.py:24: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  coding['sample'] = folder\n",
      "/tmp/ipykernel_3946535/3563408073.py:25: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  coding['Entry'] = coding['accessions'].apply(\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "MM12_A.tsv\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_3946535/3563408073.py:17: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  noncoding['transcript_id'] = noncoding['accessions'].str.split(\":\", expand=True)[0]\n",
      "/tmp/ipykernel_3946535/3563408073.py:18: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  noncoding['transcript_id'] = noncoding['transcript_id'].str.split(\".\", expand=True)[0]\n",
      "/tmp/ipykernel_3946535/3563408073.py:19: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  noncoding['sample'] = folder\n",
      "/tmp/ipykernel_3946535/3563408073.py:24: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  coding['sample'] = folder\n",
      "/tmp/ipykernel_3946535/3563408073.py:25: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  coding['Entry'] = coding['accessions'].apply(\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "MM20_A.tsv\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_3946535/3563408073.py:17: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  noncoding['transcript_id'] = noncoding['accessions'].str.split(\":\", expand=True)[0]\n",
      "/tmp/ipykernel_3946535/3563408073.py:18: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  noncoding['transcript_id'] = noncoding['transcript_id'].str.split(\".\", expand=True)[0]\n",
      "/tmp/ipykernel_3946535/3563408073.py:19: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  noncoding['sample'] = folder\n",
      "/tmp/ipykernel_3946535/3563408073.py:24: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  coding['sample'] = folder\n",
      "/tmp/ipykernel_3946535/3563408073.py:25: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  coding['Entry'] = coding['accessions'].apply(\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "MM40_A.tsv\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_3946535/3563408073.py:17: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  noncoding['transcript_id'] = noncoding['accessions'].str.split(\":\", expand=True)[0]\n",
      "/tmp/ipykernel_3946535/3563408073.py:18: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  noncoding['transcript_id'] = noncoding['transcript_id'].str.split(\".\", expand=True)[0]\n",
      "/tmp/ipykernel_3946535/3563408073.py:19: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  noncoding['sample'] = folder\n",
      "/tmp/ipykernel_3946535/3563408073.py:24: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  coding['sample'] = folder\n",
      "/tmp/ipykernel_3946535/3563408073.py:25: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  coding['Entry'] = coding['accessions'].apply(\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "MM41_A.tsv\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_3946535/3563408073.py:17: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  noncoding['transcript_id'] = noncoding['accessions'].str.split(\":\", expand=True)[0]\n",
      "/tmp/ipykernel_3946535/3563408073.py:18: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  noncoding['transcript_id'] = noncoding['transcript_id'].str.split(\".\", expand=True)[0]\n",
      "/tmp/ipykernel_3946535/3563408073.py:19: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  noncoding['sample'] = folder\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "MM26_A.tsv\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_3946535/3563408073.py:17: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  noncoding['transcript_id'] = noncoding['accessions'].str.split(\":\", expand=True)[0]\n",
      "/tmp/ipykernel_3946535/3563408073.py:18: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  noncoding['transcript_id'] = noncoding['transcript_id'].str.split(\".\", expand=True)[0]\n",
      "/tmp/ipykernel_3946535/3563408073.py:19: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  noncoding['sample'] = folder\n",
      "/tmp/ipykernel_3946535/3563408073.py:24: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  coding['sample'] = folder\n",
      "/tmp/ipykernel_3946535/3563408073.py:25: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  coding['Entry'] = coding['accessions'].apply(\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "MM15_A.tsv\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_3946535/3563408073.py:17: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  noncoding['transcript_id'] = noncoding['accessions'].str.split(\":\", expand=True)[0]\n",
      "/tmp/ipykernel_3946535/3563408073.py:18: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  noncoding['transcript_id'] = noncoding['transcript_id'].str.split(\".\", expand=True)[0]\n",
      "/tmp/ipykernel_3946535/3563408073.py:19: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  noncoding['sample'] = folder\n",
      "/tmp/ipykernel_3946535/3563408073.py:24: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  coding['sample'] = folder\n",
      "/tmp/ipykernel_3946535/3563408073.py:25: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  coding['Entry'] = coding['accessions'].apply(\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "MM27_A.tsv\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_3946535/3563408073.py:17: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  noncoding['transcript_id'] = noncoding['accessions'].str.split(\":\", expand=True)[0]\n",
      "/tmp/ipykernel_3946535/3563408073.py:18: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  noncoding['transcript_id'] = noncoding['transcript_id'].str.split(\".\", expand=True)[0]\n",
      "/tmp/ipykernel_3946535/3563408073.py:19: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  noncoding['sample'] = folder\n",
      "/tmp/ipykernel_3946535/3563408073.py:24: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  coding['sample'] = folder\n",
      "/tmp/ipykernel_3946535/3563408073.py:25: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  coding['Entry'] = coding['accessions'].apply(\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sequence</th>\n",
       "      <th>sample</th>\n",
       "      <th>chr</th>\n",
       "      <th>accessions</th>\n",
       "      <th>gene_id</th>\n",
       "      <th>gene_name_x</th>\n",
       "      <th>gene_name_y</th>\n",
       "      <th>n</th>\n",
       "      <th>transcript_type</th>\n",
       "      <th>gene_type_x</th>\n",
       "      <th>protein_references</th>\n",
       "      <th>orfID_TSTR_RiboSeq</th>\n",
       "      <th>ctype</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>KEADPTGHSYVL</td>\n",
       "      <td>MM36</td>\n",
       "      <td>X</td>\n",
       "      <td>sp|P43355|MAGA1_HUMAN</td>\n",
       "      <td>ENSG00000198681</td>\n",
       "      <td>MAGEA1</td>\n",
       "      <td>MAGEA1</td>\n",
       "      <td>40.0</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>unique</td>\n",
       "      <td></td>\n",
       "      <td>BRCA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>KEADPTGHSYVL</td>\n",
       "      <td>MM36</td>\n",
       "      <td>X</td>\n",
       "      <td>sp|P43355|MAGA1_HUMAN</td>\n",
       "      <td>ENSG00000198681</td>\n",
       "      <td>MAGEA1</td>\n",
       "      <td>MAGEA1</td>\n",
       "      <td>40.0</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>unique</td>\n",
       "      <td></td>\n",
       "      <td>BLCA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>KEADPTGHSYVL</td>\n",
       "      <td>MM36</td>\n",
       "      <td>X</td>\n",
       "      <td>sp|P43355|MAGA1_HUMAN</td>\n",
       "      <td>ENSG00000198681</td>\n",
       "      <td>MAGEA1</td>\n",
       "      <td>MAGEA1</td>\n",
       "      <td>40.0</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>unique</td>\n",
       "      <td></td>\n",
       "      <td>LUAD</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>KEADPTGHSYVL</td>\n",
       "      <td>MM36</td>\n",
       "      <td>X</td>\n",
       "      <td>sp|P43355|MAGA1_HUMAN</td>\n",
       "      <td>ENSG00000198681</td>\n",
       "      <td>MAGEA1</td>\n",
       "      <td>MAGEA1</td>\n",
       "      <td>40.0</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>unique</td>\n",
       "      <td></td>\n",
       "      <td>PRAD</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>KEADPTGHSYVL</td>\n",
       "      <td>MM36</td>\n",
       "      <td>X</td>\n",
       "      <td>sp|P43355|MAGA1_HUMAN</td>\n",
       "      <td>ENSG00000198681</td>\n",
       "      <td>MAGEA1</td>\n",
       "      <td>MAGEA1</td>\n",
       "      <td>40.0</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>unique</td>\n",
       "      <td></td>\n",
       "      <td>LUSC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1414</th>\n",
       "      <td>RVVEMLLHR</td>\n",
       "      <td>MM34</td>\n",
       "      <td>13</td>\n",
       "      <td>ENST00000813109.1:13:-|9|877:205:448|noncoding...</td>\n",
       "      <td>ENSG00000287861</td>\n",
       "      <td>ENSG00000287861</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>lncRNA</td>\n",
       "      <td>lncRNA</td>\n",
       "      <td>unique</td>\n",
       "      <td>ENST00000813109.1:13:-|10|877:247:448|noncodin...</td>\n",
       "      <td>BLCA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1415</th>\n",
       "      <td>RVVEMLLHR</td>\n",
       "      <td>MM34</td>\n",
       "      <td>13</td>\n",
       "      <td>ENST00000813109.1:13:-|9|877:205:448|noncoding...</td>\n",
       "      <td>ENSG00000287861</td>\n",
       "      <td>ENSG00000287861</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>lncRNA</td>\n",
       "      <td>lncRNA</td>\n",
       "      <td>unique</td>\n",
       "      <td>ENST00000813109.1:13:-|10|877:247:448|noncodin...</td>\n",
       "      <td>LUAD</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1416</th>\n",
       "      <td>RVVEMLLHR</td>\n",
       "      <td>MM34</td>\n",
       "      <td>13</td>\n",
       "      <td>ENST00000813109.1:13:-|9|877:205:448|noncoding...</td>\n",
       "      <td>ENSG00000287861</td>\n",
       "      <td>ENSG00000287861</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>lncRNA</td>\n",
       "      <td>lncRNA</td>\n",
       "      <td>unique</td>\n",
       "      <td>ENST00000813109.1:13:-|10|877:247:448|noncodin...</td>\n",
       "      <td>KIRC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1417</th>\n",
       "      <td>RVVEMLLHR</td>\n",
       "      <td>MM34</td>\n",
       "      <td>13</td>\n",
       "      <td>ENST00000813109.1:13:-|9|877:205:448|noncoding...</td>\n",
       "      <td>ENSG00000287861</td>\n",
       "      <td>ENSG00000287861</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>lncRNA</td>\n",
       "      <td>lncRNA</td>\n",
       "      <td>unique</td>\n",
       "      <td>ENST00000813109.1:13:-|10|877:247:448|noncodin...</td>\n",
       "      <td>PRAD</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1418</th>\n",
       "      <td>RVVEMLLHR</td>\n",
       "      <td>MM34</td>\n",
       "      <td>13</td>\n",
       "      <td>ENST00000813109.1:13:-|9|877:205:448|noncoding...</td>\n",
       "      <td>ENSG00000287861</td>\n",
       "      <td>ENSG00000287861</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>lncRNA</td>\n",
       "      <td>lncRNA</td>\n",
       "      <td>unique</td>\n",
       "      <td>ENST00000813109.1:13:-|10|877:247:448|noncodin...</td>\n",
       "      <td>LUSC</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1419 rows × 13 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "          sequence sample chr  \\\n",
       "0     KEADPTGHSYVL   MM36   X   \n",
       "1     KEADPTGHSYVL   MM36   X   \n",
       "2     KEADPTGHSYVL   MM36   X   \n",
       "3     KEADPTGHSYVL   MM36   X   \n",
       "4     KEADPTGHSYVL   MM36   X   \n",
       "...            ...    ...  ..   \n",
       "1414     RVVEMLLHR   MM34  13   \n",
       "1415     RVVEMLLHR   MM34  13   \n",
       "1416     RVVEMLLHR   MM34  13   \n",
       "1417     RVVEMLLHR   MM34  13   \n",
       "1418     RVVEMLLHR   MM34  13   \n",
       "\n",
       "                                             accessions          gene_id  \\\n",
       "0                                 sp|P43355|MAGA1_HUMAN  ENSG00000198681   \n",
       "1                                 sp|P43355|MAGA1_HUMAN  ENSG00000198681   \n",
       "2                                 sp|P43355|MAGA1_HUMAN  ENSG00000198681   \n",
       "3                                 sp|P43355|MAGA1_HUMAN  ENSG00000198681   \n",
       "4                                 sp|P43355|MAGA1_HUMAN  ENSG00000198681   \n",
       "...                                                 ...              ...   \n",
       "1414  ENST00000813109.1:13:-|9|877:205:448|noncoding...  ENSG00000287861   \n",
       "1415  ENST00000813109.1:13:-|9|877:205:448|noncoding...  ENSG00000287861   \n",
       "1416  ENST00000813109.1:13:-|9|877:205:448|noncoding...  ENSG00000287861   \n",
       "1417  ENST00000813109.1:13:-|9|877:205:448|noncoding...  ENSG00000287861   \n",
       "1418  ENST00000813109.1:13:-|9|877:205:448|noncoding...  ENSG00000287861   \n",
       "\n",
       "          gene_name_x gene_name_y     n transcript_type     gene_type_x  \\\n",
       "0              MAGEA1      MAGEA1  40.0  protein_coding  protein_coding   \n",
       "1              MAGEA1      MAGEA1  40.0  protein_coding  protein_coding   \n",
       "2              MAGEA1      MAGEA1  40.0  protein_coding  protein_coding   \n",
       "3              MAGEA1      MAGEA1  40.0  protein_coding  protein_coding   \n",
       "4              MAGEA1      MAGEA1  40.0  protein_coding  protein_coding   \n",
       "...               ...         ...   ...             ...             ...   \n",
       "1414  ENSG00000287861         NaN   NaN          lncRNA          lncRNA   \n",
       "1415  ENSG00000287861         NaN   NaN          lncRNA          lncRNA   \n",
       "1416  ENSG00000287861         NaN   NaN          lncRNA          lncRNA   \n",
       "1417  ENSG00000287861         NaN   NaN          lncRNA          lncRNA   \n",
       "1418  ENSG00000287861         NaN   NaN          lncRNA          lncRNA   \n",
       "\n",
       "     protein_references                                 orfID_TSTR_RiboSeq  \\\n",
       "0                unique                                                      \n",
       "1                unique                                                      \n",
       "2                unique                                                      \n",
       "3                unique                                                      \n",
       "4                unique                                                      \n",
       "...                 ...                                                ...   \n",
       "1414             unique  ENST00000813109.1:13:-|10|877:247:448|noncodin...   \n",
       "1415             unique  ENST00000813109.1:13:-|10|877:247:448|noncodin...   \n",
       "1416             unique  ENST00000813109.1:13:-|10|877:247:448|noncodin...   \n",
       "1417             unique  ENST00000813109.1:13:-|10|877:247:448|noncodin...   \n",
       "1418             unique  ENST00000813109.1:13:-|10|877:247:448|noncodin...   \n",
       "\n",
       "     ctype  \n",
       "0     BRCA  \n",
       "1     BLCA  \n",
       "2     LUAD  \n",
       "3     PRAD  \n",
       "4     LUSC  \n",
       "...    ...  \n",
       "1414  BLCA  \n",
       "1415  LUAD  \n",
       "1416  KIRC  \n",
       "1417  PRAD  \n",
       "1418  LUSC  \n",
       "\n",
       "[1419 rows x 13 columns]"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "dir=\"/users/genomics/marta/241205_Immunopeptidomics_HCC/Melanoma_Bassani/FDR5percent\"\n",
    "\n",
    "beijer_output_FDR5 = pd.DataFrame()\n",
    "coding_TSTR = pd.DataFrame()\n",
    "\n",
    "folders = [ f.name for f in os.scandir(dir) if f.is_dir() ]\n",
    "\n",
    "for folder in folders:\n",
    "    for file in os.listdir(os.path.join(dir,folder)):\n",
    "        statinfo = os.stat(os.path.join(dir,folder,file))\n",
    "        \n",
    "        if file.endswith(\"_A.tsv\") and statinfo.st_size > 129:\n",
    "            print(file)\n",
    "            output = pd.read_csv(os.path.join(dir,folder,file), sep=\"\\t\")\n",
    "            output['sequence'] = output['sequence'].str.replace('(Oxidation)', '')\n",
    "            noncoding = output[output['accessions'].str.contains(\"ENS\") | output['accessions'].str.contains(\"TCONS\") ]\n",
    "            noncoding['transcript_id'] = noncoding['accessions'].str.split(\":\", expand=True)[0]\n",
    "            noncoding['transcript_id'] = noncoding['transcript_id'].str.split(\".\", expand=True)[0]\n",
    "            noncoding['sample'] = folder\n",
    "\n",
    "            ## protein coding genes\n",
    "            coding = output[output['accessions'].str.contains(pattern_coding_genes, na=False)] \n",
    "            coding\n",
    "            coding['sample'] = folder\n",
    "            coding['Entry'] = coding['accessions'].apply(\n",
    "                            lambda x: re.search(pattern_coding_genes, x).group(0) if re.search(pattern_coding_genes, x) else None\n",
    "                        )\n",
    "            coding = annotated_geneNames[['Entry','transcript_id']].merge(coding, on=\"Entry\")\n",
    "            coding.drop(\"Entry\", axis=1, inplace=True)\n",
    "            coding_TSTR = pd.concat([coding,coding_TSTR])    \n",
    "            beijer_output_FDR5 = pd.concat([noncoding,beijer_output_FDR5])\n",
    "\n",
    "## add protein-coding\n",
    "coding_TSTR = coding_TSTR.merge(PCG_TSTR_SciAdv[['Entry','transcript_id']])\n",
    "coding_TSTR = coding_TSTR[beijer_output_FDR5.columns]\n",
    "beijer_output_FDR5_TSA = pd.concat([coding_TSTR, beijer_output_FDR5])\n",
    "beijer_output_FDR5_TSA = beijer_output_FDR5_TSA[beijer_output_FDR5_TSA['target_decoy'] == \"target\"]\n",
    "\n",
    "## TSA Science Advances\n",
    "beijer_output_FDR5_TSA = beijer_output_FDR5_TSA.merge(annot, on=\"transcript_id\")\n",
    "beijer_output_FDR5_TSA = beijer_output_FDR5_TSA.merge(TSASciAdv, on=\"gene_id\", how=\"left\")\n",
    "\n",
    "beijer_output_FDR5_TSA = beijer_output_FDR5_TSA[['sequence','sample','chr','accessions','gene_id','gene_name_x','gene_name_y','n','transcript_type','gene_type_x','protein_references']].sort_values(by=\"n\", ascending=False)\n",
    "beijer_output_FDR5_TSA.to_csv(os.path.join(dir,\"FDR5percent_TSA_analysis.csv\"), index=None)\n",
    "\n",
    "## TSTR\n",
    "beijer_output_FDR5_TSA_TSTR = beijer_output_FDR5_TSA.merge(TSTR, on=\"gene_id\")\n",
    "beijer_output_FDR5_TSA_TSTR.to_csv(os.path.join(dir,\"FDR5percent_TSA_TSTR_analysis.csv\"), index=None)\n",
    "beijer_output_FDR5_TSA_TSTR"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Merge results"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_3946535/2722277600.py:14: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  whole_output_ordered.drop_duplicates(inplace=True)\n"
     ]
    }
   ],
   "source": [
    "## Merge them all (except PT)\n",
    "dir = \"/users/genomics/marta/241205_Immunopeptidomics_HCC\"\n",
    "data_source = [\"Melanoma_Chong\",\"Melanoma_Bassani\"]\n",
    "\n",
    "whole_output = pd.DataFrame()\n",
    "\n",
    "for source in data_source:\n",
    "    input = pd.read_csv(os.path.join(dir,source,\"FDR5percent/FDR5percent_TSA_TSTR_analysis.csv\"))\n",
    "    input['source'] = source\n",
    "\n",
    "    whole_output = pd.concat([whole_output, input])\n",
    "\n",
    "whole_output_ordered = whole_output[['sequence','accessions','orfID_TSTR_RiboSeq','protein_references','gene_name_x','gene_name_y','gene_type_x','gene_id','source','sample','gene_id','ctype','n','chr']]\n",
    "whole_output_ordered.drop_duplicates(inplace=True)\n",
    "\n",
    "## add immunopep\n",
    "immunopep = pd.read_csv(\"/projects_eg/projects/marta/TestisRestricted_Microproteins_TSA/with_TranscriptomeReconstruction/Multimap_altORFs/Q5_immunopeptidomics/human/TOv3x_5percent_TestisRestrictedGTEx_Translated_Ctypes_log2ratio3x_immuno.csv\")\n",
    "immunopep = immunopep[[\"Peptide\", \"orfID\",\"ORFpep\",\"source\"]]\n",
    "immunopep.columns = [\"Peptide\", \"orfID\",\"ORFpep\",\"source_literature\"]\n",
    "\n",
    "whole_output_ordered_immuno = whole_output_ordered.merge(immunopep, left_on=[\"sequence\"], right_on=\"Peptide\", how=\"left\")\n",
    "whole_output_ordered_immuno.drop_duplicates(inplace=True)\n",
    "whole_output_ordered_immuno.to_csv(os.path.join(dir,\"results/Melanoma_hits_combined.csv\"), index=None)\n",
    "\n",
    "# ## remove redundancies\n",
    "noredundant = whole_output_ordered[[\"sequence\",\"accessions\",\"sample\"]].drop_duplicates().groupby([\"sequence\",\"accessions\"]).count().reset_index()[[\"sequence\",\"accessions\",\"sample\"]]\n",
    "noredundant.columns = [\"sequence\",\"accessions\",\"num_samples_with_hit\"]\n",
    "noredundant['study'] = \"Melanoma_MHCquant_Camarena\"\n",
    "\n",
    "noredundant_full = noredundant.merge(whole_output_ordered.drop([\"sample\",\"source\",\"ctype\"], axis=1), on=[\"sequence\",\"accessions\"])\n",
    "noredundant_full.drop_duplicates(inplace=True)\n",
    "\n",
    "## how many ctypes \n",
    "howmany_ctypes = whole_output_ordered[[\"sequence\",\"accessions\",\"ctype\"]].drop_duplicates().groupby([\"sequence\",\"accessions\"], as_index=False).agg({\n",
    "    'ctype': lambda x: ','.join(sorted(x.unique()))  # Concatenate unique values\n",
    "})\n",
    "howmany_ctypes.columns = [\"sequence\",\"accessions\",\"which_ctypes\"]\n",
    "noredundant_full = noredundant_full.merge(howmany_ctypes, on=[\"sequence\",\"accessions\"], how=\"outer\")\n",
    "noredundant_full\n",
    "\n",
    "## immuno\n",
    "immuno_small = whole_output_ordered_immuno[[\"sequence\",\"accessions\",\"orfID\",\"source_literature\"]].drop_duplicates()\n",
    "immuno_grouped = immuno_small.groupby(['sequence', 'accessions', 'orfID'], as_index=False).agg({\n",
    "    'source_literature': lambda x: ','.join(sorted(x.unique()))  # Concatenate unique values\n",
    "})\n",
    "noredundant_full = noredundant_full.merge(immuno_grouped, on=[\"sequence\",\"accessions\"], how=\"outer\")\n",
    "noredundant_full.drop_duplicates(inplace=True)\n",
    "noredundant_full.to_csv(os.path.join(dir,\"results/Melanoma_hits_combined_noredundant.csv\"), index=None)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.10"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
