---
title: "raw CNV"
author: "Marta E. Camarena"
date: "2025-02-18"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
library(tidyr)
library(TCGAgistic)
library(maftools)
library(rtracklayer)
library(GenomicRanges)
library(dplyr)
library(TCGAbiolinks)
library(ggplot2)
library(ggbeeswarm)
library(rtracklayer)

cancers = c("BRCA","BLCA","LUAD","KIRC","PRAD","LUSC","COAD","LIHC")
TCGAcancers = c("TCGA-BRCA","TCGA-BLCA","TCGA-LUAD","TCGA-KIRC","TCGA-PRAD","TCGA-LUSC","TCGA-COAD","TCGA-LIHC")

save_wd="/projects_eg/projects/marta/TestisRestricted_Microproteins_TSA/with_TranscriptomeReconstruction/Multimap_altORFs/rawCNV/human"

gtf = rtracklayer::import("/users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/human/newReference_Resconstructed/gencode.v47.gffcompare.TestisLiverBrain.annotation.sorted.1transcript.sorted.gtf")
gtf = as.data.frame(gtf)
gtf = gtf %>% subset(type == "transcript") %>% select(seqnames, start, end, transcript_id)
gtf$transcript_id = gsub("\\..*","",gtf$transcript_id)

annot = read.csv("/users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/human/newReference_Resconstructed/transID_geneID_isoforms_selected.1to1.csv")
tumorReactDIR = "/users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/cancers/log2ratio3x/cancertypes/onlyStep1"
# bed = "/users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/cancers/log2ratio3x/cancertypes/onlyStep1/annotations_coding_noncoding/TSTR_ORFs.chr.bed"
```

## Import candidates

```{r candidates, echo=F}
candidates = read.csv(file.path(tumorReactDIR,"TSTR_candidatesORFs_fullcharacterized.csv"))
genes = candidates %>% dplyr::select(gene_name, transcript_id, gene_id, gene_type, coding_noncoding_chr) %>% subset(coding_noncoding_chr != "other ncORFs") %>% unique()

genes = merge(genes, gtf, by="transcript_id", all.x=T)
genes$chr = gsub("chr","",genes$seqnames)
genes = unique(genes)
table(genes$coding_noncoding_chr)
genes = genes %>% select(-seqnames)
```

## Download raw CNV

Segment_Mean is the log2 ratio of copy number for a particular genomic segment.

Log2 Ratio Interpretation:

- 0: Diploid (normal two copies).

- Positive values: Amplifications (e.g., +1 ~ 4 copies, +2 ~ 8 copies).

- Negative values: Deletions (e.g., -1 ~ 1 copy, -2 ~ 0 copies or complete loss).


```{r downloading, echo=F}
# cnvdata_complete = data.frame()
# # for(proj in TCGAcancers) {
# for(proj in c("TCGA-KIRC","TCGA-PRAD","TCGA-LUSC","TCGA-COAD","TCGA-LIHC")) {
# 
#   # Query for CNV segment data (raw copy number data)
#   query <- GDCquery(
#     project = proj,  # Change to your cancer type
#     # project = TCGAcancers,  # Change to your cancer type
#     data.category = "Copy Number Variation",
#     data.type = "Copy Number Segment",
#     workflow.type = "DNAcopy",
#     access = "open",
#     sample.type = "Primary Tumor"
#   )
#   # getResults(query)
#   
#   # Download the data
#   GDCdownload(query,
#               directory = file.path(tumorReactDIR,"rawCNV/data"))
#   # Prepare the data
#   cnv_data <- GDCprepare(query,
#                          directory = file.path(tumorReactDIR,"rawCNV/data"))
#   
#   cnv_data$ctype = proj
#   cnvdata_complete = rbind(cnvdata_complete, cnv_data)
# }
# saveRDS(cnvdata_complete, file.path(tumorReactDIR,"rawCNV/data/cnvdata_8cancertypes.rds"))
cnvdata_complete = readRDS(file.path(tumorReactDIR,"rawCNV/data/cnvdata_8cancertypes.rds"))
```

```{r mapGenes, echo=F}
# Convert CNV data into a GRanges object
cnv_gr <- GRanges(
  seqnames = cnvdata_complete$Chromosome,  
  ranges = IRanges(
    start = cnvdata_complete$Start,  
    end = cnvdata_complete$End       
  ),
  segment_mean = cnvdata_complete$Segment_Mean,
  ctype = cnvdata_complete$ctype
)

# Convert gene data into a GRanges object
genes_gr <- GRanges(
  seqnames = genes$chr, 
  ranges = IRanges(
    start = genes$start,       
    end = genes$end            
  ),
  gene = genes$gene_name     
)

# Find overlaps
overlaps <- findOverlaps(genes_gr, cnv_gr)
# Extract overlapping segments and genes
gene_cnv_data <- data.frame(
  gene_name = genes_gr$gene[queryHits(overlaps)],              # Gene name
  Chromosome = seqnames(genes_gr)[queryHits(overlaps)],   # Gene chromosome
  Gene_Start = start(genes_gr)[queryHits(overlaps)],      # Gene start
  Gene_End = end(genes_gr)[queryHits(overlaps)],          # Gene end
  Segment_Start = start(cnv_gr)[subjectHits(overlaps)],   # CNV segment start
  Segment_End = end(cnv_gr)[subjectHits(overlaps)],       # CNV segment end
  Segment_Mean = cnv_gr$segment_mean[subjectHits(overlaps)], # Segment mean
  ctype = cnv_gr$ctype[subjectHits(overlaps)]
)
gene_cnv_data = merge(gene_cnv_data, genes %>% dplyr::select(gene_name, coding_noncoding_chr), by="gene_name")

## by category
gene_cnv_data <- gene_cnv_data %>%
  mutate(
    Category = case_when(
      Segment_Mean < -2 ~ "Extreme Deletion",
      Segment_Mean >= -2 & Segment_Mean < -1 ~ "Deleted",
      Segment_Mean >= -1 & Segment_Mean < 1 ~ "Slight Change",
      Segment_Mean >= 1 & Segment_Mean < 2 ~ "Amplification",
      Segment_Mean >= 2 ~ "Extreme Amplification",
      TRUE ~ "Other"
    )
  )

# Summarize 
gene_summary <- gene_cnv_data %>%
  group_by(gene_name, coding_noncoding_chr, ctype) %>%  # Add Sample_ID if applicable
  summarize(
    # Count how many times each category appears for this gene
    Extreme_Deletion = sum(Category == "Extreme Deletion"),
    Deleted = sum(Category == "Deleted"),
    Slight_Change = sum(Category == "Slight Change"),
    Amplification = sum(Category == "Amplification"),
    Extreme_Amplification = sum(Category == "Extreme Amplification"),
    
    Avg_Segment_Mean = mean(Segment_Mean),
    Min_Segment_Mean = min(Segment_Mean),  # Smallest value (deepest deletion)
    Max_Segment_Mean = max(Segment_Mean)  # Largest value (highest amplification)
  )
write.csv(gene_summary, file.path(tumorReactDIR,"rawCNV/results_ctypes.csv"), row.names = F)
```

```{r plots, echo=F}
for(proj in TCGAcancers) {
  data_to_plot = gene_summary %>%
    dplyr::select(-c(Avg_Segment_Mean, Min_Segment_Mean, Max_Segment_Mean)) %>%
    pivot_longer(cols=-c(gene_name, coding_noncoding_chr, ctype), values_to = "nums", names_to = "Category") %>% subset(Category != "Slight_Change") %>% 
    subset(ctype == proj) %>% 
    subset(coding_noncoding_chr != "Noncoding-X") %>%
    subset(nums > 0)
    
  print(ggplot(data_to_plot, aes(x=gene_name, y=nums, fill=coding_noncoding_chr)) +
    geom_bar(stat="identity", position="dodge") +
    scale_fill_manual(values=c("CT-X" = "#833437", "CT-nonX" = "#ead890", "Noncoding-nonX" = "#646c31")) + 
    labs(x="Genes",
         y="Number of ocurrences",
         title=paste0("CNV in ",proj)) +
    theme_minimal() +
    theme(legend.position = "top",
          # axis.text.x = element_text(angle=90)) +
          axis.text.x = element_blank()) +
    facet_wrap(~ Category, scales="free"))
  ggsave(paste0(save_wd,"/CNV_",proj,".jpeg"))
  
  segmentMean_to_plot = gene_cnv_data %>%
    subset(ctype == proj) %>% 
    subset(coding_noncoding_chr != "Noncoding-X")
  
  ggplot(segmentMean_to_plot, aes(x=gene_name, y=Segment_Mean, color=coding_noncoding_chr)) +
    geom_quasirandom() +
    geom_hline(yintercept = 1) +
    geom_hline(yintercept = -1) +
    scale_color_manual(values=c("CT-X" = "#833437", "CT-nonX" = "#ead890", "Noncoding-nonX" = "#646c31")) + 
    theme_minimal() +
    theme(axis.text.x=element_blank(),
          legend.position = "none") +
    facet_wrap(~ coding_noncoding_chr, nrow=3, scales="free_x")
    ggsave(paste0(save_wd,"/segmentMean_CNV_",proj,".jpeg"))

  extrems = segmentMean_to_plot %>% subset(Segment_Mean > 1 | Segment_Mean < -1)
  ggplot(extrems, aes(x=gene_name, y=Segment_Mean, color=coding_noncoding_chr)) +
    geom_quasirandom() +
    geom_hline(yintercept = 1) +
    geom_hline(yintercept = -1) +
    scale_color_manual(values=c("CT-X" = "#833437", "CT-nonX" = "#ead890", "Noncoding-nonX" = "#646c31")) + 
    theme_minimal() +
    theme(axis.text.x=element_text(angle=90),
          legend.position = "none") +
    facet_wrap(~ coding_noncoding_chr, nrow=3, scales="free_x")
    ggsave(paste0(save_wd,"/segmentMeanExtrems_CNV_",proj,".jpeg"))

  ### amplifications only
  amplifications = segmentMean_to_plot %>% subset(Segment_Mean > 1)
  ggplot(amplifications, aes(x=gene_name, y=Segment_Mean, color=coding_noncoding_chr)) +
    geom_quasirandom(size=.5) +
    scale_color_manual(values=c("CT-X" = "#833437", "CT-nonX" = "#ead890", "Noncoding-nonX" = "#646c31")) + 
    labs(title=paste0("Amplifications in TSTR | ", proj)) +
    theme_minimal() +
    theme(axis.text.x=element_text(angle=90),
          legend.position = "none") +
    facet_wrap(~ coding_noncoding_chr, nrow=3, scales="free_x")
    ggsave(paste0(save_wd,"/amplifications_CNV_",proj,".jpeg"))

  ## amplified in more than 5 observations
  amplifications_5_counts = segmentMean_to_plot %>% subset(Segment_Mean > 1) %>% dplyr::group_by(coding_noncoding_chr, gene_name, Category) %>% dplyr::count() %>% subset(n>5)
  amplifications_5 = segmentMean_to_plot %>% subset(Segment_Mean > 1) %>% subset(gene_name %in% amplifications_5_counts$gene_name)
  
  if(nrow(amplifications_5) > 0) {
    print(ggplot(amplifications_5, aes(x=gene_name, y=Segment_Mean, color=coding_noncoding_chr, fill=coding_noncoding_chr)) +
      geom_quasirandom() +
      geom_boxplot(color="black", alpha=.3, width=.2) +
      scale_color_manual(values=c("CT-X" = "#833437", "CT-nonX" = "#ead890", "Noncoding-nonX" = "#646c31")) + 
      scale_fill_manual(values=c("CT-X" = "#833437", "CT-nonX" = "#ead890", "Noncoding-nonX" = "#646c31")) + 
      labs(title=paste0("Amplifications in TSTR >5 times | ",proj)) +
      theme_minimal() +
      theme(axis.text.x=element_text(angle=90),
            legend.position = "none") +
      facet_wrap(~ coding_noncoding_chr, nrow=3, scales="free_x"))
      ggsave(paste0(save_wd,"/amplifications5_CNV_",proj,".jpeg"))
  }
}
```

```{r amplifications, echo=F}
segmentMean_to_plot = gene_cnv_data %>%
    subset(coding_noncoding_chr != "Noncoding-X")

amplifications_5_counts = segmentMean_to_plot %>% 
  subset(Segment_Mean > 1) %>% 
  dplyr::group_by(coding_noncoding_chr, gene_name, Category) %>% 
  dplyr::count() %>% 
  subset(n>5)
amplifications_5 = segmentMean_to_plot %>% 
  subset(Segment_Mean > 1) %>% 
  subset(gene_name %in% amplifications_5_counts$gene_name)

## Noncoding-nonX
ggplot(amplifications_5 %>% subset(coding_noncoding_chr == "Noncoding-nonX"), aes(x=gene_name, y=Segment_Mean, color=ctype)) +
  geom_jitter(size=1) +
  geom_boxplot(color="black", alpha=.3, width=.5, fill=NA) +
  labs(title="Amplifications in TSTR >5 times | Noncoding-nonX") +
  theme_minimal() +
  theme(axis.text.x=element_text(angle=90),
        legend.position = "none") 
ggsave(file.path(save_wd,"amplifications5_CNV_allctypes_noncodingnonX.jpeg"), width=15.8, height=8.47)

## CT-nonX
ggplot(amplifications_5 %>% subset(coding_noncoding_chr == "CT-nonX"), aes(x=gene_name, y=Segment_Mean, color=ctype)) +
  geom_jitter(size=1) +
  geom_boxplot(color="black", alpha=.3, width=.5, fill=NA) +
  labs(title="Amplifications in TSTR >5 times | CT-nonX") +
  theme_minimal() +
  theme(axis.text.x=element_text(angle=90),
        legend.position = "none") 
ggsave(file.path(save_wd,"amplifications5_CNV_allctypes_CTnonX.jpeg"), width=15.8, height=8.47)
```

```{r controlnegatiu, echo=F}
## candidate genes and their length
candidates_length = candidates %>% 
  select(gene_id, transcript_id, gene_name, gene_type, coding_noncoding_chr, length) %>% 
  unique() %>% subset(coding_noncoding_chr != "other ncORFs")

## all genes from annotation and their length
genes_length = annot %>% select(gene_id, gene_name, gene_type, transcript_id) %>% unique()
counts = read.csv("/users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/human/featureCounts_gffcompare/table_of_counts_TPMs_withLength.csv")
counts = counts %>% select(gene_id, Length)
names(counts) = c("transcript_id","length")

genes_length = merge(genes_length, counts, by="transcript_id")

## observation
ref = genes_length %>% select(gene_id, gene_name, length, gene_type)
ref$source = "all"
cand = candidates_length  %>% select(gene_id, gene_name, length, gene_type)
cand$source = "candidates"
to_plot = rbind(ref, cand)

ggplot(to_plot, aes(x=length, color=source)) +
  geom_density() +
  scale_x_continuous(trans="log10")

## per gene_type
ggplot(to_plot, aes(x=length, color=source)) +
  geom_density() +
  scale_x_continuous(trans="log10") +
  facet_wrap(~ gene_type, scales="free_y")

############################# Length distribution ############################## 
set.seed(123)
# Step 1: Define Bins Based on Reference Distribution
# Divide the reference length in 19 bins (20 breakpoints)
ref_breaks <- quantile(candidates_length$length, probs = seq(0, 1, length.out = 20), na.rm = TRUE)  # Handle NAs
ref_breaks[length(ref_breaks)] <- ref_breaks[length(ref_breaks)] + 1  # Ensure all values fit in bins
# Assign bins
# categorize each gene into one of the bins based on length both for the reference and the pool 
ref_bins <- cut(candidates_length$length, breaks = ref_breaks, include.lowest = TRUE)
pool_bins <- cut(genes_length$length, breaks = ref_breaks, include.lowest = TRUE)

# Step 2: Determine Sample Sizes Per Bin
# number of genes per bin
ref_bin_counts <- table(ref_bins)

# Step 3: Sample from the Pool Proportionally
# take a random sample from genes_length, ensuring that we match the reference distribution
sampled_genes <- genes_length %>%
  mutate(bin = cut(length, breaks = ref_breaks, include.lowest = TRUE)) %>%
  filter(!is.na(bin)) %>%  # Remove NAs
  group_by(bin) %>%
  sample_n(size = pmin(ref_bin_counts[as.character(bin)], n()), replace = FALSE) %>%
  ungroup()
table(sampled_genes$gene_type)

# Step 4: Validate the Distribution
ggplot() +
  geom_density(data = candidates_length, aes(x = length, y = ..density..), color = "blue", alpha = 0.5) +
  geom_density(data = sampled_genes, aes(x = length, y = ..density..), color = "red", alpha = 0.5) +
  labs(title = "Comparison of Gene Length Distributions", x = "Gene Length", y = "Density") +
  theme_minimal()
```

```{r cnv_control, echo=F}
sampled_genes = merge(sampled_genes, gtf, by="transcript_id", all.x=T)
sampled_genes$chr = gsub("chr","",sampled_genes$seqnames)
sampled_genes = unique(sampled_genes)
sampled_genes = sampled_genes %>% select(-seqnames)
sampled_genes = sampled_genes %>% mutate(coding_noncoding_chr = case_when(gene_type == "protein_coding" & chr == "X" ~ "CT-X",
                                                           gene_type == "protein_coding" & chr != "X" ~ "CT-nonX",
                                                           gene_type != "protein_coding" & chr == "X" ~ "Noncoding-X",
                                                           gene_type != "protein_coding" & chr != "X" ~ "Noncoding-nonX"))
# Convert CNV data into a GRanges object
cnv_gr_ctrl <- GRanges(
  seqnames = cnvdata_complete$Chromosome,  
  ranges = IRanges(
    start = cnvdata_complete$Start,  
    end = cnvdata_complete$End       
  ),
  segment_mean = cnvdata_complete$Segment_Mean,
  ctype = cnvdata_complete$ctype
)

# Convert gene data into a GRanges object
genes_gr_ctrl <- GRanges(
  seqnames = sampled_genes$chr, 
  ranges = IRanges(
    start = sampled_genes$start,       
    end = sampled_genes$end            
  ),
  gene = sampled_genes$gene_name     
)

# Find overlaps
overlaps_ctrl <- findOverlaps(genes_gr_ctrl, cnv_gr_ctrl)
# Extract overlapping segments and genes
gene_cnv_data_ctrl <- data.frame(
  gene_name = genes_gr_ctrl$gene[queryHits(overlaps)],              # Gene name
  Chromosome = seqnames(genes_gr_ctrl)[queryHits(overlaps)],   # Gene chromosome
  Gene_Start = start(genes_gr_ctrl)[queryHits(overlaps)],      # Gene start
  Gene_End = end(genes_gr_ctrl)[queryHits(overlaps)],          # Gene end
  Segment_Start = start(cnv_gr_ctrl)[subjectHits(overlaps)],   # CNV segment start
  Segment_End = end(cnv_gr_ctrl)[subjectHits(overlaps)],       # CNV segment end
  Segment_Mean = cnv_gr_ctrl$segment_mean[subjectHits(overlaps)], # Segment mean
  ctype = cnv_gr_ctrl$ctype[subjectHits(overlaps)]
)
gene_cnv_data_ctrl = merge(gene_cnv_data_ctrl, sampled_genes %>% dplyr::select(gene_name, coding_noncoding_chr), by="gene_name")

## by category
gene_cnv_data_ctrl <- gene_cnv_data_ctrl %>%
  mutate(
    Category = case_when(
      Segment_Mean < -2 ~ "Extreme Deletion",
      Segment_Mean >= -2 & Segment_Mean < -1 ~ "Deleted",
      Segment_Mean >= -1 & Segment_Mean < 1 ~ "Slight Change",
      Segment_Mean >= 1 & Segment_Mean < 2 ~ "Amplification",
      Segment_Mean >= 2 ~ "Extreme Amplification",
      TRUE ~ "Other"
    )
  )

# Summarize 
gene_summary_ctrl <- gene_cnv_data_ctrl %>%
  group_by(gene_name, ctype, coding_noncoding_chr) %>%  # Add Sample_ID if applicable
  summarize(
    # Count how many times each category appears for this gene
    Extreme_Deletion = sum(Category == "Extreme Deletion"),
    Deleted = sum(Category == "Deleted"),
    Slight_Change = sum(Category == "Slight Change"),
    Amplification = sum(Category == "Amplification"),
    Extreme_Amplification = sum(Category == "Extreme Amplification"),
    
    Avg_Segment_Mean = mean(Segment_Mean),
    Min_Segment_Mean = min(Segment_Mean),  # Smallest value (deepest deletion)
    Max_Segment_Mean = max(Segment_Mean)  # Largest value (highest amplification)
  )
write.csv(gene_summary_ctrl, file.path(tumorReactDIR,"rawCNV/ctrlResults_ctypes.csv"), row.names = F)
```

```{r plots_ctrl, echo=F}
for(proj in TCGAcancers) {
  data_to_plot = gene_summary_ctrl %>%
    dplyr::select(-c(Avg_Segment_Mean, Min_Segment_Mean, Max_Segment_Mean)) %>%
    pivot_longer(cols=-c(gene_name, coding_noncoding_chr, ctype), values_to = "nums", names_to = "Category") %>% subset(Category != "Slight_Change") %>% 
    subset(ctype == proj) %>% 
    subset(nums > 0)
    
  print(ggplot(data_to_plot, aes(x=gene_name, y=nums, fill=coding_noncoding_chr)) +
    geom_bar(stat="identity", position="dodge") +
    scale_fill_manual(values=c("CT-X" = "#833437", "CT-nonX" = "#ead890", "Noncoding-nonX" = "#646c31", "Noncoding-X"="#06402B")) + 
    labs(x="Genes",
         y="Number of ocurrences",
         title=paste0("CONTROL CNV in ",proj)) +
    theme_minimal() +
    theme(legend.position = "top",
          # axis.text.x = element_text(angle=90)) +
          axis.text.x = element_blank()) +
    facet_wrap(~ Category, scales="free"), height=8, width=8)
  ggsave(paste0(save_wd,"/CTRL_CNV_",proj,".jpeg"))
  
  segmentMean_to_plot = gene_cnv_data_ctrl %>%
    subset(ctype == proj)
  
  ggplot(segmentMean_to_plot, aes(x=gene_name, y=Segment_Mean, color=coding_noncoding_chr)) +
    geom_quasirandom() +
    geom_hline(yintercept = 1) +
    geom_hline(yintercept = -1) +
    scale_color_manual(values=c("CT-X" = "#833437", "CT-nonX" = "#ead890", "Noncoding-nonX" = "#646c31", "Noncoding-X"="#06402B")) + 
    theme_minimal() +
    theme(axis.text.x=element_blank(),
          legend.position = "none") +
    facet_wrap(~ coding_noncoding_chr, nrow=3, scales="free_x")
    ggsave(paste0(save_wd,"/segmentMean_CNV_",proj,".jpeg"), height = 9.52, width=7.10)

  extrems = segmentMean_to_plot %>% subset(Segment_Mean > 1 | Segment_Mean < -1)
  ggplot(extrems, aes(x=gene_name, y=Segment_Mean, color=coding_noncoding_chr)) +
    geom_quasirandom() +
    geom_hline(yintercept = 1) +
    geom_hline(yintercept = -1) +
    ggtitle("CONTROL") +
    scale_color_manual(values=c("CT-X" = "#833437", "CT-nonX" = "#ead890", "Noncoding-nonX" = "#646c31", "Noncoding-X"="#06402B")) + 
    theme_minimal() +
    theme(axis.text.x=element_blank(),,
          legend.position = "none") +
    facet_wrap(~ coding_noncoding_chr, nrow=3, scales="free_x")
    ggsave(paste0(save_wd,"/CTRL_segmentMeanExtrems_CNV_",proj,".jpeg"), height = 9.52, width=7.10)

  ### amplifications only
  amplifications = segmentMean_to_plot %>% subset(Segment_Mean > 1)
  ggplot(amplifications, aes(x=gene_name, y=Segment_Mean, color=coding_noncoding_chr)) +
    geom_quasirandom(size=.5) +
    scale_color_manual(values=c("CT-X" = "#833437", "CT-nonX" = "#ead890", "Noncoding-nonX" = "#646c31", "Noncoding-X"="#06402B")) + 
    labs(title=paste0("CTRL Amplifications in TSTR | ", proj)) +
    theme_minimal() +
    theme(axis.text.x=element_text(angle=90),
          legend.position = "none") +
    facet_wrap(~ coding_noncoding_chr, nrow=3, scales="free_x")
    ggsave(paste0(save_wd,"/CTRL_amplifications_CNV_",proj,".jpeg"), height = 9.52, width=12.97)

  ## amplified in more than 5 observations
  amplifications_5_counts = segmentMean_to_plot %>% subset(Segment_Mean > 1) %>% dplyr::group_by(coding_noncoding_chr, gene_name, Category) %>% dplyr::count() %>% subset(n>5)
  amplifications_5 = segmentMean_to_plot %>% subset(Segment_Mean > 1) %>% subset(gene_name %in% amplifications_5_counts$gene_name)
  
  if(nrow(amplifications_5) > 0) {
    print(ggplot(amplifications_5, aes(x=gene_name, y=Segment_Mean, color=coding_noncoding_chr, fill=coding_noncoding_chr)) +
      geom_quasirandom() +
      geom_boxplot(color="black", alpha=.3, width=.2) +
      scale_color_manual(values=c("CT-X" = "#833437", "CT-nonX" = "#ead890", "Noncoding-nonX" = "#646c31", "Noncoding-X"="#06402B")) + 
      scale_fill_manual(values=c("CT-X" = "#833437", "CT-nonX" = "#ead890", "Noncoding-nonX" = "#646c31", "Noncoding-X"="#06402B")) + 
      labs(title=paste0("CONTROL Amplifications in TSTR >5 times | ",proj)) +
      theme_minimal() +
      theme(axis.text.x=element_text(angle=90),
            legend.position = "none") +
      facet_wrap(~ coding_noncoding_chr, nrow=3, scales="free_x"))
      ggsave(paste0(save_wd,"/CTRL_amplifications5_CNV_",proj,".jpeg"), height = 9.52, width=8.97)
  }
}
```