{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import os\n",
    "import pandas as pd\n",
    "\n",
    "riboDir = \"/users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/human/RiboNovel_MultMap_1to1\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash -s \"$riboDir\"\n",
    "\n",
    "module load ucsc_tools\n",
    "module load Python/3.8.6-GCCcore-10.2.0\n",
    "module load BEDTools\n",
    "\n",
    "for file in $1/RibORF/*/repre.valid.ORF.genepred.txt; do\n",
    "\n",
    "    echo $file\n",
    "\n",
    "    out=${file%%.txt}\n",
    "\n",
    "    # Transform the genePred into GTF\n",
    "    genePredToGtf file source=\"ORF\" $file ${out}.tmp.gtf\n",
    "    # Extract only the CDS lines of the GTF\n",
    "    grep -P \"\\tCDS\\t\" ${out}.tmp.gtf > ${out}.tmp.CDS.gtf\n",
    "\n",
    "    # Regenerate a GenePred file only of the CDS but this time the ORFs are correctly mapped\n",
    "    gtfToGenePred ${out}.tmp.CDS.gtf ${out}.tmp.genePred\n",
    "    genePredToBed ${out}.tmp.genePred ${out}.tmp.bed\n",
    "\n",
    "    mv ${out}.tmp.bed ${out}.bed\n",
    "\n",
    "    # then we have to slop by 3 \n",
    "    bedtools slop -i ${out}.bed \\\n",
    "    -g /projects_eg/projects/marta/chromosomes_order.NOchr.txt \\\n",
    "    -s -l 0 -r 3 > ${out}.slopped.bed\n",
    "\n",
    "    # And then we adjust with my python script\n",
    "    python /projects_eg/projects/chris/Pipelines/ORF_conservation/adjust_bed12_after_slop.py \\\n",
    "    -i ${out}.slopped.bed \\\n",
    "    -o ${out}.slopped.adjusted.bed\n",
    "\n",
    "    rm ${out}.tmp.CDS.gtf ${out}.tmp.gtf\n",
    "\n",
    "    # Re-name the files\n",
    "    #mv candidateORF.genepred.txt candidateORF.CDS.genePred\n",
    "    mv ${out}.txt ${out}.CDS.genepred.txt\n",
    "    mv ${out}.tmp.genePred ${out}.CDS.genePred\n",
    "\n",
    "    # mv candidateORF.fa candidateORF.CDS.fa\n",
    "done"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash -s \"$riboDir\"\n",
    "\n",
    "## one bed file per tissue\n",
    "\n",
    "tissues=\"testis liver brain\"\n",
    "\n",
    "for t in $tissues; do\n",
    "\n",
    "    echo $t\n",
    "\n",
    "    cat $1/RibORF/human*${t}*/repre.valid.ORF.genepred.slopped.adjusted.bed > $1/RibORF/${t}.bed\n",
    "    sort $1/RibORF/${t}.bed | uniq > $1/RibORF/valid.ORF.${t}.bed\n",
    "\n",
    "done"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Brain Only in R"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene_id</th>\n",
       "      <th>gene_name</th>\n",
       "      <th>orfID</th>\n",
       "      <th>transcript_id</th>\n",
       "      <th>gene_type</th>\n",
       "      <th>orfType</th>\n",
       "      <th>length</th>\n",
       "      <th>geneORFtype</th>\n",
       "      <th>length_aa</th>\n",
       "      <th>start_codon</th>\n",
       "      <th>ORFpep</th>\n",
       "      <th>TranslatedLiver</th>\n",
       "      <th>TranslatedTestis</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ENSG00000022355</td>\n",
       "      <td>GABRA1</td>\n",
       "      <td>ENST00000428797.7:5:+|11|4285:227:341|uORF|CTG</td>\n",
       "      <td>ENST00000428797</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>uORF</td>\n",
       "      <td>114</td>\n",
       "      <td>protein_coding_uORF</td>\n",
       "      <td>38</td>\n",
       "      <td>CTG</td>\n",
       "      <td>MARPDSDSQTRAGSSLSTILSPRLFPGLKRSCVQRGP*</td>\n",
       "      <td>no</td>\n",
       "      <td>no</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ENSG00000022355</td>\n",
       "      <td>GABRA1</td>\n",
       "      <td>ENST00000428797.7:5:+|1|4285:42:96|uORF|CTG</td>\n",
       "      <td>ENST00000428797</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>uORF</td>\n",
       "      <td>54</td>\n",
       "      <td>protein_coding_uORF</td>\n",
       "      <td>18</td>\n",
       "      <td>CTG</td>\n",
       "      <td>MHLPVSELAFIHMQAVV*</td>\n",
       "      <td>no</td>\n",
       "      <td>no</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ENSG00000022355</td>\n",
       "      <td>GABRA1</td>\n",
       "      <td>ENST00000428797.7:5:+|4|4285:109:196|uORF|CTG</td>\n",
       "      <td>ENST00000428797</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>uORF</td>\n",
       "      <td>87</td>\n",
       "      <td>protein_coding_uORF</td>\n",
       "      <td>29</td>\n",
       "      <td>CTG</td>\n",
       "      <td>MQIGYWEANLGVKSSAKEHAESMMAQTK*</td>\n",
       "      <td>no</td>\n",
       "      <td>no</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ENSG00000101958</td>\n",
       "      <td>GLRA2</td>\n",
       "      <td>ENST00000218075.9:X:+|7|3208:338:452|uORF|ATG</td>\n",
       "      <td>ENST00000218075</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>uORF</td>\n",
       "      <td>114</td>\n",
       "      <td>protein_coding_uORF</td>\n",
       "      <td>38</td>\n",
       "      <td>ATG</td>\n",
       "      <td>MMPRTGTFSFFSANCTKPNLFLIFKETRFLPNFESGQ*</td>\n",
       "      <td>no</td>\n",
       "      <td>no</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ENSG00000101958</td>\n",
       "      <td>GLRA2</td>\n",
       "      <td>ENST00000218075.9:X:+|15|3208:523:1882|canonic...</td>\n",
       "      <td>ENST00000218075</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>canonical</td>\n",
       "      <td>1359</td>\n",
       "      <td>protein_coding_canonical</td>\n",
       "      <td>453</td>\n",
       "      <td>ATG</td>\n",
       "      <td>MNRQLVNILTALFAFFLETNHFRTAFCKDHDSRSGKQPSQTLSPSD...</td>\n",
       "      <td>no</td>\n",
       "      <td>no</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>116</th>\n",
       "      <td>ENSG00000288563</td>\n",
       "      <td>ENSG00000288563</td>\n",
       "      <td>ENST00000673949.1:4:-|3|4225:62:107|noncoding|ATG</td>\n",
       "      <td>ENST00000673949</td>\n",
       "      <td>lncRNA</td>\n",
       "      <td>noncoding</td>\n",
       "      <td>45</td>\n",
       "      <td>lncRNA_noncoding</td>\n",
       "      <td>15</td>\n",
       "      <td>ATG</td>\n",
       "      <td>MKRIQLVGISREDA*</td>\n",
       "      <td>no</td>\n",
       "      <td>no</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>117</th>\n",
       "      <td>ENSG00000288563</td>\n",
       "      <td>ENSG00000288563</td>\n",
       "      <td>ENST00000673949.1:4:-|7|4225:99:192|noncoding|ATG</td>\n",
       "      <td>ENST00000673949</td>\n",
       "      <td>lncRNA</td>\n",
       "      <td>noncoding</td>\n",
       "      <td>93</td>\n",
       "      <td>lncRNA_noncoding</td>\n",
       "      <td>31</td>\n",
       "      <td>ATG</td>\n",
       "      <td>MHDSFMRRLMNKGLKMMMKSTECSNPACQC*</td>\n",
       "      <td>no</td>\n",
       "      <td>no</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>118</th>\n",
       "      <td>ENSG00000289441</td>\n",
       "      <td>ENSG00000289441</td>\n",
       "      <td>ENST00000728103.1:10:+|31|1182:529:700|noncodi...</td>\n",
       "      <td>ENST00000728103</td>\n",
       "      <td>lncRNA</td>\n",
       "      <td>noncoding</td>\n",
       "      <td>171</td>\n",
       "      <td>lncRNA_noncoding</td>\n",
       "      <td>57</td>\n",
       "      <td>ATG</td>\n",
       "      <td>MPTEPPHRKTGTRMALCKDSHQECQSQRLRDQRTQKGDLKNSCLGA...</td>\n",
       "      <td>no</td>\n",
       "      <td>no</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>119</th>\n",
       "      <td>ENSG00000289462</td>\n",
       "      <td>ENSG00000289462</td>\n",
       "      <td>ENST00000689020.2:5:-|18|2391:364:619|noncodin...</td>\n",
       "      <td>ENST00000689020</td>\n",
       "      <td>lncRNA</td>\n",
       "      <td>noncoding</td>\n",
       "      <td>255</td>\n",
       "      <td>lncRNA_noncoding</td>\n",
       "      <td>85</td>\n",
       "      <td>ATG</td>\n",
       "      <td>MPLQGLREGSARRLVGNAFLPLNTRLISLKHTMYLSCPKDDDRWFF...</td>\n",
       "      <td>no</td>\n",
       "      <td>no</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>120</th>\n",
       "      <td>XLOC_000131</td>\n",
       "      <td>XLOC_000131</td>\n",
       "      <td>TCONS_00000143:10:+|63|9125:2167:2290|noncodin...</td>\n",
       "      <td>TCONS_00000143</td>\n",
       "      <td>novel</td>\n",
       "      <td>noncoding</td>\n",
       "      <td>123</td>\n",
       "      <td>novel_noncoding</td>\n",
       "      <td>41</td>\n",
       "      <td>ATG</td>\n",
       "      <td>MLSEISRIILIVDNKDAPGIWVEDVVKNNKEDKNYTYQGK*</td>\n",
       "      <td>no</td>\n",
       "      <td>no</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>121 rows × 13 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "             gene_id        gene_name  \\\n",
       "0    ENSG00000022355           GABRA1   \n",
       "1    ENSG00000022355           GABRA1   \n",
       "2    ENSG00000022355           GABRA1   \n",
       "3    ENSG00000101958            GLRA2   \n",
       "4    ENSG00000101958            GLRA2   \n",
       "..               ...              ...   \n",
       "116  ENSG00000288563  ENSG00000288563   \n",
       "117  ENSG00000288563  ENSG00000288563   \n",
       "118  ENSG00000289441  ENSG00000289441   \n",
       "119  ENSG00000289462  ENSG00000289462   \n",
       "120      XLOC_000131      XLOC_000131   \n",
       "\n",
       "                                                 orfID    transcript_id  \\\n",
       "0       ENST00000428797.7:5:+|11|4285:227:341|uORF|CTG  ENST00000428797   \n",
       "1          ENST00000428797.7:5:+|1|4285:42:96|uORF|CTG  ENST00000428797   \n",
       "2        ENST00000428797.7:5:+|4|4285:109:196|uORF|CTG  ENST00000428797   \n",
       "3        ENST00000218075.9:X:+|7|3208:338:452|uORF|ATG  ENST00000218075   \n",
       "4    ENST00000218075.9:X:+|15|3208:523:1882|canonic...  ENST00000218075   \n",
       "..                                                 ...              ...   \n",
       "116  ENST00000673949.1:4:-|3|4225:62:107|noncoding|ATG  ENST00000673949   \n",
       "117  ENST00000673949.1:4:-|7|4225:99:192|noncoding|ATG  ENST00000673949   \n",
       "118  ENST00000728103.1:10:+|31|1182:529:700|noncodi...  ENST00000728103   \n",
       "119  ENST00000689020.2:5:-|18|2391:364:619|noncodin...  ENST00000689020   \n",
       "120  TCONS_00000143:10:+|63|9125:2167:2290|noncodin...   TCONS_00000143   \n",
       "\n",
       "          gene_type    orfType  length               geneORFtype  length_aa  \\\n",
       "0    protein_coding       uORF     114       protein_coding_uORF         38   \n",
       "1    protein_coding       uORF      54       protein_coding_uORF         18   \n",
       "2    protein_coding       uORF      87       protein_coding_uORF         29   \n",
       "3    protein_coding       uORF     114       protein_coding_uORF         38   \n",
       "4    protein_coding  canonical    1359  protein_coding_canonical        453   \n",
       "..              ...        ...     ...                       ...        ...   \n",
       "116          lncRNA  noncoding      45          lncRNA_noncoding         15   \n",
       "117          lncRNA  noncoding      93          lncRNA_noncoding         31   \n",
       "118          lncRNA  noncoding     171          lncRNA_noncoding         57   \n",
       "119          lncRNA  noncoding     255          lncRNA_noncoding         85   \n",
       "120           novel  noncoding     123           novel_noncoding         41   \n",
       "\n",
       "    start_codon                                             ORFpep  \\\n",
       "0           CTG             MARPDSDSQTRAGSSLSTILSPRLFPGLKRSCVQRGP*   \n",
       "1           CTG                                 MHLPVSELAFIHMQAVV*   \n",
       "2           CTG                      MQIGYWEANLGVKSSAKEHAESMMAQTK*   \n",
       "3           ATG             MMPRTGTFSFFSANCTKPNLFLIFKETRFLPNFESGQ*   \n",
       "4           ATG  MNRQLVNILTALFAFFLETNHFRTAFCKDHDSRSGKQPSQTLSPSD...   \n",
       "..          ...                                                ...   \n",
       "116         ATG                                    MKRIQLVGISREDA*   \n",
       "117         ATG                    MHDSFMRRLMNKGLKMMMKSTECSNPACQC*   \n",
       "118         ATG  MPTEPPHRKTGTRMALCKDSHQECQSQRLRDQRTQKGDLKNSCLGA...   \n",
       "119         ATG  MPLQGLREGSARRLVGNAFLPLNTRLISLKHTMYLSCPKDDDRWFF...   \n",
       "120         ATG          MLSEISRIILIVDNKDAPGIWVEDVVKNNKEDKNYTYQGK*   \n",
       "\n",
       "    TranslatedLiver TranslatedTestis  \n",
       "0                no               no  \n",
       "1                no               no  \n",
       "2                no               no  \n",
       "3                no               no  \n",
       "4                no               no  \n",
       "..              ...              ...  \n",
       "116              no               no  \n",
       "117              no               no  \n",
       "118              no               no  \n",
       "119              no               no  \n",
       "120              no               no  \n",
       "\n",
       "[121 rows x 13 columns]"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "brainOnly = pd.read_csv(\"/projects_eg/projects/marta/TestisRestricted_Microproteins_TSA/with_TranscriptomeReconstruction/Multimap_altORFs/Q2_TestisRestricted/human/brain/brainRestricted_GTEx_translatedONLYbrain.csv\")\n",
    "brainOnly"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "bedBrain = pd.read_csv(\"/users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/human/RiboNovel_MultMap_1to1/RibORF/valid.ORF.brain.bed\", sep=\"\\t\", header=None)\n",
    "\n",
    "bedBrainOnly = bedBrain[bedBrain[3].isin(brainOnly.orfID.values.tolist())]\n",
    "bedBrainOnly.to_csv(\"/projects_eg/projects/marta/250225_BrainMPs_Gabriel/brainOnly_GTEx_Wang.bed\", sep=\"\\t\", index=None, header=None)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Brain & Testis"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [],
   "source": [
    "bedBrain = pd.read_csv(\"/users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/human/RiboNovel_MultMap_1to1/RibORF/valid.ORF.brain.bed\", sep=\"\\t\", header=None)\n",
    "bedTestis = pd.read_csv(\"/users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/human/RiboNovel_MultMap_1to1/RibORF/valid.ORF.testis.bed\", sep=\"\\t\", header=None)\n",
    "bedBrainTestis = pd.concat([bedBrain, bedTestis])\n",
    "\n",
    "bothBrainTestis = pd.read_csv(\"/projects_eg/projects/marta/TestisRestricted_Microproteins_TSA/with_TranscriptomeReconstruction/Multimap_altORFs/Q2_TestisRestricted/human/brain/brainTestis_Restricted_GTEx_translated.csv\")\n",
    "bedBrainTestis = bedBrainTestis[bedBrainTestis[3].isin(bothBrainTestis.orfID.values.tolist())]\n",
    "bedBrainTestis.to_csv(\"/projects_eg/projects/marta/250225_BrainMPs_Gabriel/BrainTestis_GTEx_Wang.bed\", sep=\"\\t\", index=None, header=None)"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.10"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
