{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import os,re,glob\n",
    "import pandas as pd\n",
    "import numpy as np\n",
    "from collections import Counter\n",
    "\n",
    "\n",
    "users_dir = \"/users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47\"\n",
    "prev_users_dir = \"/users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction\"\n",
    "specie = \"human\"\n",
    "GENOMEDIR = \"/data/genomics/marta/genomes\"\n"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Bams are processed by Sara --> look for the scripts\n",
    "\n",
    "Outputs are in:\n",
    "\n",
    "`/users/genomics/saraa/projectTestis/STAR/{human/macaca/mouse}`"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Bam2Fastq + QC + STAR shiva\n",
    "\n",
    "/users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/macaca/STAR for macaca\n",
    "\n",
    "/datasets/marta/Wang2020/RNASeq/mouse* for mouse\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [
    {
     "ename": "SyntaxError",
     "evalue": "invalid syntax (2449622322.py, line 14)",
     "output_type": "error",
     "traceback": [
      "\u001b[0;36m  Cell \u001b[0;32mIn[20], line 14\u001b[0;36m\u001b[0m\n\u001b[0;31m    declare -a bam_array\u001b[0m\n\u001b[0m               ^\u001b[0m\n\u001b[0;31mSyntaxError\u001b[0m\u001b[0;31m:\u001b[0m invalid syntax\n"
     ]
    }
   ],
   "source": [
    "#!/bin/bash\n",
    "\n",
    "#SBATCH -p long,bigmem            # Partition to submit to\n",
    "#SBATCH --cpus-per-task=8\n",
    "#SBATCH --mem-per-cpu 10Gb           # Memory in MB\n",
    "#SBATCH -J stringtie           # job name\n",
    "#SBATCH -o /users/genomics/marta/logs/stringtie_%A_%a.out    # File to which standard out will be written\n",
    "#SBATCH -e /users/genomics/marta/logs/stringtie_%A_%a.err    # File to which standard err will be written\n",
    "\n",
    "####Change output and output error paths to redirect the log files generated\n",
    "\n",
    "#export PATH=/genomics/users/irubia/tools/stringtie2/:$PATH\n",
    "specie=\"human\"\n",
    "declare -a bam_array\n",
    "\n",
    "\n",
    "BAMDIR=/users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/$specie/STAR\n",
    "for file in $BAMDIR/*bam; do\n",
    "    bam_array=(${bam_array[@]} $file) # Add new element at the end of the array\n",
    "done\n",
    "\n",
    "\n",
    "\n",
    "i=$(($SLURM_ARRAY_TASK_ID -1))\n",
    "specie=\"human\"\n",
    "OUTDIR=/users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/$specie/stringtieReconstruction\n",
    "mkdir -p $OUTDIR\n",
    "bam=${bam_array[i]}\n",
    "echo $bam\n",
    "\n",
    "if [ \"$specie\" == \"macaca\" ]; then\n",
    "  REF=/datasets/Common/ReferenceGenomes/Macaca/10.111/GenomesAndAnnotation/Macaca_mulatta.Mmul_10.111.gtf\n",
    "elif [ \"$specie\" == \"mouse\" ]; then\n",
    "  REF=/datasets/Common/ReferenceGenomes/Mouse/mm39/GenomesAndAnnotation/gencode.vM32.annotation.gtf\n",
    "elif [ \"$specie\" == \"human\" ]; then\n",
    "  # REF=/datasets/Common/ReferenceGenomes/Human/GRCh38/GenomesAndAnnotation/gencode.v38.primary_assembly.annotation.gtf\n",
    "  REF=/data/genomics/marta/genomes/Annot_files_GTF/GENCODE_47_M36/gencode.v47.primary_assembly.annotation.fixed.gtf\n",
    "elif [ \"$specie\" == \"platypus\" ]; then\n",
    "  REF=/datasets/Common/ReferenceGenomes/Platypus/mOrnaAna1/GenomesAndAnnotation/Ornithorhynchus_anatinus.mOrnAna1.p.v1.112.chr.gtf\n",
    "elif [ \"$specie\" == \"opossum\" ]; then\n",
    "  REF=/datasets/Common/ReferenceGenomes/Opossum/ASSM229v1/GenomesAndAnnotation/Monodelphis_domestica.ASM229v1.112.chr.gtf\n",
    "elif [ \"$specie\" == \"chicken\" ]; then\n",
    "  REF=/datasets/Common/ReferenceGenomes/Chicken/GRCg7b/GenomesAndAnnotation/Gallus_gallus.bGalGal1.mat.broiler.GRCg7b.112.chr.gtf\n",
    "fi\n",
    "\n",
    "sample=${bam%%Aligned*}\n",
    "sample=${sample##*/}\n",
    "\n",
    "module load stringtie/2.2.0\n",
    "## in this case, we assume it's always fr\n",
    "stringtie $bam -G $REF --conservative --fr -o $OUTDIR/${sample}.gtf -C $OUTDIR/${sample}_cov.gtf\n"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Make summary"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/human/stringtieReconstruction/transcriptomeReconstruction.csv\n"
     ]
    }
   ],
   "source": [
    "%%bash -s \"$users_dir\" \"$specie\"\n",
    "## Total transcriptome + Presumably annotated + Presumably novel\n",
    "OUT=$1/$2/stringtieReconstruction/transcriptomeReconstruction.csv\n",
    "echo $OUT\n",
    "if [ -f \"$OUT\" ] ; then\n",
    "    rm \"$OUT\"\n",
    "fi\n",
    "echo -e \"Sample\\tTotal_transcripts\\tAnnotated_transcripts\\tNovel_transcript\" >> $OUT\n",
    "\n",
    "for file in $1/$2/stringtieReconstruction/*gtf; do\n",
    "    if [[ ${file} != *\"cov\"* ]]; then \n",
    "        name=\"${file%%.*}\"\n",
    "        name=\"${name##*/}\"\n",
    "        echo -e $name\"\\t\"$(awk '($3 == \"transcript\") {print}' $file | wc -l)\"\\t\"$(awk '($3 == \"transcript\") {print}' $file | grep \"reference\" | wc -l)\"\\t\"$(awk '($3 == \"transcript\") {print}' $file | grep -v \"reference\" | wc -l) >> $OUT\n",
    "    fi\n",
    "done\n"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "From here, I **need** to know if what is considered as novel is really novel. Intersect with annotated genes. Those that do not overlap at all, are novels for sure: maintain.\n",
    "\n",
    "For **human**:\n",
    "\n",
    "`$GENOMEDIR/Annot_files_GTF/gencode.v38.primary_assembly.annotation.gtf`\n",
    "\n",
    "\n",
    "`$GENOMEDIR/Annot_files_GTF/GENCODE_47_M36/gencode.v47.primary_assembly.annotation.fixed.gtf`\n",
    "\n",
    "For **macaca**:\n",
    "\n",
    "`/datasets/Common/ReferenceGenomes/Macaca/10.111/GenomesAndAnnotation/Macaca_mulatta.Mmul_10.111.bed`"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash -s \"$users_dir\" \"$specie\"\n",
    "\n",
    "OUTDIR=$1/$2/stringtieReconstruction\n",
    "\n",
    "mkdir $OUTDIR/NOoriented\n",
    "mkdir $OUTDIR/oriented\n",
    "\n",
    "for file in $OUTDIR/*.gtf; do \n",
    "    if [[ ${file} != *\"cov\"* ]]; then \n",
    "        if [[ ${file} != *\"gene\"* ]]; then \n",
    "            name=${file##*/}\n",
    "            name=${name%%.*}\n",
    "            #separation of novel transcripts oriented not oriented\n",
    "            awk '($7 == \".\") {print}' $file > $OUTDIR/NOoriented/${name}_NOoriented.gtf\n",
    "            awk '($7 != \".\") {print}' $file > $OUTDIR/oriented/${name}_oriented.gtf\n",
    "        fi\n",
    "    fi\n",
    "done\n"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "**FORCING STRANDNESS**\n",
    "\n",
    "For the novel transcripts oriented, we force orientation when checking the overlapping with the reference."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "***** WARNING: File /data/genomics/marta/genomes/Annot_files_GTF/GENCODE_47_M36/gencode.v47.primary_assembly.annotation.fixed.gtf has inconsistent naming convention for record:\n",
      "GL000195.1\tHAVANA\tgene\t2579\t87094\t.\t-\t.\tgene_id \"ENSG00000303853.1\"; ID \"ENSG00000303853.1\"; gene_name \"ENSG00000303853\"; gene_type \"lncRNA\"; level \"2\"; tag \"ncRNA_host\" \"overlapping_locus\";\n",
      "\n",
      "***** WARNING: File /data/genomics/marta/genomes/Annot_files_GTF/GENCODE_47_M36/gencode.v47.primary_assembly.annotation.fixed.gtf has inconsistent naming convention for record:\n",
      "GL000195.1\tHAVANA\tgene\t2579\t87094\t.\t-\t.\tgene_id \"ENSG00000303853.1\"; ID \"ENSG00000303853.1\"; gene_name \"ENSG00000303853\"; gene_type \"lncRNA\"; level \"2\"; tag \"ncRNA_host\" \"overlapping_locus\";\n",
      "\n",
      "***** WARNING: File /data/genomics/marta/genomes/Annot_files_GTF/GENCODE_47_M36/gencode.v47.primary_assembly.annotation.fixed.gtf has inconsistent naming convention for record:\n",
      "GL000195.1\tHAVANA\tgene\t2579\t87094\t.\t-\t.\tgene_id \"ENSG00000303853.1\"; ID \"ENSG00000303853.1\"; gene_name \"ENSG00000303853\"; gene_type \"lncRNA\"; level \"2\"; tag \"ncRNA_host\" \"overlapping_locus\";\n",
      "\n",
      "***** WARNING: File /data/genomics/marta/genomes/Annot_files_GTF/GENCODE_47_M36/gencode.v47.primary_assembly.annotation.fixed.gtf has inconsistent naming convention for record:\n",
      "GL000195.1\tHAVANA\tgene\t2579\t87094\t.\t-\t.\tgene_id \"ENSG00000303853.1\"; ID \"ENSG00000303853.1\"; gene_name \"ENSG00000303853\"; gene_type \"lncRNA\"; level \"2\"; tag \"ncRNA_host\" \"overlapping_locus\";\n",
      "\n",
      "***** WARNING: File /data/genomics/marta/genomes/Annot_files_GTF/GENCODE_47_M36/gencode.v47.primary_assembly.annotation.fixed.gtf has inconsistent naming convention for record:\n",
      "GL000195.1\tHAVANA\tgene\t2579\t87094\t.\t-\t.\tgene_id \"ENSG00000303853.1\"; ID \"ENSG00000303853.1\"; gene_name \"ENSG00000303853\"; gene_type \"lncRNA\"; level \"2\"; tag \"ncRNA_host\" \"overlapping_locus\";\n",
      "\n",
      "***** WARNING: File /data/genomics/marta/genomes/Annot_files_GTF/GENCODE_47_M36/gencode.v47.primary_assembly.annotation.fixed.gtf has inconsistent naming convention for record:\n",
      "GL000195.1\tHAVANA\tgene\t2579\t87094\t.\t-\t.\tgene_id \"ENSG00000303853.1\"; ID \"ENSG00000303853.1\"; gene_name \"ENSG00000303853\"; gene_type \"lncRNA\"; level \"2\"; tag \"ncRNA_host\" \"overlapping_locus\";\n",
      "\n",
      "***** WARNING: File /data/genomics/marta/genomes/Annot_files_GTF/GENCODE_47_M36/gencode.v47.primary_assembly.annotation.fixed.gtf has inconsistent naming convention for record:\n",
      "GL000195.1\tHAVANA\tgene\t2579\t87094\t.\t-\t.\tgene_id \"ENSG00000303853.1\"; ID \"ENSG00000303853.1\"; gene_name \"ENSG00000303853\"; gene_type \"lncRNA\"; level \"2\"; tag \"ncRNA_host\" \"overlapping_locus\";\n",
      "\n",
      "***** WARNING: File /data/genomics/marta/genomes/Annot_files_GTF/GENCODE_47_M36/gencode.v47.primary_assembly.annotation.fixed.gtf has inconsistent naming convention for record:\n",
      "GL000195.1\tHAVANA\tgene\t2579\t87094\t.\t-\t.\tgene_id \"ENSG00000303853.1\"; ID \"ENSG00000303853.1\"; gene_name \"ENSG00000303853\"; gene_type \"lncRNA\"; level \"2\"; tag \"ncRNA_host\" \"overlapping_locus\";\n",
      "\n",
      "***** WARNING: File /data/genomics/marta/genomes/Annot_files_GTF/GENCODE_47_M36/gencode.v47.primary_assembly.annotation.fixed.gtf has inconsistent naming convention for record:\n",
      "GL000195.1\tHAVANA\tgene\t2579\t87094\t.\t-\t.\tgene_id \"ENSG00000303853.1\"; ID \"ENSG00000303853.1\"; gene_name \"ENSG00000303853\"; gene_type \"lncRNA\"; level \"2\"; tag \"ncRNA_host\" \"overlapping_locus\";\n",
      "\n",
      "***** WARNING: File /data/genomics/marta/genomes/Annot_files_GTF/GENCODE_47_M36/gencode.v47.primary_assembly.annotation.fixed.gtf has inconsistent naming convention for record:\n",
      "GL000195.1\tHAVANA\tgene\t2579\t87094\t.\t-\t.\tgene_id \"ENSG00000303853.1\"; ID \"ENSG00000303853.1\"; gene_name \"ENSG00000303853\"; gene_type \"lncRNA\"; level \"2\"; tag \"ncRNA_host\" \"overlapping_locus\";\n",
      "\n",
      "***** WARNING: File /data/genomics/marta/genomes/Annot_files_GTF/GENCODE_47_M36/gencode.v47.primary_assembly.annotation.fixed.gtf has inconsistent naming convention for record:\n",
      "GL000195.1\tHAVANA\tgene\t2579\t87094\t.\t-\t.\tgene_id \"ENSG00000303853.1\"; ID \"ENSG00000303853.1\"; gene_name \"ENSG00000303853\"; gene_type \"lncRNA\"; level \"2\"; tag \"ncRNA_host\" \"overlapping_locus\";\n",
      "\n",
      "***** WARNING: File /data/genomics/marta/genomes/Annot_files_GTF/GENCODE_47_M36/gencode.v47.primary_assembly.annotation.fixed.gtf has inconsistent naming convention for record:\n",
      "GL000195.1\tHAVANA\tgene\t2579\t87094\t.\t-\t.\tgene_id \"ENSG00000303853.1\"; ID \"ENSG00000303853.1\"; gene_name \"ENSG00000303853\"; gene_type \"lncRNA\"; level \"2\"; tag \"ncRNA_host\" \"overlapping_locus\";\n",
      "\n",
      "***** WARNING: File /data/genomics/marta/genomes/Annot_files_GTF/GENCODE_47_M36/gencode.v47.primary_assembly.annotation.fixed.gtf has inconsistent naming convention for record:\n",
      "GL000195.1\tHAVANA\tgene\t2579\t87094\t.\t-\t.\tgene_id \"ENSG00000303853.1\"; ID \"ENSG00000303853.1\"; gene_name \"ENSG00000303853\"; gene_type \"lncRNA\"; level \"2\"; tag \"ncRNA_host\" \"overlapping_locus\";\n",
      "\n",
      "***** WARNING: File /data/genomics/marta/genomes/Annot_files_GTF/GENCODE_47_M36/gencode.v47.primary_assembly.annotation.fixed.gtf has inconsistent naming convention for record:\n",
      "GL000195.1\tHAVANA\tgene\t2579\t87094\t.\t-\t.\tgene_id \"ENSG00000303853.1\"; ID \"ENSG00000303853.1\"; gene_name \"ENSG00000303853\"; gene_type \"lncRNA\"; level \"2\"; tag \"ncRNA_host\" \"overlapping_locus\";\n",
      "\n",
      "***** WARNING: File /data/genomics/marta/genomes/Annot_files_GTF/GENCODE_47_M36/gencode.v47.primary_assembly.annotation.fixed.gtf has inconsistent naming convention for record:\n",
      "GL000195.1\tHAVANA\tgene\t2579\t87094\t.\t-\t.\tgene_id \"ENSG00000303853.1\"; ID \"ENSG00000303853.1\"; gene_name \"ENSG00000303853\"; gene_type \"lncRNA\"; level \"2\"; tag \"ncRNA_host\" \"overlapping_locus\";\n",
      "\n",
      "***** WARNING: File /data/genomics/marta/genomes/Annot_files_GTF/GENCODE_47_M36/gencode.v47.primary_assembly.annotation.fixed.gtf has inconsistent naming convention for record:\n",
      "GL000195.1\tHAVANA\tgene\t2579\t87094\t.\t-\t.\tgene_id \"ENSG00000303853.1\"; ID \"ENSG00000303853.1\"; gene_name \"ENSG00000303853\"; gene_type \"lncRNA\"; level \"2\"; tag \"ncRNA_host\" \"overlapping_locus\";\n",
      "\n",
      "***** WARNING: File /data/genomics/marta/genomes/Annot_files_GTF/GENCODE_47_M36/gencode.v47.primary_assembly.annotation.fixed.gtf has inconsistent naming convention for record:\n",
      "GL000195.1\tHAVANA\tgene\t2579\t87094\t.\t-\t.\tgene_id \"ENSG00000303853.1\"; ID \"ENSG00000303853.1\"; gene_name \"ENSG00000303853\"; gene_type \"lncRNA\"; level \"2\"; tag \"ncRNA_host\" \"overlapping_locus\";\n",
      "\n",
      "***** WARNING: File /data/genomics/marta/genomes/Annot_files_GTF/GENCODE_47_M36/gencode.v47.primary_assembly.annotation.fixed.gtf has inconsistent naming convention for record:\n",
      "GL000195.1\tHAVANA\tgene\t2579\t87094\t.\t-\t.\tgene_id \"ENSG00000303853.1\"; ID \"ENSG00000303853.1\"; gene_name \"ENSG00000303853\"; gene_type \"lncRNA\"; level \"2\"; tag \"ncRNA_host\" \"overlapping_locus\";\n",
      "\n"
     ]
    }
   ],
   "source": [
    "%%bash -s \"$users_dir\" \"$specie\" \"$GENOMEDIR\"\n",
    "\n",
    "OUTDIR=$1/$2/stringtieReconstruction\n",
    "\n",
    "module load BEDTools/2.30.0-GCC-10.2.0\n",
    "\n",
    "if [ \"$2\" == \"macaca\" ]; then\n",
    "  REF=/datasets/Common/ReferenceGenomes/Macaca/10.111/GenomesAndAnnotation/Macaca_mulatta.Mmul_10.111.gtf\n",
    "elif [ \"$2\" == \"mouse\" ]; then\n",
    "  REF=/datasets/Common/ReferenceGenomes/Mouse/mm39/GenomesAndAnnotation/gencode.vM32.annotation.gtf\n",
    "elif [ \"$2\" == \"human\" ]; then\n",
    "  REF=/data/genomics/marta/genomes/Annot_files_GTF/GENCODE_47_M36/gencode.v47.primary_assembly.annotation.fixed.gtf\n",
    "elif [ \"$2\" == \"platypus\" ]; then\n",
    "  REF=/datasets/Common/ReferenceGenomes/Platypus/mOrnaAna1/GenomesAndAnnotation/Ornithorhynchus_anatinus.mOrnAna1.p.v1.112.chr.gtf\n",
    "elif [ \"$2\" == \"opossum\" ]; then\n",
    "  REF=/datasets/Common/ReferenceGenomes/Opossum/ASSM229v1/GenomesAndAnnotation/Monodelphis_domestica.ASM229v1.112.chr.gtf\n",
    "elif [ \"$2\" == \"chicken\" ]; then\n",
    "  REF=/datasets/Common/ReferenceGenomes/Chicken/GRCg7b/GenomesAndAnnotation/Gallus_gallus.bGalGal1.mat.broiler.GRCg7b.112.chr.gtf\n",
    "fi\n",
    "\n",
    "for file in $OUTDIR/oriented/*oriented.gtf; do \n",
    "    name=${file##*/}\n",
    "    name=${name%%.*}\n",
    "    #select novel transcripts that do not overlap at all with the gencode annotation (-v)\n",
    "    bedtools intersect -nonamecheck -v -s -a $file -b $REF > $OUTDIR/oriented/${name}_novel.gtf\n",
    "done\n"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "**WITHOUT FORCING STRANDNESS**\n",
    "\n",
    "For the novel transcripts NOT oriented, we do not force orientation when checking the overlapping with the reference."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash -s \"$users_dir\" \"$specie\" \"$GENOMEDIR\"\n",
    "\n",
    "OUTDIR=$1/$2/stringtieReconstruction\n",
    "\n",
    "module load BEDTools/2.30.0-GCC-10.2.0\n",
    "\n",
    "if [ \"$2\" == \"macaca\" ]; then\n",
    "  REF=/datasets/Common/ReferenceGenomes/Macaca/10.111/GenomesAndAnnotation/Macaca_mulatta.Mmul_10.111.gtf\n",
    "elif [ \"$2\" == \"mouse\" ]; then\n",
    "  REF=/datasets/Common/ReferenceGenomes/Mouse/mm39/GenomesAndAnnotation/gencode.vM32.annotation.gtf\n",
    "elif [ \"$2\" == \"human\" ]; then\n",
    "  REF=/data/genomics/marta/genomes/Annot_files_GTF/GENCODE_47_M36/gencode.v47.primary_assembly.annotation.fixed.gtf\n",
    "elif [ \"$2\" == \"platypus\" ]; then\n",
    "  REF=/datasets/Common/ReferenceGenomes/Platypus/mOrnaAna1/GenomesAndAnnotation/Ornithorhynchus_anatinus.mOrnAna1.p.v1.112.chr.gtf\n",
    "elif [ \"$2\" == \"opossum\" ]; then\n",
    "  REF=/datasets/Common/ReferenceGenomes/Opossum/ASSM229v1/GenomesAndAnnotation/Monodelphis_domestica.ASM229v1.112.chr.gtf\n",
    "elif [ \"$2\" == \"chicken\" ]; then\n",
    "  REF=/datasets/Common/ReferenceGenomes/Chicken/GRCg7b/GenomesAndAnnotation/Gallus_gallus.bGalGal1.mat.broiler.GRCg7b.112.chr.gtf\n",
    "fi\n",
    "\n",
    "for file in $OUTDIR/NOoriented/*oriented.gtf; do \n",
    "    name=${file##*/}\n",
    "    name=${name%%.*}\n",
    "\n",
    "    #select novel transcripts that do not overlap at all with the gencode annotation (-v)\n",
    "    bedtools intersect -nonamecheck -v -a $file -b $REF > $OUTDIR/NOoriented/${name}_novel.gtf\n",
    "done"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "I want to see if there are \"alone exons\", I don't want them, only full transcripts. Check transcript line, store geneid (STRG.XX), see if later exons contain the same. If so, store in new file, if not, skip."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "for file in os.listdir(os.path.join(users_dir,specie,\"stringtieReconstruction/NOoriented\")):\n",
    "    if file.endswith(\"novel.gtf\"):\n",
    "        outname=file[:-4]+\"_complete.gtf\"\n",
    "        with open(os.path.join(os.path.join(users_dir,specie,\"stringtieReconstruction/NOoriented\"),file)) as inp:\n",
    "            with open(os.path.join(os.path.join(users_dir,specie,\"stringtieReconstruction/NOoriented\"),outname), 'w') as out:\n",
    "                lines = inp.readlines()\n",
    "                t_gene_id = \"\"\n",
    "                for line in lines:\n",
    "                    tab_split_line=line.split(\"\\t\")\n",
    "                    coma_split_line=line.split(\";\")\n",
    "                    gene_id=re.findall(r'\"([^\"]*)\"',coma_split_line[0])\n",
    "                    if tab_split_line[2] == \"transcript\":\n",
    "                        t_gene_id = gene_id[0]\n",
    "                        out.write(line)\n",
    "                    if tab_split_line[2] == \"exon\":\n",
    "                        if gene_id[0] == t_gene_id:\n",
    "                            out.write(line)\n",
    "                        else:\n",
    "                            continue\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "for file in os.listdir(os.path.join(users_dir,specie,\"stringtieReconstruction/oriented\")):\n",
    "    if file.endswith(\"novel.gtf\"):\n",
    "        outname=file[:-4]+\"_complete.gtf\"\n",
    "        with open(os.path.join(os.path.join(users_dir,specie,\"stringtieReconstruction/oriented\"),file)) as inp:\n",
    "            with open(os.path.join(os.path.join(users_dir,specie,\"stringtieReconstruction/oriented\"),outname), 'w') as out:\n",
    "                lines = inp.readlines()\n",
    "                t_gene_id = \"\"\n",
    "                for line in lines:\n",
    "                    tab_split_line=line.split(\"\\t\")\n",
    "                    coma_split_line=line.split(\";\")\n",
    "                    gene_id=re.findall(r'\"([^\"]*)\"',coma_split_line[0])\n",
    "                    if tab_split_line[2] == \"transcript\":\n",
    "                        t_gene_id = gene_id[0]\n",
    "                        out.write(line)\n",
    "                    if tab_split_line[2] == \"exon\":\n",
    "                        if gene_id[0] == t_gene_id:\n",
    "                            out.write(line)\n",
    "                        else:\n",
    "                            continue\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "mkdir: cannot create directory ‘/users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/human/stringtieReconstruction/intersect_novel’: File exists\n"
     ]
    }
   ],
   "source": [
    "%%bash -s \"$users_dir\" \"$specie\" \"$GENOMEDIR\"\n",
    "\n",
    "## Concatenate both oriented and unoriented\n",
    "OUTDIR=$1/$2/stringtieReconstruction/intersect_novel\n",
    "mkdir $OUTDIR\n",
    "\n",
    "\n",
    "for sample in $(cat $1/$2/stringtieReconstruction/transcriptomeReconstruction.csv | tail -n +2 | cut -f1); do\n",
    "    #concatenate both results in order to obtain a gtf file with all novel transcripts considered\n",
    "    cat $1/$2/stringtieReconstruction/oriented/${sample}_oriented_novel_complete.gtf $1/$2/stringtieReconstruction/NOoriented/${sample}_NOoriented_novel_complete.gtf > $OUTDIR/${sample}_novel.gtf\n",
    "done\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/human/stringtieReconstruction/clean_transcriptomeReconstruction.csv\n"
     ]
    }
   ],
   "source": [
    "%%bash -s \"$users_dir\" \"$specie\"\n",
    "## Total transcriptome + Presumably annotated + Presumably novel\n",
    "OUT=$1/$2/stringtieReconstruction/clean_transcriptomeReconstruction.csv\n",
    "OUTDIR=$1/$2/stringtieReconstruction/intersect_novel\n",
    "\n",
    "echo $OUT\n",
    "if [ -f \"$OUT\" ] ; then\n",
    "    rm \"$OUT\"\n",
    "fi\n",
    "echo -e \"Sample\\tNovel_transcript\" >> $OUT\n",
    "\n",
    "for file in $OUTDIR/*_novel.gtf; do\n",
    "        name=\"${file%%_novel.*}\"\n",
    "        name=\"${name##*/}\"\n",
    "        echo -e $name\"\\t\"$(awk '($3 == \"transcript\") {print}' $file | wc -l) >> $OUT\n",
    "done\n"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Now we use gffcompare so that we unify all novel transcripts into a single transcriptome"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "mkdir: cannot create directory ‘/users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/human/gffcompare’: File exists\n",
      "Loading query file #1: /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/human/stringtieReconstruction/intersect_novel/human_brain_rna_1_novel.gtf\n",
      "Loading query file #2: /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/human/stringtieReconstruction/intersect_novel/human_brain_rna_2_novel.gtf\n",
      "Loading query file #3: /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/human/stringtieReconstruction/intersect_novel/human_brain_rna_3_novel.gtf\n",
      "Loading query file #4: /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/human/stringtieReconstruction/intersect_novel/human_liver_rna_1_novel.gtf\n",
      "Loading query file #5: /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/human/stringtieReconstruction/intersect_novel/human_liver_rna_2_novel.gtf\n",
      "Loading query file #6: /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/human/stringtieReconstruction/intersect_novel/human_liver_rna_3_novel.gtf\n",
      "Loading query file #7: /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/human/stringtieReconstruction/intersect_novel/human_testis_rna_1_novel.gtf\n",
      "Loading query file #8: /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/human/stringtieReconstruction/intersect_novel/human_testis_rna_2_novel.gtf\n",
      "Loading query file #9: /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/human/stringtieReconstruction/intersect_novel/human_testis_rna_3_novel.gtf\n"
     ]
    }
   ],
   "source": [
    "%%bash -s \"$users_dir\" \"$specie\"\n",
    "\n",
    "OUTDIR=$1/$2/gffcompare\n",
    "mkdir $OUTDIR\n",
    "\n",
    "module load gffcompare/0.12.2\n",
    "\n",
    "ls $1/$2/stringtieReconstruction/intersect_novel/*_novel.gtf > $OUTDIR/list_novel_assembled_files_stranded.txt\n",
    "\n",
    "cd $OUTDIR\n",
    "gffcompare -i list_novel_assembled_files_stranded.txt -o gffcmp.assembled.stranded\n",
    "\n",
    "\n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "    570 +\n",
      "    613 -\n"
     ]
    }
   ],
   "source": [
    "%%bash -s \"$users_dir\" \"$specie\"\n",
    "\n",
    "echo -e \"\\n\"$proj\n",
    "awk '$3 == \"transcript\" {print}' $1/$2/gffcompare/gffcmp.assembled.stranded.combined.gtf | cut -f 7 | sort | uniq -c\n"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Let's create a new reference file with novel reconstructed transcripts"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "mkdir: cannot create directory ‘/users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/human/newReference_Resconstructed’: File exists\n"
     ]
    }
   ],
   "source": [
    "%%bash -s \"$users_dir\" \"$specie\" \"$GENOMEDIR\"\n",
    "\n",
    "export PATH=/data/genomics/marta/tools/gff3sort-master:$PATH\n",
    "\n",
    "## reference \n",
    "REF=$3/Annot_files_GTF/GENCODE_47_M36/gencode.v47.primary_assembly.annotation.fixed.gtf \n",
    "\n",
    "OUTDIR=$1/$2/newReference_Resconstructed\n",
    "mkdir $OUTDIR\n",
    "\n",
    "## gffcompare\n",
    "gffcompare=$1/$2/gffcompare/gffcmp.assembled.stranded.combined.gtf\n",
    "\n",
    "## concat\n",
    "cat $gffcompare $REF > $OUTDIR/gencode.v47.gffcompare.TestisLiverBrain.annotation.gtf\n",
    "\n",
    "#sort the newly assembled transcriptome\n",
    "gff3sort.pl --precise --chr_order natural $OUTDIR/gencode.v47.gffcompare.TestisLiverBrain.annotation.gtf > $OUTDIR/gencode.v47.gffcompare.TestisLiverBrain.annotation.sorted.gtf\n",
    "rm $OUTDIR/gencode.v47.gffcompare.TestisLiverBrain.annotation.gtf\n"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Get simplified annotation\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [],
   "source": [
    "## Author: Marta Espinosa Camarena\n",
    "## Date: 21st May 24\n",
    "\n",
    "## This scripts attemps to build a simplified version of an annotation file.\n",
    "## By simplified we mean that we keep only 1 transcript per gene, being the longest transcript the selected one.\n",
    "## Running mode:\n",
    "## python get_simplified_annotation.py $absolute_path_to_annotation $outdir\n",
    "## make sure the output directory has writing permissions\n",
    "\n",
    "import pandas as pd\n",
    "import sys, os\n",
    "from pathlib import Path\n",
    "\n",
    "##----------------- importing requested data -----------------##\n",
    "# annotation_GTF = sys.argv[1]\n",
    "annotation_GTF = \"/users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/human/newReference_Resconstructed/gencode.v47.gffcompare.TestisLiverBrain.annotation.sorted.gtf\" ## testing\n",
    "\n",
    "# outdir = sys.argv[2]\n",
    "outdir = \"/users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/human/newReference_Resconstructed\"\n",
    "outfile = os.path.basename(annotation_GTF)\n",
    "outfile = outfile[:-4]\n",
    "\n",
    "## reading reference file and separate by transcripts and genes\n",
    "reference = pd.read_csv(annotation_GTF, sep=\"\\t\", header=None, comment=\"#\")\n",
    "reference['transcript_id'] = reference[8].str.extract('transcript_id \"([^\"]*)\"', expand=True)\n",
    "reference['gene_id'] = reference[8].str.extract('gene_id \"([^\"]*)\"', expand=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_1959167/101842013.py:6: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  exons['length'] = exons[4] - exons[3]\n"
     ]
    }
   ],
   "source": [
    "reference['transcript_type'] = reference[8].str.extract('transcript_type \"([^\"]*)\"', expand=True)\n",
    "reference['gene_type'] = reference[8].str.extract('gene_type \"([^\"]*)\"', expand=True)\n",
    "\n",
    "## subset exons and calculate length\n",
    "exons = reference[reference[2] == \"exon\"]\n",
    "exons['length'] = exons[4] - exons[3]\n",
    "genes = reference[reference[2] == \"gene\"]\n",
    "\n",
    "## calculate length transcripts\n",
    "length_tid = exons.groupby(['transcript_id'], as_index=False)[['length']].sum()\n",
    "\n",
    "\n",
    "## make dataframe with geneid_transcriptid_length\n",
    "ref_w_length = length_tid.merge(reference, on=[\"transcript_id\"], how=\"right\").drop_duplicates()\n",
    "# ref_w_length = ref_w_length[['transcript_id','gene_id','length']]\n",
    "# ref_w_length.drop_duplicates(inplace=True)\n",
    "# ref_w_length.dropna(inplace=True)\n",
    "\n",
    "## select longest transcript per gene\n",
    "## Group by 'gene_id' and get the indices of the rows with the maximum 'length'\n",
    "## select datafrem rows corresponding to longest trancript_id - if different transcripts for the same gene with same length, we keep the first one. \n",
    "ref_w_length\n",
    "## For protein_coding genes we keep the longest protein coding transcript\n",
    "filtered_ref_w_length = ref_w_length[\n",
    "    (ref_w_length['gene_type'] != 'protein_coding') | \n",
    "    ((ref_w_length['gene_type'] == 'protein_coding') & (ref_w_length['transcript_type'] == 'protein_coding'))\n",
    "]\n",
    "\n",
    "max_length_indices = filtered_ref_w_length.groupby('gene_id')['length'].idxmax()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>transcript_id</th>\n",
       "      <th>length</th>\n",
       "      <th>0</th>\n",
       "      <th>1</th>\n",
       "      <th>2</th>\n",
       "      <th>3</th>\n",
       "      <th>4</th>\n",
       "      <th>5</th>\n",
       "      <th>6</th>\n",
       "      <th>7</th>\n",
       "      <th>8</th>\n",
       "      <th>gene_id</th>\n",
       "      <th>transcript_type</th>\n",
       "      <th>gene_type</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>84487</th>\n",
       "      <td>ENST00000612152.4</td>\n",
       "      <td>3789.0</td>\n",
       "      <td>chrX</td>\n",
       "      <td>ENSEMBL</td>\n",
       "      <td>transcript</td>\n",
       "      <td>100627109</td>\n",
       "      <td>100637104</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>.</td>\n",
       "      <td>gene_id \"ENSG00000000003.16\"; transcript_id \"E...</td>\n",
       "      <td>ENSG00000000003.16</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>protein_coding</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>84459</th>\n",
       "      <td>ENST00000373031.5</td>\n",
       "      <td>1198.0</td>\n",
       "      <td>chrX</td>\n",
       "      <td>HAVANA</td>\n",
       "      <td>transcript</td>\n",
       "      <td>100584936</td>\n",
       "      <td>100599885</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>gene_id \"ENSG00000000005.6\"; transcript_id \"EN...</td>\n",
       "      <td>ENSG00000000005.6</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>protein_coding</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3946569</th>\n",
       "      <td>ENST00000371582.8</td>\n",
       "      <td>1151.0</td>\n",
       "      <td>chr20</td>\n",
       "      <td>HAVANA</td>\n",
       "      <td>exon</td>\n",
       "      <td>50934867</td>\n",
       "      <td>50935236</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>.</td>\n",
       "      <td>gene_id \"ENSG00000000419.14\"; transcript_id \"E...</td>\n",
       "      <td>ENSG00000000419.14</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>protein_coding</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>436469</th>\n",
       "      <td>ENST00000367771.11</td>\n",
       "      <td>6295.0</td>\n",
       "      <td>chr1</td>\n",
       "      <td>HAVANA</td>\n",
       "      <td>exon</td>\n",
       "      <td>169849631</td>\n",
       "      <td>169853772</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>.</td>\n",
       "      <td>gene_id \"ENSG00000000457.14\"; transcript_id \"E...</td>\n",
       "      <td>ENSG00000000457.14</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>protein_coding</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>436221</th>\n",
       "      <td>ENST00000286031.10</td>\n",
       "      <td>4331.0</td>\n",
       "      <td>chr1</td>\n",
       "      <td>HAVANA</td>\n",
       "      <td>exon</td>\n",
       "      <td>169795409</td>\n",
       "      <td>169795983</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>gene_id \"ENSG00000000460.17\"; transcript_id \"E...</td>\n",
       "      <td>ENSG00000000460.17</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>protein_coding</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>95227</th>\n",
       "      <td>TCONS_00001174</td>\n",
       "      <td>491.0</td>\n",
       "      <td>chrX</td>\n",
       "      <td>StringTie</td>\n",
       "      <td>transcript</td>\n",
       "      <td>109054114</td>\n",
       "      <td>109054605</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>.</td>\n",
       "      <td>transcript_id \"TCONS_00001174\"; gene_id \"XLOC_...</td>\n",
       "      <td>XLOC_001035</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>100636</th>\n",
       "      <td>TCONS_00001181</td>\n",
       "      <td>216.0</td>\n",
       "      <td>chrX</td>\n",
       "      <td>StringTie</td>\n",
       "      <td>exon</td>\n",
       "      <td>119077902</td>\n",
       "      <td>119078118</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>.</td>\n",
       "      <td>transcript_id \"TCONS_00001181\"; gene_id \"XLOC_...</td>\n",
       "      <td>XLOC_001036</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>135923</th>\n",
       "      <td>TCONS_00001173</td>\n",
       "      <td>314.0</td>\n",
       "      <td>chrX</td>\n",
       "      <td>StringTie</td>\n",
       "      <td>transcript</td>\n",
       "      <td>155419406</td>\n",
       "      <td>155419720</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>.</td>\n",
       "      <td>transcript_id \"TCONS_00001173\"; gene_id \"XLOC_...</td>\n",
       "      <td>XLOC_001037</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>141020</th>\n",
       "      <td>TCONS_00001183</td>\n",
       "      <td>321.0</td>\n",
       "      <td>chrY</td>\n",
       "      <td>StringTie</td>\n",
       "      <td>transcript</td>\n",
       "      <td>9262142</td>\n",
       "      <td>9262463</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>transcript_id \"TCONS_00001183\"; gene_id \"XLOC_...</td>\n",
       "      <td>XLOC_001038</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>144364</th>\n",
       "      <td>TCONS_00001182</td>\n",
       "      <td>258.0</td>\n",
       "      <td>chrY</td>\n",
       "      <td>StringTie</td>\n",
       "      <td>transcript</td>\n",
       "      <td>13917646</td>\n",
       "      <td>13917904</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>transcript_id \"TCONS_00001182\"; gene_id \"XLOC_...</td>\n",
       "      <td>XLOC_001039</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>79623 rows × 14 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "              transcript_id  length      0          1           2          3  \\\n",
       "84487     ENST00000612152.4  3789.0   chrX    ENSEMBL  transcript  100627109   \n",
       "84459     ENST00000373031.5  1198.0   chrX     HAVANA  transcript  100584936   \n",
       "3946569   ENST00000371582.8  1151.0  chr20     HAVANA        exon   50934867   \n",
       "436469   ENST00000367771.11  6295.0   chr1     HAVANA        exon  169849631   \n",
       "436221   ENST00000286031.10  4331.0   chr1     HAVANA        exon  169795409   \n",
       "...                     ...     ...    ...        ...         ...        ...   \n",
       "95227        TCONS_00001174   491.0   chrX  StringTie  transcript  109054114   \n",
       "100636       TCONS_00001181   216.0   chrX  StringTie        exon  119077902   \n",
       "135923       TCONS_00001173   314.0   chrX  StringTie  transcript  155419406   \n",
       "141020       TCONS_00001183   321.0   chrY  StringTie  transcript    9262142   \n",
       "144364       TCONS_00001182   258.0   chrY  StringTie  transcript   13917646   \n",
       "\n",
       "                 4  5  6  7  \\\n",
       "84487    100637104  .  -  .   \n",
       "84459    100599885  .  +  .   \n",
       "3946569   50935236  .  -  .   \n",
       "436469   169853772  .  -  .   \n",
       "436221   169795983  .  +  .   \n",
       "...            ... .. .. ..   \n",
       "95227    109054605  .  -  .   \n",
       "100636   119078118  .  -  .   \n",
       "135923   155419720  .  -  .   \n",
       "141020     9262463  .  +  .   \n",
       "144364    13917904  .  +  .   \n",
       "\n",
       "                                                         8  \\\n",
       "84487    gene_id \"ENSG00000000003.16\"; transcript_id \"E...   \n",
       "84459    gene_id \"ENSG00000000005.6\"; transcript_id \"EN...   \n",
       "3946569  gene_id \"ENSG00000000419.14\"; transcript_id \"E...   \n",
       "436469   gene_id \"ENSG00000000457.14\"; transcript_id \"E...   \n",
       "436221   gene_id \"ENSG00000000460.17\"; transcript_id \"E...   \n",
       "...                                                    ...   \n",
       "95227    transcript_id \"TCONS_00001174\"; gene_id \"XLOC_...   \n",
       "100636   transcript_id \"TCONS_00001181\"; gene_id \"XLOC_...   \n",
       "135923   transcript_id \"TCONS_00001173\"; gene_id \"XLOC_...   \n",
       "141020   transcript_id \"TCONS_00001183\"; gene_id \"XLOC_...   \n",
       "144364   transcript_id \"TCONS_00001182\"; gene_id \"XLOC_...   \n",
       "\n",
       "                    gene_id transcript_type       gene_type  \n",
       "84487    ENSG00000000003.16  protein_coding  protein_coding  \n",
       "84459     ENSG00000000005.6  protein_coding  protein_coding  \n",
       "3946569  ENSG00000000419.14  protein_coding  protein_coding  \n",
       "436469   ENSG00000000457.14  protein_coding  protein_coding  \n",
       "436221   ENSG00000000460.17  protein_coding  protein_coding  \n",
       "...                     ...             ...             ...  \n",
       "95227           XLOC_001035             NaN             NaN  \n",
       "100636          XLOC_001036             NaN             NaN  \n",
       "135923          XLOC_001037             NaN             NaN  \n",
       "141020          XLOC_001038             NaN             NaN  \n",
       "144364          XLOC_001039             NaN             NaN  \n",
       "\n",
       "[79623 rows x 14 columns]"
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "longest_transcripts_gid = ref_w_length.loc[max_length_indices]\n",
    "longest_transcripts_gid"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_1959167/1651782170.py:2: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  longest_transcripts_gid.drop_duplicates(inplace=True) ## no duplications\n"
     ]
    }
   ],
   "source": [
    "longest_transcripts_gid = longest_transcripts_gid[['gene_id','transcript_id','length']]\n",
    "longest_transcripts_gid.drop_duplicates(inplace=True) ## no duplications\n",
    "\n",
    "reference_longest = longest_transcripts_gid.merge(ref_w_length[ref_w_length[2] != \"gene\"], on=['gene_id','transcript_id','length'], how=\"left\")\n",
    "\n",
    "\n",
    "full_reference = pd.concat([reference_longest, genes])\n",
    "full_reference.drop(['transcript_id','gene_id','length','transcript_type','gene_type'], axis=1, inplace=True)\n",
    "full_reference.dropna(inplace=True)\n",
    "full_reference.to_csv(outdir+\"/\"+outfile+\".1transcript.gtf\", sep=\"\\t\", header=None, index=None,  quoting = 3)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>0</th>\n",
       "      <th>1</th>\n",
       "      <th>2</th>\n",
       "      <th>3</th>\n",
       "      <th>4</th>\n",
       "      <th>5</th>\n",
       "      <th>6</th>\n",
       "      <th>7</th>\n",
       "      <th>8</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chrX</td>\n",
       "      <td>ENSEMBL</td>\n",
       "      <td>transcript</td>\n",
       "      <td>100627109</td>\n",
       "      <td>100637104</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>.</td>\n",
       "      <td>gene_id \"ENSG00000000003.16\"; transcript_id \"E...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chrX</td>\n",
       "      <td>ENSEMBL</td>\n",
       "      <td>exon</td>\n",
       "      <td>100627109</td>\n",
       "      <td>100629986</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>.</td>\n",
       "      <td>gene_id \"ENSG00000000003.16\"; transcript_id \"E...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chrX</td>\n",
       "      <td>ENSEMBL</td>\n",
       "      <td>three_prime_utr</td>\n",
       "      <td>100627109</td>\n",
       "      <td>100629986</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>.</td>\n",
       "      <td>gene_id \"ENSG00000000003.16\"; transcript_id \"E...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chrX</td>\n",
       "      <td>ENSEMBL</td>\n",
       "      <td>exon</td>\n",
       "      <td>100630759</td>\n",
       "      <td>100630866</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>.</td>\n",
       "      <td>gene_id \"ENSG00000000003.16\"; transcript_id \"E...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chrX</td>\n",
       "      <td>ENSEMBL</td>\n",
       "      <td>three_prime_utr</td>\n",
       "      <td>100630759</td>\n",
       "      <td>100630797</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>.</td>\n",
       "      <td>gene_id \"ENSG00000000003.16\"; transcript_id \"E...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4114863</th>\n",
       "      <td>KI270751.1</td>\n",
       "      <td>HAVANA</td>\n",
       "      <td>gene</td>\n",
       "      <td>2579</td>\n",
       "      <td>73732</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>.</td>\n",
       "      <td>gene_id \"ENSG00000303867.1\"; ID \"ENSG000003038...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4114911</th>\n",
       "      <td>KI270751.1</td>\n",
       "      <td>HAVANA</td>\n",
       "      <td>gene</td>\n",
       "      <td>34096</td>\n",
       "      <td>71851</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>gene_id \"ENSG00000303902.1\"; ID \"ENSG000003039...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4115234</th>\n",
       "      <td>KI270751.1</td>\n",
       "      <td>HAVANA</td>\n",
       "      <td>gene</td>\n",
       "      <td>133801</td>\n",
       "      <td>139253</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>.</td>\n",
       "      <td>gene_id \"ENSG00000306528.1\"; ID \"ENSG000003065...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4115241</th>\n",
       "      <td>KI270753.1</td>\n",
       "      <td>HAVANA</td>\n",
       "      <td>gene</td>\n",
       "      <td>43135</td>\n",
       "      <td>44491</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>gene_id \"ENSG00000297844.1\"; ID \"ENSG000002978...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4115247</th>\n",
       "      <td>KI270755.1</td>\n",
       "      <td>HAVANA</td>\n",
       "      <td>gene</td>\n",
       "      <td>26350</td>\n",
       "      <td>27723</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>gene_id \"ENSG00000309258.1\"; ID \"ENSG000003092...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>782883 rows × 9 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                  0        1                2          3          4  5  6  7  \\\n",
       "0              chrX  ENSEMBL       transcript  100627109  100637104  .  -  .   \n",
       "1              chrX  ENSEMBL             exon  100627109  100629986  .  -  .   \n",
       "2              chrX  ENSEMBL  three_prime_utr  100627109  100629986  .  -  .   \n",
       "3              chrX  ENSEMBL             exon  100630759  100630866  .  -  .   \n",
       "4              chrX  ENSEMBL  three_prime_utr  100630759  100630797  .  -  .   \n",
       "...             ...      ...              ...        ...        ... .. .. ..   \n",
       "4114863  KI270751.1   HAVANA             gene       2579      73732  .  -  .   \n",
       "4114911  KI270751.1   HAVANA             gene      34096      71851  .  +  .   \n",
       "4115234  KI270751.1   HAVANA             gene     133801     139253  .  -  .   \n",
       "4115241  KI270753.1   HAVANA             gene      43135      44491  .  +  .   \n",
       "4115247  KI270755.1   HAVANA             gene      26350      27723  .  +  .   \n",
       "\n",
       "                                                         8  \n",
       "0        gene_id \"ENSG00000000003.16\"; transcript_id \"E...  \n",
       "1        gene_id \"ENSG00000000003.16\"; transcript_id \"E...  \n",
       "2        gene_id \"ENSG00000000003.16\"; transcript_id \"E...  \n",
       "3        gene_id \"ENSG00000000003.16\"; transcript_id \"E...  \n",
       "4        gene_id \"ENSG00000000003.16\"; transcript_id \"E...  \n",
       "...                                                    ...  \n",
       "4114863  gene_id \"ENSG00000303867.1\"; ID \"ENSG000003038...  \n",
       "4114911  gene_id \"ENSG00000303902.1\"; ID \"ENSG000003039...  \n",
       "4115234  gene_id \"ENSG00000306528.1\"; ID \"ENSG000003065...  \n",
       "4115241  gene_id \"ENSG00000297844.1\"; ID \"ENSG000002978...  \n",
       "4115247  gene_id \"ENSG00000309258.1\"; ID \"ENSG000003092...  \n",
       "\n",
       "[782883 rows x 9 columns]"
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "full_reference"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash -s \"$users_dir\" \"$specie\"\n",
    "\n",
    "export PATH=/data/genomics/marta/tools/gff3sort-master:$PATH\n",
    "\n",
    "\n",
    "reference=$1/$2/newReference_Resconstructed/gencode.v47.gffcompare.TestisLiverBrain.annotation.sorted.1transcript.gtf\n",
    "out=$1/$2/newReference_Resconstructed/gencode.v47.gffcompare.TestisLiverBrain.annotation.sorted.1transcript.sorted.gtf\n",
    "\n",
    "\n",
    "gff3sort.pl --precise --chr_order natural $reference > $out\n"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Quantification\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash -s \"$users_dir\" \"$specie\"\n",
    "\n",
    "# sed 's/^chr//g' $1/$2/newReference_Resconstructed/gencode.v47.gffcompare.TestisLiverBrain.annotation.sorted.gtf > $1/$2/newReference_Resconstructed/gencode.v47.gffcompare.TestisLiverBrain.annotation.sorted.NOchr.gtf\n",
    "# AnnotGTF=$1/$2/newReference_Resconstructed/gencode.v47.gffcompare.TestisLiverBrain.annotation.sorted.NOchr.gtf\n",
    "## 1 transcript\n",
    "sed 's/^chr//g' $1/$2/newReference_Resconstructed/gencode.v47.gffcompare.TestisLiverBrain.annotation.sorted.1transcript.sorted.gtf > $1/$2/newReference_Resconstructed/gencode.v47.gffcompare.TestisLiverBrain.annotation.sorted.1transcript.sorted.NOchr.gtf\n",
    "AnnotGTF=$1/$2/newReference_Resconstructed/gencode.v47.gffcompare.TestisLiverBrain.annotation.sorted.1transcript.sorted.NOchr.gtf\n",
    "\n",
    "## formatting gtf - Cova's fix\n",
    "sed 's/\"\"/\"/g' $AnnotGTF > $1/$2/newReference_Resconstructed/test.gtf\n",
    "sed 's/\\(transcript_id \"[^\"]*\\).*/\\1\"/' $1/$2/newReference_Resconstructed/test.gtf > $1/$2/newReference_Resconstructed/gencode.v47.gffcompare.TestisLiverBrain.annotation.sorted.1transcript.sorted.NOchr.fixed.gtf\n",
    "rm $1/$2/newReference_Resconstructed/test.gtf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "mkdir: cannot create directory ‘/users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/human/featureCounts_gffcompare’: File exists\n",
      "\n",
      "        ==========     _____ _    _ ____  _____  ______          _____  \n",
      "        =====         / ____| |  | |  _ \\|  __ \\|  ____|   /\\   |  __ \\ \n",
      "          =====      | (___ | |  | | |_) | |__) | |__     /  \\  | |  | |\n",
      "            ====      \\___ \\| |  | |  _ <|  _  /|  __|   / /\\ \\ | |  | |\n",
      "              ====    ____) | |__| | |_) | | \\ \\| |____ / ____ \\| |__| |\n",
      "        ==========   |_____/ \\____/|____/|_|  \\_\\______/_/    \\_\\_____/\n",
      "\t  v2.0.3\n",
      "\n",
      "//========================== featureCounts setting ===========================\\\\\n",
      "||                                                                            ||\n",
      "||             Input files : 9 BAM files                                      ||\n",
      "||                                                                            ||\n",
      "||                           human_brain_rna_1Aligned.sortedByCoord.out.bam   ||\n",
      "||                           human_brain_rna_2Aligned.sortedByCoord.out.bam   ||\n",
      "||                           human_brain_rna_3Aligned.sortedByCoord.out.bam   ||\n",
      "||                           human_liver_rna_1Aligned.sortedByCoord.out.bam   ||\n",
      "||                           human_liver_rna_2Aligned.sortedByCoord.out.bam   ||\n",
      "||                           human_liver_rna_3Aligned.sortedByCoord.out.bam   ||\n",
      "||                           human_testis_rna_1Aligned.sortedByCoord.out.bam  ||\n",
      "||                           human_testis_rna_2Aligned.sortedByCoord.out.bam  ||\n",
      "||                           human_testis_rna_3Aligned.sortedByCoord.out.bam  ||\n",
      "||                                                                            ||\n",
      "||             Output file : gffcompare_stranded_featureCounts.txt            ||\n",
      "||                 Summary : gffcompare_stranded_featureCounts.txt.summary    ||\n",
      "||              Paired-end : no                                               ||\n",
      "||        Count read pairs : no                                               ||\n",
      "||              Annotation : gencode.v47.gffcompare.TestisLiverBrain.anno ... ||\n",
      "||      Dir for temp files : /users/genomics/marta/TestisProject_SaraRazq ... ||\n",
      "||                                                                            ||\n",
      "||                 Threads : 10                                               ||\n",
      "||                   Level : meta-feature level                               ||\n",
      "||      Multimapping reads : not counted                                      ||\n",
      "|| Multi-overlapping reads : counted                                          ||\n",
      "||   Min overlapping bases : 1                                                ||\n",
      "||                                                                            ||\n",
      "\\\\============================================================================//\n",
      "\n",
      "//================================= Running ==================================\\\\\n",
      "||                                                                            ||\n",
      "|| Load annotation file gencode.v47.gffcompare.TestisLiverBrain.annotatio ... ||\n",
      "||    Features : 339928                                                       ||\n",
      "||    Meta-features : 79623                                                   ||\n",
      "||    Chromosomes/contigs : 74                                                ||\n",
      "||                                                                            ||\n",
      "|| Process BAM file human_brain_rna_1Aligned.sortedByCoord.out.bam...         ||\n",
      "||    Strand specific : stranded                                              ||\n",
      "||    Single-end reads are included.                                          ||\n",
      "||    Total alignments : 46273364                                             ||\n",
      "||    Successfully assigned alignments : 26663708 (57.6%)                     ||\n",
      "||    Running time : 0.29 minutes                                             ||\n",
      "||                                                                            ||\n",
      "|| Process BAM file human_brain_rna_2Aligned.sortedByCoord.out.bam...         ||\n",
      "||    Strand specific : stranded                                              ||\n",
      "||    Single-end reads are included.                                          ||\n",
      "||    Total alignments : 35698682                                             ||\n",
      "||    Successfully assigned alignments : 30834811 (86.4%)                     ||\n",
      "||    Running time : 0.23 minutes                                             ||\n",
      "||                                                                            ||\n",
      "|| Process BAM file human_brain_rna_3Aligned.sortedByCoord.out.bam...         ||\n",
      "||    Strand specific : stranded                                              ||\n",
      "||    Single-end reads are included.                                          ||\n",
      "||    Total alignments : 76816768                                             ||\n",
      "||    Successfully assigned alignments : 45205451 (58.8%)                     ||\n",
      "||    Running time : 0.51 minutes                                             ||\n",
      "||                                                                            ||\n",
      "|| Process BAM file human_liver_rna_1Aligned.sortedByCoord.out.bam...         ||\n",
      "||    Strand specific : stranded                                              ||\n",
      "||    Single-end reads are included.                                          ||\n",
      "||    Total alignments : 28826009                                             ||\n",
      "||    Successfully assigned alignments : 25146474 (87.2%)                     ||\n",
      "||    Running time : 0.17 minutes                                             ||\n",
      "||                                                                            ||\n",
      "|| Process BAM file human_liver_rna_2Aligned.sortedByCoord.out.bam...         ||\n",
      "||    Strand specific : stranded                                              ||\n",
      "||    Single-end reads are included.                                          ||\n",
      "||    Total alignments : 15281093                                             ||\n",
      "||    Successfully assigned alignments : 12995373 (85.0%)                     ||\n",
      "||    Running time : 0.30 minutes                                             ||\n",
      "||                                                                            ||\n",
      "|| Process BAM file human_liver_rna_3Aligned.sortedByCoord.out.bam...         ||\n",
      "||    Strand specific : stranded                                              ||\n",
      "||    Single-end reads are included.                                          ||\n",
      "||    Total alignments : 14285878                                             ||\n",
      "||    Successfully assigned alignments : 11347975 (79.4%)                     ||\n",
      "||    Running time : 0.09 minutes                                             ||\n",
      "||                                                                            ||\n",
      "|| Process BAM file human_testis_rna_1Aligned.sortedByCoord.out.bam...        ||\n",
      "||    Strand specific : stranded                                              ||\n",
      "||    Single-end reads are included.                                          ||\n",
      "||    Total alignments : 33997991                                             ||\n",
      "||    Successfully assigned alignments : 20874073 (61.4%)                     ||\n",
      "||    Running time : 0.21 minutes                                             ||\n",
      "||                                                                            ||\n",
      "|| Process BAM file human_testis_rna_2Aligned.sortedByCoord.out.bam...        ||\n",
      "||    Strand specific : stranded                                              ||\n",
      "||    Single-end reads are included.                                          ||\n",
      "||    Total alignments : 46379629                                             ||\n",
      "||    Successfully assigned alignments : 28119433 (60.6%)                     ||\n",
      "||    Running time : 0.26 minutes                                             ||\n",
      "||                                                                            ||\n",
      "|| Process BAM file human_testis_rna_3Aligned.sortedByCoord.out.bam...        ||\n",
      "||    Strand specific : stranded                                              ||\n",
      "||    Single-end reads are included.                                          ||\n",
      "||    Total alignments : 45311096                                             ||\n",
      "||    Successfully assigned alignments : 35511344 (78.4%)                     ||\n",
      "||    Running time : 0.24 minutes                                             ||\n",
      "||                                                                            ||\n",
      "|| Write the final count table.                                               ||\n",
      "|| Write the read assignment summary.                                         ||\n",
      "||                                                                            ||\n",
      "|| Summary of counting results can be found in file \"/users/genomics/marta/T  ||\n",
      "|| estisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/human/featu  ||\n",
      "|| reCounts_gffcompare/gffcompare_stranded_featureCounts.txt.summary\"         ||\n",
      "||                                                                            ||\n",
      "\\\\============================================================================//\n",
      "\n"
     ]
    }
   ],
   "source": [
    "%%bash -s \"$users_dir\" \"$specie\" \"$prev_users_dir\"\n",
    "\n",
    "###PREPARING NEEDED DATA\n",
    "# PROJECT=$1\n",
    "# DIR=$2/$PROJECT\n",
    "# p=$3 #single-end or paired-end dataset\n",
    "p=\"single-end\"\n",
    "strand=\"secondstrand\"\n",
    "CLUSTERDIR=$1/$2/featureCounts_gffcompare\n",
    "mkdir $CLUSTERDIR\n",
    "\n",
    "# AnnotGTF=$1/$2/newReference_Resconstructed/gencode.v38.gffcompare.TestisLiverBrain.annotation.sorted.1transcript.sorted.gtf\n",
    "AnnotGTF=$1/$2/newReference_Resconstructed/gencode.v47.gffcompare.TestisLiverBrain.annotation.sorted.1transcript.sorted.NOchr.fixed.gtf\n",
    "\n",
    "BAMDIR=$1/$2/STAR\n",
    "\n",
    "module load Subread/2.0.3\n",
    "########################\n",
    "\n",
    "# countReadPairs may need to be removed in case of single-end reads\n",
    "\n",
    "if [ $p == \"paired-end\" ]; then\n",
    "    if [ $strand == \"firststrand\" ]; then\n",
    "        featureCounts -T 10 -p -s 2 -g transcript_id -O --countReadPairs -a $AnnotGTF -o ${CLUSTERDIR}/gffcompare_stranded_featureCounts.txt $BAMDIR/*Aligned.sortedByCoord.out.bam \n",
    "    elif [ $strand == \"secondstrand\" ]; then\n",
    "        featureCounts -T 10 -p -s 1 -g transcript_id -O --countReadPairs -a $AnnotGTF -o ${CLUSTERDIR}/gffcompare_stranded_featureCounts.txt $BAMDIR/*Aligned.sortedByCoord.out.bam \n",
    "    elif [ $strand == \"unstranded\" ]; then\n",
    "        featureCounts -T 10 -p -s 0 -g transcript_id -O --countReadPairs -a $AnnotGTF -o ${CLUSTERDIR}/gffcompare_stranded_featureCounts.txt $BAMDIR/*Aligned.sortedByCoord.out.bam \n",
    "    fi\n",
    "fi\n",
    "if [ $p == \"single-end\" ]; then\n",
    "    if [ $strand == \"firststrand\" ]; then\n",
    "        featureCounts -T 10 -s 2 -g transcript_id -O -a $AnnotGTF -o ${CLUSTERDIR}/gffcompare_stranded_featureCounts.txt $BAMDIR/*Aligned.sortedByCoord.out.bam \n",
    "    elif [ $strand == \"secondstrand\" ]; then\n",
    "        featureCounts -T 10 -s 1 -g transcript_id -O -a $AnnotGTF -o ${CLUSTERDIR}/gffcompare_stranded_featureCounts.txt $BAMDIR/*Aligned.sortedByCoord.out.bam \n",
    "    elif [ $strand == \"unstranded\" ]; then\n",
    "        featureCounts -T 10 -s 0 -g transcript_id -O -a $AnnotGTF -o ${CLUSTERDIR}/gffcompare_stranded_featureCounts.txt $BAMDIR/*Aligned.sortedByCoord.out.bam \n",
    "    fi\n",
    "fi\n",
    "\n"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## TPMs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [],
   "source": [
    "from rna_seq_normalization import Normalization as Norm"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "15\n"
     ]
    }
   ],
   "source": [
    "table_of_counts = pd.DataFrame()\n",
    "### modify headers\n",
    "\n",
    "df=pd.read_csv(os.path.join(users_dir,specie,\"featureCounts_gffcompare/gffcompare_stranded_featureCounts.txt\"), skiprows=1, sep=\"\\t\")\n",
    "df['Geneid']=df['Geneid'].str.split('.').str[0]\n",
    "filter_col = [col for col in df if col.startswith('/')]\n",
    "for col in filter_col:\n",
    "    new_col=col.split(\"Aligned\")[0]\n",
    "    new_col=new_col.split(\"/\")[-1]\n",
    "\n",
    "    df.rename(columns={col:new_col}, inplace=True)\n",
    "    \n",
    "cols_to_select = [\"Geneid\",\"Chr\",\"Start\",\"End\",\"Strand\",\"Length\"]\n",
    "\n",
    "print(len(df.columns))\n",
    "df.rename(columns={'Geneid':'gene_id'}, inplace=True)\n",
    "\n",
    "if table_of_counts.empty:\n",
    "    table_of_counts = df\n",
    "else:\n",
    "    table_of_counts = table_of_counts.merge(df, on=[\"gene_id\",\"Chr\",\"Start\",\"End\",\"Strand\",\"Length\"])\n",
    "\n",
    "table_of_counts.to_csv(os.path.join(users_dir,specie,\"featureCounts_gffcompare/table_of_counts.csv\"), index=None)\n",
    "\n",
    "# get TPMs\n",
    "length = table_of_counts['Length']\n",
    "genes = table_of_counts['gene_id']\n",
    "# we are only interested in the columns with counts\n",
    "counts = table_of_counts\n",
    "counts.drop([\"Chr\",\"Start\",\"End\",\"Strand\",\"Length\",\"gene_id\"],axis=1, inplace=True)\n",
    "# calculate TPMs\n",
    "tpm_df = Norm.tpm(counts, length)\n",
    "# add gene_names again\n",
    "tpms = pd.concat([genes,tpm_df], axis=1)\n",
    "tpms.to_csv(os.path.join(users_dir,specie,\"featureCounts_gffcompare/table_of_counts_TPMs.csv\"), index=None)\n",
    "# add length again\n",
    "tpms_length = pd.concat([genes,length,tpm_df], axis=1)\n",
    "tpms_length.to_csv(os.path.join(users_dir,specie,\"featureCounts_gffcompare/table_of_counts_TPMs_withLength.csv\"), index=None)"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Run RiboNovel with the same annotation"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Generate annotation file"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_1959167/4029716907.py:1: DtypeWarning: Columns (0) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  gtf=pd.read_csv(os.path.join(users_dir,specie,\"newReference_Resconstructed/gencode.v47.gffcompare.TestisLiverBrain.annotation.sorted.1transcript.sorted.NOchr.gtf\"), sep=\"\\t\", header=None)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>0</th>\n",
       "      <th>1</th>\n",
       "      <th>2</th>\n",
       "      <th>3</th>\n",
       "      <th>4</th>\n",
       "      <th>5</th>\n",
       "      <th>6</th>\n",
       "      <th>7</th>\n",
       "      <th>8</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>M</td>\n",
       "      <td>StringTie</td>\n",
       "      <td>transcript</td>\n",
       "      <td>1</td>\n",
       "      <td>577</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>.</td>\n",
       "      <td>transcript_id \"TCONS_00001163\"; gene_id \"XLOC_...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>M</td>\n",
       "      <td>ENSEMBL</td>\n",
       "      <td>transcript</td>\n",
       "      <td>577</td>\n",
       "      <td>647</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>gene_id \"ENSG00000210049.1\"; transcript_id \"EN...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>M</td>\n",
       "      <td>ENSEMBL</td>\n",
       "      <td>transcript</td>\n",
       "      <td>648</td>\n",
       "      <td>1601</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>gene_id \"ENSG00000211459.2\"; transcript_id \"EN...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>M</td>\n",
       "      <td>StringTie</td>\n",
       "      <td>transcript</td>\n",
       "      <td>683</td>\n",
       "      <td>1035</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>.</td>\n",
       "      <td>transcript_id \"TCONS_00001164\"; gene_id \"XLOC_...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>M</td>\n",
       "      <td>ENSEMBL</td>\n",
       "      <td>transcript</td>\n",
       "      <td>1602</td>\n",
       "      <td>1670</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>gene_id \"ENSG00000210077.1\"; transcript_id \"EN...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>782862</th>\n",
       "      <td>KI270751.1</td>\n",
       "      <td>HAVANA</td>\n",
       "      <td>transcript</td>\n",
       "      <td>34175</td>\n",
       "      <td>73722</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>.</td>\n",
       "      <td>gene_id \"ENSG00000303867.1\"; transcript_id \"EN...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>782865</th>\n",
       "      <td>KI270751.1</td>\n",
       "      <td>HAVANA</td>\n",
       "      <td>transcript</td>\n",
       "      <td>44958</td>\n",
       "      <td>53274</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>gene_id \"ENSG00000303902.1\"; transcript_id \"EN...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>782871</th>\n",
       "      <td>KI270751.1</td>\n",
       "      <td>HAVANA</td>\n",
       "      <td>transcript</td>\n",
       "      <td>133801</td>\n",
       "      <td>139253</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>.</td>\n",
       "      <td>gene_id \"ENSG00000306528.1\"; transcript_id \"EN...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>782875</th>\n",
       "      <td>KI270753.1</td>\n",
       "      <td>HAVANA</td>\n",
       "      <td>transcript</td>\n",
       "      <td>43135</td>\n",
       "      <td>44491</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>gene_id \"ENSG00000297844.1\"; transcript_id \"EN...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>782879</th>\n",
       "      <td>KI270755.1</td>\n",
       "      <td>HAVANA</td>\n",
       "      <td>transcript</td>\n",
       "      <td>26350</td>\n",
       "      <td>27723</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>gene_id \"ENSG00000309258.1\"; transcript_id \"EN...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>79623 rows × 9 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                 0          1           2       3       4  5  6  7  \\\n",
       "1                M  StringTie  transcript       1     577  .  -  .   \n",
       "3                M    ENSEMBL  transcript     577     647  .  +  .   \n",
       "7                M    ENSEMBL  transcript     648    1601  .  +  .   \n",
       "8                M  StringTie  transcript     683    1035  .  -  .   \n",
       "11               M    ENSEMBL  transcript    1602    1670  .  +  .   \n",
       "...            ...        ...         ...     ...     ... .. .. ..   \n",
       "782862  KI270751.1     HAVANA  transcript   34175   73722  .  -  .   \n",
       "782865  KI270751.1     HAVANA  transcript   44958   53274  .  +  .   \n",
       "782871  KI270751.1     HAVANA  transcript  133801  139253  .  -  .   \n",
       "782875  KI270753.1     HAVANA  transcript   43135   44491  .  +  .   \n",
       "782879  KI270755.1     HAVANA  transcript   26350   27723  .  +  .   \n",
       "\n",
       "                                                        8  \n",
       "1       transcript_id \"TCONS_00001163\"; gene_id \"XLOC_...  \n",
       "3       gene_id \"ENSG00000210049.1\"; transcript_id \"EN...  \n",
       "7       gene_id \"ENSG00000211459.2\"; transcript_id \"EN...  \n",
       "8       transcript_id \"TCONS_00001164\"; gene_id \"XLOC_...  \n",
       "11      gene_id \"ENSG00000210077.1\"; transcript_id \"EN...  \n",
       "...                                                   ...  \n",
       "782862  gene_id \"ENSG00000303867.1\"; transcript_id \"EN...  \n",
       "782865  gene_id \"ENSG00000303902.1\"; transcript_id \"EN...  \n",
       "782871  gene_id \"ENSG00000306528.1\"; transcript_id \"EN...  \n",
       "782875  gene_id \"ENSG00000297844.1\"; transcript_id \"EN...  \n",
       "782879  gene_id \"ENSG00000309258.1\"; transcript_id \"EN...  \n",
       "\n",
       "[79623 rows x 9 columns]"
      ]
     },
     "execution_count": 23,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gtf=pd.read_csv(os.path.join(users_dir,specie,\"newReference_Resconstructed/gencode.v47.gffcompare.TestisLiverBrain.annotation.sorted.1transcript.sorted.NOchr.gtf\"), sep=\"\\t\", header=None)\n",
    "## keep only transcripts\n",
    "gtf = gtf[gtf[2] == \"transcript\"]\n",
    "gtf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_1959167/2607630529.py:3: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  gtf_novels['transcript_id'] = gtf_novels[8].str.split(\";\", expand=True)[0]\n",
      "/tmp/ipykernel_1959167/2607630529.py:4: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  gtf_novels['transcript_id'] = gtf_novels['transcript_id'].str.split('\"', expand=True)[1]\n",
      "/tmp/ipykernel_1959167/2607630529.py:6: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  gtf_novels['gene_id'] = gtf_novels[8].str.split(\";\", expand=True)[1]\n",
      "/tmp/ipykernel_1959167/2607630529.py:7: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  gtf_novels['gene_id'] = gtf_novels['gene_id'].str.split('\"', expand=True)[1]\n",
      "/tmp/ipykernel_1959167/2607630529.py:9: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  gtf_novels['gene_name'] = gtf_novels['gene_id']\n",
      "/tmp/ipykernel_1959167/2607630529.py:10: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  gtf_novels['transcript_type'] = \"novel\"\n",
      "/tmp/ipykernel_1959167/2607630529.py:11: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  gtf_novels['gene_type'] = \"novel\"\n",
      "/tmp/ipykernel_1959167/2607630529.py:12: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  gtf_novels['chr'] = gtf_novels[0]\n",
      "/tmp/ipykernel_1959167/2607630529.py:15: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  annotation_novels.drop_duplicates(inplace=True)\n",
      "/tmp/ipykernel_1959167/2607630529.py:19: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  gtf_annotated['gene_id'] = gtf_annotated[8].str.split(\";\", expand=True)[0]\n",
      "/tmp/ipykernel_1959167/2607630529.py:20: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  gtf_annotated['gene_id'] = gtf_annotated['gene_id'].str.split('\"', expand=True)[1]\n",
      "/tmp/ipykernel_1959167/2607630529.py:22: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  gtf_annotated['transcript_id'] = gtf_annotated[8].str.split(\";\", expand=True)[1]\n",
      "/tmp/ipykernel_1959167/2607630529.py:23: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  gtf_annotated['transcript_id'] = gtf_annotated['transcript_id'].str.split('\"', expand=True)[1]\n",
      "/tmp/ipykernel_1959167/2607630529.py:25: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  gtf_annotated['gene_name'] = gtf_annotated[8].str.split(\"gene_name\", expand=True)[1]\n",
      "/tmp/ipykernel_1959167/2607630529.py:26: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  gtf_annotated['gene_name'] = gtf_annotated['gene_name'].str.split('\"', expand=True)[1]\n",
      "/tmp/ipykernel_1959167/2607630529.py:28: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  gtf_annotated['transcript_type'] = gtf_annotated[8].str.split(\"transcript_type\", expand=True)[1]\n",
      "/tmp/ipykernel_1959167/2607630529.py:29: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  gtf_annotated['transcript_type'] = gtf_annotated['transcript_type'].str.split('\"', expand=True)[1]\n",
      "/tmp/ipykernel_1959167/2607630529.py:31: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  gtf_annotated['gene_type'] = gtf_annotated[8].str.split(\"gene_type\", expand=True)[1]\n",
      "/tmp/ipykernel_1959167/2607630529.py:32: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  gtf_annotated['gene_type'] = gtf_annotated['gene_type'].str.split('\"', expand=True)[1]\n",
      "/tmp/ipykernel_1959167/2607630529.py:34: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  gtf_annotated['chr'] = gtf_annotated[0]\n",
      "/tmp/ipykernel_1959167/2607630529.py:37: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  annotation_annotated.drop_duplicates(inplace=True)\n",
      "/tmp/ipykernel_1959167/2607630529.py:49: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  annotation_selected_v['gene_id'] = annotation_selected_v['gene_id'].str.split(\".\", expand=True)[0]\n",
      "/tmp/ipykernel_1959167/2607630529.py:50: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  annotation_selected_v['transcript_id'] = annotation_selected_v['transcript_id'].str.split(\".\", expand=True)[0]\n"
     ]
    }
   ],
   "source": [
    "## novels\n",
    "gtf_novels = gtf[gtf[8].str.contains(\"TCONS\")]\n",
    "gtf_novels['transcript_id'] = gtf_novels[8].str.split(\";\", expand=True)[0]\n",
    "gtf_novels['transcript_id'] = gtf_novels['transcript_id'].str.split('\"', expand=True)[1]\n",
    "\n",
    "gtf_novels['gene_id'] = gtf_novels[8].str.split(\";\", expand=True)[1]\n",
    "gtf_novels['gene_id'] = gtf_novels['gene_id'].str.split('\"', expand=True)[1]\n",
    "\n",
    "gtf_novels['gene_name'] = gtf_novels['gene_id']\n",
    "gtf_novels['transcript_type'] = \"novel\"\n",
    "gtf_novels['gene_type'] = \"novel\"\n",
    "gtf_novels['chr'] = gtf_novels[0]\n",
    "\n",
    "annotation_novels = gtf_novels[['chr','transcript_id','gene_id','gene_name','transcript_type','gene_type']]\n",
    "annotation_novels.drop_duplicates(inplace=True)\n",
    "\n",
    "## remove novels\n",
    "gtf_annotated = gtf[~gtf[8].str.contains(\"TCONS\")]\n",
    "gtf_annotated['gene_id'] = gtf_annotated[8].str.split(\";\", expand=True)[0]\n",
    "gtf_annotated['gene_id'] = gtf_annotated['gene_id'].str.split('\"', expand=True)[1]\n",
    "\n",
    "gtf_annotated['transcript_id'] = gtf_annotated[8].str.split(\";\", expand=True)[1]\n",
    "gtf_annotated['transcript_id'] = gtf_annotated['transcript_id'].str.split('\"', expand=True)[1]\n",
    "\n",
    "gtf_annotated['gene_name'] = gtf_annotated[8].str.split(\"gene_name\", expand=True)[1]\n",
    "gtf_annotated['gene_name'] = gtf_annotated['gene_name'].str.split('\"', expand=True)[1]\n",
    "\n",
    "gtf_annotated['transcript_type'] = gtf_annotated[8].str.split(\"transcript_type\", expand=True)[1]\n",
    "gtf_annotated['transcript_type'] = gtf_annotated['transcript_type'].str.split('\"', expand=True)[1]\n",
    "\n",
    "gtf_annotated['gene_type'] = gtf_annotated[8].str.split(\"gene_type\", expand=True)[1]\n",
    "gtf_annotated['gene_type'] = gtf_annotated['gene_type'].str.split('\"', expand=True)[1]\n",
    "\n",
    "gtf_annotated['chr'] = gtf_annotated[0]\n",
    "\n",
    "annotation_annotated = gtf_annotated[['chr','transcript_id','gene_id','gene_name','transcript_type','gene_type']]\n",
    "annotation_annotated.drop_duplicates(inplace=True)\n",
    "\n",
    "annotation = pd.concat([annotation_annotated, annotation_novels])\n",
    "annotation.to_csv(os.path.join(users_dir,specie,\"newReference_Resconstructed/transID_geneID_isoforms_all.1to1.csv\"), index=None)\n",
    "\n",
    "###### lncRNA | processed pseudogene | novel | protein coding \n",
    "types=[\"lncRNA\",\"processed_pseudogene\",\"novel\",\"protein_coding\"]\n",
    "annotation_selected = annotation[annotation['transcript_type'].isin(types)]\n",
    "annotation_selected.to_csv(os.path.join(users_dir,specie,\"newReference_Resconstructed/transID_geneID_isoforms_selected_version.1to1.csv\"), index=None)\n",
    "\n",
    "## no version\n",
    "annotation_selected_v = annotation_selected\n",
    "annotation_selected_v['gene_id'] = annotation_selected_v['gene_id'].str.split(\".\", expand=True)[0]\n",
    "annotation_selected_v['transcript_id'] = annotation_selected_v['transcript_id'].str.split(\".\", expand=True)[0]\n",
    "annotation_selected_v.to_csv(os.path.join(users_dir,specie,\"newReference_Resconstructed/transID_geneID_isoforms_selected.1to1.csv\"), index=None)\n"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.10"
  },
  "orig_nbformat": 4
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
