{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "import os, re\n",
    "import pandas as pd\n",
    "\n",
    "IPdir = \"/users/genomics/marta/241205_Immunopeptidomics_HCC\"\n",
    "PREDdir = \"/users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/cancers/log2ratio3x/cancertypes/onlyStep1/netMHCpan/FrequentHLA-I\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "254 unique hits in TSTR genes with immunopeptidomics data (HCC + lung + melanoma, FDR5%)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sequence</th>\n",
       "      <th>accessions</th>\n",
       "      <th>num_samples_with_hit</th>\n",
       "      <th>study</th>\n",
       "      <th>orfID_TSTR_RiboSeq</th>\n",
       "      <th>protein_references</th>\n",
       "      <th>gene_name_x</th>\n",
       "      <th>gene_name_y</th>\n",
       "      <th>gene_type_x</th>\n",
       "      <th>gene_id</th>\n",
       "      <th>gene_id.1</th>\n",
       "      <th>n</th>\n",
       "      <th>chr</th>\n",
       "      <th>which_ctypes</th>\n",
       "      <th>orfID</th>\n",
       "      <th>source_literature</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>AEEPVTEAEML</td>\n",
       "      <td>sp|Q9UBF1|MAGC2_HUMAN</td>\n",
       "      <td>1</td>\n",
       "      <td>Melanoma_MHCquant_Camarena</td>\n",
       "      <td>NaN</td>\n",
       "      <td>unique</td>\n",
       "      <td>MAGEC2</td>\n",
       "      <td>MAGEC2</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>ENSG00000046774</td>\n",
       "      <td>ENSG00000046774</td>\n",
       "      <td>25.0</td>\n",
       "      <td>X</td>\n",
       "      <td>BLCA,BRCA,KIRC,LIHC,LUAD,LUSC,PRAD</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>AEMLGSVV</td>\n",
       "      <td>sp|P43357|MAGA3_HUMAN;sp|P43360|MAGA6_HUMAN;tr...</td>\n",
       "      <td>1</td>\n",
       "      <td>Melanoma_MHCquant_Camarena</td>\n",
       "      <td>NaN</td>\n",
       "      <td>non-unique</td>\n",
       "      <td>MAGEA3</td>\n",
       "      <td>MAGEA3</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>ENSG00000221867</td>\n",
       "      <td>ENSG00000221867</td>\n",
       "      <td>39.0</td>\n",
       "      <td>X</td>\n",
       "      <td>BLCA,BRCA,COAD,LIHC,LUAD,LUSC,PRAD</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>AETSKMKVL</td>\n",
       "      <td>sp|O15479|MAGB2_HUMAN;sp|O15481|MAGB4_HUMAN;sp...</td>\n",
       "      <td>1</td>\n",
       "      <td>Melanoma_MHCquant_Camarena</td>\n",
       "      <td>NaN</td>\n",
       "      <td>non-unique</td>\n",
       "      <td>MAGEB2</td>\n",
       "      <td>MAGEB2</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>ENSG00000099399</td>\n",
       "      <td>ENSG00000099399</td>\n",
       "      <td>19.0</td>\n",
       "      <td>X</td>\n",
       "      <td>BLCA,BRCA,COAD,KIRC,LIHC,LUAD,LUSC</td>\n",
       "      <td>ENST00000378988.5:X:+|4|1607:99:1059|canonical...</td>\n",
       "      <td>IEAtlas</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>AETSYVKVL</td>\n",
       "      <td>sp|P43355|MAGA1_HUMAN;sp|P43358|MAGA4_HUMAN;sp...</td>\n",
       "      <td>2</td>\n",
       "      <td>Melanoma_MHCquant_Camarena</td>\n",
       "      <td>NaN</td>\n",
       "      <td>non-unique</td>\n",
       "      <td>MAGEA1</td>\n",
       "      <td>MAGEA1</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>ENSG00000198681</td>\n",
       "      <td>ENSG00000198681</td>\n",
       "      <td>40.0</td>\n",
       "      <td>X</td>\n",
       "      <td>BLCA,BRCA,COAD,LIHC,LUAD,LUSC,PRAD</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>AFAPFPTQF</td>\n",
       "      <td>ENST00000636096.3:X:+|109|1990:1901:1970|dORF|...</td>\n",
       "      <td>1</td>\n",
       "      <td>Melanoma_MHCquant_Camarena</td>\n",
       "      <td>NaN</td>\n",
       "      <td>non-unique</td>\n",
       "      <td>ENSG00000283599</td>\n",
       "      <td>BX276092.9</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>ENSG00000283599</td>\n",
       "      <td>ENSG00000283599</td>\n",
       "      <td>13.0</td>\n",
       "      <td>X</td>\n",
       "      <td>BLCA,BRCA,COAD,LIHC,LUAD,LUSC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>293</th>\n",
       "      <td>YNDEDIYLF</td>\n",
       "      <td>sp|Q5TAQ9|DCAF8_HUMAN;sp|A6NGE4|DC8L1_HUMAN</td>\n",
       "      <td>1</td>\n",
       "      <td>Melanoma_MHCquant_Camarena</td>\n",
       "      <td>NaN</td>\n",
       "      <td>non-unique</td>\n",
       "      <td>DCAF8L1</td>\n",
       "      <td>DCAF8L1</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>ENSG00000226372</td>\n",
       "      <td>ENSG00000226372</td>\n",
       "      <td>4.0</td>\n",
       "      <td>X</td>\n",
       "      <td>BLCA,LIHC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>294</th>\n",
       "      <td>YNDEDIYLF</td>\n",
       "      <td>sp|Q5TAQ9|DCAF8_HUMAN;sp|A6NGE4|DC8L1_HUMAN</td>\n",
       "      <td>1</td>\n",
       "      <td>Lung_MHCquant_Camarena</td>\n",
       "      <td>NaN</td>\n",
       "      <td>non-unique</td>\n",
       "      <td>DCAF8L1</td>\n",
       "      <td>DCAF8L1</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>ENSG00000226372</td>\n",
       "      <td>ENSG00000226372</td>\n",
       "      <td>4.0</td>\n",
       "      <td>X</td>\n",
       "      <td>BLCA,LIHC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>295</th>\n",
       "      <td>YNDEDIYLF</td>\n",
       "      <td>sp|Q5TAQ9|DCAF8_HUMAN;sp|A6NGE4|DC8L1_HUMAN</td>\n",
       "      <td>1</td>\n",
       "      <td>HCC_MHCquant_Camarena</td>\n",
       "      <td>NaN</td>\n",
       "      <td>non-unique</td>\n",
       "      <td>DCAF8L1</td>\n",
       "      <td>DCAF8L1</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>ENSG00000226372</td>\n",
       "      <td>ENSG00000226372</td>\n",
       "      <td>4.0</td>\n",
       "      <td>X</td>\n",
       "      <td>BLCA,LIHC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>296</th>\n",
       "      <td>YPPLHEWVL</td>\n",
       "      <td>sp|P43357|MAGA3_HUMAN</td>\n",
       "      <td>1</td>\n",
       "      <td>Melanoma_MHCquant_Camarena</td>\n",
       "      <td>NaN</td>\n",
       "      <td>unique</td>\n",
       "      <td>MAGEA3</td>\n",
       "      <td>MAGEA3</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>ENSG00000221867</td>\n",
       "      <td>ENSG00000221867</td>\n",
       "      <td>39.0</td>\n",
       "      <td>X</td>\n",
       "      <td>BLCA,BRCA,COAD,LIHC,LUAD,LUSC,PRAD</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>297</th>\n",
       "      <td>YSEKISYVY</td>\n",
       "      <td>sp|Q16384|SSX1_HUMAN;sp|Q7RTT4|SSX8_HUMAN;ENST...</td>\n",
       "      <td>1</td>\n",
       "      <td>HCC_MHCquant_Camarena</td>\n",
       "      <td>NaN</td>\n",
       "      <td>non-unique</td>\n",
       "      <td>SSX1</td>\n",
       "      <td>SSX1</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>ENSG00000126752</td>\n",
       "      <td>ENSG00000126752</td>\n",
       "      <td>38.0</td>\n",
       "      <td>X</td>\n",
       "      <td>BLCA,BRCA,LIHC,LUAD,PRAD</td>\n",
       "      <td>ENST00000376919.4:X:+|3|1223:62:629|canonical|ATG</td>\n",
       "      <td>deBeijer</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>298 rows Ã— 16 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        sequence                                         accessions  \\\n",
       "0    AEEPVTEAEML                              sp|Q9UBF1|MAGC2_HUMAN   \n",
       "1       AEMLGSVV  sp|P43357|MAGA3_HUMAN;sp|P43360|MAGA6_HUMAN;tr...   \n",
       "2      AETSKMKVL  sp|O15479|MAGB2_HUMAN;sp|O15481|MAGB4_HUMAN;sp...   \n",
       "3      AETSYVKVL  sp|P43355|MAGA1_HUMAN;sp|P43358|MAGA4_HUMAN;sp...   \n",
       "4      AFAPFPTQF  ENST00000636096.3:X:+|109|1990:1901:1970|dORF|...   \n",
       "..           ...                                                ...   \n",
       "293    YNDEDIYLF        sp|Q5TAQ9|DCAF8_HUMAN;sp|A6NGE4|DC8L1_HUMAN   \n",
       "294    YNDEDIYLF        sp|Q5TAQ9|DCAF8_HUMAN;sp|A6NGE4|DC8L1_HUMAN   \n",
       "295    YNDEDIYLF        sp|Q5TAQ9|DCAF8_HUMAN;sp|A6NGE4|DC8L1_HUMAN   \n",
       "296    YPPLHEWVL                              sp|P43357|MAGA3_HUMAN   \n",
       "297    YSEKISYVY  sp|Q16384|SSX1_HUMAN;sp|Q7RTT4|SSX8_HUMAN;ENST...   \n",
       "\n",
       "     num_samples_with_hit                       study orfID_TSTR_RiboSeq  \\\n",
       "0                       1  Melanoma_MHCquant_Camarena                NaN   \n",
       "1                       1  Melanoma_MHCquant_Camarena                NaN   \n",
       "2                       1  Melanoma_MHCquant_Camarena                NaN   \n",
       "3                       2  Melanoma_MHCquant_Camarena                NaN   \n",
       "4                       1  Melanoma_MHCquant_Camarena                NaN   \n",
       "..                    ...                         ...                ...   \n",
       "293                     1  Melanoma_MHCquant_Camarena                NaN   \n",
       "294                     1      Lung_MHCquant_Camarena                NaN   \n",
       "295                     1       HCC_MHCquant_Camarena                NaN   \n",
       "296                     1  Melanoma_MHCquant_Camarena                NaN   \n",
       "297                     1       HCC_MHCquant_Camarena                NaN   \n",
       "\n",
       "    protein_references      gene_name_x gene_name_y     gene_type_x  \\\n",
       "0               unique           MAGEC2      MAGEC2  protein_coding   \n",
       "1           non-unique           MAGEA3      MAGEA3  protein_coding   \n",
       "2           non-unique           MAGEB2      MAGEB2  protein_coding   \n",
       "3           non-unique           MAGEA1      MAGEA1  protein_coding   \n",
       "4           non-unique  ENSG00000283599  BX276092.9  protein_coding   \n",
       "..                 ...              ...         ...             ...   \n",
       "293         non-unique          DCAF8L1     DCAF8L1  protein_coding   \n",
       "294         non-unique          DCAF8L1     DCAF8L1  protein_coding   \n",
       "295         non-unique          DCAF8L1     DCAF8L1  protein_coding   \n",
       "296             unique           MAGEA3      MAGEA3  protein_coding   \n",
       "297         non-unique             SSX1        SSX1  protein_coding   \n",
       "\n",
       "             gene_id        gene_id.1     n chr  \\\n",
       "0    ENSG00000046774  ENSG00000046774  25.0   X   \n",
       "1    ENSG00000221867  ENSG00000221867  39.0   X   \n",
       "2    ENSG00000099399  ENSG00000099399  19.0   X   \n",
       "3    ENSG00000198681  ENSG00000198681  40.0   X   \n",
       "4    ENSG00000283599  ENSG00000283599  13.0   X   \n",
       "..               ...              ...   ...  ..   \n",
       "293  ENSG00000226372  ENSG00000226372   4.0   X   \n",
       "294  ENSG00000226372  ENSG00000226372   4.0   X   \n",
       "295  ENSG00000226372  ENSG00000226372   4.0   X   \n",
       "296  ENSG00000221867  ENSG00000221867  39.0   X   \n",
       "297  ENSG00000126752  ENSG00000126752  38.0   X   \n",
       "\n",
       "                           which_ctypes  \\\n",
       "0    BLCA,BRCA,KIRC,LIHC,LUAD,LUSC,PRAD   \n",
       "1    BLCA,BRCA,COAD,LIHC,LUAD,LUSC,PRAD   \n",
       "2    BLCA,BRCA,COAD,KIRC,LIHC,LUAD,LUSC   \n",
       "3    BLCA,BRCA,COAD,LIHC,LUAD,LUSC,PRAD   \n",
       "4         BLCA,BRCA,COAD,LIHC,LUAD,LUSC   \n",
       "..                                  ...   \n",
       "293                           BLCA,LIHC   \n",
       "294                           BLCA,LIHC   \n",
       "295                           BLCA,LIHC   \n",
       "296  BLCA,BRCA,COAD,LIHC,LUAD,LUSC,PRAD   \n",
       "297            BLCA,BRCA,LIHC,LUAD,PRAD   \n",
       "\n",
       "                                                 orfID source_literature  \n",
       "0                                                  NaN               NaN  \n",
       "1                                                  NaN               NaN  \n",
       "2    ENST00000378988.5:X:+|4|1607:99:1059|canonical...           IEAtlas  \n",
       "3                                                  NaN               NaN  \n",
       "4                                                  NaN               NaN  \n",
       "..                                                 ...               ...  \n",
       "293                                                NaN               NaN  \n",
       "294                                                NaN               NaN  \n",
       "295                                                NaN               NaN  \n",
       "296                                                NaN               NaN  \n",
       "297  ENST00000376919.4:X:+|3|1223:62:629|canonical|ATG          deBeijer  \n",
       "\n",
       "[298 rows x 16 columns]"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "immunopeptidomics_results = pd.read_csv(os.path.join(IPdir,\"results/merged/merged_results_HCC_LUNG_Melanoma.csv\"))\n",
    "\n",
    "print(len(set(immunopeptidomics_results.sequence.values.tolist())),\"unique hits in TSTR genes with immunopeptidomics data (HCC + lung + melanoma, FDR5%)\")\n",
    "immunopeptidomics_results\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## WB / SB"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "CTnonx\n",
      "17910 predicted binders in  CTnonx TSTR genes for any common HLA-I allele\n",
      "28 overlapping of hits CTnonx TSTR genes for any common HLA-I allele\n",
      "altORFs\n",
      "555 predicted binders in  altORFs TSTR genes for any common HLA-I allele\n",
      "0 overlapping of hits altORFs TSTR genes for any common HLA-I allele\n",
      "CTx\n",
      "4942 predicted binders in  CTx TSTR genes for any common HLA-I allele\n",
      "89 overlapping of hits CTx TSTR genes for any common HLA-I allele\n",
      "Noncodingnonx\n",
      "1759 predicted binders in  Noncodingnonx TSTR genes for any common HLA-I allele\n",
      "2 overlapping of hits Noncodingnonx TSTR genes for any common HLA-I allele\n"
     ]
    }
   ],
   "source": [
    "## exact hit\n",
    "for file in os.listdir(PREDdir):\n",
    "    if file.endswith(\"WBSB.out\"):\n",
    "        print(file.split(\".\")[0])\n",
    "        WB = pd.read_csv(os.path.join(PREDdir,file), header=None, sep='\\s+')\n",
    "        WB.drop([0,3,9,4,5,6,7,8,16,17], axis=1, inplace=True)\n",
    "        WB.columns = [\"MHC\",\"Peptide\",\"Identity\",\"Score_EL\",\"%Rank_EL\",\"Score_BA\",\"%Rank_BA\",\"Aff(nM)\"]\n",
    "        print(len(set(WB.Peptide.values.tolist())),\"predicted binders in \",file.split(\".\")[0],\"TSTR genes for any common HLA-I allele\")\n",
    "        \n",
    "        # Filter immunopeptidomics_results to keep rows where sequence is the same as WB peptide\n",
    "        WB_IP = WB[['Peptide']].merge(immunopeptidomics_results, left_on=\"Peptide\", right_on=\"sequence\")\n",
    "        print(len(set(WB_IP.Peptide.values.tolist())),\"overlapping of hits\",file.split(\".\")[0],\"TSTR genes for any common HLA-I allele\")\n",
    "        # print(WB_IP.drop_duplicates())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "CTnonx\n",
      "17910 predicted binders in  CTnonx TSTR genes for any common HLA-I allele\n",
      "35 overlapping of hits CTnonx TSTR genes for any common HLA-I allele\n",
      "altORFs\n",
      "555 predicted binders in  altORFs TSTR genes for any common HLA-I allele\n",
      "1 overlapping of hits altORFs TSTR genes for any common HLA-I allele\n",
      "CTx\n",
      "4942 predicted binders in  CTx TSTR genes for any common HLA-I allele\n",
      "119 overlapping of hits CTx TSTR genes for any common HLA-I allele\n",
      "Noncodingnonx\n",
      "1759 predicted binders in  Noncodingnonx TSTR genes for any common HLA-I allele\n",
      "4 overlapping of hits Noncodingnonx TSTR genes for any common HLA-I allele\n"
     ]
    }
   ],
   "source": [
    "## WB contained in IP hit\n",
    "for file in os.listdir(PREDdir):\n",
    "    if file.endswith(\"WBSB.out\"):\n",
    "        print(file.split(\".\")[0])\n",
    "        WB = pd.read_csv(os.path.join(PREDdir,file), header=None, sep='\\s+')\n",
    "        WB.drop([0,3,9,4,5,6,7,8,16,17], axis=1, inplace=True)\n",
    "        WB.columns = [\"MHC\",\"Peptide\",\"Identity\",\"Score_EL\",\"%Rank_EL\",\"Score_BA\",\"%Rank_BA\",\"Aff(nM)\"]\n",
    "        print(len(set(WB.Peptide.values.tolist())),\"predicted binders in \",file.split(\".\")[0],\"TSTR genes for any common HLA-I allele\")\n",
    "\n",
    "        # Filter immunopeptidomics_results to keep rows where sequence is the same as or contains a WB peptide\n",
    "        WB_peptides = WB.Peptide.values.tolist()\n",
    "        WB_IP = immunopeptidomics_results[\n",
    "                    immunopeptidomics_results['sequence'].apply(lambda seq: any(peptide in seq for peptide in WB_peptides))]     \n",
    "           \n",
    "        print(len(set(WB_IP.sequence.values.tolist())),\"overlapping of hits\",file.split(\".\")[0],\"TSTR genes for any common HLA-I allele\")\n",
    "        # print(WB_IP.drop_duplicates())"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## SB"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "CTnonx\n",
      "7421 predicted strong binders in  CTnonx TSTR genes for any common HLA-I allele\n",
      "15 overlapping of hits CTnonx TSTR genes for any common HLA-I allele\n",
      "Noncodingnonx\n",
      "715 predicted strong binders in  Noncodingnonx TSTR genes for any common HLA-I allele\n",
      "1 overlapping of hits Noncodingnonx TSTR genes for any common HLA-I allele\n",
      "CTx\n",
      "2015 predicted strong binders in  CTx TSTR genes for any common HLA-I allele\n",
      "70 overlapping of hits CTx TSTR genes for any common HLA-I allele\n",
      "altORFs\n",
      "221 predicted strong binders in  altORFs TSTR genes for any common HLA-I allele\n",
      "0 overlapping of hits altORFs TSTR genes for any common HLA-I allele\n"
     ]
    }
   ],
   "source": [
    "## exact hit\n",
    "for file in os.listdir(PREDdir):\n",
    "    if file.endswith(\".SB.out\"):\n",
    "        print(file.split(\".\")[0])\n",
    "        SB = pd.read_csv(os.path.join(PREDdir,file), header=None, sep='\\s+')\n",
    "        SB.drop([0,3,9,4,5,6,7,8,16,17], axis=1, inplace=True)\n",
    "        SB.columns = [\"MHC\",\"Peptide\",\"Identity\",\"Score_EL\",\"%Rank_EL\",\"Score_BA\",\"%Rank_BA\",\"Aff(nM)\"]\n",
    "        print(len(set(SB.Peptide.values.tolist())),\"predicted strong binders in \",file.split(\".\")[0],\"TSTR genes for any common HLA-I allele\")\n",
    "        \n",
    "        # Filter immunopeptidomics_results to keep rows where sequence is the same as WB peptide\n",
    "        SB_IP = SB[['Peptide']].merge(immunopeptidomics_results, left_on=\"Peptide\", right_on=\"sequence\")\n",
    "        print(len(set(SB_IP.Peptide.values.tolist())),\"overlapping of hits\",file.split(\".\")[0],\"TSTR genes for any common HLA-I allele\")\n",
    "        # print(SB_IP.drop_duplicates())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "CTnonx\n",
      "7421 predicted strong binders in  CTnonx TSTR genes for any common HLA-I allele\n",
      "18 overlapping of hits CTnonx TSTR genes for any common HLA-I allele\n",
      "Noncodingnonx\n",
      "715 predicted strong binders in  Noncodingnonx TSTR genes for any common HLA-I allele\n",
      "2 overlapping of hits Noncodingnonx TSTR genes for any common HLA-I allele\n",
      "CTx\n",
      "2015 predicted strong binders in  CTx TSTR genes for any common HLA-I allele\n",
      "93 overlapping of hits CTx TSTR genes for any common HLA-I allele\n",
      "altORFs\n",
      "221 predicted strong binders in  altORFs TSTR genes for any common HLA-I allele\n",
      "1 overlapping of hits altORFs TSTR genes for any common HLA-I allele\n"
     ]
    }
   ],
   "source": [
    "## WB contained in IP hit\n",
    "for file in os.listdir(PREDdir):\n",
    "    if file.endswith(\".SB.out\"):\n",
    "        print(file.split(\".\")[0])\n",
    "        SB = pd.read_csv(os.path.join(PREDdir,file), header=None, sep='\\s+')\n",
    "        SB.drop([0,3,9,4,5,6,7,8,16,17], axis=1, inplace=True)\n",
    "        SB.columns = [\"MHC\",\"Peptide\",\"Identity\",\"Score_EL\",\"%Rank_EL\",\"Score_BA\",\"%Rank_BA\",\"Aff(nM)\"]\n",
    "        print(len(set(SB.Peptide.values.tolist())),\"predicted strong binders in \",file.split(\".\")[0],\"TSTR genes for any common HLA-I allele\")\n",
    "\n",
    "        # Filter immunopeptidomics_results to keep rows where sequence is the same as or contains a WB peptide\n",
    "        SB_peptides = SB.Peptide.values.tolist()\n",
    "        SB_IP = immunopeptidomics_results[\n",
    "                    immunopeptidomics_results['sequence'].apply(lambda seq: any(peptide in seq for peptide in SB_peptides))]     \n",
    "           \n",
    "        print(len(set(SB_IP.sequence.values.tolist())),\"overlapping of hits\",file.split(\".\")[0],\"TSTR genes for any common HLA-I allele\")\n",
    "        # print(SB_IP.drop_duplicates())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.10"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
