---
title: "Q1. ncORFs in testis"
author: "Marta Espinosa"
date: "2024-05-13"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
library(dplyr)
library(ggplot2)
library(stringr)
library(purrr)
library(ggvenn)
library(ggpubr)
library(ggside)
library(bioseq)
library(rcartocolor)
library(ggbeeswarm)
library(ggbreak)

plots_wd = "/projects_eg/projects/marta/TestisRestricted_Microproteins_TSA/with_TranscriptomeReconstruction/Multimap_altORFs/Q1_TestisORFs/human/plots"
save_wd = "/projects_eg/projects/marta/TestisRestricted_Microproteins_TSA/with_TranscriptomeReconstruction/Multimap_altORFs/Q1_TestisORFs/human"
# riboseq_dir = "/users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/human/240724_RiboNovel"
riboseq_dir = "/users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/human/RiboNovel_MultMap_1to1"

## annotatioin
annot = read.csv("/users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/human/newReference_Resconstructed/transID_geneID_isoforms_selected.1to1.csv")
```

## Evidence of translation in testis according to the Ribo-Seq data in Wang et al. (2020) 

- We keep representative ORF if several are detected (check ribORF paper for selection of representative)

- pvalue >= 0.6

- minimum length of 33 nucleotides, including stop

- Both canonical ORFs and alternative ORFs such as dORFs, uORFS, doORFs, uoORFs are maintained for protein coding genes

- Only reproducible results on translated ORFs are expected, so we only keep ORFs that are translated in 1 out of 3 testis samples

```{r riboNovel_input, echo=F}
##-------------------------- SAMPLE 1 --------------------------##
ribORF_humanTestis_1 = read.table(file.path(riboseq_dir,"RibORF/human_testis_ribo_1_r1/pred.pvalue.parameters.txt"), header=T)

repre_ribORF_humanTestis_1 = read.table(file.path(riboseq_dir,"RibORF/human_testis_ribo_1_r1/repre.valid.ORF.genepred.txt"), header=F)

ribORF_humanTestis_1 = ribORF_humanTestis_1 %>% subset(orfID %in% repre_ribORF_humanTestis_1$V1)
ribORF_humanTestis_1$sample = "Testis1"

## get transcriptID
ribORF_humanTestis_1$transcript_id = gsub(":.*","", ribORF_humanTestis_1$orfID)
ribORF_humanTestis_1$transcript_id = gsub("\\..*","", ribORF_humanTestis_1$transcript_id)
nrow(ribORF_humanTestis_1)
 
##-------------------------- SAMPLE 2 --------------------------##
ribORF_humanTestis_2 = read.table(file.path(riboseq_dir,"RibORF/human_testis_ribo_2_r1/pred.pvalue.parameters.txt"), header=T)

repre_ribORF_humanTestis_2 =  read.table(file.path(riboseq_dir,"RibORF/human_testis_ribo_2_r1/repre.valid.ORF.genepred.txt"), header=F)

ribORF_humanTestis_2 = ribORF_humanTestis_2 %>% subset(orfID %in% repre_ribORF_humanTestis_2$V1)
ribORF_humanTestis_2$sample = "Testis2"
## get transcriptID
ribORF_humanTestis_2$transcript_id = gsub(":.*","", ribORF_humanTestis_2$orfID)
ribORF_humanTestis_2$transcript_id = gsub("\\..*","", ribORF_humanTestis_2$transcript_id)

nrow(ribORF_humanTestis_2)

##-------------------------- SAMPLE 3 --------------------------##
ribORF_humanTestis_3 = read.table(file.path(riboseq_dir,"RibORF/human_testis_ribo_3_r1/pred.pvalue.parameters.txt"), header=T)

repre_ribORF_humanTestis_3 =  read.table(file.path(riboseq_dir,"RibORF/human_testis_ribo_3_r1/repre.valid.ORF.genepred.txt"), header=F)

ribORF_humanTestis_3 = ribORF_humanTestis_3 %>% subset(orfID %in% repre_ribORF_humanTestis_3$V1)
ribORF_humanTestis_3$sample = "Testis3"
## get transcriptID
ribORF_humanTestis_3$transcript_id = gsub(":.*","", ribORF_humanTestis_3$orfID)
ribORF_humanTestis_3$transcript_id = gsub("\\..*","", ribORF_humanTestis_3$transcript_id)
nrow(ribORF_humanTestis_3)

### Concatenate ncORFs for all 2 samples
ribORF_humanTestis = rbind(ribORF_humanTestis_3, ribORF_humanTestis_2, ribORF_humanTestis_1)

ggvenn(list("Sample1"=ribORF_humanTestis_1$orfID,
                 "Sample2"=ribORF_humanTestis_2$orfID,
                 "Sample3"=ribORF_humanTestis_3$orfID),
       fill_alpha=0.5, stroke_color="gray", set_name_size = 4, text_size=3, fill_color = c("#7876B1"," red","#91D1C2")) +
  ggtitle("Testis-expressed ORFs")
ggsave(paste0(plots_wd,"/PNG/venn_TestisExpressedORFs.png"))
ggsave(paste0(plots_wd,"/PDF/venn_TestisExpressedORFs.pdf"))

### only novels
ggvenn(list("Sample1"=ribORF_humanTestis_1 %>% subset(grepl("TCONS",transcript_id)) %>% pull(orfID),
                 "Sample2"=ribORF_humanTestis_2%>% subset(grepl("TCONS",transcript_id)) %>% pull(orfID),
                 "Sample3"=ribORF_humanTestis_3%>% subset(grepl("TCONS",transcript_id)) %>% pull(orfID)),
       fill_alpha=0.5, stroke_color="gray", set_name_size = 4, text_size=3, fill_color = c("#7876B1","red","#91D1C2")) +
  ggtitle("Novel Testis-expressed ORFs")
ggsave(paste0(plots_wd,"/PNG/venn_NovelTestisExpressedORFs.png"))
ggsave(paste0(plots_wd,"/PDF/venn_NovelTestisExpressedORFs.pdf"))
```


```{r ORFs, echo=F}
## get gene_id, gene_name, gene_type
ribORF_humanTestis_Annotated = merge(annot, ribORF_humanTestis, by="transcript_id")
# nrow(ribORF_humanTestis_1_Annotated)
## get orfType
ribORF_humanTestis_Annotated$orfType = sapply(strsplit(ribORF_humanTestis_Annotated$orfID, "\\|"), function(x) x[4])

## how many orfTypes per gene_type?
ribORF_humanTestis_Annotated %>% subset(gene_type == "lncRNA" | gene_type == "processed_pseudogene" | gene_type == "protein_coding" | gene_type == "novel") %>% group_by(gene_type, orfType) %>% count()

## for protein-coding we only keep the canonical
ribORF_humanTestis_Annotated_biotypes = ribORF_humanTestis_Annotated %>%
  subset(gene_type == "lncRNA" | gene_type == "processed_pseudogene" | gene_type == "protein_coding" | gene_type == "novel")
table(ribORF_humanTestis_Annotated_biotypes$orfType, ribORF_humanTestis_Annotated_biotypes$gene_type)

## understanding the protein coding ORFs
PCG_ORFs = ribORF_humanTestis_Annotated_biotypes %>% subset(gene_type == "protein_coding") %>% select(transcript_id, gene_id, gene_type, gene_name, orfID, orfType) %>% unique()
table(PCG_ORFs$orfType)

ribORF_humanTestis_Annotated_biotypes = ribORF_humanTestis_Annotated_biotypes %>% 
  subset(orfType == "uORF" | orfType == "dORF" | orfType == "ouORF" | orfType == "odORF" | orfType == "noncoding" | orfType == "canonical")


for(g in 1:length(unique(ribORF_humanTestis_Annotated_biotypes$gene_type))){
  temp_to_plot = ribORF_humanTestis_Annotated_biotypes %>% subset(gene_type == unique(ribORF_humanTestis_Annotated_biotypes$gene_type)[g])
  ggvenn(list("Sample1"=temp_to_plot %>% subset(sample == "Testis1") %>% pull(orfID),
                 "Sample2"=temp_to_plot %>% subset(sample == "Testis2") %>% pull(orfID),
                 "Sample3"=temp_to_plot %>% subset(sample == "Testis3") %>% pull(orfID)),
       fill_alpha=0.5, stroke_color="gray", set_name_size = 4, text_size=3, fill_color = c("#7876B1","red","#91D1C2")) +
  ggtitle(paste0("Testis-expressed", unique(ribORF_humanTestis_Annotated_biotypes$gene_type)[g]," ORFs"))
  ggsave(paste0(plots_wd,"/PNG/venn_TestisExpressedORFs_",unique(ribORF_humanTestis_Annotated_biotypes$gene_type)[g],".png"))
  ggsave(paste0(plots_wd,"/PDF/venn_TestisExpressedORFs_",unique(ribORF_humanTestis_Annotated_biotypes$gene_type)[g],".pdf"))
}
```

#### How many ORFs are translated in at least 1 out of the 3 testis samples?

ncORFs from lncRNA and processed pseudogenes + canonical ORFs.

```{r translated_in_1, echo=F}
## of the selected, select those that are translated in at least 2 out of 3 testis samples
translated_in_1 = ribORF_humanTestis_Annotated_biotypes %>%
  select(orfID,gene_id,gene_type,sample, orfType) %>%
  group_by(orfID,gene_id,gene_type,orfType) %>%
  count() %>% 
  subset(n>=1) %>% unique()

table(translated_in_1$gene_type)
table(translated_in_1$gene_type, translated_in_1$orfType)

translated_in_1_genes = ribORF_humanTestis_Annotated_biotypes %>%
  select(gene_id,gene_type,sample, orfType) %>%
  group_by(gene_id,gene_type,orfType) %>%
  count() %>% 
  subset(n>=1) %>% unique()

table(translated_in_1_genes$gene_type)
table(translated_in_1_genes$gene_type, translated_in_1_genes$orfType)

ribORF_humanTestis_in1 = ribORF_humanTestis_Annotated_biotypes %>% subset(orfID %in% translated_in_1$orfID) 
write.csv(ribORF_humanTestis_in1, file.path(save_wd,"ribORF_humanTestis_translated_in_1.fulltable.csv"), row.names = F)

ribORF_humanTestis_in1 = ribORF_humanTestis_in1 %>% select(orfID, gene_id, gene_name, transcript_id, gene_type, orfType, length) %>% unique()
write.csv(ribORF_humanTestis_in1, file.path(save_wd,"ribORF_humanTestis_translated_in_1.simplified.csv"), row.names = F)

ORFs = as.data.frame(table(ribORF_humanTestis_in1$gene_type))
names(ORFs) = c("gene_type","num_ORFs")

## that correspond to:
ribORF_humanTestis_in1_genes = ribORF_humanTestis_in1 %>% select(gene_id, gene_type) %>% unique()
genes = as.data.frame(table(ribORF_humanTestis_in1_genes$gene_type))
names(genes) = c("gene_type","num_genes_w_ORFs")

ORFs_in_genes = merge(ORFs, genes, by="gene_type")
ORFs_in_genes$ratio = ORFs_in_genes$num_ORFs / ORFs_in_genes$num_genes_w_ORFs
ORFs_in_genes

### geneORFtype
ribORF_humanTestis_in1 = ribORF_humanTestis_in1 %>% mutate(geneORFtype = paste0(gene_type,"_",orfType))
```

```{r ORFs_per_transcript, echo=F}
num_ORFs_per_transcript = ribORF_humanTestis_in1 %>% 
  group_by(transcript_id, geneORFtype) %>% 
  count()
names(num_ORFs_per_transcript)[3] = "n_ORFs"
```

```{r ORFs_per_biotype, echo=F}
num_ORFs_per_biotype = num_ORFs_per_transcript %>% group_by(geneORFtype, n_ORFs) %>% count()
num_ORFs_per_biotype %>% head

num_ORFs_per_biotype = num_ORFs_per_biotype %>% mutate(num_ORFs_modif = case_when(n_ORFs >= 10 ~ "morethan10", TRUE ~ as.character(n_ORFs)))
ggplot(num_ORFs_per_biotype, aes(x=num_ORFs_modif, y=n)) +
  geom_bar(stat="identity") +
  scale_y_continuous(trans="log10") +
  # scale_fill_manual(values=c("protein_coding"="#CC79A7",
  #                          "lncRNA" = "#009E73",
  #                          "processed_pseudogene" = "#0090B2",
  #                          "novel" = "#E69F00")) +
  labs(title="Number of ORFs per transcript",
       fill="Biotype") +
  theme_classic() +
  theme(legend.position = "top") +
  facet_wrap(~ geneORFtype, nrow=2)
ggsave(file.path(plots_wd,"PNG/nORFs_per_transcript.png"), height=4.45, width=8.24)
ggsave(file.path(plots_wd,"PDF/nORFs_per_transcript.pdf"), height=4.45, width=8.24)
```

```{r altORFs_per_transcript}
num_altORFs_per_biotype = num_ORFs_per_transcript %>% subset(geneORFtype == "protein_coding_dORF" | geneORFtype == "protein_coding_uORF" | geneORFtype == "protein_coding_ouORF" | geneORFtype == "protein_coding_odORF") %>% group_by(geneORFtype, n_ORFs) %>% count()
num_altORFs_per_biotype %>% head

num_altORFs_per_biotype = num_altORFs_per_biotype %>% mutate(num_ORFs_modif = case_when(n_ORFs >= 10 ~ "morethan10", TRUE ~ as.character(n_ORFs)))
ggplot(num_altORFs_per_biotype, aes(x=num_ORFs_modif, y=n)) +
  geom_bar(stat="identity") +
  scale_y_continuous(trans="log10") +
  # scale_fill_manual(values=c("protein_coding"="#CC79A7",
  #                          "lncRNA" = "#009E73",
  #                          "processed_pseudogene" = "#0090B2",
  #                          "novel" = "#E69F00")) +
  labs(title="Number of alternative ORFs per transcript",
       fill="Biotype") +
  theme_classic() +
  theme(legend.position = "top") +
  facet_wrap(~ geneORFtype, nrow=2)
ggsave(file.path(plots_wd,"PNG/alt_nORFs_per_transcript.png"), height=4.45, width=8.24)
ggsave(file.path(plots_wd,"PDF/alt_nORFs_per_transcript.pdf"), height=4.45, width=8.24)
```

Questions: How many ncORFs in lncRNAs/processed pseudogenes are translated in testis? Which is their length, composition and translation level in comparison to canonical ORFs? 

#### Analysis translated ncORFs with respect to canonical ORFs

**Length**

```{r length, echo=F}
# my_comparisons = list(c("lncRNA","processed_pseudogene"), c("protein_coding","processed_pseudogene"), c("protein_coding","lncRNA"), c("lncRNA","novel"), c("novel","processed_pseudogene"), c("novel","protein_coding"))
ggplot(ribORF_humanTestis_in1, aes(x=reorder(geneORFtype, length), y=length, fill=geneORFtype)) +
  geom_boxplot() +
  scale_fill_manual(values=c("protein_coding_canonical"="#ead890",
                         "lncRNA_noncoding" = "#cd4f38",
                         "processed_pseudogene_noncoding" = "#e48c2a",
                                               "novel_noncoding" = "#637c68",
                         "protein_coding_uORF"="#899da4",
                         "protein_coding_ouORF"="#c5d0ca",
                         "protein_coding_dORF"="#ba968a",
                         "protein_coding_odORF"="#f2c695")) +
  scale_y_continuous(trans="log10") +
  labs(x="Biotype",
       y="Length (nt)",
       title="Length of translated ncORFs\nand canonical ORFs") +
  theme_classic() +
  theme(legend.position = "none",
        axis.text.x = element_text(angle=90))  +
  facet_grid(~ gene_type, scales="free_x", space="free")
ggsave(file.path(plots_wd,"PNG/nORFs_length.png"), height=5.30, width=5.75)
ggsave(file.path(plots_wd,"PDF/nORFs_length.pdf"), height=5.30, width=5.75)

## ncORFs vs canonical ORFs
# wilcox.test(ribORF_humanTestis_in1$length ~ ribORF_humanTestis_in1$orfType)
```

```{r length_2, echo=F}
ribORF_humanTestis_in1$length_aa = ribORF_humanTestis_in1$length/3
ggplot(ribORF_humanTestis_in1, aes(x=length_aa, fill=gene_type, color="black")) +
  geom_density(alpha=.5, color="black") +
  geom_xsideboxplot(orientation = "y", width=4, color="black") +
  scale_fill_manual(values=c("protein_coding"="#ead890",
                         "lncRNA" = "#cd4f38",
                         "processed_pseudogene" = "#e48c2a",
                                               "novel" = "#637c68")) +
  scale_x_continuous(trans="log10") +
  geom_vline(xintercept=100) +
  labs(fill="Biotype",
       color="Biotype",
       x="Length (aa)",
       title="Length of translated ncORFs and canonical ORFs") +
  theme_classic() +
  theme(legend.position = "none")
# ggsave(file.path(plots_wd,"PNG/nORFs_length_combined.png"), height=5.69, width=4.72)
# ggsave(file.path(plots_wd,"PDF/nORFs_length_combined.pdf"), height=5.69, width=4.72)

ggplot(ribORF_humanTestis_in1, aes(x=length_aa, fill=gene_type, color="black")) +
  geom_density(alpha=.5, color="black") +
  scale_fill_manual(values=c("protein_coding"="#ead890",
                         "lncRNA" = "#cd4f38",
                         "processed_pseudogene" = "#e48c2a",
                                               "novel" = "#637c68")) +
  scale_x_continuous(trans="log10") +
  geom_vline(xintercept=100) +
  labs(fill="Biotype",
       color="Biotype",
       x="Length (aa)",
       title="Length of translated ncORFs and canonical ORFs") +
  theme_classic() +
  theme(legend.position = "none")
# ggsave(file.path(plots_wd,"PNG/nORFs_length_density.png"))
# ggsave(file.path(plots_wd,"PDF/nORFs_length_density.pdf"))

```


**Starting codons**

```{r startcodons, echo=F}
ribORF_humanTestis_in1$start_codon = str_sub(ribORF_humanTestis_in1$orfID,-3)

ggplot(ribORF_humanTestis_in1, aes(x=gene_type, fill=start_codon)) +
  geom_bar(position="fill") +
  labs(x="Biotype",
       fill="Starting codon",
       title="Starting codon of ncORFs and canonical ORFs") +
  scale_fill_manual(values=c("#F8806A","#EB4F48","#DC3440","#AE123A")) +
  theme_classic() +
  theme(axis.text.x = element_text(angle=45))
ggsave(file.path(plots_wd,"PNG/startCodons.png"), height=4.69, width=4.72)
ggsave(file.path(plots_wd,"PDF/startCodons.pdf"), height=4.69, width=4.72)

ggplot(ribORF_humanTestis_in1, aes(x=geneORFtype, fill=start_codon)) +
  geom_bar(position="fill") +
  labs(x="Biotype",
       fill="Starting codon",
       title="Starting codon of ncORFs and canonical ORFs") +
  scale_fill_manual(values=c("#F8806A","#EB4F48","#DC3440","#AE123A")) +
  theme_classic() +
  theme(axis.text.x = element_text(angle=45)) + 
  facet_grid(~ gene_type, scales="free_x", space="free")
ggsave(file.path(plots_wd,"PNG/startCodons_geneORFtype.png"), height=4.69, width=4.72)
ggsave(file.path(plots_wd,"PDF/startCodons_geneORFtype.pdf"), height=4.69, width=4.72)
```


```{r translate_seq, echo=F}
# library(Biostrings)
# 
# # Function to translate DNA sequence to amino acids
# translate_dna_to_aa <- function(dna_sequence) {
#   aa_seq <- translate(DNAString(dna_sequence))
#   return(as.character(aa_seq))
# }
# 
# fastaFile = read.table(file.path(riboseq_dir,"Annotation/candidateORF.fa"), header=F)
# 
# ## subset headers and seqs
# orfIDs = fastaFile %>% subset(grepl(">", V1))
# seqs = fastaFile %>% subset(!grepl(">",V1))
# 
# fastaCandidates = data.frame("orfID" = orfIDs,
#                              "ORFseq"= seqs)
# names(fastaCandidates) = c("orfID","ORFseq")
# fastaCandidates$orfID = gsub(">","",fastaCandidates$orfID)
# 
# ## select those that are in subset of interest --> ribORF_humanTestis_in1
# nrow(ribORF_humanTestis_in1)
# fastaCandidates_subset = fastaCandidates %>% subset(orfID %in% ribORF_humanTestis_in1$orfID)
# nrow(fastaCandidates_subset)
# 
# ## add the fasta sequences to the dataframe
# ribORF_humanTestis_in1_fastas = merge(fastaCandidates_subset, ribORF_humanTestis_in1, by="orfID")
# nrow(ribORF_humanTestis_in1_fastas)
# 
# ## translate the ORFs
# ribORF_humanTestis_in1_fastas$ORFpep = sapply(ribORF_humanTestis_in1_fastas$ORFseq, translate_dna_to_aa)
# write.csv(ribORF_humanTestis_in1_fastas, file.path(save_wd, "ribORF_humanTestis_in1.csv"), row.names = F)
ribORF_humanTestis_in1_fastas = read.csv(file.path(save_wd, "ribORF_humanTestis_in1.csv"))
```


**Composition**

```{r composition_aa, echo=F}
# ## log2ratio to see if ncORFs are relatively enriched in any aminoacid
# # Function to calculate log2ratio of relative frequency
# calculate_log2ratio_df <- function(group1_seqs, group2_seqs) {
#   all_amino_acids <- c("A","R","N","D","C","Q","E","G","H","U","L","K","M","F","P","S","T","W","Y","V", "*")
#   log2ratios <- numeric(length(all_amino_acids))
#   
#   for (aa in all_amino_acids) {
#     count_group1 <- sum(sapply(group1_seqs, function(seq) sum(unlist(strsplit(as.character(seq), "")) == aa)))
#     count_group2 <- sum(sapply(group2_seqs, function(seq) sum(unlist(strsplit(as.character(seq), "")) == aa)))
#     
#     freq_group1 <- count_group1 / sum(sapply(group1_seqs, nchar))
#     freq_group2 <- count_group2 / sum(sapply(group2_seqs, nchar))
#     
#     log2ratios[aa] <- log2(freq_group1 / freq_group2)
#   }
#   
#   log2ratio_df <- data.frame(Aminoacid = all_amino_acids, log2ratio = log2ratios)
#   return(log2ratio_df)
# }
# 
# ## select ncORFs
# ncORFs_seqs = ribORF_humanTestis_in1_fastas %>% subset(orfType == "noncoding")
# ncORFs_seqs = as.character(ncORFs_seqs$ORFpep)
# 
# ## ORFs
# ORFs_seqs = ribORF_humanTestis_in1_fastas %>% subset(orfType == "canonical")
# ORFs_seqs = as.character(ORFs_seqs$ORFpep)
# 
# ## compute log2ratio per letter
# log2ratios = calculate_log2ratio_df(ncORFs_seqs, ORFs_seqs)
# print(log2ratios)
# 
# hph=c("A","V","L","I","M","F","Y","W")
# polar=c("S","T","N","Q")
# positively_charged=c("R","H","K")
# negatively_charged=c("D","E")
# special_cases=c("C","G","P")
# 
# log2ratios = log2ratios %>% 
#   mutate(aa_type = case_when(Aminoacid %in% hph ~ "hidrophobic",
#                              Aminoacid %in% polar ~ "polar",
#                              Aminoacid %in% positively_charged ~ "positive",
#                              Aminoacid %in% negatively_charged ~ "negative",
#                              Aminoacid %in% special_cases ~ "special")) %>%
#   subset(Aminoacid != "*")
# log2ratios = na.omit(log2ratios) 
# 
# # ggthemr("dust")
# ggplot(log2ratios, aes(x=Aminoacid, y=log2ratio, fill = aa_type)) +
#   geom_bar(stat = "identity") +
#   labs(y="Log2Ratio (ncORFs/canonical ORFs)",
#        title="Relative enrichement of aminoacids in ncORFs") +
#   scale_fill_carto_d(palette = "Burg") +
#   theme_classic() +
#   theme(legend.position = "none") +
#   facet_grid(~ aa_type, space="free", scales = "free_x")
# ggsave(file.path(plots_wd,"PNG/log2ratio_aa_ncORFsVSORFs.png"), height=4.72, width=6.02)
# ggsave(file.path(plots_wd,"PDF/log2ratio_aa_ncORFsVSORFs.pdf"), height=4.72, width=6.02)
# 
# # ggthemr("dust")
# ggplot(log2ratios, aes(x=reorder(Aminoacid,log2ratio), y=log2ratio, fill = aa_type)) +
#   geom_bar(stat = "identity") +
#   labs(y="Log2Ratio (ncORFs/canonical ORFs)",
#        x="Aminoacid",
#        title="Relative enrichement of aminoacids in ncORFs",
#        fill="Aminoacid type") +
#     scale_fill_carto_d(palette = "Burg") +
#   theme_classic() +
#   theme(legend.position = "top")
# ggsave(file.path(plots_wd,"PNG/log2ratio_aa_ncORFsVSORFs.ordered.png"), height=4.72, width=6.02)
# ggsave(file.path(plots_wd,"PDF/log2ratio_aa_ncORFsVSORFs.ordered.pdf"), height=4.72, width=6.02)
```

**Translation efficiency**

```{r translation_eficiency, echo=F}
TE_dir=
orf_TE_1 = read.csv("/users/genomics/marta/TestisProject_SaraRazquin/JC_RiboSeq_TPMs/Outputs/with_TranscriptomeReconstruction/Expression/v47_altORFs_MultMap/ribORFsimp.testis_1.tsv", sep="\t")
orf_TE_2 = read.csv("/users/genomics/marta/TestisProject_SaraRazquin/JC_RiboSeq_TPMs/Outputs/with_TranscriptomeReconstruction/Expression/v47_altORFs_MultMap/ribORFsimp.testis_2.tsv", sep="\t")
orf_TE_3 = read.csv("/users/genomics/marta/TestisProject_SaraRazquin/JC_RiboSeq_TPMs/Outputs/with_TranscriptomeReconstruction/Expression/v47_altORFs_MultMap/ribORFsimp.testis_3.tsv", sep="\t")

orf_TE = rbind(orf_TE_1, orf_TE_2, orf_TE_3)

## get transcript_id
orf_TE$transcript_id = gsub(":.*","",orf_TE$orfID)
orf_TE$transcript_id = gsub("\\..*","",orf_TE$transcript_id)

## we keep only those that are in the interesting subset
nrow(ribORF_humanTestis_in1)
orf_TE_in1 = merge(orf_TE, ribORF_humanTestis_in1 %>% select(orfID, geneORFtype, gene_id, gene_name), by="orfID")
nrow(orf_TE_in1)
```

```{r RNA_Ribo, echo=F}
TPMs_RNA = read.csv("/users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/human/featureCounts_gffcompare/table_of_counts_TPMs_withLength.csv")

testis_TPMs = TPMs_RNA %>% select(gene_id, Length, human_testis_rna_1, human_testis_rna_2, human_testis_rna_3)
names(testis_TPMs)[1] = "transcript_id"
testis_TPMs_with_ORFs = testis_TPMs %>% subset(transcript_id %in% orf_TE_in1$transcript_id)

#----------- sample 1 ---------#
## riboseq
orf_testis_1 = orf_TE %>% subset(Sample == "testis_1") %>% subset(orfID %in% orf_TE_in1$orfID) %>% select(orfID, transcript_id, f1Tpm)
names(orf_testis_1)[3] = "Ribo_TPM"
orf_testis_1 = merge(orf_testis_1, orf_TE_in1 %>% select(orfID, geneORFtype, gene_id, gene_name, transcript_id), by=c("orfID","transcript_id"))

## rnaseq
rna_testis_1 = testis_TPMs_with_ORFs %>% select(transcript_id, human_testis_rna_1)
names(rna_testis_1)[2] = "RNA_TPM"

testis_1 = merge(rna_testis_1, orf_testis_1, by=c("transcript_id"))

## TE
testis_1$TE = testis_1$Ribo_TPM / testis_1$RNA_TPM
testis_1$sample = "Testis_1"

#----------- sample 2 ---------#
## riboseq
orf_testis_2 = orf_TE %>% subset(Sample == "testis_2") %>% subset(orfID %in% orf_TE_in1$orfID) %>% select(orfID, transcript_id, f1Tpm)
names(orf_testis_2)[3] = "Ribo_TPM"
orf_testis_2 = merge(orf_testis_2, orf_TE_in1 %>% select(orfID, geneORFtype, gene_id, gene_name, transcript_id), by=c("orfID","transcript_id"))

## rnaseq
rna_testis_2 = testis_TPMs_with_ORFs %>% select(transcript_id, human_testis_rna_2)
names(rna_testis_2)[2] = "RNA_TPM"

testis_2 = merge(rna_testis_2, orf_testis_2, by=c("transcript_id"))

## TE
testis_2$TE = testis_2$Ribo_TPM / testis_2$RNA_TPM
testis_2$sample = "Testis_2"

#----------- sample 3 ---------#
## riboseq
orf_testis_3 = orf_TE %>% subset(Sample == "testis_3") %>% subset(orfID %in% orf_TE_in1$orfID) %>% select(orfID, transcript_id, f1Tpm)
names(orf_testis_3)[3] = "Ribo_TPM"
orf_testis_3 = merge(orf_testis_3, orf_TE_in1 %>% select(orfID, geneORFtype, gene_id, gene_name, transcript_id), by=c("orfID","transcript_id"))

## rnaseq
rna_testis_3 = testis_TPMs_with_ORFs %>% select(transcript_id, human_testis_rna_3) 
names(rna_testis_3)[2] = "RNA_TPM"

testis_3 = merge(rna_testis_3, orf_testis_3, by=c("transcript_id"))

## TE
testis_3$TE = testis_3$Ribo_TPM / testis_3$RNA_TPM
testis_3$sample = "Testis_3"
#--------------- merging ---------------#
testis_TE = rbind(testis_1, testis_2, testis_3)
write.csv(testis_TE, file.path(save_wd, "translationEfficiency_testis.csv"), row.names=F)
testis_TE = testis_TE %>% subset(Ribo_TPM > 0 & RNA_TPM > 0)
write.csv(testis_TE, file.path(save_wd, "translationEfficiency_testis.withoutZeros.csv"), row.names=F)

```

```{r TE_plot, echo=F}
ggplot(testis_TE, aes(x=sample, y=TE, fill=geneORFtype)) +
  geom_boxplot() +
  scale_fill_manual(values=c("protein_coding_canonical"="#ead890",
                         "lncRNA_noncoding" = "#cd4f38",
                         "processed_pseudogene_noncoding" = "#e48c2a",
                                               "novel_noncoding" = "#637c68",
                         "protein_coding_uORF"="#899da4",
                         "protein_coding_ouORF"="#c5d0ca",
                         "protein_coding_dORF"="#ba968a",
                         "protein_coding_odORF"="#f2c695")) +
    scale_x_discrete(labels = function(x) sub(".*_", "", x)) +  # Extract everything after the last "_" for display
labs(x="Sample",
  y="Translation Efficiency",
  title="Translation Efficiency") +
  scale_y_continuous(trans="log10") +
  theme_classic()

ggsave(file.path(plots_wd,"PNG/translation_efficiency_persample.png"), height=4.72, width=6.02)
ggsave(file.path(plots_wd,"PDF/translation_efficiency_persample.pdf"), height=4.72, width=6.02)
```

```{r TE_plot_mean, echo=F}
## if an ORF is translated in more than 1 sample, compute the mean
testis_TE = as.data.frame(testis_TE)
testis_TE$TE = as.numeric(testis_TE$TE)
mean_testis_TE = testis_TE %>% unique() %>% ungroup() %>%
  group_by(orfID) %>% 
  mutate(mean_TE = mean(TE)) %>%  
  ungroup() %>% select(orfID, geneORFtype, gene_id, gene_name, mean_TE) %>% unique()
write.csv(mean_testis_TE, file.path(save_wd, "mean_translationEfficiency_testis.csv"), row.names=F)

ggplot(mean_testis_TE, aes(x=geneORFtype, y=log10(mean_TE))) +
  # geom_violin(aes(fill=geneORFtype), alpha=.2) +
  geom_quasirandom(dodge.width = 0.9, size=1, alpha=.3,  varwidth = TRUE, aes(color=geneORFtype)) +
  geom_boxplot(width=0.25, aes(color = "black", fill=geneORFtype), alpha=.8) +
  scale_fill_manual(values=c("protein_coding_canonical"="#ead890",
                         "lncRNA_noncoding" = "#cd4f38",
                         "processed_pseudogene_noncoding" = "#e48c2a",
                                               "novel_noncoding" = "#637c68",
                         "protein_coding_uORF"="#899da4",
                         "protein_coding_ouORF"="#c5d0ca",
                         "protein_coding_dORF"="#ba968a",
                         "protein_coding_odORF"="#f2c695")) +
  scale_color_manual(values=c("protein_coding_canonical"="#ead890",
                         "lncRNA_noncoding" = "#cd4f38",
                         "processed_pseudogene_noncoding" = "#e48c2a",
                                               "novel_noncoding" = "#637c68",
                         "protein_coding_uORF"="#899da4",
                         "protein_coding_ouORF"="#c5d0ca",
                         "protein_coding_dORF"="#ba968a",
                         "protein_coding_odORF"="#f2c695")) +  
  labs(x="Biotype",
  y="Translation Efficiency (log10TE)",
  title="Translation Efficiency") +
    scale_x_discrete(labels = function(x) sub(".*_", "", x)) +  # Extract everything after the last "_" for display
# scale_y_continuous(trans="log10") +
  theme_classic() +
  theme(legend.position="none")

ggsave(file.path(plots_wd,"PNG/translation_efficiency_meanTE.log10.png"))
ggsave(file.path(plots_wd,"PDF/translation_efficiency_meanTE.log10.pdf"))

ggplot(mean_testis_TE, aes(x=geneORFtype, y=mean_TE)) +
  # geom_violin(aes(fill=gene_type), alpha=.2) +
  geom_quasirandom(dodge.width = 0.9, size=1, alpha=.3,  varwidth = TRUE, aes(color=geneORFtype)) +
  geom_boxplot(width=0.25, aes(color = "black", fill=geneORFtype), alpha=.8) +
    scale_x_discrete(labels = function(x) sub(".*_", "", x)) +  # Extract everything after the last "_" for display
scale_fill_manual(values=c("protein_coding_canonical"="#ead890",
                         "lncRNA_noncoding" = "#cd4f38",
                         "processed_pseudogene_noncoding" = "#e48c2a",
                                               "novel_noncoding" = "#637c68",
                         "protein_coding_uORF"="#899da4",
                         "protein_coding_ouORF"="#c5d0ca",
                         "protein_coding_dORF"="#ba968a",
                         "protein_coding_odORF"="#f2c695")) +
  scale_color_manual(values=c("protein_coding_canonical"="#ead890",
                         "lncRNA_noncoding" = "#cd4f38",
                         "processed_pseudogene_noncoding" = "#e48c2a",
                                               "novel_noncoding" = "#637c68",
                         "protein_coding_uORF"="#899da4",
                         "protein_coding_ouORF"="#c5d0ca",
                         "protein_coding_dORF"="#ba968a",
                         "protein_coding_odORF"="#f2c695")) +  
  labs(x="Biotype",
  y="Translation Efficiency",
  title="Translation Efficiency") +
  scale_y_continuous(trans="log10") +
  theme_classic() +
  theme(legend.position="none")

ggsave(file.path(plots_wd,"PNG/translation_efficiency_meanTE.png"), height=4.72, width=5.02)
ggsave(file.path(plots_wd,"PDF/translation_efficiency_meanTE.pdf"), height=4.72, width=5.02)

ggplot(mean_testis_TE, aes(x=geneORFtype, y=log10(mean_TE))) +
  # geom_violin(aes(fill=geneORFtype), alpha=.2) +
  geom_boxplot(aes(fill=geneORFtype), alpha=.8, notch=T) +
  scale_fill_manual(values=c("protein_coding_canonical"="#ead890",
                         "lncRNA_noncoding" = "#cd4f38",
                         "processed_pseudogene_noncoding" = "#e48c2a",
                                               "novel_noncoding" = "#637c68",
                         "protein_coding_uORF"="#899da4",
                         "protein_coding_ouORF"="#c5d0ca",
                         "protein_coding_dORF"="#ba968a",
                         "protein_coding_odORF"="#f2c695")) +
    scale_x_discrete(labels = function(x) sub(".*_", "", x)) +  # Extract everything after the last "_" for display
scale_color_manual(values=c("protein_coding_canonical"="#ead890",
                         "lncRNA_noncoding" = "#cd4f38",
                         "processed_pseudogene_noncoding" = "#e48c2a",
                                               "novel_noncoding" = "#637c68",
                         "protein_coding_uORF"="#899da4",
                         "protein_coding_ouORF"="#c5d0ca",
                         "protein_coding_dORF"="#ba968a",
                         "protein_coding_odORF"="#f2c695")) +  
  labs(x="Biotype",
  y="Translation Efficiency (log10TE)",
  title="Translation Efficiency") +
  # scale_y_continuous(trans="log10") +
  theme_classic() +
  theme(legend.position="right",
        axis.text.x = element_blank())
ggsave(file.path(plots_wd,"PNG/translation_efficiency_meanTE.simple.log10.png"))
ggsave(file.path(plots_wd,"PDF/translation_efficiency_meanTE.simple.log10.pdf"))
```

## Of the translated, how many are NOT translated in liver or brain?

```{r translation_brain, echo=F}
##-------------------------- SAMPLE 1 --------------------------##
ribORF_humanBrain_1 = read.table(file.path(riboseq_dir,"RibORF/human_brain_ribo_1_r1/pred.pvalue.parameters.txt"), header=T)

repre_ribORF_humanBrain_1=  read.table(file.path(riboseq_dir,"RibORF/human_brain_ribo_1_r1/repre.valid.ORF.genepred.txt"), header=F)

ribORF_humanBrain_1 = ribORF_humanBrain_1 %>% subset(orfID %in% repre_ribORF_humanBrain_1$V1)
ribORF_humanBrain_1$sample = "Brain1"
## get transcriptID
ribORF_humanBrain_1$transcript_id = gsub(":.*", "", ribORF_humanBrain_1$orfID)
ribORF_humanBrain_1$transcript_id = gsub("\\..*", "", ribORF_humanBrain_1$transcript_id)
# nrow(ribORF_humanBrain_1)

##-------------------------- SAMPLE 2 --------------------------##
ribORF_humanBrain_1 = read.table(file.path(riboseq_dir,"RibORF/human_brain_ribo_2_r1/pred.pvalue.parameters.txt"), header=T)

repre_ribORF_humanBrain_1=  read.table(file.path(riboseq_dir,"RibORF/human_brain_ribo_2_r1/repre.valid.ORF.genepred.txt"), header=F)

ribORF_humanBrain_1 = ribORF_humanBrain_1 %>% subset(orfID %in% repre_ribORF_humanBrain_1$V1)
ribORF_humanBrain_1$sample = "Brain1"
## get transcriptID
ribORF_humanBrain_1$transcript_id = gsub(":.*","", ribORF_humanBrain_1$orfID)
ribORF_humanBrain_1$transcript_id = gsub("\\..*", "",ribORF_humanBrain_1$transcript_id)
# nrow(ribORF_humanBrain_1)

##-------------------------- SAMPLE 3 --------------------------##
ribORF_humanBrain_3 = read.table(file.path(riboseq_dir,"RibORF/human_brain_ribo_3_r1/pred.pvalue.parameters.txt"), header=T)

repre_ribORF_humanBrain_3=  read.table(file.path(riboseq_dir,"RibORF/human_brain_ribo_3_r1/repre.valid.ORF.genepred.txt"), header=F)

ribORF_humanBrain_3 = ribORF_humanBrain_3 %>% subset(orfID %in% repre_ribORF_humanBrain_3$V1)
ribORF_humanBrain_3$sample = "Brain3"
## get transcriptID
ribORF_humanBrain_3$transcript_id = gsub(":.*","", ribORF_humanBrain_3$orfID)
ribORF_humanBrain_3$transcript_id = gsub("\\..*","", ribORF_humanBrain_3$transcript_id)

# nrow(ribORF_humanBrain_3)

### Concatenate ncORFs for all 2 samples
ribORF_humanBrain = rbind(ribORF_humanBrain_3, ribORF_humanBrain_1, ribORF_humanBrain_1)


## get gene_id, gene_name, gene_type
ribORF_humanBrain_Annotated = merge(annot, ribORF_humanBrain, by="transcript_id")
# nrow(ribORF_humanBrain_Annotated)
## get orfType
ribORF_humanBrain_Annotated$orfType = sapply(strsplit(ribORF_humanBrain_Annotated$orfID, "\\|"), function(x) x[4])

## how many orfTypes per gene_type?
# ribORF_humanBrain_Annotated %>% subset(gene_type == "lncRNA" | gene_type == "processed_pseudogene" | gene_type == "protein_coding") %>% group_by(gene_type, orfType) %>% count()

## for protein-coding we only keep the canonical
ribORF_humanBrain_Annotated_biotypes = ribORF_humanBrain_Annotated %>%
  subset(gene_type == "lncRNA" | gene_type == "processed_pseudogene" | gene_type == "protein_coding" | gene_type == "novel") %>%
  filter(!(gene_type == "protein_coding" & orfType != "canonical"))
write.csv(ribORF_humanBrain_Annotated_biotypes, file.path(save_wd, "ribORF_humanBrain_translated.csv"), row.names=F) 
```

```{r translation_liver, echo=F}
##-------------------------- SAMPLE 1 --------------------------##
ribORF_humanLiver_1 = read.table(file.path(riboseq_dir,"RibORF/human_liver_ribo_1_r1/pred.pvalue.parameters.txt"), header=T)

repre_ribORF_humanLiver_1=  read.table(file.path(riboseq_dir,"RibORF/human_liver_ribo_1_r1/repre.valid.ORF.genepred.txt"), header=F)

ribORF_humanLiver_1 = ribORF_humanLiver_1 %>% subset(orfID %in% repre_ribORF_humanLiver_1$V1)
ribORF_humanLiver_1$sample = "Liver1"
## get transcriptID
ribORF_humanLiver_1$transcript_id = gsub(":.*","", ribORF_humanLiver_1$orfID)
ribORF_humanLiver_1$transcript_id = gsub("\\..*","", ribORF_humanLiver_1$transcript_id)
# nrow(ribORF_humanLiver_1)

##-------------------------- SAMPLE 2 --------------------------##
ribORF_humanLiver_2 = read.table(file.path(riboseq_dir,"RibORF/human_liver_ribo_2_r1/pred.pvalue.parameters.txt"), header=T)

repre_ribORF_humanLiver_2=  read.table(file.path(riboseq_dir,"RibORF/human_liver_ribo_2_r1/repre.valid.ORF.genepred.txt"), header=F)

ribORF_humanLiver_2 = ribORF_humanLiver_2 %>% subset(orfID %in% repre_ribORF_humanLiver_2$V1)
ribORF_humanLiver_2$sample = "Liver2"
## get transcriptID
ribORF_humanLiver_2$transcript_id = gsub(":.*","", ribORF_humanLiver_2$orfID)
ribORF_humanLiver_2$transcript_id = gsub("\\..*","", ribORF_humanLiver_2$transcript_id)

# nrow(ribORF_humanLiver_2)

##-------------------------- SAMPLE 3 --------------------------##
ribORF_humanLiver_3 = read.table(file.path(riboseq_dir,"RibORF/human_liver_ribo_3_r1/pred.pvalue.parameters.txt"), header=T)

repre_ribORF_humanLiver_3=  read.table(file.path(riboseq_dir,"RibORF/human_liver_ribo_3_r1/repre.valid.ORF.genepred.txt"), header=F)

ribORF_humanLiver_3 = ribORF_humanLiver_3 %>% subset(orfID %in% repre_ribORF_humanLiver_3$V1)
ribORF_humanLiver_3$sample = "Liver3"
## get transcriptID
ribORF_humanLiver_3$transcript_id = gsub(":.*","", ribORF_humanLiver_3$orfID)
ribORF_humanLiver_3$transcript_id = gsub("\\..*","", ribORF_humanLiver_3$transcript_id)
# nrow(ribORF_humanLiver_3)

### Concatenate ncORFs for all 2 samples
ribORF_humanLiver = rbind(ribORF_humanLiver_3, ribORF_humanLiver_2, ribORF_humanLiver_1)


## get gene_id, gene_name, gene_type
ribORF_humanLiver_Annotated = merge(annot, ribORF_humanLiver, by="transcript_id")
# nrow(ribORF_humanLiver_Annotated)
## get orfType
ribORF_humanLiver_Annotated$orfType = sapply(strsplit(ribORF_humanLiver_Annotated$orfID, "\\|"), function(x) x[4])

## how many orfTypes per gene_type?
# ribORF_humanLiver_Annotated %>% subset(gene_type == "lncRNA" | gene_type == "processed_pseudogene" | gene_type == "protein_coding") %>% group_by(gene_type, orfType) %>% count()

## for protein-coding we only keep the canonical
ribORF_humanLiver_Annotated_biotypes = ribORF_humanLiver_Annotated %>%
  subset(gene_type == "lncRNA" | gene_type == "processed_pseudogene" | gene_type == "protein_coding" | gene_type == "novel") %>%
  filter(!(gene_type == "protein_coding" & orfType != "canonical"))
write.csv(ribORF_humanLiver_Annotated_biotypes, file.path(save_wd, "ribORF_humanLiver_translated.csv"), row.names=F) 
```

## CTDB 

```{r CTDB, echo=F}
CTDB = read.csv("/projects_eg/projects/marta/CTdatabase_list.csv", sep="\t")
CTDB = CTDB %>% mutate(X_A = case_when(grepl("^X", chr) ~ "X",
                                       TRUE ~ "A"))
print(nrow(CTDB))
print(nrow(ribORF_humanTestis_in1 %>% subset(gene_name %in% CTDB$gene_name)))
CTDB_translated_in_testis = merge(ribORF_humanTestis_in1, CTDB, by="gene_name")
table(CTDB_translated_in_testis$X_A)
## non-X are more dominantly expressed in germ-cell differentation late stages such as spermatocytes
## CT-X are generally expressed in the speratogonia, wchich are proliferating germ cells.

notFound = CTDB %>% subset(!gene_name %in% CTDB_translated_in_testis$gene_name)
print(nrow(notFound))
## From those not translated, are they not translated or not found because of a different name?
notTranslated = notFound %>% subset(gene_name %in% annot$gene_name)
print(nrow(notTranslated))

notFound_diffName = notFound %>% subset(!gene_name %in% annot$gene_name)
print(nrow(notFound_diffName))
```
