{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import os,re,glob,csv\n",
    "import pandas as pd\n",
    "import numpy as np\n",
    "from collections import Counter\n",
    "from rna_seq_normalization import Normalization as Norm\n",
    "from functools import reduce\n",
    "\n",
    "users_dir = \"/users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47\"\n",
    "specie = \"human\"\n",
    "# GENOMEDIR = \"/genomics/users/marta/genomes\"\n",
    "GENOMEDIR = \"/data/genomics/marta/genomes\"\n",
    "\n",
    "### cancer data\n",
    "cancer_dir = users_dir + \"/cancers\"\n",
    "raw_cancer_dir=\"/users/genomics/marta/TCGA_RNASeq\"\n",
    "tcga_projects=[\"TCGA-BRCA\",\"TCGA-LUSC\",\"TCGA-PRAD\",\"TCGA-KIRC\",\"TCGA-LUAD\",\"TCGA-BLCA\"]#,\"TCGA-LIHC\"]\n",
    "other_projects=[\"GSE102101_KIRC\",\"GSE133624_BLCA\",\"GSE22260_PRAD\",\"PRJEB2449_PRAD\",\"SRP238334_KIRC\",\"GSE214846_LIHC\",\"GSE229705_LUAD\",\"SRP107326_COAD\",\"TCGA_COAD\"]\n",
    "# deleted=[\"GSE103001_BRCA\",\"GSE89223_PRAD\"]\n",
    "manuscript_projects = [\"liver_adjacent_totalRNA_LIHC\",\"hcc_normal_totalRNA_LIHC\",\"GSE193567_LIHC\",\"LIHC_TCGA_LIHC\"]\n",
    "all_projects = tcga_projects + other_projects + manuscript_projects\n",
    "\n",
    "cancertypes = [\"BRCA\",\"BLCA\",\"LUAD\",\"KIRC\",\"PRAD\",\"LUSC\",\"LIHC\",\"COAD\"]\n",
    "## annotation file\n",
    "annotation=\"/users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/human/newReference_Resconstructed/gencode.v47.gffcompare.TestisLiverBrain.annotation.sorted.1transcript.sorted.NOchr.gtf\"\n",
    "transcript_gene=pd.read_csv(\"/users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/human/newReference_Resconstructed/transID_geneID_isoforms_selected.1to1.csv\")\n",
    "\n",
    "def count_greater_than_one(row):\n",
    "    return (row > 1).sum()"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Quantify with TCGAData and new reference\n",
    "`featureCounts_newRef_cancer.sh`\n",
    "\n",
    "output in: `/users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/cancers/featureCounts`"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## TPM"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "for file in os.listdir(os.path.join(cancer_dir,\"featureCounts\")):\n",
    "    if file.endswith(\".txt\"):\n",
    "\n",
    "        proj = file.split(\"featureCounts_\")[-1]\n",
    "        cancer_type = proj[:-4]\n",
    "        print(cancer_type)\n",
    "        toc = pd.read_csv(os.path.join(cancer_dir,\"featureCounts\",file), sep=\"\\t\", comment=\"#\")\n",
    "        toc = toc[toc['Geneid'].str.contains('PAR_')==False]\n",
    "        toc['Geneid']=toc['Geneid'].str.split('.').str[0]\n",
    "        toc.rename(columns={'Geneid':'transcript_id'}, inplace=True)\n",
    "\n",
    "        filter_col = [col for col in toc if col.startswith('/')]\n",
    "        for col in filter_col:\n",
    "            new_col=col.split(\"Aligned\")[0]\n",
    "            new_col=new_col.split(\"/\")[-1]\n",
    "            if proj in tcga_projects:\n",
    "                new_col=cancer_type+new_col[4:]\n",
    "            toc.rename(columns={col:new_col}, inplace=True)\n",
    "\n",
    "        length = toc['Length']\n",
    "        genes = toc['transcript_id']\n",
    "        # we are only interested in the columns with counts\n",
    "        counts = toc\n",
    "        counts.drop([\"Chr\",\"Start\",\"End\",\"Strand\",\"Length\",\"transcript_id\"],axis=1, inplace=True)\n",
    "        # calculate TPMs\n",
    "        tpm_df = Norm.tpm(counts, length)\n",
    "        # add transcript_id and length again\n",
    "        tpms = pd.concat([genes,tpm_df, length], axis=1)\n",
    "        tpms.to_csv(os.path.join(cancer_dir,\"featureCounts/table_of_counts_TPMs_\")+cancer_type+\".csv\", index=None)\n",
    " "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "## merge single-end and paired-end TCGA COAD\n",
    "SE_COAD = pd.read_csv(os.path.join(cancer_dir,\"featureCounts/table_of_counts_TPMs_TCGA_COAD_SE.csv\"))\n",
    "print(len(SE_COAD.columns))\n",
    "PE_COAD = pd.read_csv(os.path.join(cancer_dir,\"featureCounts/table_of_counts_TPMs_TCGA_COAD_PE.csv\"))\n",
    "print(len(PE_COAD.columns))\n",
    "\n",
    "COAD = SE_COAD.merge(PE_COAD, on=[\"transcript_id\",\"Length\"])\n",
    "COAD.to_csv(os.path.join(cancer_dir,\"featureCounts/table_of_counts_TPMs_TCGA_COAD.csv\"), index=None)\n",
    "print(len(COAD.columns))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "## filter patients we are interested in, from those not TCGA\n",
    "for proj in all_projects:\n",
    "    print(proj)\n",
    "    if proj in tcga_projects:\n",
    "        cancer_type = proj[5:]\n",
    "\n",
    "        fc = pd.read_csv(os.path.join(cancer_dir,\"featureCounts/table_of_counts_TPMs_\")+proj+\".csv\")\n",
    "        tumor_transcripts = list()\n",
    "        patients=pd.read_csv(os.path.join(\"/users/genomics/marta/TCGA_RNASeq\",proj,\"results/QC_patients1.csv\"))\n",
    "        print(len(patients),\" patients\")\n",
    "        # patients['normal'] = patients['normal'].str.replace('TCGA', cancer_type)\n",
    "        # patients['tumor'] = patients['tumor'].str.replace('TCGA', cancer_type)\n",
    "\n",
    "    elif proj in other_projects:\n",
    "        fc = pd.read_csv(os.path.join(cancer_dir,\"featureCounts/table_of_counts_TPMs_\")+proj+\".csv\")\n",
    "        print(fc)\n",
    "        tumor_transcripts = list()\n",
    "        patients=pd.read_csv(os.path.join(\"/users/genomics/marta/cancers_RNASeq\",proj,\"results/patients.csv\"))\n",
    "        print(len(patients),\" patients\")\n",
    "\n",
    "        samples_to_keep = patients.tumor.values.tolist() + patients.normal.values.tolist() + ['Length','transcript_id']\n",
    "        fc_selected = fc[fc.columns.intersection(samples_to_keep)]\n",
    "\n",
    "    elif proj in manuscript_projects:\n",
    "        fc = pd.read_csv(os.path.join(cancer_dir,\"featureCounts/table_of_counts_TPMs_\")+proj+\".csv\")\n",
    "        tumor_transcripts = list()\n",
    "\n",
    "        if \"liver_adjacent_totalRNA\" in proj:\n",
    "            patients=pd.read_csv(os.path.join(\"/projects_eg/projects/marta/liver_adjacent_totalRNA/results/clean_patients.csv\"))\n",
    "            # patients.rename(columns={'adjacent':'normal'}, inplace=True)\n",
    "        elif \"hcc_normal_totalRNA\" in proj:\n",
    "            patients=pd.read_csv(os.path.join(\"/projects_eg/projects/marta/hcc_normal_totalRNA/results/clean_patients.csv\"))\n",
    "            # patients.rename(columns={'adjacent':'normal'}, inplace=True)\n",
    "        elif \"GSE193567\" in proj:\n",
    "            patients=pd.read_csv(os.path.join(\"/projects_eg/projects/marta/GSE193567/results/clean_patients.csv\"))        \n",
    "            # patients.rename(columns={'adjacent':'normal'}, inplace=True)\n",
    "        elif \"LIHC_TCGA\" in proj:\n",
    "            patients=pd.read_csv(os.path.join(\"/projects_eg/projects/marta/LIHC_TCGA/results/clean_patients.csv\"))    \n",
    "            patients['normal'] = patients['patient'] + \"_normal\"\n",
    "            patients['tumor'] = patients['patient'] + \"_tumor\"\n",
    "        print(len(patients),\" patients\")\n",
    "\n",
    "    for index,patient in patients.iterrows():\n",
    "        if proj in tcga_projects:\n",
    "            patient_fc = fc[[\"transcript_id\",\"Length\",patient['tumor']]]\n",
    "\n",
    "            tumor_patient_fc = patient_fc[patient_fc[patient['tumor']] >= 1 ]\n",
    "            tumor_transcripts.extend(tumor_patient_fc.transcript_id.values.tolist())\n",
    "        if proj in other_projects:\n",
    "            patient_fc = fc[[\"transcript_id\",\"Length\",patient['tumor']]]\n",
    "\n",
    "            tumor_patient_fc = patient_fc[patient_fc[patient['tumor']] >= 1 ]\n",
    "            tumor_transcripts.extend(tumor_patient_fc.transcript_id.values.tolist())       \n",
    "        if proj in manuscript_projects:\n",
    "            patient_fc = fc[[\"transcript_id\",\"Length\",patient['tumor']]]\n",
    "\n",
    "            tumor_patient_fc = patient_fc[patient_fc[patient['tumor']] >= 1 ]\n",
    "            tumor_transcripts.extend(tumor_patient_fc.transcript_id.values.tolist())                        \n",
    "\n",
    "    tumor1FPKM = fc[fc['transcript_id'].isin(tumor_transcripts)]\n",
    "    # print(tumor1FPKM)\n",
    "    known = tumor1FPKM[tumor1FPKM['transcript_id'].str.contains('ENST')]\n",
    "    novel = tumor1FPKM[tumor1FPKM['transcript_id'].str.contains('TCONS')]\n",
    "    merged = tumor1FPKM.merge(transcript_gene, on=['transcript_id'], how=\"inner\")\n",
    "    merged = merged[merged['gene_type'].isin(['lncRNA','processed_pseudogene','novel','protein_coding'])]\n",
    "\n",
    "    lncRNA = merged[merged['gene_type'].isin(['lncRNA','processed_pseudogene'])]\n",
    "    print(\"lncRNA: \",len(lncRNA))\n",
    "\n",
    "    cds = merged[merged['gene_type'] == \"protein_coding\"]\n",
    "    print(\"PROTEIN CODING: \",len(cds))\n",
    "\n",
    "    novel = merged[merged['gene_type'] == \"novel\"]\n",
    "    ## length limitations\n",
    "    novel = novel[novel['Length'].astype(int) < 91667]\n",
    "    novel = novel[novel['Length'].astype(int) > 300]\n",
    "\n",
    "    print(\"NOVEL: \",len(novel))\n",
    "\n",
    "    # # tumor1FPKM = pd.concat([novel, lncRNA, cds], ignore_index = True)\n",
    "    # merged[['transcript_id','gene_id','gene_name','gene_type','Length']].to_csv(os.path.join(cancer_dir,\"tumorexpressed/tumor_1FPKM_\")+cancer_type+\".csv\",index=False)\n",
    "\n",
    "    ## keep only tumor samples\n",
    "    if proj in tcga_projects:\n",
    "        tumorONLY_merged = merged[merged.columns.drop(list(merged.filter(regex='normal')))]\n",
    "    else:\n",
    "        tumorONLY_merged = merged.drop(columns=patients.normal.values.tolist())\n",
    "\n",
    "    tumorONLY_merged = tumorONLY_merged[['transcript_id','gene_id','gene_id', 'gene_type', 'gene_name','Length', *patients.tumor.values.tolist()]]\n",
    "    tumorONLY_merged.to_csv(os.path.join(cancer_dir,\"tumorexpressed/tumor_1FPKM_table_of_counts_\")+proj+\".csv\",index=False)\n",
    "\n",
    "    ###### GET TUMOR-EXPRESSED > 1 TPM\n",
    "    # Filter only integer columns\n",
    "    int_columns = tumorONLY_merged.select_dtypes(include='int64')\n",
    "    \n",
    "    # Create a new column with the count of int columns per row whose value is > 1\n",
    "    # tumorONLY_merged['n'] = int_columns.apply(lambda row: (row > 1).sum(), axis=1)\n",
    "    tumorONLY_merged['n'] = tumorONLY_merged.drop(columns=['transcript_id','gene_id','gene_name','gene_type','Length']).apply(count_greater_than_one, axis=1)\n",
    "    tumorONLY_merged = tumorONLY_merged[['transcript_id','gene_id','gene_name','gene_type','Length','n']]\n",
    "    tumorONLY_merged.sort_values(by=['n'], ascending=False).to_csv(os.path.join(cancer_dir,\"tumorexpressed/tumor_1FPKM_n_\")+proj+\".csv\",index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "## create a merged table of counts per ctype and a big patients file\n",
    "## CAN THEY BE CONSIDERED AS ONE?\n",
    "for ctype in cancertypes:\n",
    "    print(ctype)\n",
    "    \n",
    "    # Lists to store DataFrames for counts and patients\n",
    "    fc_list = []\n",
    "    patients_list = []\n",
    "    \n",
    "    for proj in all_projects:\n",
    "        if proj in tcga_projects and ctype in proj:\n",
    "            print(proj)\n",
    "            # Read the CSV files for TCGA projects\n",
    "            patients = pd.read_csv(os.path.join(\"/users/genomics/marta/TCGA_RNASeq\", proj, \"results/QC_patients1.csv\"))\n",
    "            patients['project'] = proj  # Add the project column\n",
    "            fc = pd.read_csv(os.path.join(cancer_dir, \"featureCounts\", f\"table_of_counts_TPMs_{proj}.csv\"))\n",
    "            fc_list.append(fc)\n",
    "            patients_list.append(patients)\n",
    "        \n",
    "        elif proj in other_projects and ctype in proj:\n",
    "            print(proj)\n",
    "            # Read the CSV files for other projects\n",
    "            patients = pd.read_csv(os.path.join(\"/users/genomics/marta/cancers_RNASeq\", proj, \"results/patients.csv\"))\n",
    "            patients['project'] = proj  # Add the project column\n",
    "            fc = pd.read_csv(os.path.join(cancer_dir, \"featureCounts\", f\"table_of_counts_TPMs_{proj}.csv\"))\n",
    "            fc_list.append(fc)\n",
    "            patients_list.append(patients)\n",
    "        \n",
    "        elif proj in manuscript_projects and ctype in proj:\n",
    "            print(proj)\n",
    "            # Read the CSV files for manuscript projects\n",
    "            if proj == \"LIHC_TCGA_LIHC\":\n",
    "                patients = pd.read_csv(os.path.join(\"/projects_eg/projects/marta\", proj[:-5], \"results/clean_patients.csv\"))\n",
    "                patients['normal'] = patients['patient'] + \"_normal\"\n",
    "                patients['tumor'] = patients['patient'] + \"_tumor\"\n",
    "\n",
    "            else:\n",
    "                patients = pd.read_csv(os.path.join(\"/projects_eg/projects/marta\", proj[:-5], \"results/clean_patients.csv\"))\n",
    "\n",
    "            patients['project'] = proj  # Add the project column\n",
    "            fc = pd.read_csv(os.path.join(cancer_dir, \"featureCounts\", f\"table_of_counts_TPMs_{proj}.csv\"))\n",
    "            fc_list.append(fc)\n",
    "            patients_list.append(patients)\n",
    "    \n",
    "    # Merge all fc DataFrames by 'transcript_id' and 'Length'\n",
    "    if fc_list:\n",
    "        merged_fc = reduce(lambda left, right: pd.merge(left, right, on=[\"transcript_id\", \"Length\"], how='outer'), fc_list)\n",
    "        # Save the merged fc DataFrame\n",
    "        merged_fc.to_csv(os.path.join(cancer_dir, f\"merged_fc_{ctype}.csv\"), index=False)\n",
    "    \n",
    "    # Combine all patient DataFrames into one big DataFrame\n",
    "    if patients_list:\n",
    "        merged_patients = pd.concat(patients_list, ignore_index=True)\n",
    "        # Save the merged patients DataFrame\n",
    "        merged_patients.to_csv(os.path.join(cancer_dir, f\"merged_patients_{ctype}.csv\"), index=False)\n"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Here to run Q3.1 - TPM distribution to know if we can treat them as one per cancer type or if there is bias according to the dataset"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Instead of selecting the tumor-specific, let's look first at the log2ratio between means\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Unnamed: 0</th>\n",
       "      <th>gene_id</th>\n",
       "      <th>gene_name</th>\n",
       "      <th>orfID</th>\n",
       "      <th>transcript_id</th>\n",
       "      <th>gene_type</th>\n",
       "      <th>orfType</th>\n",
       "      <th>length</th>\n",
       "      <th>geneORFtype</th>\n",
       "      <th>length_aa</th>\n",
       "      <th>start_codon</th>\n",
       "      <th>ORFpep</th>\n",
       "      <th>TranslatedLiver</th>\n",
       "      <th>TranslatedBrain</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>0</td>\n",
       "      <td>ENSG00000039600</td>\n",
       "      <td>SOX30</td>\n",
       "      <td>ENST00000265007.11:5:-|14|3283:361:2623|canoni...</td>\n",
       "      <td>ENST00000265007</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>canonical</td>\n",
       "      <td>2262</td>\n",
       "      <td>protein_coding_canonical</td>\n",
       "      <td>754</td>\n",
       "      <td>ATG</td>\n",
       "      <td>MERARPEPPPQPRPLRPAPPPLPVEGTSFWAAAMEPPPSSPTLSAA...</td>\n",
       "      <td>no</td>\n",
       "      <td>no</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>ENSG00000039600</td>\n",
       "      <td>SOX30</td>\n",
       "      <td>ENST00000265007.11:5:-|9|3283:233:491|ouORF|CTG</td>\n",
       "      <td>ENST00000265007</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>ouORF</td>\n",
       "      <td>258</td>\n",
       "      <td>protein_coding_ouORF</td>\n",
       "      <td>86</td>\n",
       "      <td>CTG</td>\n",
       "      <td>MRFEPRRVLGSKGLARLLTERGRGKVANKRLSSQSWPGRPPPPWRE...</td>\n",
       "      <td>no</td>\n",
       "      <td>no</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>2</td>\n",
       "      <td>ENSG00000046774</td>\n",
       "      <td>MAGEC2</td>\n",
       "      <td>ENST00000247452.4:X:-|1|1994:91:148|uORF|CTG</td>\n",
       "      <td>ENST00000247452</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>uORF</td>\n",
       "      <td>57</td>\n",
       "      <td>protein_coding_uORF</td>\n",
       "      <td>19</td>\n",
       "      <td>CTG</td>\n",
       "      <td>MASPQGEGPEEELRDLPP*</td>\n",
       "      <td>no</td>\n",
       "      <td>no</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>3</td>\n",
       "      <td>ENSG00000046774</td>\n",
       "      <td>MAGEC2</td>\n",
       "      <td>ENST00000247452.4:X:-|15|1994:349:1471|canonic...</td>\n",
       "      <td>ENST00000247452</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>canonical</td>\n",
       "      <td>1122</td>\n",
       "      <td>protein_coding_canonical</td>\n",
       "      <td>374</td>\n",
       "      <td>ATG</td>\n",
       "      <td>MPPVPGVPFRNVDNDSPTSVELEDWVDAQHPTDEEEEEASSASSTL...</td>\n",
       "      <td>no</td>\n",
       "      <td>no</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>4</td>\n",
       "      <td>ENSG00000046774</td>\n",
       "      <td>MAGEC2</td>\n",
       "      <td>ENST00000247452.4:X:-|2|1994:165:240|uORF|CTG</td>\n",
       "      <td>ENST00000247452</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>uORF</td>\n",
       "      <td>75</td>\n",
       "      <td>protein_coding_uORF</td>\n",
       "      <td>25</td>\n",
       "      <td>CTG</td>\n",
       "      <td>MYCAAVRLVLQEPGGDELGVRHTA*</td>\n",
       "      <td>no</td>\n",
       "      <td>no</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1029</th>\n",
       "      <td>1067</td>\n",
       "      <td>XLOC_002365</td>\n",
       "      <td>XLOC_002365</td>\n",
       "      <td>TCONS_00002556:9:-|365|10655:7423:7471|noncodi...</td>\n",
       "      <td>TCONS_00002556</td>\n",
       "      <td>novel</td>\n",
       "      <td>noncoding</td>\n",
       "      <td>48</td>\n",
       "      <td>novel_noncoding</td>\n",
       "      <td>16</td>\n",
       "      <td>ATG</td>\n",
       "      <td>MLPLGSVFPEPRRLR*</td>\n",
       "      <td>no</td>\n",
       "      <td>no</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1030</th>\n",
       "      <td>1068</td>\n",
       "      <td>XLOC_002365</td>\n",
       "      <td>XLOC_002365</td>\n",
       "      <td>TCONS_00002556:9:-|130|10655:3631:3685|noncodi...</td>\n",
       "      <td>TCONS_00002556</td>\n",
       "      <td>novel</td>\n",
       "      <td>noncoding</td>\n",
       "      <td>54</td>\n",
       "      <td>novel_noncoding</td>\n",
       "      <td>18</td>\n",
       "      <td>ATG</td>\n",
       "      <td>MNCFPNRIWIFHHHLTL*</td>\n",
       "      <td>no</td>\n",
       "      <td>no</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1031</th>\n",
       "      <td>1069</td>\n",
       "      <td>XLOC_002365</td>\n",
       "      <td>XLOC_002365</td>\n",
       "      <td>TCONS_00002556:9:-|498|10655:9144:9303|noncodi...</td>\n",
       "      <td>TCONS_00002556</td>\n",
       "      <td>novel</td>\n",
       "      <td>noncoding</td>\n",
       "      <td>159</td>\n",
       "      <td>novel_noncoding</td>\n",
       "      <td>53</td>\n",
       "      <td>ATG</td>\n",
       "      <td>MALPEGATHWRHLGLLSQMVFQQVGVGISLQRPGPAPEGGEVSLGL...</td>\n",
       "      <td>no</td>\n",
       "      <td>no</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1032</th>\n",
       "      <td>1070</td>\n",
       "      <td>XLOC_002366</td>\n",
       "      <td>XLOC_002366</td>\n",
       "      <td>TCONS_00002557:9:-|14|437:244:337|noncoding|CTG</td>\n",
       "      <td>TCONS_00002557</td>\n",
       "      <td>novel</td>\n",
       "      <td>noncoding</td>\n",
       "      <td>93</td>\n",
       "      <td>novel_noncoding</td>\n",
       "      <td>31</td>\n",
       "      <td>CTG</td>\n",
       "      <td>MKPIFRLEFLLLFFFLSYCVSPSPESTGVM*</td>\n",
       "      <td>no</td>\n",
       "      <td>no</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1033</th>\n",
       "      <td>1071</td>\n",
       "      <td>XLOC_002421</td>\n",
       "      <td>XLOC_002421</td>\n",
       "      <td>TCONS_00002655:X:-|6|569:39:201|noncoding|CTG</td>\n",
       "      <td>TCONS_00002655</td>\n",
       "      <td>novel</td>\n",
       "      <td>noncoding</td>\n",
       "      <td>162</td>\n",
       "      <td>novel_noncoding</td>\n",
       "      <td>54</td>\n",
       "      <td>CTG</td>\n",
       "      <td>MVSLLRFLKPSCYSVLATERDRCLTRDKTYFCLVAEYLHLAFQISP...</td>\n",
       "      <td>no</td>\n",
       "      <td>no</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1034 rows Ã— 14 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      Unnamed: 0          gene_id    gene_name  \\\n",
       "0              0  ENSG00000039600        SOX30   \n",
       "1              1  ENSG00000039600        SOX30   \n",
       "2              2  ENSG00000046774       MAGEC2   \n",
       "3              3  ENSG00000046774       MAGEC2   \n",
       "4              4  ENSG00000046774       MAGEC2   \n",
       "...          ...              ...          ...   \n",
       "1029        1067      XLOC_002365  XLOC_002365   \n",
       "1030        1068      XLOC_002365  XLOC_002365   \n",
       "1031        1069      XLOC_002365  XLOC_002365   \n",
       "1032        1070      XLOC_002366  XLOC_002366   \n",
       "1033        1071      XLOC_002421  XLOC_002421   \n",
       "\n",
       "                                                  orfID    transcript_id  \\\n",
       "0     ENST00000265007.11:5:-|14|3283:361:2623|canoni...  ENST00000265007   \n",
       "1       ENST00000265007.11:5:-|9|3283:233:491|ouORF|CTG  ENST00000265007   \n",
       "2          ENST00000247452.4:X:-|1|1994:91:148|uORF|CTG  ENST00000247452   \n",
       "3     ENST00000247452.4:X:-|15|1994:349:1471|canonic...  ENST00000247452   \n",
       "4         ENST00000247452.4:X:-|2|1994:165:240|uORF|CTG  ENST00000247452   \n",
       "...                                                 ...              ...   \n",
       "1029  TCONS_00002556:9:-|365|10655:7423:7471|noncodi...   TCONS_00002556   \n",
       "1030  TCONS_00002556:9:-|130|10655:3631:3685|noncodi...   TCONS_00002556   \n",
       "1031  TCONS_00002556:9:-|498|10655:9144:9303|noncodi...   TCONS_00002556   \n",
       "1032    TCONS_00002557:9:-|14|437:244:337|noncoding|CTG   TCONS_00002557   \n",
       "1033      TCONS_00002655:X:-|6|569:39:201|noncoding|CTG   TCONS_00002655   \n",
       "\n",
       "           gene_type    orfType  length               geneORFtype  length_aa  \\\n",
       "0     protein_coding  canonical    2262  protein_coding_canonical        754   \n",
       "1     protein_coding      ouORF     258      protein_coding_ouORF         86   \n",
       "2     protein_coding       uORF      57       protein_coding_uORF         19   \n",
       "3     protein_coding  canonical    1122  protein_coding_canonical        374   \n",
       "4     protein_coding       uORF      75       protein_coding_uORF         25   \n",
       "...              ...        ...     ...                       ...        ...   \n",
       "1029           novel  noncoding      48           novel_noncoding         16   \n",
       "1030           novel  noncoding      54           novel_noncoding         18   \n",
       "1031           novel  noncoding     159           novel_noncoding         53   \n",
       "1032           novel  noncoding      93           novel_noncoding         31   \n",
       "1033           novel  noncoding     162           novel_noncoding         54   \n",
       "\n",
       "     start_codon                                             ORFpep  \\\n",
       "0            ATG  MERARPEPPPQPRPLRPAPPPLPVEGTSFWAAAMEPPPSSPTLSAA...   \n",
       "1            CTG  MRFEPRRVLGSKGLARLLTERGRGKVANKRLSSQSWPGRPPPPWRE...   \n",
       "2            CTG                                MASPQGEGPEEELRDLPP*   \n",
       "3            ATG  MPPVPGVPFRNVDNDSPTSVELEDWVDAQHPTDEEEEEASSASSTL...   \n",
       "4            CTG                          MYCAAVRLVLQEPGGDELGVRHTA*   \n",
       "...          ...                                                ...   \n",
       "1029         ATG                                   MLPLGSVFPEPRRLR*   \n",
       "1030         ATG                                 MNCFPNRIWIFHHHLTL*   \n",
       "1031         ATG  MALPEGATHWRHLGLLSQMVFQQVGVGISLQRPGPAPEGGEVSLGL...   \n",
       "1032         CTG                    MKPIFRLEFLLLFFFLSYCVSPSPESTGVM*   \n",
       "1033         CTG  MVSLLRFLKPSCYSVLATERDRCLTRDKTYFCLVAEYLHLAFQISP...   \n",
       "\n",
       "     TranslatedLiver TranslatedBrain  \n",
       "0                 no              no  \n",
       "1                 no              no  \n",
       "2                 no              no  \n",
       "3                 no              no  \n",
       "4                 no              no  \n",
       "...              ...             ...  \n",
       "1029              no              no  \n",
       "1030              no              no  \n",
       "1031              no              no  \n",
       "1032              no              no  \n",
       "1033              no              no  \n",
       "\n",
       "[1034 rows x 14 columns]"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "testisRestr = pd.read_csv(\"/projects_eg/projects/marta/TestisRestricted_Microproteins_TSA/with_TranscriptomeReconstruction/Multimap_altORFs/Q2_TestisRestricted/human/testisRestricted_GTEx_translatedONLYtestis.noProteome.csv\")\n",
    "testisRestr"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "BRCA\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_1060059/3311690612.py:56: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  log2ratio3x_1TPM['ctype'] = ctype\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "BLCA\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_1060059/3311690612.py:56: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  log2ratio3x_1TPM['ctype'] = ctype\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "LUAD\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_1060059/3311690612.py:56: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  log2ratio3x_1TPM['ctype'] = ctype\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "KIRC\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_1060059/3311690612.py:56: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  log2ratio3x_1TPM['ctype'] = ctype\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PRAD\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_1060059/3311690612.py:56: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  log2ratio3x_1TPM['ctype'] = ctype\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "LUSC\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_1060059/3311690612.py:56: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  log2ratio3x_1TPM['ctype'] = ctype\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "LIHC\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_1060059/3311690612.py:56: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  log2ratio3x_1TPM['ctype'] = ctype\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "COAD\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_1060059/3311690612.py:56: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  log2ratio3x_1TPM['ctype'] = ctype\n"
     ]
    }
   ],
   "source": [
    "## log2ratio3x & 1 TPM\n",
    "log2ratio3x_general = pd.DataFrame(columns=[\"gene_name\",\"gene_id\",\"gene_type\"])\n",
    "log2ratio3x_1TPM_general = pd.DataFrame(columns=[\"gene_name\",\"gene_id\",\"gene_type\",\"ctype\"])\n",
    "log2ratio3x_1TPM_general_5percent = pd.DataFrame(columns=[\"gene_name\",\"gene_id\",\"gene_type\"])\n",
    "\n",
    "for ctype in cancertypes:\n",
    "\n",
    "    print(ctype)\n",
    "    tumorspecific_candidates = pd.DataFrame()\n",
    "\n",
    "    patients = pd.read_csv(cancer_dir+\"/merged_patients_\"+ctype+\".csv\")\n",
    "    patients_long = patients.melt(id_vars=['patient', 'project'], value_vars=['normal', 'tumor'], \n",
    "                    var_name='normal_tumor', value_name='sample')\n",
    "    patients_long['normal_tumor'] = patients_long['normal_tumor'].replace({'normal': 'normal', 'tumor': 'tumor'})\n",
    "    \n",
    "    fc=pd.read_csv(cancer_dir+\"/merged_fc_\"+ctype+\".csv\")\n",
    "\n",
    "    fc_info = fc.merge(transcript_gene, on=['transcript_id'], how=\"inner\")\n",
    "    fc_info = fc_info[fc_info['gene_type'].isin(['lncRNA','processed_pseudogene','novel','protein_coding'])]\n",
    "\n",
    "    fc_info_long = pd.melt(fc_info, id_vars=[\"transcript_id\",\"Length\", \"gene_id\", \"gene_type\", \"gene_name\"], ignore_index=False).reset_index()\n",
    "    fc_info_long.rename(columns={\"variable\":\"sample\",\"value\":\"TPM\"}, inplace=True)\n",
    "    fc_info_long = fc_info_long.merge(patients_long, on=[\"sample\"])\n",
    "\n",
    "    means = fc_info_long.groupby([\"gene_name\",\"gene_id\",\"gene_type\",\"normal_tumor\"])['TPM'].mean().reset_index()\n",
    "    means_pivot = means.pivot_table(index=['gene_name', 'gene_id', 'gene_type'], \n",
    "                          columns='normal_tumor', \n",
    "                          values='TPM').reset_index()\n",
    "\n",
    "    # Avoid division by zero by adding a small constant (epsilon)\n",
    "    epsilon = 1e-6  # A small value to prevent division by zero\n",
    "    means_pivot['tumor'] = pd.to_numeric(means_pivot['tumor'], errors='coerce')\n",
    "    means_pivot['normal'] = pd.to_numeric(means_pivot['normal'], errors='coerce')\n",
    "    means_pivot['log2ratio'] = np.log2((means_pivot['tumor'] + epsilon) / (means_pivot['normal'] + epsilon))\n",
    "\n",
    "    log2ratio3x = means_pivot[means_pivot['log2ratio'] >= np.log2(3)]\n",
    "    log2ratio3x_testisRestr = log2ratio3x[log2ratio3x['gene_id'].isin(testisRestr.gene_id.values.tolist())]\n",
    "\n",
    "    # try:\n",
    "    #     os.makedirs(os.path.join(cancer_dir,\"log2ratio3x/cancertypes\",ctype))\n",
    "    # except:\n",
    "    #     continue\n",
    "    log2ratio3x_testisRestr.to_csv(os.path.join(cancer_dir,\"log2ratio3x/cancertypes\",ctype,\"log2ratio3x.csv\"), index=None)\n",
    "    log2ratio3x_general = pd.concat([log2ratio3x_general, log2ratio3x_testisRestr[[\"gene_name\",\"gene_id\",\"gene_type\"]]])\n",
    "\n",
    "    ### > 1 TPM\n",
    "    max = fc_info_long.groupby([\"gene_name\",\"gene_id\",\"gene_type\",\"normal_tumor\"])['TPM'].max().reset_index()\n",
    "    ## tumor samples max has to be greater than 1TPM\n",
    "    max = max[max['normal_tumor'] == \"tumor\"]\n",
    "    max.rename(columns={'TPM':'max_TPM'}, inplace=True)\n",
    "\n",
    "    TPM1 = max[max['max_TPM'] > 1]\n",
    "    log2ratio3x_1TPM = log2ratio3x_testisRestr[log2ratio3x_testisRestr['gene_id'].isin(TPM1.gene_id.values.tolist())]\n",
    "    log2ratio3x_1TPM.to_csv(os.path.join(cancer_dir,\"log2ratio3x/cancertypes\",ctype,\"log2ratio3x_1TPM.csv\"), index=None)\n",
    "    log2ratio3x_1TPM['ctype'] = ctype\n",
    "    log2ratio3x_1TPM_general = pd.concat([log2ratio3x_1TPM_general, log2ratio3x_1TPM[[\"gene_name\",\"gene_id\",\"gene_type\",\"ctype\"]]])\n",
    "\n",
    "    ### > 1 TPM - 5%\n",
    "    # TAA_1TPM5percent = pd.read_csv(os.path.join(cancer_dir,\"tumorexpressed/cancertypes/tumor_1FPKM_n5percent_pancancer.csv\"))\n",
    "    \n",
    "    # log2ratio3x_1TPM = log2ratio3x_testisRestr[log2ratio3x_testisRestr['gene_id'].isin(TAA_1TPM5percent.gene_id.values.tolist())]\n",
    "    # log2ratio3x_1TPM.to_csv(os.path.join(cancer_dir,\"log2ratio3x/cancertypes\",ctype,\"log2ratio3x_1TPM5percent.csv\"), index=None)\n",
    "    # log2ratio3x_1TPM_general_5percent = pd.concat([log2ratio3x_1TPM_general_5percent, log2ratio3x_1TPM[[\"gene_name\",\"gene_id\",\"gene_type\"]]])\n",
    "\n",
    "log2ratio3x_1TPM_general.to_csv(os.path.join(cancer_dir,\"log2ratio3x/cancertypes/log2ratio3x_1TPM_genes.csv\"), index=None)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene_name</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>gene_type</th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>lncRNA</th>\n",
       "      <td>162</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>novel</th>\n",
       "      <td>91</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>processed_pseudogene</th>\n",
       "      <td>15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>protein_coding</th>\n",
       "      <td>555</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                      gene_name\n",
       "gene_type                      \n",
       "lncRNA                      162\n",
       "novel                        91\n",
       "processed_pseudogene         15\n",
       "protein_coding              555"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "log2ratio3x_1TPM_general.drop_duplicates(inplace=True)\n",
    "log2ratio3x_1TPM_general[['gene_name','gene_type']].groupby(\"gene_type\").count()\n",
    "# log2ratio3x_1TPM_general"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th>gene_name</th>\n",
       "      <th>length_aa</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ctype</th>\n",
       "      <th>geneORFtype</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th rowspan=\"8\" valign=\"top\">BLCA</th>\n",
       "      <th>lncRNA_noncoding</th>\n",
       "      <td>27</td>\n",
       "      <td>27</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>novel_noncoding</th>\n",
       "      <td>14</td>\n",
       "      <td>14</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>processed_pseudogene_noncoding</th>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>protein_coding_canonical</th>\n",
       "      <td>79</td>\n",
       "      <td>79</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>protein_coding_dORF</th>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>protein_coding_odORF</th>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>protein_coding_ouORF</th>\n",
       "      <td>9</td>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>protein_coding_uORF</th>\n",
       "      <td>14</td>\n",
       "      <td>14</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"6\" valign=\"top\">BRCA</th>\n",
       "      <th>lncRNA_noncoding</th>\n",
       "      <td>16</td>\n",
       "      <td>16</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>novel_noncoding</th>\n",
       "      <td>7</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>protein_coding_canonical</th>\n",
       "      <td>72</td>\n",
       "      <td>72</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>protein_coding_dORF</th>\n",
       "      <td>5</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>protein_coding_ouORF</th>\n",
       "      <td>9</td>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>protein_coding_uORF</th>\n",
       "      <td>12</td>\n",
       "      <td>12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"8\" valign=\"top\">COAD</th>\n",
       "      <th>lncRNA_noncoding</th>\n",
       "      <td>49</td>\n",
       "      <td>49</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>novel_noncoding</th>\n",
       "      <td>18</td>\n",
       "      <td>18</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>processed_pseudogene_noncoding</th>\n",
       "      <td>4</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>protein_coding_canonical</th>\n",
       "      <td>51</td>\n",
       "      <td>51</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>protein_coding_dORF</th>\n",
       "      <td>3</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>protein_coding_odORF</th>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>protein_coding_ouORF</th>\n",
       "      <td>7</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>protein_coding_uORF</th>\n",
       "      <td>13</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"6\" valign=\"top\">KIRC</th>\n",
       "      <th>lncRNA_noncoding</th>\n",
       "      <td>16</td>\n",
       "      <td>16</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>novel_noncoding</th>\n",
       "      <td>13</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>protein_coding_canonical</th>\n",
       "      <td>25</td>\n",
       "      <td>25</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>protein_coding_dORF</th>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>protein_coding_ouORF</th>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>protein_coding_uORF</th>\n",
       "      <td>4</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"8\" valign=\"top\">LIHC</th>\n",
       "      <th>lncRNA_noncoding</th>\n",
       "      <td>51</td>\n",
       "      <td>51</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>novel_noncoding</th>\n",
       "      <td>27</td>\n",
       "      <td>27</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>processed_pseudogene_noncoding</th>\n",
       "      <td>6</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>protein_coding_canonical</th>\n",
       "      <td>130</td>\n",
       "      <td>130</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>protein_coding_dORF</th>\n",
       "      <td>12</td>\n",
       "      <td>12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>protein_coding_odORF</th>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>protein_coding_ouORF</th>\n",
       "      <td>13</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>protein_coding_uORF</th>\n",
       "      <td>23</td>\n",
       "      <td>23</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"7\" valign=\"top\">LUAD</th>\n",
       "      <th>lncRNA_noncoding</th>\n",
       "      <td>27</td>\n",
       "      <td>27</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>novel_noncoding</th>\n",
       "      <td>13</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>processed_pseudogene_noncoding</th>\n",
       "      <td>3</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>protein_coding_canonical</th>\n",
       "      <td>68</td>\n",
       "      <td>68</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>protein_coding_dORF</th>\n",
       "      <td>4</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>protein_coding_ouORF</th>\n",
       "      <td>9</td>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>protein_coding_uORF</th>\n",
       "      <td>18</td>\n",
       "      <td>18</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"7\" valign=\"top\">LUSC</th>\n",
       "      <th>lncRNA_noncoding</th>\n",
       "      <td>27</td>\n",
       "      <td>27</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>novel_noncoding</th>\n",
       "      <td>25</td>\n",
       "      <td>25</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>processed_pseudogene_noncoding</th>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>protein_coding_canonical</th>\n",
       "      <td>92</td>\n",
       "      <td>92</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>protein_coding_dORF</th>\n",
       "      <td>3</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>protein_coding_ouORF</th>\n",
       "      <td>12</td>\n",
       "      <td>12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>protein_coding_uORF</th>\n",
       "      <td>15</td>\n",
       "      <td>15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"7\" valign=\"top\">PRAD</th>\n",
       "      <th>lncRNA_noncoding</th>\n",
       "      <td>9</td>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>novel_noncoding</th>\n",
       "      <td>5</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>processed_pseudogene_noncoding</th>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>protein_coding_canonical</th>\n",
       "      <td>22</td>\n",
       "      <td>22</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>protein_coding_dORF</th>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>protein_coding_ouORF</th>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>protein_coding_uORF</th>\n",
       "      <td>9</td>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                      gene_name  length_aa\n",
       "ctype geneORFtype                                         \n",
       "BLCA  lncRNA_noncoding                       27         27\n",
       "      novel_noncoding                        14         14\n",
       "      processed_pseudogene_noncoding          1          1\n",
       "      protein_coding_canonical               79         79\n",
       "      protein_coding_dORF                     1          1\n",
       "      protein_coding_odORF                    1          1\n",
       "      protein_coding_ouORF                    9          9\n",
       "      protein_coding_uORF                    14         14\n",
       "BRCA  lncRNA_noncoding                       16         16\n",
       "      novel_noncoding                         7          7\n",
       "      protein_coding_canonical               72         72\n",
       "      protein_coding_dORF                     5          5\n",
       "      protein_coding_ouORF                    9          9\n",
       "      protein_coding_uORF                    12         12\n",
       "COAD  lncRNA_noncoding                       49         49\n",
       "      novel_noncoding                        18         18\n",
       "      processed_pseudogene_noncoding          4          4\n",
       "      protein_coding_canonical               51         51\n",
       "      protein_coding_dORF                     3          3\n",
       "      protein_coding_odORF                    1          1\n",
       "      protein_coding_ouORF                    7          7\n",
       "      protein_coding_uORF                    13         13\n",
       "KIRC  lncRNA_noncoding                       16         16\n",
       "      novel_noncoding                        13         13\n",
       "      protein_coding_canonical               25         25\n",
       "      protein_coding_dORF                     1          1\n",
       "      protein_coding_ouORF                    1          1\n",
       "      protein_coding_uORF                     4          4\n",
       "LIHC  lncRNA_noncoding                       51         51\n",
       "      novel_noncoding                        27         27\n",
       "      processed_pseudogene_noncoding          6          6\n",
       "      protein_coding_canonical              130        130\n",
       "      protein_coding_dORF                    12         12\n",
       "      protein_coding_odORF                    1          1\n",
       "      protein_coding_ouORF                   13         13\n",
       "      protein_coding_uORF                    23         23\n",
       "LUAD  lncRNA_noncoding                       27         27\n",
       "      novel_noncoding                        13         13\n",
       "      processed_pseudogene_noncoding          3          3\n",
       "      protein_coding_canonical               68         68\n",
       "      protein_coding_dORF                     4          4\n",
       "      protein_coding_ouORF                    9          9\n",
       "      protein_coding_uORF                    18         18\n",
       "LUSC  lncRNA_noncoding                       27         27\n",
       "      novel_noncoding                        25         25\n",
       "      processed_pseudogene_noncoding          2          2\n",
       "      protein_coding_canonical               92         92\n",
       "      protein_coding_dORF                     3          3\n",
       "      protein_coding_ouORF                   12         12\n",
       "      protein_coding_uORF                    15         15\n",
       "PRAD  lncRNA_noncoding                        9          9\n",
       "      novel_noncoding                         5          5\n",
       "      processed_pseudogene_noncoding          1          1\n",
       "      protein_coding_canonical               22         22\n",
       "      protein_coding_dORF                     1          1\n",
       "      protein_coding_ouORF                    2          2\n",
       "      protein_coding_uORF                     9          9"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "## ORF level   \n",
    "log2ratio3x_1TPM_general = pd.read_csv(os.path.join(cancer_dir,\"log2ratio3x/cancertypes/log2ratio3x_1TPM_genes.csv\"))\n",
    "tumorReact_ORFs = log2ratio3x_1TPM_general.merge(testisRestr, on=[\"gene_name\",\"gene_id\",\"gene_type\"])\n",
    "tumorReact_ORFs = tumorReact_ORFs[['gene_name','gene_id','transcript_id','gene_type','length','ctype','orfType','geneORFtype','length_aa','start_codon','ORFpep']]\n",
    "tumorReact_ORFs.to_csv(os.path.join(cancer_dir,\"log2ratio3x/cancertypes/log2ratio3x_1TPM_ORFs.csv\"), index=None)\n",
    "tumorReact_ORFs.groupby(\"geneORFtype\").count()\n",
    "tumorReact_ORFs[['gene_name','length_aa','geneORFtype']].drop_duplicates().groupby(\"geneORFtype\").count()\n",
    "tumorReact_ORFs[['gene_name','length_aa','geneORFtype','ctype']].drop_duplicates().groupby([\"ctype\",\"geneORFtype\"]).count()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "## OVEREXPRESSED TUMOR 3X NORMAL - per cancer type - 5%\n",
    "tumorspecific_candidates_All = pd.DataFrame()\n",
    "patients_overexpressed_5percent_general = pd.DataFrame(columns=[\"gene_name\",\"gene_id\"])\n",
    "TSTR_all_5percent = pd.DataFrame(columns=[\"gene_name\",\"gene_id\"])\n",
    "for ctype in cancertypes:\n",
    "# for ctype in [\"BLCA\"]:\n",
    "    print(ctype)\n",
    "\n",
    "    tumorspecific_candidates_ctype = pd.DataFrame()\n",
    "\n",
    "    patients = pd.read_csv(cancer_dir+\"/merged_patients_\"+ctype+\".csv\")\n",
    "    patients_long = pd.melt(patients, id_vars=[\"patient\",\"project\"], ignore_index=False).reset_index()\n",
    "    patients_long.drop(\"index\", axis=1, inplace=True)\n",
    "    patients_long.rename(columns={\"variable\":\"normal_tumor\",\"value\":\"sample\"}, inplace=True)\n",
    "\n",
    "    fc=pd.read_csv(cancer_dir+\"/merged_fc_\"+ctype+\".csv\")\n",
    "\n",
    "    log2ratio_1TPM = pd.read_csv(os.path.join(cancer_dir,\"log2ratio3x/cancertypes\",ctype,\"log2ratio3x_1TPM.csv\"))\n",
    "    fc_info = fc.merge(transcript_gene, on=['transcript_id'], how=\"inner\")\n",
    "    fc_info = fc_info[fc_info['gene_type'].isin(['lncRNA','processed_pseudogene','novel','protein_coding'])]\n",
    "    # print(\"log2ratio 1TPM: \", len(fc_info))\n",
    "\n",
    "    fc_info_long = pd.melt(fc_info, id_vars=[\"transcript_id\",\"Length\", \"gene_id\", \"gene_type\", \"gene_name\"], ignore_index=False).reset_index()\n",
    "    fc_info_long.rename(columns={\"variable\":\"sample\",\"value\":\"TPM\"}, inplace=True)\n",
    "    fc_info_long = fc_info_long.merge(patients_long, on=[\"sample\"])\n",
    "\n",
    "    ## create full file with TPM per patient and ctype\n",
    "    fc_info_long['ctype'] = ctype\n",
    "\n",
    "    for index,patient in patients.iterrows():\n",
    "        log2ratio_1TPM_TOv3x = fc_info[[\"transcript_id\",\"gene_id\",\"gene_name\",\"gene_type\",\"Length\",patient['tumor'],patient['normal']]] \n",
    "        log2ratio_1TPM_TOv3x = log2ratio_1TPM_TOv3x[log2ratio_1TPM_TOv3x['gene_id'].isin(log2ratio_1TPM.gene_id.values.tolist())]\n",
    "        # Selecting the rows where the value of tumor is double that of normal sample\n",
    "        ## overexpressed\n",
    "        tumor1TPM_overexpressed = log2ratio_1TPM_TOv3x[log2ratio_1TPM_TOv3x.iloc[:, 5] >= 3 * log2ratio_1TPM_TOv3x.iloc[:, 6]]\n",
    "        # print(\"log2ratio 1TPM Overexpressed 3x: \", len(tumor1TPM_overexpressed))\n",
    "        try:\n",
    "            os.makedirs(os.path.join(cancer_dir,\"log2ratio3x/cancertypes\",ctype,patient['patient']))\n",
    "        except:\n",
    "            pass\n",
    "\n",
    "        tumor1TPM_overexpressed.to_csv(os.path.join(cancer_dir,\"log2ratio3x/cancertypes\",ctype,patient['patient'],\"tumor_3xnormal_TPMs.csv\"), index=None)\n",
    "        tumor1TPM_overexpressed['patient_overexpr'] = patient['patient']\n",
    "        tumor1TPM_overexpressed['ctype'] = ctype\n",
    "        tumorspecific_candidates_ctype = pd.concat([tumorspecific_candidates_ctype,tumor1TPM_overexpressed[[\"transcript_id\",\"gene_id\",\"gene_name\",\"gene_type\",\"Length\",\"patient_overexpr\",\"ctype\"]]])\n",
    "    \n",
    "        tumorspecific_candidates_All = pd.concat([tumorspecific_candidates_All,tumor1TPM_overexpressed[[\"transcript_id\",\"gene_id\",\"gene_name\",\"gene_type\",\"Length\",\"patient_overexpr\",\"ctype\"]]])\n",
    "\n",
    "    ## in how many patients of a specific ctype is the gene overexpressed?\n",
    "    patients_overexpressed = tumorspecific_candidates_ctype[['gene_id','transcript_id','gene_name','gene_type','Length','ctype']].groupby(['gene_id','transcript_id','gene_name','gene_type','Length','ctype']).size().reset_index(name='num_patients_overexpr')\n",
    "    patients_overexpressed['percentage_num_patients_overexpr'] = round(patients_overexpressed['num_patients_overexpr']/len(set(patients.patient.values.tolist()))*100,2)\n",
    "    # print(patients_overexpressed['percentage_num_patients_overexpr'].max())\n",
    "    patients_overexpressed.to_csv(os.path.join(cancer_dir,\"log2ratio3x/cancertypes\",ctype,\"tumor_3xnormal_TPMs.csv\"), index=None)\n",
    "    \n",
    "    patients_overexpressed_5percent = patients_overexpressed[patients_overexpressed['percentage_num_patients_overexpr'] >= 5]\n",
    "    patients_overexpressed_5percent.to_csv(os.path.join(cancer_dir,\"log2ratio3x/cancertypes\",ctype,\"tumor_3xnormal_TPMs_5percent.csv\"), index=None)\n",
    "    print(len(patients_overexpressed_5percent),\" genes expressed 3x in tumor than in normal (>1TPM) in at least 5% of the patients\")\n",
    "    patients_overexpressed_5percent_general = pd.concat([patients_overexpressed_5percent_general,patients_overexpressed_5percent])\n",
    "\n",
    "    # fc_info_long_5percent = fc_info_long[fc_info_long['gene_id'].isin(tumor1TPM_overexpressed.gene_id.values.tolist())]\n",
    "\n",
    "    # TSTR_all_5percent = pd.concat([TSTR_all_5percent, fc_info_long_5percent])\n",
    "\n",
    "patients_overexpressed_5percent_general.to_csv(os.path.join(cancer_dir,\"log2ratio3x/cancertypes/TOv3x_5percent_TestisRestrictedGTEx_Translated_Ctypes_log2ratio3xMEANgenes.csv\"))\n",
    "tumorspecific_candidates_5percent = tumorspecific_candidates_All.merge(patients_overexpressed_5percent_general[[\"ctype\",\"gene_id\",\"percentage_num_patients_overexpr\"]], on=[\"ctype\",\"gene_id\"])\n",
    "tumorspecific_candidates_5percent.to_csv(os.path.join(cancer_dir,\"log2ratio3x/cancertypes/TOv3x_5percent_TestisRestrictedGTEx_Translated_Ctypes_log2ratio3xMEANgenes_patients.csv\"))\n",
    "\n",
    "# TSTR_all_5percent.to_csv(os.path.join(cancer_dir,\"log2ratio3x/cancertypes/TSTR_Expression.csv\"))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "## get table of counts of candidates\n",
    "candidates = pd.read_csv(os.path.join(cancer_dir,\"log2ratio3x/cancertypes/TOv3x_5percent_TestisRestrictedGTEx_Translated_Ctypes_log2ratio3xMEANgenes_patients.csv\"))\n",
    "candidates\n",
    "\n",
    "full_table_of_counts = pd.DataFrame()\n",
    "for ctype in cancertypes:\n",
    "    patients = pd.read_csv(cancer_dir+\"/merged_patients_\"+ctype+\".csv\")\n",
    "    patients_long = pd.melt(patients, id_vars=[\"patient\",\"project\"], ignore_index=False).reset_index()\n",
    "    patients_long.drop(\"index\", axis=1, inplace=True)\n",
    "    patients_long.rename(columns={\"variable\":\"normal_tumor\",\"value\":\"sample\"}, inplace=True)\n",
    "\n",
    "    fc=pd.read_csv(cancer_dir+\"/merged_fc_\"+ctype+\".csv\")\n",
    "\n",
    "    log2ratio_1TPM = pd.read_csv(os.path.join(cancer_dir,\"log2ratio3x/cancertypes\",ctype,\"log2ratio3x_1TPM.csv\"))\n",
    "    fc_info = fc.merge(transcript_gene, on=['transcript_id'], how=\"inner\")\n",
    "    fc_info = fc_info[fc_info['gene_type'].isin(['lncRNA','processed_pseudogene','novel','protein_coding'])]\n",
    "    # print(\"log2ratio 1TPM: \", len(fc_info))\n",
    "\n",
    "    fc_info_long = pd.melt(fc_info, id_vars=[\"transcript_id\",\"Length\", \"gene_id\", \"gene_type\", \"gene_name\"], ignore_index=False).reset_index()\n",
    "    fc_info_long.rename(columns={\"variable\":\"sample\",\"value\":\"TPM\"}, inplace=True)\n",
    "    fc_info_long = fc_info_long.merge(patients_long, on=[\"sample\"])\n",
    "\n",
    "    ## create full file with TPM per patient and ctype\n",
    "    fc_info_long['ctype'] = ctype\n",
    "\n",
    "    full_table_of_counts = pd.concat([full_table_of_counts, fc_info_long])\n",
    "\n",
    "# full_table_of_counts.to_csv(os.path.join(cancer_dir,\"full_table_of_counts.csv\"))\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "merged = pd.merge(\n",
    "    full_table_of_counts,\n",
    "    candidates[['gene_id', 'patient_overexpr']],  # Only keep relevant columns for the merge\n",
    "    left_on=['gene_id', 'patient'],\n",
    "    right_on=['gene_id', 'patient_overexpr'],\n",
    "    how='left'\n",
    ")\n",
    "merged = merged[merged['gene_id'].isin(candidates.gene_id.values.tolist())]\n",
    "merged.to_csv(os.path.join(cancer_dir,\"log2ratio3x/cancertypes/TSTR_Expression.csv\"), index=False)\n",
    "\n",
    "# Create the new column 'forced_TPM' using numpy.where\n",
    "merged['forced_TPM'] = np.where(merged['patient_overexpr'].notna(), merged['TPM'], 0)\n",
    "merged.drop(columns=['patient_overexpr','index','Length','sample','TPM'], inplace=True)\n",
    "merged_tumor = merged[merged['normal_tumor'] == \"tumor\"]\n",
    "merged_tumor.to_csv(os.path.join(cancer_dir,\"log2ratio3x/cancertypes/TSTR_Expression_tumor.csv\"), index=False)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene_name</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>gene_type</th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>lncRNA</th>\n",
       "      <td>95</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>novel</th>\n",
       "      <td>46</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>processed_pseudogene</th>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>protein_coding</th>\n",
       "      <td>202</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                      gene_name\n",
       "gene_type                      \n",
       "lncRNA                       95\n",
       "novel                        46\n",
       "processed_pseudogene          7\n",
       "protein_coding              202"
      ]
     },
     "execution_count": 46,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "patients_overexpressed_5percent_general[['gene_name','gene_type']].drop_duplicates().groupby(\"gene_type\").count()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th>gene_name</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>gene_type</th>\n",
       "      <th>ctype</th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th rowspan=\"8\" valign=\"top\">lncRNA</th>\n",
       "      <th>BLCA</th>\n",
       "      <td>17</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>BRCA</th>\n",
       "      <td>12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>COAD</th>\n",
       "      <td>37</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>KIRC</th>\n",
       "      <td>12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>LIHC</th>\n",
       "      <td>38</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>LUAD</th>\n",
       "      <td>22</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>LUSC</th>\n",
       "      <td>16</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PRAD</th>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"8\" valign=\"top\">novel</th>\n",
       "      <th>BLCA</th>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>BRCA</th>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>COAD</th>\n",
       "      <td>15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>KIRC</th>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>LIHC</th>\n",
       "      <td>22</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>LUAD</th>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>LUSC</th>\n",
       "      <td>16</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PRAD</th>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"6\" valign=\"top\">processed_pseudogene</th>\n",
       "      <th>BLCA</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>COAD</th>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>LIHC</th>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>LUAD</th>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>LUSC</th>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PRAD</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"8\" valign=\"top\">protein_coding</th>\n",
       "      <th>BLCA</th>\n",
       "      <td>81</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>BRCA</th>\n",
       "      <td>75</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>COAD</th>\n",
       "      <td>52</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>KIRC</th>\n",
       "      <td>25</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>LIHC</th>\n",
       "      <td>134</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>LUAD</th>\n",
       "      <td>71</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>LUSC</th>\n",
       "      <td>94</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PRAD</th>\n",
       "      <td>23</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                            gene_name\n",
       "gene_type            ctype           \n",
       "lncRNA               BLCA          17\n",
       "                     BRCA          12\n",
       "                     COAD          37\n",
       "                     KIRC          12\n",
       "                     LIHC          38\n",
       "                     LUAD          22\n",
       "                     LUSC          16\n",
       "                     PRAD           8\n",
       "novel                BLCA           8\n",
       "                     BRCA           6\n",
       "                     COAD          15\n",
       "                     KIRC          10\n",
       "                     LIHC          22\n",
       "                     LUAD          10\n",
       "                     LUSC          16\n",
       "                     PRAD           4\n",
       "processed_pseudogene BLCA           1\n",
       "                     COAD           3\n",
       "                     LIHC           5\n",
       "                     LUAD           3\n",
       "                     LUSC           2\n",
       "                     PRAD           1\n",
       "protein_coding       BLCA          81\n",
       "                     BRCA          75\n",
       "                     COAD          52\n",
       "                     KIRC          25\n",
       "                     LIHC         134\n",
       "                     LUAD          71\n",
       "                     LUSC          94\n",
       "                     PRAD          23"
      ]
     },
     "execution_count": 33,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "patients_overexpressed_5percent_general[['gene_name','gene_type','ctype']].drop_duplicates().groupby([\"gene_type\",\"ctype\"]).count()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene_name</th>\n",
       "      <th>gene_id</th>\n",
       "      <th>transcript_id</th>\n",
       "      <th>gene_type</th>\n",
       "      <th>length</th>\n",
       "      <th>num_patients_overexpr</th>\n",
       "      <th>percentage_num_patients_overexpr</th>\n",
       "      <th>ctype</th>\n",
       "      <th>orfType</th>\n",
       "      <th>geneORFtype</th>\n",
       "      <th>length_aa</th>\n",
       "      <th>start_codon</th>\n",
       "      <th>ORFpep</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>MAGEC2</td>\n",
       "      <td>ENSG00000046774</td>\n",
       "      <td>ENST00000247452</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>57</td>\n",
       "      <td>101.0</td>\n",
       "      <td>92.66</td>\n",
       "      <td>BRCA</td>\n",
       "      <td>uORF</td>\n",
       "      <td>protein_coding_uORF</td>\n",
       "      <td>19</td>\n",
       "      <td>CTG</td>\n",
       "      <td>MASPQGEGPEEELRDLPP*</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>MAGEC2</td>\n",
       "      <td>ENSG00000046774</td>\n",
       "      <td>ENST00000247452</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>1122</td>\n",
       "      <td>101.0</td>\n",
       "      <td>92.66</td>\n",
       "      <td>BRCA</td>\n",
       "      <td>canonical</td>\n",
       "      <td>protein_coding_canonical</td>\n",
       "      <td>374</td>\n",
       "      <td>ATG</td>\n",
       "      <td>MPPVPGVPFRNVDNDSPTSVELEDWVDAQHPTDEEEEEASSASSTL...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>MAGEC2</td>\n",
       "      <td>ENSG00000046774</td>\n",
       "      <td>ENST00000247452</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>75</td>\n",
       "      <td>101.0</td>\n",
       "      <td>92.66</td>\n",
       "      <td>BRCA</td>\n",
       "      <td>uORF</td>\n",
       "      <td>protein_coding_uORF</td>\n",
       "      <td>25</td>\n",
       "      <td>CTG</td>\n",
       "      <td>MYCAAVRLVLQEPGGDELGVRHTA*</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>MAGEC2</td>\n",
       "      <td>ENSG00000046774</td>\n",
       "      <td>ENST00000247452</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>54</td>\n",
       "      <td>101.0</td>\n",
       "      <td>92.66</td>\n",
       "      <td>BRCA</td>\n",
       "      <td>uORF</td>\n",
       "      <td>protein_coding_uORF</td>\n",
       "      <td>18</td>\n",
       "      <td>CTG</td>\n",
       "      <td>MLDLIIHIPVDTFTCCS*</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>MAGEC2</td>\n",
       "      <td>ENSG00000046774</td>\n",
       "      <td>ENST00000247452</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>57</td>\n",
       "      <td>33.0</td>\n",
       "      <td>86.84</td>\n",
       "      <td>BLCA</td>\n",
       "      <td>uORF</td>\n",
       "      <td>protein_coding_uORF</td>\n",
       "      <td>19</td>\n",
       "      <td>CTG</td>\n",
       "      <td>MASPQGEGPEEELRDLPP*</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1098</th>\n",
       "      <td>XLOC_001819</td>\n",
       "      <td>XLOC_001819</td>\n",
       "      <td>TCONS_00002062</td>\n",
       "      <td>novel</td>\n",
       "      <td>711</td>\n",
       "      <td>105.0</td>\n",
       "      <td>72.92</td>\n",
       "      <td>COAD</td>\n",
       "      <td>noncoding</td>\n",
       "      <td>novel_noncoding</td>\n",
       "      <td>237</td>\n",
       "      <td>ATG</td>\n",
       "      <td>MSSPPGLQLHGPQIGTSAPHQISSCTALKLEYQLHPGAPGAQPSNC...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1099</th>\n",
       "      <td>XLOC_001820</td>\n",
       "      <td>XLOC_001820</td>\n",
       "      <td>TCONS_00002063</td>\n",
       "      <td>novel</td>\n",
       "      <td>48</td>\n",
       "      <td>59.0</td>\n",
       "      <td>40.97</td>\n",
       "      <td>COAD</td>\n",
       "      <td>noncoding</td>\n",
       "      <td>novel_noncoding</td>\n",
       "      <td>16</td>\n",
       "      <td>ATG</td>\n",
       "      <td>MFQSEGCAAGYPVGS*</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1100</th>\n",
       "      <td>XLOC_002017</td>\n",
       "      <td>XLOC_002017</td>\n",
       "      <td>TCONS_00002162</td>\n",
       "      <td>novel</td>\n",
       "      <td>51</td>\n",
       "      <td>128.0</td>\n",
       "      <td>88.89</td>\n",
       "      <td>COAD</td>\n",
       "      <td>noncoding</td>\n",
       "      <td>novel_noncoding</td>\n",
       "      <td>17</td>\n",
       "      <td>GTG</td>\n",
       "      <td>VSTKHLSKYGLGRLRD*</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1101</th>\n",
       "      <td>XLOC_002017</td>\n",
       "      <td>XLOC_002017</td>\n",
       "      <td>TCONS_00002162</td>\n",
       "      <td>novel</td>\n",
       "      <td>294</td>\n",
       "      <td>128.0</td>\n",
       "      <td>88.89</td>\n",
       "      <td>COAD</td>\n",
       "      <td>noncoding</td>\n",
       "      <td>novel_noncoding</td>\n",
       "      <td>98</td>\n",
       "      <td>ATG</td>\n",
       "      <td>MIPELVVSSVDEVKKESLENYARASHCLSPQVTHLFPSQPTLAGTN...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1102</th>\n",
       "      <td>XLOC_002151</td>\n",
       "      <td>XLOC_002151</td>\n",
       "      <td>TCONS_00002326</td>\n",
       "      <td>novel</td>\n",
       "      <td>60</td>\n",
       "      <td>129.0</td>\n",
       "      <td>89.58</td>\n",
       "      <td>COAD</td>\n",
       "      <td>noncoding</td>\n",
       "      <td>novel_noncoding</td>\n",
       "      <td>20</td>\n",
       "      <td>ATG</td>\n",
       "      <td>MKQDCREKQRQEMDLPISS*</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1103 rows Ã— 13 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        gene_name          gene_id    transcript_id       gene_type  length  \\\n",
       "0          MAGEC2  ENSG00000046774  ENST00000247452  protein_coding      57   \n",
       "1          MAGEC2  ENSG00000046774  ENST00000247452  protein_coding    1122   \n",
       "2          MAGEC2  ENSG00000046774  ENST00000247452  protein_coding      75   \n",
       "3          MAGEC2  ENSG00000046774  ENST00000247452  protein_coding      54   \n",
       "4          MAGEC2  ENSG00000046774  ENST00000247452  protein_coding      57   \n",
       "...           ...              ...              ...             ...     ...   \n",
       "1098  XLOC_001819      XLOC_001819   TCONS_00002062           novel     711   \n",
       "1099  XLOC_001820      XLOC_001820   TCONS_00002063           novel      48   \n",
       "1100  XLOC_002017      XLOC_002017   TCONS_00002162           novel      51   \n",
       "1101  XLOC_002017      XLOC_002017   TCONS_00002162           novel     294   \n",
       "1102  XLOC_002151      XLOC_002151   TCONS_00002326           novel      60   \n",
       "\n",
       "      num_patients_overexpr  percentage_num_patients_overexpr ctype  \\\n",
       "0                     101.0                             92.66  BRCA   \n",
       "1                     101.0                             92.66  BRCA   \n",
       "2                     101.0                             92.66  BRCA   \n",
       "3                     101.0                             92.66  BRCA   \n",
       "4                      33.0                             86.84  BLCA   \n",
       "...                     ...                               ...   ...   \n",
       "1098                  105.0                             72.92  COAD   \n",
       "1099                   59.0                             40.97  COAD   \n",
       "1100                  128.0                             88.89  COAD   \n",
       "1101                  128.0                             88.89  COAD   \n",
       "1102                  129.0                             89.58  COAD   \n",
       "\n",
       "        orfType               geneORFtype  length_aa start_codon  \\\n",
       "0          uORF       protein_coding_uORF         19         CTG   \n",
       "1     canonical  protein_coding_canonical        374         ATG   \n",
       "2          uORF       protein_coding_uORF         25         CTG   \n",
       "3          uORF       protein_coding_uORF         18         CTG   \n",
       "4          uORF       protein_coding_uORF         19         CTG   \n",
       "...         ...                       ...        ...         ...   \n",
       "1098  noncoding           novel_noncoding        237         ATG   \n",
       "1099  noncoding           novel_noncoding         16         ATG   \n",
       "1100  noncoding           novel_noncoding         17         GTG   \n",
       "1101  noncoding           novel_noncoding         98         ATG   \n",
       "1102  noncoding           novel_noncoding         20         ATG   \n",
       "\n",
       "                                                 ORFpep  \n",
       "0                                   MASPQGEGPEEELRDLPP*  \n",
       "1     MPPVPGVPFRNVDNDSPTSVELEDWVDAQHPTDEEEEEASSASSTL...  \n",
       "2                             MYCAAVRLVLQEPGGDELGVRHTA*  \n",
       "3                                    MLDLIIHIPVDTFTCCS*  \n",
       "4                                   MASPQGEGPEEELRDLPP*  \n",
       "...                                                 ...  \n",
       "1098  MSSPPGLQLHGPQIGTSAPHQISSCTALKLEYQLHPGAPGAQPSNC...  \n",
       "1099                                   MFQSEGCAAGYPVGS*  \n",
       "1100                                  VSTKHLSKYGLGRLRD*  \n",
       "1101  MIPELVVSSVDEVKKESLENYARASHCLSPQVTHLFPSQPTLAGTN...  \n",
       "1102                               MKQDCREKQRQEMDLPISS*  \n",
       "\n",
       "[1103 rows x 13 columns]"
      ]
     },
     "execution_count": 47,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "## ORF level   \n",
    "patients_overexpressed_5percent_general = pd.read_csv(os.path.join(cancer_dir,\"log2ratio3x/cancertypes/TOv3x_5percent_TestisRestrictedGTEx_Translated_Ctypes_log2ratio3xMEANgenes.csv\"))\n",
    "tumorReact_ORFs = patients_overexpressed_5percent_general.merge(testisRestr, on=[\"gene_name\",\"gene_id\",\"gene_type\",\"transcript_id\"])\n",
    "tumorReact_ORFs = tumorReact_ORFs[['gene_name','gene_id','transcript_id','gene_type','length','num_patients_overexpr','percentage_num_patients_overexpr','ctype','orfType','geneORFtype','length_aa','start_codon','ORFpep']]\n",
    "tumorReact_ORFs.to_csv(os.path.join(cancer_dir,\"log2ratio3x/cancertypes/TOv3x_5percent_TestisRestrictedGTEx_Translated_Ctypes_log2ratio3xMEAN.csv\"), index=None)\n",
    "tumorReact_ORFs.groupby(\"gene_type\").count()\n",
    "\n",
    "tumorReact_ORFs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene_name</th>\n",
       "      <th>length_aa</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>geneORFtype</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>lncRNA_noncoding</th>\n",
       "      <td>126</td>\n",
       "      <td>126</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>novel_noncoding</th>\n",
       "      <td>61</td>\n",
       "      <td>61</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>processed_pseudogene_noncoding</th>\n",
       "      <td>8</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>protein_coding_canonical</th>\n",
       "      <td>193</td>\n",
       "      <td>193</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>protein_coding_dORF</th>\n",
       "      <td>12</td>\n",
       "      <td>12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>protein_coding_odORF</th>\n",
       "      <td>3</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>protein_coding_ouORF</th>\n",
       "      <td>18</td>\n",
       "      <td>18</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>protein_coding_uORF</th>\n",
       "      <td>39</td>\n",
       "      <td>39</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                gene_name  length_aa\n",
       "geneORFtype                                         \n",
       "lncRNA_noncoding                      126        126\n",
       "novel_noncoding                        61         61\n",
       "processed_pseudogene_noncoding          8          8\n",
       "protein_coding_canonical              193        193\n",
       "protein_coding_dORF                    12         12\n",
       "protein_coding_odORF                    3          3\n",
       "protein_coding_ouORF                   18         18\n",
       "protein_coding_uORF                    39         39"
      ]
     },
     "execution_count": 48,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "tumorReact_ORFs[['gene_name','length_aa','geneORFtype']].drop_duplicates().groupby(\"geneORFtype\").count()\n",
    "# tumorReact_ORFs[['gene_name','length_aa','geneORFtype','ctype']].drop_duplicates().groupby([\"ctype\",\"geneORFtype\"]).count()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'ADAD1',\n",
       " 'ADAM29',\n",
       " 'C12orf50',\n",
       " 'CCDC172',\n",
       " 'COX7B2',\n",
       " 'DCAF8L2',\n",
       " 'DYDC1',\n",
       " 'FSHR',\n",
       " 'IQCM',\n",
       " 'LRRC74A',\n",
       " 'LYPD4',\n",
       " 'MAGEA1',\n",
       " 'MAGEB1',\n",
       " 'MAGEB3',\n",
       " 'MAGEC1',\n",
       " 'MAGEC2',\n",
       " 'MEIOSIN',\n",
       " 'OR14A2',\n",
       " 'PLCZ1',\n",
       " 'PRSS54',\n",
       " 'RFPL4B',\n",
       " 'RHOXF2B',\n",
       " 'SMIM47',\n",
       " 'SPATA12',\n",
       " 'SPMIP9',\n",
       " 'TEX13C',\n",
       " 'TPTE',\n",
       " 'TRPC5OS',\n",
       " 'USP26'}"
      ]
     },
     "execution_count": 37,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "set(tumorReact_ORFs[tumorReact_ORFs['geneORFtype'] == \"protein_coding_uORF\"].gene_name.values.tolist())"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Generate bed/gtf file for candidates (TSTR) to look for chromatin acessibility (Cova)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "TSTR = pd.read_csv(os.path.join(cancer_dir,\"log2ratio3x/cancertypes/TOv3x_5percent_TestisRestrictedGTEx_Translated_Ctypes_log2ratio3xMEANgenes.csv\"))\n",
    "TSTR\n",
    "print(len(set(TSTR.transcript_id.values.tolist())))\n",
    "\n",
    "annotation_gtf = pd.read_csv(annotation, sep=\"\\t\", comment=\"#\", header=None)\n",
    "annotation_gtf = annotation_gtf[annotation_gtf[2] == \"transcript\"]\n",
    "annotation_gtf\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Convert the list of values into a regex pattern\n",
    "pattern = '|'.join(TSTR.transcript_id.values.tolist())\n",
    "\n",
    "# Filter the DataFrame\n",
    "filtered_annotation = annotation_gtf[annotation_gtf[8].str.contains(pattern)]\n",
    "\n",
    "filtered_annotation.to_csv(os.path.join(cancer_dir,\"log2ratio3x/cancertypes/TSTR_annotation.gtf\"), sep=\"\\t\", index=None, header=None, quoting=csv.QUOTE_NONE)\n"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.10"
  },
  "orig_nbformat": 4
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
