{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import os,re,glob,csv\n",
    "import pandas as pd\n",
    "import numpy as np\n",
    "from collections import Counter\n",
    "from rna_seq_normalization import Normalization as Norm\n",
    "from functools import reduce\n",
    "from Bio.Seq import Seq\n",
    "from Bio.SeqRecord import SeqRecord\n",
    "from Bio import SeqIO\n",
    "\n",
    "users_dir = \"/users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47\"\n",
    "specie = \"human\"\n",
    "# GENOMEDIR = \"/genomics/users/marta/genomes\"\n",
    "GENOMEDIR = \"/data/genomics/marta/genomes\"\n",
    "\n",
    "### cancer data\n",
    "cancer_dir = users_dir + \"/cancers\"\n",
    "raw_cancer_dir=\"/users/genomics/marta/TCGA_RNASeq\"\n",
    "tcga_projects=[\"TCGA-BRCA\",\"TCGA-LUSC\",\"TCGA-PRAD\",\"TCGA-KIRC\",\"TCGA-LUAD\",\"TCGA-BLCA\"]#,\"TCGA-LIHC\"]\n",
    "other_projects=[\"GSE102101_KIRC\",\"GSE133624_BLCA\",\"GSE22260_PRAD\",\"PRJEB2449_PRAD\",\"SRP238334_KIRC\",\"GSE214846_LIHC\",\"GSE229705_LUAD\",\"SRP107326_COAD\",\"TCGA_COAD\"]\n",
    "# deleted=[\"GSE103001_BRCA\",\"GSE89223_PRAD\"]\n",
    "manuscript_projects = [\"liver_adjacent_totalRNA_LIHC\",\"hcc_normal_totalRNA_LIHC\",\"GSE193567_LIHC\",\"LIHC_TCGA_LIHC\"]\n",
    "all_projects = tcga_projects + other_projects + manuscript_projects\n",
    "\n",
    "cancertypes = [\"BRCA\",\"BLCA\",\"LUAD\",\"KIRC\",\"PRAD\",\"LUSC\",\"LIHC\",\"COAD\"]\n",
    "## annotation file\n",
    "annotation=\"/users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/human/newReference_Resconstructed/gencode.v47.gffcompare.TestisLiverBrain.annotation.sorted.1transcript.sorted.NOchr.gtf\"\n",
    "transcript_gene=pd.read_csv(\"/users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/human/newReference_Resconstructed/transID_geneID_isoforms_selected.1to1.csv\")\n",
    "\n",
    "def count_greater_than_one(row):\n",
    "    return (row > 1).sum()"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Quantify with TCGAData and new reference\n",
    "`featureCounts_newRef_cancer.sh`\n",
    "\n",
    "output in: `/users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/cancers/featureCounts`"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## TPM"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "TCGA-BRCA\n",
      "GSE102101_KIRC\n",
      "TCGA-LUSC\n",
      "TCGA-PRAD\n",
      "TCGA-BLCA\n",
      "GSE229705_LUAD\n",
      "PRJEB2449_PRAD\n",
      "GSE133624_BLCA\n",
      "TCGA_COAD_SE\n",
      "GSE22260_PRAD\n",
      "GSE214846_LIHC\n",
      "TCGA_COAD_PE\n",
      "liver_adjacent_totalRNA_LIHC\n",
      "TCGA-KIRC\n",
      "hcc_normal_totalRNA_LIHC\n",
      "SRP107326_COAD\n",
      "TCGA-LUAD\n",
      "GSE193567_LIHC\n",
      "SRP238334_KIRC\n",
      "LIHC_TCGA_LIHC\n"
     ]
    }
   ],
   "source": [
    "for file in os.listdir(os.path.join(cancer_dir,\"featureCounts\")):\n",
    "    if file.endswith(\".txt\"):\n",
    "\n",
    "        proj = file.split(\"featureCounts_\")[-1]\n",
    "        cancer_type = proj[:-4]\n",
    "        print(cancer_type)\n",
    "        toc = pd.read_csv(os.path.join(cancer_dir,\"featureCounts\",file), sep=\"\\t\", comment=\"#\")\n",
    "        toc = toc[toc['Geneid'].str.contains('PAR_')==False]\n",
    "        toc['Geneid']=toc['Geneid'].str.split('.').str[0]\n",
    "        toc.rename(columns={'Geneid':'transcript_id'}, inplace=True)\n",
    "\n",
    "        filter_col = [col for col in toc if col.startswith('/')]\n",
    "        for col in filter_col:\n",
    "            new_col=col.split(\"Aligned\")[0]\n",
    "            new_col=new_col.split(\"/\")[-1]\n",
    "            if proj in tcga_projects:\n",
    "                new_col=cancer_type+new_col[4:]\n",
    "            toc.rename(columns={col:new_col}, inplace=True)\n",
    "\n",
    "        length = toc['Length']\n",
    "        genes = toc['transcript_id']\n",
    "        # we are only interested in the columns with counts\n",
    "        counts = toc\n",
    "        counts.drop([\"Chr\",\"Start\",\"End\",\"Strand\",\"Length\",\"transcript_id\"],axis=1, inplace=True)\n",
    "        # calculate TPMs\n",
    "        tpm_df = Norm.tpm(counts, length)\n",
    "        # add transcript_id and length again\n",
    "        tpms = pd.concat([genes,tpm_df, length], axis=1)\n",
    "        tpms.to_csv(os.path.join(cancer_dir,\"featureCounts/table_of_counts_TPMs_\")+cancer_type+\".csv\", index=None)\n",
    " "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "19\n",
      "66\n",
      "83\n"
     ]
    }
   ],
   "source": [
    "## merge single-end and paired-end TCGA COAD\n",
    "SE_COAD = pd.read_csv(os.path.join(cancer_dir,\"featureCounts/table_of_counts_TPMs_TCGA_COAD_SE.csv\"))\n",
    "print(len(SE_COAD.columns))\n",
    "PE_COAD = pd.read_csv(os.path.join(cancer_dir,\"featureCounts/table_of_counts_TPMs_TCGA_COAD_PE.csv\"))\n",
    "print(len(PE_COAD.columns))\n",
    "\n",
    "COAD = SE_COAD.merge(PE_COAD, on=[\"transcript_id\",\"Length\"])\n",
    "COAD.to_csv(os.path.join(cancer_dir,\"featureCounts/table_of_counts_TPMs_TCGA_COAD.csv\"), index=None)\n",
    "print(len(COAD.columns))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "TCGA-BRCA\n",
      "109  patients\n",
      "lncRNA:  15803\n",
      "PROTEIN CODING:  17313\n",
      "NOVEL:  304\n",
      "TCGA-LUSC\n",
      "49  patients\n",
      "lncRNA:  14578\n",
      "PROTEIN CODING:  17328\n",
      "NOVEL:  271\n",
      "TCGA-PRAD\n",
      "52  patients\n",
      "lncRNA:  10409\n",
      "PROTEIN CODING:  15949\n",
      "NOVEL:  235\n",
      "TCGA-KIRC\n",
      "71  patients\n",
      "lncRNA:  11415\n",
      "PROTEIN CODING:  16419\n",
      "NOVEL:  250\n",
      "TCGA-LUAD\n",
      "56  patients\n",
      "lncRNA:  14375\n",
      "PROTEIN CODING:  17107\n",
      "NOVEL:  278\n",
      "TCGA-BLCA\n",
      "18  patients\n",
      "lncRNA:  11285\n",
      "PROTEIN CODING:  16530\n",
      "NOVEL:  222\n",
      "GSE102101_KIRC\n",
      "         transcript_id   SRR5885319   SRR5885320   SRR5885321   SRR5885322  \\\n",
      "0       TCONS_00001163   830.787897   202.965330   666.785038   360.316898   \n",
      "1      ENST00000387314  2477.853965   272.093328  1653.609499   313.237274   \n",
      "2      ENST00000389680  4034.349225   855.853788  3014.283347   703.347008   \n",
      "3       TCONS_00001164  7730.138459  1868.191546  5779.664200  1457.492971   \n",
      "4      ENST00000387342  1515.029280    43.292633  1467.940510   140.294062   \n",
      "...                ...          ...          ...          ...          ...   \n",
      "79618  ENST00000797680     0.000000     0.000000     0.000000     0.000000   \n",
      "79619  ENST00000797955     0.000000     0.000000     0.000000     0.015446   \n",
      "79620  ENST00000819228     0.000000     0.000000     0.000000     0.000000   \n",
      "79621  ENST00000751368     0.000000     0.000000     0.000000     0.000000   \n",
      "79622  ENST00000839853     0.000000     0.000000     0.000000     0.000000   \n",
      "\n",
      "        SRR5885323   SRR5885324   SRR5885325   SRR5885326   SRR5885327  ...  \\\n",
      "0       840.900950   146.906407   629.282905   522.964794   528.046633  ...   \n",
      "1      1868.728051    63.991912  3326.478827  1164.856298  2603.398721  ...   \n",
      "2      4043.380952  2285.570025  3707.631731  1666.492568  5233.063600  ...   \n",
      "3      6670.210951  4968.397862  7213.956564  3072.723470  9342.914036  ...   \n",
      "4      1870.150329   115.430146  2112.885077   678.024752  2573.555873  ...   \n",
      "...            ...          ...          ...          ...          ...  ...   \n",
      "79618     0.000000     0.000000     0.000000     0.000000     0.000000  ...   \n",
      "79619     0.000000     0.527984     0.000000     0.000000     0.000000  ...   \n",
      "79620     0.000000     0.000000     0.000000     0.000000     0.000000  ...   \n",
      "79621     0.000000     0.000000     0.000000     0.000000     0.000000  ...   \n",
      "79622     0.000000     0.000000     0.072677     0.000000     0.039815  ...   \n",
      "\n",
      "        SRR5885330   SRR5885331   SRR5885332   SRR5885333   SRR5885334  \\\n",
      "0       361.149039   716.406623   549.857754   937.075708   133.530755   \n",
      "1       722.929355  2521.032641   259.554239  1761.252822   311.035068   \n",
      "2      1110.828835  4123.570687  1122.442898  4512.397338   822.116969   \n",
      "3      2158.384895  7906.207378  2493.685630  8250.447153  1739.841848   \n",
      "4       608.277295  1787.413833   146.879700  2586.154822   709.002759   \n",
      "...            ...          ...          ...          ...          ...   \n",
      "79618     0.000000     0.000000     0.006345     0.000000     0.000000   \n",
      "79619     0.000000     0.000000     0.000000     0.000000     0.383919   \n",
      "79620     0.000000     0.000000     0.000000     0.000000     0.000000   \n",
      "79621     0.000000     0.000000     0.000000     0.000000     0.000000   \n",
      "79622     0.022995     0.249778     0.000000     0.081359     0.023234   \n",
      "\n",
      "        SRR5885335   SRR5885336   SRR5885337   SRR5885338  Length  \n",
      "0       312.679275   388.212798   315.188731   272.759151     577  \n",
      "1       757.296744   233.090331   202.833909   151.619038      71  \n",
      "2      2982.843805  1202.701813   785.838458   788.351117     954  \n",
      "3      5630.858298  2586.306756  1638.687286  1693.872600     353  \n",
      "4      1511.210320   322.254315   114.580700    68.374084      69  \n",
      "...            ...          ...          ...          ...     ...  \n",
      "79618     0.000000     0.005162     0.000000     0.000000    2817  \n",
      "79619     0.000000     0.027148     0.000000     0.032440    1607  \n",
      "79620     0.000000     0.000000     0.000000     0.000000     567  \n",
      "79621     0.000000     0.000000     0.000000     0.000000     769  \n",
      "79622     0.000000     0.000000     0.218034     0.066748     781  \n",
      "\n",
      "[79623 rows x 22 columns]\n",
      "10  patients\n",
      "lncRNA:  9166\n",
      "PROTEIN CODING:  15356\n",
      "NOVEL:  189\n",
      "GSE133624_BLCA\n",
      "         transcript_id   SRR9623608    SRR9623609   SRR9623610   SRR9623611  \\\n",
      "0       TCONS_00001163    94.710059    755.279868   131.571140    81.735996   \n",
      "1      ENST00000387314   237.933726   1448.570432   512.759418   227.360352   \n",
      "2      ENST00000389680  2130.740302  21293.019278  3817.605207  1669.423003   \n",
      "3       TCONS_00001164     9.196596     30.243868    35.903748     6.276626   \n",
      "4      ENST00000387342  2633.015430   8520.494556   724.247502  2750.829818   \n",
      "...                ...          ...           ...          ...          ...   \n",
      "79618  ENST00000797680     0.000000      0.000000     0.000000     0.074907   \n",
      "79619  ENST00000797955     0.000000      0.206320     0.000000     0.131309   \n",
      "79620  ENST00000819228     0.000000      0.000000     0.000000     0.000000   \n",
      "79621  ENST00000751368     0.000000      0.000000     0.000000     0.000000   \n",
      "79622  ENST00000839853     0.076976      0.000000     0.054456     0.135092   \n",
      "\n",
      "        SRR9623612   SRR9623613   SRR9623614   SRR9623615   SRR9623616  ...  \\\n",
      "0       221.820677   305.204319   157.987621   354.472352    29.359201  ...   \n",
      "1      1118.381814   497.500838   539.908001  2659.414753   331.135839  ...   \n",
      "2      3328.302099  3774.682251  1980.600320  5869.803196  2249.472809  ...   \n",
      "3        14.223091    28.627207     9.459355    31.737706     7.195437  ...   \n",
      "4      1018.703110   681.889724  1304.689062  1667.244833  7712.148555  ...   \n",
      "...            ...          ...          ...          ...          ...  ...   \n",
      "79618     0.000000     0.000000     0.000000     0.000000     0.000000  ...   \n",
      "79619     0.048066     0.000000     0.000000     0.000000     0.000000  ...   \n",
      "79620     0.000000     0.081755     0.000000     0.000000     0.000000  ...   \n",
      "79621     0.000000     0.000000     0.000000     0.000000     0.000000  ...   \n",
      "79622     0.000000     0.000000     0.513058     0.174938     0.000000  ...   \n",
      "\n",
      "        SRR9623664   SRR9623665   SRR9623666   SRR9623667   SRR9623668  \\\n",
      "0        59.060252    83.220292   135.273242    35.871146   138.115354   \n",
      "1       365.521317   445.661837   138.151505   446.044868   337.387440   \n",
      "2      3326.073985  6886.881131  5502.030499  2919.401640  4779.265523   \n",
      "3         4.461071    21.229903     8.079863     4.872131    11.751331   \n",
      "4      2996.148455  3510.748427   582.738355   529.452812  2782.414782   \n",
      "...            ...          ...          ...          ...          ...   \n",
      "79618     0.000000     0.000000     0.000000     0.105264     0.000000   \n",
      "79619     0.000000     0.000000     0.043289     0.000000     0.000000   \n",
      "79620     0.000000     0.000000     0.000000     0.000000     0.000000   \n",
      "79621     0.000000     0.000000     0.000000     0.000000     0.000000   \n",
      "79622     0.000000     0.177696     0.000000     0.000000     0.074809   \n",
      "\n",
      "        SRR9623669   SRR9623670   SRR9623671   SRR9623672  Length  \n",
      "0       102.530259   401.339245   232.607473    51.853672     577  \n",
      "1       390.631476   187.041381  1723.291470   303.697235      71  \n",
      "2      5742.143261  6212.995190  4725.088922  4507.334426     954  \n",
      "3        11.597494    14.042571    12.182474     9.578194     353  \n",
      "4      1644.585703  2517.095047  2497.014171  1382.211777      69  \n",
      "...            ...          ...          ...          ...     ...  \n",
      "79618     0.000000     0.000000     0.000000     0.000000    2817  \n",
      "79619     0.000000     0.000000     0.000000     0.000000    1607  \n",
      "79620     0.000000     0.000000     0.000000     0.000000     567  \n",
      "79621     0.000000     0.000000     0.000000     0.000000     769  \n",
      "79622     0.000000     0.000000     0.000000     0.000000     781  \n",
      "\n",
      "[79623 rows x 67 columns]\n",
      "20  patients\n",
      "lncRNA:  11836\n",
      "PROTEIN CODING:  16670\n",
      "NOVEL:  272\n",
      "GSE22260_PRAD\n",
      "         transcript_id    SRR057629    SRR057630    SRR057631    SRR057632  \\\n",
      "0       TCONS_00001163   505.580155   358.188817   610.393830   485.691114   \n",
      "1      ENST00000387314   307.404897   260.978564   335.470578   302.790879   \n",
      "2      ENST00000389680   963.113987  1492.577693  1714.270582  1722.298825   \n",
      "3       TCONS_00001164  2072.036019  3018.257752  3454.086355  3363.708291   \n",
      "4      ENST00000387342   123.440072   340.843241   246.204803   105.710377   \n",
      "...                ...          ...          ...          ...          ...   \n",
      "79618  ENST00000797680     0.000000     0.000000     0.000000     0.136278   \n",
      "79619  ENST00000797955     0.000000     0.000000     0.000000     0.000000   \n",
      "79620  ENST00000819228     0.000000     0.000000     0.000000     0.000000   \n",
      "79621  ENST00000751368     0.000000     0.000000     0.000000     0.000000   \n",
      "79622  ENST00000839853     0.000000     0.000000     0.000000     0.000000   \n",
      "\n",
      "         SRR057633    SRR057634    SRR057635    SRR057636    SRR057637  ...  \\\n",
      "0       828.704060   560.157860   667.025397   312.207345   599.186986  ...   \n",
      "1       192.041008   279.694482   465.586379   327.150559   375.332796  ...   \n",
      "2      1882.648813  1315.704436  1697.877370  1288.262967  1886.251261  ...   \n",
      "3      3590.868564  2455.857983  3444.792783  2639.344030  3722.940942  ...   \n",
      "4       118.110178   992.419200   182.507290   235.643229   391.293744  ...   \n",
      "...            ...          ...          ...          ...          ...  ...   \n",
      "79618     0.000000     0.000000     0.000000     0.000000     0.000000  ...   \n",
      "79619     0.000000     0.000000     0.489771     0.000000     0.109098  ...   \n",
      "79620     0.000000     0.000000     0.000000     0.000000     0.000000  ...   \n",
      "79621     0.000000     0.000000     0.000000     0.000000     0.000000  ...   \n",
      "79622     0.000000     0.876785     0.000000     0.000000     0.000000  ...   \n",
      "\n",
      "         SRR057650    SRR057651    SRR057652    SRR057653    SRR057654  \\\n",
      "0       516.492661   804.660160   562.653201   729.659957   522.969092   \n",
      "1       183.335245   210.944531   272.748468   834.339745   233.490104   \n",
      "2      1528.896188  1994.455848  1166.589158  1917.166681  1349.364683   \n",
      "3      3138.239075  3857.406587  2087.859750  3907.654339  2685.310848   \n",
      "4       407.085353   325.588298   726.399159   413.930976   245.656988   \n",
      "...            ...          ...          ...          ...          ...   \n",
      "79618     0.000000     0.000000     0.000000     0.000000     0.000000   \n",
      "79619     0.000000     0.000000     0.000000     0.000000     0.000000   \n",
      "79620     0.000000     0.000000     0.000000     0.000000     0.000000   \n",
      "79621     0.000000     0.000000     0.000000     0.000000     0.000000   \n",
      "79622     0.000000     0.000000     0.000000     0.000000     0.000000   \n",
      "\n",
      "         SRR057655    SRR057656    SRR057657    SRR057658  Length  \n",
      "0       877.597660   507.158228   467.438946   406.611887     577  \n",
      "1       215.695726   146.917247   240.897254   152.474043      71  \n",
      "2      1001.207811   876.875194  1045.364330  1153.553617     954  \n",
      "3      1924.909674  1550.315861  1977.231537  2121.593491     353  \n",
      "4       424.595745   264.557507   333.684324   336.935393      69  \n",
      "...            ...          ...          ...          ...     ...  \n",
      "79618     0.000000     0.000000     0.000000     0.000000    2817  \n",
      "79619     0.000000     0.000000     0.000000     0.000000    1607  \n",
      "79620     0.000000     0.000000     0.000000     0.000000     567  \n",
      "79621     0.000000     0.000000     0.000000     0.000000     769  \n",
      "79622     0.213138     0.000000     0.000000     0.000000     781  \n",
      "\n",
      "[79623 rows x 32 columns]\n",
      "10  patients\n",
      "lncRNA:  14672\n",
      "PROTEIN CODING:  15751\n",
      "NOVEL:  303\n",
      "PRJEB2449_PRAD\n",
      "         transcript_id    ERR031017    ERR031018    ERR031019    ERR031022  \\\n",
      "0       TCONS_00001163  1643.835910   978.789108   944.501504  1025.865785   \n",
      "1      ENST00000387314  1402.709285   804.954081  1190.264761   969.624769   \n",
      "2      ENST00000389680  4327.329773  4648.985452  2538.209165  3772.605735   \n",
      "3       TCONS_00001164  7588.449706  8742.091412  4208.475893  7355.942465   \n",
      "4      ENST00000387342   690.970358   444.598922   793.912293   252.167464   \n",
      "...                ...          ...          ...          ...          ...   \n",
      "79618  ENST00000797680     0.000000     0.013944     0.000000     0.026797   \n",
      "79619  ENST00000797955     0.000000     0.000000     0.000000     0.187893   \n",
      "79620  ENST00000819228     0.000000     0.000000     0.000000     0.000000   \n",
      "79621  ENST00000751368     0.000000     0.000000     0.000000     0.000000   \n",
      "79622  ENST00000839853     0.000000     0.000000     0.049499     0.000000   \n",
      "\n",
      "         ERR031023    ERR031024    ERR031025    ERR031026    ERR031027  ...  \\\n",
      "0      2122.008103  1547.429044  1180.249343   961.788729  1550.371791  ...   \n",
      "1      1106.098041   577.082310   563.351254   461.398405  1519.047376  ...   \n",
      "2      4290.634950  2827.066640  1670.330423  2793.355123  3447.920474  ...   \n",
      "3      7492.350658  4939.289524  3160.066075  5597.451175  6214.241117  ...   \n",
      "4       759.611548   907.400502   276.494488   204.953357   579.972275  ...   \n",
      "...            ...          ...          ...          ...          ...  ...   \n",
      "79618     0.000000     0.000000     0.000000     0.000000     0.000000  ...   \n",
      "79619     0.000000     0.024706     0.000000     0.000000     0.000000  ...   \n",
      "79620     0.000000     0.000000     0.000000     0.000000     0.000000  ...   \n",
      "79621     0.000000     0.000000     0.000000     0.000000     0.000000  ...   \n",
      "79622     0.000000     0.000000     0.181395     0.000000     0.099301  ...   \n",
      "\n",
      "         ERR031041    ERR031042    ERR031043    ERR031044    ERR299295  \\\n",
      "0      1040.254433   972.609374   980.946943  1121.929401   834.896548   \n",
      "1       468.617058   529.232969   313.151174   391.959735   589.796319   \n",
      "2      2345.702629  4555.461681  2417.183816  3085.858361  2523.420999   \n",
      "3      4181.565559  6268.583531  4853.391762  6063.849003  4195.546282   \n",
      "4       279.168514   170.207754   173.030726   135.570046   255.152150   \n",
      "...            ...          ...          ...          ...          ...   \n",
      "79618     0.000000     0.000000     0.000000     0.000000     0.000000   \n",
      "79619     0.000000     0.000000     0.000000     0.000000     0.241920   \n",
      "79620     0.000000     0.000000     0.000000     0.000000     0.000000   \n",
      "79621     0.000000     0.000000     0.000000     0.000000     0.000000   \n",
      "79622     0.000000     0.162130     0.294790     0.171105     0.000000   \n",
      "\n",
      "         ERR299296    ERR299297    ERR299298    ERR299299  Length  \n",
      "0      1001.201081  1267.499117   908.156784  1027.491712     577  \n",
      "1      1253.562972   450.013028  1198.733223  1217.351758      71  \n",
      "2      3571.769519  4211.367772  4709.190787  2555.225196     954  \n",
      "3      6752.083281  7361.466780  8697.278924  4502.260338     353  \n",
      "4       326.409235   373.813193   534.397089   334.615596      69  \n",
      "...            ...          ...          ...          ...     ...  \n",
      "79618     0.000000     0.000000     0.000000     0.000000    2817  \n",
      "79619     0.000000     0.000000     0.000000     0.263220    1607  \n",
      "79620     0.000000     0.000000     0.000000     0.000000     567  \n",
      "79621     0.000000     0.000000     0.000000     0.000000     769  \n",
      "79622     0.544107     0.049588     0.292946     0.000000     781  \n",
      "\n",
      "[79623 rows x 30 columns]\n",
      "13  patients\n",
      "lncRNA:  9563\n",
      "PROTEIN CODING:  14609\n",
      "NOVEL:  214\n",
      "SRP238334_KIRC\n",
      "         transcript_id  SRR10744251  SRR10744252  SRR10744253  SRR10744254  \\\n",
      "0       TCONS_00001163    28.342008    93.105607    22.649516    18.746208   \n",
      "1      ENST00000387314     0.000000     2.067341     0.000000     0.000000   \n",
      "2      ENST00000389680     0.113148     0.153859     0.512545     0.086221   \n",
      "3       TCONS_00001164     0.000000     0.000000     0.000000     0.000000   \n",
      "4      ENST00000387342     0.000000     0.000000     0.000000     0.000000   \n",
      "...                ...          ...          ...          ...          ...   \n",
      "79618  ENST00000797680     0.000000     0.000000     0.000000     0.000000   \n",
      "79619  ENST00000797955     0.033585     0.000000     0.000000     0.153557   \n",
      "79620  ENST00000819228     0.000000     0.000000     0.000000     0.000000   \n",
      "79621  ENST00000751368     0.000000     0.000000     0.000000     0.000000   \n",
      "79622  ENST00000839853     0.207317     0.000000     0.113833     0.000000   \n",
      "\n",
      "       SRR10744255  SRR10744256  SRR10744257  SRR10744258  SRR10744259  ...  \\\n",
      "0        20.892104    69.348241    18.613853    25.218071    28.681637  ...   \n",
      "1         0.000000     0.390288     0.000000     0.000000     0.000000  ...   \n",
      "2         0.306947     1.917077     1.030668     4.238391     0.276084  ...   \n",
      "3         0.000000     0.078500     0.000000     0.000000     0.000000  ...   \n",
      "4         0.000000     0.000000     0.000000     0.000000     0.000000  ...   \n",
      "...            ...          ...          ...          ...          ...  ...   \n",
      "79618     0.000000     0.000000     0.000000     0.000000     0.000000  ...   \n",
      "79619     0.121480     2.707247     3.153426     0.000000     0.000000  ...   \n",
      "79620     0.000000     0.000000     0.000000     0.000000     0.000000  ...   \n",
      "79621     0.000000     0.000000     0.000000     0.000000     0.000000  ...   \n",
      "79622     0.000000     0.000000     0.048422     0.000000     0.000000  ...   \n",
      "\n",
      "       SRR10744431  SRR10744432  SRR10744433  SRR10744434  SRR10744435  \\\n",
      "0       240.710024   288.552252   209.823362   236.873928   160.682948   \n",
      "1         1.250363     1.040370     3.036297     1.607307     2.370791   \n",
      "2         1.675014     1.277562     1.943357     2.990534     1.305675   \n",
      "3         0.000000     0.000000     0.000000     0.000000     0.000000   \n",
      "4         0.000000     0.535263     0.000000     0.000000     0.000000   \n",
      "...            ...          ...          ...          ...          ...   \n",
      "79618     0.000000     0.000000     0.000000     0.000000     0.000000   \n",
      "79619     0.000000     0.000000     0.000000     0.000000     0.000000   \n",
      "79620     0.000000     0.000000     0.000000     0.000000     0.000000   \n",
      "79621     0.000000     0.000000     0.000000     0.000000     0.000000   \n",
      "79622     0.056835     0.141869     0.607259     0.000000     0.043105   \n",
      "\n",
      "       SRR10744436  SRR10744437  SRR10744438  SRR10744439  Length  \n",
      "0       359.507501     6.440832   134.453368    31.208939     577  \n",
      "1         0.505735    35.593305     2.801719     0.000000      71  \n",
      "2         0.715133    90.532706     1.709812     0.235213     954  \n",
      "3         0.000000   192.450587     0.000000     0.000000     353  \n",
      "4         0.000000    17.235292     0.000000     0.000000      69  \n",
      "...            ...          ...          ...          ...     ...  \n",
      "79618     0.000000     0.000000     0.000000     0.000000    2817  \n",
      "79619     0.000000     0.000000     0.000000     0.000000    1607  \n",
      "79620     0.000000     0.000000     0.000000     0.000000     567  \n",
      "79621     0.000000     0.000000     0.000000     0.000000     769  \n",
      "79622     0.000000     0.000000     0.101881     0.143658     781  \n",
      "\n",
      "[79623 rows x 191 columns]\n",
      "61  patients\n",
      "lncRNA:  13055\n",
      "PROTEIN CODING:  16295\n",
      "NOVEL:  312\n",
      "GSE214846_LIHC\n",
      "         transcript_id  SRR21814873  SRR21814874  SRR21814875  SRR21814876  \\\n",
      "0       TCONS_00001163   228.207492   130.894376    84.957114   104.507643   \n",
      "1      ENST00000387314   368.592456   240.142084   269.783732   296.903144   \n",
      "2      ENST00000389680  1839.720826  3860.125457  2087.235929  3033.831994   \n",
      "3       TCONS_00001164  3736.242379  7353.434684  4098.254624  6456.937170   \n",
      "4      ENST00000387342   346.907632   346.695457  1279.242485   299.423195   \n",
      "...                ...          ...          ...          ...          ...   \n",
      "79618  ENST00000797680     0.000000     0.000000     0.000000     0.000000   \n",
      "79619  ENST00000797955     0.000000     0.000000     0.000000     0.000000   \n",
      "79620  ENST00000819228     0.000000     0.000000     0.000000     0.000000   \n",
      "79621  ENST00000751368     0.000000     0.000000     0.101667     0.000000   \n",
      "79622  ENST00000839853     0.000000     0.000000     0.000000     0.000000   \n",
      "\n",
      "       SRR21814877  SRR21814878  SRR21814879  SRR21814880  SRR21814881  ...  \\\n",
      "0        98.084335   146.395108    89.279234   207.558076   108.525760  ...   \n",
      "1       186.914880   237.073603   126.652333   364.642214   196.100150  ...   \n",
      "2      2160.672406  3493.534787  1984.847660  2720.019666  1755.473402  ...   \n",
      "3      4538.027811  6070.806155  4103.285827  5746.455961  3851.799633  ...   \n",
      "4       283.919704   243.199292   302.067450   712.818387   199.776410  ...   \n",
      "...            ...          ...          ...          ...          ...  ...   \n",
      "79618     0.000000     0.000000     0.024745     0.000000     0.000000  ...   \n",
      "79619     0.000000     0.000000     0.000000     0.000000     0.000000  ...   \n",
      "79620     0.000000     0.000000     0.000000     0.000000     0.000000  ...   \n",
      "79621     0.000000     0.000000     0.000000     0.000000     0.000000  ...   \n",
      "79622     0.000000     0.000000     0.000000     0.000000     0.000000  ...   \n",
      "\n",
      "        SRR21814994  SRR21814995  SRR21814996  SRR21814997  SRR21814998  \\\n",
      "0         41.607753   147.596335    57.573411   106.853992   196.037386   \n",
      "1         60.344315   208.124371   271.951652   323.455792  1121.808991   \n",
      "2       6876.702144  4115.899262   720.816207  2318.064207  2958.904251   \n",
      "3      12962.155792  8583.265429  1588.194375  5107.907081  6027.479855   \n",
      "4        308.325976  1624.282271   321.699284   321.836894  1332.468154   \n",
      "...             ...          ...          ...          ...          ...   \n",
      "79618      0.000000     0.000000     0.026985     0.000000     0.000000   \n",
      "79619      0.000000     0.088843     0.047304     0.000000     0.000000   \n",
      "79620      0.000000     0.000000     0.000000     0.000000     0.000000   \n",
      "79621      0.000000     0.000000     0.000000     0.000000     0.000000   \n",
      "79622      0.000000     0.091403     0.000000     0.000000     0.000000   \n",
      "\n",
      "       SRR21814999  SRR21815000  SRR21815001  SRR21815002  Length  \n",
      "0        29.940187    88.619653    39.487418   210.139958     577  \n",
      "1        99.822249   128.605482   193.176241   174.120892      71  \n",
      "2      1061.621127  1582.723030  1122.101213  4287.153570     954  \n",
      "3      2478.324161  3599.430287  2491.123941  9424.229201     353  \n",
      "4       554.664496   357.929736  2599.289055   704.315093      69  \n",
      "...            ...          ...          ...          ...     ...  \n",
      "79618     0.000000     0.277833     0.000000     0.000000    2817  \n",
      "79619     0.000000     0.000000     0.000000     0.000000    1607  \n",
      "79620     0.312495     0.000000     0.000000     0.000000     567  \n",
      "79621     0.000000     0.000000     0.000000     0.000000     769  \n",
      "79622     0.000000     0.000000     0.000000     0.077976     781  \n",
      "\n",
      "[79623 rows x 132 columns]\n",
      "65  patients\n",
      "lncRNA:  16435\n",
      "PROTEIN CODING:  16979\n",
      "NOVEL:  304\n",
      "GSE229705_LUAD\n",
      "         transcript_id   SRR24166098    SRR24166099    SRR24166100  \\\n",
      "0       TCONS_00001163   1220.031292    1275.252019     769.348922   \n",
      "1      ENST00000387314    961.341020    3703.472337    2067.498413   \n",
      "2      ENST00000389680  38781.552160   80209.365485   67543.658833   \n",
      "3       TCONS_00001164  94663.622967  186834.503447  161666.687864   \n",
      "4      ENST00000387342   4103.659169    8032.833400    8767.443965   \n",
      "...                ...           ...            ...            ...   \n",
      "79618  ENST00000797680      0.000000       0.000000       0.000000   \n",
      "79619  ENST00000797955      0.000000       0.000000       0.000000   \n",
      "79620  ENST00000819228      0.000000       0.000000       0.000000   \n",
      "79621  ENST00000751368      0.000000       0.000000       0.000000   \n",
      "79622  ENST00000839853      0.000000       0.000000       0.000000   \n",
      "\n",
      "         SRR24166101   SRR24166102    SRR24166103    SRR24166104  SRR24166105  \\\n",
      "0         394.476234    514.177177     927.310370     729.127933    56.707084   \n",
      "1        1382.216865   1271.964259    4288.619597    1408.674432    25.681759   \n",
      "2       87557.842612  27890.927198  111707.466322   78656.936868   431.003985   \n",
      "3      207847.821521  66149.101847  268775.077021  179808.954152   903.667293   \n",
      "4       12851.325852   1523.818463    8411.284392    4503.903723    13.702452   \n",
      "...              ...           ...            ...            ...          ...   \n",
      "79618       0.000000      0.000000       0.000000       0.000000     0.000000   \n",
      "79619       0.000000      0.000000       0.107232       0.000000     0.000000   \n",
      "79620       0.000000      0.000000       0.000000       0.000000     0.000000   \n",
      "79621       0.000000      0.000000       0.000000       0.000000     0.000000   \n",
      "79622       0.263983      0.000000       0.000000       0.000000     0.000000   \n",
      "\n",
      "        SRR24166106  ...    SRR24166335    SRR24166336    SRR24166337  \\\n",
      "0       1274.436593  ...    1490.651846     384.966134     661.049147   \n",
      "1       2350.635176  ...    3260.957384    1790.756923    1546.515340   \n",
      "2      39453.829812  ...   63487.641855   55127.789885   51382.580002   \n",
      "3      91857.849588  ...  151882.914947  128461.905446  124560.444217   \n",
      "4       3584.176665  ...    6975.861925    2146.390306    2358.079319   \n",
      "...             ...  ...            ...            ...            ...   \n",
      "79618      0.000000  ...       0.000000       0.000000       0.000000   \n",
      "79619      0.000000  ...       0.000000       0.000000       0.000000   \n",
      "79620      0.000000  ...       0.000000       0.000000       0.000000   \n",
      "79621      0.000000  ...       0.000000       0.000000       0.000000   \n",
      "79622      0.000000  ...       0.130022       0.863464       0.000000   \n",
      "\n",
      "         SRR24166338   SRR24166339    SRR24166340   SRR24166341  \\\n",
      "0         751.935936    944.492325    2150.794460    948.147422   \n",
      "1        2386.046113    819.366736    2537.083087   2093.524280   \n",
      "2       62923.761309  19119.565738   79754.444867  34651.910268   \n",
      "3      152293.572347  47722.903466  196184.783253  84608.898013   \n",
      "4        5091.972457   2147.020442    5550.347353   2837.184491   \n",
      "...              ...           ...            ...           ...   \n",
      "79618       0.000000      0.000000       0.000000      0.000000   \n",
      "79619       0.000000      0.000000       0.000000      0.000000   \n",
      "79620       0.000000      0.000000       0.000000      0.000000   \n",
      "79621       0.000000      0.000000       0.000000      0.000000   \n",
      "79622       0.000000      0.000000       0.000000      0.000000   \n",
      "\n",
      "         SRR24166342    SRR24166343  Length  \n",
      "0        2746.193727    1704.785136     577  \n",
      "1        3448.507809    2319.766583      71  \n",
      "2       86163.731698   45361.724973     954  \n",
      "3      208494.163466  111003.448302     353  \n",
      "4       16106.892965    4937.474202      69  \n",
      "...              ...            ...     ...  \n",
      "79618       0.000000       0.000000    2817  \n",
      "79619       0.000000       0.000000    1607  \n",
      "79620       0.000000       0.000000     567  \n",
      "79621       0.000000       0.000000     769  \n",
      "79622       0.000000       0.000000     781  \n",
      "\n",
      "[79623 rows x 248 columns]\n",
      "123  patients\n",
      "lncRNA:  26865\n",
      "PROTEIN CODING:  17307\n",
      "NOVEL:  448\n",
      "SRP107326_COAD\n",
      "         transcript_id    SRR5580910   SRR5580911   SRR5580912   SRR5580913  \\\n",
      "0       TCONS_00001163   1352.749553   357.552449   204.747110    46.960666   \n",
      "1      ENST00000387314   5586.858442  1920.032847  3116.212603   204.810457   \n",
      "2      ENST00000389680  10637.842408  3642.201838  8790.035105  2957.616877   \n",
      "3       TCONS_00001164    102.941405    32.886566    13.830216    11.975050   \n",
      "4      ENST00000387342   2444.688631  1850.703406  3372.133445  2339.659195   \n",
      "...                ...           ...          ...          ...          ...   \n",
      "79618  ENST00000797680      0.000000     0.000000     0.000000     0.000000   \n",
      "79619  ENST00000797955      0.000000     0.000000     0.000000     0.000000   \n",
      "79620  ENST00000819228      0.000000     0.000000     0.000000     0.000000   \n",
      "79621  ENST00000751368      0.000000     0.000000     0.000000     0.000000   \n",
      "79622  ENST00000839853      0.000000     0.000000     0.177335     0.054125   \n",
      "\n",
      "        SRR5580914   SRR5580915   SRR5580916   SRR5580917   SRR5580920  ...  \\\n",
      "0       662.433437   210.511811    98.873922    52.844190   444.512464  ...   \n",
      "1      2898.935276   435.223752   797.603025   791.288299  1041.231363  ...   \n",
      "2      8462.184027  3049.440996  3907.039359  2105.978863  4876.413055  ...   \n",
      "3        62.517103    22.275275    43.885374    22.788838    14.664175  ...   \n",
      "4      3453.559701  2094.246799   796.348715  2834.741269   688.526683  ...   \n",
      "...            ...          ...          ...          ...          ...  ...   \n",
      "79618     0.000000     0.000000     0.000000     0.000000     0.000000  ...   \n",
      "79619     0.000000     0.000000     0.000000     0.000000     0.000000  ...   \n",
      "79620     0.000000     0.000000     0.000000     0.000000     0.000000  ...   \n",
      "79621     0.000000     0.000000     0.000000     0.000000     0.000000  ...   \n",
      "79622     0.142711     0.000000     0.000000     0.083066     0.000000  ...   \n",
      "\n",
      "        SRR5581286   SRR5581287   SRR5581289   SRR5581300   SRR5581301  \\\n",
      "0       147.785835   329.411257   133.043543   171.385786  1960.719180   \n",
      "1       648.304173   836.115716  2625.898457   127.158665   840.219462   \n",
      "2      3641.627375  3858.130233  4974.231908  1444.143894  5274.185236   \n",
      "3      1774.060426  2375.170909    21.357243    60.699951    18.777331   \n",
      "4      1192.205333  1237.907237  2066.365323   390.788679  1966.744946   \n",
      "...            ...          ...          ...          ...          ...   \n",
      "79618     0.000000     0.000000     0.000000     0.000000     0.000000   \n",
      "79619     0.000000     0.000000     0.000000     0.000000     0.000000   \n",
      "79620     0.000000     0.000000     0.000000     0.000000     0.000000   \n",
      "79621     0.000000     0.000000     0.000000     0.000000     0.000000   \n",
      "79622     0.000000     0.095852     0.000000     0.000000     0.000000   \n",
      "\n",
      "        SRR5581322   SRR5581323   SRR5581324   SRR5581325  Length  \n",
      "0       223.198527   152.027809   451.091050    68.420046     577  \n",
      "1       492.750437   193.381608  5002.118405   384.379103      71  \n",
      "2      2106.233349  1367.006543  8270.772336  2203.431325     954  \n",
      "3      3473.924718  2102.182765   152.806419    31.348175     353  \n",
      "4       579.829123  1135.245614  2296.323336  1505.145236      69  \n",
      "...            ...          ...          ...          ...     ...  \n",
      "79618     0.000000     0.000000     0.000000     0.000000    2817  \n",
      "79619     0.000000     0.000000     0.000000     0.000000    1607  \n",
      "79620     0.000000     0.000000     0.000000     0.000000     567  \n",
      "79621     0.000000     0.000000     0.000000     0.000000     769  \n",
      "79622     0.149568     0.075129     0.180094     0.095736     781  \n",
      "\n",
      "[79623 rows x 210 columns]\n",
      "104  patients\n",
      "lncRNA:  12010\n",
      "PROTEIN CODING:  16460\n",
      "NOVEL:  278\n",
      "TCGA_COAD\n",
      "         transcript_id  03fec96d-2ca4-40ba-9ba5-e1904285a66b  \\\n",
      "0       TCONS_00001163                            163.929274   \n",
      "1      ENST00000387314                             40.928232   \n",
      "2      ENST00000389680                            695.330552   \n",
      "3       TCONS_00001164                           1266.085298   \n",
      "4      ENST00000387342                            105.286394   \n",
      "...                ...                                   ...   \n",
      "79618  ENST00000797680                              0.000000   \n",
      "79619  ENST00000797955                              0.000000   \n",
      "79620  ENST00000819228                              0.000000   \n",
      "79621  ENST00000751368                              0.000000   \n",
      "79622  ENST00000839853                              0.000000   \n",
      "\n",
      "       1f04bbb9-7526-495d-a40d-f49c6bbf7402  \\\n",
      "0                                266.428268   \n",
      "1                                 78.190926   \n",
      "2                               1279.719527   \n",
      "3                               2511.891656   \n",
      "4                                115.953211   \n",
      "...                                     ...   \n",
      "79618                              0.000000   \n",
      "79619                              0.000000   \n",
      "79620                              0.000000   \n",
      "79621                              0.000000   \n",
      "79622                              0.000000   \n",
      "\n",
      "       3f805d4a-24d4-4df9-90f5-5dcf9f979fd3  \\\n",
      "0                                164.020056   \n",
      "1                                 29.450422   \n",
      "2                                560.434478   \n",
      "3                               1169.496259   \n",
      "4                                118.581096   \n",
      "...                                     ...   \n",
      "79618                              0.000000   \n",
      "79619                              0.000000   \n",
      "79620                              0.000000   \n",
      "79621                              0.000000   \n",
      "79622                              0.000000   \n",
      "\n",
      "       417630a9-9bfc-4e9f-84d7-47f62f8f37f1  \\\n",
      "0                                392.136618   \n",
      "1                                 43.035047   \n",
      "2                               1098.155944   \n",
      "3                               2212.327142   \n",
      "4                                493.919520   \n",
      "...                                     ...   \n",
      "79618                              0.000000   \n",
      "79619                              0.000000   \n",
      "79620                              0.000000   \n",
      "79621                              0.000000   \n",
      "79622                              0.000000   \n",
      "\n",
      "       43ba1922-8387-40ec-9dca-308d2e60ed45  \\\n",
      "0                                286.598999   \n",
      "1                                 48.777412   \n",
      "2                               1233.506535   \n",
      "3                               2470.266182   \n",
      "4                                 99.336850   \n",
      "...                                     ...   \n",
      "79618                              0.000000   \n",
      "79619                              0.000000   \n",
      "79620                              0.000000   \n",
      "79621                              0.000000   \n",
      "79622                              0.000000   \n",
      "\n",
      "       5e64bb5c-a7bd-4b23-a39c-d4ff5ffe2252  \\\n",
      "0                                202.900132   \n",
      "1                                 27.853392   \n",
      "2                                956.457442   \n",
      "3                               2300.840073   \n",
      "4                                 80.250062   \n",
      "...                                     ...   \n",
      "79618                              0.000000   \n",
      "79619                              0.000000   \n",
      "79620                              0.000000   \n",
      "79621                              0.000000   \n",
      "79622                              0.000000   \n",
      "\n",
      "       699eb7a7-c181-4061-b21d-45f86aa569b4  \\\n",
      "0                                279.112612   \n",
      "1                                 92.415266   \n",
      "2                                878.532716   \n",
      "3                               1651.626906   \n",
      "4                                182.791742   \n",
      "...                                     ...   \n",
      "79618                              0.000000   \n",
      "79619                              0.000000   \n",
      "79620                              0.000000   \n",
      "79621                              0.000000   \n",
      "79622                              0.000000   \n",
      "\n",
      "       abb984a8-1f9b-426b-ae7f-f6e3df4597f5  \\\n",
      "0                                261.578165   \n",
      "1                                146.763883   \n",
      "2                               1577.388433   \n",
      "3                               3156.927365   \n",
      "4                                151.017908   \n",
      "...                                     ...   \n",
      "79618                              0.000000   \n",
      "79619                              0.000000   \n",
      "79620                              0.000000   \n",
      "79621                              0.000000   \n",
      "79622                              0.000000   \n",
      "\n",
      "       abf49519-3682-4c89-9197-77219375a029  ...  \\\n",
      "0                                232.946391  ...   \n",
      "1                                 19.833416  ...   \n",
      "2                                420.237650  ...   \n",
      "3                                703.687347  ...   \n",
      "4                                285.716165  ...   \n",
      "...                                     ...  ...   \n",
      "79618                              0.000000  ...   \n",
      "79619                              0.000000  ...   \n",
      "79620                              0.000000  ...   \n",
      "79621                              0.000000  ...   \n",
      "79622                              0.000000  ...   \n",
      "\n",
      "       d48667cd-c1ce-4602-8cfa-84a84e474479  \\\n",
      "0                                 69.147035   \n",
      "1                                 85.647292   \n",
      "2                                595.559906   \n",
      "3                               1134.174221   \n",
      "4                                982.157907   \n",
      "...                                     ...   \n",
      "79618                              0.000000   \n",
      "79619                              0.000000   \n",
      "79620                              0.000000   \n",
      "79621                              0.000000   \n",
      "79622                              0.000000   \n",
      "\n",
      "       d58ca2b2-a17c-47f4-b5d0-84e24cd2574c  \\\n",
      "0                                103.178471   \n",
      "1                                231.522910   \n",
      "2                                743.751531   \n",
      "3                               1260.789412   \n",
      "4                                347.569722   \n",
      "...                                     ...   \n",
      "79618                              0.000000   \n",
      "79619                              0.000000   \n",
      "79620                              0.000000   \n",
      "79621                              0.000000   \n",
      "79622                              0.000000   \n",
      "\n",
      "       dee5c1f1-db2b-44cc-8b1f-5867a736055a  \\\n",
      "0                                399.004088   \n",
      "1                                197.864570   \n",
      "2                                967.579738   \n",
      "3                               2207.434253   \n",
      "4                               2248.637670   \n",
      "...                                     ...   \n",
      "79618                              0.000000   \n",
      "79619                              0.000000   \n",
      "79620                              0.000000   \n",
      "79621                              0.000000   \n",
      "79622                              0.000000   \n",
      "\n",
      "       df187a91-a1f5-42ad-bb2f-8ca57fa5dffc  \\\n",
      "0                                103.778707   \n",
      "1                                122.812911   \n",
      "2                                778.050023   \n",
      "3                               1445.331309   \n",
      "4                                261.313051   \n",
      "...                                     ...   \n",
      "79618                              0.000000   \n",
      "79619                              0.000000   \n",
      "79620                              0.000000   \n",
      "79621                              0.000000   \n",
      "79622                              0.000000   \n",
      "\n",
      "       e1ac23ad-cdcf-4195-9ed8-1c40d9729a17  \\\n",
      "0                                665.495999   \n",
      "1                                928.906530   \n",
      "2                               4489.012639   \n",
      "3                               7655.490824   \n",
      "4                                557.364577   \n",
      "...                                     ...   \n",
      "79618                              0.000000   \n",
      "79619                              0.000000   \n",
      "79620                              0.000000   \n",
      "79621                              0.000000   \n",
      "79622                              0.000000   \n",
      "\n",
      "       e730e97c-c9af-45a1-9bed-3ea3d5a31075  \\\n",
      "0                                223.525485   \n",
      "1                                343.155850   \n",
      "2                               1457.514315   \n",
      "3                               2906.078476   \n",
      "4                                669.472665   \n",
      "...                                     ...   \n",
      "79618                              0.000000   \n",
      "79619                              0.000000   \n",
      "79620                              0.000000   \n",
      "79621                              0.000000   \n",
      "79622                              0.000000   \n",
      "\n",
      "       e907b5dd-1f1f-4d20-b049-b300c812406e  \\\n",
      "0                                140.104095   \n",
      "1                                124.967462   \n",
      "2                               2119.276999   \n",
      "3                               4662.680497   \n",
      "4                                820.933023   \n",
      "...                                     ...   \n",
      "79618                              0.000000   \n",
      "79619                              0.000000   \n",
      "79620                              0.000000   \n",
      "79621                              0.000000   \n",
      "79622                              0.000000   \n",
      "\n",
      "       ec6de687-aed9-43a8-a158-12f183fbf917  \\\n",
      "0                                334.266232   \n",
      "1                               1179.299206   \n",
      "2                               1840.466982   \n",
      "3                               3005.896156   \n",
      "4                                756.218123   \n",
      "...                                     ...   \n",
      "79618                              0.000000   \n",
      "79619                              0.000000   \n",
      "79620                              0.000000   \n",
      "79621                              0.000000   \n",
      "79622                              0.000000   \n",
      "\n",
      "       ef895091-fa01-458e-9279-09ec9d38b17b  \\\n",
      "0                                384.452038   \n",
      "1                                293.756561   \n",
      "2                                837.456873   \n",
      "3                               1757.817417   \n",
      "4                               1283.900336   \n",
      "...                                     ...   \n",
      "79618                              0.000000   \n",
      "79619                              0.000000   \n",
      "79620                              0.000000   \n",
      "79621                              0.000000   \n",
      "79622                              0.000000   \n",
      "\n",
      "       f27ac5c0-97f7-4f30-b762-992a5f393f6b  \n",
      "0                                172.439142  \n",
      "1                                264.954075  \n",
      "2                               1451.713676  \n",
      "3                               2638.273625  \n",
      "4                                353.844002  \n",
      "...                                     ...  \n",
      "79618                              0.000000  \n",
      "79619                              0.000000  \n",
      "79620                              0.000000  \n",
      "79621                              0.000000  \n",
      "79622                              0.000000  \n",
      "\n",
      "[79623 rows x 83 columns]\n",
      "40  patients\n",
      "lncRNA:  16105\n",
      "PROTEIN CODING:  16234\n",
      "NOVEL:  315\n",
      "liver_adjacent_totalRNA_LIHC\n",
      "11  patients\n",
      "lncRNA:  8135\n",
      "PROTEIN CODING:  14870\n",
      "NOVEL:  243\n",
      "hcc_normal_totalRNA_LIHC\n",
      "50  patients\n",
      "lncRNA:  15697\n",
      "PROTEIN CODING:  16965\n",
      "NOVEL:  327\n",
      "GSE193567_LIHC\n",
      "9  patients\n",
      "lncRNA:  5366\n",
      "PROTEIN CODING:  14292\n",
      "NOVEL:  188\n",
      "LIHC_TCGA_LIHC\n",
      "47  patients\n",
      "lncRNA:  10557\n",
      "PROTEIN CODING:  16146\n",
      "NOVEL:  208\n"
     ]
    }
   ],
   "source": [
    "## filter patients we are interested in, from those not TCGA\n",
    "for proj in all_projects:\n",
    "    print(proj)\n",
    "    if proj in tcga_projects:\n",
    "        cancer_type = proj[5:]\n",
    "\n",
    "        fc = pd.read_csv(os.path.join(cancer_dir,\"featureCounts/table_of_counts_TPMs_\")+proj+\".csv\")\n",
    "        tumor_transcripts = list()\n",
    "        patients=pd.read_csv(os.path.join(\"/users/genomics/marta/TCGA_RNASeq\",proj,\"results/QC_patients1.csv\"))\n",
    "        print(len(patients),\" patients\")\n",
    "        # patients['normal'] = patients['normal'].str.replace('TCGA', cancer_type)\n",
    "        # patients['tumor'] = patients['tumor'].str.replace('TCGA', cancer_type)\n",
    "\n",
    "    elif proj in other_projects:\n",
    "        fc = pd.read_csv(os.path.join(cancer_dir,\"featureCounts/table_of_counts_TPMs_\")+proj+\".csv\")\n",
    "        print(fc)\n",
    "        tumor_transcripts = list()\n",
    "        patients=pd.read_csv(os.path.join(\"/users/genomics/marta/cancers_RNASeq\",proj,\"results/patients.csv\"))\n",
    "        print(len(patients),\" patients\")\n",
    "\n",
    "        samples_to_keep = patients.tumor.values.tolist() + patients.normal.values.tolist() + ['Length','transcript_id']\n",
    "        fc_selected = fc[fc.columns.intersection(samples_to_keep)]\n",
    "\n",
    "    elif proj in manuscript_projects:\n",
    "        fc = pd.read_csv(os.path.join(cancer_dir,\"featureCounts/table_of_counts_TPMs_\")+proj+\".csv\")\n",
    "        tumor_transcripts = list()\n",
    "\n",
    "        if \"liver_adjacent_totalRNA\" in proj:\n",
    "            patients=pd.read_csv(os.path.join(\"/projects_eg/projects/marta/liver_adjacent_totalRNA/results/clean_patients.csv\"))\n",
    "            # patients.rename(columns={'adjacent':'normal'}, inplace=True)\n",
    "        elif \"hcc_normal_totalRNA\" in proj:\n",
    "            patients=pd.read_csv(os.path.join(\"/projects_eg/projects/marta/hcc_normal_totalRNA/results/clean_patients.csv\"))\n",
    "            # patients.rename(columns={'adjacent':'normal'}, inplace=True)\n",
    "        elif \"GSE193567\" in proj:\n",
    "            patients=pd.read_csv(os.path.join(\"/projects_eg/projects/marta/GSE193567/results/clean_patients.csv\"))        \n",
    "            # patients.rename(columns={'adjacent':'normal'}, inplace=True)\n",
    "        elif \"LIHC_TCGA\" in proj:\n",
    "            patients=pd.read_csv(os.path.join(\"/projects_eg/projects/marta/LIHC_TCGA/results/clean_patients.csv\"))    \n",
    "            patients['normal'] = patients['patient'] + \"_normal\"\n",
    "            patients['tumor'] = patients['patient'] + \"_tumor\"\n",
    "        print(len(patients),\" patients\")\n",
    "\n",
    "    for index,patient in patients.iterrows():\n",
    "        if proj in tcga_projects:\n",
    "            patient_fc = fc[[\"transcript_id\",\"Length\",patient['tumor']]]\n",
    "\n",
    "            tumor_patient_fc = patient_fc[patient_fc[patient['tumor']] >= 1 ]\n",
    "            tumor_transcripts.extend(tumor_patient_fc.transcript_id.values.tolist())\n",
    "        if proj in other_projects:\n",
    "            patient_fc = fc[[\"transcript_id\",\"Length\",patient['tumor']]]\n",
    "\n",
    "            tumor_patient_fc = patient_fc[patient_fc[patient['tumor']] >= 1 ]\n",
    "            tumor_transcripts.extend(tumor_patient_fc.transcript_id.values.tolist())       \n",
    "        if proj in manuscript_projects:\n",
    "            patient_fc = fc[[\"transcript_id\",\"Length\",patient['tumor']]]\n",
    "\n",
    "            tumor_patient_fc = patient_fc[patient_fc[patient['tumor']] >= 1 ]\n",
    "            tumor_transcripts.extend(tumor_patient_fc.transcript_id.values.tolist())                        \n",
    "\n",
    "    tumor1FPKM = fc[fc['transcript_id'].isin(tumor_transcripts)]\n",
    "    # print(tumor1FPKM)\n",
    "    known = tumor1FPKM[tumor1FPKM['transcript_id'].str.contains('ENST')]\n",
    "    novel = tumor1FPKM[tumor1FPKM['transcript_id'].str.contains('TCONS')]\n",
    "    merged = tumor1FPKM.merge(transcript_gene, on=['transcript_id'], how=\"inner\")\n",
    "    merged = merged[merged['gene_type'].isin(['lncRNA','processed_pseudogene','novel','protein_coding'])]\n",
    "\n",
    "    lncRNA = merged[merged['gene_type'].isin(['lncRNA','processed_pseudogene'])]\n",
    "    print(\"lncRNA: \",len(lncRNA))\n",
    "\n",
    "    cds = merged[merged['gene_type'] == \"protein_coding\"]\n",
    "    print(\"PROTEIN CODING: \",len(cds))\n",
    "\n",
    "    novel = merged[merged['gene_type'] == \"novel\"]\n",
    "    ## length limitations\n",
    "    novel = novel[novel['Length'].astype(int) < 91667]\n",
    "    novel = novel[novel['Length'].astype(int) > 300]\n",
    "\n",
    "    print(\"NOVEL: \",len(novel))\n",
    "\n",
    "    # # tumor1FPKM = pd.concat([novel, lncRNA, cds], ignore_index = True)\n",
    "    # merged[['transcript_id','gene_id','gene_name','gene_type','Length']].to_csv(os.path.join(cancer_dir,\"tumorexpressed/tumor_1FPKM_\")+cancer_type+\".csv\",index=False)\n",
    "\n",
    "    ## keep only tumor samples\n",
    "    if proj in tcga_projects:\n",
    "        tumorONLY_merged = merged[merged.columns.drop(list(merged.filter(regex='normal')))]\n",
    "    else:\n",
    "        tumorONLY_merged = merged.drop(columns=patients.normal.values.tolist())\n",
    "\n",
    "    tumorONLY_merged = tumorONLY_merged[['transcript_id','gene_id','gene_id', 'gene_type', 'gene_name','Length', *patients.tumor.values.tolist()]]\n",
    "    tumorONLY_merged.to_csv(os.path.join(cancer_dir,\"tumorexpressed/tumor_1FPKM_table_of_counts_\")+proj+\".csv\",index=False)\n",
    "\n",
    "    ###### GET TUMOR-EXPRESSED > 1 TPM\n",
    "    # Filter only integer columns\n",
    "    int_columns = tumorONLY_merged.select_dtypes(include='int64')\n",
    "    \n",
    "    # Create a new column with the count of int columns per row whose value is > 1\n",
    "    # tumorONLY_merged['n'] = int_columns.apply(lambda row: (row > 1).sum(), axis=1)\n",
    "    tumorONLY_merged['n'] = tumorONLY_merged.drop(columns=['transcript_id','gene_id','gene_name','gene_type','Length']).apply(count_greater_than_one, axis=1)\n",
    "    tumorONLY_merged = tumorONLY_merged[['transcript_id','gene_id','gene_name','gene_type','Length','n']]\n",
    "    tumorONLY_merged.sort_values(by=['n'], ascending=False).to_csv(os.path.join(cancer_dir,\"tumorexpressed/tumor_1FPKM_n_\")+proj+\".csv\",index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "BRCA\n",
      "TCGA-BRCA\n",
      "BLCA\n",
      "TCGA-BLCA\n",
      "GSE133624_BLCA\n",
      "LUAD\n",
      "TCGA-LUAD\n",
      "GSE229705_LUAD\n",
      "KIRC\n",
      "TCGA-KIRC\n",
      "GSE102101_KIRC\n",
      "SRP238334_KIRC\n",
      "PRAD\n",
      "TCGA-PRAD\n",
      "GSE22260_PRAD\n",
      "PRJEB2449_PRAD\n",
      "LUSC\n",
      "TCGA-LUSC\n",
      "LIHC\n",
      "GSE214846_LIHC\n",
      "liver_adjacent_totalRNA_LIHC\n",
      "hcc_normal_totalRNA_LIHC\n",
      "GSE193567_LIHC\n",
      "LIHC_TCGA_LIHC\n",
      "COAD\n",
      "SRP107326_COAD\n",
      "TCGA_COAD\n"
     ]
    }
   ],
   "source": [
    "## create a merged table of counts per ctype and a big patients file\n",
    "## CAN THEY BE CONSIDERED AS ONE?\n",
    "for ctype in cancertypes:\n",
    "    print(ctype)\n",
    "    \n",
    "    # Lists to store DataFrames for counts and patients\n",
    "    fc_list = []\n",
    "    patients_list = []\n",
    "    \n",
    "    for proj in all_projects:\n",
    "        if proj in tcga_projects and ctype in proj:\n",
    "            print(proj)\n",
    "            # Read the CSV files for TCGA projects\n",
    "            patients = pd.read_csv(os.path.join(\"/users/genomics/marta/TCGA_RNASeq\", proj, \"results/QC_patients1.csv\"))\n",
    "            patients['project'] = proj  # Add the project column\n",
    "            fc = pd.read_csv(os.path.join(cancer_dir, \"featureCounts\", f\"table_of_counts_TPMs_{proj}.csv\"))\n",
    "            fc_list.append(fc)\n",
    "            patients_list.append(patients)\n",
    "        \n",
    "        elif proj in other_projects and ctype in proj:\n",
    "            print(proj)\n",
    "            # Read the CSV files for other projects\n",
    "            patients = pd.read_csv(os.path.join(\"/users/genomics/marta/cancers_RNASeq\", proj, \"results/patients.csv\"))\n",
    "            patients['project'] = proj  # Add the project column\n",
    "            fc = pd.read_csv(os.path.join(cancer_dir, \"featureCounts\", f\"table_of_counts_TPMs_{proj}.csv\"))\n",
    "            fc_list.append(fc)\n",
    "            patients_list.append(patients)\n",
    "        \n",
    "        elif proj in manuscript_projects and ctype in proj:\n",
    "            print(proj)\n",
    "            # Read the CSV files for manuscript projects\n",
    "            if proj == \"LIHC_TCGA_LIHC\":\n",
    "                patients = pd.read_csv(os.path.join(\"/projects_eg/projects/marta\", proj[:-5], \"results/clean_patients.csv\"))\n",
    "                patients['normal'] = patients['patient'] + \"_normal\"\n",
    "                patients['tumor'] = patients['patient'] + \"_tumor\"\n",
    "\n",
    "            else:\n",
    "                patients = pd.read_csv(os.path.join(\"/projects_eg/projects/marta\", proj[:-5], \"results/clean_patients.csv\"))\n",
    "\n",
    "            patients['project'] = proj  # Add the project column\n",
    "            fc = pd.read_csv(os.path.join(cancer_dir, \"featureCounts\", f\"table_of_counts_TPMs_{proj}.csv\"))\n",
    "            fc_list.append(fc)\n",
    "            patients_list.append(patients)\n",
    "    \n",
    "    # Merge all fc DataFrames by 'transcript_id' and 'Length'\n",
    "    if fc_list:\n",
    "        merged_fc = reduce(lambda left, right: pd.merge(left, right, on=[\"transcript_id\", \"Length\"], how='outer'), fc_list)\n",
    "        # Save the merged fc DataFrame\n",
    "        merged_fc.to_csv(os.path.join(cancer_dir, f\"merged_fc_{ctype}.csv\"), index=False)\n",
    "    \n",
    "    # Combine all patient DataFrames into one big DataFrame\n",
    "    if patients_list:\n",
    "        merged_patients = pd.concat(patients_list, ignore_index=True)\n",
    "        # Save the merged patients DataFrame\n",
    "        merged_patients.to_csv(os.path.join(cancer_dir, f\"merged_patients_{ctype}.csv\"), index=False)\n"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Here to run Q3.1 - TPM distribution to know if we can treat them as one per cancer type or if there is bias according to the dataset"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Instead of selecting the tumor-specific, let's look first at the log2ratio between means\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Unnamed: 0</th>\n",
       "      <th>gene_id</th>\n",
       "      <th>gene_name</th>\n",
       "      <th>orfID</th>\n",
       "      <th>transcript_id</th>\n",
       "      <th>gene_type</th>\n",
       "      <th>orfType</th>\n",
       "      <th>length</th>\n",
       "      <th>geneORFtype</th>\n",
       "      <th>length_aa</th>\n",
       "      <th>start_codon</th>\n",
       "      <th>ORFpep</th>\n",
       "      <th>TranslatedLiver</th>\n",
       "      <th>TranslatedBrain</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>0</td>\n",
       "      <td>ENSG00000039600</td>\n",
       "      <td>SOX30</td>\n",
       "      <td>ENST00000265007.11:5:-|14|3283:361:2623|canoni...</td>\n",
       "      <td>ENST00000265007</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>canonical</td>\n",
       "      <td>2262</td>\n",
       "      <td>protein_coding_canonical</td>\n",
       "      <td>754</td>\n",
       "      <td>ATG</td>\n",
       "      <td>MERARPEPPPQPRPLRPAPPPLPVEGTSFWAAAMEPPPSSPTLSAA...</td>\n",
       "      <td>no</td>\n",
       "      <td>no</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>ENSG00000039600</td>\n",
       "      <td>SOX30</td>\n",
       "      <td>ENST00000265007.11:5:-|9|3283:233:491|ouORF|CTG</td>\n",
       "      <td>ENST00000265007</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>ouORF</td>\n",
       "      <td>258</td>\n",
       "      <td>protein_coding_ouORF</td>\n",
       "      <td>86</td>\n",
       "      <td>CTG</td>\n",
       "      <td>MRFEPRRVLGSKGLARLLTERGRGKVANKRLSSQSWPGRPPPPWRE...</td>\n",
       "      <td>no</td>\n",
       "      <td>no</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>2</td>\n",
       "      <td>ENSG00000046774</td>\n",
       "      <td>MAGEC2</td>\n",
       "      <td>ENST00000247452.4:X:-|15|1994:349:1471|canonic...</td>\n",
       "      <td>ENST00000247452</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>canonical</td>\n",
       "      <td>1122</td>\n",
       "      <td>protein_coding_canonical</td>\n",
       "      <td>374</td>\n",
       "      <td>ATG</td>\n",
       "      <td>MPPVPGVPFRNVDNDSPTSVELEDWVDAQHPTDEEEEEASSASSTL...</td>\n",
       "      <td>no</td>\n",
       "      <td>no</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>3</td>\n",
       "      <td>ENSG00000046774</td>\n",
       "      <td>MAGEC2</td>\n",
       "      <td>ENST00000247452.4:X:-|2|1994:165:240|uORF|CTG</td>\n",
       "      <td>ENST00000247452</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>uORF</td>\n",
       "      <td>75</td>\n",
       "      <td>protein_coding_uORF</td>\n",
       "      <td>25</td>\n",
       "      <td>CTG</td>\n",
       "      <td>MYCAAVRLVLQEPGGDELGVRHTA*</td>\n",
       "      <td>no</td>\n",
       "      <td>no</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>4</td>\n",
       "      <td>ENSG00000046774</td>\n",
       "      <td>MAGEC2</td>\n",
       "      <td>ENST00000247452.4:X:-|1|1994:91:148|uORF|CTG</td>\n",
       "      <td>ENST00000247452</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>uORF</td>\n",
       "      <td>57</td>\n",
       "      <td>protein_coding_uORF</td>\n",
       "      <td>19</td>\n",
       "      <td>CTG</td>\n",
       "      <td>MASPQGEGPEEELRDLPP*</td>\n",
       "      <td>no</td>\n",
       "      <td>no</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>936</th>\n",
       "      <td>981</td>\n",
       "      <td>XLOC_000999</td>\n",
       "      <td>XLOC_000999</td>\n",
       "      <td>TCONS_00001142:9:-|259|6071:5083:5176|noncodin...</td>\n",
       "      <td>TCONS_00001142</td>\n",
       "      <td>novel</td>\n",
       "      <td>noncoding</td>\n",
       "      <td>93</td>\n",
       "      <td>novel_noncoding</td>\n",
       "      <td>31</td>\n",
       "      <td>ATG</td>\n",
       "      <td>MPSLKFGKEALSFIHSLPFGFLSWRKFFII*</td>\n",
       "      <td>no</td>\n",
       "      <td>no</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>937</th>\n",
       "      <td>982</td>\n",
       "      <td>XLOC_001006</td>\n",
       "      <td>XLOC_001006</td>\n",
       "      <td>TCONS_00001162:9:-|216|3510:2703:2862|noncodin...</td>\n",
       "      <td>TCONS_00001162</td>\n",
       "      <td>novel</td>\n",
       "      <td>noncoding</td>\n",
       "      <td>159</td>\n",
       "      <td>novel_noncoding</td>\n",
       "      <td>53</td>\n",
       "      <td>ATG</td>\n",
       "      <td>MALPEGATHWRHLGLLSQMVFQQVGVGISLQRPGPAPEGGEVSLGL...</td>\n",
       "      <td>no</td>\n",
       "      <td>no</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>938</th>\n",
       "      <td>983</td>\n",
       "      <td>XLOC_001006</td>\n",
       "      <td>XLOC_001006</td>\n",
       "      <td>TCONS_00001162:9:-|252|3510:3325:3391|noncodin...</td>\n",
       "      <td>TCONS_00001162</td>\n",
       "      <td>novel</td>\n",
       "      <td>noncoding</td>\n",
       "      <td>66</td>\n",
       "      <td>novel_noncoding</td>\n",
       "      <td>22</td>\n",
       "      <td>TTG</td>\n",
       "      <td>MPISTEGQKQGNEPPSGYRKE*</td>\n",
       "      <td>no</td>\n",
       "      <td>no</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>939</th>\n",
       "      <td>984</td>\n",
       "      <td>XLOC_001006</td>\n",
       "      <td>XLOC_001006</td>\n",
       "      <td>TCONS_00001162:9:-|83|3510:982:1030|noncoding|ATG</td>\n",
       "      <td>TCONS_00001162</td>\n",
       "      <td>novel</td>\n",
       "      <td>noncoding</td>\n",
       "      <td>48</td>\n",
       "      <td>novel_noncoding</td>\n",
       "      <td>16</td>\n",
       "      <td>ATG</td>\n",
       "      <td>MLPLGSVFPEPRRLR*</td>\n",
       "      <td>no</td>\n",
       "      <td>no</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>940</th>\n",
       "      <td>985</td>\n",
       "      <td>XLOC_001006</td>\n",
       "      <td>XLOC_001006</td>\n",
       "      <td>TCONS_00001162:9:-|227|3510:2871:2922|noncodin...</td>\n",
       "      <td>TCONS_00001162</td>\n",
       "      <td>novel</td>\n",
       "      <td>noncoding</td>\n",
       "      <td>51</td>\n",
       "      <td>novel_noncoding</td>\n",
       "      <td>17</td>\n",
       "      <td>GTG</td>\n",
       "      <td>VATGALFLSSCSWVST*</td>\n",
       "      <td>no</td>\n",
       "      <td>no</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>941 rows × 14 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     Unnamed: 0          gene_id    gene_name  \\\n",
       "0             0  ENSG00000039600        SOX30   \n",
       "1             1  ENSG00000039600        SOX30   \n",
       "2             2  ENSG00000046774       MAGEC2   \n",
       "3             3  ENSG00000046774       MAGEC2   \n",
       "4             4  ENSG00000046774       MAGEC2   \n",
       "..          ...              ...          ...   \n",
       "936         981      XLOC_000999  XLOC_000999   \n",
       "937         982      XLOC_001006  XLOC_001006   \n",
       "938         983      XLOC_001006  XLOC_001006   \n",
       "939         984      XLOC_001006  XLOC_001006   \n",
       "940         985      XLOC_001006  XLOC_001006   \n",
       "\n",
       "                                                 orfID    transcript_id  \\\n",
       "0    ENST00000265007.11:5:-|14|3283:361:2623|canoni...  ENST00000265007   \n",
       "1      ENST00000265007.11:5:-|9|3283:233:491|ouORF|CTG  ENST00000265007   \n",
       "2    ENST00000247452.4:X:-|15|1994:349:1471|canonic...  ENST00000247452   \n",
       "3        ENST00000247452.4:X:-|2|1994:165:240|uORF|CTG  ENST00000247452   \n",
       "4         ENST00000247452.4:X:-|1|1994:91:148|uORF|CTG  ENST00000247452   \n",
       "..                                                 ...              ...   \n",
       "936  TCONS_00001142:9:-|259|6071:5083:5176|noncodin...   TCONS_00001142   \n",
       "937  TCONS_00001162:9:-|216|3510:2703:2862|noncodin...   TCONS_00001162   \n",
       "938  TCONS_00001162:9:-|252|3510:3325:3391|noncodin...   TCONS_00001162   \n",
       "939  TCONS_00001162:9:-|83|3510:982:1030|noncoding|ATG   TCONS_00001162   \n",
       "940  TCONS_00001162:9:-|227|3510:2871:2922|noncodin...   TCONS_00001162   \n",
       "\n",
       "          gene_type    orfType  length               geneORFtype  length_aa  \\\n",
       "0    protein_coding  canonical    2262  protein_coding_canonical        754   \n",
       "1    protein_coding      ouORF     258      protein_coding_ouORF         86   \n",
       "2    protein_coding  canonical    1122  protein_coding_canonical        374   \n",
       "3    protein_coding       uORF      75       protein_coding_uORF         25   \n",
       "4    protein_coding       uORF      57       protein_coding_uORF         19   \n",
       "..              ...        ...     ...                       ...        ...   \n",
       "936           novel  noncoding      93           novel_noncoding         31   \n",
       "937           novel  noncoding     159           novel_noncoding         53   \n",
       "938           novel  noncoding      66           novel_noncoding         22   \n",
       "939           novel  noncoding      48           novel_noncoding         16   \n",
       "940           novel  noncoding      51           novel_noncoding         17   \n",
       "\n",
       "    start_codon                                             ORFpep  \\\n",
       "0           ATG  MERARPEPPPQPRPLRPAPPPLPVEGTSFWAAAMEPPPSSPTLSAA...   \n",
       "1           CTG  MRFEPRRVLGSKGLARLLTERGRGKVANKRLSSQSWPGRPPPPWRE...   \n",
       "2           ATG  MPPVPGVPFRNVDNDSPTSVELEDWVDAQHPTDEEEEEASSASSTL...   \n",
       "3           CTG                          MYCAAVRLVLQEPGGDELGVRHTA*   \n",
       "4           CTG                                MASPQGEGPEEELRDLPP*   \n",
       "..          ...                                                ...   \n",
       "936         ATG                    MPSLKFGKEALSFIHSLPFGFLSWRKFFII*   \n",
       "937         ATG  MALPEGATHWRHLGLLSQMVFQQVGVGISLQRPGPAPEGGEVSLGL...   \n",
       "938         TTG                             MPISTEGQKQGNEPPSGYRKE*   \n",
       "939         ATG                                   MLPLGSVFPEPRRLR*   \n",
       "940         GTG                                  VATGALFLSSCSWVST*   \n",
       "\n",
       "    TranslatedLiver TranslatedBrain  \n",
       "0                no              no  \n",
       "1                no              no  \n",
       "2                no              no  \n",
       "3                no              no  \n",
       "4                no              no  \n",
       "..              ...             ...  \n",
       "936              no              no  \n",
       "937              no              no  \n",
       "938              no              no  \n",
       "939              no              no  \n",
       "940              no              no  \n",
       "\n",
       "[941 rows x 14 columns]"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "testisRestr = pd.read_csv(\"/projects_eg/projects/marta/TestisRestricted_Microproteins_TSA/with_TranscriptomeReconstruction/Multimap_altORFs/Q2_TestisRestricted/human/testisRestricted_GTEx_translatedONLYtestis.noProteome.csv\")\n",
    "testisRestr"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "## log2ratio3x & 1 TPM\n",
    "log2ratio3x_general = pd.DataFrame(columns=[\"gene_name\",\"gene_id\",\"gene_type\"])\n",
    "log2ratio3x_1TPM_general = pd.DataFrame(columns=[\"gene_name\",\"gene_id\",\"gene_type\",\"ctype\"])\n",
    "log2ratio3x_1TPM_general_5percent = pd.DataFrame(columns=[\"gene_name\",\"gene_id\",\"gene_type\"])\n",
    "\n",
    "for ctype in cancertypes:\n",
    "\n",
    "    print(ctype)\n",
    "    tumorspecific_candidates = pd.DataFrame()\n",
    "\n",
    "    patients = pd.read_csv(cancer_dir+\"/merged_patients_\"+ctype+\".csv\")\n",
    "    patients_long = patients.melt(id_vars=['patient', 'project'], value_vars=['normal', 'tumor'], \n",
    "                    var_name='normal_tumor', value_name='sample')\n",
    "    patients_long['normal_tumor'] = patients_long['normal_tumor'].replace({'normal': 'normal', 'tumor': 'tumor'})\n",
    "    \n",
    "    fc=pd.read_csv(cancer_dir+\"/merged_fc_\"+ctype+\".csv\")\n",
    "\n",
    "    fc_info = fc.merge(transcript_gene, on=['transcript_id'], how=\"inner\")\n",
    "    fc_info = fc_info[fc_info['gene_type'].isin(['lncRNA','processed_pseudogene','novel','protein_coding'])]\n",
    "\n",
    "    fc_info_long = pd.melt(fc_info, id_vars=[\"transcript_id\",\"Length\", \"gene_id\", \"gene_type\", \"gene_name\"], ignore_index=False).reset_index()\n",
    "    fc_info_long.rename(columns={\"variable\":\"sample\",\"value\":\"TPM\"}, inplace=True)\n",
    "    fc_info_long = fc_info_long.merge(patients_long, on=[\"sample\"])\n",
    "\n",
    "    means = fc_info_long.groupby([\"gene_name\",\"gene_id\",\"gene_type\",\"normal_tumor\"])['TPM'].mean().reset_index()\n",
    "    means_pivot = means.pivot_table(index=['gene_name', 'gene_id', 'gene_type'], \n",
    "                          columns='normal_tumor', \n",
    "                          values='TPM').reset_index()\n",
    "\n",
    "    # Avoid division by zero by adding a small constant (epsilon)\n",
    "    epsilon = 1e-6  # A small value to prevent division by zero\n",
    "    means_pivot['tumor'] = pd.to_numeric(means_pivot['tumor'], errors='coerce')\n",
    "    means_pivot['normal'] = pd.to_numeric(means_pivot['normal'], errors='coerce')\n",
    "    means_pivot['log2ratio'] = np.log2((means_pivot['tumor'] + epsilon) / (means_pivot['normal'] + epsilon))\n",
    "\n",
    "    log2ratio3x = means_pivot[means_pivot['log2ratio'] >= np.log2(3)]\n",
    "    log2ratio3x_testisRestr = log2ratio3x[log2ratio3x['gene_id'].isin(testisRestr.gene_id.values.tolist())]\n",
    "\n",
    "    # try:\n",
    "    #     os.makedirs(os.path.join(cancer_dir,\"log2ratio3x/cancertypes\",ctype))\n",
    "    # except:\n",
    "    #     continue\n",
    "    log2ratio3x_testisRestr.to_csv(os.path.join(cancer_dir,\"log2ratio3x/cancertypes\",ctype,\"log2ratio3x.csv\"), index=None)\n",
    "    log2ratio3x_general = pd.concat([log2ratio3x_general, log2ratio3x_testisRestr[[\"gene_name\",\"gene_id\",\"gene_type\"]]])\n",
    "\n",
    "    ### > 1 TPM\n",
    "    max = fc_info_long.groupby([\"gene_name\",\"gene_id\",\"gene_type\",\"normal_tumor\"])['TPM'].max().reset_index()\n",
    "    ## tumor samples max has to be greater than 1TPM\n",
    "    max = max[max['normal_tumor'] == \"tumor\"]\n",
    "    max.rename(columns={'TPM':'max_TPM'}, inplace=True)\n",
    "\n",
    "    TPM1 = max[max['max_TPM'] > 1]\n",
    "    log2ratio3x_1TPM = log2ratio3x_testisRestr[log2ratio3x_testisRestr['gene_id'].isin(TPM1.gene_id.values.tolist())]\n",
    "    log2ratio3x_1TPM.to_csv(os.path.join(cancer_dir,\"log2ratio3x/cancertypes\",ctype,\"log2ratio3x_1TPM.csv\"), index=None)\n",
    "    log2ratio3x_1TPM['ctype'] = ctype\n",
    "    log2ratio3x_1TPM_general = pd.concat([log2ratio3x_1TPM_general, log2ratio3x_1TPM[[\"gene_name\",\"gene_id\",\"gene_type\",\"ctype\"]]])\n",
    "\n",
    "    ### > 1 TPM - 5%\n",
    "    # TAA_1TPM5percent = pd.read_csv(os.path.join(cancer_dir,\"tumorexpressed/cancertypes/tumor_1FPKM_n5percent_pancancer.csv\"))\n",
    "    \n",
    "    # log2ratio3x_1TPM = log2ratio3x_testisRestr[log2ratio3x_testisRestr['gene_id'].isin(TAA_1TPM5percent.gene_id.values.tolist())]\n",
    "    # log2ratio3x_1TPM.to_csv(os.path.join(cancer_dir,\"log2ratio3x/cancertypes\",ctype,\"log2ratio3x_1TPM5percent.csv\"), index=None)\n",
    "    # log2ratio3x_1TPM_general_5percent = pd.concat([log2ratio3x_1TPM_general_5percent, log2ratio3x_1TPM[[\"gene_name\",\"gene_id\",\"gene_type\"]]])\n",
    "\n",
    "log2ratio3x_1TPM_general.to_csv(os.path.join(cancer_dir,\"log2ratio3x/cancertypes/log2ratio3x_1TPM_genes.csv\"), index=None)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene_name</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>gene_type</th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>lncRNA</th>\n",
       "      <td>184</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>novel</th>\n",
       "      <td>29</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>processed_pseudogene</th>\n",
       "      <td>15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>protein_coding</th>\n",
       "      <td>568</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                      gene_name\n",
       "gene_type                      \n",
       "lncRNA                      184\n",
       "novel                        29\n",
       "processed_pseudogene         15\n",
       "protein_coding              568"
      ]
     },
     "execution_count": 31,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "log2ratio3x_1TPM_general.drop_duplicates(inplace=True)\n",
    "log2ratio3x_1TPM_general[['gene_name','gene_type']].groupby(\"gene_type\").count()\n",
    "# log2ratio3x_1TPM_general"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene_name</th>\n",
       "      <th>length_aa</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>geneORFtype</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>lncRNA_noncoding</th>\n",
       "      <td>129</td>\n",
       "      <td>129</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>novel_noncoding</th>\n",
       "      <td>20</td>\n",
       "      <td>20</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>processed_pseudogene_noncoding</th>\n",
       "      <td>8</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>protein_coding_canonical</th>\n",
       "      <td>194</td>\n",
       "      <td>194</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>protein_coding_dORF</th>\n",
       "      <td>11</td>\n",
       "      <td>11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>protein_coding_odORF</th>\n",
       "      <td>5</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>protein_coding_ouORF</th>\n",
       "      <td>19</td>\n",
       "      <td>19</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>protein_coding_uORF</th>\n",
       "      <td>42</td>\n",
       "      <td>42</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                gene_name  length_aa\n",
       "geneORFtype                                         \n",
       "lncRNA_noncoding                      129        129\n",
       "novel_noncoding                        20         20\n",
       "processed_pseudogene_noncoding          8          8\n",
       "protein_coding_canonical              194        194\n",
       "protein_coding_dORF                    11         11\n",
       "protein_coding_odORF                    5          5\n",
       "protein_coding_ouORF                   19         19\n",
       "protein_coding_uORF                    42         42"
      ]
     },
     "execution_count": 37,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "## ORF level   \n",
    "log2ratio3x_1TPM_general = pd.read_csv(os.path.join(cancer_dir,\"log2ratio3x/cancertypes/log2ratio3x_1TPM_genes.csv\"))\n",
    "log2ratio3x_1TPM_general = log2ratio3x_1TPM_general.merge(transcript_gene[[\"gene_id\",\"transcript_id\"]], on=[\"gene_id\"])\n",
    "log2ratio3x_1TPM_general.to_csv(os.path.join(cancer_dir,\"log2ratio3x/cancertypes/onlyStep1/log2ratio3x_1TPM_genes.csv\"), index=None)\n",
    "\n",
    "tumorReact_ORFs = log2ratio3x_1TPM_general.merge(testisRestr, on=[\"transcript_id\",\"gene_name\",\"gene_id\",\"gene_type\"])\n",
    "tumorReact_ORFs = tumorReact_ORFs[['gene_name','gene_id','transcript_id','gene_type','orfID','length','ctype','orfType','geneORFtype','length_aa','start_codon','ORFpep']]\n",
    "tumorReact_ORFs.to_csv(os.path.join(cancer_dir,\"log2ratio3x/cancertypes/log2ratio3x_1TPM_ORFs.csv\"), index=None)\n",
    "tumorReact_ORFs.to_csv(os.path.join(cancer_dir,\"log2ratio3x/cancertypes/onlyStep1/log2ratio3x_1TPM_ORFs.csv\"), index=None)\n",
    "tumorReact_ORFs.groupby(\"geneORFtype\").count()\n",
    "tumorReact_ORFs[['gene_name','length_aa','geneORFtype']].drop_duplicates().groupby(\"geneORFtype\").count()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th>gene_name</th>\n",
       "      <th>length_aa</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ctype</th>\n",
       "      <th>geneORFtype</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th rowspan=\"5\" valign=\"top\">BLCA</th>\n",
       "      <th>lncRNA_noncoding</th>\n",
       "      <td>33</td>\n",
       "      <td>33</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>novel_noncoding</th>\n",
       "      <td>7</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>processed_pseudogene_noncoding</th>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>protein_coding_canonical</th>\n",
       "      <td>86</td>\n",
       "      <td>86</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>protein_coding_dORF</th>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"5\" valign=\"top\">PRAD</th>\n",
       "      <th>processed_pseudogene_noncoding</th>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>protein_coding_canonical</th>\n",
       "      <td>21</td>\n",
       "      <td>21</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>protein_coding_dORF</th>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>protein_coding_ouORF</th>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>protein_coding_uORF</th>\n",
       "      <td>9</td>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>61 rows × 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                      gene_name  length_aa\n",
       "ctype geneORFtype                                         \n",
       "BLCA  lncRNA_noncoding                       33         33\n",
       "      novel_noncoding                         7          7\n",
       "      processed_pseudogene_noncoding          1          1\n",
       "      protein_coding_canonical               86         86\n",
       "      protein_coding_dORF                     1          1\n",
       "...                                         ...        ...\n",
       "PRAD  processed_pseudogene_noncoding          1          1\n",
       "      protein_coding_canonical               21         21\n",
       "      protein_coding_dORF                     1          1\n",
       "      protein_coding_ouORF                    2          2\n",
       "      protein_coding_uORF                     9          9\n",
       "\n",
       "[61 rows x 2 columns]"
      ]
     },
     "execution_count": 34,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "tumorReact_ORFs[['gene_name','length_aa','geneORFtype','ctype']].drop_duplicates().groupby([\"ctype\",\"geneORFtype\"]).count()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "## get table of counts of candidates\n",
    "candidates = pd.read_csv(os.path.join(cancer_dir,\"log2ratio3x/cancertypes/log2ratio3x_1TPM_genes.csv\"))\n",
    "\n",
    "full_table_of_counts = pd.DataFrame()\n",
    "for ctype in cancertypes:\n",
    "    patients = pd.read_csv(cancer_dir+\"/merged_patients_\"+ctype+\".csv\")\n",
    "    patients_long = pd.melt(patients, id_vars=[\"patient\",\"project\"], ignore_index=False).reset_index()\n",
    "    patients_long.drop(\"index\", axis=1, inplace=True)\n",
    "    patients_long.rename(columns={\"variable\":\"normal_tumor\",\"value\":\"sample\"}, inplace=True)\n",
    "\n",
    "    fc=pd.read_csv(cancer_dir+\"/merged_fc_\"+ctype+\".csv\")\n",
    "\n",
    "    log2ratio_1TPM = pd.read_csv(os.path.join(cancer_dir,\"log2ratio3x/cancertypes\",ctype,\"log2ratio3x_1TPM.csv\"))\n",
    "    fc_info = fc.merge(transcript_gene, on=['transcript_id'], how=\"inner\")\n",
    "    fc_info = fc_info[fc_info['gene_type'].isin(['lncRNA','processed_pseudogene','novel','protein_coding'])]\n",
    "    # print(\"log2ratio 1TPM: \", len(fc_info))\n",
    "\n",
    "    fc_info_long = pd.melt(fc_info, id_vars=[\"transcript_id\",\"Length\", \"gene_id\", \"gene_type\", \"gene_name\"], ignore_index=False).reset_index()\n",
    "    fc_info_long.rename(columns={\"variable\":\"sample\",\"value\":\"TPM\"}, inplace=True)\n",
    "    fc_info_long = fc_info_long.merge(patients_long, on=[\"sample\"])\n",
    "\n",
    "    ## create full file with TPM per patient and ctype\n",
    "    fc_info_long['ctype'] = ctype\n",
    "\n",
    "    full_table_of_counts = pd.concat([full_table_of_counts, fc_info_long])\n",
    "\n",
    "full_table_of_counts.to_csv(os.path.join(cancer_dir,\"full_table_of_counts.csv\"))\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_3928551/2343107821.py:1: DtypeWarning: Columns (9) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  full_table_of_counts = pd.read_csv(os.path.join(cancer_dir,\"full_table_of_counts.csv\"))\n",
      "/tmp/ipykernel_3928551/2343107821.py:15: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  merged_tumor.drop(columns=['index','Length','sample'], inplace=True)\n",
      "/tmp/ipykernel_3928551/2343107821.py:19: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  merged_normal.drop(columns=['index','Length','sample'], inplace=True)\n"
     ]
    }
   ],
   "source": [
    "full_table_of_counts = pd.read_csv(os.path.join(cancer_dir,\"full_table_of_counts.csv\"))\n",
    "\n",
    "merged = pd.merge(\n",
    "    full_table_of_counts,\n",
    "    candidates[['gene_id']],  # Only keep relevant columns for the merge\n",
    "    left_on=['gene_id'],\n",
    "    right_on=['gene_id'],\n",
    "    how='left'\n",
    ")\n",
    "merged = merged[merged['gene_id'].isin(candidates.gene_id.values.tolist())]\n",
    "merged.to_csv(os.path.join(cancer_dir,\"log2ratio3x/cancertypes/onlyStep1/TSTR_Expression.csv\"), index=False)\n",
    "\n",
    "# Create the new column 'forced_TPM' using numpy.where\n",
    "merged_tumor = merged[merged['normal_tumor'] == \"tumor\"]\n",
    "merged_tumor.drop(columns=['index','Length','sample'], inplace=True)\n",
    "merged_tumor.to_csv(os.path.join(cancer_dir,\"log2ratio3x/cancertypes/onlyStep1/TSTR_Expression_tumor.csv\"), index=False)\n",
    "\n",
    "merged_normal = merged[merged['normal_tumor'] == \"normal\"]\n",
    "merged_normal.drop(columns=['index','Length','sample'], inplace=True)\n",
    "merged_normal.to_csv(os.path.join(cancer_dir,\"log2ratio3x/cancertypes/onlyStep1/TSTR_Expression_normal.csv\"), index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene_name</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>gene_type</th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>lncRNA</th>\n",
       "      <td>102</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>novel</th>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>processed_pseudogene</th>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>protein_coding</th>\n",
       "      <td>204</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                      gene_name\n",
       "gene_type                      \n",
       "lncRNA                      102\n",
       "novel                        13\n",
       "processed_pseudogene          7\n",
       "protein_coding              204"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "candidates[['gene_name','gene_type']].drop_duplicates().groupby(\"gene_type\").count()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th>gene_name</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>gene_type</th>\n",
       "      <th>ctype</th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th rowspan=\"8\" valign=\"top\">lncRNA</th>\n",
       "      <th>BLCA</th>\n",
       "      <td>25</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>BRCA</th>\n",
       "      <td>11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>COAD</th>\n",
       "      <td>38</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>KIRC</th>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>LIHC</th>\n",
       "      <td>42</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>LUAD</th>\n",
       "      <td>25</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>LUSC</th>\n",
       "      <td>21</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PRAD</th>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"8\" valign=\"top\">novel</th>\n",
       "      <th>BLCA</th>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>BRCA</th>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>COAD</th>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>KIRC</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>LIHC</th>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>LUAD</th>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>LUSC</th>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PRAD</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"6\" valign=\"top\">processed_pseudogene</th>\n",
       "      <th>BLCA</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>COAD</th>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>LIHC</th>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>LUAD</th>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>LUSC</th>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PRAD</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"8\" valign=\"top\">protein_coding</th>\n",
       "      <th>BLCA</th>\n",
       "      <td>90</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>BRCA</th>\n",
       "      <td>74</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>COAD</th>\n",
       "      <td>51</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>KIRC</th>\n",
       "      <td>27</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>LIHC</th>\n",
       "      <td>135</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>LUAD</th>\n",
       "      <td>73</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>LUSC</th>\n",
       "      <td>96</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PRAD</th>\n",
       "      <td>22</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                            gene_name\n",
       "gene_type            ctype           \n",
       "lncRNA               BLCA          25\n",
       "                     BRCA          11\n",
       "                     COAD          38\n",
       "                     KIRC          13\n",
       "                     LIHC          42\n",
       "                     LUAD          25\n",
       "                     LUSC          21\n",
       "                     PRAD           9\n",
       "novel                BLCA           2\n",
       "                     BRCA           3\n",
       "                     COAD           6\n",
       "                     KIRC           1\n",
       "                     LIHC           5\n",
       "                     LUAD           4\n",
       "                     LUSC           7\n",
       "                     PRAD           1\n",
       "processed_pseudogene BLCA           1\n",
       "                     COAD           3\n",
       "                     LIHC           5\n",
       "                     LUAD           3\n",
       "                     LUSC           2\n",
       "                     PRAD           1\n",
       "protein_coding       BLCA          90\n",
       "                     BRCA          74\n",
       "                     COAD          51\n",
       "                     KIRC          27\n",
       "                     LIHC         135\n",
       "                     LUAD          73\n",
       "                     LUSC          96\n",
       "                     PRAD          22"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "candidates[['gene_name','gene_type','ctype']].drop_duplicates().groupby([\"gene_type\",\"ctype\"]).count()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene_name</th>\n",
       "      <th>gene_id</th>\n",
       "      <th>transcript_id</th>\n",
       "      <th>gene_type</th>\n",
       "      <th>length</th>\n",
       "      <th>orfType</th>\n",
       "      <th>length_aa</th>\n",
       "      <th>start_codon</th>\n",
       "      <th>ORFpep</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>geneORFtype</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>lncRNA_noncoding</th>\n",
       "      <td>130</td>\n",
       "      <td>130</td>\n",
       "      <td>130</td>\n",
       "      <td>130</td>\n",
       "      <td>130</td>\n",
       "      <td>130</td>\n",
       "      <td>130</td>\n",
       "      <td>130</td>\n",
       "      <td>130</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>novel_noncoding</th>\n",
       "      <td>20</td>\n",
       "      <td>20</td>\n",
       "      <td>20</td>\n",
       "      <td>20</td>\n",
       "      <td>20</td>\n",
       "      <td>20</td>\n",
       "      <td>20</td>\n",
       "      <td>20</td>\n",
       "      <td>20</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>processed_pseudogene_noncoding</th>\n",
       "      <td>8</td>\n",
       "      <td>8</td>\n",
       "      <td>8</td>\n",
       "      <td>8</td>\n",
       "      <td>8</td>\n",
       "      <td>8</td>\n",
       "      <td>8</td>\n",
       "      <td>8</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>protein_coding_canonical</th>\n",
       "      <td>194</td>\n",
       "      <td>194</td>\n",
       "      <td>194</td>\n",
       "      <td>194</td>\n",
       "      <td>194</td>\n",
       "      <td>194</td>\n",
       "      <td>194</td>\n",
       "      <td>194</td>\n",
       "      <td>194</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>protein_coding_dORF</th>\n",
       "      <td>11</td>\n",
       "      <td>11</td>\n",
       "      <td>11</td>\n",
       "      <td>11</td>\n",
       "      <td>11</td>\n",
       "      <td>11</td>\n",
       "      <td>11</td>\n",
       "      <td>11</td>\n",
       "      <td>11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>protein_coding_odORF</th>\n",
       "      <td>5</td>\n",
       "      <td>5</td>\n",
       "      <td>5</td>\n",
       "      <td>5</td>\n",
       "      <td>5</td>\n",
       "      <td>5</td>\n",
       "      <td>5</td>\n",
       "      <td>5</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>protein_coding_ouORF</th>\n",
       "      <td>19</td>\n",
       "      <td>19</td>\n",
       "      <td>19</td>\n",
       "      <td>19</td>\n",
       "      <td>19</td>\n",
       "      <td>19</td>\n",
       "      <td>19</td>\n",
       "      <td>19</td>\n",
       "      <td>19</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>protein_coding_uORF</th>\n",
       "      <td>42</td>\n",
       "      <td>42</td>\n",
       "      <td>42</td>\n",
       "      <td>42</td>\n",
       "      <td>42</td>\n",
       "      <td>42</td>\n",
       "      <td>42</td>\n",
       "      <td>42</td>\n",
       "      <td>42</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                gene_name  gene_id  transcript_id  gene_type  \\\n",
       "geneORFtype                                                                    \n",
       "lncRNA_noncoding                      130      130            130        130   \n",
       "novel_noncoding                        20       20             20         20   \n",
       "processed_pseudogene_noncoding          8        8              8          8   \n",
       "protein_coding_canonical              194      194            194        194   \n",
       "protein_coding_dORF                    11       11             11         11   \n",
       "protein_coding_odORF                    5        5              5          5   \n",
       "protein_coding_ouORF                   19       19             19         19   \n",
       "protein_coding_uORF                    42       42             42         42   \n",
       "\n",
       "                                length  orfType  length_aa  start_codon  \\\n",
       "geneORFtype                                                               \n",
       "lncRNA_noncoding                   130      130        130          130   \n",
       "novel_noncoding                     20       20         20           20   \n",
       "processed_pseudogene_noncoding       8        8          8            8   \n",
       "protein_coding_canonical           194      194        194          194   \n",
       "protein_coding_dORF                 11       11         11           11   \n",
       "protein_coding_odORF                 5        5          5            5   \n",
       "protein_coding_ouORF                19       19         19           19   \n",
       "protein_coding_uORF                 42       42         42           42   \n",
       "\n",
       "                                ORFpep  \n",
       "geneORFtype                             \n",
       "lncRNA_noncoding                   130  \n",
       "novel_noncoding                     20  \n",
       "processed_pseudogene_noncoding       8  \n",
       "protein_coding_canonical           194  \n",
       "protein_coding_dORF                 11  \n",
       "protein_coding_odORF                 5  \n",
       "protein_coding_ouORF                19  \n",
       "protein_coding_uORF                 42  "
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "tumorReact_ORFs.drop(\"ctype\", axis=1).drop_duplicates().groupby(\"geneORFtype\").count()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'ADAD1',\n",
       " 'ADAM29',\n",
       " 'C12orf50',\n",
       " 'CCDC172',\n",
       " 'COX7B2',\n",
       " 'DCAF8L2',\n",
       " 'DYDC1',\n",
       " 'FSHR',\n",
       " 'IQCM',\n",
       " 'LRRC74A',\n",
       " 'LYPD4',\n",
       " 'MAGEA1',\n",
       " 'MAGEB1',\n",
       " 'MAGEB3',\n",
       " 'MAGEC1',\n",
       " 'MAGEC2',\n",
       " 'MEIOSIN',\n",
       " 'OR14A2',\n",
       " 'PLCZ1',\n",
       " 'PLSCR2',\n",
       " 'PRSS54',\n",
       " 'RFPL4B',\n",
       " 'RHOXF2B',\n",
       " 'SMIM47',\n",
       " 'SPANXB1',\n",
       " 'SPMIP9',\n",
       " 'TBC1D21',\n",
       " 'TEX13C',\n",
       " 'TPTE',\n",
       " 'TRPC5OS',\n",
       " 'USP26'}"
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "set(tumorReact_ORFs[tumorReact_ORFs['geneORFtype'] == \"protein_coding_uORF\"].gene_name.values.tolist())"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Generate bed/gtf file for candidates (TSTR) to look for chromatin acessibility (Cova)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "326\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_3928551/860995836.py:5: DtypeWarning: Columns (0) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  annotation_gtf = pd.read_csv(annotation, sep=\"\\t\", comment=\"#\", header=None)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>0</th>\n",
       "      <th>1</th>\n",
       "      <th>2</th>\n",
       "      <th>3</th>\n",
       "      <th>4</th>\n",
       "      <th>5</th>\n",
       "      <th>6</th>\n",
       "      <th>7</th>\n",
       "      <th>8</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>M</td>\n",
       "      <td>StringTie</td>\n",
       "      <td>transcript</td>\n",
       "      <td>1</td>\n",
       "      <td>577</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>.</td>\n",
       "      <td>transcript_id \"TCONS_00001163\"; gene_id \"XLOC_...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>M</td>\n",
       "      <td>ENSEMBL</td>\n",
       "      <td>transcript</td>\n",
       "      <td>577</td>\n",
       "      <td>647</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>gene_id \"ENSG00000210049.1\"; transcript_id \"EN...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>M</td>\n",
       "      <td>ENSEMBL</td>\n",
       "      <td>transcript</td>\n",
       "      <td>648</td>\n",
       "      <td>1601</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>gene_id \"ENSG00000211459.2\"; transcript_id \"EN...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>M</td>\n",
       "      <td>StringTie</td>\n",
       "      <td>transcript</td>\n",
       "      <td>683</td>\n",
       "      <td>1035</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>.</td>\n",
       "      <td>transcript_id \"TCONS_00001164\"; gene_id \"XLOC_...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>M</td>\n",
       "      <td>ENSEMBL</td>\n",
       "      <td>transcript</td>\n",
       "      <td>1602</td>\n",
       "      <td>1670</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>gene_id \"ENSG00000210077.1\"; transcript_id \"EN...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>782862</th>\n",
       "      <td>KI270751.1</td>\n",
       "      <td>HAVANA</td>\n",
       "      <td>transcript</td>\n",
       "      <td>34175</td>\n",
       "      <td>73722</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>.</td>\n",
       "      <td>gene_id \"ENSG00000303867.1\"; transcript_id \"EN...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>782865</th>\n",
       "      <td>KI270751.1</td>\n",
       "      <td>HAVANA</td>\n",
       "      <td>transcript</td>\n",
       "      <td>44958</td>\n",
       "      <td>53274</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>gene_id \"ENSG00000303902.1\"; transcript_id \"EN...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>782871</th>\n",
       "      <td>KI270751.1</td>\n",
       "      <td>HAVANA</td>\n",
       "      <td>transcript</td>\n",
       "      <td>133801</td>\n",
       "      <td>139253</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>.</td>\n",
       "      <td>gene_id \"ENSG00000306528.1\"; transcript_id \"EN...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>782875</th>\n",
       "      <td>KI270753.1</td>\n",
       "      <td>HAVANA</td>\n",
       "      <td>transcript</td>\n",
       "      <td>43135</td>\n",
       "      <td>44491</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>gene_id \"ENSG00000297844.1\"; transcript_id \"EN...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>782879</th>\n",
       "      <td>KI270755.1</td>\n",
       "      <td>HAVANA</td>\n",
       "      <td>transcript</td>\n",
       "      <td>26350</td>\n",
       "      <td>27723</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>gene_id \"ENSG00000309258.1\"; transcript_id \"EN...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>79623 rows × 9 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                 0          1           2       3       4  5  6  7  \\\n",
       "1                M  StringTie  transcript       1     577  .  -  .   \n",
       "3                M    ENSEMBL  transcript     577     647  .  +  .   \n",
       "7                M    ENSEMBL  transcript     648    1601  .  +  .   \n",
       "8                M  StringTie  transcript     683    1035  .  -  .   \n",
       "11               M    ENSEMBL  transcript    1602    1670  .  +  .   \n",
       "...            ...        ...         ...     ...     ... .. .. ..   \n",
       "782862  KI270751.1     HAVANA  transcript   34175   73722  .  -  .   \n",
       "782865  KI270751.1     HAVANA  transcript   44958   53274  .  +  .   \n",
       "782871  KI270751.1     HAVANA  transcript  133801  139253  .  -  .   \n",
       "782875  KI270753.1     HAVANA  transcript   43135   44491  .  +  .   \n",
       "782879  KI270755.1     HAVANA  transcript   26350   27723  .  +  .   \n",
       "\n",
       "                                                        8  \n",
       "1       transcript_id \"TCONS_00001163\"; gene_id \"XLOC_...  \n",
       "3       gene_id \"ENSG00000210049.1\"; transcript_id \"EN...  \n",
       "7       gene_id \"ENSG00000211459.2\"; transcript_id \"EN...  \n",
       "8       transcript_id \"TCONS_00001164\"; gene_id \"XLOC_...  \n",
       "11      gene_id \"ENSG00000210077.1\"; transcript_id \"EN...  \n",
       "...                                                   ...  \n",
       "782862  gene_id \"ENSG00000303867.1\"; transcript_id \"EN...  \n",
       "782865  gene_id \"ENSG00000303902.1\"; transcript_id \"EN...  \n",
       "782871  gene_id \"ENSG00000306528.1\"; transcript_id \"EN...  \n",
       "782875  gene_id \"ENSG00000297844.1\"; transcript_id \"EN...  \n",
       "782879  gene_id \"ENSG00000309258.1\"; transcript_id \"EN...  \n",
       "\n",
       "[79623 rows x 9 columns]"
      ]
     },
     "execution_count": 35,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "TSTR = pd.read_csv(os.path.join(cancer_dir,\"log2ratio3x/cancertypes/onlyStep1/log2ratio3x_1TPM_genes.csv\"))\n",
    "TSTR.drop(\"ctype\", axis=1).drop_duplicates().groupby(\"gene_type\").count()\n",
    "print(len(set(TSTR.transcript_id.values.tolist())))\n",
    "\n",
    "annotation_gtf = pd.read_csv(annotation, sep=\"\\t\", comment=\"#\", header=None)\n",
    "annotation_gtf = annotation_gtf[annotation_gtf[2] == \"transcript\"]\n",
    "annotation_gtf\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Convert the list of values into a regex pattern\n",
    "pattern = '|'.join(TSTR.transcript_id.values.tolist())\n",
    "\n",
    "# Filter the DataFrame\n",
    "filtered_annotation = annotation_gtf[annotation_gtf[8].str.contains(pattern)]\n",
    "\n",
    "filtered_annotation.to_csv(os.path.join(cancer_dir,\"log2ratio3x/cancertypes/onlyStep1/TSTR_annotation.gtf\"), sep=\"\\t\", index=None, header=None, quoting=csv.QUOTE_NONE)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chr</th>\n",
       "      <th>transcript_id</th>\n",
       "      <th>gene_id</th>\n",
       "      <th>gene_name</th>\n",
       "      <th>transcript_type</th>\n",
       "      <th>gene_type</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>M</td>\n",
       "      <td>ENST00000361390</td>\n",
       "      <td>ENSG00000198888</td>\n",
       "      <td>MT-ND1</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>protein_coding</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>M</td>\n",
       "      <td>ENST00000361453</td>\n",
       "      <td>ENSG00000198763</td>\n",
       "      <td>MT-ND2</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>protein_coding</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>M</td>\n",
       "      <td>ENST00000361624</td>\n",
       "      <td>ENSG00000198804</td>\n",
       "      <td>MT-CO1</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>protein_coding</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>M</td>\n",
       "      <td>ENST00000361739</td>\n",
       "      <td>ENSG00000198712</td>\n",
       "      <td>MT-CO2</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>protein_coding</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>M</td>\n",
       "      <td>ENST00000361851</td>\n",
       "      <td>ENSG00000228253</td>\n",
       "      <td>MT-ATP8</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>protein_coding</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>65159</th>\n",
       "      <td>KI270442.1</td>\n",
       "      <td>TCONS_00000004</td>\n",
       "      <td>XLOC_000003</td>\n",
       "      <td>XLOC_000003</td>\n",
       "      <td>novel</td>\n",
       "      <td>novel</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>65160</th>\n",
       "      <td>KI270442.1</td>\n",
       "      <td>TCONS_00000005</td>\n",
       "      <td>XLOC_000004</td>\n",
       "      <td>XLOC_000004</td>\n",
       "      <td>novel</td>\n",
       "      <td>novel</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>65161</th>\n",
       "      <td>KI270466.1</td>\n",
       "      <td>TCONS_00000006</td>\n",
       "      <td>XLOC_000005</td>\n",
       "      <td>XLOC_000005</td>\n",
       "      <td>novel</td>\n",
       "      <td>novel</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>65162</th>\n",
       "      <td>KI270467.1</td>\n",
       "      <td>TCONS_00000008</td>\n",
       "      <td>XLOC_000006</td>\n",
       "      <td>XLOC_000006</td>\n",
       "      <td>novel</td>\n",
       "      <td>novel</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>65163</th>\n",
       "      <td>KI270467.1</td>\n",
       "      <td>TCONS_00000007</td>\n",
       "      <td>XLOC_000007</td>\n",
       "      <td>XLOC_000007</td>\n",
       "      <td>novel</td>\n",
       "      <td>novel</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>65164 rows × 6 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "              chr    transcript_id          gene_id    gene_name  \\\n",
       "0               M  ENST00000361390  ENSG00000198888       MT-ND1   \n",
       "1               M  ENST00000361453  ENSG00000198763       MT-ND2   \n",
       "2               M  ENST00000361624  ENSG00000198804       MT-CO1   \n",
       "3               M  ENST00000361739  ENSG00000198712       MT-CO2   \n",
       "4               M  ENST00000361851  ENSG00000228253      MT-ATP8   \n",
       "...           ...              ...              ...          ...   \n",
       "65159  KI270442.1   TCONS_00000004      XLOC_000003  XLOC_000003   \n",
       "65160  KI270442.1   TCONS_00000005      XLOC_000004  XLOC_000004   \n",
       "65161  KI270466.1   TCONS_00000006      XLOC_000005  XLOC_000005   \n",
       "65162  KI270467.1   TCONS_00000008      XLOC_000006  XLOC_000006   \n",
       "65163  KI270467.1   TCONS_00000007      XLOC_000007  XLOC_000007   \n",
       "\n",
       "      transcript_type       gene_type  \n",
       "0      protein_coding  protein_coding  \n",
       "1      protein_coding  protein_coding  \n",
       "2      protein_coding  protein_coding  \n",
       "3      protein_coding  protein_coding  \n",
       "4      protein_coding  protein_coding  \n",
       "...               ...             ...  \n",
       "65159           novel           novel  \n",
       "65160           novel           novel  \n",
       "65161           novel           novel  \n",
       "65162           novel           novel  \n",
       "65163           novel           novel  \n",
       "\n",
       "[65164 rows x 6 columns]"
      ]
     },
     "execution_count": 24,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "transcript_gene"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene_name</th>\n",
       "      <th>gene_id</th>\n",
       "      <th>transcript_id</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>coding_noncoding_chr</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>CT-X</th>\n",
       "      <td>67</td>\n",
       "      <td>67</td>\n",
       "      <td>67</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CT-nonX</th>\n",
       "      <td>137</td>\n",
       "      <td>137</td>\n",
       "      <td>137</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Noncoding-X</th>\n",
       "      <td>3</td>\n",
       "      <td>3</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Noncoding-nonX</th>\n",
       "      <td>119</td>\n",
       "      <td>119</td>\n",
       "      <td>119</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                      gene_name  gene_id  transcript_id\n",
       "coding_noncoding_chr                                   \n",
       "CT-X                         67       67             67\n",
       "CT-nonX                     137      137            137\n",
       "Noncoding-X                   3        3              3\n",
       "Noncoding-nonX              119      119            119"
      ]
     },
     "execution_count": 36,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "TSTR_annot = TSTR.merge(transcript_gene, on=[\"gene_id\",\"gene_name\",\"transcript_id\",\"gene_type\"])\n",
    "TSTR_annot.drop(\"ctype\", axis=1).drop_duplicates().groupby(\"gene_type\").count()\n",
    "\n",
    "TSTR_annot['coding_noncoding_chr'] = np.select(\n",
    "    [\n",
    "        (TSTR_annot['gene_type'] == \"protein_coding\") & (TSTR_annot['chr'] == \"X\"),\n",
    "        (TSTR_annot['gene_type'] == \"protein_coding\") & (TSTR_annot['chr'] != \"X\"),\n",
    "        (TSTR_annot['gene_type'] != \"protein_coding\") & (TSTR_annot['chr'] == \"X\"),\n",
    "        (TSTR_annot['gene_type'] != \"protein_coding\") & (TSTR_annot['chr'] != \"X\"),\n",
    "    ],\n",
    "    [\n",
    "        \"CT-X\",\n",
    "        \"CT-nonX\",\n",
    "        \"Noncoding-X\",\n",
    "        \"Noncoding-nonX\"\n",
    "    ]\n",
    ")\n",
    "\n",
    "TSTR_annot = TSTR_annot[['gene_name','gene_id','transcript_id','coding_noncoding_chr']].drop_duplicates()\n",
    "TSTR_annot\n",
    "TSTR_annot.groupby(\"coding_noncoding_chr\").count()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {},
   "outputs": [],
   "source": [
    "## CT-X\n",
    "CTx = TSTR_annot[TSTR_annot['coding_noncoding_chr'] == \"CT-X\"]\n",
    "CTx_annotation = annotation_gtf[annotation_gtf[8].str.contains('|'.join(CTx.transcript_id.values.tolist()))]\n",
    "CTx_annotation.to_csv(os.path.join(cancer_dir,\"log2ratio3x/cancertypes/onlyStep1/annotations_coding_noncoding/CTx_annotation.gtf\"), sep=\"\\t\", index=None, header=None, quoting=csv.QUOTE_NONE)\n",
    "\n",
    "## CT-nonX\n",
    "CTnonx = TSTR_annot[TSTR_annot['coding_noncoding_chr'] == \"CT-nonX\"]\n",
    "CTnonx_annotation = annotation_gtf[annotation_gtf[8].str.contains('|'.join(CTnonx.transcript_id.values.tolist()))]\n",
    "CTnonx_annotation.to_csv(os.path.join(cancer_dir,\"log2ratio3x/cancertypes/onlyStep1/annotations_coding_noncoding/CTnonx_annotation.gtf\"), sep=\"\\t\", index=None, header=None, quoting=csv.QUOTE_NONE)\n",
    "\n",
    "## Noncoding-nonX\n",
    "Noncodingnonx = TSTR_annot[TSTR_annot['coding_noncoding_chr'] == \"Noncoding-nonX\"]\n",
    "Noncodingnonx_annotation = annotation_gtf[annotation_gtf[8].str.contains('|'.join(Noncodingnonx.transcript_id.values.tolist()))]\n",
    "Noncodingnonx_annotation.to_csv(os.path.join(cancer_dir,\"log2ratio3x/cancertypes/onlyStep1/annotations_coding_noncoding/Noncodingnonx_annotation.gtf\"), sep=\"\\t\", index=None, header=None, quoting=csv.QUOTE_NONE)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning: If your Wiggle data is a significant portion of available system memory, use the --max-mem and --sort-tmpdir options, or use --do-not-sort to disable post-conversion sorting. See --help for more information.\n",
      "Warning: If your Wiggle data is a significant portion of available system memory, use the --max-mem and --sort-tmpdir options, or use --do-not-sort to disable post-conversion sorting. See --help for more information.\n",
      "Warning: If your Wiggle data is a significant portion of available system memory, use the --max-mem and --sort-tmpdir options, or use --do-not-sort to disable post-conversion sorting. See --help for more information.\n"
     ]
    }
   ],
   "source": [
    "%%bash -s \"$cancer_dir\"\n",
    "\n",
    "module load BEDOPS\n",
    "\n",
    "for file in $1/log2ratio3x/cancertypes/onlyStep1/annotations_coding_noncoding/*gtf; do\n",
    "\n",
    "    outfile=${file%%.gtf*}\n",
    "    outfile=${outfile}.bed\n",
    "    gtf2bed < $file > $outfile\n",
    "\n",
    "done"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "metadata": {},
   "outputs": [],
   "source": [
    "## Get fasta of ORFs per group\n",
    "dir=\"/users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/cancers/log2ratio3x/cancertypes/onlyStep1/annotations_coding_noncoding\"\n",
    "ORFs = pd.read_csv(\"/users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/cancers/log2ratio3x/cancertypes/onlyStep1/TSTR_candidatesORFs_fullcharacterized.csv\")\n",
    "ORFs\n",
    "def create_seqrecord_PROThuman(row):\n",
    "    return SeqRecord(Seq(row['ORFpep']), id=row['orfID'], description=\"\")\n",
    "\n",
    "## CT-X\n",
    "CTx = ORFs[ORFs['coding_noncoding_chr'] == \"CT-X\"][['orfID','ORFpep']].drop_duplicates()\n",
    "seq_records_CTx = CTx.apply(create_seqrecord_PROThuman, axis=1).tolist()\n",
    "\n",
    "with open(os.path.join(dir,\"CTx.fasta\"), \"w\") as fasta_file:\n",
    "    for record in seq_records_CTx:\n",
    "        fasta_file.write(f\">{record.id}\\n{record.seq}\\n\")\n",
    "\n",
    "## CT-nonX\n",
    "CTnonx = ORFs[ORFs['coding_noncoding_chr'] == \"CT-nonX\"][['orfID','ORFpep']].drop_duplicates()\n",
    "seq_records_CTnonx = CTnonx.apply(create_seqrecord_PROThuman, axis=1).tolist()\n",
    "with open(os.path.join(dir,\"CTnonx.fasta\"), \"w\") as fasta_file:\n",
    "    for record in seq_records_CTnonx:\n",
    "        fasta_file.write(f\">{record.id}\\n{record.seq}\\n\")\n",
    "\n",
    "## NC-nonX\n",
    "NCnonx = ORFs[ORFs['coding_noncoding_chr'] == \"Noncoding-nonX\"][['orfID','ORFpep']].drop_duplicates()\n",
    "seq_records_NCnonx = NCnonx.apply(create_seqrecord_PROThuman, axis=1).tolist()\n",
    "with open(os.path.join(dir,\"Noncodingnonx.fasta\"), \"w\") as fasta_file:\n",
    "    for record in seq_records_NCnonx:\n",
    "        fasta_file.write(f\">{record.id}\\n{record.seq}\\n\")\n",
    "\n",
    "## altORFs\n",
    "altORFs = ORFs[ORFs['coding_noncoding_chr'] == \"other ncORFs\"][['orfID','ORFpep']].drop_duplicates()\n",
    "seq_records_altORFs = altORFs.apply(create_seqrecord_PROThuman, axis=1).tolist()\n",
    "with open(os.path.join(dir,\"altORFs.fasta\"), \"w\") as fasta_file:\n",
    "    for record in seq_records_altORFs:\n",
    "        fasta_file.write(f\">{record.id}\\n{record.seq}\\n\")\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_3928551/1397546842.py:57: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  candidates_bed[0] = \"chr\"+candidates_bed[0]\n"
     ]
    }
   ],
   "source": [
    "## Get bed of ORFs per group\n",
    "dir=\"/users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/cancers/log2ratio3x/cancertypes/onlyStep1/annotations_coding_noncoding\"\n",
    "ORFs = pd.read_csv(\"/users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/cancers/log2ratio3x/cancertypes/onlyStep1/TSTR_candidatesORFs_fullcharacterized.csv\")\n",
    "bed = pd.read_csv(\"/users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/human/RiboNovel_MultMap_1to1/RibORF/repre.valid.ORF.genepred.testis.bed\", header=None, sep=\"\\t\")\n",
    "\n",
    "\n",
    "## Canonical\n",
    "CT = ORFs[(ORFs['gene_type'] == \"protein_coding\") & (ORFs['coding_noncoding_chr'] != \"other ncORFs\")].orfID.values.tolist()\n",
    "CT_bed = bed[bed[3].isin(CT)]\n",
    "## ORF coordinates instead of transcript coordinates\n",
    "# CT_bed[1] = CT_bed[6]\n",
    "# CT_bed[2] = CT_bed[7]\n",
    "CT_bed.to_csv(os.path.join(dir,\"canonical_ORFs.bed\"), header=None, index=None, sep=\"\\t\")\n",
    "\n",
    "## CT-X\n",
    "# CTx = ORFs[ORFs['coding_noncoding_chr'] == \"CT-X\"].orfID.values.tolist()\n",
    "# CTx_bed = bed[bed[3].isin(CTx)]\n",
    "# CTx_bed.to_csv(os.path.join(dir,\"CTx_ORFs.bed\"), header=None, index=None, sep=\"\\t\")\n",
    "\n",
    "## CT-nonX\n",
    "# CTnonx = ORFs[ORFs['coding_noncoding_chr'] == \"CT-nonX\"].orfID.values.tolist()\n",
    "# CTnonx_bed = bed[bed[3].isin(CTnonx)]\n",
    "# CTnonx_bed.to_csv(os.path.join(dir,\"CTnonx_ORFs.bed\"), header=None, index=None, sep=\"\\t\")\n",
    "\n",
    "# lncORF\n",
    "lncORF = ORFs[ORFs['gene_type'] != \"protein_coding\"].orfID.values.tolist()\n",
    "lncORF_bed = bed[bed[3].isin(lncORF)]\n",
    "## ORF coordinates instead of transcript coordinates\n",
    "# lncORF_bed[1] = lncORF_bed[6]\n",
    "# lncORF_bed[2] = lncORF_bed[7]\n",
    "lncORF_bed.to_csv(os.path.join(dir,\"lncORF_ORFs.bed\"), header=None, index=None, sep=\"\\t\")\n",
    "\n",
    "## other ncORFs\n",
    "other_ncORFs = ORFs[ORFs['coding_noncoding_chr'] == \"other ncORFs\"].orfID.values.tolist()\n",
    "other_ncORFs_bed = bed[bed[3].isin(other_ncORFs)]\n",
    "## ORF coordinates instead of transcript coordinates\n",
    "# other_ncORFs_bed[1] = other_ncORFs_bed[6]\n",
    "# other_ncORFs_bed[2] = other_ncORFs_bed[7]\n",
    "other_ncORFs_bed.to_csv(os.path.join(dir,\"other_ncORFs_ORFs.bed\"), header=None, index=None, sep=\"\\t\")\n",
    "\n",
    "## ncORFs\n",
    "ncORFs = ORFs[~ORFs['coding_noncoding_chr'].str.contains(\"CT\")].orfID.values.tolist()\n",
    "ncORFs_bed = bed[bed[3].isin(ncORFs)]\n",
    "## ORF coordinates instead of transcript coordinates\n",
    "# ncORFs_bed[1] = ncORFs_bed[6]\n",
    "# ncORFs_bed[2] = ncORFs_bed[7]\n",
    "ncORFs_bed.to_csv(os.path.join(dir,\"ncORFs_ORFs.bed\"), header=None, index=None, sep=\"\\t\")\n",
    "\n",
    "\n",
    "## all ORF candidates\n",
    "candidates_orfIDs = ORFs.orfID.values.tolist()\n",
    "candidates_bed = bed[bed[3].isin(candidates_orfIDs)]\n",
    "## ORF coordinates instead of transcript coordinates\n",
    "# candidates_bed[1] = candidates_bed[6]\n",
    "# candidates_bed[2] = candidates_bed[7]\n",
    "candidates_bed.to_csv(os.path.join(dir,\"TSTR_ORFs.bed\"), header=None, index=None, sep=\"\\t\")\n",
    "candidates_bed[0] = \"chr\"+candidates_bed[0]\n",
    "candidates_bed.to_csv(os.path.join(dir,\"TSTR_ORFs.chr.bed\"), header=None, index=None, sep=\"\\t\")\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "\n",
    "#add chr\n",
    "\n",
    "dir=\"/users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/cancers/log2ratio3x/cancertypes/onlyStep1/annotations_coding_noncoding\"\n",
    "sed -e 's/^/chr/' $dir/TSTR_ORFs.bed > $dir/TSTR_ORFs.chr.bed "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## NetMHCpan\n",
    "\n",
    "How many putative peptides could arise from the sequences of the candidates? (Fully dependant on their length)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "metadata": {},
   "outputs": [],
   "source": [
    "def extract_9mers(fasta_file, output_file):\n",
    "    with open(output_file, 'w') as out_f:\n",
    "        for record in SeqIO.parse(fasta_file, \"fasta\"):\n",
    "            sequence = str(record.seq)\n",
    "            for i in range(len(sequence) - 8):  # Extract 9-mers\n",
    "                peptide = sequence[i:i+9]\n",
    "                out_f.write(f\">{record.id}_pos{i}\\n{peptide}\\n\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "metadata": {},
   "outputs": [],
   "source": [
    "CTx = os.path.join(cancer_dir,\"log2ratio3x/cancertypes/onlyStep1/annotations_coding_noncoding/CTx.fasta\")\n",
    "CTnonx = os.path.join(cancer_dir,\"log2ratio3x/cancertypes/onlyStep1/annotations_coding_noncoding/CTnonx.fasta\")\n",
    "NCnonx = os.path.join(cancer_dir,\"log2ratio3x/cancertypes/onlyStep1/annotations_coding_noncoding/Noncodingnonx.fasta\")\n",
    "altORFs = os.path.join(cancer_dir,\"log2ratio3x/cancertypes/onlyStep1/annotations_coding_noncoding/altORFs.fasta\")\n",
    "\n",
    "files = list([CTx, CTnonx, NCnonx, altORFs])\n",
    "files\n",
    "\n",
    "for f in files:\n",
    "    outfile = f.split(\".fasta\")[0] + \"_9mers.fasta\"\n",
    "\n",
    "    extract_9mers(f, outfile)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "CT-X\n",
      "19068\n",
      "CT-nonX\n",
      "65025\n",
      "NC-nonX\n",
      "6397\n",
      "altORFs\n",
      "1986\n"
     ]
    }
   ],
   "source": [
    "%%bash -s \"$cancer_dir\"\n",
    "\n",
    "echo \"CT-X\"\n",
    "grep -v '^>' $1/log2ratio3x/cancertypes/onlyStep1/annotations_coding_noncoding/CTx_9mers.fasta | sort | uniq | wc -l\n",
    "\n",
    "echo \"CT-nonX\"\n",
    "grep -v '^>' $1/log2ratio3x/cancertypes/onlyStep1/annotations_coding_noncoding/CTnonx_9mers.fasta | sort | uniq | wc -l\n",
    "\n",
    "echo \"NC-nonX\"\n",
    "grep -v '^>' $1/log2ratio3x/cancertypes/onlyStep1/annotations_coding_noncoding/Noncodingnonx_9mers.fasta | sort | uniq | wc -l\n",
    "\n",
    "echo \"altORFs\"\n",
    "grep -v '^>' $1/log2ratio3x/cancertypes/onlyStep1/annotations_coding_noncoding/altORFs_9mers.fasta | sort | uniq | wc -l"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "**NetMHCpan**\n",
    "\n",
    "HLA-A02:01"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 52,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "CTx\n",
      "initial\n",
      "19068\n",
      "SB\n",
      "200\n",
      "WB/SB\n",
      "603\n",
      "\n",
      "\n",
      "CTnonx\n",
      "initial\n",
      "65025\n",
      "SB\n",
      "948\n",
      "WB/SB\n",
      "2579\n",
      "\n",
      "\n",
      "Noncodingnonx\n",
      "initial\n",
      "6397\n",
      "SB\n",
      "53\n",
      "WB/SB\n",
      "199\n",
      "\n",
      "\n",
      "altORFs\n",
      "initial\n",
      "1986\n",
      "SB\n",
      "19\n",
      "WB/SB\n",
      "61\n",
      "\n",
      "\n"
     ]
    }
   ],
   "source": [
    "%%bash -s \"$cancer_dir\"\n",
    "\n",
    "## HLA-A02:01\n",
    "dir=$1/log2ratio3x/cancertypes/onlyStep1/netMHCpan/HLA0201\n",
    "mkdir -p $dir\n",
    "\n",
    "CTx=\"$1/log2ratio3x/cancertypes/onlyStep1/annotations_coding_noncoding/CTx.fasta\"\n",
    "CTnonx=\"$1/log2ratio3x/cancertypes/onlyStep1/annotations_coding_noncoding/CTnonx.fasta\"\n",
    "NCnonx=\"$1/log2ratio3x/cancertypes/onlyStep1/annotations_coding_noncoding/Noncodingnonx.fasta\"\n",
    "altORFs=\"$1/log2ratio3x/cancertypes/onlyStep1/annotations_coding_noncoding/altORFs.fasta\"\n",
    "\n",
    "samples_array=(\"$CTx\" \"$CTnonx\" \"$NCnonx\" \"$altORFs\")\n",
    "\n",
    "#=======================================#\n",
    "# Run netMHCpan.sh + Create output xlsx #\n",
    "#=======================================#\n",
    "\n",
    "# Set path for software\n",
    "module load netMHCpan/4.1\n",
    "module load R/4.0.4-foss-2020b\n",
    "\n",
    "for file in \"${samples_array[@]}\"; do\n",
    "\n",
    "    name=${file##*/}\n",
    "    name=${name%%.fasta*}\n",
    "\n",
    "    echo $name\n",
    "\n",
    "    ## netMHCpan\n",
    "    netMHCpan -a HLA-A02:01 -f $file -l 9 -BA -xls -xlsfile ${dir}/${name}.netMHCpan.BA.xls >  ${dir}/${name}.netMHCpan.BA.out\n",
    "\n",
    "    echo \"initial\"\n",
    "    grep -v \"^>\" $1/log2ratio3x/cancertypes/onlyStep1/annotations_coding_noncoding/${name}_9mers.fasta | sort | uniq | wc -l\n",
    "\n",
    "    echo \"SB\"\n",
    "    grep \"= SB\" ${dir}/${name}.netMHCpan.BA.out > ${dir}/${name}.SB.out\n",
    "    awk '{print $3}' ${dir}/${name}.SB.out | sort | uniq | wc -l\n",
    "\n",
    "    echo \"WB/SB\"\n",
    "    grep \"= WB\\|= SB\" ${dir}/${name}.netMHCpan.BA.out > ${dir}/${name}.WBSB.out\n",
    "    awk '{print $3}' ${dir}/${name}.WBSB.out | sort | uniq | wc -l\n",
    "    echo -e \"\\n\"\n",
    "    # ## create output xlsx\n",
    "    # excelFile=${dir}/${name}.netMHCpan.BA.xls\n",
    "    # Rscript /data/genomics/marta/scripts/parse_netmhcpan_EL_BA_out_marta.r $excelFile\n",
    "\n",
    "done\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Most frequent alleles\n",
    "\n",
    "https://doi.org/10.1080/22221751.2021.1978823"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "CTx\n",
      "initial\n",
      "19068\n",
      "SB\n",
      "2015\n",
      "WB/SB\n",
      "4942\n",
      "\n",
      "\n",
      "CTnonx\n",
      "initial\n",
      "65025\n",
      "SB\n",
      "7421\n",
      "WB/SB\n",
      "17910\n",
      "\n",
      "\n",
      "Noncodingnonx\n",
      "initial\n",
      "6397\n",
      "SB\n",
      "715\n",
      "WB/SB\n",
      "1759\n",
      "\n",
      "\n",
      "altORFs\n",
      "initial\n",
      "1986\n",
      "SB\n",
      "221\n",
      "WB/SB\n",
      "555\n",
      "\n",
      "\n"
     ]
    }
   ],
   "source": [
    "%%bash -s \"$cancer_dir\"\n",
    "\n",
    "## Most common alleles\n",
    "dir=$1/log2ratio3x/cancertypes/onlyStep1/netMHCpan/FrequentHLA-I\n",
    "mkdir -p $dir\n",
    "\n",
    "CTx=\"$1/log2ratio3x/cancertypes/onlyStep1/annotations_coding_noncoding/CTx.fasta\"\n",
    "CTnonx=\"$1/log2ratio3x/cancertypes/onlyStep1/annotations_coding_noncoding/CTnonx.fasta\"\n",
    "NCnonx=\"$1/log2ratio3x/cancertypes/onlyStep1/annotations_coding_noncoding/Noncodingnonx.fasta\"\n",
    "altORFs=\"$1/log2ratio3x/cancertypes/onlyStep1/annotations_coding_noncoding/altORFs.fasta\"\n",
    "\n",
    "samples_array=(\"$CTx\" \"$CTnonx\" \"$NCnonx\" \"$altORFs\")\n",
    "\n",
    "#=======================================#\n",
    "# Run netMHCpan.sh + Create output xlsx #\n",
    "#=======================================#\n",
    "\n",
    "# Set path for software\n",
    "module load netMHCpan/4.1\n",
    "module load R/4.0.4-foss-2020b\n",
    "\n",
    "for file in \"${samples_array[@]}\"; do\n",
    "\n",
    "    name=${file##*/}\n",
    "    name=${name%%.fasta*}\n",
    "\n",
    "    echo $name\n",
    "\n",
    "    ## netMHCpan\n",
    "    netMHCpan -a \"$( < /projects_eg/projects/marta/VaccineCandidates/frequence_HLAalleles.csv)\" -f $file -l 9 -BA -xls -xlsfile ${dir}/${name}.netMHCpan.BA.xls >  ${dir}/${name}.netMHCpan.BA.out\n",
    "\n",
    "    echo \"initial\"\n",
    "    grep -v \"^>\" $1/log2ratio3x/cancertypes/onlyStep1/annotations_coding_noncoding/${name}_9mers.fasta | sort | uniq | wc -l\n",
    "\n",
    "    echo \"SB\"\n",
    "    grep \"= SB\" ${dir}/${name}.netMHCpan.BA.out > ${dir}/${name}.SB.out\n",
    "    awk '{print $3}' ${dir}/${name}.SB.out | sort | uniq | wc -l\n",
    "\n",
    "    echo \"WB/SB\"\n",
    "    grep \"= WB\\|= SB\" ${dir}/${name}.netMHCpan.BA.out > ${dir}/${name}.WBSB.out\n",
    "    awk '{print $3}' ${dir}/${name}.WBSB.out | sort | uniq | wc -l\n",
    "    echo -e \"\\n\"\n",
    "    # ## create output xlsx\n",
    "    # excelFile=${dir}/${name}.netMHCpan.BA.xls\n",
    "    # Rscript /data/genomics/marta/scripts/parse_netmhcpan_EL_BA_out_marta.r $excelFile\n",
    "\n",
    "done\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.10"
  },
  "orig_nbformat": 4
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
