---
title: "TSA TestisRestrctedGTEX Translated | Matrix"
author: "Marta Espinosa"
date: "2024-09-12"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
library(dplyr)
library(tidyr)
library(ggplot2)
library(stringr)
library(purrr)
library(ggvenn)
library(ggpubr)
library(ggside)
library(bioseq)
library(rcartocolor)
library(ggbeeswarm)
library(ggbreak)
library("pheatmap")
library(cluster)
library(factoextra)
library(cluster)  # for silhouette score
library(stats)    # for hclust and dist
library(dynamicTreeCut)
library(pheatmap)

annot = read.csv("/users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/human/newReference_Resconstructed/transID_geneID_isoforms_selected.1to1.csv")

Q2_wd = "/projects_eg/projects/marta/TestisRestricted_Microproteins_TSA/with_TranscriptomeReconstruction/Multimap_altORFs/Q2_TestisRestricted/human"

pre_save_wd = "/projects_eg/projects/marta/TestisRestricted_Microproteins_TSA/with_TranscriptomeReconstruction/Multimap_altORFs/Q3_TestisRestrictedTumors/human"
plots_wd = "/projects_eg/projects/marta/TestisRestricted_Microproteins_TSA/with_TranscriptomeReconstruction/Multimap_altORFs/Q4_TestisRestricted_TumorSpecific/human/plots"
save_wd = "/projects_eg/projects/marta/TestisRestricted_Microproteins_TSA/with_TranscriptomeReconstruction/Multimap_altORFs/Q4_TestisRestricted_TumorSpecific/human"
heatmaps_wd = "/projects_eg/projects/marta/TestisRestricted_Microproteins_TSA/with_TranscriptomeReconstruction/Multimap_altORFs/Q4_TestisRestricted_TumorSpecific/human/plots/heatmaps"
tumorReactDIR = "/users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/cancers/log2ratio3x/cancertypes"

cancers = c("BRCA","BLCA","LUAD","KIRC","PRAD","LUSC","COAD","LIHC")
tcga_projects=c("TCGA-BRCA","TCGA-LUSC","TCGA-PRAD","TCGA-KIRC","TCGA-LUAD","TCGA-BLCA")#,"TCGA-LIHC"]
other_projects=c("GSE102101_KIRC","GSE133624_BLCA","GSE22260_PRAD","PRJEB2449_PRAD","SRP238334_KIRC","GSE214846_LIHC","GSE229705_LUAD","TCGA_COAD","SRP107326_COAD")
manuscript_projects = c("liver_adjacent_totalRNA_LIHC","hcc_normal_totalRNA_LIHC","GSE193567_LIHC","LIHC_TCGA_LIHC")
# deleted_projects=c("GSE103001_BRCA"GSE89223_PRAD")
all_projects = c(tcga_projects,other_projects,manuscript_projects)
ctype_patients = data.frame(ctype = c("BRCA","BLCA","LUAD","KIRC","PRAD","LUSC","COAD","LIHC"),
                            num_patients = c(109,38,179,142,49,75,182,144))

cancers_dir = "/users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/cancers"

ctypes_data = read.csv("/projects_eg/projects/marta/TestisRestricted_Microproteins_TSA/with_TranscriptomeReconstruction/used_ctypes_info.csv", sep="\t")

# Colors
mycolors = list(
  "organ_system" = c("core_gastrointestinal" = "#2c715f", "developmental_gastrointestinal" = "#44a57c", "gynecologic" = "#60620c", "thoracic" = "#58a449", "urologic" = "#1d271c"),
  "histopathology" = c("adenocarcinoma" = "#278b9a", "other" = "#e75b64","squamous" = "#d8af39"),
  "tissue" = c("bladder" = "#28231d", "breast" = "#5e2d30", "lung" = "#008e90", "colon" = "#d87427", "liver" = "#c5a387", "prostate" = "#67b8d6", "kidney" = "#e9d097"),
  "X_nonX" = c("X" = c("#833437"), "nonX" = "#646c31"),
  "gene_type" = c("protein_coding"="#ead890",
                       "lncRNA" = "#cd4f38",
                       "processed_pseudogene" = "#e48c2a",
                                             "novel" = "#637c68"))

CTDB = read.csv("/projects_eg/projects/marta/CTdatabase_list.csv", sep="\t")

save_pheatmap_pdf <- function(x, filename, width=7, height=7) {
   stopifnot(!missing(x))
   stopifnot(!missing(filename))
   pdf(filename, width=width, height=height)
   grid::grid.newpage()
   grid::grid.draw(x$gtable)
   dev.off()
}

# Extract the correlation coefficients
# res2$r
# Extract p-values
# res2$P
# A simple function to format the correlation matrix
# ++++++++++++++++++++++++++++
# flattenCorrMatrix
# ++++++++++++++++++++++++++++
# cormat : matrix of the correlation coefficients
# pmat : matrix of the correlation p-values
flattenCorrMatrix <- function(cormat, pmat) {
  ut <- upper.tri(cormat)
  data.frame(
    row = rownames(cormat)[row(cormat)[ut]],
    column = rownames(cormat)[col(cormat)[ut]],
    cor  =(cormat)[ut],
    p = pmat[ut]
  )
}
# flattenCorrMatrix(res2$r, res2$P)
```

## Work with matrices

```{r import_data, echo=F}
TSA_TestisGTEx_Translated = read.csv(file.path(tumorReactDIR,"TOv3x_5percent_TestisRestrictedGTEx_Translated_Ctypes_log2ratio3xMEAN.csv"))
TSA_TestisGTEx_Translated = TSA_TestisGTEx_Translated %>% subset(ctype != "KIRP")
temp = TSA_TestisGTEx_Translated %>% select(gene_id, gene_type) %>% unique()
table(temp$gene_type)

TSTR_expr = read.csv("/users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/cancers/log2ratio3x/cancertypes/TSTR_Expression_tumor.csv")
table(TSTR_expr %>% select(gene_id, gene_type) %>% unique() %>% pull(gene_type))
```

```{r patients_matrix_ctype, echo=F}
TSA_TestisGTEx_Translated = TSA_TestisGTEx_Translated %>% mutate(n_percent = case_when(ctype == "BRCA" ~ num_patients_overexpr/109,
                                                                                       ctype == "BLCA" ~ num_patients_overexpr/38,
                                                                                       ctype == "LUAD" ~ num_patients_overexpr/179,
                                                                                       ctype == "KIRC" ~ num_patients_overexpr/142,
                                                                                       ctype == "LUSC" ~ num_patients_overexpr/49,
                                                                                       ctype == "PRAD" ~ num_patients_overexpr/75,
                                                                                       ctype == "LIHC" ~ num_patients_overexpr/182,
                                                                                       ctype == "COAD" ~ num_patients_overexpr/144))



##### lncRNAs ######
patient_ctype_mat_lncRNA = TSA_TestisGTEx_Translated %>% subset(gene_type != "protein_coding") %>% select(gene_name, ctype, n_percent) %>% unique()
patient_ctype_mat_lncRNA$n_percent = str_remove_all(patient_ctype_mat_lncRNA$n_percent," ")
patient_ctype_mat_lncRNA$n_percent = as.numeric(patient_ctype_mat_lncRNA$n_percent)
patient_ctype_mat_lncRNA_wide = patient_ctype_mat_lncRNA %>% pivot_wider(names_from = "ctype", values_from = "n_percent")
patient_ctype_mat_lncRNA_wide$max = apply(patient_ctype_mat_lncRNA_wide[,2:ncol(patient_ctype_mat_lncRNA_wide)], 1, max, na.rm=TRUE)
patient_ctype_mat_lncRNA_wide = patient_ctype_mat_lncRNA_wide %>% subset(max > 0.05) %>% select(-max) ## 5% of the patients in one cancer type
## replace NAs by zeros
patient_ctype_mat_lncRNA_wide[is.na(patient_ctype_mat_lncRNA_wide)] = 0
lncRNAs_mat = patient_ctype_mat_lncRNA_wide$gene_name
patient_ctype_mat_lncRNA_wide = as.matrix(patient_ctype_mat_lncRNA_wide)

##### CTA ######
patient_ctype_mat_CTA = TSA_TestisGTEx_Translated %>% subset(gene_type == "protein_coding") %>% select(gene_name, ctype, n_percent) %>% unique()
patient_ctype_mat_CTA$n_percent = str_remove_all(patient_ctype_mat_CTA$n_percent," ")
patient_ctype_mat_CTA$n_percent = as.numeric(patient_ctype_mat_CTA$n_percent)
patient_ctype_mat_CTA_wide = patient_ctype_mat_CTA %>% pivot_wider(names_from = "ctype", values_from = "n_percent")
patient_ctype_mat_CTA_wide$max = apply(patient_ctype_mat_CTA_wide[,2:ncol(patient_ctype_mat_CTA_wide)], 1, max, na.rm=TRUE)
patient_ctype_mat_CTA_wide = patient_ctype_mat_CTA_wide %>% subset(max > 0.05) %>% select(-max) ## 5% of the patients in one cancer type
## replace NAs by zeros
patient_ctype_mat_CTA_wide[is.na(patient_ctype_mat_CTA_wide)] = 0
CTAs_mat = patient_ctype_mat_CTA_wide$gene_name
patient_ctype_mat_CTA_wide = as.matrix(patient_ctype_mat_CTA_wide)

## select geneName and geneType
gname_gtype = TSA_TestisGTEx_Translated %>% select(gene_name, gene_type) %>% unique()

## row annotation
row_annotation = annot %>% select(gene_name, gene_type, chr) %>% subset(gene_name %in% TSA_TestisGTEx_Translated$gene_name)
rownames(row_annotation) = row_annotation$gene_name
row_annotation = row_annotation %>% mutate(X_nonX = case_when(chr == "X" ~ "X",
                                                            TRUE ~ "nonX"))
row_annotation = row_annotation %>% select(gene_name, X_nonX, gene_type)
rownames(row_annotation) = row_annotation$gene_name
row_annotation = row_annotation %>% select(-c(gene_name))


## col annotation
col_annot = ctypes_data
rownames(col_annot) = col_annot$ctype
col_annot = col_annot %>% select(-c(ctype))
# save.image(file=file.path(save_wd,"Q4.2_matrix.RData"))
```

## Matrix for expression


```{r read_data, echo=F}
## select only the patients for which the gene is TSA
data_completed_TSA_wide = TSTR_expr %>% subset(forced_TPM > 0)

## we compute the median of the patients that express the gene in a tumor-specific manner
data_completed_TSA_median = data_completed_TSA_wide %>% group_by(gene_name,ctype) %>% mutate(medianTPM = median(forced_TPM)) %>% ungroup()
data_completed_TSA_median = merge(data_completed_TSA_median, annot, by=c("gene_id","transcript_id","gene_type","gene_name"))

## counting
counting_genes = data_completed_TSA_median %>% select(gene_id, gene_type, ctype) %>% unique()
table(counting_genes$ctype, counting_genes$gene_type)
```

```{r median_heatmap, echo=F}
## keep expression data to plot
df_expr = data_completed_TSA_median %>% select(gene_name, ctype, medianTPM) %>% unique() %>% pivot_wider(names_from = ctype, values_from = medianTPM) %>% unique() %>% subset(gene_name %in% rownames(row_annotation))
df_expr = as.data.frame(df_expr)
df_expr[is.na(df_expr)] = 0

df_expr_values = mutate_all(df_expr[,-1], function(x) as.numeric(as.character(x)))
## convert to log2medianTPM
df_log = log2(df_expr_values+1)

## convert to matrix
matrix = data.matrix(df_log)
rownames(matrix) = df_expr$gene_name
matrix = na.omit(matrix)
nozeros_matrix_log = matrix[,colSums(matrix) > 0]


## heatmap_all_geneTypes
ph = pheatmap(nozeros_matrix_log, clustering_distance_cols = "correlation",clustering_distance_rows = "euclidean", clustering_method = "complete",
       # treeheight_row = 0, treeheight_col = 0,
       annotation_row = row_annotation,
         annotation_col = col_annot,  # Annotation data frame
       annotation_names_col = FALSE,
       annotation_names_row = FALSE,
       show_colnames = TRUE,
       # cutree_cols = 3,
       annotation_colors = mycolors,
       color=colorRampPalette(c("lightyellow", "#de2d26"))(50),
       fontsize = 6,
       main=paste0("Median logTPM of Testis/Tumor Specific genes (TOv3x)"))
save_pheatmap_pdf(ph, paste0(heatmaps_wd,"/PDF/pheatmap_medianTPM_all_TOv3x5percentlog2ratio3x.pdf"))

lncRNA = data_completed_TSA_median %>% subset(gene_type != "protein_coding") %>% select(gene_name) %>% unique() %>% subset(gene_name %in% df_expr$gene_name)
CTA = data_completed_TSA_median %>% subset(gene_type == "protein_coding") %>% select(gene_name) %>% unique() %>% subset(gene_name %in% df_expr$gene_name)
  
## CTA
CTA_matrix = nozeros_matrix_log[CTA$gene_name, ]
CTA_matrix_non_zero <- CTA_matrix[, apply(CTA_matrix, 2, function(x) any(x != 0))]

cta_ph = pheatmap(CTA_matrix_non_zero, clustering_distance_cols = "correlation",clustering_distance_rows = "euclidean", clustering_method = "complete",
       # treeheight_row = 0, treeheight_col = 0,
       annotation_row = row_annotation,
         annotation_col = col_annot,  # Annotation data frame
       annotation_names_col = FALSE,
       annotation_names_row = FALSE,
       show_colnames = TRUE,
       # cutree_cols = 3,
       annotation_colors = mycolors,
       color=colorRampPalette(c("lightyellow", "#de2d26"))(50),
       fontsize = 6,
       main="Median logTPM of Testis/Tumor Specific genes (TOv3x) | CTA")
save_pheatmap_pdf(cta_ph, paste0(heatmaps_wd,"/PDF/pheatmap_medianTPM_CTA_TOv3x5percentlog2ratio3x.pdf"))
col.order = cta_ph$tree_col$order
ordered_labels_col_CTA = cta_ph$tree_col$labels[col.order]

row.order = cta_ph$tree_row$order
ordered_labels_row_CTA = cta_ph$tree_row$labels[row.order]

## lncRNA
lncRNA_matrix = nozeros_matrix_log[lncRNA$gene_name, ]
lncRNA_matrix_non_zero <- lncRNA_matrix[, apply(lncRNA_matrix, 2, function(x) any(x != 0))]

lncRNA_ph = pheatmap(lncRNA_matrix_non_zero, clustering_distance_cols = "correlation",clustering_distance_rows = "euclidean", clustering_method = "complete",
       # treeheight_row = 0, treeheight_col = 0,
       annotation_row = row_annotation,
         annotation_col = col_annot,  # Annotation data frame
       annotation_names_col = FALSE,
       annotation_names_row = FALSE,
       show_colnames = TRUE,
       # cutree_cols = 3,
       annotation_colors = mycolors,
       color=colorRampPalette(c("lightyellow", "#de2d26"))(50),
       fontsize = 6,
       main="Median logTPM of Testis/Tumor Specific genes (TOv3x) | lncRNA")
col.order = lncRNA_ph$tree_col$order
ordered_labels_col_lncRNA = lncRNA_ph$tree_col$labels[col.order]

row.order = lncRNA_ph$tree_row$order
ordered_labels_row_lncRNA = lncRNA_ph$tree_row$labels[row.order]
save_pheatmap_pdf(lncRNA_ph, paste0(heatmaps_wd,"/PDF/pheatmap_medianTPM_lncRNA_TOv3x5percentlog2ratio3x.pdf"))

####### X / vs nonX
chrX = data_completed_TSA_median %>% subset(chr == "X") %>% select(gene_name) %>% unique() %>% subset(gene_name %in% df_expr$gene_name)
nrow(chrX)
chrA = data_completed_TSA_median %>% subset(chr != "X") %>% select(gene_name) %>% unique() %>% subset(gene_name %in% df_expr$gene_name)
nrow(chrA)

## chrX
chrX_matrix = nozeros_matrix_log[chrX$gene_name, ]
chrX_matrix_non_zero <- chrX_matrix[, apply(chrX_matrix, 2, function(x) any(x != 0))]

chrX_ph = pheatmap(chrX_matrix_non_zero, clustering_distance_cols = "correlation",clustering_distance_rows = "euclidean", clustering_method = "complete",
       # treeheight_row = 0, treeheight_col = 0,
       annotation_row = row_annotation,
         annotation_col = col_annot,  # Annotation data frame
       annotation_names_col = FALSE,
       annotation_names_row = FALSE,
       show_colnames = TRUE,
       # cutree_cols = 3,
       annotation_colors = mycolors,
       color=colorRampPalette(c("lightyellow", "#de2d26"))(50),
       fontsize = 6,
       main="Median logTPM of Testis/Tumor Specific genes (TOv3x) | chrX")
save_pheatmap_pdf(chrX_ph, paste0(heatmaps_wd,"/PDF/pheatmap_medianTPM_chrX_TOv3x5percentlog2ratio3x.pdf"))
col.order = chrX_ph$tree_col$order
ordered_labels_col_chrX = chrX_ph$tree_col$labels[col.order]

row.order = chrX_ph$tree_row$order
ordered_labels_row_chrX = chrX_ph$tree_row$labels[row.order]

## chrA
chrA_matrix = nozeros_matrix_log[chrA$gene_name, ]
chrA_matrix_non_zero <- chrA_matrix[, apply(chrA_matrix, 2, function(x) any(x != 0))]

chrA_ph = pheatmap(chrA_matrix_non_zero, clustering_distance_cols = "correlation",clustering_distance_rows = "euclidean", clustering_method = "complete",
       # treeheight_row = 0, treeheight_col = 0,
       annotation_row = row_annotation,
         annotation_col = col_annot,  # Annotation data frame
       annotation_names_col = FALSE,
       annotation_names_row = FALSE,
       show_colnames = TRUE,
       # cutree_cols = 3,
       annotation_colors = mycolors,
       color=colorRampPalette(c("lightyellow", "#de2d26"))(50),
       fontsize = 6,
       main="Median logTPM of Testis/Tumor Specific genes (TOv3x) | autosomes")
col.order = chrA_ph$tree_col$order
ordered_labels_col_chrA = chrA_ph$tree_col$labels[col.order]

row.order = chrA_ph$tree_row$order
ordered_labels_row_chrA = chrA_ph$tree_row$labels[row.order]
save_pheatmap_pdf(chrA_ph, paste0(heatmaps_wd,"/PDF/pheatmap_medianTPM_chrA_TOv3x5percentlog2ratio3x.pdf"))
```

## Correlation

```{r correlation, echo=F}
library(Hmisc)
transposed = t(nozeros_matrix_log)
res2 <- rcorr(transposed)
# Extract the correlation coefficients
# res2$r
# # Extract p-values
# res2$P
# A simple function to format the correlation matrix
# ++++++++++++++++++++++++++++
# flattenCorrMatrix
# ++++++++++++++++++++++++++++
# cormat : matrix of the correlation coefficients
# pmat : matrix of the correlation p-values
flattenCorrMatrix <- function(cormat, pmat) {
  ut <- upper.tri(cormat)
  data.frame(
    row = rownames(cormat)[row(cormat)[ut]],
    column = rownames(cormat)[col(cormat)[ut]],
    cor  =(cormat)[ut],
    p = pmat[ut]
  )
}
# flattenCorrMatrix(res2$r, res2$P)


library(corrplot)
cormat <- round(cor(transposed),2)

corrplot(cormat, type = "lower", order = "hclust", method="circle",
         tl.col = "black", col = COL2('BrBG', 10), tl.cex=0.6)


### getting correlation values
cormat = data.frame(cormat)
cormat$A = rownames(cormat)
cormat = cormat %>% pivot_longer(cols=-c(A), names_to = "B", values_to = "correlation")

biotypes = data_completed_TSA_median %>% select(gene_name, gene_type) %>% distinct()
lncRNA_biotypes = biotypes %>% subset(gene_type != "protein_coding")
proteincoding_biotypes = biotypes %>% subset(gene_type == "protein_coding")

cormat = cormat %>% mutate(A_biotype = case_when(A %in% proteincoding_biotypes$gene_name ~ "protein_coding", 
                                                 TRUE ~ "non-coding")) 
cormat = cormat %>%
  mutate(B_biotype = case_when(B %in% lncRNA_biotypes$gene_name ~ "non-coding", 
                               TRUE ~ "protein_coding")) 

cormat_lncRNA = cormat %>% subset(A_biotype == "non-coding" & B_biotype == "non-coding")

cormat_proteincoding = cormat %>% subset(A_biotype == "protein_coding" & B_biotype == "protein_coding")
summary(cormat_lncRNA$correlation)
summary(cormat_proteincoding$correlation)
```

```{r correlation_nonCTA, echo=F}
lncRNA_transposed = t(lncRNA_matrix)
res2 <- rcorr(lncRNA_transposed)
# A simple function to format the correlation matrix
# ++++++++++++++++++++++++++++
# flattenCorrMatrix
# ++++++++++++++++++++++++++++
# cormat : matrix of the correlation coefficients
# pmat : matrix of the correlation p-values
flattenCorrMatrix <- function(cormat, pmat) {
  ut <- upper.tri(cormat)
  data.frame(
    row = rownames(cormat)[row(cormat)[ut]],
    column = rownames(cormat)[col(cormat)[ut]],
    cor  =(cormat)[ut],
    p = pmat[ut]
  )
}
# flattenCorrMatrix(res2$r, res2$P)


lncRNA_cormat <- round(cor(lncRNA_transposed),2)
dim(lncRNA_cormat)
corrplot(lncRNA_cormat, type = "lower", order = "hclust", method="circle",
         tl.col = "black", col = COL2('BrBG', 10), tl.cex=0.6)
```

```{r correlation_CTA, echo=F}
CTA_transposed = t(CTA_matrix)
res2 <- rcorr(CTA_transposed)
# A simple function to format the correlation matrix
# ++++++++++++++++++++++++++++
# flattenCorrMatrix
# ++++++++++++++++++++++++++++
# cormat : matrix of the correlation coefficients
# pmat : matrix of the correlation p-values
flattenCorrMatrix <- function(cormat, pmat) {
  ut <- upper.tri(cormat)
  data.frame(
    row = rownames(cormat)[row(cormat)[ut]],
    column = rownames(cormat)[col(cormat)[ut]],
    cor  =(cormat)[ut],
    p = pmat[ut]
  )
}
# flattenCorrMatrix(res2$r, res2$P)


CTA_cormat <- round(cor(CTA_transposed),2)
dim(CTA_cormat)
corrplot(CTA_cormat, type = "lower", order = "hclust", method="circle",
         tl.col = "black", col = COL2('BrBG', 10), tl.cex=0.6)

CTA_cormat = data.frame(CTA_cormat)
CTA_cormat$A = rownames(CTA_cormat)
CTA_cormat = CTA_cormat %>% pivot_longer(cols=-c(A), names_to = "B", values_to = "correlation")
## which group we want to compare?
x_nonX_df = data_completed_TSA_median %>% select(gene_name, chr) %>% mutate(X_nonX = case_when(chr == "X" ~ "X",
                                                            TRUE ~ "nonX")) %>% select(-chr)
chr_X = x_nonX_df %>% subset(X_nonX == "X")
chr_nonX = x_nonX_df %>% subset(X_nonX == "nonX")


CTA_cormat = CTA_cormat %>% mutate(A_chr = case_when(A %in% chr_X$gene_name ~ "X", 
                                                 TRUE ~ "non-X")) 
CTA_cormat = CTA_cormat %>%
  mutate(B_chr = case_when(B %in% chr_nonX$gene_name ~ "non-X", 
                               TRUE ~ "X")) 

CTA_cormat_nonX = CTA_cormat %>% subset(A_chr == "non-X" & B_chr == "non-X")
CTA_cormat_X = CTA_cormat %>% subset(A_chr == "X" & B_chr == "X")
summary(CTA_cormat_nonX$correlation)
summary(CTA_cormat_X$correlation)

```

```{r correlation_X, echo=F}
CTA_matrix_X = CTA_matrix[rownames(CTA_matrix) %in% unique(chr_X$gene_name),]

CTAX_transposed = t(CTA_matrix_X)
res2 <- rcorr(CTAX_transposed)
# A simple function to format the correlation matrix
# ++++++++++++++++++++++++++++
# flattenCorrMatrix
# ++++++++++++++++++++++++++++
# cormat : matrix of the correlation coefficients
# pmat : matrix of the correlation p-values
flattenCorrMatrix <- function(cormat, pmat) {
  ut <- upper.tri(cormat)
  data.frame(
    row = rownames(cormat)[row(cormat)[ut]],
    column = rownames(cormat)[col(cormat)[ut]],
    cor  =(cormat)[ut],
    p = pmat[ut]
  )
}
# flattenCorrMatrix(res2$r, res2$P)


CTAX_cormat <- round(cor(CTAX_transposed),2)
dim(CTAX_cormat)
corrplot(CTAX_cormat, type = "lower", order = "hclust", method="circle",
         tl.col = "black", col = COL2('BrBG', 10), tl.cex=0.6)

### non-X
CTA_matrix_nonX = CTA_matrix[rownames(CTA_matrix) %in% unique(chr_nonX$gene_name),]
CTAnonX_transposed = t(CTA_matrix_nonX)
res2 <- rcorr(CTAnonX_transposed)
# A simple function to format the correlation matrix
# ++++++++++++++++++++++++++++
# flattenCorrMatrix
# ++++++++++++++++++++++++++++
# cormat : matrix of the correlation coefficients
# pmat : matrix of the correlation p-values
flattenCorrMatrix <- function(cormat, pmat) {
  ut <- upper.tri(cormat)
  data.frame(
    row = rownames(cormat)[row(cormat)[ut]],
    column = rownames(cormat)[col(cormat)[ut]],
    cor  =(cormat)[ut],
    p = pmat[ut]
  )
}
# flattenCorrMatrix(res2$r, res2$P)


CTAnonX_cormat <- round(cor(CTAnonX_transposed),2)
dim(CTAnonX_cormat)
corrplot(CTAnonX_cormat, type = "lower", order = "hclust", method="circle",
         tl.col = "black", col =COL2('BrBG', 10), tl.cex=0.6)

```


## Burden of CT per ctype (CTAB)

```{r CTAB, echo=F}
## How many CT of the defined in CTdatabase are expressed > 1 TPM in at least 10% of the patients - tumor samples?
burden_TTS = data_completed_TSA_wide %>% 
  select(ctype, gene_name, gene_type, forced_TPM, patient) %>% 
  subset (forced_TPM > 0) %>% 
  mutate(coding_noncoding = case_when(gene_type == "protein_coding" ~ "coding",
                                      TRUE ~ "noncoding")) %>% 
  group_by(ctype, coding_noncoding, patient) %>%
  count()

burden_TTS_coding_noncoding = burden_TTS %>% pivot_wider(names_from = "coding_noncoding", values_from = "n")
burden_TTS_coding_noncoding[is.na(burden_TTS_coding_noncoding)] = 0
burden_TTS_coding_noncoding$coding = as.numeric(burden_TTS_coding_noncoding$coding)
burden_TTS_coding_noncoding$noncoding = as.numeric(burden_TTS_coding_noncoding$noncoding)

ggplot(burden_TTS_coding_noncoding, aes(x = coding, y=noncoding)) +
  geom_point(size=1) +
  stat_smooth(method = "lm",
              formula = y ~ x, 
              geom = "smooth", color="red") +
  stat_cor(p.accuracy = 0.001, r.accuracy = 0.01, size=2)+
  labs(x="Burden of codingTTS",
       y="Burden of non-codingTTS",
       title="Correlation between coding and non-coding TTS burden?") +
  theme_minimal() +
  facet_wrap(~ ctype, scales="free", nrow=2)
ggsave(file.path(plots_wd, "PNG/TOv3xlog2ratio3xburden_coding_vs_noncoding.png"), width=7, height=4)
ggsave(file.path(plots_wd, "PDF/TOv3xlog2ratio3xburden_coding_vs_noncoding.pdf"), width=7, height=4)


## How many CT of the defined in CTdatabase are tumor-specific in at least 10% of the patients?
```

## Matrix for patients

```{r patients_matrix_ctype2, echo=F}
patients_data = read.csv(file.path(cancers_dir, "log2ratio3x/cancertypes/TOv3x_5percent_TestisRestrictedGTEx_Translated_Ctypes_log2ratio3xMEANgenes_patients.csv"))
names(patients_data)[7] = "patient"
patients_data = merge(patients_data, annot, by=c("transcript_id","gene_id","gene_type","gene_name"))

##### lncRNAs ######
patient_ctype_mat_lncRNA = patients_data %>% subset(gene_type != "protein_coding") %>% select(gene_name, ctype, percentage_num_patients_overexpr) %>% unique()
patient_ctype_mat_lncRNA$percentage_num_patients_overexpr = as.numeric(patient_ctype_mat_lncRNA$percentage_num_patients_overexpr)
patient_ctype_mat_lncRNA_wide = patient_ctype_mat_lncRNA %>% pivot_wider(names_from = "ctype", values_from = "percentage_num_patients_overexpr")
patient_ctype_mat_lncRNA_wide$max = apply(patient_ctype_mat_lncRNA_wide[,2:ncol(patient_ctype_mat_lncRNA_wide)], 1, max, na.rm=TRUE)
patient_ctype_mat_lncRNA_wide = patient_ctype_mat_lncRNA_wide %>% subset(max > 0.05) %>% select(-max) ## 5% of the patients in one cancer type
## replace NAs by zeros
patient_ctype_mat_lncRNA_wide[is.na(patient_ctype_mat_lncRNA_wide)] = 0
lncRNAs_mat = patient_ctype_mat_lncRNA_wide$gene_name
patient_ctype_mat_lncRNA_wide = as.matrix(patient_ctype_mat_lncRNA_wide)

##### CTA ######
patient_ctype_mat_CTA = patients_data %>% subset(gene_type == "protein_coding") %>% select(gene_name, ctype, percentage_num_patients_overexpr) %>% unique()
patient_ctype_mat_CTA$percentage_num_patients_overexpr = as.numeric(patient_ctype_mat_CTA$percentage_num_patients_overexpr)
patient_ctype_mat_CTA_wide = patient_ctype_mat_CTA %>% pivot_wider(names_from = "ctype", values_from = "percentage_num_patients_overexpr")
patient_ctype_mat_CTA_wide$max = apply(patient_ctype_mat_CTA_wide[,2:ncol(patient_ctype_mat_CTA_wide)], 1, max, na.rm=TRUE)
patient_ctype_mat_CTA_wide = patient_ctype_mat_CTA_wide %>% subset(max > 0.05) %>% select(-max) ## 5% of the patients in one cancer type
## replace NAs by zeros
patient_ctype_mat_CTA_wide[is.na(patient_ctype_mat_CTA_wide)] = 0
CTAs_mat = patient_ctype_mat_CTA_wide$gene_name
patient_ctype_mat_CTA_wide = as.matrix(patient_ctype_mat_CTA_wide)

##### chrX ######
patient_ctype_mat_chrX = patients_data %>% subset(chr == "X") %>% select(gene_name, ctype, percentage_num_patients_overexpr) %>% unique()
patient_ctype_mat_chrX$percentage_num_patients_overexpr = as.numeric(patient_ctype_mat_chrX$percentage_num_patients_overexpr)
patient_ctype_mat_chrX_wide = patient_ctype_mat_chrX %>% pivot_wider(names_from = "ctype", values_from = "percentage_num_patients_overexpr")
patient_ctype_mat_chrX_wide$max = apply(patient_ctype_mat_chrX_wide[,2:ncol(patient_ctype_mat_chrX_wide)], 1, max, na.rm=TRUE)
patient_ctype_mat_chrX_wide = patient_ctype_mat_chrX_wide %>% subset(max > 0.05) %>% select(-max) ## 5% of the patients in one cancer type
## replace NAs by zeros
patient_ctype_mat_chrX_wide[is.na(patient_ctype_mat_chrX_wide)] = 0
chrXs_mat = patient_ctype_mat_chrX_wide$gene_name
patient_ctype_mat_chrX_wide = as.matrix(patient_ctype_mat_chrX_wide)

##### CTA ######
patient_ctype_mat_chrA = patients_data %>% subset(chr != "X") %>% select(gene_name, ctype, percentage_num_patients_overexpr) %>% unique()
patient_ctype_mat_chrA$percentage_num_patients_overexpr = str_remove_all(patient_ctype_mat_chrA$percentage_num_patients_overexpr," ")
patient_ctype_mat_chrA$percentage_num_patients_overexpr = as.numeric(patient_ctype_mat_chrA$percentage_num_patients_overexpr)
patient_ctype_mat_chrA_wide = patient_ctype_mat_chrA %>% pivot_wider(names_from = "ctype", values_from = "percentage_num_patients_overexpr")
patient_ctype_mat_chrA_wide$max = apply(patient_ctype_mat_chrA_wide[,2:ncol(patient_ctype_mat_chrA_wide)], 1, max, na.rm=TRUE)
patient_ctype_mat_chrA_wide = patient_ctype_mat_chrA_wide %>% subset(max > 0.05) %>% select(-max) ## 5% of the patients in one cancer type
## replace NAs by zeros
patient_ctype_mat_chrA_wide[is.na(patient_ctype_mat_chrA_wide)] = 0
chrAs_mat = patient_ctype_mat_chrA_wide$gene_name
patient_ctype_mat_chrA_wide = as.matrix(patient_ctype_mat_chrA_wide)
## select geneName and geneType
gname_gtype = TSA_TestisGTEx_Translated %>% select(gene_name, gene_type) %>% unique()

## row annotation
row_annotation = annot %>% select(gene_name, gene_type, chr) %>% subset(gene_name %in% TSA_TestisGTEx_Translated$gene_name)
rownames(row_annotation) = row_annotation$gene_name
row_annotation = row_annotation %>% mutate(X_nonX = case_when(chr == "X" ~ "X",
                                                            TRUE ~ "nonX"))
row_annotation = row_annotation %>% select(gene_name, X_nonX, gene_type)
rownames(row_annotation) = row_annotation$gene_name
row_annotation = row_annotation %>% select(-c(gene_name))


## col annotation
col_annot = ctypes_data
rownames(col_annot) = col_annot$ctype
col_annot = col_annot %>% select(-c(ctype))
# save.image(file=file.path(save_wd,"Q4.2_matrix.RData"))
```

```{r heatmaps_lncRNA, echo=F}
### lncRNAs
# Remove leading/trailing spaces
patient_ctype_mat_lncRNA_wide = apply(patient_ctype_mat_lncRNA_wide, 2, function(x) gsub("^\\s+|\\s+$ ","", x))
# Set the first column (gene_name) as rownames
rownames(patient_ctype_mat_lncRNA_wide) = patient_ctype_mat_lncRNA_wide[, 1]
patient_ctype_mat_lncRNA_wide = patient_ctype_mat_lncRNA_wide[, -1]
# Convert the values (except the gene names) to numeric
patient_ctype_mat_lncRNA_wide = apply(patient_ctype_mat_lncRNA_wide, 2, as.numeric)
rownames(patient_ctype_mat_lncRNA_wide) = lncRNAs_mat
## remove those where everything is 0
patient_ctype_mat_lncRNA_wide = patient_ctype_mat_lncRNA_wide[,colSums(patient_ctype_mat_lncRNA_wide) > 0.1]

ordered_labels_row_lncRNA = intersect(ordered_labels_row_lncRNA, rownames(patient_ctype_mat_lncRNA_wide))

ph_lncRNA_patients_matrix = patient_ctype_mat_lncRNA_wide[ordered_labels_row_lncRNA, ordered_labels_col_lncRNA]
# Numbers greater than 0.1 are kept, others are replaced with an empty string
display_numbers_custom = ifelse(ph_lncRNA_patients_matrix > 0.1, round(ph_lncRNA_patients_matrix,2), "")


ph_patients = pheatmap(ph_lncRNA_patients_matrix,  cluster_rows = F, cluster_cols = F, scale="none",
         # display_numbers = display_numbers_custom,  # Only display numbers > 0.1
         annotation_names_col = FALSE, 
         annotation_names_row = FALSE,
         show_colnames = TRUE,
        show_rownames = F,
         annotation_col = col_annot,  # Annotation data frame
         annotation_row = row_annotation,
         annotation_colors = mycolors,  # Custom colors
         color = colorRampPalette(c("lightyellow", "#addd8e","#31a354"))(50),
         fontsize = 6,
         main="TOv3x lncRNA | percentage of patients (at least 5% in one cancer type)")
save_pheatmap_pdf(ph_patients, paste0(heatmaps_wd,"/PDF/TOv3xlog2ratio3x_pheatmap_patients_5percent_lncRNA.pdf"))

```

```{r heatmaps_CTA, echo=F}
### CTA
# Remove leading/trailing spaces
patient_ctype_mat_CTA_wide = apply(patient_ctype_mat_CTA_wide, 2, function(x) gsub("^\\s+|\\s+$", "", x))

# Set the first column (gene_name) as rownames
rownames(patient_ctype_mat_CTA_wide) = patient_ctype_mat_CTA_wide[, 1]
patient_ctype_mat_CTA_wide = patient_ctype_mat_CTA_wide[, -1]
# Convert the values (except the gene names) to numeric
patient_ctype_mat_CTA_wide = apply(patient_ctype_mat_CTA_wide, 2, as.numeric)
rownames(patient_ctype_mat_CTA_wide) = CTAs_mat
## remove those where everything is 0
patient_ctype_mat_CTA_wide = patient_ctype_mat_CTA_wide[,colSums(patient_ctype_mat_CTA_wide) > 0]


# Numbers greater than 0.1 are kept, others are replaced with an empty string
ordered_labels_row_CTA = intersect(ordered_labels_row_CTA, rownames(patient_ctype_mat_CTA_wide))

ph_CTA_patients_matrix = patient_ctype_mat_CTA_wide[ordered_labels_row_CTA, ordered_labels_col_CTA]
# Numbers greater than 0.1 are kept, others are replaced with an empty string
display_numbers_custom = ifelse(ph_CTA_patients_matrix > 0.1, round(ph_CTA_patients_matrix,2), "")

ph_CTA_patients = pheatmap(ph_CTA_patients_matrix, cluster_rows = F, cluster_cols = F,scale="none",
         # display_numbers = display_numbers_custom,  # Only display numbers > 0.1
         annotation_names_col = FALSE, 
         annotation_names_row = FALSE,
         show_colnames = TRUE,
        show_rownames = F,
         annotation_col = col_annot,  # Annotation data frame
         annotation_row = row_annotation,
         annotation_colors = mycolors,  # Custom colors
         color = colorRampPalette(c("lightyellow", "#addd8e","#31a354"))(50),
         fontsize = 6,
         main="TOv3x CTA | percentage of patients (at least 5% in one cancer type)")
save_pheatmap_pdf(ph_CTA_patients, paste0(heatmaps_wd,"/PDF/TOv3xlog2ratio3x_pheatmap_patients_5percent_CTA.pdf"))

```

```{r heatmaps_chrX, echo=F}
### chrX
# Remove leading/trailing spaces
patient_ctype_mat_chrX_wide = apply(patient_ctype_mat_chrX_wide, 2, function(x) gsub("^\\s+|\\s+$", "", x))

# Set the first column (gene_name) as rownames
rownames(patient_ctype_mat_chrX_wide) = patient_ctype_mat_chrX_wide[, 1]
patient_ctype_mat_chrX_wide = patient_ctype_mat_chrX_wide[, -1]
# Convert the values (except the gene names) to numeric
patient_ctype_mat_chrX_wide = apply(patient_ctype_mat_chrX_wide, 2, as.numeric)
rownames(patient_ctype_mat_chrX_wide) = chrXs_mat
## remove those where everything is 0
patient_ctype_mat_chrX_wide = patient_ctype_mat_chrX_wide[,colSums(patient_ctype_mat_chrX_wide) > 0]


# Numbers greater than 0.1 are kept, others are replaced with an empty string
ordered_labels_row_chrX = intersect(ordered_labels_row_chrX, rownames(patient_ctype_mat_chrX_wide))

ph_chrX_patients_matrix = patient_ctype_mat_chrX_wide[ordered_labels_row_chrX, ordered_labels_col_chrX]
# Numbers greater than 0.1 are kept, others are replaced with an empty string
display_numbers_custom = ifelse(ph_chrX_patients_matrix > 0.1, round(ph_chrX_patients_matrix,2), "")

ph_chrX_patients = pheatmap(ph_chrX_patients_matrix, cluster_rows = F, cluster_cols = F,scale="none",
         # display_numbers = display_numbers_custom,  # Only display numbers > 0.1
         annotation_names_col = FALSE, 
         annotation_names_row = FALSE,
         show_colnames = TRUE,
         annotation_col = col_annot,  # Annotation data frame
         annotation_row = row_annotation,
         annotation_colors = mycolors,  # Custom colors
         color = colorRampPalette(c("lightyellow", "#addd8e","#31a354"))(50),
         fontsize = 6,
         main="TOv3x chrX | percentage of patients (at least 5% in one cancer type)")
save_pheatmap_pdf(ph_chrX_patients, paste0(heatmaps_wd,"/PDF/TOv3xlog2ratio3x_pheatmap_patients_5percent_chrX.pdf"))

```

```{r heatmaps_chrA, echo=F}
### chrA
# Remove leading/trailing spaces
patient_ctype_mat_chrA_wide = apply(patient_ctype_mat_chrA_wide, 2, function(x) gsub("^\\s+|\\s+$", "", x))

# Set the first column (gene_name) as rownames
rownames(patient_ctype_mat_chrA_wide) = patient_ctype_mat_chrA_wide[, 1]
patient_ctype_mat_chrA_wide = patient_ctype_mat_chrA_wide[, -1]
# Convert the values (except the gene names) to numeric
patient_ctype_mat_chrA_wide = apply(patient_ctype_mat_chrA_wide, 2, as.numeric)
rownames(patient_ctype_mat_chrA_wide) = chrAs_mat
## remove those where everything is 0
patient_ctype_mat_chrA_wide = patient_ctype_mat_chrA_wide[,colSums(patient_ctype_mat_chrA_wide) > 0]


# Numbers greater than 0.1 are kept, others are replaced with an empty string
ordered_labels_row_chrA = intersect(ordered_labels_row_chrA, rownames(patient_ctype_mat_chrA_wide))

ph_chrA_patients_matrix = patient_ctype_mat_chrA_wide[ordered_labels_row_chrA, ordered_labels_col_chrA]
# Numbers greater than 0.1 are kept, others are replaced with an empty string
display_numbers_custom = ifelse(ph_chrA_patients_matrix > 0.1, round(ph_chrA_patients_matrix,2), "")

ph_chrA_patients = pheatmap(ph_chrA_patients_matrix, cluster_rows = F, cluster_cols = F,scale="none",
         # display_numbers = display_numbers_custom,  # Only display numbers > 0.1
         annotation_names_col = FALSE, 
         annotation_names_row = FALSE,
         show_colnames = TRUE,
        show_rownames = F,
         annotation_col = col_annot,  # Annotation data frame
         annotation_row = row_annotation,
         annotation_colors = mycolors,  # Custom colors
         color = colorRampPalette(c("lightyellow", "#addd8e","#31a354"))(50),
         fontsize = 6,
         main="TOv3x chrA | percentage of patients (at least 5% in one cancer type)")
save_pheatmap_pdf(ph_chrA_patients, paste0(heatmaps_wd,"/PDF/TOv3xlog2ratio3x_pheatmap_patients_5percent_chrA.pdf"))

```