{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Evolutionary origin\n",
    "\n",
    "Macaca - Mouse - Human"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>specie</th>\n",
       "      <th>age</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>human</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>macaca</td>\n",
       "      <td>25</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>mouse</td>\n",
       "      <td>90</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>opossum</td>\n",
       "      <td>180</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>platypus</td>\n",
       "      <td>200</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>chicken</td>\n",
       "      <td>310</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "     specie  age\n",
       "1     human    0\n",
       "2    macaca   25\n",
       "3     mouse   90\n",
       "4   opossum  180\n",
       "5  platypus  200\n",
       "6   chicken  310"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import os,re,glob\n",
    "import pandas as pd\n",
    "import numpy as np\n",
    "from collections import Counter\n",
    "from Bio.Seq import Seq\n",
    "from Bio.SeqRecord import SeqRecord\n",
    "from Bio import SeqIO\n",
    "\n",
    "species_dir = \"/users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction\"\n",
    "users_dir = \"/users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47\"\n",
    "\n",
    "GENOMEDIR = \"/genomics/users/marta/genomes\"\n",
    "species=[\"mouse\",\"macaca\",\"platypus\",\"chicken\",\"opossum\"]\n",
    "\n",
    "## annotation file\n",
    "annotation=\"/users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/human/newReference_Resconstructed/gencode.v47.gffcompare.TestisLiverBrain.annotation.sorted.1transcript.sorted.NOchr.gtf\"\n",
    "transcript_gene=pd.read_csv(\"/users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/human/newReference_Resconstructed/transID_geneID_isoforms_selected.1to1.csv\")\n",
    "\n",
    "## evoDir\n",
    "evoDir=os.path.join(users_dir,\"EvolutionaryOrigin_MMseqs\")\n",
    "\n",
    "def translate_dna_to_protein(dna_seq):\n",
    "    return str(Seq(dna_seq).translate())\n",
    "\n",
    "# Function to save DF in fasta format\n",
    "def create_seqrecord_PROT(row):\n",
    "    return SeqRecord(Seq(row['protein']), id=row['header'][:-4], description=\"\")\n",
    "def create_seqrecord_DNA(row):\n",
    "    return SeqRecord(Seq(row['seq']), id=row['header'], description=\"\")\n",
    "\n",
    "species_and_years_dict = {\"specie\" : [\"human\",\"macaca\",\"mouse\",\"opossum\",\"platypus\",\"chicken\"],\n",
    "                          \"age\" : [0,25,90,180,200,310]}\n",
    "species_and_years = pd.DataFrame(data=species_and_years_dict, index=[1,2,3,4,5,6])\n",
    "species_and_years\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "# ## get repre sequences - both DNA and PROTEINS\n",
    "# for s in species:\n",
    "#     print(s)\n",
    "#     riboseq_dir = os.path.join(users_dir,s,\"RiboSeq/RiboQC_RiboNovel\")\n",
    "\n",
    "#     ## get all candidates\n",
    "#     print(\"Reading all the candidate ORF sequences\")\n",
    "#     df = pd.read_csv(os.path.join(riboseq_dir,\"Annotation/candidateORF.fa\"), header=None, sep=\"\\t\")\n",
    "#     ## convert to df\n",
    "#     candidates_fasta = pd.DataFrame({'header':df[0].iloc[::2].values, 'seq':df[0].iloc[1::2].values})\n",
    "#     candidates_fasta['header'] = candidates_fasta['header'].str[1:]\n",
    "#     candidates_fasta = candidates_fasta[~candidates_fasta['header'].str.contains(\"iORF\")]\n",
    "\n",
    "#     all_repre_candidates = pd.DataFrame()\n",
    "#     for sample in os.listdir(os.path.join(riboseq_dir,\"RibORF\")):\n",
    "#         if sample.endswith(\"r1\"):\n",
    "#             print(sample)\n",
    "\n",
    "#             repre = pd.read_csv(os.path.join(riboseq_dir,\"RibORF\",sample,\"repre.valid.pred.pvalue.parameters.txt\"), sep=\"\\t\")\n",
    "#             repre_fasta = candidates_fasta[candidates_fasta['header'].isin(repre.orfID.values.tolist())]\n",
    "#             repre_fasta['header'] = s + \"_\" +repre_fasta['header']\n",
    "\n",
    "#             ## keep only the coding and the non-coding\n",
    "#             repre_fasta = repre_fasta[repre_fasta['header'].str.contains(\"canonical\")| repre_fasta['header'].str.contains(\"noncoding\")]\n",
    "\n",
    "#             repre_fasta['protein'] = repre_fasta['seq'].apply(translate_dna_to_protein)\n",
    "#             all_repre_candidates = pd.concat([all_repre_candidates, repre_fasta])\n",
    "\n",
    "#             ## translate to protein\n",
    "#             print(\"Translating to protein\")\n",
    "#             out_proteins = os.path.join(riboseq_dir,\"RibORF\",sample,\"repre.valid.pred.pvalue.parameters.PROTEIN.fa\")\n",
    "#             seq_records_proteins = repre_fasta.apply(create_seqrecord_PROT, axis=1).tolist()\n",
    "#             SeqIO.write(seq_records_proteins, out_proteins, \"fasta\")\n",
    "\n",
    "#             ## keep the dna sequences as well\n",
    "#             print(\"Saving the original DNA ORF\")\n",
    "#             out_dna = os.path.join(riboseq_dir,\"RibORF\",sample,\"repre.valid.pred.pvalue.parameters.DNA.fa\")\n",
    "#             seq_records_dna = repre_fasta.apply(create_seqrecord_DNA, axis=1).tolist()\n",
    "#             SeqIO.write(seq_records_dna, out_dna, \"fasta\")\n",
    "\n",
    "#             ## get only the nonredundant\n",
    "#             print(\"Getting the non-redundant\")\n",
    "#             all_repre_candidates.drop_duplicates(inplace=True)\n",
    "#             out_nonredundant_proteins = os.path.join(riboseq_dir,\"RibORF\")+\"/\"+s+\"_repre.valid.pred.pvalue.parameters.PROTEIN.noredunant.fa\"\n",
    "#             seq_records_proteins = all_repre_candidates.apply(create_seqrecord_PROT, axis=1).tolist()\n",
    "#             SeqIO.write(seq_records_proteins, out_nonredundant_proteins, \"fasta\")"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## MMseqs2\n",
    "\n",
    "Against all expressed in mouse, macaca, platypus, opossum in any tissue (non-redunant set)"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### TESTIS-EXPRESSED"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "27251"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "## Testis-expressed ORFs human\n",
    "TestisExpressed_translatedINtestis = pd.read_csv(\"/projects_eg/projects/marta/TestisRestricted_Microproteins_TSA/with_TranscriptomeReconstruction/Multimap_altORFs/Q1_TestisORFs/human/ribORF_humanTestis_in1.csv\")\n",
    "TestisExpressed_translatedINtestis\n",
    "\n",
    "def create_seqrecord_PROThuman(row):\n",
    "    return SeqRecord(Seq(row['ORFpep']), id=row['orfID'], description=\"\")\n",
    "\n",
    "out_nonredundant_proteins = \"/projects_eg/projects/marta/TestisRestricted_Microproteins_TSA/with_TranscriptomeReconstruction/Multimap_altORFs/Q1_TestisORFs/human/ribORF_humanTestis_in1.fa\"\n",
    "seq_records_proteins = TestisExpressed_translatedINtestis.apply(create_seqrecord_PROThuman, axis=1).tolist()\n",
    "SeqIO.write(seq_records_proteins, out_nonredundant_proteins, \"fasta\")"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Creating database of human (expressed in testis)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/MMseqsDB/ribORF_humanTestis_in1_DB exists and will be overwritten\n",
      "createdb /projects_eg/projects/marta/TestisRestricted_Microproteins_TSA/with_TranscriptomeReconstruction/Multimap_altORFs/Q1_TestisORFs/human/ribORF_humanTestis_in1.fa /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/MMseqsDB/ribORF_humanTestis_in1_DB \n",
      "\n",
      "MMseqs Version:       \t14.7e284\n",
      "Database type         \t0\n",
      "Shuffle input database\ttrue\n",
      "Createdb mode         \t0\n",
      "Write lookup file     \t1\n",
      "Offset of numeric ids \t0\n",
      "Compressed            \t0\n",
      "Verbosity             \t3\n",
      "\n",
      "Converting sequences\n",
      "[==\n",
      "Time for merging to ribORF_humanTestis_in1_DB_h: 0h 0m 0s 684ms\n",
      "Time for merging to ribORF_humanTestis_in1_DB: 0h 0m 2s 772ms\n",
      "Database type: Aminoacid\n",
      "Time for processing: 0h 0m 15s 217ms\n"
     ]
    }
   ],
   "source": [
    "%%bash -s \"$evoDir\" \"$users_dir\"\n",
    "\n",
    "# mkdir $1/MMseqsDB\n",
    "\n",
    "module load Miniconda3/4.9.2\n",
    "source /soft/system/software/Miniconda3/4.9.2/bin/activate MMseqs2\n",
    "\n",
    "mkdir -p $1/MMseqsDB\n",
    "\n",
    "mmseqs createdb /projects_eg/projects/marta/TestisRestricted_Microproteins_TSA/with_TranscriptomeReconstruction/Multimap_altORFs/Q1_TestisORFs/human/ribORF_humanTestis_in1.fa $1/MMseqsDB/ribORF_humanTestis_in1_DB\n"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Creating a joint database with all the specie, everything translated, no matter the tissue"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/MMseqsDB/speciesDB exists and will be overwritten\n",
      "createdb /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/chicken/RiboSeq/RiboQC_RiboNovel/RibORF/chicken_repre.valid.pred.pvalue.parameters.PROTEIN.noredunant.fa /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/macaca/RiboSeq/RiboQC_RiboNovel/RibORF/macaca_repre.valid.pred.pvalue.parameters.PROTEIN.noredunant.fa /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/mouse/RiboSeq/RiboQC_RiboNovel/RibORF/mouse_repre.valid.pred.pvalue.parameters.PROTEIN.noredunant.fa /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/opossum/RiboSeq/RiboQC_RiboNovel/RibORF/opossum_repre.valid.pred.pvalue.parameters.PROTEIN.noredunant.fa /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/platypus/RiboSeq/RiboQC_RiboNovel/RibORF/platypus_repre.valid.pred.pvalue.parameters.PROTEIN.noredunant.fa /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/MMseqsDB/speciesDB \n",
      "\n",
      "MMseqs Version:       \t14.7e284\n",
      "Database type         \t0\n",
      "Shuffle input database\ttrue\n",
      "Createdb mode         \t0\n",
      "Write lookup file     \t1\n",
      "Offset of numeric ids \t0\n",
      "Compressed            \t0\n",
      "Verbosity             \t3\n",
      "\n",
      "Converting sequences\n",
      "[===========\n",
      "Time for merging to speciesDB_h: 0h 0m 4s 649ms\n",
      "Time for merging to speciesDB: 0h 0m 10s 801ms\n",
      "Database type: Aminoacid\n",
      "Time for processing: 0h 1m 25s 544ms\n"
     ]
    }
   ],
   "source": [
    "%%bash -s \"$evoDir\" \"$species_dir\"\n",
    "\n",
    "module load Miniconda3/4.9.2\n",
    "source /soft/system/software/Miniconda3/4.9.2/bin/activate MMseqs2\n",
    "\n",
    "# species=\"platypus mouse macaca chicken\"\n",
    "# for s in $species; do\n",
    "#     echo $s\n",
    "\n",
    "#     ### create DB\n",
    "#     mmseqs createdb $2/$s/RiboSeq/RiboQC_RiboNovel/RibORF/repre.valid.pred.pvalue.parameters.PROTEIN.noredunant.fa $1/MMseqsDB/${s}_noredunantProts_DB\n",
    "\n",
    "# done\n",
    "\n",
    "mmseqs createdb $2/*/RiboSeq/RiboQC_RiboNovel*/RibORF/*_repre.valid.pred.pvalue.parameters.PROTEIN.noredunant.fa $1/MMseqsDB/speciesDB\n",
    "\n"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Creating a joint database with all the specie AND human for clustering, everything translated for the species, no matter the tissue, but only translated in 1 for testis"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/MMseqsDB/species_and_humanDB exists and will be overwritten\n",
      "createdb /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/chicken/RiboSeq/RiboQC_RiboNovel/RibORF/chicken_repre.valid.pred.pvalue.parameters.PROTEIN.noredunant.fa /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/macaca/RiboSeq/RiboQC_RiboNovel/RibORF/macaca_repre.valid.pred.pvalue.parameters.PROTEIN.noredunant.fa /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/mouse/RiboSeq/RiboQC_RiboNovel/RibORF/mouse_repre.valid.pred.pvalue.parameters.PROTEIN.noredunant.fa /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/opossum/RiboSeq/RiboQC_RiboNovel/RibORF/opossum_repre.valid.pred.pvalue.parameters.PROTEIN.noredunant.fa /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/platypus/RiboSeq/RiboQC_RiboNovel/RibORF/platypus_repre.valid.pred.pvalue.parameters.PROTEIN.noredunant.fa /projects_eg/projects/marta/TestisRestricted_Microproteins_TSA/with_TranscriptomeReconstruction/Multimap_altORFs/Q1_TestisORFs/human/ribORF_humanTestis_in1.fa /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/MMseqsDB/species_and_humanDB \n",
      "\n",
      "MMseqs Version:       \t14.7e284\n",
      "Database type         \t0\n",
      "Shuffle input database\ttrue\n",
      "Createdb mode         \t0\n",
      "Write lookup file     \t1\n",
      "Offset of numeric ids \t0\n",
      "Compressed            \t0\n",
      "Verbosity             \t3\n",
      "\n",
      "Converting sequences\n",
      "[==============\n",
      "Time for merging to species_and_humanDB_h: 0h 0m 7s 960ms\n",
      "Time for merging to species_and_humanDB: 0h 0m 14s 694ms\n",
      "Database type: Aminoacid\n",
      "Time for processing: 0h 1m 51s 770ms\n"
     ]
    }
   ],
   "source": [
    "%%bash -s \"$evoDir\" \"$species_dir\"\n",
    "\n",
    "module load Miniconda3/4.9.2\n",
    "source /soft/system/software/Miniconda3/4.9.2/bin/activate MMseqs2\n",
    "\n",
    "mmseqs createdb $2/*/RiboSeq/RiboQC_RiboNovel*/RibORF/*_repre.valid.pred.pvalue.parameters.PROTEIN.noredunant.fa /projects_eg/projects/marta/TestisRestricted_Microproteins_TSA/with_TranscriptomeReconstruction/Multimap_altORFs/Q1_TestisORFs/human/ribORF_humanTestis_in1.fa $1/MMseqsDB/species_and_humanDB\n",
    "\n"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Searching"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [],
   "source": [
    "# %%bash -s \"$evoDir\" \"$users_dir\"\n",
    "\n",
    "# module load Miniconda3/4.9.2\n",
    "# source /soft/system/software/Miniconda3/4.9.2/bin/activate MMseqs2\n",
    "# mkdir -p $1/Searching\n",
    "\n",
    "# rm $1/Searching/resultDB*\n",
    "# ## The alignment consists of two steps the prefilter and alignment. To run the search, type:\n",
    "# mmseqs search -a -s 6 $1/MMseqsDB/ribORF_humanTestis_in1_DB $1/MMseqsDB/speciesDB $1/Searching/resultDB $1/tmp\n",
    "\n",
    "# # Then, convert the result database into a BLAST tab formatted file (option -m 8 in legacy blast, -outfmt 6 in blast+):\n",
    "# # Format alignment output\tquery,target,fident,alnlen,mismatch,gapopen,qstart,qend,tstart,tend,evalue,bits\n",
    "\n",
    "# mmseqs convertalis $1/MMseqsDB/ribORF_humanTestis_in1_DB $1/MMseqsDB/speciesDB $1/Searching/resultDB $1/Searching/resultDB.m8"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Clustering"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "cluster -c 0.5 --min-seq-id 0.5 /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/MMseqsDB/species_and_humanDB /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/Clustering/species_and_human_cluDB /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/tmp \n",
      "\n",
      "MMseqs Version:                     \t14.7e284\n",
      "Substitution matrix                 \taa:blosum62.out,nucl:nucleotide.out\n",
      "Seed substitution matrix            \taa:VTML80.out,nucl:nucleotide.out\n",
      "Sensitivity                         \t4\n",
      "k-mer length                        \t0\n",
      "k-score                             \tseq:2147483647,prof:2147483647\n",
      "Alphabet size                       \taa:21,nucl:5\n",
      "Max sequence length                 \t65535\n",
      "Max results per query               \t20\n",
      "Split database                      \t0\n",
      "Split mode                          \t2\n",
      "Split memory limit                  \t0\n",
      "Coverage threshold                  \t0.5\n",
      "Coverage mode                       \t0\n",
      "Compositional bias                  \t1\n",
      "Compositional bias                  \t1\n",
      "Diagonal scoring                    \ttrue\n",
      "Exact k-mer matching                \t0\n",
      "Mask residues                       \t1\n",
      "Mask residues probability           \t0.9\n",
      "Mask lower case residues            \t0\n",
      "Minimum diagonal score              \t15\n",
      "Selected taxa                       \t\n",
      "Include identical seq. id.          \tfalse\n",
      "Spaced k-mers                       \t1\n",
      "Preload mode                        \t0\n",
      "Pseudo count a                      \tsubstitution:1.100,context:1.400\n",
      "Pseudo count b                      \tsubstitution:4.100,context:5.800\n",
      "Spaced k-mer pattern                \t\n",
      "Local temporary path                \t\n",
      "Threads                             \t24\n",
      "Compressed                          \t0\n",
      "Verbosity                           \t3\n",
      "Add backtrace                       \tfalse\n",
      "Alignment mode                      \t3\n",
      "Alignment mode                      \t0\n",
      "Allow wrapped scoring               \tfalse\n",
      "E-value threshold                   \t0.001\n",
      "Seq. id. threshold                  \t0.5\n",
      "Min alignment length                \t0\n",
      "Seq. id. mode                       \t0\n",
      "Alternative alignments              \t0\n",
      "Max reject                          \t2147483647\n",
      "Max accept                          \t2147483647\n",
      "Score bias                          \t0\n",
      "Realign hits                        \tfalse\n",
      "Realign score bias                  \t-0.2\n",
      "Realign max seqs                    \t2147483647\n",
      "Correlation score weight            \t0\n",
      "Gap open cost                       \taa:11,nucl:5\n",
      "Gap extension cost                  \taa:1,nucl:2\n",
      "Zdrop                               \t40\n",
      "Rescore mode                        \t0\n",
      "Remove hits by seq. id. and coverage\tfalse\n",
      "Sort results                        \t0\n",
      "Cluster mode                        \t0\n",
      "Max connected component depth       \t1000\n",
      "Similarity type                     \t2\n",
      "Single step clustering              \tfalse\n",
      "Cascaded clustering steps           \t3\n",
      "Cluster reassign                    \tfalse\n",
      "Remove temporary files              \tfalse\n",
      "Force restart with latest tmp       \tfalse\n",
      "MPI runner                          \t\n",
      "k-mers per sequence                 \t21\n",
      "Scale k-mers per sequence           \taa:0.000,nucl:0.200\n",
      "Adjust k-mer length                 \tfalse\n",
      "Shift hash                          \t67\n",
      "Include only extendable             \tfalse\n",
      "Skip repeating k-mers               \tfalse\n",
      "\n",
      "Set cluster sensitivity to -s 3.000000\n",
      "Set cluster mode SET COVER\n",
      "Set cluster iterations to 3\n",
      "linclust /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/MMseqsDB/species_and_humanDB /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/tmp/5843596209453182034/clu_redundancy /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/tmp/5843596209453182034/linclust --cluster-mode 0 --max-iterations 1000 --similarity-type 2 --threads 24 --compressed 0 -v 3 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' -a 0 --alignment-mode 3 --alignment-output-mode 0 --wrapped-scoring 0 -e 0.001 --min-seq-id 0.5 --min-aln-len 0 --seq-id-mode 0 --alt-ali 0 -c 0.5 --cov-mode 0 --max-seq-len 65535 --comp-bias-corr 1 --comp-bias-corr-scale 1 --max-rejected 2147483647 --max-accept 2147483647 --add-self-matches 0 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --score-bias 0 --realign 0 --realign-score-bias -0.2 --realign-max-seqs 2147483647 --corr-score-weight 0 --gap-open aa:11,nucl:5 --gap-extend aa:1,nucl:2 --zdrop 40 --alph-size aa:13,nucl:5 --kmer-per-seq 21 --spaced-kmer-mode 1 --kmer-per-seq-scale aa:0.000,nucl:0.200 --adjust-kmer-len 0 --mask 0 --mask-prob 0.9 --mask-lower-case 0 -k 0 --hash-shift 67 --split-memory-limit 0 --include-only-extendable 0 --ignore-multi-kmer 0 --rescore-mode 0 --filter-hits 0 --sort-results 0 --remove-tmp-files 0 --force-reuse 0 \n",
      "\n",
      "kmermatcher /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/MMseqsDB/species_and_humanDB /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/tmp/5843596209453182034/linclust/4944694312900497178/pref --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' --alph-size aa:13,nucl:5 --min-seq-id 0.5 --kmer-per-seq 21 --spaced-kmer-mode 1 --kmer-per-seq-scale aa:0.000,nucl:0.200 --adjust-kmer-len 0 --mask 0 --mask-prob 0.9 --mask-lower-case 0 --cov-mode 0 -k 0 -c 0.5 --max-seq-len 65535 --hash-shift 67 --split-memory-limit 0 --include-only-extendable 0 --ignore-multi-kmer 0 --threads 24 --compressed 0 -v 3 \n",
      "\n",
      "kmermatcher /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/MMseqsDB/species_and_humanDB /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/tmp/5843596209453182034/linclust/4944694312900497178/pref --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' --alph-size aa:13,nucl:5 --min-seq-id 0.5 --kmer-per-seq 21 --spaced-kmer-mode 1 --kmer-per-seq-scale aa:0.000,nucl:0.200 --adjust-kmer-len 0 --mask 0 --mask-prob 0.9 --mask-lower-case 0 --cov-mode 0 -k 0 -c 0.5 --max-seq-len 65535 --hash-shift 67 --split-memory-limit 0 --include-only-extendable 0 --ignore-multi-kmer 0 --threads 24 --compressed 0 -v 3 \n",
      "\n",
      "Database size: 142664 type: Aminoacid\n",
      "Reduced amino acid alphabet: (A S T) (C) (D B N) (E Q Z) (F Y) (G) (H) (I V) (K R) (L J M) (P) (W) (X) \n",
      "\n",
      "Generate k-mers list for 1 split\n",
      "[=================================================================] 142.66K 0s 538ms\n",
      "Sort kmer 0h 0m 0s 189ms\n",
      "Sort by rep. sequence 0h 0m 0s 104ms\n",
      "Time for fill: 0h 0m 0s 49ms\n",
      "Time for merging to pref: 0h 0m 2s 929ms\n",
      "Time for processing: 0h 0m 4s 465ms\n",
      "rescorediagonal /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/MMseqsDB/species_and_humanDB /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/MMseqsDB/species_and_humanDB /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/tmp/5843596209453182034/linclust/4944694312900497178/pref /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/tmp/5843596209453182034/linclust/4944694312900497178/pref_rescore1 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' --rescore-mode 0 --wrapped-scoring 0 --filter-hits 0 -e 0.001 -c 0.5 -a 0 --cov-mode 0 --min-seq-id 0.5 --min-aln-len 0 --seq-id-mode 0 --add-self-matches 0 --sort-results 0 --db-load-mode 0 --threads 24 --compressed 0 -v 3 \n",
      "\n",
      "[=================================================================] 142.66K 0s 103ms\n",
      "Time for merging to pref_rescore1: 0h 0m 3s 394ms\n",
      "Time for processing: 0h 0m 23s 99ms\n",
      "clust /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/MMseqsDB/species_and_humanDB /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/tmp/5843596209453182034/linclust/4944694312900497178/pref_rescore1 /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/tmp/5843596209453182034/linclust/4944694312900497178/pre_clust --cluster-mode 0 --max-iterations 1000 --similarity-type 2 --threads 24 --compressed 0 -v 3 \n",
      "\n",
      "Clustering mode: Set Cover\n",
      "[=================================================================] 142.66K 0s 42ms\n",
      "Sort entries\n",
      "Find missing connections\n",
      "Found 90501 new connections.\n",
      "Reconstruct initial order\n",
      "[=================================================================] 142.66K 0s 18ms\n",
      "Add missing connections\n",
      "[=================================================================] 142.66K 0s 12ms\n",
      "\n",
      "Time for read in: 0h 0m 0s 213ms\n",
      "Total time: 0h 0m 2s 62ms\n",
      "\n",
      "Size of the sequence database: 142664\n",
      "Size of the alignment database: 142664\n",
      "Number of clusters: 93656\n",
      "\n",
      "Writing results 0h 0m 0s 30ms\n",
      "Time for merging to pre_clust: 0h 0m 0s 415ms\n",
      "Time for processing: 0h 0m 3s 276ms\n",
      "createsubdb /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/tmp/5843596209453182034/linclust/4944694312900497178/order_redundancy /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/MMseqsDB/species_and_humanDB /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/tmp/5843596209453182034/linclust/4944694312900497178/input_step_redundancy -v 3 --subdb-mode 1 \n",
      "\n",
      "Time for merging to input_step_redundancy: 0h 0m 0s 691ms\n",
      "Time for processing: 0h 0m 2s 629ms\n",
      "createsubdb /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/tmp/5843596209453182034/linclust/4944694312900497178/order_redundancy /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/tmp/5843596209453182034/linclust/4944694312900497178/pref /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/tmp/5843596209453182034/linclust/4944694312900497178/pref_filter1 -v 3 --subdb-mode 1 \n",
      "\n",
      "Time for merging to pref_filter1: 0h 0m 0s 852ms\n",
      "Time for processing: 0h 0m 3s 473ms\n",
      "filterdb /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/tmp/5843596209453182034/linclust/4944694312900497178/pref_filter1 /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/tmp/5843596209453182034/linclust/4944694312900497178/pref_filter2 --filter-file /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/tmp/5843596209453182034/linclust/4944694312900497178/order_redundancy --threads 24 --compressed 0 -v 3 \n",
      "\n",
      "Filtering using file(s)\n",
      "[=================================================================] 93.66K 0s 50ms\n",
      "Time for merging to pref_filter2: 0h 0m 4s 38ms\n",
      "Time for processing: 0h 0m 21s 858ms\n",
      "rescorediagonal /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/tmp/5843596209453182034/linclust/4944694312900497178/input_step_redundancy /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/tmp/5843596209453182034/linclust/4944694312900497178/input_step_redundancy /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/tmp/5843596209453182034/linclust/4944694312900497178/pref_filter2 /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/tmp/5843596209453182034/linclust/4944694312900497178/pref_rescore2 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' --rescore-mode 1 --wrapped-scoring 0 --filter-hits 1 -e 0.001 -c 0.5 -a 0 --cov-mode 0 --min-seq-id 0.5 --min-aln-len 0 --seq-id-mode 0 --add-self-matches 0 --sort-results 0 --db-load-mode 0 --threads 24 --compressed 0 -v 3 \n",
      "\n",
      "[=================================================================] 93.66K 2s 69ms\n",
      "Time for merging to pref_rescore2: 0h 0m 0s 763ms\n",
      "Time for processing: 0h 0m 13s 844ms\n",
      "align /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/tmp/5843596209453182034/linclust/4944694312900497178/input_step_redundancy /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/tmp/5843596209453182034/linclust/4944694312900497178/input_step_redundancy /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/tmp/5843596209453182034/linclust/4944694312900497178/pref_rescore2 /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/tmp/5843596209453182034/linclust/4944694312900497178/aln --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' -a 0 --alignment-mode 3 --alignment-output-mode 0 --wrapped-scoring 0 -e 0.001 --min-seq-id 0.5 --min-aln-len 0 --seq-id-mode 0 --alt-ali 0 -c 0.5 --cov-mode 0 --max-seq-len 65535 --comp-bias-corr 1 --comp-bias-corr-scale 1 --max-rejected 2147483647 --max-accept 2147483647 --add-self-matches 0 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --score-bias 0 --realign 0 --realign-score-bias -0.2 --realign-max-seqs 2147483647 --corr-score-weight 0 --gap-open aa:11,nucl:5 --gap-extend aa:1,nucl:2 --zdrop 40 --threads 24 --compressed 0 -v 3 \n",
      "\n",
      "Compute score, coverage and sequence identity\n",
      "Query database size: 93656 type: Aminoacid\n",
      "Target database size: 93656 type: Aminoacid\n",
      "Calculation of alignments\n",
      "[=================================================================] 93.66K 7s 865ms\n",
      "Time for merging to aln: 0h 0m 5s 2ms\n",
      "116136 alignments calculated\n",
      "110503 sequence pairs passed the thresholds (0.951497 of overall calculated)\n",
      "1.179882 hits per query sequence\n",
      "Time for processing: 0h 0m 28s 391ms\n",
      "clust /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/tmp/5843596209453182034/linclust/4944694312900497178/input_step_redundancy /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/tmp/5843596209453182034/linclust/4944694312900497178/aln /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/tmp/5843596209453182034/linclust/4944694312900497178/clust --cluster-mode 0 --max-iterations 1000 --similarity-type 2 --threads 24 --compressed 0 -v 3 \n",
      "\n",
      "Clustering mode: Set Cover\n",
      "[=================================================================] 93.66K 0s 8ms\n",
      "Sort entries\n",
      "Find missing connections\n",
      "Found 16847 new connections.\n",
      "Reconstruct initial order\n",
      "[=================================================================] 93.66K 0s 9ms\n",
      "Add missing connections\n",
      "[=================================================================] 93.66K 0s 3ms\n",
      "\n",
      "Time for read in: 0h 0m 0s 165ms\n",
      "Total time: 0h 0m 0s 383ms\n",
      "\n",
      "Size of the sequence database: 93656\n",
      "Size of the alignment database: 93656\n",
      "Number of clusters: 80750\n",
      "\n",
      "Writing results 0h 0m 0s 21ms\n",
      "Time for merging to clust: 0h 0m 1s 362ms\n",
      "Time for processing: 0h 0m 3s 576ms\n",
      "mergeclusters /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/MMseqsDB/species_and_humanDB /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/tmp/5843596209453182034/clu_redundancy /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/tmp/5843596209453182034/linclust/4944694312900497178/pre_clust /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/tmp/5843596209453182034/linclust/4944694312900497178/clust --threads 24 --compressed 0 -v 3 \n",
      "\n",
      "Clustering step 1\n",
      "[=================================================================] 93.66K 0s 781ms\n",
      "Clustering step 2\n",
      "[=================================================================] 80.75K 0s 903ms\n",
      "Write merged clustering\n",
      "[=================================================================] 142.66K 20s 804ms\n",
      "Time for merging to clu_redundancy: 0h 0m 8s 805ms\n",
      "Time for processing: 0h 0m 35s 474ms\n",
      "createsubdb /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/tmp/5843596209453182034/clu_redundancy /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/MMseqsDB/species_and_humanDB /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/tmp/5843596209453182034/input_step_redundancy -v 3 --subdb-mode 1 \n",
      "\n",
      "Time for merging to input_step_redundancy: 0h 0m 2s 219ms\n",
      "Time for processing: 0h 0m 9s 310ms\n",
      "prefilter /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/tmp/5843596209453182034/input_step_redundancy /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/tmp/5843596209453182034/input_step_redundancy /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/tmp/5843596209453182034/pref_step0 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' --seed-sub-mat 'aa:VTML80.out,nucl:nucleotide.out' -s 1 -k 0 --k-score seq:2147483647,prof:2147483647 --alph-size aa:21,nucl:5 --max-seq-len 65535 --max-seqs 20 --split 0 --split-mode 2 --split-memory-limit 0 -c 0.5 --cov-mode 0 --comp-bias-corr 0 --comp-bias-corr-scale 1 --diag-score 0 --exact-kmer-matching 0 --mask 1 --mask-prob 0.9 --mask-lower-case 0 --min-ungapped-score 0 --add-self-matches 0 --spaced-kmer-mode 1 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --threads 24 --compressed 0 -v 3 \n",
      "\n",
      "Query database size: 80750 type: Aminoacid\n",
      "Estimated memory consumption: 1G\n",
      "Target database size: 80750 type: Aminoacid\n",
      "Index table k-mer threshold: 154 at k-mer size 6 \n",
      "Index table: counting k-mers\n",
      "[=================================================================] 80.75K 0s 660ms\n",
      "Index table: Masked residues: 921833\n",
      "Index table: fill\n",
      "[=================================================================] 80.75K 0s 250ms\n",
      "Index statistics\n",
      "Entries:          8872509\n",
      "DB size:          539 MB\n",
      "Avg k-mer size:   0.138633\n",
      "Top 10 k-mers\n",
      "    LYDMNY\t149\n",
      "    TFDAFS\t117\n",
      "    VADTEW\t89\n",
      "    EWLDQK\t64\n",
      "    YRVLYR\t57\n",
      "    ECKFSS\t55\n",
      "    GGDCDS\t48\n",
      "    LYEMNY\t47\n",
      "    VAYSEW\t46\n",
      "    LREHEQ\t40\n",
      "Time for index table init: 0h 0m 1s 499ms\n",
      "Process prefiltering step 1 of 1\n",
      "\n",
      "k-mer similarity threshold: 154\n",
      "Starting prefiltering scores calculation (step 1 of 1)\n",
      "Query db start 1 to 80750\n",
      "Target db start 1 to 80750\n",
      "[=================================================================] 80.75K 1s 336ms\n",
      "\n",
      "2.277450 k-mers per position\n",
      "371 DB matches per sequence\n",
      "0 overflows\n",
      "0 queries produce too many hits (truncated result)\n",
      "3 sequences passed prefiltering per query sequence\n",
      "2 median result list length\n",
      "0 sequences with 0 size result lists\n",
      "Time for merging to pref_step0: 0h 0m 3s 136ms\n",
      "Time for processing: 0h 0m 16s 482ms\n",
      "align /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/tmp/5843596209453182034/input_step_redundancy /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/tmp/5843596209453182034/input_step_redundancy /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/tmp/5843596209453182034/pref_step0 /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/tmp/5843596209453182034/aln_step0 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' -a 0 --alignment-mode 3 --alignment-output-mode 0 --wrapped-scoring 0 -e 0.001 --min-seq-id 0.5 --min-aln-len 0 --seq-id-mode 0 --alt-ali 0 -c 0.5 --cov-mode 0 --max-seq-len 65535 --comp-bias-corr 0 --comp-bias-corr-scale 1 --max-rejected 2147483647 --max-accept 2147483647 --add-self-matches 0 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --score-bias 0 --realign 0 --realign-score-bias -0.2 --realign-max-seqs 2147483647 --corr-score-weight 0 --gap-open aa:11,nucl:5 --gap-extend aa:1,nucl:2 --zdrop 40 --threads 24 --compressed 0 -v 3 \n",
      "\n",
      "Compute score, coverage and sequence identity\n",
      "Query database size: 80750 type: Aminoacid\n",
      "Target database size: 80750 type: Aminoacid\n",
      "Calculation of alignments\n",
      "[=================================================================] 80.75K 23s 96ms\n",
      "Time for merging to aln_step0: 0h 0m 2s 266ms\n",
      "205573 alignments calculated\n",
      "102465 sequence pairs passed the thresholds (0.498436 of overall calculated)\n",
      "1.268916 hits per query sequence\n",
      "Time for processing: 0h 0m 44s 823ms\n",
      "clust /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/tmp/5843596209453182034/input_step_redundancy /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/tmp/5843596209453182034/aln_step0 /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/tmp/5843596209453182034/clu_step0 --cluster-mode 0 --max-iterations 1000 --similarity-type 2 --threads 24 --compressed 0 -v 3 \n",
      "\n",
      "Clustering mode: Set Cover\n",
      "[=================================================================] 80.75K 0s 47ms\n",
      "Sort entries\n",
      "Find missing connections\n",
      "Found 1911 new connections.\n",
      "Reconstruct initial order\n",
      "[=================================================================] 80.75K 0s 13ms\n",
      "Add missing connections\n",
      "[=================================================================] 80.75K 0s 2ms\n",
      "\n",
      "Time for read in: 0h 0m 0s 262ms\n",
      "Total time: 0h 0m 2s 170ms\n",
      "\n",
      "Size of the sequence database: 80750\n",
      "Size of the alignment database: 80750\n",
      "Number of clusters: 72859\n",
      "\n",
      "Writing results 0h 0m 0s 401ms\n",
      "Time for merging to clu_step0: 0h 0m 1s 547ms\n",
      "Time for processing: 0h 0m 5s 905ms\n",
      "createsubdb /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/tmp/5843596209453182034/clu_step0 /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/tmp/5843596209453182034/input_step_redundancy /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/tmp/5843596209453182034/input_step1 -v 3 --subdb-mode 1 \n",
      "\n",
      "Time for merging to input_step1: 0h 0m 3s 51ms\n",
      "Time for processing: 0h 0m 9s 165ms\n",
      "prefilter /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/tmp/5843596209453182034/input_step1 /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/tmp/5843596209453182034/input_step1 /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/tmp/5843596209453182034/pref_step1 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' --seed-sub-mat 'aa:VTML80.out,nucl:nucleotide.out' -s 2 -k 0 --k-score seq:2147483647,prof:2147483647 --alph-size aa:21,nucl:5 --max-seq-len 65535 --max-seqs 20 --split 0 --split-mode 2 --split-memory-limit 0 -c 0.5 --cov-mode 0 --comp-bias-corr 1 --comp-bias-corr-scale 1 --diag-score 1 --exact-kmer-matching 0 --mask 1 --mask-prob 0.9 --mask-lower-case 0 --min-ungapped-score 15 --add-self-matches 0 --spaced-kmer-mode 1 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --threads 24 --compressed 0 -v 3 \n",
      "\n",
      "Query database size: 72859 type: Aminoacid\n",
      "Estimated memory consumption: 1G\n",
      "Target database size: 72859 type: Aminoacid\n",
      "Index table k-mer threshold: 145 at k-mer size 6 \n",
      "Index table: counting k-mers\n",
      "[=================================================================] 72.86K 0s 685ms\n",
      "Index table: Masked residues: 603789\n",
      "Index table: fill\n",
      "[=================================================================] 72.86K 0s 315ms\n",
      "Index statistics\n",
      "Entries:          9449238\n",
      "DB size:          542 MB\n",
      "Avg k-mer size:   0.147644\n",
      "Top 10 k-mers\n",
      "    DVVFEE\t112\n",
      "    LYDMNY\t64\n",
      "    TFDAFT\t52\n",
      "    LLGPGK\t39\n",
      "    GPELLR\t39\n",
      "    LLPQLY\t37\n",
      "    VADTEW\t33\n",
      "    PLPPPP\t30\n",
      "    SPPPSP\t30\n",
      "    GGDCDS\t30\n",
      "Time for index table init: 0h 0m 1s 642ms\n",
      "Process prefiltering step 1 of 1\n",
      "\n",
      "k-mer similarity threshold: 145\n",
      "Starting prefiltering scores calculation (step 1 of 1)\n",
      "Query db start 1 to 72859\n",
      "Target db start 1 to 72859\n",
      "[=================================================================] 72.86K 3s 86ms\n",
      "\n",
      "6.007843 k-mers per position\n",
      "316 DB matches per sequence\n",
      "0 overflows\n",
      "0 queries produce too many hits (truncated result)\n",
      "3 sequences passed prefiltering per query sequence\n",
      "1 median result list length\n",
      "0 sequences with 0 size result lists\n",
      "Time for merging to pref_step1: 0h 0m 7s 209ms\n",
      "Time for processing: 0h 0m 35s 900ms\n",
      "align /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/tmp/5843596209453182034/input_step1 /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/tmp/5843596209453182034/input_step1 /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/tmp/5843596209453182034/pref_step1 /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/tmp/5843596209453182034/aln_step1 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' -a 0 --alignment-mode 3 --alignment-output-mode 0 --wrapped-scoring 0 -e 0.001 --min-seq-id 0.5 --min-aln-len 0 --seq-id-mode 0 --alt-ali 0 -c 0.5 --cov-mode 0 --max-seq-len 65535 --comp-bias-corr 1 --comp-bias-corr-scale 1 --max-rejected 2147483647 --max-accept 2147483647 --add-self-matches 0 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --score-bias 0 --realign 0 --realign-score-bias -0.2 --realign-max-seqs 2147483647 --corr-score-weight 0 --gap-open aa:11,nucl:5 --gap-extend aa:1,nucl:2 --zdrop 40 --threads 24 --compressed 0 -v 3 \n",
      "\n",
      "Compute score, coverage and sequence identity\n",
      "Query database size: 72859 type: Aminoacid\n",
      "Target database size: 72859 type: Aminoacid\n",
      "Calculation of alignments\n",
      "[=================================================================] 72.86K 9s 353ms\n",
      "Time for merging to aln_step1: 0h 0m 8s 685ms\n",
      "154485 alignments calculated\n",
      "73621 sequence pairs passed the thresholds (0.476558 of overall calculated)\n",
      "1.010459 hits per query sequence\n",
      "Time for processing: 0h 0m 40s 863ms\n",
      "clust /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/tmp/5843596209453182034/input_step1 /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/tmp/5843596209453182034/aln_step1 /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/tmp/5843596209453182034/clu_step1 --cluster-mode 0 --max-iterations 1000 --similarity-type 2 --threads 24 --compressed 0 -v 3 \n",
      "\n",
      "Clustering mode: Set Cover\n",
      "[=================================================================] 72.86K 0s 42ms\n",
      "Sort entries\n",
      "Find missing connections\n",
      "Found 232 new connections.\n",
      "Reconstruct initial order\n",
      "[=================================================================] 72.86K 0s 37ms\n",
      "Add missing connections\n",
      "[=================================================================] 72.86K 0s 1ms\n",
      "\n",
      "Time for read in: 0h 0m 0s 179ms\n",
      "Total time: 0h 0m 2s 414ms\n",
      "\n",
      "Size of the sequence database: 72859\n",
      "Size of the alignment database: 72859\n",
      "Number of clusters: 72397\n",
      "\n",
      "Writing results 0h 0m 0s 21ms\n",
      "Time for merging to clu_step1: 0h 0m 1s 143ms\n",
      "Time for processing: 0h 0m 5s 780ms\n",
      "createsubdb /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/tmp/5843596209453182034/clu_step1 /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/tmp/5843596209453182034/input_step1 /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/tmp/5843596209453182034/input_step2 -v 3 --subdb-mode 1 \n",
      "\n",
      "Time for merging to input_step2: 0h 0m 3s 668ms\n",
      "Time for processing: 0h 0m 6s 674ms\n",
      "prefilter /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/tmp/5843596209453182034/input_step2 /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/tmp/5843596209453182034/input_step2 /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/tmp/5843596209453182034/pref_step2 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' --seed-sub-mat 'aa:VTML80.out,nucl:nucleotide.out' -s 3 -k 0 --k-score seq:2147483647,prof:2147483647 --alph-size aa:21,nucl:5 --max-seq-len 65535 --max-seqs 20 --split 0 --split-mode 2 --split-memory-limit 0 -c 0.5 --cov-mode 0 --comp-bias-corr 1 --comp-bias-corr-scale 1 --diag-score 1 --exact-kmer-matching 0 --mask 1 --mask-prob 0.9 --mask-lower-case 0 --min-ungapped-score 15 --add-self-matches 0 --spaced-kmer-mode 1 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --threads 24 --compressed 0 -v 3 \n",
      "\n",
      "Query database size: 72397 type: Aminoacid\n",
      "Estimated memory consumption: 1G\n",
      "Target database size: 72397 type: Aminoacid\n",
      "Index table k-mer threshold: 136 at k-mer size 6 \n",
      "Index table: counting k-mers\n",
      "[=================================================================] 72.40K 0s 712ms\n",
      "Index table: Masked residues: 579555\n",
      "Index table: fill\n",
      "[=================================================================] 72.40K 0s 297ms\n",
      "Index statistics\n",
      "Entries:          12137047\n",
      "DB size:          557 MB\n",
      "Avg k-mer size:   0.189641\n",
      "Top 10 k-mers\n",
      "    DVVFEE\t90\n",
      "    VLQLLP\t67\n",
      "    LYDMNY\t53\n",
      "    VMEYLV\t48\n",
      "    TFDAFS\t45\n",
      "    TFDAFT\t43\n",
      "    TKQLLV\t38\n",
      "    CQDGLV\t34\n",
      "    PPPPLP\t30\n",
      "    PLPPPP\t30\n",
      "Time for index table init: 0h 0m 2s 563ms\n",
      "Process prefiltering step 1 of 1\n",
      "\n",
      "k-mer similarity threshold: 136\n",
      "Starting prefiltering scores calculation (step 1 of 1)\n",
      "Query db start 1 to 72397\n",
      "Target db start 1 to 72397\n",
      "[=================================================================] 72.40K 2s 812ms\n",
      "\n",
      "17.161285 k-mers per position\n",
      "761 DB matches per sequence\n",
      "0 overflows\n",
      "0 queries produce too many hits (truncated result)\n",
      "5 sequences passed prefiltering per query sequence\n",
      "3 median result list length\n",
      "0 sequences with 0 size result lists\n",
      "Time for merging to pref_step2: 0h 0m 11s 52ms\n",
      "Time for processing: 0h 0m 50s 45ms\n",
      "align /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/tmp/5843596209453182034/input_step2 /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/tmp/5843596209453182034/input_step2 /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/tmp/5843596209453182034/pref_step2 /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/tmp/5843596209453182034/aln_step2 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' -a 0 --alignment-mode 3 --alignment-output-mode 0 --wrapped-scoring 0 -e 0.001 --min-seq-id 0.5 --min-aln-len 0 --seq-id-mode 0 --alt-ali 0 -c 0.5 --cov-mode 0 --max-seq-len 65535 --comp-bias-corr 1 --comp-bias-corr-scale 1 --max-rejected 2147483647 --max-accept 2147483647 --add-self-matches 0 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --score-bias 0 --realign 0 --realign-score-bias -0.2 --realign-max-seqs 2147483647 --corr-score-weight 0 --gap-open aa:11,nucl:5 --gap-extend aa:1,nucl:2 --zdrop 40 --threads 24 --compressed 0 -v 3 \n",
      "\n",
      "Compute score, coverage and sequence identity\n",
      "Query database size: 72397 type: Aminoacid\n",
      "Target database size: 72397 type: Aminoacid\n",
      "Calculation of alignments\n",
      "[=================================================================] 72.40K 10s 10ms\n",
      "Time for merging to aln_step2: 0h 0m 8s 587ms\n",
      "208854 alignments calculated\n",
      "72540 sequence pairs passed the thresholds (0.347324 of overall calculated)\n",
      "1.001975 hits per query sequence\n",
      "Time for processing: 0h 0m 39s 493ms\n",
      "clust /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/tmp/5843596209453182034/input_step2 /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/tmp/5843596209453182034/aln_step2 /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/tmp/5843596209453182034/clu_step2 --cluster-mode 0 --max-iterations 1000 --similarity-type 2 --threads 24 --compressed 0 -v 3 \n",
      "\n",
      "Clustering mode: Set Cover\n",
      "[=================================================================] 72.40K 0s 32ms\n",
      "Sort entries\n",
      "Find missing connections\n",
      "Found 51 new connections.\n",
      "Reconstruct initial order\n",
      "[=================================================================] 72.40K 0s 3ms\n",
      "Add missing connections\n",
      "[=================================================================] 72.40K 0s 1ms\n",
      "\n",
      "Time for read in: 0h 0m 0s 159ms\n",
      "Total time: 0h 0m 1s 59ms\n",
      "\n",
      "Size of the sequence database: 72397\n",
      "Size of the alignment database: 72397\n",
      "Number of clusters: 72301\n",
      "\n",
      "Writing results 0h 0m 0s 19ms\n",
      "Time for merging to clu_step2: 0h 0m 2s 472ms\n",
      "Time for processing: 0h 0m 4s 27ms\n",
      "mergeclusters /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/MMseqsDB/species_and_humanDB /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/Clustering/species_and_human_cluDB /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/tmp/5843596209453182034/clu_redundancy /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/tmp/5843596209453182034/clu_step0 /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/tmp/5843596209453182034/clu_step1 /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/tmp/5843596209453182034/clu_step2 --threads 24 --compressed 0 -v 3 \n",
      "\n",
      "Clustering step 1\n",
      "[=================================================================] 80.75K 1s 64ms\n",
      "Clustering step 2\n",
      "[=================================================================] 72.86K 1s 146ms\n",
      "Clustering step 3\n",
      "[=================================================================] 72.40K 1s 752ms\n",
      "Clustering step 4\n",
      "[=================================================================] 72.30K 1s 826ms\n",
      "Write merged clustering\n",
      "[=================================================================] 142.66K 17s 145ms\n",
      "Time for merging to species_and_human_cluDB: 0h 0m 7s 834ms\n",
      "Time for processing: 0h 0m 33s 669ms\n",
      "createtsv /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/MMseqsDB/species_and_humanDB /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/MMseqsDB/species_and_humanDB /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/Clustering/species_and_human_cluDB /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/Clustering/species_and_human_cluDB.tsv \n",
      "\n",
      "MMseqs Version:                 \t14.7e284\n",
      "First sequence as representative\tfalse\n",
      "Target column                   \t1\n",
      "Add full header                 \tfalse\n",
      "Sequence source                 \t0\n",
      "Database output                 \tfalse\n",
      "Threads                         \t24\n",
      "Compressed                      \t0\n",
      "Verbosity                       \t3\n",
      "\n",
      "Time for merging to species_and_human_cluDB.tsv: 0h 0m 16s 626ms\n",
      "Time for processing: 0h 0m 53s 122ms\n"
     ]
    }
   ],
   "source": [
    "%%bash -s \"$evoDir\" \"$users_dir\"\n",
    "\n",
    "module load Miniconda3/4.9.2\n",
    "source /soft/system/software/Miniconda3/4.9.2/bin/activate MMseqs2\n",
    "\n",
    "mkdir -p $1/Clustering\n",
    "\n",
    "rm -r $1/Clustering/species_and_human_cluDB*\n",
    "\n",
    "## cluster all species and human\n",
    "mmseqs cluster -c 0.5 --min-seq-id 0.5 $1/MMseqsDB/species_and_humanDB $1/Clustering/species_and_human_cluDB $1/tmp\n",
    "mmseqs createtsv $1/MMseqsDB/species_and_humanDB $1/MMseqsDB/species_and_humanDB $1/Clustering/species_and_human_cluDB $1/Clustering/species_and_human_cluDB.tsv"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Analysis"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "all human\n",
      "26375\n",
      "all human after annotation\n",
      "26375\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene_name</th>\n",
       "      <th>0</th>\n",
       "      <th>1</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>gene_type</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>lncRNA</th>\n",
       "      <td>2318</td>\n",
       "      <td>2318</td>\n",
       "      <td>2318</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>novel</th>\n",
       "      <td>130</td>\n",
       "      <td>130</td>\n",
       "      <td>130</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>processed_pseudogene</th>\n",
       "      <td>1557</td>\n",
       "      <td>1557</td>\n",
       "      <td>1557</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>protein_coding</th>\n",
       "      <td>22370</td>\n",
       "      <td>22370</td>\n",
       "      <td>22370</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                      gene_name      0      1\n",
       "gene_type                                    \n",
       "lncRNA                     2318   2318   2318\n",
       "novel                       130    130    130\n",
       "processed_pseudogene       1557   1557   1557\n",
       "protein_coding            22370  22370  22370"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "clustering = pd.read_csv(os.path.join(evoDir,\"Clustering/species_and_human_cluDB.tsv\"), header=None, sep=\"\\t\")\n",
    "clustering = clustering[clustering[0].str.startswith(\"ENST\") | clustering[0].str.startswith(\"TCONS\")]\n",
    "print(\"all human\")\n",
    "print(len(clustering))\n",
    "clustering['transcript_id'] = clustering[0].str.split(\":\", expand=True)[0]\n",
    "clustering['transcript_id'] = clustering[\"transcript_id\"].str.split(\".\", expand=True)[0]\n",
    "\n",
    "clustering = clustering.merge(transcript_gene, on=\"transcript_id\")\n",
    "print(\"all human after annotation\")\n",
    "print(len(clustering))\n",
    "small_clust = clustering[[\"gene_name\",\"gene_type\",0,1]]\n",
    "small_clust.groupby(\"gene_type\").count()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene_name</th>\n",
       "      <th>gene_type</th>\n",
       "      <th>0</th>\n",
       "      <th>1</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>R3HCC1L</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>ENST00000298999.8:10:+|18|3393:305:350|ouORF|TTG</td>\n",
       "      <td>ENST00000298999.8:10:+|18|3393:305:350|ouORF|TTG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>R3HCC1L</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>ENST00000298999.8:10:+|8|3393:157:268|uORF|GTG</td>\n",
       "      <td>ENST00000298999.8:10:+|8|3393:157:268|uORF|GTG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>R3HCC1L</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>ENST00000298999.8:10:+|2|3393:78:282|uORF|CTG</td>\n",
       "      <td>ENST00000298999.8:10:+|2|3393:78:282|uORF|CTG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>R3HCC1L</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>ENST00000298999.8:10:+|151|3393:3093:3348|dORF...</td>\n",
       "      <td>ENST00000298999.8:10:+|151|3393:3093:3348|dORF...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>CHODL</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>ENST00000299295.7:21:+|58|2545:1089:1329|odORF...</td>\n",
       "      <td>ENST00000299295.7:21:+|58|2545:1089:1329|odORF...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26370</th>\n",
       "      <td>FOXRED2</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>ENST00000684868.1:22:-|6|5640:116:494|uORF|CTG</td>\n",
       "      <td>ENST00000684868.1:22:-|6|5640:116:494|uORF|CTG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26371</th>\n",
       "      <td>EXOSC8</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>ENST00000685624.1:13:+|32|4097:704:755|uORF|ATG</td>\n",
       "      <td>ENST00000685624.1:13:+|32|4097:704:755|uORF|ATG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26372</th>\n",
       "      <td>SNHG33</td>\n",
       "      <td>lncRNA</td>\n",
       "      <td>ENST00000686765.2:3:+|2|1688:28:175|noncoding|ATG</td>\n",
       "      <td>ENST00000686765.2:3:+|2|1688:28:175|noncoding|ATG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26373</th>\n",
       "      <td>LINC01871</td>\n",
       "      <td>lncRNA</td>\n",
       "      <td>ENST00000690653.3:2:+|23|2292:494:677|noncodin...</td>\n",
       "      <td>ENST00000690653.3:2:+|23|2292:494:677|noncodin...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26374</th>\n",
       "      <td>ENSG00000286039</td>\n",
       "      <td>lncRNA</td>\n",
       "      <td>ENST00000692288.2:4:+|65|3183:1349:1841|noncod...</td>\n",
       "      <td>ENST00000692288.2:4:+|65|3183:1349:1841|noncod...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>26375 rows × 4 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "             gene_name       gene_type  \\\n",
       "0              R3HCC1L  protein_coding   \n",
       "1              R3HCC1L  protein_coding   \n",
       "2              R3HCC1L  protein_coding   \n",
       "3              R3HCC1L  protein_coding   \n",
       "4                CHODL  protein_coding   \n",
       "...                ...             ...   \n",
       "26370          FOXRED2  protein_coding   \n",
       "26371           EXOSC8  protein_coding   \n",
       "26372           SNHG33          lncRNA   \n",
       "26373        LINC01871          lncRNA   \n",
       "26374  ENSG00000286039          lncRNA   \n",
       "\n",
       "                                                       0  \\\n",
       "0       ENST00000298999.8:10:+|18|3393:305:350|ouORF|TTG   \n",
       "1         ENST00000298999.8:10:+|8|3393:157:268|uORF|GTG   \n",
       "2          ENST00000298999.8:10:+|2|3393:78:282|uORF|CTG   \n",
       "3      ENST00000298999.8:10:+|151|3393:3093:3348|dORF...   \n",
       "4      ENST00000299295.7:21:+|58|2545:1089:1329|odORF...   \n",
       "...                                                  ...   \n",
       "26370     ENST00000684868.1:22:-|6|5640:116:494|uORF|CTG   \n",
       "26371    ENST00000685624.1:13:+|32|4097:704:755|uORF|ATG   \n",
       "26372  ENST00000686765.2:3:+|2|1688:28:175|noncoding|ATG   \n",
       "26373  ENST00000690653.3:2:+|23|2292:494:677|noncodin...   \n",
       "26374  ENST00000692288.2:4:+|65|3183:1349:1841|noncod...   \n",
       "\n",
       "                                                       1  \n",
       "0       ENST00000298999.8:10:+|18|3393:305:350|ouORF|TTG  \n",
       "1         ENST00000298999.8:10:+|8|3393:157:268|uORF|GTG  \n",
       "2          ENST00000298999.8:10:+|2|3393:78:282|uORF|CTG  \n",
       "3      ENST00000298999.8:10:+|151|3393:3093:3348|dORF...  \n",
       "4      ENST00000299295.7:21:+|58|2545:1089:1329|odORF...  \n",
       "...                                                  ...  \n",
       "26370     ENST00000684868.1:22:-|6|5640:116:494|uORF|CTG  \n",
       "26371    ENST00000685624.1:13:+|32|4097:704:755|uORF|ATG  \n",
       "26372  ENST00000686765.2:3:+|2|1688:28:175|noncoding|ATG  \n",
       "26373  ENST00000690653.3:2:+|23|2292:494:677|noncodin...  \n",
       "26374  ENST00000692288.2:4:+|65|3183:1349:1841|noncod...  \n",
       "\n",
       "[26375 rows x 4 columns]"
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "small_clust"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "noncoding all\n",
      "4005\n",
      "3308\n",
      "altORFs all\n",
      "7827\n",
      "7516\n",
      "Canonical all\n",
      "14543\n",
      "2909\n"
     ]
    }
   ],
   "source": [
    "## non-coding\n",
    "noncoding = clustering[clustering[0].str.contains(\"noncoding\")]\n",
    "noncoding.to_csv(os.path.join(evoDir,\"Clustering/noncodingHuman_speciesDB.tsv\"), header=None, index=None, sep=\"\\t\")\n",
    "print(\"noncoding all\")\n",
    "print(len(noncoding))\n",
    "print(len(set(noncoding[0].values.tolist())))\n",
    "\n",
    "## altORFs\n",
    "altORFs = clustering[clustering[0].str.contains(\"ORF\")]\n",
    "altORFs.to_csv(os.path.join(evoDir,\"Clustering/altORFsHuman_speciesDB.tsv\"), header=None, index=None, sep=\"\\t\")\n",
    "print(\"altORFs all\")\n",
    "print(len(altORFs))\n",
    "print(len(set(altORFs[0].values.tolist())))\n",
    "\n",
    "## canonical\n",
    "canonical = clustering[clustering[0].str.contains(\"canonical\")]\n",
    "print(\"Canonical all\")\n",
    "print(len(canonical))\n",
    "print(len(set(canonical[0].values.tolist())))\n",
    "canonical.to_csv(os.path.join(evoDir,\"Clustering/canonicalHuman_speciesDB.tsv\"), header=None, index=None, sep=\"\\t\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### NONCODING"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Conserved noncoding human ORFs\n",
      "176\n",
      "Non-Conserved noncoding human ORFs\n",
      "3308\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene_name</th>\n",
       "      <th>gene_type</th>\n",
       "      <th>orfID</th>\n",
       "      <th>1</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>190</th>\n",
       "      <td>ENSG00000260422</td>\n",
       "      <td>lncRNA</td>\n",
       "      <td>ENST00000564697.1:6:-|17|1639:356:590|noncodin...</td>\n",
       "      <td>ENST00000564697.1:6:-|17|1639:356:590|noncodin...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>193</th>\n",
       "      <td>ENSG00000290665</td>\n",
       "      <td>lncRNA</td>\n",
       "      <td>ENST00000567226.2:15:-|6|2266:104:320|noncodin...</td>\n",
       "      <td>ENST00000567226.2:15:-|6|2266:104:320|noncodin...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>210</th>\n",
       "      <td>LINC02864</td>\n",
       "      <td>lncRNA</td>\n",
       "      <td>ENST00000581862.7:18:-|12|3689:297:567|noncodi...</td>\n",
       "      <td>ENST00000581862.7:18:-|12|3689:297:567|noncodi...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>212</th>\n",
       "      <td>RDM1P5</td>\n",
       "      <td>lncRNA</td>\n",
       "      <td>ENST00000583447.6:17:+|10|2309:144:324|noncodi...</td>\n",
       "      <td>ENST00000583447.6:17:+|10|2309:144:324|noncodi...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>241</th>\n",
       "      <td>SINHCAFP2</td>\n",
       "      <td>processed_pseudogene</td>\n",
       "      <td>ENST00000591086.1:18:-|2|664:24:237|noncoding|ATG</td>\n",
       "      <td>ENST00000591086.1:18:-|2|664:24:237|noncoding|ATG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26360</th>\n",
       "      <td>DUSP5-DT</td>\n",
       "      <td>lncRNA</td>\n",
       "      <td>ENST00000666621.1:10:-|25|3011:473:665|noncodi...</td>\n",
       "      <td>ENST00000666621.1:10:-|25|3011:473:665|noncodi...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26361</th>\n",
       "      <td>MAP4K3-DT</td>\n",
       "      <td>lncRNA</td>\n",
       "      <td>ENST00000671403.1:2:+|11|4190:235:391|noncodin...</td>\n",
       "      <td>ENST00000671403.1:2:+|11|4190:235:391|noncodin...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26372</th>\n",
       "      <td>SNHG33</td>\n",
       "      <td>lncRNA</td>\n",
       "      <td>ENST00000686765.2:3:+|2|1688:28:175|noncoding|ATG</td>\n",
       "      <td>ENST00000686765.2:3:+|2|1688:28:175|noncoding|ATG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26373</th>\n",
       "      <td>LINC01871</td>\n",
       "      <td>lncRNA</td>\n",
       "      <td>ENST00000690653.3:2:+|23|2292:494:677|noncodin...</td>\n",
       "      <td>ENST00000690653.3:2:+|23|2292:494:677|noncodin...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26374</th>\n",
       "      <td>ENSG00000286039</td>\n",
       "      <td>lncRNA</td>\n",
       "      <td>ENST00000692288.2:4:+|65|3183:1349:1841|noncod...</td>\n",
       "      <td>ENST00000692288.2:4:+|65|3183:1349:1841|noncod...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>3527 rows × 4 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "             gene_name             gene_type  \\\n",
       "190    ENSG00000260422                lncRNA   \n",
       "193    ENSG00000290665                lncRNA   \n",
       "210          LINC02864                lncRNA   \n",
       "212             RDM1P5                lncRNA   \n",
       "241          SINHCAFP2  processed_pseudogene   \n",
       "...                ...                   ...   \n",
       "26360         DUSP5-DT                lncRNA   \n",
       "26361        MAP4K3-DT                lncRNA   \n",
       "26372           SNHG33                lncRNA   \n",
       "26373        LINC01871                lncRNA   \n",
       "26374  ENSG00000286039                lncRNA   \n",
       "\n",
       "                                                   orfID  \\\n",
       "190    ENST00000564697.1:6:-|17|1639:356:590|noncodin...   \n",
       "193    ENST00000567226.2:15:-|6|2266:104:320|noncodin...   \n",
       "210    ENST00000581862.7:18:-|12|3689:297:567|noncodi...   \n",
       "212    ENST00000583447.6:17:+|10|2309:144:324|noncodi...   \n",
       "241    ENST00000591086.1:18:-|2|664:24:237|noncoding|ATG   \n",
       "...                                                  ...   \n",
       "26360  ENST00000666621.1:10:-|25|3011:473:665|noncodi...   \n",
       "26361  ENST00000671403.1:2:+|11|4190:235:391|noncodin...   \n",
       "26372  ENST00000686765.2:3:+|2|1688:28:175|noncoding|ATG   \n",
       "26373  ENST00000690653.3:2:+|23|2292:494:677|noncodin...   \n",
       "26374  ENST00000692288.2:4:+|65|3183:1349:1841|noncod...   \n",
       "\n",
       "                                                       1  \n",
       "190    ENST00000564697.1:6:-|17|1639:356:590|noncodin...  \n",
       "193    ENST00000567226.2:15:-|6|2266:104:320|noncodin...  \n",
       "210    ENST00000581862.7:18:-|12|3689:297:567|noncodi...  \n",
       "212    ENST00000583447.6:17:+|10|2309:144:324|noncodi...  \n",
       "241    ENST00000591086.1:18:-|2|664:24:237|noncoding|ATG  \n",
       "...                                                  ...  \n",
       "26360  ENST00000666621.1:10:-|25|3011:473:665|noncodi...  \n",
       "26361  ENST00000671403.1:2:+|11|4190:235:391|noncodin...  \n",
       "26372  ENST00000686765.2:3:+|2|1688:28:175|noncoding|ATG  \n",
       "26373  ENST00000690653.3:2:+|23|2292:494:677|noncodin...  \n",
       "26374  ENST00000692288.2:4:+|65|3183:1349:1841|noncod...  \n",
       "\n",
       "[3527 rows x 4 columns]"
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# if column 1 starts with a specie, it is conserved\n",
    "noncoding_conserved = noncoding[~noncoding[1].str.startswith(\"ENST\")]\n",
    "noncoding_conserved = noncoding_conserved[~noncoding_conserved[1].str.startswith(\"TCONS\")]\n",
    "noncoding_conserved.to_csv(os.path.join(evoDir,\"Clustering/noncodingHuman_Conserved.tsv\"), header=None, index=None, sep=\"\\t\")\n",
    "print(\"Conserved noncoding human ORFs\")\n",
    "print(len(set(noncoding_conserved[0].values.tolist())))\n",
    "noncoding_conserved = noncoding_conserved[['gene_name','gene_type',0,1]]\n",
    "noncoding_conserved.columns = ['gene_name','gene_type','orfID',1]\n",
    "\n",
    "# if column 1 starts with a human gene, it is not conserved\n",
    "noncoding_NonConserved = noncoding[noncoding[1].str.startswith(\"ENST\") | noncoding[1].str.startswith(\"TCONS\")]\n",
    "noncoding_NonConserved.to_csv(os.path.join(evoDir,\"Clustering/noncodingHuman_NonConserved.tsv\"), header=None, index=None, sep=\"\\t\")\n",
    "print(\"Non-Conserved noncoding human ORFs\")\n",
    "print(len(set(noncoding_NonConserved[0].values.tolist())))\n",
    "noncoding_NonConserved = noncoding_NonConserved[['gene_name','gene_type',0,1]]\n",
    "noncoding_NonConserved.columns = ['gene_name','gene_type','orfID',1]\n",
    "noncoding_NonConserved"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "176\n",
      "176\n",
      "176\n",
      "176\n",
      "176\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_2049236/3235198060.py:8: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  noncoding_conserved_to_matrix['presence'] = 1\n",
      "/tmp/ipykernel_2049236/3235198060.py:25: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  noncoding_conserved_matrix_ordered['sum'] = noncoding_conserved_matrix_ordered.select_dtypes(include='number').sum(axis=1)\n"
     ]
    }
   ],
   "source": [
    "## conserved non-coding\n",
    "noncoding_conserved_to_matrix = noncoding_conserved\n",
    "print(len(set(noncoding_conserved_to_matrix.orfID.values.tolist())))\n",
    "# Assuming noncoding_conserved_to_matrix is already defined\n",
    "noncoding_conserved_to_matrix['specie'] = noncoding_conserved_to_matrix[1].str.split(\"_\", expand=True)[0]\n",
    "noncoding_conserved_to_matrix = noncoding_conserved_to_matrix[['orfID', 'specie']]\n",
    "noncoding_conserved_to_matrix.groupby('specie').count()\n",
    "noncoding_conserved_to_matrix['presence'] = 1\n",
    "## control\n",
    "print(len(set(noncoding_conserved_to_matrix.orfID.values.tolist())))\n",
    "\n",
    "# Pivoting the data without creating an 'id', and keeping 'orfID' as the unique identifier\n",
    "noncoding_conserved_matrix = noncoding_conserved_to_matrix.pivot_table(index='orfID', columns='specie', values='presence', aggfunc='max')\n",
    "\n",
    "# Fill NaN values with 0\n",
    "noncoding_conserved_matrix = noncoding_conserved_matrix.fillna(0)\n",
    "\n",
    "# Reset index so 'orfID' becomes a column instead of an index\n",
    "noncoding_conserved_matrix = noncoding_conserved_matrix.reset_index()\n",
    "\n",
    "# Reorder columns for the final matrix\n",
    "noncoding_conserved_matrix_ordered = noncoding_conserved_matrix[[\"orfID\", \"macaca\", \"mouse\", \"opossum\", \"platypus\", \"chicken\"]]\n",
    "\n",
    "# Calculate sum across species columns to identify how many species are present\n",
    "noncoding_conserved_matrix_ordered['sum'] = noncoding_conserved_matrix_ordered.select_dtypes(include='number').sum(axis=1)\n",
    "\n",
    "# Sort by the sum column to order by the number of species present\n",
    "noncoding_conserved_matrix_ordered = noncoding_conserved_matrix_ordered.sort_values(by=\"sum\", ascending=False)\n",
    "print(len(set(noncoding_conserved_matrix_ordered.orfID.values.tolist())))\n",
    "\n",
    "# Display the result\n",
    "noncoding_conserved_matrix_ordered =  noncoding_conserved_matrix_ordered.merge(noncoding_conserved[['gene_name','gene_type','orfID']], on=\"orfID\")\n",
    "noncoding_conserved_matrix_ordered.drop_duplicates(inplace=True)\n",
    "\n",
    "noncoding_conserved_matrix_ordered.to_csv(os.path.join(evoDir,\"Clustering/noncodingHuman_Conserved_matrix.tsv\"), sep=\"\\t\", index=None)\n",
    "print(len(set(noncoding_conserved_matrix_ordered.orfID.values.tolist())))\n",
    "\n",
    "# Display the result\n",
    "noncoding_conserved_matrix_ordered =  noncoding_conserved_matrix_ordered.merge(noncoding_conserved[['gene_name','gene_type','orfID']], on=\"orfID\")\n",
    "noncoding_conserved_matrix_ordered.drop_duplicates(inplace=True)\n",
    "\n",
    "noncoding_conserved_matrix_ordered.to_csv(os.path.join(evoDir,\"Clustering/noncodingHuman_Conserved_matrix.tsv\"), sep=\"\\t\", index=None)\n",
    "print(len(set(noncoding_conserved_matrix_ordered.orfID.values.tolist())))\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "          orfID  presence\n",
      "specie                   \n",
      "chicken      38        38\n",
      "human      3308      3308\n",
      "macaca      111       111\n",
      "mouse        92        92\n",
      "opossum      61        61\n",
      "platypus     51        51\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>orfID</th>\n",
       "      <th>specie</th>\n",
       "      <th>presence</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>211</th>\n",
       "      <td>ENST00000581862.7:18:-|12|3689:297:567|noncodi...</td>\n",
       "      <td>macaca</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>286</th>\n",
       "      <td>ENST00000636308.1:7:+|49|4646:635:1763|noncodi...</td>\n",
       "      <td>macaca</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>359</th>\n",
       "      <td>ENST00000701281.2:7:+|2|1243:33:507|noncoding|ATG</td>\n",
       "      <td>macaca</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>360</th>\n",
       "      <td>ENST00000701281.2:7:+|2|1243:33:507|noncoding|ATG</td>\n",
       "      <td>mouse</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>642</th>\n",
       "      <td>ENST00000683452.1:3:+|19|3753:358:1843|noncodi...</td>\n",
       "      <td>macaca</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26360</th>\n",
       "      <td>ENST00000666621.1:10:-|25|3011:473:665|noncodi...</td>\n",
       "      <td>human</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26361</th>\n",
       "      <td>ENST00000671403.1:2:+|11|4190:235:391|noncodin...</td>\n",
       "      <td>human</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26372</th>\n",
       "      <td>ENST00000686765.2:3:+|2|1688:28:175|noncoding|ATG</td>\n",
       "      <td>human</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26373</th>\n",
       "      <td>ENST00000690653.3:2:+|23|2292:494:677|noncodin...</td>\n",
       "      <td>human</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26374</th>\n",
       "      <td>ENST00000692288.2:4:+|65|3183:1349:1841|noncod...</td>\n",
       "      <td>human</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>3661 rows × 3 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                   orfID  specie  presence\n",
       "211    ENST00000581862.7:18:-|12|3689:297:567|noncodi...  macaca         1\n",
       "286    ENST00000636308.1:7:+|49|4646:635:1763|noncodi...  macaca         1\n",
       "359    ENST00000701281.2:7:+|2|1243:33:507|noncoding|ATG  macaca         1\n",
       "360    ENST00000701281.2:7:+|2|1243:33:507|noncoding|ATG   mouse         1\n",
       "642    ENST00000683452.1:3:+|19|3753:358:1843|noncodi...  macaca         1\n",
       "...                                                  ...     ...       ...\n",
       "26360  ENST00000666621.1:10:-|25|3011:473:665|noncodi...   human         1\n",
       "26361  ENST00000671403.1:2:+|11|4190:235:391|noncodin...   human         1\n",
       "26372  ENST00000686765.2:3:+|2|1688:28:175|noncoding|ATG   human         1\n",
       "26373  ENST00000690653.3:2:+|23|2292:494:677|noncodin...   human         1\n",
       "26374  ENST00000692288.2:4:+|65|3183:1349:1841|noncod...   human         1\n",
       "\n",
       "[3661 rows x 3 columns]"
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "### all noncoding\n",
    "noncoding_conserved_touse = noncoding_conserved\n",
    "noncoding_conserved_touse['specie'] = noncoding_conserved_touse[1].str.split(\"_\", expand=True)[0]\n",
    "# noncoding_NonConserved.drop(\"specie\",axis=1, inplace=True)\n",
    "noncoding_to_matrix_raw = pd.concat([noncoding_conserved, noncoding_NonConserved])\n",
    "noncoding_to_matrix = noncoding_to_matrix_raw[['orfID', 'specie']]\n",
    "noncoding_to_matrix = noncoding_to_matrix.fillna('human').drop_duplicates()\n",
    "noncoding_to_matrix['presence'] = 1\n",
    "print(noncoding_to_matrix.groupby(\"specie\").count())\n",
    "noncoding_to_matrix"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "3308\n",
      "3308\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_2049236/1563675081.py:15: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  noncoding_matrix_ordered['sum'] = noncoding_matrix_ordered.select_dtypes(include='number').sum(axis=1)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>orfID</th>\n",
       "      <th>human</th>\n",
       "      <th>macaca</th>\n",
       "      <th>mouse</th>\n",
       "      <th>opossum</th>\n",
       "      <th>platypus</th>\n",
       "      <th>chicken</th>\n",
       "      <th>sum</th>\n",
       "      <th>gene_name</th>\n",
       "      <th>gene_type</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ENST00000456500.1:2:+|1|1047:1:1048|noncoding|ATG</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>6.0</td>\n",
       "      <td>ENSG00000225402</td>\n",
       "      <td>processed_pseudogene</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>ENST00000429606.1:5:-|1|459:1:412|noncoding|ATG</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>6.0</td>\n",
       "      <td>ENSG00000229738</td>\n",
       "      <td>processed_pseudogene</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>ENST00000397254.2:X:+|26|1590:721:1591|noncodi...</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>6.0</td>\n",
       "      <td>RBM39P1</td>\n",
       "      <td>processed_pseudogene</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>ENST00000439215.1:1:+|10|681:226:682|noncoding...</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>6.0</td>\n",
       "      <td>ATP6V1E1P1</td>\n",
       "      <td>processed_pseudogene</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>ENST00000398050.3:10:-|1|1076:1:538|noncoding|ATG</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>6.0</td>\n",
       "      <td>ALDOAP2</td>\n",
       "      <td>processed_pseudogene</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4000</th>\n",
       "      <td>ENST00000510731.1:5:-|38|1099:564:885|noncodin...</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>ENSG00000249541</td>\n",
       "      <td>processed_pseudogene</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4001</th>\n",
       "      <td>ENST00000510896.1:5:-|42|1379:722:1361|noncodi...</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>HNRNPKP1</td>\n",
       "      <td>processed_pseudogene</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4002</th>\n",
       "      <td>ENST00000510897.7:8:+|162|3536:3164:3212|nonco...</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>LINC00964</td>\n",
       "      <td>lncRNA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4003</th>\n",
       "      <td>ENST00000510897.7:8:+|1|3536:20:77|noncoding|CTG</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>LINC00964</td>\n",
       "      <td>lncRNA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4004</th>\n",
       "      <td>TCONS_00001162:9:-|83|3510:982:1030|noncoding|ATG</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>XLOC_001006</td>\n",
       "      <td>novel</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>3308 rows × 10 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                  orfID  human  macaca  mouse  \\\n",
       "0     ENST00000456500.1:2:+|1|1047:1:1048|noncoding|ATG    1.0     1.0    1.0   \n",
       "7       ENST00000429606.1:5:-|1|459:1:412|noncoding|ATG    1.0     1.0    1.0   \n",
       "14    ENST00000397254.2:X:+|26|1590:721:1591|noncodi...    1.0     1.0    1.0   \n",
       "21    ENST00000439215.1:1:+|10|681:226:682|noncoding...    1.0     1.0    1.0   \n",
       "31    ENST00000398050.3:10:-|1|1076:1:538|noncoding|ATG    1.0     1.0    1.0   \n",
       "...                                                 ...    ...     ...    ...   \n",
       "4000  ENST00000510731.1:5:-|38|1099:564:885|noncodin...    1.0     0.0    0.0   \n",
       "4001  ENST00000510896.1:5:-|42|1379:722:1361|noncodi...    1.0     0.0    0.0   \n",
       "4002  ENST00000510897.7:8:+|162|3536:3164:3212|nonco...    1.0     0.0    0.0   \n",
       "4003   ENST00000510897.7:8:+|1|3536:20:77|noncoding|CTG    1.0     0.0    0.0   \n",
       "4004  TCONS_00001162:9:-|83|3510:982:1030|noncoding|ATG    1.0     0.0    0.0   \n",
       "\n",
       "      opossum  platypus  chicken  sum        gene_name             gene_type  \n",
       "0         1.0       1.0      1.0  6.0  ENSG00000225402  processed_pseudogene  \n",
       "7         1.0       1.0      1.0  6.0  ENSG00000229738  processed_pseudogene  \n",
       "14        1.0       1.0      1.0  6.0          RBM39P1  processed_pseudogene  \n",
       "21        1.0       1.0      1.0  6.0       ATP6V1E1P1  processed_pseudogene  \n",
       "31        1.0       1.0      1.0  6.0          ALDOAP2  processed_pseudogene  \n",
       "...       ...       ...      ...  ...              ...                   ...  \n",
       "4000      0.0       0.0      0.0  1.0  ENSG00000249541  processed_pseudogene  \n",
       "4001      0.0       0.0      0.0  1.0         HNRNPKP1  processed_pseudogene  \n",
       "4002      0.0       0.0      0.0  1.0        LINC00964                lncRNA  \n",
       "4003      0.0       0.0      0.0  1.0        LINC00964                lncRNA  \n",
       "4004      0.0       0.0      0.0  1.0      XLOC_001006                 novel  \n",
       "\n",
       "[3308 rows x 10 columns]"
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Pivoting the data without creating an 'id', and keeping 'orfID' as the unique identifier\n",
    "noncoding_matrix = noncoding_to_matrix.pivot_table(index='orfID', columns='specie', values='presence', aggfunc='max')\n",
    "# Fill NaN values with 0\n",
    "noncoding_matrix = noncoding_matrix.fillna(0)\n",
    "# Reset index so 'orfID' becomes a column instead of an index\n",
    "noncoding_matrix = noncoding_matrix.reset_index()\n",
    "\n",
    "# Rename columns as necessary\n",
    "noncoding_matrix.columns = [\"orfID\", \"chicken\", \"human\", \"macaca\", \"mouse\", \"opossum\", \"platypus\"]\n",
    "print(len(noncoding_matrix))\n",
    "# Reorder columns for the final matrix\n",
    "noncoding_matrix_ordered = noncoding_matrix[[\"orfID\", \"human\",\"macaca\", \"mouse\", \"opossum\", \"platypus\", \"chicken\"]]\n",
    "\n",
    "# # Calculate sum across species columns to identify how many species are present\n",
    "noncoding_matrix_ordered['sum'] = noncoding_matrix_ordered.select_dtypes(include='number').sum(axis=1)\n",
    "\n",
    "# Sort by the sum column to order by the number of species present\n",
    "noncoding_matrix_ordered = noncoding_matrix_ordered.sort_values(by=\"sum\", ascending=False)\n",
    "\n",
    "# Display the result\n",
    "noncoding_matrix_ordered = noncoding_matrix_ordered.merge(noncoding_to_matrix_raw[['gene_name','gene_type','orfID']], on=\"orfID\")\n",
    "noncoding_matrix_ordered.drop_duplicates(inplace=True)\n",
    "\n",
    "noncoding_matrix_ordered.to_csv(os.path.join(evoDir,\"Clustering/noncodingHuman_matrix.tsv\"), sep=\"\\t\", index=None)\n",
    "print(len(set(noncoding_matrix_ordered.orfID.values.tolist())))\n",
    "noncoding_matrix_ordered"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_2049236/3967721755.py:17: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  noncoding_matrix_ordered['sum'] = noncoding_matrix_ordered.select_dtypes(include='number').sum(axis=1)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "3308\n"
     ]
    }
   ],
   "source": [
    "\n",
    "# Pivoting the data without creating an 'id', and keeping 'orfID' as the unique identifier\n",
    "noncoding_matrix = noncoding_to_matrix.pivot_table(index='orfID', columns='specie', values='presence', aggfunc='max')\n",
    "noncoding_matrix\n",
    "# Fill NaN values with 0\n",
    "noncoding_matrix = noncoding_matrix.fillna(0)\n",
    "\n",
    "# Reset index so 'orfID' becomes a column instead of an index\n",
    "noncoding_matrix = noncoding_matrix.reset_index()\n",
    "\n",
    "# Rename columns as necessary\n",
    "noncoding_matrix.columns = [\"orfID\", \"chicken\", \"human\", \"macaca\", \"mouse\", \"opossum\", \"platypus\"]\n",
    "\n",
    "# Reorder columns for the final matrix\n",
    "noncoding_matrix_ordered = noncoding_matrix[[\"orfID\", \"human\",\"macaca\", \"mouse\", \"opossum\", \"platypus\", \"chicken\"]]\n",
    "\n",
    "# Calculate sum across species columns to identify how many species are present\n",
    "noncoding_matrix_ordered['sum'] = noncoding_matrix_ordered.select_dtypes(include='number').sum(axis=1)\n",
    "\n",
    "# Sort by the sum column to order by the number of species present\n",
    "noncoding_matrix_ordered = noncoding_matrix_ordered.sort_values(by=\"sum\", ascending=False)\n",
    "\n",
    "# Display the result\n",
    "noncoding_matrix_ordered =  noncoding_matrix_ordered.merge(noncoding_to_matrix_raw[['gene_name','gene_type','orfID']], on=\"orfID\")\n",
    "noncoding_matrix_ordered.drop_duplicates(inplace=True)\n",
    "\n",
    "noncoding_matrix_ordered.to_csv(os.path.join(evoDir,\"Clustering/noncodingHuman_matrix.tsv\"), sep=\"\\t\", index=None)\n",
    "print(len(set(noncoding_matrix_ordered.orfID.values.tolist())))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### alternative ORFs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Conserved altORFs human ORFs\n",
      "92\n",
      "Not-conserved altORFs human ORFs\n",
      "7516\n"
     ]
    }
   ],
   "source": [
    "# if column 1 starts with a specie, it is conserved\n",
    "altORFs_conserved = altORFs[~altORFs[1].str.startswith(\"ENST\")]\n",
    "altORFs_conserved.to_csv(os.path.join(evoDir,\"Clustering/altORFsHuman_Conserved.tsv\"), header=None, index=None, sep=\"\\t\")\n",
    "print(\"Conserved altORFs human ORFs\")\n",
    "print(len(set(altORFs_conserved[0].values.tolist())))\n",
    "altORFs_conserved = altORFs_conserved[['gene_name','gene_type',0,1]]\n",
    "altORFs_conserved.columns = ['gene_name','gene_type','orfID',1]\n",
    "\n",
    "# if column 1 starts with a human gene, it is not conserved\n",
    "altORFs_NonConserved = altORFs[altORFs[1].str.startswith(\"ENST\")]\n",
    "altORFs_NonConserved.to_csv(os.path.join(evoDir,\"Clustering/altORFsHuman_NonConserved.tsv\"), header=None, index=None, sep=\"\\t\")\n",
    "print(\"Not-conserved altORFs human ORFs\")\n",
    "print(len(set(altORFs_NonConserved[0].values.tolist())))\n",
    "altORFs_NonConserved = altORFs_NonConserved[['gene_name','gene_type',0,1]]\n",
    "altORFs_NonConserved.columns = ['gene_name','gene_type','orfID',1]\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "92\n",
      "92\n",
      "92\n",
      "92\n",
      "92\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_2049236/2466750026.py:8: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  altORFs_conserved_to_matrix['presence'] = 1\n"
     ]
    }
   ],
   "source": [
    "## conserved altORFs\n",
    "altORFs_conserved_to_matrix = altORFs_conserved\n",
    "print(len(set(altORFs_conserved_to_matrix.orfID.values.tolist())))\n",
    "# Assuming noncoding_conserved_to_matrix is already defined\n",
    "altORFs_conserved_to_matrix['specie'] = altORFs_conserved_to_matrix[1].str.split(\"_\", expand=True)[0]\n",
    "altORFs_conserved_to_matrix = altORFs_conserved_to_matrix[['orfID', 'specie']]\n",
    "altORFs_conserved_to_matrix.groupby('specie').count()\n",
    "altORFs_conserved_to_matrix['presence'] = 1\n",
    "## control\n",
    "print(len(set(altORFs_conserved_to_matrix.orfID.values.tolist())))\n",
    "\n",
    "# Pivoting the data without creating an 'id', and keeping 'orfID' as the unique identifier\n",
    "altORFs_conserved_matrix = altORFs_conserved_to_matrix.pivot_table(index='orfID', columns='specie', values='presence', aggfunc='max')\n",
    "\n",
    "# Fill NaN values with 0\n",
    "altORFs_conserved_matrix = altORFs_conserved_matrix.fillna(0)\n",
    "\n",
    "# Reset index so 'orfID' becomes a column instead of an index\n",
    "altORFs_conserved_matrix = altORFs_conserved_matrix.reset_index()\n",
    "\n",
    "# Reorder columns for the final matrix\n",
    "altORFs_conserved_matrix_ordered = altORFs_conserved_matrix[[\"orfID\", \"macaca\", \"mouse\", \"opossum\", \"platypus\", \"chicken\"]]\n",
    "\n",
    "# Calculate sum across species columns to identify how many species are present\n",
    "altORFs_conserved_matrix_ordered['sum'] = altORFs_conserved_matrix_ordered.select_dtypes(include='number').sum(axis=1)\n",
    "\n",
    "# Sort by the sum column to order by the number of species present\n",
    "altORFs_conserved_matrix_ordered = altORFs_conserved_matrix_ordered.sort_values(by=\"sum\", ascending=False)\n",
    "print(len(set(altORFs_conserved_matrix_ordered.orfID.values.tolist())))\n",
    "\n",
    "# Display the result\n",
    "altORFs_conserved_matrix_ordered =  altORFs_conserved_matrix_ordered.merge(altORFs_conserved[['gene_name','gene_type','orfID']], on=\"orfID\")\n",
    "altORFs_conserved_matrix_ordered.drop_duplicates(inplace=True)\n",
    "\n",
    "altORFs_conserved_matrix_ordered.to_csv(os.path.join(evoDir,\"Clustering/altORFsHuman_Conserved_matrix.tsv\"), sep=\"\\t\", index=None)\n",
    "print(len(set(altORFs_conserved_matrix_ordered.orfID.values.tolist())))\n",
    "\n",
    "# Display the result\n",
    "altORFs_conserved_matrix_ordered =  altORFs_conserved_matrix_ordered.merge(altORFs_conserved[['gene_name','gene_type','orfID']], on=\"orfID\")\n",
    "altORFs_conserved_matrix_ordered.drop_duplicates(inplace=True)\n",
    "\n",
    "altORFs_conserved_matrix_ordered.to_csv(os.path.join(evoDir,\"Clustering/altORFsHuman_Conserved_matrix.tsv\"), sep=\"\\t\", index=None)\n",
    "print(len(set(altORFs_conserved_matrix_ordered.orfID.values.tolist())))\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "          orfID  presence\n",
      "specie                   \n",
      "chicken       6         6\n",
      "human      7518      7518\n",
      "macaca       44        44\n",
      "mouse        37        37\n",
      "opossum      20        20\n",
      "platypus      9         9\n"
     ]
    }
   ],
   "source": [
    "### all altORFs\n",
    "altORFs_conserved_touse = altORFs_conserved\n",
    "altORFs_conserved_touse['specie'] = altORFs_conserved_touse[1].str.split(\"_\", expand=True)[0]\n",
    "altORFs_to_matrix_raw = pd.concat([altORFs_conserved, altORFs_NonConserved])\n",
    "altORFs_to_matrix = altORFs_to_matrix_raw[['orfID', 'specie']]\n",
    "altORFs_to_matrix = altORFs_to_matrix.fillna('human').drop_duplicates()\n",
    "altORFs_to_matrix['presence'] = 1\n",
    "altORFs_to_matrix = altORFs_to_matrix.replace(\"TCONS\", \"human\")\n",
    "print(altORFs_to_matrix.groupby(\"specie\").count())\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "7516\n",
      "7516\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_2049236/70459802.py:15: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  altORFs_matrix_ordered['sum'] = altORFs_matrix_ordered.select_dtypes(include='number').sum(axis=1)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>orfID</th>\n",
       "      <th>human</th>\n",
       "      <th>macaca</th>\n",
       "      <th>mouse</th>\n",
       "      <th>opossum</th>\n",
       "      <th>platypus</th>\n",
       "      <th>chicken</th>\n",
       "      <th>sum</th>\n",
       "      <th>gene_name</th>\n",
       "      <th>gene_type</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ENST00000532219.5:5:+|456|8246:7706:7940|odORF...</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>6.0</td>\n",
       "      <td>ANKHD1-EIF4EBP3</td>\n",
       "      <td>protein_coding</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>ENST00000618666.4:12:-|46|6915:1015:2341|dORF|ATG</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>6.0</td>\n",
       "      <td>CCNT1</td>\n",
       "      <td>protein_coding</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>ENST00000695752.1:1:+|41|1548:541:1222|odORF|ATG</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>C1QC</td>\n",
       "      <td>protein_coding</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>ENST00000628377.2:3:-|53|2214:875:2054|dORF|ATG</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>HACL1</td>\n",
       "      <td>protein_coding</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>ENST00000643349.2:11:-|42|2861:727:1270|dORF|ATG</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>ENSG00000284779</td>\n",
       "      <td>protein_coding</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7822</th>\n",
       "      <td>ENST00000339697.5:6:-|159|3125:2425:2530|dORF|CTG</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>SERINC1</td>\n",
       "      <td>protein_coding</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7823</th>\n",
       "      <td>ENST00000339656.8:19:+|3|6885:22:76|uORF|ATG</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>ZNF587</td>\n",
       "      <td>protein_coding</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7824</th>\n",
       "      <td>ENST00000339656.8:19:+|113|6885:1827:2277|odOR...</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>ZNF587</td>\n",
       "      <td>protein_coding</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7825</th>\n",
       "      <td>ENST00000339618.8:17:+|109|3828:1745:1805|odOR...</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>ALDH3A2</td>\n",
       "      <td>protein_coding</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7826</th>\n",
       "      <td>ENST00000850623.1:11:-|3|8048:24:234|ouORF|ATG</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NPAT</td>\n",
       "      <td>protein_coding</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>7516 rows × 10 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                  orfID  human  macaca  mouse  \\\n",
       "0     ENST00000532219.5:5:+|456|8246:7706:7940|odORF...    1.0     1.0    1.0   \n",
       "13    ENST00000618666.4:12:-|46|6915:1015:2341|dORF|ATG    1.0     1.0    1.0   \n",
       "19     ENST00000695752.1:1:+|41|1548:541:1222|odORF|ATG    1.0     1.0    1.0   \n",
       "24      ENST00000628377.2:3:-|53|2214:875:2054|dORF|ATG    1.0     0.0    1.0   \n",
       "29     ENST00000643349.2:11:-|42|2861:727:1270|dORF|ATG    1.0     1.0    1.0   \n",
       "...                                                 ...    ...     ...    ...   \n",
       "7822  ENST00000339697.5:6:-|159|3125:2425:2530|dORF|CTG    1.0     0.0    0.0   \n",
       "7823       ENST00000339656.8:19:+|3|6885:22:76|uORF|ATG    1.0     0.0    0.0   \n",
       "7824  ENST00000339656.8:19:+|113|6885:1827:2277|odOR...    1.0     0.0    0.0   \n",
       "7825  ENST00000339618.8:17:+|109|3828:1745:1805|odOR...    1.0     0.0    0.0   \n",
       "7826     ENST00000850623.1:11:-|3|8048:24:234|ouORF|ATG    1.0     0.0    0.0   \n",
       "\n",
       "      opossum  platypus  chicken  sum        gene_name       gene_type  \n",
       "0         1.0       1.0      1.0  6.0  ANKHD1-EIF4EBP3  protein_coding  \n",
       "13        1.0       1.0      1.0  6.0            CCNT1  protein_coding  \n",
       "19        1.0       0.0      1.0  5.0             C1QC  protein_coding  \n",
       "24        1.0       1.0      1.0  5.0            HACL1  protein_coding  \n",
       "29        0.0       1.0      1.0  5.0  ENSG00000284779  protein_coding  \n",
       "...       ...       ...      ...  ...              ...             ...  \n",
       "7822      0.0       0.0      0.0  1.0          SERINC1  protein_coding  \n",
       "7823      0.0       0.0      0.0  1.0           ZNF587  protein_coding  \n",
       "7824      0.0       0.0      0.0  1.0           ZNF587  protein_coding  \n",
       "7825      0.0       0.0      0.0  1.0          ALDH3A2  protein_coding  \n",
       "7826      0.0       0.0      0.0  1.0             NPAT  protein_coding  \n",
       "\n",
       "[7516 rows x 10 columns]"
      ]
     },
     "execution_count": 23,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Pivoting the data without creating an 'id', and keeping 'orfID' as the unique identifier\n",
    "altORFs_matrix = altORFs_to_matrix.pivot_table(index='orfID', columns='specie', values='presence', aggfunc='max')\n",
    "# Fill NaN values with 0\n",
    "altORFs_matrix = altORFs_matrix.fillna(0)\n",
    "# Reset index so 'orfID' becomes a column instead of an index\n",
    "altORFs_matrix = altORFs_matrix.reset_index()\n",
    "altORFs_matrix\n",
    "# Rename columns as necessary\n",
    "altORFs_matrix.columns = [\"orfID\", \"chicken\", \"human\", \"macaca\", \"mouse\", \"opossum\", \"platypus\"]\n",
    "print(len(altORFs_matrix))\n",
    "# Reorder columns for the final matrix\n",
    "altORFs_matrix_ordered = altORFs_matrix[[\"orfID\", \"human\",\"macaca\", \"mouse\", \"opossum\", \"platypus\", \"chicken\"]]\n",
    "\n",
    "# # Calculate sum across species columns to identify how many species are present\n",
    "altORFs_matrix_ordered['sum'] = altORFs_matrix_ordered.select_dtypes(include='number').sum(axis=1)\n",
    "\n",
    "# Sort by the sum column to order by the number of species present\n",
    "altORFs_matrix_ordered = altORFs_matrix_ordered.sort_values(by=\"sum\", ascending=False)\n",
    "\n",
    "# Display the result\n",
    "altORFs_matrix_ordered = altORFs_matrix_ordered.merge(altORFs_to_matrix_raw[['gene_name','gene_type','orfID']], on=\"orfID\")\n",
    "altORFs_matrix_ordered.drop_duplicates(inplace=True)\n",
    "\n",
    "altORFs_matrix_ordered.to_csv(os.path.join(evoDir,\"Clustering/altORFsHuman_matrix.tsv\"), sep=\"\\t\", index=None)\n",
    "print(len(set(altORFs_matrix_ordered.orfID.values.tolist())))\n",
    "altORFs_matrix_ordered"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "\n",
    "### CANONICAL"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Conserved canonical human ORFs\n",
      "2594\n",
      "AltORFs noncoding human ORFs\n",
      "2909\n"
     ]
    }
   ],
   "source": [
    "# if column 1 starts with a specie, it is conserved\n",
    "canonical_conserved = canonical[~canonical[1].str.startswith(\"ENST\")]\n",
    "canonical_conserved.to_csv(os.path.join(evoDir,\"Clustering/canonicalHuman_Conserved.tsv\"), header=None, index=None, sep=\"\\t\")\n",
    "print(\"Conserved canonical human ORFs\")\n",
    "print(len(set(canonical_conserved[0].values.tolist())))\n",
    "canonical_conserved = canonical_conserved[['gene_name','gene_type',0,1]]\n",
    "canonical_conserved.columns = ['gene_name','gene_type','orfID',1]\n",
    "\n",
    "# if column 1 starts with a human gene, it is not conserved\n",
    "canonical_NonConserved = canonical[canonical[1].str.startswith(\"ENST\")]\n",
    "canonical_NonConserved.to_csv(os.path.join(evoDir,\"Clustering/canonicalHuman_NonConserved.tsv\"), header=None, index=None, sep=\"\\t\")\n",
    "print(\"AltORFs noncoding human ORFs\")\n",
    "print(len(set(canonical_NonConserved[0].values.tolist())))\n",
    "canonical_NonConserved = canonical_NonConserved[['gene_name','gene_type',0,1]]\n",
    "canonical_NonConserved.columns = ['gene_name','gene_type','orfID',1]\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2594\n",
      "2594\n",
      "2594\n",
      "2594\n",
      "2594\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_2049236/3948074870.py:8: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  canonical_conserved_to_matrix['presence'] = 1\n",
      "/tmp/ipykernel_2049236/3948074870.py:25: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  canonical_conserved_matrix_ordered['sum'] = canonical_conserved_matrix_ordered.select_dtypes(include='number').sum(axis=1)\n"
     ]
    }
   ],
   "source": [
    "## conserved canonical\n",
    "canonical_conserved_to_matrix = canonical_conserved\n",
    "print(len(set(canonical_conserved_to_matrix.orfID.values.tolist())))\n",
    "# Assuming noncoding_conserved_to_matrix is already defined\n",
    "canonical_conserved_to_matrix['specie'] = canonical_conserved_to_matrix[1].str.split(\"_\", expand=True)[0]\n",
    "canonical_conserved_to_matrix = canonical_conserved_to_matrix[['orfID', 'specie']]\n",
    "canonical_conserved_to_matrix.groupby('specie').count()\n",
    "canonical_conserved_to_matrix['presence'] = 1\n",
    "## control\n",
    "print(len(set(canonical_conserved_to_matrix.orfID.values.tolist())))\n",
    "\n",
    "# Pivoting the data without creating an 'id', and keeping 'orfID' as the unique identifier\n",
    "canonical_conserved_matrix = canonical_conserved_to_matrix.pivot_table(index='orfID', columns='specie', values='presence', aggfunc='max')\n",
    "\n",
    "# Fill NaN values with 0\n",
    "canonical_conserved_matrix = canonical_conserved_matrix.fillna(0)\n",
    "\n",
    "# Reset index so 'orfID' becomes a column instead of an index\n",
    "canonical_conserved_matrix = canonical_conserved_matrix.reset_index()\n",
    "\n",
    "# Reorder columns for the final matrix\n",
    "canonical_conserved_matrix_ordered = canonical_conserved_matrix[[\"orfID\", \"macaca\", \"mouse\", \"opossum\", \"platypus\", \"chicken\"]]\n",
    "\n",
    "# Calculate sum across species columns to identify how many species are present\n",
    "canonical_conserved_matrix_ordered['sum'] = canonical_conserved_matrix_ordered.select_dtypes(include='number').sum(axis=1)\n",
    "\n",
    "# Sort by the sum column to order by the number of species present\n",
    "canonical_conserved_matrix_ordered = canonical_conserved_matrix_ordered.sort_values(by=\"sum\", ascending=False)\n",
    "print(len(set(canonical_conserved_matrix_ordered.orfID.values.tolist())))\n",
    "\n",
    "# Display the result\n",
    "canonical_conserved_matrix_ordered =  canonical_conserved_matrix_ordered.merge(canonical_conserved[['gene_name','gene_type','orfID']], on=\"orfID\")\n",
    "canonical_conserved_matrix_ordered.drop_duplicates(inplace=True)\n",
    "\n",
    "canonical_conserved_matrix_ordered.to_csv(os.path.join(evoDir,\"Clustering/canonicalHuman_Conserved_matrix.tsv\"), sep=\"\\t\", index=None)\n",
    "print(len(set(canonical_conserved_matrix_ordered.orfID.values.tolist())))\n",
    "\n",
    "# Display the result\n",
    "canonical_conserved_matrix_ordered =  canonical_conserved_matrix_ordered.merge(canonical_conserved[['gene_name','gene_type','orfID']], on=\"orfID\")\n",
    "canonical_conserved_matrix_ordered.drop_duplicates(inplace=True)\n",
    "\n",
    "canonical_conserved_matrix_ordered.to_csv(os.path.join(evoDir,\"Clustering/canonicalHuman_Conserved_matrix.tsv\"), sep=\"\\t\", index=None)\n",
    "print(len(set(canonical_conserved_matrix_ordered.orfID.values.tolist())))\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "          orfID  presence\n",
      "specie                   \n",
      "chicken    1466      1466\n",
      "human      2909      2909\n",
      "macaca     2320      2320\n",
      "mouse      1989      1989\n",
      "opossum    1629      1629\n",
      "platypus   1465      1465\n"
     ]
    }
   ],
   "source": [
    "### all canonical\n",
    "canonical_conserved_touse = canonical_conserved\n",
    "canonical_conserved_touse['specie'] = canonical_conserved_touse[1].str.split(\"_\", expand=True)[0]\n",
    "canonical_to_matrix_raw = pd.concat([canonical_conserved, canonical_NonConserved])\n",
    "canonical_to_matrix = canonical_to_matrix_raw[['orfID', 'specie']]\n",
    "canonical_to_matrix = canonical_to_matrix.fillna('human').drop_duplicates()\n",
    "canonical_to_matrix['presence'] = 1\n",
    "canonical_to_matrix = canonical_to_matrix.replace(\"TCONS\", \"human\")\n",
    "print(canonical_to_matrix.groupby(\"specie\").count())\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2909\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_2049236/1133186206.py:15: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  canonical_matrix_ordered['sum'] = canonical_matrix_ordered.select_dtypes(include='number').sum(axis=1)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2909\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>orfID</th>\n",
       "      <th>human</th>\n",
       "      <th>macaca</th>\n",
       "      <th>mouse</th>\n",
       "      <th>opossum</th>\n",
       "      <th>platypus</th>\n",
       "      <th>chicken</th>\n",
       "      <th>sum</th>\n",
       "      <th>gene_name</th>\n",
       "      <th>gene_type</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ENST00000850822.1:Y:-|1|3290:12:1563|canonical...</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>6.0</td>\n",
       "      <td>GTPBP6</td>\n",
       "      <td>protein_coding</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>ENST00000398805.8:9:+|4|3174:57:1314|canonical...</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>6.0</td>\n",
       "      <td>HSDL2</td>\n",
       "      <td>protein_coding</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>ENST00000400590.8:13:-|5|5355:133:1231|canonic...</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>6.0</td>\n",
       "      <td>ZDHHC20</td>\n",
       "      <td>protein_coding</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>ENST00000301328.9:17:-|3|1809:25:967|canonical...</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>6.0</td>\n",
       "      <td>GLOD4</td>\n",
       "      <td>protein_coding</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>35</th>\n",
       "      <td>ENST00000400405.4:13:-|6|7075:125:935|canonica...</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>6.0</td>\n",
       "      <td>SIAH3</td>\n",
       "      <td>protein_coding</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14535</th>\n",
       "      <td>ENST00000557436.2:14:+|10|2956:277:589|canonic...</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>PTGER2</td>\n",
       "      <td>protein_coding</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14536</th>\n",
       "      <td>ENST00000553909.1:14:+|9|1498:191:287|canonica...</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>ENSG00000259171</td>\n",
       "      <td>protein_coding</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14537</th>\n",
       "      <td>ENST00000370629.7:X:+|8|1852:92:878|canonical|ATG</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>CD40LG</td>\n",
       "      <td>protein_coding</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14538</th>\n",
       "      <td>ENST00000552878.5:12:-|7|1052:84:633|canonical...</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>FKBP11</td>\n",
       "      <td>protein_coding</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14539</th>\n",
       "      <td>ENST00000370518.4:X:+|5|408:61:355|canonical|ATG</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>SPANXA2</td>\n",
       "      <td>protein_coding</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2909 rows × 10 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                   orfID  human  macaca  \\\n",
       "0      ENST00000850822.1:Y:-|1|3290:12:1563|canonical...    1.0     1.0   \n",
       "6      ENST00000398805.8:9:+|4|3174:57:1314|canonical...    1.0     1.0   \n",
       "12     ENST00000400590.8:13:-|5|5355:133:1231|canonic...    1.0     1.0   \n",
       "29     ENST00000301328.9:17:-|3|1809:25:967|canonical...    1.0     1.0   \n",
       "35     ENST00000400405.4:13:-|6|7075:125:935|canonica...    1.0     1.0   \n",
       "...                                                  ...    ...     ...   \n",
       "14535  ENST00000557436.2:14:+|10|2956:277:589|canonic...    1.0     0.0   \n",
       "14536  ENST00000553909.1:14:+|9|1498:191:287|canonica...    1.0     0.0   \n",
       "14537  ENST00000370629.7:X:+|8|1852:92:878|canonical|ATG    1.0     0.0   \n",
       "14538  ENST00000552878.5:12:-|7|1052:84:633|canonical...    1.0     0.0   \n",
       "14539   ENST00000370518.4:X:+|5|408:61:355|canonical|ATG    1.0     0.0   \n",
       "\n",
       "       mouse  opossum  platypus  chicken  sum        gene_name       gene_type  \n",
       "0        1.0      1.0       1.0      1.0  6.0           GTPBP6  protein_coding  \n",
       "6        1.0      1.0       1.0      1.0  6.0            HSDL2  protein_coding  \n",
       "12       1.0      1.0       1.0      1.0  6.0          ZDHHC20  protein_coding  \n",
       "29       1.0      1.0       1.0      1.0  6.0            GLOD4  protein_coding  \n",
       "35       1.0      1.0       1.0      1.0  6.0            SIAH3  protein_coding  \n",
       "...      ...      ...       ...      ...  ...              ...             ...  \n",
       "14535    0.0      0.0       0.0      0.0  1.0           PTGER2  protein_coding  \n",
       "14536    0.0      0.0       0.0      0.0  1.0  ENSG00000259171  protein_coding  \n",
       "14537    0.0      0.0       0.0      0.0  1.0           CD40LG  protein_coding  \n",
       "14538    0.0      0.0       0.0      0.0  1.0           FKBP11  protein_coding  \n",
       "14539    0.0      0.0       0.0      0.0  1.0          SPANXA2  protein_coding  \n",
       "\n",
       "[2909 rows x 10 columns]"
      ]
     },
     "execution_count": 27,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Pivoting the data without creating an 'id', and keeping 'orfID' as the unique identifier\n",
    "canonical_matrix = canonical_to_matrix.pivot_table(index='orfID', columns='specie', values='presence', aggfunc='max')\n",
    "# Fill NaN values with 0\n",
    "canonical_matrix = canonical_matrix.fillna(0)\n",
    "# Reset index so 'orfID' becomes a column instead of an index\n",
    "canonical_matrix = canonical_matrix.reset_index()\n",
    "canonical_matrix\n",
    "# Rename columns as necessary\n",
    "canonical_matrix.columns = [\"orfID\", \"chicken\", \"human\", \"macaca\", \"mouse\", \"opossum\", \"platypus\"]\n",
    "print(len(canonical_matrix))\n",
    "# Reorder columns for the final matrix\n",
    "canonical_matrix_ordered = canonical_matrix[[\"orfID\", \"human\",\"macaca\", \"mouse\", \"opossum\", \"platypus\", \"chicken\"]]\n",
    "\n",
    "# # Calculate sum across species columns to identify how many species are present\n",
    "canonical_matrix_ordered['sum'] = canonical_matrix_ordered.select_dtypes(include='number').sum(axis=1)\n",
    "\n",
    "# Sort by the sum column to order by the number of species present\n",
    "canonical_matrix_ordered = canonical_matrix_ordered.sort_values(by=\"sum\", ascending=False)\n",
    "\n",
    "# Display the result\n",
    "canonical_matrix_ordered = canonical_matrix_ordered.merge(canonical_to_matrix_raw[['gene_name','gene_type','orfID']], on=\"orfID\")\n",
    "canonical_matrix_ordered.drop_duplicates(inplace=True)\n",
    "\n",
    "canonical_matrix_ordered.to_csv(os.path.join(evoDir,\"Clustering/canonicalHuman_matrix.tsv\"), sep=\"\\t\", index=None)\n",
    "print(len(set(canonical_matrix_ordered.orfID.values.tolist())))\n",
    "canonical_matrix_ordered"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Find the oldest specie containg the ORF translated"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {},
   "outputs": [],
   "source": [
    "species_ordered = ['chicken','platypus','opossum','mouse','macaca','human']\n",
    "def find_last_species(row):\n",
    "    for sp in species_ordered:\n",
    "        if row[sp] == 1.0:  # Check if the value is 1.0\n",
    "            return sp\n",
    "    return None  # Return None if no species has a 1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Apply the function to each row to create a new column 'oldest_species'\n",
    "canonical_matrix_ordered['oldest_species'] = canonical_matrix_ordered.apply(find_last_species, axis=1)\n",
    "canonical_matrix_ordered = species_and_years.merge(canonical_matrix_ordered, left_on=[\"specie\"], right_on=\"oldest_species\")\n",
    "canonical_matrix_ordered[['orfID','gene_name','gene_type','oldest_species','age']].to_csv(os.path.join(evoDir,\"Clustering/canonicalHuman_matrix_oldestAge.tsv\"), sep=\"\\t\", index=None)\n",
    "\n",
    "noncoding_matrix_ordered['oldest_species'] = noncoding_matrix_ordered.apply(find_last_species, axis=1)\n",
    "noncoding_matrix_ordered = species_and_years.merge(noncoding_matrix_ordered, left_on=[\"specie\"], right_on=\"oldest_species\")\n",
    "noncoding_matrix_ordered[['orfID','gene_name','gene_type','oldest_species','age']].to_csv(os.path.join(evoDir,\"Clustering/noncodingHuman_matrix_oldestAge.tsv\"), sep=\"\\t\", index=None)\n",
    "\n",
    "\n",
    "altORFs_matrix_ordered['oldest_species'] = altORFs_matrix_ordered.apply(find_last_species, axis=1)\n",
    "altORFs_matrix_ordered = species_and_years.merge(altORFs_matrix_ordered, left_on=[\"specie\"], right_on=\"oldest_species\")\n",
    "altORFs_matrix_ordered[['orfID','gene_name','gene_type','oldest_species','age']].to_csv(os.path.join(evoDir,\"Clustering/altORFsHuman_matrix_oldestAge.tsv\"), sep=\"\\t\", index=None)\n",
    "\n",
    "\n",
    "complete_matrix_age = pd.concat([noncoding_matrix_ordered[['orfID','gene_name','gene_type','oldest_species','age']],altORFs_matrix_ordered[['orfID','gene_name','gene_type','oldest_species','age']], canonical_matrix_ordered[['orfID','gene_name','gene_type','oldest_species','age']]])\n",
    "complete_matrix_age.to_csv(os.path.join(evoDir,\"Clustering/allHuman_matrix_oldestAge.tsv\"), sep=\"\\t\", index=None)\n"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.10"
  },
  "orig_nbformat": 4
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
