{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Evolutionary origin\n",
    "\n",
    "Macaca - Mouse - Human"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>specie</th>\n",
       "      <th>age</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>human</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>macaca</td>\n",
       "      <td>25</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>mouse</td>\n",
       "      <td>90</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>opossum</td>\n",
       "      <td>180</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>platypus</td>\n",
       "      <td>200</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>chicken</td>\n",
       "      <td>310</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "     specie  age\n",
       "1     human    0\n",
       "2    macaca   25\n",
       "3     mouse   90\n",
       "4   opossum  180\n",
       "5  platypus  200\n",
       "6   chicken  310"
      ]
     },
     "execution_count": 1,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import os,re,glob\n",
    "import pandas as pd\n",
    "import numpy as np\n",
    "from collections import Counter\n",
    "from Bio.Seq import Seq\n",
    "from Bio.SeqRecord import SeqRecord\n",
    "from Bio import SeqIO\n",
    "\n",
    "species_dir = \"/users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction\"\n",
    "users_dir = \"/users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47\"\n",
    "\n",
    "GENOMEDIR = \"/genomics/users/marta/genomes\"\n",
    "species=[\"mouse\",\"macaca\",\"platypus\",\"chicken\",\"opossum\"]\n",
    "\n",
    "## annotation file\n",
    "annotation=\"/users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/human/newReference_Resconstructed/gencode.v47.gffcompare.TestisLiverBrain.annotation.sorted.1transcript.sorted.NOchr.gtf\"\n",
    "transcript_gene=pd.read_csv(\"/users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/human/newReference_Resconstructed/transID_geneID_isoforms_selected.1to1.csv\")\n",
    "\n",
    "## evoDir\n",
    "evoDir=os.path.join(users_dir,\"EvolutionaryOrigin_MMseqs\")\n",
    "\n",
    "def translate_dna_to_protein(dna_seq):\n",
    "    return str(Seq(dna_seq).translate())\n",
    "\n",
    "# Function to save DF in fasta format\n",
    "def create_seqrecord_PROT(row):\n",
    "    return SeqRecord(Seq(row['protein']), id=row['header'][:-4], description=\"\")\n",
    "def create_seqrecord_DNA(row):\n",
    "    return SeqRecord(Seq(row['seq']), id=row['header'], description=\"\")\n",
    "\n",
    "species_and_years_dict = {\"specie\" : [\"human\",\"macaca\",\"mouse\",\"opossum\",\"platypus\",\"chicken\"],\n",
    "                          \"age\" : [0,25,90,180,200,310]}\n",
    "species_and_years = pd.DataFrame(data=species_and_years_dict, index=[1,2,3,4,5,6])\n",
    "species_and_years\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "## get repre sequences - both DNA and PROTEINS\n",
    "for s in species:\n",
    "    print(s)\n",
    "    riboseq_dir = os.path.join(users_dir,s,\"RiboSeq/RiboQC_RiboNovel\")\n",
    "\n",
    "    ## get all candidates\n",
    "    print(\"Reading all the candidate ORF sequences\")\n",
    "    df = pd.read_csv(os.path.join(riboseq_dir,\"Annotation/candidateORF.fa\"), header=None, sep=\"\\t\")\n",
    "    ## convert to df\n",
    "    candidates_fasta = pd.DataFrame({'header':df[0].iloc[::2].values, 'seq':df[0].iloc[1::2].values})\n",
    "    candidates_fasta['header'] = candidates_fasta['header'].str[1:]\n",
    "    candidates_fasta = candidates_fasta[~candidates_fasta['header'].str.contains(\"iORF\")]\n",
    "\n",
    "    all_repre_candidates = pd.DataFrame()\n",
    "    for sample in os.listdir(os.path.join(riboseq_dir,\"RibORF\")):\n",
    "        if sample.endswith(\"r1\"):\n",
    "            print(sample)\n",
    "\n",
    "            repre = pd.read_csv(os.path.join(riboseq_dir,\"RibORF\",sample,\"repre.valid.pred.pvalue.parameters.txt\"), sep=\"\\t\")\n",
    "            repre_fasta = candidates_fasta[candidates_fasta['header'].isin(repre.orfID.values.tolist())]\n",
    "            repre_fasta['header'] = s + \"_\" +repre_fasta['header']\n",
    "\n",
    "            ## keep only the coding and the non-coding\n",
    "            repre_fasta = repre_fasta[repre_fasta['header'].str.contains(\"canonical\")| repre_fasta['header'].str.contains(\"noncoding\")]\n",
    "\n",
    "            repre_fasta['protein'] = repre_fasta['seq'].apply(translate_dna_to_protein)\n",
    "            all_repre_candidates = pd.concat([all_repre_candidates, repre_fasta])\n",
    "\n",
    "            ## translate to protein\n",
    "            print(\"Translating to protein\")\n",
    "            out_proteins = os.path.join(riboseq_dir,\"RibORF\",sample,\"repre.valid.pred.pvalue.parameters.PROTEIN.fa\")\n",
    "            seq_records_proteins = repre_fasta.apply(create_seqrecord_PROT, axis=1).tolist()\n",
    "            SeqIO.write(seq_records_proteins, out_proteins, \"fasta\")\n",
    "\n",
    "            ## keep the dna sequences as well\n",
    "            print(\"Saving the original DNA ORF\")\n",
    "            out_dna = os.path.join(riboseq_dir,\"RibORF\",sample,\"repre.valid.pred.pvalue.parameters.DNA.fa\")\n",
    "            seq_records_dna = repre_fasta.apply(create_seqrecord_DNA, axis=1).tolist()\n",
    "            SeqIO.write(seq_records_dna, out_dna, \"fasta\")\n",
    "\n",
    "            ## get only the nonredundant\n",
    "            print(\"Getting the non-redundant\")\n",
    "            all_repre_candidates.drop_duplicates(inplace=True)\n",
    "            out_nonredundant_proteins = os.path.join(riboseq_dir,\"RibORF\")+\"/\"+s+\"_repre.valid.pred.pvalue.parameters.PROTEIN.noredunant.fa\"\n",
    "            seq_records_proteins = all_repre_candidates.apply(create_seqrecord_PROT, axis=1).tolist()\n",
    "            SeqIO.write(seq_records_proteins, out_nonredundant_proteins, \"fasta\")"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## MMseqs2\n",
    "\n",
    "Against all expressed in mouse, macaca, platypus, opossum in any tissue (non-redunant set)"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### TESTIS-EXPRESSED"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "27106"
      ]
     },
     "execution_count": 2,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "## Testis-expressed ORFs human\n",
    "TestisExpressed_translatedINtestis = pd.read_csv(\"/projects_eg/projects/marta/TestisRestricted_Microproteins_TSA/with_TranscriptomeReconstruction/Multimap_altORFs/Q1_TestisORFs/human/ribORF_humanTestis_in1.csv\")\n",
    "TestisExpressed_translatedINtestis\n",
    "\n",
    "def create_seqrecord_PROThuman(row):\n",
    "    return SeqRecord(Seq(row['ORFpep']), id=row['orfID'], description=\"\")\n",
    "\n",
    "out_nonredundant_proteins = \"/projects_eg/projects/marta/TestisRestricted_Microproteins_TSA/with_TranscriptomeReconstruction/Multimap_altORFs/Q1_TestisORFs/human/ribORF_humanTestis_in1.fa\"\n",
    "seq_records_proteins = TestisExpressed_translatedINtestis.apply(create_seqrecord_PROThuman, axis=1).tolist()\n",
    "SeqIO.write(seq_records_proteins, out_nonredundant_proteins, \"fasta\")"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Creating database of human (expressed in testis)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/MMseqsDB/ribORF_humanTestis_in1_DB exists and will be overwritten\n",
      "createdb /projects_eg/projects/marta/TestisRestricted_Microproteins_TSA/with_TranscriptomeReconstruction/Multimap_altORFs/Q1_TestisORFs/human/ribORF_humanTestis_in1.fa /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/MMseqsDB/ribORF_humanTestis_in1_DB \n",
      "\n",
      "MMseqs Version:       \t14.7e284\n",
      "Database type         \t0\n",
      "Shuffle input database\ttrue\n",
      "Createdb mode         \t0\n",
      "Write lookup file     \t1\n",
      "Offset of numeric ids \t0\n",
      "Compressed            \t0\n",
      "Verbosity             \t3\n",
      "\n",
      "Converting sequences\n",
      "[==\n",
      "Time for merging to ribORF_humanTestis_in1_DB_h: 0h 0m 3s 213ms\n",
      "Time for merging to ribORF_humanTestis_in1_DB: 0h 0m 10s 476ms\n",
      "Database type: Aminoacid\n",
      "Time for processing: 0h 0m 58s 317ms\n"
     ]
    }
   ],
   "source": [
    "%%bash -s \"$evoDir\" \"$users_dir\"\n",
    "\n",
    "# mkdir $1/MMseqsDB\n",
    "\n",
    "module load Miniconda3/4.9.2\n",
    "source /soft/system/software/Miniconda3/4.9.2/bin/activate MMseqs2\n",
    "\n",
    "mkdir -p $1/MMseqsDB\n",
    "\n",
    "mmseqs createdb /projects_eg/projects/marta/TestisRestricted_Microproteins_TSA/with_TranscriptomeReconstruction/Multimap_altORFs/Q1_TestisORFs/human/ribORF_humanTestis_in1.fa $1/MMseqsDB/ribORF_humanTestis_in1_DB\n"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Creating a joint database with all the specie, everything translated, no matter the tissue"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/MMseqsDB/speciesDB exists and will be overwritten\n",
      "createdb /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/chicken/RiboSeq/RiboQC_RiboNovel/RibORF/chicken_repre.valid.pred.pvalue.parameters.PROTEIN.noredunant.fa /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/macaca/RiboSeq/RiboQC_RiboNovel/RibORF/macaca_repre.valid.pred.pvalue.parameters.PROTEIN.noredunant.fa /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/mouse/RiboSeq/RiboQC_RiboNovel/RibORF/mouse_repre.valid.pred.pvalue.parameters.PROTEIN.noredunant.fa /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/opossum/RiboSeq/RiboQC_RiboNovel/RibORF/opossum_repre.valid.pred.pvalue.parameters.PROTEIN.noredunant.fa /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/platypus/RiboSeq/RiboQC_RiboNovel/RibORF/platypus_repre.valid.pred.pvalue.parameters.PROTEIN.noredunant.fa /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/MMseqsDB/speciesDB \n",
      "\n",
      "MMseqs Version:       \t14.7e284\n",
      "Database type         \t0\n",
      "Shuffle input database\ttrue\n",
      "Createdb mode         \t0\n",
      "Write lookup file     \t1\n",
      "Offset of numeric ids \t0\n",
      "Compressed            \t0\n",
      "Verbosity             \t3\n",
      "\n",
      "Converting sequences\n",
      "[===========\n",
      "Time for merging to speciesDB_h: 0h 0m 3s 657ms\n",
      "Time for merging to speciesDB: 0h 0m 3s 350ms\n",
      "Database type: Aminoacid\n",
      "Time for processing: 0h 1m 2s 511ms\n"
     ]
    }
   ],
   "source": [
    "%%bash -s \"$evoDir\" \"$species_dir\"\n",
    "\n",
    "module load Miniconda3/4.9.2\n",
    "source /soft/system/software/Miniconda3/4.9.2/bin/activate MMseqs2\n",
    "\n",
    "# species=\"platypus mouse macaca chicken\"\n",
    "# for s in $species; do\n",
    "#     echo $s\n",
    "\n",
    "#     ### create DB\n",
    "#     mmseqs createdb $2/$s/RiboSeq/RiboQC_RiboNovel/RibORF/repre.valid.pred.pvalue.parameters.PROTEIN.noredunant.fa $1/MMseqsDB/${s}_noredunantProts_DB\n",
    "\n",
    "# done\n",
    "\n",
    "mmseqs createdb $2/*/RiboSeq/RiboQC_RiboNovel*/RibORF/*_repre.valid.pred.pvalue.parameters.PROTEIN.noredunant.fa $1/MMseqsDB/speciesDB\n",
    "\n"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Creating a joint database with all the specie AND human for clustering, everything translated for the species, no matter the tissue, but only translated in 1 for testis"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/MMseqsDB/species_and_humanDB exists and will be overwritten\n",
      "createdb /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/chicken/RiboSeq/RiboQC_RiboNovel/RibORF/chicken_repre.valid.pred.pvalue.parameters.PROTEIN.noredunant.fa /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/macaca/RiboSeq/RiboQC_RiboNovel/RibORF/macaca_repre.valid.pred.pvalue.parameters.PROTEIN.noredunant.fa /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/mouse/RiboSeq/RiboQC_RiboNovel/RibORF/mouse_repre.valid.pred.pvalue.parameters.PROTEIN.noredunant.fa /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/opossum/RiboSeq/RiboQC_RiboNovel/RibORF/opossum_repre.valid.pred.pvalue.parameters.PROTEIN.noredunant.fa /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/platypus/RiboSeq/RiboQC_RiboNovel/RibORF/platypus_repre.valid.pred.pvalue.parameters.PROTEIN.noredunant.fa /projects_eg/projects/marta/TestisRestricted_Microproteins_TSA/with_TranscriptomeReconstruction/Multimap_altORFs/Q1_TestisORFs/human/ribORF_humanTestis_in1.fa /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/MMseqsDB/species_and_humanDB \n",
      "\n",
      "MMseqs Version:       \t14.7e284\n",
      "Database type         \t0\n",
      "Shuffle input database\ttrue\n",
      "Createdb mode         \t0\n",
      "Write lookup file     \t1\n",
      "Offset of numeric ids \t0\n",
      "Compressed            \t0\n",
      "Verbosity             \t3\n",
      "\n",
      "Converting sequences\n",
      "[==============\n",
      "Time for merging to species_and_humanDB_h: 0h 0m 4s 375ms\n",
      "Time for merging to species_and_humanDB: 0h 0m 4s 853ms\n",
      "Database type: Aminoacid\n",
      "Time for processing: 0h 1m 15s 574ms\n"
     ]
    }
   ],
   "source": [
    "%%bash -s \"$evoDir\" \"$species_dir\"\n",
    "\n",
    "module load Miniconda3/4.9.2\n",
    "source /soft/system/software/Miniconda3/4.9.2/bin/activate MMseqs2\n",
    "\n",
    "mmseqs createdb $2/*/RiboSeq/RiboQC_RiboNovel*/RibORF/*_repre.valid.pred.pvalue.parameters.PROTEIN.noredunant.fa /projects_eg/projects/marta/TestisRestricted_Microproteins_TSA/with_TranscriptomeReconstruction/Multimap_altORFs/Q1_TestisORFs/human/ribORF_humanTestis_in1.fa $1/MMseqsDB/species_and_humanDB\n",
    "\n"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Searching"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "rm: cannot remove '/users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/Searching/resultDB*': No such file or directory\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Create directory /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/tmp\n",
      "search -a -s 6 /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/MMseqsDB/ribORF_humanTestis_in1_DB /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/MMseqsDB/speciesDB /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/Searching/resultDB /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/tmp \n",
      "\n",
      "MMseqs Version:                        \t14.7e284\n",
      "Substitution matrix                    \taa:blosum62.out,nucl:nucleotide.out\n",
      "Add backtrace                          \ttrue\n",
      "Alignment mode                         \t2\n",
      "Alignment mode                         \t0\n",
      "Allow wrapped scoring                  \tfalse\n",
      "E-value threshold                      \t0.001\n",
      "Seq. id. threshold                     \t0\n",
      "Min alignment length                   \t0\n",
      "Seq. id. mode                          \t0\n",
      "Alternative alignments                 \t0\n",
      "Coverage threshold                     \t0\n",
      "Coverage mode                          \t0\n",
      "Max sequence length                    \t65535\n",
      "Compositional bias                     \t1\n",
      "Compositional bias                     \t1\n",
      "Max reject                             \t2147483647\n",
      "Max accept                             \t2147483647\n",
      "Include identical seq. id.             \tfalse\n",
      "Preload mode                           \t0\n",
      "Pseudo count a                         \tsubstitution:1.100,context:1.400\n",
      "Pseudo count b                         \tsubstitution:4.100,context:5.800\n",
      "Score bias                             \t0\n",
      "Realign hits                           \tfalse\n",
      "Realign score bias                     \t-0.2\n",
      "Realign max seqs                       \t2147483647\n",
      "Correlation score weight               \t0\n",
      "Gap open cost                          \taa:11,nucl:5\n",
      "Gap extension cost                     \taa:1,nucl:2\n",
      "Zdrop                                  \t40\n",
      "Threads                                \t24\n",
      "Compressed                             \t0\n",
      "Verbosity                              \t3\n",
      "Seed substitution matrix               \taa:VTML80.out,nucl:nucleotide.out\n",
      "Sensitivity                            \t6\n",
      "k-mer length                           \t0\n",
      "k-score                                \tseq:2147483647,prof:2147483647\n",
      "Alphabet size                          \taa:21,nucl:5\n",
      "Max results per query                  \t300\n",
      "Split database                         \t0\n",
      "Split mode                             \t2\n",
      "Split memory limit                     \t0\n",
      "Diagonal scoring                       \ttrue\n",
      "Exact k-mer matching                   \t0\n",
      "Mask residues                          \t1\n",
      "Mask residues probability              \t0.9\n",
      "Mask lower case residues               \t0\n",
      "Minimum diagonal score                 \t15\n",
      "Selected taxa                          \t\n",
      "Spaced k-mers                          \t1\n",
      "Spaced k-mer pattern                   \t\n",
      "Local temporary path                   \t\n",
      "Rescore mode                           \t0\n",
      "Remove hits by seq. id. and coverage   \tfalse\n",
      "Sort results                           \t0\n",
      "Mask profile                           \t1\n",
      "Profile E-value threshold              \t0.1\n",
      "Global sequence weighting              \tfalse\n",
      "Allow deletions                        \tfalse\n",
      "Filter MSA                             \t1\n",
      "Use filter only at N seqs              \t0\n",
      "Maximum seq. id. threshold             \t0.9\n",
      "Minimum seq. id.                       \t0.0\n",
      "Minimum score per column               \t-20\n",
      "Minimum coverage                       \t0\n",
      "Select N most diverse seqs             \t1000\n",
      "Pseudo count mode                      \t0\n",
      "Gap pseudo count                       \t10\n",
      "Min codons in orf                      \t30\n",
      "Max codons in length                   \t32734\n",
      "Max orf gaps                           \t2147483647\n",
      "Contig start mode                      \t2\n",
      "Contig end mode                        \t2\n",
      "Orf start mode                         \t1\n",
      "Forward frames                         \t1,2,3\n",
      "Reverse frames                         \t1,2,3\n",
      "Translation table                      \t1\n",
      "Translate orf                          \t0\n",
      "Use all table starts                   \tfalse\n",
      "Offset of numeric ids                  \t0\n",
      "Create lookup                          \t0\n",
      "Add orf stop                           \tfalse\n",
      "Overlap between sequences              \t0\n",
      "Sequence split mode                    \t1\n",
      "Header split mode                      \t0\n",
      "Chain overlapping alignments           \t0\n",
      "Merge query                            \t1\n",
      "Search type                            \t0\n",
      "Search iterations                      \t1\n",
      "Start sensitivity                      \t4\n",
      "Search steps                           \t1\n",
      "Exhaustive search mode                 \tfalse\n",
      "Filter results during exhaustive search\t0\n",
      "Strand selection                       \t1\n",
      "LCA search mode                        \tfalse\n",
      "Disk space limit                       \t0\n",
      "MPI runner                             \t\n",
      "Force restart with latest tmp          \tfalse\n",
      "Remove temporary files                 \tfalse\n",
      "\n",
      "prefilter /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/MMseqsDB/ribORF_humanTestis_in1_DB /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/MMseqsDB/speciesDB /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/tmp/12946947319972804254/pref_0 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' --seed-sub-mat 'aa:VTML80.out,nucl:nucleotide.out' -k 0 --k-score seq:2147483647,prof:2147483647 --alph-size aa:21,nucl:5 --max-seq-len 65535 --max-seqs 300 --split 0 --split-mode 2 --split-memory-limit 0 -c 0 --cov-mode 0 --comp-bias-corr 1 --comp-bias-corr-scale 1 --diag-score 1 --exact-kmer-matching 0 --mask 1 --mask-prob 0.9 --mask-lower-case 0 --min-ungapped-score 15 --add-self-matches 0 --spaced-kmer-mode 1 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --threads 24 --compressed 0 -v 3 -s 6.0 \n",
      "\n",
      "Query database size: 27106 type: Aminoacid\n",
      "Estimated memory consumption: 1G\n",
      "Target database size: 115413 type: Aminoacid\n",
      "Index table k-mer threshold: 109 at k-mer size 6 \n",
      "Index table: counting k-mers\n",
      "[=================================================================] 115.41K 0s 691ms\n",
      "Index table: Masked residues: 1929565\n",
      "Index table: fill\n",
      "[=================================================================] 115.41K 0s 446ms\n",
      "Index statistics\n",
      "Entries:          42415036\n",
      "DB size:          730 MB\n",
      "Avg k-mer size:   0.662735\n",
      "Top 10 k-mers\n",
      "    DVVFEE\t478\n",
      "    LLLLLL\t278\n",
      "    TFDAFT\t246\n",
      "    VMEYLV\t232\n",
      "    LYDMNY\t224\n",
      "    LLPQLY\t183\n",
      "    VHSGRT\t150\n",
      "    LGGFVY\t134\n",
      "    SSSSGS\t132\n",
      "    IGGFVY\t132\n",
      "Time for index table init: 0h 0m 1s 607ms\n",
      "Process prefiltering step 1 of 1\n",
      "\n",
      "k-mer similarity threshold: 109\n",
      "Starting prefiltering scores calculation (step 1 of 1)\n",
      "Query db start 1 to 27106\n",
      "Target db start 1 to 115413\n",
      "[=================================================================] 27.11K 12s 803ms\n",
      "\n",
      "377.570766 k-mers per position\n",
      "94917 DB matches per sequence\n",
      "2 overflows\n",
      "0 queries produce too many hits (truncated result)\n",
      "209 sequences passed prefiltering per query sequence\n",
      "300 median result list length\n",
      "308 sequences with 0 size result lists\n",
      "Time for merging to pref_0: 0h 0m 0s 264ms\n",
      "Time for processing: 0h 0m 16s 609ms\n",
      "align /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/MMseqsDB/ribORF_humanTestis_in1_DB /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/MMseqsDB/speciesDB /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/tmp/12946947319972804254/pref_0 /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/Searching/resultDB --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' -a 1 --alignment-mode 2 --alignment-output-mode 0 --wrapped-scoring 0 -e 0.001 --min-seq-id 0 --min-aln-len 0 --seq-id-mode 0 --alt-ali 0 -c 0 --cov-mode 0 --max-seq-len 65535 --comp-bias-corr 1 --comp-bias-corr-scale 1 --max-rejected 2147483647 --max-accept 2147483647 --add-self-matches 0 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --score-bias 0 --realign 0 --realign-score-bias -0.2 --realign-max-seqs 2147483647 --corr-score-weight 0 --gap-open aa:11,nucl:5 --gap-extend aa:1,nucl:2 --zdrop 40 --threads 24 --compressed 0 -v 3 \n",
      "\n",
      "Compute score, coverage and sequence identity\n",
      "Query database size: 27106 type: Aminoacid\n",
      "Target database size: 115413 type: Aminoacid\n",
      "Calculation of alignments\n",
      "[=================================================================] 27.11K 1m 9s 480ms\n",
      "Time for merging to resultDB: 0h 0m 0s 292ms\n",
      "5665541 alignments calculated\n",
      "1185782 sequence pairs passed the thresholds (0.209297 of overall calculated)\n",
      "43.746109 hits per query sequence\n",
      "Time for processing: 0h 1m 11s 978ms\n",
      "convertalis /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/MMseqsDB/ribORF_humanTestis_in1_DB /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/MMseqsDB/speciesDB /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/Searching/resultDB /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/Searching/resultDB.m8 \n",
      "\n",
      "MMseqs Version:        \t14.7e284\n",
      "Substitution matrix    \taa:blosum62.out,nucl:nucleotide.out\n",
      "Alignment format       \t0\n",
      "Format alignment output\tquery,target,fident,alnlen,mismatch,gapopen,qstart,qend,tstart,tend,evalue,bits\n",
      "Translation table      \t1\n",
      "Gap open cost          \taa:11,nucl:5\n",
      "Gap extension cost     \taa:1,nucl:2\n",
      "Database output        \tfalse\n",
      "Preload mode           \t0\n",
      "Search type            \t0\n",
      "Threads                \t24\n",
      "Compressed             \t0\n",
      "Verbosity              \t3\n",
      "\n",
      "[=================================================================] 27.11K 1s 245ms\n",
      "Time for merging to resultDB.m8: 0h 0m 3s 373ms\n",
      "Time for processing: 0h 0m 7s 317ms\n"
     ]
    }
   ],
   "source": [
    "# %%bash -s \"$evoDir\" \"$users_dir\"\n",
    "\n",
    "# module load Miniconda3/4.9.2\n",
    "# source /soft/system/software/Miniconda3/4.9.2/bin/activate MMseqs2\n",
    "# mkdir -p $1/Searching\n",
    "\n",
    "# rm $1/Searching/resultDB*\n",
    "# ## The alignment consists of two steps the prefilter and alignment. To run the search, type:\n",
    "# mmseqs search -a -s 6 $1/MMseqsDB/ribORF_humanTestis_in1_DB $1/MMseqsDB/speciesDB $1/Searching/resultDB $1/tmp\n",
    "\n",
    "# # Then, convert the result database into a BLAST tab formatted file (option -m 8 in legacy blast, -outfmt 6 in blast+):\n",
    "# # Format alignment output\tquery,target,fident,alnlen,mismatch,gapopen,qstart,qend,tstart,tend,evalue,bits\n",
    "\n",
    "# mmseqs convertalis $1/MMseqsDB/ribORF_humanTestis_in1_DB $1/MMseqsDB/speciesDB $1/Searching/resultDB $1/Searching/resultDB.m8"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Clustering"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "cluster -c 0.5 --min-seq-id 0.5 /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/MMseqsDB/species_and_humanDB /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/Clustering/species_and_human_cluDB /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/tmp \n",
      "\n",
      "MMseqs Version:                     \t14.7e284\n",
      "Substitution matrix                 \taa:blosum62.out,nucl:nucleotide.out\n",
      "Seed substitution matrix            \taa:VTML80.out,nucl:nucleotide.out\n",
      "Sensitivity                         \t4\n",
      "k-mer length                        \t0\n",
      "k-score                             \tseq:2147483647,prof:2147483647\n",
      "Alphabet size                       \taa:21,nucl:5\n",
      "Max sequence length                 \t65535\n",
      "Max results per query               \t20\n",
      "Split database                      \t0\n",
      "Split mode                          \t2\n",
      "Split memory limit                  \t0\n",
      "Coverage threshold                  \t0.5\n",
      "Coverage mode                       \t0\n",
      "Compositional bias                  \t1\n",
      "Compositional bias                  \t1\n",
      "Diagonal scoring                    \ttrue\n",
      "Exact k-mer matching                \t0\n",
      "Mask residues                       \t1\n",
      "Mask residues probability           \t0.9\n",
      "Mask lower case residues            \t0\n",
      "Minimum diagonal score              \t15\n",
      "Selected taxa                       \t\n",
      "Include identical seq. id.          \tfalse\n",
      "Spaced k-mers                       \t1\n",
      "Preload mode                        \t0\n",
      "Pseudo count a                      \tsubstitution:1.100,context:1.400\n",
      "Pseudo count b                      \tsubstitution:4.100,context:5.800\n",
      "Spaced k-mer pattern                \t\n",
      "Local temporary path                \t\n",
      "Threads                             \t24\n",
      "Compressed                          \t0\n",
      "Verbosity                           \t3\n",
      "Add backtrace                       \tfalse\n",
      "Alignment mode                      \t3\n",
      "Alignment mode                      \t0\n",
      "Allow wrapped scoring               \tfalse\n",
      "E-value threshold                   \t0.001\n",
      "Seq. id. threshold                  \t0.5\n",
      "Min alignment length                \t0\n",
      "Seq. id. mode                       \t0\n",
      "Alternative alignments              \t0\n",
      "Max reject                          \t2147483647\n",
      "Max accept                          \t2147483647\n",
      "Score bias                          \t0\n",
      "Realign hits                        \tfalse\n",
      "Realign score bias                  \t-0.2\n",
      "Realign max seqs                    \t2147483647\n",
      "Correlation score weight            \t0\n",
      "Gap open cost                       \taa:11,nucl:5\n",
      "Gap extension cost                  \taa:1,nucl:2\n",
      "Zdrop                               \t40\n",
      "Rescore mode                        \t0\n",
      "Remove hits by seq. id. and coverage\tfalse\n",
      "Sort results                        \t0\n",
      "Cluster mode                        \t0\n",
      "Max connected component depth       \t1000\n",
      "Similarity type                     \t2\n",
      "Single step clustering              \tfalse\n",
      "Cascaded clustering steps           \t3\n",
      "Cluster reassign                    \tfalse\n",
      "Remove temporary files              \tfalse\n",
      "Force restart with latest tmp       \tfalse\n",
      "MPI runner                          \t\n",
      "k-mers per sequence                 \t21\n",
      "Scale k-mers per sequence           \taa:0.000,nucl:0.200\n",
      "Adjust k-mer length                 \tfalse\n",
      "Shift hash                          \t67\n",
      "Include only extendable             \tfalse\n",
      "Skip repeating k-mers               \tfalse\n",
      "\n",
      "Set cluster sensitivity to -s 3.000000\n",
      "Set cluster mode SET COVER\n",
      "Set cluster iterations to 3\n",
      "linclust /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/MMseqsDB/species_and_humanDB /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/tmp/8741965568736131637/clu_redundancy /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/tmp/8741965568736131637/linclust --cluster-mode 0 --max-iterations 1000 --similarity-type 2 --threads 24 --compressed 0 -v 3 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' -a 0 --alignment-mode 3 --alignment-output-mode 0 --wrapped-scoring 0 -e 0.001 --min-seq-id 0.5 --min-aln-len 0 --seq-id-mode 0 --alt-ali 0 -c 0.5 --cov-mode 0 --max-seq-len 65535 --comp-bias-corr 1 --comp-bias-corr-scale 1 --max-rejected 2147483647 --max-accept 2147483647 --add-self-matches 0 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --score-bias 0 --realign 0 --realign-score-bias -0.2 --realign-max-seqs 2147483647 --corr-score-weight 0 --gap-open aa:11,nucl:5 --gap-extend aa:1,nucl:2 --zdrop 40 --alph-size aa:13,nucl:5 --kmer-per-seq 21 --spaced-kmer-mode 1 --kmer-per-seq-scale aa:0.000,nucl:0.200 --adjust-kmer-len 0 --mask 0 --mask-prob 0.9 --mask-lower-case 0 -k 0 --hash-shift 67 --split-memory-limit 0 --include-only-extendable 0 --ignore-multi-kmer 0 --rescore-mode 0 --filter-hits 0 --sort-results 0 --remove-tmp-files 0 --force-reuse 0 \n",
      "\n",
      "kmermatcher /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/MMseqsDB/species_and_humanDB /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/tmp/8741965568736131637/linclust/9950589562075420023/pref --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' --alph-size aa:13,nucl:5 --min-seq-id 0.5 --kmer-per-seq 21 --spaced-kmer-mode 1 --kmer-per-seq-scale aa:0.000,nucl:0.200 --adjust-kmer-len 0 --mask 0 --mask-prob 0.9 --mask-lower-case 0 --cov-mode 0 -k 0 -c 0.5 --max-seq-len 65535 --hash-shift 67 --split-memory-limit 0 --include-only-extendable 0 --ignore-multi-kmer 0 --threads 24 --compressed 0 -v 3 \n",
      "\n",
      "kmermatcher /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/MMseqsDB/species_and_humanDB /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/tmp/8741965568736131637/linclust/9950589562075420023/pref --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' --alph-size aa:13,nucl:5 --min-seq-id 0.5 --kmer-per-seq 21 --spaced-kmer-mode 1 --kmer-per-seq-scale aa:0.000,nucl:0.200 --adjust-kmer-len 0 --mask 0 --mask-prob 0.9 --mask-lower-case 0 --cov-mode 0 -k 0 -c 0.5 --max-seq-len 65535 --hash-shift 67 --split-memory-limit 0 --include-only-extendable 0 --ignore-multi-kmer 0 --threads 24 --compressed 0 -v 3 \n",
      "\n",
      "Database size: 142519 type: Aminoacid\n",
      "Reduced amino acid alphabet: (A S T) (C) (D B N) (E Q Z) (F Y) (G) (H) (I V) (K R) (L J M) (P) (W) (X) \n",
      "\n",
      "Generate k-mers list for 1 split\n",
      "[=================================================================] 142.52K 0s 706ms\n",
      "Sort kmer 0h 0m 0s 324ms\n",
      "Sort by rep. sequence 0h 0m 0s 260ms\n",
      "Time for fill: 0h 0m 0s 100ms\n",
      "Time for merging to pref: 0h 0m 0s 773ms\n",
      "Time for processing: 0h 0m 6s 640ms\n",
      "rescorediagonal /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/MMseqsDB/species_and_humanDB /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/MMseqsDB/species_and_humanDB /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/tmp/8741965568736131637/linclust/9950589562075420023/pref /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/tmp/8741965568736131637/linclust/9950589562075420023/pref_rescore1 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' --rescore-mode 0 --wrapped-scoring 0 --filter-hits 0 -e 0.001 -c 0.5 -a 0 --cov-mode 0 --min-seq-id 0.5 --min-aln-len 0 --seq-id-mode 0 --add-self-matches 0 --sort-results 0 --db-load-mode 0 --threads 24 --compressed 0 -v 3 \n",
      "\n",
      "[================================================================] 142.52K 0s 207ms\n",
      "=Time for merging to pref_rescore1: 0h 0m 2s 941ms\n",
      "Time for processing: 0h 0m 20s 309ms\n",
      "clust /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/MMseqsDB/species_and_humanDB /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/tmp/8741965568736131637/linclust/9950589562075420023/pref_rescore1 /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/tmp/8741965568736131637/linclust/9950589562075420023/pre_clust --cluster-mode 0 --max-iterations 1000 --similarity-type 2 --threads 24 --compressed 0 -v 3 \n",
      "\n",
      "Clustering mode: Set Cover\n",
      "[=================================================================] 142.52K 0s 39ms\n",
      "Sort entries\n",
      "Find missing connections\n",
      "Found 90541 new connections.\n",
      "Reconstruct initial order\n",
      "[=================================================================] 142.52K 0s 38ms\n",
      "Add missing connections\n",
      "[=================================================================] 142.52K 0s 28ms\n",
      "\n",
      "Time for read in: 0h 0m 0s 403ms\n",
      "Total time: 0h 0m 1s 481ms\n",
      "\n",
      "Size of the sequence database: 142519\n",
      "Size of the alignment database: 142519\n",
      "Number of clusters: 93442\n",
      "\n",
      "Writing results 0h 0m 0s 597ms\n",
      "Time for merging to pre_clust: 0h 0m 0s 802ms\n",
      "Time for processing: 0h 0m 5s 152ms\n",
      "createsubdb /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/tmp/8741965568736131637/linclust/9950589562075420023/order_redundancy /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/MMseqsDB/species_and_humanDB /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/tmp/8741965568736131637/linclust/9950589562075420023/input_step_redundancy -v 3 --subdb-mode 1 \n",
      "\n",
      "Time for merging to input_step_redundancy: 0h 0m 0s 960ms\n",
      "Time for processing: 0h 0m 3s 297ms\n",
      "createsubdb /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/tmp/8741965568736131637/linclust/9950589562075420023/order_redundancy /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/tmp/8741965568736131637/linclust/9950589562075420023/pref /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/tmp/8741965568736131637/linclust/9950589562075420023/pref_filter1 -v 3 --subdb-mode 1 \n",
      "\n",
      "Time for merging to pref_filter1: 0h 0m 0s 951ms\n",
      "Time for processing: 0h 0m 1s 647ms\n",
      "filterdb /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/tmp/8741965568736131637/linclust/9950589562075420023/pref_filter1 /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/tmp/8741965568736131637/linclust/9950589562075420023/pref_filter2 --filter-file /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/tmp/8741965568736131637/linclust/9950589562075420023/order_redundancy --threads 24 --compressed 0 -v 3 \n",
      "\n",
      "Filtering using file(s)\n",
      "[=================================================================] 93.44K 0s 150ms\n",
      "Time for merging to pref_filter2: 0h 0m 5s 386ms\n",
      "Time for processing: 0h 0m 29s 536ms\n",
      "rescorediagonal /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/tmp/8741965568736131637/linclust/9950589562075420023/input_step_redundancy /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/tmp/8741965568736131637/linclust/9950589562075420023/input_step_redundancy /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/tmp/8741965568736131637/linclust/9950589562075420023/pref_filter2 /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/tmp/8741965568736131637/linclust/9950589562075420023/pref_rescore2 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' --rescore-mode 1 --wrapped-scoring 0 --filter-hits 1 -e 0.001 -c 0.5 -a 0 --cov-mode 0 --min-seq-id 0.5 --min-aln-len 0 --seq-id-mode 0 --add-self-matches 0 --sort-results 0 --db-load-mode 0 --threads 24 --compressed 0 -v 3 \n",
      "\n",
      "[=================================================================] 93.44K 1s 559ms\n",
      "Time for merging to pref_rescore2: 0h 0m 4s 310ms\n",
      "Time for processing: 0h 0m 22s 235ms\n",
      "align /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/tmp/8741965568736131637/linclust/9950589562075420023/input_step_redundancy /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/tmp/8741965568736131637/linclust/9950589562075420023/input_step_redundancy /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/tmp/8741965568736131637/linclust/9950589562075420023/pref_rescore2 /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/tmp/8741965568736131637/linclust/9950589562075420023/aln --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' -a 0 --alignment-mode 3 --alignment-output-mode 0 --wrapped-scoring 0 -e 0.001 --min-seq-id 0.5 --min-aln-len 0 --seq-id-mode 0 --alt-ali 0 -c 0.5 --cov-mode 0 --max-seq-len 65535 --comp-bias-corr 1 --comp-bias-corr-scale 1 --max-rejected 2147483647 --max-accept 2147483647 --add-self-matches 0 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --score-bias 0 --realign 0 --realign-score-bias -0.2 --realign-max-seqs 2147483647 --corr-score-weight 0 --gap-open aa:11,nucl:5 --gap-extend aa:1,nucl:2 --zdrop 40 --threads 24 --compressed 0 -v 3 \n",
      "\n",
      "Compute score, coverage and sequence identity\n",
      "Query database size: 93442 type: Aminoacid\n",
      "Target database size: 93442 type: Aminoacid\n",
      "Calculation of alignments\n",
      "[=================================================================] 93.44K 15s 605ms\n",
      "Time for merging to aln: 0h 0m 3s 328ms\n",
      "115946 alignments calculated\n",
      "110333 sequence pairs passed the thresholds (0.951590 of overall calculated)\n",
      "1.180765 hits per query sequence\n",
      "Time for processing: 0h 0m 33s 316ms\n",
      "clust /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/tmp/8741965568736131637/linclust/9950589562075420023/input_step_redundancy /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/tmp/8741965568736131637/linclust/9950589562075420023/aln /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/tmp/8741965568736131637/linclust/9950589562075420023/clust --cluster-mode 0 --max-iterations 1000 --similarity-type 2 --threads 24 --compressed 0 -v 3 \n",
      "\n",
      "Clustering mode: Set Cover\n",
      "[=================================================================] 93.44K 0s 51ms\n",
      "Sort entries\n",
      "Find missing connections\n",
      "Found 16891 new connections.\n",
      "Reconstruct initial order\n",
      "[=================================================================] 93.44K 0s 31ms\n",
      "Add missing connections\n",
      "[=================================================================] 93.44K 0s 10ms\n",
      "\n",
      "Time for read in: 0h 0m 0s 333ms\n",
      "Total time: 0h 0m 0s 761ms\n",
      "\n",
      "Size of the sequence database: 93442\n",
      "Size of the alignment database: 93442\n",
      "Number of clusters: 80530\n",
      "\n",
      "Writing results 0h 0m 0s 37ms\n",
      "Time for merging to clust: 0h 0m 1s 416ms\n",
      "Time for processing: 0h 0m 2s 799ms\n",
      "mergeclusters /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/MMseqsDB/species_and_humanDB /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/tmp/8741965568736131637/clu_redundancy /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/tmp/8741965568736131637/linclust/9950589562075420023/pre_clust /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/tmp/8741965568736131637/linclust/9950589562075420023/clust --threads 24 --compressed 0 -v 3 \n",
      "\n",
      "Clustering step 1\n",
      "[=================================================================] 93.44K 0s 163ms\n",
      "Clustering step 2\n",
      "[=================================================================] 80.53K 0s 325ms\n",
      "Write merged clustering\n",
      "[=================================================================] 142.52K 14s 155ms\n",
      "Time for merging to clu_redundancy: 0h 0m 1s 872ms\n",
      "Time for processing: 0h 0m 19s 63ms\n",
      "createsubdb /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/tmp/8741965568736131637/clu_redundancy /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/MMseqsDB/species_and_humanDB /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/tmp/8741965568736131637/input_step_redundancy -v 3 --subdb-mode 1 \n",
      "\n",
      "Time for merging to input_step_redundancy: 0h 0m 2s 97ms\n",
      "Time for processing: 0h 0m 5s 289ms\n",
      "prefilter /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/tmp/8741965568736131637/input_step_redundancy /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/tmp/8741965568736131637/input_step_redundancy /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/tmp/8741965568736131637/pref_step0 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' --seed-sub-mat 'aa:VTML80.out,nucl:nucleotide.out' -s 1 -k 0 --k-score seq:2147483647,prof:2147483647 --alph-size aa:21,nucl:5 --max-seq-len 65535 --max-seqs 20 --split 0 --split-mode 2 --split-memory-limit 0 -c 0.5 --cov-mode 0 --comp-bias-corr 0 --comp-bias-corr-scale 1 --diag-score 0 --exact-kmer-matching 0 --mask 1 --mask-prob 0.9 --mask-lower-case 0 --min-ungapped-score 0 --add-self-matches 0 --spaced-kmer-mode 1 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --threads 24 --compressed 0 -v 3 \n",
      "\n",
      "Query database size: 80530 type: Aminoacid\n",
      "Estimated memory consumption: 1G\n",
      "Target database size: 80530 type: Aminoacid\n",
      "Index table k-mer threshold: 154 at k-mer size 6 \n",
      "Index table: counting k-mers\n",
      "[=================================================================] 80.53K 1s 430ms\n",
      "Index table: Masked residues: 919286\n",
      "Index table: fill\n",
      "[=================================================================] 80.53K 0s 583ms\n",
      "Index statistics\n",
      "Entries:          8856495\n",
      "DB size:          538 MB\n",
      "Avg k-mer size:   0.138383\n",
      "Top 10 k-mers\n",
      "    LYDMNY\t153\n",
      "    TFDAFT\t126\n",
      "    VADTEW\t85\n",
      "    IHDKNI\t65\n",
      "    EWLDQK\t65\n",
      "    RDMERN\t61\n",
      "    YRVLYR\t60\n",
      "    ECKFSS\t59\n",
      "    GGDCDS\t47\n",
      "    CQDGLV\t46\n",
      "Time for index table init: 0h 0m 3s 176ms\n",
      "Process prefiltering step 1 of 1\n",
      "\n",
      "k-mer similarity threshold: 154\n",
      "Starting prefiltering scores calculation (step 1 of 1)\n",
      "Query db start 1 to 80530\n",
      "Target db start 1 to 80530\n",
      "[=================================================================] 80.53K 2s 714ms\n",
      "\n",
      "2.278508 k-mers per position\n",
      "372 DB matches per sequence\n",
      "0 overflows\n",
      "0 queries produce too many hits (truncated result)\n",
      "3 sequences passed prefiltering per query sequence\n",
      "2 median result list length\n",
      "0 sequences with 0 size result lists\n",
      "Time for merging to pref_step0: 0h 0m 2s 370ms\n",
      "Time for processing: 0h 0m 25s 89ms\n",
      "align /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/tmp/8741965568736131637/input_step_redundancy /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/tmp/8741965568736131637/input_step_redundancy /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/tmp/8741965568736131637/pref_step0 /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/tmp/8741965568736131637/aln_step0 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' -a 0 --alignment-mode 3 --alignment-output-mode 0 --wrapped-scoring 0 -e 0.001 --min-seq-id 0.5 --min-aln-len 0 --seq-id-mode 0 --alt-ali 0 -c 0.5 --cov-mode 0 --max-seq-len 65535 --comp-bias-corr 0 --comp-bias-corr-scale 1 --max-rejected 2147483647 --max-accept 2147483647 --add-self-matches 0 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --score-bias 0 --realign 0 --realign-score-bias -0.2 --realign-max-seqs 2147483647 --corr-score-weight 0 --gap-open aa:11,nucl:5 --gap-extend aa:1,nucl:2 --zdrop 40 --threads 24 --compressed 0 -v 3 \n",
      "\n",
      "Compute score, coverage and sequence identity\n",
      "Query database size: 80530 type: Aminoacid\n",
      "Target database size: 80530 type: Aminoacid\n",
      "Calculation of alignments\n",
      "[=================================================================] 80.53K 43s 339ms\n",
      "Time for merging to aln_step0: 0h 0m 5s 129ms\n",
      "205162 alignments calculated\n",
      "102107 sequence pairs passed the thresholds (0.497690 of overall calculated)\n",
      "1.267937 hits per query sequence\n",
      "Time for processing: 0h 1m 12s 653ms\n",
      "clust /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/tmp/8741965568736131637/input_step_redundancy /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/tmp/8741965568736131637/aln_step0 /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/tmp/8741965568736131637/clu_step0 --cluster-mode 0 --max-iterations 1000 --similarity-type 2 --threads 24 --compressed 0 -v 3 \n",
      "\n",
      "Clustering mode: Set Cover\n",
      "[=================================================================] 80.53K 0s 35ms\n",
      "Sort entries\n",
      "Find missing connections\n",
      "Found 1889 new connections.\n",
      "Reconstruct initial order\n",
      "[=================================================================] 80.53K 0s 100ms\n",
      "Add missing connections\n",
      "[=================================================================] 80.53K 0s 3ms\n",
      "\n",
      "Time for read in: 0h 0m 0s 365ms\n",
      "Total time: 0h 0m 1s 920ms\n",
      "\n",
      "Size of the sequence database: 80530\n",
      "Size of the alignment database: 80530\n",
      "Number of clusters: 72672\n",
      "\n",
      "Writing results 0h 0m 0s 45ms\n",
      "Time for merging to clu_step0: 0h 0m 0s 925ms\n",
      "Time for processing: 0h 0m 4s 877ms\n",
      "createsubdb /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/tmp/8741965568736131637/clu_step0 /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/tmp/8741965568736131637/input_step_redundancy /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/tmp/8741965568736131637/input_step1 -v 3 --subdb-mode 1 \n",
      "\n",
      "Time for merging to input_step1: 0h 0m 1s 646ms\n",
      "Time for processing: 0h 0m 4s 929ms\n",
      "prefilter /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/tmp/8741965568736131637/input_step1 /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/tmp/8741965568736131637/input_step1 /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/tmp/8741965568736131637/pref_step1 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' --seed-sub-mat 'aa:VTML80.out,nucl:nucleotide.out' -s 2 -k 0 --k-score seq:2147483647,prof:2147483647 --alph-size aa:21,nucl:5 --max-seq-len 65535 --max-seqs 20 --split 0 --split-mode 2 --split-memory-limit 0 -c 0.5 --cov-mode 0 --comp-bias-corr 1 --comp-bias-corr-scale 1 --diag-score 1 --exact-kmer-matching 0 --mask 1 --mask-prob 0.9 --mask-lower-case 0 --min-ungapped-score 15 --add-self-matches 0 --spaced-kmer-mode 1 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --threads 24 --compressed 0 -v 3 \n",
      "\n",
      "Query database size: 72672 type: Aminoacid\n",
      "Estimated memory consumption: 1G\n",
      "Target database size: 72672 type: Aminoacid\n",
      "Index table k-mer threshold: 145 at k-mer size 6 \n",
      "Index table: counting k-mers\n",
      "[=================================================================] 72.67K 1s 221ms\n",
      "Index table: Masked residues: 603037\n",
      "Index table: fill\n",
      "[=================================================================] 72.67K 0s 554ms\n",
      "Index statistics\n",
      "Entries:          9444792\n",
      "DB size:          542 MB\n",
      "Avg k-mer size:   0.147575\n",
      "Top 10 k-mers\n",
      "    DVVFEE\t113\n",
      "    QRLRML\t60\n",
      "    LYDMNY\t60\n",
      "    TFDAFT\t54\n",
      "    VMEYLV\t54\n",
      "    CQDGLV\t34\n",
      "    VADTEW\t34\n",
      "    LLPQLY\t34\n",
      "    VHSGRT\t30\n",
      "    PAPPSP\t28\n",
      "Time for index table init: 0h 0m 2s 863ms\n",
      "Process prefiltering step 1 of 1\n",
      "\n",
      "k-mer similarity threshold: 145\n",
      "Starting prefiltering scores calculation (step 1 of 1)\n",
      "Query db start 1 to 72672\n",
      "Target db start 1 to 72672\n",
      "[=================================================================] 72.67K 3s 612ms\n",
      "\n",
      "6.012728 k-mers per position\n",
      "316 DB matches per sequence\n",
      "0 overflows\n",
      "0 queries produce too many hits (truncated result)\n",
      "3 sequences passed prefiltering per query sequence\n",
      "1 median result list length\n",
      "0 sequences with 0 size result lists\n",
      "Time for merging to pref_step1: 0h 0m 2s 445ms\n",
      "Time for processing: 0h 0m 28s 618ms\n",
      "align /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/tmp/8741965568736131637/input_step1 /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/tmp/8741965568736131637/input_step1 /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/tmp/8741965568736131637/pref_step1 /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/tmp/8741965568736131637/aln_step1 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' -a 0 --alignment-mode 3 --alignment-output-mode 0 --wrapped-scoring 0 -e 0.001 --min-seq-id 0.5 --min-aln-len 0 --seq-id-mode 0 --alt-ali 0 -c 0.5 --cov-mode 0 --max-seq-len 65535 --comp-bias-corr 1 --comp-bias-corr-scale 1 --max-rejected 2147483647 --max-accept 2147483647 --add-self-matches 0 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --score-bias 0 --realign 0 --realign-score-bias -0.2 --realign-max-seqs 2147483647 --corr-score-weight 0 --gap-open aa:11,nucl:5 --gap-extend aa:1,nucl:2 --zdrop 40 --threads 24 --compressed 0 -v 3 \n",
      "\n",
      "Compute score, coverage and sequence identity\n",
      "Query database size: 72672 type: Aminoacid\n",
      "Target database size: 72672 type: Aminoacid\n",
      "Calculation of alignments\n",
      "[=================================================================] 72.67K 15s 450ms\n",
      "Time for merging to aln_step1: 0h 0m 1s 742ms\n",
      "154144 alignments calculated\n",
      "73442 sequence pairs passed the thresholds (0.476451 of overall calculated)\n",
      "1.010596 hits per query sequence\n",
      "Time for processing: 0h 0m 28s 211ms\n",
      "clust /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/tmp/8741965568736131637/input_step1 /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/tmp/8741965568736131637/aln_step1 /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/tmp/8741965568736131637/clu_step1 --cluster-mode 0 --max-iterations 1000 --similarity-type 2 --threads 24 --compressed 0 -v 3 \n",
      "\n",
      "Clustering mode: Set Cover\n",
      "[=================================================================] 72.67K 0s 30ms\n",
      "Sort entries\n",
      "Find missing connections\n",
      "Found 238 new connections.\n",
      "Reconstruct initial order\n",
      "[=================================================================] 72.67K 0s 34ms\n",
      "Add missing connections\n",
      "[=================================================================] 72.67K 0s 1ms\n",
      "\n",
      "Time for read in: 0h 0m 0s 273ms\n",
      "Total time: 0h 0m 0s 848ms\n",
      "\n",
      "Size of the sequence database: 72672\n",
      "Size of the alignment database: 72672\n",
      "Number of clusters: 72202\n",
      "\n",
      "Writing results 0h 0m 0s 36ms\n",
      "Time for merging to clu_step1: 0h 0m 1s 975ms\n",
      "Time for processing: 0h 0m 3s 439ms\n",
      "createsubdb /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/tmp/8741965568736131637/clu_step1 /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/tmp/8741965568736131637/input_step1 /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/tmp/8741965568736131637/input_step2 -v 3 --subdb-mode 1 \n",
      "\n",
      "Time for merging to input_step2: 0h 0m 1s 995ms\n",
      "Time for processing: 0h 0m 6s 611ms\n",
      "prefilter /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/tmp/8741965568736131637/input_step2 /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/tmp/8741965568736131637/input_step2 /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/tmp/8741965568736131637/pref_step2 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' --seed-sub-mat 'aa:VTML80.out,nucl:nucleotide.out' -s 3 -k 0 --k-score seq:2147483647,prof:2147483647 --alph-size aa:21,nucl:5 --max-seq-len 65535 --max-seqs 20 --split 0 --split-mode 2 --split-memory-limit 0 -c 0.5 --cov-mode 0 --comp-bias-corr 1 --comp-bias-corr-scale 1 --diag-score 1 --exact-kmer-matching 0 --mask 1 --mask-prob 0.9 --mask-lower-case 0 --min-ungapped-score 15 --add-self-matches 0 --spaced-kmer-mode 1 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --threads 24 --compressed 0 -v 3 \n",
      "\n",
      "Query database size: 72202 type: Aminoacid\n",
      "Estimated memory consumption: 1G\n",
      "Target database size: 72202 type: Aminoacid\n",
      "Index table k-mer threshold: 136 at k-mer size 6 \n",
      "Index table: counting k-mers\n",
      "[=================================================================] 72.20K 1s 40ms\n",
      "Index table: Masked residues: 579549\n",
      "Index table: fill\n",
      "[=================================================================] 72.20K 0s 610ms\n",
      "Index statistics\n",
      "Entries:          12133590\n",
      "DB size:          557 MB\n",
      "Avg k-mer size:   0.189587\n",
      "Top 10 k-mers\n",
      "    DVVFEE\t86\n",
      "    VLQLLP\t65\n",
      "    QRLRML\t48\n",
      "    QIELEQ\t48\n",
      "    LYDMNY\t47\n",
      "    VMEYLV\t44\n",
      "    TKQLLV\t38\n",
      "    CQDGLV\t34\n",
      "    APPPPP\t33\n",
      "    RDARLV\t31\n",
      "Time for index table init: 0h 0m 2s 726ms\n",
      "Process prefiltering step 1 of 1\n",
      "\n",
      "k-mer similarity threshold: 136\n",
      "Starting prefiltering scores calculation (step 1 of 1)\n",
      "Query db start 1 to 72202\n",
      "Target db start 1 to 72202\n",
      "[=================================================================] 72.20K 5s 684ms\n",
      "\n",
      "17.172991 k-mers per position\n",
      "762 DB matches per sequence\n",
      "0 overflows\n",
      "0 queries produce too many hits (truncated result)\n",
      "5 sequences passed prefiltering per query sequence\n",
      "3 median result list length\n",
      "0 sequences with 0 size result lists\n",
      "Time for merging to pref_step2: 0h 0m 2s 863ms\n",
      "Time for processing: 0h 0m 24s 123ms\n",
      "align /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/tmp/8741965568736131637/input_step2 /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/tmp/8741965568736131637/input_step2 /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/tmp/8741965568736131637/pref_step2 /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/tmp/8741965568736131637/aln_step2 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' -a 0 --alignment-mode 3 --alignment-output-mode 0 --wrapped-scoring 0 -e 0.001 --min-seq-id 0.5 --min-aln-len 0 --seq-id-mode 0 --alt-ali 0 -c 0.5 --cov-mode 0 --max-seq-len 65535 --comp-bias-corr 1 --comp-bias-corr-scale 1 --max-rejected 2147483647 --max-accept 2147483647 --add-self-matches 0 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --score-bias 0 --realign 0 --realign-score-bias -0.2 --realign-max-seqs 2147483647 --corr-score-weight 0 --gap-open aa:11,nucl:5 --gap-extend aa:1,nucl:2 --zdrop 40 --threads 24 --compressed 0 -v 3 \n",
      "\n",
      "Compute score, coverage and sequence identity\n",
      "Query database size: 72202 type: Aminoacid\n",
      "Target database size: 72202 type: Aminoacid\n",
      "Calculation of alignments\n",
      "[=================================================================] 72.20K 17s 11ms\n",
      "Time for merging to aln_step2: 0h 0m 2s 109ms\n",
      "208618 alignments calculated\n",
      "72334 sequence pairs passed the thresholds (0.346729 of overall calculated)\n",
      "1.001828 hits per query sequence\n",
      "Time for processing: 0h 0m 38s 642ms\n",
      "clust /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/tmp/8741965568736131637/input_step2 /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/tmp/8741965568736131637/aln_step2 /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/tmp/8741965568736131637/clu_step2 --cluster-mode 0 --max-iterations 1000 --similarity-type 2 --threads 24 --compressed 0 -v 3 \n",
      "\n",
      "Clustering mode: Set Cover\n",
      "================================================================[=] 72.20K 0s 41ms\n",
      "Sort entries\n",
      "Find missing connections\n",
      "Found 48 new connections.\n",
      "Reconstruct initial order\n",
      "[=================================================================] 72.20K 0s 50ms\n",
      "Add missing connections\n",
      "[=================================================================] 72.20K 0s 1ms\n",
      "\n",
      "Time for read in: 0h 0m 0s 422ms\n",
      "Total time: 0h 0m 2s 336ms\n",
      "\n",
      "Size of the sequence database: 72202\n",
      "Size of the alignment database: 72202\n",
      "Number of clusters: 72112\n",
      "\n",
      "Writing results 0h 0m 0s 263ms\n",
      "Time for merging to clu_step2: 0h 0m 1s 274ms\n",
      "Time for processing: 0h 0m 5s 679ms\n",
      "mergeclusters /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/MMseqsDB/species_and_humanDB /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/Clustering/species_and_human_cluDB /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/tmp/8741965568736131637/clu_redundancy /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/tmp/8741965568736131637/clu_step0 /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/tmp/8741965568736131637/clu_step1 /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/tmp/8741965568736131637/clu_step2 --threads 24 --compressed 0 -v 3 \n",
      "\n",
      "Clustering step 1\n",
      "[=================================================================] 80.53K 0s 685ms\n",
      "Clustering step 2\n",
      "[=================================================================] 72.67K 1s 141ms\n",
      "Clustering step 3\n",
      "[=================================================================] 72.20K 1s 285ms\n",
      "Clustering step 4\n",
      "[=================================================================] 72.11K 2s 194ms\n",
      "Write merged clustering\n",
      "[=================================================================] 142.52K 11s 568ms\n",
      "Time for merging to species_and_human_cluDB: 0h 0m 4s 964ms\n",
      "Time for processing: 0h 0m 20s 492ms\n",
      "createtsv /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/MMseqsDB/species_and_humanDB /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/MMseqsDB/species_and_humanDB /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/Clustering/species_and_human_cluDB /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/Clustering/species_and_human_cluDB.tsv \n",
      "\n",
      "MMseqs Version:                 \t14.7e284\n",
      "First sequence as representative\tfalse\n",
      "Target column                   \t1\n",
      "Add full header                 \tfalse\n",
      "Sequence source                 \t0\n",
      "Database output                 \tfalse\n",
      "Threads                         \t24\n",
      "Compressed                      \t0\n",
      "Verbosity                       \t3\n",
      "\n",
      "Time for merging to species_and_human_cluDB.tsv: 0h 0m 9s 647ms\n",
      "Time for processing: 0h 0m 29s 750ms\n"
     ]
    }
   ],
   "source": [
    "%%bash -s \"$evoDir\" \"$users_dir\"\n",
    "\n",
    "module load Miniconda3/4.9.2\n",
    "source /soft/system/software/Miniconda3/4.9.2/bin/activate MMseqs2\n",
    "\n",
    "mkdir -p $1/Clustering\n",
    "\n",
    "rm -r $1/Clustering/species_and_human_cluDB*\n",
    "\n",
    "## cluster all species and human\n",
    "mmseqs cluster -c 0.5 --min-seq-id 0.5 $1/MMseqsDB/species_and_humanDB $1/Clustering/species_and_human_cluDB $1/tmp\n",
    "mmseqs createtsv $1/MMseqsDB/species_and_humanDB $1/MMseqsDB/species_and_humanDB $1/Clustering/species_and_human_cluDB $1/Clustering/species_and_human_cluDB.tsv"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Analysis"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 71,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "all human\n",
      "25985\n",
      "all human after annotation\n",
      "25985\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene_name</th>\n",
       "      <th>0</th>\n",
       "      <th>1</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>gene_type</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>lncRNA</th>\n",
       "      <td>2311</td>\n",
       "      <td>2311</td>\n",
       "      <td>2311</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>novel</th>\n",
       "      <td>310</td>\n",
       "      <td>310</td>\n",
       "      <td>310</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>processed_pseudogene</th>\n",
       "      <td>1455</td>\n",
       "      <td>1455</td>\n",
       "      <td>1455</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>protein_coding</th>\n",
       "      <td>21909</td>\n",
       "      <td>21909</td>\n",
       "      <td>21909</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                      gene_name      0      1\n",
       "gene_type                                    \n",
       "lncRNA                     2311   2311   2311\n",
       "novel                       310    310    310\n",
       "processed_pseudogene       1455   1455   1455\n",
       "protein_coding            21909  21909  21909"
      ]
     },
     "execution_count": 71,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "clustering = pd.read_csv(os.path.join(evoDir,\"Clustering/species_and_human_cluDB.tsv\"), header=None, sep=\"\\t\")\n",
    "clustering = clustering[clustering[0].str.startswith(\"ENST\") | clustering[0].str.startswith(\"TCONS\")]\n",
    "print(\"all human\")\n",
    "print(len(clustering))\n",
    "clustering['transcript_id'] = clustering[0].str.split(\":\", expand=True)[0]\n",
    "clustering['transcript_id'] = clustering[\"transcript_id\"].str.split(\".\", expand=True)[0]\n",
    "\n",
    "clustering = clustering.merge(transcript_gene, on=\"transcript_id\")\n",
    "print(\"all human after annotation\")\n",
    "print(len(clustering))\n",
    "small_clust = clustering[[\"gene_name\",\"gene_type\",0,1]]\n",
    "small_clust.groupby(\"gene_type\").count()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 72,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene_name</th>\n",
       "      <th>gene_type</th>\n",
       "      <th>0</th>\n",
       "      <th>1</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>STARD3NL</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>ENST00000009041.12:7:+|10|1703:194:239|ouORF|ATG</td>\n",
       "      <td>ENST00000009041.12:7:+|10|1703:194:239|ouORF|ATG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>AKAP11</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>ENST00000025301.4:13:+|2|9915:45:105|uORF|ATG</td>\n",
       "      <td>ENST00000025301.4:13:+|2|9915:45:105|uORF|ATG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>NFE2L3</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>ENST00000056233.4:7:+|17|3740:302:2387|canonic...</td>\n",
       "      <td>ENST00000056233.4:7:+|17|3740:302:2387|canonic...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>NFE2L3</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>ENST00000056233.4:7:+|17|3740:302:2387|canonic...</td>\n",
       "      <td>mouse_ENSMUST00000005103.12:chr6:+|24|2544:238...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>NFE2L3</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>ENST00000056233.4:7:+|17|3740:302:2387|canonic...</td>\n",
       "      <td>macaca_ENSMMUT00000005121:3:-|22|2456:375:2457...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25980</th>\n",
       "      <td>POLG-DT</td>\n",
       "      <td>lncRNA</td>\n",
       "      <td>ENST00000837147.1:15:+|17|1247:426:630|noncodi...</td>\n",
       "      <td>ENST00000837147.1:15:+|17|1247:426:630|noncodi...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25981</th>\n",
       "      <td>XLOC_000656</td>\n",
       "      <td>novel</td>\n",
       "      <td>TCONS_00000740:13:+|17|1854:344:740|noncoding|CTG</td>\n",
       "      <td>TCONS_00000740:13:+|17|1854:344:740|noncoding|CTG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25982</th>\n",
       "      <td>XLOC_000656</td>\n",
       "      <td>novel</td>\n",
       "      <td>TCONS_00000740:13:+|17|1854:344:740|noncoding|CTG</td>\n",
       "      <td>ENST00000685848.2:13:+|17|1854:344:740|noncodi...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25983</th>\n",
       "      <td>XLOC_002098</td>\n",
       "      <td>novel</td>\n",
       "      <td>TCONS_00002320:7:+|2|352:56:149|noncoding|ATG</td>\n",
       "      <td>TCONS_00002320:7:+|2|352:56:149|noncoding|ATG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25984</th>\n",
       "      <td>XLOC_002098</td>\n",
       "      <td>novel</td>\n",
       "      <td>TCONS_00002320:7:+|2|352:56:149|noncoding|ATG</td>\n",
       "      <td>ENST00000756632.1:7:+|2|352:56:149|noncoding|ATG</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>25985 rows  4 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "         gene_name       gene_type  \\\n",
       "0         STARD3NL  protein_coding   \n",
       "1           AKAP11  protein_coding   \n",
       "2           NFE2L3  protein_coding   \n",
       "3           NFE2L3  protein_coding   \n",
       "4           NFE2L3  protein_coding   \n",
       "...            ...             ...   \n",
       "25980      POLG-DT          lncRNA   \n",
       "25981  XLOC_000656           novel   \n",
       "25982  XLOC_000656           novel   \n",
       "25983  XLOC_002098           novel   \n",
       "25984  XLOC_002098           novel   \n",
       "\n",
       "                                                       0  \\\n",
       "0       ENST00000009041.12:7:+|10|1703:194:239|ouORF|ATG   \n",
       "1          ENST00000025301.4:13:+|2|9915:45:105|uORF|ATG   \n",
       "2      ENST00000056233.4:7:+|17|3740:302:2387|canonic...   \n",
       "3      ENST00000056233.4:7:+|17|3740:302:2387|canonic...   \n",
       "4      ENST00000056233.4:7:+|17|3740:302:2387|canonic...   \n",
       "...                                                  ...   \n",
       "25980  ENST00000837147.1:15:+|17|1247:426:630|noncodi...   \n",
       "25981  TCONS_00000740:13:+|17|1854:344:740|noncoding|CTG   \n",
       "25982  TCONS_00000740:13:+|17|1854:344:740|noncoding|CTG   \n",
       "25983      TCONS_00002320:7:+|2|352:56:149|noncoding|ATG   \n",
       "25984      TCONS_00002320:7:+|2|352:56:149|noncoding|ATG   \n",
       "\n",
       "                                                       1  \n",
       "0       ENST00000009041.12:7:+|10|1703:194:239|ouORF|ATG  \n",
       "1          ENST00000025301.4:13:+|2|9915:45:105|uORF|ATG  \n",
       "2      ENST00000056233.4:7:+|17|3740:302:2387|canonic...  \n",
       "3      mouse_ENSMUST00000005103.12:chr6:+|24|2544:238...  \n",
       "4      macaca_ENSMMUT00000005121:3:-|22|2456:375:2457...  \n",
       "...                                                  ...  \n",
       "25980  ENST00000837147.1:15:+|17|1247:426:630|noncodi...  \n",
       "25981  TCONS_00000740:13:+|17|1854:344:740|noncoding|CTG  \n",
       "25982  ENST00000685848.2:13:+|17|1854:344:740|noncodi...  \n",
       "25983      TCONS_00002320:7:+|2|352:56:149|noncoding|ATG  \n",
       "25984   ENST00000756632.1:7:+|2|352:56:149|noncoding|ATG  \n",
       "\n",
       "[25985 rows x 4 columns]"
      ]
     },
     "execution_count": 72,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "small_clust"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 73,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "noncoding all\n",
      "4076\n",
      "3279\n",
      "altORFs all\n",
      "7705\n",
      "7369\n",
      "Canonical all\n",
      "14204\n",
      "2899\n"
     ]
    }
   ],
   "source": [
    "## non-coding\n",
    "noncoding = clustering[clustering[0].str.contains(\"noncoding\")]\n",
    "noncoding.to_csv(os.path.join(evoDir,\"Clustering/noncodingHuman_speciesDB.tsv\"), header=None, index=None, sep=\"\\t\")\n",
    "print(\"noncoding all\")\n",
    "print(len(noncoding))\n",
    "print(len(set(noncoding[0].values.tolist())))\n",
    "\n",
    "## altORFs\n",
    "altORFs = clustering[clustering[0].str.contains(\"ORF\")]\n",
    "altORFs.to_csv(os.path.join(evoDir,\"Clustering/altORFsHuman_speciesDB.tsv\"), header=None, index=None, sep=\"\\t\")\n",
    "print(\"altORFs all\")\n",
    "print(len(altORFs))\n",
    "print(len(set(altORFs[0].values.tolist())))\n",
    "\n",
    "## canonical\n",
    "canonical = clustering[clustering[0].str.contains(\"canonical\")]\n",
    "print(\"Canonical all\")\n",
    "print(len(canonical))\n",
    "print(len(set(canonical[0].values.tolist())))\n",
    "canonical.to_csv(os.path.join(evoDir,\"Clustering/canonicalHuman_speciesDB.tsv\"), header=None, index=None, sep=\"\\t\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### NONCODING"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 74,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Conserved noncoding human ORFs\n",
      "178\n",
      "Non-Conserved noncoding human ORFs\n",
      "3279\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene_name</th>\n",
       "      <th>gene_type</th>\n",
       "      <th>orfID</th>\n",
       "      <th>1</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>170</th>\n",
       "      <td>ENSG00000217612</td>\n",
       "      <td>processed_pseudogene</td>\n",
       "      <td>ENST00000404707.2:6:-|73|1879:1490:1748|noncod...</td>\n",
       "      <td>ENST00000404707.2:6:-|73|1879:1490:1748|noncod...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>182</th>\n",
       "      <td>ENSG00000232952</td>\n",
       "      <td>processed_pseudogene</td>\n",
       "      <td>ENST00000415279.1:1:+|4|1779:134:167|noncoding...</td>\n",
       "      <td>ENST00000415279.1:1:+|4|1779:134:167|noncoding...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>183</th>\n",
       "      <td>ENSG00000232952</td>\n",
       "      <td>processed_pseudogene</td>\n",
       "      <td>ENST00000415279.1:1:+|6|1779:167:503|noncoding...</td>\n",
       "      <td>ENST00000415279.1:1:+|6|1779:167:503|noncoding...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>184</th>\n",
       "      <td>ENSG00000232952</td>\n",
       "      <td>processed_pseudogene</td>\n",
       "      <td>ENST00000415279.1:1:+|90|1779:1594:1684|noncod...</td>\n",
       "      <td>ENST00000415279.1:1:+|90|1779:1594:1684|noncod...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>185</th>\n",
       "      <td>ENSG00000232952</td>\n",
       "      <td>processed_pseudogene</td>\n",
       "      <td>ENST00000415279.1:1:+|25|1779:571:1231|noncodi...</td>\n",
       "      <td>ENST00000415279.1:1:+|25|1779:571:1231|noncodi...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25980</th>\n",
       "      <td>POLG-DT</td>\n",
       "      <td>lncRNA</td>\n",
       "      <td>ENST00000837147.1:15:+|17|1247:426:630|noncodi...</td>\n",
       "      <td>ENST00000837147.1:15:+|17|1247:426:630|noncodi...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25981</th>\n",
       "      <td>XLOC_000656</td>\n",
       "      <td>novel</td>\n",
       "      <td>TCONS_00000740:13:+|17|1854:344:740|noncoding|CTG</td>\n",
       "      <td>TCONS_00000740:13:+|17|1854:344:740|noncoding|CTG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25982</th>\n",
       "      <td>XLOC_000656</td>\n",
       "      <td>novel</td>\n",
       "      <td>TCONS_00000740:13:+|17|1854:344:740|noncoding|CTG</td>\n",
       "      <td>ENST00000685848.2:13:+|17|1854:344:740|noncodi...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25983</th>\n",
       "      <td>XLOC_002098</td>\n",
       "      <td>novel</td>\n",
       "      <td>TCONS_00002320:7:+|2|352:56:149|noncoding|ATG</td>\n",
       "      <td>TCONS_00002320:7:+|2|352:56:149|noncoding|ATG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25984</th>\n",
       "      <td>XLOC_002098</td>\n",
       "      <td>novel</td>\n",
       "      <td>TCONS_00002320:7:+|2|352:56:149|noncoding|ATG</td>\n",
       "      <td>ENST00000756632.1:7:+|2|352:56:149|noncoding|ATG</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>3591 rows  4 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "             gene_name             gene_type  \\\n",
       "170    ENSG00000217612  processed_pseudogene   \n",
       "182    ENSG00000232952  processed_pseudogene   \n",
       "183    ENSG00000232952  processed_pseudogene   \n",
       "184    ENSG00000232952  processed_pseudogene   \n",
       "185    ENSG00000232952  processed_pseudogene   \n",
       "...                ...                   ...   \n",
       "25980          POLG-DT                lncRNA   \n",
       "25981      XLOC_000656                 novel   \n",
       "25982      XLOC_000656                 novel   \n",
       "25983      XLOC_002098                 novel   \n",
       "25984      XLOC_002098                 novel   \n",
       "\n",
       "                                                   orfID  \\\n",
       "170    ENST00000404707.2:6:-|73|1879:1490:1748|noncod...   \n",
       "182    ENST00000415279.1:1:+|4|1779:134:167|noncoding...   \n",
       "183    ENST00000415279.1:1:+|6|1779:167:503|noncoding...   \n",
       "184    ENST00000415279.1:1:+|90|1779:1594:1684|noncod...   \n",
       "185    ENST00000415279.1:1:+|25|1779:571:1231|noncodi...   \n",
       "...                                                  ...   \n",
       "25980  ENST00000837147.1:15:+|17|1247:426:630|noncodi...   \n",
       "25981  TCONS_00000740:13:+|17|1854:344:740|noncoding|CTG   \n",
       "25982  TCONS_00000740:13:+|17|1854:344:740|noncoding|CTG   \n",
       "25983      TCONS_00002320:7:+|2|352:56:149|noncoding|ATG   \n",
       "25984      TCONS_00002320:7:+|2|352:56:149|noncoding|ATG   \n",
       "\n",
       "                                                       1  \n",
       "170    ENST00000404707.2:6:-|73|1879:1490:1748|noncod...  \n",
       "182    ENST00000415279.1:1:+|4|1779:134:167|noncoding...  \n",
       "183    ENST00000415279.1:1:+|6|1779:167:503|noncoding...  \n",
       "184    ENST00000415279.1:1:+|90|1779:1594:1684|noncod...  \n",
       "185    ENST00000415279.1:1:+|25|1779:571:1231|noncodi...  \n",
       "...                                                  ...  \n",
       "25980  ENST00000837147.1:15:+|17|1247:426:630|noncodi...  \n",
       "25981  TCONS_00000740:13:+|17|1854:344:740|noncoding|CTG  \n",
       "25982  ENST00000685848.2:13:+|17|1854:344:740|noncodi...  \n",
       "25983      TCONS_00002320:7:+|2|352:56:149|noncoding|ATG  \n",
       "25984   ENST00000756632.1:7:+|2|352:56:149|noncoding|ATG  \n",
       "\n",
       "[3591 rows x 4 columns]"
      ]
     },
     "execution_count": 74,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# if column 1 starts with a specie, it is conserved\n",
    "noncoding_conserved = noncoding[~noncoding[1].str.startswith(\"ENST\")]\n",
    "noncoding_conserved = noncoding_conserved[~noncoding_conserved[1].str.startswith(\"TCONS\")]\n",
    "noncoding_conserved.to_csv(os.path.join(evoDir,\"Clustering/noncodingHuman_Conserved.tsv\"), header=None, index=None, sep=\"\\t\")\n",
    "print(\"Conserved noncoding human ORFs\")\n",
    "print(len(set(noncoding_conserved[0].values.tolist())))\n",
    "noncoding_conserved = noncoding_conserved[['gene_name','gene_type',0,1]]\n",
    "noncoding_conserved.columns = ['gene_name','gene_type','orfID',1]\n",
    "\n",
    "# if column 1 starts with a human gene, it is not conserved\n",
    "noncoding_NonConserved = noncoding[noncoding[1].str.startswith(\"ENST\") | noncoding[1].str.startswith(\"TCONS\")]\n",
    "noncoding_NonConserved.to_csv(os.path.join(evoDir,\"Clustering/noncodingHuman_NonConserved.tsv\"), header=None, index=None, sep=\"\\t\")\n",
    "print(\"Non-Conserved noncoding human ORFs\")\n",
    "print(len(set(noncoding_NonConserved[0].values.tolist())))\n",
    "noncoding_NonConserved = noncoding_NonConserved[['gene_name','gene_type',0,1]]\n",
    "noncoding_NonConserved.columns = ['gene_name','gene_type','orfID',1]\n",
    "noncoding_NonConserved"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 75,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "178\n",
      "178\n",
      "178\n",
      "178\n",
      "178\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_2664244/3235198060.py:8: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  noncoding_conserved_to_matrix['presence'] = 1\n",
      "/tmp/ipykernel_2664244/3235198060.py:25: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  noncoding_conserved_matrix_ordered['sum'] = noncoding_conserved_matrix_ordered.select_dtypes(include='number').sum(axis=1)\n"
     ]
    }
   ],
   "source": [
    "## conserved non-coding\n",
    "noncoding_conserved_to_matrix = noncoding_conserved\n",
    "print(len(set(noncoding_conserved_to_matrix.orfID.values.tolist())))\n",
    "# Assuming noncoding_conserved_to_matrix is already defined\n",
    "noncoding_conserved_to_matrix['specie'] = noncoding_conserved_to_matrix[1].str.split(\"_\", expand=True)[0]\n",
    "noncoding_conserved_to_matrix = noncoding_conserved_to_matrix[['orfID', 'specie']]\n",
    "noncoding_conserved_to_matrix.groupby('specie').count()\n",
    "noncoding_conserved_to_matrix['presence'] = 1\n",
    "## control\n",
    "print(len(set(noncoding_conserved_to_matrix.orfID.values.tolist())))\n",
    "\n",
    "# Pivoting the data without creating an 'id', and keeping 'orfID' as the unique identifier\n",
    "noncoding_conserved_matrix = noncoding_conserved_to_matrix.pivot_table(index='orfID', columns='specie', values='presence', aggfunc='max')\n",
    "\n",
    "# Fill NaN values with 0\n",
    "noncoding_conserved_matrix = noncoding_conserved_matrix.fillna(0)\n",
    "\n",
    "# Reset index so 'orfID' becomes a column instead of an index\n",
    "noncoding_conserved_matrix = noncoding_conserved_matrix.reset_index()\n",
    "\n",
    "# Reorder columns for the final matrix\n",
    "noncoding_conserved_matrix_ordered = noncoding_conserved_matrix[[\"orfID\", \"macaca\", \"mouse\", \"opossum\", \"platypus\", \"chicken\"]]\n",
    "\n",
    "# Calculate sum across species columns to identify how many species are present\n",
    "noncoding_conserved_matrix_ordered['sum'] = noncoding_conserved_matrix_ordered.select_dtypes(include='number').sum(axis=1)\n",
    "\n",
    "# Sort by the sum column to order by the number of species present\n",
    "noncoding_conserved_matrix_ordered = noncoding_conserved_matrix_ordered.sort_values(by=\"sum\", ascending=False)\n",
    "print(len(set(noncoding_conserved_matrix_ordered.orfID.values.tolist())))\n",
    "\n",
    "# Display the result\n",
    "noncoding_conserved_matrix_ordered =  noncoding_conserved_matrix_ordered.merge(noncoding_conserved[['gene_name','gene_type','orfID']], on=\"orfID\")\n",
    "noncoding_conserved_matrix_ordered.drop_duplicates(inplace=True)\n",
    "\n",
    "noncoding_conserved_matrix_ordered.to_csv(os.path.join(evoDir,\"Clustering/noncodingHuman_Conserved_matrix.tsv\"), sep=\"\\t\", index=None)\n",
    "print(len(set(noncoding_conserved_matrix_ordered.orfID.values.tolist())))\n",
    "\n",
    "# Display the result\n",
    "noncoding_conserved_matrix_ordered =  noncoding_conserved_matrix_ordered.merge(noncoding_conserved[['gene_name','gene_type','orfID']], on=\"orfID\")\n",
    "noncoding_conserved_matrix_ordered.drop_duplicates(inplace=True)\n",
    "\n",
    "noncoding_conserved_matrix_ordered.to_csv(os.path.join(evoDir,\"Clustering/noncodingHuman_Conserved_matrix.tsv\"), sep=\"\\t\", index=None)\n",
    "print(len(set(noncoding_conserved_matrix_ordered.orfID.values.tolist())))\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 76,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "          orfID  presence\n",
      "specie                   \n",
      "chicken      42        42\n",
      "human      3279      3279\n",
      "macaca      116       116\n",
      "mouse        93        93\n",
      "opossum      59        59\n",
      "platypus     50        50\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>orfID</th>\n",
       "      <th>specie</th>\n",
       "      <th>presence</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>189</th>\n",
       "      <td>ENST00000416599.5:13:-|4|3494:38:398|noncoding...</td>\n",
       "      <td>macaca</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>360</th>\n",
       "      <td>ENST00000612689.1:19:+|37|2813:798:1206|noncod...</td>\n",
       "      <td>macaca</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>784</th>\n",
       "      <td>ENST00000659090.2:12:-|18|2245:444:516|noncodi...</td>\n",
       "      <td>macaca</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>786</th>\n",
       "      <td>ENST00000659090.2:12:-|16|2245:365:443|noncodi...</td>\n",
       "      <td>macaca</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1366</th>\n",
       "      <td>ENST00000399435.2:3:+|1|453:3:255|noncoding|TTG</td>\n",
       "      <td>opossum</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25978</th>\n",
       "      <td>ENST00000819284.1:1:+|11|1237:225:420|noncodin...</td>\n",
       "      <td>human</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25979</th>\n",
       "      <td>ENST00000828720.1:19:-|9|1805:71:146|noncoding...</td>\n",
       "      <td>human</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25980</th>\n",
       "      <td>ENST00000837147.1:15:+|17|1247:426:630|noncodi...</td>\n",
       "      <td>human</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25981</th>\n",
       "      <td>TCONS_00000740:13:+|17|1854:344:740|noncoding|CTG</td>\n",
       "      <td>human</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25983</th>\n",
       "      <td>TCONS_00002320:7:+|2|352:56:149|noncoding|ATG</td>\n",
       "      <td>human</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>3639 rows  3 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                   orfID   specie  presence\n",
       "189    ENST00000416599.5:13:-|4|3494:38:398|noncoding...   macaca         1\n",
       "360    ENST00000612689.1:19:+|37|2813:798:1206|noncod...   macaca         1\n",
       "784    ENST00000659090.2:12:-|18|2245:444:516|noncodi...   macaca         1\n",
       "786    ENST00000659090.2:12:-|16|2245:365:443|noncodi...   macaca         1\n",
       "1366     ENST00000399435.2:3:+|1|453:3:255|noncoding|TTG  opossum         1\n",
       "...                                                  ...      ...       ...\n",
       "25978  ENST00000819284.1:1:+|11|1237:225:420|noncodin...    human         1\n",
       "25979  ENST00000828720.1:19:-|9|1805:71:146|noncoding...    human         1\n",
       "25980  ENST00000837147.1:15:+|17|1247:426:630|noncodi...    human         1\n",
       "25981  TCONS_00000740:13:+|17|1854:344:740|noncoding|CTG    human         1\n",
       "25983      TCONS_00002320:7:+|2|352:56:149|noncoding|ATG    human         1\n",
       "\n",
       "[3639 rows x 3 columns]"
      ]
     },
     "execution_count": 76,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "### all noncoding\n",
    "noncoding_conserved_touse = noncoding_conserved\n",
    "noncoding_conserved_touse['specie'] = noncoding_conserved_touse[1].str.split(\"_\", expand=True)[0]\n",
    "# noncoding_NonConserved.drop(\"specie\",axis=1, inplace=True)\n",
    "noncoding_to_matrix_raw = pd.concat([noncoding_conserved, noncoding_NonConserved])\n",
    "noncoding_to_matrix = noncoding_to_matrix_raw[['orfID', 'specie']]\n",
    "noncoding_to_matrix = noncoding_to_matrix.fillna('human').drop_duplicates()\n",
    "noncoding_to_matrix['presence'] = 1\n",
    "print(noncoding_to_matrix.groupby(\"specie\").count())\n",
    "noncoding_to_matrix"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 77,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "3279\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "3279\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_2664244/1563675081.py:15: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  noncoding_matrix_ordered['sum'] = noncoding_matrix_ordered.select_dtypes(include='number').sum(axis=1)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>orfID</th>\n",
       "      <th>human</th>\n",
       "      <th>macaca</th>\n",
       "      <th>mouse</th>\n",
       "      <th>opossum</th>\n",
       "      <th>platypus</th>\n",
       "      <th>chicken</th>\n",
       "      <th>sum</th>\n",
       "      <th>gene_name</th>\n",
       "      <th>gene_type</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ENST00000605113.1:17:+|9|538:101:536|noncoding...</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>6.0</td>\n",
       "      <td>NDUFB8P2</td>\n",
       "      <td>processed_pseudogene</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>ENST00000406554.2:6:-|1|418:16:253|noncoding|ATG</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>6.0</td>\n",
       "      <td>ENSG00000220522</td>\n",
       "      <td>processed_pseudogene</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>ENST00000540175.1:12:+|1|1423:34:1423|noncodin...</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>6.0</td>\n",
       "      <td>ENSG00000256663</td>\n",
       "      <td>processed_pseudogene</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>ENST00000433675.1:2:-|1|447:1:448|noncoding|ATG</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>6.0</td>\n",
       "      <td>RPS15P4</td>\n",
       "      <td>processed_pseudogene</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37</th>\n",
       "      <td>ENST00000433698.2:9:-|1|843:1:685|noncoding|ATG</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>6.0</td>\n",
       "      <td>EEF1DP2</td>\n",
       "      <td>processed_pseudogene</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4069</th>\n",
       "      <td>ENST00000514947.1:5:-|23|1732:433:601|noncodin...</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>ENSG00000250080</td>\n",
       "      <td>processed_pseudogene</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4070</th>\n",
       "      <td>ENST00000515045.1:5:-|3|853:61:256|noncoding|ATG</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>ENSG00000290968</td>\n",
       "      <td>lncRNA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4072</th>\n",
       "      <td>ENST00000515049.1:17:+|13|698:192:531|noncodin...</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>ENSG00000249176</td>\n",
       "      <td>processed_pseudogene</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4073</th>\n",
       "      <td>ENST00000515049.1:17:+|43|698:572:620|noncodin...</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>ENSG00000249176</td>\n",
       "      <td>processed_pseudogene</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4074</th>\n",
       "      <td>TCONS_00002655:X:-|6|569:39:201|noncoding|CTG</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>XLOC_002421</td>\n",
       "      <td>novel</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>3279 rows  10 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                  orfID  human  macaca  mouse  \\\n",
       "0     ENST00000605113.1:17:+|9|538:101:536|noncoding...    1.0     1.0    1.0   \n",
       "7      ENST00000406554.2:6:-|1|418:16:253|noncoding|ATG    1.0     1.0    1.0   \n",
       "14    ENST00000540175.1:12:+|1|1423:34:1423|noncodin...    1.0     1.0    1.0   \n",
       "28      ENST00000433675.1:2:-|1|447:1:448|noncoding|ATG    1.0     1.0    1.0   \n",
       "37      ENST00000433698.2:9:-|1|843:1:685|noncoding|ATG    1.0     1.0    1.0   \n",
       "...                                                 ...    ...     ...    ...   \n",
       "4069  ENST00000514947.1:5:-|23|1732:433:601|noncodin...    1.0     0.0    0.0   \n",
       "4070   ENST00000515045.1:5:-|3|853:61:256|noncoding|ATG    1.0     0.0    0.0   \n",
       "4072  ENST00000515049.1:17:+|13|698:192:531|noncodin...    1.0     0.0    0.0   \n",
       "4073  ENST00000515049.1:17:+|43|698:572:620|noncodin...    1.0     0.0    0.0   \n",
       "4074      TCONS_00002655:X:-|6|569:39:201|noncoding|CTG    1.0     0.0    0.0   \n",
       "\n",
       "      opossum  platypus  chicken  sum        gene_name             gene_type  \n",
       "0         1.0       1.0      1.0  6.0         NDUFB8P2  processed_pseudogene  \n",
       "7         1.0       1.0      1.0  6.0  ENSG00000220522  processed_pseudogene  \n",
       "14        1.0       1.0      1.0  6.0  ENSG00000256663  processed_pseudogene  \n",
       "28        1.0       1.0      1.0  6.0          RPS15P4  processed_pseudogene  \n",
       "37        1.0       1.0      1.0  6.0          EEF1DP2  processed_pseudogene  \n",
       "...       ...       ...      ...  ...              ...                   ...  \n",
       "4069      0.0       0.0      0.0  1.0  ENSG00000250080  processed_pseudogene  \n",
       "4070      0.0       0.0      0.0  1.0  ENSG00000290968                lncRNA  \n",
       "4072      0.0       0.0      0.0  1.0  ENSG00000249176  processed_pseudogene  \n",
       "4073      0.0       0.0      0.0  1.0  ENSG00000249176  processed_pseudogene  \n",
       "4074      0.0       0.0      0.0  1.0      XLOC_002421                 novel  \n",
       "\n",
       "[3279 rows x 10 columns]"
      ]
     },
     "execution_count": 77,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Pivoting the data without creating an 'id', and keeping 'orfID' as the unique identifier\n",
    "noncoding_matrix = noncoding_to_matrix.pivot_table(index='orfID', columns='specie', values='presence', aggfunc='max')\n",
    "# Fill NaN values with 0\n",
    "noncoding_matrix = noncoding_matrix.fillna(0)\n",
    "# Reset index so 'orfID' becomes a column instead of an index\n",
    "noncoding_matrix = noncoding_matrix.reset_index()\n",
    "\n",
    "# Rename columns as necessary\n",
    "noncoding_matrix.columns = [\"orfID\", \"chicken\", \"human\", \"macaca\", \"mouse\", \"opossum\", \"platypus\"]\n",
    "print(len(noncoding_matrix))\n",
    "# Reorder columns for the final matrix\n",
    "noncoding_matrix_ordered = noncoding_matrix[[\"orfID\", \"human\",\"macaca\", \"mouse\", \"opossum\", \"platypus\", \"chicken\"]]\n",
    "\n",
    "# # Calculate sum across species columns to identify how many species are present\n",
    "noncoding_matrix_ordered['sum'] = noncoding_matrix_ordered.select_dtypes(include='number').sum(axis=1)\n",
    "\n",
    "# Sort by the sum column to order by the number of species present\n",
    "noncoding_matrix_ordered = noncoding_matrix_ordered.sort_values(by=\"sum\", ascending=False)\n",
    "\n",
    "# Display the result\n",
    "noncoding_matrix_ordered = noncoding_matrix_ordered.merge(noncoding_to_matrix_raw[['gene_name','gene_type','orfID']], on=\"orfID\")\n",
    "noncoding_matrix_ordered.drop_duplicates(inplace=True)\n",
    "\n",
    "noncoding_matrix_ordered.to_csv(os.path.join(evoDir,\"Clustering/noncodingHuman_matrix.tsv\"), sep=\"\\t\", index=None)\n",
    "print(len(set(noncoding_matrix_ordered.orfID.values.tolist())))\n",
    "noncoding_matrix_ordered"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 78,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "3279\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_2664244/3967721755.py:17: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  noncoding_matrix_ordered['sum'] = noncoding_matrix_ordered.select_dtypes(include='number').sum(axis=1)\n"
     ]
    }
   ],
   "source": [
    "\n",
    "# Pivoting the data without creating an 'id', and keeping 'orfID' as the unique identifier\n",
    "noncoding_matrix = noncoding_to_matrix.pivot_table(index='orfID', columns='specie', values='presence', aggfunc='max')\n",
    "noncoding_matrix\n",
    "# Fill NaN values with 0\n",
    "noncoding_matrix = noncoding_matrix.fillna(0)\n",
    "\n",
    "# Reset index so 'orfID' becomes a column instead of an index\n",
    "noncoding_matrix = noncoding_matrix.reset_index()\n",
    "\n",
    "# Rename columns as necessary\n",
    "noncoding_matrix.columns = [\"orfID\", \"chicken\", \"human\", \"macaca\", \"mouse\", \"opossum\", \"platypus\"]\n",
    "\n",
    "# Reorder columns for the final matrix\n",
    "noncoding_matrix_ordered = noncoding_matrix[[\"orfID\", \"human\",\"macaca\", \"mouse\", \"opossum\", \"platypus\", \"chicken\"]]\n",
    "\n",
    "# Calculate sum across species columns to identify how many species are present\n",
    "noncoding_matrix_ordered['sum'] = noncoding_matrix_ordered.select_dtypes(include='number').sum(axis=1)\n",
    "\n",
    "# Sort by the sum column to order by the number of species present\n",
    "noncoding_matrix_ordered = noncoding_matrix_ordered.sort_values(by=\"sum\", ascending=False)\n",
    "\n",
    "# Display the result\n",
    "noncoding_matrix_ordered =  noncoding_matrix_ordered.merge(noncoding_to_matrix_raw[['gene_name','gene_type','orfID']], on=\"orfID\")\n",
    "noncoding_matrix_ordered.drop_duplicates(inplace=True)\n",
    "\n",
    "noncoding_matrix_ordered.to_csv(os.path.join(evoDir,\"Clustering/noncodingHuman_matrix.tsv\"), sep=\"\\t\", index=None)\n",
    "print(len(set(noncoding_matrix_ordered.orfID.values.tolist())))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### alternative ORFs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 79,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Conserved altORFs human ORFs\n",
      "106\n",
      "Not-conserved altORFs human ORFs\n",
      "7369\n"
     ]
    }
   ],
   "source": [
    "# if column 1 starts with a specie, it is conserved\n",
    "altORFs_conserved = altORFs[~altORFs[1].str.startswith(\"ENST\")]\n",
    "altORFs_conserved.to_csv(os.path.join(evoDir,\"Clustering/altORFsHuman_Conserved.tsv\"), header=None, index=None, sep=\"\\t\")\n",
    "print(\"Conserved altORFs human ORFs\")\n",
    "print(len(set(altORFs_conserved[0].values.tolist())))\n",
    "altORFs_conserved = altORFs_conserved[['gene_name','gene_type',0,1]]\n",
    "altORFs_conserved.columns = ['gene_name','gene_type','orfID',1]\n",
    "\n",
    "# if column 1 starts with a human gene, it is not conserved\n",
    "altORFs_NonConserved = altORFs[altORFs[1].str.startswith(\"ENST\")]\n",
    "altORFs_NonConserved.to_csv(os.path.join(evoDir,\"Clustering/altORFsHuman_NonConserved.tsv\"), header=None, index=None, sep=\"\\t\")\n",
    "print(\"Not-conserved altORFs human ORFs\")\n",
    "print(len(set(altORFs_NonConserved[0].values.tolist())))\n",
    "altORFs_NonConserved = altORFs_NonConserved[['gene_name','gene_type',0,1]]\n",
    "altORFs_NonConserved.columns = ['gene_name','gene_type','orfID',1]\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 80,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "106\n",
      "106\n",
      "106\n",
      "106\n",
      "106\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_2664244/2466750026.py:8: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  altORFs_conserved_to_matrix['presence'] = 1\n"
     ]
    }
   ],
   "source": [
    "## conserved altORFs\n",
    "altORFs_conserved_to_matrix = altORFs_conserved\n",
    "print(len(set(altORFs_conserved_to_matrix.orfID.values.tolist())))\n",
    "# Assuming noncoding_conserved_to_matrix is already defined\n",
    "altORFs_conserved_to_matrix['specie'] = altORFs_conserved_to_matrix[1].str.split(\"_\", expand=True)[0]\n",
    "altORFs_conserved_to_matrix = altORFs_conserved_to_matrix[['orfID', 'specie']]\n",
    "altORFs_conserved_to_matrix.groupby('specie').count()\n",
    "altORFs_conserved_to_matrix['presence'] = 1\n",
    "## control\n",
    "print(len(set(altORFs_conserved_to_matrix.orfID.values.tolist())))\n",
    "\n",
    "# Pivoting the data without creating an 'id', and keeping 'orfID' as the unique identifier\n",
    "altORFs_conserved_matrix = altORFs_conserved_to_matrix.pivot_table(index='orfID', columns='specie', values='presence', aggfunc='max')\n",
    "\n",
    "# Fill NaN values with 0\n",
    "altORFs_conserved_matrix = altORFs_conserved_matrix.fillna(0)\n",
    "\n",
    "# Reset index so 'orfID' becomes a column instead of an index\n",
    "altORFs_conserved_matrix = altORFs_conserved_matrix.reset_index()\n",
    "\n",
    "# Reorder columns for the final matrix\n",
    "altORFs_conserved_matrix_ordered = altORFs_conserved_matrix[[\"orfID\", \"macaca\", \"mouse\", \"opossum\", \"platypus\", \"chicken\"]]\n",
    "\n",
    "# Calculate sum across species columns to identify how many species are present\n",
    "altORFs_conserved_matrix_ordered['sum'] = altORFs_conserved_matrix_ordered.select_dtypes(include='number').sum(axis=1)\n",
    "\n",
    "# Sort by the sum column to order by the number of species present\n",
    "altORFs_conserved_matrix_ordered = altORFs_conserved_matrix_ordered.sort_values(by=\"sum\", ascending=False)\n",
    "print(len(set(altORFs_conserved_matrix_ordered.orfID.values.tolist())))\n",
    "\n",
    "# Display the result\n",
    "altORFs_conserved_matrix_ordered =  altORFs_conserved_matrix_ordered.merge(altORFs_conserved[['gene_name','gene_type','orfID']], on=\"orfID\")\n",
    "altORFs_conserved_matrix_ordered.drop_duplicates(inplace=True)\n",
    "\n",
    "altORFs_conserved_matrix_ordered.to_csv(os.path.join(evoDir,\"Clustering/altORFsHuman_Conserved_matrix.tsv\"), sep=\"\\t\", index=None)\n",
    "print(len(set(altORFs_conserved_matrix_ordered.orfID.values.tolist())))\n",
    "\n",
    "# Display the result\n",
    "altORFs_conserved_matrix_ordered =  altORFs_conserved_matrix_ordered.merge(altORFs_conserved[['gene_name','gene_type','orfID']], on=\"orfID\")\n",
    "altORFs_conserved_matrix_ordered.drop_duplicates(inplace=True)\n",
    "\n",
    "altORFs_conserved_matrix_ordered.to_csv(os.path.join(evoDir,\"Clustering/altORFsHuman_Conserved_matrix.tsv\"), sep=\"\\t\", index=None)\n",
    "print(len(set(altORFs_conserved_matrix_ordered.orfID.values.tolist())))\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 81,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "          orfID  presence\n",
      "specie                   \n",
      "chicken       8         8\n",
      "human      7371      7371\n",
      "macaca       44        44\n",
      "mouse        47        47\n",
      "opossum      25        25\n",
      "platypus     13        13\n"
     ]
    }
   ],
   "source": [
    "### all altORFs\n",
    "altORFs_conserved_touse = altORFs_conserved\n",
    "altORFs_conserved_touse['specie'] = altORFs_conserved_touse[1].str.split(\"_\", expand=True)[0]\n",
    "altORFs_to_matrix_raw = pd.concat([altORFs_conserved, altORFs_NonConserved])\n",
    "altORFs_to_matrix = altORFs_to_matrix_raw[['orfID', 'specie']]\n",
    "altORFs_to_matrix = altORFs_to_matrix.fillna('human').drop_duplicates()\n",
    "altORFs_to_matrix['presence'] = 1\n",
    "altORFs_to_matrix = altORFs_to_matrix.replace(\"TCONS\", \"human\")\n",
    "print(altORFs_to_matrix.groupby(\"specie\").count())\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 82,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "7369\n",
      "7369\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_2664244/70459802.py:15: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  altORFs_matrix_ordered['sum'] = altORFs_matrix_ordered.select_dtypes(include='number').sum(axis=1)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>orfID</th>\n",
       "      <th>human</th>\n",
       "      <th>macaca</th>\n",
       "      <th>mouse</th>\n",
       "      <th>opossum</th>\n",
       "      <th>platypus</th>\n",
       "      <th>chicken</th>\n",
       "      <th>sum</th>\n",
       "      <th>gene_name</th>\n",
       "      <th>gene_type</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ENST00000618666.4:12:-|46|6915:1015:2341|dORF|ATG</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>6.0</td>\n",
       "      <td>CCNT1</td>\n",
       "      <td>protein_coding</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>ENST00000532219.5:5:+|456|8246:7706:7940|odORF...</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>6.0</td>\n",
       "      <td>ANKHD1-EIF4EBP3</td>\n",
       "      <td>protein_coding</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>ENST00000619721.6:11:-|203|3949:2829:3561|dORF...</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>6.0</td>\n",
       "      <td>MFRP</td>\n",
       "      <td>protein_coding</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>ENST00000695752.1:1:+|41|1548:541:1222|odORF|ATG</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>C1QC</td>\n",
       "      <td>protein_coding</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>ENST00000643349.2:11:-|42|2861:727:1270|dORF|ATG</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>ENSG00000284779</td>\n",
       "      <td>protein_coding</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7700</th>\n",
       "      <td>ENST00000339697.5:6:-|159|3125:2425:2530|dORF|CTG</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>SERINC1</td>\n",
       "      <td>protein_coding</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7701</th>\n",
       "      <td>ENST00000339656.8:19:+|3|6885:22:76|uORF|ATG</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>ZNF587</td>\n",
       "      <td>protein_coding</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7702</th>\n",
       "      <td>ENST00000339656.8:19:+|113|6885:1827:2277|odOR...</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>ZNF587</td>\n",
       "      <td>protein_coding</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7703</th>\n",
       "      <td>ENST00000339618.8:17:+|109|3828:1745:1805|odOR...</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>ALDH3A2</td>\n",
       "      <td>protein_coding</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7704</th>\n",
       "      <td>ENST00000850623.1:11:-|3|8048:24:234|ouORF|ATG</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NPAT</td>\n",
       "      <td>protein_coding</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>7369 rows  10 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                  orfID  human  macaca  mouse  \\\n",
       "0     ENST00000618666.4:12:-|46|6915:1015:2341|dORF|ATG    1.0     1.0    1.0   \n",
       "6     ENST00000532219.5:5:+|456|8246:7706:7940|odORF...    1.0     1.0    1.0   \n",
       "19    ENST00000619721.6:11:-|203|3949:2829:3561|dORF...    1.0     1.0    1.0   \n",
       "26     ENST00000695752.1:1:+|41|1548:541:1222|odORF|ATG    1.0     1.0    1.0   \n",
       "31     ENST00000643349.2:11:-|42|2861:727:1270|dORF|ATG    1.0     1.0    1.0   \n",
       "...                                                 ...    ...     ...    ...   \n",
       "7700  ENST00000339697.5:6:-|159|3125:2425:2530|dORF|CTG    1.0     0.0    0.0   \n",
       "7701       ENST00000339656.8:19:+|3|6885:22:76|uORF|ATG    1.0     0.0    0.0   \n",
       "7702  ENST00000339656.8:19:+|113|6885:1827:2277|odOR...    1.0     0.0    0.0   \n",
       "7703  ENST00000339618.8:17:+|109|3828:1745:1805|odOR...    1.0     0.0    0.0   \n",
       "7704     ENST00000850623.1:11:-|3|8048:24:234|ouORF|ATG    1.0     0.0    0.0   \n",
       "\n",
       "      opossum  platypus  chicken  sum        gene_name       gene_type  \n",
       "0         1.0       1.0      1.0  6.0            CCNT1  protein_coding  \n",
       "6         1.0       1.0      1.0  6.0  ANKHD1-EIF4EBP3  protein_coding  \n",
       "19        1.0       1.0      1.0  6.0             MFRP  protein_coding  \n",
       "26        1.0       0.0      1.0  5.0             C1QC  protein_coding  \n",
       "31        0.0       1.0      1.0  5.0  ENSG00000284779  protein_coding  \n",
       "...       ...       ...      ...  ...              ...             ...  \n",
       "7700      0.0       0.0      0.0  1.0          SERINC1  protein_coding  \n",
       "7701      0.0       0.0      0.0  1.0           ZNF587  protein_coding  \n",
       "7702      0.0       0.0      0.0  1.0           ZNF587  protein_coding  \n",
       "7703      0.0       0.0      0.0  1.0          ALDH3A2  protein_coding  \n",
       "7704      0.0       0.0      0.0  1.0             NPAT  protein_coding  \n",
       "\n",
       "[7369 rows x 10 columns]"
      ]
     },
     "execution_count": 82,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Pivoting the data without creating an 'id', and keeping 'orfID' as the unique identifier\n",
    "altORFs_matrix = altORFs_to_matrix.pivot_table(index='orfID', columns='specie', values='presence', aggfunc='max')\n",
    "# Fill NaN values with 0\n",
    "altORFs_matrix = altORFs_matrix.fillna(0)\n",
    "# Reset index so 'orfID' becomes a column instead of an index\n",
    "altORFs_matrix = altORFs_matrix.reset_index()\n",
    "altORFs_matrix\n",
    "# Rename columns as necessary\n",
    "altORFs_matrix.columns = [\"orfID\", \"chicken\", \"human\", \"macaca\", \"mouse\", \"opossum\", \"platypus\"]\n",
    "print(len(altORFs_matrix))\n",
    "# Reorder columns for the final matrix\n",
    "altORFs_matrix_ordered = altORFs_matrix[[\"orfID\", \"human\",\"macaca\", \"mouse\", \"opossum\", \"platypus\", \"chicken\"]]\n",
    "\n",
    "# # Calculate sum across species columns to identify how many species are present\n",
    "altORFs_matrix_ordered['sum'] = altORFs_matrix_ordered.select_dtypes(include='number').sum(axis=1)\n",
    "\n",
    "# Sort by the sum column to order by the number of species present\n",
    "altORFs_matrix_ordered = altORFs_matrix_ordered.sort_values(by=\"sum\", ascending=False)\n",
    "\n",
    "# Display the result\n",
    "altORFs_matrix_ordered = altORFs_matrix_ordered.merge(altORFs_to_matrix_raw[['gene_name','gene_type','orfID']], on=\"orfID\")\n",
    "altORFs_matrix_ordered.drop_duplicates(inplace=True)\n",
    "\n",
    "altORFs_matrix_ordered.to_csv(os.path.join(evoDir,\"Clustering/altORFsHuman_matrix.tsv\"), sep=\"\\t\", index=None)\n",
    "print(len(set(altORFs_matrix_ordered.orfID.values.tolist())))\n",
    "altORFs_matrix_ordered"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "\n",
    "### CANONICAL"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 83,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Conserved canonical human ORFs\n",
      "2585\n",
      "AltORFs noncoding human ORFs\n",
      "2899\n"
     ]
    }
   ],
   "source": [
    "# if column 1 starts with a specie, it is conserved\n",
    "canonical_conserved = canonical[~canonical[1].str.startswith(\"ENST\")]\n",
    "canonical_conserved.to_csv(os.path.join(evoDir,\"Clustering/canonicalHuman_Conserved.tsv\"), header=None, index=None, sep=\"\\t\")\n",
    "print(\"Conserved canonical human ORFs\")\n",
    "print(len(set(canonical_conserved[0].values.tolist())))\n",
    "canonical_conserved = canonical_conserved[['gene_name','gene_type',0,1]]\n",
    "canonical_conserved.columns = ['gene_name','gene_type','orfID',1]\n",
    "\n",
    "# if column 1 starts with a human gene, it is not conserved\n",
    "canonical_NonConserved = canonical[canonical[1].str.startswith(\"ENST\")]\n",
    "canonical_NonConserved.to_csv(os.path.join(evoDir,\"Clustering/canonicalHuman_NonConserved.tsv\"), header=None, index=None, sep=\"\\t\")\n",
    "print(\"AltORFs noncoding human ORFs\")\n",
    "print(len(set(canonical_NonConserved[0].values.tolist())))\n",
    "canonical_NonConserved = canonical_NonConserved[['gene_name','gene_type',0,1]]\n",
    "canonical_NonConserved.columns = ['gene_name','gene_type','orfID',1]\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 84,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2585\n",
      "2585\n",
      "2585\n",
      "2585\n",
      "2585\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_2664244/3948074870.py:8: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  canonical_conserved_to_matrix['presence'] = 1\n",
      "/tmp/ipykernel_2664244/3948074870.py:25: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  canonical_conserved_matrix_ordered['sum'] = canonical_conserved_matrix_ordered.select_dtypes(include='number').sum(axis=1)\n"
     ]
    }
   ],
   "source": [
    "## conserved canonical\n",
    "canonical_conserved_to_matrix = canonical_conserved\n",
    "print(len(set(canonical_conserved_to_matrix.orfID.values.tolist())))\n",
    "# Assuming noncoding_conserved_to_matrix is already defined\n",
    "canonical_conserved_to_matrix['specie'] = canonical_conserved_to_matrix[1].str.split(\"_\", expand=True)[0]\n",
    "canonical_conserved_to_matrix = canonical_conserved_to_matrix[['orfID', 'specie']]\n",
    "canonical_conserved_to_matrix.groupby('specie').count()\n",
    "canonical_conserved_to_matrix['presence'] = 1\n",
    "## control\n",
    "print(len(set(canonical_conserved_to_matrix.orfID.values.tolist())))\n",
    "\n",
    "# Pivoting the data without creating an 'id', and keeping 'orfID' as the unique identifier\n",
    "canonical_conserved_matrix = canonical_conserved_to_matrix.pivot_table(index='orfID', columns='specie', values='presence', aggfunc='max')\n",
    "\n",
    "# Fill NaN values with 0\n",
    "canonical_conserved_matrix = canonical_conserved_matrix.fillna(0)\n",
    "\n",
    "# Reset index so 'orfID' becomes a column instead of an index\n",
    "canonical_conserved_matrix = canonical_conserved_matrix.reset_index()\n",
    "\n",
    "# Reorder columns for the final matrix\n",
    "canonical_conserved_matrix_ordered = canonical_conserved_matrix[[\"orfID\", \"macaca\", \"mouse\", \"opossum\", \"platypus\", \"chicken\"]]\n",
    "\n",
    "# Calculate sum across species columns to identify how many species are present\n",
    "canonical_conserved_matrix_ordered['sum'] = canonical_conserved_matrix_ordered.select_dtypes(include='number').sum(axis=1)\n",
    "\n",
    "# Sort by the sum column to order by the number of species present\n",
    "canonical_conserved_matrix_ordered = canonical_conserved_matrix_ordered.sort_values(by=\"sum\", ascending=False)\n",
    "print(len(set(canonical_conserved_matrix_ordered.orfID.values.tolist())))\n",
    "\n",
    "# Display the result\n",
    "canonical_conserved_matrix_ordered =  canonical_conserved_matrix_ordered.merge(canonical_conserved[['gene_name','gene_type','orfID']], on=\"orfID\")\n",
    "canonical_conserved_matrix_ordered.drop_duplicates(inplace=True)\n",
    "\n",
    "canonical_conserved_matrix_ordered.to_csv(os.path.join(evoDir,\"Clustering/canonicalHuman_Conserved_matrix.tsv\"), sep=\"\\t\", index=None)\n",
    "print(len(set(canonical_conserved_matrix_ordered.orfID.values.tolist())))\n",
    "\n",
    "# Display the result\n",
    "canonical_conserved_matrix_ordered =  canonical_conserved_matrix_ordered.merge(canonical_conserved[['gene_name','gene_type','orfID']], on=\"orfID\")\n",
    "canonical_conserved_matrix_ordered.drop_duplicates(inplace=True)\n",
    "\n",
    "canonical_conserved_matrix_ordered.to_csv(os.path.join(evoDir,\"Clustering/canonicalHuman_Conserved_matrix.tsv\"), sep=\"\\t\", index=None)\n",
    "print(len(set(canonical_conserved_matrix_ordered.orfID.values.tolist())))\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 85,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "          orfID  presence\n",
      "specie                   \n",
      "chicken    1440      1440\n",
      "human      2899      2899\n",
      "macaca     2316      2316\n",
      "mouse      1954      1954\n",
      "opossum    1610      1610\n",
      "platypus   1473      1473\n"
     ]
    }
   ],
   "source": [
    "### all canonical\n",
    "canonical_conserved_touse = canonical_conserved\n",
    "canonical_conserved_touse['specie'] = canonical_conserved_touse[1].str.split(\"_\", expand=True)[0]\n",
    "canonical_to_matrix_raw = pd.concat([canonical_conserved, canonical_NonConserved])\n",
    "canonical_to_matrix = canonical_to_matrix_raw[['orfID', 'specie']]\n",
    "canonical_to_matrix = canonical_to_matrix.fillna('human').drop_duplicates()\n",
    "canonical_to_matrix['presence'] = 1\n",
    "canonical_to_matrix = canonical_to_matrix.replace(\"TCONS\", \"human\")\n",
    "print(canonical_to_matrix.groupby(\"specie\").count())\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 86,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2899\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_2664244/1133186206.py:15: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  canonical_matrix_ordered['sum'] = canonical_matrix_ordered.select_dtypes(include='number').sum(axis=1)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2899\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>orfID</th>\n",
       "      <th>human</th>\n",
       "      <th>macaca</th>\n",
       "      <th>mouse</th>\n",
       "      <th>opossum</th>\n",
       "      <th>platypus</th>\n",
       "      <th>chicken</th>\n",
       "      <th>sum</th>\n",
       "      <th>gene_name</th>\n",
       "      <th>gene_type</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ENST00000440944.6:9:+|1|3566:158:2186|canonica...</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>6.0</td>\n",
       "      <td>GPSM1</td>\n",
       "      <td>protein_coding</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>ENST00000334701.11:14:-|9|3510:283:2848|canoni...</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>6.0</td>\n",
       "      <td>HSP90AA1</td>\n",
       "      <td>protein_coding</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>ENST00000451261.7:X:+|24|3485:439:2335|canonic...</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>6.0</td>\n",
       "      <td>DCAF8L2</td>\n",
       "      <td>protein_coding</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>34</th>\n",
       "      <td>ENST00000335025.12:22:+|6|4929:92:1655|canonic...</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>6.0</td>\n",
       "      <td>PPIL2</td>\n",
       "      <td>protein_coding</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>40</th>\n",
       "      <td>ENST00000449105.8:2:-|7|4144:303:1932|canonica...</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>6.0</td>\n",
       "      <td>HNRNPLL</td>\n",
       "      <td>protein_coding</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14196</th>\n",
       "      <td>ENST00000370518.4:X:+|5|408:61:355|canonical|ATG</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>SPANXA2</td>\n",
       "      <td>protein_coding</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14200</th>\n",
       "      <td>ENST00000488965.1:1:+|2|3926:48:228|canonical|ATG</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>SCP2</td>\n",
       "      <td>protein_coding</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14201</th>\n",
       "      <td>ENST00000432920.2:1:-|6|1625:169:1249|canonica...</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>ENSG00000255835</td>\n",
       "      <td>protein_coding</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14202</th>\n",
       "      <td>ENST00000240316.5:17:-|2|2634:32:1763|canonica...</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>COIL</td>\n",
       "      <td>protein_coding</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14203</th>\n",
       "      <td>ENST00000252979.6:20:-|6|4237:145:2401|canonic...</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>ZNF337</td>\n",
       "      <td>protein_coding</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2899 rows  10 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                   orfID  human  macaca  \\\n",
       "0      ENST00000440944.6:9:+|1|3566:158:2186|canonica...    1.0     1.0   \n",
       "11     ENST00000334701.11:14:-|9|3510:283:2848|canoni...    1.0     1.0   \n",
       "23     ENST00000451261.7:X:+|24|3485:439:2335|canonic...    1.0     1.0   \n",
       "34     ENST00000335025.12:22:+|6|4929:92:1655|canonic...    1.0     1.0   \n",
       "40     ENST00000449105.8:2:-|7|4144:303:1932|canonica...    1.0     1.0   \n",
       "...                                                  ...    ...     ...   \n",
       "14196   ENST00000370518.4:X:+|5|408:61:355|canonical|ATG    1.0     0.0   \n",
       "14200  ENST00000488965.1:1:+|2|3926:48:228|canonical|ATG    1.0     0.0   \n",
       "14201  ENST00000432920.2:1:-|6|1625:169:1249|canonica...    1.0     0.0   \n",
       "14202  ENST00000240316.5:17:-|2|2634:32:1763|canonica...    1.0     0.0   \n",
       "14203  ENST00000252979.6:20:-|6|4237:145:2401|canonic...    1.0     0.0   \n",
       "\n",
       "       mouse  opossum  platypus  chicken  sum        gene_name       gene_type  \n",
       "0        1.0      1.0       1.0      1.0  6.0            GPSM1  protein_coding  \n",
       "11       1.0      1.0       1.0      1.0  6.0         HSP90AA1  protein_coding  \n",
       "23       1.0      1.0       1.0      1.0  6.0          DCAF8L2  protein_coding  \n",
       "34       1.0      1.0       1.0      1.0  6.0            PPIL2  protein_coding  \n",
       "40       1.0      1.0       1.0      1.0  6.0          HNRNPLL  protein_coding  \n",
       "...      ...      ...       ...      ...  ...              ...             ...  \n",
       "14196    0.0      0.0       0.0      0.0  1.0          SPANXA2  protein_coding  \n",
       "14200    0.0      0.0       0.0      0.0  1.0             SCP2  protein_coding  \n",
       "14201    0.0      0.0       0.0      0.0  1.0  ENSG00000255835  protein_coding  \n",
       "14202    0.0      0.0       0.0      0.0  1.0             COIL  protein_coding  \n",
       "14203    0.0      0.0       0.0      0.0  1.0           ZNF337  protein_coding  \n",
       "\n",
       "[2899 rows x 10 columns]"
      ]
     },
     "execution_count": 86,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Pivoting the data without creating an 'id', and keeping 'orfID' as the unique identifier\n",
    "canonical_matrix = canonical_to_matrix.pivot_table(index='orfID', columns='specie', values='presence', aggfunc='max')\n",
    "# Fill NaN values with 0\n",
    "canonical_matrix = canonical_matrix.fillna(0)\n",
    "# Reset index so 'orfID' becomes a column instead of an index\n",
    "canonical_matrix = canonical_matrix.reset_index()\n",
    "canonical_matrix\n",
    "# Rename columns as necessary\n",
    "canonical_matrix.columns = [\"orfID\", \"chicken\", \"human\", \"macaca\", \"mouse\", \"opossum\", \"platypus\"]\n",
    "print(len(canonical_matrix))\n",
    "# Reorder columns for the final matrix\n",
    "canonical_matrix_ordered = canonical_matrix[[\"orfID\", \"human\",\"macaca\", \"mouse\", \"opossum\", \"platypus\", \"chicken\"]]\n",
    "\n",
    "# # Calculate sum across species columns to identify how many species are present\n",
    "canonical_matrix_ordered['sum'] = canonical_matrix_ordered.select_dtypes(include='number').sum(axis=1)\n",
    "\n",
    "# Sort by the sum column to order by the number of species present\n",
    "canonical_matrix_ordered = canonical_matrix_ordered.sort_values(by=\"sum\", ascending=False)\n",
    "\n",
    "# Display the result\n",
    "canonical_matrix_ordered = canonical_matrix_ordered.merge(canonical_to_matrix_raw[['gene_name','gene_type','orfID']], on=\"orfID\")\n",
    "canonical_matrix_ordered.drop_duplicates(inplace=True)\n",
    "\n",
    "canonical_matrix_ordered.to_csv(os.path.join(evoDir,\"Clustering/canonicalHuman_matrix.tsv\"), sep=\"\\t\", index=None)\n",
    "print(len(set(canonical_matrix_ordered.orfID.values.tolist())))\n",
    "canonical_matrix_ordered"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Find the oldest specie containg the ORF translated"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 87,
   "metadata": {},
   "outputs": [],
   "source": [
    "species_ordered = ['chicken','platypus','opossum','mouse','macaca','human']\n",
    "def find_last_species(row):\n",
    "    for sp in species_ordered:\n",
    "        if row[sp] == 1.0:  # Check if the value is 1.0\n",
    "            return sp\n",
    "    return None  # Return None if no species has a 1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 88,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Apply the function to each row to create a new column 'oldest_species'\n",
    "canonical_matrix_ordered['oldest_species'] = canonical_matrix_ordered.apply(find_last_species, axis=1)\n",
    "canonical_matrix_ordered = species_and_years.merge(canonical_matrix_ordered, left_on=[\"specie\"], right_on=\"oldest_species\")\n",
    "canonical_matrix_ordered[['orfID','gene_name','gene_type','oldest_species','age']].to_csv(os.path.join(evoDir,\"Clustering/canonicalHuman_matrix_oldestAge.tsv\"), sep=\"\\t\", index=None)\n",
    "\n",
    "noncoding_matrix_ordered['oldest_species'] = noncoding_matrix_ordered.apply(find_last_species, axis=1)\n",
    "noncoding_matrix_ordered = species_and_years.merge(noncoding_matrix_ordered, left_on=[\"specie\"], right_on=\"oldest_species\")\n",
    "noncoding_matrix_ordered[['orfID','gene_name','gene_type','oldest_species','age']].to_csv(os.path.join(evoDir,\"Clustering/noncodingHuman_matrix_oldestAge.tsv\"), sep=\"\\t\", index=None)\n",
    "\n",
    "\n",
    "altORFs_matrix_ordered['oldest_species'] = altORFs_matrix_ordered.apply(find_last_species, axis=1)\n",
    "altORFs_matrix_ordered = species_and_years.merge(altORFs_matrix_ordered, left_on=[\"specie\"], right_on=\"oldest_species\")\n",
    "altORFs_matrix_ordered[['orfID','gene_name','gene_type','oldest_species','age']].to_csv(os.path.join(evoDir,\"Clustering/altORFsHuman_matrix_oldestAge.tsv\"), sep=\"\\t\", index=None)\n",
    "\n",
    "\n",
    "complete_matrix_age = pd.concat([noncoding_matrix_ordered[['orfID','gene_name','gene_type','oldest_species','age']],altORFs_matrix_ordered[['orfID','gene_name','gene_type','oldest_species','age']], canonical_matrix_ordered[['orfID','gene_name','gene_type','oldest_species','age']]])\n",
    "complete_matrix_age.to_csv(os.path.join(evoDir,\"Clustering/allHuman_matrix_oldestAge.tsv\"), sep=\"\\t\", index=None)\n"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.10"
  },
  "orig_nbformat": 4
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
