{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "import os,re\n",
    "import pandas as pd\n",
    "from collections import Counter\n",
    "from rna_seq_normalization import Normalization as Norm\n",
    "from functools import reduce\n",
    "\n",
    "annot = pd.read_csv(\"/users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/human/newReference_Resconstructed/transID_geneID_isoforms_selected.1to1.csv\")\n",
    "Ref_DIR = \"/users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/human/newReference_Resconstructed\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash -s \"$Ref_DIR\"\n",
    "\n",
    "\n",
    "sed 's/\"\"/\"/g' $1/gencode.v47.gffcompare.TestisLiverBrain.annotation.sorted.1transcript.sorted.NOchr.gtf > test.gtf\n",
    "sed 's/\\(transcript_id \"[^\"]*\\).*/\\1\"/' test.gtf > $1/gencode.v47.gffcompare.TestisLiverBrain.annotation.sorted.1transcript.sorted.NOchr.fixed.gtf\n",
    "rm test.gtf"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Quantify in thymus"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "        ==========     _____ _    _ ____  _____  ______          _____  \n",
      "        =====         / ____| |  | |  _ \\|  __ \\|  ____|   /\\   |  __ \\ \n",
      "          =====      | (___ | |  | | |_) | |__) | |__     /  \\  | |  | |\n",
      "            ====      \\___ \\| |  | |  _ <|  _  /|  __|   / /\\ \\ | |  | |\n",
      "              ====    ____) | |__| | |_) | | \\ \\| |____ / ____ \\| |__| |\n",
      "        ==========   |_____/ \\____/|____/|_|  \\_\\______/_/    \\_\\_____/\n",
      "\t  v2.0.3\n",
      "\n",
      "//========================== featureCounts setting ===========================\\\\\n",
      "||                                                                            ||\n",
      "||             Input files : 30 BAM files                                     ||\n",
      "||                                                                            ||\n",
      "||                           SRR8668611Aligned.sortedByCoord.out.bam          ||\n",
      "||                           SRR8668612Aligned.sortedByCoord.out.bam          ||\n",
      "||                           SRR8668613Aligned.sortedByCoord.out.bam          ||\n",
      "||                           SRR8668614Aligned.sortedByCoord.out.bam          ||\n",
      "||                           SRR8668615Aligned.sortedByCoord.out.bam          ||\n",
      "||                           SRR8668616Aligned.sortedByCoord.out.bam          ||\n",
      "||                           SRR8668617Aligned.sortedByCoord.out.bam          ||\n",
      "||                           SRR8668618Aligned.sortedByCoord.out.bam          ||\n",
      "||                           SRR8668619Aligned.sortedByCoord.out.bam          ||\n",
      "||                           SRR8668620Aligned.sortedByCoord.out.bam          ||\n",
      "||                           SRR8668621Aligned.sortedByCoord.out.bam          ||\n",
      "||                           SRR8668622Aligned.sortedByCoord.out.bam          ||\n",
      "||                           SRR8668623Aligned.sortedByCoord.out.bam          ||\n",
      "||                           SRR8668624Aligned.sortedByCoord.out.bam          ||\n",
      "||                           SRR8668625Aligned.sortedByCoord.out.bam          ||\n",
      "||                           SRR8668626Aligned.sortedByCoord.out.bam          ||\n",
      "||                           SRR8668627Aligned.sortedByCoord.out.bam          ||\n",
      "||                           SRR8668628Aligned.sortedByCoord.out.bam          ||\n",
      "||                           SRR8668629Aligned.sortedByCoord.out.bam          ||\n",
      "||                           SRR8668630Aligned.sortedByCoord.out.bam          ||\n",
      "||                           SRR8668631Aligned.sortedByCoord.out.bam          ||\n",
      "||                           SRR8668632Aligned.sortedByCoord.out.bam          ||\n",
      "||                           SRR8668633Aligned.sortedByCoord.out.bam          ||\n",
      "||                           SRR8668634Aligned.sortedByCoord.out.bam          ||\n",
      "||                           SRR18941362Aligned.sortedByCoord.out.bam         ||\n",
      "||                           SRR18941363Aligned.sortedByCoord.out.bam         ||\n",
      "||                           SRR18941364Aligned.sortedByCoord.out.bam         ||\n",
      "||                           SRR18941365Aligned.sortedByCoord.out.bam         ||\n",
      "||                           SRR18941366Aligned.sortedByCoord.out.bam         ||\n",
      "||                           SRR18941367Aligned.sortedByCoord.out.bam         ||\n",
      "||                                                                            ||\n",
      "||             Output file : gffcompare_stranded_featureCounts.txt            ||\n",
      "||                 Summary : gffcompare_stranded_featureCounts.txt.summary    ||\n",
      "||              Paired-end : yes                                              ||\n",
      "||        Count read pairs : yes                                              ||\n",
      "||              Annotation : gencode.v47.gffcompare.TestisLiverBrain.anno ... ||\n",
      "||      Dir for temp files : /users/genomics/marta/TestisProject_SaraRazq ... ||\n",
      "||                                                                            ||\n",
      "||                 Threads : 10                                               ||\n",
      "||                   Level : meta-feature level                               ||\n",
      "||      Multimapping reads : not counted                                      ||\n",
      "|| Multi-overlapping reads : counted                                          ||\n",
      "||   Min overlapping bases : 1                                                ||\n",
      "||                                                                            ||\n",
      "\\\\============================================================================//\n",
      "\n",
      "//================================= Running ==================================\\\\\n",
      "||                                                                            ||\n",
      "|| Load annotation file gencode.v47.gffcompare.TestisLiverBrain.annotatio ... ||\n",
      "||    Features : 341348                                                       ||\n",
      "||    Meta-features : 81010                                                   ||\n",
      "||    Chromosomes/contigs : 74                                                ||\n",
      "||                                                                            ||\n",
      "|| Process BAM file SRR8668611Aligned.sortedByCoord.out.bam...                ||\n",
      "||    Strand specific : reversely stranded                                    ||\n",
      "||    Paired-end reads are included.                                          ||\n",
      "||    Total alignments : 14323394                                             ||\n",
      "||    Successfully assigned alignments : 11346345 (79.2%)                     ||\n",
      "||    Running time : 0.33 minutes                                             ||\n",
      "||                                                                            ||\n",
      "|| Process BAM file SRR8668612Aligned.sortedByCoord.out.bam...                ||\n",
      "||    Strand specific : reversely stranded                                    ||\n",
      "||    Paired-end reads are included.                                          ||\n",
      "||    Total alignments : 14290397                                             ||\n",
      "||    Successfully assigned alignments : 11328200 (79.3%)                     ||\n",
      "||    Running time : 0.34 minutes                                             ||\n",
      "||                                                                            ||\n",
      "|| Process BAM file SRR8668613Aligned.sortedByCoord.out.bam...                ||\n",
      "||    Strand specific : reversely stranded                                    ||\n",
      "||    Paired-end reads are included.                                          ||\n",
      "||    Total alignments : 14669485                                             ||\n",
      "||    Successfully assigned alignments : 11625982 (79.3%)                     ||\n",
      "||    Running time : 0.36 minutes                                             ||\n",
      "||                                                                            ||\n",
      "|| Process BAM file SRR8668614Aligned.sortedByCoord.out.bam...                ||\n",
      "||    Strand specific : reversely stranded                                    ||\n",
      "||    Paired-end reads are included.                                          ||\n",
      "||    Total alignments : 14241669                                             ||\n",
      "||    Successfully assigned alignments : 11291261 (79.3%)                     ||\n",
      "||    Running time : 0.40 minutes                                             ||\n",
      "||                                                                            ||\n",
      "|| Process BAM file SRR8668615Aligned.sortedByCoord.out.bam...                ||\n",
      "||    Strand specific : reversely stranded                                    ||\n",
      "||    Paired-end reads are included.                                          ||\n",
      "||    Total alignments : 13118328                                             ||\n",
      "||    Successfully assigned alignments : 10317083 (78.6%)                     ||\n",
      "||    Running time : 1.33 minutes                                             ||\n",
      "||                                                                            ||\n",
      "|| Process BAM file SRR8668616Aligned.sortedByCoord.out.bam...                ||\n",
      "||    Strand specific : reversely stranded                                    ||\n",
      "||    Paired-end reads are included.                                          ||\n",
      "||    Total alignments : 13008581                                             ||\n",
      "||    Successfully assigned alignments : 10233092 (78.7%)                     ||\n",
      "||    Running time : 1.15 minutes                                             ||\n",
      "||                                                                            ||\n",
      "|| Process BAM file SRR8668617Aligned.sortedByCoord.out.bam...                ||\n",
      "||    Strand specific : reversely stranded                                    ||\n",
      "||    Paired-end reads are included.                                          ||\n",
      "||    Total alignments : 13429055                                             ||\n",
      "||    Successfully assigned alignments : 10566930 (78.7%)                     ||\n",
      "||    Running time : 0.49 minutes                                             ||\n",
      "||                                                                            ||\n",
      "|| Process BAM file SRR8668618Aligned.sortedByCoord.out.bam...                ||\n",
      "||    Strand specific : reversely stranded                                    ||\n",
      "||    Paired-end reads are included.                                          ||\n",
      "||    Total alignments : 13007491                                             ||\n",
      "||    Successfully assigned alignments : 10235794 (78.7%)                     ||\n",
      "||    Running time : 0.48 minutes                                             ||\n",
      "||                                                                            ||\n",
      "|| Process BAM file SRR8668619Aligned.sortedByCoord.out.bam...                ||\n",
      "||    Strand specific : reversely stranded                                    ||\n",
      "||    Paired-end reads are included.                                          ||\n",
      "||    Total alignments : 20222557                                             ||\n",
      "||    Successfully assigned alignments : 16775751 (83.0%)                     ||\n",
      "||    Running time : 1.88 minutes                                             ||\n",
      "||                                                                            ||\n",
      "|| Process BAM file SRR8668620Aligned.sortedByCoord.out.bam...                ||\n",
      "||    Strand specific : reversely stranded                                    ||\n",
      "||    Paired-end reads are included.                                          ||\n",
      "||    Total alignments : 20104349                                             ||\n",
      "||    Successfully assigned alignments : 16692231 (83.0%)                     ||\n",
      "||    Running time : 1.52 minutes                                             ||\n",
      "||                                                                            ||\n",
      "|| Process BAM file SRR8668621Aligned.sortedByCoord.out.bam...                ||\n",
      "||    Strand specific : reversely stranded                                    ||\n",
      "||    Paired-end reads are included.                                          ||\n",
      "||    Total alignments : 20273457                                             ||\n",
      "||    Successfully assigned alignments : 16828561 (83.0%)                     ||\n",
      "||    Running time : 2.38 minutes                                             ||\n",
      "||                                                                            ||\n",
      "|| Process BAM file SRR8668622Aligned.sortedByCoord.out.bam...                ||\n",
      "||    Strand specific : reversely stranded                                    ||\n",
      "||    Paired-end reads are included.                                          ||\n",
      "||    Total alignments : 19991944                                             ||\n",
      "||    Successfully assigned alignments : 16591151 (83.0%)                     ||\n",
      "||    Running time : 1.40 minutes                                             ||\n",
      "||                                                                            ||\n",
      "|| Process BAM file SRR8668623Aligned.sortedByCoord.out.bam...                ||\n",
      "||    Strand specific : reversely stranded                                    ||\n",
      "||    Paired-end reads are included.                                          ||\n",
      "||    Total alignments : 25424739                                             ||\n",
      "||    Successfully assigned alignments : 21716189 (85.4%)                     ||\n",
      "||    Running time : 3.76 minutes                                             ||\n",
      "||                                                                            ||\n",
      "|| Process BAM file SRR8668624Aligned.sortedByCoord.out.bam...                ||\n",
      "||    Strand specific : reversely stranded                                    ||\n",
      "||    Paired-end reads are included.                                          ||\n",
      "||    Total alignments : 25248963                                             ||\n",
      "||    Successfully assigned alignments : 21579662 (85.5%)                     ||\n",
      "||    Running time : 5.06 minutes                                             ||\n",
      "||                                                                            ||\n",
      "|| Process BAM file SRR8668625Aligned.sortedByCoord.out.bam...                ||\n",
      "||    Strand specific : reversely stranded                                    ||\n",
      "||    Paired-end reads are included.                                          ||\n",
      "||    Total alignments : 25466756                                             ||\n",
      "||    Successfully assigned alignments : 21768466 (85.5%)                     ||\n",
      "||    Running time : 5.73 minutes                                             ||\n",
      "||                                                                            ||\n",
      "|| Process BAM file SRR8668626Aligned.sortedByCoord.out.bam...                ||\n",
      "||    Strand specific : reversely stranded                                    ||\n",
      "||    Paired-end reads are included.                                          ||\n",
      "||    Total alignments : 25101355                                             ||\n",
      "||    Successfully assigned alignments : 21448430 (85.4%)                     ||\n",
      "||    Running time : 5.94 minutes                                             ||\n",
      "||                                                                            ||\n",
      "|| Process BAM file SRR8668627Aligned.sortedByCoord.out.bam...                ||\n",
      "||    Strand specific : reversely stranded                                    ||\n",
      "||    Paired-end reads are included.                                          ||\n",
      "||    Total alignments : 14335170                                             ||\n",
      "||    Successfully assigned alignments : 12464555 (87.0%)                     ||\n",
      "||    Running time : 1.87 minutes                                             ||\n",
      "||                                                                            ||\n",
      "|| Process BAM file SRR8668628Aligned.sortedByCoord.out.bam...                ||\n",
      "||    Strand specific : reversely stranded                                    ||\n",
      "||    Paired-end reads are included.                                          ||\n",
      "||    Total alignments : 14043222                                             ||\n",
      "||    Successfully assigned alignments : 12214675 (87.0%)                     ||\n",
      "||    Running time : 2.20 minutes                                             ||\n",
      "||                                                                            ||\n",
      "|| Process BAM file SRR8668629Aligned.sortedByCoord.out.bam...                ||\n",
      "||    Strand specific : reversely stranded                                    ||\n",
      "||    Paired-end reads are included.                                          ||\n",
      "||    Total alignments : 14365008                                             ||\n",
      "||    Successfully assigned alignments : 12498098 (87.0%)                     ||\n",
      "||    Running time : 4.02 minutes                                             ||\n",
      "||                                                                            ||\n",
      "|| Process BAM file SRR8668630Aligned.sortedByCoord.out.bam...                ||\n",
      "||    Strand specific : reversely stranded                                    ||\n",
      "||    Paired-end reads are included.                                          ||\n",
      "||    Total alignments : 13966065                                             ||\n",
      "||    Successfully assigned alignments : 12147125 (87.0%)                     ||\n",
      "||    Running time : 2.21 minutes                                             ||\n",
      "||                                                                            ||\n",
      "|| Process BAM file SRR8668631Aligned.sortedByCoord.out.bam...                ||\n",
      "||    Strand specific : reversely stranded                                    ||\n",
      "||    Paired-end reads are included.                                          ||\n",
      "||    Total alignments : 21587241                                             ||\n",
      "||    Successfully assigned alignments : 18502772 (85.7%)                     ||\n",
      "||    Running time : 2.64 minutes                                             ||\n",
      "||                                                                            ||\n",
      "|| Process BAM file SRR8668632Aligned.sortedByCoord.out.bam...                ||\n",
      "||    Strand specific : reversely stranded                                    ||\n",
      "||    Paired-end reads are included.                                          ||\n",
      "||    Total alignments : 21502897                                             ||\n",
      "||    Successfully assigned alignments : 18441818 (85.8%)                     ||\n",
      "||    Running time : 3.13 minutes                                             ||\n",
      "||                                                                            ||\n",
      "|| Process BAM file SRR8668633Aligned.sortedByCoord.out.bam...                ||\n",
      "||    Strand specific : reversely stranded                                    ||\n",
      "||    Paired-end reads are included.                                          ||\n",
      "||    Total alignments : 21702446                                             ||\n",
      "||    Successfully assigned alignments : 18617214 (85.8%)                     ||\n",
      "||    Running time : 3.53 minutes                                             ||\n",
      "||                                                                            ||\n",
      "|| Process BAM file SRR8668634Aligned.sortedByCoord.out.bam...                ||\n",
      "||    Strand specific : reversely stranded                                    ||\n",
      "||    Paired-end reads are included.                                          ||\n",
      "||    Total alignments : 21349413                                             ||\n",
      "||    Successfully assigned alignments : 18306090 (85.7%)                     ||\n",
      "||    Running time : 3.32 minutes                                             ||\n",
      "||                                                                            ||\n",
      "|| Process BAM file SRR18941362Aligned.sortedByCoord.out.bam...               ||\n",
      "||    Strand specific : reversely stranded                                    ||\n",
      "||    Paired-end reads are included.                                          ||\n",
      "||    Total alignments : 73502276                                             ||\n",
      "||    Successfully assigned alignments : 56826945 (77.3%)                     ||\n",
      "||    Running time : 5.32 minutes                                             ||\n",
      "||                                                                            ||\n",
      "|| Process BAM file SRR18941363Aligned.sortedByCoord.out.bam...               ||\n",
      "||    Strand specific : reversely stranded                                    ||\n",
      "||    Paired-end reads are included.                                          ||\n",
      "||    Total alignments : 70831124                                             ||\n",
      "||    Successfully assigned alignments : 57656796 (81.4%)                     ||\n",
      "||    Running time : 3.53 minutes                                             ||\n",
      "||                                                                            ||\n",
      "|| Process BAM file SRR18941364Aligned.sortedByCoord.out.bam...               ||\n",
      "||    Strand specific : reversely stranded                                    ||\n",
      "||    Paired-end reads are included.                                          ||\n",
      "||    Total alignments : 71334282                                             ||\n",
      "||    Successfully assigned alignments : 56839146 (79.7%)                     ||\n",
      "||    Running time : 2.64 minutes                                             ||\n",
      "||                                                                            ||\n",
      "|| Process BAM file SRR18941365Aligned.sortedByCoord.out.bam...               ||\n",
      "||    Strand specific : reversely stranded                                    ||\n",
      "||    Paired-end reads are included.                                          ||\n",
      "||    Total alignments : 76583894                                             ||\n",
      "||    Successfully assigned alignments : 61313481 (80.1%)                     ||\n",
      "||    Running time : 3.18 minutes                                             ||\n",
      "||                                                                            ||\n",
      "|| Process BAM file SRR18941366Aligned.sortedByCoord.out.bam...               ||\n",
      "||    Strand specific : reversely stranded                                    ||\n",
      "||    Paired-end reads are included.                                          ||\n",
      "||    Total alignments : 148726438                                            ||\n",
      "||    Successfully assigned alignments : 111772726 (75.2%)                    ||\n",
      "||    Running time : 8.23 minutes                                             ||\n",
      "||                                                                            ||\n",
      "|| Process BAM file SRR18941367Aligned.sortedByCoord.out.bam...               ||\n",
      "||    Strand specific : reversely stranded                                    ||\n",
      "||    Paired-end reads are included.                                          ||\n",
      "||    Total alignments : 153956013                                            ||\n",
      "||    Successfully assigned alignments : 115339852 (74.9%)                    ||\n",
      "||    Running time : 10.26 minutes                                            ||\n",
      "||                                                                            ||\n",
      "|| Write the final count table.                                               ||\n",
      "|| Write the read assignment summary.                                         ||\n",
      "||                                                                            ||\n",
      "|| Summary of counting results can be found in file \"/users/genomics/marta/T  ||\n",
      "|| estisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/human/quant  ||\n",
      "|| ification_thymus/gffcompare_stranded_featureCounts.txt.summary\"            ||\n",
      "||                                                                            ||\n",
      "\\\\============================================================================//\n",
      "\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "###PREPARING NEEDED DATA\n",
    "thymus_dir=/projects_eg/projects/marta/thymus\n",
    "outdir=/users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/human/quantification_thymus\n",
    "\n",
    "mkdir -p $outdir\n",
    "AnnotGTF=/users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/human/newReference_Resconstructed/gencode.v47.gffcompare.TestisLiverBrain.annotation.sorted.1transcript.sorted.NOchr.fixed.gtf\n",
    "\n",
    "module load Subread/2.0.3\n",
    "########################\n",
    "\n",
    "# countReadPairs may need to be removed in case of single-end reads\n",
    "featureCounts -T 10 -p -s 2 -g transcript_id -O --countReadPairs -a $AnnotGTF -o ${outdir}/gffcompare_stranded_featureCounts.txt $thymus_dir/GSE*/analysis/05_STAR/uniquely_mapped_2pass_BAM_files/*Aligned.sortedByCoord.out.bam \n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "### modify headers\n",
    "file = \"/users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/human/quantification_thymus/gffcompare_stranded_featureCounts.txt\"\n",
    "\n",
    "toc = pd.read_csv(file, sep=\"\\t\", comment=\"#\")\n",
    "toc = toc[toc['Geneid'].str.contains('PAR_')==False]\n",
    "toc['Geneid']=toc['Geneid'].str.split('.').str[0]\n",
    "toc.rename(columns={'Geneid':'transcript_id'}, inplace=True)\n",
    "\n",
    "filter_col = [col for col in toc if col.startswith('/')]\n",
    "for col in filter_col:\n",
    "    new_col=col.split(\"Aligned\")[0]\n",
    "    new_col=new_col.split(\"/\")[-1]\n",
    "    toc.rename(columns={col:new_col}, inplace=True)\n",
    "\n",
    "length = toc['Length']\n",
    "genes = toc['transcript_id']\n",
    "# we are only interested in the columns with counts\n",
    "counts = toc\n",
    "counts.drop([\"Chr\",\"Start\",\"End\",\"Strand\",\"Length\",\"transcript_id\"],axis=1, inplace=True)\n",
    "# calculate TPMs\n",
    "tpm_df = Norm.tpm(counts, length)\n",
    "# add transcript_id and length again\n",
    "tpms = pd.concat([genes,tpm_df, length], axis=1)\n",
    "tpms.to_csv(\"/users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/human/quantification_thymus/table_of_counts_TPMs_thymus.csv\", index=None)\n",
    " "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Candidates"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_3478424/717917834.py:3: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  tumorReact_genes.drop_duplicates(inplace=True)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>transcript_id</th>\n",
       "      <th>gene_id</th>\n",
       "      <th>gene_name</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ENST00000247452</td>\n",
       "      <td>ENSG00000046774</td>\n",
       "      <td>MAGEC2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ENST00000376979</td>\n",
       "      <td>ENSG00000101435</td>\n",
       "      <td>CST9L</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ENST00000375406</td>\n",
       "      <td>ENSG00000117148</td>\n",
       "      <td>ACTL8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ENST00000376919</td>\n",
       "      <td>ENSG00000126752</td>\n",
       "      <td>SSX1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ENST00000326279</td>\n",
       "      <td>ENSG00000131914</td>\n",
       "      <td>LIN28A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>239</th>\n",
       "      <td>ENST00000749288</td>\n",
       "      <td>ENSG00000287164</td>\n",
       "      <td>ENSG00000287164</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>240</th>\n",
       "      <td>TCONS_00000085</td>\n",
       "      <td>XLOC_000211</td>\n",
       "      <td>XLOC_000211</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>243</th>\n",
       "      <td>TCONS_00001755</td>\n",
       "      <td>XLOC_001654</td>\n",
       "      <td>XLOC_001654</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>246</th>\n",
       "      <td>ENST00000219301</td>\n",
       "      <td>ENSG00000103023</td>\n",
       "      <td>PRSS54</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>255</th>\n",
       "      <td>ENST00000691416</td>\n",
       "      <td>ENSG00000188306</td>\n",
       "      <td>LRRIQ4</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>140 rows × 3 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       transcript_id          gene_id        gene_name\n",
       "0    ENST00000247452  ENSG00000046774           MAGEC2\n",
       "1    ENST00000376979  ENSG00000101435            CST9L\n",
       "2    ENST00000375406  ENSG00000117148            ACTL8\n",
       "3    ENST00000376919  ENSG00000126752             SSX1\n",
       "4    ENST00000326279  ENSG00000131914           LIN28A\n",
       "..               ...              ...              ...\n",
       "239  ENST00000749288  ENSG00000287164  ENSG00000287164\n",
       "240   TCONS_00000085      XLOC_000211      XLOC_000211\n",
       "243   TCONS_00001755      XLOC_001654      XLOC_001654\n",
       "246  ENST00000219301  ENSG00000103023           PRSS54\n",
       "255  ENST00000691416  ENSG00000188306           LRRIQ4\n",
       "\n",
       "[140 rows x 3 columns]"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "tumorReact = pd.read_csv(\"/users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/cancers/log2ratio3x/cancertypes/TOv3x_5percent_TestisRestrictedGTEx_Translated_Ctypes_log2ratio3xMEANgenes.csv\")\n",
    "tumorReact_genes = tumorReact[['transcript_id','gene_id','gene_name']]\n",
    "tumorReact_genes.drop_duplicates(inplace=True)\n",
    "tumorReact_genes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_3478424/1848666162.py:2: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  tpms_candidates.drop([\"Length\"], axis=1, inplace=True)\n",
      "/tmp/ipykernel_3478424/1848666162.py:6: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  tpms_candidates['mean'] = tpms_candidates.iloc[:, 1:].mean(axis=1)\n",
      "/tmp/ipykernel_3478424/1848666162.py:7: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  tpms_candidates['median'] = tpms_candidates.iloc[:, 1:].median(axis=1)\n",
      "/tmp/ipykernel_3478424/1848666162.py:8: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  tpms_candidates['max'] = tpms_candidates.iloc[:, 1:].max(axis=1)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>transcript_id</th>\n",
       "      <th>mean</th>\n",
       "      <th>median</th>\n",
       "      <th>max</th>\n",
       "      <th>chr</th>\n",
       "      <th>gene_id</th>\n",
       "      <th>gene_name</th>\n",
       "      <th>transcript_type</th>\n",
       "      <th>gene_type</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ENST00000381089</td>\n",
       "      <td>0.160622</td>\n",
       "      <td>0.106713</td>\n",
       "      <td>0.856997</td>\n",
       "      <td>X</td>\n",
       "      <td>ENSG00000169059</td>\n",
       "      <td>VCX3A</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>protein_coding</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ENST00000381059</td>\n",
       "      <td>0.140349</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.836275</td>\n",
       "      <td>X</td>\n",
       "      <td>ENSG00000182583</td>\n",
       "      <td>VCX</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>protein_coding</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ENST00000543214</td>\n",
       "      <td>0.049054</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.287198</td>\n",
       "      <td>X</td>\n",
       "      <td>ENSG00000183304</td>\n",
       "      <td>FAM9A</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>protein_coding</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ENST00000327968</td>\n",
       "      <td>0.000760</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.012205</td>\n",
       "      <td>X</td>\n",
       "      <td>ENSG00000184735</td>\n",
       "      <td>DDX53</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>protein_coding</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ENST00000412172</td>\n",
       "      <td>0.000430</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.012894</td>\n",
       "      <td>X</td>\n",
       "      <td>ENSG00000224960</td>\n",
       "      <td>PPP4R3C</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>protein_coding</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>135</th>\n",
       "      <td>ENST00000622113</td>\n",
       "      <td>0.009798</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.084074</td>\n",
       "      <td>21</td>\n",
       "      <td>ENSG00000274391</td>\n",
       "      <td>TPTE</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>protein_coding</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>136</th>\n",
       "      <td>ENST00000400424</td>\n",
       "      <td>0.856670</td>\n",
       "      <td>0.706154</td>\n",
       "      <td>2.367336</td>\n",
       "      <td>21</td>\n",
       "      <td>ENSG00000177398</td>\n",
       "      <td>UMODL1</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>protein_coding</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>137</th>\n",
       "      <td>TCONS_00001672</td>\n",
       "      <td>0.111527</td>\n",
       "      <td>0.055174</td>\n",
       "      <td>0.686810</td>\n",
       "      <td>21</td>\n",
       "      <td>XLOC_001518</td>\n",
       "      <td>XLOC_001518</td>\n",
       "      <td>novel</td>\n",
       "      <td>novel</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>138</th>\n",
       "      <td>ENST00000332271</td>\n",
       "      <td>0.286123</td>\n",
       "      <td>0.069327</td>\n",
       "      <td>1.005759</td>\n",
       "      <td>22</td>\n",
       "      <td>ENSG00000184571</td>\n",
       "      <td>PIWIL3</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>protein_coding</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>139</th>\n",
       "      <td>ENST00000569912</td>\n",
       "      <td>0.418934</td>\n",
       "      <td>0.223653</td>\n",
       "      <td>2.421724</td>\n",
       "      <td>22</td>\n",
       "      <td>ENSG00000261202</td>\n",
       "      <td>TNRC6B-DT</td>\n",
       "      <td>lncRNA</td>\n",
       "      <td>lncRNA</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>140 rows × 9 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       transcript_id      mean    median       max chr          gene_id  \\\n",
       "0    ENST00000381089  0.160622  0.106713  0.856997   X  ENSG00000169059   \n",
       "1    ENST00000381059  0.140349  0.000000  0.836275   X  ENSG00000182583   \n",
       "2    ENST00000543214  0.049054  0.000000  0.287198   X  ENSG00000183304   \n",
       "3    ENST00000327968  0.000760  0.000000  0.012205   X  ENSG00000184735   \n",
       "4    ENST00000412172  0.000430  0.000000  0.012894   X  ENSG00000224960   \n",
       "..               ...       ...       ...       ...  ..              ...   \n",
       "135  ENST00000622113  0.009798  0.000000  0.084074  21  ENSG00000274391   \n",
       "136  ENST00000400424  0.856670  0.706154  2.367336  21  ENSG00000177398   \n",
       "137   TCONS_00001672  0.111527  0.055174  0.686810  21      XLOC_001518   \n",
       "138  ENST00000332271  0.286123  0.069327  1.005759  22  ENSG00000184571   \n",
       "139  ENST00000569912  0.418934  0.223653  2.421724  22  ENSG00000261202   \n",
       "\n",
       "       gene_name transcript_type       gene_type  \n",
       "0          VCX3A  protein_coding  protein_coding  \n",
       "1            VCX  protein_coding  protein_coding  \n",
       "2          FAM9A  protein_coding  protein_coding  \n",
       "3          DDX53  protein_coding  protein_coding  \n",
       "4        PPP4R3C  protein_coding  protein_coding  \n",
       "..           ...             ...             ...  \n",
       "135         TPTE  protein_coding  protein_coding  \n",
       "136       UMODL1  protein_coding  protein_coding  \n",
       "137  XLOC_001518           novel           novel  \n",
       "138       PIWIL3  protein_coding  protein_coding  \n",
       "139    TNRC6B-DT          lncRNA          lncRNA  \n",
       "\n",
       "[140 rows x 9 columns]"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "tpms_candidates = tpms[tpms['transcript_id'].isin(tumorReact_genes.transcript_id.values.tolist())]\n",
    "tpms_candidates.drop([\"Length\"], axis=1, inplace=True)\n",
    "tpms_candidates.to_csv(\"/users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/human/quantification_thymus/table_of_counts_TPMs_thymus_TOv3xlog2ratio3xmean.csv\")\n",
    "\n",
    "## compute mean, median, max\n",
    "tpms_candidates['mean'] = tpms_candidates.iloc[:, 1:].mean(axis=1)\n",
    "tpms_candidates['median'] = tpms_candidates.iloc[:, 1:].median(axis=1)\n",
    "tpms_candidates['max'] = tpms_candidates.iloc[:, 1:].max(axis=1)\n",
    "\n",
    "tpms_candidates_thymus = tpms_candidates[['transcript_id','mean','median','max']]\n",
    "tpms_candidates_thymus = tpms_candidates_thymus.merge(annot, on=\"transcript_id\")\n",
    "tpms_candidates_thymus.to_csv(\"/users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/human/quantification_thymus/thymus_summary_TOv3xlog2ratio3xmean.csv\")\n",
    "tpms_candidates_thymus"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Unnamed: 0</th>\n",
       "      <th>transcript_id</th>\n",
       "      <th>mean</th>\n",
       "      <th>median</th>\n",
       "      <th>max</th>\n",
       "      <th>chr</th>\n",
       "      <th>gene_id</th>\n",
       "      <th>gene_name</th>\n",
       "      <th>transcript_type</th>\n",
       "      <th>gene_type</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>48</th>\n",
       "      <td>48</td>\n",
       "      <td>TCONS_00001486</td>\n",
       "      <td>1.899517</td>\n",
       "      <td>0.355686</td>\n",
       "      <td>13.296815</td>\n",
       "      <td>2</td>\n",
       "      <td>XLOC_001350</td>\n",
       "      <td>XLOC_001350</td>\n",
       "      <td>novel</td>\n",
       "      <td>novel</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>51</th>\n",
       "      <td>51</td>\n",
       "      <td>ENST00000295453</td>\n",
       "      <td>1.150093</td>\n",
       "      <td>0.663230</td>\n",
       "      <td>7.558909</td>\n",
       "      <td>2</td>\n",
       "      <td>ENSG00000163286</td>\n",
       "      <td>ALPG</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>protein_coding</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>53</th>\n",
       "      <td>53</td>\n",
       "      <td>TCONS_00001755</td>\n",
       "      <td>1.495611</td>\n",
       "      <td>1.413604</td>\n",
       "      <td>3.625657</td>\n",
       "      <td>3</td>\n",
       "      <td>XLOC_001654</td>\n",
       "      <td>XLOC_001654</td>\n",
       "      <td>novel</td>\n",
       "      <td>novel</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>104</th>\n",
       "      <td>104</td>\n",
       "      <td>ENST00000377208</td>\n",
       "      <td>11.883755</td>\n",
       "      <td>10.270708</td>\n",
       "      <td>46.168181</td>\n",
       "      <td>13</td>\n",
       "      <td>ENSG00000152192</td>\n",
       "      <td>POU4F1</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>protein_coding</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>114</th>\n",
       "      <td>114</td>\n",
       "      <td>ENST00000357424</td>\n",
       "      <td>1.164594</td>\n",
       "      <td>0.129854</td>\n",
       "      <td>9.202288</td>\n",
       "      <td>17</td>\n",
       "      <td>ENSG00000159224</td>\n",
       "      <td>GIP</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>protein_coding</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>118</th>\n",
       "      <td>118</td>\n",
       "      <td>ENST00000663592</td>\n",
       "      <td>1.316451</td>\n",
       "      <td>1.193847</td>\n",
       "      <td>3.683874</td>\n",
       "      <td>18</td>\n",
       "      <td>ENSG00000267374</td>\n",
       "      <td>MIR924HG</td>\n",
       "      <td>lncRNA</td>\n",
       "      <td>lncRNA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>126</th>\n",
       "      <td>126</td>\n",
       "      <td>ENST00000636757</td>\n",
       "      <td>1.217590</td>\n",
       "      <td>0.958908</td>\n",
       "      <td>6.367341</td>\n",
       "      <td>19</td>\n",
       "      <td>ENSG00000261341</td>\n",
       "      <td>SMIM47</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>protein_coding</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>127</th>\n",
       "      <td>127</td>\n",
       "      <td>ENST00000424985</td>\n",
       "      <td>67.902782</td>\n",
       "      <td>47.846210</td>\n",
       "      <td>143.544888</td>\n",
       "      <td>19</td>\n",
       "      <td>ENSG00000180043</td>\n",
       "      <td>GARIN5B</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>protein_coding</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "     Unnamed: 0    transcript_id       mean     median         max chr  \\\n",
       "48           48   TCONS_00001486   1.899517   0.355686   13.296815   2   \n",
       "51           51  ENST00000295453   1.150093   0.663230    7.558909   2   \n",
       "53           53   TCONS_00001755   1.495611   1.413604    3.625657   3   \n",
       "104         104  ENST00000377208  11.883755  10.270708   46.168181  13   \n",
       "114         114  ENST00000357424   1.164594   0.129854    9.202288  17   \n",
       "118         118  ENST00000663592   1.316451   1.193847    3.683874  18   \n",
       "126         126  ENST00000636757   1.217590   0.958908    6.367341  19   \n",
       "127         127  ENST00000424985  67.902782  47.846210  143.544888  19   \n",
       "\n",
       "             gene_id    gene_name transcript_type       gene_type  \n",
       "48       XLOC_001350  XLOC_001350           novel           novel  \n",
       "51   ENSG00000163286         ALPG  protein_coding  protein_coding  \n",
       "53       XLOC_001654  XLOC_001654           novel           novel  \n",
       "104  ENSG00000152192       POU4F1  protein_coding  protein_coding  \n",
       "114  ENSG00000159224          GIP  protein_coding  protein_coding  \n",
       "118  ENSG00000267374     MIR924HG          lncRNA          lncRNA  \n",
       "126  ENSG00000261341       SMIM47  protein_coding  protein_coding  \n",
       "127  ENSG00000180043      GARIN5B  protein_coding  protein_coding  "
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "tpms_candidates_thymus = pd.read_csv(\"/users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/human/quantification_thymus/thymus_summary_TOv3xlog2ratio3xmean.csv\")\n",
    "tpms_candidates_thymus[tpms_candidates_thymus['mean'] > 1]"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.10"
  },
  "orig_nbformat": 4
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
