{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Evolutionary origin\n",
    "\n",
    "Macaca - Mouse - Human"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>specie</th>\n",
       "      <th>age</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>human</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>macaca</td>\n",
       "      <td>25</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>mouse</td>\n",
       "      <td>90</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>opossum</td>\n",
       "      <td>180</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>platypus</td>\n",
       "      <td>200</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>chicken</td>\n",
       "      <td>310</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "     specie  age\n",
       "1     human    0\n",
       "2    macaca   25\n",
       "3     mouse   90\n",
       "4   opossum  180\n",
       "5  platypus  200\n",
       "6   chicken  310"
      ]
     },
     "execution_count": 1,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import os,re,glob\n",
    "import pandas as pd\n",
    "import numpy as np\n",
    "from collections import Counter\n",
    "from Bio.Seq import Seq\n",
    "from Bio.SeqRecord import SeqRecord\n",
    "from Bio import SeqIO\n",
    "\n",
    "species_dir = \"/users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction\"\n",
    "users_dir = \"/users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47\"\n",
    "\n",
    "GENOMEDIR = \"/genomics/users/marta/genomes\"\n",
    "species=[\"mouse\",\"macaca\",\"platypus\",\"chicken\",\"opossum\"]\n",
    "\n",
    "## annotation file\n",
    "annotation=\"/users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/human/newReference_Resconstructed/gencode.v47.gffcompare.TestisLiverBrain.annotation.sorted.1transcript.sorted.NOchr.gtf\"\n",
    "transcript_gene=pd.read_csv(\"/users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/human/newReference_Resconstructed/transID_geneID_isoforms_selected.1to1.csv\")\n",
    "\n",
    "## evoDir\n",
    "evoDir=os.path.join(users_dir,\"EvolutionaryOrigin_MMseqs\")\n",
    "\n",
    "def translate_dna_to_protein(dna_seq):\n",
    "    return str(Seq(dna_seq).translate())\n",
    "\n",
    "# Function to save DF in fasta format\n",
    "def create_seqrecord_PROT(row):\n",
    "    return SeqRecord(Seq(row['protein']), id=row['header'][:-4], description=\"\")\n",
    "def create_seqrecord_DNA(row):\n",
    "    return SeqRecord(Seq(row['seq']), id=row['header'], description=\"\")\n",
    "\n",
    "species_and_years_dict = {\"specie\" : [\"human\",\"macaca\",\"mouse\",\"opossum\",\"platypus\",\"chicken\"],\n",
    "                          \"age\" : [0,25,90,180,200,310]}\n",
    "species_and_years = pd.DataFrame(data=species_and_years_dict, index=[1,2,3,4,5,6])\n",
    "species_and_years\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "## get repre sequences - both DNA and PROTEINS\n",
    "for s in species:\n",
    "    print(s)\n",
    "    riboseq_dir = os.path.join(users_dir,s,\"RiboSeq/RiboQC_RiboNovel\")\n",
    "\n",
    "    ## get all candidates\n",
    "    print(\"Reading all the candidate ORF sequences\")\n",
    "    df = pd.read_csv(os.path.join(riboseq_dir,\"Annotation/candidateORF.fa\"), header=None, sep=\"\\t\")\n",
    "    ## convert to df\n",
    "    candidates_fasta = pd.DataFrame({'header':df[0].iloc[::2].values, 'seq':df[0].iloc[1::2].values})\n",
    "    candidates_fasta['header'] = candidates_fasta['header'].str[1:]\n",
    "    candidates_fasta = candidates_fasta[~candidates_fasta['header'].str.contains(\"iORF\")]\n",
    "\n",
    "    all_repre_candidates = pd.DataFrame()\n",
    "    for sample in os.listdir(os.path.join(riboseq_dir,\"RibORF\")):\n",
    "        if sample.endswith(\"r1\"):\n",
    "            print(sample)\n",
    "\n",
    "            repre = pd.read_csv(os.path.join(riboseq_dir,\"RibORF\",sample,\"repre.valid.pred.pvalue.parameters.txt\"), sep=\"\\t\")\n",
    "            repre_fasta = candidates_fasta[candidates_fasta['header'].isin(repre.orfID.values.tolist())]\n",
    "            repre_fasta['header'] = s + \"_\" +repre_fasta['header']\n",
    "\n",
    "            ## keep only the coding and the non-coding\n",
    "            repre_fasta = repre_fasta[repre_fasta['header'].str.contains(\"canonical\")| repre_fasta['header'].str.contains(\"noncoding\")]\n",
    "\n",
    "            repre_fasta['protein'] = repre_fasta['seq'].apply(translate_dna_to_protein)\n",
    "            all_repre_candidates = pd.concat([all_repre_candidates, repre_fasta])\n",
    "\n",
    "            ## translate to protein\n",
    "            print(\"Translating to protein\")\n",
    "            out_proteins = os.path.join(riboseq_dir,\"RibORF\",sample,\"repre.valid.pred.pvalue.parameters.PROTEIN.fa\")\n",
    "            seq_records_proteins = repre_fasta.apply(create_seqrecord_PROT, axis=1).tolist()\n",
    "            SeqIO.write(seq_records_proteins, out_proteins, \"fasta\")\n",
    "\n",
    "            ## keep the dna sequences as well\n",
    "            print(\"Saving the original DNA ORF\")\n",
    "            out_dna = os.path.join(riboseq_dir,\"RibORF\",sample,\"repre.valid.pred.pvalue.parameters.DNA.fa\")\n",
    "            seq_records_dna = repre_fasta.apply(create_seqrecord_DNA, axis=1).tolist()\n",
    "            SeqIO.write(seq_records_dna, out_dna, \"fasta\")\n",
    "\n",
    "            ## get only the nonredundant\n",
    "            print(\"Getting the non-redundant\")\n",
    "            all_repre_candidates.drop_duplicates(inplace=True)\n",
    "            out_nonredundant_proteins = os.path.join(riboseq_dir,\"RibORF\")+\"/\"+s+\"_repre.valid.pred.pvalue.parameters.PROTEIN.noredunant.fa\"\n",
    "            seq_records_proteins = all_repre_candidates.apply(create_seqrecord_PROT, axis=1).tolist()\n",
    "            SeqIO.write(seq_records_proteins, out_nonredundant_proteins, \"fasta\")"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## MMseqs2\n",
    "\n",
    "Against all expressed in mouse, macaca, platypus, opossum in any tissue (non-redunant set)"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### TESTIS-EXPRESSED"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "ename": "ParserError",
     "evalue": "Error tokenizing data. C error: Calling read(nbytes) on source failed. Try engine='python'.",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mParserError\u001b[0m                               Traceback (most recent call last)",
      "Cell \u001b[0;32mIn[3], line 2\u001b[0m\n\u001b[1;32m      1\u001b[0m \u001b[39m## Testis-expressed ORFs human\u001b[39;00m\n\u001b[0;32m----> 2\u001b[0m TestisExpressed_translatedINtestis \u001b[39m=\u001b[39m pd\u001b[39m.\u001b[39;49mread_csv(\u001b[39m\"\u001b[39;49m\u001b[39m/projects_eg/projects/marta/TestisRestricted_Microproteins_TSA/with_TranscriptomeReconstruction/Multimap_altORFs/Q1_TestisORFs/human/ribORF_humanTestis_in1.csv\u001b[39;49m\u001b[39m\"\u001b[39;49m)\n\u001b[1;32m      3\u001b[0m TestisExpressed_translatedINtestis\n\u001b[1;32m      5\u001b[0m \u001b[39mdef\u001b[39;00m \u001b[39mcreate_seqrecord_PROThuman\u001b[39m(row):\n",
      "File \u001b[0;32m~/.local/lib/python3.8/site-packages/pandas/io/parsers/readers.py:912\u001b[0m, in \u001b[0;36mread_csv\u001b[0;34m(filepath_or_buffer, sep, delimiter, header, names, index_col, usecols, dtype, engine, converters, true_values, false_values, skipinitialspace, skiprows, skipfooter, nrows, na_values, keep_default_na, na_filter, verbose, skip_blank_lines, parse_dates, infer_datetime_format, keep_date_col, date_parser, date_format, dayfirst, cache_dates, iterator, chunksize, compression, thousands, decimal, lineterminator, quotechar, quoting, doublequote, escapechar, comment, encoding, encoding_errors, dialect, on_bad_lines, delim_whitespace, low_memory, memory_map, float_precision, storage_options, dtype_backend)\u001b[0m\n\u001b[1;32m    899\u001b[0m kwds_defaults \u001b[39m=\u001b[39m _refine_defaults_read(\n\u001b[1;32m    900\u001b[0m     dialect,\n\u001b[1;32m    901\u001b[0m     delimiter,\n\u001b[0;32m   (...)\u001b[0m\n\u001b[1;32m    908\u001b[0m     dtype_backend\u001b[39m=\u001b[39mdtype_backend,\n\u001b[1;32m    909\u001b[0m )\n\u001b[1;32m    910\u001b[0m kwds\u001b[39m.\u001b[39mupdate(kwds_defaults)\n\u001b[0;32m--> 912\u001b[0m \u001b[39mreturn\u001b[39;00m _read(filepath_or_buffer, kwds)\n",
      "File \u001b[0;32m~/.local/lib/python3.8/site-packages/pandas/io/parsers/readers.py:583\u001b[0m, in \u001b[0;36m_read\u001b[0;34m(filepath_or_buffer, kwds)\u001b[0m\n\u001b[1;32m    580\u001b[0m     \u001b[39mreturn\u001b[39;00m parser\n\u001b[1;32m    582\u001b[0m \u001b[39mwith\u001b[39;00m parser:\n\u001b[0;32m--> 583\u001b[0m     \u001b[39mreturn\u001b[39;00m parser\u001b[39m.\u001b[39;49mread(nrows)\n",
      "File \u001b[0;32m~/.local/lib/python3.8/site-packages/pandas/io/parsers/readers.py:1704\u001b[0m, in \u001b[0;36mTextFileReader.read\u001b[0;34m(self, nrows)\u001b[0m\n\u001b[1;32m   1697\u001b[0m nrows \u001b[39m=\u001b[39m validate_integer(\u001b[39m\"\u001b[39m\u001b[39mnrows\u001b[39m\u001b[39m\"\u001b[39m, nrows)\n\u001b[1;32m   1698\u001b[0m \u001b[39mtry\u001b[39;00m:\n\u001b[1;32m   1699\u001b[0m     \u001b[39m# error: \"ParserBase\" has no attribute \"read\"\u001b[39;00m\n\u001b[1;32m   1700\u001b[0m     (\n\u001b[1;32m   1701\u001b[0m         index,\n\u001b[1;32m   1702\u001b[0m         columns,\n\u001b[1;32m   1703\u001b[0m         col_dict,\n\u001b[0;32m-> 1704\u001b[0m     ) \u001b[39m=\u001b[39m \u001b[39mself\u001b[39;49m\u001b[39m.\u001b[39;49m_engine\u001b[39m.\u001b[39;49mread(  \u001b[39m# type: ignore[attr-defined]\u001b[39;49;00m\n\u001b[1;32m   1705\u001b[0m         nrows\n\u001b[1;32m   1706\u001b[0m     )\n\u001b[1;32m   1707\u001b[0m \u001b[39mexcept\u001b[39;00m \u001b[39mException\u001b[39;00m:\n\u001b[1;32m   1708\u001b[0m     \u001b[39mself\u001b[39m\u001b[39m.\u001b[39mclose()\n",
      "File \u001b[0;32m~/.local/lib/python3.8/site-packages/pandas/io/parsers/c_parser_wrapper.py:234\u001b[0m, in \u001b[0;36mCParserWrapper.read\u001b[0;34m(self, nrows)\u001b[0m\n\u001b[1;32m    232\u001b[0m \u001b[39mtry\u001b[39;00m:\n\u001b[1;32m    233\u001b[0m     \u001b[39mif\u001b[39;00m \u001b[39mself\u001b[39m\u001b[39m.\u001b[39mlow_memory:\n\u001b[0;32m--> 234\u001b[0m         chunks \u001b[39m=\u001b[39m \u001b[39mself\u001b[39;49m\u001b[39m.\u001b[39;49m_reader\u001b[39m.\u001b[39;49mread_low_memory(nrows)\n\u001b[1;32m    235\u001b[0m         \u001b[39m# destructive to chunks\u001b[39;00m\n\u001b[1;32m    236\u001b[0m         data \u001b[39m=\u001b[39m _concatenate_chunks(chunks)\n",
      "File \u001b[0;32m~/.local/lib/python3.8/site-packages/pandas/_libs/parsers.pyx:814\u001b[0m, in \u001b[0;36mpandas._libs.parsers.TextReader.read_low_memory\u001b[0;34m()\u001b[0m\n",
      "File \u001b[0;32m~/.local/lib/python3.8/site-packages/pandas/_libs/parsers.pyx:875\u001b[0m, in \u001b[0;36mpandas._libs.parsers.TextReader._read_rows\u001b[0;34m()\u001b[0m\n",
      "File \u001b[0;32m~/.local/lib/python3.8/site-packages/pandas/_libs/parsers.pyx:850\u001b[0m, in \u001b[0;36mpandas._libs.parsers.TextReader._tokenize_rows\u001b[0;34m()\u001b[0m\n",
      "File \u001b[0;32m~/.local/lib/python3.8/site-packages/pandas/_libs/parsers.pyx:861\u001b[0m, in \u001b[0;36mpandas._libs.parsers.TextReader._check_tokenize_status\u001b[0;34m()\u001b[0m\n",
      "File \u001b[0;32m~/.local/lib/python3.8/site-packages/pandas/_libs/parsers.pyx:2029\u001b[0m, in \u001b[0;36mpandas._libs.parsers.raise_parser_error\u001b[0;34m()\u001b[0m\n",
      "\u001b[0;31mParserError\u001b[0m: Error tokenizing data. C error: Calling read(nbytes) on source failed. Try engine='python'."
     ]
    }
   ],
   "source": [
    "## Testis-expressed ORFs human\n",
    "TestisExpressed_translatedINtestis = pd.read_csv(\"/projects_eg/projects/marta/TestisRestricted_Microproteins_TSA/with_TranscriptomeReconstruction/Multimap_altORFs/Q1_TestisORFs/human/ribORF_humanTestis_in1.csv\")\n",
    "TestisExpressed_translatedINtestis\n",
    "\n",
    "def create_seqrecord_PROThuman(row):\n",
    "    return SeqRecord(Seq(row['ORFpep']), id=row['orfID'], description=\"\")\n",
    "\n",
    "out_nonredundant_proteins = \"/projects_eg/projects/marta/TestisRestricted_Microproteins_TSA/with_TranscriptomeReconstruction/Multimap_altORFs/Q1_TestisORFs/human/ribORF_humanTestis_in1.fa\"\n",
    "seq_records_proteins = TestisExpressed_translatedINtestis.apply(create_seqrecord_PROThuman, axis=1).tolist()\n",
    "SeqIO.write(seq_records_proteins, out_nonredundant_proteins, \"fasta\")"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Creating database of human (expressed in testis)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Process is interrupted.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "# >>>>>>>>>>>>>>>>>>>>>> ERROR REPORT <<<<<<<<<<<<<<<<<<<<<<\n",
      "\n",
      "    Traceback (most recent call last):\n",
      "      File \"/soft/system/software/Miniconda3/4.9.2/lib/python3.8/site-packages/conda/exceptions.py\", line 1124, in __call__\n",
      "        return func(*args, **kwargs)\n",
      "      File \"/soft/system/software/Miniconda3/4.9.2/lib/python3.8/site-packages/conda/cli/main.py\", line 94, in main_sourced\n",
      "        print(activator.execute(), end=\"\")\n",
      "    BrokenPipeError: [Errno 32] Broken pipe\n",
      "\n",
      "`$ /soft/system/software/Miniconda3/4.9.2/bin/conda shell.posix activate MMseqs2`\n",
      "\n",
      "  environment variables:\n",
      "                 CIO_TEST=<not set>\n",
      "                CONDA_EXE=/soft/system/software/Miniconda3/4.9.2/bin/conda\n",
      "         CONDA_PYTHON_EXE=/soft/system/software/Miniconda3/4.9.2/bin/python\n",
      "               CONDA_ROOT=/soft/system/software/Miniconda3/4.9.2\n",
      "              CONDA_SHLVL=0\n",
      "           CURL_CA_BUNDLE=<not set>\n",
      "               LD_PRELOAD=<not set>\n",
      "  LMOD_DEFAULT_MODULEPATH=/soft/system/modules/all:/etc/lmod/modules:/usr/share/lmod/lmod/module\n",
      "                          files\n",
      "                  MANPATH=/soft/system/software/Miniconda3/4.9.2/share/man:/usr/share/lmod/lmod/\n",
      "                          share/man::\n",
      "               MODULEPATH=/soft/system/modules/all:/etc/lmod/modules:/usr/share/lmod/lmod/module\n",
      "                          files\n",
      "                     PATH=/soft/system/software/Miniconda3/4.9.2/condabin:/soft/system/software/\n",
      "                          Miniconda3/4.9.2:/soft/system/software/Miniconda3/4.9.2/bin:/bin:/home\n",
      "                          /marta/.local/bin:/home/marta/.vscode-\n",
      "                          server/bin/5e805b79fcb6ba4c2d23712967df89a089da575b/bin/remote-\n",
      "                          cli:/genomics/users/marta/tools/hisat-\n",
      "                          genotype:/genomics/users/marta/tools/hisat-\n",
      "                          genotype/hisat2:/genomics/users/marta/tools/hisat-\n",
      "                          genotype:/genomics/users/marta/tools/hisat-genotype/hisat2:/usr/local/\n",
      "                          sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/loca\n",
      "                          l/games:/snap/bin:/home/marta/.dotnet/tools:/opt/bin:/home/marta/.loca\n",
      "                          l/bin:/home/marta/bin:/home/marta/.dotnet/tools:/opt/bin:/home/marta/.\n",
      "                          local/bin:/home/marta/bin:/home/marta/.vscode-\n",
      "                          server/bin/5e805b79fcb6ba4c2d23712967df89a089da575b/bin/remote-\n",
      "                          cli:/genomics/users/marta/tools/hisat-\n",
      "                          genotype:/genomics/users/marta/tools/hisat-\n",
      "                          genotype/hisat2:/genomics/users/marta/tools/hisat-\n",
      "                          genotype:/genomics/users/marta/tools/hisat-genotype/hisat2:/usr/local/\n",
      "                          sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/loca\n",
      "                          l/games:/snap/bin:/home/marta/.dotnet/tools:/opt/bin:/home/marta/.loca\n",
      "                          l/bin:/home/marta/bin:/home/marta/.dotnet/tools:/opt/bin:/home/marta/.\n",
      "                          local/bin:/home/marta/bin\n",
      "          PKG_CONFIG_PATH=/soft/system/software/Miniconda3/4.9.2/lib/pkgconfig\n",
      "         PYTHONIOENCODING=utf-8\n",
      "               PYTHONPATH=/genomics/users/marta/tools/hisat-\n",
      "                          genotype/hisatgenotype_modules:/genomics/users/marta/tools/hisat-\n",
      "                          genotype/hisatgenotype_modules::/genomics/users/marta/tools/hisat-\n",
      "                          genotype/hisatgenotype_modules:/genomics/users/marta/tools/hisat-\n",
      "                          genotype/hisatgenotype_modules:\n",
      "         PYTHONUNBUFFERED=1\n",
      "       REQUESTS_CA_BUNDLE=<not set>\n",
      "            SSL_CERT_FILE=<not set>\n",
      "\n",
      "     active environment : None\n",
      "            shell level : 0\n",
      "       user config file : /home/marta/.condarc\n",
      " populated config files : \n",
      "          conda version : 23.1.0\n",
      "    conda-build version : not installed\n",
      "         python version : 3.8.16.final.0\n",
      "       virtual packages : __archspec=1=x86_64\n",
      "                          __cuda=12.2=0\n",
      "                          __glibc=2.31=0\n",
      "                          __linux=5.15.0=0\n",
      "                          __unix=0=0\n",
      "       base environment : /soft/system/software/Miniconda3/4.9.2  (read only)\n",
      "      conda av data dir : /soft/system/software/Miniconda3/4.9.2/etc/conda\n",
      "  conda av metadata url : None\n",
      "           channel URLs : https://repo.anaconda.com/pkgs/main/linux-64\n",
      "                          https://repo.anaconda.com/pkgs/main/noarch\n",
      "                          https://repo.anaconda.com/pkgs/r/linux-64\n",
      "                          https://repo.anaconda.com/pkgs/r/noarch\n",
      "          package cache : /soft/system/software/Miniconda3/4.9.2/pkgs\n",
      "                          /home/marta/.conda/pkgs\n",
      "       envs directories : /home/marta/.conda/envs\n",
      "                          /soft/system/software/Miniconda3/4.9.2/envs\n",
      "               platform : linux-64\n",
      "             user-agent : conda/23.1.0 requests/2.28.1 CPython/3.8.16 Linux/5.15.0-126-generic ubuntu/20.04.6 glibc/2.31\n",
      "                UID:GID : 2334:2300\n",
      "             netrc file : None\n",
      "           offline mode : False\n",
      "\n",
      "\n",
      "An unexpected error has occurred. Conda has prepared the above report.\n",
      "\n"
     ]
    }
   ],
   "source": [
    "%%bash -s \"$evoDir\" \"$users_dir\"\n",
    "\n",
    "# mkdir $1/MMseqsDB\n",
    "\n",
    "module load Miniconda3/4.9.2\n",
    "source /soft/system/software/Miniconda3/4.9.2/bin/activate MMseqs2\n",
    "\n",
    "mkdir -p $1/MMseqsDB\n",
    "\n",
    "mmseqs createdb /projects_eg/projects/marta/TestisRestricted_Microproteins_TSA/with_TranscriptomeReconstruction/Multimap_altORFs/Q1_TestisORFs/human/ribORF_humanTestis_in1.fa $1/MMseqsDB/ribORF_humanTestis_in1_DB\n"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Creating a joint database with all the specie, everything translated, no matter the tissue"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "createdb /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/chicken/RiboSeq/RiboQC_RiboNovel/RibORF/chicken_repre.valid.pred.pvalue.parameters.PROTEIN.noredunant.fa /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/macaca/RiboSeq/RiboQC_RiboNovel/RibORF/macaca_repre.valid.pred.pvalue.parameters.PROTEIN.noredunant.fa /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/mouse/RiboSeq/RiboQC_RiboNovel/RibORF/mouse_repre.valid.pred.pvalue.parameters.PROTEIN.noredunant.fa /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/opossum/RiboSeq/RiboQC_RiboNovel/RibORF/opossum_repre.valid.pred.pvalue.parameters.PROTEIN.noredunant.fa /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/platypus/RiboSeq/RiboQC_RiboNovel/RibORF/platypus_repre.valid.pred.pvalue.parameters.PROTEIN.noredunant.fa /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/MMseqsDB/speciesDB \n",
      "\n",
      "MMseqs Version:       \t14.7e284\n",
      "Database type         \t0\n",
      "Shuffle input database\ttrue\n",
      "Createdb mode         \t0\n",
      "Write lookup file     \t1\n",
      "Offset of numeric ids \t0\n",
      "Compressed            \t0\n",
      "Verbosity             \t3\n",
      "\n",
      "Converting sequences\n",
      "[===========\n",
      "Time for merging to speciesDB_h: 0h 0m 1s 478ms\n",
      "Time for merging to speciesDB: 0h 0m 1s 659ms\n",
      "Database type: Aminoacid\n",
      "Time for processing: 0h 0m 8s 155ms\n"
     ]
    }
   ],
   "source": [
    "%%bash -s \"$evoDir\" \"$species_dir\"\n",
    "\n",
    "module load Miniconda3/4.9.2\n",
    "source /soft/system/software/Miniconda3/4.9.2/bin/activate MMseqs2\n",
    "\n",
    "# species=\"platypus mouse macaca chicken\"\n",
    "# for s in $species; do\n",
    "#     echo $s\n",
    "\n",
    "#     ### create DB\n",
    "#     mmseqs createdb $2/$s/RiboSeq/RiboQC_RiboNovel/RibORF/repre.valid.pred.pvalue.parameters.PROTEIN.noredunant.fa $1/MMseqsDB/${s}_noredunantProts_DB\n",
    "\n",
    "# done\n",
    "\n",
    "mmseqs createdb $2/*/RiboSeq/RiboQC_RiboNovel*/RibORF/*_repre.valid.pred.pvalue.parameters.PROTEIN.noredunant.fa $1/MMseqsDB/speciesDB\n",
    "\n"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Creating a joint database with all the specie AND human for clustering, everything translated for the species, no matter the tissue, but only translated in 1 for testis"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "createdb /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/chicken/RiboSeq/RiboQC_RiboNovel/RibORF/chicken_repre.valid.pred.pvalue.parameters.PROTEIN.noredunant.fa /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/macaca/RiboSeq/RiboQC_RiboNovel/RibORF/macaca_repre.valid.pred.pvalue.parameters.PROTEIN.noredunant.fa /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/mouse/RiboSeq/RiboQC_RiboNovel/RibORF/mouse_repre.valid.pred.pvalue.parameters.PROTEIN.noredunant.fa /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/opossum/RiboSeq/RiboQC_RiboNovel/RibORF/opossum_repre.valid.pred.pvalue.parameters.PROTEIN.noredunant.fa /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/platypus/RiboSeq/RiboQC_RiboNovel/RibORF/platypus_repre.valid.pred.pvalue.parameters.PROTEIN.noredunant.fa /projects_eg/projects/marta/TestisRestricted_Microproteins_TSA/with_TranscriptomeReconstruction/Multimap_altORFs/Q1_TestisORFs/human/ribORF_humanTestis_in1.fa /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/MMseqsDB/species_and_humanDB \n",
      "\n",
      "MMseqs Version:       \t14.7e284\n",
      "Database type         \t0\n",
      "Shuffle input database\ttrue\n",
      "Createdb mode         \t0\n",
      "Write lookup file     \t1\n",
      "Offset of numeric ids \t0\n",
      "Compressed            \t0\n",
      "Verbosity             \t3\n",
      "\n",
      "Converting sequences\n",
      "[==============\n",
      "Time for merging to species_and_humanDB_h: 0h 0m 0s 746ms\n",
      "Time for merging to species_and_humanDB: 0h 0m 1s 253ms\n",
      "Database type: Aminoacid\n",
      "Time for processing: 0h 0m 6s 742ms\n"
     ]
    }
   ],
   "source": [
    "%%bash -s \"$evoDir\" \"$species_dir\"\n",
    "\n",
    "module load Miniconda3/4.9.2\n",
    "source /soft/system/software/Miniconda3/4.9.2/bin/activate MMseqs2\n",
    "\n",
    "mmseqs createdb $2/*/RiboSeq/RiboQC_RiboNovel*/RibORF/*_repre.valid.pred.pvalue.parameters.PROTEIN.noredunant.fa /projects_eg/projects/marta/TestisRestricted_Microproteins_TSA/with_TranscriptomeReconstruction/Multimap_altORFs/Q1_TestisORFs/human/ribORF_humanTestis_in1.fa $1/MMseqsDB/species_and_humanDB\n",
    "\n"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Searching"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "rm: cannot remove '/users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/Searching/resultDB*': No such file or directory\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Create directory /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/tmp\n",
      "search -a -s 6 /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/MMseqsDB/ribORF_humanTestis_in1_DB /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/MMseqsDB/speciesDB /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/Searching/resultDB /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/tmp \n",
      "\n",
      "MMseqs Version:                        \t14.7e284\n",
      "Substitution matrix                    \taa:blosum62.out,nucl:nucleotide.out\n",
      "Add backtrace                          \ttrue\n",
      "Alignment mode                         \t2\n",
      "Alignment mode                         \t0\n",
      "Allow wrapped scoring                  \tfalse\n",
      "E-value threshold                      \t0.001\n",
      "Seq. id. threshold                     \t0\n",
      "Min alignment length                   \t0\n",
      "Seq. id. mode                          \t0\n",
      "Alternative alignments                 \t0\n",
      "Coverage threshold                     \t0\n",
      "Coverage mode                          \t0\n",
      "Max sequence length                    \t65535\n",
      "Compositional bias                     \t1\n",
      "Compositional bias                     \t1\n",
      "Max reject                             \t2147483647\n",
      "Max accept                             \t2147483647\n",
      "Include identical seq. id.             \tfalse\n",
      "Preload mode                           \t0\n",
      "Pseudo count a                         \tsubstitution:1.100,context:1.400\n",
      "Pseudo count b                         \tsubstitution:4.100,context:5.800\n",
      "Score bias                             \t0\n",
      "Realign hits                           \tfalse\n",
      "Realign score bias                     \t-0.2\n",
      "Realign max seqs                       \t2147483647\n",
      "Correlation score weight               \t0\n",
      "Gap open cost                          \taa:11,nucl:5\n",
      "Gap extension cost                     \taa:1,nucl:2\n",
      "Zdrop                                  \t40\n",
      "Threads                                \t24\n",
      "Compressed                             \t0\n",
      "Verbosity                              \t3\n",
      "Seed substitution matrix               \taa:VTML80.out,nucl:nucleotide.out\n",
      "Sensitivity                            \t6\n",
      "k-mer length                           \t0\n",
      "k-score                                \tseq:2147483647,prof:2147483647\n",
      "Alphabet size                          \taa:21,nucl:5\n",
      "Max results per query                  \t300\n",
      "Split database                         \t0\n",
      "Split mode                             \t2\n",
      "Split memory limit                     \t0\n",
      "Diagonal scoring                       \ttrue\n",
      "Exact k-mer matching                   \t0\n",
      "Mask residues                          \t1\n",
      "Mask residues probability              \t0.9\n",
      "Mask lower case residues               \t0\n",
      "Minimum diagonal score                 \t15\n",
      "Selected taxa                          \t\n",
      "Spaced k-mers                          \t1\n",
      "Spaced k-mer pattern                   \t\n",
      "Local temporary path                   \t\n",
      "Rescore mode                           \t0\n",
      "Remove hits by seq. id. and coverage   \tfalse\n",
      "Sort results                           \t0\n",
      "Mask profile                           \t1\n",
      "Profile E-value threshold              \t0.1\n",
      "Global sequence weighting              \tfalse\n",
      "Allow deletions                        \tfalse\n",
      "Filter MSA                             \t1\n",
      "Use filter only at N seqs              \t0\n",
      "Maximum seq. id. threshold             \t0.9\n",
      "Minimum seq. id.                       \t0.0\n",
      "Minimum score per column               \t-20\n",
      "Minimum coverage                       \t0\n",
      "Select N most diverse seqs             \t1000\n",
      "Pseudo count mode                      \t0\n",
      "Gap pseudo count                       \t10\n",
      "Min codons in orf                      \t30\n",
      "Max codons in length                   \t32734\n",
      "Max orf gaps                           \t2147483647\n",
      "Contig start mode                      \t2\n",
      "Contig end mode                        \t2\n",
      "Orf start mode                         \t1\n",
      "Forward frames                         \t1,2,3\n",
      "Reverse frames                         \t1,2,3\n",
      "Translation table                      \t1\n",
      "Translate orf                          \t0\n",
      "Use all table starts                   \tfalse\n",
      "Offset of numeric ids                  \t0\n",
      "Create lookup                          \t0\n",
      "Add orf stop                           \tfalse\n",
      "Overlap between sequences              \t0\n",
      "Sequence split mode                    \t1\n",
      "Header split mode                      \t0\n",
      "Chain overlapping alignments           \t0\n",
      "Merge query                            \t1\n",
      "Search type                            \t0\n",
      "Search iterations                      \t1\n",
      "Start sensitivity                      \t4\n",
      "Search steps                           \t1\n",
      "Exhaustive search mode                 \tfalse\n",
      "Filter results during exhaustive search\t0\n",
      "Strand selection                       \t1\n",
      "LCA search mode                        \tfalse\n",
      "Disk space limit                       \t0\n",
      "MPI runner                             \t\n",
      "Force restart with latest tmp          \tfalse\n",
      "Remove temporary files                 \tfalse\n",
      "\n",
      "prefilter /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/MMseqsDB/ribORF_humanTestis_in1_DB /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/MMseqsDB/speciesDB /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/tmp/12946947319972804254/pref_0 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' --seed-sub-mat 'aa:VTML80.out,nucl:nucleotide.out' -k 0 --k-score seq:2147483647,prof:2147483647 --alph-size aa:21,nucl:5 --max-seq-len 65535 --max-seqs 300 --split 0 --split-mode 2 --split-memory-limit 0 -c 0 --cov-mode 0 --comp-bias-corr 1 --comp-bias-corr-scale 1 --diag-score 1 --exact-kmer-matching 0 --mask 1 --mask-prob 0.9 --mask-lower-case 0 --min-ungapped-score 15 --add-self-matches 0 --spaced-kmer-mode 1 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --threads 24 --compressed 0 -v 3 -s 6.0 \n",
      "\n",
      "Query database size: 27106 type: Aminoacid\n",
      "Estimated memory consumption: 1G\n",
      "Target database size: 115413 type: Aminoacid\n",
      "Index table k-mer threshold: 109 at k-mer size 6 \n",
      "Index table: counting k-mers\n",
      "[=================================================================] 115.41K 0s 691ms\n",
      "Index table: Masked residues: 1929565\n",
      "Index table: fill\n",
      "[=================================================================] 115.41K 0s 446ms\n",
      "Index statistics\n",
      "Entries:          42415036\n",
      "DB size:          730 MB\n",
      "Avg k-mer size:   0.662735\n",
      "Top 10 k-mers\n",
      "    DVVFEE\t478\n",
      "    LLLLLL\t278\n",
      "    TFDAFT\t246\n",
      "    VMEYLV\t232\n",
      "    LYDMNY\t224\n",
      "    LLPQLY\t183\n",
      "    VHSGRT\t150\n",
      "    LGGFVY\t134\n",
      "    SSSSGS\t132\n",
      "    IGGFVY\t132\n",
      "Time for index table init: 0h 0m 1s 607ms\n",
      "Process prefiltering step 1 of 1\n",
      "\n",
      "k-mer similarity threshold: 109\n",
      "Starting prefiltering scores calculation (step 1 of 1)\n",
      "Query db start 1 to 27106\n",
      "Target db start 1 to 115413\n",
      "[=================================================================] 27.11K 12s 803ms\n",
      "\n",
      "377.570766 k-mers per position\n",
      "94917 DB matches per sequence\n",
      "2 overflows\n",
      "0 queries produce too many hits (truncated result)\n",
      "209 sequences passed prefiltering per query sequence\n",
      "300 median result list length\n",
      "308 sequences with 0 size result lists\n",
      "Time for merging to pref_0: 0h 0m 0s 264ms\n",
      "Time for processing: 0h 0m 16s 609ms\n",
      "align /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/MMseqsDB/ribORF_humanTestis_in1_DB /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/MMseqsDB/speciesDB /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/tmp/12946947319972804254/pref_0 /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/Searching/resultDB --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' -a 1 --alignment-mode 2 --alignment-output-mode 0 --wrapped-scoring 0 -e 0.001 --min-seq-id 0 --min-aln-len 0 --seq-id-mode 0 --alt-ali 0 -c 0 --cov-mode 0 --max-seq-len 65535 --comp-bias-corr 1 --comp-bias-corr-scale 1 --max-rejected 2147483647 --max-accept 2147483647 --add-self-matches 0 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --score-bias 0 --realign 0 --realign-score-bias -0.2 --realign-max-seqs 2147483647 --corr-score-weight 0 --gap-open aa:11,nucl:5 --gap-extend aa:1,nucl:2 --zdrop 40 --threads 24 --compressed 0 -v 3 \n",
      "\n",
      "Compute score, coverage and sequence identity\n",
      "Query database size: 27106 type: Aminoacid\n",
      "Target database size: 115413 type: Aminoacid\n",
      "Calculation of alignments\n",
      "[=================================================================] 27.11K 1m 9s 480ms\n",
      "Time for merging to resultDB: 0h 0m 0s 292ms\n",
      "5665541 alignments calculated\n",
      "1185782 sequence pairs passed the thresholds (0.209297 of overall calculated)\n",
      "43.746109 hits per query sequence\n",
      "Time for processing: 0h 1m 11s 978ms\n",
      "convertalis /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/MMseqsDB/ribORF_humanTestis_in1_DB /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/MMseqsDB/speciesDB /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/Searching/resultDB /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/Searching/resultDB.m8 \n",
      "\n",
      "MMseqs Version:        \t14.7e284\n",
      "Substitution matrix    \taa:blosum62.out,nucl:nucleotide.out\n",
      "Alignment format       \t0\n",
      "Format alignment output\tquery,target,fident,alnlen,mismatch,gapopen,qstart,qend,tstart,tend,evalue,bits\n",
      "Translation table      \t1\n",
      "Gap open cost          \taa:11,nucl:5\n",
      "Gap extension cost     \taa:1,nucl:2\n",
      "Database output        \tfalse\n",
      "Preload mode           \t0\n",
      "Search type            \t0\n",
      "Threads                \t24\n",
      "Compressed             \t0\n",
      "Verbosity              \t3\n",
      "\n",
      "[=================================================================] 27.11K 1s 245ms\n",
      "Time for merging to resultDB.m8: 0h 0m 3s 373ms\n",
      "Time for processing: 0h 0m 7s 317ms\n"
     ]
    }
   ],
   "source": [
    "%%bash -s \"$evoDir\" \"$users_dir\"\n",
    "\n",
    "module load Miniconda3/4.9.2\n",
    "source /soft/system/software/Miniconda3/4.9.2/bin/activate MMseqs2\n",
    "mkdir -p $1/Searching\n",
    "\n",
    "rm $1/Searching/resultDB*\n",
    "## The alignment consists of two steps the prefilter and alignment. To run the search, type:\n",
    "mmseqs search -a -s 6 $1/MMseqsDB/ribORF_humanTestis_in1_DB $1/MMseqsDB/speciesDB $1/Searching/resultDB $1/tmp\n",
    "\n",
    "# Then, convert the result database into a BLAST tab formatted file (option -m 8 in legacy blast, -outfmt 6 in blast+):\n",
    "# Format alignment output\tquery,target,fident,alnlen,mismatch,gapopen,qstart,qend,tstart,tend,evalue,bits\n",
    "\n",
    "mmseqs convertalis $1/MMseqsDB/ribORF_humanTestis_in1_DB $1/MMseqsDB/speciesDB $1/Searching/resultDB $1/Searching/resultDB.m8"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Clustering"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "rm: cannot remove '/users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/Clustering/species_and_human_cluDB*': No such file or directory\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "cluster -c 0.5 --min-seq-id 0.5 /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/MMseqsDB/species_and_humanDB /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/Clustering/species_and_human_cluDB /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/tmp \n",
      "\n",
      "MMseqs Version:                     \t14.7e284\n",
      "Substitution matrix                 \taa:blosum62.out,nucl:nucleotide.out\n",
      "Seed substitution matrix            \taa:VTML80.out,nucl:nucleotide.out\n",
      "Sensitivity                         \t4\n",
      "k-mer length                        \t0\n",
      "k-score                             \tseq:2147483647,prof:2147483647\n",
      "Alphabet size                       \taa:21,nucl:5\n",
      "Max sequence length                 \t65535\n",
      "Max results per query               \t20\n",
      "Split database                      \t0\n",
      "Split mode                          \t2\n",
      "Split memory limit                  \t0\n",
      "Coverage threshold                  \t0.5\n",
      "Coverage mode                       \t0\n",
      "Compositional bias                  \t1\n",
      "Compositional bias                  \t1\n",
      "Diagonal scoring                    \ttrue\n",
      "Exact k-mer matching                \t0\n",
      "Mask residues                       \t1\n",
      "Mask residues probability           \t0.9\n",
      "Mask lower case residues            \t0\n",
      "Minimum diagonal score              \t15\n",
      "Selected taxa                       \t\n",
      "Include identical seq. id.          \tfalse\n",
      "Spaced k-mers                       \t1\n",
      "Preload mode                        \t0\n",
      "Pseudo count a                      \tsubstitution:1.100,context:1.400\n",
      "Pseudo count b                      \tsubstitution:4.100,context:5.800\n",
      "Spaced k-mer pattern                \t\n",
      "Local temporary path                \t\n",
      "Threads                             \t24\n",
      "Compressed                          \t0\n",
      "Verbosity                           \t3\n",
      "Add backtrace                       \tfalse\n",
      "Alignment mode                      \t3\n",
      "Alignment mode                      \t0\n",
      "Allow wrapped scoring               \tfalse\n",
      "E-value threshold                   \t0.001\n",
      "Seq. id. threshold                  \t0.5\n",
      "Min alignment length                \t0\n",
      "Seq. id. mode                       \t0\n",
      "Alternative alignments              \t0\n",
      "Max reject                          \t2147483647\n",
      "Max accept                          \t2147483647\n",
      "Score bias                          \t0\n",
      "Realign hits                        \tfalse\n",
      "Realign score bias                  \t-0.2\n",
      "Realign max seqs                    \t2147483647\n",
      "Correlation score weight            \t0\n",
      "Gap open cost                       \taa:11,nucl:5\n",
      "Gap extension cost                  \taa:1,nucl:2\n",
      "Zdrop                               \t40\n",
      "Rescore mode                        \t0\n",
      "Remove hits by seq. id. and coverage\tfalse\n",
      "Sort results                        \t0\n",
      "Cluster mode                        \t0\n",
      "Max connected component depth       \t1000\n",
      "Similarity type                     \t2\n",
      "Single step clustering              \tfalse\n",
      "Cascaded clustering steps           \t3\n",
      "Cluster reassign                    \tfalse\n",
      "Remove temporary files              \tfalse\n",
      "Force restart with latest tmp       \tfalse\n",
      "MPI runner                          \t\n",
      "k-mers per sequence                 \t21\n",
      "Scale k-mers per sequence           \taa:0.000,nucl:0.200\n",
      "Adjust k-mer length                 \tfalse\n",
      "Shift hash                          \t67\n",
      "Include only extendable             \tfalse\n",
      "Skip repeating k-mers               \tfalse\n",
      "\n",
      "Set cluster sensitivity to -s 3.000000\n",
      "Set cluster mode SET COVER\n",
      "Set cluster iterations to 3\n",
      "linclust /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/MMseqsDB/species_and_humanDB /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/tmp/8690177688940149169/clu_redundancy /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/tmp/8690177688940149169/linclust --cluster-mode 0 --max-iterations 1000 --similarity-type 2 --threads 24 --compressed 0 -v 3 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' -a 0 --alignment-mode 3 --alignment-output-mode 0 --wrapped-scoring 0 -e 0.001 --min-seq-id 0.5 --min-aln-len 0 --seq-id-mode 0 --alt-ali 0 -c 0.5 --cov-mode 0 --max-seq-len 65535 --comp-bias-corr 1 --comp-bias-corr-scale 1 --max-rejected 2147483647 --max-accept 2147483647 --add-self-matches 0 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --score-bias 0 --realign 0 --realign-score-bias -0.2 --realign-max-seqs 2147483647 --corr-score-weight 0 --gap-open aa:11,nucl:5 --gap-extend aa:1,nucl:2 --zdrop 40 --alph-size aa:13,nucl:5 --kmer-per-seq 21 --spaced-kmer-mode 1 --kmer-per-seq-scale aa:0.000,nucl:0.200 --adjust-kmer-len 0 --mask 0 --mask-prob 0.9 --mask-lower-case 0 -k 0 --hash-shift 67 --split-memory-limit 0 --include-only-extendable 0 --ignore-multi-kmer 0 --rescore-mode 0 --filter-hits 0 --sort-results 0 --remove-tmp-files 0 --force-reuse 0 \n",
      "\n",
      "kmermatcher /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/MMseqsDB/species_and_humanDB /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/tmp/8690177688940149169/linclust/9500176059251995611/pref --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' --alph-size aa:13,nucl:5 --min-seq-id 0.5 --kmer-per-seq 21 --spaced-kmer-mode 1 --kmer-per-seq-scale aa:0.000,nucl:0.200 --adjust-kmer-len 0 --mask 0 --mask-prob 0.9 --mask-lower-case 0 --cov-mode 0 -k 0 -c 0.5 --max-seq-len 65535 --hash-shift 67 --split-memory-limit 0 --include-only-extendable 0 --ignore-multi-kmer 0 --threads 24 --compressed 0 -v 3 \n",
      "\n",
      "kmermatcher /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/MMseqsDB/species_and_humanDB /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/tmp/8690177688940149169/linclust/9500176059251995611/pref --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' --alph-size aa:13,nucl:5 --min-seq-id 0.5 --kmer-per-seq 21 --spaced-kmer-mode 1 --kmer-per-seq-scale aa:0.000,nucl:0.200 --adjust-kmer-len 0 --mask 0 --mask-prob 0.9 --mask-lower-case 0 --cov-mode 0 -k 0 -c 0.5 --max-seq-len 65535 --hash-shift 67 --split-memory-limit 0 --include-only-extendable 0 --ignore-multi-kmer 0 --threads 24 --compressed 0 -v 3 \n",
      "\n",
      "Database size: 142519 type: Aminoacid\n",
      "Reduced amino acid alphabet: (A S T) (C) (D B N) (E Q Z) (F Y) (G) (H) (I V) (K R) (L J M) (P) (W) (X) \n",
      "\n",
      "Generate k-mers list for 1 split\n",
      "[=================================================================] 142.52K 0s 634ms\n",
      "Sort kmer 0h 0m 0s 34ms\n",
      "Sort by rep. sequence 0h 0m 0s 12ms\n",
      "Time for fill: 0h 0m 0s 47ms\n",
      "Time for merging to pref: 0h 0m 0s 43ms\n",
      "Time for processing: 0h 0m 0s 998ms\n",
      "rescorediagonal /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/MMseqsDB/species_and_humanDB /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/MMseqsDB/species_and_humanDB /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/tmp/8690177688940149169/linclust/9500176059251995611/pref /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/tmp/8690177688940149169/linclust/9500176059251995611/pref_rescore1 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' --rescore-mode 0 --wrapped-scoring 0 --filter-hits 0 -e 0.001 -c 0.5 -a 0 --cov-mode 0 --min-seq-id 0.5 --min-aln-len 0 --seq-id-mode 0 --add-self-matches 0 --sort-results 0 --db-load-mode 0 --threads 24 --compressed 0 -v 3 \n",
      "\n",
      "[=================================================================] 142.52K 0s 79ms\n",
      "Time for merging to pref_rescore1: 0h 0m 0s 532ms\n",
      "Time for processing: 0h 0m 1s 578ms\n",
      "clust /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/MMseqsDB/species_and_humanDB /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/tmp/8690177688940149169/linclust/9500176059251995611/pref_rescore1 /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/tmp/8690177688940149169/linclust/9500176059251995611/pre_clust --cluster-mode 0 --max-iterations 1000 --similarity-type 2 --threads 24 --compressed 0 -v 3 \n",
      "\n",
      "Clustering mode: Set Cover\n",
      "[=================================================================] 142.52K 0s 8ms\n",
      "Sort entries\n",
      "Find missing connections\n",
      "Found 90541 new connections.\n",
      "Reconstruct initial order\n",
      "[=================================================================] 142.52K 0s 10ms\n",
      "Add missing connections\n",
      "[=================================================================] 142.52K 0s 5ms\n",
      "\n",
      "Time for read in: 0h 0m 0s 53ms\n",
      "Total time: 0h 0m 0s 144ms\n",
      "\n",
      "Size of the sequence database: 142519\n",
      "Size of the alignment database: 142519\n",
      "Number of clusters: 93442\n",
      "\n",
      "Writing results 0h 0m 0s 17ms\n",
      "Time for merging to pre_clust: 0h 0m 0s 94ms\n",
      "Time for processing: 0h 0m 0s 372ms\n",
      "createsubdb /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/tmp/8690177688940149169/linclust/9500176059251995611/order_redundancy /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/MMseqsDB/species_and_humanDB /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/tmp/8690177688940149169/linclust/9500176059251995611/input_step_redundancy -v 3 --subdb-mode 1 \n",
      "\n",
      "Time for merging to input_step_redundancy: 0h 0m 0s 76ms\n",
      "Time for processing: 0h 0m 0s 281ms\n",
      "createsubdb /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/tmp/8690177688940149169/linclust/9500176059251995611/order_redundancy /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/tmp/8690177688940149169/linclust/9500176059251995611/pref /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/tmp/8690177688940149169/linclust/9500176059251995611/pref_filter1 -v 3 --subdb-mode 1 \n",
      "\n",
      "Time for merging to pref_filter1: 0h 0m 0s 73ms\n",
      "Time for processing: 0h 0m 0s 249ms\n",
      "filterdb /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/tmp/8690177688940149169/linclust/9500176059251995611/pref_filter1 /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/tmp/8690177688940149169/linclust/9500176059251995611/pref_filter2 --filter-file /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/tmp/8690177688940149169/linclust/9500176059251995611/order_redundancy --threads 24 --compressed 0 -v 3 \n",
      "\n",
      "Filtering using file(s)\n",
      "[=================================================================] 93.44K 0s 26ms\n",
      "Time for merging to pref_filter2: 0h 0m 0s 398ms\n",
      "Time for processing: 0h 0m 1s 622ms\n",
      "rescorediagonal /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/tmp/8690177688940149169/linclust/9500176059251995611/input_step_redundancy /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/tmp/8690177688940149169/linclust/9500176059251995611/input_step_redundancy /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/tmp/8690177688940149169/linclust/9500176059251995611/pref_filter2 /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/tmp/8690177688940149169/linclust/9500176059251995611/pref_rescore2 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' --rescore-mode 1 --wrapped-scoring 0 --filter-hits 1 -e 0.001 -c 0.5 -a 0 --cov-mode 0 --min-seq-id 0.5 --min-aln-len 0 --seq-id-mode 0 --add-self-matches 0 --sort-results 0 --db-load-mode 0 --threads 24 --compressed 0 -v 3 \n",
      "\n",
      "[=================================================================] 93.44K 0s 90ms\n",
      "Time for merging to pref_rescore2: 0h 0m 0s 439ms\n",
      "Time for processing: 0h 0m 1s 732ms\n",
      "align /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/tmp/8690177688940149169/linclust/9500176059251995611/input_step_redundancy /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/tmp/8690177688940149169/linclust/9500176059251995611/input_step_redundancy /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/tmp/8690177688940149169/linclust/9500176059251995611/pref_rescore2 /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/tmp/8690177688940149169/linclust/9500176059251995611/aln --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' -a 0 --alignment-mode 3 --alignment-output-mode 0 --wrapped-scoring 0 -e 0.001 --min-seq-id 0.5 --min-aln-len 0 --seq-id-mode 0 --alt-ali 0 -c 0.5 --cov-mode 0 --max-seq-len 65535 --comp-bias-corr 1 --comp-bias-corr-scale 1 --max-rejected 2147483647 --max-accept 2147483647 --add-self-matches 0 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --score-bias 0 --realign 0 --realign-score-bias -0.2 --realign-max-seqs 2147483647 --corr-score-weight 0 --gap-open aa:11,nucl:5 --gap-extend aa:1,nucl:2 --zdrop 40 --threads 24 --compressed 0 -v 3 \n",
      "\n",
      "Compute score, coverage and sequence identity\n",
      "Query database size: 93442 type: Aminoacid\n",
      "Target database size: 93442 type: Aminoacid\n",
      "Calculation of alignments\n",
      "[=================================================================] 93.44K 3s 874ms\n",
      "Time for merging to aln: 0h 0m 0s 425ms\n",
      "115946 alignments calculated\n",
      "110333 sequence pairs passed the thresholds (0.951590 of overall calculated)\n",
      "1.180765 hits per query sequence\n",
      "Time for processing: 0h 0m 5s 590ms\n",
      "clust /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/tmp/8690177688940149169/linclust/9500176059251995611/input_step_redundancy /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/tmp/8690177688940149169/linclust/9500176059251995611/aln /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/tmp/8690177688940149169/linclust/9500176059251995611/clust --cluster-mode 0 --max-iterations 1000 --similarity-type 2 --threads 24 --compressed 0 -v 3 \n",
      "\n",
      "Clustering mode: Set Cover\n",
      "[=================================================================] 93.44K 0s 5ms\n",
      "Sort entries\n",
      "Find missing connections\n",
      "Found 16891 new connections.\n",
      "Reconstruct initial order\n",
      "[=================================================================] 93.44K 0s 8ms\n",
      "Add missing connections\n",
      "[=================================================================] 93.44K 0s 3ms\n",
      "\n",
      "Time for read in: 0h 0m 0s 42ms\n",
      "Total time: 0h 0m 0s 215ms\n",
      "\n",
      "Size of the sequence database: 93442\n",
      "Size of the alignment database: 93442\n",
      "Number of clusters: 80530\n",
      "\n",
      "Writing results 0h 0m 0s 15ms\n",
      "Time for merging to clust: 0h 0m 0s 58ms\n",
      "Time for processing: 0h 0m 0s 402ms\n",
      "mergeclusters /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/MMseqsDB/species_and_humanDB /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/tmp/8690177688940149169/clu_redundancy /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/tmp/8690177688940149169/linclust/9500176059251995611/pre_clust /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/tmp/8690177688940149169/linclust/9500176059251995611/clust --threads 24 --compressed 0 -v 3 \n",
      "\n",
      "Clustering step 1\n",
      "[=================================================================] 93.44K 0s 22ms\n",
      "Clustering step 2\n",
      "[=================================================================] 80.53K 0s 71ms\n",
      "Write merged clustering\n",
      "[=================================================================] 142.52K 1s 153ms\n",
      "Time for merging to clu_redundancy: 0h 0m 0s 374ms\n",
      "Time for processing: 0h 0m 1s 702ms\n",
      "createsubdb /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/tmp/8690177688940149169/clu_redundancy /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/MMseqsDB/species_and_humanDB /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/tmp/8690177688940149169/input_step_redundancy -v 3 --subdb-mode 1 \n",
      "\n",
      "Time for merging to input_step_redundancy: 0h 0m 0s 44ms\n",
      "Time for processing: 0h 0m 0s 310ms\n",
      "prefilter /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/tmp/8690177688940149169/input_step_redundancy /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/tmp/8690177688940149169/input_step_redundancy /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/tmp/8690177688940149169/pref_step0 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' --seed-sub-mat 'aa:VTML80.out,nucl:nucleotide.out' -s 1 -k 0 --k-score seq:2147483647,prof:2147483647 --alph-size aa:21,nucl:5 --max-seq-len 65535 --max-seqs 20 --split 0 --split-mode 2 --split-memory-limit 0 -c 0.5 --cov-mode 0 --comp-bias-corr 0 --comp-bias-corr-scale 1 --diag-score 0 --exact-kmer-matching 0 --mask 1 --mask-prob 0.9 --mask-lower-case 0 --min-ungapped-score 0 --add-self-matches 0 --spaced-kmer-mode 1 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --threads 24 --compressed 0 -v 3 \n",
      "\n",
      "Query database size: 80530 type: Aminoacid\n",
      "Estimated memory consumption: 1G\n",
      "Target database size: 80530 type: Aminoacid\n",
      "Index table k-mer threshold: 154 at k-mer size 6 \n",
      "Index table: counting k-mers\n",
      "[=================================================================] 80.53K 0s 228ms\n",
      "Index table: Masked residues: 919286\n",
      "Index table: fill\n",
      "[=================================================================] 80.53K 0s 108ms\n",
      "Index statistics\n",
      "Entries:          8856495\n",
      "DB size:          538 MB\n",
      "Avg k-mer size:   0.138383\n",
      "Top 10 k-mers\n",
      "    LYDMNY\t153\n",
      "    TFDAFT\t126\n",
      "    VADTEW\t85\n",
      "    IHDKNI\t65\n",
      "    EWLDQK\t65\n",
      "    RDMERN\t61\n",
      "    YRVLYR\t60\n",
      "    ECKFSS\t59\n",
      "    GGDCDS\t47\n",
      "    CQDGLV\t46\n",
      "Time for index table init: 0h 0m 0s 781ms\n",
      "Process prefiltering step 1 of 1\n",
      "\n",
      "k-mer similarity threshold: 154\n",
      "Starting prefiltering scores calculation (step 1 of 1)\n",
      "Query db start 1 to 80530\n",
      "Target db start 1 to 80530\n",
      "[=================================================================] 80.53K 0s 355ms\n",
      "\n",
      "2.278508 k-mers per position\n",
      "372 DB matches per sequence\n",
      "0 overflows\n",
      "0 queries produce too many hits (truncated result)\n",
      "3 sequences passed prefiltering per query sequence\n",
      "2 median result list length\n",
      "0 sequences with 0 size result lists\n",
      "Time for merging to pref_step0: 0h 0m 0s 378ms\n",
      "Time for processing: 0h 0m 2s 904ms\n",
      "align /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/tmp/8690177688940149169/input_step_redundancy /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/tmp/8690177688940149169/input_step_redundancy /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/tmp/8690177688940149169/pref_step0 /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/tmp/8690177688940149169/aln_step0 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' -a 0 --alignment-mode 3 --alignment-output-mode 0 --wrapped-scoring 0 -e 0.001 --min-seq-id 0.5 --min-aln-len 0 --seq-id-mode 0 --alt-ali 0 -c 0.5 --cov-mode 0 --max-seq-len 65535 --comp-bias-corr 0 --comp-bias-corr-scale 1 --max-rejected 2147483647 --max-accept 2147483647 --add-self-matches 0 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --score-bias 0 --realign 0 --realign-score-bias -0.2 --realign-max-seqs 2147483647 --corr-score-weight 0 --gap-open aa:11,nucl:5 --gap-extend aa:1,nucl:2 --zdrop 40 --threads 24 --compressed 0 -v 3 \n",
      "\n",
      "Compute score, coverage and sequence identity\n",
      "Query database size: 80530 type: Aminoacid\n",
      "Target database size: 80530 type: Aminoacid\n",
      "Calculation of alignments\n",
      "[=================================================================] 80.53K 12s 907ms\n",
      "Time for merging to aln_step0: 0h 0m 0s 369ms\n",
      "205162 alignments calculated\n",
      "102107 sequence pairs passed the thresholds (0.497690 of overall calculated)\n",
      "1.267937 hits per query sequence\n",
      "Time for processing: 0h 0m 14s 566ms\n",
      "clust /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/tmp/8690177688940149169/input_step_redundancy /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/tmp/8690177688940149169/aln_step0 /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/tmp/8690177688940149169/clu_step0 --cluster-mode 0 --max-iterations 1000 --similarity-type 2 --threads 24 --compressed 0 -v 3 \n",
      "\n",
      "Clustering mode: Set Cover\n",
      "[=================================================================] 80.53K 0s 6ms\n",
      "Sort entries\n",
      "Find missing connections\n",
      "Found 1889 new connections.\n",
      "Reconstruct initial order\n",
      "[=================================================================] 80.53K 0s 7ms\n",
      "Add missing connections\n",
      "[=================================================================] 80.53K 0s 2ms\n",
      "\n",
      "Time for read in: 0h 0m 0s 31ms\n",
      "Total time: 0h 0m 0s 124ms\n",
      "\n",
      "Size of the sequence database: 80530\n",
      "Size of the alignment database: 80530\n",
      "Number of clusters: 72672\n",
      "\n",
      "Writing results 0h 0m 0s 12ms\n",
      "Time for merging to clu_step0: 0h 0m 0s 80ms\n",
      "Time for processing: 0h 0m 0s 288ms\n",
      "createsubdb /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/tmp/8690177688940149169/clu_step0 /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/tmp/8690177688940149169/input_step_redundancy /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/tmp/8690177688940149169/input_step1 -v 3 --subdb-mode 1 \n",
      "\n",
      "Time for merging to input_step1: 0h 0m 0s 71ms\n",
      "Time for processing: 0h 0m 0s 299ms\n",
      "prefilter /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/tmp/8690177688940149169/input_step1 /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/tmp/8690177688940149169/input_step1 /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/tmp/8690177688940149169/pref_step1 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' --seed-sub-mat 'aa:VTML80.out,nucl:nucleotide.out' -s 2 -k 0 --k-score seq:2147483647,prof:2147483647 --alph-size aa:21,nucl:5 --max-seq-len 65535 --max-seqs 20 --split 0 --split-mode 2 --split-memory-limit 0 -c 0.5 --cov-mode 0 --comp-bias-corr 1 --comp-bias-corr-scale 1 --diag-score 1 --exact-kmer-matching 0 --mask 1 --mask-prob 0.9 --mask-lower-case 0 --min-ungapped-score 15 --add-self-matches 0 --spaced-kmer-mode 1 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --threads 24 --compressed 0 -v 3 \n",
      "\n",
      "Query database size: 72672 type: Aminoacid\n",
      "Estimated memory consumption: 1G\n",
      "Target database size: 72672 type: Aminoacid\n",
      "Index table k-mer threshold: 145 at k-mer size 6 \n",
      "Index table: counting k-mers\n",
      "[=================================================================] 72.67K 0s 192ms\n",
      "Index table: Masked residues: 603037\n",
      "Index table: fill\n",
      "[=================================================================] 72.67K 0s 114ms\n",
      "Index statistics\n",
      "Entries:          9444792\n",
      "DB size:          542 MB\n",
      "Avg k-mer size:   0.147575\n",
      "Top 10 k-mers\n",
      "    DVVFEE\t113\n",
      "    QRLRML\t60\n",
      "    LYDMNY\t60\n",
      "    TFDAFT\t54\n",
      "    VMEYLV\t54\n",
      "    CQDGLV\t34\n",
      "    VADTEW\t34\n",
      "    LLPQLY\t34\n",
      "    VHSGRT\t30\n",
      "    PAPPSP\t28\n",
      "Time for index table init: 0h 0m 0s 737ms\n",
      "Process prefiltering step 1 of 1\n",
      "\n",
      "k-mer similarity threshold: 145\n",
      "Starting prefiltering scores calculation (step 1 of 1)\n",
      "Query db start 1 to 72672\n",
      "Target db start 1 to 72672\n",
      "[=================================================================] 72.67K 0s 794ms\n",
      "\n",
      "6.012728 k-mers per position\n",
      "316 DB matches per sequence\n",
      "0 overflows\n",
      "0 queries produce too many hits (truncated result)\n",
      "3 sequences passed prefiltering per query sequence\n",
      "1 median result list length\n",
      "0 sequences with 0 size result lists\n",
      "Time for merging to pref_step1: 0h 0m 0s 419ms\n",
      "Time for processing: 0h 0m 3s 70ms\n",
      "align /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/tmp/8690177688940149169/input_step1 /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/tmp/8690177688940149169/input_step1 /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/tmp/8690177688940149169/pref_step1 /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/tmp/8690177688940149169/aln_step1 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' -a 0 --alignment-mode 3 --alignment-output-mode 0 --wrapped-scoring 0 -e 0.001 --min-seq-id 0.5 --min-aln-len 0 --seq-id-mode 0 --alt-ali 0 -c 0.5 --cov-mode 0 --max-seq-len 65535 --comp-bias-corr 1 --comp-bias-corr-scale 1 --max-rejected 2147483647 --max-accept 2147483647 --add-self-matches 0 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --score-bias 0 --realign 0 --realign-score-bias -0.2 --realign-max-seqs 2147483647 --corr-score-weight 0 --gap-open aa:11,nucl:5 --gap-extend aa:1,nucl:2 --zdrop 40 --threads 24 --compressed 0 -v 3 \n",
      "\n",
      "Compute score, coverage and sequence identity\n",
      "Query database size: 72672 type: Aminoacid\n",
      "Target database size: 72672 type: Aminoacid\n",
      "Calculation of alignments\n",
      "[=================================================================] 72.67K 4s 452ms\n",
      "Time for merging to aln_step1: 0h 0m 0s 413ms\n",
      "154144 alignments calculated\n",
      "73442 sequence pairs passed the thresholds (0.476451 of overall calculated)\n",
      "1.010596 hits per query sequence\n",
      "Time for processing: 0h 0m 6s 133ms\n",
      "clust /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/tmp/8690177688940149169/input_step1 /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/tmp/8690177688940149169/aln_step1 /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/tmp/8690177688940149169/clu_step1 --cluster-mode 0 --max-iterations 1000 --similarity-type 2 --threads 24 --compressed 0 -v 3 \n",
      "\n",
      "Clustering mode: Set Cover\n",
      "[=================================================================] 72.67K 0s 5ms\n",
      "Sort entries\n",
      "Find missing connections\n",
      "Found 238 new connections.\n",
      "Reconstruct initial order\n",
      "[=================================================================] 72.67K 0s 3ms\n",
      "Add missing connections\n",
      "[=================================================================] 72.67K 0s 2ms\n",
      "\n",
      "Time for read in: 0h 0m 0s 45ms\n",
      "Total time: 0h 0m 0s 149ms\n",
      "\n",
      "Size of the sequence database: 72672\n",
      "Size of the alignment database: 72672\n",
      "Number of clusters: 72202\n",
      "\n",
      "Writing results 0h 0m 0s 12ms\n",
      "Time for merging to clu_step1: 0h 0m 0s 35ms\n",
      "Time for processing: 0h 0m 0s 303ms\n",
      "createsubdb /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/tmp/8690177688940149169/clu_step1 /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/tmp/8690177688940149169/input_step1 /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/tmp/8690177688940149169/input_step2 -v 3 --subdb-mode 1 \n",
      "\n",
      "Time for merging to input_step2: 0h 0m 0s 135ms\n",
      "Time for processing: 0h 0m 0s 388ms\n",
      "prefilter /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/tmp/8690177688940149169/input_step2 /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/tmp/8690177688940149169/input_step2 /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/tmp/8690177688940149169/pref_step2 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' --seed-sub-mat 'aa:VTML80.out,nucl:nucleotide.out' -s 3 -k 0 --k-score seq:2147483647,prof:2147483647 --alph-size aa:21,nucl:5 --max-seq-len 65535 --max-seqs 20 --split 0 --split-mode 2 --split-memory-limit 0 -c 0.5 --cov-mode 0 --comp-bias-corr 1 --comp-bias-corr-scale 1 --diag-score 1 --exact-kmer-matching 0 --mask 1 --mask-prob 0.9 --mask-lower-case 0 --min-ungapped-score 15 --add-self-matches 0 --spaced-kmer-mode 1 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --threads 24 --compressed 0 -v 3 \n",
      "\n",
      "Query database size: 72202 type: Aminoacid\n",
      "Estimated memory consumption: 1G\n",
      "Target database size: 72202 type: Aminoacid\n",
      "Index table k-mer threshold: 136 at k-mer size 6 \n",
      "Index table: counting k-mers\n",
      "[=================================================================] 72.20K 0s 196ms\n",
      "Index table: Masked residues: 579549\n",
      "Index table: fill\n",
      "[=================================================================] 72.20K 0s 140ms\n",
      "Index statistics\n",
      "Entries:          12133590\n",
      "DB size:          557 MB\n",
      "Avg k-mer size:   0.189587\n",
      "Top 10 k-mers\n",
      "    DVVFEE\t86\n",
      "    VLQLLP\t65\n",
      "    QRLRML\t48\n",
      "    QIELEQ\t48\n",
      "    LYDMNY\t47\n",
      "    VMEYLV\t44\n",
      "    TKQLLV\t38\n",
      "    CQDGLV\t34\n",
      "    APPPPP\t33\n",
      "    RDARLV\t31\n",
      "Time for index table init: 0h 0m 0s 763ms\n",
      "Process prefiltering step 1 of 1\n",
      "\n",
      "k-mer similarity threshold: 136\n",
      "Starting prefiltering scores calculation (step 1 of 1)\n",
      "Query db start 1 to 72202\n",
      "Target db start 1 to 72202\n",
      "[=================================================================] 72.20K 1s 365ms\n",
      "\n",
      "17.172991 k-mers per position\n",
      "762 DB matches per sequence\n",
      "0 overflows\n",
      "0 queries produce too many hits (truncated result)\n",
      "5 sequences passed prefiltering per query sequence\n",
      "3 median result list length\n",
      "0 sequences with 0 size result lists\n",
      "Time for merging to pref_step2: 0h 0m 0s 328ms\n",
      "Time for processing: 0h 0m 3s 891ms\n",
      "align /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/tmp/8690177688940149169/input_step2 /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/tmp/8690177688940149169/input_step2 /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/tmp/8690177688940149169/pref_step2 /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/tmp/8690177688940149169/aln_step2 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' -a 0 --alignment-mode 3 --alignment-output-mode 0 --wrapped-scoring 0 -e 0.001 --min-seq-id 0.5 --min-aln-len 0 --seq-id-mode 0 --alt-ali 0 -c 0.5 --cov-mode 0 --max-seq-len 65535 --comp-bias-corr 1 --comp-bias-corr-scale 1 --max-rejected 2147483647 --max-accept 2147483647 --add-self-matches 0 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --score-bias 0 --realign 0 --realign-score-bias -0.2 --realign-max-seqs 2147483647 --corr-score-weight 0 --gap-open aa:11,nucl:5 --gap-extend aa:1,nucl:2 --zdrop 40 --threads 24 --compressed 0 -v 3 \n",
      "\n",
      "Compute score, coverage and sequence identity\n",
      "Query database size: 72202 type: Aminoacid\n",
      "Target database size: 72202 type: Aminoacid\n",
      "Calculation of alignments\n",
      "[=================================================================] 72.20K 4s 911ms\n",
      "Time for merging to aln_step2: 0h 0m 0s 418ms\n",
      "208618 alignments calculated\n",
      "72334 sequence pairs passed the thresholds (0.346729 of overall calculated)\n",
      "1.001828 hits per query sequence\n",
      "Time for processing: 0h 0m 6s 302ms\n",
      "clust /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/tmp/8690177688940149169/input_step2 /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/tmp/8690177688940149169/aln_step2 /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/tmp/8690177688940149169/clu_step2 --cluster-mode 0 --max-iterations 1000 --similarity-type 2 --threads 24 --compressed 0 -v 3 \n",
      "\n",
      "Clustering mode: Set Cover\n",
      "[=================================================================] 72.20K 0s 7ms\n",
      "Sort entries\n",
      "Find missing connections\n",
      "Found 48 new connections.\n",
      "Reconstruct initial order\n",
      "[=================================================================] 72.20K 0s 7ms\n",
      "Add missing connections\n",
      "[=================================================================] 72.20K 0s 1ms\n",
      "\n",
      "Time for read in: 0h 0m 0s 40ms\n",
      "Total time: 0h 0m 0s 114ms\n",
      "\n",
      "Size of the sequence database: 72202\n",
      "Size of the alignment database: 72202\n",
      "Number of clusters: 72112\n",
      "\n",
      "Writing results 0h 0m 0s 11ms\n",
      "Time for merging to clu_step2: 0h 0m 0s 65ms\n",
      "Time for processing: 0h 0m 0s 493ms\n",
      "mergeclusters /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/MMseqsDB/species_and_humanDB /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/Clustering/species_and_human_cluDB /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/tmp/8690177688940149169/clu_redundancy /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/tmp/8690177688940149169/clu_step0 /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/tmp/8690177688940149169/clu_step1 /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/tmp/8690177688940149169/clu_step2 --threads 24 --compressed 0 -v 3 \n",
      "\n",
      "Clustering step 1\n",
      "[=================================================================] 80.53K 0s 70ms\n",
      "Clustering step 2\n",
      "[=================================================================] 72.67K 0s 101ms\n",
      "Clustering step 3\n",
      "[=================================================================] 72.20K 0s 130ms\n",
      "Clustering step 4\n",
      "[=================================================================] 72.11K 0s 166ms\n",
      "Write merged clustering\n",
      "[=================================================================] 142.52K 0s 974ms\n",
      "Time for merging to species_and_human_cluDB: 0h 0m 0s 386ms\n",
      "Time for processing: 0h 0m 1s 559ms\n",
      "createtsv /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/MMseqsDB/species_and_humanDB /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/MMseqsDB/species_and_humanDB /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/Clustering/species_and_human_cluDB /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/v47/EvolutionaryOrigin_MMseqs/Clustering/species_and_human_cluDB.tsv \n",
      "\n",
      "MMseqs Version:                 \t14.7e284\n",
      "First sequence as representative\tfalse\n",
      "Target column                   \t1\n",
      "Add full header                 \tfalse\n",
      "Sequence source                 \t0\n",
      "Database output                 \tfalse\n",
      "Threads                         \t24\n",
      "Compressed                      \t0\n",
      "Verbosity                       \t3\n",
      "\n",
      "Time for merging to species_and_human_cluDB.tsv: 0h 0m 0s 673ms\n",
      "Time for processing: 0h 0m 1s 817ms\n"
     ]
    }
   ],
   "source": [
    "%%bash -s \"$evoDir\" \"$users_dir\"\n",
    "\n",
    "module load Miniconda3/4.9.2\n",
    "source /soft/system/software/Miniconda3/4.9.2/bin/activate MMseqs2\n",
    "\n",
    "mkdir -p $1/Clustering\n",
    "\n",
    "rm -r $1/Clustering/species_and_human_cluDB*\n",
    "\n",
    "## cluster all species and human\n",
    "mmseqs cluster -c 0.5 --min-seq-id 0.5 $1/MMseqsDB/species_and_humanDB $1/Clustering/species_and_human_cluDB $1/tmp\n",
    "mmseqs createtsv $1/MMseqsDB/species_and_humanDB $1/MMseqsDB/species_and_humanDB $1/Clustering/species_and_human_cluDB $1/Clustering/species_and_human_cluDB.tsv"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Analysis"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "all human\n",
      "25675\n",
      "all human after annotation\n",
      "25675\n",
      "noncoding all\n",
      "3766\n",
      "3027\n",
      "Conserved noncoding human ORFs\n",
      "216\n",
      "Non-Conserved noncoding human ORFs\n",
      "3027\n",
      "Canonical all\n",
      "14204\n",
      "2899\n",
      "Conserved canonical human ORFs\n",
      "2585\n",
      "Conserved canonical human ORFs as noncoding in other species\n",
      "309\n",
      "Conserved canonical human ORFs as canonical in other species\n",
      "2573\n",
      "Non-Conserved canonical human ORFs\n",
      "2899\n"
     ]
    }
   ],
   "source": [
    "clustering = pd.read_csv(os.path.join(evoDir,\"Clustering/species_and_human_cluDB.tsv\"), header=None, sep=\"\\t\")\n",
    "clustering = clustering[clustering[0].str.startswith(\"ENST\")]\n",
    "print(\"all human\")\n",
    "print(len(clustering))\n",
    "clustering['transcript_id'] = clustering[0].str.split(\":\", expand=True)[0]\n",
    "clustering['transcript_id'] = clustering[0].str.split(\".\", expand=True)[0]\n",
    "\n",
    "clustering = clustering.merge(transcript_gene, on=\"transcript_id\")\n",
    "print(\"all human after annotation\")\n",
    "print(len(clustering))\n",
    "clustering = clustering[[\"gene_name\",\"gene_type\",0,1]]\n",
    "\n",
    "## non-coding\n",
    "noncoding = clustering[clustering[0].str.contains(\"noncoding\")]\n",
    "noncoding.to_csv(os.path.join(evoDir,\"Clustering/noncodingHuman_speciesDB.tsv\"), header=None, index=None, sep=\"\\t\")\n",
    "print(\"noncoding all\")\n",
    "print(len(noncoding))\n",
    "print(len(set(noncoding[0].values.tolist())))\n",
    "\n",
    "noncoding_conserved = noncoding[~noncoding[1].str.startswith(\"ENST\")]\n",
    "noncoding_conserved.to_csv(os.path.join(evoDir,\"Clustering/noncodingHuman_Conserved.tsv\"), header=None, index=None, sep=\"\\t\")\n",
    "print(\"Conserved noncoding human ORFs\")\n",
    "print(len(set(noncoding_conserved[0].values.tolist())))\n",
    "noncoding_conserved.columns = ['gene_name','gene_type','orfID',1]\n",
    "\n",
    "noncoding_NonConserved = noncoding[noncoding[1].str.startswith(\"ENST\")]\n",
    "noncoding_NonConserved.to_csv(os.path.join(evoDir,\"Clustering/noncodingHuman_NonConserved.tsv\"), header=None, index=None, sep=\"\\t\")\n",
    "print(\"Non-Conserved noncoding human ORFs\")\n",
    "print(len(set(noncoding_NonConserved[0].values.tolist())))\n",
    "noncoding_NonConserved.columns = ['gene_name','gene_type','orfID',1]\n",
    "\n",
    "\n",
    "## canonical\n",
    "canonical = clustering[clustering[0].str.contains(\"canonical\")]\n",
    "print(\"Canonical all\")\n",
    "print(len(canonical))\n",
    "print(len(set(canonical[0].values.tolist())))\n",
    "canonical.to_csv(os.path.join(evoDir,\"Clustering/canonicalHuman_speciesDB.tsv\"), header=None, index=None, sep=\"\\t\")\n",
    "\n",
    "canonical_conserved = canonical[~canonical[1].str.startswith(\"ENST\")]\n",
    "canonical_conserved.to_csv(os.path.join(evoDir,\"Clustering/canonicalHuman_Conserved.tsv\"), header=None, index=None, sep=\"\\t\")\n",
    "print(\"Conserved canonical human ORFs\")\n",
    "print(len(set(canonical_conserved[0].values.tolist())))\n",
    "canonical_conserved.columns = ['gene_name','gene_type','orfID',1]\n",
    "\n",
    "canonical_conserved_noncoding = canonical_conserved[~canonical_conserved[1].str.contains(\"canonical\")]\n",
    "canonical_conserved_noncoding.to_csv(os.path.join(evoDir,\"Clustering/canonicalHuman_ConservedAsNoncoding.tsv\"), header=None, index=None, sep=\"\\t\")\n",
    "print(\"Conserved canonical human ORFs as noncoding in other species\")\n",
    "print(len(set(canonical_conserved_noncoding['orfID'].values.tolist())))\n",
    "\n",
    "canonical_conserved_canonical = canonical_conserved[canonical_conserved[1].str.contains(\"canonical\")]\n",
    "canonical_conserved_canonical.to_csv(os.path.join(evoDir,\"Clustering/canonicalHuman_ConservedAsCanonical.tsv\"), header=None, index=None, sep=\"\\t\")\n",
    "print(\"Conserved canonical human ORFs as canonical in other species\")\n",
    "print(len(set(canonical_conserved_canonical['orfID'].values.tolist())))\n",
    "\n",
    "canonical_NonConserved = canonical[canonical[1].str.startswith(\"ENST\")]\n",
    "canonical_NonConserved.to_csv(os.path.join(evoDir,\"Clustering/canonicalHuman_NonConserved.tsv\"), header=None, index=None, sep=\"\\t\")\n",
    "print(\"Non-Conserved canonical human ORFs\")\n",
    "print(len(set(canonical_NonConserved[0].values.tolist())))\n",
    "canonical_NonConserved.columns = ['gene_name','gene_type','orfID',1]\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene_name</th>\n",
       "      <th>gene_type</th>\n",
       "      <th>orfID</th>\n",
       "      <th>1</th>\n",
       "      <th>specie</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>ENSG00000288700</td>\n",
       "      <td>lncRNA</td>\n",
       "      <td>ENST00000683452.1:3:+|19|3753:358:1843|noncodi...</td>\n",
       "      <td>macaca_ENSMMUT00000083577:2:+|3|2913:26:2021|c...</td>\n",
       "      <td>macaca</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>ENSG00000288700</td>\n",
       "      <td>lncRNA</td>\n",
       "      <td>ENST00000683452.1:3:+|19|3753:358:1843|noncodi...</td>\n",
       "      <td>chicken_ENSGALT00010059861:9:+|25|2797:297:243...</td>\n",
       "      <td>chicken</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>ENSG00000288700</td>\n",
       "      <td>lncRNA</td>\n",
       "      <td>ENST00000683452.1:3:+|19|3753:358:1843|noncodi...</td>\n",
       "      <td>opossum_ENSMODT00000064442:4:+|17|2707:380:235...</td>\n",
       "      <td>opossum</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>ENSG00000288700</td>\n",
       "      <td>lncRNA</td>\n",
       "      <td>ENST00000683452.1:3:+|19|3753:358:1843|noncodi...</td>\n",
       "      <td>mouse_ENSMUST00000162960.8:chr9:-|12|5861:295:...</td>\n",
       "      <td>mouse</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>ENSG00000288700</td>\n",
       "      <td>lncRNA</td>\n",
       "      <td>ENST00000683452.1:3:+|19|3753:358:1843|noncodi...</td>\n",
       "      <td>platypus_ENSOANT00000040792:1:+|44|2106:562:21...</td>\n",
       "      <td>platypus</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25597</th>\n",
       "      <td>ENSG00000258435</td>\n",
       "      <td>lncRNA</td>\n",
       "      <td>ENST00000718702.1:12:+|6|955:107:200|noncoding...</td>\n",
       "      <td>TCONS_00000607:12:+|6|619:107:200|noncoding|ATG</td>\n",
       "      <td>TCONS</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25602</th>\n",
       "      <td>ENSG00000298182</td>\n",
       "      <td>lncRNA</td>\n",
       "      <td>ENST00000753709.1:11:-|44|1051:563:803|noncodi...</td>\n",
       "      <td>TCONS_00000477:11:-|4|587:101:341|noncoding|ATG</td>\n",
       "      <td>TCONS</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25609</th>\n",
       "      <td>ENSG00000303265</td>\n",
       "      <td>lncRNA</td>\n",
       "      <td>ENST00000793280.1:18:+|12|1392:147:285|noncodi...</td>\n",
       "      <td>TCONS_00001196:18:+|16|1060:196:334|noncoding|ATG</td>\n",
       "      <td>TCONS</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25612</th>\n",
       "      <td>ENSG00000305541</td>\n",
       "      <td>lncRNA</td>\n",
       "      <td>ENST00000811623.1:1:-|1|798:54:429|noncoding|CTG</td>\n",
       "      <td>TCONS_00000085:1:-|1|798:54:429|noncoding|CTG</td>\n",
       "      <td>TCONS</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25614</th>\n",
       "      <td>ENSG00000306258</td>\n",
       "      <td>lncRNA</td>\n",
       "      <td>ENST00000816520.1:19:+|15|1238:381:660|noncodi...</td>\n",
       "      <td>TCONS_00001285:19:+|15|1238:381:660|noncoding|ATG</td>\n",
       "      <td>TCONS</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>526 rows × 5 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "             gene_name gene_type  \\\n",
       "9      ENSG00000288700    lncRNA   \n",
       "11     ENSG00000288700    lncRNA   \n",
       "12     ENSG00000288700    lncRNA   \n",
       "13     ENSG00000288700    lncRNA   \n",
       "14     ENSG00000288700    lncRNA   \n",
       "...                ...       ...   \n",
       "25597  ENSG00000258435    lncRNA   \n",
       "25602  ENSG00000298182    lncRNA   \n",
       "25609  ENSG00000303265    lncRNA   \n",
       "25612  ENSG00000305541    lncRNA   \n",
       "25614  ENSG00000306258    lncRNA   \n",
       "\n",
       "                                                   orfID  \\\n",
       "9      ENST00000683452.1:3:+|19|3753:358:1843|noncodi...   \n",
       "11     ENST00000683452.1:3:+|19|3753:358:1843|noncodi...   \n",
       "12     ENST00000683452.1:3:+|19|3753:358:1843|noncodi...   \n",
       "13     ENST00000683452.1:3:+|19|3753:358:1843|noncodi...   \n",
       "14     ENST00000683452.1:3:+|19|3753:358:1843|noncodi...   \n",
       "...                                                  ...   \n",
       "25597  ENST00000718702.1:12:+|6|955:107:200|noncoding...   \n",
       "25602  ENST00000753709.1:11:-|44|1051:563:803|noncodi...   \n",
       "25609  ENST00000793280.1:18:+|12|1392:147:285|noncodi...   \n",
       "25612   ENST00000811623.1:1:-|1|798:54:429|noncoding|CTG   \n",
       "25614  ENST00000816520.1:19:+|15|1238:381:660|noncodi...   \n",
       "\n",
       "                                                       1    specie  \n",
       "9      macaca_ENSMMUT00000083577:2:+|3|2913:26:2021|c...    macaca  \n",
       "11     chicken_ENSGALT00010059861:9:+|25|2797:297:243...   chicken  \n",
       "12     opossum_ENSMODT00000064442:4:+|17|2707:380:235...   opossum  \n",
       "13     mouse_ENSMUST00000162960.8:chr9:-|12|5861:295:...     mouse  \n",
       "14     platypus_ENSOANT00000040792:1:+|44|2106:562:21...  platypus  \n",
       "...                                                  ...       ...  \n",
       "25597    TCONS_00000607:12:+|6|619:107:200|noncoding|ATG     TCONS  \n",
       "25602    TCONS_00000477:11:-|4|587:101:341|noncoding|ATG     TCONS  \n",
       "25609  TCONS_00001196:18:+|16|1060:196:334|noncoding|ATG     TCONS  \n",
       "25612      TCONS_00000085:1:-|1|798:54:429|noncoding|CTG     TCONS  \n",
       "25614  TCONS_00001285:19:+|15|1238:381:660|noncoding|ATG     TCONS  \n",
       "\n",
       "[526 rows x 5 columns]"
      ]
     },
     "execution_count": 20,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "noncoding_conserved"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "216\n",
      "174\n",
      "174\n",
      "174\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_229902/2640938826.py:4: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  noncoding_conserved_to_matrix['specie'] = noncoding_conserved_to_matrix[1].str.split(\"_\", expand=True)[0]\n",
      "/tmp/ipykernel_229902/2640938826.py:6: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  noncoding_conserved_to_matrix['presence'] = 1\n",
      "/tmp/ipykernel_229902/2640938826.py:28: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  noncoding_conserved_matrix_ordered['sum'] = noncoding_conserved_matrix_ordered.select_dtypes(include='number').sum(axis=1)\n"
     ]
    }
   ],
   "source": [
    "noncoding_conserved_to_matrix = noncoding_conserved\n",
    "print(len(set(noncoding_conserved_to_matrix.orfID.values.tolist())))\n",
    "# Assuming noncoding_conserved_to_matrix is already defined\n",
    "noncoding_conserved_to_matrix['specie'] = noncoding_conserved_to_matrix[1].str.split(\"_\", expand=True)[0]\n",
    "noncoding_conserved_to_matrix = noncoding_conserved_to_matrix[['orfID', 'specie']]\n",
    "noncoding_conserved_to_matrix['presence'] = 1\n",
    "noncoding_conserved_to_matrix = noncoding_conserved_to_matrix[noncoding_conserved_to_matrix['specie'] != \"TCONS\"]\n",
    "\n",
    "print(len(set(noncoding_conserved_to_matrix.orfID.values.tolist())))\n",
    "noncoding_conserved_to_matrix\n",
    "\n",
    "# Pivoting the data without creating an 'id', and keeping 'orfID' as the unique identifier\n",
    "noncoding_conserved_matrix = noncoding_conserved_to_matrix.pivot_table(index='orfID', columns='specie', values='presence', aggfunc='max')\n",
    "\n",
    "# Fill NaN values with 0\n",
    "noncoding_conserved_matrix = noncoding_conserved_matrix.fillna(0)\n",
    "\n",
    "# Reset index so 'orfID' becomes a column instead of an index\n",
    "noncoding_conserved_matrix = noncoding_conserved_matrix.reset_index()\n",
    "\n",
    "# Rename columns as necessary\n",
    "noncoding_conserved_matrix.columns = [\"orfID\", \"chicken\", \"macaca\", \"mouse\", \"opossum\", \"platypus\"]\n",
    "\n",
    "# Reorder columns for the final matrix\n",
    "noncoding_conserved_matrix_ordered = noncoding_conserved_matrix[[\"orfID\", \"macaca\", \"mouse\", \"opossum\", \"platypus\", \"chicken\"]]\n",
    "\n",
    "# Calculate sum across species columns to identify how many species are present\n",
    "noncoding_conserved_matrix_ordered['sum'] = noncoding_conserved_matrix_ordered.select_dtypes(include='number').sum(axis=1)\n",
    "\n",
    "# Sort by the sum column to order by the number of species present\n",
    "noncoding_conserved_matrix_ordered = noncoding_conserved_matrix_ordered.sort_values(by=\"sum\", ascending=False)\n",
    "print(len(set(noncoding_conserved_matrix_ordered.orfID.values.tolist())))\n",
    "\n",
    "# Display the result\n",
    "noncoding_conserved_matrix_ordered =  noncoding_conserved_matrix_ordered.merge(noncoding_conserved[['gene_name','gene_type','orfID']], on=\"orfID\")\n",
    "noncoding_conserved_matrix_ordered.drop_duplicates(inplace=True)\n",
    "\n",
    "noncoding_conserved_matrix_ordered.to_csv(os.path.join(evoDir,\"Clustering/noncodingHuman_Conserved_matrix.tsv\"), sep=\"\\t\", index=None)\n",
    "print(len(set(noncoding_conserved_matrix_ordered.orfID.values.tolist())))\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "3027\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_229902/4251941356.py:3: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  noncoding_conserved_touse['specie'] = noncoding_conserved_touse[1].str.split(\"_\", expand=True)[0]\n",
      "/tmp/ipykernel_229902/4251941356.py:37: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  noncoding_matrix_ordered['sum'] = noncoding_matrix_ordered.select_dtypes(include='number').sum(axis=1)\n"
     ]
    }
   ],
   "source": [
    "### all noncoding\n",
    "noncoding_conserved_touse = noncoding_conserved\n",
    "noncoding_conserved_touse['specie'] = noncoding_conserved_touse[1].str.split(\"_\", expand=True)[0]\n",
    "# noncoding_NonConserved.drop(\"specie\",axis=1, inplace=True)\n",
    "noncoding_to_matrix_raw = pd.concat([noncoding_conserved, noncoding_NonConserved])\n",
    "\n",
    "noncoding_to_matrix = noncoding_to_matrix_raw[['orfID', 'specie']]\n",
    "noncoding_to_matrix = noncoding_to_matrix[noncoding_to_matrix['specie'] != \"TCONS\"]\n",
    "noncoding_to_matrix['presence'] = np.where(noncoding_to_matrix['specie'].isna(), 0, 1)\n",
    "\n",
    "\n",
    "human = pd.DataFrame({\n",
    "    'orfID': noncoding_to_matrix['orfID'].unique(),\n",
    "    'specie': 'human',\n",
    "    'presence': 1\n",
    "})\n",
    "\n",
    "noncoding_to_matrix = pd.concat([human, noncoding_to_matrix], ignore_index=True)\n",
    "noncoding_to_matrix.dropna(inplace=True)\n",
    "\n",
    "# Pivoting the data without creating an 'id', and keeping 'orfID' as the unique identifier\n",
    "noncoding_matrix = noncoding_to_matrix.pivot_table(index='orfID', columns='specie', values='presence', aggfunc='max')\n",
    "noncoding_matrix\n",
    "# Fill NaN values with 0\n",
    "noncoding_matrix = noncoding_matrix.fillna(0)\n",
    "\n",
    "# Reset index so 'orfID' becomes a column instead of an index\n",
    "noncoding_matrix = noncoding_matrix.reset_index()\n",
    "\n",
    "# Rename columns as necessary\n",
    "noncoding_matrix.columns = [\"orfID\", \"chicken\", \"human\", \"macaca\", \"mouse\", \"opossum\", \"platypus\"]\n",
    "\n",
    "# Reorder columns for the final matrix\n",
    "noncoding_matrix_ordered = noncoding_matrix[[\"orfID\", \"human\",\"macaca\", \"mouse\", \"opossum\", \"platypus\", \"chicken\"]]\n",
    "\n",
    "# Calculate sum across species columns to identify how many species are present\n",
    "noncoding_matrix_ordered['sum'] = noncoding_matrix_ordered.select_dtypes(include='number').sum(axis=1)\n",
    "\n",
    "# Sort by the sum column to order by the number of species present\n",
    "noncoding_matrix_ordered = noncoding_matrix_ordered.sort_values(by=\"sum\", ascending=False)\n",
    "\n",
    "# Display the result\n",
    "noncoding_matrix_ordered =  noncoding_matrix_ordered.merge(noncoding_to_matrix_raw[['gene_name','gene_type','orfID']], on=\"orfID\")\n",
    "noncoding_matrix_ordered.drop_duplicates(inplace=True)\n",
    "\n",
    "noncoding_matrix_ordered.to_csv(os.path.join(evoDir,\"Clustering/noncodingHuman_matrix.tsv\"), sep=\"\\t\", index=None)\n",
    "print(len(set(noncoding_matrix_ordered.orfID.values.tolist())))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_229902/1360690153.py:4: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  canonical_conserved_to_matrix['specie'] = canonical_conserved_to_matrix[1].str.split(\"_\", expand=True)[0]\n",
      "/tmp/ipykernel_229902/1360690153.py:6: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  canonical_conserved_to_matrix['presence'] = 1\n",
      "/tmp/ipykernel_229902/1360690153.py:25: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  canonical_conserved_matrix_ordered['sum'] = canonical_conserved_matrix_ordered.select_dtypes(include='number').sum(axis=1)\n"
     ]
    }
   ],
   "source": [
    "canonical_conserved_to_matrix = canonical_conserved\n",
    "\n",
    "# Assuming noncoding_conserved_to_matrix is already defined\n",
    "canonical_conserved_to_matrix['specie'] = canonical_conserved_to_matrix[1].str.split(\"_\", expand=True)[0]\n",
    "canonical_conserved_to_matrix = canonical_conserved_to_matrix[['orfID', 'specie']]\n",
    "canonical_conserved_to_matrix['presence'] = 1\n",
    "canonical_conserved_to_matrix = canonical_conserved_to_matrix[canonical_conserved_to_matrix['specie'] != \"TCONS\"]\n",
    "\n",
    "# Pivoting the data without creating an 'id', and keeping 'orfID' as the unique identifier\n",
    "canonical_conserved_matrix = canonical_conserved_to_matrix.pivot_table(index='orfID', columns='specie', values='presence', aggfunc='max')\n",
    "\n",
    "# Fill NaN values with 0\n",
    "canonical_conserved_matrix = canonical_conserved_matrix.fillna(0)\n",
    "\n",
    "# Reset index so 'orfID' becomes a column instead of an index\n",
    "canonical_conserved_matrix = canonical_conserved_matrix.reset_index()\n",
    "\n",
    "# Rename columns as necessary\n",
    "canonical_conserved_matrix.columns = [\"orfID\", \"chicken\", \"macaca\", \"mouse\", \"opossum\", \"platypus\"]\n",
    "\n",
    "# Reorder columns for the final matrix\n",
    "canonical_conserved_matrix_ordered = canonical_conserved_matrix[[\"orfID\", \"macaca\", \"mouse\", \"opossum\", \"platypus\", \"chicken\"]]\n",
    "\n",
    "# Calculate sum across species columns to identify how many species are present\n",
    "canonical_conserved_matrix_ordered['sum'] = canonical_conserved_matrix_ordered.select_dtypes(include='number').sum(axis=1)\n",
    "\n",
    "# Sort by the sum column to order by the number of species present\n",
    "canonical_conserved_matrix_ordered = canonical_conserved_matrix_ordered.sort_values(by=\"sum\", ascending=False)\n",
    "\n",
    "# Display the result\n",
    "canonical_conserved_matrix_ordered =  canonical_conserved_matrix_ordered.merge(canonical_conserved[['gene_name','gene_type','orfID']], on=\"orfID\")\n",
    "canonical_conserved_matrix_ordered.drop_duplicates(inplace=True)\n",
    "\n",
    "canonical_conserved_matrix_ordered.to_csv(os.path.join(evoDir,\"Clustering/canonicalHuman_Conserved_matrix.tsv\"), sep=\"\\t\", index=None)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_229902/2620241481.py:3: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  canonical_conserved_touse['specie'] = canonical_conserved_touse[1].str.split(\"_\", expand=True)[0]\n",
      "/tmp/ipykernel_229902/2620241481.py:37: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  canonical_matrix_ordered['sum'] = canonical_matrix_ordered.select_dtypes(include='number').sum(axis=1)\n"
     ]
    }
   ],
   "source": [
    "### all canonical\n",
    "canonical_conserved_touse = canonical_conserved\n",
    "canonical_conserved_touse['specie'] = canonical_conserved_touse[1].str.split(\"_\", expand=True)[0]\n",
    "# noncoding_NonConserved.drop(\"specie\",axis=1, inplace=True)\n",
    "canonical_to_matrix_raw = pd.concat([canonical_conserved, canonical_NonConserved])\n",
    "\n",
    "canonical_to_matrix = canonical_to_matrix_raw[['orfID', 'specie']]\n",
    "canonical_to_matrix = canonical_to_matrix[canonical_to_matrix['specie'] != \"TCONS\"]\n",
    "canonical_to_matrix['presence'] = np.where(canonical_to_matrix['specie'].isna(), 0, 1)\n",
    "\n",
    "\n",
    "human = pd.DataFrame({\n",
    "    'orfID': canonical_to_matrix['orfID'].unique(),\n",
    "    'specie': 'human',\n",
    "    'presence': 1\n",
    "})\n",
    "\n",
    "canonical_to_matrix = pd.concat([human, canonical_to_matrix], ignore_index=True)\n",
    "canonical_to_matrix.dropna(inplace=True)\n",
    "\n",
    "# Pivoting the data without creating an 'id', and keeping 'orfID' as the unique identifier\n",
    "canonical_matrix = canonical_to_matrix.pivot_table(index='orfID', columns='specie', values='presence', aggfunc='max')\n",
    "canonical_matrix\n",
    "# Fill NaN values with 0\n",
    "canonical_matrix = canonical_matrix.fillna(0)\n",
    "\n",
    "# Reset index so 'orfID' becomes a column instead of an index\n",
    "canonical_matrix = canonical_matrix.reset_index()\n",
    "\n",
    "# Rename columns as necessary\n",
    "canonical_matrix.columns = [\"orfID\", \"chicken\", \"human\", \"macaca\", \"mouse\", \"opossum\", \"platypus\"]\n",
    "\n",
    "# Reorder columns for the final matrix\n",
    "canonical_matrix_ordered = canonical_matrix[[\"orfID\", \"human\",\"macaca\", \"mouse\", \"opossum\", \"platypus\", \"chicken\"]]\n",
    "\n",
    "# Calculate sum across species columns to identify how many species are present\n",
    "canonical_matrix_ordered['sum'] = canonical_matrix_ordered.select_dtypes(include='number').sum(axis=1)\n",
    "\n",
    "# Sort by the sum column to order by the number of species present\n",
    "canonical_matrix_ordered = canonical_matrix_ordered.sort_values(by=\"sum\", ascending=False)\n",
    "\n",
    "# Display the result\n",
    "canonical_matrix_ordered =  canonical_matrix_ordered.merge(canonical_to_matrix_raw[['gene_name','gene_type','orfID']], on=\"orfID\")\n",
    "canonical_matrix_ordered.drop_duplicates(inplace=True)\n",
    "\n",
    "canonical_matrix_ordered.to_csv(os.path.join(evoDir,\"Clustering/canonicalHuman_matrix.tsv\"), sep=\"\\t\", index=None)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Find the oldest specie containg the ORF translated"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {},
   "outputs": [],
   "source": [
    "species_ordered = ['chicken','platypus','opossum','mouse','macaca','human']\n",
    "def find_last_species(row):\n",
    "    for sp in species_ordered:\n",
    "        if row[sp] == 1.0:  # Check if the value is 1.0\n",
    "            return sp\n",
    "    return None  # Return None if no species has a 1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Apply the function to each row to create a new column 'oldest_species'\n",
    "canonical_matrix_ordered['oldest_species'] = canonical_matrix_ordered.apply(find_last_species, axis=1)\n",
    "canonical_matrix_ordered = species_and_years.merge(canonical_matrix_ordered, left_on=[\"specie\"], right_on=\"oldest_species\")\n",
    "canonical_matrix_ordered[['orfID','gene_name','gene_type','oldest_species','age']].to_csv(os.path.join(evoDir,\"Clustering/canonicalHuman_matrix_oldestAge.tsv\"), sep=\"\\t\", index=None)\n",
    "\n",
    "noncoding_matrix_ordered['oldest_species'] = noncoding_matrix_ordered.apply(find_last_species, axis=1)\n",
    "noncoding_matrix_ordered = species_and_years.merge(noncoding_matrix_ordered, left_on=[\"specie\"], right_on=\"oldest_species\")\n",
    "noncoding_matrix_ordered[['orfID','gene_name','gene_type','oldest_species','age']].to_csv(os.path.join(evoDir,\"Clustering/noncodingHuman_matrix_oldestAge.tsv\"), sep=\"\\t\", index=None)\n",
    "\n",
    "complete_matrix_age = pd.concat([noncoding_matrix_ordered[['orfID','gene_name','gene_type','oldest_species','age']], canonical_matrix_ordered[['orfID','gene_name','gene_type','oldest_species','age']]])\n",
    "complete_matrix_age.to_csv(os.path.join(evoDir,\"Clustering/allHuman_matrix_oldestAge.tsv\"), sep=\"\\t\", index=None)\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Liver-EXPRESSED"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "ename": "KeyError",
     "evalue": "'ORFpep'",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mKeyError\u001b[0m                                  Traceback (most recent call last)",
      "File \u001b[0;32m~/.local/lib/python3.8/site-packages/pandas/core/indexes/base.py:3653\u001b[0m, in \u001b[0;36mIndex.get_loc\u001b[0;34m(self, key)\u001b[0m\n\u001b[1;32m   3652\u001b[0m \u001b[39mtry\u001b[39;00m:\n\u001b[0;32m-> 3653\u001b[0m     \u001b[39mreturn\u001b[39;00m \u001b[39mself\u001b[39;49m\u001b[39m.\u001b[39;49m_engine\u001b[39m.\u001b[39;49mget_loc(casted_key)\n\u001b[1;32m   3654\u001b[0m \u001b[39mexcept\u001b[39;00m \u001b[39mKeyError\u001b[39;00m \u001b[39mas\u001b[39;00m err:\n",
      "File \u001b[0;32m~/.local/lib/python3.8/site-packages/pandas/_libs/index.pyx:147\u001b[0m, in \u001b[0;36mpandas._libs.index.IndexEngine.get_loc\u001b[0;34m()\u001b[0m\n",
      "File \u001b[0;32m~/.local/lib/python3.8/site-packages/pandas/_libs/index.pyx:176\u001b[0m, in \u001b[0;36mpandas._libs.index.IndexEngine.get_loc\u001b[0;34m()\u001b[0m\n",
      "File \u001b[0;32mpandas/_libs/hashtable_class_helper.pxi:7080\u001b[0m, in \u001b[0;36mpandas._libs.hashtable.PyObjectHashTable.get_item\u001b[0;34m()\u001b[0m\n",
      "File \u001b[0;32mpandas/_libs/hashtable_class_helper.pxi:7088\u001b[0m, in \u001b[0;36mpandas._libs.hashtable.PyObjectHashTable.get_item\u001b[0;34m()\u001b[0m\n",
      "\u001b[0;31mKeyError\u001b[0m: 'ORFpep'",
      "\nThe above exception was the direct cause of the following exception:\n",
      "\u001b[0;31mKeyError\u001b[0m                                  Traceback (most recent call last)",
      "Cell \u001b[0;32mIn[6], line 9\u001b[0m\n\u001b[1;32m      6\u001b[0m     \u001b[39mreturn\u001b[39;00m SeqRecord(Seq(row[\u001b[39m'\u001b[39m\u001b[39mORFpep\u001b[39m\u001b[39m'\u001b[39m]), \u001b[39mid\u001b[39m\u001b[39m=\u001b[39mrow[\u001b[39m'\u001b[39m\u001b[39morfID\u001b[39m\u001b[39m'\u001b[39m], description\u001b[39m=\u001b[39m\u001b[39m\"\u001b[39m\u001b[39m\"\u001b[39m)\n\u001b[1;32m      8\u001b[0m out_nonredundant_proteins \u001b[39m=\u001b[39m \u001b[39m\"\u001b[39m\u001b[39m/projects_eg/projects/marta/TestisRestricted_Microproteins_TSA/with_TranscriptomeReconstruction/Multimap_altORFs/Q1_TestisORFs/human/ribORF_humanLiver_translated.fa\u001b[39m\u001b[39m\"\u001b[39m\n\u001b[0;32m----> 9\u001b[0m seq_records_proteins \u001b[39m=\u001b[39m LiverExpressed_translatedINliver\u001b[39m.\u001b[39;49mapply(create_seqrecord_PROThuman, axis\u001b[39m=\u001b[39;49m\u001b[39m1\u001b[39;49m)\u001b[39m.\u001b[39mtolist()\n\u001b[1;32m     10\u001b[0m SeqIO\u001b[39m.\u001b[39mwrite(seq_records_proteins, out_nonredundant_proteins, \u001b[39m\"\u001b[39m\u001b[39mfasta\u001b[39m\u001b[39m\"\u001b[39m)\n",
      "File \u001b[0;32m~/.local/lib/python3.8/site-packages/pandas/core/frame.py:9423\u001b[0m, in \u001b[0;36mDataFrame.apply\u001b[0;34m(self, func, axis, raw, result_type, args, **kwargs)\u001b[0m\n\u001b[1;32m   9412\u001b[0m \u001b[39mfrom\u001b[39;00m \u001b[39mpandas\u001b[39;00m\u001b[39m.\u001b[39;00m\u001b[39mcore\u001b[39;00m\u001b[39m.\u001b[39;00m\u001b[39mapply\u001b[39;00m \u001b[39mimport\u001b[39;00m frame_apply\n\u001b[1;32m   9414\u001b[0m op \u001b[39m=\u001b[39m frame_apply(\n\u001b[1;32m   9415\u001b[0m     \u001b[39mself\u001b[39m,\n\u001b[1;32m   9416\u001b[0m     func\u001b[39m=\u001b[39mfunc,\n\u001b[0;32m   (...)\u001b[0m\n\u001b[1;32m   9421\u001b[0m     kwargs\u001b[39m=\u001b[39mkwargs,\n\u001b[1;32m   9422\u001b[0m )\n\u001b[0;32m-> 9423\u001b[0m \u001b[39mreturn\u001b[39;00m op\u001b[39m.\u001b[39;49mapply()\u001b[39m.\u001b[39m__finalize__(\u001b[39mself\u001b[39m, method\u001b[39m=\u001b[39m\u001b[39m\"\u001b[39m\u001b[39mapply\u001b[39m\u001b[39m\"\u001b[39m)\n",
      "File \u001b[0;32m~/.local/lib/python3.8/site-packages/pandas/core/apply.py:678\u001b[0m, in \u001b[0;36mFrameApply.apply\u001b[0;34m(self)\u001b[0m\n\u001b[1;32m    675\u001b[0m \u001b[39melif\u001b[39;00m \u001b[39mself\u001b[39m\u001b[39m.\u001b[39mraw:\n\u001b[1;32m    676\u001b[0m     \u001b[39mreturn\u001b[39;00m \u001b[39mself\u001b[39m\u001b[39m.\u001b[39mapply_raw()\n\u001b[0;32m--> 678\u001b[0m \u001b[39mreturn\u001b[39;00m \u001b[39mself\u001b[39;49m\u001b[39m.\u001b[39;49mapply_standard()\n",
      "File \u001b[0;32m~/.local/lib/python3.8/site-packages/pandas/core/apply.py:798\u001b[0m, in \u001b[0;36mFrameApply.apply_standard\u001b[0;34m(self)\u001b[0m\n\u001b[1;32m    797\u001b[0m \u001b[39mdef\u001b[39;00m \u001b[39mapply_standard\u001b[39m(\u001b[39mself\u001b[39m):\n\u001b[0;32m--> 798\u001b[0m     results, res_index \u001b[39m=\u001b[39m \u001b[39mself\u001b[39;49m\u001b[39m.\u001b[39;49mapply_series_generator()\n\u001b[1;32m    800\u001b[0m     \u001b[39m# wrap results\u001b[39;00m\n\u001b[1;32m    801\u001b[0m     \u001b[39mreturn\u001b[39;00m \u001b[39mself\u001b[39m\u001b[39m.\u001b[39mwrap_results(results, res_index)\n",
      "File \u001b[0;32m~/.local/lib/python3.8/site-packages/pandas/core/apply.py:814\u001b[0m, in \u001b[0;36mFrameApply.apply_series_generator\u001b[0;34m(self)\u001b[0m\n\u001b[1;32m    811\u001b[0m \u001b[39mwith\u001b[39;00m option_context(\u001b[39m\"\u001b[39m\u001b[39mmode.chained_assignment\u001b[39m\u001b[39m\"\u001b[39m, \u001b[39mNone\u001b[39;00m):\n\u001b[1;32m    812\u001b[0m     \u001b[39mfor\u001b[39;00m i, v \u001b[39min\u001b[39;00m \u001b[39menumerate\u001b[39m(series_gen):\n\u001b[1;32m    813\u001b[0m         \u001b[39m# ignore SettingWithCopy here in case the user mutates\u001b[39;00m\n\u001b[0;32m--> 814\u001b[0m         results[i] \u001b[39m=\u001b[39m \u001b[39mself\u001b[39;49m\u001b[39m.\u001b[39;49mf(v)\n\u001b[1;32m    815\u001b[0m         \u001b[39mif\u001b[39;00m \u001b[39misinstance\u001b[39m(results[i], ABCSeries):\n\u001b[1;32m    816\u001b[0m             \u001b[39m# If we have a view on v, we need to make a copy because\u001b[39;00m\n\u001b[1;32m    817\u001b[0m             \u001b[39m#  series_generator will swap out the underlying data\u001b[39;00m\n\u001b[1;32m    818\u001b[0m             results[i] \u001b[39m=\u001b[39m results[i]\u001b[39m.\u001b[39mcopy(deep\u001b[39m=\u001b[39m\u001b[39mFalse\u001b[39;00m)\n",
      "Cell \u001b[0;32mIn[6], line 6\u001b[0m, in \u001b[0;36mcreate_seqrecord_PROThuman\u001b[0;34m(row)\u001b[0m\n\u001b[1;32m      5\u001b[0m \u001b[39mdef\u001b[39;00m \u001b[39mcreate_seqrecord_PROThuman\u001b[39m(row):\n\u001b[0;32m----> 6\u001b[0m     \u001b[39mreturn\u001b[39;00m SeqRecord(Seq(row[\u001b[39m'\u001b[39;49m\u001b[39mORFpep\u001b[39;49m\u001b[39m'\u001b[39;49m]), \u001b[39mid\u001b[39m\u001b[39m=\u001b[39mrow[\u001b[39m'\u001b[39m\u001b[39morfID\u001b[39m\u001b[39m'\u001b[39m], description\u001b[39m=\u001b[39m\u001b[39m\"\u001b[39m\u001b[39m\"\u001b[39m)\n",
      "File \u001b[0;32m~/.local/lib/python3.8/site-packages/pandas/core/series.py:1007\u001b[0m, in \u001b[0;36mSeries.__getitem__\u001b[0;34m(self, key)\u001b[0m\n\u001b[1;32m   1004\u001b[0m     \u001b[39mreturn\u001b[39;00m \u001b[39mself\u001b[39m\u001b[39m.\u001b[39m_values[key]\n\u001b[1;32m   1006\u001b[0m \u001b[39melif\u001b[39;00m key_is_scalar:\n\u001b[0;32m-> 1007\u001b[0m     \u001b[39mreturn\u001b[39;00m \u001b[39mself\u001b[39;49m\u001b[39m.\u001b[39;49m_get_value(key)\n\u001b[1;32m   1009\u001b[0m \u001b[39mif\u001b[39;00m is_hashable(key):\n\u001b[1;32m   1010\u001b[0m     \u001b[39m# Otherwise index.get_value will raise InvalidIndexError\u001b[39;00m\n\u001b[1;32m   1011\u001b[0m     \u001b[39mtry\u001b[39;00m:\n\u001b[1;32m   1012\u001b[0m         \u001b[39m# For labels that don't resolve as scalars like tuples and frozensets\u001b[39;00m\n",
      "File \u001b[0;32m~/.local/lib/python3.8/site-packages/pandas/core/series.py:1116\u001b[0m, in \u001b[0;36mSeries._get_value\u001b[0;34m(self, label, takeable)\u001b[0m\n\u001b[1;32m   1113\u001b[0m     \u001b[39mreturn\u001b[39;00m \u001b[39mself\u001b[39m\u001b[39m.\u001b[39m_values[label]\n\u001b[1;32m   1115\u001b[0m \u001b[39m# Similar to Index.get_value, but we do not fall back to positional\u001b[39;00m\n\u001b[0;32m-> 1116\u001b[0m loc \u001b[39m=\u001b[39m \u001b[39mself\u001b[39;49m\u001b[39m.\u001b[39;49mindex\u001b[39m.\u001b[39;49mget_loc(label)\n\u001b[1;32m   1118\u001b[0m \u001b[39mif\u001b[39;00m is_integer(loc):\n\u001b[1;32m   1119\u001b[0m     \u001b[39mreturn\u001b[39;00m \u001b[39mself\u001b[39m\u001b[39m.\u001b[39m_values[loc]\n",
      "File \u001b[0;32m~/.local/lib/python3.8/site-packages/pandas/core/indexes/base.py:3655\u001b[0m, in \u001b[0;36mIndex.get_loc\u001b[0;34m(self, key)\u001b[0m\n\u001b[1;32m   3653\u001b[0m     \u001b[39mreturn\u001b[39;00m \u001b[39mself\u001b[39m\u001b[39m.\u001b[39m_engine\u001b[39m.\u001b[39mget_loc(casted_key)\n\u001b[1;32m   3654\u001b[0m \u001b[39mexcept\u001b[39;00m \u001b[39mKeyError\u001b[39;00m \u001b[39mas\u001b[39;00m err:\n\u001b[0;32m-> 3655\u001b[0m     \u001b[39mraise\u001b[39;00m \u001b[39mKeyError\u001b[39;00m(key) \u001b[39mfrom\u001b[39;00m \u001b[39merr\u001b[39;00m\n\u001b[1;32m   3656\u001b[0m \u001b[39mexcept\u001b[39;00m \u001b[39mTypeError\u001b[39;00m:\n\u001b[1;32m   3657\u001b[0m     \u001b[39m# If we have a listlike key, _check_indexing_error will raise\u001b[39;00m\n\u001b[1;32m   3658\u001b[0m     \u001b[39m#  InvalidIndexError. Otherwise we fall through and re-raise\u001b[39;00m\n\u001b[1;32m   3659\u001b[0m     \u001b[39m#  the TypeError.\u001b[39;00m\n\u001b[1;32m   3660\u001b[0m     \u001b[39mself\u001b[39m\u001b[39m.\u001b[39m_check_indexing_error(key)\n",
      "\u001b[0;31mKeyError\u001b[0m: 'ORFpep'"
     ]
    }
   ],
   "source": [
    "## Liver-expressed ORFs human\n",
    "LiverExpressed_translatedINliver = pd.read_csv(\"/projects_eg/projects/marta/TestisRestricted_Microproteins_TSA/with_TranscriptomeReconstruction/Multimap_altORFs/Q1_TestisORFs/human/ribORF_humanLiver_translated.csv\")\n",
    "LiverExpressed_translatedINliver\n",
    "\n",
    "def create_seqrecord_PROThuman(row):\n",
    "    return SeqRecord(Seq(row['ORFpep']), id=row['orfID'], description=\"\")\n",
    "\n",
    "out_nonredundant_proteins = \"/projects_eg/projects/marta/TestisRestricted_Microproteins_TSA/with_TranscriptomeReconstruction/Multimap_altORFs/Q1_TestisORFs/human/ribORF_humanLiver_translated.fa\"\n",
    "seq_records_proteins = LiverExpressed_translatedINliver.apply(create_seqrecord_PROThuman, axis=1).tolist()\n",
    "SeqIO.write(seq_records_proteins, out_nonredundant_proteins, \"fasta\")"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Creating database of human (expressed in liver)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 62,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/EvolutionaryOrigin_MMseqs/MMseqsDB/ribORF_humanLiver_in1_DB exists and will be overwritten\n",
      "createdb /projects_eg/projects/marta/TestisRestricted_Microproteins_TSA/with_TranscriptomeReconstruction/Q1_TestisORFs/human/ribORF_humanLiver.fa /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/EvolutionaryOrigin_MMseqs/MMseqsDB/ribORF_humanLiver_in1_DB \n",
      "\n",
      "MMseqs Version:       \t14.7e284\n",
      "Database type         \t0\n",
      "Shuffle input database\ttrue\n",
      "Createdb mode         \t0\n",
      "Write lookup file     \t1\n",
      "Offset of numeric ids \t0\n",
      "Compressed            \t0\n",
      "Verbosity             \t3\n",
      "\n",
      "Converting sequences\n",
      "[=\n",
      "Time for merging to ribORF_humanLiver_in1_DB_h: 0h 0m 7s 140ms\n",
      "Time for merging to ribORF_humanLiver_in1_DB: 0h 0m 20s 525ms\n",
      "Database type: Aminoacid\n",
      "Time for processing: 0h 1m 59s 895ms\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "createdb /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/chicken/RiboSeq/RiboQC_RiboNovel/RibORF/chicken_repre.valid.pred.pvalue.parameters.PROTEIN.noredunant.fa /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/macaca/RiboSeq/RiboQC_RiboNovel/RibORF/macaca_repre.valid.pred.pvalue.parameters.PROTEIN.noredunant.fa /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/mouse/RiboSeq/RiboQC_RiboNovel/RibORF/mouse_repre.valid.pred.pvalue.parameters.PROTEIN.noredunant.fa /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/opossum/RiboSeq/RiboQC_RiboNovel/RibORF/opossum_repre.valid.pred.pvalue.parameters.PROTEIN.noredunant.fa /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/platypus/RiboSeq/RiboQC_RiboNovel/RibORF/platypus_repre.valid.pred.pvalue.parameters.PROTEIN.noredunant.fa /projects_eg/projects/marta/TestisRestricted_Microproteins_TSA/with_TranscriptomeReconstruction/Q1_TestisORFs/human/ribORF_humanBrain.fa /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/EvolutionaryOrigin_MMseqs/MMseqsDB/species_and_humanBrainDB \n",
      "\n",
      "MMseqs Version:       \t14.7e284\n",
      "Database type         \t0\n",
      "Shuffle input database\ttrue\n",
      "Createdb mode         \t0\n",
      "Write lookup file     \t1\n",
      "Offset of numeric ids \t0\n",
      "Compressed            \t0\n",
      "Verbosity             \t3\n",
      "\n",
      "Converting sequences\n",
      "[============\n",
      "Time for merging to species_and_humanBrainDB_h: 0h 0m 26s 2ms\n",
      "Time for merging to species_and_humanBrainDB: 0h 1m 49s 35ms\n",
      "Database type: Aminoacid\n",
      "Time for processing: 0h 3m 54s 42ms\n"
     ]
    }
   ],
   "source": [
    "%%bash -s \"$evoDir\" \"$users_dir\"\n",
    "\n",
    "# mkdir $1/MMseqsDB\n",
    "\n",
    "module load Miniconda3/4.9.2\n",
    "source /soft/system/software/Miniconda3/4.9.2/bin/activate MMseqs2\n",
    "\n",
    "mmseqs createdb /projects_eg/projects/marta/TestisRestricted_Microproteins_TSA/with_TranscriptomeReconstruction/Q1_TestisORFs/human/ribORF_humanLiver.fa $1/MMseqsDB/ribORF_humanLiver_in1_DB\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 63,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/EvolutionaryOrigin_MMseqs/MMseqsDB/species_and_humanLiverDB exists and will be overwritten\n",
      "createdb /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/chicken/RiboSeq/RiboQC_RiboNovel/RibORF/chicken_repre.valid.pred.pvalue.parameters.PROTEIN.noredunant.fa /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/macaca/RiboSeq/RiboQC_RiboNovel/RibORF/macaca_repre.valid.pred.pvalue.parameters.PROTEIN.noredunant.fa /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/mouse/RiboSeq/RiboQC_RiboNovel/RibORF/mouse_repre.valid.pred.pvalue.parameters.PROTEIN.noredunant.fa /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/opossum/RiboSeq/RiboQC_RiboNovel/RibORF/opossum_repre.valid.pred.pvalue.parameters.PROTEIN.noredunant.fa /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/platypus/RiboSeq/RiboQC_RiboNovel/RibORF/platypus_repre.valid.pred.pvalue.parameters.PROTEIN.noredunant.fa /projects_eg/projects/marta/TestisRestricted_Microproteins_TSA/with_TranscriptomeReconstruction/Q1_TestisORFs/human/ribORF_humanLiver.fa /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/EvolutionaryOrigin_MMseqs/MMseqsDB/species_and_humanLiverDB \n",
      "\n",
      "MMseqs Version:       \t14.7e284\n",
      "Database type         \t0\n",
      "Shuffle input database\ttrue\n",
      "Createdb mode         \t0\n",
      "Write lookup file     \t1\n",
      "Offset of numeric ids \t0\n",
      "Compressed            \t0\n",
      "Verbosity             \t3\n",
      "\n",
      "Converting sequences\n",
      "[============\n",
      "Time for merging to species_and_humanLiverDB_h: 0h 0m 22s 379ms\n",
      "Time for merging to species_and_humanLiverDB: 0h 1m 21s 54ms\n",
      "Database type: Aminoacid\n",
      "Time for processing: 0h 3m 46s 971ms\n"
     ]
    }
   ],
   "source": [
    "%%bash -s \"$evoDir\" \"$users_dir\"\n",
    "\n",
    "module load Miniconda3/4.9.2\n",
    "source /soft/system/software/Miniconda3/4.9.2/bin/activate MMseqs2\n",
    "\n",
    "mmseqs createdb $2/*/RiboSeq/RiboQC_RiboNovel*/RibORF/*_repre.valid.pred.pvalue.parameters.PROTEIN.noredunant.fa /projects_eg/projects/marta/TestisRestricted_Microproteins_TSA/with_TranscriptomeReconstruction/Q1_TestisORFs/human/ribORF_humanLiver.fa $1/MMseqsDB/species_and_humanLiverDB\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 64,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "cluster -c 0.5 --min-seq-id 0.5 /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/EvolutionaryOrigin_MMseqs/MMseqsDB/species_and_humanLiverDB /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/EvolutionaryOrigin_MMseqs/Clustering/species_and_humanLiver_cluDB /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/EvolutionaryOrigin_MMseqs/tmp \n",
      "\n",
      "MMseqs Version:                     \t14.7e284\n",
      "Substitution matrix                 \taa:blosum62.out,nucl:nucleotide.out\n",
      "Seed substitution matrix            \taa:VTML80.out,nucl:nucleotide.out\n",
      "Sensitivity                         \t4\n",
      "k-mer length                        \t0\n",
      "k-score                             \tseq:2147483647,prof:2147483647\n",
      "Alphabet size                       \taa:21,nucl:5\n",
      "Max sequence length                 \t65535\n",
      "Max results per query               \t20\n",
      "Split database                      \t0\n",
      "Split mode                          \t2\n",
      "Split memory limit                  \t0\n",
      "Coverage threshold                  \t0.5\n",
      "Coverage mode                       \t0\n",
      "Compositional bias                  \t1\n",
      "Compositional bias                  \t1\n",
      "Diagonal scoring                    \ttrue\n",
      "Exact k-mer matching                \t0\n",
      "Mask residues                       \t1\n",
      "Mask residues probability           \t0.9\n",
      "Mask lower case residues            \t0\n",
      "Minimum diagonal score              \t15\n",
      "Selected taxa                       \t\n",
      "Include identical seq. id.          \tfalse\n",
      "Spaced k-mers                       \t1\n",
      "Preload mode                        \t0\n",
      "Pseudo count a                      \tsubstitution:1.100,context:1.400\n",
      "Pseudo count b                      \tsubstitution:4.100,context:5.800\n",
      "Spaced k-mer pattern                \t\n",
      "Local temporary path                \t\n",
      "Threads                             \t24\n",
      "Compressed                          \t0\n",
      "Verbosity                           \t3\n",
      "Add backtrace                       \tfalse\n",
      "Alignment mode                      \t3\n",
      "Alignment mode                      \t0\n",
      "Allow wrapped scoring               \tfalse\n",
      "E-value threshold                   \t0.001\n",
      "Seq. id. threshold                  \t0.5\n",
      "Min alignment length                \t0\n",
      "Seq. id. mode                       \t0\n",
      "Alternative alignments              \t0\n",
      "Max reject                          \t2147483647\n",
      "Max accept                          \t2147483647\n",
      "Score bias                          \t0\n",
      "Realign hits                        \tfalse\n",
      "Realign score bias                  \t-0.2\n",
      "Realign max seqs                    \t2147483647\n",
      "Correlation score weight            \t0\n",
      "Gap open cost                       \taa:11,nucl:5\n",
      "Gap extension cost                  \taa:1,nucl:2\n",
      "Zdrop                               \t40\n",
      "Rescore mode                        \t0\n",
      "Remove hits by seq. id. and coverage\tfalse\n",
      "Sort results                        \t0\n",
      "Cluster mode                        \t0\n",
      "Max connected component depth       \t1000\n",
      "Similarity type                     \t2\n",
      "Single step clustering              \tfalse\n",
      "Cascaded clustering steps           \t3\n",
      "Cluster reassign                    \tfalse\n",
      "Remove temporary files              \tfalse\n",
      "Force restart with latest tmp       \tfalse\n",
      "MPI runner                          \t\n",
      "k-mers per sequence                 \t21\n",
      "Scale k-mers per sequence           \taa:0.000,nucl:0.200\n",
      "Adjust k-mer length                 \tfalse\n",
      "Shift hash                          \t67\n",
      "Include only extendable             \tfalse\n",
      "Skip repeating k-mers               \tfalse\n",
      "\n",
      "Set cluster sensitivity to -s 3.000000\n",
      "Set cluster mode SET COVER\n",
      "Set cluster iterations to 3\n",
      "/users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/EvolutionaryOrigin_MMseqs/Clustering/species_and_humanLiver_cluDB.dbtype exists already!\n",
      "createtsv /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/EvolutionaryOrigin_MMseqs/MMseqsDB/species_and_humanLiverDB /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/EvolutionaryOrigin_MMseqs/MMseqsDB/species_and_humanLiverDB /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/EvolutionaryOrigin_MMseqs/Clustering/species_and_humanLiver_cluDB /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/EvolutionaryOrigin_MMseqs/Clustering/species_and_humanLiver_cluDB.tsv \n",
      "\n",
      "MMseqs Version:                 \t14.7e284\n",
      "First sequence as representative\tfalse\n",
      "Target column                   \t1\n",
      "Add full header                 \tfalse\n",
      "Sequence source                 \t0\n",
      "Database output                 \tfalse\n",
      "Threads                         \t24\n",
      "Compressed                      \t0\n",
      "Verbosity                       \t3\n",
      "\n",
      "Time for merging to species_and_humanLiver_cluDB.tsv: 0h 0m 4s 860ms\n",
      "Time for processing: 0h 0m 19s 763ms\n"
     ]
    }
   ],
   "source": [
    "%%bash -s \"$evoDir\" \"$users_dir\"\n",
    "\n",
    "module load Miniconda3/4.9.2\n",
    "source /soft/system/software/Miniconda3/4.9.2/bin/activate MMseqs2\n",
    "\n",
    "## cluster all species and human\n",
    "mmseqs cluster -c 0.5 --min-seq-id 0.5 $1/MMseqsDB/species_and_humanLiverDB $1/Clustering/species_and_humanLiver_cluDB $1/tmp\n",
    "mmseqs createtsv $1/MMseqsDB/species_and_humanLiverDB $1/MMseqsDB/species_and_humanLiverDB $1/Clustering/species_and_humanLiver_cluDB $1/Clustering/species_and_humanLiver_cluDB.tsv"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Analysis"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 65,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "all human\n",
      "10317\n",
      "all human after annotation\n",
      "10317\n",
      "noncoding all\n",
      "798\n",
      "683\n",
      "Conserved noncoding human ORFs\n",
      "36\n",
      "Non-Conserved noncoding human ORFs\n",
      "683\n",
      "Canonical all\n",
      "9519\n",
      "1952\n",
      "Conserved canonical human ORFs\n",
      "1663\n",
      "Conserved canonical human ORFs as noncoding in other species\n",
      "1643\n",
      "Conserved canonical human ORFs as canonical in other species\n",
      "204\n",
      "Non-Conserved canonical human ORFs\n",
      "1952\n"
     ]
    }
   ],
   "source": [
    "clustering = pd.read_csv(os.path.join(evoDir,\"Clustering/species_and_humanLiver_cluDB.tsv\"), header=None, sep=\"\\t\")\n",
    "clustering = clustering[clustering[0].str.startswith(\"ENST\")]\n",
    "print(\"all human\")\n",
    "print(len(clustering))\n",
    "clustering['transcript_id'] = clustering[0].str.split(\":\", expand=True)[0]\n",
    "clustering['transcript_id'] = clustering[0].str.split(\".\", expand=True)[0]\n",
    "\n",
    "clustering = clustering.merge(transcript_gene, on=\"transcript_id\")\n",
    "print(\"all human after annotation\")\n",
    "print(len(clustering))\n",
    "clustering = clustering[[\"gene_name\",\"gene_type\",0,1]]\n",
    "\n",
    "## non-coding\n",
    "noncoding = clustering[clustering[0].str.contains(\"noncoding\")]\n",
    "noncoding.to_csv(os.path.join(evoDir,\"Clustering/LIVER_noncodingHuman_speciesDB.tsv\"), header=None, index=None, sep=\"\\t\")\n",
    "print(\"noncoding all\")\n",
    "print(len(noncoding))\n",
    "print(len(set(noncoding[0].values.tolist())))\n",
    "\n",
    "noncoding_conserved = noncoding[~noncoding[1].str.startswith(\"ENST\")]\n",
    "noncoding_conserved.to_csv(os.path.join(evoDir,\"Clustering/LIVER_noncodingHuman_Conserved.tsv\"), header=None, index=None, sep=\"\\t\")\n",
    "print(\"Conserved noncoding human ORFs\")\n",
    "print(len(set(noncoding_conserved[0].values.tolist())))\n",
    "noncoding_conserved.columns = ['gene_name','gene_type','orfID',1]\n",
    "\n",
    "noncoding_NonConserved = noncoding[noncoding[1].str.startswith(\"ENST\")]\n",
    "noncoding_NonConserved.to_csv(os.path.join(evoDir,\"Clustering/LIVER_noncodingHuman_NonConserved.tsv\"), header=None, index=None, sep=\"\\t\")\n",
    "print(\"Non-Conserved noncoding human ORFs\")\n",
    "print(len(set(noncoding_NonConserved[0].values.tolist())))\n",
    "noncoding_NonConserved.columns = ['gene_name','gene_type','orfID',1]\n",
    "\n",
    "\n",
    "## canonical\n",
    "canonical = clustering[clustering[0].str.contains(\"canonical\")]\n",
    "print(\"Canonical all\")\n",
    "print(len(canonical))\n",
    "print(len(set(canonical[0].values.tolist())))\n",
    "canonical.to_csv(os.path.join(evoDir,\"Clustering/LIVER_canonicalHuman_speciesDB.tsv\"), header=None, index=None, sep=\"\\t\")\n",
    "\n",
    "canonical_conserved = canonical[~canonical[1].str.startswith(\"ENST\")]\n",
    "canonical_conserved.to_csv(os.path.join(evoDir,\"Clustering/LIVER_canonicalHuman_Conserved.tsv\"), header=None, index=None, sep=\"\\t\")\n",
    "print(\"Conserved canonical human ORFs\")\n",
    "print(len(set(canonical_conserved[0].values.tolist())))\n",
    "canonical_conserved.columns = ['gene_name','gene_type','orfID',1]\n",
    "\n",
    "canonical_conserved_noncoding = canonical_conserved[~canonical_conserved[1].str.contains(\"noncoding\")]\n",
    "canonical_conserved_noncoding.to_csv(os.path.join(evoDir,\"Clustering/LIVER_canonicalHuman_ConservedAsNoncoding.tsv\"), header=None, index=None, sep=\"\\t\")\n",
    "print(\"Conserved canonical human ORFs as noncoding in other species\")\n",
    "print(len(set(canonical_conserved_noncoding['orfID'].values.tolist())))\n",
    "\n",
    "canonical_conserved_canonical = canonical_conserved[~canonical_conserved[1].str.contains(\"canonical\")]\n",
    "canonical_conserved_canonical.to_csv(os.path.join(evoDir,\"Clustering/LIVER_canonicalHuman_ConservedAsCanonical.tsv\"), header=None, index=None, sep=\"\\t\")\n",
    "print(\"Conserved canonical human ORFs as canonical in other species\")\n",
    "print(len(set(canonical_conserved_canonical['orfID'].values.tolist())))\n",
    "\n",
    "canonical_NonConserved = canonical[canonical[1].str.startswith(\"ENST\")]\n",
    "canonical_NonConserved.to_csv(os.path.join(evoDir,\"Clustering/LIVER_canonicalHuman_NonConserved.tsv\"), header=None, index=None, sep=\"\\t\")\n",
    "print(\"Non-Conserved canonical human ORFs\")\n",
    "print(len(set(canonical_NonConserved[0].values.tolist())))\n",
    "canonical_NonConserved.columns = ['gene_name','gene_type','orfID',1]\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 66,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene_name</th>\n",
       "      <th>gene_type</th>\n",
       "      <th>orfID</th>\n",
       "      <th>1</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>AC002116.8</td>\n",
       "      <td>lncRNA</td>\n",
       "      <td>ENST00000473572.2:19:-|56|1130:683:1046|noncod...</td>\n",
       "      <td>opossum_ENSMODT00000033099:4:-|1|879:6:639|can...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>AC002116.8</td>\n",
       "      <td>lncRNA</td>\n",
       "      <td>ENST00000473572.2:19:-|56|1130:683:1046|noncod...</td>\n",
       "      <td>macaca_ENSMMUT00000038552:19:-|1|801:1:802|can...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>AC002116.8</td>\n",
       "      <td>lncRNA</td>\n",
       "      <td>ENST00000473572.2:19:-|56|1130:683:1046|noncod...</td>\n",
       "      <td>platypus_ENSOANT00000056035:5:-|30|1579:380:11...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>200</th>\n",
       "      <td>RP11-197N18.2</td>\n",
       "      <td>lncRNA</td>\n",
       "      <td>ENST00000540866.2:12:+|141|5618:1896:2205|nonc...</td>\n",
       "      <td>mouse_ENSMUST00000130140.8:chr5:+|94|1838:1216...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>217</th>\n",
       "      <td>RP11-15H20.7</td>\n",
       "      <td>lncRNA</td>\n",
       "      <td>ENST00000594653.1:19:-|10|574:128:269|noncodin...</td>\n",
       "      <td>macaca_ENSMMUT00000104102:15:-|19|2098:297:444...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9232</th>\n",
       "      <td>AC016292.3</td>\n",
       "      <td>processed_pseudogene</td>\n",
       "      <td>ENST00000451056.1:17:-|1|408:1:409|noncoding|ATG</td>\n",
       "      <td>opossum_ENSMODT00000062873:2:+|76|1618:965:161...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9704</th>\n",
       "      <td>RP11-186B7.4</td>\n",
       "      <td>lncRNA</td>\n",
       "      <td>ENST00000581621.1:17:+|261|4554:3517:4042|nonc...</td>\n",
       "      <td>mouse_ENSMUST00000018918.12:chr11:-|4|1230:92:...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9706</th>\n",
       "      <td>RP11-186B7.4</td>\n",
       "      <td>lncRNA</td>\n",
       "      <td>ENST00000581621.1:17:+|261|4554:3517:4042|nonc...</td>\n",
       "      <td>macaca_ENSMMUT00000018950:16:+|2|3788:41:1106|...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9707</th>\n",
       "      <td>RP11-186B7.4</td>\n",
       "      <td>lncRNA</td>\n",
       "      <td>ENST00000581621.1:17:+|261|4554:3517:4042|nonc...</td>\n",
       "      <td>platypus_ENSOANT00000072290:X5:-|32|2765:470:1...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9708</th>\n",
       "      <td>RP11-186B7.4</td>\n",
       "      <td>lncRNA</td>\n",
       "      <td>ENST00000581621.1:17:+|261|4554:3517:4042|nonc...</td>\n",
       "      <td>opossum_ENSMODT00000008995:2:-|60|2189:930:190...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>99 rows × 4 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "          gene_name             gene_type  \\\n",
       "12       AC002116.8                lncRNA   \n",
       "13       AC002116.8                lncRNA   \n",
       "15       AC002116.8                lncRNA   \n",
       "200   RP11-197N18.2                lncRNA   \n",
       "217    RP11-15H20.7                lncRNA   \n",
       "...             ...                   ...   \n",
       "9232     AC016292.3  processed_pseudogene   \n",
       "9704   RP11-186B7.4                lncRNA   \n",
       "9706   RP11-186B7.4                lncRNA   \n",
       "9707   RP11-186B7.4                lncRNA   \n",
       "9708   RP11-186B7.4                lncRNA   \n",
       "\n",
       "                                                  orfID  \\\n",
       "12    ENST00000473572.2:19:-|56|1130:683:1046|noncod...   \n",
       "13    ENST00000473572.2:19:-|56|1130:683:1046|noncod...   \n",
       "15    ENST00000473572.2:19:-|56|1130:683:1046|noncod...   \n",
       "200   ENST00000540866.2:12:+|141|5618:1896:2205|nonc...   \n",
       "217   ENST00000594653.1:19:-|10|574:128:269|noncodin...   \n",
       "...                                                 ...   \n",
       "9232   ENST00000451056.1:17:-|1|408:1:409|noncoding|ATG   \n",
       "9704  ENST00000581621.1:17:+|261|4554:3517:4042|nonc...   \n",
       "9706  ENST00000581621.1:17:+|261|4554:3517:4042|nonc...   \n",
       "9707  ENST00000581621.1:17:+|261|4554:3517:4042|nonc...   \n",
       "9708  ENST00000581621.1:17:+|261|4554:3517:4042|nonc...   \n",
       "\n",
       "                                                      1  \n",
       "12    opossum_ENSMODT00000033099:4:-|1|879:6:639|can...  \n",
       "13    macaca_ENSMMUT00000038552:19:-|1|801:1:802|can...  \n",
       "15    platypus_ENSOANT00000056035:5:-|30|1579:380:11...  \n",
       "200   mouse_ENSMUST00000130140.8:chr5:+|94|1838:1216...  \n",
       "217   macaca_ENSMMUT00000104102:15:-|19|2098:297:444...  \n",
       "...                                                 ...  \n",
       "9232  opossum_ENSMODT00000062873:2:+|76|1618:965:161...  \n",
       "9704  mouse_ENSMUST00000018918.12:chr11:-|4|1230:92:...  \n",
       "9706  macaca_ENSMMUT00000018950:16:+|2|3788:41:1106|...  \n",
       "9707  platypus_ENSOANT00000072290:X5:-|32|2765:470:1...  \n",
       "9708  opossum_ENSMODT00000008995:2:-|60|2189:930:190...  \n",
       "\n",
       "[99 rows x 4 columns]"
      ]
     },
     "execution_count": 66,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "noncoding_conserved"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 67,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "36\n",
      "36\n",
      "{'mouse', 'macaca', 'opossum', 'platypus', 'chicken'}\n",
      "36\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_1203003/3156784893.py:5: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  noncoding_conserved_to_matrix['specie'] = noncoding_conserved_to_matrix[1].str.split(\"_\", expand=True)[0]\n",
      "/tmp/ipykernel_1203003/3156784893.py:24: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  noncoding_conserved_matrix_ordered['sum'] = noncoding_conserved_matrix_ordered.select_dtypes(include='number').sum(axis=1)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "36\n"
     ]
    }
   ],
   "source": [
    "noncoding_conserved_to_matrix = noncoding_conserved\n",
    "noncoding_conserved_to_matrix\n",
    "print(len(set(noncoding_conserved_to_matrix.orfID.values.tolist())))\n",
    "# Assuming noncoding_conserved_to_matrix is already defined\n",
    "noncoding_conserved_to_matrix['specie'] = noncoding_conserved_to_matrix[1].str.split(\"_\", expand=True)[0]\n",
    "noncoding_conserved_to_matrix = noncoding_conserved_to_matrix[~noncoding_conserved_to_matrix['specie'].str.contains(\"TCONS\")]\n",
    "noncoding_conserved_to_matrix = noncoding_conserved_to_matrix[['orfID', 'specie']]\n",
    "noncoding_conserved_to_matrix['presence'] = 1\n",
    "print(len(set(noncoding_conserved_to_matrix.orfID.values.tolist())))\n",
    "print(set(noncoding_conserved_to_matrix.specie.values.tolist()))\n",
    "## Pivoting the data without creating an 'id', and keeping 'orfID' as the unique identifier\n",
    "noncoding_conserved_matrix = noncoding_conserved_to_matrix.pivot_table(index='orfID', columns='specie', values='presence', aggfunc='max').reset_index()\n",
    "# Rename columns as necessary\n",
    "noncoding_conserved_matrix.columns = [\"orfID\", \"chicken\", \"macaca\", \"mouse\", \"opossum\", \"platypus\"]\n",
    "# Fill NaN values with 0\n",
    "noncoding_conserved_matrix = noncoding_conserved_matrix.fillna(0)\n",
    "\n",
    "# Reset index so 'orfID' becomes a column instead of an index\n",
    "\n",
    "# Reorder columns for the final matrix\n",
    "noncoding_conserved_matrix_ordered = noncoding_conserved_matrix[[\"orfID\", \"macaca\", \"mouse\", \"opossum\", \"platypus\", \"chicken\"]]\n",
    "\n",
    "# Calculate sum across species columns to identify how many species are present\n",
    "noncoding_conserved_matrix_ordered['sum'] = noncoding_conserved_matrix_ordered.select_dtypes(include='number').sum(axis=1)\n",
    "\n",
    "# Sort by the sum column to order by the number of species present\n",
    "noncoding_conserved_matrix_ordered = noncoding_conserved_matrix_ordered.sort_values(by=\"sum\", ascending=False)\n",
    "print(len(set(noncoding_conserved_matrix_ordered.orfID.values.tolist())))\n",
    "\n",
    "# Display the result\n",
    "noncoding_conserved_matrix_ordered =  noncoding_conserved_matrix_ordered.merge(noncoding_conserved[['gene_name','gene_type','orfID']], on=\"orfID\")\n",
    "noncoding_conserved_matrix_ordered.drop_duplicates(inplace=True)\n",
    "\n",
    "noncoding_conserved_matrix_ordered.to_csv(os.path.join(evoDir,\"Clustering/LIVER_noncodingHuman_Conserved_matrix.tsv\"), sep=\"\\t\", index=None)\n",
    "print(len(set(noncoding_conserved_matrix_ordered.orfID.values.tolist())))\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 68,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "683\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_1203003/2931333229.py:10: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  noncoding_to_matrix['presence'] = np.where(noncoding_to_matrix['specie'].isna(), 0, 1)\n",
      "/tmp/ipykernel_1203003/2931333229.py:39: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  noncoding_matrix_ordered['sum'] = noncoding_matrix_ordered.select_dtypes(include='number').sum(axis=1)\n"
     ]
    }
   ],
   "source": [
    "### all noncoding\n",
    "# noncoding_conserved_touse = noncoding_conserved\n",
    "# noncoding_conserved_touse['specie'] = noncoding_conserved_touse[1].str.split(\"_\", expand=True)[0]\n",
    "# noncoding_conserved_touse = noncoding_conserved_touse[~noncoding_conserved_touse['specie'].str.contains(\"TCONS\")]\n",
    "\n",
    "noncoding_to_matrix_raw = pd.concat([noncoding_conserved, noncoding_NonConserved])\n",
    "noncoding_to_matrix_raw\n",
    "noncoding_to_matrix = noncoding_to_matrix_raw[['orfID', 'specie']]\n",
    "# noncoding_to_matrix\n",
    "noncoding_to_matrix['presence'] = np.where(noncoding_to_matrix['specie'].isna(), 0, 1)\n",
    "\n",
    "\n",
    "human = pd.DataFrame({\n",
    "    'orfID': noncoding_to_matrix['orfID'].unique(),\n",
    "    'specie': 'human',\n",
    "    'presence': 1\n",
    "})\n",
    "\n",
    "noncoding_to_matrix = pd.concat([human, noncoding_to_matrix], ignore_index=True)\n",
    "noncoding_to_matrix.dropna(inplace=True)\n",
    "noncoding_to_matrix = noncoding_to_matrix[~noncoding_to_matrix['specie'].str.contains(\"TCONS\")]\n",
    "\n",
    "# Pivoting the data without creating an 'id', and keeping 'orfID' as the unique identifier\n",
    "noncoding_matrix = noncoding_to_matrix.pivot_table(index='orfID', columns='specie', values='presence', aggfunc='max')\n",
    "noncoding_matrix\n",
    "# Fill NaN values with 0\n",
    "noncoding_matrix = noncoding_matrix.fillna(0)\n",
    "\n",
    "# Reset index so 'orfID' becomes a column instead of an index\n",
    "noncoding_matrix = noncoding_matrix.reset_index()\n",
    "noncoding_matrix\n",
    "# Rename columns as necessary\n",
    "noncoding_matrix.columns = [\"orfID\", \"chicken\", \"human\", \"macaca\", \"mouse\", \"opossum\", \"platypus\"]\n",
    "\n",
    "# Reorder columns for the final matrix\n",
    "noncoding_matrix_ordered = noncoding_matrix[[\"orfID\", \"human\",\"macaca\", \"mouse\", \"opossum\", \"platypus\", \"chicken\"]]\n",
    "\n",
    "# Calculate sum across species columns to identify how many species are present\n",
    "noncoding_matrix_ordered['sum'] = noncoding_matrix_ordered.select_dtypes(include='number').sum(axis=1)\n",
    "\n",
    "# Sort by the sum column to order by the number of species present\n",
    "noncoding_matrix_ordered = noncoding_matrix_ordered.sort_values(by=\"sum\", ascending=False)\n",
    "\n",
    "# Display the result\n",
    "noncoding_matrix_ordered =  noncoding_matrix_ordered.merge(noncoding_to_matrix_raw[['gene_name','gene_type','orfID']], on=\"orfID\")\n",
    "noncoding_matrix_ordered.drop_duplicates(inplace=True)\n",
    "\n",
    "noncoding_matrix_ordered.to_csv(os.path.join(evoDir,\"Clustering/LIVER_noncodingHuman_matrix.tsv\"), sep=\"\\t\", index=None)\n",
    "print(len(set(noncoding_matrix_ordered.orfID.values.tolist())))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 69,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_1203003/3108424402.py:4: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  canonical_conserved_to_matrix['specie'] = canonical_conserved_to_matrix[1].str.split(\"_\", expand=True)[0]\n",
      "/tmp/ipykernel_1203003/3108424402.py:6: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  canonical_conserved_to_matrix['presence'] = 1\n",
      "/tmp/ipykernel_1203003/3108424402.py:26: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  canonical_conserved_matrix_ordered['sum'] = canonical_conserved_matrix_ordered.select_dtypes(include='number').sum(axis=1)\n"
     ]
    }
   ],
   "source": [
    "canonical_conserved_to_matrix = canonical_conserved\n",
    "\n",
    "# Assuming noncoding_conserved_to_matrix is already defined\n",
    "canonical_conserved_to_matrix['specie'] = canonical_conserved_to_matrix[1].str.split(\"_\", expand=True)[0]\n",
    "canonical_conserved_to_matrix = canonical_conserved_to_matrix[['orfID', 'specie']]\n",
    "canonical_conserved_to_matrix['presence'] = 1\n",
    "\n",
    "canonical_conserved_to_matrix = canonical_conserved_to_matrix[~canonical_conserved_to_matrix['specie'].str.contains(\"TCONS\")]\n",
    "\n",
    "# Pivoting the data without creating an 'id', and keeping 'orfID' as the unique identifier\n",
    "canonical_conserved_matrix = canonical_conserved_to_matrix.pivot_table(index='orfID', columns='specie', values='presence', aggfunc='max')\n",
    "\n",
    "# Fill NaN values with 0\n",
    "canonical_conserved_matrix = canonical_conserved_matrix.fillna(0)\n",
    "\n",
    "# Reset index so 'orfID' becomes a column instead of an index\n",
    "canonical_conserved_matrix = canonical_conserved_matrix.reset_index()\n",
    "\n",
    "# Rename columns as necessary\n",
    "canonical_conserved_matrix.columns = [\"orfID\", \"chicken\", \"macaca\", \"mouse\", \"opossum\", \"platypus\"]\n",
    "\n",
    "# Reorder columns for the final matrix\n",
    "canonical_conserved_matrix_ordered = canonical_conserved_matrix[[\"orfID\", \"macaca\", \"mouse\", \"opossum\", \"platypus\", \"chicken\"]]\n",
    "\n",
    "# Calculate sum across species columns to identify how many species are present\n",
    "canonical_conserved_matrix_ordered['sum'] = canonical_conserved_matrix_ordered.select_dtypes(include='number').sum(axis=1)\n",
    "\n",
    "# Sort by the sum column to order by the number of species present\n",
    "canonical_conserved_matrix_ordered = canonical_conserved_matrix_ordered.sort_values(by=\"sum\", ascending=False)\n",
    "\n",
    "# Display the result\n",
    "canonical_conserved_matrix_ordered =  canonical_conserved_matrix_ordered.merge(canonical_conserved[['gene_name','gene_type','orfID']], on=\"orfID\")\n",
    "canonical_conserved_matrix_ordered.drop_duplicates(inplace=True)\n",
    "\n",
    "canonical_conserved_matrix_ordered.to_csv(os.path.join(evoDir,\"Clustering/LIVER_canonicalHuman_Conserved_matrix.tsv\"), sep=\"\\t\", index=None)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 70,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_1203003/3550657156.py:3: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  canonical_conserved_touse['specie'] = canonical_conserved_touse[1].str.split(\"_\", expand=True)[0]\n",
      "/tmp/ipykernel_1203003/3550657156.py:8: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  canonical_to_matrix['presence'] = np.where(canonical_to_matrix['specie'].isna(), 0, 1)\n",
      "/tmp/ipykernel_1203003/3550657156.py:36: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  canonical_matrix_ordered['sum'] = canonical_matrix_ordered.select_dtypes(include='number').sum(axis=1)\n"
     ]
    }
   ],
   "source": [
    "### all canonical\n",
    "canonical_conserved_touse = canonical_conserved\n",
    "canonical_conserved_touse['specie'] = canonical_conserved_touse[1].str.split(\"_\", expand=True)[0]\n",
    "# noncoding_NonConserved.drop(\"specie\",axis=1, inplace=True)\n",
    "canonical_to_matrix_raw = pd.concat([canonical_conserved, canonical_NonConserved])\n",
    "\n",
    "canonical_to_matrix = canonical_to_matrix_raw[['orfID', 'specie']]\n",
    "canonical_to_matrix['presence'] = np.where(canonical_to_matrix['specie'].isna(), 0, 1)\n",
    "\n",
    "\n",
    "human = pd.DataFrame({\n",
    "    'orfID': canonical_to_matrix['orfID'].unique(),\n",
    "    'specie': 'human',\n",
    "    'presence': 1\n",
    "})\n",
    "\n",
    "canonical_to_matrix = pd.concat([human, canonical_to_matrix], ignore_index=True)\n",
    "canonical_to_matrix.dropna(inplace=True)\n",
    "canonical_to_matrix = canonical_to_matrix[~canonical_to_matrix['specie'].str.contains(\"TCONS\")]\n",
    "# Pivoting the data without creating an 'id', and keeping 'orfID' as the unique identifier\n",
    "canonical_matrix = canonical_to_matrix.pivot_table(index='orfID', columns='specie', values='presence', aggfunc='max')\n",
    "canonical_matrix\n",
    "# Fill NaN values with 0\n",
    "canonical_matrix = canonical_matrix.fillna(0)\n",
    "\n",
    "# Reset index so 'orfID' becomes a column instead of an index\n",
    "canonical_matrix = canonical_matrix.reset_index()\n",
    "\n",
    "# Rename columns as necessary\n",
    "canonical_matrix.columns = [\"orfID\", \"chicken\", \"human\", \"macaca\", \"mouse\", \"opossum\", \"platypus\"]\n",
    "\n",
    "# Reorder columns for the final matrix\n",
    "canonical_matrix_ordered = canonical_matrix[[\"orfID\", \"human\",\"macaca\", \"mouse\", \"opossum\", \"platypus\", \"chicken\"]]\n",
    "\n",
    "# Calculate sum across species columns to identify how many species are present\n",
    "canonical_matrix_ordered['sum'] = canonical_matrix_ordered.select_dtypes(include='number').sum(axis=1)\n",
    "\n",
    "# Sort by the sum column to order by the number of species present\n",
    "canonical_matrix_ordered = canonical_matrix_ordered.sort_values(by=\"sum\", ascending=False)\n",
    "\n",
    "# Display the result\n",
    "canonical_matrix_ordered =  canonical_matrix_ordered.merge(canonical_to_matrix_raw[['gene_name','gene_type','orfID']], on=\"orfID\")\n",
    "canonical_matrix_ordered.drop_duplicates(inplace=True)\n",
    "\n",
    "canonical_matrix_ordered.to_csv(os.path.join(evoDir,\"Clustering/LIVER_canonicalHuman_matrix.tsv\"), sep=\"\\t\", index=None)"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Find the oldest specie containg the ORF translated"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 71,
   "metadata": {},
   "outputs": [],
   "source": [
    "species_ordered = ['chicken','platypus','opossum','mouse','macaca','human']\n",
    "def find_last_species(row):\n",
    "    for sp in species_ordered:\n",
    "        if row[sp] == 1.0:  # Check if the value is 1.0\n",
    "            return sp\n",
    "    return None  # Return None if no species has a 1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 72,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Apply the function to each row to create a new column 'oldest_species'\n",
    "canonical_matrix_ordered['oldest_species'] = canonical_matrix_ordered.apply(find_last_species, axis=1)\n",
    "canonical_matrix_ordered = species_and_years.merge(canonical_matrix_ordered, left_on=[\"specie\"], right_on=\"oldest_species\")\n",
    "canonical_matrix_ordered[['orfID','gene_name','gene_type','oldest_species','age']].to_csv(os.path.join(evoDir,\"Clustering/LIVER_canonicalHuman_matrix_oldestAge.tsv\"), sep=\"\\t\", index=None)\n",
    "\n",
    "noncoding_matrix_ordered['oldest_species'] = noncoding_matrix_ordered.apply(find_last_species, axis=1)\n",
    "noncoding_matrix_ordered = species_and_years.merge(noncoding_matrix_ordered, left_on=[\"specie\"], right_on=\"oldest_species\")\n",
    "noncoding_matrix_ordered[['orfID','gene_name','gene_type','oldest_species','age']].to_csv(os.path.join(evoDir,\"Clustering/LIVER_noncodingHuman_matrix_oldestAge.tsv\"), sep=\"\\t\", index=None)\n",
    "\n",
    "complete_matrix_age = pd.concat([noncoding_matrix_ordered[['orfID','gene_name','gene_type','oldest_species','age']], canonical_matrix_ordered[['orfID','gene_name','gene_type','oldest_species','age']]])\n",
    "complete_matrix_age.to_csv(os.path.join(evoDir,\"Clustering/LIVER_allHuman_matrix_oldestAge.tsv\"), sep=\"\\t\", index=None)\n"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Brain-EXPRESSED"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 73,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "13297"
      ]
     },
     "execution_count": 73,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "## Brain-expressed ORFs human\n",
    "BrainExpressed_translatedINbrain = pd.read_csv(\"/projects_eg/projects/marta/TestisRestricted_Microproteins_TSA/with_TranscriptomeReconstruction/ribORF_humanBrain_in1.csv\")\n",
    "BrainExpressed_translatedINbrain\n",
    "\n",
    "def create_seqrecord_PROThuman(row):\n",
    "    return SeqRecord(Seq(row['ORFpep']), id=row['orfID'], description=\"\")\n",
    "\n",
    "out_nonredundant_proteins = \"/projects_eg/projects/marta/TestisRestricted_Microproteins_TSA/with_TranscriptomeReconstruction/Q1_TestisORFs/human/ribORF_humanBrain.fa\"\n",
    "seq_records_proteins = BrainExpressed_translatedINbrain.apply(create_seqrecord_PROThuman, axis=1).tolist()\n",
    "SeqIO.write(seq_records_proteins, out_nonredundant_proteins, \"fasta\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 74,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/EvolutionaryOrigin_MMseqs/MMseqsDB/ribORF_humanBrain_in1_DB exists and will be overwritten\n",
      "createdb /projects_eg/projects/marta/TestisRestricted_Microproteins_TSA/with_TranscriptomeReconstruction/Q1_TestisORFs/human/ribORF_humanBrain.fa /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/EvolutionaryOrigin_MMseqs/MMseqsDB/ribORF_humanBrain_in1_DB \n",
      "\n",
      "MMseqs Version:       \t14.7e284\n",
      "Database type         \t0\n",
      "Shuffle input database\ttrue\n",
      "Createdb mode         \t0\n",
      "Write lookup file     \t1\n",
      "Offset of numeric ids \t0\n",
      "Compressed            \t0\n",
      "Verbosity             \t3\n",
      "\n",
      "Converting sequences\n",
      "[=\n",
      "Time for merging to ribORF_humanBrain_in1_DB_h: 0h 0m 7s 812ms\n",
      "Time for merging to ribORF_humanBrain_in1_DB: 0h 0m 11s 606ms\n",
      "Database type: Aminoacid\n",
      "Time for processing: 0h 1m 28s 705ms\n"
     ]
    }
   ],
   "source": [
    "%%bash -s \"$evoDir\" \"$users_dir\"\n",
    "\n",
    "# mkdir $1/MMseqsDB\n",
    "\n",
    "module load Miniconda3/4.9.2\n",
    "source /soft/system/software/Miniconda3/4.9.2/bin/activate MMseqs2\n",
    "\n",
    "mmseqs createdb /projects_eg/projects/marta/TestisRestricted_Microproteins_TSA/with_TranscriptomeReconstruction/Q1_TestisORFs/human/ribORF_humanBrain.fa $1/MMseqsDB/ribORF_humanBrain_in1_DB\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 75,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/EvolutionaryOrigin_MMseqs/MMseqsDB/species_and_humanBrainDB exists and will be overwritten\n",
      "createdb /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/chicken/RiboSeq/RiboQC_RiboNovel/RibORF/chicken_repre.valid.pred.pvalue.parameters.PROTEIN.noredunant.fa /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/macaca/RiboSeq/RiboQC_RiboNovel/RibORF/macaca_repre.valid.pred.pvalue.parameters.PROTEIN.noredunant.fa /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/mouse/RiboSeq/RiboQC_RiboNovel/RibORF/mouse_repre.valid.pred.pvalue.parameters.PROTEIN.noredunant.fa /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/opossum/RiboSeq/RiboQC_RiboNovel/RibORF/opossum_repre.valid.pred.pvalue.parameters.PROTEIN.noredunant.fa /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/platypus/RiboSeq/RiboQC_RiboNovel/RibORF/platypus_repre.valid.pred.pvalue.parameters.PROTEIN.noredunant.fa /projects_eg/projects/marta/TestisRestricted_Microproteins_TSA/with_TranscriptomeReconstruction/Q1_TestisORFs/human/ribORF_humanBrain.fa /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/EvolutionaryOrigin_MMseqs/MMseqsDB/species_and_humanBrainDB \n",
      "\n",
      "MMseqs Version:       \t14.7e284\n",
      "Database type         \t0\n",
      "Shuffle input database\ttrue\n",
      "Createdb mode         \t0\n",
      "Write lookup file     \t1\n",
      "Offset of numeric ids \t0\n",
      "Compressed            \t0\n",
      "Verbosity             \t3\n",
      "\n",
      "Converting sequences\n",
      "[============\n",
      "Time for merging to species_and_humanBrainDB_h: 0h 0m 9s 618ms\n",
      "Time for merging to species_and_humanBrainDB: 0h 1m 23s 993ms\n",
      "Database type: Aminoacid\n",
      "Time for processing: 0h 2m 56s 423ms\n"
     ]
    }
   ],
   "source": [
    "%%bash -s \"$evoDir\" \"$users_dir\"\n",
    "\n",
    "module load Miniconda3/4.9.2\n",
    "source /soft/system/software/Miniconda3/4.9.2/bin/activate MMseqs2\n",
    "\n",
    "mmseqs createdb $2/*/RiboSeq/RiboQC_RiboNovel*/RibORF/*_repre.valid.pred.pvalue.parameters.PROTEIN.noredunant.fa /projects_eg/projects/marta/TestisRestricted_Microproteins_TSA/with_TranscriptomeReconstruction/Q1_TestisORFs/human/ribORF_humanBrain.fa $1/MMseqsDB/species_and_humanBrainDB\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 76,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "cluster -c 0.5 --min-seq-id 0.5 /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/EvolutionaryOrigin_MMseqs/MMseqsDB/species_and_humanBrainDB /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/EvolutionaryOrigin_MMseqs/Clustering/species_and_humanBrain_cluDB /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/EvolutionaryOrigin_MMseqs/tmp \n",
      "\n",
      "MMseqs Version:                     \t14.7e284\n",
      "Substitution matrix                 \taa:blosum62.out,nucl:nucleotide.out\n",
      "Seed substitution matrix            \taa:VTML80.out,nucl:nucleotide.out\n",
      "Sensitivity                         \t4\n",
      "k-mer length                        \t0\n",
      "k-score                             \tseq:2147483647,prof:2147483647\n",
      "Alphabet size                       \taa:21,nucl:5\n",
      "Max sequence length                 \t65535\n",
      "Max results per query               \t20\n",
      "Split database                      \t0\n",
      "Split mode                          \t2\n",
      "Split memory limit                  \t0\n",
      "Coverage threshold                  \t0.5\n",
      "Coverage mode                       \t0\n",
      "Compositional bias                  \t1\n",
      "Compositional bias                  \t1\n",
      "Diagonal scoring                    \ttrue\n",
      "Exact k-mer matching                \t0\n",
      "Mask residues                       \t1\n",
      "Mask residues probability           \t0.9\n",
      "Mask lower case residues            \t0\n",
      "Minimum diagonal score              \t15\n",
      "Selected taxa                       \t\n",
      "Include identical seq. id.          \tfalse\n",
      "Spaced k-mers                       \t1\n",
      "Preload mode                        \t0\n",
      "Pseudo count a                      \tsubstitution:1.100,context:1.400\n",
      "Pseudo count b                      \tsubstitution:4.100,context:5.800\n",
      "Spaced k-mer pattern                \t\n",
      "Local temporary path                \t\n",
      "Threads                             \t24\n",
      "Compressed                          \t0\n",
      "Verbosity                           \t3\n",
      "Add backtrace                       \tfalse\n",
      "Alignment mode                      \t3\n",
      "Alignment mode                      \t0\n",
      "Allow wrapped scoring               \tfalse\n",
      "E-value threshold                   \t0.001\n",
      "Seq. id. threshold                  \t0.5\n",
      "Min alignment length                \t0\n",
      "Seq. id. mode                       \t0\n",
      "Alternative alignments              \t0\n",
      "Max reject                          \t2147483647\n",
      "Max accept                          \t2147483647\n",
      "Score bias                          \t0\n",
      "Realign hits                        \tfalse\n",
      "Realign score bias                  \t-0.2\n",
      "Realign max seqs                    \t2147483647\n",
      "Correlation score weight            \t0\n",
      "Gap open cost                       \taa:11,nucl:5\n",
      "Gap extension cost                  \taa:1,nucl:2\n",
      "Zdrop                               \t40\n",
      "Rescore mode                        \t0\n",
      "Remove hits by seq. id. and coverage\tfalse\n",
      "Sort results                        \t0\n",
      "Cluster mode                        \t0\n",
      "Max connected component depth       \t1000\n",
      "Similarity type                     \t2\n",
      "Single step clustering              \tfalse\n",
      "Cascaded clustering steps           \t3\n",
      "Cluster reassign                    \tfalse\n",
      "Remove temporary files              \tfalse\n",
      "Force restart with latest tmp       \tfalse\n",
      "MPI runner                          \t\n",
      "k-mers per sequence                 \t21\n",
      "Scale k-mers per sequence           \taa:0.000,nucl:0.200\n",
      "Adjust k-mer length                 \tfalse\n",
      "Shift hash                          \t67\n",
      "Include only extendable             \tfalse\n",
      "Skip repeating k-mers               \tfalse\n",
      "\n",
      "Set cluster sensitivity to -s 3.000000\n",
      "Set cluster mode SET COVER\n",
      "Set cluster iterations to 3\n",
      "linclust /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/EvolutionaryOrigin_MMseqs/MMseqsDB/species_and_humanBrainDB /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/EvolutionaryOrigin_MMseqs/tmp/9674412578754287220/clu_redundancy /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/EvolutionaryOrigin_MMseqs/tmp/9674412578754287220/linclust --cluster-mode 0 --max-iterations 1000 --similarity-type 2 --threads 24 --compressed 0 -v 3 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' -a 0 --alignment-mode 3 --alignment-output-mode 0 --wrapped-scoring 0 -e 0.001 --min-seq-id 0.5 --min-aln-len 0 --seq-id-mode 0 --alt-ali 0 -c 0.5 --cov-mode 0 --max-seq-len 65535 --comp-bias-corr 1 --comp-bias-corr-scale 1 --max-rejected 2147483647 --max-accept 2147483647 --add-self-matches 0 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --score-bias 0 --realign 0 --realign-score-bias -0.2 --realign-max-seqs 2147483647 --corr-score-weight 0 --gap-open aa:11,nucl:5 --gap-extend aa:1,nucl:2 --zdrop 40 --alph-size aa:13,nucl:5 --kmer-per-seq 21 --spaced-kmer-mode 1 --kmer-per-seq-scale aa:0.000,nucl:0.200 --adjust-kmer-len 0 --mask 0 --mask-prob 0.9 --mask-lower-case 0 -k 0 --hash-shift 67 --split-memory-limit 0 --include-only-extendable 0 --ignore-multi-kmer 0 --rescore-mode 0 --filter-hits 0 --sort-results 0 --remove-tmp-files 0 --force-reuse 0 \n",
      "\n",
      "kmermatcher /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/EvolutionaryOrigin_MMseqs/MMseqsDB/species_and_humanBrainDB /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/EvolutionaryOrigin_MMseqs/tmp/9674412578754287220/linclust/7810710425537572980/pref --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' --alph-size aa:13,nucl:5 --min-seq-id 0.5 --kmer-per-seq 21 --spaced-kmer-mode 1 --kmer-per-seq-scale aa:0.000,nucl:0.200 --adjust-kmer-len 0 --mask 0 --mask-prob 0.9 --mask-lower-case 0 --cov-mode 0 -k 0 -c 0.5 --max-seq-len 65535 --hash-shift 67 --split-memory-limit 0 --include-only-extendable 0 --ignore-multi-kmer 0 --threads 24 --compressed 0 -v 3 \n",
      "\n",
      "kmermatcher /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/EvolutionaryOrigin_MMseqs/MMseqsDB/species_and_humanBrainDB /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/EvolutionaryOrigin_MMseqs/tmp/9674412578754287220/linclust/7810710425537572980/pref --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' --alph-size aa:13,nucl:5 --min-seq-id 0.5 --kmer-per-seq 21 --spaced-kmer-mode 1 --kmer-per-seq-scale aa:0.000,nucl:0.200 --adjust-kmer-len 0 --mask 0 --mask-prob 0.9 --mask-lower-case 0 --cov-mode 0 -k 0 -c 0.5 --max-seq-len 65535 --hash-shift 67 --split-memory-limit 0 --include-only-extendable 0 --ignore-multi-kmer 0 --threads 24 --compressed 0 -v 3 \n",
      "\n",
      "Database size: 128710 type: Aminoacid\n",
      "Reduced amino acid alphabet: (A S T) (C) (D B N) (E Q Z) (F Y) (G) (H) (I V) (K R) (L J M) (P) (W) (X) \n",
      "\n",
      "Generate k-mers list for 1 split\n",
      "[=================================================================] 128.71K 1m 8s 2ms\n",
      "Sort kmer 0h 0m 0s 36ms\n",
      "Sort by rep. sequence 0h 0m 0s 10ms\n",
      "Time for fill: 0h 0m 0s 49ms\n",
      "Time for merging to pref: 0h 0m 4s 362ms\n",
      "Time for processing: 0h 1m 12s 829ms\n",
      "rescorediagonal /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/EvolutionaryOrigin_MMseqs/MMseqsDB/species_and_humanBrainDB /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/EvolutionaryOrigin_MMseqs/MMseqsDB/species_and_humanBrainDB /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/EvolutionaryOrigin_MMseqs/tmp/9674412578754287220/linclust/7810710425537572980/pref /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/EvolutionaryOrigin_MMseqs/tmp/9674412578754287220/linclust/7810710425537572980/pref_rescore1 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' --rescore-mode 0 --wrapped-scoring 0 --filter-hits 0 -e 0.001 -c 0.5 -a 0 --cov-mode 0 --min-seq-id 0.5 --min-aln-len 0 --seq-id-mode 0 --add-self-matches 0 --sort-results 0 --db-load-mode 0 --threads 24 --compressed 0 -v 3 \n",
      "\n",
      "[=================================================================] 128.71K 0s 55ms\n",
      "Time for merging to pref_rescore1: 0h 0m 4s 34ms\n",
      "Time for processing: 0h 0m 29s 460ms\n",
      "clust /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/EvolutionaryOrigin_MMseqs/MMseqsDB/species_and_humanBrainDB /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/EvolutionaryOrigin_MMseqs/tmp/9674412578754287220/linclust/7810710425537572980/pref_rescore1 /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/EvolutionaryOrigin_MMseqs/tmp/9674412578754287220/linclust/7810710425537572980/pre_clust --cluster-mode 0 --max-iterations 1000 --similarity-type 2 --threads 24 --compressed 0 -v 3 \n",
      "\n",
      "Clustering mode: Set Cover\n",
      "[=================================================================] 128.71K 0s 13ms\n",
      "Sort entries\n",
      "Find missing connections\n",
      "Found 82800 new connections.\n",
      "Reconstruct initial order\n",
      "[=================================================================] 128.71K 0s 11ms\n",
      "Add missing connections\n",
      "[=================================================================] 128.71K 0s 5ms\n",
      "\n",
      "Time for read in: 0h 0m 0s 85ms\n",
      "Total time: 0h 0m 2s 534ms\n",
      "\n",
      "Size of the sequence database: 128710\n",
      "Size of the alignment database: 128710\n",
      "Number of clusters: 83669\n",
      "\n",
      "Writing results 0h 0m 0s 19ms\n",
      "Time for merging to pre_clust: 0h 0m 0s 802ms\n",
      "Time for processing: 0h 0m 5s 533ms\n",
      "createsubdb /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/EvolutionaryOrigin_MMseqs/tmp/9674412578754287220/linclust/7810710425537572980/order_redundancy /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/EvolutionaryOrigin_MMseqs/MMseqsDB/species_and_humanBrainDB /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/EvolutionaryOrigin_MMseqs/tmp/9674412578754287220/linclust/7810710425537572980/input_step_redundancy -v 3 --subdb-mode 1 \n",
      "\n",
      "Time for merging to input_step_redundancy: 0h 0m 1s 51ms\n",
      "Time for processing: 0h 0m 7s 672ms\n",
      "createsubdb /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/EvolutionaryOrigin_MMseqs/tmp/9674412578754287220/linclust/7810710425537572980/order_redundancy /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/EvolutionaryOrigin_MMseqs/tmp/9674412578754287220/linclust/7810710425537572980/pref /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/EvolutionaryOrigin_MMseqs/tmp/9674412578754287220/linclust/7810710425537572980/pref_filter1 -v 3 --subdb-mode 1 \n",
      "\n",
      "Time for merging to pref_filter1: 0h 0m 2s 411ms\n",
      "Time for processing: 0h 0m 6s 560ms\n",
      "filterdb /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/EvolutionaryOrigin_MMseqs/tmp/9674412578754287220/linclust/7810710425537572980/pref_filter1 /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/EvolutionaryOrigin_MMseqs/tmp/9674412578754287220/linclust/7810710425537572980/pref_filter2 --filter-file /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/EvolutionaryOrigin_MMseqs/tmp/9674412578754287220/linclust/7810710425537572980/order_redundancy --threads 24 --compressed 0 -v 3 \n",
      "\n",
      "Filtering using file(s)\n",
      "[=================================================================] 83.67K 0s 25ms\n",
      "Time for merging to pref_filter2: 0h 0m 2s 247ms\n",
      "Time for processing: 0h 0m 28s 217ms\n",
      "rescorediagonal /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/EvolutionaryOrigin_MMseqs/tmp/9674412578754287220/linclust/7810710425537572980/input_step_redundancy /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/EvolutionaryOrigin_MMseqs/tmp/9674412578754287220/linclust/7810710425537572980/input_step_redundancy /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/EvolutionaryOrigin_MMseqs/tmp/9674412578754287220/linclust/7810710425537572980/pref_filter2 /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/EvolutionaryOrigin_MMseqs/tmp/9674412578754287220/linclust/7810710425537572980/pref_rescore2 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' --rescore-mode 1 --wrapped-scoring 0 --filter-hits 1 -e 0.001 -c 0.5 -a 0 --cov-mode 0 --min-seq-id 0.5 --min-aln-len 0 --seq-id-mode 0 --add-self-matches 0 --sort-results 0 --db-load-mode 0 --threads 24 --compressed 0 -v 3 \n",
      "\n",
      "[=================================================================] 83.67K 12s 538ms\n",
      "Time for merging to pref_rescore2: 0h 0m 1s 217ms\n",
      "Time for processing: 0h 0m 26s 460ms\n",
      "align /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/EvolutionaryOrigin_MMseqs/tmp/9674412578754287220/linclust/7810710425537572980/input_step_redundancy /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/EvolutionaryOrigin_MMseqs/tmp/9674412578754287220/linclust/7810710425537572980/input_step_redundancy /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/EvolutionaryOrigin_MMseqs/tmp/9674412578754287220/linclust/7810710425537572980/pref_rescore2 /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/EvolutionaryOrigin_MMseqs/tmp/9674412578754287220/linclust/7810710425537572980/aln --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' -a 0 --alignment-mode 3 --alignment-output-mode 0 --wrapped-scoring 0 -e 0.001 --min-seq-id 0.5 --min-aln-len 0 --seq-id-mode 0 --alt-ali 0 -c 0.5 --cov-mode 0 --max-seq-len 65535 --comp-bias-corr 1 --comp-bias-corr-scale 1 --max-rejected 2147483647 --max-accept 2147483647 --add-self-matches 0 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --score-bias 0 --realign 0 --realign-score-bias -0.2 --realign-max-seqs 2147483647 --corr-score-weight 0 --gap-open aa:11,nucl:5 --gap-extend aa:1,nucl:2 --zdrop 40 --threads 24 --compressed 0 -v 3 \n",
      "\n",
      "Compute score, coverage and sequence identity\n",
      "Query database size: 83669 type: Aminoacid\n",
      "Target database size: 83669 type: Aminoacid\n",
      "Calculation of alignments\n",
      "[=================================================================] 83.67K 7s 307ms\n",
      "Time for merging to aln: 0h 0m 7s 182ms\n",
      "105954 alignments calculated\n",
      "100501 sequence pairs passed the thresholds (0.948534 of overall calculated)\n",
      "1.201174 hits per query sequence\n",
      "Time for processing: 0h 0m 37s 464ms\n",
      "clust /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/EvolutionaryOrigin_MMseqs/tmp/9674412578754287220/linclust/7810710425537572980/input_step_redundancy /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/EvolutionaryOrigin_MMseqs/tmp/9674412578754287220/linclust/7810710425537572980/aln /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/EvolutionaryOrigin_MMseqs/tmp/9674412578754287220/linclust/7810710425537572980/clust --cluster-mode 0 --max-iterations 1000 --similarity-type 2 --threads 24 --compressed 0 -v 3 \n",
      "\n",
      "Clustering mode: Set Cover\n",
      "[=================================================================] 83.67K 0s 9ms\n",
      "Sort entries\n",
      "Find missing connections\n",
      "Found 16832 new connections.\n",
      "Reconstruct initial order\n",
      "[=================================================================] 83.67K 0s 9ms\n",
      "Add missing connections\n",
      "[=================================================================] 83.67K 0s 2ms\n",
      "\n",
      "Time for read in: 0h 0m 0s 57ms\n",
      "Total time: 0h 0m 2s 856ms\n",
      "\n",
      "Size of the sequence database: 83669\n",
      "Size of the alignment database: 83669\n",
      "Number of clusters: 70830\n",
      "\n",
      "Writing results 0h 0m 0s 14ms\n",
      "Time for merging to clust: 0h 0m 3s 31ms\n",
      "Time for processing: 0h 0m 7s 214ms\n",
      "mergeclusters /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/EvolutionaryOrigin_MMseqs/MMseqsDB/species_and_humanBrainDB /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/EvolutionaryOrigin_MMseqs/tmp/9674412578754287220/clu_redundancy /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/EvolutionaryOrigin_MMseqs/tmp/9674412578754287220/linclust/7810710425537572980/pre_clust /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/EvolutionaryOrigin_MMseqs/tmp/9674412578754287220/linclust/7810710425537572980/clust --threads 24 --compressed 0 -v 3 \n",
      "\n",
      "Clustering step 1\n",
      "[=================================================================] 83.67K 3s 872ms\n",
      "Clustering step 2\n",
      "[=================================================================] 70.83K 3s 916ms\n",
      "Write merged clustering\n",
      "[=================================================================] 128.71K 20s 97ms\n",
      "Time for merging to clu_redundancy: 0h 0m 2s 470ms\n",
      "Time for processing: 0h 0m 29s 948ms\n",
      "createsubdb /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/EvolutionaryOrigin_MMseqs/tmp/9674412578754287220/clu_redundancy /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/EvolutionaryOrigin_MMseqs/MMseqsDB/species_and_humanBrainDB /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/EvolutionaryOrigin_MMseqs/tmp/9674412578754287220/input_step_redundancy -v 3 --subdb-mode 1 \n",
      "\n",
      "Time for merging to input_step_redundancy: 0h 0m 1s 749ms\n",
      "Time for processing: 0h 0m 5s 770ms\n",
      "prefilter /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/EvolutionaryOrigin_MMseqs/tmp/9674412578754287220/input_step_redundancy /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/EvolutionaryOrigin_MMseqs/tmp/9674412578754287220/input_step_redundancy /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/EvolutionaryOrigin_MMseqs/tmp/9674412578754287220/pref_step0 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' --seed-sub-mat 'aa:VTML80.out,nucl:nucleotide.out' -s 1 -k 0 --k-score seq:2147483647,prof:2147483647 --alph-size aa:21,nucl:5 --max-seq-len 65535 --max-seqs 20 --split 0 --split-mode 2 --split-memory-limit 0 -c 0.5 --cov-mode 0 --comp-bias-corr 0 --comp-bias-corr-scale 1 --diag-score 0 --exact-kmer-matching 0 --mask 1 --mask-prob 0.9 --mask-lower-case 0 --min-ungapped-score 0 --add-self-matches 0 --spaced-kmer-mode 1 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --threads 24 --compressed 0 -v 3 \n",
      "\n",
      "Query database size: 70830 type: Aminoacid\n",
      "Estimated memory consumption: 1G\n",
      "Target database size: 70830 type: Aminoacid\n",
      "Index table k-mer threshold: 154 at k-mer size 6 \n",
      "Index table: counting k-mers\n",
      "[=================================================================] 70.83K 0s 219ms\n",
      "Index table: Masked residues: 891942\n",
      "Index table: fill\n",
      "[=================================================================] 70.83K 0s 105ms\n",
      "Index statistics\n",
      "Entries:          8535084\n",
      "DB size:          537 MB\n",
      "Avg k-mer size:   0.133361\n",
      "Top 10 k-mers\n",
      "    LYDMNY\t148\n",
      "    TFDAFT\t126\n",
      "    VADTEW\t88\n",
      "    EWLDQK\t63\n",
      "    YRVLYR\t60\n",
      "    ECKFSS\t58\n",
      "    VAYSEW\t47\n",
      "    LYEMNY\t46\n",
      "    VADSEW\t42\n",
      "    QRHGPY\t39\n",
      "Time for index table init: 0h 0m 0s 766ms\n",
      "Process prefiltering step 1 of 1\n",
      "\n",
      "k-mer similarity threshold: 154\n",
      "Starting prefiltering scores calculation (step 1 of 1)\n",
      "Query db start 1 to 70830\n",
      "Target db start 1 to 70830\n",
      "[=================================================================] 70.83K 0s 341ms\n",
      "\n",
      "2.267984 k-mers per position\n",
      "395 DB matches per sequence\n",
      "0 overflows\n",
      "0 queries produce too many hits (truncated result)\n",
      "3 sequences passed prefiltering per query sequence\n",
      "2 median result list length\n",
      "0 sequences with 0 size result lists\n",
      "Time for merging to pref_step0: 0h 0m 8s 613ms\n",
      "Time for processing: 0h 0m 48s 990ms\n",
      "align /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/EvolutionaryOrigin_MMseqs/tmp/9674412578754287220/input_step_redundancy /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/EvolutionaryOrigin_MMseqs/tmp/9674412578754287220/input_step_redundancy /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/EvolutionaryOrigin_MMseqs/tmp/9674412578754287220/pref_step0 /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/EvolutionaryOrigin_MMseqs/tmp/9674412578754287220/aln_step0 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' -a 0 --alignment-mode 3 --alignment-output-mode 0 --wrapped-scoring 0 -e 0.001 --min-seq-id 0.5 --min-aln-len 0 --seq-id-mode 0 --alt-ali 0 -c 0.5 --cov-mode 0 --max-seq-len 65535 --comp-bias-corr 0 --comp-bias-corr-scale 1 --max-rejected 2147483647 --max-accept 2147483647 --add-self-matches 0 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --score-bias 0 --realign 0 --realign-score-bias -0.2 --realign-max-seqs 2147483647 --corr-score-weight 0 --gap-open aa:11,nucl:5 --gap-extend aa:1,nucl:2 --zdrop 40 --threads 24 --compressed 0 -v 3 \n",
      "\n",
      "Compute score, coverage and sequence identity\n",
      "Query database size: 70830 type: Aminoacid\n",
      "Target database size: 70830 type: Aminoacid\n",
      "Calculation of alignments\n",
      "[=================================================================] 70.83K 14s 534ms\n",
      "Time for merging to aln_step0: 0h 0m 2s 444ms\n",
      "189693 alignments calculated\n",
      "91386 sequence pairs passed the thresholds (0.481757 of overall calculated)\n",
      "1.290216 hits per query sequence\n",
      "Time for processing: 0h 0m 40s 505ms\n",
      "clust /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/EvolutionaryOrigin_MMseqs/tmp/9674412578754287220/input_step_redundancy /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/EvolutionaryOrigin_MMseqs/tmp/9674412578754287220/aln_step0 /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/EvolutionaryOrigin_MMseqs/tmp/9674412578754287220/clu_step0 --cluster-mode 0 --max-iterations 1000 --similarity-type 2 --threads 24 --compressed 0 -v 3 \n",
      "\n",
      "Clustering mode: Set Cover\n",
      "[=================================================================] 70.83K 0s 7ms\n",
      "Sort entries\n",
      "Find missing connections\n",
      "Found 2028 new connections.\n",
      "Reconstruct initial order\n",
      "[=================================================================] 70.83K 0s 9ms\n",
      "Add missing connections\n",
      "[=================================================================] 70.83K 0s 1ms\n",
      "\n",
      "Time for read in: 0h 0m 0s 53ms\n",
      "Total time: 0h 0m 3s 130ms\n",
      "\n",
      "Size of the sequence database: 70830\n",
      "Size of the alignment database: 70830\n",
      "Number of clusters: 63421\n",
      "\n",
      "Writing results 0h 0m 0s 15ms\n",
      "Time for merging to clu_step0: 0h 0m 1s 869ms\n",
      "Time for processing: 0h 0m 5s 251ms\n",
      "createsubdb /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/EvolutionaryOrigin_MMseqs/tmp/9674412578754287220/clu_step0 /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/EvolutionaryOrigin_MMseqs/tmp/9674412578754287220/input_step_redundancy /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/EvolutionaryOrigin_MMseqs/tmp/9674412578754287220/input_step1 -v 3 --subdb-mode 1 \n",
      "\n",
      "Time for merging to input_step1: 0h 0m 1s 679ms\n",
      "Time for processing: 0h 0m 6s 396ms\n",
      "prefilter /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/EvolutionaryOrigin_MMseqs/tmp/9674412578754287220/input_step1 /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/EvolutionaryOrigin_MMseqs/tmp/9674412578754287220/input_step1 /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/EvolutionaryOrigin_MMseqs/tmp/9674412578754287220/pref_step1 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' --seed-sub-mat 'aa:VTML80.out,nucl:nucleotide.out' -s 2 -k 0 --k-score seq:2147483647,prof:2147483647 --alph-size aa:21,nucl:5 --max-seq-len 65535 --max-seqs 20 --split 0 --split-mode 2 --split-memory-limit 0 -c 0.5 --cov-mode 0 --comp-bias-corr 1 --comp-bias-corr-scale 1 --diag-score 1 --exact-kmer-matching 0 --mask 1 --mask-prob 0.9 --mask-lower-case 0 --min-ungapped-score 15 --add-self-matches 0 --spaced-kmer-mode 1 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --threads 24 --compressed 0 -v 3 \n",
      "\n",
      "Query database size: 63421 type: Aminoacid\n",
      "Estimated memory consumption: 1G\n",
      "Target database size: 63421 type: Aminoacid\n",
      "Index table k-mer threshold: 145 at k-mer size 6 \n",
      "Index table: counting k-mers\n",
      "[=================================================================] 63.42K 0s 176ms\n",
      "Index table: Masked residues: 586121\n",
      "Index table: fill\n",
      "[=================================================================] 63.42K 0s 109ms\n",
      "Index statistics\n",
      "Entries:          9119842\n",
      "DB size:          540 MB\n",
      "Avg k-mer size:   0.142498\n",
      "Top 10 k-mers\n",
      "    DVVFEE\t113\n",
      "    LYDMNY\t61\n",
      "    QIELEQ\t55\n",
      "    TFDAFT\t53\n",
      "    VMEYLV\t51\n",
      "    CQDGLV\t33\n",
      "    LLPQLY\t33\n",
      "    PPPSPP\t29\n",
      "    PEHKIL\t28\n",
      "    PPPPGP\t28\n",
      "Time for index table init: 0h 0m 0s 749ms\n",
      "Process prefiltering step 1 of 1\n",
      "\n",
      "k-mer similarity threshold: 145\n",
      "Starting prefiltering scores calculation (step 1 of 1)\n",
      "Query db start 1 to 63421\n",
      "Target db start 1 to 63421\n",
      "[=================================================================] 63.42K 0s 683ms\n",
      "\n",
      "6.077798 k-mers per position\n",
      "341 DB matches per sequence\n",
      "0 overflows\n",
      "0 queries produce too many hits (truncated result)\n",
      "3 sequences passed prefiltering per query sequence\n",
      "1 median result list length\n",
      "0 sequences with 0 size result lists\n",
      "Time for merging to pref_step1: 0h 0m 4s 267ms\n",
      "Time for processing: 0h 0m 21s 593ms\n",
      "align /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/EvolutionaryOrigin_MMseqs/tmp/9674412578754287220/input_step1 /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/EvolutionaryOrigin_MMseqs/tmp/9674412578754287220/input_step1 /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/EvolutionaryOrigin_MMseqs/tmp/9674412578754287220/pref_step1 /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/EvolutionaryOrigin_MMseqs/tmp/9674412578754287220/aln_step1 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' -a 0 --alignment-mode 3 --alignment-output-mode 0 --wrapped-scoring 0 -e 0.001 --min-seq-id 0.5 --min-aln-len 0 --seq-id-mode 0 --alt-ali 0 -c 0.5 --cov-mode 0 --max-seq-len 65535 --comp-bias-corr 1 --comp-bias-corr-scale 1 --max-rejected 2147483647 --max-accept 2147483647 --add-self-matches 0 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --score-bias 0 --realign 0 --realign-score-bias -0.2 --realign-max-seqs 2147483647 --corr-score-weight 0 --gap-open aa:11,nucl:5 --gap-extend aa:1,nucl:2 --zdrop 40 --threads 24 --compressed 0 -v 3 \n",
      "\n",
      "Compute score, coverage and sequence identity\n",
      "Query database size: 63421 type: Aminoacid\n",
      "Target database size: 63421 type: Aminoacid\n",
      "Calculation of alignments\n",
      "[=================================================================] 63.42K 6s 187ms\n",
      "Time for merging to aln_step1: 0h 0m 5s 703ms\n",
      "141053 alignments calculated\n",
      "64191 sequence pairs passed the thresholds (0.455084 of overall calculated)\n",
      "1.012141 hits per query sequence\n",
      "Time for processing: 0h 0m 41s 223ms\n",
      "clust /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/EvolutionaryOrigin_MMseqs/tmp/9674412578754287220/input_step1 /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/EvolutionaryOrigin_MMseqs/tmp/9674412578754287220/aln_step1 /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/EvolutionaryOrigin_MMseqs/tmp/9674412578754287220/clu_step1 --cluster-mode 0 --max-iterations 1000 --similarity-type 2 --threads 24 --compressed 0 -v 3 \n",
      "\n",
      "Clustering mode: Set Cover\n",
      "[=================================================================] 63.42K 0s 13ms\n",
      "Sort entries\n",
      "Find missing connections\n",
      "Found 240 new connections.\n",
      "Reconstruct initial order\n",
      "[=================================================================] 63.42K 0s 7ms\n",
      "Add missing connections\n",
      "[=================================================================] 63.42K 0s 1ms\n",
      "\n",
      "Time for read in: 0h 0m 0s 50ms\n",
      "Total time: 0h 0m 4s 366ms\n",
      "\n",
      "Size of the sequence database: 63421\n",
      "Size of the alignment database: 63421\n",
      "Number of clusters: 62976\n",
      "\n",
      "Writing results 0h 0m 0s 11ms\n",
      "Time for merging to clu_step1: 0h 0m 0s 880ms\n",
      "Time for processing: 0h 0m 8s 412ms\n",
      "createsubdb /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/EvolutionaryOrigin_MMseqs/tmp/9674412578754287220/clu_step1 /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/EvolutionaryOrigin_MMseqs/tmp/9674412578754287220/input_step1 /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/EvolutionaryOrigin_MMseqs/tmp/9674412578754287220/input_step2 -v 3 --subdb-mode 1 \n",
      "\n",
      "Time for merging to input_step2: 0h 0m 2s 490ms\n",
      "Time for processing: 0h 0m 9s 404ms\n",
      "prefilter /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/EvolutionaryOrigin_MMseqs/tmp/9674412578754287220/input_step2 /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/EvolutionaryOrigin_MMseqs/tmp/9674412578754287220/input_step2 /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/EvolutionaryOrigin_MMseqs/tmp/9674412578754287220/pref_step2 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' --seed-sub-mat 'aa:VTML80.out,nucl:nucleotide.out' -s 3 -k 0 --k-score seq:2147483647,prof:2147483647 --alph-size aa:21,nucl:5 --max-seq-len 65535 --max-seqs 20 --split 0 --split-mode 2 --split-memory-limit 0 -c 0.5 --cov-mode 0 --comp-bias-corr 1 --comp-bias-corr-scale 1 --diag-score 1 --exact-kmer-matching 0 --mask 1 --mask-prob 0.9 --mask-lower-case 0 --min-ungapped-score 15 --add-self-matches 0 --spaced-kmer-mode 1 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --threads 24 --compressed 0 -v 3 \n",
      "\n",
      "Query database size: 62976 type: Aminoacid\n",
      "Estimated memory consumption: 1G\n",
      "Target database size: 62976 type: Aminoacid\n",
      "Index table k-mer threshold: 136 at k-mer size 6 \n",
      "Index table: counting k-mers\n",
      "[=================================================================] 62.98K 0s 189ms\n",
      "Index table: Masked residues: 560710\n",
      "Index table: fill\n",
      "[=================================================================] 62.98K 0s 134ms\n",
      "Index statistics\n",
      "Entries:          11741870\n",
      "DB size:          555 MB\n",
      "Avg k-mer size:   0.183467\n",
      "Top 10 k-mers\n",
      "    DVVFEE\t84\n",
      "    VLQLLP\t69\n",
      "    QIELEQ\t51\n",
      "    LYDMNY\t46\n",
      "    TFDAFS\t42\n",
      "    TFDAFT\t38\n",
      "    VMEYLV\t38\n",
      "    GPELLR\t34\n",
      "    CQDGLV\t33\n",
      "    RDARLV\t32\n",
      "Time for index table init: 0h 0m 0s 756ms\n",
      "Process prefiltering step 1 of 1\n",
      "\n",
      "k-mer similarity threshold: 136\n",
      "Starting prefiltering scores calculation (step 1 of 1)\n",
      "Query db start 1 to 62976\n",
      "Target db start 1 to 62976\n",
      "[=================================================================] 62.98K 1s 180ms\n",
      "\n",
      "17.379572 k-mers per position\n",
      "817 DB matches per sequence\n",
      "0 overflows\n",
      "0 queries produce too many hits (truncated result)\n",
      "5 sequences passed prefiltering per query sequence\n",
      "3 median result list length\n",
      "0 sequences with 0 size result lists\n",
      "Time for merging to pref_step2: 0h 0m 2s 653ms\n",
      "Time for processing: 0h 0m 31s 667ms\n",
      "align /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/EvolutionaryOrigin_MMseqs/tmp/9674412578754287220/input_step2 /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/EvolutionaryOrigin_MMseqs/tmp/9674412578754287220/input_step2 /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/EvolutionaryOrigin_MMseqs/tmp/9674412578754287220/pref_step2 /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/EvolutionaryOrigin_MMseqs/tmp/9674412578754287220/aln_step2 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' -a 0 --alignment-mode 3 --alignment-output-mode 0 --wrapped-scoring 0 -e 0.001 --min-seq-id 0.5 --min-aln-len 0 --seq-id-mode 0 --alt-ali 0 -c 0.5 --cov-mode 0 --max-seq-len 65535 --comp-bias-corr 1 --comp-bias-corr-scale 1 --max-rejected 2147483647 --max-accept 2147483647 --add-self-matches 0 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --score-bias 0 --realign 0 --realign-score-bias -0.2 --realign-max-seqs 2147483647 --corr-score-weight 0 --gap-open aa:11,nucl:5 --gap-extend aa:1,nucl:2 --zdrop 40 --threads 24 --compressed 0 -v 3 \n",
      "\n",
      "Compute score, coverage and sequence identity\n",
      "Query database size: 62976 type: Aminoacid\n",
      "Target database size: 62976 type: Aminoacid\n",
      "Calculation of alignments\n",
      "[=================================================================] 62.98K 8s 897ms\n",
      "Time for merging to aln_step2: 0h 0m 3s 977ms\n",
      "190585 alignments calculated\n",
      "63099 sequence pairs passed the thresholds (0.331081 of overall calculated)\n",
      "1.001953 hits per query sequence\n",
      "Time for processing: 0h 0m 34s 919ms\n",
      "clust /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/EvolutionaryOrigin_MMseqs/tmp/9674412578754287220/input_step2 /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/EvolutionaryOrigin_MMseqs/tmp/9674412578754287220/aln_step2 /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/EvolutionaryOrigin_MMseqs/tmp/9674412578754287220/clu_step2 --cluster-mode 0 --max-iterations 1000 --similarity-type 2 --threads 24 --compressed 0 -v 3 \n",
      "\n",
      "Clustering mode: Set Cover\n",
      "[=================================================================] 62.98K 0s 7ms\n",
      "Sort entries\n",
      "Find missing connections\n",
      "Found 41 new connections.\n",
      "Reconstruct initial order\n",
      "[=================================================================] 62.98K 0s 8ms\n",
      "Add missing connections\n",
      "[=================================================================] 62.98K 0s 1ms\n",
      "\n",
      "Time for read in: 0h 0m 0s 65ms\n",
      "Total time: 0h 0m 2s 584ms\n",
      "\n",
      "Size of the sequence database: 62976\n",
      "Size of the alignment database: 62976\n",
      "Number of clusters: 62896\n",
      "\n",
      "Writing results 0h 0m 0s 14ms\n",
      "Time for merging to clu_step2: 0h 0m 1s 467ms\n",
      "Time for processing: 0h 0m 7s 742ms\n",
      "mergeclusters /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/EvolutionaryOrigin_MMseqs/MMseqsDB/species_and_humanBrainDB /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/EvolutionaryOrigin_MMseqs/Clustering/species_and_humanBrain_cluDB /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/EvolutionaryOrigin_MMseqs/tmp/9674412578754287220/clu_redundancy /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/EvolutionaryOrigin_MMseqs/tmp/9674412578754287220/clu_step0 /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/EvolutionaryOrigin_MMseqs/tmp/9674412578754287220/clu_step1 /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/EvolutionaryOrigin_MMseqs/tmp/9674412578754287220/clu_step2 --threads 24 --compressed 0 -v 3 \n",
      "\n",
      "Clustering step 1\n",
      "[=================================================================] 70.83K 2s 87ms\n",
      "Clustering step 2\n",
      "[=================================================================] 63.42K 2s 121ms\n",
      "Clustering step 3\n",
      "[=================================================================] 62.98K 2s 152ms\n",
      "Clustering step 4\n",
      "[=================================================================] 62.90K 2s 192ms\n",
      "Write merged clustering\n",
      "[=================================================================] 128.71K 18s 165ms\n",
      "Time for merging to species_and_humanBrain_cluDB: 0h 0m 3s 432ms\n",
      "Time for processing: 0h 0m 32s 828ms\n",
      "createtsv /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/EvolutionaryOrigin_MMseqs/MMseqsDB/species_and_humanBrainDB /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/EvolutionaryOrigin_MMseqs/MMseqsDB/species_and_humanBrainDB /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/EvolutionaryOrigin_MMseqs/Clustering/species_and_humanBrain_cluDB /users/genomics/marta/TestisProject_SaraRazquin/with_TranscriptomeReconstruction/EvolutionaryOrigin_MMseqs/Clustering/species_and_humanBrain_cluDB.tsv \n",
      "\n",
      "MMseqs Version:                 \t14.7e284\n",
      "First sequence as representative\tfalse\n",
      "Target column                   \t1\n",
      "Add full header                 \tfalse\n",
      "Sequence source                 \t0\n",
      "Database output                 \tfalse\n",
      "Threads                         \t24\n",
      "Compressed                      \t0\n",
      "Verbosity                       \t3\n",
      "\n",
      "Time for merging to species_and_humanBrain_cluDB.tsv: 0h 0m 14s 991ms\n",
      "Time for processing: 0h 0m 51s 734ms\n"
     ]
    }
   ],
   "source": [
    "%%bash -s \"$evoDir\" \"$users_dir\"\n",
    "\n",
    "module load Miniconda3/4.9.2\n",
    "source /soft/system/software/Miniconda3/4.9.2/bin/activate MMseqs2\n",
    "\n",
    "## cluster all species and human\n",
    "mmseqs cluster -c 0.5 --min-seq-id 0.5 $1/MMseqsDB/species_and_humanBrainDB $1/Clustering/species_and_humanBrain_cluDB $1/tmp\n",
    "mmseqs createtsv $1/MMseqsDB/species_and_humanBrainDB $1/MMseqsDB/species_and_humanBrainDB $1/Clustering/species_and_humanBrain_cluDB $1/Clustering/species_and_humanBrain_cluDB.tsv"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Analysis"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 77,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "all human\n",
      "12417\n",
      "all human after annotation\n",
      "12417\n",
      "noncoding all\n",
      "1400\n",
      "1175\n",
      "Conserved noncoding human ORFs\n",
      "72\n",
      "Non-Conserved noncoding human ORFs\n",
      "1175\n",
      "Canonical all\n",
      "11017\n",
      "2194\n",
      "Conserved canonical human ORFs\n",
      "1872\n",
      "Conserved canonical human ORFs as noncoding in other species\n",
      "1852\n",
      "Conserved canonical human ORFs as canonical in other species\n",
      "235\n",
      "Non-Conserved canonical human ORFs\n",
      "2194\n"
     ]
    }
   ],
   "source": [
    "clustering = pd.read_csv(os.path.join(evoDir,\"Clustering/species_and_humanBrain_cluDB.tsv\"), header=None, sep=\"\\t\")\n",
    "clustering = clustering[clustering[0].str.startswith(\"ENST\")]\n",
    "print(\"all human\")\n",
    "print(len(clustering))\n",
    "clustering['transcript_id'] = clustering[0].str.split(\":\", expand=True)[0]\n",
    "clustering['transcript_id'] = clustering[0].str.split(\".\", expand=True)[0]\n",
    "\n",
    "clustering = clustering.merge(transcript_gene, on=\"transcript_id\")\n",
    "print(\"all human after annotation\")\n",
    "print(len(clustering))\n",
    "clustering = clustering[[\"gene_name\",\"gene_type\",0,1]]\n",
    "\n",
    "## non-coding\n",
    "noncoding = clustering[clustering[0].str.contains(\"noncoding\")]\n",
    "noncoding.to_csv(os.path.join(evoDir,\"Clustering/BRAIN_noncodingHuman_speciesDB.tsv\"), header=None, index=None, sep=\"\\t\")\n",
    "print(\"noncoding all\")\n",
    "print(len(noncoding))\n",
    "print(len(set(noncoding[0].values.tolist())))\n",
    "\n",
    "noncoding_conserved = noncoding[~noncoding[1].str.startswith(\"ENST\")]\n",
    "noncoding_conserved.to_csv(os.path.join(evoDir,\"Clustering/BRAIN_noncodingHuman_Conserved.tsv\"), header=None, index=None, sep=\"\\t\")\n",
    "print(\"Conserved noncoding human ORFs\")\n",
    "print(len(set(noncoding_conserved[0].values.tolist())))\n",
    "noncoding_conserved.columns = ['gene_name','gene_type','orfID',1]\n",
    "\n",
    "noncoding_NonConserved = noncoding[noncoding[1].str.startswith(\"ENST\")]\n",
    "noncoding_NonConserved.to_csv(os.path.join(evoDir,\"Clustering/BRAIN_noncodingHuman_NonConserved.tsv\"), header=None, index=None, sep=\"\\t\")\n",
    "print(\"Non-Conserved noncoding human ORFs\")\n",
    "print(len(set(noncoding_NonConserved[0].values.tolist())))\n",
    "noncoding_NonConserved.columns = ['gene_name','gene_type','orfID',1]\n",
    "\n",
    "\n",
    "## canonical\n",
    "canonical = clustering[clustering[0].str.contains(\"canonical\")]\n",
    "print(\"Canonical all\")\n",
    "print(len(canonical))\n",
    "print(len(set(canonical[0].values.tolist())))\n",
    "canonical.to_csv(os.path.join(evoDir,\"Clustering/BRAIN_canonicalHuman_speciesDB.tsv\"), header=None, index=None, sep=\"\\t\")\n",
    "\n",
    "canonical_conserved = canonical[~canonical[1].str.startswith(\"ENST\")]\n",
    "canonical_conserved.to_csv(os.path.join(evoDir,\"Clustering/BRAIN_canonicalHuman_Conserved.tsv\"), header=None, index=None, sep=\"\\t\")\n",
    "print(\"Conserved canonical human ORFs\")\n",
    "print(len(set(canonical_conserved[0].values.tolist())))\n",
    "canonical_conserved.columns = ['gene_name','gene_type','orfID',1]\n",
    "\n",
    "canonical_conserved_noncoding = canonical_conserved[~canonical_conserved[1].str.contains(\"noncoding\")]\n",
    "canonical_conserved_noncoding.to_csv(os.path.join(evoDir,\"Clustering/BRAIN_canonicalHuman_ConservedAsNoncoding.tsv\"), header=None, index=None, sep=\"\\t\")\n",
    "print(\"Conserved canonical human ORFs as noncoding in other species\")\n",
    "print(len(set(canonical_conserved_noncoding['orfID'].values.tolist())))\n",
    "\n",
    "canonical_conserved_canonical = canonical_conserved[~canonical_conserved[1].str.contains(\"canonical\")]\n",
    "canonical_conserved_canonical.to_csv(os.path.join(evoDir,\"Clustering/BRAIN_canonicalHuman_ConservedAsCanonical.tsv\"), header=None, index=None, sep=\"\\t\")\n",
    "print(\"Conserved canonical human ORFs as canonical in other species\")\n",
    "print(len(set(canonical_conserved_canonical['orfID'].values.tolist())))\n",
    "\n",
    "canonical_NonConserved = canonical[canonical[1].str.startswith(\"ENST\")]\n",
    "canonical_NonConserved.to_csv(os.path.join(evoDir,\"Clustering/BRAIN_canonicalHuman_NonConserved.tsv\"), header=None, index=None, sep=\"\\t\")\n",
    "print(\"Non-Conserved canonical human ORFs\")\n",
    "print(len(set(canonical_NonConserved[0].values.tolist())))\n",
    "canonical_NonConserved.columns = ['gene_name','gene_type','orfID',1]\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 78,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "72\n",
      "72\n",
      "{'mouse', 'macaca', 'opossum', 'platypus', 'chicken'}\n",
      "72\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_1203003/4120325751.py:5: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  noncoding_conserved_to_matrix['specie'] = noncoding_conserved_to_matrix[1].str.split(\"_\", expand=True)[0]\n",
      "/tmp/ipykernel_1203003/4120325751.py:24: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  noncoding_conserved_matrix_ordered['sum'] = noncoding_conserved_matrix_ordered.select_dtypes(include='number').sum(axis=1)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "72\n"
     ]
    }
   ],
   "source": [
    "noncoding_conserved_to_matrix = noncoding_conserved\n",
    "noncoding_conserved_to_matrix\n",
    "print(len(set(noncoding_conserved_to_matrix.orfID.values.tolist())))\n",
    "# Assuming noncoding_conserved_to_matrix is already defined\n",
    "noncoding_conserved_to_matrix['specie'] = noncoding_conserved_to_matrix[1].str.split(\"_\", expand=True)[0]\n",
    "noncoding_conserved_to_matrix = noncoding_conserved_to_matrix[~noncoding_conserved_to_matrix['specie'].str.contains(\"TCONS\")]\n",
    "noncoding_conserved_to_matrix = noncoding_conserved_to_matrix[['orfID', 'specie']]\n",
    "noncoding_conserved_to_matrix['presence'] = 1\n",
    "print(len(set(noncoding_conserved_to_matrix.orfID.values.tolist())))\n",
    "print(set(noncoding_conserved_to_matrix.specie.values.tolist()))\n",
    "## Pivoting the data without creating an 'id', and keeping 'orfID' as the unique identifier\n",
    "noncoding_conserved_matrix = noncoding_conserved_to_matrix.pivot_table(index='orfID', columns='specie', values='presence', aggfunc='max').reset_index()\n",
    "# Rename columns as necessary\n",
    "noncoding_conserved_matrix.columns = [\"orfID\", \"chicken\", \"macaca\", \"mouse\", \"opossum\", \"platypus\"]\n",
    "# Fill NaN values with 0\n",
    "noncoding_conserved_matrix = noncoding_conserved_matrix.fillna(0)\n",
    "\n",
    "# Reset index so 'orfID' becomes a column instead of an index\n",
    "\n",
    "# Reorder columns for the final matrix\n",
    "noncoding_conserved_matrix_ordered = noncoding_conserved_matrix[[\"orfID\", \"macaca\", \"mouse\", \"opossum\", \"platypus\", \"chicken\"]]\n",
    "\n",
    "# Calculate sum across species columns to identify how many species are present\n",
    "noncoding_conserved_matrix_ordered['sum'] = noncoding_conserved_matrix_ordered.select_dtypes(include='number').sum(axis=1)\n",
    "\n",
    "# Sort by the sum column to order by the number of species present\n",
    "noncoding_conserved_matrix_ordered = noncoding_conserved_matrix_ordered.sort_values(by=\"sum\", ascending=False)\n",
    "print(len(set(noncoding_conserved_matrix_ordered.orfID.values.tolist())))\n",
    "\n",
    "# Display the result\n",
    "noncoding_conserved_matrix_ordered =  noncoding_conserved_matrix_ordered.merge(noncoding_conserved[['gene_name','gene_type','orfID']], on=\"orfID\")\n",
    "noncoding_conserved_matrix_ordered.drop_duplicates(inplace=True)\n",
    "\n",
    "noncoding_conserved_matrix_ordered.to_csv(os.path.join(evoDir,\"Clustering/BRAIN_noncodingHuman_Conserved_matrix.tsv\"), sep=\"\\t\", index=None)\n",
    "print(len(set(noncoding_conserved_matrix_ordered.orfID.values.tolist())))\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 79,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_1203003/27345827.py:10: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  noncoding_to_matrix['presence'] = np.where(noncoding_to_matrix['specie'].isna(), 0, 1)\n",
      "/tmp/ipykernel_1203003/27345827.py:39: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  noncoding_matrix_ordered['sum'] = noncoding_matrix_ordered.select_dtypes(include='number').sum(axis=1)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "1175\n"
     ]
    }
   ],
   "source": [
    "### all noncoding\n",
    "# noncoding_conserved_touse = noncoding_conserved\n",
    "# noncoding_conserved_touse['specie'] = noncoding_conserved_touse[1].str.split(\"_\", expand=True)[0]\n",
    "# noncoding_conserved_touse = noncoding_conserved_touse[~noncoding_conserved_touse['specie'].str.contains(\"TCONS\")]\n",
    "\n",
    "noncoding_to_matrix_raw = pd.concat([noncoding_conserved, noncoding_NonConserved])\n",
    "noncoding_to_matrix_raw\n",
    "noncoding_to_matrix = noncoding_to_matrix_raw[['orfID', 'specie']]\n",
    "# noncoding_to_matrix\n",
    "noncoding_to_matrix['presence'] = np.where(noncoding_to_matrix['specie'].isna(), 0, 1)\n",
    "\n",
    "\n",
    "human = pd.DataFrame({\n",
    "    'orfID': noncoding_to_matrix['orfID'].unique(),\n",
    "    'specie': 'human',\n",
    "    'presence': 1\n",
    "})\n",
    "\n",
    "noncoding_to_matrix = pd.concat([human, noncoding_to_matrix], ignore_index=True)\n",
    "noncoding_to_matrix.dropna(inplace=True)\n",
    "noncoding_to_matrix = noncoding_to_matrix[~noncoding_to_matrix['specie'].str.contains(\"TCONS\")]\n",
    "\n",
    "# Pivoting the data without creating an 'id', and keeping 'orfID' as the unique identifier\n",
    "noncoding_matrix = noncoding_to_matrix.pivot_table(index='orfID', columns='specie', values='presence', aggfunc='max')\n",
    "noncoding_matrix\n",
    "# Fill NaN values with 0\n",
    "noncoding_matrix = noncoding_matrix.fillna(0)\n",
    "\n",
    "# Reset index so 'orfID' becomes a column instead of an index\n",
    "noncoding_matrix = noncoding_matrix.reset_index()\n",
    "noncoding_matrix\n",
    "# Rename columns as necessary\n",
    "noncoding_matrix.columns = [\"orfID\", \"chicken\", \"human\", \"macaca\", \"mouse\", \"opossum\", \"platypus\"]\n",
    "\n",
    "# Reorder columns for the final matrix\n",
    "noncoding_matrix_ordered = noncoding_matrix[[\"orfID\", \"human\",\"macaca\", \"mouse\", \"opossum\", \"platypus\", \"chicken\"]]\n",
    "\n",
    "# Calculate sum across species columns to identify how many species are present\n",
    "noncoding_matrix_ordered['sum'] = noncoding_matrix_ordered.select_dtypes(include='number').sum(axis=1)\n",
    "\n",
    "# Sort by the sum column to order by the number of species present\n",
    "noncoding_matrix_ordered = noncoding_matrix_ordered.sort_values(by=\"sum\", ascending=False)\n",
    "\n",
    "# Display the result\n",
    "noncoding_matrix_ordered =  noncoding_matrix_ordered.merge(noncoding_to_matrix_raw[['gene_name','gene_type','orfID']], on=\"orfID\")\n",
    "noncoding_matrix_ordered.drop_duplicates(inplace=True)\n",
    "\n",
    "noncoding_matrix_ordered.to_csv(os.path.join(evoDir,\"Clustering/BRAIN_noncodingHuman_matrix.tsv\"), sep=\"\\t\", index=None)\n",
    "print(len(set(noncoding_matrix_ordered.orfID.values.tolist())))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 80,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_1203003/1759624414.py:4: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  canonical_conserved_to_matrix['specie'] = canonical_conserved_to_matrix[1].str.split(\"_\", expand=True)[0]\n",
      "/tmp/ipykernel_1203003/1759624414.py:6: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  canonical_conserved_to_matrix['presence'] = 1\n",
      "/tmp/ipykernel_1203003/1759624414.py:26: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  canonical_conserved_matrix_ordered['sum'] = canonical_conserved_matrix_ordered.select_dtypes(include='number').sum(axis=1)\n"
     ]
    }
   ],
   "source": [
    "canonical_conserved_to_matrix = canonical_conserved\n",
    "\n",
    "# Assuming noncoding_conserved_to_matrix is already defined\n",
    "canonical_conserved_to_matrix['specie'] = canonical_conserved_to_matrix[1].str.split(\"_\", expand=True)[0]\n",
    "canonical_conserved_to_matrix = canonical_conserved_to_matrix[['orfID', 'specie']]\n",
    "canonical_conserved_to_matrix['presence'] = 1\n",
    "\n",
    "canonical_conserved_to_matrix = canonical_conserved_to_matrix[~canonical_conserved_to_matrix['specie'].str.contains(\"TCONS\")]\n",
    "\n",
    "# Pivoting the data without creating an 'id', and keeping 'orfID' as the unique identifier\n",
    "canonical_conserved_matrix = canonical_conserved_to_matrix.pivot_table(index='orfID', columns='specie', values='presence', aggfunc='max')\n",
    "\n",
    "# Fill NaN values with 0\n",
    "canonical_conserved_matrix = canonical_conserved_matrix.fillna(0)\n",
    "\n",
    "# Reset index so 'orfID' becomes a column instead of an index\n",
    "canonical_conserved_matrix = canonical_conserved_matrix.reset_index()\n",
    "\n",
    "# Rename columns as necessary\n",
    "canonical_conserved_matrix.columns = [\"orfID\", \"chicken\", \"macaca\", \"mouse\", \"opossum\", \"platypus\"]\n",
    "\n",
    "# Reorder columns for the final matrix\n",
    "canonical_conserved_matrix_ordered = canonical_conserved_matrix[[\"orfID\", \"macaca\", \"mouse\", \"opossum\", \"platypus\", \"chicken\"]]\n",
    "\n",
    "# Calculate sum across species columns to identify how many species are present\n",
    "canonical_conserved_matrix_ordered['sum'] = canonical_conserved_matrix_ordered.select_dtypes(include='number').sum(axis=1)\n",
    "\n",
    "# Sort by the sum column to order by the number of species present\n",
    "canonical_conserved_matrix_ordered = canonical_conserved_matrix_ordered.sort_values(by=\"sum\", ascending=False)\n",
    "\n",
    "# Display the result\n",
    "canonical_conserved_matrix_ordered =  canonical_conserved_matrix_ordered.merge(canonical_conserved[['gene_name','gene_type','orfID']], on=\"orfID\")\n",
    "canonical_conserved_matrix_ordered.drop_duplicates(inplace=True)\n",
    "\n",
    "canonical_conserved_matrix_ordered.to_csv(os.path.join(evoDir,\"Clustering/BRAIN_canonicalHuman_Conserved_matrix.tsv\"), sep=\"\\t\", index=None)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 81,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_1203003/2906229079.py:3: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  canonical_conserved_touse['specie'] = canonical_conserved_touse[1].str.split(\"_\", expand=True)[0]\n",
      "/tmp/ipykernel_1203003/2906229079.py:8: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  canonical_to_matrix['presence'] = np.where(canonical_to_matrix['specie'].isna(), 0, 1)\n",
      "/tmp/ipykernel_1203003/2906229079.py:36: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  canonical_matrix_ordered['sum'] = canonical_matrix_ordered.select_dtypes(include='number').sum(axis=1)\n"
     ]
    }
   ],
   "source": [
    "### all canonical\n",
    "canonical_conserved_touse = canonical_conserved\n",
    "canonical_conserved_touse['specie'] = canonical_conserved_touse[1].str.split(\"_\", expand=True)[0]\n",
    "# noncoding_NonConserved.drop(\"specie\",axis=1, inplace=True)\n",
    "canonical_to_matrix_raw = pd.concat([canonical_conserved, canonical_NonConserved])\n",
    "\n",
    "canonical_to_matrix = canonical_to_matrix_raw[['orfID', 'specie']]\n",
    "canonical_to_matrix['presence'] = np.where(canonical_to_matrix['specie'].isna(), 0, 1)\n",
    "\n",
    "\n",
    "human = pd.DataFrame({\n",
    "    'orfID': canonical_to_matrix['orfID'].unique(),\n",
    "    'specie': 'human',\n",
    "    'presence': 1\n",
    "})\n",
    "\n",
    "canonical_to_matrix = pd.concat([human, canonical_to_matrix], ignore_index=True)\n",
    "canonical_to_matrix.dropna(inplace=True)\n",
    "canonical_to_matrix = canonical_to_matrix[~canonical_to_matrix['specie'].str.contains(\"TCONS\")]\n",
    "# Pivoting the data without creating an 'id', and keeping 'orfID' as the unique identifier\n",
    "canonical_matrix = canonical_to_matrix.pivot_table(index='orfID', columns='specie', values='presence', aggfunc='max')\n",
    "canonical_matrix\n",
    "# Fill NaN values with 0\n",
    "canonical_matrix = canonical_matrix.fillna(0)\n",
    "\n",
    "# Reset index so 'orfID' becomes a column instead of an index\n",
    "canonical_matrix = canonical_matrix.reset_index()\n",
    "\n",
    "# Rename columns as necessary\n",
    "canonical_matrix.columns = [\"orfID\", \"chicken\", \"human\", \"macaca\", \"mouse\", \"opossum\", \"platypus\"]\n",
    "\n",
    "# Reorder columns for the final matrix\n",
    "canonical_matrix_ordered = canonical_matrix[[\"orfID\", \"human\",\"macaca\", \"mouse\", \"opossum\", \"platypus\", \"chicken\"]]\n",
    "\n",
    "# Calculate sum across species columns to identify how many species are present\n",
    "canonical_matrix_ordered['sum'] = canonical_matrix_ordered.select_dtypes(include='number').sum(axis=1)\n",
    "\n",
    "# Sort by the sum column to order by the number of species present\n",
    "canonical_matrix_ordered = canonical_matrix_ordered.sort_values(by=\"sum\", ascending=False)\n",
    "\n",
    "# Display the result\n",
    "canonical_matrix_ordered =  canonical_matrix_ordered.merge(canonical_to_matrix_raw[['gene_name','gene_type','orfID']], on=\"orfID\")\n",
    "canonical_matrix_ordered.drop_duplicates(inplace=True)\n",
    "\n",
    "canonical_matrix_ordered.to_csv(os.path.join(evoDir,\"Clustering/BRAIN_canonicalHuman_matrix.tsv\"), sep=\"\\t\", index=None)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Find the oldest specie containg the ORF translated"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 82,
   "metadata": {},
   "outputs": [],
   "source": [
    "species_ordered = ['chicken','platypus','opossum','mouse','macaca','human']\n",
    "def find_last_species(row):\n",
    "    for sp in species_ordered:\n",
    "        if row[sp] == 1.0:  # Check if the value is 1.0\n",
    "            return sp\n",
    "    return None  # Return None if no species has a 1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 83,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Apply the function to each row to create a new column 'oldest_species'\n",
    "canonical_matrix_ordered['oldest_species'] = canonical_matrix_ordered.apply(find_last_species, axis=1)\n",
    "canonical_matrix_ordered = species_and_years.merge(canonical_matrix_ordered, left_on=[\"specie\"], right_on=\"oldest_species\")\n",
    "canonical_matrix_ordered[['orfID','gene_name','gene_type','oldest_species','age']].to_csv(os.path.join(evoDir,\"Clustering/BRAIN_canonicalHuman_matrix_oldestAge.tsv\"), sep=\"\\t\", index=None)\n",
    "\n",
    "noncoding_matrix_ordered['oldest_species'] = noncoding_matrix_ordered.apply(find_last_species, axis=1)\n",
    "noncoding_matrix_ordered = species_and_years.merge(noncoding_matrix_ordered, left_on=[\"specie\"], right_on=\"oldest_species\")\n",
    "noncoding_matrix_ordered[['orfID','gene_name','gene_type','oldest_species','age']].to_csv(os.path.join(evoDir,\"Clustering/BRAIN_noncodingHuman_matrix_oldestAge.tsv\"), sep=\"\\t\", index=None)\n",
    "\n",
    "complete_matrix_age = pd.concat([noncoding_matrix_ordered[['orfID','gene_name','gene_type','oldest_species','age']], canonical_matrix_ordered[['orfID','gene_name','gene_type','oldest_species','age']]])\n",
    "complete_matrix_age.to_csv(os.path.join(evoDir,\"Clustering/BRAIN_allHuman_matrix_oldestAge.tsv\"), sep=\"\\t\", index=None)\n"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.10"
  },
  "orig_nbformat": 4
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
